--- a/net/minecraft/world/entity/animal/camel/Camel.java
+++ b/net/minecraft/world/entity/animal/camel/Camel.java
@@ -430,7 +_,7 @@
          }
 
          Vec3 vec3 = (new Vec3(0.0D, 0.0D, (double)f)).m_82524_(-this.f_20883_ * ((float)Math.PI / 180F));
-         p_289551_.m_20372_(p_289564_, this.m_20185_() + vec3.f_82479_, this.m_20186_() + (double)f1, this.m_20189_() + vec3.f_82481_);
+         p_289551_.accept(p_289564_, this.getBigX().add(vec3.getX()), this.getBigY().add(java.math.BigDecimal.valueOf(f1)), this.getBigZ().add(vec3.getZ()));
          this.m_246824_(p_289564_);
       }
    }
