--- a/net/minecraft/advancements/critereon/DistanceTrigger.java
+++ b/net/minecraft/advancements/critereon/DistanceTrigger.java
@@ -61,10 +_,10 @@
       }
 
       public boolean m_186188_(ServerLevel p_186189_, Vec3 p_186190_, Vec3 p_186191_) {
-         if (!this.f_186181_.m_52617_(p_186189_, p_186190_.f_82479_, p_186190_.f_82480_, p_186190_.f_82481_)) {
+         if (!this.f_186181_.m_52617_(p_186189_, p_186190_.m_7096_(), p_186190_.m_7098_(), p_186190_.m_7094_())) {
             return false;
          } else {
-            return this.f_186182_.m_26255_(p_186190_.f_82479_, p_186190_.f_82480_, p_186190_.f_82481_, p_186191_.f_82479_, p_186191_.f_82480_, p_186191_.f_82481_);
+            return this.f_186182_.m_26255_(p_186190_.m_7096_(), p_186190_.m_7098_(), p_186190_.m_7094_(), p_186191_.m_7096_(), p_186191_.m_7098_(), p_186191_.m_7094_());
          }
       }
    }
