--- a/net/minecraft/advancements/critereon/EntityPredicate.java
+++ b/net/minecraft/advancements/critereon/EntityPredicate.java
@@ -95,7 +_,7 @@
             return false;
          } else {
             if (this.f_150285_ != LocationPredicate.f_52592_) {
-               Vec3 vec3 = Vec3.m_82512_(p_36610_.m_216999_());
+               Vec3 vec3 = Vec3.atCenterOf(p_36610_.m_216999_());
                if (!this.f_150285_.m_52617_(p_36608_, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_())) {
                   return false;
                }
