--- a/net/minecraft/advancements/critereon/ItemInteractWithBlockTrigger.java
+++ b/net/minecraft/advancements/critereon/ItemInteractWithBlockTrigger.java
@@ -52,7 +_,7 @@
       }
 
       public boolean m_220070_(BlockState p_220071_, ServerLevel p_220072_, BlockPos p_220073_, ItemStack p_220074_) {
-         return !this.f_220058_.m_52617_(p_220072_, (double)p_220073_.m_123341_() + 0.5D, (double)p_220073_.m_123342_() + 0.5D, (double)p_220073_.m_123343_() + 0.5D) ? false : this.f_220059_.m_45049_(p_220074_);
+         return !this.f_220058_.m_52617_(p_220072_, (double)p_220073_.getX() + 0.5D, (double)p_220073_.getY() + 0.5D, (double)p_220073_.getZ() + 0.5D) ? false : this.f_220059_.m_45049_(p_220074_);
       }
 
       public JsonObject m_7683_(SerializationContext p_220069_) {
