--- a/net/minecraft/advancements/critereon/LevitationTrigger.java
+++ b/net/minecraft/advancements/critereon/LevitationTrigger.java
@@ -39,7 +_,7 @@
       }
 
       public boolean m_49140_(ServerPlayer p_49141_, Vec3 p_49142_, int p_49143_) {
-         if (!this.f_49134_.m_26255_(p_49142_.f_82479_, p_49142_.f_82480_, p_49142_.f_82481_, p_49141_.m_20185_(), p_49141_.m_20186_(), p_49141_.m_20189_())) {
+         if (!this.f_49134_.m_26255_(p_49142_.m_7096_(), p_49142_.m_7098_(), p_49142_.m_7094_(), p_49141_.m_20185_(), p_49141_.m_20186_(), p_49141_.m_20189_())) {
             return false;
          } else {
             return this.f_49135_.m_55390_(p_49143_);
