--- a/net/minecraft/advancements/critereon/PlacedBlockTrigger.java
+++ b/net/minecraft/advancements/critereon/PlacedBlockTrigger.java
@@ -77,7 +_,7 @@
             return false;
          } else if (!this.f_59496_.m_67667_(p_59508_)) {
             return false;
-         } else if (!this.f_59497_.m_52617_(p_59510_, (double)p_59509_.m_123341_(), (double)p_59509_.m_123342_(), (double)p_59509_.m_123343_())) {
+         } else if (!this.f_59497_.m_52617_(p_59510_, (double)p_59509_.getX(), (double)p_59509_.getY(), (double)p_59509_.getZ())) {
             return false;
          } else {
             return this.f_59498_.m_45049_(p_59511_);
