--- a/net/minecraft/advancements/critereon/PlayerPredicate.java
+++ b/net/minecraft/advancements/critereon/PlayerPredicate.java
@@ -108,7 +_,7 @@
          if (this.f_156744_ != EntityPredicate.f_36550_) {
             Vec3 vec3 = serverplayer.m_146892_();
             Vec3 vec31 = serverplayer.m_20252_(1.0F);
-            Vec3 vec32 = vec3.m_82520_(vec31.f_82479_ * 100.0D, vec31.f_82480_ * 100.0D, vec31.f_82481_ * 100.0D);
+            Vec3 vec32 = vec3.m_82520_(vec31.m_7096_() * 100.0D, vec31.m_7098_() * 100.0D, vec31.m_7094_() * 100.0D);
             EntityHitResult entityhitresult = ProjectileUtil.m_150175_(serverplayer.m_9236_(), serverplayer, vec3, vec32, (new AABB(vec3, vec32)).m_82400_(1.0D), (p_156765_) -> {
                return !p_156765_.m_5833_();
             }, 0.0F);
