--- a/net/minecraft/advancements/critereon/UsedEnderEyeTrigger.java
+++ b/net/minecraft/advancements/critereon/UsedEnderEyeTrigger.java
@@ -18,8 +_,8 @@
    }
 
    public void m_73935_(ServerPlayer p_73936_, BlockPos p_73937_) {
-      double d0 = p_73936_.m_20185_() - (double)p_73937_.m_123341_();
-      double d1 = p_73936_.m_20189_() - (double)p_73937_.m_123343_();
+      double d0 = p_73936_.m_20185_() - (double)p_73937_.getX();
+      double d1 = p_73936_.m_20189_() - (double)p_73937_.getZ();
       double d2 = d0 * d0 + d1 * d1;
       this.m_66234_(p_73936_, (p_73934_) -> {
          return p_73934_.m_73951_(d2);
