--- a/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/BeeDebugRenderer.java
@@ -80,7 +_,7 @@
       });
    }
 
-   public void m_7790_(PoseStack p_113061_, MultiBufferSource p_113062_, double p_113063_, double p_113064_, double p_113065_) {
+   public void render(PoseStack p_113061_, MultiBufferSource p_113062_, me.alphamode.mclong.math.BigDecimal p_113063_, me.alphamode.mclong.math.BigDecimal p_113064_, me.alphamode.mclong.math.BigDecimal p_113065_) {
       this.m_113136_();
       this.m_113126_();
       this.m_269283_(p_113061_, p_113062_);
@@ -114,21 +_,21 @@
       this.m_269561_(p_270886_, p_270808_);
 
       for(BlockPos blockpos1 : this.f_113049_.keySet()) {
-         if (blockpos.m_123314_(blockpos1, 30.0D)) {
+         if (blockpos.closerThan(blockpos1, 30.0D)) {
             m_269172_(p_270886_, p_270808_, blockpos1);
          }
       }
 
       Map<BlockPos, Set<UUID>> map = this.m_113146_();
       this.f_113049_.values().forEach((p_269692_) -> {
-         if (blockpos.m_123314_(p_269692_.f_113180_, 30.0D)) {
+         if (blockpos.closerThan(p_269692_.f_113180_, 30.0D)) {
             Set<UUID> set = map.get(p_269692_.f_113180_);
             this.m_269169_(p_270886_, p_270808_, p_269692_, (Collection<UUID>)(set == null ? Sets.newHashSet() : set));
          }
 
       });
       this.m_113155_().forEach((p_269699_, p_269700_) -> {
-         if (blockpos.m_123314_(p_269699_, 30.0D)) {
+         if (blockpos.closerThan(p_269699_, 30.0D)) {
             this.m_269584_(p_270886_, p_270808_, p_269699_, p_269700_);
          }
 
@@ -207,7 +_,7 @@
 
    private void m_269467_(PoseStack p_270424_, MultiBufferSource p_270123_, BeeDebugRenderer.BeeInfo p_270137_) {
       if (p_270137_.f_113160_ != null) {
-         PathfindingRenderer.m_269027_(p_270424_, p_270123_, p_270137_.f_113160_, 0.5F, false, false, this.m_113154_().m_90583_().m_7096_(), this.m_113154_().m_90583_().m_7098_(), this.m_113154_().m_90583_().m_7094_());
+         PathfindingRenderer.renderPath(p_270424_, p_270123_, p_270137_.f_113160_, 0.5F, false, false, this.m_113154_().m_90583_().getX(), this.m_113154_().m_90583_().getY(), this.m_113154_().m_90583_().getZ());
       }
 
    }
@@ -251,9 +_,9 @@
    private static void m_269380_(PoseStack p_270438_, MultiBufferSource p_270244_, String p_270486_, BlockPos p_270062_, int p_270574_, int p_270228_) {
       double d0 = 1.3D;
       double d1 = 0.2D;
-      double d2 = (double)p_270062_.m_123341_() + 0.5D;
-      double d3 = (double)p_270062_.m_123342_() + 1.3D + (double)p_270574_ * 0.2D;
-      double d4 = (double)p_270062_.m_123343_() + 0.5D;
+      double d2 = (double)p_270062_.getX() + 0.5D;
+      double d3 = (double)p_270062_.getY() + 1.3D + (double)p_270574_ * 0.2D;
+      double d4 = (double)p_270062_.getZ() + 0.5D;
       DebugRenderer.m_269439_(p_270438_, p_270244_, p_270486_, d2, d3, d4, p_270228_, 0.02F, true, 0.0F, true);
    }
 
@@ -261,9 +_,9 @@
       double d0 = 2.4D;
       double d1 = 0.25D;
       BlockPos blockpos = BlockPos.m_274446_(p_270548_);
-      double d2 = (double)blockpos.m_123341_() + 0.5D;
+      double d2 = (double)blockpos.getX() + 0.5D;
       double d3 = p_270548_.m_7098_() + 2.4D + (double)p_270592_ * 0.25D;
-      double d4 = (double)blockpos.m_123343_() + 0.5D;
+      double d4 = (double)blockpos.getZ() + 0.5D;
       float f = 0.5F;
       DebugRenderer.m_269439_(p_270426_, p_270600_, p_270198_, d2, d3, d4, p_270792_, p_270938_, false, 0.5F, true);
    }
@@ -277,9 +_,9 @@
    }
 
    private String m_113068_(BeeDebugRenderer.BeeInfo p_113069_, BlockPos p_113070_) {
-      double d0 = Math.sqrt(p_113070_.m_203193_(p_113069_.f_113159_));
+      double d0 = Math.sqrt(p_113070_.distToCenterSqr(p_113069_.f_113159_));
       double d1 = (double)Math.round(d0 * 10.0D) / 10.0D;
-      return p_113070_.m_123344_() + " (dist " + d1 + ")";
+      return p_113070_.toShortString() + " (dist " + d1 + ")";
    }
 
    private boolean m_113142_(BeeDebugRenderer.BeeInfo p_113143_) {
@@ -290,7 +_,7 @@
       Player player = this.f_113048_.f_91074_;
       BlockPos blockpos = BlockPos.m_274561_(player.m_20185_(), p_113148_.f_113159_.m_7098_(), player.m_20189_());
       BlockPos blockpos1 = BlockPos.m_274446_(p_113148_.f_113159_);
-      return blockpos.m_123314_(blockpos1, 30.0D);
+      return blockpos.closerThan(blockpos1, 30.0D);
    }
 
    private Collection<UUID> m_113129_(BlockPos p_113130_) {
