--- a/net/minecraft/client/renderer/debug/BrainDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/BrainDebugRenderer.java
@@ -139,19 +_,19 @@
       });
 
       for(BlockPos blockpos1 : this.f_113195_.keySet()) {
-         if (blockpos.m_123314_(blockpos1, 30.0D)) {
+         if (blockpos.closerThan(blockpos1, 30.0D)) {
             m_113274_(blockpos1);
          }
       }
 
       this.f_113195_.values().forEach((p_113238_) -> {
-         if (blockpos.m_123314_(p_113238_.f_113333_, 30.0D)) {
+         if (blockpos.closerThan(p_113238_.f_113333_, 30.0D)) {
             this.m_113272_(p_113238_);
          }
 
       });
       this.m_113279_().forEach((p_113241_, p_113242_) -> {
-         if (blockpos.m_123314_(p_113241_, 30.0D)) {
+         if (blockpos.closerThan(p_113241_, 30.0D)) {
             this.m_113243_(p_113241_, p_113242_);
          }
 
@@ -282,9 +_,9 @@
    private static void m_113257_(String p_113258_, BlockPos p_113259_, int p_113260_, int p_113261_) {
       double d0 = 1.3D;
       double d1 = 0.2D;
-      double d2 = (double)p_113259_.m_123341_() + 0.5D;
-      double d3 = (double)p_113259_.m_123342_() + 1.3D + (double)p_113260_ * 0.2D;
-      double d4 = (double)p_113259_.m_123343_() + 0.5D;
+      double d2 = (double)p_113259_.getX() + 0.5D;
+      double d3 = (double)p_113259_.getY() + 1.3D + (double)p_113260_ * 0.2D;
+      double d4 = (double)p_113259_.getZ() + 0.5D;
       DebugRenderer.m_113490_(p_113258_, d2, d3, d4, p_113261_, 0.02F, true, 0.0F, true);
    }
 
@@ -292,9 +_,9 @@
       double d0 = 2.4D;
       double d1 = 0.25D;
       BlockPos blockpos = new BlockPos(p_113247_);
-      double d2 = (double)blockpos.m_123341_() + 0.5D;
+      double d2 = (double)blockpos.getX() + 0.5D;
       double d3 = p_113247_.m_7098_() + 2.4D + (double)p_113248_ * 0.25D;
-      double d4 = (double)blockpos.m_123343_() + 0.5D;
+      double d4 = (double)blockpos.getZ() + 0.5D;
       float f = 0.5F;
       DebugRenderer.m_113490_(p_113249_, d2, d3, d4, p_113250_, p_113251_, false, 0.5F, true);
    }
@@ -315,7 +_,7 @@
       Player player = this.f_113194_.f_91074_;
       BlockPos blockpos = new BlockPos(player.m_20185_(), p_113281_.f_113300_.m_7098_(), player.m_20189_());
       BlockPos blockpos1 = new BlockPos(p_113281_.f_113300_);
-      return blockpos.m_123314_(blockpos1, 30.0D);
+      return blockpos.closerThan(blockpos1, 30.0D);
    }
 
    private Collection<UUID> m_113284_(BlockPos p_113285_) {
