--- a/net/minecraft/client/renderer/debug/ChunkDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/ChunkDebugRenderer.java
@@ -47,7 +_,7 @@
 
       if (this.f_113366_ != null) {
          Map<ChunkPos, String> map = this.f_113366_.f_113379_.getNow((Map<ChunkPos, String>)null);
-         double d1 = this.f_113363_.f_91063_.m_109153_().m_90583_().f_82480_ * 0.85D;
+         double d1 = this.f_113363_.f_91063_.m_109153_().m_90583_().m_7098_() * 0.85D;
 
          for(Map.Entry<ChunkPos, String> entry : this.f_113366_.f_113378_.entrySet()) {
             ChunkPos chunkpos = entry.getKey();
@@ -60,7 +_,7 @@
             int i = 0;
 
             for(String s1 : astring) {
-               DebugRenderer.m_269439_(p_113370_, p_113371_, s1, (double)SectionPos.m_175554_(chunkpos.f_45578_, 8), d1 + (double)i, (double)SectionPos.m_175554_(chunkpos.f_45579_, 8), -1, 0.15F, true, 0.0F, true);
+               DebugRenderer.m_269439_(p_113370_, p_113371_, s1, (double)SectionPos.sectionToBlockCoord(chunkpos.x(), 8), d1 + (double)i, (double)SectionPos.sectionToBlockCoord(chunkpos.z(), 8), -1, 0.15F, true, 0.0F, true);
                i -= 2;
             }
          }
@@ -76,16 +_,16 @@
       ChunkData(IntegratedServer p_113382_, double p_113383_, double p_113384_) {
          ClientLevel clientlevel = ChunkDebugRenderer.this.f_113363_.f_91073_;
          ResourceKey<Level> resourcekey = clientlevel.m_46472_();
-         int i = SectionPos.m_175552_(p_113383_);
-         int j = SectionPos.m_175552_(p_113384_);
+         long i = SectionPos.posToSectionCoord(p_113383_);
+         long j = SectionPos.posToSectionCoord(p_113384_);
          ImmutableMap.Builder<ChunkPos, String> builder = ImmutableMap.builder();
          ClientChunkCache clientchunkcache = clientlevel.m_7726_();
 
-         for(int k = i - 12; k <= i + 12; ++k) {
-            for(int l = j - 12; l <= j + 12; ++l) {
+         for(long k = i - 12; k <= i + 12; ++k) {
+            for(long l = j - 12; l <= j + 12; ++l) {
                ChunkPos chunkpos = new ChunkPos(k, l);
                String s = "";
-               LevelChunk levelchunk = clientchunkcache.m_62227_(k, l, false);
+               LevelChunk levelchunk = clientchunkcache.getChunk(k, l, false);
                s = s + "Client: ";
                if (levelchunk == null) {
                   s = s + "0n/a\n";
@@ -107,8 +_,8 @@
                ImmutableMap.Builder<ChunkPos, String> builder1 = ImmutableMap.builder();
                ServerChunkCache serverchunkcache = serverlevel.m_7726_();
 
-               for(int i1 = i - 12; i1 <= i + 12; ++i1) {
-                  for(int j1 = j - 12; j1 <= j + 12; ++j1) {
+               for(long i1 = i - 12; i1 <= i + 12; ++i1) {
+                  for(long j1 = j - 12; j1 <= j + 12; ++j1) {
                      ChunkPos chunkpos1 = new ChunkPos(i1, j1);
                      builder1.put(chunkpos1, "Server: " + serverchunkcache.m_8448_(chunkpos1));
                   }
