--- a/net/minecraft/client/renderer/debug/CollisionBoxRenderer.java
+++ b/net/minecraft/client/renderer/debug/CollisionBoxRenderer.java
@@ -36,7 +_,7 @@
       VertexConsumer vertexconsumer = p_113409_.m_6299_(RenderType.m_110504_());
 
       for(VoxelShape voxelshape : this.f_113402_) {
-         LevelRenderer.m_285900_(p_113408_, vertexconsumer, voxelshape, -p_113410_, -p_113411_, -p_113412_, 1.0F, 1.0F, 1.0F, 1.0F, true);
+         LevelRenderer.renderVoxelShape(p_113408_, vertexconsumer, voxelshape, me.alphamode.mclong.math.BigDecimal.valueOf(-p_113410_), me.alphamode.mclong.math.BigDecimal.valueOf(-p_113411_), me.alphamode.mclong.math.BigDecimal.valueOf(-p_113412_), 1.0F, 1.0F, 1.0F, 1.0F, true);
       }
 
    }
