--- a/net/minecraft/client/renderer/debug/DebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/DebugRenderer.java
@@ -96,12 +_,12 @@
       return this.f_113431_;
    }
 
-   public void m_113457_(PoseStack p_113458_, MultiBufferSource.BufferSource p_113459_, double p_113460_, double p_113461_, double p_113462_) {
+   public void render(PoseStack p_113458_, MultiBufferSource.BufferSource p_113459_, me.alphamode.mclong.math.BigDecimal p_113460_, me.alphamode.mclong.math.BigDecimal p_113461_, me.alphamode.mclong.math.BigDecimal p_113462_) {
       if (this.f_113431_ && !Minecraft.m_91087_().m_91299_()) {
-         this.f_113415_.m_7790_(p_113458_, p_113459_, p_113460_, p_113461_, p_113462_);
+         this.f_113415_.render(p_113458_, p_113459_, p_113460_, p_113461_, p_113462_);
       }
 
-      this.f_113430_.m_7790_(p_113458_, p_113459_, p_113460_, p_113461_, p_113462_);
+      this.f_113430_.render(p_113458_, p_113459_, p_113460_, p_113461_, p_113462_);
    }
 
    public static Optional<Entity> m_113448_(@Nullable Entity p_113449_, int p_113450_) {
@@ -144,7 +_,7 @@
    }
 
    public static void m_269311_(PoseStack p_271017_, MultiBufferSource p_270356_, AABB p_270833_, float p_270850_, float p_270249_, float p_270654_, float p_270476_) {
-      m_269008_(p_271017_, p_270356_, p_270833_.f_82288_, p_270833_.f_82289_, p_270833_.f_82290_, p_270833_.f_82291_, p_270833_.f_82292_, p_270833_.f_82293_, p_270850_, p_270249_, p_270654_, p_270476_);
+      m_269008_(p_271017_, p_270356_, p_270833_.f_82288_.doubleValue(), p_270833_.f_82289_.doubleValue(), p_270833_.f_82290_.doubleValue(), p_270833_.f_82291_.doubleValue(), p_270833_.f_82292_.doubleValue(), p_270833_.f_82293_.doubleValue(), p_270850_, p_270249_, p_270654_, p_270476_);
    }
 
    public static void m_269008_(PoseStack p_270616_, MultiBufferSource p_270769_, double p_270653_, double p_270967_, double p_270556_, double p_270724_, double p_270427_, double p_270138_, float p_270391_, float p_270093_, float p_270312_, float p_270567_) {
@@ -169,9 +_,9 @@
       Camera camera = minecraft.f_91063_.m_109153_();
       if (camera.m_90593_() && minecraft.m_91290_().f_114360_ != null) {
          Font font = minecraft.f_91062_;
-         double d0 = camera.m_90583_().f_82479_;
-         double d1 = camera.m_90583_().f_82480_;
-         double d2 = camera.m_90583_().f_82481_;
+         double d0 = camera.m_90583_().m_7096_();
+         double d1 = camera.m_90583_().m_7098_();
+         double d2 = camera.m_90583_().m_7094_();
          p_270649_.m_85836_();
          p_270649_.m_252880_((float)(p_270942_ - d0), (float)(p_270292_ - d1) + 0.07F, (float)(p_270885_ - d2));
          p_270649_.m_252931_((new Matrix4f()).rotation(camera.m_253121_()));
@@ -185,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public interface SimpleDebugRenderer {
-      void m_7790_(PoseStack p_113507_, MultiBufferSource p_113508_, double p_113509_, double p_113510_, double p_113511_);
+      void render(PoseStack p_113507_, MultiBufferSource p_113508_, me.alphamode.mclong.math.BigDecimal p_113509_, me.alphamode.mclong.math.BigDecimal p_113510_, me.alphamode.mclong.math.BigDecimal p_113511_);
 
       default void m_5630_() {
       }
