--- a/net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
+++ b/net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
@@ -35,18 +_,18 @@
       this.f_173816_ = p_173822_;
    }
 
-   public void m_7790_(PoseStack p_173846_, MultiBufferSource p_173847_, double p_173848_, double p_173849_, double p_173850_) {
+   public void render(PoseStack pPoseStack, MultiBufferSource pBuffer, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
       Level level = this.f_173816_.f_91073_;
       if (level == null) {
          this.f_173819_.clear();
          this.f_173820_.clear();
       } else {
-         Vec3 vec3 = new Vec3(p_173848_, 0.0D, p_173850_);
+         Vec3 vec3 = new Vec3(pCamX, me.alphamode.mclong.math.BigDecimal.ZERO, pCamZ);
          this.f_173819_.removeIf(GameEventListenerRenderer.TrackedGameEvent::m_173868_);
          this.f_173820_.removeIf((p_234512_) -> {
             return p_234512_.m_234542_(level, vec3);
          });
-         VertexConsumer vertexconsumer = p_173847_.m_6299_(RenderType.m_110504_());
+         VertexConsumer vertexconsumer = pBuffer.m_6299_(RenderType.m_110504_());
 
          for(GameEventListenerRenderer.TrackedListener gameeventlistenerrenderer$trackedlistener : this.f_173820_) {
             gameeventlistenerrenderer$trackedlistener.m_173875_(level).ifPresent((p_269731_) -> {
@@ -56,36 +_,36 @@
                double d10 = p_269731_.m_7096_() + (double)gameeventlistenerrenderer$trackedlistener.m_142078_();
                double d11 = p_269731_.m_7098_() + (double)gameeventlistenerrenderer$trackedlistener.m_142078_();
                double d12 = p_269731_.m_7094_() + (double)gameeventlistenerrenderer$trackedlistener.m_142078_();
-               LevelRenderer.m_285900_(p_173846_, vertexconsumer, Shapes.m_83064_(new AABB(d7, d8, d9, d10, d11, d12)), -p_173848_, -p_173849_, -p_173850_, 1.0F, 1.0F, 0.0F, 0.35F, true);
+               LevelRenderer.renderVoxelShape(pPoseStack, vertexconsumer, Shapes.m_83064_(new AABB(d7, d8, d9, d10, d11, d12)), pCamX.negate(), pCamY.negate(), pCamZ.negate(), 1.0F, 1.0F, 0.0F, 0.35F, true);
             });
          }
 
-         VertexConsumer vertexconsumer1 = p_173847_.m_6299_(RenderType.m_269313_());
+         VertexConsumer vertexconsumer1 = pBuffer.m_6299_(RenderType.m_269313_());
 
          for(GameEventListenerRenderer.TrackedListener gameeventlistenerrenderer$trackedlistener1 : this.f_173820_) {
             gameeventlistenerrenderer$trackedlistener1.m_173875_(level).ifPresent((p_269724_) -> {
-               LevelRenderer.m_269208_(p_173846_, vertexconsumer1, p_269724_.m_7096_() - 0.25D - p_173848_, p_269724_.m_7098_() - p_173849_, p_269724_.m_7094_() - 0.25D - p_173850_, p_269724_.m_7096_() + 0.25D - p_173848_, p_269724_.m_7098_() - p_173849_ + 1.0D, p_269724_.m_7094_() + 0.25D - p_173850_, 1.0F, 1.0F, 0.0F, 0.35F);
+               LevelRenderer.m_269208_(pPoseStack, vertexconsumer1, p_269724_.m_7096_() - 0.25D - pCamX.doubleValue(), p_269724_.m_7098_() - pCamY.doubleValue(), p_269724_.m_7094_() - 0.25D - pCamZ.doubleValue(), p_269724_.m_7096_() + 0.25D - pCamX.doubleValue(), p_269724_.m_7098_() - pCamY.doubleValue() + 1.0D, p_269724_.m_7094_() + 0.25D - pCamZ.doubleValue(), 1.0F, 1.0F, 0.0F, 0.35F);
             });
          }
 
          for(GameEventListenerRenderer.TrackedListener gameeventlistenerrenderer$trackedlistener2 : this.f_173820_) {
             gameeventlistenerrenderer$trackedlistener2.m_173875_(level).ifPresent((p_274713_) -> {
-               DebugRenderer.m_269569_(p_173846_, p_173847_, "Listener Origin", p_274713_.m_7096_(), p_274713_.m_7098_() + (double)1.8F, p_274713_.m_7094_(), -1, 0.025F);
-               DebugRenderer.m_269569_(p_173846_, p_173847_, BlockPos.m_274446_(p_274713_).toString(), p_274713_.m_7096_(), p_274713_.m_7098_() + 1.5D, p_274713_.m_7094_(), -6959665, 0.025F);
+               DebugRenderer.m_269569_(pPoseStack, pBuffer, "Listener Origin", p_274713_.m_7096_(), p_274713_.m_7098_() + (double)1.8F, p_274713_.m_7094_(), -1, 0.025F);
+               DebugRenderer.m_269569_(pPoseStack, pBuffer, BlockPos.containing(p_274713_).toString(), p_274713_.m_7096_(), p_274713_.m_7098_() + 1.5D, p_274713_.m_7094_(), -6959665, 0.025F);
             });
          }
 
          for(GameEventListenerRenderer.TrackedGameEvent gameeventlistenerrenderer$trackedgameevent : this.f_173819_) {
             Vec3 vec31 = gameeventlistenerrenderer$trackedgameevent.f_173863_;
             double d0 = (double)0.2F;
-            double d1 = vec31.f_82479_ - (double)0.2F;
-            double d2 = vec31.f_82480_ - (double)0.2F;
-            double d3 = vec31.f_82481_ - (double)0.2F;
-            double d4 = vec31.f_82479_ + (double)0.2F;
-            double d5 = vec31.f_82480_ + (double)0.2F + 0.5D;
-            double d6 = vec31.f_82481_ + (double)0.2F;
-            m_269429_(p_173846_, p_173847_, new AABB(d1, d2, d3, d4, d5, d6), 1.0F, 1.0F, 1.0F, 0.2F);
-            DebugRenderer.m_269569_(p_173846_, p_173847_, gameeventlistenerrenderer$trackedgameevent.f_173862_.m_157821_(), vec31.f_82479_, vec31.f_82480_ + (double)0.85F, vec31.f_82481_, -7564911, 0.0075F);
+            double d1 = vec31.m_7096_() - (double)0.2F;
+            double d2 = vec31.m_7098_() - (double)0.2F;
+            double d3 = vec31.m_7094_() - (double)0.2F;
+            double d4 = vec31.m_7096_() + (double)0.2F;
+            double d5 = vec31.m_7098_() + (double)0.2F + 0.5D;
+            double d6 = vec31.m_7094_() + (double)0.2F;
+            m_269429_(pPoseStack, pBuffer, new AABB(d1, d2, d3, d4, d5, d6), 1.0F, 1.0F, 1.0F, 0.2F);
+            DebugRenderer.m_269569_(pPoseStack, pBuffer, gameeventlistenerrenderer$trackedgameevent.f_173862_.m_157821_(), vec31.m_7096_(), vec31.m_7098_() + (double)0.85F, vec31.m_7094_(), -7564911, 0.0075F);
          }
 
       }
