--- a/net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
@@ -22,7 +_,7 @@
       this.f_113512_.clear();
    }
 
-   public void m_7790_(PoseStack p_113519_, MultiBufferSource p_113520_, double p_113521_, double p_113522_, double p_113523_) {
+   public void render(PoseStack p_113519_, MultiBufferSource p_113520_, me.alphamode.mclong.math.BigDecimal p_113521_, me.alphamode.mclong.math.BigDecimal p_113522_, me.alphamode.mclong.math.BigDecimal p_113523_) {
       long i = Util.m_137550_();
       this.f_113512_.entrySet().removeIf((p_113517_) -> {
          return i > (p_113517_.getValue()).f_113534_;
@@ -35,9 +_,9 @@
    private void m_269452_(PoseStack p_270274_, MultiBufferSource p_271018_, BlockPos p_270918_, GameTestDebugRenderer.Marker p_270827_) {
       DebugRenderer.m_269371_(p_270274_, p_271018_, p_270918_, 0.02F, p_270827_.m_113539_(), p_270827_.m_113540_(), p_270827_.m_113541_(), p_270827_.m_113542_() * 0.75F);
       if (!p_270827_.f_113533_.isEmpty()) {
-         double d0 = (double)p_270918_.m_123341_() + 0.5D;
-         double d1 = (double)p_270918_.m_123342_() + 1.2D;
-         double d2 = (double)p_270918_.m_123343_() + 0.5D;
+         double d0 = (double)p_270918_.getX() + 0.5D;
+         double d1 = (double)p_270918_.getY() + 1.2D;
+         double d2 = (double)p_270918_.getZ() + 0.5D;
          DebugRenderer.m_269439_(p_270274_, p_271018_, p_270827_.f_113533_, d0, d1, d2, -1, 0.01F, true, 0.0F, true);
       }
 
