--- a/net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
@@ -39,9 +_,9 @@
       RenderSystem.m_69472_();
       DebugRenderer.m_113463_(p_113530_, 0.02F, p_113531_.m_113539_(), p_113531_.m_113540_(), p_113531_.m_113541_(), p_113531_.m_113542_());
       if (!p_113531_.f_113533_.isEmpty()) {
-         double d0 = (double)p_113530_.m_123341_() + 0.5D;
-         double d1 = (double)p_113530_.m_123342_() + 1.2D;
-         double d2 = (double)p_113530_.m_123343_() + 0.5D;
+         double d0 = (double)p_113530_.getX() + 0.5D;
+         double d1 = (double)p_113530_.getY() + 1.2D;
+         double d2 = (double)p_113530_.getZ() + 0.5D;
          DebugRenderer.m_113490_(p_113531_.f_113533_, d0, d1, d2, -1, 0.01F, true, 0.0F, true);
       }
 
