--- a/net/minecraft/client/renderer/debug/GoalSelectorDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/GoalSelectorDebugRenderer.java
@@ -39,10 +_,10 @@
       this.f_113544_.forEach((p_269742_, p_269743_) -> {
          for(int i = 0; i < p_269743_.size(); ++i) {
             GoalSelectorDebugRenderer.DebugGoal goalselectordebugrenderer$debuggoal = p_269743_.get(i);
-            if (blockpos.m_123314_(goalselectordebugrenderer$debuggoal.f_113561_, 160.0D)) {
-               double d0 = (double)goalselectordebugrenderer$debuggoal.f_113561_.m_123341_() + 0.5D;
-               double d1 = (double)goalselectordebugrenderer$debuggoal.f_113561_.m_123342_() + 2.0D + (double)i * 0.25D;
-               double d2 = (double)goalselectordebugrenderer$debuggoal.f_113561_.m_123343_() + 0.5D;
+            if (blockpos.closerThan(goalselectordebugrenderer$debuggoal.f_113561_, 160.0D)) {
+               double d0 = (double)goalselectordebugrenderer$debuggoal.f_113561_.getX() + 0.5D;
+               double d1 = (double)goalselectordebugrenderer$debuggoal.f_113561_.getY() + 2.0D + (double)i * 0.25D;
+               double d2 = (double)goalselectordebugrenderer$debuggoal.f_113561_.getZ() + 0.5D;
                int j = goalselectordebugrenderer$debuggoal.f_113564_ ? -16711936 : -3355444;
                DebugRenderer.m_269271_(p_113552_, p_113553_, goalselectordebugrenderer$debuggoal.f_113563_, d0, d1, d2, j);
             }
