--- a/net/minecraft/client/renderer/debug/HeightMapRenderer.java
+++ b/net/minecraft/client/renderer/debug/HeightMapRenderer.java
@@ -27,14 +_,14 @@
       this.f_113570_ = p_113572_;
    }
 
-   public void m_7790_(PoseStack p_113576_, MultiBufferSource p_113577_, double p_113578_, double p_113579_, double p_113580_) {
+   public void render(PoseStack pPoseStack, MultiBufferSource pBuffer, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
       LevelAccessor levelaccessor = this.f_113570_.f_91073_;
-      VertexConsumer vertexconsumer = p_113577_.m_6299_(RenderType.m_269313_());
-      BlockPos blockpos = BlockPos.m_274561_(p_113578_, 0.0D, p_113580_);
+      VertexConsumer vertexconsumer = pBuffer.m_6299_(RenderType.m_269313_());
+      BlockPos blockpos = BlockPos.containing(pCamX, me.alphamode.mclong.math.BigDecimal.ZERO, pCamZ);
 
       for(int i = -2; i <= 2; ++i) {
          for(int j = -2; j <= 2; ++j) {
-            ChunkAccess chunkaccess = levelaccessor.m_46865_(blockpos.m_7918_(i * 16, 0, j * 16));
+            ChunkAccess chunkaccess = levelaccessor.m_46865_(blockpos.offset(i * 16, 0, j * 16));
 
             for(Map.Entry<Heightmap.Types, Heightmap> entry : chunkaccess.m_6890_()) {
                Heightmap.Types heightmap$types = entry.getKey();
@@ -43,10 +_,10 @@
 
                for(int k = 0; k < 16; ++k) {
                   for(int l = 0; l < 16; ++l) {
-                     int i1 = SectionPos.m_175554_(chunkpos.f_45578_, k);
-                     int j1 = SectionPos.m_175554_(chunkpos.f_45579_, l);
-                     float f = (float)((double)((float)levelaccessor.m_6924_(heightmap$types, i1, j1) + (float)heightmap$types.ordinal() * 0.09375F) - p_113579_);
-                     LevelRenderer.m_269208_(p_113576_, vertexconsumer, (double)((float)i1 + 0.25F) - p_113578_, (double)f, (double)((float)j1 + 0.25F) - p_113580_, (double)((float)i1 + 0.75F) - p_113578_, (double)(f + 0.09375F), (double)((float)j1 + 0.75F) - p_113580_, vector3f.x(), vector3f.y(), vector3f.z(), 1.0F);
+                     me.alphamode.mclong.math.BigInteger i1 = SectionPos.sectionToBlockCoord(chunkpos.getX(), k);
+                     me.alphamode.mclong.math.BigInteger j1 = SectionPos.sectionToBlockCoord(chunkpos.getZ(), l);
+                     float f = (float)((double)((float)levelaccessor.getHeight(heightmap$types, i1, j1) + (float)heightmap$types.ordinal() * 0.09375F) - pCamY.doubleValue());
+                     LevelRenderer.m_269208_(pPoseStack, vertexconsumer, (double)((float)i1.longValue() + 0.25F) - pCamX.doubleValue(), (double)f, (double)((float)j1.longValue() + 0.25F) - pCamZ.doubleValue(), (double)((float)i1.longValue() + 0.75F) - pCamX.doubleValue(), (double)(f + 0.09375F), (double)((float)j1.longValue() + 0.75F) - pCamZ.doubleValue(), vector3f.x(), vector3f.y(), vector3f.z(), 1.0F);
                   }
                }
             }
