--- a/net/minecraft/client/renderer/debug/LightDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/LightDebugRenderer.java
@@ -27,17 +_,17 @@
       BlockPos blockpos = BlockPos.m_274561_(p_113589_, p_113590_, p_113591_);
       LongSet longset = new LongOpenHashSet();
 
-      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.m_7918_(-10, -10, -10), blockpos.m_7918_(10, 10, 10))) {
+      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.offset(-10, -10, -10), blockpos.offset(10, 10, 10))) {
          int i = level.m_45517_(LightLayer.SKY, blockpos1);
          float f = (float)(15 - i) / 15.0F * 0.5F + 0.16F;
          int j = Mth.m_14169_(f, 0.9F, 0.9F);
          long k = SectionPos.m_123235_(blockpos1.m_121878_());
          if (longset.add(k)) {
-            DebugRenderer.m_269569_(p_113587_, p_113588_, level.m_7726_().m_7827_().m_75816_(LightLayer.SKY, SectionPos.m_123184_(k)), (double)SectionPos.m_175554_(SectionPos.m_123213_(k), 8), (double)SectionPos.m_175554_(SectionPos.m_123225_(k), 8), (double)SectionPos.m_175554_(SectionPos.m_123230_(k), 8), 16711680, 0.3F);
+            DebugRenderer.m_269569_(p_113587_, p_113588_, level.m_7726_().m_7827_().m_75816_(LightLayer.SKY, SectionPos.m_123184_(k)), (double)SectionPos.sectionToBlockCoord(SectionPos.x(k), 8), (double)SectionPos.sectionToBlockCoord(SectionPos.y(k), 8), (double)SectionPos.sectionToBlockCoord(SectionPos.z(k), 8), 16711680, 0.3F);
          }
 
          if (i != 15) {
-            DebugRenderer.m_269271_(p_113587_, p_113588_, String.valueOf(i), (double)blockpos1.m_123341_() + 0.5D, (double)blockpos1.m_123342_() + 0.25D, (double)blockpos1.m_123343_() + 0.5D, j);
+            DebugRenderer.m_269271_(p_113587_, p_113588_, String.valueOf(i), (double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.25D, (double)blockpos1.getZ() + 0.5D, j);
          }
       }
 
