--- a/net/minecraft/client/renderer/debug/PathfindingRenderer.java
+++ b/net/minecraft/client/renderer/debug/PathfindingRenderer.java
@@ -60,7 +_,7 @@
       m_269170_(p_270399_, p_270359_.m_6299_(RenderType.m_269399_(6.0D)), p_270189_, p_270187_, p_270252_, p_270371_);
       BlockPos blockpos = p_270189_.m_77406_();
       if (m_113634_(blockpos, p_270187_, p_270252_, p_270371_) <= 80.0F) {
-         DebugRenderer.m_269311_(p_270399_, p_270359_, (new AABB((double)((float)blockpos.m_123341_() + 0.25F), (double)((float)blockpos.m_123342_() + 0.25F), (double)blockpos.m_123343_() + 0.25D, (double)((float)blockpos.m_123341_() + 0.75F), (double)((float)blockpos.m_123342_() + 0.75F), (double)((float)blockpos.m_123343_() + 0.75F))).m_82386_(-p_270187_, -p_270252_, -p_270371_), 0.0F, 1.0F, 0.0F, 0.5F);
+         DebugRenderer.m_269311_(p_270399_, p_270359_, (new AABB((double)((float)blockpos.getX() + 0.25F), (double)((float)blockpos.getY() + 0.25F), (double)blockpos.getZ() + 0.25D, (double)((float)blockpos.getX() + 0.75F), (double)((float)blockpos.getY() + 0.75F), (double)((float)blockpos.getZ() + 0.75F))).m_82386_(-p_270187_, -p_270252_, -p_270371_), 0.0F, 1.0F, 0.0F, 0.5F);
 
          for(int i = 0; i < p_270189_.m_77398_(); ++i) {
             Node node = p_270189_.m_77375_(i);
@@ -114,6 +_,6 @@
    }
 
    private static float m_113634_(BlockPos p_113635_, double p_113636_, double p_113637_, double p_113638_) {
-      return (float)(Math.abs((double)p_113635_.m_123341_() - p_113636_) + Math.abs((double)p_113635_.m_123342_() - p_113637_) + Math.abs((double)p_113635_.m_123343_() - p_113638_));
+      return (float)(Math.abs((double)p_113635_.getX() - p_113636_) + Math.abs((double)p_113635_.getY() - p_113637_) + Math.abs((double)p_113635_.getZ() - p_113638_));
    }
 }
