--- a/net/minecraft/client/renderer/debug/RaidDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/RaidDebugRenderer.java
@@ -25,27 +_,27 @@
       this.f_113648_ = p_113664_;
    }
 
-   public void m_7790_(PoseStack p_113652_, MultiBufferSource p_113653_, double p_113654_, double p_113655_, double p_113656_) {
+   public void render(PoseStack pPoseStack, MultiBufferSource pBuffer, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
       BlockPos blockpos = this.m_113665_().m_90588_();
 
       for(BlockPos blockpos1 : this.f_113648_) {
-         if (blockpos.m_123314_(blockpos1, 160.0D)) {
-            m_269099_(p_113652_, p_113653_, blockpos1);
+         if (blockpos.closerThan(blockpos1, 160.0D)) {
+            m_269099_(pPoseStack, pBuffer, blockpos1);
          }
       }
 
    }
 
    private static void m_269099_(PoseStack p_270914_, MultiBufferSource p_270517_, BlockPos p_270208_) {
-      DebugRenderer.m_269451_(p_270914_, p_270517_, p_270208_.m_7918_(-1, -1, -1), p_270208_.m_7918_(1, 1, 1), 1.0F, 0.0F, 0.0F, 0.15F);
+      DebugRenderer.m_269451_(p_270914_, p_270517_, p_270208_.offset(-1, -1, -1), p_270208_.offset(1, 1, 1), 1.0F, 0.0F, 0.0F, 0.15F);
       int i = -65536;
       m_269257_(p_270914_, p_270517_, "Raid center", p_270208_, -65536);
    }
 
    private static void m_269257_(PoseStack p_270092_, MultiBufferSource p_270518_, String p_270237_, BlockPos p_270941_, int p_270307_) {
-      double d0 = (double)p_270941_.m_123341_() + 0.5D;
-      double d1 = (double)p_270941_.m_123342_() + 1.3D;
-      double d2 = (double)p_270941_.m_123343_() + 0.5D;
+      double d0 = (double)p_270941_.getX() + 0.5D;
+      double d1 = (double)p_270941_.getY() + 1.3D;
+      double d2 = (double)p_270941_.getZ() + 0.5D;
       DebugRenderer.m_269439_(p_270092_, p_270518_, p_270237_, d0, d1, d2, p_270307_, 0.04F, true, 0.0F, true);
    }
 
