--- a/net/minecraft/client/renderer/debug/SolidFaceRenderer.java
+++ b/net/minecraft/client/renderer/debug/SolidFaceRenderer.java
@@ -24,24 +_,24 @@
       this.f_113666_ = p_113668_;
    }
 
-   public void m_7790_(PoseStack p_113670_, MultiBufferSource p_113671_, double p_113672_, double p_113673_, double p_113674_) {
+   public void render(PoseStack p_113670_, MultiBufferSource p_113671_, me.alphamode.mclong.math.BigDecimal p_113672_, me.alphamode.mclong.math.BigDecimal p_113673_, me.alphamode.mclong.math.BigDecimal p_113674_) {
       Matrix4f matrix4f = p_113670_.m_85850_().m_252922_();
       BlockGetter blockgetter = this.f_113666_.f_91074_.m_9236_();
-      BlockPos blockpos = BlockPos.m_274561_(p_113672_, p_113673_, p_113674_);
+      BlockPos blockpos = BlockPos.containing(p_113672_, p_113673_, p_113674_);
 
-      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.m_7918_(-6, -6, -6), blockpos.m_7918_(6, 6, 6))) {
+      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.offset(-6, -6, -6), blockpos.offset(6, 6, 6))) {
          BlockState blockstate = blockgetter.m_8055_(blockpos1);
          if (!blockstate.m_60713_(Blocks.f_50016_)) {
             VoxelShape voxelshape = blockstate.m_60808_(blockgetter, blockpos1);
 
             for(AABB aabb : voxelshape.m_83299_()) {
                AABB aabb1 = aabb.m_82338_(blockpos1).m_82400_(0.002D);
-               float f = (float)(aabb1.f_82288_ - p_113672_);
-               float f1 = (float)(aabb1.f_82289_ - p_113673_);
-               float f2 = (float)(aabb1.f_82290_ - p_113674_);
-               float f3 = (float)(aabb1.f_82291_ - p_113672_);
-               float f4 = (float)(aabb1.f_82292_ - p_113673_);
-               float f5 = (float)(aabb1.f_82293_ - p_113674_);
+               float f = (aabb1.f_82288_.subtract(p_113672_)).floatValue();
+               float f1 = (aabb1.f_82289_.subtract(p_113673_)).floatValue();
+               float f2 = (aabb1.f_82290_.subtract(p_113674_)).floatValue();
+               float f3 = (aabb1.f_82291_.subtract(p_113672_)).floatValue();
+               float f4 = (aabb1.f_82292_.subtract(p_113673_)).floatValue();
+               float f5 = (aabb1.f_82293_.subtract(p_113674_)).floatValue();
                float f6 = 1.0F;
                float f7 = 0.0F;
                float f8 = 0.0F;
