--- a/net/minecraft/client/renderer/debug/SolidFaceRenderer.java
+++ b/net/minecraft/client/renderer/debug/SolidFaceRenderer.java
@@ -29,7 +_,7 @@
       BlockGetter blockgetter = this.f_113666_.f_91074_.f_19853_;
       BlockPos blockpos = BlockPos.m_274561_(p_113672_, p_113673_, p_113674_);
 
-      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.m_7918_(-6, -6, -6), blockpos.m_7918_(6, 6, 6))) {
+      for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.offset(-6, -6, -6), blockpos.offset(6, 6, 6))) {
          BlockState blockstate = blockgetter.m_8055_(blockpos1);
          if (!blockstate.m_60713_(Blocks.f_50016_)) {
             VoxelShape voxelshape = blockstate.m_60808_(blockgetter, blockpos1);
