--- a/net/minecraft/client/renderer/debug/SupportBlockRenderer.java
+++ b/net/minecraft/client/renderer/debug/SupportBlockRenderer.java
@@ -70,13 +_,13 @@
    }
 
    private void m_285847_(BlockPos p_286268_, PoseStack p_286592_, double p_286463_, double p_286552_, double p_286660_, MultiBufferSource p_286314_, double p_286880_, float p_286918_, float p_286304_, float p_286672_) {
-      double d0 = (double)p_286268_.m_123341_() - p_286463_ - 2.0D * p_286880_;
-      double d1 = (double)p_286268_.m_123342_() - p_286552_ - 2.0D * p_286880_;
-      double d2 = (double)p_286268_.m_123343_() - p_286660_ - 2.0D * p_286880_;
+      double d0 = (double)p_286268_.getX() - p_286463_ - 2.0D * p_286880_;
+      double d1 = (double)p_286268_.getY() - p_286552_ - 2.0D * p_286880_;
+      double d2 = (double)p_286268_.getZ() - p_286660_ - 2.0D * p_286880_;
       double d3 = d0 + 1.0D + 4.0D * p_286880_;
       double d4 = d1 + 1.0D + 4.0D * p_286880_;
       double d5 = d2 + 1.0D + 4.0D * p_286880_;
       LevelRenderer.m_109608_(p_286592_, p_286314_.m_6299_(RenderType.m_110504_()), d0, d1, d2, d3, d4, d5, p_286918_, p_286304_, p_286672_, 0.4F);
-      LevelRenderer.m_285900_(p_286592_, p_286314_.m_6299_(RenderType.m_110504_()), this.f_285628_.f_91073_.m_8055_(p_286268_).m_60742_(this.f_285628_.f_91073_, p_286268_, CollisionContext.m_82749_()).m_83216_((double)p_286268_.m_123341_(), (double)p_286268_.m_123342_(), (double)p_286268_.m_123343_()), -p_286463_, -p_286552_, -p_286660_, p_286918_, p_286304_, p_286672_, 1.0F, false);
+      LevelRenderer.renderVoxelShape(p_286592_, p_286314_.m_6299_(RenderType.m_110504_()), this.f_285628_.f_91073_.m_8055_(p_286268_).m_60742_(this.f_285628_.f_91073_, p_286268_, CollisionContext.m_82749_()).move(p_286268_.getBigDecX(), p_286268_.getBigDecY(), p_286268_.getBigDecZ()), me.alphamode.mclong.math.BigDecimal.valueOf(-p_286463_), me.alphamode.mclong.math.BigDecimal.valueOf(-p_286552_), me.alphamode.mclong.math.BigDecimal.valueOf(-p_286660_), p_286918_, p_286304_, p_286672_, 1.0F, false);
    }
 }
