--- a/net/minecraft/client/renderer/debug/WaterDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/WaterDebugRenderer.java
@@ -32,15 +_,15 @@
       for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.m_7918_(-10, -10, -10), blockpos.m_7918_(10, 10, 10))) {
          FluidState fluidstate = levelreader.m_6425_(blockpos1);
          if (fluidstate.m_205070_(FluidTags.f_13131_)) {
-            double d0 = (double)((float)blockpos1.m_123342_() + fluidstate.m_76155_(levelreader, blockpos1));
-            DebugRenderer.m_113451_((new AABB((double)((float)blockpos1.m_123341_() + 0.01F), (double)((float)blockpos1.m_123342_() + 0.01F), (double)((float)blockpos1.m_123343_() + 0.01F), (double)((float)blockpos1.m_123341_() + 0.99F), d0, (double)((float)blockpos1.m_123343_() + 0.99F))).m_82386_(-p_113721_, -p_113722_, -p_113723_), 1.0F, 1.0F, 1.0F, 0.2F);
+            double d0 = (double)((float)blockpos1.getY() + fluidstate.m_76155_(levelreader, blockpos1));
+            DebugRenderer.m_113451_((new AABB((double)((float)blockpos1.getX() + 0.01F), (double)((float)blockpos1.getY() + 0.01F), (double)((float)blockpos1.getZ() + 0.01F), (double)((float)blockpos1.getX() + 0.99F), d0, (double)((float)blockpos1.getZ() + 0.99F))).m_82386_(-p_113721_, -p_113722_, -p_113723_), 1.0F, 1.0F, 1.0F, 0.2F);
          }
       }
 
       for(BlockPos blockpos2 : BlockPos.m_121940_(blockpos.m_7918_(-10, -10, -10), blockpos.m_7918_(10, 10, 10))) {
          FluidState fluidstate1 = levelreader.m_6425_(blockpos2);
          if (fluidstate1.m_205070_(FluidTags.f_13131_)) {
-            DebugRenderer.m_113477_(String.valueOf(fluidstate1.m_76186_()), (double)blockpos2.m_123341_() + 0.5D, (double)((float)blockpos2.m_123342_() + fluidstate1.m_76155_(levelreader, blockpos2)), (double)blockpos2.m_123343_() + 0.5D, -16777216);
+            DebugRenderer.m_113477_(String.valueOf(fluidstate1.m_76186_()), (double)blockpos2.getX() + 0.5D, (double)((float)blockpos2.getY() + fluidstate1.m_76155_(levelreader, blockpos2)), (double)blockpos2.getZ() + 0.5D, -16777216);
          }
       }
 
