--- a/net/minecraft/world/level/material/FluidState.java
+++ b/net/minecraft/world/level/material/FluidState.java
@@ -61,7 +_,7 @@
    public boolean m_76171_(BlockGetter p_76172_, BlockPos p_76173_) {
       for(int i = -1; i <= 1; ++i) {
          for(int j = -1; j <= 1; ++j) {
-            BlockPos blockpos = p_76173_.m_7918_(i, 0, j);
+            BlockPos blockpos = p_76173_.offset(i, 0, j);
             FluidState fluidstate = p_76172_.m_6425_(blockpos);
             if (!fluidstate.m_76152_().m_6212_(this.m_76152_()) && !p_76172_.m_8055_(blockpos).m_60804_(p_76172_, blockpos)) {
                return true;
