--- a/net/minecraft/world/level/material/WaterFluid.java
+++ b/net/minecraft/world/level/material/WaterFluid.java
@@ -41,10 +_,10 @@
    public void m_213811_(Level p_230606_, BlockPos p_230607_, FluidState p_230608_, RandomSource p_230609_) {
       if (!p_230608_.m_76170_() && !p_230608_.m_61143_(f_75947_)) {
          if (p_230609_.m_188503_(64) == 0) {
-            p_230606_.m_7785_((double)p_230607_.m_123341_() + 0.5D, (double)p_230607_.m_123342_() + 0.5D, (double)p_230607_.m_123343_() + 0.5D, SoundEvents.f_12540_, SoundSource.BLOCKS, p_230609_.m_188501_() * 0.25F + 0.75F, p_230609_.m_188501_() + 0.5F, false);
+            p_230606_.playLocalSound(p_230607_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_230607_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_230607_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), SoundEvents.f_12540_, SoundSource.BLOCKS, p_230609_.m_188501_() * 0.25F + 0.75F, p_230609_.m_188501_() + 0.5F, false);
          }
       } else if (p_230609_.m_188503_(10) == 0) {
-         p_230606_.m_7106_(ParticleTypes.f_123768_, (double)p_230607_.m_123341_() + p_230609_.m_188500_(), (double)p_230607_.m_123342_() + p_230609_.m_188500_(), (double)p_230607_.m_123343_() + p_230609_.m_188500_(), 0.0D, 0.0D, 0.0D);
+         p_230606_.addParticle(ParticleTypes.f_123768_, p_230607_.getBigDecX().add(p_230609_.m_188500_()), p_230607_.getBigDecY().add(p_230609_.m_188500_()), p_230607_.getBigDecZ().add(p_230609_.m_188500_()), 0.0D, 0.0D, 0.0D);
       }
 
    }
