--- a/net/minecraft/CrashReportCategory.java
+++ b/net/minecraft/CrashReportCategory.java
@@ -23,7 +_,7 @@
    }
 
    public static String m_178947_(LevelHeightAccessor p_178948_, BlockPos p_178949_) {
-      return m_178942_(p_178948_, p_178949_.m_123341_(), p_178949_.m_123342_(), p_178949_.m_123343_());
+      return m_178937_(p_178948_, p_178949_.getX(), p_178949_.getY(), p_178949_.getZ());
    }
 
    public static String m_178942_(LevelHeightAccessor p_178943_, int p_178944_, int p_178945_, int p_178946_) {
@@ -38,18 +_,18 @@
       stringbuilder.append(", ");
 
       try {
-         int i = SectionPos.m_123171_(p_178944_);
-         int j = SectionPos.m_123171_(p_178945_);
-         int k = SectionPos.m_123171_(p_178946_);
+         long i = SectionPos.blockToSectionCoord(p_178944_);
+         long j = SectionPos.blockToSectionCoord(p_178945_);
+         long k = SectionPos.blockToSectionCoord(p_178946_);
          int l = p_178944_ & 15;
          int i1 = p_178945_ & 15;
          int j1 = p_178946_ & 15;
-         int k1 = SectionPos.m_123223_(i);
-         int l1 = p_178943_.m_141937_();
-         int i2 = SectionPos.m_123223_(k);
-         int j2 = SectionPos.m_123223_(i + 1) - 1;
-         int k2 = p_178943_.m_151558_() - 1;
-         int l2 = SectionPos.m_123223_(k + 1) - 1;
+         long k1 = SectionPos.sectionToBlockCoord(i);
+         long l1 = p_178943_.getMinBuildHeight();
+         long i2 = SectionPos.sectionToBlockCoord(k);
+         long j2 = SectionPos.sectionToBlockCoord(i + 1) - 1;
+         long k2 = p_178943_.getMaxBuildHeight() - 1;
+         long l2 = SectionPos.sectionToBlockCoord(k + 1) - 1;
          stringbuilder.append(String.format(Locale.ROOT, "Section: (at %d,%d,%d in %d,%d,%d; chunk contains blocks %d,%d,%d to %d,%d,%d)", l, i1, j1, i, j, k, k1, l1, i2, j2, k2, l2));
       } catch (Throwable throwable1) {
          stringbuilder.append("(Error finding chunk loc)");
@@ -65,10 +_,10 @@
          int i4 = (i3 + 1 << 5) - 1;
          int j4 = (j3 + 1 << 5) - 1;
          int k4 = i3 << 9;
-         int l4 = p_178943_.m_141937_();
+         long l4 = p_178943_.getMinBuildHeight();
          int i5 = j3 << 9;
          int j5 = (i3 + 1 << 9) - 1;
-         int k5 = p_178943_.m_151558_() - 1;
+         long k5 = p_178943_.getMaxBuildHeight() - 1;
          int l5 = (j3 + 1 << 9) - 1;
          stringbuilder.append(String.format(Locale.ROOT, "Region: (%d,%d; contains chunks %d,%d to %d,%d, blocks %d,%d,%d to %d,%d,%d)", i3, j3, k3, l3, i4, j4, k4, l4, i5, j5, k5, l5));
       } catch (Throwable throwable) {
