--- a/net/minecraft/SharedConstants.java
+++ b/net/minecraft/SharedConstants.java
@@ -186,8 +_,8 @@
    }
 
    public static boolean m_183707_(ChunkPos p_183708_) {
-      int i = p_183708_.m_45604_();
-      int j = p_183708_.m_45605_();
+      long i = p_183708_.getMinBlockX();
+      long j = p_183708_.getMinBlockZ();
       if (!f_183698_) {
          return false;
       } else {
@@ -196,18 +_,18 @@
    }
 
    public static void m_214358_() {
-      DataFixerOptimizationOption datafixeroptimizationoption;
-      switch (f_214354_) {
-         case INITIALIZED_UNOPTIMIZED:
-            throw new IllegalStateException("Tried to enable datafixer optimization after unoptimized initialization");
-         case INITIALIZED_OPTIMIZED:
-            datafixeroptimizationoption = DataFixerOptimizationOption.INITIALIZED_OPTIMIZED;
-            break;
-         default:
-            datafixeroptimizationoption = DataFixerOptimizationOption.UNINITIALIZED_OPTIMIZED;
-      }
-
-      f_214354_ = datafixeroptimizationoption;
+//      DataFixerOptimizationOption datafixeroptimizationoption;
+//      switch (DATAFIXER_OPTIMIZATION_OPTION) {
+//         case INITIALIZED_UNOPTIMIZED:
+//            throw new IllegalStateException("Tried to enable datafixer optimization after unoptimized initialization");
+//         case INITIALIZED_OPTIMIZED:
+//            datafixeroptimizationoption = DataFixerOptimizationOption.INITIALIZED_OPTIMIZED;
+//            break;
+//         default:
+//            datafixeroptimizationoption = DataFixerOptimizationOption.UNINITIALIZED_OPTIMIZED;
+//      }
+//
+//      DATAFIXER_OPTIMIZATION_OPTION = datafixeroptimizationoption;
    }
 
    static {
