--- a/net/minecraft/SharedConstants.java
+++ b/net/minecraft/SharedConstants.java
@@ -23,7 +_,7 @@
    public static final String f_183702_ = "main";
    /** @deprecated */
    @Deprecated
-   public static final String f_142952_ = "1.20.1";
+   public static final String f_142952_ = "1.20.1 Long";
    /** @deprecated */
    @Deprecated
    public static final int f_142954_ = 763;
@@ -127,7 +_,7 @@
    public static final long f_136181_ = Duration.ofMillis(300L).toNanos();
    public static final boolean f_279621_ = false;
    public static boolean f_136182_ = true;
-   public static boolean f_136183_;
+   public static boolean f_136183_ = true;
    public static Set<DSL.TypeReference> f_273866_ = Set.of();
    public static final int f_142916_ = 16;
    public static final int f_142917_ = 256;
@@ -196,8 +_,8 @@
    }
 
    public static boolean m_183707_(ChunkPos p_183708_) {
-      int i = p_183708_.m_45604_();
-      int j = p_183708_.m_45605_();
+      long i = p_183708_.getMinBlockX();
+      long j = p_183708_.getMinBlockZ();
       if (!f_183698_) {
          return false;
       } else {
