--- a/net/minecraft/SharedConstants.java
+++ b/net/minecraft/SharedConstants.java
@@ -23,7 +_,7 @@
    public static final String f_183702_ = "main";
    /** @deprecated */
    @Deprecated
-   public static final String f_142952_ = "1.20.1";
+   public static final String f_142952_ = "1.20.1 Long";
    /** @deprecated */
    @Deprecated
    public static final int f_142954_ = 763;
@@ -196,8 +_,8 @@
    }
 
    public static boolean m_183707_(ChunkPos p_183708_) {
-      int i = p_183708_.m_45604_();
-      int j = p_183708_.m_45605_();
+      long i = p_183708_.getMinBlockX();
+      long j = p_183708_.getMinBlockZ();
       if (!f_183698_) {
          return false;
       } else {
