--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -199,20 +_,20 @@
    private long f_9749_;
    private int f_9750_;
    private int f_9751_;
-   private double f_9753_;
-   private double f_9754_;
-   private double f_9755_;
-   private double f_9756_;
-   private double f_9757_;
-   private double f_9758_;
+   private me.alphamode.mclong.math.BigDecimal f_9753_;
+   private me.alphamode.mclong.math.BigDecimal f_9754_;
+   private me.alphamode.mclong.math.BigDecimal f_9755_;
+   private me.alphamode.mclong.math.BigDecimal f_9756_;
+   private me.alphamode.mclong.math.BigDecimal f_9757_;
+   private me.alphamode.mclong.math.BigDecimal f_9758_;
    @Nullable
    private Entity f_9759_;
-   private double f_9760_;
-   private double f_9761_;
-   private double f_9762_;
-   private double f_9763_;
-   private double f_9764_;
-   private double f_9765_;
+   private me.alphamode.mclong.math.BigDecimal f_9760_;
+   private me.alphamode.mclong.math.BigDecimal f_9761_;
+   private me.alphamode.mclong.math.BigDecimal f_9762_;
+   private me.alphamode.mclong.math.BigDecimal f_9763_;
+   private me.alphamode.mclong.math.BigDecimal f_9764_;
+   private me.alphamode.mclong.math.BigDecimal f_9765_;
    @Nullable
    private Vec3 f_9766_;
    private int f_9767_;
@@ -250,11 +_,11 @@
       }
 
       this.m_9953_();
-      this.f_9743_.f_19854_ = this.f_9743_.m_20185_();
-      this.f_9743_.f_19855_ = this.f_9743_.m_20186_();
-      this.f_9743_.f_19856_ = this.f_9743_.m_20189_();
+      this.f_9743_.f_19854_ = this.f_9743_.getBigX();
+      this.f_9743_.f_19855_ = this.f_9743_.getBigY();
+      this.f_9743_.f_19856_ = this.f_9743_.getBigZ();
       this.f_9743_.m_9240_();
-      this.f_9743_.m_19890_(this.f_9753_, this.f_9754_, this.f_9755_, this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
+      this.f_9743_.absMoveTo(this.f_9753_, this.f_9754_, this.f_9755_, this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
       ++this.f_9746_;
       this.f_9741_ = this.f_9740_;
       if (this.f_9736_ && !this.f_9743_.m_5803_() && !this.f_9743_.m_20159_() && !this.f_9743_.m_21224_()) {
@@ -270,12 +_,12 @@
 
       this.f_9759_ = this.f_9743_.m_20201_();
       if (this.f_9759_ != this.f_9743_ && this.f_9759_.m_6688_() == this.f_9743_) {
-         this.f_9760_ = this.f_9759_.m_20185_();
-         this.f_9761_ = this.f_9759_.m_20186_();
-         this.f_9762_ = this.f_9759_.m_20189_();
-         this.f_9763_ = this.f_9759_.m_20185_();
-         this.f_9764_ = this.f_9759_.m_20186_();
-         this.f_9765_ = this.f_9759_.m_20189_();
+         this.f_9760_ = this.f_9759_.getBigX();
+         this.f_9761_ = this.f_9759_.getBigY();
+         this.f_9762_ = this.f_9759_.getBigZ();
+         this.f_9763_ = this.f_9759_.getBigX();
+         this.f_9764_ = this.f_9759_.getBigY();
+         this.f_9765_ = this.f_9759_.getBigZ();
          if (this.f_9738_ && this.f_9743_.m_20201_().m_6688_() == this.f_9743_) {
             if (++this.f_9739_ > 80) {
                f_9744_.warn("{} was kicked for floating a vehicle too long!", (Object)this.f_9743_.m_7755_().getString());
@@ -321,12 +_,12 @@
    }
 
    public void m_9953_() {
-      this.f_9753_ = this.f_9743_.m_20185_();
-      this.f_9754_ = this.f_9743_.m_20186_();
-      this.f_9755_ = this.f_9743_.m_20189_();
-      this.f_9756_ = this.f_9743_.m_20185_();
-      this.f_9757_ = this.f_9743_.m_20186_();
-      this.f_9758_ = this.f_9743_.m_20189_();
+      this.f_9753_ = this.f_9743_.getBigX();
+      this.f_9754_ = this.f_9743_.getBigY();
+      this.f_9755_ = this.f_9743_.getBigZ();
+      this.f_9756_ = this.f_9743_.getBigX();
+      this.f_9757_ = this.f_9743_.getBigY();
+      this.f_9758_ = this.f_9743_.getBigZ();
    }
 
    public boolean m_6198_() {
@@ -369,37 +_,37 @@
       this.f_9743_.m_8980_(p_9893_.m_134355_(), p_9893_.m_134358_(), p_9893_.m_134359_(), p_9893_.m_134360_());
    }
 
-   private static boolean m_143663_(double p_143664_, double p_143665_, double p_143666_, float p_143667_, float p_143668_) {
-      return Double.isNaN(p_143664_) || Double.isNaN(p_143665_) || Double.isNaN(p_143666_) || !Floats.isFinite(p_143668_) || !Floats.isFinite(p_143667_);
-   }
-
-   private static double m_143609_(double p_143610_) {
-      return Mth.m_14008_(p_143610_, -3.0E7D, 3.0E7D);
-   }
-
-   private static double m_143653_(double p_143654_) {
-      return Mth.m_14008_(p_143654_, -2.0E7D, 2.0E7D);
+   private static boolean containsInvalidValues(me.alphamode.mclong.math.BigDecimal p_143664_, me.alphamode.mclong.math.BigDecimal p_143665_, me.alphamode.mclong.math.BigDecimal p_143666_, float p_143667_, float p_143668_) {
+      return !Floats.isFinite(p_143668_) || !Floats.isFinite(p_143667_);
+   }
+
+   private static me.alphamode.mclong.math.BigDecimal clampHorizontal(me.alphamode.mclong.math.BigDecimal p_143610_) {
+      return p_143610_;
+   }
+
+   private static me.alphamode.mclong.math.BigDecimal clampVertical(me.alphamode.mclong.math.BigDecimal p_143654_) {
+      return p_143654_;
    }
 
    public void m_5659_(ServerboundMoveVehiclePacket p_9876_) {
       PacketUtils.m_131359_(p_9876_, this, this.f_9743_.m_284548_());
-      if (m_143663_(p_9876_.m_134199_(), p_9876_.m_134202_(), p_9876_.m_134203_(), p_9876_.m_134204_(), p_9876_.m_134205_())) {
+      if (containsInvalidValues(p_9876_.getX(), p_9876_.getY(), p_9876_.getZ(), p_9876_.m_134204_(), p_9876_.m_134205_())) {
          this.m_9942_(Component.m_237115_("multiplayer.disconnect.invalid_vehicle_movement"));
       } else {
          Entity entity = this.f_9743_.m_20201_();
          if (entity != this.f_9743_ && entity.m_6688_() == this.f_9743_ && entity == this.f_9759_) {
             ServerLevel serverlevel = this.f_9743_.m_284548_();
-            double d0 = entity.m_20185_();
-            double d1 = entity.m_20186_();
-            double d2 = entity.m_20189_();
-            double d3 = m_143609_(p_9876_.m_134199_());
-            double d4 = m_143653_(p_9876_.m_134202_());
-            double d5 = m_143609_(p_9876_.m_134203_());
+            me.alphamode.mclong.math.BigDecimal d0 = entity.getBigX();
+            me.alphamode.mclong.math.BigDecimal d1 = entity.getBigY();
+            me.alphamode.mclong.math.BigDecimal d2 = entity.getBigZ();
+            me.alphamode.mclong.math.BigDecimal d3 = clampHorizontal(p_9876_.getX());
+            me.alphamode.mclong.math.BigDecimal d4 = clampVertical(p_9876_.getY());
+            me.alphamode.mclong.math.BigDecimal d5 = clampHorizontal(p_9876_.getZ());
             float f = Mth.m_14177_(p_9876_.m_134204_());
             float f1 = Mth.m_14177_(p_9876_.m_134205_());
-            double d6 = d3 - this.f_9760_;
-            double d7 = d4 - this.f_9761_;
-            double d8 = d5 - this.f_9762_;
+            double d6 = d3.subtract(this.f_9760_).doubleValue();
+            double d7 = d4.subtract(this.f_9761_).doubleValue();
+            double d8 = d5.subtract(this.f_9762_).doubleValue();
             double d9 = entity.m_20184_().m_82556_();
             double d10 = d6 * d6 + d7 * d7 + d8 * d8;
             if (d10 - d9 > 100.0D && !this.m_9956_()) {
@@ -409,9 +_,9 @@
             }
 
             boolean flag = serverlevel.m_45756_(entity, entity.m_20191_().m_82406_(0.0625D));
-            d6 = d3 - this.f_9763_;
-            d7 = d4 - this.f_9764_ - 1.0E-6D;
-            d8 = d5 - this.f_9765_;
+            d6 = d3.subtract(this.f_9763_).doubleValue();
+            d7 = d4.subtract(this.f_9764_).doubleValue() - 1.0E-6D;
+            d8 = d5.subtract(this.f_9765_).doubleValue();
             boolean flag1 = entity.f_201939_;
             if (entity instanceof LivingEntity) {
                LivingEntity livingentity = (LivingEntity)entity;
@@ -421,13 +_,13 @@
             }
 
             entity.m_6478_(MoverType.PLAYER, new Vec3(d6, d7, d8));
-            d6 = d3 - entity.m_20185_();
-            d7 = d4 - entity.m_20186_();
+            d6 = d3.subtract(entity.getBigX()).doubleValue();
+            d7 = d4.subtract(entity.getBigY()).doubleValue();
             if (d7 > -0.5D || d7 < 0.5D) {
                d7 = 0.0D;
             }
 
-            d8 = d5 - entity.m_20189_();
+            d8 = d5.subtract(entity.getBigZ()).doubleValue();
             d10 = d6 * d6 + d7 * d7 + d8 * d8;
             boolean flag2 = false;
             if (d10 > 0.0625D) {
@@ -435,20 +_,20 @@
                f_9744_.warn("{} (vehicle of {}) moved wrongly! {}", entity.m_7755_().getString(), this.f_9743_.m_7755_().getString(), Math.sqrt(d10));
             }
 
-            entity.m_19890_(d3, d4, d5, f, f1);
+            entity.absMoveTo(d3, d4, d5, f, f1);
             boolean flag3 = serverlevel.m_45756_(entity, entity.m_20191_().m_82406_(0.0625D));
             if (flag && (flag2 || !flag3)) {
-               entity.m_19890_(d0, d1, d2, f, f1);
+               entity.absMoveTo(d0, d1, d2, f, f1);
                this.f_9742_.m_129512_(new ClientboundMoveVehiclePacket(entity));
                return;
             }
 
             this.f_9743_.m_284548_().m_7726_().m_8385_(this.f_9743_);
-            this.f_9743_.m_36378_(this.f_9743_.m_20185_() - d0, this.f_9743_.m_20186_() - d1, this.f_9743_.m_20189_() - d2);
+            this.f_9743_.m_36378_(this.f_9743_.getBigX().subtract(d0).doubleValue(), this.f_9743_.getBigY().subtract(d1).doubleValue(), this.f_9743_.getBigZ().subtract(d2).doubleValue());
             this.f_9738_ = d7 >= -0.03125D && !flag1 && !this.f_9745_.m_129915_() && !entity.m_20068_() && this.m_9793_(entity);
-            this.f_9763_ = entity.m_20185_();
-            this.f_9764_ = entity.m_20186_();
-            this.f_9765_ = entity.m_20189_();
+            this.f_9763_ = entity.getBigX();
+            this.f_9764_ = entity.getBigY();
+            this.f_9765_ = entity.getBigZ();
          }
 
       }
@@ -466,10 +_,10 @@
             return;
          }
 
-         this.f_9743_.m_19890_(this.f_9766_.f_82479_, this.f_9766_.f_82480_, this.f_9766_.f_82481_, this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
-         this.f_9756_ = this.f_9766_.f_82479_;
-         this.f_9757_ = this.f_9766_.f_82480_;
-         this.f_9758_ = this.f_9766_.f_82481_;
+         this.f_9743_.absMoveTo(this.f_9766_.getX(), this.f_9766_.getY(), this.f_9766_.getZ(), this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
+         this.f_9756_ = this.f_9766_.getX();
+         this.f_9757_ = this.f_9766_.getY();
+         this.f_9758_ = this.f_9766_.getZ();
          if (this.f_9743_.m_8958_()) {
             this.f_9743_.m_8959_();
          }
@@ -645,7 +_,7 @@
             structureblockentity.m_59860_(p_9919_.m_134633_());
             structureblockentity.m_59868_(p_9919_.m_134634_());
             structureblockentity.m_59885_(p_9919_.m_134635_());
-            structureblockentity.m_155797_(p_9919_.m_179787_());
+            structureblockentity.setStructureSize(p_9919_.getSize());
             structureblockentity.m_59881_(p_9919_.m_134637_());
             structureblockentity.m_59883_(p_9919_.m_134638_());
             structureblockentity.m_59887_(p_9919_.m_134639_());
@@ -834,7 +_,7 @@
 
    public void m_7185_(ServerboundMovePlayerPacket p_9874_) {
       PacketUtils.m_131359_(p_9874_, this, this.f_9743_.m_284548_());
-      if (m_143663_(p_9874_.m_134129_(0.0D), p_9874_.m_134140_(0.0D), p_9874_.m_134146_(0.0D), p_9874_.m_134131_(0.0F), p_9874_.m_134142_(0.0F))) {
+      if (containsInvalidValues(p_9874_.getX(me.alphamode.mclong.math.BigDecimal.ZERO), p_9874_.getY(me.alphamode.mclong.math.BigDecimal.ZERO), p_9874_.getZ(me.alphamode.mclong.math.BigDecimal.ZERO), p_9874_.m_134131_(0.0F), p_9874_.m_134142_(0.0F))) {
          this.m_9942_(Component.m_237115_("multiplayer.disconnect.invalid_player_movement"));
       } else {
          ServerLevel serverlevel = this.f_9743_.m_284548_();
@@ -846,31 +_,31 @@
             if (this.f_9766_ != null) {
                if (this.f_9746_ - this.f_9735_ > 20) {
                   this.f_9735_ = this.f_9746_;
-                  this.m_9774_(this.f_9766_.f_82479_, this.f_9766_.f_82480_, this.f_9766_.f_82481_, this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
+                  this.teleport(this.f_9766_.getX(), this.f_9766_.getY(), this.f_9766_.getZ(), this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
                }
 
             } else {
                this.f_9735_ = this.f_9746_;
-               double d0 = m_143609_(p_9874_.m_134129_(this.f_9743_.m_20185_()));
-               double d1 = m_143653_(p_9874_.m_134140_(this.f_9743_.m_20186_()));
-               double d2 = m_143609_(p_9874_.m_134146_(this.f_9743_.m_20189_()));
+               me.alphamode.mclong.math.BigDecimal d0 = clampHorizontal(p_9874_.getX(this.f_9743_.getBigX()));
+               me.alphamode.mclong.math.BigDecimal d1 = clampVertical(p_9874_.getY(this.f_9743_.getBigY()));
+               me.alphamode.mclong.math.BigDecimal d2 = clampHorizontal(p_9874_.getZ(this.f_9743_.getBigZ()));
                float f = Mth.m_14177_(p_9874_.m_134131_(this.f_9743_.m_146908_()));
                float f1 = Mth.m_14177_(p_9874_.m_134142_(this.f_9743_.m_146909_()));
                if (this.f_9743_.m_20159_()) {
-                  this.f_9743_.m_19890_(this.f_9743_.m_20185_(), this.f_9743_.m_20186_(), this.f_9743_.m_20189_(), f, f1);
+                  this.f_9743_.absMoveTo(this.f_9743_.getBigX(), this.f_9743_.getBigY(), this.f_9743_.getBigZ(), f, f1);
                   this.f_9743_.m_284548_().m_7726_().m_8385_(this.f_9743_);
                } else {
-                  double d3 = this.f_9743_.m_20185_();
-                  double d4 = this.f_9743_.m_20186_();
-                  double d5 = this.f_9743_.m_20189_();
-                  double d6 = d0 - this.f_9753_;
-                  double d7 = d1 - this.f_9754_;
-                  double d8 = d2 - this.f_9755_;
+                  me.alphamode.mclong.math.BigDecimal d3 = this.f_9743_.getBigX();
+                  me.alphamode.mclong.math.BigDecimal d4 = this.f_9743_.getBigY();
+                  me.alphamode.mclong.math.BigDecimal d5 = this.f_9743_.getBigZ();
+                  me.alphamode.mclong.math.BigDecimal d6 = d0.subtract(this.f_9753_);
+                  me.alphamode.mclong.math.BigDecimal d7 = d1.subtract(this.f_9754_);
+                  me.alphamode.mclong.math.BigDecimal d8 = d2.subtract(this.f_9755_);
                   double d9 = this.f_9743_.m_20184_().m_82556_();
-                  double d10 = d6 * d6 + d7 * d7 + d8 * d8;
+                  double d10 = d6.doubleValue() * d6.doubleValue() + d7.doubleValue() * d7.doubleValue() + d8.doubleValue() * d8.doubleValue();
                   if (this.f_9743_.m_5803_()) {
                      if (d10 > 1.0D) {
-                        this.m_9774_(this.f_9743_.m_20185_(), this.f_9743_.m_20186_(), this.f_9743_.m_20189_(), f, f1);
+                        this.teleport(this.f_9743_.getBigX(), this.f_9743_.getBigY(), this.f_9743_.getBigZ(), f, f1);
                      }
 
                   } else {
@@ -885,53 +_,53 @@
                         float f2 = this.f_9743_.m_21255_() ? 300.0F : 100.0F;
                         if (d10 - d9 > (double)(f2 * (float)i) && !this.m_9956_()) {
                            f_9744_.warn("{} moved too quickly! {},{},{}", this.f_9743_.m_7755_().getString(), d6, d7, d8);
-                           this.m_9774_(this.f_9743_.m_20185_(), this.f_9743_.m_20186_(), this.f_9743_.m_20189_(), this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
+                           this.teleport(this.f_9743_.getBigX(), this.f_9743_.getBigY(), this.f_9743_.getBigZ(), this.f_9743_.m_146908_(), this.f_9743_.m_146909_());
                            return;
                         }
                      }
 
                      AABB aabb = this.f_9743_.m_20191_();
-                     d6 = d0 - this.f_9756_;
-                     d7 = d1 - this.f_9757_;
-                     d8 = d2 - this.f_9758_;
-                     boolean flag = d7 > 0.0D;
+                     d6 = d0.subtract(this.f_9756_);
+                     d7 = d1.subtract(this.f_9757_);
+                     d8 = d2.subtract(this.f_9758_);
+                     boolean flag = d7.doubleValue() > 0.0D;
                      if (this.f_9743_.m_20096_() && !p_9874_.m_134139_() && flag) {
                         this.f_9743_.m_6135_();
                      }
 
                      boolean flag1 = this.f_9743_.f_201939_;
                      this.f_9743_.m_6478_(MoverType.PLAYER, new Vec3(d6, d7, d8));
-                     d6 = d0 - this.f_9743_.m_20185_();
-                     d7 = d1 - this.f_9743_.m_20186_();
-                     if (d7 > -0.5D || d7 < 0.5D) {
-                        d7 = 0.0D;
+                     d6 = d0.subtract(this.f_9743_.getBigX());
+                     d7 = d1.subtract(this.f_9743_.getBigY());
+                     if (d7.doubleValue() > -0.5D || d7.doubleValue() < 0.5D) {
+                        d7 = d7.subtract(me.alphamode.mclong.math.BigDecimal.ZERO);
                      }
 
-                     d8 = d2 - this.f_9743_.m_20189_();
-                     d10 = d6 * d6 + d7 * d7 + d8 * d8;
+                     d8 = d2.subtract(this.f_9743_.getBigZ());
+                     d10 = d6.doubleValue() * d6.doubleValue() + d7.doubleValue() * d7.doubleValue() + d8.doubleValue() * d8.doubleValue();
                      boolean flag2 = false;
                      if (!this.f_9743_.m_8958_() && d10 > 0.0625D && !this.f_9743_.m_5803_() && !this.f_9743_.f_8941_.m_9295_() && this.f_9743_.f_8941_.m_9290_() != GameType.SPECTATOR) {
                         flag2 = true;
                         f_9744_.warn("{} moved wrongly!", (Object)this.f_9743_.m_7755_().getString());
                      }
 
-                     if (this.f_9743_.f_19794_ || this.f_9743_.m_5803_() || (!flag2 || !serverlevel.m_45756_(this.f_9743_, aabb)) && !this.m_288208_(serverlevel, aabb, d0, d1, d2)) {
-                        this.f_9743_.m_19890_(d0, d1, d2, f, f1);
-                        this.f_9736_ = d7 >= -0.03125D && !flag1 && this.f_9743_.f_8941_.m_9290_() != GameType.SPECTATOR && !this.f_9745_.m_129915_() && !this.f_9743_.m_150110_().f_35936_ && !this.f_9743_.m_21023_(MobEffects.f_19620_) && !this.f_9743_.m_21255_() && !this.f_9743_.m_21209_() && this.m_9793_(this.f_9743_);
+                     if (this.f_9743_.f_19794_ || this.f_9743_.m_5803_() || (!flag2 || !serverlevel.m_45756_(this.f_9743_, aabb)) && !this.isPlayerCollidingWithAnythingNew(serverlevel, aabb, d0, d1, d2)) {
+                        this.f_9743_.absMoveTo(d0, d1, d2, f, f1);
+                        this.f_9736_ = d7.doubleValue() >= -0.03125D && !flag1 && this.f_9743_.f_8941_.m_9290_() != GameType.SPECTATOR && !this.f_9745_.m_129915_() && !this.f_9743_.m_150110_().f_35936_ && !this.f_9743_.m_21023_(MobEffects.f_19620_) && !this.f_9743_.m_21255_() && !this.f_9743_.m_21209_() && this.m_9793_(this.f_9743_);
                         this.f_9743_.m_284548_().m_7726_().m_8385_(this.f_9743_);
-                        this.f_9743_.m_289599_(this.f_9743_.m_20185_() - d3, this.f_9743_.m_20186_() - d4, this.f_9743_.m_20189_() - d5, p_9874_.m_134139_());
-                        this.f_9743_.m_289603_(p_9874_.m_134139_(), new Vec3(this.f_9743_.m_20185_() - d3, this.f_9743_.m_20186_() - d4, this.f_9743_.m_20189_() - d5));
+                        this.f_9743_.doCheckFallDamage(this.f_9743_.getBigX().subtract(d3), this.f_9743_.getBigY().subtract(d4), this.f_9743_.getBigZ().subtract(d5), p_9874_.m_134139_());
+                        this.f_9743_.m_289603_(p_9874_.m_134139_(), new Vec3(this.f_9743_.getBigX().subtract(d3), this.f_9743_.getBigY().subtract(d4), this.f_9743_.getBigZ().subtract(d5)));
                         if (flag) {
                            this.f_9743_.m_183634_();
                         }
 
-                        this.f_9743_.m_36378_(this.f_9743_.m_20185_() - d3, this.f_9743_.m_20186_() - d4, this.f_9743_.m_20189_() - d5);
-                        this.f_9756_ = this.f_9743_.m_20185_();
-                        this.f_9757_ = this.f_9743_.m_20186_();
-                        this.f_9758_ = this.f_9743_.m_20189_();
+                        this.f_9743_.m_36378_(this.f_9743_.getBigX().subtract(d3).doubleValue(), this.f_9743_.getBigY().subtract(d4).doubleValue(), this.f_9743_.getBigZ().subtract(d5).doubleValue());
+                        this.f_9756_ = this.f_9743_.getBigX();
+                        this.f_9757_ = this.f_9743_.getBigY();
+                        this.f_9758_ = this.f_9743_.getBigZ();
                      } else {
-                        this.m_9774_(d3, d4, d5, f, f1);
-                        this.f_9743_.m_289599_(this.f_9743_.m_20185_() - d3, this.f_9743_.m_20186_() - d4, this.f_9743_.m_20189_() - d5, p_9874_.m_134139_());
+                        this.teleport(d3, d4, d5, f, f1);
+                        this.f_9743_.doCheckFallDamage(this.f_9743_.getBigX().subtract(d3), this.f_9743_.getBigY().subtract(d4), this.f_9743_.getBigZ().subtract(d5), p_9874_.m_134139_());
                      }
                   }
                }
@@ -940,8 +_,8 @@
       }
    }
 
-   private boolean m_288208_(LevelReader p_289008_, AABB p_288986_, double p_288990_, double p_288991_, double p_288967_) {
-      AABB aabb = this.f_9743_.m_20191_().m_82386_(p_288990_ - this.f_9743_.m_20185_(), p_288991_ - this.f_9743_.m_20186_(), p_288967_ - this.f_9743_.m_20189_());
+   private boolean isPlayerCollidingWithAnythingNew(LevelReader p_289008_, AABB p_288986_, me.alphamode.mclong.math.BigDecimal p_288990_, me.alphamode.mclong.math.BigDecimal p_288991_, me.alphamode.mclong.math.BigDecimal p_288967_) {
+      AABB aabb = this.f_9743_.m_20191_().move(p_288990_.subtract(this.f_9743_.getBigX()), p_288991_.subtract(this.f_9743_.getBigY()), p_288967_.subtract(this.f_9743_.getBigZ()));
       Iterable<VoxelShape> iterable = p_289008_.m_186431_(this.f_9743_, aabb.m_82406_((double)1.0E-5F));
       VoxelShape voxelshape = Shapes.m_83064_(p_288986_.m_82406_((double)1.0E-5F));
 
@@ -954,14 +_,14 @@
       return false;
    }
 
-   public void m_9774_(double p_9775_, double p_9776_, double p_9777_, float p_9778_, float p_9779_) {
-      this.m_9780_(p_9775_, p_9776_, p_9777_, p_9778_, p_9779_, Collections.emptySet());
+   public void teleport(me.alphamode.mclong.math.BigDecimal p_9775_, me.alphamode.mclong.math.BigDecimal p_9776_, me.alphamode.mclong.math.BigDecimal p_9777_, float p_9778_, float p_9779_) {
+      this.teleport(p_9775_, p_9776_, p_9777_, p_9778_, p_9779_, Collections.emptySet());
    }
 
-   public void m_9780_(double p_9781_, double p_9782_, double p_9783_, float p_9784_, float p_9785_, Set<RelativeMovement> p_9786_) {
-      double d0 = p_9786_.contains(RelativeMovement.X) ? this.f_9743_.m_20185_() : 0.0D;
-      double d1 = p_9786_.contains(RelativeMovement.Y) ? this.f_9743_.m_20186_() : 0.0D;
-      double d2 = p_9786_.contains(RelativeMovement.Z) ? this.f_9743_.m_20189_() : 0.0D;
+   public void teleport(me.alphamode.mclong.math.BigDecimal p_9781_, me.alphamode.mclong.math.BigDecimal p_9782_, me.alphamode.mclong.math.BigDecimal p_9783_, float p_9784_, float p_9785_, Set<RelativeMovement> p_9786_) {
+      me.alphamode.mclong.math.BigDecimal d0 = p_9786_.contains(RelativeMovement.X) ? this.f_9743_.getBigX() : me.alphamode.mclong.math.BigDecimal.ZERO;
+      me.alphamode.mclong.math.BigDecimal d1 = p_9786_.contains(RelativeMovement.Y) ? this.f_9743_.getBigY() : me.alphamode.mclong.math.BigDecimal.ZERO;
+      me.alphamode.mclong.math.BigDecimal d2 = p_9786_.contains(RelativeMovement.Z) ? this.f_9743_.getBigZ() : me.alphamode.mclong.math.BigDecimal.ZERO;
       float f = p_9786_.contains(RelativeMovement.Y_ROT) ? this.f_9743_.m_146908_() : 0.0F;
       float f1 = p_9786_.contains(RelativeMovement.X_ROT) ? this.f_9743_.m_146909_() : 0.0F;
       this.f_9766_ = new Vec3(p_9781_, p_9782_, p_9783_);
@@ -970,8 +_,8 @@
       }
 
       this.f_9735_ = this.f_9746_;
-      this.f_9743_.m_19890_(p_9781_, p_9782_, p_9783_, p_9784_, p_9785_);
-      this.f_9743_.f_8906_.m_9829_(new ClientboundPlayerPositionPacket(p_9781_ - d0, p_9782_ - d1, p_9783_ - d2, p_9784_ - f, p_9785_ - f1, p_9786_, this.f_9767_));
+      this.f_9743_.absMoveTo(p_9781_, p_9782_, p_9783_, p_9784_, p_9785_);
+      this.f_9743_.f_8906_.m_9829_(new ClientboundPlayerPositionPacket(p_9781_.subtract(d0), p_9782_.subtract(d1), p_9783_.subtract(d2), p_9784_ - f, p_9785_ - f1, p_9786_, this.f_9767_));
    }
 
    public void m_7502_(ServerboundPlayerActionPacket p_9889_) {
@@ -1007,7 +_,7 @@
          case START_DESTROY_BLOCK:
          case ABORT_DESTROY_BLOCK:
          case STOP_DESTROY_BLOCK:
-            this.f_9743_.f_8941_.m_214168_(blockpos, serverboundplayeractionpacket$action, p_9889_.m_134284_(), this.f_9743_.m_9236_().m_151558_(), p_9889_.m_237987_());
+            this.f_9743_.f_8941_.handleBlockBreakAction(blockpos, serverboundplayeractionpacket$action, p_9889_.m_134284_(), this.f_9743_.m_9236_().getMaxBuildHeight(), p_9889_.m_237987_());
             this.f_9743_.f_8906_.m_215201_(p_9889_.m_237987_());
             return;
          default:
@@ -1034,18 +_,18 @@
          BlockHitResult blockhitresult = p_9930_.m_134706_();
          Vec3 vec3 = blockhitresult.m_82450_();
          BlockPos blockpos = blockhitresult.m_82425_();
-         Vec3 vec31 = Vec3.m_82512_(blockpos);
+         Vec3 vec31 = Vec3.atCenterOf(blockpos);
          if (!(this.f_9743_.m_146892_().m_82557_(vec31) > f_215198_)) {
             Vec3 vec32 = vec3.m_82546_(vec31);
             double d0 = 1.0000001D;
             if (Math.abs(vec32.m_7096_()) < 1.0000001D && Math.abs(vec32.m_7098_()) < 1.0000001D && Math.abs(vec32.m_7094_()) < 1.0000001D) {
                Direction direction = blockhitresult.m_82434_();
                this.f_9743_.m_9243_();
-               int i = this.f_9743_.m_9236_().m_151558_();
-               if (blockpos.m_123342_() < i) {
-                  if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) < 64.0D && serverlevel.m_7966_(this.f_9743_, blockpos)) {
+               long i = this.f_9743_.m_9236_().getMaxBuildHeight();
+               if (blockpos.getY() < i) {
+                  if (this.f_9766_ == null && this.f_9743_.m_20275_((double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.5D, (double)blockpos.getZ() + 0.5D) < 64.0D && serverlevel.m_7966_(this.f_9743_, blockpos)) {
                      InteractionResult interactionresult = this.f_9743_.f_8941_.m_7179_(this.f_9743_, serverlevel, itemstack, interactionhand, blockhitresult);
-                     if (direction == Direction.UP && !interactionresult.m_19077_() && blockpos.m_123342_() >= i - 1 && m_9790_(this.f_9743_, itemstack)) {
+                     if (direction == Direction.UP && !interactionresult.m_19077_() && blockpos.getY() >= i - 1 && m_9790_(this.f_9743_, itemstack)) {
                         Component component = Component.m_237110_("build.tooHigh", i - 1).m_130940_(ChatFormatting.RED);
                         this.f_9743_.m_240418_(component, true);
                      } else if (interactionresult.m_19080_()) {
@@ -1058,7 +_,7 @@
                }
 
                this.f_9743_.f_8906_.m_9829_(new ClientboundBlockUpdatePacket(serverlevel, blockpos));
-               this.f_9743_.f_8906_.m_9829_(new ClientboundBlockUpdatePacket(serverlevel, blockpos.m_121945_(direction)));
+               this.f_9743_.f_8906_.m_9829_(new ClientboundBlockUpdatePacket(serverlevel, blockpos.relative(direction)));
             } else {
                f_9744_.warn("Rejecting UseItemOnPacket from {}: Location {} too far away from hit block {}.", this.f_9743_.m_36316_().getName(), vec3, blockpos);
             }
@@ -1088,7 +_,7 @@
          for(ServerLevel serverlevel : this.f_9745_.m_129785_()) {
             Entity entity = p_9928_.m_134681_(serverlevel);
             if (entity != null) {
-               this.f_9743_.m_8999_(serverlevel, entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), entity.m_146908_(), entity.m_146909_());
+               this.f_9743_.teleportTo(serverlevel, entity.getBigX(), entity.getBigY(), entity.getBigZ(), entity.m_146908_(), entity.m_146909_());
                return;
             }
          }
