--- a/net/minecraft/util/ParticleUtils.java
+++ b/net/minecraft/util/ParticleUtils.java
@@ -33,7 +_,7 @@
    }
 
    public static void m_144967_(Direction.Axis p_144968_, Level p_144969_, BlockPos p_144970_, double p_144971_, ParticleOptions p_144972_, UniformInt p_144973_) {
-      Vec3 vec3 = Vec3.m_82512_(p_144970_);
+      Vec3 vec3 = Vec3.atCenterOf(p_144970_);
       boolean flag = p_144968_ == Direction.Axis.X;
       boolean flag1 = p_144968_ == Direction.Axis.Y;
       boolean flag2 = p_144968_ == Direction.Axis.Z;
@@ -52,7 +_,7 @@
    }
 
    public static void m_216306_(Level p_216307_, BlockPos p_216308_, Direction p_216309_, ParticleOptions p_216310_, Vec3 p_216311_, double p_216312_) {
-      Vec3 vec3 = Vec3.m_82512_(p_216308_);
+      Vec3 vec3 = Vec3.atCenterOf(p_216308_);
       int i = p_216309_.m_122429_();
       int j = p_216309_.m_122430_();
       int k = p_216309_.m_122431_();
