--- a/net/minecraft/util/SpawnUtil.java
+++ b/net/minecraft/util/SpawnUtil.java
@@ -19,7 +_,7 @@
       for(int i = 0; i < p_216408_; ++i) {
          int j = Mth.m_216287_(p_216406_.f_46441_, -p_216409_, p_216409_);
          int k = Mth.m_216287_(p_216406_.f_46441_, -p_216409_, p_216409_);
-         blockpos$mutableblockpos.m_122154_(p_216407_, j, p_216410_, k);
+         blockpos$mutableblockpos.setWithOffset(p_216407_, j, p_216410_, k);
          if (p_216406_.m_6857_().m_61937_(blockpos$mutableblockpos) && m_216398_(p_216406_, p_216410_, blockpos$mutableblockpos, p_216411_)) {
             T t = p_216404_.m_262451_(p_216406_, (CompoundTag)null, (Consumer<T>)null, blockpos$mutableblockpos, p_216405_, false, false);
             if (t != null) {
@@ -37,12 +_,12 @@
    }
 
    private static boolean m_216398_(ServerLevel p_216399_, int p_216400_, BlockPos.MutableBlockPos p_216401_, SpawnUtil.Strategy p_216402_) {
-      BlockPos.MutableBlockPos blockpos$mutableblockpos = (new BlockPos.MutableBlockPos()).m_122190_(p_216401_);
+      BlockPos.MutableBlockPos blockpos$mutableblockpos = (new BlockPos.MutableBlockPos()).set(p_216401_);
       BlockState blockstate = p_216399_.m_8055_(blockpos$mutableblockpos);
 
       for(int i = p_216400_; i >= -p_216400_; --i) {
          p_216401_.m_122173_(Direction.DOWN);
-         blockpos$mutableblockpos.m_122159_(p_216401_, Direction.UP);
+         blockpos$mutableblockpos.setWithOffset(p_216401_, Direction.UP);
          BlockState blockstate1 = p_216399_.m_8055_(p_216401_);
          if (p_216402_.m_216427_(p_216399_, p_216401_, blockstate1, blockpos$mutableblockpos, blockstate)) {
             p_216401_.m_122173_(Direction.UP);
