--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -439,29 +_,34 @@
    }
 
    public BlockPos m_130135_() {
-      return BlockPos.m_122022_(this.readLong());
+      return new BlockPos(this.readLong(), this.readLong(), this.readLong());
    }
 
    public FriendlyByteBuf m_130064_(BlockPos p_130065_) {
-      this.writeLong(p_130065_.m_121878_());
+      this.writeLong(p_130065_.getX());
+      this.writeLong(p_130065_.getY());
+      this.writeLong(p_130065_.getZ());
       return this;
    }
 
    public ChunkPos m_178383_() {
-      return new ChunkPos(this.readLong());
+      return new ChunkPos(this.readLong(), this.readLong());
    }
 
    public FriendlyByteBuf m_178341_(ChunkPos p_178342_) {
-      this.writeLong(p_178342_.m_45588_());
+      this.writeLong(p_178342_.x());
+      this.writeLong(p_178342_.z());
       return this;
    }
 
    public SectionPos m_130157_() {
-      return SectionPos.m_123184_(this.readLong());
+      return SectionPos.of(this.readLong(), this.readLong(), this.readLong());
    }
 
    public FriendlyByteBuf m_178343_(SectionPos p_178344_) {
-      this.writeLong(p_178344_.m_123252_());
+      this.writeLong(p_178344_.x());
+      this.writeLong(p_178344_.y());
+      this.writeLong(p_178344_.z());
       return this;
    }
 
@@ -753,7 +_,7 @@
       float f1 = this.readFloat();
       float f2 = this.readFloat();
       boolean flag = this.readBoolean();
-      return new BlockHitResult(new Vec3((double)blockpos.m_123341_() + (double)f, (double)blockpos.m_123342_() + (double)f1, (double)blockpos.m_123343_() + (double)f2), direction, blockpos, flag);
+      return new BlockHitResult(new Vec3((double)blockpos.getX() + (double)f, (double)blockpos.getY() + (double)f1, (double)blockpos.getZ() + (double)f2), direction, blockpos, flag);
    }
 
    public void m_130062_(BlockHitResult p_130063_) {
@@ -761,9 +_,9 @@
       this.m_130064_(blockpos);
       this.m_130068_(p_130063_.m_82434_());
       Vec3 vec3 = p_130063_.m_82450_();
-      this.writeFloat((float)(vec3.f_82479_ - (double)blockpos.m_123341_()));
-      this.writeFloat((float)(vec3.f_82480_ - (double)blockpos.m_123342_()));
-      this.writeFloat((float)(vec3.f_82481_ - (double)blockpos.m_123343_()));
+      this.writeFloat((float)(vec3.m_7096_() - (double)blockpos.getX()));
+      this.writeFloat((float)(vec3.m_7098_() - (double)blockpos.getY()));
+      this.writeFloat((float)(vec3.m_7094_() - (double)blockpos.getZ()));
       this.writeBoolean(p_130063_.m_82436_());
    }
 
