--- a/net/minecraft/world/level/entity/EntitySectionStorage.java
+++ b/net/minecraft/world/level/entity/EntitySectionStorage.java
@@ -34,16 +_,16 @@
 
    public void m_188362_(AABB p_188363_, Consumer<EntitySection<T>> p_188364_) {
       int i = 2;
-      int j = SectionPos.m_175552_(p_188363_.f_82288_ - 2.0D);
-      int k = SectionPos.m_175552_(p_188363_.f_82289_ - 4.0D);
-      int l = SectionPos.m_175552_(p_188363_.f_82290_ - 2.0D);
-      int i1 = SectionPos.m_175552_(p_188363_.f_82291_ + 2.0D);
-      int j1 = SectionPos.m_175552_(p_188363_.f_82292_ + 0.0D);
-      int k1 = SectionPos.m_175552_(p_188363_.f_82293_ + 2.0D);
+      long j = SectionPos.posToSectionCoord(p_188363_.f_82288_ - 2.0D);
+      long k = SectionPos.posToSectionCoord(p_188363_.f_82289_ - 4.0D);
+      long l = SectionPos.posToSectionCoord(p_188363_.f_82290_ - 2.0D);
+      long i1 = SectionPos.posToSectionCoord(p_188363_.f_82291_ + 2.0D);
+      long j1 = SectionPos.posToSectionCoord(p_188363_.f_82292_ + 0.0D);
+      long k1 = SectionPos.posToSectionCoord(p_188363_.f_82293_ + 2.0D);
 
-      for(int l1 = j; l1 <= i1; ++l1) {
-         long i2 = SectionPos.m_123209_(l1, 0, 0);
-         long j2 = SectionPos.m_123209_(l1, -1, -1);
+      for(long l1 = j; l1 <= i1; ++l1) {
+         long i2 = SectionPos.asLong(l1, 0, 0);
+         long j2 = SectionPos.asLong(l1, -1, -1);
          LongIterator longiterator = this.f_156853_.subSet(i2, j2 + 1L).iterator();
 
          while(longiterator.hasNext()) {
@@ -74,8 +_,8 @@
    }
 
    private LongSortedSet m_156858_(int p_156859_, int p_156860_) {
-      long i = SectionPos.m_123209_(p_156859_, 0, p_156860_);
-      long j = SectionPos.m_123209_(p_156859_, -1, p_156860_);
+      long i = SectionPos.asLong(p_156859_, 0, p_156860_);
+      long j = SectionPos.asLong(p_156859_, -1, p_156860_);
       return this.f_156853_.subSet(i, j + 1L);
    }
 
@@ -84,7 +_,7 @@
    }
 
    private static long m_156899_(long p_156900_) {
-      return ChunkPos.m_45589_(SectionPos.m_123213_(p_156900_), SectionPos.m_123230_(p_156900_));
+      return ChunkPos.asLong(SectionPos.m_123213_(p_156900_), SectionPos.m_123230_(p_156900_));
    }
 
    public EntitySection<T> m_156893_(long p_156894_) {
