--- a/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
+++ b/net/minecraft/world/level/entity/PersistentEntitySectionManager.java
@@ -332,7 +_,7 @@
             EntitySection<T> entitysection = this.f_157495_.m_156895_(p_157521_);
             if (entitysection != null) {
                try {
-                  csvoutput.m_13624_(SectionPos.m_123213_(p_157521_), SectionPos.m_123225_(p_157521_), SectionPos.m_123230_(p_157521_), entitysection.m_156848_(), persistententitysectionmanager$chunkloadstatus, entitysection.m_156849_());
+                  csvoutput.m_13624_(SectionPos.x(p_157521_), SectionPos.y(p_157521_), SectionPos.z(p_157521_), entitysection.m_156848_(), persistententitysectionmanager$chunkloadstatus, entitysection.m_156849_());
                } catch (IOException ioexception) {
                   throw new UncheckedIOException(ioexception);
                }
