--- a/net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
@@ -32,12 +_,12 @@
       double d1 = p_123375_.m_7096_() + (double)direction.m_122429_() * d0;
       double d2 = p_123375_.m_7098_() + (double)((float)direction.m_122430_() * 1.125F);
       double d3 = p_123375_.m_7094_() + (double)direction.m_122431_() * d0;
-      BlockPos blockpos = p_123375_.m_7961_().m_121945_(direction);
+      BlockPos blockpos = p_123375_.m_7961_().relative(direction);
       double d4;
       if (level.m_6425_(blockpos).m_205070_(FluidTags.f_13131_)) {
          d4 = 1.0D;
       } else {
-         if (!level.m_8055_(blockpos).m_60795_() || !level.m_6425_(blockpos.m_7495_()).m_205070_(FluidTags.f_13131_)) {
+         if (!level.m_8055_(blockpos).m_60795_() || !level.m_6425_(blockpos.below()).m_205070_(FluidTags.f_13131_)) {
             return this.f_123368_.m_6115_(p_123375_, p_123376_);
          }
 
