--- a/net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
@@ -25,13 +_,13 @@
    }
 
    public static void m_123378_(Level p_123379_, ItemStack p_123380_, int p_123381_, Direction p_123382_, Position p_123383_) {
-      double d0 = p_123383_.m_7096_();
-      double d1 = p_123383_.m_7098_();
-      double d2 = p_123383_.m_7094_();
+      me.alphamode.mclong.math.BigDecimal d0 = p_123383_.getX();
+      me.alphamode.mclong.math.BigDecimal d1 = p_123383_.getY();
+      me.alphamode.mclong.math.BigDecimal d2 = p_123383_.getZ();
       if (p_123382_.m_122434_() == Direction.Axis.Y) {
-         d1 -= 0.125D;
+         d1 = d1.subtract(0.125D);
       } else {
-         d1 -= 0.15625D;
+         d1 = d1.subtract(0.15625D);
       }
 
       ItemEntity itementity = new ItemEntity(p_123379_, d0, d1, d2, p_123380_);
