--- a/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -89,14 +_,14 @@
    static void m_123402_() {
       DispenserBlock.m_52672_(Items.f_42412_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123407_, Position p_123408_, ItemStack p_123409_) {
-            Arrow arrow = new Arrow(p_123407_, p_123408_.m_7096_(), p_123408_.m_7098_(), p_123408_.m_7094_());
+            Arrow arrow = new Arrow(p_123407_, p_123408_.getX(), p_123408_.getY(), p_123408_.getZ());
             arrow.f_36705_ = AbstractArrow.Pickup.ALLOWED;
             return arrow;
          }
       });
       DispenserBlock.m_52672_(Items.f_42738_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123420_, Position p_123421_, ItemStack p_123422_) {
-            Arrow arrow = new Arrow(p_123420_, p_123421_.m_7096_(), p_123421_.m_7098_(), p_123421_.m_7094_());
+            Arrow arrow = new Arrow(p_123420_, p_123421_.getX(), p_123421_.getY(), p_123421_.getZ());
             arrow.m_36878_(p_123422_);
             arrow.f_36705_ = AbstractArrow.Pickup.ALLOWED;
             return arrow;
@@ -104,28 +_,28 @@
       });
       DispenserBlock.m_52672_(Items.f_42737_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123456_, Position p_123457_, ItemStack p_123458_) {
-            AbstractArrow abstractarrow = new SpectralArrow(p_123456_, p_123457_.m_7096_(), p_123457_.m_7098_(), p_123457_.m_7094_());
+            AbstractArrow abstractarrow = new SpectralArrow(p_123456_, p_123457_.getX(), p_123457_.getY(), p_123457_.getZ());
             abstractarrow.f_36705_ = AbstractArrow.Pickup.ALLOWED;
             return abstractarrow;
          }
       });
       DispenserBlock.m_52672_(Items.f_42521_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123468_, Position p_123469_, ItemStack p_123470_) {
-            return Util.m_137469_(new ThrownEgg(p_123468_, p_123469_.m_7096_(), p_123469_.m_7098_(), p_123469_.m_7094_()), (p_123466_) -> {
+            return Util.m_137469_(new ThrownEgg(p_123468_, p_123469_.getX(), p_123469_.getY(), p_123469_.getZ()), (p_123466_) -> {
                p_123466_.m_37446_(p_123470_);
             });
          }
       });
       DispenserBlock.m_52672_(Items.f_42452_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123476_, Position p_123477_, ItemStack p_123478_) {
-            return Util.m_137469_(new Snowball(p_123476_, p_123477_.m_7096_(), p_123477_.m_7098_(), p_123477_.m_7094_()), (p_123474_) -> {
+            return Util.m_137469_(new Snowball(p_123476_, p_123477_.getX(), p_123477_.getY(), p_123477_.getZ()), (p_123474_) -> {
                p_123474_.m_37446_(p_123478_);
             });
          }
       });
       DispenserBlock.m_52672_(Items.f_42612_, new AbstractProjectileDispenseBehavior() {
          protected Projectile m_6895_(Level p_123485_, Position p_123486_, ItemStack p_123487_) {
-            return Util.m_137469_(new ThrownExperienceBottle(p_123485_, p_123486_.m_7096_(), p_123486_.m_7098_(), p_123486_.m_7094_()), (p_123483_) -> {
+            return Util.m_137469_(new ThrownExperienceBottle(p_123485_, p_123486_.getX(), p_123486_.getY(), p_123486_.getZ()), (p_123483_) -> {
                p_123483_.m_37446_(p_123487_);
             });
          }
@@ -142,7 +_,7 @@
          public ItemStack m_6115_(BlockSource p_123491_, ItemStack p_123492_) {
             return (new AbstractProjectileDispenseBehavior() {
                protected Projectile m_6895_(Level p_123501_, Position p_123502_, ItemStack p_123503_) {
-                  return Util.m_137469_(new ThrownPotion(p_123501_, p_123502_.m_7096_(), p_123502_.m_7098_(), p_123502_.m_7094_()), (p_123499_) -> {
+                  return Util.m_137469_(new ThrownPotion(p_123501_, p_123502_.getX(), p_123502_.getY(), p_123502_.getZ()), (p_123499_) -> {
                      p_123499_.m_37446_(p_123503_);
                   });
                }
@@ -161,7 +_,7 @@
          public ItemStack m_6115_(BlockSource p_123507_, ItemStack p_123508_) {
             return (new AbstractProjectileDispenseBehavior() {
                protected Projectile m_6895_(Level p_123517_, Position p_123518_, ItemStack p_123519_) {
-                  return Util.m_137469_(new ThrownPotion(p_123517_, p_123518_.m_7096_(), p_123518_.m_7098_(), p_123518_.m_7094_()), (p_123515_) -> {
+                  return Util.m_137469_(new ThrownPotion(p_123517_, p_123518_.getX(), p_123518_.getY(), p_123518_.getZ()), (p_123515_) -> {
                      p_123515_.m_37446_(p_123519_);
                   });
                }
@@ -182,7 +_,7 @@
             EntityType<?> entitytype = ((SpawnEggItem)p_123524_.m_41720_()).m_43228_(p_123524_.m_41783_());
 
             try {
-               entitytype.m_20592_(p_123523_.m_7727_(), p_123524_, (Player)null, p_123523_.m_7961_().m_121945_(direction), MobSpawnType.DISPENSER, direction != Direction.UP, false);
+               entitytype.m_20592_(p_123523_.m_7727_(), p_123524_, (Player)null, p_123523_.m_7961_().relative(direction), MobSpawnType.DISPENSER, direction != Direction.UP, false);
             } catch (Exception exception) {
                f_181892_.error("Error while dispensing spawn egg from dispenser at {}", p_123523_.m_7961_(), exception);
                return ItemStack.f_41583_;
@@ -201,7 +_,7 @@
       DispenserBlock.m_52672_(Items.f_42650_, new DefaultDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_123461_, ItemStack p_123462_) {
             Direction direction = p_123461_.m_6414_().m_61143_(DispenserBlock.f_52659_);
-            BlockPos blockpos = p_123461_.m_7961_().m_121945_(direction);
+            BlockPos blockpos = p_123461_.m_7961_().relative(direction);
             ServerLevel serverlevel = p_123461_.m_7727_();
             Consumer<ArmorStand> consumer = EntityType.m_264081_((p_277236_) -> {
                p_277236_.m_146922_(direction.m_122435_());
@@ -216,7 +_,7 @@
       });
       DispenserBlock.m_52672_(Items.f_42450_, new OptionalDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_123529_, ItemStack p_123530_) {
-            BlockPos blockpos = p_123529_.m_7961_().m_121945_(p_123529_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123529_.m_7961_().relative(p_123529_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             List<LivingEntity> list = p_123529_.m_7727_().m_6443_(LivingEntity.class, new AABB(blockpos), (p_123527_) -> {
                if (!(p_123527_ instanceof Saddleable saddleable)) {
                   return false;
@@ -236,7 +_,7 @@
       });
       DefaultDispenseItemBehavior defaultdispenseitembehavior1 = new OptionalDispenseItemBehavior() {
          protected ItemStack m_7498_(BlockSource p_123535_, ItemStack p_123536_) {
-            BlockPos blockpos = p_123535_.m_7961_().m_121945_(p_123535_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123535_.m_7961_().relative(p_123535_.m_6414_().m_61143_(DispenserBlock.f_52659_));
 
             for(AbstractHorse abstracthorse : p_123535_.m_7727_().m_6443_(AbstractHorse.class, new AABB(blockpos), (p_289248_) -> {
                return p_289248_.m_6084_() && p_289248_.m_7482_();
@@ -273,7 +_,7 @@
       DispenserBlock.m_52672_(Items.f_42134_, defaultdispenseitembehavior1);
       DispenserBlock.m_52672_(Items.f_42009_, new OptionalDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_123541_, ItemStack p_123542_) {
-            BlockPos blockpos = p_123541_.m_7961_().m_121945_(p_123541_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123541_.m_7961_().relative(p_123541_.m_6414_().m_61143_(DispenserBlock.f_52659_));
 
             for(AbstractChestedHorse abstractchestedhorse : p_123541_.m_7727_().m_6443_(AbstractChestedHorse.class, new AABB(blockpos), (p_289249_) -> {
                return p_289249_.m_6084_() && !p_289249_.m_30502_();
@@ -291,7 +_,7 @@
       DispenserBlock.m_52672_(Items.f_42688_, new DefaultDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_123547_, ItemStack p_123548_) {
             Direction direction = p_123547_.m_6414_().m_61143_(DispenserBlock.f_52659_);
-            FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(p_123547_.m_7727_(), p_123548_, p_123547_.m_7096_(), p_123547_.m_7098_(), p_123547_.m_7096_(), true);
+            FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(p_123547_.m_7727_(), p_123548_, p_123547_.getX(), p_123547_.getY(), p_123547_.getX(), true);
             DispenseItemBehavior.m_123395_(p_123547_, fireworkrocketentity, direction);
             fireworkrocketentity.m_6686_((double)direction.m_122429_(), (double)direction.m_122430_(), (double)direction.m_122431_(), 0.5F, 1.0F);
             p_123547_.m_7727_().m_7967_(fireworkrocketentity);
@@ -307,9 +_,9 @@
          public ItemStack m_7498_(BlockSource p_123556_, ItemStack p_123557_) {
             Direction direction = p_123556_.m_6414_().m_61143_(DispenserBlock.f_52659_);
             Position position = DispenserBlock.m_52720_(p_123556_);
-            double d0 = position.m_7096_() + (double)((float)direction.m_122429_() * 0.3F);
-            double d1 = position.m_7098_() + (double)((float)direction.m_122430_() * 0.3F);
-            double d2 = position.m_7094_() + (double)((float)direction.m_122431_() * 0.3F);
+            me.alphamode.mclong.math.BigDecimal d0 = position.getX().add((double)((float)direction.m_122429_() * 0.3F));
+            me.alphamode.mclong.math.BigDecimal d1 = position.getY().add((double)((float)direction.m_122430_() * 0.3F));
+            me.alphamode.mclong.math.BigDecimal d2 = position.getZ().add((double)((float)direction.m_122431_() * 0.3F));
             Level level = p_123556_.m_7727_();
             RandomSource randomsource = level.f_46441_;
             double d3 = randomsource.m_216328_((double)direction.m_122429_(), 0.11485000000000001D);
@@ -350,7 +_,7 @@
 
          public ItemStack m_7498_(BlockSource p_123561_, ItemStack p_123562_) {
             DispensibleContainerItem dispensiblecontaineritem = (DispensibleContainerItem)p_123562_.m_41720_();
-            BlockPos blockpos = p_123561_.m_7961_().m_121945_(p_123561_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123561_.m_7961_().relative(p_123561_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             Level level = p_123561_.m_7727_();
             if (dispensiblecontaineritem.m_142073_((Player)null, level, blockpos, (BlockHitResult)null)) {
                dispensiblecontaineritem.m_142131_((Player)null, level, p_123562_, blockpos);
@@ -374,7 +_,7 @@
 
          public ItemStack m_7498_(BlockSource p_123566_, ItemStack p_123567_) {
             LevelAccessor levelaccessor = p_123566_.m_7727_();
-            BlockPos blockpos = p_123566_.m_7961_().m_121945_(p_123566_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123566_.m_7961_().relative(p_123566_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             BlockState blockstate = levelaccessor.m_8055_(blockpos);
             Block block = blockstate.m_60734_();
             if (block instanceof BucketPickup) {
@@ -405,7 +_,7 @@
             Level level = p_123412_.m_7727_();
             this.m_123573_(true);
             Direction direction = p_123412_.m_6414_().m_61143_(DispenserBlock.f_52659_);
-            BlockPos blockpos = p_123412_.m_7961_().m_121945_(direction);
+            BlockPos blockpos = p_123412_.m_7961_().relative(direction);
             BlockState blockstate = level.m_8055_(blockpos);
             if (BaseFireBlock.m_49255_(level, blockpos, direction)) {
                level.m_46597_(blockpos, BaseFireBlock.m_49245_(level, blockpos));
@@ -433,7 +_,7 @@
          protected ItemStack m_7498_(BlockSource p_123416_, ItemStack p_123417_) {
             this.m_123573_(true);
             Level level = p_123416_.m_7727_();
-            BlockPos blockpos = p_123416_.m_7961_().m_121945_(p_123416_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123416_.m_7961_().relative(p_123416_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             if (!BoneMealItem.m_40627_(p_123417_, level, blockpos) && !BoneMealItem.m_40631_(p_123417_, level, blockpos, (Direction)null)) {
                this.m_123573_(false);
             } else if (!level.f_46443_) {
@@ -446,10 +_,10 @@
       DispenserBlock.m_52672_(Blocks.f_50077_, new DefaultDispenseItemBehavior() {
          protected ItemStack m_7498_(BlockSource p_123425_, ItemStack p_123426_) {
             Level level = p_123425_.m_7727_();
-            BlockPos blockpos = p_123425_.m_7961_().m_121945_(p_123425_.m_6414_().m_61143_(DispenserBlock.f_52659_));
-            PrimedTnt primedtnt = new PrimedTnt(level, (double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_(), (double)blockpos.m_123343_() + 0.5D, (LivingEntity)null);
+            BlockPos blockpos = p_123425_.m_7961_().relative(p_123425_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            PrimedTnt primedtnt = new PrimedTnt(level, blockpos.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), blockpos.getBigDecY(), blockpos.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), (LivingEntity)null);
             level.m_7967_(primedtnt);
-            level.m_6263_((Player)null, primedtnt.m_20185_(), primedtnt.m_20186_(), primedtnt.m_20189_(), SoundEvents.f_12512_, SoundSource.BLOCKS, 1.0F, 1.0F);
+            level.playSound((Player)null, primedtnt.getBigX(), primedtnt.getBigY(), primedtnt.getBigZ(), SoundEvents.f_12512_, SoundSource.BLOCKS, 1.0F, 1.0F);
             level.m_142346_((Entity)null, GameEvent.f_157810_, blockpos);
             p_123426_.m_41774_(1);
             return p_123426_;
@@ -471,7 +_,7 @@
          protected ItemStack m_7498_(BlockSource p_123433_, ItemStack p_123434_) {
             Level level = p_123433_.m_7727_();
             Direction direction = p_123433_.m_6414_().m_61143_(DispenserBlock.f_52659_);
-            BlockPos blockpos = p_123433_.m_7961_().m_121945_(direction);
+            BlockPos blockpos = p_123433_.m_7961_().relative(direction);
             if (level.m_46859_(blockpos) && WitherSkullBlock.m_58267_(level, blockpos, p_123434_)) {
                level.m_7731_(blockpos, Blocks.f_50312_.m_49966_().m_61124_(SkullBlock.f_56314_, Integer.valueOf(RotationSegment.m_245225_(direction))), 3);
                level.m_142346_((Entity)null, GameEvent.f_157797_, blockpos);
@@ -492,7 +_,7 @@
       DispenserBlock.m_52672_(Blocks.f_50143_, new OptionalDispenseItemBehavior() {
          protected ItemStack m_7498_(BlockSource p_123437_, ItemStack p_123438_) {
             Level level = p_123437_.m_7727_();
-            BlockPos blockpos = p_123437_.m_7961_().m_121945_(p_123437_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123437_.m_7961_().relative(p_123437_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             CarvedPumpkinBlock carvedpumpkinblock = (CarvedPumpkinBlock)Blocks.f_50143_;
             if (level.m_46859_(blockpos) && carvedpumpkinblock.m_51381_(level, blockpos)) {
                if (!level.f_46443_) {
@@ -535,7 +_,7 @@
          public ItemStack m_7498_(BlockSource p_123444_, ItemStack p_123445_) {
             this.m_123573_(false);
             ServerLevel serverlevel = p_123444_.m_7727_();
-            BlockPos blockpos = p_123444_.m_7961_().m_121945_(p_123444_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_123444_.m_7961_().relative(p_123444_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             BlockState blockstate = serverlevel.m_8055_(blockpos);
             if (blockstate.m_204338_(BlockTags.f_13072_, (p_123442_) -> {
                return p_123442_.m_61138_(BeehiveBlock.f_49564_) && p_123442_.m_60734_() instanceof BeehiveBlock;
@@ -554,7 +_,7 @@
       DispenserBlock.m_52672_(Items.f_42054_, new OptionalDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_123452_, ItemStack p_123453_) {
             Direction direction = p_123452_.m_6414_().m_61143_(DispenserBlock.f_52659_);
-            BlockPos blockpos = p_123452_.m_7961_().m_121945_(direction);
+            BlockPos blockpos = p_123452_.m_7961_().relative(direction);
             Level level = p_123452_.m_7727_();
             BlockState blockstate = level.m_8055_(blockpos);
             this.m_123573_(true);
@@ -575,7 +_,7 @@
       DispenserBlock.m_52672_(Items.f_42574_.m_5456_(), new ShearsDispenseItemBehavior());
       DispenserBlock.m_52672_(Items.f_42784_, new OptionalDispenseItemBehavior() {
          public ItemStack m_7498_(BlockSource p_175747_, ItemStack p_175748_) {
-            BlockPos blockpos = p_175747_.m_7961_().m_121945_(p_175747_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+            BlockPos blockpos = p_175747_.m_7961_().relative(p_175747_.m_6414_().m_61143_(DispenserBlock.f_52659_));
             Level level = p_175747_.m_7727_();
             BlockState blockstate = level.m_8055_(blockpos);
             Optional<BlockState> optional = HoneycombItem.m_150878_(blockstate);
@@ -599,13 +_,13 @@
             } else {
                ServerLevel serverlevel = p_235896_.m_7727_();
                BlockPos blockpos = p_235896_.m_7961_();
-               BlockPos blockpos1 = p_235896_.m_7961_().m_121945_(p_235896_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+               BlockPos blockpos1 = p_235896_.m_7961_().relative(p_235896_.m_6414_().m_61143_(DispenserBlock.f_52659_));
                if (!serverlevel.m_8055_(blockpos1).m_204336_(BlockTags.f_215828_)) {
                   return this.f_235893_.m_6115_(p_235896_, p_235897_);
                } else {
                   if (!serverlevel.f_46443_) {
                      for(int i = 0; i < 5; ++i) {
-                        serverlevel.m_8767_(ParticleTypes.f_123769_, (double)blockpos.m_123341_() + serverlevel.f_46441_.m_188500_(), (double)(blockpos.m_123342_() + 1), (double)blockpos.m_123343_() + serverlevel.f_46441_.m_188500_(), 1, 0.0D, 0.0D, 0.0D, 1.0D);
+                        serverlevel.sendParticles(ParticleTypes.f_123769_, blockpos.getBigDecX().add(serverlevel.f_46441_.m_188500_()), blockpos.getBigDecY().add(me.alphamode.mclong.math.BigDecimal.ONE), blockpos.getBigDecZ().add(serverlevel.f_46441_.m_188500_()), 1, 0.0D, 0.0D, 0.0D, 1.0D);
                      }
                   }
 
@@ -620,6 +_,6 @@
    }
 
    static void m_123395_(BlockSource p_123396_, Entity p_123397_, Direction p_123398_) {
-      p_123397_.m_6034_(p_123396_.m_7096_() + (double)p_123398_.m_122429_() * (0.5000099999997474D - (double)p_123397_.m_20205_() / 2.0D), p_123396_.m_7098_() + (double)p_123398_.m_122430_() * (0.5000099999997474D - (double)p_123397_.m_20206_() / 2.0D) - (double)p_123397_.m_20206_() / 2.0D, p_123396_.m_7094_() + (double)p_123398_.m_122431_() * (0.5000099999997474D - (double)p_123397_.m_20205_() / 2.0D));
+      p_123397_.setPos(p_123396_.getX().add((double)p_123398_.m_122429_() * (0.5000099999997474D - (double)p_123397_.m_20205_() / 2.0D)), p_123396_.getY().add((double)p_123398_.m_122430_() * (0.5000099999997474D - (double)p_123397_.m_20206_() / 2.0D) - (double)p_123397_.m_20206_() / 2.0D), p_123396_.getZ().add((double)p_123398_.m_122431_() * (0.5000099999997474D - (double)p_123397_.m_20205_() / 2.0D)));
    }
 }
