--- a/net/minecraft/core/dispenser/ShearsDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/ShearsDispenseItemBehavior.java
@@ -24,7 +_,7 @@
    protected ItemStack m_7498_(BlockSource p_123580_, ItemStack p_123581_) {
       ServerLevel serverlevel = p_123580_.m_7727_();
       if (!serverlevel.m_5776_()) {
-         BlockPos blockpos = p_123580_.m_7961_().m_121945_(p_123580_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+         BlockPos blockpos = p_123580_.m_7961_().relative(p_123580_.m_6414_().m_61143_(DispenserBlock.f_52659_));
          this.m_123573_(m_123576_(serverlevel, blockpos) || m_123582_(serverlevel, blockpos));
          if (this.m_123570_() && p_123581_.m_220157_(1, serverlevel.m_213780_(), (ServerPlayer)null)) {
             p_123581_.m_41764_(0);
