--- a/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
@@ -427,7 +_,7 @@
       if (this.f_58857_.m_7702_(this.f_58858_) != this) {
          return false;
       } else {
-         return p_58340_.m_20275_((double)this.f_58858_.m_123341_() + 0.5D, (double)this.f_58858_.m_123342_() + 0.5D, (double)this.f_58858_.m_123343_() + 0.5D) <= 64.0D;
+         return p_58340_.m_20275_((double)this.f_58858_.getX() + 0.5D, (double)this.f_58858_.getY() + 0.5D, (double)this.f_58858_.getZ() + 0.5D) <= 64.0D;
       }
    }
 
