--- a/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BarrelBlockEntity.java
@@ -113,9 +_,9 @@
 
    void m_58600_(BlockState p_58601_, SoundEvent p_58602_) {
       Vec3i vec3i = p_58601_.m_61143_(BarrelBlock.f_49042_).m_122436_();
-      double d0 = (double)this.f_58858_.m_123341_() + 0.5D + (double)vec3i.m_123341_() / 2.0D;
-      double d1 = (double)this.f_58858_.m_123342_() + 0.5D + (double)vec3i.m_123342_() / 2.0D;
-      double d2 = (double)this.f_58858_.m_123343_() + 0.5D + (double)vec3i.m_123343_() / 2.0D;
-      this.f_58857_.m_6263_((Player)null, d0, d1, d2, p_58602_, SoundSource.BLOCKS, 0.5F, this.f_58857_.f_46441_.m_188501_() * 0.1F + 0.9F);
+      me.alphamode.mclong.math.BigDecimal d0 = this.f_58858_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i.m_123341_() / 2.0D);
+      me.alphamode.mclong.math.BigDecimal d1 = this.f_58858_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i.m_123342_() / 2.0D);
+      me.alphamode.mclong.math.BigDecimal d2 = this.f_58858_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i.m_123343_() / 2.0D);
+      this.f_58857_.playSound((Player)null, d0, d1, d2, p_58602_, SoundSource.BLOCKS, 0.5F, this.f_58857_.f_46441_.m_188501_() * 0.1F + 0.9F);
    }
 }
