--- a/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -59,7 +_,7 @@
       if (this.f_58857_ == null) {
          return false;
       } else {
-         for(BlockPos blockpos : BlockPos.m_121940_(this.f_58858_.m_7918_(-1, -1, -1), this.f_58858_.m_7918_(1, 1, 1))) {
+         for(BlockPos blockpos : BlockPos.m_121940_(this.f_58858_.offset(-1, -1, -1), this.f_58858_.offset(1, 1, 1))) {
             if (this.f_58857_.m_8055_(blockpos).m_60734_() instanceof FireBlock) {
                return true;
             }
@@ -142,7 +_,7 @@
             }
 
             BlockPos blockpos = this.m_58899_();
-            this.f_58857_.m_6263_((Player)null, (double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), SoundEvents.f_11695_, SoundSource.BLOCKS, 1.0F, 1.0F);
+            this.f_58857_.playSound((Player)null, blockpos.getBigDecX(), blockpos.getBigDecY(), blockpos.getBigDecZ(), SoundEvents.f_11695_, SoundSource.BLOCKS, 1.0F, 1.0F);
             this.f_58857_.m_220407_(GameEvent.f_157792_, blockpos, GameEvent.Context.m_223719_(p_58745_, this.m_58900_()));
          }
 
@@ -164,7 +_,7 @@
          compoundtag.m_128365_("HivePos", NbtUtils.m_129224_(p_155138_));
          compoundtag.m_128379_("NoGravity", true);
          Direction direction = p_155139_.m_61143_(BeehiveBlock.f_49563_);
-         BlockPos blockpos = p_155138_.m_121945_(direction);
+         BlockPos blockpos = p_155138_.relative(direction);
          boolean flag = !p_155137_.m_8055_(blockpos).m_60812_(p_155137_, blockpos).m_83281_();
          if (flag && p_155142_ != BeehiveBlockEntity.BeeReleaseStatus.EMERGENCY) {
             return false;
@@ -206,10 +_,10 @@
 
                      float f = entity.m_20205_();
                      double d3 = flag ? 0.0D : 0.55D + (double)(f / 2.0F);
-                     double d0 = (double)p_155138_.m_123341_() + 0.5D + d3 * (double)direction.m_122429_();
-                     double d1 = (double)p_155138_.m_123342_() + 0.5D - (double)(entity.m_20206_() / 2.0F);
-                     double d2 = (double)p_155138_.m_123343_() + 0.5D + d3 * (double)direction.m_122431_();
-                     entity.m_7678_(d0, d1, d2, entity.m_146908_(), entity.m_146909_());
+                     me.alphamode.mclong.math.BigDecimal d0 = p_155138_.getBigDecX().add(0.5D + d3 * (double)direction.m_122429_());
+                     me.alphamode.mclong.math.BigDecimal d1 = p_155138_.getBigDecY().add(0.5D - (double)(entity.m_20206_() / 2.0F));
+                     me.alphamode.mclong.math.BigDecimal d2 = p_155138_.getBigDecZ().add(0.5D + d3 * (double)direction.m_122431_());
+                     entity.moveTo(d0, d1, d2, entity.m_146908_(), entity.m_146909_());
                   }
 
                   p_155137_.m_5594_((Player)null, p_155138_, SoundEvents.f_11696_, SoundSource.BLOCKS, 1.0F, 1.0F);
@@ -269,10 +_,10 @@
    public static void m_155144_(Level p_155145_, BlockPos p_155146_, BlockState p_155147_, BeehiveBlockEntity p_155148_) {
       m_155149_(p_155145_, p_155146_, p_155147_, p_155148_.f_58732_, p_155148_.f_58733_);
       if (!p_155148_.f_58732_.isEmpty() && p_155145_.m_213780_().m_188500_() < 0.005D) {
-         double d0 = (double)p_155146_.m_123341_() + 0.5D;
-         double d1 = (double)p_155146_.m_123342_();
-         double d2 = (double)p_155146_.m_123343_() + 0.5D;
-         p_155145_.m_6263_((Player)null, d0, d1, d2, SoundEvents.f_11698_, SoundSource.BLOCKS, 1.0F, 1.0F);
+         me.alphamode.mclong.math.BigDecimal d0 = p_155146_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB);
+         me.alphamode.mclong.math.BigDecimal d1 = p_155146_.getBigDecY();
+         me.alphamode.mclong.math.BigDecimal d2 = p_155146_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB);
+         p_155145_.playSound((Player)null, d0, d1, d2, SoundEvents.f_11698_, SoundSource.BLOCKS, 1.0F, 1.0F);
       }
 
       DebugPackets.m_179510_(p_155145_, p_155146_, p_155147_, p_155148_);
