--- a/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -31,7 +_,7 @@
    }
 
    public static BlockPos m_187472_(CompoundTag p_187473_) {
-      return new BlockPos(p_187473_.m_128451_("x"), p_187473_.m_128451_("y"), p_187473_.m_128451_("z"));
+      return new BlockPos(p_187473_.m_128454_("x"), p_187473_.m_128454_("y"), p_187473_.m_128454_("z"));
    }
 
    @Nullable
@@ -90,9 +_,9 @@
 
    private void m_187478_(CompoundTag p_187479_) {
       this.m_187474_(p_187479_);
-      p_187479_.m_128405_("x", this.f_58858_.m_123341_());
-      p_187479_.m_128405_("y", this.f_58858_.m_123342_());
-      p_187479_.m_128405_("z", this.f_58858_.m_123343_());
+      p_187479_.m_128356_("x", this.f_58858_.getX());
+      p_187479_.m_128356_("y", this.f_58858_.getY());
+      p_187479_.m_128356_("z", this.f_58858_.getZ());
    }
 
    @Nullable
