--- a/net/minecraft/world/level/block/entity/BrewingStandBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BrewingStandBlockEntity.java
@@ -176,7 +_,7 @@
          if (itemstack.m_41619_()) {
             itemstack = itemstack1;
          } else {
-            Containers.m_18992_(p_155291_, (double)p_155292_.m_123341_(), (double)p_155292_.m_123342_(), (double)p_155292_.m_123343_(), itemstack1);
+            Containers.m_18992_(p_155291_, (double)p_155292_.getX(), (double)p_155292_.getY(), (double)p_155292_.getZ(), itemstack1);
          }
       }
 
@@ -222,7 +_,7 @@
       if (this.f_58857_.m_7702_(this.f_58858_) != this) {
          return false;
       } else {
-         return !(p_59000_.m_20275_((double)this.f_58858_.m_123341_() + 0.5D, (double)this.f_58858_.m_123342_() + 0.5D, (double)this.f_58858_.m_123343_() + 0.5D) > 64.0D);
+         return !(p_59000_.m_20275_((double)this.f_58858_.getX() + 0.5D, (double)this.f_58858_.getY() + 0.5D, (double)this.f_58858_.getZ() + 0.5D) > 64.0D);
       }
    }
 
