--- a/net/minecraft/world/level/block/entity/CalibratedSculkSensorBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/CalibratedSculkSensorBlockEntity.java
@@ -35,7 +_,7 @@
 
       private int m_280214_(Level p_282204_, BlockPos p_282397_, BlockState p_282240_) {
          Direction direction = p_282240_.m_61143_(CalibratedSculkSensorBlock.f_276692_).m_122424_();
-         return p_282204_.m_277185_(p_282397_.m_121945_(direction), direction);
+         return p_282204_.m_277185_(p_282397_.relative(direction), direction);
       }
    }
 }
