--- a/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/ConduitBlockEntity.java
@@ -169,14 +_,14 @@
    private static void m_155443_(Level p_155444_, BlockPos p_155445_, List<BlockPos> p_155446_) {
       int i = p_155446_.size();
       int j = i / 7 * 16;
-      int k = p_155445_.m_123341_();
-      int l = p_155445_.m_123342_();
-      int i1 = p_155445_.m_123343_();
-      AABB aabb = (new AABB((double)k, (double)l, (double)i1, (double)(k + 1), (double)(l + 1), (double)(i1 + 1))).m_82400_((double)j).m_82363_(0.0D, (double)p_155444_.m_141928_(), 0.0D);
+      long k = p_155445_.getX();
+      long l = p_155445_.getY();
+      long i1 = p_155445_.getZ();
+      AABB aabb = (new AABB((double)k, (double)l, (double)i1, (double)(k + 1), (double)(l + 1), (double)(i1 + 1))).m_82400_((double)j).m_82363_(0.0D, (double)p_155444_.getHeight(), 0.0D);
       List<Player> list = p_155444_.m_45976_(Player.class, aabb);
       if (!list.isEmpty()) {
          for(Player player : list) {
-            if (p_155445_.m_123314_(player.m_20183_(), (double)j) && player.m_20070_()) {
+            if (p_155445_.closerThan(player.m_20183_(), (double)j) && player.m_20070_()) {
                player.m_7292_(new MobEffectInstance(MobEffects.f_19592_, 260, 0, true, true));
             }
          }
@@ -199,7 +_,7 @@
          if (!list.isEmpty()) {
             p_155413_.f_59189_ = list.get(p_155409_.f_46441_.m_188503_(list.size()));
          }
-      } else if (!p_155413_.f_59189_.m_6084_() || !p_155410_.m_123314_(p_155413_.f_59189_.m_20183_(), 8.0D)) {
+      } else if (!p_155413_.f_59189_.m_6084_() || !p_155410_.closerThan(p_155413_.f_59189_.m_20183_(), 8.0D)) {
          p_155413_.f_59189_ = null;
       }
 
@@ -227,9 +_,9 @@
    }
 
    private static AABB m_155431_(BlockPos p_155432_) {
-      int i = p_155432_.m_123341_();
-      int j = p_155432_.m_123342_();
-      int k = p_155432_.m_123343_();
+      long i = p_155432_.getX();
+      long j = p_155432_.getY();
+      long k = p_155432_.getZ();
       return (new AABB((double)i, (double)j, (double)k, (double)(i + 1), (double)(j + 1), (double)(k + 1))).m_82400_(8.0D);
    }
 
@@ -245,14 +_,14 @@
       RandomSource randomsource = p_155419_.f_46441_;
       double d0 = (double)(Mth.m_14031_((float)(p_155423_ + 35) * 0.1F) / 2.0F + 0.5F);
       d0 = (d0 * d0 + d0) * (double)0.3F;
-      Vec3 vec3 = new Vec3((double)p_155420_.m_123341_() + 0.5D, (double)p_155420_.m_123342_() + 1.5D + d0, (double)p_155420_.m_123343_() + 0.5D);
+      Vec3 vec3 = new Vec3((double)p_155420_.getX() + 0.5D, (double)p_155420_.getY() + 1.5D + d0, (double)p_155420_.getZ() + 0.5D);
 
       for(BlockPos blockpos : p_155421_) {
          if (randomsource.m_188503_(50) == 0) {
-            BlockPos blockpos1 = blockpos.m_121996_(p_155420_);
-            float f = -0.5F + randomsource.m_188501_() + (float)blockpos1.m_123341_();
-            float f1 = -2.0F + randomsource.m_188501_() + (float)blockpos1.m_123342_();
-            float f2 = -0.5F + randomsource.m_188501_() + (float)blockpos1.m_123343_();
+            BlockPos blockpos1 = blockpos.subtract(p_155420_);
+            float f = -0.5F + randomsource.m_188501_() + (float)blockpos1.getX();
+            float f1 = -2.0F + randomsource.m_188501_() + (float)blockpos1.getY();
+            float f2 = -0.5F + randomsource.m_188501_() + (float)blockpos1.getZ();
             p_155419_.m_7106_(ParticleTypes.f_123775_, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, (double)f, (double)f1, (double)f2);
          }
       }
