--- a/net/minecraft/world/level/block/entity/EnchantmentTableBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/EnchantmentTableBlockEntity.java
@@ -48,10 +_,10 @@
    public static void m_155503_(Level p_155504_, BlockPos p_155505_, BlockState p_155506_, EnchantmentTableBlockEntity p_155507_) {
       p_155507_.f_59257_ = p_155507_.f_59256_;
       p_155507_.f_59259_ = p_155507_.f_59258_;
-      Player player = p_155504_.m_45924_((double)p_155505_.m_123341_() + 0.5D, (double)p_155505_.m_123342_() + 0.5D, (double)p_155505_.m_123343_() + 0.5D, 3.0D, false);
+      Player player = p_155504_.m_45924_((double)p_155505_.getX() + 0.5D, (double)p_155505_.getY() + 0.5D, (double)p_155505_.getZ() + 0.5D, 3.0D, false);
       if (player != null) {
-         double d0 = player.m_20185_() - ((double)p_155505_.m_123341_() + 0.5D);
-         double d1 = player.m_20189_() - ((double)p_155505_.m_123343_() + 0.5D);
+         double d0 = player.m_20185_() - ((double)p_155505_.getX() + 0.5D);
+         double d1 = player.m_20189_() - ((double)p_155505_.getZ() + 0.5D);
          p_155507_.f_59260_ = (float)Mth.m_14136_(d1, d0);
          p_155507_.f_59256_ += 0.1F;
          if (p_155507_.f_59256_ < 0.5F || f_59261_.m_188503_(40) == 0) {
