--- a/net/minecraft/world/level/block/entity/EnderChestBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/EnderChestBlockEntity.java
@@ -13,11 +_,11 @@
    private final ChestLidController f_155510_ = new ChestLidController();
    private final ContainerOpenersCounter f_155511_ = new ContainerOpenersCounter() {
       protected void m_142292_(Level p_155531_, BlockPos p_155532_, BlockState p_155533_) {
-         p_155531_.m_6263_((Player)null, (double)p_155532_.m_123341_() + 0.5D, (double)p_155532_.m_123342_() + 0.5D, (double)p_155532_.m_123343_() + 0.5D, SoundEvents.f_11889_, SoundSource.BLOCKS, 0.5F, p_155531_.f_46441_.m_188501_() * 0.1F + 0.9F);
+         p_155531_.m_6263_((Player)null, (double)p_155532_.getX() + 0.5D, (double)p_155532_.getY() + 0.5D, (double)p_155532_.getZ() + 0.5D, SoundEvents.f_11889_, SoundSource.BLOCKS, 0.5F, p_155531_.f_46441_.m_188501_() * 0.1F + 0.9F);
       }
 
       protected void m_142289_(Level p_155541_, BlockPos p_155542_, BlockState p_155543_) {
-         p_155541_.m_6263_((Player)null, (double)p_155542_.m_123341_() + 0.5D, (double)p_155542_.m_123342_() + 0.5D, (double)p_155542_.m_123343_() + 0.5D, SoundEvents.f_11888_, SoundSource.BLOCKS, 0.5F, p_155541_.f_46441_.m_188501_() * 0.1F + 0.9F);
+         p_155541_.m_6263_((Player)null, (double)p_155542_.getX() + 0.5D, (double)p_155542_.getY() + 0.5D, (double)p_155542_.getZ() + 0.5D, SoundEvents.f_11888_, SoundSource.BLOCKS, 0.5F, p_155541_.f_46441_.m_188501_() * 0.1F + 0.9F);
       }
 
       protected void m_142148_(Level p_155535_, BlockPos p_155536_, BlockState p_155537_, int p_155538_, int p_155539_) {
