--- a/net/minecraft/world/level/block/entity/JigsawBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/JigsawBlockEntity.java
@@ -103,7 +_,7 @@
    }
 
    public void m_59420_(ServerLevel p_59421_, int p_59422_, boolean p_59423_) {
-      BlockPos blockpos = this.m_58899_().m_121945_(this.m_58900_().m_61143_(JigsawBlock.f_54222_).m_122625_());
+      BlockPos blockpos = this.m_58899_().relative(this.m_58900_().m_61143_(JigsawBlock.f_54222_).m_122625_());
       Registry<StructureTemplatePool> registry = p_59421_.m_9598_().m_175515_(Registries.f_256948_);
       Holder<StructureTemplatePool> holder = registry.m_246971_(this.f_59413_);
       JigsawPlacement.m_227203_(p_59421_, holder, this.f_59412_, p_59422_, blockpos, p_59423_);
