--- a/net/minecraft/world/level/block/entity/LecternBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/LecternBlockEntity.java
@@ -81,7 +_,7 @@
          if (LecternBlockEntity.this.f_58857_.m_7702_(LecternBlockEntity.this.f_58858_) != LecternBlockEntity.this) {
             return false;
          } else {
-            return p_59588_.m_20275_((double)LecternBlockEntity.this.f_58858_.m_123341_() + 0.5D, (double)LecternBlockEntity.this.f_58858_.m_123342_() + 0.5D, (double)LecternBlockEntity.this.f_58858_.m_123343_() + 0.5D) > 64.0D ? false : LecternBlockEntity.this.m_59567_();
+            return p_59588_.m_20275_((double)LecternBlockEntity.this.f_58858_.getX() + 0.5D, (double)LecternBlockEntity.this.f_58858_.getY() + 0.5D, (double)LecternBlockEntity.this.f_58858_.getZ() + 0.5D) > 64.0D ? false : LecternBlockEntity.this.m_59567_();
          }
       }
 
@@ -179,7 +_,7 @@
          component = p_59535_.m_5446_();
       }
 
-      Vec3 vec3 = Vec3.m_82512_(this.f_58858_);
+      Vec3 vec3 = Vec3.atCenterOf(this.f_58858_);
       return new CommandSourceStack(CommandSource.f_80164_, vec3, Vec2.f_82462_, (ServerLevel)this.f_58857_, 2, s, component, this.f_58857_.m_7654_(), p_59535_);
    }
 
