--- a/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
@@ -72,7 +_,7 @@
          }
 
          this.f_59605_ = null;
-         LootContext.Builder lootcontext$builder = (new LootContext.Builder((ServerLevel)this.f_58857_)).m_78972_(LootContextParams.f_81460_, Vec3.m_82512_(this.f_58858_)).m_78965_(this.f_59606_);
+         LootContext.Builder lootcontext$builder = (new LootContext.Builder((ServerLevel)this.f_58857_)).m_78972_(LootContextParams.f_81460_, Vec3.atCenterOf(this.f_58858_)).m_78965_(this.f_59606_);
          if (p_59641_ != null) {
             lootcontext$builder.m_78963_(p_59641_.m_36336_()).m_78972_(LootContextParams.f_81455_, p_59641_);
          }
@@ -126,7 +_,7 @@
       if (this.f_58857_.m_7702_(this.f_58858_) != this) {
          return false;
       } else {
-         return !(p_59619_.m_20275_((double)this.f_58858_.m_123341_() + 0.5D, (double)this.f_58858_.m_123342_() + 0.5D, (double)this.f_58858_.m_123343_() + 0.5D) > 64.0D);
+         return !(p_59619_.m_20275_((double)this.f_58858_.getX() + 0.5D, (double)this.f_58858_.getY() + 0.5D, (double)this.f_58858_.getZ() + 0.5D) > 64.0D);
       }
    }
 
