--- a/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/RandomizableContainerBlockEntity.java
@@ -73,7 +_,7 @@
          }
 
          this.f_59605_ = null;
-         LootParams.Builder lootparams$builder = (new LootParams.Builder((ServerLevel)this.f_58857_)).m_287286_(LootContextParams.f_81460_, Vec3.m_82512_(this.f_58858_));
+         LootParams.Builder lootparams$builder = (new LootParams.Builder((ServerLevel)this.f_58857_)).m_287286_(LootContextParams.f_81460_, Vec3.atCenterOf(this.f_58858_));
          if (p_59641_ != null) {
             lootparams$builder.m_287239_(p_59641_.m_36336_()).m_287286_(LootContextParams.f_81455_, p_59641_);
          }
