--- a/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SculkShriekerBlockEntity.java
@@ -149,7 +_,7 @@
       BlockPos blockpos = this.m_58899_();
       BlockState blockstate = this.m_58900_();
       p_222845_.m_7731_(blockpos, blockstate.m_61124_(SculkShriekerBlock.f_222152_, Boolean.valueOf(true)), 2);
-      p_222845_.m_186460_(blockpos, blockstate.m_60734_(), 90);
+      p_222845_.scheduleTick(blockpos, blockstate.m_60734_(), 90);
       p_222845_.m_46796_(3007, blockpos, 0);
       p_222845_.m_220407_(GameEvent.f_223701_, blockpos, GameEvent.Context.m_223717_(p_222846_));
    }
@@ -164,7 +_,7 @@
             this.m_280188_(p_222840_);
          }
 
-         Warden.m_219375_(p_222840_, Vec3.m_82512_(this.m_58899_()), (Entity)null, 40);
+         Warden.m_219375_(p_222840_, Vec3.atCenterOf(this.m_58899_()), (Entity)null, 40);
       }
 
    }
@@ -173,10 +_,10 @@
       SoundEvent soundevent = f_222829_.get(this.f_222831_);
       if (soundevent != null) {
          BlockPos blockpos = this.m_58899_();
-         int i = blockpos.m_123341_() + Mth.m_216287_(p_281300_.f_46441_, -10, 10);
-         int j = blockpos.m_123342_() + Mth.m_216287_(p_281300_.f_46441_, -10, 10);
-         int k = blockpos.m_123343_() + Mth.m_216287_(p_281300_.f_46441_, -10, 10);
-         p_281300_.m_6263_((Player)null, (double)i, (double)j, (double)k, soundevent, SoundSource.HOSTILE, 5.0F, 1.0F);
+         me.alphamode.mclong.math.BigDecimal i = blockpos.getBigDecX().add(Mth.m_216287_(this.f_58857_.f_46441_, -10, 10));
+         me.alphamode.mclong.math.BigDecimal j = blockpos.getBigDecY().add(Mth.m_216287_(this.f_58857_.f_46441_, -10, 10));
+         me.alphamode.mclong.math.BigDecimal k = blockpos.getBigDecZ().add(Mth.m_216287_(this.f_58857_.f_46441_, -10, 10));
+         p_281300_.playSound((Player)null, i, j, k, soundevent, SoundSource.HOSTILE, 5.0F, 1.0F);
       }
 
    }
