--- a/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
@@ -156,7 +_,7 @@
          p_155833_.f_59927_ = 100;
          if (p_155833_.f_59928_ == null && p_155829_.m_46472_() == Level.f_46430_) {
             BlockPos blockpos = m_155818_(serverlevel, p_155830_);
-            blockpos = blockpos.m_6630_(10);
+            blockpos = blockpos.above(10);
             f_59925_.debug("Creating portal at {}", (Object)blockpos);
             m_155821_(serverlevel, blockpos, EndGatewayConfiguration.m_67650_(p_155830_, false));
             p_155833_.f_59928_ = blockpos;
@@ -182,7 +_,7 @@
             }
 
             entity.m_20091_();
-            entity.m_20324_((double)blockpos1.m_123341_() + 0.5D, (double)blockpos1.m_123342_(), (double)blockpos1.m_123343_() + 0.5D);
+            entity.m_20324_((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY(), (double)blockpos1.getZ() + 0.5D);
          }
 
          m_155849_(p_155829_, p_155830_, p_155831_, p_155833_);
@@ -190,9 +_,9 @@
    }
 
    private static BlockPos m_155825_(Level p_155826_, BlockPos p_155827_) {
-      BlockPos blockpos = m_59942_(p_155826_, p_155827_.m_7918_(0, 2, 0), 5, false);
+      BlockPos blockpos = m_59942_(p_155826_, p_155827_.offset(0, 2, 0), 5, false);
       f_59925_.debug("Best exit position for portal at {} is {}", p_155827_, blockpos);
-      return blockpos.m_7494_();
+      return blockpos.above();
    }
 
    private static BlockPos m_155818_(ServerLevel p_155819_, BlockPos p_155820_) {
@@ -216,7 +_,7 @@
    }
 
    private static Vec3 m_155841_(ServerLevel p_155842_, BlockPos p_155843_) {
-      Vec3 vec3 = (new Vec3((double)p_155843_.m_123341_(), 0.0D, (double)p_155843_.m_123343_())).m_82541_();
+      Vec3 vec3 = (new Vec3((double)p_155843_.getX(), 0.0D, (double)p_155843_.getZ())).m_82541_();
       int i = 1024;
       Vec3 vec31 = vec3.m_82490_(1024.0D);
 
@@ -233,7 +_,7 @@
    }
 
    private static boolean m_155815_(ServerLevel p_155816_, Vec3 p_155817_) {
-      return m_59947_(p_155816_, p_155817_).m_280525_() == -1;
+      return m_59947_(p_155816_, p_155817_).getHighestSectionPosition() <= p_155816_.getMinBuildHeight();
    }
 
    private static BlockPos m_59942_(BlockGetter p_59943_, BlockPos p_59944_, int p_59945_, boolean p_59946_) {
@@ -242,8 +_,8 @@
       for(int i = -p_59945_; i <= p_59945_; ++i) {
          for(int j = -p_59945_; j <= p_59945_; ++j) {
             if (i != 0 || j != 0 || p_59946_) {
-               for(int k = p_59943_.m_151558_() - 1; k > (blockpos == null ? p_59943_.m_141937_() : blockpos.m_123342_()); --k) {
-                  BlockPos blockpos1 = new BlockPos(p_59944_.m_123341_() + i, k, p_59944_.m_123343_() + j);
+               for(long k = p_59943_.getMaxBuildHeight() - 1; k > (blockpos == null ? p_59943_.getMinBuildHeight() : blockpos.getY()); --k) {
+                  BlockPos blockpos1 = new BlockPos(p_59944_.getX() + i, k, p_59944_.getZ() + j);
                   BlockState blockstate = p_59943_.m_8055_(blockpos1);
                   if (blockstate.m_60838_(p_59943_, blockpos1) && (p_59946_ || !blockstate.m_60713_(Blocks.f_50752_))) {
                      blockpos = blockpos1;
@@ -258,24 +_,24 @@
    }
 
    private static LevelChunk m_59947_(Level p_59948_, Vec3 p_59949_) {
-      return p_59948_.m_6325_(Mth.m_14107_(p_59949_.f_82479_ / 16.0D), Mth.m_14107_(p_59949_.f_82481_ / 16.0D));
+      return p_59948_.getChunk(Mth.m_14107_(p_59949_.f_82479_ / 16.0D), Mth.m_14107_(p_59949_.f_82481_ / 16.0D));
    }
 
    @Nullable
    private static BlockPos m_59953_(LevelChunk p_59954_) {
       ChunkPos chunkpos = p_59954_.m_7697_();
-      BlockPos blockpos = new BlockPos(chunkpos.m_45604_(), 30, chunkpos.m_45605_());
-      int i = p_59954_.m_62098_() + 16 - 1;
-      BlockPos blockpos1 = new BlockPos(chunkpos.m_45608_(), i, chunkpos.m_45609_());
+      BlockPos blockpos = new BlockPos(chunkpos.getMinBlockX(), 30, chunkpos.getMinBlockZ());
+      long i = p_59954_.getHighestSectionPosition() + 16 - 1;
+      BlockPos blockpos1 = new BlockPos(chunkpos.getMaxBlockX(), i, chunkpos.getMaxBlockZ());
       BlockPos blockpos2 = null;
       double d0 = 0.0D;
 
       for(BlockPos blockpos3 : BlockPos.m_121940_(blockpos, blockpos1)) {
          BlockState blockstate = p_59954_.m_8055_(blockpos3);
-         BlockPos blockpos4 = blockpos3.m_7494_();
-         BlockPos blockpos5 = blockpos3.m_6630_(2);
+         BlockPos blockpos4 = blockpos3.above();
+         BlockPos blockpos5 = blockpos3.above(2);
          if (blockstate.m_60713_(Blocks.f_50259_) && !p_59954_.m_8055_(blockpos4).m_60838_(p_59954_, blockpos4) && !p_59954_.m_8055_(blockpos5).m_60838_(p_59954_, blockpos5)) {
-            double d1 = blockpos3.m_203198_(0.0D, 0.0D, 0.0D);
+            double d1 = blockpos3.distToCenterSqr(0.0D, 0.0D, 0.0D);
             if (blockpos2 == null || d1 < d0) {
                blockpos2 = blockpos3;
                d0 = d1;
@@ -291,7 +_,7 @@
    }
 
    public boolean m_6665_(Direction p_59959_) {
-      return Block.m_152444_(this.m_58900_(), this.f_58857_, this.m_58899_(), p_59959_, this.m_58899_().m_121945_(p_59959_));
+      return Block.m_152444_(this.m_58900_(), this.f_58857_, this.m_58899_(), p_59959_, this.m_58899_().relative(p_59959_));
    }
 
    public int m_59975_() {
