--- a/net/minecraft/world/level/block/entity/TrappedChestBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TrappedChestBlockEntity.java
@@ -15,7 +_,7 @@
       if (p_155868_ != p_155869_) {
          Block block = p_155867_.m_60734_();
          p_155865_.m_46672_(p_155866_, block);
-         p_155865_.m_46672_(p_155866_.m_7495_(), block);
+         p_155865_.m_46672_(p_155866_.below(), block);
       }
 
    }
