--- a/net/minecraft/server/rcon/RconConsoleSource.java
+++ b/net/minecraft/server/rcon/RconConsoleSource.java
@@ -29,7 +_,7 @@
 
    public CommandSourceStack m_11514_() {
       ServerLevel serverlevel = this.f_11502_.m_129783_();
-      return new CommandSourceStack(this, Vec3.m_82528_(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "Rcon", f_11500_, this.f_11502_, (Entity)null);
+      return new CommandSourceStack(this, Vec3.atLowerCornerOf(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "Rcon", f_11500_, this.f_11502_, (Entity)null);
    }
 
    public void m_213846_(Component p_215653_) {
