--- a/net/minecraft/world/entity/npc/AbstractVillager.java
+++ b/net/minecraft/world/entity/npc/AbstractVillager.java
@@ -183,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.02D;
          double d1 = this.f_19796_.m_188583_() * 0.02D;
          double d2 = this.f_19796_.m_188583_() * 0.02D;
-         this.m_9236_().m_7106_(p_35288_, this.m_20208_(1.0D), this.m_20187_() + 1.0D, this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(p_35288_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.math.BigDecimal.ONE), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
