--- a/net/minecraft/world/entity/npc/CatSpawner.java
+++ b/net/minecraft/world/entity/npc/CatSpawner.java
@@ -37,9 +_,9 @@
                RandomSource randomsource = p_35330_.f_46441_;
                int i = (8 + randomsource.m_188503_(24)) * (randomsource.m_188499_() ? -1 : 1);
                int j = (8 + randomsource.m_188503_(24)) * (randomsource.m_188499_() ? -1 : 1);
-               BlockPos blockpos = player.m_20183_().m_7918_(i, 0, j);
+               BlockPos blockpos = player.m_20183_().offset(i, 0, j);
                int k = 10;
-               if (!p_35330_.m_151572_(blockpos.m_123341_() - 10, blockpos.m_123343_() - 10, blockpos.m_123341_() + 10, blockpos.m_123343_() + 10)) {
+               if (!p_35330_.hasChunksAt(blockpos.getBigX().subtract(me.alphamode.mclong.core.BigConstants.Ints.TEN), blockpos.getBigZ().subtract(me.alphamode.mclong.core.BigConstants.Ints.TEN), blockpos.getBigX().add(me.alphamode.mclong.core.BigConstants.Ints.TEN), blockpos.getBigZ().add(me.alphamode.mclong.core.BigConstants.Ints.TEN))) {
                   return 0;
                } else {
                   if (NaturalSpawner.m_47051_(SpawnPlacements.Type.ON_GROUND, p_35330_, blockpos, EntityType.f_20553_)) {
