--- a/net/minecraft/world/entity/npc/Villager.java
+++ b/net/minecraft/world/entity/npc/Villager.java
@@ -535,7 +_,7 @@
       }
 
       if (p_35475_.m_45383_()) {
-         this.m_9236_().m_7967_(new ExperienceOrb(this.m_9236_(), this.m_20185_(), this.m_20186_() + 0.5D, this.m_20189_(), i));
+         this.m_9236_().m_7967_(new ExperienceOrb(this.m_9236_(), this.getBigX(), this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigZ(), i));
       }
 
    }
