--- a/net/minecraft/world/entity/npc/Villager.java
+++ b/net/minecraft/world/entity/npc/Villager.java
@@ -535,7 +_,7 @@
       }
 
       if (p_35475_.m_45383_()) {
-         this.m_9236_().m_7967_(new ExperienceOrb(this.m_9236_(), this.m_20185_(), this.m_20186_() + 0.5D, this.m_20189_(), i));
+         this.m_9236_().m_7967_(new ExperienceOrb(this.m_9236_(), this.getBigX(), this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigZ(), i));
       }
 
    }
@@ -726,7 +_,7 @@
          f_201976_.info("Villager {} was struck by lightning {}.", this, p_35410_);
          Witch witch = EntityType.f_20495_.m_20615_(p_35409_);
          if (witch != null) {
-            witch.m_7678_(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_146908_(), this.m_146909_());
+            witch.moveTo(this.getBigX(), this.getBigY(), this.getBigZ(), this.m_146908_(), this.m_146909_());
             witch.m_6518_(p_35409_, p_35409_.m_6436_(witch.m_20183_()), MobSpawnType.CONVERSION, (SpawnGroupData)null, (CompoundTag)null);
             witch.m_21557_(this.m_21525_());
             if (this.m_8077_()) {
