--- a/net/minecraft/world/entity/npc/VillagerTrades.java
+++ b/net/minecraft/world/entity/npc/VillagerTrades.java
@@ -379,7 +_,7 @@
             ServerLevel serverlevel = (ServerLevel)p_219708_.f_19853_;
             BlockPos blockpos = serverlevel.m_215011_(this.f_35806_, p_219708_.m_20183_(), 100, true);
             if (blockpos != null) {
-               ItemStack itemstack = MapItem.m_42886_(serverlevel, blockpos.m_123341_(), blockpos.m_123343_(), (byte)2, true, true);
+               ItemStack itemstack = MapItem.create(serverlevel, blockpos.getX(), blockpos.getZ(), (byte)2, true, true);
                MapItem.m_42850_(serverlevel, itemstack);
                MapItemSavedData.m_77925_(itemstack, blockpos, "+", this.f_35807_);
                itemstack.m_41714_(Component.m_237115_(this.f_207765_));
