--- a/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
+++ b/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
@@ -139,9 +_,9 @@
       BlockPos blockpos = null;
 
       for(int i = 0; i < 10; ++i) {
-         int j = p_35930_.m_123341_() + this.f_35908_.m_188503_(p_35931_ * 2) - p_35931_;
-         int k = p_35930_.m_123343_() + this.f_35908_.m_188503_(p_35931_ * 2) - p_35931_;
-         int l = p_35929_.m_6924_(Heightmap.Types.WORLD_SURFACE, j, k);
+         long j = p_35930_.getX() + this.f_35908_.m_188503_(p_35931_ * 2) - p_35931_;
+         long k = p_35930_.getZ() + this.f_35908_.m_188503_(p_35931_ * 2) - p_35931_;
+         long l = p_35929_.getHeight(Heightmap.Types.WORLD_SURFACE, j, k);
          BlockPos blockpos1 = new BlockPos(j, l, k);
          if (NaturalSpawner.m_47051_(SpawnPlacements.Type.ON_GROUND, p_35929_, blockpos1, EntityType.f_20494_)) {
             blockpos = blockpos1;
