--- a/net/minecraft/world/level/levelgen/blockpredicates/ReplaceablePredicate.java
+++ b/net/minecraft/world/level/levelgen/blockpredicates/ReplaceablePredicate.java
@@ -10,7 +_,7 @@
       return m_190546_(p_190529_).apply(p_190529_, ReplaceablePredicate::new);
    });
 
-   public ReplaceablePredicate(Vec3i p_190524_) {
+   public ReplaceablePredicate(me.alphamode.mclong.core.Vec3l p_190524_) {
       super(p_190524_);
    }
 
