--- a/net/minecraft/world/entity/ai/control/BodyRotationControl.java
+++ b/net/minecraft/world/entity/ai/control/BodyRotationControl.java
@@ -58,8 +_,8 @@
    }
 
    private boolean m_24884_() {
-      double d0 = this.f_24875_.m_20185_() - this.f_24875_.f_19854_;
-      double d1 = this.f_24875_.m_20189_() - this.f_24875_.f_19856_;
+      double d0 = this.f_24875_.getBigX().subtract(this.f_24875_.f_19854_).doubleValue();
+      double d1 = this.f_24875_.getBigZ().subtract(this.f_24875_.f_19856_).doubleValue();
       return d0 * d0 + d1 * d1 > (double)2.5000003E-7F;
    }
 }
