--- a/net/minecraft/world/entity/ai/control/SmoothSwimmingMoveControl.java
+++ b/net/minecraft/world/entity/ai/control/SmoothSwimmingMoveControl.java
@@ -28,9 +_,9 @@
       }
 
       if (this.f_24981_ == MoveControl.Operation.MOVE_TO && !this.f_24974_.m_21573_().m_26571_()) {
-         double d0 = this.f_24975_ - this.f_24974_.m_20185_();
-         double d1 = this.f_24976_ - this.f_24974_.m_20186_();
-         double d2 = this.f_24977_ - this.f_24974_.m_20189_();
+         double d0 = this.f_24975_.subtract(this.f_24974_.getBigX()).doubleValue();
+         double d1 = this.f_24976_.subtract(this.f_24974_.getBigY()).doubleValue();
+         double d2 = this.f_24977_.subtract(this.f_24974_.getBigZ()).doubleValue();
          double d3 = d0 * d0 + d1 * d1 + d2 * d2;
          if (d3 < (double)2.5000003E-7F) {
             this.f_24974_.m_21564_(0.0F);
