--- a/net/minecraft/world/level/dimension/end/EndDragonFight.java
+++ b/net/minecraft/world/level/dimension/end/EndDragonFight.java
@@ -99,7 +_,7 @@
    public EndDragonFight(ServerLevel p_289771_, long p_289793_, EndDragonFight.Data p_289768_, BlockPos p_289794_) {
       this.f_64061_ = p_289771_;
       this.f_286985_ = p_289794_;
-      this.f_286991_ = EntitySelector.f_20402_.and(EntitySelector.m_20410_((double)p_289794_.m_123341_(), (double)(128 + p_289794_.m_123342_()), (double)p_289794_.m_123343_(), 192.0D));
+      this.f_286991_ = EntitySelector.f_20402_.and(EntitySelector.withinDistance(p_289794_.getBigDecX(), (p_289794_.getBigDecY().add(128)), p_289794_.getBigDecZ(), 192.0D));
       this.f_64071_ = p_289768_.f_289710_;
       this.f_64070_ = p_289768_.f_289702_.orElse((UUID)null);
       this.f_64068_ = p_289768_.f_289711_;
@@ -240,7 +_,7 @@
    private boolean m_64104_() {
       for(int i = -8; i <= 8; ++i) {
          for(int j = -8; j <= 8; ++j) {
-            LevelChunk levelchunk = this.f_64061_.m_6325_(i, j);
+            LevelChunk levelchunk = this.f_64061_.getChunk(me.alphamode.mclong.math.BigInteger.valueOf(i), me.alphamode.mclong.math.BigInteger.valueOf(j));
 
             for(BlockEntity blockentity : levelchunk.m_62954_().values()) {
                if (blockentity instanceof TheEndPortalBlockEntity) {
@@ -257,9 +_,9 @@
    private BlockPattern.BlockPatternMatch m_64105_() {
       ChunkPos chunkpos = new ChunkPos(this.f_286985_);
 
-      for(int i = -8 + chunkpos.f_45578_; i <= 8 + chunkpos.f_45578_; ++i) {
-         for(int j = -8 + chunkpos.f_45579_; j <= 8 + chunkpos.f_45579_; ++j) {
-            LevelChunk levelchunk = this.f_64061_.m_6325_(i, j);
+      for(long i = -8 + chunkpos.x(); i <= 8 + chunkpos.x(); ++i) {
+         for(long j = -8 + chunkpos.z(); j <= 8 + chunkpos.z(); ++j) {
+            LevelChunk levelchunk = this.f_64061_.getChunk(me.alphamode.mclong.math.BigInteger.valueOf(i), me.alphamode.mclong.math.BigInteger.valueOf(j));
 
             for(BlockEntity blockentity : levelchunk.m_62954_().values()) {
                if (blockentity instanceof TheEndPortalBlockEntity) {
@@ -278,10 +_,10 @@
       }
 
       BlockPos blockpos1 = EndPodiumFeature.m_287210_(this.f_286985_);
-      int k = this.f_64061_.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos1).m_123342_();
+      long k = this.f_64061_.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos1).getY();
 
-      for(int l = k; l >= this.f_64061_.m_141937_(); --l) {
-         BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch1 = this.f_64063_.m_61184_(this.f_64061_, new BlockPos(blockpos1.m_123341_(), l, blockpos1.m_123343_()));
+      for(long l = k; l >= this.f_64061_.getMinBuildHeight(); --l) {
+         BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch1 = this.f_64063_.m_61184_(this.f_64061_, new BlockPos(blockpos1.getX(), l, blockpos1.getZ()));
          if (blockpattern$blockpatternmatch1 != null) {
             if (this.f_64072_ == null) {
                this.f_64072_ = blockpattern$blockpatternmatch1.m_61229_(3, 3, 3).m_61176_();
@@ -300,9 +_,9 @@
       } else {
          ChunkPos chunkpos = new ChunkPos(this.f_286985_);
 
-         for(int i = -8 + chunkpos.f_45578_; i <= 8 + chunkpos.f_45578_; ++i) {
-            for(int j = 8 + chunkpos.f_45579_; j <= 8 + chunkpos.f_45579_; ++j) {
-               ChunkAccess chunkaccess = this.f_64061_.m_6522_(i, j, ChunkStatus.f_62326_, false);
+         for(long i = -8 + chunkpos.x(); i <= 8 + chunkpos.x(); ++i) {
+            for(long j = 8 + chunkpos.z(); j <= 8 + chunkpos.z(); ++j) {
+               ChunkAccess chunkaccess = this.f_64061_.getChunk(me.alphamode.mclong.math.BigInteger.valueOf(i), me.alphamode.mclong.math.BigInteger.valueOf(j), ChunkStatus.f_62326_, false);
                if (!(chunkaccess instanceof LevelChunk)) {
                   return false;
                }
@@ -390,7 +_,7 @@
    private void m_64093_(boolean p_64094_) {
       EndPodiumFeature endpodiumfeature = new EndPodiumFeature(p_64094_);
       if (this.f_64072_ == null) {
-         for(this.f_64072_ = this.f_64061_.m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.m_287210_(this.f_286985_)).m_7495_(); this.f_64061_.m_8055_(this.f_64072_).m_60713_(Blocks.f_50752_) && this.f_64072_.m_123342_() > this.f_64061_.m_5736_(); this.f_64072_ = this.f_64072_.m_7495_()) {
+         for(this.f_64072_ = this.f_64061_.m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.m_287210_(this.f_286985_)).below(); this.f_64061_.m_8055_(this.f_64072_).m_60713_(Blocks.f_50752_) && this.f_64072_.getY() > this.f_64061_.m_5736_(); this.f_64072_ = this.f_64072_.below()) {
          }
       }
 
@@ -399,13 +_,13 @@
 
    @Nullable
    private EnderDragon m_64110_() {
-      this.f_64061_.m_46745_(new BlockPos(this.f_286985_.m_123341_(), 128 + this.f_286985_.m_123342_(), this.f_286985_.m_123343_()));
+      this.f_64061_.m_46745_(new BlockPos(this.f_286985_.getX(), 128 + this.f_286985_.getY(), this.f_286985_.getZ()));
       EnderDragon enderdragon = EntityType.f_20565_.m_20615_(this.f_64061_);
       if (enderdragon != null) {
          enderdragon.m_287231_(this);
          enderdragon.m_287266_(this.f_286985_);
          enderdragon.m_31157_().m_31416_(EnderDragonPhase.f_31377_);
-         enderdragon.m_7678_((double)this.f_286985_.m_123341_(), (double)(128 + this.f_286985_.m_123342_()), (double)this.f_286985_.m_123343_(), this.f_64061_.f_46441_.m_188501_() * 360.0F, 0.0F);
+         enderdragon.moveTo(this.f_286985_.getBigDecX(), (this.f_286985_.getBigY().add(f_156736_).toBigDecimal()), this.f_286985_.getBigDecZ(), this.f_64061_.f_46441_.m_188501_() * 360.0F, 0.0F);
          this.f_64061_.m_7967_(enderdragon);
          this.f_64070_ = enderdragon.m_20148_();
       }
@@ -466,10 +_,10 @@
          }
 
          List<EndCrystal> list1 = Lists.newArrayList();
-         BlockPos blockpos1 = blockpos.m_6630_(1);
+         BlockPos blockpos1 = blockpos.above(1);
 
          for(Direction direction : Direction.Plane.HORIZONTAL) {
-            List<EndCrystal> list = this.f_64061_.m_45976_(EndCrystal.class, new AABB(blockpos1.m_5484_(direction, 2)));
+            List<EndCrystal> list = this.f_64061_.m_45976_(EndCrystal.class, new AABB(blockpos1.relative(direction, 2)));
             if (list.isEmpty()) {
                return;
             }
