--- a/net/minecraft/nbt/visitors/CollectToTag.java
+++ b/net/minecraft/nbt/visitors/CollectToTag.java
@@ -85,6 +_,11 @@
       return StreamTagVisitor.ValueResult.CONTINUE;
    }
 
+   public StreamTagVisitor.ValueResult visit(it.unimi.dsi.fastutil.longs.Long2ByteMap pEntry) {
+      this.m_197682_(new me.alphamode.mclong.nbt.LongByteArrayTag(pEntry));
+      return StreamTagVisitor.ValueResult.CONTINUE;
+   }
+
    public StreamTagVisitor.ValueResult m_196376_(int[] p_197697_) {
       this.m_197682_(new IntArrayTag(p_197697_));
       return StreamTagVisitor.ValueResult.CONTINUE;
