--- a/net/minecraft/nbt/visitors/SkipAll.java
+++ b/net/minecraft/nbt/visitors/SkipAll.java
@@ -43,6 +_,10 @@
       return StreamTagVisitor.ValueResult.CONTINUE;
    }
 
+   default StreamTagVisitor.ValueResult visit(it.unimi.dsi.fastutil.longs.Long2ByteMap p_197741_) {
+      return StreamTagVisitor.ValueResult.CONTINUE;
+   }
+
    default StreamTagVisitor.ValueResult m_196376_(int[] p_197743_) {
       return StreamTagVisitor.ValueResult.CONTINUE;
    }
