--- a/net/minecraft/world/level/block/piston/MovingPistonBlock.java
+++ b/net/minecraft/world/level/block/piston/MovingPistonBlock.java
@@ -67,7 +_,7 @@
    }
 
    public void m_6786_(LevelAccessor p_60061_, BlockPos p_60062_, BlockState p_60063_) {
-      BlockPos blockpos = p_60062_.m_121945_(p_60063_.m_61143_(f_60046_).m_122424_());
+      BlockPos blockpos = p_60062_.relative(p_60063_.m_61143_(f_60046_).m_122424_());
       BlockState blockstate = p_60061_.m_8055_(blockpos);
       if (blockstate.m_60734_() instanceof PistonBaseBlock && blockstate.m_61143_(PistonBaseBlock.f_60153_)) {
          p_60061_.m_7471_(blockpos, false);
@@ -85,7 +_,7 @@
    }
 
    public List<ItemStack> m_49635_(BlockState p_287650_, LootParams.Builder p_287754_) {
-      PistonMovingBlockEntity pistonmovingblockentity = this.m_60053_(p_287754_.m_287258_(), BlockPos.m_274446_(p_287754_.m_287261_(LootContextParams.f_81460_)));
+      PistonMovingBlockEntity pistonmovingblockentity = this.m_60053_(p_287754_.m_287258_(), BlockPos.containing(p_287754_.m_287261_(LootContextParams.f_81460_)));
       return pistonmovingblockentity == null ? Collections.emptyList() : pistonmovingblockentity.m_60400_().m_287290_(p_287754_);
    }
 
