--- a/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
+++ b/net/minecraft/world/level/block/piston/PistonMovingBlockEntity.java
@@ -228,7 +_,7 @@
 
    private static AABB m_155925_(BlockPos p_155926_, AABB p_155927_, PistonMovingBlockEntity p_155928_) {
       double d0 = (double)p_155928_.m_60390_(p_155928_.f_60339_);
-      return p_155927_.m_82386_((double)p_155926_.m_123341_() + d0 * (double)p_155928_.f_60335_.m_122429_(), (double)p_155926_.m_123342_() + d0 * (double)p_155928_.f_60335_.m_122430_(), (double)p_155926_.m_123343_() + d0 * (double)p_155928_.f_60335_.m_122431_());
+      return p_155927_.m_82386_((double)p_155926_.getX() + d0 * (double)p_155928_.f_60335_.m_122429_(), (double)p_155926_.getY() + d0 * (double)p_155928_.f_60335_.m_122430_(), (double)p_155926_.getZ() + d0 * (double)p_155928_.f_60335_.m_122431_());
    }
 
    private static void m_155920_(BlockPos p_155921_, Entity p_155922_, Direction p_155923_, double p_155924_) {
