--- a/net/minecraft/world/level/block/piston/PistonMath.java
+++ b/net/minecraft/world/level/block/piston/PistonMath.java
@@ -5,23 +_,23 @@
 
 public class PistonMath {
    public static AABB m_60328_(AABB p_60329_, Direction p_60330_, double p_60331_) {
-      double d0 = p_60331_ * (double)p_60330_.m_122421_().m_122540_();
-      double d1 = Math.min(d0, 0.0D);
-      double d2 = Math.max(d0, 0.0D);
+      me.alphamode.mclong.math.BigDecimal d0 = me.alphamode.mclong.math.BigDecimal.valueOf(p_60331_ * (double)p_60330_.m_122421_().m_122540_());
+      me.alphamode.mclong.math.BigDecimal d1 = d0.min(me.alphamode.mclong.math.BigDecimal.ZERO);
+      me.alphamode.mclong.math.BigDecimal d2 = d0.max(me.alphamode.mclong.math.BigDecimal.ZERO);
       switch (p_60330_) {
          case WEST:
-            return new AABB(p_60329_.f_82288_ + d1, p_60329_.f_82289_, p_60329_.f_82290_, p_60329_.f_82288_ + d2, p_60329_.f_82292_, p_60329_.f_82293_);
+            return new AABB(p_60329_.f_82288_.add(d1), p_60329_.f_82289_, p_60329_.f_82290_, p_60329_.f_82288_.add(d2), p_60329_.f_82292_, p_60329_.f_82293_);
          case EAST:
-            return new AABB(p_60329_.f_82291_ + d1, p_60329_.f_82289_, p_60329_.f_82290_, p_60329_.f_82291_ + d2, p_60329_.f_82292_, p_60329_.f_82293_);
+            return new AABB(p_60329_.f_82291_.add(d1), p_60329_.f_82289_, p_60329_.f_82290_, p_60329_.f_82291_.add(d2), p_60329_.f_82292_, p_60329_.f_82293_);
          case DOWN:
-            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_ + d1, p_60329_.f_82290_, p_60329_.f_82291_, p_60329_.f_82289_ + d2, p_60329_.f_82293_);
+            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_.add(d1), p_60329_.f_82290_, p_60329_.f_82291_, p_60329_.f_82289_.add(d2), p_60329_.f_82293_);
          case UP:
          default:
-            return new AABB(p_60329_.f_82288_, p_60329_.f_82292_ + d1, p_60329_.f_82290_, p_60329_.f_82291_, p_60329_.f_82292_ + d2, p_60329_.f_82293_);
+            return new AABB(p_60329_.f_82288_, p_60329_.f_82292_.add(d1), p_60329_.f_82290_, p_60329_.f_82291_, p_60329_.f_82292_.add(d2), p_60329_.f_82293_);
          case NORTH:
-            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_, p_60329_.f_82290_ + d1, p_60329_.f_82291_, p_60329_.f_82292_, p_60329_.f_82290_ + d2);
+            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_, p_60329_.f_82290_.add(d1), p_60329_.f_82291_, p_60329_.f_82292_, p_60329_.f_82290_.add(d2));
          case SOUTH:
-            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_, p_60329_.f_82293_ + d1, p_60329_.f_82291_, p_60329_.f_82292_, p_60329_.f_82293_ + d2);
+            return new AABB(p_60329_.f_82288_, p_60329_.f_82289_, p_60329_.f_82293_.add(d1), p_60329_.f_82291_, p_60329_.f_82292_, p_60329_.f_82293_.add(d2));
       }
    }
 }
