--- a/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
@@ -31,9 +_,9 @@
       this.f_31032_ = this.f_19796_.m_188503_(100000);
    }
 
-   public EndCrystal(Level p_31040_, double p_31041_, double p_31042_, double p_31043_) {
-      this(EntityType.f_20564_, p_31040_);
-      this.m_6034_(p_31041_, p_31042_, p_31043_);
+   public EndCrystal(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      this(EntityType.f_20564_, pLevel);
+      this.setPos(pX, pY, pZ);
    }
 
    protected Entity.MovementEmission m_142319_() {
@@ -89,7 +_,7 @@
             this.m_142687_(Entity.RemovalReason.KILLED);
             if (!p_31050_.m_269533_(DamageTypeTags.f_268415_)) {
                DamageSource damagesource = p_31050_.m_7639_() != null ? this.m_269291_().m_269036_(this, p_31050_.m_7639_()) : null;
-               this.m_9236_().m_254877_(this, damagesource, (ExplosionDamageCalculator)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), 6.0F, false, Level.ExplosionInteraction.BLOCK);
+               this.m_9236_().explode(this, damagesource, (ExplosionDamageCalculator)null, this.getBigX(), this.getBigY(), this.getBigZ(), 6.0F, false, Level.ExplosionInteraction.BLOCK);
             }
 
             this.m_31047_(p_31050_);
