--- a/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
@@ -89,7 +_,7 @@
             this.m_142687_(Entity.RemovalReason.KILLED);
             if (!p_31050_.m_269533_(DamageTypeTags.f_268415_)) {
                DamageSource damagesource = p_31050_.m_7639_() != null ? this.m_269291_().m_269036_(this, p_31050_.m_7639_()) : null;
-               this.m_9236_().m_254877_(this, damagesource, (ExplosionDamageCalculator)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), 6.0F, false, Level.ExplosionInteraction.BLOCK);
+               this.m_9236_().explode(this, damagesource, (ExplosionDamageCalculator)null, this.getBigX(), this.getBigY(), this.getBigZ(), 6.0F, false, Level.ExplosionInteraction.BLOCK);
             }
 
             this.m_31047_(p_31050_);
