--- a/net/minecraft/client/model/ElytraModel.java
+++ b/net/minecraft/client/model/ElytraModel.java
@@ -49,9 +_,9 @@
       if (p_102544_.m_21255_()) {
          float f4 = 1.0F;
          Vec3 vec3 = p_102544_.m_20184_();
-         if (vec3.f_82480_ < 0.0D) {
+         if (vec3.m_7098_() < 0.0D) {
             Vec3 vec31 = vec3.m_82541_();
-            f4 = 1.0F - (float)Math.pow(-vec31.f_82480_, 1.5D);
+            f4 = 1.0F - (float)Math.pow(-vec31.m_7098_(), 1.5D);
          }
 
          f = f4 * 0.34906584F + (1.0F - f4) * f;
