--- a/net/minecraft/client/model/GuardianModel.java
+++ b/net/minecraft/client/model/GuardianModel.java
@@ -93,7 +_,7 @@
       if (entity != null) {
          Vec3 vec3 = entity.m_20299_(0.0F);
          Vec3 vec31 = p_102719_.m_20299_(0.0F);
-         double d0 = vec3.f_82480_ - vec31.f_82480_;
+         double d0 = vec3.m_7098_() - vec31.m_7098_();
          if (d0 > 0.0D) {
             this.f_102702_.f_104201_ = 0.0F;
          } else {
@@ -101,8 +_,8 @@
          }
 
          Vec3 vec32 = p_102719_.m_20252_(0.0F);
-         vec32 = new Vec3(vec32.f_82479_, 0.0D, vec32.f_82481_);
-         Vec3 vec33 = (new Vec3(vec31.f_82479_ - vec3.f_82479_, 0.0D, vec31.f_82481_ - vec3.f_82481_)).m_82541_().m_82524_(((float)Math.PI / 2F));
+         vec32 = new Vec3(vec32.m_7096_(), 0.0D, vec32.m_7094_());
+         Vec3 vec33 = (new Vec3(vec31.m_7096_() - vec3.m_7096_(), 0.0D, vec31.m_7094_() - vec3.m_7094_())).m_82541_().m_82524_(((float)Math.PI / 2F));
          double d1 = vec32.m_82526_(vec33);
          this.f_102702_.f_104200_ = Mth.m_14116_((float)Math.abs(d1)) * 2.0F * (float)Math.signum(d1);
       }
