--- a/net/minecraft/client/player/LocalPlayer.java
+++ b/net/minecraft/client/player/LocalPlayer.java
@@ -103,9 +_,9 @@
    private final ClientRecipeBook f_108592_;
    private final List<AmbientSoundHandler> f_108593_ = Lists.newArrayList();
    private int f_108594_ = 0;
-   private double f_108595_;
-   private double f_108596_;
-   private double f_108597_;
+   private me.alphamode.mclong.math.BigDecimal f_108595_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_108596_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_108597_ = me.alphamode.mclong.math.BigDecimal.ZERO;
    private float f_108598_;
    private float f_108599_;
    private boolean f_108600_;
@@ -184,7 +_,7 @@
    }
 
    public void m_8119_() {
-      if (this.m_9236_().m_151577_(this.m_146903_(), this.m_146907_())) {
+      if (this.m_9236_().hasChunkAt(this.getBigBlockX(), this.getBigBlockZ())) {
          super.m_8119_();
          if (this.m_20159_()) {
             this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.Rot(this.m_146908_(), this.m_146909_(), this.m_20096_()));
@@ -225,9 +_,9 @@
       }
 
       if (this.m_108636_()) {
-         double d4 = this.m_20185_() - this.f_108595_;
-         double d0 = this.m_20186_() - this.f_108596_;
-         double d1 = this.m_20189_() - this.f_108597_;
+         double d4 = this.getBigX().subtract(this.f_108595_).doubleValue();
+         double d0 = this.getBigY().subtract(this.f_108596_).doubleValue();
+         double d1 = this.getBigZ().subtract(this.f_108597_).doubleValue();
          double d2 = (double)(this.m_146908_() - this.f_108598_);
          double d3 = (double)(this.m_146909_() - this.f_108599_);
          ++this.f_108604_;
@@ -235,12 +_,12 @@
          boolean flag2 = d2 != 0.0D || d3 != 0.0D;
          if (this.m_20159_()) {
             Vec3 vec3 = this.m_20184_();
-            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.PosRot(vec3.f_82479_, -999.0D, vec3.f_82481_, this.m_146908_(), this.m_146909_(), this.m_20096_()));
+            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.PosRot(vec3.getX(), me.alphamode.mclong.math.BigDecimal.valueOf(-999.0D), vec3.getZ(), this.m_146908_(), this.m_146909_(), this.m_20096_()));
             flag1 = false;
          } else if (flag1 && flag2) {
-            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.PosRot(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_146908_(), this.m_146909_(), this.m_20096_()));
+            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.PosRot(this.getBigX(), this.getBigY(), this.getBigZ(), this.m_146908_(), this.m_146909_(), this.m_20096_()));
          } else if (flag1) {
-            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.Pos(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_20096_()));
+            this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.Pos(this.getBigX(), this.getBigY(), this.getBigZ(), this.m_20096_()));
          } else if (flag2) {
             this.f_108617_.m_104955_(new ServerboundMovePlayerPacket.Rot(this.m_146908_(), this.m_146909_(), this.m_20096_()));
          } else if (this.f_108600_ != this.m_20096_()) {
@@ -248,9 +_,9 @@
          }
 
          if (flag1) {
-            this.f_108595_ = this.m_20185_();
-            this.f_108596_ = this.m_20186_();
-            this.f_108597_ = this.m_20189_();
+            this.f_108595_ = this.getBigX();
+            this.f_108596_ = this.getBigY();
+            this.f_108597_ = this.getBigZ();
             this.f_108604_ = 0;
          }
 
@@ -395,19 +_,19 @@
       this.f_108619_.m_240442_().m_240494_(p_108696_, p_108697_);
    }
 
-   private void m_108704_(double p_108705_, double p_108706_) {
-      BlockPos blockpos = BlockPos.m_274561_(p_108705_, this.m_20186_(), p_108706_);
+   private void moveTowardsClosestSpace(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pZ) {
+      BlockPos blockpos = BlockPos.containing(pX, this.getBigY(), pZ);
       if (this.m_108746_(blockpos)) {
-         double d0 = p_108705_ - (double)blockpos.m_123341_();
-         double d1 = p_108706_ - (double)blockpos.m_123343_();
+         me.alphamode.mclong.math.BigDecimal d0 = pX .subtract(blockpos.getBigDecX());
+         me.alphamode.mclong.math.BigDecimal d1 = pZ .subtract(blockpos.getBigDecZ());
          Direction direction = null;
-         double d2 = Double.MAX_VALUE;
+         me.alphamode.mclong.math.BigDecimal d2 = me.alphamode.mclong.math.BigDecimal.MAX_VALUE;
          Direction[] adirection = new Direction[]{Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH};
 
          for(Direction direction1 : adirection) {
-            double d3 = direction1.m_122434_().m_6150_(d0, 0.0D, d1);
-            double d4 = direction1.m_122421_() == Direction.AxisDirection.POSITIVE ? 1.0D - d3 : d3;
-            if (d4 < d2 && !this.m_108746_(blockpos.m_121945_(direction1))) {
+            me.alphamode.mclong.math.BigDecimal d3 = direction1.m_122434_().choose(d0, me.alphamode.mclong.math.BigDecimal.ZERO, d1);
+            me.alphamode.mclong.math.BigDecimal d4 = direction1.m_122421_() == Direction.AxisDirection.POSITIVE ? me.alphamode.mclong.math.BigDecimal.ONE.subtract(d3) : d3;
+            if (d4.compareTo(d2) < 0 && !this.m_108746_(blockpos.relative(direction1))) {
                d2 = d4;
                direction = direction1;
             }
@@ -416,9 +_,9 @@
          if (direction != null) {
             Vec3 vec3 = this.m_20184_();
             if (direction.m_122434_() == Direction.Axis.X) {
-               this.m_20334_(0.1D * (double)direction.m_122429_(), vec3.f_82480_, vec3.f_82481_);
+               this.m_20334_(0.1D * (double)direction.m_122429_(), vec3.m_7098_(), vec3.m_7094_());
             } else {
-               this.m_20334_(vec3.f_82479_, vec3.f_82480_, 0.1D * (double)direction.m_122431_());
+               this.m_20334_(vec3.m_7096_(), vec3.m_7098_(), 0.1D * (double)direction.m_122431_());
             }
          }
 
@@ -427,7 +_,7 @@
 
    private boolean m_108746_(BlockPos p_108747_) {
       AABB aabb = this.m_20191_();
-      AABB aabb1 = (new AABB((double)p_108747_.m_123341_(), aabb.f_82289_, (double)p_108747_.m_123343_(), (double)p_108747_.m_123341_() + 1.0D, aabb.f_82292_, (double)p_108747_.m_123343_() + 1.0D)).m_82406_(1.0E-7D);
+      AABB aabb1 = (new AABB(p_108747_.getBigDecX(), aabb.f_82289_, p_108747_.getBigDecZ(), p_108747_.getBigDecX().add(me.alphamode.mclong.math.BigDecimal.ONE), aabb.f_82292_, p_108747_.getBigDecZ().add(me.alphamode.mclong.math.BigDecimal.ONE))).m_82406_(1.0E-7D);
       return this.m_9236_().m_186437_(this, aabb1);
    }
 
@@ -459,11 +_,11 @@
    }
 
    public void m_5496_(SoundEvent p_108651_, float p_108652_, float p_108653_) {
-      this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), p_108651_, this.m_5720_(), p_108652_, p_108653_, false);
+      this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), p_108651_, this.m_5720_(), p_108652_, p_108653_, false);
    }
 
    public void m_6330_(SoundEvent p_108655_, SoundSource p_108656_, float p_108657_, float p_108658_) {
-      this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), p_108655_, p_108656_, p_108657_, p_108658_, false);
+      this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), p_108655_, p_108656_, p_108657_, p_108658_, false);
    }
 
    public boolean m_21515_() {
@@ -603,8 +_,8 @@
    public void m_172530_() {
       this.m_20124_(Pose.STANDING);
       if (this.m_9236_() != null) {
-         for(double d0 = this.m_20186_(); d0 > (double)this.m_9236_().m_141937_() && d0 < (double)this.m_9236_().m_151558_(); ++d0) {
-            this.m_6034_(this.m_20185_(), d0, this.m_20189_());
+         for(double d0 = this.m_20186_(); d0 > (double)this.m_9236_().getMinBuildHeight() && d0 < (double)this.m_9236_().getMaxBuildHeight(); ++d0) {
+            this.setPos(this.getBigX(), me.alphamode.mclong.math.BigDecimal.valueOf(d0), this.getBigZ());
             if (this.m_9236_().m_45786_(this)) {
                break;
             }
@@ -648,10 +_,10 @@
       }
 
       if (!this.f_19794_) {
-         this.m_108704_(this.m_20185_() - (double)this.m_20205_() * 0.35D, this.m_20189_() + (double)this.m_20205_() * 0.35D);
-         this.m_108704_(this.m_20185_() - (double)this.m_20205_() * 0.35D, this.m_20189_() - (double)this.m_20205_() * 0.35D);
-         this.m_108704_(this.m_20185_() + (double)this.m_20205_() * 0.35D, this.m_20189_() - (double)this.m_20205_() * 0.35D);
-         this.m_108704_(this.m_20185_() + (double)this.m_20205_() * 0.35D, this.m_20189_() + (double)this.m_20205_() * 0.35D);
+         this.moveTowardsClosestSpace(this.getBigX().subtract(this.m_20205_() * 0.35D), this.getBigZ().add(this.m_20205_() * 0.35D));
+         this.moveTowardsClosestSpace(this.getBigX().subtract(this.m_20205_() * 0.35D), this.getBigZ().subtract(this.m_20205_() * 0.35D));
+         this.moveTowardsClosestSpace(this.getBigX().add(this.m_20205_() * 0.35D), this.getBigZ().subtract(this.m_20205_() * 0.35D));
+         this.moveTowardsClosestSpace(this.getBigX().add(this.m_20205_() * 0.35D), this.getBigZ().add(this.m_20205_() * 0.35D));
       }
 
       if (flag1) {
@@ -838,10 +_,10 @@
    }
 
    public void m_6478_(MoverType p_108670_, Vec3 p_108671_) {
-      double d0 = this.m_20185_();
-      double d1 = this.m_20189_();
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigZ();
       super.m_6478_(p_108670_, p_108671_);
-      this.m_108743_((float)(this.m_20185_() - d0), (float)(this.m_20189_() - d1));
+      this.m_108743_((float)(this.getBigX().subtract(d0)).floatValue(), (float)(this.getBigZ().subtract(d1)).floatValue());
    }
 
    public boolean m_108638_() {
@@ -861,7 +_,7 @@
             float f3 = f * vec2.f_82471_;
             float f4 = Mth.m_14031_(this.m_146908_() * ((float)Math.PI / 180F));
             float f5 = Mth.m_14089_(this.m_146908_() * ((float)Math.PI / 180F));
-            vec32 = new Vec3((double)(f2 * f5 - f3 * f4), vec32.f_82480_, (double)(f3 * f5 + f2 * f4));
+            vec32 = new Vec3((double)(f2 * f5 - f3 * f4), vec32.m_7098_(), (double)(f3 * f5 + f2 * f4));
             f1 = (float)vec32.m_82556_();
             if (f1 <= 0.001F) {
                return;
@@ -871,13 +_,13 @@
          float f12 = Mth.m_264536_(f1);
          Vec3 vec312 = vec32.m_82490_((double)f12);
          Vec3 vec313 = this.m_20156_();
-         float f13 = (float)(vec313.f_82479_ * vec312.f_82479_ + vec313.f_82481_ * vec312.f_82481_);
+         float f13 = (float)(vec313.m_7096_() * vec312.m_7096_() + vec313.m_7094_() * vec312.m_7094_());
          if (!(f13 < -0.15F)) {
             CollisionContext collisioncontext = CollisionContext.m_82750_(this);
-            BlockPos blockpos = BlockPos.m_274561_(this.m_20185_(), this.m_20191_().f_82292_, this.m_20189_());
+            BlockPos blockpos = BlockPos.containing(this.getBigX(), this.m_20191_().f_82292_, this.getBigZ());
             BlockState blockstate = this.m_9236_().m_8055_(blockpos);
             if (blockstate.m_60742_(this.m_9236_(), blockpos, collisioncontext).m_83281_()) {
-               blockpos = blockpos.m_7494_();
+               blockpos = blockpos.above();
                BlockState blockstate1 = this.m_9236_().m_8055_(blockpos);
                if (blockstate1.m_60742_(this.m_9236_(), blockpos, collisioncontext).m_83281_()) {
                   float f6 = 7.0F;
@@ -908,23 +_,23 @@
                   while(iterator.hasNext()) {
                      AABB aabb1 = iterator.next();
                      if (aabb1.m_82335_(vec37, vec38) || aabb1.m_82335_(vec39, vec310)) {
-                        f11 = (float)aabb1.f_82292_;
+                        f11 = aabb1.f_82292_.floatValue();
                         Vec3 vec311 = aabb1.m_82399_();
-                        BlockPos blockpos1 = BlockPos.m_274446_(vec311);
+                        BlockPos blockpos1 = BlockPos.containing(vec311);
 
                         for(int i = 1; (float)i < f7; ++i) {
-                           BlockPos blockpos2 = blockpos1.m_6630_(i);
+                           BlockPos blockpos2 = blockpos1.above(i);
                            BlockState blockstate2 = this.m_9236_().m_8055_(blockpos2);
                            VoxelShape voxelshape;
                            if (!(voxelshape = blockstate2.m_60742_(this.m_9236_(), blockpos2, collisioncontext)).m_83281_()) {
-                              f11 = (float)voxelshape.m_83297_(Direction.Axis.Y) + (float)blockpos2.m_123342_();
+                              f11 = (float)voxelshape.m_83297_(Direction.Axis.Y) + (float)blockpos2.getY();
                               if ((double)f11 - this.m_20186_() > (double)f7) {
                                  return;
                               }
                            }
 
                            if (i > 1) {
-                              blockpos = blockpos.m_7494_();
+                              blockpos = blockpos.above();
                               BlockState blockstate3 = this.m_9236_().m_8055_(blockpos);
                               if (!blockstate3.m_60742_(this.m_9236_(), blockpos, collisioncontext).m_83281_()) {
                                  return;
@@ -954,9 +_,9 @@
       double d2 = (double)this.f_20900_ * d1 - (double)this.f_20902_ * d0;
       double d3 = (double)this.f_20902_ * d1 + (double)this.f_20900_ * d0;
       double d4 = Mth.m_144952_(d2) + Mth.m_144952_(d3);
-      double d5 = Mth.m_144952_(p_197411_.f_82479_) + Mth.m_144952_(p_197411_.f_82481_);
+      double d5 = Mth.m_144952_(p_197411_.m_7096_()) + Mth.m_144952_(p_197411_.m_7094_());
       if (!(d4 < (double)1.0E-5F) && !(d5 < (double)1.0E-5F)) {
-         double d6 = d2 * p_197411_.f_82479_ + d3 * p_197411_.f_82481_;
+         double d6 = d2 * p_197411_.m_7096_() + d3 * p_197411_.m_7094_();
          double d7 = Math.acos(d6 / Math.sqrt(d4 * d5));
          return d7 < (double)0.13962634F;
       } else {
@@ -1024,12 +_,12 @@
          return this.f_36076_;
       } else {
          if (!flag && flag1) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12537_, SoundSource.AMBIENT, 1.0F, 1.0F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12537_, SoundSource.AMBIENT, 1.0F, 1.0F, false);
             this.f_108619_.m_91106_().m_120367_(new UnderwaterAmbientSoundInstances.UnderwaterAmbientSoundInstance(this));
          }
 
          if (flag && !flag1) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12590_, SoundSource.AMBIENT, 1.0F, 1.0F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12590_, SoundSource.AMBIENT, 1.0F, 1.0F, false);
          }
 
          return this.f_36076_;
