--- a/net/minecraft/client/player/RemotePlayer.java
+++ b/net/minecraft/client/player/RemotePlayer.java
@@ -58,7 +_,7 @@
       }
 
       if (this.f_271363_ > 0) {
-         this.m_246865_(new Vec3((this.f_271343_.f_82479_ - this.m_20184_().f_82479_) / (double)this.f_271363_, (this.f_271343_.f_82480_ - this.m_20184_().f_82480_) / (double)this.f_271363_, (this.f_271343_.f_82481_ - this.m_20184_().f_82481_) / (double)this.f_271363_));
+         this.m_246865_(new Vec3((this.f_271343_.m_7096_() - this.m_20184_().m_7096_()) / (double)this.f_271363_, (this.f_271343_.m_7098_() - this.m_20184_().m_7098_()) / (double)this.f_271363_, (this.f_271343_.m_7094_() - this.m_20184_().m_7094_()) / (double)this.f_271363_));
          --this.f_271363_;
       }
 
