--- a/net/minecraft/client/KeyboardHandler.java
+++ b/net/minecraft/client/KeyboardHandler.java
@@ -278,7 +_,7 @@
          stringbuilder.append((Object)p_90902_);
       }
 
-      String s = String.format(Locale.ROOT, "/setblock %d %d %d %s", p_90901_.m_123341_(), p_90901_.m_123342_(), p_90901_.m_123343_(), stringbuilder);
+      String s = String.format(Locale.ROOT, "/setblock %d %d %d %s", p_90901_.getX(), p_90901_.getY(), p_90901_.getZ(), stringbuilder);
       this.m_90911_(s);
    }
 
