--- a/net/minecraft/client/KeyboardHandler.java
+++ b/net/minecraft/client/KeyboardHandler.java
@@ -279,7 +_,7 @@
          stringbuilder.append((Object)p_90902_);
       }
 
-      String s = String.format(Locale.ROOT, "/setblock %d %d %d %s", p_90901_.m_123341_(), p_90901_.m_123342_(), p_90901_.m_123343_(), stringbuilder);
+      String s = String.format(Locale.ROOT, "/setblock %d %d %d %s", p_90901_.getX(), p_90901_.getY(), p_90901_.getZ(), stringbuilder);
       this.m_90911_(s);
    }
 
@@ -290,9 +_,9 @@
          p_90925_.m_128473_("Pos");
          p_90925_.m_128473_("Dimension");
          String s1 = NbtUtils.m_178061_(p_90925_).getString();
-         s = String.format(Locale.ROOT, "/summon %s %.2f %.2f %.2f %s", p_90923_.toString(), p_90924_.f_82479_, p_90924_.f_82480_, p_90924_.f_82481_, s1);
+         s = String.format(Locale.ROOT, "/summon %s %.2f %.2f %.2f %s", p_90923_.toString(), p_90924_.getX(), p_90924_.getY(), p_90924_.getZ(), s1);
       } else {
-         s = String.format(Locale.ROOT, "/summon %s %.2f %.2f %.2f", p_90923_.toString(), p_90924_.f_82479_, p_90924_.f_82480_, p_90924_.f_82481_);
+         s = String.format(Locale.ROOT, "/summon %s %.2f %.2f %.2f", p_90923_.toString(), p_90924_.getX(), p_90924_.getY(), p_90924_.getZ());
       }
 
       this.m_90911_(s);
