--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -1799,7 +_,7 @@
 
          this.f_91026_.m_6182_("animateTick");
          if (!this.f_91012_ && this.f_91073_ != null) {
-            this.f_91073_.m_104784_(this.f_91074_.m_146903_(), this.f_91074_.m_146904_(), this.f_91074_.m_146907_());
+            this.f_91073_.animateTick(this.f_91074_.m_20183_().getBigX(), this.f_91074_.m_20183_().getBigY(), this.f_91074_.m_20183_().getBigZ());
          }
 
          this.f_91026_.m_6182_("particles");
@@ -1892,6 +_,16 @@
 
       if (this.f_91080_ == null && this.f_91081_ == null && this.f_91066_.f_92100_.m_90859_()) {
          this.m_91326_("/");
+      }
+
+      if (me.alphamode.mclong.core.MCLongKeyBinds.toggleBigDecimal.m_90859_()) {
+         me.alphamode.mclong.core.BigConstants.BIG_MODE = !me.alphamode.mclong.core.BigConstants.BIG_MODE;
+         f_91074_.m_213846_(Component.m_237113_("Toggled Big Decimal Mode: " + me.alphamode.mclong.core.BigConstants.BIG_MODE));
+      }
+
+      if (me.alphamode.mclong.core.MCLongKeyBinds.toggleBigInteger.m_90859_()) {
+//         me.alphamode.mclong.core.BigConstants.Ints.BIG_MODE = !me.alphamode.mclong.core.BigConstants.Ints.BIG_MODE;
+         f_91074_.m_213846_(Component.m_237113_("Toggled Big Integer Mode: " + me.alphamode.mclong.core.BigConstants.Ints.BIG_MODE));
       }
 
       boolean flag2 = false;
