--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -1768,7 +_,7 @@
 
          this.f_91026_.m_6182_("animateTick");
          if (!this.f_91012_ && this.f_91073_ != null) {
-            this.f_91073_.m_104784_(this.f_91074_.m_146903_(), this.f_91074_.m_146904_(), this.f_91074_.m_146907_());
+            this.f_91073_.animateTick(this.f_91074_.getBlockX(), this.f_91074_.getBlockY(), this.f_91074_.getBlockZ());
          }
 
          this.f_91026_.m_6182_("particles");
