--- a/net/minecraft/world/level/levelgen/feature/rootplacers/RootPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/rootplacers/RootPlacer.java
@@ -53,7 +_,7 @@
          p_225875_.accept(p_225877_, this.m_225870_(p_225874_, p_225877_, this.f_225861_.m_213972_(p_225876_, p_225877_)));
          if (this.f_225862_.isPresent()) {
             AboveRootPlacement aboverootplacement = this.f_225862_.get();
-            BlockPos blockpos = p_225877_.m_7494_();
+            BlockPos blockpos = p_225877_.above();
             if (p_225876_.m_188501_() < aboverootplacement.f_225755_() && p_225874_.m_7433_(blockpos, BlockBehaviour.BlockStateBase::m_60795_)) {
                p_225875_.accept(blockpos, this.m_225870_(p_225874_, blockpos, aboverootplacement.f_225754_().m_213972_(p_225876_, blockpos)));
             }
@@ -74,6 +_,6 @@
    }
 
    public BlockPos m_225891_(BlockPos p_225892_, RandomSource p_225893_) {
-      return p_225892_.m_6630_(this.f_225860_.m_214085_(p_225893_));
+      return p_225892_.above(this.f_225860_.m_214085_(p_225893_));
    }
 }
