--- a/net/minecraft/commands/arguments/EntityAnchorArgument.java
+++ b/net/minecraft/commands/arguments/EntityAnchorArgument.java
@@ -60,7 +_,7 @@
          return p_90389_;
       }),
       EYES("eyes", (p_90382_, p_90383_) -> {
-         return new Vec3(p_90382_.f_82479_, p_90382_.f_82480_ + (double)p_90383_.m_20192_(), p_90382_.f_82481_);
+         return new Vec3(p_90382_.m_7096_(), p_90382_.m_7098_() + (double)p_90383_.m_20192_(), p_90382_.m_7094_());
       });
 
       static final Map<String, EntityAnchorArgument.Anchor> f_90366_ = Util.m_137469_(Maps.newHashMap(), (p_90387_) -> {
