--- a/net/minecraft/client/color/block/BlockColors.java
+++ b/net/minecraft/client/color/block/BlockColors.java
@@ -34,7 +_,7 @@
    public static BlockColors m_92574_() {
       BlockColors blockcolors = new BlockColors();
       blockcolors.m_92589_((p_276233_, p_276234_, p_276235_, p_276236_) -> {
-         return p_276234_ != null && p_276235_ != null ? BiomeColors.m_108793_(p_276234_, p_276233_.m_61143_(DoublePlantBlock.f_52858_) == DoubleBlockHalf.UPPER ? p_276235_.m_7495_() : p_276235_) : GrassColor.m_276205_();
+         return p_276234_ != null && p_276235_ != null ? BiomeColors.m_108793_(p_276234_, p_276233_.m_61143_(DoublePlantBlock.f_52858_) == DoubleBlockHalf.UPPER ? p_276235_.below() : p_276235_) : GrassColor.m_276205_();
       }, Blocks.f_50360_, Blocks.f_50359_);
       blockcolors.m_92586_(DoublePlantBlock.f_52858_, Blocks.f_50360_, Blocks.f_50359_);
       blockcolors.m_92589_((p_276237_, p_276238_, p_276239_, p_276240_) -> {
