--- a/net/minecraft/client/color/block/BlockTintCache.java
+++ b/net/minecraft/client/color/block/BlockTintCache.java
@@ -17,7 +_,7 @@
 public class BlockTintCache {
    private static final int f_168641_ = 256;
    private final ThreadLocal<BlockTintCache.LatestCacheInfo> f_92650_ = ThreadLocal.withInitial(BlockTintCache.LatestCacheInfo::new);
-   private final Long2ObjectLinkedOpenHashMap<BlockTintCache.CacheData> f_92651_ = new Long2ObjectLinkedOpenHashMap<>(256, 0.25F);
+   private final it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, BlockTintCache.CacheData> f_92651_ = new it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<>(256, 0.25F);
    private final ReentrantReadWriteLock f_92652_ = new ReentrantReadWriteLock();
    private final ToIntFunction<BlockPos> f_193809_;
 
@@ -26,36 +_,36 @@
    }
 
    public int m_193812_(BlockPos p_193813_) {
-      int i = SectionPos.m_123171_(p_193813_.m_123341_());
-      int j = SectionPos.m_123171_(p_193813_.m_123343_());
+      me.alphamode.mclong.math.BigInteger i = SectionPos.blockToSectionCoord(p_193813_.getBigX());
+      me.alphamode.mclong.math.BigInteger j = SectionPos.blockToSectionCoord(p_193813_.getBigZ());
       BlockTintCache.LatestCacheInfo blocktintcache$latestcacheinfo = this.f_92650_.get();
-      if (blocktintcache$latestcacheinfo.f_92665_ != i || blocktintcache$latestcacheinfo.f_92666_ != j || blocktintcache$latestcacheinfo.f_92667_ == null || blocktintcache$latestcacheinfo.f_92667_.m_262488_()) {
+      if (!blocktintcache$latestcacheinfo.f_92665_.equals(i) || !blocktintcache$latestcacheinfo.f_92666_.equals(j) || blocktintcache$latestcacheinfo.f_92667_ == null || blocktintcache$latestcacheinfo.f_92667_.m_262488_()) {
          blocktintcache$latestcacheinfo.f_92665_ = i;
          blocktintcache$latestcacheinfo.f_92666_ = j;
-         blocktintcache$latestcacheinfo.f_92667_ = this.m_193814_(i, j);
+         blocktintcache$latestcacheinfo.f_92667_ = this.findOrCreateChunkCache(i, j);
       }
 
-      int[] aint = blocktintcache$latestcacheinfo.f_92667_.m_193823_(p_193813_.m_123342_());
-      int k = p_193813_.m_123341_() & 15;
-      int l = p_193813_.m_123343_() & 15;
-      int i1 = l << 4 | k;
-      int j1 = aint[i1];
+      int[] aint = blocktintcache$latestcacheinfo.f_92667_.getLayer(p_193813_.getY());
+      long k = p_193813_.getBigX().and(me.alphamode.mclong.core.BigConstants.Ints.FIFTEEN).longValue();
+      long l = p_193813_.getBigZ().and(me.alphamode.mclong.core.BigConstants.Ints.FIFTEEN).longValue();
+      long i1 = l << 4 | k;
+      int j1 = aint[(int) i1];
       if (j1 != -1) {
          return j1;
       } else {
          int k1 = this.f_193809_.applyAsInt(p_193813_);
-         aint[i1] = k1;
+         aint[(int) i1] = k1;
          return k1;
       }
    }
 
-   public void m_92655_(int p_92656_, int p_92657_) {
+   public void invalidateForChunk(me.alphamode.mclong.math.BigInteger p_92656_, me.alphamode.mclong.math.BigInteger p_92657_) {
       try {
          this.f_92652_.writeLock().lock();
 
          for(int i = -1; i <= 1; ++i) {
             for(int j = -1; j <= 1; ++j) {
-               long k = ChunkPos.m_45589_(p_92656_ + i, p_92657_ + j);
+               ChunkPos k = new ChunkPos(p_92656_.add(i), p_92657_.add(j));
                BlockTintCache.CacheData blocktintcache$cachedata = this.f_92651_.remove(k);
                if (blocktintcache$cachedata != null) {
                   blocktintcache$cachedata.m_262378_();
@@ -79,8 +_,8 @@
 
    }
 
-   private BlockTintCache.CacheData m_193814_(int p_193815_, int p_193816_) {
-      long i = ChunkPos.m_45589_(p_193815_, p_193816_);
+   private BlockTintCache.CacheData findOrCreateChunkCache(me.alphamode.mclong.math.BigInteger p_193815_, me.alphamode.mclong.math.BigInteger p_193816_) {
+      ChunkPos i = new ChunkPos(p_193815_, p_193816_);
       this.f_92652_.readLock().lock();
 
       try {
@@ -120,12 +_,12 @@
 
    @OnlyIn(Dist.CLIENT)
    static class CacheData {
-      private final Int2ObjectArrayMap<int[]> f_193817_ = new Int2ObjectArrayMap<>(16);
+      private final it.unimi.dsi.fastutil.longs.Long2ObjectArrayMap<int[]> f_193817_ = new it.unimi.dsi.fastutil.longs.Long2ObjectArrayMap<>(16);
       private final ReentrantReadWriteLock f_193818_ = new ReentrantReadWriteLock();
       private static final int f_193819_ = Mth.m_144944_(16);
       private volatile boolean f_262289_;
 
-      public int[] m_193823_(int p_193824_) {
+      public int[] getLayer(long p_193824_) {
          this.f_193818_.readLock().lock();
 
          try {
@@ -168,8 +_,8 @@
 
    @OnlyIn(Dist.CLIENT)
    static class LatestCacheInfo {
-      public int f_92665_ = Integer.MIN_VALUE;
-      public int f_92666_ = Integer.MIN_VALUE;
+      public me.alphamode.mclong.math.BigInteger f_92665_ = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MIN_VALUE);
+      public me.alphamode.mclong.math.BigInteger f_92666_ = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MIN_VALUE);
       @Nullable
       BlockTintCache.CacheData f_92667_;
 
