--- a/net/minecraft/client/color/block/BlockColors.java
+++ b/net/minecraft/client/color/block/BlockColors.java
@@ -34,7 +_,7 @@
    public static BlockColors m_92574_() {
       BlockColors blockcolors = new BlockColors();
       blockcolors.m_92589_((p_92646_, p_92647_, p_92648_, p_92649_) -> {
-         return p_92647_ != null && p_92648_ != null ? BiomeColors.m_108793_(p_92647_, p_92646_.m_61143_(DoublePlantBlock.f_52858_) == DoubleBlockHalf.UPPER ? p_92648_.m_7495_() : p_92648_) : -1;
+         return p_92647_ != null && p_92648_ != null ? BiomeColors.m_108793_(p_92647_, p_92646_.m_61143_(DoublePlantBlock.f_52858_) == DoubleBlockHalf.UPPER ? p_92648_.below() : p_92648_) : -1;
       }, Blocks.f_50360_, Blocks.f_50359_);
       blockcolors.m_92586_(DoublePlantBlock.f_52858_, Blocks.f_50360_, Blocks.f_50359_);
       blockcolors.m_92589_((p_92641_, p_92642_, p_92643_, p_92644_) -> {
