--- a/net/minecraft/world/entity/item/FallingBlockEntity.java
+++ b/net/minecraft/world/entity/item/FallingBlockEntity.java
@@ -69,14 +_,14 @@
       this.f_19850_ = true;
       this.m_6034_(p_31954_, p_31955_, p_31956_);
       this.m_20256_(Vec3.f_82478_);
-      this.f_19854_ = p_31954_;
-      this.f_19855_ = p_31955_;
-      this.f_19856_ = p_31956_;
+      this.f_19854_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_31954_);
+      this.f_19855_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_31955_);
+      this.f_19856_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_31956_);
       this.m_31959_(this.m_20183_());
    }
 
    public static FallingBlockEntity m_201971_(Level p_201972_, BlockPos p_201973_, BlockState p_201974_) {
-      FallingBlockEntity fallingblockentity = new FallingBlockEntity(p_201972_, (double)p_201973_.m_123341_() + 0.5D, (double)p_201973_.m_123342_(), (double)p_201973_.m_123343_() + 0.5D, p_201974_.m_61138_(BlockStateProperties.f_61362_) ? p_201974_.m_61124_(BlockStateProperties.f_61362_, Boolean.valueOf(false)) : p_201974_);
+      FallingBlockEntity fallingblockentity = new FallingBlockEntity(p_201972_, (double)p_201973_.getX() + 0.5D, (double)p_201973_.getY(), (double)p_201973_.getZ() + 0.5D, p_201974_.m_61138_(BlockStateProperties.f_61362_) ? p_201974_.m_61124_(BlockStateProperties.f_61362_, Boolean.valueOf(false)) : p_201974_);
       p_201972_.m_7731_(p_201973_, p_201974_.m_60819_().m_76188_(), 3);
       p_201972_.m_7967_(fallingblockentity);
       return fallingblockentity;
@@ -131,7 +_,7 @@
             }
 
             if (!this.m_20096_() && !flag1) {
-               if (!this.m_9236_().f_46443_ && (this.f_31942_ > 100 && (blockpos.m_123342_() <= this.m_9236_().m_141937_() || blockpos.m_123342_() > this.m_9236_().m_151558_()) || this.f_31942_ > 600)) {
+               if (!this.m_9236_().f_46443_ && (this.f_31942_ > 100 && (blockpos.getY() <= this.m_9236_().getMinBuildHeight() || blockpos.getY() > this.m_9236_().getMaxBuildHeight()) || this.f_31942_ > 600)) {
                   if (this.f_31943_ && this.m_9236_().m_46469_().m_46207_(GameRules.f_46137_)) {
                      this.m_19998_(block);
                   }
@@ -144,7 +_,7 @@
                if (!blockstate.m_60713_(Blocks.f_50110_)) {
                   if (!this.f_31947_) {
                      boolean flag2 = blockstate.m_60629_(new DirectionalPlaceContext(this.m_9236_(), blockpos, Direction.DOWN, ItemStack.f_41583_, Direction.UP));
-                     boolean flag3 = FallingBlock.m_53241_(this.m_9236_().m_8055_(blockpos.m_7495_())) && (!flag || !flag1);
+                     boolean flag3 = FallingBlock.m_53241_(this.m_9236_().m_8055_(blockpos.below())) && (!flag || !flag1);
                      boolean flag4 = this.f_31946_.m_60710_(this.m_9236_(), blockpos) && !flag3;
                      if (flag2 && flag4) {
                         if (this.f_31946_.m_61138_(BlockStateProperties.f_61362_) && this.m_9236_().m_6425_(blockpos).m_76152_() == Fluids.f_76193_) {
