--- a/net/minecraft/world/entity/item/ItemEntity.java
+++ b/net/minecraft/world/entity/item/ItemEntity.java
@@ -140,7 +_,7 @@
             }
          }
 
-         boolean flag = Mth.m_14107_(this.f_19854_) != Mth.m_14107_(this.m_20185_()) || Mth.m_14107_(this.f_19855_) != Mth.m_14107_(this.m_20186_()) || Mth.m_14107_(this.f_19856_) != Mth.m_14107_(this.m_20189_());
+         boolean flag = Mth.m_14134_(this.f_19854_) != Mth.m_14134_(this.m_20185_()) || Mth.m_14134_(this.f_19855_) != Mth.m_14134_(this.m_20186_()) || Mth.m_14134_(this.f_19856_) != Mth.m_14134_(this.m_20189_());
          int i = flag ? 2 : 40;
          if (this.f_19797_ % i == 0 && !this.m_9236_().f_46443_ && this.m_32070_()) {
             this.m_32069_();
