--- a/net/minecraft/world/entity/item/PrimedTnt.java
+++ b/net/minecraft/world/entity/item/PrimedTnt.java
@@ -32,9 +_,9 @@
       double d0 = p_32079_.f_46441_.m_188500_() * (double)((float)Math.PI * 2F);
       this.m_20334_(-Math.sin(d0) * 0.02D, (double)0.2F, -Math.cos(d0) * 0.02D);
       this.m_32085_(80);
-      this.f_19854_ = p_32080_;
-      this.f_19855_ = p_32081_;
-      this.f_19856_ = p_32082_;
+      this.f_19854_ = java.math.BigDecimal.valueOf(p_32080_);
+      this.f_19855_ = java.math.BigDecimal.valueOf(p_32081_);
+      this.f_19856_ = java.math.BigDecimal.valueOf(p_32082_);
       this.f_32072_ = p_32083_;
    }
 
