--- a/net/minecraft/world/level/biome/CheckerboardColumnBiomeSource.java
+++ b/net/minecraft/world/level/biome/CheckerboardColumnBiomeSource.java
@@ -28,7 +_,7 @@
       return f_48230_;
    }
 
-   public Holder<Biome> m_203407_(int p_204248_, int p_204249_, int p_204250_, Climate.Sampler p_204251_) {
+   public Holder<Biome> getNoiseBiome(long p_204248_, long p_204249_, long p_204250_, Climate.Sampler p_204251_) {
       return this.f_48231_.m_203662_(Math.floorMod((p_204248_ >> this.f_48232_) + (p_204250_ >> this.f_48232_), this.f_48231_.m_203632_()));
    }
 }
