--- a/net/minecraft/world/level/biome/FixedBiomeSource.java
+++ b/net/minecraft/world/level/biome/FixedBiomeSource.java
@@ -30,11 +_,11 @@
       return f_48251_;
    }
 
-   public Holder<Biome> m_203407_(int p_204265_, int p_204266_, int p_204267_, Climate.Sampler p_204268_) {
+   public Holder<Biome> getNoiseBiome(long p_204265_, long p_204266_, long p_204267_, Climate.Sampler p_204268_) {
       return this.f_48252_;
    }
 
-   public Holder<Biome> m_203495_(int p_204261_, int p_204262_, int p_204263_) {
+   public Holder<Biome> getNoiseBiome(long p_204261_, long p_204262_, long p_204263_) {
       return this.f_48252_;
    }
 
