--- a/net/minecraft/world/level/biome/MultiNoiseBiomeSource.java
+++ b/net/minecraft/world/level/biome/MultiNoiseBiomeSource.java
@@ -66,8 +_,8 @@
       return this.f_48438_.isPresent() && Objects.equals(this.f_48438_.get().f_48541_(), p_187064_);
    }
 
-   public Holder<Biome> m_203407_(int p_204272_, int p_204273_, int p_204274_, Climate.Sampler p_204275_) {
-      return this.m_204269_(p_204275_.m_183445_(p_204272_, p_204273_, p_204274_));
+   public Holder<Biome> getNoiseBiome(long p_204272_, long p_204273_, long p_204274_, Climate.Sampler p_204275_) {
+      return this.m_204269_(p_204275_.sample(p_204272_, p_204273_, p_204274_));
    }
 
    @VisibleForDebug
@@ -76,10 +_,10 @@
    }
 
    public void m_207301_(List<String> p_207895_, BlockPos p_207896_, Climate.Sampler p_207897_) {
-      int i = QuartPos.m_175400_(p_207896_.m_123341_());
-      int j = QuartPos.m_175400_(p_207896_.m_123342_());
-      int k = QuartPos.m_175400_(p_207896_.m_123343_());
-      Climate.TargetPoint climate$targetpoint = p_207897_.m_183445_(i, j, k);
+      long i = QuartPos.fromBlock(p_207896_.getX());
+      long j = QuartPos.fromBlock(p_207896_.getY());
+      long k = QuartPos.fromBlock(p_207896_.getZ());
+      Climate.TargetPoint climate$targetpoint = p_207897_.sample(i, j, k);
       float f = Climate.m_186796_(climate$targetpoint.f_187005_());
       float f1 = Climate.m_186796_(climate$targetpoint.f_187006_());
       float f2 = Climate.m_186796_(climate$targetpoint.f_187003_());
