--- a/net/minecraft/world/entity/monster/Blaze.java
+++ b/net/minecraft/world/entity/monster/Blaze.java
@@ -75,7 +_,7 @@
    }
 
    public void m_8107_() {
-      if (!this.m_20096_() && this.m_20184_().f_82480_ < 0.0D) {
+      if (!this.m_20096_() && this.m_20184_().m_7098_() < 0.0D) {
          this.m_20256_(this.m_20184_().m_82542_(1.0D, 0.6D, 1.0D));
       }
 
@@ -106,7 +_,7 @@
       LivingEntity livingentity = this.m_5448_();
       if (livingentity != null && livingentity.m_20188_() > this.m_20188_() + (double)this.f_32214_ && this.m_6779_(livingentity)) {
          Vec3 vec3 = this.m_20184_();
-         this.m_20256_(this.m_20184_().m_82520_(0.0D, ((double)0.3F - vec3.f_82480_) * (double)0.3F, 0.0D));
+         this.m_20256_(this.m_20184_().m_82520_(0.0D, ((double)0.3F - vec3.m_7098_()) * (double)0.3F, 0.0D));
          this.f_19812_ = true;
       }
 
