--- a/net/minecraft/world/entity/monster/Blaze.java
+++ b/net/minecraft/world/entity/monster/Blaze.java
@@ -75,17 +_,17 @@
    }
 
    public void m_8107_() {
-      if (!this.m_20096_() && this.m_20184_().f_82480_ < 0.0D) {
+      if (!this.m_20096_() && this.m_20184_().m_7098_() < 0.0D) {
          this.m_20256_(this.m_20184_().m_82542_(1.0D, 0.6D, 1.0D));
       }
 
       if (this.m_9236_().f_46443_) {
          if (this.f_19796_.m_188503_(24) == 0 && !this.m_20067_()) {
-            this.m_9236_().m_7785_(this.m_20185_() + 0.5D, this.m_20186_() + 0.5D, this.m_20189_() + 0.5D, SoundEvents.f_11702_, this.m_5720_(), 1.0F + this.f_19796_.m_188501_(), this.f_19796_.m_188501_() * 0.7F + 0.3F, false);
+            this.m_9236_().playLocalSound(this.getBigX().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigZ().add(me.alphamode.mclong.core.BigConstants.AABB), SoundEvents.f_11702_, this.m_5720_(), 1.0F + this.f_19796_.m_188501_(), this.f_19796_.m_188501_() * 0.7F + 0.3F, false);
          }
 
          for(int i = 0; i < 2; ++i) {
-            this.m_9236_().m_7106_(ParticleTypes.f_123755_, this.m_20208_(0.5D), this.m_20187_(), this.m_20262_(0.5D), 0.0D, 0.0D, 0.0D);
+            this.m_9236_().addParticle(ParticleTypes.f_123755_, this.getRandomX(0.5D), this.getRandomY(), this.getRandomZ(0.5D), 0.0D, 0.0D, 0.0D);
          }
       }
 
@@ -106,7 +_,7 @@
       LivingEntity livingentity = this.m_5448_();
       if (livingentity != null && livingentity.m_20188_() > this.m_20188_() + (double)this.f_32214_ && this.m_6779_(livingentity)) {
          Vec3 vec3 = this.m_20184_();
-         this.m_20256_(this.m_20184_().m_82520_(0.0D, ((double)0.3F - vec3.f_82480_) * (double)0.3F, 0.0D));
+         this.m_20256_(this.m_20184_().m_82520_(0.0D, ((double)0.3F - vec3.m_7098_()) * (double)0.3F, 0.0D));
          this.f_19812_ = true;
       }
 
@@ -183,11 +_,11 @@
                   this.f_32242_.m_7327_(livingentity);
                }
 
-               this.f_32242_.m_21566_().m_6849_(livingentity.m_20185_(), livingentity.m_20186_(), livingentity.m_20189_(), 1.0D);
+               this.f_32242_.m_21566_().setWantedPosition(livingentity.getBigX(), livingentity.getBigY(), livingentity.getBigZ(), 1.0D);
             } else if (d0 < this.m_32252_() * this.m_32252_() && flag) {
-               double d1 = livingentity.m_20185_() - this.f_32242_.m_20185_();
-               double d2 = livingentity.m_20227_(0.5D) - this.f_32242_.m_20227_(0.5D);
-               double d3 = livingentity.m_20189_() - this.f_32242_.m_20189_();
+               double d1 = livingentity.getBigX().subtract(this.f_32242_.getBigX()).doubleValue();
+               double d2 = livingentity.getBigY(0.5D).subtract(this.f_32242_.getBigY(0.5D)).doubleValue();
+               double d3 = livingentity.getBigZ().subtract(this.f_32242_.getBigZ()).doubleValue();
                if (this.f_32244_ <= 0) {
                   ++this.f_32243_;
                   if (this.f_32243_ == 1) {
@@ -209,7 +_,7 @@
 
                      for(int i = 0; i < 1; ++i) {
                         SmallFireball smallfireball = new SmallFireball(this.f_32242_.m_9236_(), this.f_32242_, this.f_32242_.m_217043_().m_216328_(d1, 2.297D * d4), d2, this.f_32242_.m_217043_().m_216328_(d3, 2.297D * d4));
-                        smallfireball.m_6034_(smallfireball.m_20185_(), this.f_32242_.m_20227_(0.5D) + 0.5D, smallfireball.m_20189_());
+                        smallfireball.setPos(smallfireball.getBigX(), this.f_32242_.getBigY(0.5D).add(me.alphamode.mclong.core.BigConstants.AABB), smallfireball.getBigZ());
                         this.f_32242_.m_9236_().m_7967_(smallfireball);
                      }
                   }
@@ -217,7 +_,7 @@
 
                this.f_32242_.m_21563_().m_24960_(livingentity, 10.0F, 10.0F);
             } else if (this.f_32245_ < 5) {
-               this.f_32242_.m_21566_().m_6849_(livingentity.m_20185_(), livingentity.m_20186_(), livingentity.m_20189_(), 1.0D);
+               this.f_32242_.m_21566_().setWantedPosition(livingentity.getBigX(), livingentity.getBigY(), livingentity.getBigZ(), 1.0D);
             }
 
             super.m_8037_();
