--- a/net/minecraft/world/entity/monster/Endermite.java
+++ b/net/minecraft/world/entity/monster/Endermite.java
@@ -120,7 +_,7 @@
 
    public static boolean m_218968_(EntityType<Endermite> p_218969_, LevelAccessor p_218970_, MobSpawnType p_218971_, BlockPos p_218972_, RandomSource p_218973_) {
       if (m_219019_(p_218969_, p_218970_, p_218971_, p_218972_, p_218973_)) {
-         Player player = p_218970_.m_45924_((double)p_218972_.m_123341_() + 0.5D, (double)p_218972_.m_123342_() + 0.5D, (double)p_218972_.m_123343_() + 0.5D, 5.0D, true);
+         Player player = p_218970_.m_45924_((double)p_218972_.getX() + 0.5D, (double)p_218972_.getY() + 0.5D, (double)p_218972_.getZ() + 0.5D, 5.0D, true);
          return player == null;
       } else {
          return false;
