--- a/net/minecraft/world/entity/monster/Evoker.java
+++ b/net/minecraft/world/entity/monster/Evoker.java
@@ -172,7 +_,7 @@
          double d0 = 0.0D;
 
          do {
-            BlockPos blockpos1 = blockpos.m_7495_();
+            BlockPos blockpos1 = blockpos.below();
             BlockState blockstate = Evoker.this.m_9236_().m_8055_(blockpos1);
             if (blockstate.m_60783_(Evoker.this.m_9236_(), blockpos1, Direction.UP)) {
                if (!Evoker.this.m_9236_().m_46859_(blockpos)) {
@@ -187,11 +_,11 @@
                break;
             }
 
-            blockpos = blockpos.m_7495_();
-         } while(blockpos.m_123342_() >= Mth.m_14107_(p_32675_) - 1);
+            blockpos = blockpos.below();
+         } while(blockpos.getY() >= Mth.m_14134_(p_32675_) - 1);
 
          if (flag) {
-            Evoker.this.m_9236_().m_7967_(new EvokerFangs(Evoker.this.m_9236_(), p_32673_, (double)blockpos.m_123342_() + d0, p_32674_, p_32677_, p_32678_, Evoker.this));
+            Evoker.this.m_9236_().m_7967_(new EvokerFangs(Evoker.this.m_9236_(), p_32673_, (double)blockpos.getY() + d0, p_32674_, p_32677_, p_32678_, Evoker.this));
          }
 
       }
@@ -240,7 +_,7 @@
          ServerLevel serverlevel = (ServerLevel)Evoker.this.m_9236_();
 
          for(int i = 0; i < 3; ++i) {
-            BlockPos blockpos = Evoker.this.m_20183_().m_7918_(-2 + Evoker.this.f_19796_.m_188503_(5), 1, -2 + Evoker.this.f_19796_.m_188503_(5));
+            BlockPos blockpos = Evoker.this.m_20183_().offset(-2 + Evoker.this.f_19796_.m_188503_(5), 1, -2 + Evoker.this.f_19796_.m_188503_(5));
             Vex vex = EntityType.f_20491_.m_20615_(Evoker.this.m_9236_());
             if (vex != null) {
                vex.m_20035_(blockpos, 0.0F, 0.0F);
