--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -196,7 +_,7 @@
                   this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_7868_(), this.m_5720_(), 1.0F, 1.0F, false);
                }
 
-               this.f_32805_ = vec3.f_82480_ < 0.0D && this.m_9236_().m_46575_(this.m_20183_().m_7495_(), this);
+               this.f_32805_ = vec3.f_82480_ < 0.0D && this.m_9236_().m_46575_(this.m_20183_().below(), this);
             } else if (this.m_32854_()) {
                if (this.f_32800_ < 0.5F) {
                   this.f_32800_ = 4.0F;
@@ -294,7 +_,7 @@
    }
 
    public static boolean m_218990_(EntityType<? extends Guardian> p_218991_, LevelAccessor p_218992_, MobSpawnType p_218993_, BlockPos p_218994_, RandomSource p_218995_) {
-      return (p_218995_.m_188503_(20) == 0 || !p_218992_.m_46861_(p_218994_)) && p_218992_.m_46791_() != Difficulty.PEACEFUL && (p_218993_ == MobSpawnType.SPAWNER || p_218992_.m_6425_(p_218994_).m_205070_(FluidTags.f_13131_)) && p_218992_.m_6425_(p_218994_.m_7495_()).m_205070_(FluidTags.f_13131_);
+      return (p_218995_.m_188503_(20) == 0 || !p_218992_.m_46861_(p_218994_)) && p_218992_.m_46791_() != Difficulty.PEACEFUL && (p_218993_ == MobSpawnType.SPAWNER || p_218992_.m_6425_(p_218994_).m_205070_(FluidTags.f_13131_)) && p_218992_.m_6425_(p_218994_.below()).m_205070_(FluidTags.f_13131_);
    }
 
    public boolean m_6469_(DamageSource p_32820_, float p_32821_) {
