--- a/net/minecraft/world/entity/monster/Illusioner.java
+++ b/net/minecraft/world/entity/monster/Illusioner.java
@@ -119,10 +_,10 @@
             }
 
             for(int l = 0; l < 16; ++l) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123796_, this.m_20208_(0.5D), this.m_20187_(), this.m_20246_(0.5D), 0.0D, 0.0D, 0.0D);
+               this.m_9236_().addParticle(ParticleTypes.f_123796_, this.getRandomX(0.5D), this.getRandomY(), this.getBigZ(0.5D), 0.0D, 0.0D, 0.0D);
             }
 
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12052_, this.m_5720_(), 1.0F, 1.0F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12052_, this.m_5720_(), 1.0F, 1.0F, false);
          }
       }
 
