--- a/net/minecraft/world/entity/monster/Slime.java
+++ b/net/minecraft/world/entity/monster/Slime.java
@@ -129,7 +_,7 @@
             float f1 = this.f_19796_.m_188501_() * 0.5F + 0.5F;
             float f2 = Mth.m_14031_(f) * (float)i * 0.5F * f1;
             float f3 = Mth.m_14089_(f) * (float)i * 0.5F * f1;
-            this.m_9236_().m_7106_(this.m_6300_(), this.m_20185_() + (double)f2, this.m_20186_(), this.m_20189_() + (double)f3, 0.0D, 0.0D, 0.0D);
+            this.m_9236_().addParticle(this.m_6300_(), this.getBigX().add(f2), this.getBigY(), this.getBigZ().add(f3), 0.0D, 0.0D, 0.0D);
          }
 
          this.m_5496_(this.m_7905_(), this.m_6121_(), ((this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F + 1.0F) / 0.8F);
@@ -151,11 +_,11 @@
    }
 
    public void m_6210_() {
-      double d0 = this.m_20185_();
-      double d1 = this.m_20186_();
-      double d2 = this.m_20189_();
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigY();
+      me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ();
       super.m_6210_();
-      this.m_6034_(d0, d1, d2);
+      this.setPos(d0, d1, d2);
    }
 
    public void m_7350_(EntityDataAccessor<?> p_33609_) {
@@ -197,7 +_,7 @@
                slime.m_21557_(flag);
                slime.m_20331_(this.m_20147_());
                slime.m_7839_(j, true);
-               slime.m_7678_(this.m_20185_() + (double)f1, this.m_20186_() + 0.5D, this.m_20189_() + (double)f2, this.f_19796_.m_188501_() * 360.0F, 0.0F);
+               slime.moveTo(this.getBigX().add((double)f1), this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigZ().add((double)f2), this.f_19796_.m_188501_() * 360.0F, 0.0F);
                this.m_9236_().m_7967_(slime);
             }
          }
@@ -258,7 +_,7 @@
 
    public static boolean m_219112_(EntityType<Slime> p_219113_, LevelAccessor p_219114_, MobSpawnType p_219115_, BlockPos p_219116_, RandomSource p_219117_) {
       if (p_219114_.m_46791_() != Difficulty.PEACEFUL) {
-         if (p_219114_.m_204166_(p_219116_).m_203656_(BiomeTags.f_215815_) && p_219116_.m_123342_() > 50 && p_219116_.m_123342_() < 70 && p_219117_.m_188501_() < 0.5F && p_219117_.m_188501_() < p_219114_.m_46940_() && p_219114_.m_46803_(p_219116_) <= p_219117_.m_188503_(8)) {
+         if (p_219114_.m_204166_(p_219116_).m_203656_(BiomeTags.f_215815_) && p_219116_.getY() > 50 && p_219116_.getY() < 70 && p_219117_.m_188501_() < 0.5F && p_219117_.m_188501_() < p_219114_.m_46940_() && p_219114_.m_46803_(p_219116_) <= p_219117_.m_188503_(8)) {
             return m_217057_(p_219113_, p_219114_, p_219115_, p_219116_, p_219117_);
          }
 
@@ -267,8 +_,8 @@
          }
 
          ChunkPos chunkpos = new ChunkPos(p_219116_);
-         boolean flag = WorldgenRandom.m_224681_(chunkpos.f_45578_, chunkpos.f_45579_, ((WorldGenLevel)p_219114_).m_7328_(), 987234911L).m_188503_(10) == 0;
-         if (p_219117_.m_188503_(10) == 0 && flag && p_219116_.m_123342_() < 40) {
+         boolean flag = WorldgenRandom.seedSlimeChunk(chunkpos.x(), chunkpos.z(), ((WorldGenLevel)p_219114_).m_7328_(), 987234911L).m_188503_(10) == 0;
+         if (p_219117_.m_188503_(10) == 0 && flag && p_219116_.getY() < 40) {
             return m_217057_(p_219113_, p_219114_, p_219115_, p_219116_, p_219117_);
          }
       }
@@ -290,7 +_,7 @@
 
    protected void m_6135_() {
       Vec3 vec3 = this.m_20184_();
-      this.m_20334_(vec3.f_82479_, (double)this.m_6118_(), vec3.f_82481_);
+      this.m_20334_(vec3.m_7096_(), (double)this.m_6118_(), vec3.m_7094_());
       this.f_19812_ = true;
    }
 
