--- a/net/minecraft/world/entity/monster/SpellcasterIllager.java
+++ b/net/minecraft/world/entity/monster/SpellcasterIllager.java
@@ -83,8 +_,8 @@
          float f = this.f_20883_ * ((float)Math.PI / 180F) + Mth.m_14089_((float)this.f_19797_ * 0.6662F) * 0.25F;
          float f1 = Mth.m_14089_(f);
          float f2 = Mth.m_14031_(f);
-         this.m_9236_().m_7106_(ParticleTypes.f_123811_, this.m_20185_() + (double)f1 * 0.6D, this.m_20186_() + 1.8D, this.m_20189_() + (double)f2 * 0.6D, d0, d1, d2);
-         this.m_9236_().m_7106_(ParticleTypes.f_123811_, this.m_20185_() - (double)f1 * 0.6D, this.m_20186_() + 1.8D, this.m_20189_() - (double)f2 * 0.6D, d0, d1, d2);
+         this.m_9236_().addParticle(ParticleTypes.f_123811_, this.getBigX().add((double)f1 * 0.6D), this.getBigY().add(1.8D), this.getBigZ().add((double)f2 * 0.6D), d0, d1, d2);
+         this.m_9236_().addParticle(ParticleTypes.f_123811_, this.getBigX().subtract((double)f1 * 0.6D), this.getBigY().add(1.8D), this.getBigZ().subtract((double)f2 * 0.6D), d0, d1, d2);
       }
 
    }
