--- a/net/minecraft/world/entity/monster/Spider.java
+++ b/net/minecraft/world/entity/monster/Spider.java
@@ -144,7 +_,7 @@
       if (randomsource.m_188503_(100) == 0) {
          Skeleton skeleton = EntityType.f_20524_.m_20615_(this.m_9236_());
          if (skeleton != null) {
-            skeleton.m_7678_(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_146908_(), 0.0F);
+            skeleton.moveTo(this.getBigX(), this.getBigY(), this.getBigZ(), this.m_146908_(), 0.0F);
             skeleton.m_6518_(p_33790_, p_33791_, p_33792_, (SpawnGroupData)null, (CompoundTag)null);
             skeleton.m_20329_(this);
          }
