--- a/net/minecraft/world/entity/monster/Stray.java
+++ b/net/minecraft/world/entity/monster/Stray.java
@@ -25,10 +_,10 @@
       BlockPos blockpos = p_219124_;
 
       do {
-         blockpos = blockpos.m_7494_();
+         blockpos = blockpos.above();
       } while(p_219122_.m_8055_(blockpos).m_60713_(Blocks.f_152499_));
 
-      return m_219013_(p_219121_, p_219122_, p_219123_, p_219124_, p_219125_) && (p_219123_ == MobSpawnType.SPAWNER || p_219122_.m_45527_(blockpos.m_7495_()));
+      return m_219013_(p_219121_, p_219122_, p_219123_, p_219124_, p_219125_) && (p_219123_ == MobSpawnType.SPAWNER || p_219122_.m_45527_(blockpos.below()));
    }
 
    protected SoundEvent m_7515_() {
