--- a/net/minecraft/world/entity/monster/Vex.java
+++ b/net/minecraft/world/entity/monster/Vex.java
@@ -110,7 +_,7 @@
    public void m_7378_(CompoundTag p_34008_) {
       super.m_7378_(p_34008_);
       if (p_34008_.m_128441_("BoundX")) {
-         this.f_33981_ = new BlockPos(p_34008_.m_128451_("BoundX"), p_34008_.m_128451_("BoundY"), p_34008_.m_128451_("BoundZ"));
+         this.f_33981_ = new BlockPos(p_34008_.m_128454_("BoundX"), p_34008_.m_128454_("BoundY"), p_34008_.m_128454_("BoundZ"));
       }
 
       if (p_34008_.m_128441_("LifeTicks")) {
@@ -122,9 +_,9 @@
    public void m_7380_(CompoundTag p_34015_) {
       super.m_7380_(p_34015_);
       if (this.f_33981_ != null) {
-         p_34015_.m_128405_("BoundX", this.f_33981_.m_123341_());
-         p_34015_.m_128405_("BoundY", this.f_33981_.m_123342_());
-         p_34015_.m_128405_("BoundZ", this.f_33981_.m_123343_());
+         p_34015_.m_128356_("BoundX", this.f_33981_.getX());
+         p_34015_.m_128356_("BoundY", this.f_33981_.getY());
+         p_34015_.m_128356_("BoundZ", this.f_33981_.getZ());
       }
 
       if (this.f_33978_) {
@@ -235,7 +_,7 @@
          LivingEntity livingentity = Vex.this.m_5448_();
          if (livingentity != null) {
             Vec3 vec3 = livingentity.m_146892_();
-            Vex.this.f_21342_.m_6849_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, 1.0D);
+            Vex.this.f_21342_.m_6849_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), 1.0D);
          }
 
          Vex.this.m_34042_(true);
@@ -260,7 +_,7 @@
                double d0 = Vex.this.m_20280_(livingentity);
                if (d0 < 9.0D) {
                   Vec3 vec3 = livingentity.m_146892_();
-                  Vex.this.f_21342_.m_6849_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, 1.0D);
+                  Vex.this.f_21342_.m_6849_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), 1.0D);
                }
             }
 
@@ -301,7 +_,7 @@
                Vex.this.m_20256_(Vex.this.m_20184_().m_82549_(vec3.m_82490_(this.f_24978_ * 0.05D / d0)));
                if (Vex.this.m_5448_() == null) {
                   Vec3 vec31 = Vex.this.m_20184_();
-                  Vex.this.m_146922_(-((float)Mth.m_14136_(vec31.f_82479_, vec31.f_82481_)) * (180F / (float)Math.PI));
+                  Vex.this.m_146922_(-((float)Mth.m_14136_(vec31.m_7096_(), vec31.m_7094_())) * (180F / (float)Math.PI));
                   Vex.this.f_20883_ = Vex.this.m_146908_();
                } else {
                   double d2 = Vex.this.m_5448_().m_20185_() - Vex.this.m_20185_();
@@ -335,11 +_,11 @@
          }
 
          for(int i = 0; i < 3; ++i) {
-            BlockPos blockpos1 = blockpos.m_7918_(Vex.this.f_19796_.m_188503_(15) - 7, Vex.this.f_19796_.m_188503_(11) - 5, Vex.this.f_19796_.m_188503_(15) - 7);
+            BlockPos blockpos1 = blockpos.offset(Vex.this.f_19796_.m_188503_(15) - 7, Vex.this.f_19796_.m_188503_(11) - 5, Vex.this.f_19796_.m_188503_(15) - 7);
             if (Vex.this.m_9236_().m_46859_(blockpos1)) {
-               Vex.this.f_21342_.m_6849_((double)blockpos1.m_123341_() + 0.5D, (double)blockpos1.m_123342_() + 0.5D, (double)blockpos1.m_123343_() + 0.5D, 0.25D);
+               Vex.this.f_21342_.m_6849_((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.5D, (double)blockpos1.getZ() + 0.5D, 0.25D);
                if (Vex.this.m_5448_() == null) {
-                  Vex.this.m_21563_().m_24950_((double)blockpos1.m_123341_() + 0.5D, (double)blockpos1.m_123342_() + 0.5D, (double)blockpos1.m_123343_() + 0.5D, 180.0F, 20.0F);
+                  Vex.this.m_21563_().m_24950_((double)blockpos1.getX() + 0.5D, (double)blockpos1.getY() + 0.5D, (double)blockpos1.getZ() + 0.5D, 180.0F, 20.0F);
                }
                break;
             }
