--- a/net/minecraft/world/entity/monster/ZombieVillager.java
+++ b/net/minecraft/world/entity/monster/ZombieVillager.java
@@ -240,7 +_,7 @@
          for(int k = (int)this.m_20185_() - 4; k < (int)this.m_20185_() + 4 && j < 14; ++k) {
             for(int l = (int)this.m_20186_() - 4; l < (int)this.m_20186_() + 4 && j < 14; ++l) {
                for(int i1 = (int)this.m_20189_() - 4; i1 < (int)this.m_20189_() + 4 && j < 14; ++i1) {
-                  BlockState blockstate = this.f_19853_.m_8055_(blockpos$mutableblockpos.m_122178_(k, l, i1));
+                  BlockState blockstate = this.f_19853_.m_8055_(blockpos$mutableblockpos.set(k, l, i1));
                   if (blockstate.m_60713_(Blocks.f_50183_) || blockstate.m_60734_() instanceof BedBlock) {
                      if (this.f_19796_.m_188501_() < 0.3F) {
                         ++i;
