--- a/net/minecraft/world/entity/monster/ZombieVillager.java
+++ b/net/minecraft/world/entity/monster/ZombieVillager.java
@@ -179,7 +_,7 @@
    public void m_7822_(byte p_34372_) {
       if (p_34372_ == 16) {
          if (!this.m_20067_()) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20188_(), this.m_20189_(), SoundEvents.f_12644_, this.m_5720_(), 1.0F + this.f_19796_.m_188501_(), this.f_19796_.m_188501_() * 0.7F + 0.3F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getEyeBigY(), this.getBigZ(), SoundEvents.f_12644_, this.m_5720_(), 1.0F + this.f_19796_.m_188501_(), this.f_19796_.m_188501_() * 0.7F + 0.3F, false);
          }
 
       } else {
@@ -240,7 +_,7 @@
          for(int k = (int)this.m_20185_() - 4; k < (int)this.m_20185_() + 4 && j < 14; ++k) {
             for(int l = (int)this.m_20186_() - 4; l < (int)this.m_20186_() + 4 && j < 14; ++l) {
                for(int i1 = (int)this.m_20189_() - 4; i1 < (int)this.m_20189_() + 4 && j < 14; ++i1) {
-                  BlockState blockstate = this.m_9236_().m_8055_(blockpos$mutableblockpos.m_122178_(k, l, i1));
+                  BlockState blockstate = this.m_9236_().m_8055_(blockpos$mutableblockpos.set(k, l, i1));
                   if (blockstate.m_60713_(Blocks.f_50183_) || blockstate.m_60734_() instanceof BedBlock) {
                      if (this.f_19796_.m_188501_() < 0.3F) {
                         ++i;
