--- a/net/minecraft/world/entity/monster/Zombie.java
+++ b/net/minecraft/world/entity/monster/Zombie.java
@@ -275,15 +_,15 @@
          }
 
          if (livingentity != null && this.f_19853_.m_46791_() == Difficulty.HARD && (double)this.f_19796_.m_188501_() < this.m_21133_(Attributes.f_22287_) && this.f_19853_.m_46469_().m_46207_(GameRules.f_46134_)) {
-            int i = Mth.m_14107_(this.m_20185_());
-            int j = Mth.m_14107_(this.m_20186_());
-            int k = Mth.m_14107_(this.m_20189_());
+            long i = Mth.m_14134_(this.m_20185_());
+            long j = Mth.m_14134_(this.m_20186_());
+            long k = Mth.m_14134_(this.m_20189_());
             Zombie zombie = new Zombie(this.f_19853_);
 
             for(int l = 0; l < 50; ++l) {
-               int i1 = i + Mth.m_216271_(this.f_19796_, 7, 40) * Mth.m_216271_(this.f_19796_, -1, 1);
-               int j1 = j + Mth.m_216271_(this.f_19796_, 7, 40) * Mth.m_216271_(this.f_19796_, -1, 1);
-               int k1 = k + Mth.m_216271_(this.f_19796_, 7, 40) * Mth.m_216271_(this.f_19796_, -1, 1);
+               long i1 = i + Mth.nextLong(this.f_19796_, 7, 40) * Mth.nextLong(this.f_19796_, -1, 1);
+               long j1 = j + Mth.nextLong(this.f_19796_, 7, 40) * Mth.nextLong(this.f_19796_, -1, 1);
+               long k1 = k + Mth.nextLong(this.f_19796_, 7, 40) * Mth.nextLong(this.f_19796_, -1, 1);
                BlockPos blockpos = new BlockPos(i1, j1, k1);
                EntityType<?> entitytype = zombie.m_6095_();
                SpawnPlacements.Type spawnplacements$type = SpawnPlacements.m_21752_(entitytype);
