--- a/net/minecraft/world/entity/monster/ZombifiedPiglin.java
+++ b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
@@ -162,7 +_,7 @@
    }
 
    public static boolean m_219173_(EntityType<ZombifiedPiglin> p_219174_, LevelAccessor p_219175_, MobSpawnType p_219176_, BlockPos p_219177_, RandomSource p_219178_) {
-      return p_219175_.m_46791_() != Difficulty.PEACEFUL && !p_219175_.m_8055_(p_219177_.m_7495_()).m_60713_(Blocks.f_50451_);
+      return p_219175_.m_46791_() != Difficulty.PEACEFUL && !p_219175_.m_8055_(p_219177_.below()).m_60713_(Blocks.f_50451_);
    }
 
    public boolean m_6914_(LevelReader p_34442_) {
