--- a/net/minecraft/world/entity/ai/behavior/warden/SonicBoom.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/SonicBoom.java
@@ -57,7 +_,7 @@
 
             for(int i = 1; i < Mth.m_14107_(vec31.m_82553_()) + 7; ++i) {
                Vec3 vec33 = vec3.m_82549_(vec32.m_82490_((double)i));
-               p_217724_.m_8767_(ParticleTypes.f_235902_, vec33.f_82479_, vec33.f_82480_, vec33.f_82481_, 1, 0.0D, 0.0D, 0.0D, 0.0D);
+               p_217724_.sendParticles(ParticleTypes.f_235902_, vec33.getX(), vec33.getY(), vec33.getZ(), 1, 0.0D, 0.0D, 0.0D, 0.0D);
             }
 
             p_217725_.m_5496_(SoundEvents.f_215771_, 3.0F, 1.0F);
