--- a/net/minecraft/client/renderer/entity/player/PlayerRenderer.java
+++ b/net/minecraft/client/renderer/entity/player/PlayerRenderer.java
@@ -202,8 +_,8 @@
          double d0 = vec31.m_165925_();
          double d1 = vec3.m_165925_();
          if (d0 > 0.0D && d1 > 0.0D) {
-            double d2 = (vec31.f_82479_ * vec3.f_82479_ + vec31.f_82481_ * vec3.f_82481_) / Math.sqrt(d0 * d1);
-            double d3 = vec31.f_82479_ * vec3.f_82481_ - vec31.f_82481_ * vec3.f_82479_;
+            double d2 = (vec31.m_7096_() * vec3.m_7096_() + vec31.m_7094_() * vec3.m_7094_()) / Math.sqrt(d0 * d1);
+            double d3 = vec31.m_7096_() * vec3.m_7094_() - vec31.m_7094_() * vec3.m_7096_();
             p_117803_.m_252781_(Axis.f_252436_.m_252961_((float)(Math.signum(d3) * Math.acos(d2))));
          }
       } else if (f > 0.0F) {
