--- a/net/minecraft/client/gui/screens/LevelLoadingScreen.java
+++ b/net/minecraft/client/gui/screens/LevelLoadingScreen.java
@@ -103,7 +_,7 @@
 
          for(int j2 = 0; j2 < l; ++j2) {
             for(int k2 = 0; k2 < l; ++k2) {
-               ChunkStatus chunkstatus = p_96151_.m_9663_(j2, k2);
+               ChunkStatus chunkstatus = p_96151_.getStatus(j2, k2);
                int l2 = j1 + j2 * i;
                int i3 = k1 + k2 * i;
                p_283467_.m_280509_(l2, i3, l2 + p_96154_, i3 + p_96154_, f_96140_.getInt(chunkstatus) | -16777216);
