--- a/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
+++ b/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
@@ -33,7 +_,7 @@
       boolean flag = this.f_202372_ || System.currentTimeMillis() > this.f_202373_ + 2000L;
       if (flag && this.f_96541_ != null && this.f_96541_.f_91074_ != null) {
          BlockPos blockpos = this.f_96541_.f_91074_.m_20183_();
-         boolean flag1 = this.f_96541_.f_91073_ != null && this.f_96541_.f_91073_.m_151562_(blockpos.m_123342_());
+         boolean flag1 = this.f_96541_.f_91073_ != null && this.f_96541_.f_91073_.isOutsideBuildHeight(blockpos.getY());
          if (flag1 || this.f_96541_.f_91060_.m_202430_(blockpos)) {
             this.m_7379_();
          }
