--- a/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
+++ b/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
@@ -43,7 +_,7 @@
             }
 
             BlockPos blockpos = this.f_96541_.f_91074_.m_20183_();
-            boolean flag = this.f_96541_.f_91073_ != null && this.f_96541_.f_91073_.m_151562_(blockpos.m_123342_());
+            boolean flag = this.f_96541_.f_91073_ != null && this.f_96541_.f_91073_.isOutsideBuildHeight(blockpos.getY());
             if (flag || this.f_96541_.f_91060_.m_202430_(blockpos) || this.f_96541_.f_91074_.m_5833_() || !this.f_96541_.f_91074_.m_6084_()) {
                this.m_7379_();
             }
