--- a/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/net/minecraft/client/gui/screens/TitleScreen.java
@@ -248,7 +_,7 @@
             this.f_96721_.m_280672_(p_282860_, this.f_96543_, this.f_96547_, i);
          }
 
-         String s = "Minecraft " + SharedConstants.m_183709_().m_132493_();
+         String s = SharedConstants.m_183709_().m_132493_();
          if (this.f_96541_.m_91402_()) {
             s = s + " Demo";
          } else {
