--- a/net/minecraft/world/level/levelgen/Aquifer.java
+++ b/net/minecraft/world/level/levelgen/Aquifer.java
@@ -22,7 +_,7 @@
       return new Aquifer() {
          @Nullable
          public BlockState m_207104_(DensityFunction.FunctionContext p_208172_, double p_208173_) {
-            return p_208173_ > 0.0D ? null : p_188375_.m_183538_(p_208172_.m_207115_(), p_208172_.m_207114_(), p_208172_.m_207113_()).m_188405_(p_208172_.m_207114_());
+            return p_208173_ > 0.0D ? null : p_188375_.computeFluid(p_208172_.blockX(), p_208172_.blockY(), p_208172_.blockZ()).at(p_208172_.blockY().longValue());
          }
 
          public boolean m_142203_() {
@@ -37,19 +_,19 @@
    boolean m_142203_();
 
    public interface FluidPicker {
-      Aquifer.FluidStatus m_183538_(int p_188397_, int p_188398_, int p_188399_);
+      Aquifer.FluidStatus computeFluid(me.alphamode.mclong.math.BigInteger p_188397_, me.alphamode.mclong.math.BigInteger p_188398_, me.alphamode.mclong.math.BigInteger p_188399_);
    }
 
    public static final class FluidStatus {
-      final int f_188400_;
+      final long f_188400_;
       final BlockState f_188401_;
 
-      public FluidStatus(int p_188403_, BlockState p_188404_) {
+      public FluidStatus(long p_188403_, BlockState p_188404_) {
          this.f_188400_ = p_188403_;
          this.f_188401_ = p_188404_;
       }
 
-      public BlockState m_188405_(int p_188406_) {
+      public BlockState at(long p_188406_) {
          return p_188406_ < this.f_188400_ ? this.f_188401_ : Blocks.f_50016_.m_49966_();
       }
    }
@@ -73,14 +_,14 @@
       protected final DensityFunction f_157996_;
       private final PositionalRandomFactory f_188410_;
       protected final Aquifer.FluidStatus[] f_157998_;
-      protected final long[] f_157999_;
+      protected final BlockPos[] f_157999_;
       private final Aquifer.FluidPicker f_188411_;
       private final DensityFunction f_223888_;
       private final DensityFunction f_223889_;
       protected boolean f_158000_;
-      protected final int f_158002_;
+      protected final me.alphamode.mclong.math.BigInteger f_158002_;
       protected final int f_158003_;
-      protected final int f_158004_;
+      protected final me.alphamode.mclong.math.BigInteger f_158004_;
       protected final int f_158005_;
       protected final int f_158006_;
       private static final int[][] f_188412_ = new int[][]{{0, 0}, {-2, -1}, {-1, -1}, {0, -1}, {1, -1}, {-3, 0}, {-2, 0}, {-1, 0}, {1, 0}, {-2, 1}, {-1, 1}, {0, 1}, {1, 1}};
@@ -94,87 +_,87 @@
          this.f_223888_ = p_223893_.f_209387_();
          this.f_223889_ = p_223893_.f_209388_();
          this.f_188410_ = p_223894_;
-         this.f_158002_ = this.m_158039_(p_223892_.m_45604_()) - 1;
+         this.f_158002_ = this.gridX(p_223892_.getMinBigBlockX()).subtract();
          this.f_188411_ = p_223897_;
-         int i = this.m_158039_(p_223892_.m_45608_()) + 1;
-         this.f_158005_ = i - this.f_158002_ + 1;
+         me.alphamode.mclong.math.BigInteger i = this.gridX(p_223892_.getMaxBlockBigX()).add();
+         this.f_158005_ = i.subtract(this.f_158002_).add().intValue();
          this.f_158003_ = this.m_158045_(p_223895_) - 1;
          int j = this.m_158045_(p_223895_ + p_223896_) + 1;
          int k = j - this.f_158003_ + 1;
-         this.f_158004_ = this.m_158047_(p_223892_.m_45605_()) - 1;
-         int l = this.m_158047_(p_223892_.m_45609_()) + 1;
-         this.f_158006_ = l - this.f_158004_ + 1;
+         this.f_158004_ = this.gridZ(p_223892_.getMinBigBlockZ()).subtract();
+         me.alphamode.mclong.math.BigInteger l = this.gridZ(p_223892_.getMaxBlockBigZ()).add();
+         this.f_158006_ = l.subtract(this.f_158004_).add().intValue();
          int i1 = this.f_158005_ * k * this.f_158006_;
          this.f_157998_ = new Aquifer.FluidStatus[i1];
-         this.f_157999_ = new long[i1];
-         Arrays.fill(this.f_157999_, Long.MAX_VALUE);
+         this.f_157999_ = new BlockPos[i1];
+         Arrays.fill(this.f_157999_, BlockPos.MAX_VALUE);
       }
 
-      protected int m_158027_(int p_158028_, int p_158029_, int p_158030_) {
-         int i = p_158028_ - this.f_158002_;
+      protected int getIndex(me.alphamode.mclong.math.BigInteger p_158028_, int p_158029_, me.alphamode.mclong.math.BigInteger p_158030_) {
+         int i = p_158028_.subtract(this.f_158002_).intValue();
          int j = p_158029_ - this.f_158003_;
-         int k = p_158030_ - this.f_158004_;
+         int k = p_158030_.subtract(this.f_158004_).intValue();
          return (j * this.f_158006_ + k) * this.f_158005_ + i;
       }
 
       @Nullable
       public BlockState m_207104_(DensityFunction.FunctionContext p_208186_, double p_208187_) {
-         int i = p_208186_.m_207115_();
-         int j = p_208186_.m_207114_();
-         int k = p_208186_.m_207113_();
+         me.alphamode.mclong.math.BigInteger i = p_208186_.blockX();
+         me.alphamode.mclong.math.BigInteger j = p_208186_.blockY();
+         me.alphamode.mclong.math.BigInteger k = p_208186_.blockZ();
          if (p_208187_ > 0.0D) {
             this.f_158000_ = false;
             return null;
          } else {
-            Aquifer.FluidStatus aquifer$fluidstatus = this.f_188411_.m_183538_(i, j, k);
-            if (aquifer$fluidstatus.m_188405_(j).m_60713_(Blocks.f_49991_)) {
+            Aquifer.FluidStatus aquifer$fluidstatus = this.f_188411_.computeFluid(i, j, k);
+            if (aquifer$fluidstatus.at(j.intValue()).m_60713_(Blocks.f_49991_)) {
                this.f_158000_ = false;
                return Blocks.f_49991_.m_49966_();
             } else {
-               int l = Math.floorDiv(i - 5, 16);
-               int i1 = Math.floorDiv(j + 1, 12);
-               int j1 = Math.floorDiv(k - 5, 16);
-               int k1 = Integer.MAX_VALUE;
-               int l1 = Integer.MAX_VALUE;
-               int i2 = Integer.MAX_VALUE;
-               long j2 = 0L;
-               long k2 = 0L;
-               long l2 = 0L;
+               me.alphamode.mclong.math.BigInteger l = me.alphamode.mclong.math.BigMath.floorDiv(i.subtract(me.alphamode.mclong.core.BigConstants.Ints.FIVE), 16);
+               me.alphamode.mclong.math.BigInteger i1 = me.alphamode.mclong.math.BigMath.floorDiv(j.add(), 12);
+               me.alphamode.mclong.math.BigInteger j1 = me.alphamode.mclong.math.BigMath.floorDiv(k.subtract(me.alphamode.mclong.core.BigConstants.Ints.FIVE), 16);
+               me.alphamode.mclong.math.BigInteger k1 = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+               me.alphamode.mclong.math.BigInteger l1 = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+               me.alphamode.mclong.math.BigInteger i2 = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+               BlockPos j2 = BlockPos.f_121853_;
+               BlockPos k2 = BlockPos.f_121853_;
+               BlockPos l2 = BlockPos.f_121853_;
 
                for(int i3 = 0; i3 <= 1; ++i3) {
                   for(int j3 = -1; j3 <= 1; ++j3) {
                      for(int k3 = 0; k3 <= 1; ++k3) {
-                        int l3 = l + i3;
-                        int i4 = i1 + j3;
-                        int j4 = j1 + k3;
-                        int k4 = this.m_158027_(l3, i4, j4);
-                        long i5 = this.f_157999_[k4];
-                        long l4;
-                        if (i5 != Long.MAX_VALUE) {
+                        me.alphamode.mclong.math.BigInteger l3 = l.add(i3);
+                        me.alphamode.mclong.math.BigInteger i4 = i1.add(j3);
+                        me.alphamode.mclong.math.BigInteger j4 = j1.add(k3);
+                        int k4 = this.getIndex(l3, i4.intValue(), j4);
+                        BlockPos i5 = this.f_157999_[k4];
+                        BlockPos l4;
+                        if (!i5.equals(BlockPos.MAX_VALUE)) {
                            l4 = i5;
                         } else {
-                           RandomSource randomsource = this.f_188410_.m_213715_(l3, i4, j4);
-                           l4 = BlockPos.m_121882_(l3 * 16 + randomsource.m_188503_(10), i4 * 12 + randomsource.m_188503_(9), j4 * 16 + randomsource.m_188503_(10));
+                           RandomSource randomsource = this.f_188410_.at(l3.longValue(), i4.longValue(), j4.longValue());
+                           l4 = new BlockPos(l3.multiply(me.alphamode.mclong.core.BigConstants.Ints.SIXTEEN).add(randomsource.m_188503_(10)), i4.multiply(me.alphamode.mclong.core.BigConstants.Ints.TWELVE).add(randomsource.m_188503_(9)), j4.multiply(me.alphamode.mclong.core.BigConstants.Ints.SIXTEEN).add(randomsource.m_188503_(10)));
                            this.f_157999_[k4] = l4;
                         }
 
-                        int i6 = BlockPos.m_121983_(l4) - i;
-                        int j5 = BlockPos.m_122008_(l4) - j;
-                        int k5 = BlockPos.m_122015_(l4) - k;
-                        int l5 = i6 * i6 + j5 * j5 + k5 * k5;
-                        if (k1 >= l5) {
+                        me.alphamode.mclong.math.BigInteger i6 = l4.getBigX().subtract(i);
+                        me.alphamode.mclong.math.BigInteger j5 = l4.getBigY().subtract(j);
+                        me.alphamode.mclong.math.BigInteger k5 = l4.getBigZ().subtract(k);
+                        me.alphamode.mclong.math.BigInteger l5 = i6.multiply(i6).add(j5.multiply(j5)).add(k5.multiply(k5));
+                        if (k1.compareTo(l5) >= 0) {
                            l2 = k2;
                            k2 = j2;
                            j2 = l4;
                            i2 = l1;
                            l1 = k1;
                            k1 = l5;
-                        } else if (l1 >= l5) {
+                        } else if (l1.compareTo(l5) >= 0) {
                            l2 = k2;
                            k2 = l4;
                            i2 = l1;
                            l1 = l5;
-                        } else if (i2 >= l5) {
+                        } else if (i2.compareTo(l5) >= 0) {
                            l2 = l4;
                            i2 = l5;
                         }
@@ -182,25 +_,25 @@
                   }
                }
 
-               Aquifer.FluidStatus aquifer$fluidstatus1 = this.m_188445_(j2);
-               double d1 = m_158024_(k1, l1);
-               BlockState blockstate = aquifer$fluidstatus1.m_188405_(j);
+               Aquifer.FluidStatus aquifer$fluidstatus1 = this.getAquiferStatus(j2);
+               double d1 = similarity(k1, l1);
+               BlockState blockstate = aquifer$fluidstatus1.at(j.intValue());
                if (d1 <= 0.0D) {
                   this.f_158000_ = d1 >= f_196979_;
                   return blockstate;
-               } else if (blockstate.m_60713_(Blocks.f_49990_) && this.f_188411_.m_183538_(i, j - 1, k).m_188405_(j - 1).m_60713_(Blocks.f_49991_)) {
+               } else if (blockstate.m_60713_(Blocks.f_49990_) && this.f_188411_.computeFluid(i, j.subtract(), k).at(j.subtract().intValue()).m_60713_(Blocks.f_49991_)) {
                   this.f_158000_ = true;
                   return blockstate;
                } else {
                   MutableDouble mutabledouble = new MutableDouble(Double.NaN);
-                  Aquifer.FluidStatus aquifer$fluidstatus2 = this.m_188445_(k2);
+                  Aquifer.FluidStatus aquifer$fluidstatus2 = this.getAquiferStatus(k2);
                   double d2 = d1 * this.m_208188_(p_208186_, mutabledouble, aquifer$fluidstatus1, aquifer$fluidstatus2);
                   if (p_208187_ + d2 > 0.0D) {
                      this.f_158000_ = false;
                      return null;
                   } else {
-                     Aquifer.FluidStatus aquifer$fluidstatus3 = this.m_188445_(l2);
-                     double d0 = m_158024_(k1, i2);
+                     Aquifer.FluidStatus aquifer$fluidstatus3 = this.getAquiferStatus(l2);
+                     double d0 = similarity(k1, i2);
                      if (d0 > 0.0D) {
                         double d3 = d1 * d0 * this.m_208188_(p_208186_, mutabledouble, aquifer$fluidstatus1, aquifer$fluidstatus3);
                         if (p_208187_ + d3 > 0.0D) {
@@ -209,7 +_,7 @@
                         }
                      }
 
-                     double d4 = m_158024_(l1, i2);
+                     double d4 = similarity(l1, i2);
                      if (d4 > 0.0D) {
                         double d5 = d1 * d4 * this.m_208188_(p_208186_, mutabledouble, aquifer$fluidstatus2, aquifer$fluidstatus3);
                         if (p_208187_ + d5 > 0.0D) {
@@ -230,22 +_,27 @@
          return this.f_158000_;
       }
 
+      protected static double similarity(me.alphamode.mclong.math.BigInteger x1, me.alphamode.mclong.math.BigInteger x2) {
+         double d0 = 25.0D;
+         return 1.0D - (double)Math.abs(x2.subtract(x1).longValue()) / 25.0D;
+      }
+
       protected static double m_158024_(int p_158025_, int p_158026_) {
          double d0 = 25.0D;
          return 1.0D - (double)Math.abs(p_158026_ - p_158025_) / 25.0D;
       }
 
       private double m_208188_(DensityFunction.FunctionContext p_208189_, MutableDouble p_208190_, Aquifer.FluidStatus p_208191_, Aquifer.FluidStatus p_208192_) {
-         int i = p_208189_.m_207114_();
-         BlockState blockstate = p_208191_.m_188405_(i);
-         BlockState blockstate1 = p_208192_.m_188405_(i);
+         me.alphamode.mclong.math.BigInteger i = p_208189_.blockY();
+         BlockState blockstate = p_208191_.at(i.longValue());
+         BlockState blockstate1 = p_208192_.at(i.longValue());
          if ((!blockstate.m_60713_(Blocks.f_49991_) || !blockstate1.m_60713_(Blocks.f_49990_)) && (!blockstate.m_60713_(Blocks.f_49990_) || !blockstate1.m_60713_(Blocks.f_49991_))) {
-            int j = Math.abs(p_208191_.f_188400_ - p_208192_.f_188400_);
+            long j = Math.abs(p_208191_.f_188400_ - p_208192_.f_188400_);
             if (j == 0) {
                return 0.0D;
             } else {
                double d0 = 0.5D * (double)(p_208191_.f_188400_ + p_208192_.f_188400_);
-               double d1 = (double)i + 0.5D - d0;
+               double d1 = (double)i.longValue() + 0.5D - d0;
                double d2 = (double)j / 2.0D;
                double d3 = 0.0D;
                double d4 = 2.5D;
@@ -293,57 +_,57 @@
          }
       }
 
-      protected int m_158039_(int p_158040_) {
-         return Math.floorDiv(p_158040_, 16);
+      protected me.alphamode.mclong.math.BigInteger gridX(me.alphamode.mclong.math.BigInteger p_158040_) {
+         return me.alphamode.mclong.math.BigMath.floorDiv(p_158040_, 16);
       }
 
       protected int m_158045_(int p_158046_) {
          return Math.floorDiv(p_158046_, 12);
       }
 
-      protected int m_158047_(int p_158048_) {
-         return Math.floorDiv(p_158048_, 16);
+      protected me.alphamode.mclong.math.BigInteger gridZ(me.alphamode.mclong.math.BigInteger p_158048_) {
+         return me.alphamode.mclong.math.BigMath.floorDiv(p_158048_, 16);
       }
 
-      private Aquifer.FluidStatus m_188445_(long p_188446_) {
-         int i = BlockPos.m_121983_(p_188446_);
-         int j = BlockPos.m_122008_(p_188446_);
-         int k = BlockPos.m_122015_(p_188446_);
-         int l = this.m_158039_(i);
-         int i1 = this.m_158045_(j);
-         int j1 = this.m_158047_(k);
-         int k1 = this.m_158027_(l, i1, j1);
+      private Aquifer.FluidStatus getAquiferStatus(BlockPos p_188446_) {
+         me.alphamode.mclong.math.BigInteger i = p_188446_.getBigX();
+         me.alphamode.mclong.math.BigInteger j = p_188446_.getBigY();
+         me.alphamode.mclong.math.BigInteger k = p_188446_.getBigZ();
+         me.alphamode.mclong.math.BigInteger l = this.gridX(i);
+         int i1 = this.m_158045_(j.intValue());
+         me.alphamode.mclong.math.BigInteger j1 = this.gridZ(k);
+         int k1 = this.getIndex(l, i1, j1);
          Aquifer.FluidStatus aquifer$fluidstatus = this.f_157998_[k1];
          if (aquifer$fluidstatus != null) {
             return aquifer$fluidstatus;
          } else {
-            Aquifer.FluidStatus aquifer$fluidstatus1 = this.m_188447_(i, j, k);
+            Aquifer.FluidStatus aquifer$fluidstatus1 = this.computeFluid(i, j, k);
             this.f_157998_[k1] = aquifer$fluidstatus1;
             return aquifer$fluidstatus1;
          }
       }
 
-      private Aquifer.FluidStatus m_188447_(int p_188448_, int p_188449_, int p_188450_) {
-         Aquifer.FluidStatus aquifer$fluidstatus = this.f_188411_.m_183538_(p_188448_, p_188449_, p_188450_);
-         int i = Integer.MAX_VALUE;
-         int j = p_188449_ + 12;
-         int k = p_188449_ - 12;
+      private Aquifer.FluidStatus computeFluid(me.alphamode.mclong.math.BigInteger p_188448_, me.alphamode.mclong.math.BigInteger p_188449_, me.alphamode.mclong.math.BigInteger p_188450_) {
+         Aquifer.FluidStatus aquifer$fluidstatus = this.f_188411_.computeFluid(p_188448_, p_188449_, p_188450_);
+         long i = java.lang.Long.MAX_VALUE;
+         me.alphamode.mclong.math.BigInteger j = p_188449_.add(me.alphamode.mclong.core.BigConstants.Ints.TWELVE);
+         me.alphamode.mclong.math.BigInteger k = p_188449_.subtract(me.alphamode.mclong.core.BigConstants.Ints.TWELVE);
          boolean flag = false;
 
          for(int[] aint : f_188412_) {
-            int l = p_188448_ + SectionPos.m_123223_(aint[0]);
-            int i1 = p_188450_ + SectionPos.m_123223_(aint[1]);
-            int j1 = this.f_188407_.m_198256_(l, i1);
-            int k1 = j1 + 8;
+            me.alphamode.mclong.math.BigInteger l = p_188448_.add(SectionPos.sectionToBlockCoord(aint[0]));
+            me.alphamode.mclong.math.BigInteger i1 = p_188450_.add(SectionPos.sectionToBlockCoord(aint[1]));
+            long j1 = this.f_188407_.preliminarySurfaceLevel(l, i1);
+            long k1 = j1 + 8;
             boolean flag1 = aint[0] == 0 && aint[1] == 0;
-            if (flag1 && k > k1) {
+            if (flag1 && k.compareTo(k1) > 0) {
                return aquifer$fluidstatus;
             }
 
-            boolean flag2 = j > k1;
+            boolean flag2 = j.compareTo(k1) > 0;
             if (flag2 || flag1) {
-               Aquifer.FluidStatus aquifer$fluidstatus1 = this.f_188411_.m_183538_(l, k1, i1);
-               if (!aquifer$fluidstatus1.m_188405_(k1).m_60795_()) {
+               Aquifer.FluidStatus aquifer$fluidstatus1 = this.f_188411_.computeFluid(l, me.alphamode.mclong.math.BigInteger.valueOf(k1), i1);
+               if (!aquifer$fluidstatus1.at(k1).m_60795_()) {
                   if (flag1) {
                      flag = true;
                   }
@@ -357,11 +_,11 @@
             i = Math.min(i, j1);
          }
 
-         int l1 = this.m_223909_(p_188448_, p_188449_, p_188450_, aquifer$fluidstatus, i, flag);
-         return new Aquifer.FluidStatus(l1, this.m_223903_(p_188448_, p_188449_, p_188450_, aquifer$fluidstatus, l1));
+         long l1 = this.computeSurfaceLevel(p_188448_, p_188449_, p_188450_, aquifer$fluidstatus, i, flag);
+         return new Aquifer.FluidStatus(l1, this.computeFluidType(p_188448_, p_188449_, p_188450_, aquifer$fluidstatus, l1));
       }
 
-      private int m_223909_(int p_223910_, int p_223911_, int p_223912_, Aquifer.FluidStatus p_223913_, int p_223914_, boolean p_223915_) {
+      private long computeSurfaceLevel(me.alphamode.mclong.math.BigInteger p_223910_, me.alphamode.mclong.math.BigInteger p_223911_, me.alphamode.mclong.math.BigInteger p_223912_, Aquifer.FluidStatus p_223913_, long p_223914_, boolean p_223915_) {
          DensityFunction.SinglePointContext densityfunction$singlepointcontext = new DensityFunction.SinglePointContext(p_223910_, p_223911_, p_223912_);
          double d0;
          double d1;
@@ -369,7 +_,7 @@
             d0 = -1.0D;
             d1 = -1.0D;
          } else {
-            int i = p_223914_ + 8 - p_223911_;
+            long i = me.alphamode.mclong.math.BigInteger.valueOf(p_223914_ + 8).subtract(p_223911_).longValue();
             int j = 64;
             double d2 = p_223915_ ? Mth.m_144851_((double)i, 0.0D, 64.0D, 1.0D, 0.0D) : 0.0D;
             double d3 = Mth.m_14008_(this.f_188408_.m_207386_(densityfunction$singlepointcontext), -1.0D, 1.0D);
@@ -379,11 +_,11 @@
             d1 = d3 - d4;
          }
 
-         int k;
+         long k;
          if (d1 > 0.0D) {
             k = p_223913_.f_188400_;
          } else if (d0 > 0.0D) {
-            k = this.m_223898_(p_223910_, p_223911_, p_223912_, p_223914_);
+            k = this.computeRandomizedFluidSurfaceLevel(p_223910_, p_223911_, p_223912_, p_223914_);
          } else {
             k = DimensionType.f_188294_;
          }
@@ -391,13 +_,13 @@
          return k;
       }
 
-      private int m_223898_(int p_223899_, int p_223900_, int p_223901_, int p_223902_) {
+      private long computeRandomizedFluidSurfaceLevel(me.alphamode.mclong.math.BigInteger p_223899_, me.alphamode.mclong.math.BigInteger p_223900_, me.alphamode.mclong.math.BigInteger p_223901_, long p_223902_) {
          int i = 16;
          int j = 40;
-         int k = Math.floorDiv(p_223899_, 16);
-         int l = Math.floorDiv(p_223900_, 40);
-         int i1 = Math.floorDiv(p_223901_, 16);
-         int j1 = l * 40 + 20;
+         me.alphamode.mclong.math.BigInteger k = me.alphamode.mclong.math.BigMath.floorDiv(p_223899_, 16);
+         me.alphamode.mclong.math.BigInteger l = me.alphamode.mclong.math.BigMath.floorDiv(p_223900_, 40);
+         me.alphamode.mclong.math.BigInteger i1 = me.alphamode.mclong.math.BigMath.floorDiv(p_223901_, 16);
+         int j1 = l.intValue() * 40 + 20;
          int k1 = 10;
          double d0 = this.f_188409_.m_207386_(new DensityFunction.SinglePointContext(k, l, i1)) * 10.0D;
          int l1 = Mth.m_184628_(d0, 3);
@@ -405,14 +_,14 @@
          return Math.min(p_223902_, i2);
       }
 
-      private BlockState m_223903_(int p_223904_, int p_223905_, int p_223906_, Aquifer.FluidStatus p_223907_, int p_223908_) {
+      private BlockState computeFluidType(me.alphamode.mclong.math.BigInteger p_223904_, me.alphamode.mclong.math.BigInteger p_223905_, me.alphamode.mclong.math.BigInteger p_223906_, Aquifer.FluidStatus p_223907_, long p_223908_) {
          BlockState blockstate = p_223907_.f_188401_;
          if (p_223908_ <= -10 && p_223908_ != DimensionType.f_188294_ && p_223907_.f_188401_ != Blocks.f_49991_.m_49966_()) {
             int i = 64;
             int j = 40;
-            int k = Math.floorDiv(p_223904_, 64);
-            int l = Math.floorDiv(p_223905_, 40);
-            int i1 = Math.floorDiv(p_223906_, 64);
+            me.alphamode.mclong.math.BigInteger k = me.alphamode.mclong.math.BigMath.floorDiv(p_223904_, 64);
+            me.alphamode.mclong.math.BigInteger l = me.alphamode.mclong.math.BigMath.floorDiv(p_223905_, 40);
+            me.alphamode.mclong.math.BigInteger i1 = me.alphamode.mclong.math.BigMath.floorDiv(p_223906_, 64);
             double d0 = this.f_157996_.m_207386_(new DensityFunction.SinglePointContext(k, l, i1));
             if (Math.abs(d0) > 0.3D) {
                blockstate = Blocks.f_49991_.m_49966_();
