--- a/net/minecraft/world/level/levelgen/Column.java
+++ b/net/minecraft/world/level/levelgen/Column.java
@@ -13,11 +_,11 @@
       return new Column.Range(p_158165_ - 1, p_158166_ + 1);
    }
 
-   public static Column.Range m_158188_(int p_158189_, int p_158190_) {
+   public static Column.Range inside(long p_158189_, long p_158190_) {
       return new Column.Range(p_158189_, p_158190_);
    }
 
-   public static Column m_158162_(int p_158163_) {
+   public static Column below(long p_158163_) {
       return new Column.Ray(p_158163_, false);
    }
 
@@ -25,7 +_,7 @@
       return new Column.Ray(p_158187_ + 1, false);
    }
 
-   public static Column m_158193_(int p_158194_) {
+   public static Column above(long p_158194_) {
       return new Column.Ray(p_158194_, true);
    }
 
@@ -37,28 +_,28 @@
       return Column.Line.f_158197_;
    }
 
-   public static Column m_158183_(OptionalInt p_158184_, OptionalInt p_158185_) {
+   public static Column create(java.util.OptionalLong p_158184_, java.util.OptionalLong p_158185_) {
       if (p_158184_.isPresent() && p_158185_.isPresent()) {
-         return m_158188_(p_158184_.getAsInt(), p_158185_.getAsInt());
+         return inside(p_158184_.getAsLong(), p_158185_.getAsLong());
       } else if (p_158184_.isPresent()) {
-         return m_158193_(p_158184_.getAsInt());
+         return above(p_158184_.getAsLong());
       } else {
-         return p_158185_.isPresent() ? m_158162_(p_158185_.getAsInt()) : m_158161_();
+         return p_158185_.isPresent() ? below(p_158185_.getAsLong()) : m_158161_();
       }
    }
 
-   public abstract OptionalInt m_142011_();
-
-   public abstract OptionalInt m_142009_();
-
-   public abstract OptionalInt m_142030_();
-
-   public Column m_158181_(OptionalInt p_158182_) {
-      return m_158183_(p_158182_, this.m_142011_());
+   public abstract java.util.OptionalLong getCeiling();
+
+   public abstract java.util.OptionalLong getFloor();
+
+   public abstract java.util.OptionalLong getHeight();
+
+   public Column withFloor(java.util.OptionalLong p_158182_) {
+      return create(p_158182_, this.getCeiling());
    }
 
-   public Column m_158191_(OptionalInt p_158192_) {
-      return m_158183_(this.m_142009_(), p_158192_);
+   public Column withCeiling(java.util.OptionalLong p_158192_) {
+      return create(this.getFloor(), p_158192_);
    }
 
    public static Optional<Column> m_158175_(LevelSimulatedReader p_158176_, BlockPos p_158177_, int p_158178_, Predicate<BlockState> p_158179_, Predicate<BlockState> p_158180_) {
@@ -66,21 +_,21 @@
       if (!p_158176_.m_7433_(p_158177_, p_158179_)) {
          return Optional.empty();
       } else {
-         int i = p_158177_.m_123342_();
-         OptionalInt optionalint = m_158167_(p_158176_, p_158178_, p_158179_, p_158180_, blockpos$mutableblockpos, i, Direction.UP);
-         OptionalInt optionalint1 = m_158167_(p_158176_, p_158178_, p_158179_, p_158180_, blockpos$mutableblockpos, i, Direction.DOWN);
-         return Optional.of(m_158183_(optionalint1, optionalint));
+         long i = p_158177_.getY();
+         java.util.OptionalLong optionalint = scanDirection(p_158176_, p_158178_, p_158179_, p_158180_, blockpos$mutableblockpos, i, Direction.UP);
+         java.util.OptionalLong optionalint1 = scanDirection(p_158176_, p_158178_, p_158179_, p_158180_, blockpos$mutableblockpos, i, Direction.DOWN);
+         return Optional.of(create(optionalint1, optionalint));
       }
    }
 
-   private static OptionalInt m_158167_(LevelSimulatedReader p_158168_, int p_158169_, Predicate<BlockState> p_158170_, Predicate<BlockState> p_158171_, BlockPos.MutableBlockPos p_158172_, int p_158173_, Direction p_158174_) {
-      p_158172_.m_142448_(p_158173_);
+   private static java.util.OptionalLong scanDirection(LevelSimulatedReader p_158168_, int p_158169_, Predicate<BlockState> p_158170_, Predicate<BlockState> p_158171_, BlockPos.MutableBlockPos p_158172_, long p_158173_, Direction p_158174_) {
+      p_158172_.setY(p_158173_);
 
       for(int i = 1; i < p_158169_ && p_158168_.m_7433_(p_158172_, p_158170_); ++i) {
          p_158172_.m_122173_(p_158174_);
       }
 
-      return p_158168_.m_7433_(p_158172_, p_158171_) ? OptionalInt.of(p_158172_.m_123342_()) : OptionalInt.empty();
+      return p_158168_.m_7433_(p_158172_, p_158171_) ? java.util.OptionalLong.of(p_158172_.getY()) : java.util.OptionalLong.empty();
    }
 
    public static final class Line extends Column {
@@ -89,16 +_,16 @@
       private Line() {
       }
 
-      public OptionalInt m_142011_() {
-         return OptionalInt.empty();
-      }
-
-      public OptionalInt m_142009_() {
-         return OptionalInt.empty();
-      }
-
-      public OptionalInt m_142030_() {
-         return OptionalInt.empty();
+      public java.util.OptionalLong getCeiling() {
+         return java.util.OptionalLong.empty();
+      }
+
+      public java.util.OptionalLong getFloor() {
+         return java.util.OptionalLong.empty();
+      }
+
+      public java.util.OptionalLong getHeight() {
+         return java.util.OptionalLong.empty();
       }
 
       public String toString() {
@@ -107,38 +_,38 @@
    }
 
    public static final class Range extends Column {
-      private final int f_158204_;
-      private final int f_158205_;
+      private final long f_158204_;
+      private final long f_158205_;
 
-      protected Range(int p_158207_, int p_158208_) {
+      protected Range(long p_158207_, long p_158208_) {
          this.f_158204_ = p_158207_;
          this.f_158205_ = p_158208_;
-         if (this.m_158214_() < 0) {
+         if (this.height() < 0) {
             throw new IllegalArgumentException("Column of negative height: " + this);
          }
       }
 
-      public OptionalInt m_142011_() {
-         return OptionalInt.of(this.f_158205_);
-      }
-
-      public OptionalInt m_142009_() {
-         return OptionalInt.of(this.f_158204_);
-      }
-
-      public OptionalInt m_142030_() {
-         return OptionalInt.of(this.m_158214_());
-      }
-
-      public int m_158212_() {
+      public java.util.OptionalLong getCeiling() {
+         return java.util.OptionalLong.of(this.f_158205_);
+      }
+
+      public java.util.OptionalLong getFloor() {
+         return java.util.OptionalLong.of(this.f_158204_);
+      }
+
+      public java.util.OptionalLong getHeight() {
+         return java.util.OptionalLong.of(this.height());
+      }
+
+      public long ceiling() {
          return this.f_158205_;
       }
 
-      public int m_158213_() {
+      public long floor() {
          return this.f_158204_;
       }
 
-      public int m_158214_() {
+      public long height() {
          return this.f_158205_ - this.f_158204_ - 1;
       }
 
@@ -148,24 +_,24 @@
    }
 
    public static final class Ray extends Column {
-      private final int f_158216_;
+      private final long f_158216_;
       private final boolean f_158217_;
 
-      public Ray(int p_158219_, boolean p_158220_) {
+      public Ray(long p_158219_, boolean p_158220_) {
          this.f_158216_ = p_158219_;
          this.f_158217_ = p_158220_;
       }
 
-      public OptionalInt m_142011_() {
-         return this.f_158217_ ? OptionalInt.empty() : OptionalInt.of(this.f_158216_);
-      }
-
-      public OptionalInt m_142009_() {
-         return this.f_158217_ ? OptionalInt.of(this.f_158216_) : OptionalInt.empty();
-      }
-
-      public OptionalInt m_142030_() {
-         return OptionalInt.empty();
+      public java.util.OptionalLong getCeiling() {
+         return this.f_158217_ ? java.util.OptionalLong.empty() : java.util.OptionalLong.of(this.f_158216_);
+      }
+
+      public java.util.OptionalLong getFloor() {
+         return this.f_158217_ ? java.util.OptionalLong.of(this.f_158216_) : java.util.OptionalLong.empty();
+      }
+
+      public java.util.OptionalLong getHeight() {
+         return java.util.OptionalLong.empty();
       }
 
       public String toString() {
