--- a/net/minecraft/world/level/levelgen/DebugLevelSource.java
+++ b/net/minecraft/world/level/levelgen/DebugLevelSource.java
@@ -58,16 +_,16 @@
    public void m_213609_(WorldGenLevel p_223983_, ChunkAccess p_223984_, StructureManager p_223985_) {
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
       ChunkPos chunkpos = p_223984_.m_7697_();
-      int i = chunkpos.f_45578_;
-      int j = chunkpos.f_45579_;
+      long i = chunkpos.x();
+      long j = chunkpos.z();
 
       for(int k = 0; k < 16; ++k) {
          for(int l = 0; l < 16; ++l) {
-            int i1 = SectionPos.m_175554_(i, k);
-            int j1 = SectionPos.m_175554_(j, l);
-            p_223983_.m_7731_(blockpos$mutableblockpos.m_122178_(i1, 60, j1), f_64113_, 2);
-            BlockState blockstate = m_64148_(i1, j1);
-            p_223983_.m_7731_(blockpos$mutableblockpos.m_122178_(i1, 70, j1), blockstate, 2);
+            long i1 = SectionPos.sectionToBlockCoord(i, k);
+            long j1 = SectionPos.sectionToBlockCoord(j, l);
+            p_223983_.m_7731_(blockpos$mutableblockpos.set(i1, 60, j1), f_64113_, 2);
+            BlockState blockstate = getBlockStateFor(i1, j1);
+            p_223983_.m_7731_(blockpos$mutableblockpos.set(i1, 70, j1), blockstate, 2);
          }
       }
 
@@ -77,24 +_,24 @@
       return CompletableFuture.completedFuture(p_223995_);
    }
 
-   public int m_214096_(int p_223964_, int p_223965_, Heightmap.Types p_223966_, LevelHeightAccessor p_223967_, RandomState p_223968_) {
+   public long getBaseHeight(me.alphamode.mclong.math.BigInteger p_223964_, me.alphamode.mclong.math.BigInteger p_223965_, Heightmap.Types p_223966_, LevelHeightAccessor p_223967_, RandomState p_223968_) {
       return 0;
    }
 
-   public NoiseColumn m_214184_(int p_223959_, int p_223960_, LevelHeightAccessor p_223961_, RandomState p_223962_) {
+   public NoiseColumn getBaseColumn(me.alphamode.mclong.math.BigInteger p_223959_, me.alphamode.mclong.math.BigInteger p_223960_, LevelHeightAccessor p_223961_, RandomState p_223962_) {
       return new NoiseColumn(0, new BlockState[0]);
    }
 
    public void m_213600_(List<String> p_223987_, RandomState p_223988_, BlockPos p_223989_) {
    }
 
-   public static BlockState m_64148_(int p_64149_, int p_64150_) {
+   public static BlockState getBlockStateFor(long p_64149_, long p_64150_) {
       BlockState blockstate = f_64112_;
       if (p_64149_ > 0 && p_64150_ > 0 && p_64149_ % 2 != 0 && p_64150_ % 2 != 0) {
          p_64149_ /= 2;
          p_64150_ /= 2;
          if (p_64149_ <= f_64115_ && p_64150_ <= f_64116_) {
-            int i = Mth.m_14040_(p_64149_ * f_64115_ + p_64150_);
+            int i = (int) Mth.abs(p_64149_ * f_64115_ + p_64150_);
             if (i < f_64114_.size()) {
                blockstate = f_64114_.get(i);
             }
@@ -110,15 +_,15 @@
    public void m_6929_(WorldGenRegion p_188511_) {
    }
 
-   public int m_142062_() {
+   public long getMinY() {
       return 0;
    }
 
-   public int m_6331_() {
+   public long getGenDepth() {
       return 384;
    }
 
-   public int m_6337_() {
+   public long getSeaLevel() {
       return 63;
    }
 }
