--- a/net/minecraft/world/level/levelgen/DensityFunction.java
+++ b/net/minecraft/world/level/levelgen/DensityFunction.java
@@ -67,11 +_,11 @@
    }
 
    public interface FunctionContext {
-      int m_207115_();
-
-      int m_207114_();
-
-      int m_207113_();
+      me.alphamode.mclong.math.BigInteger blockX();
+
+      me.alphamode.mclong.math.BigInteger blockY();
+
+      me.alphamode.mclong.math.BigInteger blockZ();
 
       default Blender m_188743_() {
          return Blender.m_190153_();
@@ -106,16 +_,16 @@
       }
    }
 
-   public static record SinglePointContext(int f_208243_, int f_208244_, int f_208245_) implements DensityFunction.FunctionContext {
-      public int m_207115_() {
+   public static record SinglePointContext(me.alphamode.mclong.math.BigInteger f_208243_, me.alphamode.mclong.math.BigInteger f_208244_, me.alphamode.mclong.math.BigInteger f_208245_) implements DensityFunction.FunctionContext {
+      public me.alphamode.mclong.math.BigInteger blockX() {
          return this.f_208243_;
       }
 
-      public int m_207114_() {
+      public me.alphamode.mclong.math.BigInteger blockY() {
          return this.f_208244_;
       }
 
-      public int m_207113_() {
+      public me.alphamode.mclong.math.BigInteger blockZ() {
          return this.f_208245_;
       }
    }
