--- a/net/minecraft/world/level/levelgen/FlatLevelSource.java
+++ b/net/minecraft/world/level/levelgen/FlatLevelSource.java
@@ -57,8 +_,8 @@
    public void m_214194_(WorldGenRegion p_224174_, StructureManager p_224175_, RandomState p_224176_, ChunkAccess p_224177_) {
    }
 
-   public int m_142051_(LevelHeightAccessor p_158279_) {
-      return p_158279_.m_141937_() + Math.min(p_158279_.m_141928_(), this.f_64165_.m_161917_().size());
+   public long getSpawnHeight(LevelHeightAccessor p_158279_) {
+      return p_158279_.getMinBuildHeight() + Math.min(p_158279_.getHeight(), this.f_64165_.m_161917_().size());
    }
 
    public CompletableFuture<ChunkAccess> m_213974_(Executor p_224183_, Blender p_224184_, RandomState p_224185_, StructureManager p_224186_, ChunkAccess p_224187_) {
@@ -67,16 +_,19 @@
       Heightmap heightmap = p_224187_.m_6005_(Heightmap.Types.OCEAN_FLOOR_WG);
       Heightmap heightmap1 = p_224187_.m_6005_(Heightmap.Types.WORLD_SURFACE_WG);
 
-      for(int i = 0; i < Math.min(p_224187_.m_141928_(), list.size()); ++i) {
+      for(int i = 0; i < Math.min(p_224187_.getHeight(), list.size()); ++i) {
          BlockState blockstate = list.get(i);
          if (blockstate != null) {
-            int j = p_224187_.m_141937_() + i;
+            long j = p_224187_.getMinBuildHeight() + i;
 
             for(int k = 0; k < 16; ++k) {
                for(int l = 0; l < 16; ++l) {
-                  p_224187_.m_6978_(blockpos$mutableblockpos.m_122178_(k, j, l), blockstate, false);
-                  heightmap.m_64249_(k, j, l, blockstate);
-                  heightmap1.m_64249_(k, j, l, blockstate);
+                  var x = me.alphamode.mclong.math.BigInteger.valueOf(k);
+                  var y = me.alphamode.mclong.math.BigInteger.valueOf(j);
+                  var z = me.alphamode.mclong.math.BigInteger.valueOf(l);
+                  p_224187_.m_6978_(blockpos$mutableblockpos.set(x, y, z), blockstate, false);
+                  heightmap.update(x, y, z, blockstate);
+                  heightmap1.update(x, y, z, blockstate);
                }
             }
          }
@@ -85,21 +_,21 @@
       return CompletableFuture.completedFuture(p_224187_);
    }
 
-   public int m_214096_(int p_224160_, int p_224161_, Heightmap.Types p_224162_, LevelHeightAccessor p_224163_, RandomState p_224164_) {
+   public long getBaseHeight(me.alphamode.mclong.math.BigInteger p_224160_, me.alphamode.mclong.math.BigInteger p_224161_, Heightmap.Types p_224162_, LevelHeightAccessor p_224163_, RandomState p_224164_) {
       List<BlockState> list = this.f_64165_.m_161917_();
 
-      for(int i = Math.min(list.size(), p_224163_.m_151558_()) - 1; i >= 0; --i) {
-         BlockState blockstate = list.get(i);
+      for(long i = Math.min(list.size(), p_224163_.getMaxBuildHeight()) - 1; i >= 0; --i) {
+         BlockState blockstate = list.get((int) i);
          if (blockstate != null && p_224162_.m_64299_().test(blockstate)) {
-            return p_224163_.m_141937_() + i + 1;
+            return p_224163_.getMinBuildHeight() + i + 1;
          }
       }
 
-      return p_224163_.m_141937_();
+      return p_224163_.getMinBuildHeight();
    }
 
-   public NoiseColumn m_214184_(int p_224155_, int p_224156_, LevelHeightAccessor p_224157_, RandomState p_224158_) {
-      return new NoiseColumn(p_224157_.m_141937_(), this.f_64165_.m_161917_().stream().limit((long)p_224157_.m_141928_()).map((p_204549_) -> {
+   public NoiseColumn getBaseColumn(me.alphamode.mclong.math.BigInteger p_224155_, me.alphamode.mclong.math.BigInteger p_224156_, LevelHeightAccessor p_224157_, RandomState p_224158_) {
+      return new NoiseColumn(p_224157_.getMinBuildHeight(), this.f_64165_.m_161917_().stream().limit((long)p_224157_.getHeight()).map((p_204549_) -> {
          return p_204549_ == null ? Blocks.f_50016_.m_49966_() : p_204549_;
       }).toArray((p_204543_) -> {
          return new BlockState[p_204543_];
@@ -115,15 +_,15 @@
    public void m_6929_(WorldGenRegion p_188545_) {
    }
 
-   public int m_142062_() {
+   public long getMinY() {
       return 0;
    }
 
-   public int m_6331_() {
+   public long getGenDepth() {
       return 384;
    }
 
-   public int m_6337_() {
+   public long getSeaLevel() {
       return -63;
    }
 }
