--- a/net/minecraft/world/level/levelgen/LegacyRandomSource.java
+++ b/net/minecraft/world/level/levelgen/LegacyRandomSource.java
@@ -56,7 +_,13 @@
       }
 
       public RandomSource m_213715_(int p_224198_, int p_224199_, int p_224200_) {
-         long i = Mth.m_14130_(p_224198_, p_224199_, p_224200_);
+         long i = Mth.getSeed(p_224198_, p_224199_, p_224200_);
+         long j = i ^ this.f_188586_;
+         return new LegacyRandomSource(j);
+      }
+
+      public RandomSource at(long pX, long pY, long pZ) {
+         long i = Mth.getSeed(pX, pY, pZ);
          long j = i ^ this.f_188586_;
          return new LegacyRandomSource(j);
       }
