--- a/net/minecraft/world/level/levelgen/NoiseRouterData.java
+++ b/net/minecraft/world/level/levelgen/NoiseRouterData.java
@@ -101,7 +_,7 @@
       DensityFunctions.Spline.Coordinate densityfunctions$spline$coordinate3 = new DensityFunctions.Spline.Coordinate(p_256393_.m_255043_(f_224429_));
       DensityFunction densityfunction = m_255105_(p_256336_, p_224479_, m_224453_(DensityFunctions.m_208293_(DensityFunctions.m_208264_((double)-0.50375F), DensityFunctions.m_224020_(TerrainProvider.m_236635_(densityfunctions$spline$coordinate, densityfunctions$spline$coordinate1, densityfunctions$spline$coordinate3, p_224484_))), DensityFunctions.m_208372_()));
       DensityFunction densityfunction1 = m_255105_(p_256336_, p_224480_, m_224453_(DensityFunctions.m_224020_(TerrainProvider.m_236629_(densityfunctions$spline$coordinate, densityfunctions$spline$coordinate1, densityfunctions$spline$coordinate2, densityfunctions$spline$coordinate3, p_224484_)), f_209444_));
-      DensityFunction densityfunction2 = m_255105_(p_256336_, p_224482_, DensityFunctions.m_208293_(DensityFunctions.m_208266_(-64, 320, 1.5D, -1.5D), densityfunction));
+      DensityFunction densityfunction2 = m_255105_(p_256336_, p_224482_, DensityFunctions.m_208293_(DensityFunctions.m_208266_(net.minecraft.world.level.dimension.DimensionDefaults.f_156634_, 320, 1.5D, -1.5D), densityfunction));
       DensityFunction densityfunction3 = m_255105_(p_256336_, p_224481_, m_224453_(DensityFunctions.m_224020_(TerrainProvider.m_236642_(densityfunctions$spline$coordinate, densityfunctions$spline$coordinate1, densityfunctions$spline$coordinate2, densityfunctions$spline$coordinate3, p_224484_)), f_209445_));
       DensityFunction densityfunction4 = DensityFunctions.m_208363_(densityfunction3, p_224476_.m_208232_());
       DensityFunction densityfunction5 = m_212271_(densityfunction1, DensityFunctions.m_208293_(densityfunction2, densityfunction4));
@@ -171,7 +_,7 @@
       DensityFunction densityfunction1 = DensityFunctions.m_208315_(densityfunction, p_255650_.m_255043_(Noises.f_189294_), DensityFunctions.WeirdScaledSampler.RarityValueMapper.TYPE2);
       DensityFunction densityfunction2 = DensityFunctions.m_208331_(p_255650_.m_255043_(Noises.f_189295_), 0.0D, (double)Math.floorDiv(-64, 8), 8.0D);
       DensityFunction densityfunction3 = m_255403_(p_256535_, f_209438_);
-      DensityFunction densityfunction4 = DensityFunctions.m_208293_(densityfunction2, DensityFunctions.m_208266_(-64, 320, 8.0D, -40.0D)).m_208229_();
+      DensityFunction densityfunction4 = DensityFunctions.m_208293_(densityfunction2, DensityFunctions.m_208266_(net.minecraft.world.level.dimension.DimensionDefaults.f_156634_, 320, 8.0D, -40.0D)).m_208229_();
       DensityFunction densityfunction5 = DensityFunctions.m_208293_(densityfunction4, densityfunction3).m_208231_();
       double d0 = 0.083D;
       DensityFunction densityfunction6 = DensityFunctions.m_208293_(densityfunction1, DensityFunctions.m_208363_(DensityFunctions.m_208264_(0.083D), densityfunction3));
@@ -226,7 +_,7 @@
       DensityFunction densityfunction18 = m_209471_(densityfunction15, DensityFunctions.m_208368_(p_256005_.m_255043_(Noises.f_189249_), 4.0D, 4.0D), i, j, 0).m_208229_();
       DensityFunction densityfunction19 = DensityFunctions.m_208293_(DensityFunctions.m_208264_((double)-0.08F), DensityFunctions.m_208382_(densityfunction17, densityfunction18));
       DensityFunction densityfunction20 = DensityFunctions.m_208322_(p_256005_.m_255043_(Noises.f_189250_));
-      return new NoiseRouter(densityfunction, densityfunction1, densityfunction2, densityfunction3, densityfunction6, densityfunction7, m_255403_(p_255681_, p_255649_ ? f_209457_ : f_209451_), m_255403_(p_255681_, p_255649_ ? f_209458_ : f_209452_), densityfunction9, m_255403_(p_255681_, f_209453_), m_224489_(p_255617_, DensityFunctions.m_208293_(densityfunction10, DensityFunctions.m_208264_(-0.703125D)).m_208220_(-64.0D, 64.0D)), densityfunction14, densityfunction16, densityfunction19, densityfunction20);
+      return new NoiseRouter(densityfunction, densityfunction1, densityfunction2, densityfunction3, densityfunction6, densityfunction7, m_255403_(p_255681_, p_255649_ ? f_209457_ : f_209451_), m_255403_(p_255681_, p_255649_ ? f_209458_ : f_209452_), densityfunction9, m_255403_(p_255681_, f_209453_), m_224489_(p_255617_, DensityFunctions.m_208293_(densityfunction10, DensityFunctions.m_208264_(-0.703125D)).m_208220_(-800.0D, 64.0D)), densityfunction14, densityfunction16, densityfunction19, densityfunction20);
    }
 
    private static NoiseRouter m_254999_(HolderGetter<DensityFunction> p_255724_, HolderGetter<NormalNoise.NoiseParameters> p_255986_, DensityFunction p_256378_) {
@@ -239,7 +_,7 @@
    }
 
    private static DensityFunction m_224489_(boolean p_224490_, DensityFunction p_224491_) {
-      return m_224443_(p_224491_, -64, 384, p_224490_ ? 16 : 80, p_224490_ ? 0 : 64, -0.078125D, 0, 24, p_224490_ ? 0.4D : 0.1171875D);
+      return m_224443_(p_224491_, net.minecraft.world.level.dimension.DimensionDefaults.f_156634_, net.minecraft.world.level.dimension.DimensionDefaults.f_156636_, p_224490_ ? 16 : 80, p_224490_ ? 0 : 64, -0.078125D, 0, 24, p_224490_ ? 0.4D : 0.1171875D);
    }
 
    private static DensityFunction m_254915_(HolderGetter<DensityFunction> p_256084_, int p_255802_, int p_255834_) {
@@ -255,7 +_,7 @@
    }
 
    protected static NoiseRouter m_255020_(HolderGetter<DensityFunction> p_256088_, HolderGetter<NormalNoise.NoiseParameters> p_255675_) {
-      return m_254999_(p_256088_, p_255675_, m_254915_(p_256088_, -64, 192));
+      return m_254999_(p_256088_, p_255675_, m_254915_(p_256088_, net.minecraft.world.level.dimension.DimensionDefaults.f_156634_, 192));
    }
 
    protected static NoiseRouter m_254860_(HolderGetter<DensityFunction> p_256633_, HolderGetter<NormalNoise.NoiseParameters> p_255902_) {
