--- a/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
+++ b/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
@@ -29,7 +_,7 @@
    public static final ResourceKey<NoiseGeneratorSettings> f_64437_ = ResourceKey.m_135785_(Registries.f_256932_, new ResourceLocation("floating_islands"));
 
    public boolean m_158567_() {
-      return this.f_158533_;
+      return false;//this.aquifersEnabled;
    }
 
    public boolean m_209369_() {
