--- a/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
+++ b/net/minecraft/world/level/levelgen/NoiseGeneratorSettings.java
@@ -67,7 +_,7 @@
    }
 
    private static NoiseGeneratorSettings m_255230_(BootstapContext<?> p_255885_) {
-      return new NoiseGeneratorSettings(NoiseSettings.f_209633_, Blocks.f_50069_.m_49966_(), Blocks.f_49990_.m_49966_(), NoiseRouterData.m_254860_(p_255885_.m_255420_(Registries.f_257040_), p_255885_.m_255420_(Registries.f_256865_)), SurfaceRuleData.m_198380_(false, false, false), List.of(), -64, false, false, false, true);
+      return new NoiseGeneratorSettings(NoiseSettings.f_209633_, Blocks.f_50069_.m_49966_(), Blocks.f_49990_.m_49966_(), NoiseRouterData.m_254860_(p_255885_.m_255420_(Registries.f_257040_), p_255885_.m_255420_(Registries.f_256865_)), SurfaceRuleData.m_198380_(false, false, false), List.of(), net.minecraft.world.level.dimension.DimensionDefaults.f_156634_, false, false, false, true);
    }
 
    public static NoiseGeneratorSettings m_238396_() {
