--- a/net/minecraft/world/level/levelgen/XoroshiroRandomSource.java
+++ b/net/minecraft/world/level/levelgen/XoroshiroRandomSource.java
@@ -72,6 +_,25 @@
       return this.f_190099_.m_190096_();
    }
 
+   public long nextLong(long bound) {
+      if (bound <= 0) {
+         throw new IllegalArgumentException("Bound must be positive");
+      } else {
+         long i = Integer.toUnsignedLong(this.m_188502_());
+         long j = i * bound;
+         long k = j & 4294967295L;
+         if (k < bound) {
+            for(long l = Long.remainderUnsigned(~bound + 1, bound); k < l; k = j & 4294967295L) {
+               i = Integer.toUnsignedLong(this.m_188502_());
+               j = i * bound;
+            }
+         }
+
+         long i1 = j >> 32;
+         return i1;
+      }
+   }
+
    public boolean m_188499_() {
       return (this.f_190099_.m_190096_() & 1L) != 0L;
    }
@@ -109,11 +_,16 @@
       }
 
       public RandomSource m_213715_(int p_224691_, int p_224692_, int p_224693_) {
-         long i = Mth.m_14130_(p_224691_, p_224692_, p_224693_);
+         long i = Mth.getSeed(p_224691_, p_224692_, p_224693_);
          long j = i ^ this.f_190123_;
          return new XoroshiroRandomSource(j, this.f_190124_);
       }
 
+      public RandomSource at(long p_224691_, long p_224692_, long p_224693_) {
+         long i = Mth.getSeed(p_224691_, p_224692_, p_224693_);
+         long j = i ^ this.f_190123_;
+         return new XoroshiroRandomSource(j, this.f_190124_);
+      }
       public RandomSource m_214111_(String p_224695_) {
          RandomSupport.Seed128bit randomsupport$seed128bit = RandomSupport.m_288212_(p_224695_);
          return new XoroshiroRandomSource(randomsupport$seed128bit.m_288194_(this.f_190123_, this.f_190124_));
