--- a/net/minecraft/world/entity/animal/sniffer/Sniffer.java
+++ b/net/minecraft/world/entity/animal/sniffer/Sniffer.java
@@ -232,13 +_,13 @@
    Optional<BlockPos> m_271905_() {
       return IntStream.range(0, 5).mapToObj((p_273771_) -> {
          return LandRandomPos.m_148488_(this, 10 + 2 * p_273771_, 3);
-      }).filter(Objects::nonNull).map(BlockPos::m_274446_).filter((p_289451_) -> {
+      }).filter(Objects::nonNull).map(BlockPos::containing).filter((p_289451_) -> {
          return this.m_9236_().m_6857_().m_61937_(p_289451_);
-      }).map(BlockPos::m_7495_).filter(this::m_271898_).findFirst();
+      }).map(BlockPos::below).filter(this::m_271898_).findFirst();
    }
 
    boolean m_272270_() {
-      return !this.m_272223_() && !this.m_278663_() && !this.m_6162_() && !this.m_20069_() && this.m_20096_() && !this.m_20159_() && this.m_271898_(this.m_284345_().m_7495_());
+      return !this.m_272223_() && !this.m_278663_() && !this.m_6162_() && !this.m_20069_() && this.m_20096_() && !this.m_20159_() && this.m_271898_(this.m_284345_().below());
    }
 
    private boolean m_271898_(BlockPos p_272757_) {
@@ -256,7 +_,7 @@
          BlockPos blockpos = this.m_284345_();
 
          for(ItemStack itemstack : list) {
-            ItemEntity itementity = new ItemEntity(serverlevel, (double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), itemstack);
+            ItemEntity itementity = new ItemEntity(serverlevel, (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), itemstack);
             itementity.m_32060_();
             serverlevel.m_7967_(itementity);
          }
@@ -269,15 +_,15 @@
       boolean flag = p_273528_.m_216981_() > 1700L && p_273528_.m_216981_() < 6000L;
       if (flag) {
          BlockPos blockpos = this.m_284345_();
-         BlockState blockstate = this.m_9236_().m_8055_(blockpos.m_7495_());
+         BlockState blockstate = this.m_9236_().m_8055_(blockpos.below());
          if (blockstate.m_60799_() != RenderShape.INVISIBLE) {
             for(int i = 0; i < 30; ++i) {
-               Vec3 vec3 = Vec3.m_82512_(blockpos).m_82520_(0.0D, (double)-0.65F, 0.0D);
-               this.m_9236_().m_7106_(new BlockParticleOption(ParticleTypes.f_123794_, blockstate), vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, 0.0D, 0.0D, 0.0D);
+               Vec3 vec3 = Vec3.atCenterOf(blockpos).m_82520_(0.0D, (double)-0.65F, 0.0D);
+               this.m_9236_().addParticle(new BlockParticleOption(ParticleTypes.f_123794_, blockstate), vec3.getX(), vec3.getY(), vec3.getZ(), 0.0D, 0.0D, 0.0D);
             }
 
             if (this.f_19797_ % 10 == 0) {
-               this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), blockstate.m_60827_().m_56778_(), this.m_5720_(), 0.5F, 0.5F, false);
+               this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), blockstate.m_60827_().m_56778_(), this.m_5720_(), 0.5F, 0.5F, false);
             }
          }
       }
@@ -359,7 +_,7 @@
 
    private void m_272136_() {
       if (this.m_9236_().m_5776_() && this.f_19797_ % 20 == 0) {
-         this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_271418_, this.m_5720_(), 1.0F, 1.0F, false);
+         this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_271418_, this.m_5720_(), 1.0F, 1.0F, false);
       }
 
    }
