--- a/net/minecraft/world/entity/animal/horse/AbstractHorse.java
+++ b/net/minecraft/world/entity/animal/horse/AbstractHorse.java
@@ -275,7 +_,7 @@
       if (!this.m_20067_()) {
          SoundEvent soundevent = this.m_7872_();
          if (soundevent != null) {
-            this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), soundevent, this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
+            this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), soundevent, this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
          }
       }
 
@@ -373,7 +_,7 @@
 
    protected void m_7355_(BlockPos p_30584_, BlockState p_30585_) {
       if (!p_30585_.m_278721_()) {
-         BlockState blockstate = this.m_9236_().m_8055_(p_30584_.m_7494_());
+         BlockState blockstate = this.m_9236_().m_8055_(p_30584_.above());
          SoundType soundtype = p_30585_.m_60827_();
          if (blockstate.m_60713_(Blocks.f_50125_)) {
             soundtype = blockstate.m_60827_();
@@ -487,7 +_,7 @@
       }
 
       if (this.m_6162_() && i > 0) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123748_, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123748_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), 0.0D, 0.0D, 0.0D);
          if (!this.m_9236_().f_46443_) {
             this.m_146758_(i);
          }
@@ -558,7 +_,7 @@
          }
 
          if (this.m_7559_()) {
-            if (!this.m_30617_() && !this.m_20160_() && this.f_19796_.m_188503_(300) == 0 && this.m_9236_().m_8055_(this.m_20183_().m_7495_()).m_60713_(Blocks.f_50440_)) {
+            if (!this.m_30617_() && !this.m_20160_() && this.f_19796_.m_188503_(300) == 0 && this.m_9236_().m_8055_(this.m_20183_().below()).m_60713_(Blocks.f_50440_)) {
                this.m_30661_(true);
             }
 
@@ -574,7 +_,7 @@
 
    protected void m_7567_() {
       if (this.m_30623_() && this.m_6162_() && !this.m_30617_()) {
-         LivingEntity livingentity = this.m_9236_().m_45963_(AbstractHorse.class, f_30526_, this, this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_20191_().m_82400_(16.0D));
+         LivingEntity livingentity = this.m_9236_().getNearestEntity(AbstractHorse.class, f_30526_, this, this.getBigX(), this.getBigY(), this.getBigZ(), this.m_20191_().m_82400_(16.0D));
          if (livingentity != null && this.m_20280_(livingentity) > 4.0D) {
             this.f_21344_.m_6570_(livingentity, 0);
          }
@@ -741,7 +_,7 @@
       this.m_19915_(vec2.f_82471_, vec2.f_82470_);
       this.f_19859_ = this.f_20883_ = this.f_20885_ = this.m_146908_();
       if (this.m_6109_()) {
-         if (p_275693_.f_82481_ <= 0.0D) {
+         if (p_275693_.m_7094_() <= 0.0D) {
             this.f_30524_ = 0;
          }
 
@@ -783,10 +_,10 @@
       double d0 = this.m_30626_() * (double)p_248808_ * (double)this.m_20098_();
       double d1 = d0 + (double)this.m_285755_();
       Vec3 vec3 = this.m_20184_();
-      this.m_20334_(vec3.f_82479_, d1, vec3.f_82481_);
+      this.m_20334_(vec3.m_7096_(), d1, vec3.m_7094_());
       this.m_30655_(true);
       this.f_19812_ = true;
-      if (p_275435_.f_82481_ > 0.0D) {
+      if (p_275435_.m_7094_() > 0.0D) {
          float f = Mth.m_14031_(this.m_146908_() * ((float)Math.PI / 180F));
          float f1 = Mth.m_14089_(this.m_146908_() * ((float)Math.PI / 180F));
          this.m_20256_(this.m_20184_().m_82520_((double)(-0.4F * f * p_248808_), 0.0D, (double)(0.4F * f1 * p_248808_)));
@@ -939,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.02D;
          double d1 = this.f_19796_.m_188583_() * 0.02D;
          double d2 = this.f_19796_.m_188583_() * 0.02D;
-         this.m_9236_().m_7106_(particleoptions, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(particleoptions, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
@@ -962,7 +_,7 @@
          float f1 = Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F));
          float f2 = 0.7F * this.f_30514_;
          float f3 = 0.15F * this.f_30514_;
-         p_289558_.m_20372_(p_289569_, this.m_20185_() + (double)(f2 * f), this.m_20186_() + this.m_6048_() + p_289569_.m_6049_() + (double)f3, this.m_20189_() - (double)(f2 * f1));
+         p_289558_.accept(p_289569_, this.getBigX().add(me.alphamode.mclong.math.BigDecimal.valueOf((f2 * f))), this.getBigY().add(me.alphamode.mclong.math.BigDecimal.valueOf(this.m_6048_() + p_289569_.m_6049_() + (double)f3)), this.getBigZ().subtract(me.alphamode.mclong.math.BigDecimal.valueOf(f2 * f1)));
          if (p_289569_ instanceof LivingEntity) {
             ((LivingEntity)p_289569_).f_20883_ = this.f_20883_;
          }
@@ -1063,24 +_,24 @@
 
    @Nullable
    private Vec3 m_30561_(Vec3 p_30562_, LivingEntity p_30563_) {
-      double d0 = this.m_20185_() + p_30562_.f_82479_;
-      double d1 = this.m_20191_().f_82289_;
-      double d2 = this.m_20189_() + p_30562_.f_82481_;
+      double d0 = this.m_20185_() + p_30562_.m_7096_();
+      double d1 = this.m_20191_().f_82289_.doubleValue();
+      double d2 = this.m_20189_() + p_30562_.m_7094_();
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(Pose pose : p_30563_.m_7431_()) {
          blockpos$mutableblockpos.m_122169_(d0, d1, d2);
-         double d3 = this.m_20191_().f_82292_ + 0.75D;
+         double d3 = this.m_20191_().f_82292_.doubleValue() + 0.75D;
 
          while(true) {
             double d4 = this.m_9236_().m_45573_(blockpos$mutableblockpos);
-            if ((double)blockpos$mutableblockpos.m_123342_() + d4 > d3) {
+            if ((double)blockpos$mutableblockpos.getY() + d4 > d3) {
                break;
             }
 
             if (DismountHelper.m_38439_(d4)) {
                AABB aabb = p_30563_.m_21270_(pose);
-               Vec3 vec3 = new Vec3(d0, (double)blockpos$mutableblockpos.m_123342_() + d4, d2);
+               Vec3 vec3 = new Vec3(d0, (double)blockpos$mutableblockpos.getY() + d4, d2);
                if (DismountHelper.m_38456_(this.m_9236_(), p_30563_, aabb.m_82383_(vec3))) {
                   p_30563_.m_20124_(pose);
                   return vec3;
@@ -1088,7 +_,7 @@
             }
 
             blockpos$mutableblockpos.m_122173_(Direction.UP);
-            if (!((double)blockpos$mutableblockpos.m_123342_() < d3)) {
+            if (!((double)blockpos$mutableblockpos.getY() < d3)) {
                break;
             }
          }
