--- a/net/minecraft/world/entity/animal/horse/Llama.java
+++ b/net/minecraft/world/entity/animal/horse/Llama.java
@@ -158,7 +_,7 @@
          float f = Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F));
          float f1 = Mth.m_14031_(this.f_20883_ * ((float)Math.PI / 180F));
          float f2 = 0.3F;
-         p_289531_.m_20372_(p_30830_, this.m_20185_() + (double)(0.3F * f1), this.m_20186_() + this.m_6048_() + p_30830_.m_6049_(), this.m_20189_() - (double)(0.3F * f));
+         p_289531_.accept(p_30830_, this.getBigX().add(java.math.BigDecimal.valueOf((0.3F * f1))), this.getBigY().add(java.math.BigDecimal.valueOf(this.m_6048_() + p_30830_.m_6049_())), this.getBigZ().subtract(java.math.BigDecimal.valueOf((0.3F * f))));
       }
    }
 
