--- a/net/minecraft/world/entity/animal/horse/Llama.java
+++ b/net/minecraft/world/entity/animal/horse/Llama.java
@@ -158,7 +_,7 @@
          float f = Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F));
          float f1 = Mth.m_14031_(this.f_20883_ * ((float)Math.PI / 180F));
          float f2 = 0.3F;
-         p_289531_.m_20372_(p_30830_, this.m_20185_() + (double)(0.3F * f1), this.m_20186_() + this.m_6048_() + p_30830_.m_6049_(), this.m_20189_() - (double)(0.3F * f));
+         p_289531_.accept(p_30830_, this.getBigX().add(me.alphamode.mclong.math.BigDecimal.valueOf((0.3F * f1))), this.getBigY().add(me.alphamode.mclong.math.BigDecimal.valueOf(this.m_6048_() + p_30830_.m_6049_())), this.getBigZ().subtract(me.alphamode.mclong.math.BigDecimal.valueOf((0.3F * f))));
       }
    }
 
@@ -200,7 +_,7 @@
       }
 
       if (this.m_6162_() && i > 0) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123748_, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123748_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), 0.0D, 0.0D, 0.0D);
          if (!this.m_9236_().f_46443_) {
             this.m_146758_(i);
          }
@@ -218,7 +_,7 @@
       if (flag && !this.m_20067_()) {
          SoundEvent soundevent = this.m_7872_();
          if (soundevent != null) {
-            this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_7872_(), this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
+            this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), this.m_7872_(), this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
          }
       }
 
@@ -370,7 +_,7 @@
       double d3 = Math.sqrt(d0 * d0 + d2 * d2) * (double)0.2F;
       llamaspit.m_6686_(d0, d1 + d3, d2, 1.5F, 10.0F);
       if (!this.m_20067_()) {
-         this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12098_, this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
+         this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12098_, this.m_5720_(), 1.0F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.2F);
       }
 
       this.m_9236_().m_7967_(llamaspit);
