--- a/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
+++ b/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
@@ -34,7 +_,7 @@
       this.f_30925_.m_146762_(0);
       LightningBolt lightningbolt = EntityType.f_20465_.m_20615_(serverlevel);
       if (lightningbolt != null) {
-         lightningbolt.m_6027_(this.f_30925_.m_20185_(), this.f_30925_.m_20186_(), this.f_30925_.m_20189_());
+         lightningbolt.moveTo(this.f_30925_.getBigX(), this.f_30925_.getBigY(), this.f_30925_.getBigZ());
          lightningbolt.m_20874_(true);
          serverlevel.m_7967_(lightningbolt);
          Skeleton skeleton = this.m_30931_(difficultyinstance, this.f_30925_);
