--- a/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
+++ b/net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
@@ -23,7 +_,7 @@
    }
 
    public boolean m_8036_() {
-      return this.f_30925_.m_9236_().m_45914_(this.f_30925_.m_20185_(), this.f_30925_.m_20186_(), this.f_30925_.m_20189_(), 10.0D);
+      return this.f_30925_.m_9236_().hasNearbyAlivePlayer(this.f_30925_.getBigX(), this.f_30925_.getBigY(), this.f_30925_.getBigZ(), 10.0D);
    }
 
    public void m_8037_() {
@@ -34,7 +_,7 @@
       this.f_30925_.m_146762_(0);
       LightningBolt lightningbolt = EntityType.f_20465_.m_20615_(serverlevel);
       if (lightningbolt != null) {
-         lightningbolt.m_6027_(this.f_30925_.m_20185_(), this.f_30925_.m_20186_(), this.f_30925_.m_20189_());
+         lightningbolt.moveTo(this.f_30925_.getBigX(), this.f_30925_.getBigY(), this.f_30925_.getBigZ());
          lightningbolt.m_20874_(true);
          serverlevel.m_7967_(lightningbolt);
          Skeleton skeleton = this.m_30931_(difficultyinstance, this.f_30925_);
@@ -63,7 +_,7 @@
       SkeletonHorse skeletonhorse = EntityType.f_20525_.m_20615_(this.f_30925_.m_9236_());
       if (skeletonhorse != null) {
          skeletonhorse.m_6518_((ServerLevel)this.f_30925_.m_9236_(), p_30930_, MobSpawnType.TRIGGERED, (SpawnGroupData)null, (CompoundTag)null);
-         skeletonhorse.m_6034_(this.f_30925_.m_20185_(), this.f_30925_.m_20186_(), this.f_30925_.m_20189_());
+         skeletonhorse.setPos(this.f_30925_.getBigX(), this.f_30925_.getBigY(), this.f_30925_.getBigZ());
          skeletonhorse.f_19802_ = 60;
          skeletonhorse.m_21530_();
          skeletonhorse.m_30651_(true);
@@ -78,7 +_,7 @@
       Skeleton skeleton = EntityType.f_20524_.m_20615_(p_30933_.m_9236_());
       if (skeleton != null) {
          skeleton.m_6518_((ServerLevel)p_30933_.m_9236_(), p_30932_, MobSpawnType.TRIGGERED, (SpawnGroupData)null, (CompoundTag)null);
-         skeleton.m_6034_(p_30933_.m_20185_(), p_30933_.m_20186_(), p_30933_.m_20189_());
+         skeleton.setPos(p_30933_.getBigX(), p_30933_.getBigY(), p_30933_.getBigZ());
          skeleton.f_19802_ = 60;
          skeleton.m_21530_();
          if (skeleton.m_6844_(EquipmentSlot.HEAD).m_41619_()) {
