--- a/net/minecraft/util/profiling/jfr/serialize/JfrResultJsonSerializer.java
+++ b/net/minecraft/util/profiling/jfr/serialize/JfrResultJsonSerializer.java
@@ -94,8 +_,8 @@
             jsonobject3.addProperty("level", p_185538_.f_185596_());
             jsonobject3.addProperty("chunkPosX", p_185538_.f_185593_().f_45578_);
             jsonobject3.addProperty("chunkPosZ", p_185538_.f_185593_().f_45579_);
-            jsonobject3.addProperty("worldPosX", p_185538_.f_185594_().f_140723_());
-            jsonobject3.addProperty("worldPosZ", p_185538_.f_185594_().f_140724_());
+            jsonobject3.addProperty("worldPosX", p_185538_.f_185594_().x());
+            jsonobject3.addProperty("worldPosZ", p_185538_.f_185594_().z());
             return jsonobject3;
          };
          jsonobject1.add("fastest", function.apply(timedstatsummary.f_185833_()));
