--- a/net/minecraft/client/renderer/blockentity/BannerRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BannerRenderer.java
@@ -90,7 +_,7 @@
       this.f_112046_.m_104301_(p_112054_, vertexconsumer, p_112056_, p_112057_);
       this.f_112047_.m_104301_(p_112054_, vertexconsumer, p_112056_, p_112057_);
       BlockPos blockpos = p_112052_.m_58899_();
-      float f2 = ((float)Math.floorMod((long)(blockpos.m_123341_() * 7 + blockpos.m_123342_() * 9 + blockpos.m_123343_() * 13) + i, 100L) + p_112053_) / 100.0F;
+      float f2 = ((float)Math.floorMod((long)(blockpos.getX() * 7 + blockpos.getY() * 9 + blockpos.getZ() * 13) + i, 100L) + p_112053_) / 100.0F;
       this.f_112045_.f_104203_ = (-0.0125F + 0.01F * Mth.m_14089_(((float)Math.PI * 2F) * f2)) * (float)Math.PI;
       this.f_112045_.f_104201_ = -32.0F;
       m_112065_(p_112054_, p_112055_, p_112056_, p_112057_, this.f_112045_, ModelBakery.f_119224_, true, list);
