--- a/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
@@ -107,6 +_,6 @@
    }
 
    public boolean m_142756_(BeaconBlockEntity p_173531_, Vec3 p_173532_) {
-      return Vec3.m_82512_(p_173531_.m_58899_()).m_82542_(1.0D, 0.0D, 1.0D).m_82509_(p_173532_.m_82542_(1.0D, 0.0D, 1.0D), (double)this.m_142163_());
+      return Vec3.atCenterOf(p_173531_.m_58899_()).m_82542_(1.0D, 0.0D, 1.0D).m_82509_(p_173532_.m_82542_(1.0D, 0.0D, 1.0D), (double)this.m_142163_());
    }
 }
