--- a/net/minecraft/client/renderer/blockentity/BlockEntityRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BlockEntityRenderer.java
@@ -20,6 +_,6 @@
    }
 
    default boolean m_142756_(T p_173568_, Vec3 p_173569_) {
-      return Vec3.m_82512_(p_173568_.m_58899_()).m_82509_(p_173569_, (double)this.m_142163_());
+      return Vec3.atCenterOf(p_173568_.m_58899_()).m_82509_(p_173569_, (double)this.m_142163_());
    }
 }
