--- a/net/minecraft/client/renderer/blockentity/BrushableBlockRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BrushableBlockRenderer.java
@@ -38,7 +_,7 @@
                   boolean flag = direction == Direction.EAST || direction == Direction.WEST;
                   p_277490_.m_252781_(Axis.f_252436_.m_252977_((float)((flag ? 90 : 0) + 11)));
                   p_277490_.m_85841_(0.5F, 0.5F, 0.5F);
-                  int j = LevelRenderer.m_109537_(p_277712_.m_58904_(), p_277712_.m_58900_(), p_277712_.m_58899_().m_121945_(direction));
+                  int j = LevelRenderer.m_109537_(p_277712_.m_58904_(), p_277712_.m_58900_(), p_277712_.m_58899_().relative(direction));
                   this.f_276519_.m_269128_(itemstack, ItemDisplayContext.FIXED, j, OverlayTexture.f_118083_, p_277490_, p_278015_, p_277712_.m_58904_(), 0);
                   p_277490_.m_85849_();
                }
