--- a/net/minecraft/client/renderer/blockentity/SignRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/SignRenderer.java
@@ -127,7 +_,7 @@
             return true;
          } else {
             Entity entity = minecraft.m_91288_();
-            return entity != null && entity.m_20238_(Vec3.m_82512_(p_173642_.m_58899_())) < (double)f_173631_;
+            return entity != null && entity.m_20238_(Vec3.atCenterOf(p_173642_.m_58899_())) < (double)f_173631_;
          }
       }
    }
