--- a/net/minecraft/client/renderer/blockentity/SignRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/SignRenderer.java
@@ -152,7 +_,7 @@
       }
 
       float f = 0.015625F * this.m_278631_();
-      p_279133_.m_85837_(p_279280_.f_82479_, p_279280_.f_82480_, p_279280_.f_82481_);
+      p_279133_.m_85837_(p_279280_.m_7096_(), p_279280_.m_7098_(), p_279280_.m_7094_());
       p_279133_.m_85841_(f, -f, f);
    }
 
@@ -170,7 +_,7 @@
             return true;
          } else {
             Entity entity = minecraft.m_91288_();
-            return entity != null && entity.m_20238_(Vec3.m_82512_(p_277741_)) < (double)f_173631_;
+            return entity != null && entity.m_20238_(Vec3.atCenterOf(p_277741_)) < (double)f_173631_;
          }
       }
    }
