--- a/net/minecraft/client/renderer/blockentity/SuspiciousSandRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/SuspiciousSandRenderer.java
@@ -38,7 +_,7 @@
                   boolean flag = direction == Direction.EAST || direction == Direction.WEST;
                   p_273698_.m_252781_(Axis.f_252436_.m_252977_((float)((flag ? 90 : 0) + 11)));
                   p_273698_.m_85841_(0.5F, 0.5F, 0.5F);
-                  int j = LevelRenderer.m_109537_(p_272652_.m_58904_(), p_272652_.m_58900_(), p_272652_.m_58899_().m_121945_(direction));
+                  int j = LevelRenderer.m_109537_(p_272652_.m_58904_(), p_272652_.m_58900_(), p_272652_.m_58899_().relative(direction));
                   this.f_271096_.m_269128_(itemstack, ItemDisplayContext.FIXED, j, OverlayTexture.f_118083_, p_273698_, p_273630_, p_272652_.m_58904_(), 0);
                   p_273698_.m_85849_();
                }
