--- a/net/minecraft/client/renderer/blockentity/TheEndGatewayRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/TheEndGatewayRenderer.java
@@ -21,7 +_,7 @@
    public void m_6922_(TheEndGatewayBlockEntity p_112613_, float p_112614_, PoseStack p_112615_, MultiBufferSource p_112616_, int p_112617_, int p_112618_) {
       if (p_112613_.m_59971_() || p_112613_.m_59972_()) {
          float f = p_112613_.m_59971_() ? p_112613_.m_59933_(p_112614_) : p_112613_.m_59967_(p_112614_);
-         double d0 = p_112613_.m_59971_() ? (double)p_112613_.m_58904_().m_151558_() : 50.0D;
+         double d0 = p_112613_.m_59971_() ? (double)p_112613_.m_58904_().getMaxBuildHeight() : 50.0D;
          f = Mth.m_14031_(f * (float)Math.PI);
          int i = Mth.m_14107_((double)f * d0);
          float[] afloat = p_112613_.m_59971_() ? DyeColor.MAGENTA.m_41068_() : DyeColor.PURPLE.m_41068_();
