--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -175,7 +_,7 @@
       this.f_36084_ = p_252153_;
       this.f_36095_ = new InventoryMenu(this.f_36093_, !p_250508_.f_46443_, this);
       this.f_36096_ = this.f_36095_;
-      this.m_7678_((double)p_250289_.m_123341_() + 0.5D, (double)(p_250289_.m_123342_() + 1), (double)p_250289_.m_123343_() + 0.5D, p_251702_, 0.0F);
+      this.m_7678_((double)p_250289_.getX() + 0.5D, (double)(p_250289_.getY() + 1), (double)p_250289_.getZ() + 0.5D, p_251702_, 0.0F);
       this.f_20896_ = 180.0F;
    }
 
@@ -258,10 +_,10 @@
       }
 
       int i = 29999999;
-      double d0 = Mth.m_14008_(this.m_20185_(), -2.9999999E7D, 2.9999999E7D);
-      double d1 = Mth.m_14008_(this.m_20189_(), -2.9999999E7D, 2.9999999E7D);
-      if (d0 != this.m_20185_() || d1 != this.m_20189_()) {
-         this.m_6034_(d0, this.m_20186_(), d1);
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigZ();
+      if (!d0.equals(this.getBigX()) || !d1.equals(this.getBigZ())) {
+         this.setPos(d0, this.getBigY(), d1);
       }
 
       ++this.f_20922_;
@@ -404,7 +_,7 @@
    }
 
    public void m_5496_(SoundEvent p_36137_, float p_36138_, float p_36139_) {
-      this.m_9236_().m_6263_(this, this.m_20185_(), this.m_20186_(), this.m_20189_(), p_36137_, this.m_5720_(), p_36138_, p_36139_);
+      this.m_9236_().playSound(this, this.getBigX(), this.getBigY(), this.getBigZ(), p_36137_, this.m_5720_(), p_36138_, p_36139_);
    }
 
    public void m_6330_(SoundEvent p_36140_, SoundSource p_36141_, float p_36142_, float p_36143_) {
@@ -438,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.02D;
          double d1 = this.f_19796_.m_188583_() * 0.02D;
          double d2 = this.f_19796_.m_188583_() * 0.02D;
-         this.m_9236_().m_7106_(p_36209_, this.m_20208_(1.0D), this.m_20187_() + 1.0D, this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(p_36209_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.math.BigDecimal.ONE), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
@@ -538,7 +_,7 @@
             return p_36280_ == EntityType.f_20508_;
          }).ifPresent((p_289488_) -> {
             if (!Parrot.m_29382_(this.m_9236_(), this)) {
-               this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), Parrot.m_218238_(this.m_9236_(), this.m_9236_().f_46441_), this.m_5720_(), 1.0F, Parrot.m_218236_(this.m_9236_().f_46441_));
+               this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), Parrot.m_218238_(this.m_9236_(), this.m_9236_().f_46441_), this.m_5720_(), 1.0F, Parrot.m_218236_(this.m_9236_().f_46441_));
             }
 
          });
@@ -998,9 +_,9 @@
    }
 
    protected Vec3 m_5763_(Vec3 p_36201_, MoverType p_36202_) {
-      if (!this.f_36077_.f_35935_ && p_36201_.f_82480_ <= 0.0D && (p_36202_ == MoverType.SELF || p_36202_ == MoverType.PLAYER) && this.m_36343_() && this.m_36386_()) {
-         double d0 = p_36201_.f_82479_;
-         double d1 = p_36201_.f_82481_;
+      if (!this.f_36077_.f_35935_ && p_36201_.m_7098_() <= 0.0D && (p_36202_ == MoverType.SELF || p_36202_ == MoverType.PLAYER) && this.m_36343_() && this.m_36386_()) {
+         double d0 = p_36201_.m_7096_();
+         double d1 = p_36201_.m_7094_();
          double d2 = 0.05D;
 
          while(d0 != 0.0D && this.m_9236_().m_45756_(this, this.m_20191_().m_82386_(d0, (double)(-this.m_274421_()), 0.0D))) {
@@ -1041,7 +_,7 @@
             }
          }
 
-         p_36201_ = new Vec3(d0, p_36201_.f_82480_, d1);
+         p_36201_ = new Vec3(d0, p_36201_.m_7098_(), d1);
       }
 
       return p_36201_;
@@ -1072,7 +_,7 @@
                int i = 0;
                i += EnchantmentHelper.m_44894_(this);
                if (this.m_20142_() && flag) {
-                  this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12314_, this.m_5720_(), 1.0F, 1.0F);
+                  this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12314_, this.m_5720_(), 1.0F, 1.0F);
                   ++i;
                   flag1 = true;
                }
@@ -1128,7 +_,7 @@
                         }
                      }
 
-                     this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12317_, this.m_5720_(), 1.0F, 1.0F);
+                     this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12317_, this.m_5720_(), 1.0F, 1.0F);
                      this.m_36346_();
                   }
 
@@ -1139,15 +_,15 @@
                   }
 
                   if (flag2) {
-                     this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12313_, this.m_5720_(), 1.0F, 1.0F);
+                     this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12313_, this.m_5720_(), 1.0F, 1.0F);
                      this.m_5704_(p_36347_);
                   }
 
                   if (!flag2 && !flag3) {
                      if (flag) {
-                        this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12316_, this.m_5720_(), 1.0F, 1.0F);
+                        this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12316_, this.m_5720_(), 1.0F, 1.0F);
                      } else {
-                        this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12318_, this.m_5720_(), 1.0F, 1.0F);
+                        this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12318_, this.m_5720_(), 1.0F, 1.0F);
                      }
                   }
 
@@ -1183,13 +_,13 @@
 
                      if (this.m_9236_() instanceof ServerLevel && f5 > 2.0F) {
                         int k = (int)((double)f5 * 0.5D);
-                        ((ServerLevel)this.m_9236_()).m_8767_(ParticleTypes.f_123798_, p_36347_.m_20185_(), p_36347_.m_20227_(0.5D), p_36347_.m_20189_(), k, 0.1D, 0.0D, 0.1D, 0.2D);
+                        ((ServerLevel)this.m_9236_()).sendParticles(ParticleTypes.f_123798_, p_36347_.getBigX(), p_36347_.getBigY(0.5D), p_36347_.getBigZ(), k, 0.1D, 0.0D, 0.1D, 0.2D);
                      }
                   }
 
                   this.m_36399_(0.1F);
                } else {
-                  this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12315_, this.m_5720_(), 1.0F, 1.0F);
+                  this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12315_, this.m_5720_(), 1.0F, 1.0F);
                   if (flag4) {
                      p_36347_.m_20095_();
                   }
@@ -1228,7 +_,7 @@
       double d0 = (double)(-Mth.m_14031_(this.m_146908_() * ((float)Math.PI / 180F)));
       double d1 = (double)Mth.m_14089_(this.m_146908_() * ((float)Math.PI / 180F));
       if (this.m_9236_() instanceof ServerLevel) {
-         ((ServerLevel)this.m_9236_()).m_8767_(ParticleTypes.f_123766_, this.m_20185_() + d0, this.m_20227_(0.5D), this.m_20189_() + d1, 0, d0, 0.0D, d1, 0.0D);
+         ((ServerLevel)this.m_9236_()).sendParticles(ParticleTypes.f_123766_, this.getBigX().add(d0), this.getBigY(0.5D), this.getBigZ().add(d1), 0, d0, 0.0D, d1, 0.0D);
       }
 
    }
@@ -1303,9 +_,9 @@
          return Optional.empty();
       } else {
          boolean flag = block.m_48673_(blockstate);
-         BlockState blockstate1 = p_36131_.m_8055_(p_36132_.m_7494_());
+         BlockState blockstate1 = p_36131_.m_8055_(p_36132_.above());
          boolean flag1 = blockstate1.m_60734_().m_48673_(blockstate1);
-         return flag && flag1 ? Optional.of(new Vec3((double)p_36132_.m_123341_() + 0.5D, (double)p_36132_.m_123342_() + 0.1D, (double)p_36132_.m_123343_() + 0.5D)) : Optional.empty();
+         return flag && flag1 ? Optional.of(new Vec3((double)p_36132_.getX() + 0.5D, (double)p_36132_.getY() + 0.1D, (double)p_36132_.getZ() + 0.5D)) : Optional.empty();
       }
    }
 
@@ -1364,30 +_,30 @@
    }
 
    public void m_7023_(Vec3 p_36359_) {
-      double d0 = this.m_20185_();
-      double d1 = this.m_20186_();
-      double d2 = this.m_20189_();
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigY();
+      me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ();
       if (this.m_6069_() && !this.m_20159_()) {
-         double d3 = this.m_20154_().f_82480_;
+         double d3 = this.m_20154_().m_7098_();
          double d4 = d3 < -0.2D ? 0.085D : 0.06D;
          if (d3 <= 0.0D || this.f_20899_ || !this.m_9236_().m_8055_(BlockPos.m_274561_(this.m_20185_(), this.m_20186_() + 1.0D - 0.1D, this.m_20189_())).m_60819_().m_76178_()) {
             Vec3 vec3 = this.m_20184_();
-            this.m_20256_(vec3.m_82520_(0.0D, (d3 - vec3.f_82480_) * d4, 0.0D));
+            this.m_20256_(vec3.m_82520_(0.0D, (d3 - vec3.m_7098_()) * d4, 0.0D));
          }
       }
 
       if (this.f_36077_.f_35935_ && !this.m_20159_()) {
-         double d5 = this.m_20184_().f_82480_;
+         double d5 = this.m_20184_().m_7098_();
          super.m_7023_(p_36359_);
          Vec3 vec31 = this.m_20184_();
-         this.m_20334_(vec31.f_82479_, d5 * 0.6D, vec31.f_82481_);
+         this.m_20334_(vec31.m_7096_(), d5 * 0.6D, vec31.m_7094_());
          this.m_183634_();
          this.m_20115_(7, false);
       } else {
          super.m_7023_(p_36359_);
       }
 
-      this.m_36378_(this.m_20185_() - d0, this.m_20186_() - d1, this.m_20189_() - d2);
+      this.m_36378_(this.getBigX().subtract(d0).doubleValue(), this.getBigY().subtract(d1).doubleValue(), this.getBigZ().subtract(d2).doubleValue());
    }
 
    public void m_5844_() {
@@ -1604,7 +_,7 @@
 
       if (p_36276_ > 0 && this.f_36078_ % 5 == 0 && (float)this.f_36111_ < (float)this.f_19797_ - 100.0F) {
          float f = this.f_36078_ > 30 ? 1.0F : (float)this.f_36078_ / 30.0F;
-         this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12275_, this.m_5720_(), f * 0.75F, 1.0F);
+         this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12275_, this.m_5720_(), f * 0.75F, 1.0F);
          this.f_36111_ = this.f_19797_;
       }
 
@@ -1651,7 +_,7 @@
       if (this.f_36077_.f_35938_) {
          return true;
       } else {
-         BlockPos blockpos = p_36205_.m_121945_(p_36206_.m_122424_());
+         BlockPos blockpos = p_36205_.relative(p_36206_.m_122424_());
          BlockInWorld blockinworld = new BlockInWorld(this.m_9236_(), blockpos, false);
          return p_36207_.m_204121_(this.m_9236_().m_9598_().m_175515_(Registries.f_256747_), blockinworld);
       }
@@ -1946,7 +_,7 @@
    public ItemStack m_5584_(Level p_36185_, ItemStack p_36186_) {
       this.m_36324_().m_38712_(p_36186_.m_41720_(), p_36186_);
       this.m_36246_(Stats.f_12982_.m_12902_(p_36186_.m_41720_()));
-      p_36185_.m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12321_, SoundSource.PLAYERS, 0.5F, p_36185_.f_46441_.m_188501_() * 0.1F + 0.9F);
+      p_36185_.playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12321_, SoundSource.PLAYERS, 0.5F, p_36185_.f_46441_.m_188501_() * 0.1F + 0.9F);
       if (this instanceof ServerPlayer) {
          CriteriaTriggers.f_10592_.m_23682_((ServerPlayer)this, p_36186_);
       }
@@ -1977,8 +_,8 @@
          double d2 = vec3.m_165925_();
          float f2;
          if (d1 > 0.0D && d2 > 0.0D) {
-            double d3 = (vec31.f_82479_ * vec3.f_82479_ + vec31.f_82481_ * vec3.f_82481_) / Math.sqrt(d1 * d2);
-            double d4 = vec31.f_82479_ * vec3.f_82481_ - vec31.f_82481_ * vec3.f_82479_;
+            double d3 = (vec31.m_7096_() * vec3.m_7096_() + vec31.m_7094_() * vec3.m_7094_()) / Math.sqrt(d1 * d2);
+            double d4 = vec31.m_7096_() * vec3.m_7094_() - vec31.m_7094_() * vec3.m_7096_();
             f2 = (float)(Math.signum(d4) * Math.acos(d3));
          } else {
             f2 = 0.0F;
