--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -170,7 +_,7 @@
          s1 = p_11262_.m_129523_().toString();
       }
 
-      f_11188_.info("{}[{}] logged in with entity id {} at ({}, {}, {})", p_11263_.m_7755_().getString(), s1, p_11263_.m_19879_(), p_11263_.m_20185_(), p_11263_.m_20186_(), p_11263_.m_20189_());
+      f_11188_.info("{}[{}] logged in with entity id {} at ({}, {}, {})", p_11263_.m_7755_().getString(), s1, p_11263_.m_19879_(), p_11263_.getBigX(), p_11263_.getBigY(), p_11263_.getBigZ());
       LevelData leveldata = serverlevel1.m_6106_();
       p_11263_.m_143427_(compoundtag);
       ServerGamePacketListenerImpl servergamepacketlistenerimpl = new ServerGamePacketListenerImpl(this.f_11195_, p_11262_, p_11263_);
@@ -198,7 +_,7 @@
       }
 
       this.m_240416_(mutablecomponent.m_130940_(ChatFormatting.YELLOW), false);
-      servergamepacketlistenerimpl.m_9774_(p_11263_.m_20185_(), p_11263_.m_20186_(), p_11263_.m_20189_(), p_11263_.m_146908_(), p_11263_.m_146909_());
+      servergamepacketlistenerimpl.teleport(p_11263_.getBigX(), p_11263_.getBigY(), p_11263_.getBigZ(), p_11263_.m_146908_(), p_11263_.m_146909_());
       ServerStatus serverstatus = this.f_11195_.m_129928_();
       if (serverstatus != null) {
          p_11263_.m_215109_(serverstatus);
@@ -449,25 +_,25 @@
          if (!blockstate.m_204336_(BlockTags.f_13038_) && !flag1) {
             f1 = f;
          } else {
-            Vec3 vec31 = Vec3.m_82539_(blockpos).m_82546_(vec3).m_82541_();
-            f1 = (float)Mth.m_14175_(Mth.m_14136_(vec31.f_82481_, vec31.f_82479_) * (double)(180F / (float)Math.PI) - 90.0D);
+            Vec3 vec31 = Vec3.atBottomCenterOf(blockpos).m_82546_(vec3).m_82541_();
+            f1 = (float)Mth.m_14175_(Mth.m_14136_(vec31.m_7094_(), vec31.m_7096_()) * (double)(180F / (float)Math.PI) - 90.0D);
          }
 
-         serverplayer.m_7678_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, f1, 0.0F);
+         serverplayer.moveTo(vec3.getX(), vec3.getY(), vec3.getZ(), f1, 0.0F);
          serverplayer.m_9158_(serverlevel1.m_46472_(), blockpos, f, flag, false);
          flag2 = !p_11238_ && flag1;
       } else if (blockpos != null) {
          serverplayer.f_8906_.m_9829_(new ClientboundGameEventPacket(ClientboundGameEventPacket.f_132153_, 0.0F));
       }
 
-      while(!serverlevel1.m_45786_(serverplayer) && serverplayer.m_20186_() < (double)serverlevel1.m_151558_()) {
+      while(!serverlevel1.m_45786_(serverplayer) && serverplayer.m_20186_() < (double)serverlevel1.getMaxBuildHeight()) {
          serverplayer.m_6034_(serverplayer.m_20185_(), serverplayer.m_20186_() + 1.0D, serverplayer.m_20189_());
       }
 
       byte b0 = (byte)(p_11238_ ? 1 : 0);
       LevelData leveldata = serverplayer.m_9236_().m_6106_();
       serverplayer.f_8906_.m_9829_(new ClientboundRespawnPacket(serverplayer.m_9236_().m_220362_(), serverplayer.m_9236_().m_46472_(), BiomeManager.m_47877_(serverplayer.m_284548_().m_7328_()), serverplayer.f_8941_.m_9290_(), serverplayer.f_8941_.m_9293_(), serverplayer.m_9236_().m_46659_(), serverplayer.m_284548_().m_8584_(), b0, serverplayer.m_219759_(), serverplayer.m_287157_()));
-      serverplayer.f_8906_.m_9774_(serverplayer.m_20185_(), serverplayer.m_20186_(), serverplayer.m_20189_(), serverplayer.m_146908_(), serverplayer.m_146909_());
+      serverplayer.f_8906_.teleport(serverplayer.getBigX(), serverplayer.getBigY(), serverplayer.getBigZ(), serverplayer.m_146908_(), serverplayer.m_146909_());
       serverplayer.f_8906_.m_9829_(new ClientboundSetDefaultSpawnPositionPacket(serverlevel1.m_220360_(), serverlevel1.m_220361_()));
       serverplayer.f_8906_.m_9829_(new ClientboundChangeDifficultyPacket(leveldata.m_5472_(), leveldata.m_5474_()));
       serverplayer.f_8906_.m_9829_(new ClientboundSetExperiencePacket(serverplayer.f_36080_, serverplayer.f_36079_, serverplayer.f_36078_));
@@ -479,7 +_,7 @@
       serverplayer.m_143429_();
       serverplayer.m_21153_(serverplayer.m_21223_());
       if (flag2) {
-         serverplayer.f_8906_.m_9829_(new ClientboundSoundPacket(SoundEvents.f_12377_, SoundSource.BLOCKS, (double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), 1.0F, 1.0F, serverlevel1.m_213780_().m_188505_()));
+         serverplayer.f_8906_.m_9829_(new ClientboundSoundPacket(SoundEvents.f_12377_, SoundSource.BLOCKS, blockpos.getBigDecX(), blockpos.getBigDecY(), blockpos.getBigDecZ(), 1.0F, 1.0F, serverlevel1.m_213780_().m_188505_()));
       }
 
       return serverplayer;
@@ -615,15 +_,15 @@
       return null;
    }
 
-   public void m_11241_(@Nullable Player p_11242_, double p_11243_, double p_11244_, double p_11245_, double p_11246_, ResourceKey<Level> p_11247_, Packet<?> p_11248_) {
+   public void broadcast(@Nullable Player pExcept, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pRadius, ResourceKey<Level> pDimension, Packet<?> pPacket) {
       for(int i = 0; i < this.f_11196_.size(); ++i) {
          ServerPlayer serverplayer = this.f_11196_.get(i);
-         if (serverplayer != p_11242_ && serverplayer.m_9236_().m_46472_() == p_11247_) {
-            double d0 = p_11243_ - serverplayer.m_20185_();
-            double d1 = p_11244_ - serverplayer.m_20186_();
-            double d2 = p_11245_ - serverplayer.m_20189_();
-            if (d0 * d0 + d1 * d1 + d2 * d2 < p_11246_ * p_11246_) {
-               serverplayer.f_8906_.m_9829_(p_11248_);
+         if (serverplayer != pExcept && serverplayer.m_9236_().m_46472_() == pDimension) {
+            me.alphamode.mclong.math.BigDecimal d0 = pX.subtract(serverplayer.getBigX());
+            me.alphamode.mclong.math.BigDecimal d1 = pY.subtract(serverplayer.getBigY());
+            me.alphamode.mclong.math.BigDecimal d2 = pZ.subtract(serverplayer.getBigZ());
+            if (d0.multiply(d0).add(d1.multiply(d1)).add(d2.multiply(d2)).compareTo(pRadius * pRadius) < 0) {
+               serverplayer.f_8906_.m_9829_(pPacket);
             }
          }
       }
