--- a/net/minecraft/world/phys/shapes/ArrayVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/ArrayVoxelShape.java
@@ -7,15 +_,15 @@
 import net.minecraft.core.Direction;
 
 public class ArrayVoxelShape extends VoxelShape {
-   private final DoubleList f_82563_;
-   private final DoubleList f_82564_;
-   private final DoubleList f_82565_;
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_82563_;
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_82564_;
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_82565_;
 
-   protected ArrayVoxelShape(DiscreteVoxelShape p_82572_, double[] p_82573_, double[] p_82574_, double[] p_82575_) {
-      this(p_82572_, (DoubleList)DoubleArrayList.wrap(Arrays.copyOf(p_82573_, p_82572_.m_82828_() + 1)), (DoubleList)DoubleArrayList.wrap(Arrays.copyOf(p_82574_, p_82572_.m_82845_() + 1)), (DoubleList)DoubleArrayList.wrap(Arrays.copyOf(p_82575_, p_82572_.m_82852_() + 1)));
+   protected ArrayVoxelShape(DiscreteVoxelShape p_82572_, me.alphamode.mclong.math.BigDecimal[] p_82573_, me.alphamode.mclong.math.BigDecimal[] p_82574_, me.alphamode.mclong.math.BigDecimal[] p_82575_) {
+      this(p_82572_, it.unimi.dsi.fastutil.objects.ObjectArrayList.wrap(Arrays.copyOf(p_82573_, p_82572_.m_82828_() + 1)), it.unimi.dsi.fastutil.objects.ObjectArrayList.wrap(Arrays.copyOf(p_82574_, p_82572_.m_82845_() + 1)), it.unimi.dsi.fastutil.objects.ObjectArrayList.wrap(Arrays.copyOf(p_82575_, p_82572_.m_82852_() + 1)));
    }
 
-   ArrayVoxelShape(DiscreteVoxelShape p_82567_, DoubleList p_82568_, DoubleList p_82569_, DoubleList p_82570_) {
+   ArrayVoxelShape(DiscreteVoxelShape p_82567_, it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_82568_, it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_82569_, it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_82570_) {
       super(p_82567_);
       int i = p_82567_.m_82828_() + 1;
       int j = p_82567_.m_82845_() + 1;
@@ -29,7 +_,7 @@
       }
    }
 
-   protected DoubleList m_7700_(Direction.Axis p_82577_) {
+   protected it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> getCoords(Direction.Axis p_82577_) {
       switch (p_82577_) {
          case X:
             return this.f_82563_;
