--- a/net/minecraft/world/phys/shapes/BitSetDiscreteVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/BitSetDiscreteVoxelShape.java
@@ -87,8 +_,8 @@
 
    }
 
-   public void m_142703_(int p_165987_, int p_165988_, int p_165989_) {
-      this.m_165942_(p_165987_, p_165988_, p_165989_, true);
+   public void fill(long pX, long pY, long pZ) {
+      this.m_165942_((int) pX, (int) pY, (int) pZ, true);
    }
 
    public boolean m_6224_() {
