--- a/net/minecraft/world/phys/shapes/CubePointRange.java
+++ b/net/minecraft/world/phys/shapes/CubePointRange.java
@@ -2,7 +_,7 @@
 
 import it.unimi.dsi.fastutil.doubles.AbstractDoubleList;
 
-public class CubePointRange extends AbstractDoubleList {
+public class CubePointRange extends it.unimi.dsi.fastutil.objects.AbstractObjectList<java.math.BigDecimal> {
    private final int f_82758_;
 
    CubePointRange(int p_82760_) {
@@ -13,8 +_,8 @@
       }
    }
 
-   public double getDouble(int p_82762_) {
-      return (double)p_82762_ / (double)this.f_82758_;
+   public java.math.BigDecimal get(int p_82762_) {
+      return java.math.BigDecimal.valueOf((double)p_82762_ / (double)this.f_82758_);
    }
 
    public int size() {
