--- a/net/minecraft/world/phys/shapes/CubeVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/CubeVoxelShape.java
@@ -9,12 +_,12 @@
       super(p_82765_);
    }
 
-   protected DoubleList m_7700_(Direction.Axis p_82767_) {
+   protected it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> getCoords(Direction.Axis p_82767_) {
       return new CubePointRange(this.f_83211_.m_82850_(p_82767_));
    }
 
-   protected int m_6595_(Direction.Axis p_82769_, double p_82770_) {
+   protected int findIndex(Direction.Axis p_82769_, me.alphamode.mclong.math.BigDecimal p_82770_) {
       int i = this.f_83211_.m_82850_(p_82769_);
-      return Mth.m_14107_(Mth.m_14008_(p_82770_ * (double)i, -1.0D, (double)i));
+      return Mth.m_14107_(Mth.m_14008_(p_82770_.doubleValue() * i, -1.0D, (double)i));
    }
 }
