--- a/net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
+++ b/net/minecraft/world/phys/shapes/DiscreteVoxelShape.java
@@ -37,7 +_,7 @@
 
    public abstract boolean m_6696_(int p_82829_, int p_82830_, int p_82831_);
 
-   public abstract void m_142703_(int p_165998_, int p_165999_, int p_166000_);
+   public abstract void fill(long p_165998_, long p_165999_, long p_166000_);
 
    public boolean m_6224_() {
       for(Direction.Axis direction$axis : f_82784_) {
