--- a/net/minecraft/world/phys/shapes/EntityCollisionContext.java
+++ b/net/minecraft/world/phys/shapes/EntityCollisionContext.java
@@ -54,7 +_,7 @@
    }
 
    public boolean m_6513_(VoxelShape p_82886_, BlockPos p_82887_, boolean p_82888_) {
-      return this.f_82867_ > (double)p_82887_.m_123342_() + p_82886_.m_83297_(Direction.Axis.Y) - (double)1.0E-5F;
+      return this.f_82867_ > (double)p_82887_.getY() + p_82886_.m_83297_(Direction.Axis.Y) - (double)1.0E-5F;
    }
 
    @Nullable
