--- a/net/minecraft/world/phys/shapes/IdenticalMerger.java
+++ b/net/minecraft/world/phys/shapes/IdenticalMerger.java
@@ -3,9 +_,9 @@
 import it.unimi.dsi.fastutil.doubles.DoubleList;
 
 public class IdenticalMerger implements IndexMerger {
-   private final DoubleList f_82901_;
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_82901_;
 
-   public IdenticalMerger(DoubleList p_82903_) {
+   public IdenticalMerger(it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_82903_) {
       this.f_82901_ = p_82903_;
    }
 
@@ -25,7 +_,7 @@
       return this.f_82901_.size();
    }
 
-   public DoubleList m_6241_() {
+   public it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> getList() {
       return this.f_82901_;
    }
 }
