--- a/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
+++ b/net/minecraft/world/phys/shapes/NonOverlappingMerger.java
@@ -3,12 +_,12 @@
 import it.unimi.dsi.fastutil.doubles.AbstractDoubleList;
 import it.unimi.dsi.fastutil.doubles.DoubleList;
 
-public class NonOverlappingMerger extends AbstractDoubleList implements IndexMerger {
-   private final DoubleList f_83008_;
-   private final DoubleList f_83009_;
+public class NonOverlappingMerger extends it.unimi.dsi.fastutil.objects.AbstractObjectList<me.alphamode.mclong.math.BigDecimal> implements IndexMerger {
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_83008_;
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_83009_;
    private final boolean f_83010_;
 
-   protected NonOverlappingMerger(DoubleList p_83012_, DoubleList p_83013_, boolean p_83014_) {
+   protected NonOverlappingMerger(it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_83012_, it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_83013_, boolean p_83014_) {
       this.f_83008_ = p_83012_;
       this.f_83009_ = p_83013_;
       this.f_83010_ = p_83014_;
@@ -44,11 +_,11 @@
       return true;
    }
 
-   public double getDouble(int p_83026_) {
-      return p_83026_ < this.f_83008_.size() ? this.f_83008_.getDouble(p_83026_) : this.f_83009_.getDouble(p_83026_ - this.f_83008_.size());
+   public me.alphamode.mclong.math.BigDecimal get(int p_83026_) {
+      return p_83026_ < this.f_83008_.size() ? this.f_83008_.get(p_83026_) : this.f_83009_.get(p_83026_ - this.f_83008_.size());
    }
 
-   public DoubleList m_6241_() {
+   public it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> getList() {
       return this;
    }
 }
