--- a/net/minecraft/world/phys/shapes/OffsetDoubleList.java
+++ b/net/minecraft/world/phys/shapes/OffsetDoubleList.java
@@ -3,17 +_,17 @@
 import it.unimi.dsi.fastutil.doubles.AbstractDoubleList;
 import it.unimi.dsi.fastutil.doubles.DoubleList;
 
-public class OffsetDoubleList extends AbstractDoubleList {
-   private final DoubleList f_83028_;
-   private final double f_83029_;
+public class OffsetDoubleList extends it.unimi.dsi.fastutil.objects.AbstractObjectList<me.alphamode.mclong.math.BigDecimal> {
+   private final it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> f_83028_;
+   private final me.alphamode.mclong.math.BigDecimal f_83029_;
 
-   public OffsetDoubleList(DoubleList p_83031_, double p_83032_) {
+   public OffsetDoubleList(it.unimi.dsi.fastutil.objects.ObjectList<me.alphamode.mclong.math.BigDecimal> p_83031_, me.alphamode.mclong.math.BigDecimal p_83032_) {
       this.f_83028_ = p_83031_;
       this.f_83029_ = p_83032_;
    }
 
-   public double getDouble(int p_83034_) {
-      return this.f_83028_.getDouble(p_83034_) + this.f_83029_;
+   public me.alphamode.mclong.math.BigDecimal get(int p_83034_) {
+      return this.f_83028_.get(p_83034_).add(this.f_83029_);
    }
 
    public int size() {
