--- a/net/minecraft/world/phys/shapes/Shapes.java
+++ b/net/minecraft/world/phys/shapes/Shapes.java
@@ -17,7 +_,7 @@
    public static final double f_166026_ = 1.0E-6D;
    private static final VoxelShape f_83037_ = Util.m_137537_(() -> {
       DiscreteVoxelShape discretevoxelshape = new BitSetDiscreteVoxelShape(1, 1, 1);
-      discretevoxelshape.m_142703_(0, 0, 0);
+      discretevoxelshape.fill(0, 0, 0);
       return new CubeVoxelShape(discretevoxelshape);
    });
    public static final VoxelShape f_83036_ = m_83048_(Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY);
