--- a/net/minecraft/world/phys/shapes/SubShape.java
+++ b/net/minecraft/world/phys/shapes/SubShape.java
@@ -27,8 +_,8 @@
       return this.f_83182_.m_6696_(this.f_83183_ + p_83206_, this.f_83184_ + p_83207_, this.f_83185_ + p_83208_);
    }
 
-   public void m_142703_(int p_166060_, int p_166061_, int p_166062_) {
-      this.f_83182_.m_142703_(this.f_83183_ + p_166060_, this.f_83184_ + p_166061_, this.f_83185_ + p_166062_);
+   public void fill(long p_166060_, long p_166061_, long p_166062_) {
+      this.f_83182_.fill(this.f_83183_ + p_166060_, this.f_83184_ + p_166061_, this.f_83185_ + p_166062_);
    }
 
    public int m_6538_(Direction.Axis p_83204_) {
