--- a/net/minecraft/world/entity/animal/AbstractFish.java
+++ b/net/minecraft/world/entity/animal/AbstractFish.java
@@ -163,9 +_,9 @@
          if (this.f_24981_ == MoveControl.Operation.MOVE_TO && !this.f_27499_.m_21573_().m_26571_()) {
             float f = (float)(this.f_24978_ * this.f_27499_.m_21133_(Attributes.f_22279_));
             this.f_27499_.m_7910_(Mth.m_14179_(0.125F, this.f_27499_.m_6113_(), f));
-            double d0 = this.f_24975_ - this.f_27499_.m_20185_();
-            double d1 = this.f_24976_ - this.f_27499_.m_20186_();
-            double d2 = this.f_24977_ - this.f_27499_.m_20189_();
+            double d0 = this.f_24975_.subtract(this.f_27499_.getBigX()).doubleValue();
+            double d1 = this.f_24976_.subtract(this.f_27499_.getBigY()).doubleValue();
+            double d2 = this.f_24977_.subtract(this.f_27499_.getBigZ()).doubleValue();
             if (d1 != 0.0D) {
                double d3 = Math.sqrt(d0 * d0 + d1 * d1 + d2 * d2);
                this.f_27499_.m_20256_(this.f_27499_.m_20184_().m_82520_(0.0D, (double)this.f_27499_.m_6113_() * (d1 / d3) * 0.1D, 0.0D));
