--- a/net/minecraft/world/entity/animal/Animal.java
+++ b/net/minecraft/world/entity/animal/Animal.java
@@ -77,7 +_,7 @@
    }
 
    public float m_5610_(BlockPos p_27573_, LevelReader p_27574_) {
-      return p_27574_.m_8055_(p_27573_.m_7495_()).m_60713_(Blocks.f_50440_) ? 10.0F : p_27574_.m_220419_(p_27573_);
+      return p_27574_.m_8055_(p_27573_.below()).m_60713_(Blocks.f_50440_) ? 10.0F : p_27574_.m_220419_(p_27573_);
    }
 
    public void m_7380_(CompoundTag p_27587_) {
@@ -100,7 +_,7 @@
    }
 
    public static boolean m_218104_(EntityType<? extends Animal> p_218105_, LevelAccessor p_218106_, MobSpawnType p_218107_, BlockPos p_218108_, RandomSource p_218109_) {
-      return p_218106_.m_8055_(p_218108_.m_7495_()).m_204336_(BlockTags.f_184228_) && m_186209_(p_218106_, p_218108_);
+      return p_218106_.m_8055_(p_218108_.below()).m_204336_(BlockTags.f_184228_) && m_186209_(p_218106_, p_218108_);
    }
 
    protected static boolean m_186209_(BlockAndTintGetter p_186210_, BlockPos p_186211_) {
