--- a/net/minecraft/world/entity/animal/Bee.java
+++ b/net/minecraft/world/entity/animal/Bee.java
@@ -246,22 +_,22 @@
       super.m_8119_();
       if (this.m_27856_() && this.m_27869_() < 10 && this.f_19796_.m_188501_() < 0.05F) {
          for(int i = 0; i < this.f_19796_.m_188503_(2) + 1; ++i) {
-            this.m_27779_(this.m_9236_(), this.m_20185_() - (double)0.3F, this.m_20185_() + (double)0.3F, this.m_20189_() - (double)0.3F, this.m_20189_() + (double)0.3F, this.m_20227_(0.5D), ParticleTypes.f_123782_);
+            this.spawnFluidParticle(this.m_9236_(), this.getBigX().subtract(me.alphamode.mclong.core.BigConstants.BEE), this.getBigX().add(me.alphamode.mclong.core.BigConstants.BEE), this.getBigZ().subtract(me.alphamode.mclong.core.BigConstants.BEE), this.getBigZ().add(me.alphamode.mclong.core.BigConstants.BEE), this.getY(me.alphamode.mclong.core.BigConstants.AABB), ParticleTypes.f_123782_);
          }
       }
 
       this.m_27867_();
    }
 
-   private void m_27779_(Level p_27780_, double p_27781_, double p_27782_, double p_27783_, double p_27784_, double p_27785_, ParticleOptions p_27786_) {
-      p_27780_.m_7106_(p_27786_, Mth.m_14139_(p_27780_.f_46441_.m_188500_(), p_27781_, p_27782_), p_27785_, Mth.m_14139_(p_27780_.f_46441_.m_188500_(), p_27783_, p_27784_), 0.0D, 0.0D, 0.0D);
+   private void spawnFluidParticle(Level pLevel, me.alphamode.mclong.math.BigDecimal pStartX, me.alphamode.mclong.math.BigDecimal pEndX, me.alphamode.mclong.math.BigDecimal pStartZ, me.alphamode.mclong.math.BigDecimal pEndZ, me.alphamode.mclong.math.BigDecimal pPosY, ParticleOptions pParticleOption) {
+      pLevel.addParticle(pParticleOption, Mth.lerp(pLevel.f_46441_.m_188500_(), pStartX, pEndX), pPosY, Mth.lerp(pLevel.f_46441_.m_188500_(), pStartZ, pEndZ), 0.0D, 0.0D, 0.0D);
    }
 
    void m_27880_(BlockPos p_27881_) {
-      Vec3 vec3 = Vec3.m_82539_(p_27881_);
+      Vec3 vec3 = Vec3.atBottomCenterOf(p_27881_);
       int i = 0;
       BlockPos blockpos = this.m_20183_();
-      int j = (int)vec3.f_82480_ - blockpos.m_123342_();
+      long j = (int)vec3.m_7098_() - blockpos.getY();
       if (j > 2) {
          i = 4;
       } else if (j < -2) {
@@ -270,7 +_,7 @@
 
       int k = 6;
       int l = 8;
-      int i1 = blockpos.m_123333_(p_27881_);
+      int i1 = blockpos.distManhattan(p_27881_);
       if (i1 < 15) {
          k = i1 / 2;
          l = i1 / 2;
@@ -279,7 +_,7 @@
       Vec3 vec31 = AirRandomPos.m_148387_(this, k, l, i, vec3, (double)((float)Math.PI / 10F));
       if (vec31 != null) {
          this.f_21344_.m_26529_(0.5F);
-         this.f_21344_.m_26519_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, 1.0D);
+         this.f_21344_.moveTo(vec31.getX(), vec31.getY(), vec31.getZ(), 1.0D);
       }
    }
 
@@ -529,7 +_,7 @@
    protected PathNavigation m_6037_(Level p_27815_) {
       FlyingPathNavigation flyingpathnavigation = new FlyingPathNavigation(this, p_27815_) {
          public boolean m_6342_(BlockPos p_27947_) {
-            return !this.f_26495_.m_8055_(p_27947_.m_7495_()).m_60795_();
+            return !this.f_26495_.m_8055_(p_27947_.below()).m_60795_();
          }
 
          public void m_7638_() {
@@ -621,7 +_,7 @@
    }
 
    boolean m_27816_(BlockPos p_27817_, int p_27818_) {
-      return p_27817_.m_123314_(this.m_20183_(), (double)p_27818_);
+      return p_27817_.closerThan(this.m_20183_(), (double)p_27818_);
    }
 
    abstract class BaseBeeGoal extends Goal {
@@ -679,7 +_,7 @@
 
    class BeeEnterHiveGoal extends Bee.BaseBeeGoal {
       public boolean m_7989_() {
-         if (Bee.this.m_27854_() && Bee.this.m_27866_() && Bee.this.f_27698_.m_203195_(Bee.this.m_20182_(), 2.0D)) {
+         if (Bee.this.m_27854_() && Bee.this.m_27866_() && Bee.this.f_27698_.closerToCenterThan(Bee.this.m_20182_(), 2.0D)) {
             BlockEntity blockentity = Bee.this.m_9236_().m_7702_(Bee.this.f_27698_);
             if (blockentity instanceof BeehiveBlockEntity) {
                BeehiveBlockEntity beehiveblockentity = (BeehiveBlockEntity)blockentity;
@@ -776,7 +_,7 @@
 
       private boolean m_27990_(BlockPos p_27991_) {
          Bee.this.f_21344_.m_26529_(10.0F);
-         Bee.this.f_21344_.m_26519_((double)p_27991_.m_123341_(), (double)p_27991_.m_123342_(), (double)p_27991_.m_123343_(), 1.0D);
+         Bee.this.f_21344_.moveTo(p_27991_.getBigDecX(), p_27991_.getBigDecY(), p_27991_.getBigDecZ(), 1.0D);
          return Bee.this.f_21344_.m_26570_() != null && Bee.this.f_21344_.m_26570_().m_77403_();
       }
 
@@ -887,7 +_,7 @@
       public void m_8037_() {
          if (Bee.this.f_19796_.m_188503_(this.m_183277_(30)) == 0) {
             for(int i = 1; i <= 2; ++i) {
-               BlockPos blockpos = Bee.this.m_20183_().m_6625_(i);
+               BlockPos blockpos = Bee.this.m_20183_().below(i);
                BlockState blockstate = Bee.this.m_9236_().m_8055_(blockpos);
                Block block = blockstate.m_60734_();
                BlockState blockstate1 = null;
@@ -972,7 +_,7 @@
             return p_218130_.m_203656_(PoiTypeTags.f_215877_);
          }, blockpos, 20, PoiManager.Occupancy.ANY);
          return stream.map(PoiRecord::m_27257_).filter(Bee.this::m_27884_).sorted(Comparator.comparingDouble((p_148811_) -> {
-            return p_148811_.m_123331_(blockpos);
+            return p_148811_.distSqr(blockpos);
          })).collect(Collectors.toList());
       }
    }
@@ -1038,7 +_,7 @@
             Optional<BlockPos> optional = this.m_28090_();
             if (optional.isPresent()) {
                Bee.this.f_27697_ = optional.get();
-               Bee.this.f_21344_.m_26519_((double)Bee.this.f_27697_.m_123341_() + 0.5D, (double)Bee.this.f_27697_.m_123342_() + 0.5D, (double)Bee.this.f_27697_.m_123343_() + 0.5D, (double)1.2F);
+               Bee.this.f_21344_.moveTo(Bee.this.f_27697_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), Bee.this.f_27697_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), Bee.this.f_27697_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), (double)1.2F);
                return true;
             } else {
                Bee.this.f_27714_ = Mth.m_216271_(Bee.this.f_19796_, 20, 60);
@@ -1103,7 +_,7 @@
          if (this.f_28068_ > 600) {
             Bee.this.f_27697_ = null;
          } else {
-            Vec3 vec3 = Vec3.m_82539_(Bee.this.f_27697_).m_82520_(0.0D, (double)0.6F, 0.0D);
+            Vec3 vec3 = Vec3.atBottomCenterOf(Bee.this.f_27697_).m_82520_(0.0D, (double)0.6F, 0.0D);
             if (vec3.m_82554_(Bee.this.m_20182_()) > 1.0D) {
                this.f_28067_ = vec3;
                this.m_28088_();
@@ -1126,7 +_,7 @@
                         flag1 = false;
                      }
 
-                     Bee.this.m_21563_().m_24946_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
+                     Bee.this.m_21563_().setLookAt(vec3.getX(), vec3.getY(), vec3.getZ());
                   }
 
                   if (flag1) {
@@ -1145,7 +_,7 @@
       }
 
       private void m_28088_() {
-         Bee.this.m_21566_().m_6849_(this.f_28067_.m_7096_(), this.f_28067_.m_7098_(), this.f_28067_.m_7094_(), (double)0.35F);
+         Bee.this.m_21566_().setWantedPosition(this.f_28067_.getX(), this.f_28067_.getY(), this.f_28067_.getZ(), (double)0.35F);
       }
 
       private float m_28089_() {
@@ -1164,8 +_,8 @@
             for(int j = 0; (double)j < p_28077_; ++j) {
                for(int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
                   for(int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
-                     blockpos$mutableblockpos.m_122154_(blockpos, k, i - 1, l);
-                     if (blockpos.m_123314_(blockpos$mutableblockpos, p_28077_) && p_28076_.test(Bee.this.m_9236_().m_8055_(blockpos$mutableblockpos))) {
+                     blockpos$mutableblockpos.setWithOffset(blockpos, k, i - 1, l);
+                     if (blockpos.closerThan(blockpos$mutableblockpos, p_28077_) && p_28076_.test(Bee.this.m_9236_().m_8055_(blockpos$mutableblockpos))) {
                         return Optional.of(blockpos$mutableblockpos);
                      }
                   }
@@ -1195,7 +_,7 @@
       public void m_8056_() {
          Vec3 vec3 = this.m_28097_();
          if (vec3 != null) {
-            Bee.this.f_21344_.m_26536_(Bee.this.f_21344_.m_7864_(BlockPos.m_274446_(vec3), 1), 1.0D);
+            Bee.this.f_21344_.m_26536_(Bee.this.f_21344_.m_7864_(BlockPos.containing(vec3), 1), 1.0D);
          }
 
       }
@@ -1204,15 +_,15 @@
       private Vec3 m_28097_() {
          Vec3 vec3;
          if (Bee.this.m_27872_() && !Bee.this.m_27816_(Bee.this.f_27698_, 22)) {
-            Vec3 vec31 = Vec3.m_82512_(Bee.this.f_27698_);
+            Vec3 vec31 = Vec3.atCenterOf(Bee.this.f_27698_);
             vec3 = vec31.m_82546_(Bee.this.m_20182_()).m_82541_();
          } else {
             vec3 = Bee.this.m_20252_(0.0F);
          }
 
          int i = 8;
-         Vec3 vec32 = HoverRandomPos.m_148465_(Bee.this, 8, 7, vec3.f_82479_, vec3.f_82481_, ((float)Math.PI / 2F), 3, 1);
-         return vec32 != null ? vec32 : AirAndWaterRandomPos.m_148357_(Bee.this, 8, 4, -2, vec3.f_82479_, vec3.f_82481_, (double)((float)Math.PI / 2F));
+         Vec3 vec32 = HoverRandomPos.m_148465_(Bee.this, 8, 7, vec3.m_7096_(), vec3.m_7094_(), ((float)Math.PI / 2F), 3, 1);
+         return vec32 != null ? vec32 : AirAndWaterRandomPos.m_148357_(Bee.this, 8, 4, -2, vec3.m_7096_(), vec3.m_7094_(), (double)((float)Math.PI / 2F));
       }
    }
 }
