--- a/net/minecraft/world/entity/animal/Chicken.java
+++ b/net/minecraft/world/entity/animal/Chicken.java
@@ -82,7 +_,7 @@
 
       this.f_28230_ *= 0.9F;
       Vec3 vec3 = this.m_20184_();
-      if (!this.m_20096_() && vec3.f_82480_ < 0.0D) {
+      if (!this.m_20096_() && vec3.m_7098_() < 0.0D) {
          this.m_20256_(vec3.m_82542_(1.0D, 0.6D, 1.0D));
       }
 
@@ -158,7 +_,7 @@
       float f1 = Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F));
       float f2 = 0.1F;
       float f3 = 0.0F;
-      p_289541_.m_20372_(p_289537_, this.m_20185_() + (double)(0.1F * f), this.m_20227_(0.5D) + p_289537_.m_6049_() + 0.0D, this.m_20189_() - (double)(0.1F * f1));
+      p_289541_.accept(p_289537_, this.getBigX().add(java.math.BigDecimal.valueOf((0.1F * f))), java.math.BigDecimal.valueOf(this.m_20227_(0.5D) + p_289537_.m_6049_() + 0.0D), this.getBigZ().subtract(java.math.BigDecimal.valueOf((0.1F * f1))));
       if (p_289537_ instanceof LivingEntity) {
          ((LivingEntity)p_289537_).f_20883_ = this.f_20883_;
       }
