--- a/net/minecraft/world/entity/animal/Dolphin.java
+++ b/net/minecraft/world/entity/animal/Dolphin.java
@@ -126,17 +_,17 @@
 
    public void m_7380_(CompoundTag p_28364_) {
       super.m_7380_(p_28364_);
-      p_28364_.m_128405_("TreasurePosX", this.m_28387_().m_123341_());
-      p_28364_.m_128405_("TreasurePosY", this.m_28387_().m_123342_());
-      p_28364_.m_128405_("TreasurePosZ", this.m_28387_().m_123343_());
+      p_28364_.m_128356_("TreasurePosX", this.m_28387_().getX());
+      p_28364_.m_128356_("TreasurePosY", this.m_28387_().getY());
+      p_28364_.m_128356_("TreasurePosZ", this.m_28387_().getZ());
       p_28364_.m_128379_("GotFish", this.m_28377_());
       p_28364_.m_128405_("Moistness", this.m_28378_());
    }
 
    public void m_7378_(CompoundTag p_28340_) {
-      int i = p_28340_.m_128451_("TreasurePosX");
-      int j = p_28340_.m_128451_("TreasurePosY");
-      int k = p_28340_.m_128451_("TreasurePosZ");
+      long i = p_28340_.m_128454_("TreasurePosX");
+      long j = p_28340_.m_128454_("TreasurePosY");
+      long k = p_28340_.m_128454_("TreasurePosZ");
       this.m_28384_(new BlockPos(i, j, k));
       super.m_7378_(p_28340_);
       this.m_28393_(p_28340_.m_128471_("GotFish"));
@@ -252,8 +_,8 @@
             float f2 = 1.2F - this.f_19796_.m_188501_() * 0.7F;
 
             for(int i = 0; i < 2; ++i) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123776_, this.m_20185_() - vec3.f_82479_ * (double)f2 + (double)f, this.m_20186_() - vec3.f_82480_, this.m_20189_() - vec3.f_82481_ * (double)f2 + (double)f1, 0.0D, 0.0D, 0.0D);
-               this.m_9236_().m_7106_(ParticleTypes.f_123776_, this.m_20185_() - vec3.f_82479_ * (double)f2 - (double)f, this.m_20186_() - vec3.f_82480_, this.m_20189_() - vec3.f_82481_ * (double)f2 - (double)f1, 0.0D, 0.0D, 0.0D);
+               this.m_9236_().addParticle(ParticleTypes.f_123776_, this.getBigX().subtract(vec3.getX().multiply(f2).add(f)), this.getBigY().subtract(vec3.getY()), this.getBigZ().subtract(vec3.getZ().multiply(f2).add(f1)), 0.0D, 0.0D, 0.0D);
+               this.m_9236_().addParticle(ParticleTypes.f_123776_, this.getBigX().subtract(vec3.getX().multiply(f2)).subtract(f), this.getBigY().subtract(vec3.getY()), this.getBigZ().subtract(vec3.getZ().multiply(f2)).subtract(f1), 0.0D, 0.0D, 0.0D);
             }
          }
 
@@ -274,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.01D;
          double d1 = this.f_19796_.m_188583_() * 0.01D;
          double d2 = this.f_19796_.m_188583_() * 0.01D;
-         this.m_9236_().m_7106_(p_28338_, this.m_20208_(1.0D), this.m_20187_() + 0.2D, this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(p_28338_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.PARTICLE_TICK), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
@@ -321,7 +_,7 @@
 
    protected boolean m_28380_() {
       BlockPos blockpos = this.m_21573_().m_26567_();
-      return blockpos != null ? blockpos.m_203195_(this.m_20182_(), 12.0D) : false;
+      return blockpos != null ? blockpos.closerToCenterThan(this.m_20182_(), 12.0D) : false;
    }
 
    public void m_7023_(Vec3 p_28383_) {
@@ -361,7 +_,7 @@
 
       public boolean m_8045_() {
          BlockPos blockpos = this.f_28399_.m_28387_();
-         return !BlockPos.m_274561_((double)blockpos.m_123341_(), this.f_28399_.m_20186_(), (double)blockpos.m_123343_()).m_203195_(this.f_28399_.m_20182_(), 4.0D) && !this.f_28400_ && this.f_28399_.m_20146_() >= 100;
+         return !BlockPos.m_274561_((double)blockpos.getX(), this.f_28399_.m_20186_(), (double)blockpos.getZ()).closerToCenterThan(this.f_28399_.m_20182_(), 4.0D) && !this.f_28400_ && this.f_28399_.m_20146_() >= 100;
       }
 
       public void m_8056_() {
@@ -382,7 +_,7 @@
 
       public void m_8041_() {
          BlockPos blockpos = this.f_28399_.m_28387_();
-         if (BlockPos.m_274561_((double)blockpos.m_123341_(), this.f_28399_.m_20186_(), (double)blockpos.m_123343_()).m_203195_(this.f_28399_.m_20182_(), 4.0D) || this.f_28400_) {
+         if (BlockPos.m_274561_((double)blockpos.getX(), this.f_28399_.m_20186_(), (double)blockpos.getZ()).closerToCenterThan(this.f_28399_.m_20182_(), 4.0D) || this.f_28400_) {
             this.f_28399_.m_28393_(false);
          }
 
@@ -391,14 +_,14 @@
       public void m_8037_() {
          Level level = this.f_28399_.m_9236_();
          if (this.f_28399_.m_28380_() || this.f_28399_.m_21573_().m_26571_()) {
-            Vec3 vec3 = Vec3.m_82512_(this.f_28399_.m_28387_());
+            Vec3 vec3 = Vec3.atCenterOf(this.f_28399_.m_28387_());
             Vec3 vec31 = DefaultRandomPos.m_148412_(this.f_28399_, 16, 1, vec3, (double)((float)Math.PI / 8F));
             if (vec31 == null) {
                vec31 = DefaultRandomPos.m_148412_(this.f_28399_, 8, 4, vec3, (double)((float)Math.PI / 2F));
             }
 
             if (vec31 != null) {
-               BlockPos blockpos = BlockPos.m_274446_(vec31);
+               BlockPos blockpos = BlockPos.containing(vec31);
                if (!level.m_6425_(blockpos).m_205070_(FluidTags.f_13131_) || !level.m_8055_(blockpos).m_60647_(level, blockpos, PathComputationType.WATER)) {
                   vec31 = DefaultRandomPos.m_148412_(this.f_28399_, 8, 5, vec3, (double)((float)Math.PI / 2F));
                }
@@ -409,8 +_,8 @@
                return;
             }
 
-            this.f_28399_.m_21563_().m_24950_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, (float)(this.f_28399_.m_8085_() + 20), (float)this.f_28399_.m_8132_());
-            this.f_28399_.m_21573_().m_26519_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, 1.3D);
+            this.f_28399_.m_21563_().m_24950_(vec31.m_7096_(), vec31.m_7098_(), vec31.m_7094_(), (float)(this.f_28399_.m_8085_() + 20), (float)this.f_28399_.m_8132_());
+            this.f_28399_.m_21573_().m_26519_(vec31.m_7096_(), vec31.m_7098_(), vec31.m_7094_(), 1.3D);
             if (level.f_46441_.m_188503_(this.m_183277_(80)) == 0) {
                level.m_7605_(this.f_28399_, (byte)38);
             }
