--- a/net/minecraft/world/entity/animal/Fox.java
+++ b/net/minecraft/world/entity/animal/Fox.java
@@ -288,7 +_,7 @@
    }
 
    public static boolean m_218175_(EntityType<Fox> p_218176_, LevelAccessor p_218177_, MobSpawnType p_218178_, BlockPos p_218179_, RandomSource p_218180_) {
-      return p_218177_.m_8055_(p_218179_.m_7495_()).m_204336_(BlockTags.f_184235_) && m_186209_(p_218177_, p_218179_);
+      return p_218177_.m_8055_(p_218179_.below()).m_204336_(BlockTags.f_184235_) && m_186209_(p_218177_, p_218179_);
    }
 
    @Nullable
