--- a/net/minecraft/world/entity/animal/IronGolem.java
+++ b/net/minecraft/world/entity/animal/IronGolem.java
@@ -116,9 +_,9 @@
       }
 
       if (this.m_20184_().m_165925_() > (double)2.5000003E-7F && this.f_19796_.m_188503_(5) == 0) {
-         int i = Mth.m_14107_(this.m_20185_());
-         int j = Mth.m_14107_(this.m_20186_() - (double)0.2F);
-         int k = Mth.m_14107_(this.m_20189_());
+         long i = Mth.m_14134_(this.m_20185_());
+         long j = Mth.m_14134_(this.m_20186_() - (double)0.2F);
+         long k = Mth.m_14134_(this.m_20189_());
          BlockState blockstate = this.f_19853_.m_8055_(new BlockPos(i, j, k));
          if (!blockstate.m_60795_()) {
             this.f_19853_.m_7106_(new BlockParticleOption(ParticleTypes.f_123794_, blockstate), this.m_20185_() + ((double)this.f_19796_.m_188501_() - 0.5D) * (double)this.m_20205_(), this.m_20186_() + 0.1D, this.m_20189_() + ((double)this.f_19796_.m_188501_() - 0.5D) * (double)this.m_20205_(), 4.0D * ((double)this.f_19796_.m_188501_() - 0.5D), 0.5D, ((double)this.f_19796_.m_188501_() - 0.5D) * 4.0D);
@@ -301,13 +_,13 @@
 
    public boolean m_6914_(LevelReader p_28853_) {
       BlockPos blockpos = this.m_20183_();
-      BlockPos blockpos1 = blockpos.m_7495_();
+      BlockPos blockpos1 = blockpos.below();
       BlockState blockstate = p_28853_.m_8055_(blockpos1);
       if (!blockstate.m_60634_(p_28853_, blockpos1, this)) {
          return false;
       } else {
          for(int i = 1; i < 3; ++i) {
-            BlockPos blockpos2 = blockpos.m_6630_(i);
+            BlockPos blockpos2 = blockpos.above(i);
             BlockState blockstate1 = p_28853_.m_8055_(blockpos2);
             if (!NaturalSpawner.m_47056_(p_28853_, blockpos2, blockstate1, blockstate1.m_60819_(), EntityType.f_20460_)) {
                return false;
