--- a/net/minecraft/world/entity/animal/MushroomCow.java
+++ b/net/minecraft/world/entity/animal/MushroomCow.java
@@ -55,11 +_,11 @@
    }
 
    public float m_5610_(BlockPos p_28933_, LevelReader p_28934_) {
-      return p_28934_.m_8055_(p_28933_.m_7495_()).m_60713_(Blocks.f_50195_) ? 10.0F : p_28934_.m_220419_(p_28933_);
+      return p_28934_.m_8055_(p_28933_.below()).m_60713_(Blocks.f_50195_) ? 10.0F : p_28934_.m_220419_(p_28933_);
    }
 
    public static boolean m_218200_(EntityType<MushroomCow> p_218201_, LevelAccessor p_218202_, MobSpawnType p_218203_, BlockPos p_218204_, RandomSource p_218205_) {
-      return p_218202_.m_8055_(p_218204_.m_7495_()).m_204336_(BlockTags.f_184231_) && m_186209_(p_218202_, p_218204_);
+      return p_218202_.m_8055_(p_218204_.below()).m_204336_(BlockTags.f_184231_) && m_186209_(p_218202_, p_218204_);
    }
 
    public void m_8038_(ServerLevel p_28921_, LightningBolt p_28922_) {
