--- a/net/minecraft/world/entity/animal/Ocelot.java
+++ b/net/minecraft/world/entity/animal/Ocelot.java
@@ -230,11 +_,11 @@
    public boolean m_6914_(LevelReader p_29005_) {
       if (p_29005_.m_45784_(this) && !p_29005_.m_46855_(this.m_20191_())) {
          BlockPos blockpos = this.m_20183_();
-         if (blockpos.m_123342_() < p_29005_.m_5736_()) {
+         if (blockpos.getY() < p_29005_.m_5736_()) {
             return false;
          }
 
-         BlockState blockstate = p_29005_.m_8055_(blockpos.m_7495_());
+         BlockState blockstate = p_29005_.m_8055_(blockpos.below());
          if (blockstate.m_60713_(Blocks.f_50440_) || blockstate.m_204336_(BlockTags.f_13035_)) {
             return true;
          }
