--- a/net/minecraft/world/entity/animal/Ocelot.java
+++ b/net/minecraft/world/entity/animal/Ocelot.java
@@ -197,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.02D;
          double d1 = this.f_19796_.m_188583_() * 0.02D;
          double d2 = this.f_19796_.m_188583_() * 0.02D;
-         this.m_9236_().m_7106_(particleoptions, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(particleoptions, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
@@ -230,11 +_,11 @@
    public boolean m_6914_(LevelReader p_29005_) {
       if (p_29005_.m_45784_(this) && !p_29005_.m_46855_(this.m_20191_())) {
          BlockPos blockpos = this.m_20183_();
-         if (blockpos.m_123342_() < p_29005_.m_5736_()) {
+         if (blockpos.getY() < p_29005_.m_5736_()) {
             return false;
          }
 
-         BlockState blockstate = p_29005_.m_8055_(blockpos.m_7495_());
+         BlockState blockstate = p_29005_.m_8055_(blockpos.below());
          if (blockstate.m_60713_(Blocks.f_50440_) || blockstate.m_204336_(BlockTags.f_13035_)) {
             return true;
          }
