--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -802,7 +_,7 @@
             for(int j = 0; j < 8; ++j) {
                for(int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
                   for(int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
-                     blockpos$mutableblockpos.m_122154_(blockpos, k, i, l);
+                     blockpos$mutableblockpos.setWithOffset(blockpos, k, i, l);
                      if (this.f_25114_.m_8055_(blockpos$mutableblockpos).m_60713_(Blocks.f_50571_)) {
                         return true;
                      }
