--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -796,7 +_,7 @@
             for(int j = 0; j < 8; ++j) {
                for(int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
                   for(int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
-                     blockpos$mutableblockpos.m_122154_(blockpos, k, i, l);
+                     blockpos$mutableblockpos.setWithOffset(blockpos, k, i, l);
                      if (this.f_25114_.m_8055_(blockpos$mutableblockpos).m_60713_(Blocks.f_50571_)) {
                         return true;
                      }
@@ -962,7 +_,7 @@
                float f2 = Mth.m_14089_(f);
                int i = (double)Math.abs(f1) > 0.5D ? Mth.m_14205_((double)f1) : 0;
                int j = (double)Math.abs(f2) > 0.5D ? Mth.m_14205_((double)f2) : 0;
-               if (this.f_29326_.f_19853_.m_8055_(this.f_29326_.m_20183_().m_7918_(i, -1, j)).m_60795_()) {
+               if (this.f_29326_.f_19853_.m_8055_(this.f_29326_.m_20183_().offset(i, -1, j)).m_60795_()) {
                   return true;
                } else if (this.f_29326_.m_29163_() && this.f_29326_.f_19796_.m_188503_(m_186073_(60)) == 1) {
                   return true;
