--- a/net/minecraft/world/entity/animal/Panda.java
+++ b/net/minecraft/world/entity/animal/Panda.java
@@ -418,7 +_,7 @@
             Vec3 vec31 = new Vec3(((double)this.f_19796_.m_188501_() - 0.5D) * 0.8D, d0, 1.0D + ((double)this.f_19796_.m_188501_() - 0.5D) * 0.4D);
             vec31 = vec31.m_82524_(-this.f_20883_ * ((float)Math.PI / 180F));
             vec31 = vec31.m_82520_(this.m_20185_(), this.m_20188_() + 1.0D, this.m_20189_());
-            this.m_9236_().m_7106_(new ItemParticleOption(ParticleTypes.f_123752_, this.m_6844_(EquipmentSlot.MAINHAND)), vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, vec3.f_82479_, vec3.f_82480_ + 0.05D, vec3.f_82481_);
+            this.m_9236_().m_7106_(new ItemParticleOption(ParticleTypes.f_123752_, this.m_6844_(EquipmentSlot.MAINHAND)), vec31.m_7096_(), vec31.m_7098_(), vec31.m_7094_(), vec3.m_7096_(), vec3.m_7098_() + 0.05D, vec3.m_7094_());
          }
       }
 
@@ -476,12 +_,12 @@
             if (this.f_29072_ == 1) {
                float f = this.m_146908_() * ((float)Math.PI / 180F);
                float f1 = this.m_6162_() ? 0.1F : 0.2F;
-               this.f_29082_ = new Vec3(vec3.f_82479_ + (double)(-Mth.m_14031_(f) * f1), 0.0D, vec3.f_82481_ + (double)(Mth.m_14089_(f) * f1));
+               this.f_29082_ = new Vec3(vec3.m_7096_() + (double)(-Mth.m_14031_(f) * f1), 0.0D, vec3.m_7094_() + (double)(Mth.m_14089_(f) * f1));
                this.m_20256_(this.f_29082_.m_82520_(0.0D, 0.27D, 0.0D));
             } else if ((float)this.f_29072_ != 7.0F && (float)this.f_29072_ != 15.0F && (float)this.f_29072_ != 23.0F) {
-               this.m_20334_(this.f_29082_.f_82479_, vec3.f_82480_, this.f_29082_.f_82481_);
+               this.m_20334_(this.f_29082_.m_7096_(), vec3.m_7098_(), this.f_29082_.m_7094_());
             } else {
-               this.m_20334_(0.0D, this.m_20096_() ? 0.27D : vec3.f_82480_, 0.0D);
+               this.m_20334_(0.0D, this.m_20096_() ? 0.27D : vec3.m_7098_(), 0.0D);
             }
          }
 
@@ -490,7 +_,7 @@
 
    private void m_29177_() {
       Vec3 vec3 = this.m_20184_();
-      this.m_9236_().m_7106_(ParticleTypes.f_123763_, this.m_20185_() - (double)(this.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14031_(this.f_20883_ * ((float)Math.PI / 180F)), this.m_20188_() - (double)0.1F, this.m_20189_() + (double)(this.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F)), vec3.f_82479_, 0.0D, vec3.f_82481_);
+      this.m_9236_().m_7106_(ParticleTypes.f_123763_, this.m_20185_() - (double)(this.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14031_(this.f_20883_ * ((float)Math.PI / 180F)), this.m_20188_() - (double)0.1F, this.m_20189_() + (double)(this.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14089_(this.f_20883_ * ((float)Math.PI / 180F)), vec3.m_7096_(), 0.0D, vec3.m_7094_());
       this.m_5496_(SoundEvents.f_12178_, 1.0F, 1.0F);
 
       for(Panda panda : this.m_9236_().m_45976_(Panda.class, this.m_20191_().m_82400_(10.0D))) {
@@ -796,7 +_,7 @@
             for(int j = 0; j < 8; ++j) {
                for(int k = 0; k <= j; k = k > 0 ? -k : 1 - k) {
                   for(int l = k < j && k > -j ? j : 0; l <= j; l = l > 0 ? -l : 1 - l) {
-                     blockpos$mutableblockpos.m_122154_(blockpos, k, i, l);
+                     blockpos$mutableblockpos.setWithOffset(blockpos, k, i, l);
                      if (this.f_25114_.m_8055_(blockpos$mutableblockpos).m_60713_(Blocks.f_50571_)) {
                         return true;
                      }
@@ -962,7 +_,7 @@
                float f2 = Mth.m_14089_(f);
                int i = (double)Math.abs(f1) > 0.5D ? Mth.m_14205_((double)f1) : 0;
                int j = (double)Math.abs(f2) > 0.5D ? Mth.m_14205_((double)f2) : 0;
-               if (this.f_29326_.m_9236_().m_8055_(this.f_29326_.m_20183_().m_7918_(i, -1, j)).m_60795_()) {
+               if (this.f_29326_.m_9236_().m_8055_(this.f_29326_.m_20183_().offset(i, -1, j)).m_60795_()) {
                   return true;
                } else if (this.f_29326_.m_29163_() && this.f_29326_.f_19796_.m_188503_(m_186073_(60)) == 1) {
                   return true;
