--- a/net/minecraft/world/entity/animal/PolarBear.java
+++ b/net/minecraft/world/entity/animal/PolarBear.java
@@ -99,7 +_,7 @@
       if (!holder.m_203656_(BiomeTags.f_215813_)) {
          return m_218104_(p_218250_, p_218251_, p_218252_, p_218253_, p_218254_);
       } else {
-         return m_186209_(p_218251_, p_218253_) && p_218251_.m_8055_(p_218253_.m_7495_()).m_204336_(BlockTags.f_215826_);
+         return m_186209_(p_218251_, p_218253_) && p_218251_.m_8055_(p_218253_.below()).m_204336_(BlockTags.f_215826_);
       }
    }
 
