--- a/net/minecraft/world/entity/animal/Rabbit.java
+++ b/net/minecraft/world/entity/animal/Rabbit.java
@@ -113,7 +_,7 @@
       Path path = this.f_21344_.m_26570_();
       if (path != null && !path.m_77392_()) {
          Vec3 vec3 = path.m_77380_(this);
-         if (vec3.f_82480_ > this.m_20186_() + 0.5D) {
+         if (vec3.m_7098_() > this.m_20186_() + 0.5D) {
             f = 0.5F;
          }
       }
@@ -206,7 +_,7 @@
                   vec3 = path.m_77380_(this);
                }
 
-               this.m_29686_(vec3.f_82479_, vec3.f_82481_);
+               this.m_29686_(vec3.m_7096_(), vec3.m_7094_());
                this.m_29716_();
             }
          } else if (!rabbit$rabbitjumpcontrol.m_29762_()) {
@@ -382,7 +_,7 @@
    }
 
    public static boolean m_218255_(EntityType<Rabbit> p_218256_, LevelAccessor p_218257_, MobSpawnType p_218258_, BlockPos p_218259_, RandomSource p_218260_) {
-      return p_218257_.m_8055_(p_218259_.m_7495_()).m_204336_(BlockTags.f_184234_) && m_186209_(p_218257_, p_218259_);
+      return p_218257_.m_8055_(p_218259_.below()).m_204336_(BlockTags.f_184234_) && m_186209_(p_218257_, p_218259_);
    }
 
    boolean m_29724_() {
@@ -541,10 +_,10 @@
 
       public void m_8037_() {
          super.m_8037_();
-         this.f_29778_.m_21563_().m_24950_((double)this.f_25602_.m_123341_() + 0.5D, (double)(this.f_25602_.m_123342_() + 1), (double)this.f_25602_.m_123343_() + 0.5D, 10.0F, (float)this.f_29778_.m_8132_());
+         this.f_29778_.m_21563_().m_24950_((double)this.f_25602_.getX() + 0.5D, (double)(this.f_25602_.getY() + 1), (double)this.f_25602_.getZ() + 0.5D, 10.0F, (float)this.f_29778_.m_8132_());
          if (this.m_25625_()) {
             Level level = this.f_29778_.m_9236_();
-            BlockPos blockpos = this.f_25602_.m_7494_();
+            BlockPos blockpos = this.f_25602_.above();
             BlockState blockstate = level.m_8055_(blockpos);
             Block block = blockstate.m_60734_();
             if (this.f_29780_ && block instanceof CarrotBlock) {
@@ -569,7 +_,7 @@
       protected boolean m_6465_(LevelReader p_29785_, BlockPos p_29786_) {
          BlockState blockstate = p_29785_.m_8055_(p_29786_);
          if (blockstate.m_60713_(Blocks.f_50093_) && this.f_29779_ && !this.f_29780_) {
-            blockstate = p_29785_.m_8055_(p_29786_.m_7494_());
+            blockstate = p_29785_.m_8055_(p_29786_.above());
             if (blockstate.m_60734_() instanceof CarrotBlock && ((CarrotBlock)blockstate.m_60734_()).m_52307_(blockstate)) {
                this.f_29780_ = true;
                return true;
