--- a/net/minecraft/world/entity/animal/SnowGolem.java
+++ b/net/minecraft/world/entity/animal/SnowGolem.java
@@ -99,9 +_,9 @@
          BlockState blockstate = Blocks.f_50125_.m_49966_();
 
          for(int i = 0; i < 4; ++i) {
-            int j = Mth.m_14107_(this.m_20185_() + (double)((float)(i % 2 * 2 - 1) * 0.25F));
-            int k = Mth.m_14107_(this.m_20186_());
-            int l = Mth.m_14107_(this.m_20189_() + (double)((float)(i / 2 % 2 * 2 - 1) * 0.25F));
+            long j = Mth.m_14134_(this.m_20185_() + (double)((float)(i % 2 * 2 - 1) * 0.25F));
+            long k = Mth.m_14134_(this.m_20186_());
+            long l = Mth.m_14134_(this.m_20189_() + (double)((float)(i / 2 % 2 * 2 - 1) * 0.25F));
             BlockPos blockpos = new BlockPos(j, k, l);
             if (this.m_9236_().m_8055_(blockpos).m_60795_() && blockstate.m_60710_(this.m_9236_(), blockpos)) {
                this.m_9236_().m_46597_(blockpos, blockstate);
