--- a/net/minecraft/world/entity/animal/TropicalFish.java
+++ b/net/minecraft/world/entity/animal/TropicalFish.java
@@ -195,7 +_,7 @@
    }
 
    public static boolean m_218266_(EntityType<TropicalFish> p_218267_, LevelAccessor p_218268_, MobSpawnType p_218269_, BlockPos p_218270_, RandomSource p_218271_) {
-      return p_218268_.m_6425_(p_218270_.m_7495_()).m_205070_(FluidTags.f_13131_) && p_218268_.m_8055_(p_218270_.m_7494_()).m_60713_(Blocks.f_49990_) && (p_218268_.m_204166_(p_218270_).m_203656_(BiomeTags.f_215812_) || WaterAnimal.m_218282_(p_218267_, p_218268_, p_218269_, p_218270_, p_218271_));
+      return p_218268_.m_6425_(p_218270_.below()).m_205070_(FluidTags.f_13131_) && p_218268_.m_8055_(p_218270_.above()).m_60713_(Blocks.f_49990_) && (p_218268_.m_204166_(p_218270_).m_203656_(BiomeTags.f_215812_) || WaterAnimal.m_218282_(p_218267_, p_218268_, p_218269_, p_218270_, p_218271_));
    }
 
    static enum Pattern {
