--- a/net/minecraft/world/entity/animal/Wolf.java
+++ b/net/minecraft/world/entity/animal/Wolf.java
@@ -506,7 +_,7 @@
    }
 
    public static boolean m_218291_(EntityType<Wolf> p_218292_, LevelAccessor p_218293_, MobSpawnType p_218294_, BlockPos p_218295_, RandomSource p_218296_) {
-      return p_218293_.m_8055_(p_218295_.m_7495_()).m_204336_(BlockTags.f_184236_) && m_186209_(p_218293_, p_218295_);
+      return p_218293_.m_8055_(p_218295_.below()).m_204336_(BlockTags.f_184236_) && m_186209_(p_218293_, p_218295_);
    }
 
    class WolfAvoidEntityGoal<T extends LivingEntity> extends AvoidEntityGoal<T> {
