--- a/net/minecraft/commands/CommandSourceStack.java
+++ b/net/minecraft/commands/CommandSourceStack.java
@@ -131,7 +_,7 @@
          return this;
       } else {
          double d0 = DimensionType.m_63908_(this.f_81290_.m_6042_(), p_81328_.m_6042_());
-         Vec3 vec3 = new Vec3(this.f_81289_.f_82479_ * d0, this.f_81289_.f_82480_, this.f_81289_.f_82481_ * d0);
+         Vec3 vec3 = new Vec3(this.f_81289_.m_7096_() * d0, this.f_81289_.m_7098_(), this.f_81289_.m_7094_() * d0);
          return new CommandSourceStack(this.f_81288_, vec3, this.f_81299_, p_81328_, this.f_81291_, this.f_81292_, this.f_81293_, this.f_81294_, this.f_81296_, this.f_81295_, this.f_81297_, this.f_81298_, this.f_230878_, this.f_241659_, this.f_279552_);
       }
    }
@@ -142,9 +_,9 @@
 
    public CommandSourceStack m_81364_(Vec3 p_81365_) {
       Vec3 vec3 = this.f_81298_.m_90379_(this);
-      double d0 = p_81365_.f_82479_ - vec3.f_82479_;
-      double d1 = p_81365_.f_82480_ - vec3.f_82480_;
-      double d2 = p_81365_.f_82481_ - vec3.f_82481_;
+      double d0 = p_81365_.m_7096_() - vec3.m_7096_();
+      double d1 = p_81365_.m_7098_() - vec3.m_7098_();
+      double d2 = p_81365_.m_7094_() - vec3.m_7094_();
       double d3 = Math.sqrt(d0 * d0 + d2 * d2);
       float f = Mth.m_14177_((float)(-(Mth.m_14136_(d1, d3) * (double)(180F / (float)Math.PI))));
       float f1 = Mth.m_14177_((float)(Mth.m_14136_(d2, d0) * (double)(180F / (float)Math.PI)) - 90.0F);
