--- a/net/minecraft/world/level/levelgen/structure/BoundingBox.java
+++ b/net/minecraft/world/level/levelgen/structure/BoundingBox.java
@@ -31,10 +_,14 @@
    private int f_162360_;
    private int f_162361_;
 
-   public BoundingBox(BlockPos p_162364_) {
+   public BoundingBox(me.alphamode.mclong.core.LegacyBlockPos p_162364_) {
       this(p_162364_.m_123341_(), p_162364_.m_123342_(), p_162364_.m_123343_(), p_162364_.m_123341_(), p_162364_.m_123342_(), p_162364_.m_123343_());
    }
 
+   public BoundingBox(BlockPos p_162364_) {
+      this((int) p_162364_.getX(), (int) p_162364_.getY(), (int) p_162364_.getZ(), (int) p_162364_.getX(), (int) p_162364_.getY(), (int) p_162364_.getZ());
+   }
+
    public BoundingBox(int p_71001_, int p_71002_, int p_71003_, int p_71004_, int p_71005_, int p_71006_) {
       this.f_162356_ = p_71001_;
       this.f_162357_ = p_71002_;
@@ -63,6 +_,10 @@
       return new BoundingBox(Math.min(p_162376_.m_123341_(), p_162377_.m_123341_()), Math.min(p_162376_.m_123342_(), p_162377_.m_123342_()), Math.min(p_162376_.m_123343_(), p_162377_.m_123343_()), Math.max(p_162376_.m_123341_(), p_162377_.m_123341_()), Math.max(p_162376_.m_123342_(), p_162377_.m_123342_()), Math.max(p_162376_.m_123343_(), p_162377_.m_123343_()));
    }
 
+   public static BoundingBox fromCorners(me.alphamode.mclong.core.Vec3l p_162376_, me.alphamode.mclong.core.Vec3l p_162377_) {
+      return new BoundingBox((int) Math.min(p_162376_.getX(), p_162377_.getX()), (int) Math.min(p_162376_.getY(), p_162377_.getY()), (int) Math.min(p_162376_.getZ(), p_162377_.getZ()), (int) Math.max(p_162376_.getX(), p_162377_.getX()), (int) Math.max(p_162376_.getY(), p_162377_.getY()), (int) Math.max(p_162376_.getZ(), p_162377_.getZ()));
+   }
+
    public static BoundingBox m_71044_() {
       return new BoundingBox(Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE);
    }
@@ -85,11 +_,22 @@
       return this.f_162359_ >= p_71050_.f_162356_ && this.f_162356_ <= p_71050_.f_162359_ && this.f_162361_ >= p_71050_.f_162358_ && this.f_162358_ <= p_71050_.f_162361_ && this.f_162360_ >= p_71050_.f_162357_ && this.f_162357_ <= p_71050_.f_162360_;
    }
 
-   public boolean m_71019_(int p_71020_, int p_71021_, int p_71022_, int p_71023_) {
+   public boolean intersects(long p_71020_, long p_71021_, long p_71022_, long p_71023_) {
       return this.f_162359_ >= p_71020_ && this.f_162356_ <= p_71022_ && this.f_162361_ >= p_71021_ && this.f_162358_ <= p_71023_;
    }
 
-   public static Optional<BoundingBox> m_162378_(Iterable<BlockPos> p_162379_) {
+   public static Optional<BoundingBox> m_162378_(Iterable<me.alphamode.mclong.core.LegacyBlockPos> p_162379_) {
+      Iterator<me.alphamode.mclong.core.LegacyBlockPos> iterator = p_162379_.iterator();
+      if (!iterator.hasNext()) {
+         return Optional.empty();
+      } else {
+         BoundingBox boundingbox = new BoundingBox(iterator.next());
+         iterator.forEachRemaining(boundingbox::encapsulate);
+         return Optional.of(boundingbox);
+      }
+   }
+
+   public static Optional<BoundingBox> unsafeEncapsulatingPositions(Iterable<BlockPos> p_162379_) {
       Iterator<BlockPos> iterator = p_162379_.iterator();
       if (!iterator.hasNext()) {
          return Optional.empty();
@@ -126,7 +_,7 @@
 
    /** @deprecated */
    @Deprecated
-   public BoundingBox m_162371_(BlockPos p_162372_) {
+   public BoundingBox encapsulate(me.alphamode.mclong.core.LegacyBlockPos p_162372_) {
       this.f_162356_ = Math.min(this.f_162356_, p_162372_.m_123341_());
       this.f_162357_ = Math.min(this.f_162357_, p_162372_.m_123342_());
       this.f_162358_ = Math.min(this.f_162358_, p_162372_.m_123343_());
@@ -138,6 +_,18 @@
 
    /** @deprecated */
    @Deprecated
+   public BoundingBox m_162371_(BlockPos p_162372_) {
+      this.f_162356_ = (int) Math.min(this.f_162356_, p_162372_.getX());
+      this.f_162357_ = (int) Math.min(this.f_162357_, p_162372_.getY());
+      this.f_162358_ = (int) Math.min(this.f_162358_, p_162372_.getZ());
+      this.f_162359_ = (int) Math.max(this.f_162359_, p_162372_.getX());
+      this.f_162360_ = (int) Math.max(this.f_162360_, p_162372_.getY());
+      this.f_162361_ = (int) Math.max(this.f_162361_, p_162372_.getZ());
+      return this;
+   }
+
+   /** @deprecated */
+   @Deprecated
    public BoundingBox m_162367_(int p_162368_, int p_162369_, int p_162370_) {
       this.f_162356_ += p_162368_;
       this.f_162357_ += p_162369_;
@@ -150,10 +_,28 @@
 
    /** @deprecated */
    @Deprecated
+   public BoundingBox move(long p_162368_, long p_162369_, long p_162370_) {
+      this.f_162356_ += p_162368_;
+      this.f_162357_ += p_162369_;
+      this.f_162358_ += p_162370_;
+      this.f_162359_ += p_162368_;
+      this.f_162360_ += p_162369_;
+      this.f_162361_ += p_162370_;
+      return this;
+   }
+
+   /** @deprecated */
+   @Deprecated
    public BoundingBox m_162373_(Vec3i p_162374_) {
       return this.m_162367_(p_162374_.m_123341_(), p_162374_.m_123342_(), p_162374_.m_123343_());
    }
 
+   /** @deprecated */
+   @Deprecated
+   public BoundingBox move(me.alphamode.mclong.core.Vec3l p_162374_) {
+      return this.m_162367_((int) p_162374_.getX(), (int) p_162374_.getY(), (int) p_162374_.getZ());
+   }
+
    public BoundingBox m_71045_(int p_71046_, int p_71047_, int p_71048_) {
       return new BoundingBox(this.f_162356_ + p_71046_, this.f_162357_ + p_71047_, this.f_162358_ + p_71048_, this.f_162359_ + p_71046_, this.f_162360_ + p_71047_, this.f_162361_ + p_71048_);
    }
@@ -162,8 +_,13 @@
       return new BoundingBox(this.m_162395_() - p_191962_, this.m_162396_() - p_191962_, this.m_162398_() - p_191962_, this.m_162399_() + p_191962_, this.m_162400_() + p_191962_, this.m_162401_() + p_191962_);
    }
 
+   @Deprecated
    public boolean m_71051_(Vec3i p_71052_) {
       return p_71052_.m_123341_() >= this.f_162356_ && p_71052_.m_123341_() <= this.f_162359_ && p_71052_.m_123343_() >= this.f_162358_ && p_71052_.m_123343_() <= this.f_162361_ && p_71052_.m_123342_() >= this.f_162357_ && p_71052_.m_123342_() <= this.f_162360_;
+   }
+
+   public boolean isInside(me.alphamode.mclong.core.Vec3l p_71052_) {
+      return p_71052_.getX() >= this.f_162356_ && p_71052_.getX() <= this.f_162359_ && p_71052_.getZ() >= this.f_162358_ && p_71052_.getZ() <= this.f_162361_ && p_71052_.getY() >= this.f_162357_ && p_71052_.getY() <= this.f_162360_;
    }
 
    public Vec3i m_71053_() {
