--- a/net/minecraft/world/level/levelgen/structure/TemplateStructurePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/TemplateStructurePiece.java
@@ -46,7 +_,7 @@
       super(p_226894_, p_226895_);
       this.m_73519_(Direction.NORTH);
       this.f_163658_ = p_226895_.m_128461_("Template");
-      this.f_73658_ = new BlockPos(p_226895_.m_128451_("TPX"), p_226895_.m_128451_("TPY"), p_226895_.m_128451_("TPZ"));
+      this.f_73658_ = new BlockPos(new me.alphamode.mclong.math.BigInteger(p_226895_.m_128461_("TPX")), new me.alphamode.mclong.math.BigInteger(p_226895_.m_128461_("TPY")), new me.alphamode.mclong.math.BigInteger(p_226895_.m_128461_("TPZ")));
       ResourceLocation resourcelocation = this.m_142415_();
       this.f_73656_ = p_226896_.m_230359_(resourcelocation);
       this.f_73657_ = p_226897_.apply(resourcelocation);
@@ -58,9 +_,9 @@
    }
 
    protected void m_183620_(StructurePieceSerializationContext p_192690_, CompoundTag p_192691_) {
-      p_192691_.m_128405_("TPX", this.f_73658_.m_123341_());
-      p_192691_.m_128405_("TPY", this.f_73658_.m_123342_());
-      p_192691_.m_128405_("TPZ", this.f_73658_.m_123343_());
+      p_192691_.m_128359_("TPX", this.f_73658_.getBigX().toString());
+      p_192691_.m_128359_("TPY", this.f_73658_.getBigY().toString());
+      p_192691_.m_128359_("TPZ", this.f_73658_.getBigZ().toString());
       p_192691_.m_128359_("Template", this.f_163658_);
    }
 
@@ -100,8 +_,8 @@
    /** @deprecated */
    @Deprecated
    public void m_6324_(int p_73668_, int p_73669_, int p_73670_) {
-      super.m_6324_(p_73668_, p_73669_, p_73670_);
-      this.f_73658_ = this.f_73658_.m_7918_(p_73668_, p_73669_, p_73670_);
+      super.move(p_73668_, p_73669_, p_73670_);
+      this.f_73658_ = this.f_73658_.offset(p_73668_, p_73669_, p_73670_);
    }
 
    public Rotation m_6830_() {
