--- a/net/minecraft/core/BlockSourceImpl.java
+++ b/net/minecraft/core/BlockSourceImpl.java
@@ -18,15 +_,27 @@
    }
 
    public double m_7096_() {
-      return (double)this.f_122211_.m_123341_() + 0.5D;
+      return (double)this.f_122211_.getX() + 0.5D;
    }
 
    public double m_7098_() {
-      return (double)this.f_122211_.m_123342_() + 0.5D;
+      return (double)this.f_122211_.getY() + 0.5D;
    }
 
    public double m_7094_() {
-      return (double)this.f_122211_.m_123343_() + 0.5D;
+      return (double)this.f_122211_.getZ() + 0.5D;
+   }
+
+   public me.alphamode.mclong.math.BigDecimal getX() {
+      return this.f_122211_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB);
+   }
+
+   public me.alphamode.mclong.math.BigDecimal getY() {
+      return this.f_122211_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB);
+   }
+
+   public me.alphamode.mclong.math.BigDecimal getZ() {
+      return this.f_122211_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB);
    }
 
    public BlockPos m_7961_() {
