--- a/net/minecraft/core/Cursor3D.java
+++ b/net/minecraft/core/Cursor3D.java
@@ -5,25 +_,25 @@
    public static final int f_175353_ = 1;
    public static final int f_175354_ = 2;
    public static final int f_175355_ = 3;
-   private final int f_122286_;
-   private final int f_122287_;
-   private final int f_122288_;
-   private final int f_122289_;
-   private final int f_122290_;
-   private final int f_122291_;
-   private final int f_122292_;
+   private final me.alphamode.mclong.math.BigInteger f_122286_;
+   private final me.alphamode.mclong.math.BigInteger f_122287_;
+   private final me.alphamode.mclong.math.BigInteger f_122288_;
+   private final long f_122289_;
+   private final long f_122290_;
+   private final long f_122291_;
+   private final long f_122292_;
    private int f_122293_;
-   private int f_122294_;
-   private int f_122295_;
-   private int f_122296_;
+   private me.alphamode.mclong.math.BigInteger f_122294_;
+   private me.alphamode.mclong.math.BigInteger f_122295_;
+   private me.alphamode.mclong.math.BigInteger f_122296_;
 
-   public Cursor3D(int p_122298_, int p_122299_, int p_122300_, int p_122301_, int p_122302_, int p_122303_) {
-      this.f_122286_ = p_122298_;
-      this.f_122287_ = p_122299_;
-      this.f_122288_ = p_122300_;
-      this.f_122289_ = p_122301_ - p_122298_ + 1;
-      this.f_122290_ = p_122302_ - p_122299_ + 1;
-      this.f_122291_ = p_122303_ - p_122300_ + 1;
+   public Cursor3D(me.alphamode.mclong.math.BigInteger pOriginX, me.alphamode.mclong.math.BigInteger pOriginY, me.alphamode.mclong.math.BigInteger pOriginZ, me.alphamode.mclong.math.BigInteger pEndX, me.alphamode.mclong.math.BigInteger pEndY, me.alphamode.mclong.math.BigInteger pEndZ) {
+      this.f_122286_ = pOriginX;
+      this.f_122287_ = pOriginY;
+      this.f_122288_ = pOriginZ;
+      this.f_122289_ = pEndX.subtract(pOriginX).longValue() + 1;
+      this.f_122290_ = pEndY.subtract(pOriginY).longValue() + 1;
+      this.f_122291_ = pEndZ.subtract(pOriginZ).longValue() + 1;
       this.f_122292_ = this.f_122289_ * this.f_122290_ * this.f_122291_;
    }
 
@@ -31,38 +_,38 @@
       if (this.f_122293_ == this.f_122292_) {
          return false;
       } else {
-         this.f_122294_ = this.f_122293_ % this.f_122289_;
-         int i = this.f_122293_ / this.f_122289_;
-         this.f_122295_ = i % this.f_122290_;
-         this.f_122296_ = i / this.f_122290_;
+         this.f_122294_ = me.alphamode.mclong.math.BigInteger.valueOf(this.f_122293_ % this.f_122289_);
+         long i = this.f_122293_ / this.f_122289_;
+         this.f_122295_ = me.alphamode.mclong.math.BigInteger.valueOf(i % this.f_122290_);
+         this.f_122296_ = me.alphamode.mclong.math.BigInteger.valueOf(i / this.f_122290_);
          ++this.f_122293_;
          return true;
       }
    }
 
-   public int m_122305_() {
-      return this.f_122286_ + this.f_122294_;
-   }
-
-   public int m_122306_() {
-      return this.f_122287_ + this.f_122295_;
-   }
-
-   public int m_122307_() {
-      return this.f_122288_ + this.f_122296_;
+   public me.alphamode.mclong.math.BigInteger nextX() {
+      return this.f_122286_.add(this.f_122294_);
+   }
+
+   public me.alphamode.mclong.math.BigInteger nextY() {
+      return this.f_122287_.add(this.f_122295_);
+   }
+
+   public me.alphamode.mclong.math.BigInteger nextZ() {
+      return this.f_122288_.add(this.f_122296_);
    }
 
    public int m_122308_() {
       int i = 0;
-      if (this.f_122294_ == 0 || this.f_122294_ == this.f_122289_ - 1) {
-         ++i;
-      }
-
-      if (this.f_122295_ == 0 || this.f_122295_ == this.f_122290_ - 1) {
-         ++i;
-      }
-
-      if (this.f_122296_ == 0 || this.f_122296_ == this.f_122291_ - 1) {
+      if (this.f_122294_.equals(me.alphamode.mclong.math.BigInteger.ZERO) || this.f_122294_.equals(me.alphamode.mclong.math.BigInteger.valueOf(this.f_122289_ - 1))) {
+         ++i;
+      }
+
+      if (this.f_122295_.equals(me.alphamode.mclong.math.BigInteger.ZERO) || this.f_122295_.equals(me.alphamode.mclong.math.BigInteger.valueOf(this.f_122290_ - 1))) {
+         ++i;
+      }
+
+      if (this.f_122296_.equals(me.alphamode.mclong.math.BigInteger.ZERO) || this.f_122296_.equals(me.alphamode.mclong.math.BigInteger.valueOf(this.f_122291_ - 1))) {
          ++i;
       }
 
