--- a/net/minecraft/core/Direction.java
+++ b/net/minecraft/core/Direction.java
@@ -507,6 +_,10 @@
          public double m_6150_(double p_122492_, double p_122493_, double p_122494_) {
             return p_122492_;
          }
+
+         public java.math.BigDecimal choose(java.math.BigDecimal x, java.math.BigDecimal y, java.math.BigDecimal z) {
+            return x;
+         }
       },
       Y("y") {
          public int m_7863_(int p_122510_, int p_122511_, int p_122512_) {
@@ -516,6 +_,10 @@
          public double m_6150_(double p_122506_, double p_122507_, double p_122508_) {
             return p_122507_;
          }
+
+         public java.math.BigDecimal choose(java.math.BigDecimal x, java.math.BigDecimal y, java.math.BigDecimal z) {
+            return y;
+         }
       },
       Z("z") {
          public int m_7863_(int p_122524_, int p_122525_, int p_122526_) {
@@ -525,6 +_,10 @@
          public double m_6150_(double p_122520_, double p_122521_, double p_122522_) {
             return p_122522_;
          }
+
+         public java.math.BigDecimal choose(java.math.BigDecimal x, java.math.BigDecimal y, java.math.BigDecimal z) {
+            return z;
+         }
       };
 
       public static final Direction.Axis[] f_122448_ = values();
@@ -588,6 +_,8 @@
       public abstract int m_7863_(int p_122466_, int p_122467_, int p_122468_);
 
       public abstract double m_6150_(double p_122463_, double p_122464_, double p_122465_);
+
+      public abstract java.math.BigDecimal choose(java.math.BigDecimal x, java.math.BigDecimal y, java.math.BigDecimal z);
    }
 
    public static enum AxisDirection {
