--- a/net/minecraft/core/Direction.java
+++ b/net/minecraft/core/Direction.java
@@ -400,8 +_,8 @@
    }
 
    @Nullable
-   public static Direction m_122378_(int p_122379_, int p_122380_, int p_122381_) {
-      return f_122350_.get(BlockPos.m_121882_(p_122379_, p_122380_, p_122381_));
+   public static Direction fromNormal(long p_122379_, long p_122380_, long p_122381_) {
+      return f_122350_.get(BlockPos.asLong(p_122379_, p_122380_, p_122381_));
    }
 
    public static Direction m_122364_(double p_122365_) {
