--- a/net/minecraft/core/PositionImpl.java
+++ b/net/minecraft/core/PositionImpl.java
@@ -22,4 +_,16 @@
    public double m_7094_() {
       return this.f_122800_;
    }
+
+   public java.math.BigDecimal getX() {
+      return java.math.BigDecimal.valueOf(this.m_7096_());
+   }
+
+   public java.math.BigDecimal getY() {
+      return java.math.BigDecimal.valueOf(this.m_7098_());
+   }
+
+   public java.math.BigDecimal getZ() {
+      return java.math.BigDecimal.valueOf(this.m_7094_());
+   }
 }
