--- a/net/minecraft/core/QuartPos.java
+++ b/net/minecraft/core/QuartPos.java
@@ -9,23 +_,39 @@
    private QuartPos() {
    }
 
-   public static int m_175400_(int p_175401_) {
-      return p_175401_ >> 2;
+   public static long fromBlock(long pValue) {
+      return pValue >> 2;
+   }
+
+   public static me.alphamode.mclong.math.BigInteger fromBlock(me.alphamode.mclong.math.BigInteger pValue) {
+      return pValue.shiftRight(2);
    }
 
    public static int m_198376_(int p_198377_) {
       return p_198377_ & 3;
    }
 
-   public static int m_175402_(int p_175403_) {
-      return p_175403_ << 2;
-   }
-
-   public static int m_175404_(int p_175405_) {
-      return p_175405_ << 2;
-   }
-
-   public static int m_175406_(int p_175407_) {
-      return p_175407_ >> 2;
+   public static long toBlock(long pValue) {
+      return pValue << 2;
+   }
+
+   public static long fromSection(long pValue) {
+      return pValue << 2;
+   }
+
+   public static me.alphamode.mclong.math.BigInteger fromSection(me.alphamode.mclong.math.BigInteger pValue) {
+      return pValue.shiftLeft(2);
+   }
+
+   public static long toSection(long pValue) {
+      return pValue >> 2;
+   }
+
+   public static me.alphamode.mclong.math.BigInteger toSection(me.alphamode.mclong.math.BigInteger pValue) {
+      return pValue.shiftRight(2);
+   }
+
+   public static me.alphamode.mclong.math.BigInteger toBlock(me.alphamode.mclong.math.BigInteger pValue) {
+      return pValue.shiftLeft(2);
    }
 }
