--- a/net/minecraft/world/entity/monster/piglin/Piglin.java
+++ b/net/minecraft/world/entity/monster/piglin/Piglin.java
@@ -137,7 +_,7 @@
    }
 
    public static boolean m_219197_(EntityType<Piglin> p_219198_, LevelAccessor p_219199_, MobSpawnType p_219200_, BlockPos p_219201_, RandomSource p_219202_) {
-      return !p_219199_.m_8055_(p_219201_.m_7495_()).m_60713_(Blocks.f_50451_);
+      return !p_219199_.m_8055_(p_219201_.below()).m_60713_(Blocks.f_50451_);
    }
 
    @Nullable
