--- a/net/minecraft/world/entity/ai/goal/AvoidEntityGoal.java
+++ b/net/minecraft/world/entity/ai/goal/AvoidEntityGoal.java
@@ -62,10 +_,10 @@
          Vec3 vec3 = DefaultRandomPos.m_148407_(this.f_25015_, 16, 7, this.f_25016_.m_20182_());
          if (vec3 == null) {
             return false;
-         } else if (this.f_25016_.m_20275_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_) < this.f_25016_.m_20280_(this.f_25015_)) {
+         } else if (this.f_25016_.m_20275_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_()) < this.f_25016_.m_20280_(this.f_25015_)) {
             return false;
          } else {
-            this.f_25018_ = this.f_25019_.m_26524_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, 0);
+            this.f_25018_ = this.f_25019_.m_26524_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), 0);
             return this.f_25018_ != null;
          }
       }
