--- a/net/minecraft/world/entity/ai/goal/BegGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BegGoal.java
@@ -53,7 +_,7 @@
    }
 
    public void m_8037_() {
-      this.f_25056_.m_21563_().m_24950_(this.f_25057_.m_20185_(), this.f_25057_.m_20188_(), this.f_25057_.m_20189_(), 10.0F, (float)this.f_25056_.m_8132_());
+      this.f_25056_.m_21563_().setLookAt(this.f_25057_.getBigX(), this.f_25057_.getEyeBigY(), this.f_25057_.getBigZ(), 10.0F, (float)this.f_25056_.m_8132_());
       --this.f_25060_;
    }
 
