--- a/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
@@ -43,7 +_,7 @@
    }
 
    public boolean m_8045_() {
-      return this.f_25082_ <= this.m_25100_() && !this.m_25200_() && this.f_25190_.m_203195_(this.f_25189_.m_20182_(), 2.0D) && this.m_25094_(this.f_25189_.m_9236_().m_46791_());
+      return this.f_25082_ <= this.m_25100_() && !this.m_25200_() && this.f_25190_.closerToCenterThan(this.f_25189_.m_20182_(), 2.0D) && this.m_25094_(this.f_25189_.m_9236_().m_46791_());
    }
 
    public void m_8041_() {
