--- a/net/minecraft/world/entity/ai/goal/ClimbOnTopOfPowderSnowGoal.java
+++ b/net/minecraft/world/entity/ai/goal/ClimbOnTopOfPowderSnowGoal.java
@@ -22,7 +_,7 @@
    public boolean m_8036_() {
       boolean flag = this.f_204052_.f_146809_ || this.f_204052_.f_146808_;
       if (flag && this.f_204052_.m_6095_().m_204039_(EntityTypeTags.f_144291_)) {
-         BlockPos blockpos = this.f_204052_.m_20183_().m_7494_();
+         BlockPos blockpos = this.f_204052_.m_20183_().above();
          BlockState blockstate = this.f_204053_.m_8055_(blockpos);
          return blockstate.m_60713_(Blocks.f_152499_) || blockstate.m_60812_(this.f_204053_, blockpos) == Shapes.m_83040_();
       } else {
