--- a/net/minecraft/world/entity/ai/goal/DolphinJumpGoal.java
+++ b/net/minecraft/world/entity/ai/goal/DolphinJumpGoal.java
@@ -40,12 +_,12 @@
    }
 
    private boolean m_25172_(BlockPos p_25173_, int p_25174_, int p_25175_, int p_25176_) {
-      BlockPos blockpos = p_25173_.m_7918_(p_25174_ * p_25176_, 0, p_25175_ * p_25176_);
+      BlockPos blockpos = p_25173_.offset(p_25174_ * p_25176_, 0, p_25175_ * p_25176_);
       return this.f_25163_.f_19853_.m_6425_(blockpos).m_205070_(FluidTags.f_13131_) && !this.f_25163_.f_19853_.m_8055_(blockpos).m_60767_().m_76334_();
    }
 
    private boolean m_25178_(BlockPos p_25179_, int p_25180_, int p_25181_, int p_25182_) {
-      return this.f_25163_.f_19853_.m_8055_(p_25179_.m_7918_(p_25180_ * p_25182_, 1, p_25181_ * p_25182_)).m_60795_() && this.f_25163_.f_19853_.m_8055_(p_25179_.m_7918_(p_25180_ * p_25182_, 2, p_25181_ * p_25182_)).m_60795_();
+      return this.f_25163_.f_19853_.m_8055_(p_25179_.offset(p_25180_ * p_25182_, 1, p_25181_ * p_25182_)).m_60795_() && this.f_25163_.f_19853_.m_8055_(p_25179_.offset(p_25180_ * p_25182_, 2, p_25181_ * p_25182_)).m_60795_();
    }
 
    public boolean m_8045_() {
