--- a/net/minecraft/world/entity/ai/goal/EatBlockGoal.java
+++ b/net/minecraft/world/entity/ai/goal/EatBlockGoal.java
@@ -32,7 +_,7 @@
          if (f_25201_.test(this.f_25203_.m_8055_(blockpos))) {
             return true;
          } else {
-            return this.f_25203_.m_8055_(blockpos.m_7495_()).m_60713_(Blocks.f_50440_);
+            return this.f_25203_.m_8055_(blockpos.below()).m_60713_(Blocks.f_50440_);
          }
       }
    }
@@ -66,7 +_,7 @@
 
             this.f_25202_.m_8035_();
          } else {
-            BlockPos blockpos1 = blockpos.m_7495_();
+            BlockPos blockpos1 = blockpos.below();
             if (this.f_25203_.m_8055_(blockpos1).m_60713_(Blocks.f_50440_)) {
                if (this.f_25203_.m_46469_().m_46207_(GameRules.f_46132_)) {
                   this.f_25203_.m_46796_(2001, blockpos1, Block.m_49956_(Blocks.f_50440_.m_49966_()));
