--- a/net/minecraft/world/entity/ai/goal/FleeSunGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FleeSunGoal.java
@@ -11,9 +_,9 @@
 
 public class FleeSunGoal extends Goal {
    protected final PathfinderMob f_25214_;
-   private double f_25215_;
-   private double f_25216_;
-   private double f_25217_;
+   private me.alphamode.mclong.math.BigDecimal f_25215_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_25216_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_25217_ = me.alphamode.mclong.math.BigDecimal.ZERO;
    private final double f_25218_;
    private final Level f_25219_;
 
@@ -43,9 +_,9 @@
       if (vec3 == null) {
          return false;
       } else {
-         this.f_25215_ = vec3.f_82479_;
-         this.f_25216_ = vec3.f_82480_;
-         this.f_25217_ = vec3.f_82481_;
+         this.f_25215_ = vec3.getX();
+         this.f_25216_ = vec3.getY();
+         this.f_25217_ = vec3.getZ();
          return true;
       }
    }
@@ -55,7 +_,7 @@
    }
 
    public void m_8056_() {
-      this.f_25214_.m_21573_().m_26519_(this.f_25215_, this.f_25216_, this.f_25217_, this.f_25218_);
+      this.f_25214_.m_21573_().moveTo(this.f_25215_, this.f_25216_, this.f_25217_, this.f_25218_);
    }
 
    @Nullable
@@ -64,9 +_,9 @@
       BlockPos blockpos = this.f_25214_.m_20183_();
 
       for(int i = 0; i < 10; ++i) {
-         BlockPos blockpos1 = blockpos.m_7918_(randomsource.m_188503_(20) - 10, randomsource.m_188503_(6) - 3, randomsource.m_188503_(20) - 10);
+         BlockPos blockpos1 = blockpos.offset(randomsource.m_188503_(20) - 10, randomsource.m_188503_(6) - 3, randomsource.m_188503_(20) - 10);
          if (!this.f_25219_.m_45527_(blockpos1) && this.f_25214_.m_21692_(blockpos1) < 0.0F) {
-            return Vec3.m_82539_(blockpos1);
+            return Vec3.atBottomCenterOf(blockpos1);
          }
       }
 
