--- a/net/minecraft/world/entity/ai/goal/FollowOwnerGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowOwnerGoal.java
@@ -103,7 +_,7 @@
          int j = this.m_25300_(-3, 3);
          int k = this.m_25300_(-1, 1);
          int l = this.m_25300_(-3, 3);
-         boolean flag = this.m_25303_(blockpos.m_123341_() + j, blockpos.m_123342_() + k, blockpos.m_123343_() + l);
+         boolean flag = this.maybeTeleportTo(blockpos.getX() + j, blockpos.getY() + k, blockpos.getZ() + l);
          if (flag) {
             return;
          }
@@ -111,7 +_,7 @@
 
    }
 
-   private boolean m_25303_(int p_25304_, int p_25305_, int p_25306_) {
+   private boolean maybeTeleportTo(long p_25304_, long p_25305_, long p_25306_) {
       if (Math.abs((double)p_25304_ - this.f_25284_.m_20185_()) < 2.0D && Math.abs((double)p_25306_ - this.f_25284_.m_20189_()) < 2.0D) {
          return false;
       } else if (!this.m_25307_(new BlockPos(p_25304_, p_25305_, p_25306_))) {
@@ -128,11 +_,11 @@
       if (blockpathtypes != BlockPathTypes.WALKABLE) {
          return false;
       } else {
-         BlockState blockstate = this.f_25285_.m_8055_(p_25308_.m_7495_());
+         BlockState blockstate = this.f_25285_.m_8055_(p_25308_.below());
          if (!this.f_25292_ && blockstate.m_60734_() instanceof LeavesBlock) {
             return false;
          } else {
-            BlockPos blockpos = p_25308_.m_121996_(this.f_25283_.m_20183_());
+            BlockPos blockpos = p_25308_.subtract(this.f_25283_.m_20183_());
             return this.f_25285_.m_45756_(this.f_25283_, this.f_25283_.m_20191_().m_82338_(blockpos));
          }
       }
