--- a/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
@@ -70,17 +_,17 @@
       if (--this.f_25233_ <= 0) {
          this.f_25233_ = this.m_183277_(10);
          if (this.f_25236_ == BoatGoals.GO_TO_BOAT) {
-            BlockPos blockpos = this.f_25235_.m_20183_().m_121945_(this.f_25235_.m_6350_().m_122424_());
+            BlockPos blockpos = this.f_25235_.m_20183_().relative(this.f_25235_.m_6350_().m_122424_());
             blockpos = blockpos.m_7918_(0, -1, 0);
-            this.f_25234_.m_21573_().m_26519_((double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), 1.0D);
+            this.f_25234_.m_21573_().m_26519_((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), 1.0D);
             if (this.f_25234_.m_20270_(this.f_25235_) < 4.0F) {
                this.f_25233_ = 0;
                this.f_25236_ = BoatGoals.GO_IN_BOAT_DIRECTION;
             }
          } else if (this.f_25236_ == BoatGoals.GO_IN_BOAT_DIRECTION) {
             Direction direction = this.f_25235_.m_6374_();
-            BlockPos blockpos1 = this.f_25235_.m_20183_().m_5484_(direction, 10);
-            this.f_25234_.m_21573_().m_26519_((double)blockpos1.m_123341_(), (double)(blockpos1.m_123342_() - 1), (double)blockpos1.m_123343_(), 1.0D);
+            BlockPos blockpos1 = this.f_25235_.m_20183_().relative(direction, 10);
+            this.f_25234_.m_21573_().m_26519_((double)blockpos1.getX(), (double)(blockpos1.getY() - 1), (double)blockpos1.getZ(), 1.0D);
             if (this.f_25234_.m_20270_(this.f_25235_) > 12.0F) {
                this.f_25233_ = 0;
                this.f_25236_ = BoatGoals.GO_TO_BOAT;
