--- a/net/minecraft/world/entity/ai/goal/GolemRandomStrollInVillageGoal.java
+++ b/net/minecraft/world/entity/ai/goal/GolemRandomStrollInVillageGoal.java
@@ -72,7 +_,7 @@
          return null;
       } else {
          BlockPos blockpos = this.m_25407_(sectionpos);
-         return blockpos == null ? null : LandRandomPos.m_148492_(this.f_25725_, 10, 7, Vec3.m_82539_(blockpos));
+         return blockpos == null ? null : LandRandomPos.m_148492_(this.f_25725_, 10, 7, Vec3.atBottomCenterOf(blockpos));
       }
    }
 
