--- a/net/minecraft/world/entity/ai/goal/LeapAtTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LeapAtTargetGoal.java
@@ -49,6 +_,6 @@
          vec31 = vec31.m_82541_().m_82490_(0.4D).m_82549_(vec3.m_82490_(0.2D));
       }
 
-      this.f_25488_.m_20334_(vec31.f_82479_, (double)this.f_25490_, vec31.f_82481_);
+      this.f_25488_.m_20334_(vec31.m_7096_(), (double)this.f_25490_, vec31.m_7094_());
    }
 }
