--- a/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+++ b/net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
@@ -104,8 +_,8 @@
             Llama llama = this.f_25497_.m_30812_();
             double d0 = (double)this.f_25497_.m_20270_(llama);
             float f = 2.0F;
-            Vec3 vec3 = (new Vec3(llama.m_20185_() - this.f_25497_.m_20185_(), llama.m_20186_() - this.f_25497_.m_20186_(), llama.m_20189_() - this.f_25497_.m_20189_())).m_82541_().m_82490_(Math.max(d0 - 2.0D, 0.0D));
-            this.f_25497_.m_21573_().m_26519_(this.f_25497_.m_20185_() + vec3.f_82479_, this.f_25497_.m_20186_() + vec3.f_82480_, this.f_25497_.m_20189_() + vec3.f_82481_, this.f_25498_);
+            Vec3 vec3 = (new Vec3(llama.getBigX().subtract(this.f_25497_.getBigX()), llama.getBigY().subtract(this.f_25497_.getBigY()), llama.getBigZ().subtract(this.f_25497_.getBigZ()))).m_82541_().m_82490_(Math.max(d0 - 2.0D, 0.0D));
+            this.f_25497_.m_21573_().moveTo(this.f_25497_.getBigX().add(vec3.getX()), this.f_25497_.getBigY().add(vec3.getY()), this.f_25497_.getBigZ().add(vec3.getZ()), this.f_25498_);
          }
       }
    }
