--- a/net/minecraft/world/entity/ai/goal/MoveBackToVillageGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveBackToVillageGoal.java
@@ -29,6 +_,6 @@
       BlockPos blockpos = this.f_25725_.m_20183_();
       SectionPos sectionpos = SectionPos.m_123199_(blockpos);
       SectionPos sectionpos1 = BehaviorUtils.m_22581_(serverlevel, sectionpos, 2);
-      return sectionpos1 != sectionpos ? DefaultRandomPos.m_148412_(this.f_25725_, 10, 7, Vec3.m_82539_(sectionpos1.m_123250_()), (double)((float)Math.PI / 2F)) : null;
+      return sectionpos1 != sectionpos ? DefaultRandomPos.m_148412_(this.f_25725_, 10, 7, Vec3.atBottomCenterOf(sectionpos1.m_123250_()), (double)((float)Math.PI / 2F)) : null;
    }
 }
