--- a/net/minecraft/world/entity/ai/goal/MoveTowardsRestrictionGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveTowardsRestrictionGoal.java
@@ -22,7 +_,7 @@
       if (this.f_25627_.m_21533_()) {
          return false;
       } else {
-         Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25627_, 16, 7, Vec3.m_82539_(this.f_25627_.m_21534_()), (double)((float)Math.PI / 2F));
+         Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25627_, 16, 7, Vec3.atBottomCenterOf(this.f_25627_.m_21534_()), (double)((float)Math.PI / 2F));
          if (vec3 == null) {
             return false;
          } else {
