--- a/net/minecraft/world/entity/ai/goal/MoveTowardsRestrictionGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveTowardsRestrictionGoal.java
@@ -22,13 +_,13 @@
       if (this.f_25627_.m_21533_()) {
          return false;
       } else {
-         Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25627_, 16, 7, Vec3.m_82539_(this.f_25627_.m_21534_()), (double)((float)Math.PI / 2F));
+         Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25627_, 16, 7, Vec3.atBottomCenterOf(this.f_25627_.m_21534_()), (double)((float)Math.PI / 2F));
          if (vec3 == null) {
             return false;
          } else {
-            this.f_25628_ = vec3.f_82479_;
-            this.f_25629_ = vec3.f_82480_;
-            this.f_25630_ = vec3.f_82481_;
+            this.f_25628_ = vec3.m_7096_();
+            this.f_25629_ = vec3.m_7098_();
+            this.f_25630_ = vec3.m_7094_();
             return true;
          }
       }
