--- a/net/minecraft/world/entity/ai/goal/MoveTowardsTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/MoveTowardsTargetGoal.java
@@ -35,9 +_,9 @@
          if (vec3 == null) {
             return false;
          } else {
-            this.f_25640_ = vec3.f_82479_;
-            this.f_25641_ = vec3.f_82480_;
-            this.f_25642_ = vec3.f_82481_;
+            this.f_25640_ = vec3.m_7096_();
+            this.f_25641_ = vec3.m_7098_();
+            this.f_25642_ = vec3.m_7094_();
             return true;
          }
       }
