--- a/net/minecraft/world/entity/ai/goal/OcelotAttackGoal.java
+++ b/net/minecraft/world/entity/ai/goal/OcelotAttackGoal.java
@@ -46,7 +_,7 @@
    public void m_8037_() {
       this.f_25654_.m_21563_().m_24960_(this.f_25655_, 30.0F, 30.0F);
       double d0 = (double)(this.f_25654_.m_20205_() * 2.0F * this.f_25654_.m_20205_() * 2.0F);
-      double d1 = this.f_25654_.m_20275_(this.f_25655_.m_20185_(), this.f_25655_.m_20186_(), this.f_25655_.m_20189_());
+      double d1 = this.f_25654_.distanceToSqr(this.f_25655_.getBigX(), this.f_25655_.getBigY(), this.f_25655_.getBigZ());
       double d2 = 0.8D;
       if (d1 > d0 && d1 < 16.0D) {
          d2 = 1.33D;
