--- a/net/minecraft/world/entity/ai/goal/OfferFlowerGoal.java
+++ b/net/minecraft/world/entity/ai/goal/OfferFlowerGoal.java
@@ -23,7 +_,7 @@
       } else if (this.f_25664_.m_217043_().m_188503_(8000) != 0) {
          return false;
       } else {
-         this.f_25665_ = this.f_25664_.m_9236_().m_45963_(Villager.class, f_25663_, this.f_25664_, this.f_25664_.m_20185_(), this.f_25664_.m_20186_(), this.f_25664_.m_20189_(), this.f_25664_.m_20191_().m_82377_(6.0D, 2.0D, 6.0D));
+         this.f_25665_ = this.f_25664_.m_9236_().getNearestEntity(Villager.class, f_25663_, this.f_25664_, this.f_25664_.getBigX(), this.f_25664_.getBigY(), this.f_25664_.getBigZ(), this.f_25664_.m_20191_().m_82377_(6.0D, 2.0D, 6.0D));
          return this.f_25665_ != null;
       }
    }
