--- a/net/minecraft/world/entity/ai/goal/PanicGoal.java
+++ b/net/minecraft/world/entity/ai/goal/PanicGoal.java
@@ -32,9 +_,9 @@
          if (this.f_25684_.m_6060_()) {
             BlockPos blockpos = this.m_198172_(this.f_25684_.m_9236_(), this.f_25684_, 5);
             if (blockpos != null) {
-               this.f_25686_ = (double)blockpos.m_123341_();
-               this.f_25687_ = (double)blockpos.m_123342_();
-               this.f_25688_ = (double)blockpos.m_123343_();
+               this.f_25686_ = (double)blockpos.getX();
+               this.f_25687_ = (double)blockpos.getY();
+               this.f_25688_ = (double)blockpos.getZ();
                return true;
             }
          }
@@ -52,9 +_,9 @@
       if (vec3 == null) {
          return false;
       } else {
-         this.f_25686_ = vec3.f_82479_;
-         this.f_25687_ = vec3.f_82480_;
-         this.f_25688_ = vec3.f_82481_;
+         this.f_25686_ = vec3.m_7096_();
+         this.f_25687_ = vec3.m_7098_();
+         this.f_25688_ = vec3.m_7094_();
          return true;
       }
    }
