--- a/net/minecraft/world/entity/ai/goal/PanicGoal.java
+++ b/net/minecraft/world/entity/ai/goal/PanicGoal.java
@@ -32,9 +_,9 @@
          if (this.f_25684_.m_6060_()) {
             BlockPos blockpos = this.m_198172_(this.f_25684_.f_19853_, this.f_25684_, 5);
             if (blockpos != null) {
-               this.f_25686_ = (double)blockpos.m_123341_();
-               this.f_25687_ = (double)blockpos.m_123342_();
-               this.f_25688_ = (double)blockpos.m_123343_();
+               this.f_25686_ = (double)blockpos.getX();
+               this.f_25687_ = (double)blockpos.getY();
+               this.f_25688_ = (double)blockpos.getZ();
                return true;
             }
          }
