--- a/net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
+++ b/net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
@@ -40,9 +_,9 @@
          }
 
          if (!this.f_25704_.m_21691_()) {
-            Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25704_, 15, 4, Vec3.m_82539_(raid.m_37780_()), (double)((float)Math.PI / 2F));
+            Vec3 vec3 = DefaultRandomPos.m_148412_(this.f_25704_, 15, 4, Vec3.atBottomCenterOf(raid.m_37780_()), (double)((float)Math.PI / 2F));
             if (vec3 != null) {
-               this.f_25704_.m_21573_().m_26519_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, 1.0D);
+               this.f_25704_.m_21573_().m_26519_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), 1.0D);
             }
          }
       }
