--- a/net/minecraft/world/entity/ai/goal/RandomLookAroundGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RandomLookAroundGoal.java
@@ -35,6 +_,6 @@
 
    public void m_8037_() {
       --this.f_25718_;
-      this.f_25715_.m_21563_().m_24946_(this.f_25715_.m_20185_() + this.f_25716_, this.f_25715_.m_20188_(), this.f_25715_.m_20189_() + this.f_25717_);
+      this.f_25715_.m_21563_().setLookAt(this.f_25715_.getBigX().add(this.f_25716_), this.f_25715_.getEyeBigY(), this.f_25715_.getBigZ().add(this.f_25717_));
    }
 }
