--- a/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RandomStrollGoal.java
@@ -51,9 +_,9 @@
          if (vec3 == null) {
             return false;
          } else {
-            this.f_25726_ = vec3.f_82479_;
-            this.f_25727_ = vec3.f_82480_;
-            this.f_25728_ = vec3.f_82481_;
+            this.f_25726_ = vec3.m_7096_();
+            this.f_25727_ = vec3.m_7098_();
+            this.f_25728_ = vec3.m_7094_();
             this.f_25731_ = false;
             return true;
          }
