--- a/net/minecraft/world/entity/ai/goal/RangedBowAttackGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RangedBowAttackGoal.java
@@ -65,7 +_,7 @@
    public void m_8037_() {
       LivingEntity livingentity = this.f_25782_.m_5448_();
       if (livingentity != null) {
-         double d0 = this.f_25782_.m_20275_(livingentity.m_20185_(), livingentity.m_20186_(), livingentity.m_20189_());
+         double d0 = this.f_25782_.distanceToSqr(livingentity.getBigX(), livingentity.getBigY(), livingentity.getBigZ());
          boolean flag = this.f_25782_.m_21574_().m_148306_(livingentity);
          boolean flag1 = this.f_25787_ > 0;
          if (flag != flag1) {
