--- a/net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
@@ -64,7 +_,7 @@
    }
 
    public void m_8037_() {
-      double d0 = this.f_25757_.m_20275_(this.f_25759_.m_20185_(), this.f_25759_.m_20186_(), this.f_25759_.m_20189_());
+      double d0 = this.f_25757_.distanceToSqr(this.f_25759_.getBigX(), this.f_25759_.getBigY(), this.f_25759_.getBigZ());
       boolean flag = this.f_25757_.m_21574_().m_148306_(this.f_25759_);
       if (flag) {
          ++this.f_25762_;
