--- a/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java
+++ b/net/minecraft/world/entity/ai/goal/RunAroundLikeCrazyGoal.java
@@ -10,9 +_,9 @@
 public class RunAroundLikeCrazyGoal extends Goal {
    private final AbstractHorse f_25884_;
    private final double f_25885_;
-   private double f_25886_;
-   private double f_25887_;
-   private double f_25888_;
+   private me.alphamode.mclong.math.BigDecimal f_25886_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_25887_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   private me.alphamode.mclong.math.BigDecimal f_25888_ = me.alphamode.mclong.math.BigDecimal.ZERO;
 
    public RunAroundLikeCrazyGoal(AbstractHorse p_25890_, double p_25891_) {
       this.f_25884_ = p_25890_;
@@ -26,9 +_,9 @@
          if (vec3 == null) {
             return false;
          } else {
-            this.f_25886_ = vec3.f_82479_;
-            this.f_25887_ = vec3.f_82480_;
-            this.f_25888_ = vec3.f_82481_;
+            this.f_25886_ = vec3.getX();
+            this.f_25887_ = vec3.getY();
+            this.f_25888_ = vec3.getZ();
             return true;
          }
       } else {
@@ -37,7 +_,7 @@
    }
 
    public void m_8056_() {
-      this.f_25884_.m_21573_().m_26519_(this.f_25886_, this.f_25887_, this.f_25888_, this.f_25885_);
+      this.f_25884_.m_21573_().moveTo(this.f_25886_, this.f_25887_, this.f_25888_, this.f_25885_);
    }
 
    public boolean m_8045_() {
