--- a/net/minecraft/world/entity/ai/goal/TryFindWaterGoal.java
+++ b/net/minecraft/world/entity/ai/goal/TryFindWaterGoal.java
@@ -19,7 +_,7 @@
    public void m_8056_() {
       BlockPos blockpos = null;
 
-      for(BlockPos blockpos1 : BlockPos.m_121976_(Mth.m_14107_(this.f_25962_.m_20185_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20186_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20189_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20185_() + 2.0D), this.f_25962_.m_146904_(), Mth.m_14107_(this.f_25962_.m_20189_() + 2.0D))) {
+      for(BlockPos blockpos1 : BlockPos.betweenClosed(Mth.m_14107_(this.f_25962_.m_20185_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20186_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20189_() - 2.0D), Mth.m_14107_(this.f_25962_.m_20185_() + 2.0D), this.f_25962_.getBlockY(), Mth.m_14107_(this.f_25962_.m_20189_() + 2.0D))) {
          if (this.f_25962_.m_9236_().m_6425_(blockpos1).m_205070_(FluidTags.f_13131_)) {
             blockpos = blockpos1;
             break;
@@ -27,7 +_,7 @@
       }
 
       if (blockpos != null) {
-         this.f_25962_.m_21566_().m_6849_((double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_(), 1.0D);
+         this.f_25962_.m_21566_().m_6849_((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), 1.0D);
       }
 
    }
