--- a/net/minecraft/world/entity/ai/goal/WaterAvoidingRandomFlyingGoal.java
+++ b/net/minecraft/world/entity/ai/goal/WaterAvoidingRandomFlyingGoal.java
@@ -15,7 +_,7 @@
    protected Vec3 m_7037_() {
       Vec3 vec3 = this.f_25725_.m_20252_(0.0F);
       int i = 8;
-      Vec3 vec31 = HoverRandomPos.m_148465_(this.f_25725_, 8, 7, vec3.f_82479_, vec3.f_82481_, ((float)Math.PI / 2F), 3, 1);
-      return vec31 != null ? vec31 : AirAndWaterRandomPos.m_148357_(this.f_25725_, 8, 4, -2, vec3.f_82479_, vec3.f_82481_, (double)((float)Math.PI / 2F));
+      Vec3 vec31 = HoverRandomPos.m_148465_(this.f_25725_, 8, 7, vec3.m_7096_(), vec3.m_7094_(), ((float)Math.PI / 2F), 3, 1);
+      return vec31 != null ? vec31 : AirAndWaterRandomPos.m_148357_(this.f_25725_, 8, 4, -2, vec3.m_7096_(), vec3.m_7094_(), (double)((float)Math.PI / 2F));
    }
 }
