--- a/net/minecraft/world/Container.java
+++ b/net/minecraft/world/Container.java
@@ -89,7 +_,7 @@
       } else if (level.m_7702_(blockpos) != p_272877_) {
          return false;
       } else {
-         return p_272670_.m_20275_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.5D, (double)blockpos.m_123343_() + 0.5D) <= (double)(p_273411_ * p_273411_);
+         return p_272670_.m_20275_((double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.5D, (double)blockpos.getZ() + 0.5D) <= (double)(p_273411_ * p_273411_);
       }
    }
 }
