--- a/net/minecraft/world/RandomSequences.java
+++ b/net/minecraft/world/RandomSequences.java
@@ -55,6 +_,11 @@
             return randomsource.m_188505_();
          }
 
+         public long nextLong(long off) {
+            RandomSequences.this.m_77762_();
+            return randomsource.nextLong(off);
+         }
+
          public boolean m_188499_() {
             RandomSequences.this.m_77762_();
             return randomsource.m_188499_();
