--- a/net/minecraft/world/entity/animal/goat/Goat.java
+++ b/net/minecraft/world/entity/animal/goat/Goat.java
@@ -329,6 +_,6 @@
    }
 
    public static boolean m_218752_(EntityType<? extends Animal> p_218753_, LevelAccessor p_218754_, MobSpawnType p_218755_, BlockPos p_218756_, RandomSource p_218757_) {
-      return p_218754_.m_8055_(p_218756_.m_7495_()).m_204336_(BlockTags.f_184230_) && m_186209_(p_218754_, p_218756_);
+      return p_218754_.m_8055_(p_218756_.below()).m_204336_(BlockTags.f_184230_) && m_186209_(p_218754_, p_218756_);
    }
 }
