--- a/net/minecraft/world/entity/animal/goat/Goat.java
+++ b/net/minecraft/world/entity/animal/goat/Goat.java
@@ -300,7 +_,7 @@
          double d0 = (double)Mth.m_216283_(this.f_19796_, -0.2F, 0.2F);
          double d1 = (double)Mth.m_216283_(this.f_19796_, 0.3F, 0.7F);
          double d2 = (double)Mth.m_216283_(this.f_19796_, -0.2F, 0.2F);
-         ItemEntity itementity = new ItemEntity(this.m_9236_(), vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), itemstack, d0, d1, d2);
+         ItemEntity itementity = new ItemEntity(this.m_9236_(), vec3.getX(), vec3.getY(), vec3.getZ(), itemstack, d0, d1, d2);
          this.m_9236_().m_7967_(itementity);
          return true;
       }
@@ -329,6 +_,6 @@
    }
 
    public static boolean m_218752_(EntityType<? extends Animal> p_218753_, LevelAccessor p_218754_, MobSpawnType p_218755_, BlockPos p_218756_, RandomSource p_218757_) {
-      return p_218754_.m_8055_(p_218756_.m_7495_()).m_204336_(BlockTags.f_184230_) && m_186209_(p_218754_, p_218756_);
+      return p_218754_.m_8055_(p_218756_.below()).m_204336_(BlockTags.f_184230_) && m_186209_(p_218754_, p_218756_);
    }
 }
