--- a/net/minecraft/nbt/NbtUtils.java
+++ b/net/minecraft/nbt/NbtUtils.java
@@ -210,14 +_,14 @@
    }
 
    public static BlockPos m_129239_(CompoundTag p_129240_) {
-      return new BlockPos(p_129240_.m_128451_("X"), p_129240_.m_128451_("Y"), p_129240_.m_128451_("Z"));
+      return new BlockPos(p_129240_.m_128454_("X"), p_129240_.m_128454_("Y"), p_129240_.m_128454_("Z"));
    }
 
    public static CompoundTag m_129224_(BlockPos p_129225_) {
       CompoundTag compoundtag = new CompoundTag();
-      compoundtag.m_128405_("X", p_129225_.m_123341_());
-      compoundtag.m_128405_("Y", p_129225_.m_123342_());
-      compoundtag.m_128405_("Z", p_129225_.m_123343_());
+      compoundtag.m_128356_("X", p_129225_.getX());
+      compoundtag.m_128356_("Y", p_129225_.getY());
+      compoundtag.m_128356_("Z", p_129225_.getZ());
       return compoundtag;
    }
 
