--- a/net/minecraft/world/entity/ai/targeting/TargetingConditions.java
+++ b/net/minecraft/world/entity/ai/targeting/TargetingConditions.java
@@ -77,8 +_,8 @@
             if (this.f_26873_ > 0.0D) {
                double d0 = this.f_26878_ ? p_26887_.m_20968_(p_26886_) : 1.0D;
                double d1 = Math.max(this.f_26873_ * d0, 2.0D);
-               double d2 = p_26886_.m_20275_(p_26887_.m_20185_(), p_26887_.m_20186_(), p_26887_.m_20189_());
-               if (d2 > d1 * d1) {
+               me.alphamode.mclong.math.BigDecimal d2 = p_26886_.distanceToSqr(p_26887_.getBigX(), p_26887_.getBigY(), p_26887_.getBigZ());
+               if (d2.compareTo(d1 * d1) > 0) {
                   return false;
                }
             }
