--- a/net/minecraft/world/item/ArmorItem.java
+++ b/net/minecraft/world/item/ArmorItem.java
@@ -45,7 +_,7 @@
    private final Multimap<Attribute, AttributeModifier> f_40383_;
 
    public static boolean m_40398_(BlockSource p_40399_, ItemStack p_40400_) {
-      BlockPos blockpos = p_40399_.m_7961_().m_121945_(p_40399_.m_6414_().m_61143_(DispenserBlock.f_52659_));
+      BlockPos blockpos = p_40399_.m_7961_().relative(p_40399_.m_6414_().m_61143_(DispenserBlock.f_52659_));
       List<LivingEntity> list = p_40399_.m_7727_().m_6443_(LivingEntity.class, new AABB(blockpos), EntitySelector.f_20408_.and(new EntitySelector.MobCanWearArmorEntitySelector(p_40400_)));
       if (list.isEmpty()) {
          return false;
