--- a/net/minecraft/world/item/ArmorStandItem.java
+++ b/net/minecraft/world/item/ArmorStandItem.java
@@ -34,8 +_,8 @@
          BlockPlaceContext blockplacecontext = new BlockPlaceContext(p_40510_);
          BlockPos blockpos = blockplacecontext.m_8083_();
          ItemStack itemstack = p_40510_.m_43722_();
-         Vec3 vec3 = Vec3.m_82539_(blockpos);
-         AABB aabb = EntityType.f_20529_.m_20680_().m_20384_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
+         Vec3 vec3 = Vec3.atBottomCenterOf(blockpos);
+         AABB aabb = EntityType.f_20529_.m_20680_().makeBoundingBox(vec3.getX(), vec3.getY(), vec3.getZ());
          if (level.m_45756_((Entity)null, aabb) && level.m_45933_((Entity)null, aabb).isEmpty()) {
             if (level instanceof ServerLevel) {
                ServerLevel serverlevel = (ServerLevel)level;
@@ -46,9 +_,9 @@
                }
 
                float f = (float)Mth.m_14143_((Mth.m_14177_(p_40510_.m_7074_() - 180.0F) + 22.5F) / 45.0F) * 45.0F;
-               armorstand.m_7678_(armorstand.m_20185_(), armorstand.m_20186_(), armorstand.m_20189_(), f, 0.0F);
+               armorstand.moveTo(armorstand.getBigX(), armorstand.getBigY(), armorstand.getBigZ(), f, 0.0F);
                serverlevel.m_47205_(armorstand);
-               level.m_6263_((Player)null, armorstand.m_20185_(), armorstand.m_20186_(), armorstand.m_20189_(), SoundEvents.f_11684_, SoundSource.BLOCKS, 0.75F, 0.8F);
+               level.playSound((Player)null, armorstand.getBigX(), armorstand.getBigY(), armorstand.getBigZ(), SoundEvents.f_11684_, SoundSource.BLOCKS, 0.75F, 0.8F);
                armorstand.m_146852_(GameEvent.f_157810_, p_40510_.m_43723_());
             }
 
