--- a/net/minecraft/world/item/BoatItem.java
+++ b/net/minecraft/world/item/BoatItem.java
@@ -73,6 +_,6 @@
    }
 
    private Boat m_220016_(Level p_220017_, HitResult p_220018_) {
-      return (Boat)(this.f_220011_ ? new ChestBoat(p_220017_, p_220018_.m_82450_().f_82479_, p_220018_.m_82450_().f_82480_, p_220018_.m_82450_().f_82481_) : new Boat(p_220017_, p_220018_.m_82450_().f_82479_, p_220018_.m_82450_().f_82480_, p_220018_.m_82450_().f_82481_));
+      return (Boat)(this.f_220011_ ? new ChestBoat(p_220017_, p_220018_.m_82450_().getX(), p_220018_.m_82450_().getY(), p_220018_.m_82450_().getZ()) : new Boat(p_220017_, p_220018_.m_82450_().getX(), p_220018_.m_82450_().getY(), p_220018_.m_82450_().getZ()));
    }
 }
