--- a/net/minecraft/world/item/BottleItem.java
+++ b/net/minecraft/world/item/BottleItem.java
@@ -34,7 +_,7 @@
       if (!list.isEmpty()) {
          AreaEffectCloud areaeffectcloud = list.get(0);
          areaeffectcloud.m_19712_(areaeffectcloud.m_19743_() - 0.5F);
-         p_40656_.m_6263_((Player)null, p_40657_.m_20185_(), p_40657_.m_20186_(), p_40657_.m_20189_(), SoundEvents.f_11771_, SoundSource.NEUTRAL, 1.0F, 1.0F);
+         p_40656_.playSound((Player)null, p_40657_.getBigX(), p_40657_.getBigY(), p_40657_.getBigZ(), SoundEvents.f_11771_, SoundSource.NEUTRAL, 1.0F, 1.0F);
          p_40656_.m_220400_(p_40657_, GameEvent.f_157816_, p_40657_.m_20182_());
          if (p_40657_ instanceof ServerPlayer) {
             ServerPlayer serverplayer = (ServerPlayer)p_40657_;
@@ -54,7 +_,7 @@
                }
 
                if (p_40656_.m_6425_(blockpos).m_205070_(FluidTags.f_13131_)) {
-                  p_40656_.m_6263_(p_40657_, p_40657_.m_20185_(), p_40657_.m_20186_(), p_40657_.m_20189_(), SoundEvents.f_11770_, SoundSource.NEUTRAL, 1.0F, 1.0F);
+                  p_40656_.playSound(p_40657_, p_40657_.getBigX(), p_40657_.getBigY(), p_40657_.getBigZ(), SoundEvents.f_11770_, SoundSource.NEUTRAL, 1.0F, 1.0F);
                   p_40656_.m_142346_(p_40657_, GameEvent.f_157816_, blockpos);
                   return InteractionResultHolder.m_19092_(this.m_40651_(itemstack, p_40657_, PotionUtils.m_43549_(new ItemStack(Items.f_42589_), Potions.f_43599_)), p_40656_.m_5776_());
                }
