--- a/net/minecraft/world/item/BowItem.java
+++ b/net/minecraft/world/item/BowItem.java
@@ -66,7 +_,7 @@
                   p_40668_.m_7967_(abstractarrow);
                }
 
-               p_40668_.m_6263_((Player)null, player.m_20185_(), player.m_20186_(), player.m_20189_(), SoundEvents.f_11687_, SoundSource.PLAYERS, 1.0F, 1.0F / (p_40668_.m_213780_().m_188501_() * 0.4F + 1.2F) + f * 0.5F);
+               p_40668_.playSound((Player)null, player.getBigX(), player.getBigY(), player.getBigZ(), SoundEvents.f_11687_, SoundSource.PLAYERS, 1.0F, 1.0F / (p_40668_.m_213780_().m_188501_() * 0.4F + 1.2F) + f * 0.5F);
                if (!flag1 && !player.m_150110_().f_35937_) {
                   itemstack.m_41774_(1);
                   if (itemstack.m_41619_()) {
