--- a/net/minecraft/world/item/BrushItem.java
+++ b/net/minecraft/world/item/BrushItem.java
@@ -115,7 +_,7 @@
       Vec3 vec3 = p_278272_.m_82450_();
 
       for(int k = 0; k < j; ++k) {
-         p_278327_.m_7106_(blockparticleoption, vec3.f_82479_ - (double)(direction == Direction.WEST ? 1.0E-6F : 0.0F), vec3.f_82480_, vec3.f_82481_ - (double)(direction == Direction.NORTH ? 1.0E-6F : 0.0F), brushitem$dustparticlesdelta.f_271456_() * (double)i * 3.0D * p_278327_.m_213780_().m_188500_(), 0.0D, brushitem$dustparticlesdelta.f_271522_() * (double)i * 3.0D * p_278327_.m_213780_().m_188500_());
+         p_278327_.addParticle(blockparticleoption, vec3.getX().subtract((double)(direction == Direction.WEST ? 1.0E-6F : 0.0F)), vec3.getY(), vec3.getZ().subtract((double)(direction == Direction.NORTH ? 1.0E-6F : 0.0F)), brushitem$dustparticlesdelta.f_271456_() * (double)i * 3.0D * p_278327_.m_213780_().m_188500_(), 0.0D, brushitem$dustparticlesdelta.f_271522_() * (double)i * 3.0D * p_278327_.m_213780_().m_188500_());
       }
 
    }
