--- a/net/minecraft/world/item/ChorusFruitItem.java
+++ b/net/minecraft/world/item/ChorusFruitItem.java
@@ -26,7 +_,7 @@
 
          for(int i = 0; i < 16; ++i) {
             double d3 = p_40714_.m_20185_() + (p_40714_.m_217043_().m_188500_() - 0.5D) * 16.0D;
-            double d4 = Mth.m_14008_(p_40714_.m_20186_() + (double)(p_40714_.m_217043_().m_188503_(16) - 8), (double)p_40713_.m_141937_(), (double)(p_40713_.m_141937_() + ((ServerLevel)p_40713_).m_143344_() - 1));
+            double d4 = Mth.m_14008_(p_40714_.m_20186_() + (double)(p_40714_.m_217043_().m_188503_(16) - 8), (double)p_40713_.getMinBuildHeight(), (double)(p_40713_.getMinBuildHeight() + ((ServerLevel)p_40713_).m_143344_() - 1));
             double d5 = p_40714_.m_20189_() + (p_40714_.m_217043_().m_188500_() - 0.5D) * 16.0D;
             if (p_40714_.m_20159_()) {
                p_40714_.m_8127_();
