--- a/net/minecraft/world/item/CrossbowItem.java
+++ b/net/minecraft/world/item/CrossbowItem.java
@@ -209,7 +_,7 @@
             crossbowattackmob.m_5811_(crossbowattackmob.m_5448_(), p_40898_, projectile, p_40904_);
          } else {
             Vec3 vec31 = p_40896_.m_20289_(1.0F);
-            Quaternionf quaternionf = (new Quaternionf()).setAngleAxis((double)(p_40904_ * ((float)Math.PI / 180F)), vec31.f_82479_, vec31.f_82480_, vec31.f_82481_);
+            Quaternionf quaternionf = (new Quaternionf()).setAngleAxis((double)(p_40904_ * ((float)Math.PI / 180F)), vec31.m_7096_(), vec31.m_7098_(), vec31.m_7094_());
             Vec3 vec3 = p_40896_.m_20252_(1.0F);
             Vector3f vector3f = vec3.m_252839_().rotate(quaternionf);
             projectile.m_6686_((double)vector3f.x(), (double)vector3f.y(), (double)vector3f.z(), p_40902_, p_40903_);
