--- a/net/minecraft/world/item/DoubleHighBlockItem.java
+++ b/net/minecraft/world/item/DoubleHighBlockItem.java
@@ -14,7 +_,7 @@
 
    protected boolean m_7429_(BlockPlaceContext p_41013_, BlockState p_41014_) {
       Level level = p_41013_.m_43725_();
-      BlockPos blockpos = p_41013_.m_8083_().m_7494_();
+      BlockPos blockpos = p_41013_.m_8083_().above();
       BlockState blockstate = level.m_46801_(blockpos) ? Blocks.f_49990_.m_49966_() : Blocks.f_50016_.m_49966_();
       level.m_7731_(blockpos, blockstate, 27);
       return super.m_7429_(p_41013_, p_41014_);
