--- a/net/minecraft/world/item/EmptyMapItem.java
+++ b/net/minecraft/world/item/EmptyMapItem.java
@@ -23,7 +_,7 @@
 
          p_41146_.m_36246_(Stats.f_12982_.m_12902_(this));
          p_41146_.f_19853_.m_6269_((Player)null, p_41146_, SoundEvents.f_12493_, p_41146_.m_5720_(), 1.0F, 1.0F);
-         ItemStack itemstack1 = MapItem.m_42886_(p_41145_, p_41146_.m_146903_(), p_41146_.m_146907_(), (byte)0, true, false);
+         ItemStack itemstack1 = MapItem.create(p_41145_, p_41146_.getBlockX(), p_41146_.getBlockZ(), (byte)0, true, false);
          if (itemstack.m_41619_()) {
             return InteractionResultHolder.m_19096_(itemstack1);
          } else {
