--- a/net/minecraft/world/item/EnderEyeItem.java
+++ b/net/minecraft/world/item/EnderEyeItem.java
@@ -46,15 +_,15 @@
             level.m_46796_(1503, blockpos, 0);
             BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = EndPortalFrameBlock.m_53077_().m_61184_(level, blockpos);
             if (blockpattern$blockpatternmatch != null) {
-               BlockPos blockpos1 = blockpattern$blockpatternmatch.m_61228_().m_7918_(-3, 0, -3);
+               BlockPos blockpos1 = blockpattern$blockpatternmatch.m_61228_().offset(-3, 0, -3);
 
                for(int i = 0; i < 3; ++i) {
                   for(int j = 0; j < 3; ++j) {
-                     level.m_7731_(blockpos1.m_7918_(i, 0, j), Blocks.f_50257_.m_49966_(), 2);
+                     level.m_7731_(blockpos1.offset(i, 0, j), Blocks.f_50257_.m_49966_(), 2);
                   }
                }
 
-               level.m_6798_(1038, blockpos1.m_7918_(1, 0, 1), 0);
+               level.m_6798_(1038, blockpos1.offset(1, 0, 1), 0);
             }
 
             return InteractionResult.CONSUME;
@@ -75,7 +_,7 @@
             ServerLevel serverlevel = (ServerLevel)p_41184_;
             BlockPos blockpos = serverlevel.m_215011_(StructureTags.f_215882_, p_41185_.m_20183_(), 100, false);
             if (blockpos != null) {
-               EyeOfEnder eyeofender = new EyeOfEnder(p_41184_, p_41185_.m_20185_(), p_41185_.m_20227_(0.5D), p_41185_.m_20189_());
+               EyeOfEnder eyeofender = new EyeOfEnder(p_41184_, p_41185_.getBigX(), p_41185_.getBigY(0.5D), p_41185_.getBigZ());
                eyeofender.m_36972_(itemstack);
                eyeofender.m_36967_(blockpos);
                p_41184_.m_214171_(GameEvent.f_157778_, eyeofender.m_20182_(), GameEvent.Context.m_223717_(p_41185_));
@@ -84,7 +_,7 @@
                   CriteriaTriggers.f_10579_.m_73935_((ServerPlayer)p_41185_, blockpos);
                }
 
-               p_41184_.m_6263_((Player)null, p_41185_.m_20185_(), p_41185_.m_20186_(), p_41185_.m_20189_(), SoundEvents.f_11898_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_41184_.m_213780_().m_188501_() * 0.4F + 0.8F));
+               p_41184_.playSound((Player)null, p_41185_.getBigX(), p_41185_.getBigY(), p_41185_.getBigZ(), SoundEvents.f_11898_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_41184_.m_213780_().m_188501_() * 0.4F + 0.8F));
                p_41184_.m_5898_((Player)null, 1003, p_41185_.m_20183_(), 0);
                if (!p_41185_.m_150110_().f_35937_) {
                   itemstack.m_41774_(1);
