--- a/net/minecraft/world/item/ExperienceBottleItem.java
+++ b/net/minecraft/world/item/ExperienceBottleItem.java
@@ -20,7 +_,7 @@
 
    public InteractionResultHolder<ItemStack> m_7203_(Level p_41196_, Player p_41197_, InteractionHand p_41198_) {
       ItemStack itemstack = p_41197_.m_21120_(p_41198_);
-      p_41196_.m_6263_((Player)null, p_41197_.m_20185_(), p_41197_.m_20186_(), p_41197_.m_20189_(), SoundEvents.f_11870_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_41196_.m_213780_().m_188501_() * 0.4F + 0.8F));
+      p_41196_.playSound((Player)null, p_41197_.getBigX(), p_41197_.getBigY(), p_41197_.getBigZ(), SoundEvents.f_11870_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_41196_.m_213780_().m_188501_() * 0.4F + 0.8F));
       if (!p_41196_.f_46443_) {
          ThrownExperienceBottle thrownexperiencebottle = new ThrownExperienceBottle(p_41196_, p_41197_);
          thrownexperiencebottle.m_37446_(itemstack);
