--- a/net/minecraft/world/item/FireChargeItem.java
+++ b/net/minecraft/world/item/FireChargeItem.java
@@ -27,7 +_,7 @@
       BlockState blockstate = level.m_8055_(blockpos);
       boolean flag = false;
       if (!CampfireBlock.m_51321_(blockstate) && !CandleBlock.m_152845_(blockstate) && !CandleCakeBlock.m_152910_(blockstate)) {
-         blockpos = blockpos.m_121945_(p_41204_.m_43719_());
+         blockpos = blockpos.relative(p_41204_.m_43719_());
          if (BaseFireBlock.m_49255_(level, blockpos, p_41204_.m_8125_())) {
             this.m_41205_(level, blockpos);
             level.m_46597_(blockpos, BaseFireBlock.m_49245_(level, blockpos));
