--- a/net/minecraft/world/item/FireworkRocketItem.java
+++ b/net/minecraft/world/item/FireworkRocketItem.java
@@ -44,7 +_,7 @@
          ItemStack itemstack = p_41216_.m_43722_();
          Vec3 vec3 = p_41216_.m_43720_();
          Direction direction = p_41216_.m_43719_();
-         FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(level, p_41216_.m_43723_(), vec3.f_82479_ + (double)direction.m_122429_() * 0.15D, vec3.f_82480_ + (double)direction.m_122430_() * 0.15D, vec3.f_82481_ + (double)direction.m_122431_() * 0.15D, itemstack);
+         FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(level, p_41216_.m_43723_(), vec3.getX().add((double)direction.m_122429_() * 0.15D), vec3.getY().add((double)direction.m_122430_() * 0.15D), vec3.getZ().add((double)direction.m_122431_() * 0.15D), itemstack);
          level.m_7967_(fireworkrocketentity);
          itemstack.m_41774_(1);
       }
