--- a/net/minecraft/world/item/FlintAndSteelItem.java
+++ b/net/minecraft/world/item/FlintAndSteelItem.java
@@ -28,7 +_,7 @@
       BlockPos blockpos = p_41297_.m_8083_();
       BlockState blockstate = level.m_8055_(blockpos);
       if (!CampfireBlock.m_51321_(blockstate) && !CandleBlock.m_152845_(blockstate) && !CandleCakeBlock.m_152910_(blockstate)) {
-         BlockPos blockpos1 = blockpos.m_121945_(p_41297_.m_43719_());
+         BlockPos blockpos1 = blockpos.relative(p_41297_.m_43719_());
          if (BaseFireBlock.m_49255_(level, blockpos1, p_41297_.m_8125_())) {
             level.m_5594_(player, blockpos1, SoundEvents.f_11942_, SoundSource.BLOCKS, 1.0F, level.m_213780_().m_188501_() * 0.4F + 0.8F);
             BlockState blockstate1 = BaseFireBlock.m_49245_(level, blockpos1);
