--- a/net/minecraft/world/item/HangingEntityItem.java
+++ b/net/minecraft/world/item/HangingEntityItem.java
@@ -26,7 +_,7 @@
    public InteractionResult m_6225_(UseOnContext p_41331_) {
       BlockPos blockpos = p_41331_.m_8083_();
       Direction direction = p_41331_.m_43719_();
-      BlockPos blockpos1 = blockpos.m_121945_(direction);
+      BlockPos blockpos1 = blockpos.relative(direction);
       Player player = p_41331_.m_43723_();
       ItemStack itemstack = p_41331_.m_43722_();
       if (player != null && !this.m_5595_(player, direction, itemstack, blockpos1)) {
