--- a/net/minecraft/world/item/ItemUtils.java
+++ b/net/minecraft/world/item/ItemUtils.java
@@ -46,7 +_,7 @@
       Level level = p_150953_.m_9236_();
       if (!level.f_46443_) {
          p_150954_.forEach((p_289504_) -> {
-            level.m_7967_(new ItemEntity(level, p_150953_.m_20185_(), p_150953_.m_20186_(), p_150953_.m_20189_(), p_289504_));
+            level.m_7967_(new ItemEntity(level, p_150953_.getBigX(), p_150953_.getBigY(), p_150953_.getBigZ(), p_289504_));
          });
       }
    }
