--- a/net/minecraft/world/item/LeadItem.java
+++ b/net/minecraft/world/item/LeadItem.java
@@ -38,9 +_,9 @@
       LeashFenceKnotEntity leashfenceknotentity = null;
       boolean flag = false;
       double d0 = 7.0D;
-      int i = p_42832_.m_123341_();
-      int j = p_42832_.m_123342_();
-      int k = p_42832_.m_123343_();
+      long i = p_42832_.getX();
+      long j = p_42832_.getY();
+      long k = p_42832_.getZ();
 
       for(Mob mob : p_42831_.m_45976_(Mob.class, new AABB((double)i - 7.0D, (double)j - 7.0D, (double)k - 7.0D, (double)i + 7.0D, (double)j + 7.0D, (double)k + 7.0D))) {
          if (mob.m_21524_() == p_42830_) {
