--- a/net/minecraft/world/item/MapItem.java
+++ b/net/minecraft/world/item/MapItem.java
@@ -45,9 +_,9 @@
       super(p_42847_);
    }
 
-   public static ItemStack m_42886_(Level p_42887_, int p_42888_, int p_42889_, byte p_42890_, boolean p_42891_, boolean p_42892_) {
+   public static ItemStack create(Level pLevel, long pLevelX, long pLevelZ, byte pScale, boolean pTrackingPosition, boolean pUnlimitedTracking) {
       ItemStack itemstack = new ItemStack(Items.f_42573_);
-      m_151111_(itemstack, p_42887_, p_42888_, p_42889_, p_42890_, p_42891_, p_42892_, p_42887_.m_46472_());
+      createAndStoreSavedData(itemstack, pLevel, pLevelX, pLevelZ, pScale, pTrackingPosition, pUnlimitedTracking, pLevel.m_46472_());
       return itemstack;
    }
 
@@ -68,10 +_,10 @@
       return compoundtag != null && compoundtag.m_128425_("map", 99) ? compoundtag.m_128451_("map") : null;
    }
 
-   private static int m_151120_(Level p_151121_, int p_151122_, int p_151123_, int p_151124_, boolean p_151125_, boolean p_151126_, ResourceKey<Level> p_151127_) {
-      MapItemSavedData mapitemsaveddata = MapItemSavedData.m_164780_((double)p_151122_, (double)p_151123_, (byte)p_151124_, p_151125_, p_151126_, p_151127_);
-      int i = p_151121_.m_7354_();
-      p_151121_.m_142325_(m_42848_(i), mapitemsaveddata);
+   private static int createNewSavedData(Level pLevel, long pX, long pZ, int pScale, boolean pTrackingPosition, boolean pUnlimitedTracking, ResourceKey<Level> pDimension) {
+      MapItemSavedData mapitemsaveddata = MapItemSavedData.m_164780_((double)pX, (double)pZ, (byte)pScale, pTrackingPosition, pUnlimitedTracking, pDimension);
+      int i = pLevel.m_7354_();
+      pLevel.m_142325_(m_42848_(i), mapitemsaveddata);
       return i;
    }
 
@@ -79,9 +_,9 @@
       p_151109_.m_41784_().m_128405_("map", p_151110_);
    }
 
-   private static void m_151111_(ItemStack p_151112_, Level p_151113_, int p_151114_, int p_151115_, int p_151116_, boolean p_151117_, boolean p_151118_, ResourceKey<Level> p_151119_) {
-      int i = m_151120_(p_151113_, p_151114_, p_151115_, p_151116_, p_151117_, p_151118_, p_151119_);
-      m_151108_(p_151112_, i);
+   private static void createAndStoreSavedData(ItemStack pStack, Level pLevel, long pX, long pZ, int pScale, boolean pTrackingPosition, boolean pUnlimitedTracking, ResourceKey<Level> pDimension) {
+      int i = createNewSavedData(pLevel, pX, pZ, pScale, pTrackingPosition, pUnlimitedTracking, pDimension);
+      m_151108_(pStack, i);
    }
 
    public static String m_42848_(int p_42849_) {
@@ -91,8 +_,8 @@
    public void m_42893_(Level p_42894_, Entity p_42895_, MapItemSavedData p_42896_) {
       if (p_42894_.m_46472_() == p_42896_.f_77887_ && p_42895_ instanceof Player) {
          int i = 1 << p_42896_.f_77890_;
-         int j = p_42896_.f_256718_;
-         int k = p_42896_.f_256789_;
+         long j = p_42896_.f_256718_;
+         long k = p_42896_.f_256789_;
          int l = Mth.m_14107_(p_42895_.m_20185_() - (double)j) / i + 64;
          int i1 = Mth.m_14107_(p_42895_.m_20189_() - (double)k) / i + 64;
          int j1 = 128 / i;
@@ -113,17 +_,17 @@
 
                for(int l1 = i1 - j1 - 1; l1 < i1 + j1; ++l1) {
                   if (k1 >= 0 && l1 >= -1 && k1 < 128 && l1 < 128) {
-                     int i2 = Mth.m_144944_(k1 - l) + Mth.m_144944_(l1 - i1);
+                     long i2 = Mth.m_144944_(k1 - l) + Mth.m_144944_(l1 - i1);
                      boolean flag1 = i2 > (j1 - 2) * (j1 - 2);
-                     int j2 = (j / i + k1 - 64) * i;
-                     int k2 = (k / i + l1 - 64) * i;
+                     long j2 = (j / i + k1 - 64) * i;
+                     long k2 = (k / i + l1 - 64) * i;
                      Multiset<MapColor> multiset = LinkedHashMultiset.create();
-                     LevelChunk levelchunk = p_42894_.m_6325_(SectionPos.m_123171_(j2), SectionPos.m_123171_(k2));
+                     LevelChunk levelchunk = p_42894_.getChunk(me.alphamode.mclong.math.BigInteger.valueOf(SectionPos.blockToSectionCoord(j2)), me.alphamode.mclong.math.BigInteger.valueOf(SectionPos.blockToSectionCoord(k2)));
                      if (!levelchunk.m_6430_()) {
                         int l2 = 0;
                         double d1 = 0.0D;
                         if (p_42894_.m_6042_().f_63856_()) {
-                           int i3 = j2 + k2 * 231871;
+                           long i3 = j2 + k2 * 231871;
                            i3 = i3 * i3 * 31287121 + i3 * 11;
                            if ((i3 >> 20 & 1) == 0) {
                               multiset.add(Blocks.f_50493_.m_49966_().m_284242_(p_42894_, BlockPos.f_121853_), 10);
@@ -135,34 +_,34 @@
                         } else {
                            for(int i4 = 0; i4 < i; ++i4) {
                               for(int j3 = 0; j3 < i; ++j3) {
-                                 blockpos$mutableblockpos.m_122178_(j2 + i4, 0, k2 + j3);
-                                 int k3 = levelchunk.m_5885_(Heightmap.Types.WORLD_SURFACE, blockpos$mutableblockpos.m_123341_(), blockpos$mutableblockpos.m_123343_()) + 1;
+                                 blockpos$mutableblockpos.set(j2 + i4, 0, k2 + j3);
+                                 long k3 = levelchunk.getHeight(Heightmap.Types.WORLD_SURFACE, blockpos$mutableblockpos.getBigX(), blockpos$mutableblockpos.getBigZ()) + 1;
                                  BlockState blockstate;
-                                 if (k3 <= p_42894_.m_141937_() + 1) {
+                                 if (k3 <= p_42894_.getMinBuildHeight() + 1) {
                                     blockstate = Blocks.f_50752_.m_49966_();
                                  } else {
                                     do {
                                        --k3;
-                                       blockpos$mutableblockpos.m_142448_(k3);
+                                       blockpos$mutableblockpos.setY(k3);
                                        blockstate = levelchunk.m_8055_(blockpos$mutableblockpos);
-                                    } while(blockstate.m_284242_(p_42894_, blockpos$mutableblockpos) == MapColor.f_283808_ && k3 > p_42894_.m_141937_());
+                                    } while(blockstate.m_284242_(p_42894_, blockpos$mutableblockpos) == MapColor.f_283808_ && k3 > p_42894_.getMinBuildHeight());
 
-                                    if (k3 > p_42894_.m_141937_() && !blockstate.m_60819_().m_76178_()) {
-                                       int l3 = k3 - 1;
-                                       blockpos$mutableblockpos1.m_122190_(blockpos$mutableblockpos);
+                                    if (k3 > p_42894_.getMinBuildHeight() && !blockstate.m_60819_().m_76178_()) {
+                                       long l3 = k3 - 1;
+                                       blockpos$mutableblockpos1.set(blockpos$mutableblockpos);
 
                                        BlockState blockstate1;
                                        do {
-                                          blockpos$mutableblockpos1.m_142448_(l3--);
+                                          blockpos$mutableblockpos1.setY(l3--);
                                           blockstate1 = levelchunk.m_8055_(blockpos$mutableblockpos1);
                                           ++l2;
-                                       } while(l3 > p_42894_.m_141937_() && !blockstate1.m_60819_().m_76178_());
+                                       } while(l3 > p_42894_.getMinBuildHeight() && !blockstate1.m_60819_().m_76178_());
 
                                        blockstate = this.m_42900_(p_42894_, blockstate, blockpos$mutableblockpos);
                                     }
                                  }
 
-                                 p_42896_.m_77930_(p_42894_, blockpos$mutableblockpos.m_123341_(), blockpos$mutableblockpos.m_123343_());
+                                 p_42896_.checkBanners(p_42894_, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ());
                                  d1 += (double)k3 / (double)(i * i);
                                  multiset.add(blockstate.m_284242_(p_42894_, blockpos$mutableblockpos));
                               }
@@ -219,16 +_,16 @@
       if (mapitemsaveddata != null) {
          if (p_42851_.m_46472_() == mapitemsaveddata.f_77887_) {
             int i = 1 << mapitemsaveddata.f_77890_;
-            int j = mapitemsaveddata.f_256718_;
-            int k = mapitemsaveddata.f_256789_;
+            long j = mapitemsaveddata.f_256718_;
+            long k = mapitemsaveddata.f_256789_;
             boolean[] aboolean = new boolean[16384];
-            int l = j / i - 64;
-            int i1 = k / i - 64;
+            long l = j / i - 64;
+            long i1 = k / i - 64;
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
             for(int j1 = 0; j1 < 128; ++j1) {
                for(int k1 = 0; k1 < 128; ++k1) {
-                  Holder<Biome> holder = p_42851_.m_204166_(blockpos$mutableblockpos.m_122178_((l + k1) * i, 0, (i1 + j1) * i));
+                  Holder<Biome> holder = p_42851_.m_204166_(blockpos$mutableblockpos.set((l + k1) * i, 0, (i1 + j1) * i));
                   aboolean[j1 * 128 + k1] = holder.m_203656_(BiomeTags.f_215803_);
                }
             }
