--- a/net/minecraft/world/item/PlaceOnWaterBlockItem.java
+++ b/net/minecraft/world/item/PlaceOnWaterBlockItem.java
@@ -21,7 +_,7 @@
 
    public InteractionResultHolder<ItemStack> m_7203_(Level p_220231_, Player p_220232_, InteractionHand p_220233_) {
       BlockHitResult blockhitresult = m_41435_(p_220231_, p_220232_, ClipContext.Fluid.SOURCE_ONLY);
-      BlockHitResult blockhitresult1 = blockhitresult.m_82430_(blockhitresult.m_82425_().m_7494_());
+      BlockHitResult blockhitresult1 = blockhitresult.m_82430_(blockhitresult.m_82425_().above());
       InteractionResult interactionresult = super.m_6225_(new UseOnContext(p_220232_, p_220233_, blockhitresult1));
       return new InteractionResultHolder<>(interactionresult, p_220232_.m_21120_(p_220233_));
    }
