--- a/net/minecraft/world/item/PotionItem.java
+++ b/net/minecraft/world/item/PotionItem.java
@@ -90,7 +_,7 @@
             ServerLevel serverlevel = (ServerLevel)level;
 
             for(int i = 0; i < 5; ++i) {
-               serverlevel.m_8767_(ParticleTypes.f_123769_, (double)blockpos.m_123341_() + level.f_46441_.m_188500_(), (double)(blockpos.m_123342_() + 1), (double)blockpos.m_123343_() + level.f_46441_.m_188500_(), 1, 0.0D, 0.0D, 0.0D, 1.0D);
+               serverlevel.sendParticles(ParticleTypes.f_123769_, blockpos.getBigDecX().add(level.f_46441_.m_188500_()), blockpos.getBigDecY().add(me.alphamode.mclong.math.BigDecimal.ONE), blockpos.getBigDecZ().add(level.f_46441_.m_188500_()), 1, 0.0D, 0.0D, 0.0D, 1.0D);
             }
          }
 
