--- a/net/minecraft/world/item/SnowballItem.java
+++ b/net/minecraft/world/item/SnowballItem.java
@@ -16,7 +_,7 @@
 
    public InteractionResultHolder<ItemStack> m_7203_(Level p_43142_, Player p_43143_, InteractionHand p_43144_) {
       ItemStack itemstack = p_43143_.m_21120_(p_43144_);
-      p_43142_.m_6263_((Player)null, p_43143_.m_20185_(), p_43143_.m_20186_(), p_43143_.m_20189_(), SoundEvents.f_12473_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_43142_.m_213780_().m_188501_() * 0.4F + 0.8F));
+      p_43142_.playSound((Player)null, p_43143_.getBigX(), p_43143_.getBigY(), p_43143_.getBigZ(), SoundEvents.f_12473_, SoundSource.NEUTRAL, 0.5F, 0.4F / (p_43142_.m_213780_().m_188501_() * 0.4F + 0.8F));
       if (!p_43142_.f_46443_) {
          Snowball snowball = new Snowball(p_43142_, p_43143_);
          snowball.m_37446_(itemstack);
