--- a/net/minecraft/world/item/SpawnEggItem.java
+++ b/net/minecraft/world/item/SpawnEggItem.java
@@ -75,7 +_,7 @@
          if (blockstate.m_60812_(level, blockpos).m_83281_()) {
             blockpos1 = blockpos;
          } else {
-            blockpos1 = blockpos.m_121945_(direction);
+            blockpos1 = blockpos.relative(direction);
          }
 
          EntityType<?> entitytype = this.m_43228_(itemstack.m_41783_());
@@ -169,7 +_,7 @@
             if (!mob.m_6162_()) {
                return Optional.empty();
             } else {
-               mob.m_7678_(p_43220_.m_7096_(), p_43220_.m_7098_(), p_43220_.m_7094_(), 0.0F, 0.0F);
+               mob.moveTo(p_43220_.getX(), p_43220_.getY(), p_43220_.getZ(), 0.0F, 0.0F);
                p_43219_.m_47205_(mob);
                if (p_43221_.m_41788_()) {
                   mob.m_6593_(p_43221_.m_41786_());
