--- a/net/minecraft/world/item/SpawnEggItem.java
+++ b/net/minecraft/world/item/SpawnEggItem.java
@@ -75,7 +_,7 @@
          if (blockstate.m_60812_(level, blockpos).m_83281_()) {
             blockpos1 = blockpos;
          } else {
-            blockpos1 = blockpos.m_121945_(direction);
+            blockpos1 = blockpos.relative(direction);
          }
 
          EntityType<?> entitytype = this.m_43228_(itemstack.m_41783_());
