--- a/net/minecraft/world/item/SplashPotionItem.java
+++ b/net/minecraft/world/item/SplashPotionItem.java
@@ -13,7 +_,7 @@
    }
 
    public InteractionResultHolder<ItemStack> m_7203_(Level p_43243_, Player p_43244_, InteractionHand p_43245_) {
-      p_43243_.m_6263_((Player)null, p_43244_.m_20185_(), p_43244_.m_20186_(), p_43244_.m_20189_(), SoundEvents.f_12437_, SoundSource.PLAYERS, 0.5F, 0.4F / (p_43243_.m_213780_().m_188501_() * 0.4F + 0.8F));
+      p_43243_.playSound((Player)null, p_43244_.getBigX(), p_43244_.getBigY(), p_43244_.getBigZ(), SoundEvents.f_12437_, SoundSource.PLAYERS, 0.5F, 0.4F / (p_43243_.m_213780_().m_188501_() * 0.4F + 0.8F));
       return super.m_7203_(p_43243_, p_43244_, p_43245_);
    }
 }
