--- a/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -280,17 +_,17 @@
          if (this.f_31429_ > 0) {
             --this.f_31429_;
             if (this.f_31429_ == 0 && this.m_9236_().m_46469_().m_46207_(GameRules.f_46132_)) {
-               int j1 = Mth.m_14107_(this.m_20186_());
-               int i2 = Mth.m_14107_(this.m_20185_());
-               int j2 = Mth.m_14107_(this.m_20189_());
+               long j1 = Mth.m_14134_(this.m_20186_());
+               long i2 = Mth.m_14134_(this.m_20185_());
+               long j2 = Mth.m_14134_(this.m_20189_());
                boolean flag = false;
 
                for(int j = -1; j <= 1; ++j) {
                   for(int k2 = -1; k2 <= 1; ++k2) {
                      for(int k = 0; k <= 3; ++k) {
-                        int l2 = i2 + j;
-                        int l = j1 + k;
-                        int i1 = j2 + k2;
+                        long l2 = i2 + j;
+                        long l = j1 + k;
+                        long i1 = j2 + k2;
                         BlockPos blockpos = new BlockPos(l2, l, i1);
                         BlockState blockstate = this.m_9236_().m_8055_(blockpos);
                         if (m_31491_(blockstate)) {
