--- a/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -143,24 +_,24 @@
       if (!this.m_9236_().f_46443_ && this.m_31512_(0) > 0) {
          Entity entity = this.m_9236_().m_6815_(this.m_31512_(0));
          if (entity != null) {
-            double d0 = vec3.f_82480_;
+            double d0 = vec3.m_7098_();
             if (this.m_20186_() < entity.m_20186_() || !this.m_7090_() && this.m_20186_() < entity.m_20186_() + 5.0D) {
                d0 = Math.max(0.0D, d0);
                d0 += 0.3D - d0 * (double)0.6F;
             }
 
-            vec3 = new Vec3(vec3.f_82479_, d0, vec3.f_82481_);
+            vec3 = new Vec3(vec3.m_7096_(), d0, vec3.m_7094_());
             Vec3 vec31 = new Vec3(entity.m_20185_() - this.m_20185_(), 0.0D, entity.m_20189_() - this.m_20189_());
             if (vec31.m_165925_() > 9.0D) {
                Vec3 vec32 = vec31.m_82541_();
-               vec3 = vec3.m_82520_(vec32.f_82479_ * 0.3D - vec3.f_82479_ * 0.6D, 0.0D, vec32.f_82481_ * 0.3D - vec3.f_82481_ * 0.6D);
+               vec3 = vec3.m_82520_(vec32.m_7096_() * 0.3D - vec3.m_7096_() * 0.6D, 0.0D, vec32.m_7094_() * 0.3D - vec3.m_7094_() * 0.6D);
             }
          }
       }
 
       this.m_20256_(vec3);
       if (vec3.m_165925_() > 0.05D) {
-         this.m_146922_((float)Mth.m_14136_(vec3.f_82481_, vec3.f_82479_) * (180F / (float)Math.PI) - 90.0F);
+         this.m_146922_((float)Mth.m_14136_(vec3.m_7094_(), vec3.m_7096_()) * (180F / (float)Math.PI) - 90.0F);
       }
 
       super.m_8107_();
@@ -280,17 +_,17 @@
          if (this.f_31429_ > 0) {
             --this.f_31429_;
             if (this.f_31429_ == 0 && this.m_9236_().m_46469_().m_46207_(GameRules.f_46132_)) {
-               int j1 = Mth.m_14107_(this.m_20186_());
-               int i2 = Mth.m_14107_(this.m_20185_());
-               int j2 = Mth.m_14107_(this.m_20189_());
+               long j1 = Mth.m_14134_(this.m_20186_());
+               long i2 = Mth.m_14134_(this.m_20185_());
+               long j2 = Mth.m_14134_(this.m_20189_());
                boolean flag = false;
 
                for(int j = -1; j <= 1; ++j) {
                   for(int k2 = -1; k2 <= 1; ++k2) {
                      for(int k = 0; k <= 3; ++k) {
-                        int l2 = i2 + j;
-                        int l = j1 + k;
-                        int i1 = j2 + k2;
+                        long l2 = i2 + j;
+                        long l = j1 + k;
+                        long i1 = j2 + k2;
                         BlockPos blockpos = new BlockPos(l2, l, i1);
                         BlockState blockstate = this.m_9236_().m_8055_(blockpos);
                         if (m_31491_(blockstate)) {
