--- a/net/minecraft/server/level/progress/StoringChunkProgressListener.java
+++ b/net/minecraft/server/level/progress/StoringChunkProgressListener.java
@@ -66,6 +_,6 @@
 
    @Nullable
    public ChunkStatus m_9663_(int p_9664_, int p_9665_) {
-      return this.f_9654_.get(ChunkPos.m_45589_(p_9664_ + this.f_9655_.f_45578_ - this.f_9657_, p_9665_ + this.f_9655_.f_45579_ - this.f_9657_));
+      return this.f_9654_.get(ChunkPos.asLong(p_9664_ + this.f_9655_.f_45578_ - this.f_9657_, p_9665_ + this.f_9655_.f_45579_ - this.f_9657_));
    }
 }
