--- a/net/minecraft/world/level/block/AbstractFurnaceBlock.java
+++ b/net/minecraft/world/level/block/AbstractFurnaceBlock.java
@@ -66,7 +_,7 @@
          if (blockentity instanceof AbstractFurnaceBlockEntity) {
             if (p_48714_ instanceof ServerLevel) {
                Containers.m_19002_(p_48714_, p_48715_, (AbstractFurnaceBlockEntity)blockentity);
-               ((AbstractFurnaceBlockEntity)blockentity).m_154995_((ServerLevel)p_48714_, Vec3.m_82512_(p_48715_));
+               ((AbstractFurnaceBlockEntity)blockentity).m_154995_((ServerLevel)p_48714_, Vec3.atCenterOf(p_48715_));
             }
 
             p_48714_.m_46717_(p_48715_, this);
