--- a/net/minecraft/world/level/block/AzaleaBlock.java
+++ b/net/minecraft/world/level/block/AzaleaBlock.java
@@ -29,7 +_,7 @@
    }
 
    public boolean m_7370_(BlockGetter p_152074_, BlockPos p_152075_, BlockState p_152076_, boolean p_152077_) {
-      return p_152074_.m_6425_(p_152075_.m_7494_()).m_76178_();
+      return p_152074_.m_6425_(p_152075_.above()).m_76178_();
    }
 
    public boolean m_214167_(Level p_220712_, RandomSource p_220713_, BlockPos p_220714_, BlockState p_220715_) {
