--- a/net/minecraft/world/level/block/AzaleaBlock.java
+++ b/net/minecraft/world/level/block/AzaleaBlock.java
@@ -30,7 +_,7 @@
    }
 
    public boolean m_7370_(LevelReader p_256329_, BlockPos p_256107_, BlockState p_255771_, boolean p_255916_) {
-      return p_256329_.m_6425_(p_256107_.m_7494_()).m_76178_();
+      return p_256329_.m_6425_(p_256107_.above()).m_76178_();
    }
 
    public boolean m_214167_(Level p_220712_, RandomSource p_220713_, BlockPos p_220714_, BlockState p_220715_) {
