--- a/net/minecraft/world/level/block/BaseCoralWallFanBlock.java
+++ b/net/minecraft/world/level/block/BaseCoralWallFanBlock.java
@@ -53,7 +_,7 @@
 
    public boolean m_7898_(BlockState p_49200_, LevelReader p_49201_, BlockPos p_49202_) {
       Direction direction = p_49200_.m_61143_(f_49192_);
-      BlockPos blockpos = p_49202_.m_121945_(direction.m_122424_());
+      BlockPos blockpos = p_49202_.relative(direction.m_122424_());
       BlockState blockstate = p_49201_.m_8055_(blockpos);
       return blockstate.m_60783_(p_49201_, blockpos, direction);
    }
