--- a/net/minecraft/world/level/block/BeehiveBlock.java
+++ b/net/minecraft/world/level/block/BeehiveBlock.java
@@ -111,7 +_,7 @@
       if (i >= 5) {
          Item item = itemstack.m_41720_();
          if (itemstack.m_150930_(Items.f_42574_)) {
-            p_49625_.m_6263_(p_49627_, p_49627_.m_20185_(), p_49627_.m_20186_(), p_49627_.m_20189_(), SoundEvents.f_11697_, SoundSource.BLOCKS, 1.0F, 1.0F);
+            p_49625_.playSound(p_49627_, p_49627_.getBigX(), p_49627_.getBigY(), p_49627_.getBigZ(), SoundEvents.f_11697_, SoundSource.BLOCKS, 1.0F, 1.0F);
             m_49600_(p_49625_, p_49626_);
             itemstack.m_41622_(1, p_49627_, (p_49571_) -> {
                p_49571_.m_21190_(p_49628_);
@@ -120,7 +_,7 @@
             p_49625_.m_142346_(p_49627_, GameEvent.f_157781_, p_49626_);
          } else if (itemstack.m_150930_(Items.f_42590_)) {
             itemstack.m_41774_(1);
-            p_49625_.m_6263_(p_49627_, p_49627_.m_20185_(), p_49627_.m_20186_(), p_49627_.m_20189_(), SoundEvents.f_11770_, SoundSource.BLOCKS, 1.0F, 1.0F);
+            p_49625_.playSound(p_49627_, p_49627_.getBigX(), p_49627_.getBigY(), p_49627_.getBigZ(), SoundEvents.f_11770_, SoundSource.BLOCKS, 1.0F, 1.0F);
             if (itemstack.m_41619_()) {
                p_49627_.m_21008_(p_49628_, new ItemStack(Items.f_42787_));
             } else if (!p_49627_.m_150109_().m_36054_(new ItemStack(Items.f_42787_))) {
@@ -191,14 +_,14 @@
          if (d0 >= 1.0D && !p_49606_.m_204336_(BlockTags.f_13049_)) {
             double d1 = voxelshape.m_83288_(Direction.Axis.Y);
             if (d1 > 0.0D) {
-               this.m_49612_(p_49604_, p_49605_, voxelshape, (double)p_49605_.m_123342_() + d1 - 0.05D);
+               this.m_49612_(p_49604_, p_49605_, voxelshape, (double)p_49605_.getY() + d1 - 0.05D);
             } else {
-               BlockPos blockpos = p_49605_.m_7495_();
+               BlockPos blockpos = p_49605_.below();
                BlockState blockstate = p_49604_.m_8055_(blockpos);
                VoxelShape voxelshape1 = blockstate.m_60812_(p_49604_, blockpos);
                double d2 = voxelshape1.m_83297_(Direction.Axis.Y);
                if ((d2 < 1.0D || !blockstate.m_60838_(p_49604_, blockpos)) && blockstate.m_60819_().m_76178_()) {
-                  this.m_49612_(p_49604_, p_49605_, voxelshape, (double)p_49605_.m_123342_() - 0.05D);
+                  this.m_49612_(p_49604_, p_49605_, voxelshape, (double)p_49605_.getY() - 0.05D);
                }
             }
          }
@@ -207,11 +_,11 @@
    }
 
    private void m_49612_(Level p_49613_, BlockPos p_49614_, VoxelShape p_49615_, double p_49616_) {
-      this.m_49576_(p_49613_, (double)p_49614_.m_123341_() + p_49615_.m_83288_(Direction.Axis.X), (double)p_49614_.m_123341_() + p_49615_.m_83297_(Direction.Axis.X), (double)p_49614_.m_123343_() + p_49615_.m_83288_(Direction.Axis.Z), (double)p_49614_.m_123343_() + p_49615_.m_83297_(Direction.Axis.Z), p_49616_);
+      this.spawnFluidParticle(p_49613_, p_49614_.getBigDecX().add(p_49615_.minB(Direction.Axis.X)), p_49614_.getBigDecX().add(p_49615_.maxB(Direction.Axis.X)), p_49614_.getBigDecZ().add(p_49615_.minB(Direction.Axis.Z)), p_49614_.getBigDecZ().add(p_49615_.maxB(Direction.Axis.Z)), p_49616_);
    }
 
-   private void m_49576_(Level p_49577_, double p_49578_, double p_49579_, double p_49580_, double p_49581_, double p_49582_) {
-      p_49577_.m_7106_(ParticleTypes.f_123779_, Mth.m_14139_(p_49577_.f_46441_.m_188500_(), p_49578_, p_49579_), p_49582_, Mth.m_14139_(p_49577_.f_46441_.m_188500_(), p_49580_, p_49581_), 0.0D, 0.0D, 0.0D);
+   private void spawnFluidParticle(Level pParticleData, me.alphamode.mclong.math.BigDecimal pX1, me.alphamode.mclong.math.BigDecimal pX2, me.alphamode.mclong.math.BigDecimal pZ1, me.alphamode.mclong.math.BigDecimal pZ2, double pY) {
+      pParticleData.addParticle(ParticleTypes.f_123779_, Mth.lerp(pParticleData.f_46441_.m_188500_(), pX1, pX2), me.alphamode.mclong.math.BigDecimal.valueOf(pY), Mth.lerp(pParticleData.f_46441_.m_188500_(), pZ1, pZ2), 0.0D, 0.0D, 0.0D);
    }
 
    public BlockState m_5573_(BlockPlaceContext p_49573_) {
@@ -254,7 +_,7 @@
                CompoundTag compoundtag1 = new CompoundTag();
                compoundtag1.m_128405_("honey_level", i);
                itemstack.m_41700_("BlockStateTag", compoundtag1);
-               ItemEntity itementity = new ItemEntity(p_49608_, (double)p_49609_.m_123341_(), (double)p_49609_.m_123342_(), (double)p_49609_.m_123343_(), itemstack);
+               ItemEntity itementity = new ItemEntity(p_49608_, p_49609_.getBigDecX(), p_49609_.getBigDecY(), p_49609_.getBigDecZ(), itemstack);
                itementity.m_32060_();
                p_49608_.m_7967_(itementity);
             }
