--- a/net/minecraft/world/level/block/BlastFurnaceBlock.java
+++ b/net/minecraft/world/level/block/BlastFurnaceBlock.java
@@ -43,9 +_,9 @@
 
    public void m_214162_(BlockState p_220818_, Level p_220819_, BlockPos p_220820_, RandomSource p_220821_) {
       if (p_220818_.m_61143_(f_48684_)) {
-         double d0 = (double)p_220820_.m_123341_() + 0.5D;
-         double d1 = (double)p_220820_.m_123342_();
-         double d2 = (double)p_220820_.m_123343_() + 0.5D;
+         double d0 = (double)p_220820_.getX() + 0.5D;
+         double d1 = (double)p_220820_.getY();
+         double d2 = (double)p_220820_.getZ() + 0.5D;
          if (p_220821_.m_188500_() < 0.1D) {
             p_220819_.m_7785_(d0, d1, d2, SoundEvents.f_11715_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
          }
