--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -118,13 +_,13 @@
    }
 
    public static BlockState m_49897_(BlockState p_49898_, BlockState p_49899_, LevelAccessor p_238252_, BlockPos p_49901_) {
-      VoxelShape voxelshape = Shapes.m_83148_(p_49898_.m_60812_(p_238252_, p_49901_), p_49899_.m_60812_(p_238252_, p_49901_), BooleanOp.f_82683_).m_83216_((double)p_49901_.m_123341_(), (double)p_49901_.m_123342_(), (double)p_49901_.m_123343_());
+      VoxelShape voxelshape = Shapes.m_83148_(p_49898_.m_60812_(p_238252_, p_49901_), p_49899_.m_60812_(p_238252_, p_49901_), BooleanOp.f_82683_).move(p_49901_.getBigDecX(), p_49901_.getBigDecY(), p_49901_.getBigDecZ());
       if (voxelshape.m_83281_()) {
          return p_49899_;
       } else {
          for(Entity entity : p_238252_.m_45933_((Entity)null, voxelshape.m_83215_())) {
-            double d0 = Shapes.m_193135_(Direction.Axis.Y, entity.m_20191_().m_82386_(0.0D, 1.0D, 0.0D), List.of(voxelshape), -1.0D);
-            entity.m_246847_(0.0D, 1.0D + d0, 0.0D);
+            me.alphamode.mclong.math.BigDecimal d0 = Shapes.collide(Direction.Axis.Y, entity.m_20191_().m_82386_(0.0D, 1.0D, 0.0D), List.of(voxelshape), me.alphamode.mclong.math.BigDecimal.ONE.negate());
+            entity.teleportRelative(me.alphamode.mclong.math.BigDecimal.ZERO, me.alphamode.mclong.math.BigDecimal.ONE.add(d0), me.alphamode.mclong.math.BigDecimal.ZERO);
          }
 
          return p_49899_;
@@ -140,7 +_,7 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(Direction direction : f_60441_) {
-         blockpos$mutableblockpos.m_122159_(p_49934_, direction);
+         blockpos$mutableblockpos.setWithOffset(p_49934_, direction);
          blockstate = blockstate.m_60728_(direction, p_49933_.m_8055_(blockpos$mutableblockpos), p_49933_, p_49934_, blockpos$mutableblockpos);
       }
 
@@ -246,12 +_,12 @@
    }
 
    public static List<ItemStack> m_49869_(BlockState p_49870_, ServerLevel p_49871_, BlockPos p_49872_, @Nullable BlockEntity p_49873_) {
-      LootParams.Builder lootparams$builder = (new LootParams.Builder(p_49871_)).m_287286_(LootContextParams.f_81460_, Vec3.m_82512_(p_49872_)).m_287286_(LootContextParams.f_81463_, ItemStack.f_41583_).m_287289_(LootContextParams.f_81462_, p_49873_);
+      LootParams.Builder lootparams$builder = (new LootParams.Builder(p_49871_)).m_287286_(LootContextParams.f_81460_, Vec3.atCenterOf(p_49872_)).m_287286_(LootContextParams.f_81463_, ItemStack.f_41583_).m_287289_(LootContextParams.f_81462_, p_49873_);
       return p_49870_.m_287290_(lootparams$builder);
    }
 
    public static List<ItemStack> m_49874_(BlockState p_49875_, ServerLevel p_49876_, BlockPos p_49877_, @Nullable BlockEntity p_49878_, @Nullable Entity p_49879_, ItemStack p_49880_) {
-      LootParams.Builder lootparams$builder = (new LootParams.Builder(p_49876_)).m_287286_(LootContextParams.f_81460_, Vec3.m_82512_(p_49877_)).m_287286_(LootContextParams.f_81463_, p_49880_).m_287289_(LootContextParams.f_81455_, p_49879_).m_287289_(LootContextParams.f_81462_, p_49878_);
+      LootParams.Builder lootparams$builder = (new LootParams.Builder(p_49876_)).m_287286_(LootContextParams.f_81460_, Vec3.atCenterOf(p_49877_)).m_287286_(LootContextParams.f_81463_, p_49880_).m_287289_(LootContextParams.f_81455_, p_49879_).m_287289_(LootContextParams.f_81462_, p_49878_);
       return p_49875_.m_287290_(lootparams$builder);
    }
 
@@ -287,9 +_,9 @@
 
    public static void m_49840_(Level p_49841_, BlockPos p_49842_, ItemStack p_49843_) {
       double d0 = (double)EntityType.f_20461_.m_20679_() / 2.0D;
-      double d1 = (double)p_49842_.m_123341_() + 0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D);
-      double d2 = (double)p_49842_.m_123342_() + 0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D) - d0;
-      double d3 = (double)p_49842_.m_123343_() + 0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D);
+      me.alphamode.mclong.math.BigDecimal d1 = p_49842_.getBigDecX().add(0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D));
+      me.alphamode.mclong.math.BigDecimal d2 = p_49842_.getBigDecY().add(0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D) - d0);
+      me.alphamode.mclong.math.BigDecimal d3 = p_49842_.getBigDecZ().add(0.5D + Mth.m_216263_(p_49841_.f_46441_, -0.25D, 0.25D));
       m_152440_(p_49841_, () -> {
          return new ItemEntity(p_49841_, d1, d2, d3, p_49843_);
       }, p_49843_);
@@ -301,9 +_,9 @@
       int k = p_152438_.m_122431_();
       double d0 = (double)EntityType.f_20461_.m_20678_() / 2.0D;
       double d1 = (double)EntityType.f_20461_.m_20679_() / 2.0D;
-      double d2 = (double)p_152437_.m_123341_() + 0.5D + (i == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)i * (0.5D + d0));
-      double d3 = (double)p_152437_.m_123342_() + 0.5D + (j == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)j * (0.5D + d1)) - d1;
-      double d4 = (double)p_152437_.m_123343_() + 0.5D + (k == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)k * (0.5D + d0));
+      me.alphamode.mclong.math.BigDecimal d2 = p_152437_.getBigX().toBigDecimal().add(0.5D + (i == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)i * (0.5D + d0)));
+      me.alphamode.mclong.math.BigDecimal d3 = p_152437_.getBigY().toBigDecimal().add(0.5D + (j == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)j * (0.5D + d1)) - d1);
+      me.alphamode.mclong.math.BigDecimal d4 = p_152437_.getBigZ().toBigDecimal().add(0.5D + (k == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.25D, 0.25D) : (double)k * (0.5D + d0)));
       double d5 = i == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.1D, 0.1D) : (double)i * 0.1D;
       double d6 = j == 0 ? Mth.m_216263_(p_152436_.f_46441_, 0.0D, 0.1D) : (double)j * 0.1D + 0.1D;
       double d7 = k == 0 ? Mth.m_216263_(p_152436_.f_46441_, -0.1D, 0.1D) : (double)k * 0.1D;
@@ -322,7 +_,7 @@
 
    public void m_49805_(ServerLevel p_49806_, BlockPos p_49807_, int p_49808_) {
       if (p_49806_.m_46469_().m_46207_(GameRules.f_46136_)) {
-         ExperienceOrb.m_147082_(p_49806_, Vec3.m_82512_(p_49807_), p_49808_);
+         ExperienceOrb.m_147082_(p_49806_, Vec3.atCenterOf(p_49807_), p_49808_);
       }
 
    }
