--- a/net/minecraft/world/level/block/BrewingStandBlock.java
+++ b/net/minecraft/world/level/block/BrewingStandBlock.java
@@ -80,9 +_,9 @@
    }
 
    public void m_214162_(BlockState p_220883_, Level p_220884_, BlockPos p_220885_, RandomSource p_220886_) {
-      double d0 = (double)p_220885_.m_123341_() + 0.4D + (double)p_220886_.m_188501_() * 0.2D;
-      double d1 = (double)p_220885_.m_123342_() + 0.7D + (double)p_220886_.m_188501_() * 0.3D;
-      double d2 = (double)p_220885_.m_123343_() + 0.4D + (double)p_220886_.m_188501_() * 0.2D;
+      double d0 = (double)p_220885_.getX() + 0.4D + (double)p_220886_.m_188501_() * 0.2D;
+      double d1 = (double)p_220885_.getY() + 0.7D + (double)p_220886_.m_188501_() * 0.3D;
+      double d2 = (double)p_220885_.getZ() + 0.4D + (double)p_220886_.m_188501_() * 0.2D;
       p_220884_.m_7106_(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
    }
 
