--- a/net/minecraft/world/level/block/BrushableBlock.java
+++ b/net/minecraft/world/level/block/BrushableBlock.java
@@ -45,11 +_,11 @@
    }
 
    public void m_6807_(BlockState p_277817_, Level p_277984_, BlockPos p_277869_, BlockState p_277926_, boolean p_277736_) {
-      p_277984_.m_186460_(p_277869_, this, 2);
+      p_277984_.scheduleTick(p_277869_, this, 2);
    }
 
    public BlockState m_7417_(BlockState p_277801_, Direction p_277455_, BlockState p_277832_, LevelAccessor p_277473_, BlockPos p_278111_, BlockPos p_277904_) {
-      p_277473_.m_186460_(p_278111_, this, 2);
+      p_277473_.scheduleTick(p_278111_, this, 2);
       return super.m_7417_(p_277801_, p_277455_, p_277832_, p_277473_, p_278111_, p_277904_);
    }
 
@@ -59,7 +_,7 @@
          brushableblockentity.m_277175_();
       }
 
-      if (FallingBlock.m_53241_(p_277779_.m_8055_(p_278019_.m_7495_())) && p_278019_.m_123342_() >= p_277779_.m_141937_()) {
+      if (FallingBlock.m_53241_(p_277779_.m_8055_(p_278019_.below())) && p_278019_.getY() >= p_277779_.getMinBuildHeight()) {
          FallingBlockEntity fallingblockentity = FallingBlockEntity.m_201971_(p_277779_, p_278019_, p_277544_);
          fallingblockentity.m_272001_();
       }
@@ -67,17 +_,17 @@
 
    public void m_142525_(Level p_278097_, BlockPos p_277734_, FallingBlockEntity p_277539_) {
       Vec3 vec3 = p_277539_.m_20191_().m_82399_();
-      p_278097_.m_46796_(2001, BlockPos.m_274446_(vec3), Block.m_49956_(p_277539_.m_31980_()));
+      p_278097_.m_46796_(2001, BlockPos.containing(vec3), Block.m_49956_(p_277539_.m_31980_()));
       p_278097_.m_220400_(p_277539_, GameEvent.f_157794_, vec3);
    }
 
    public void m_214162_(BlockState p_277390_, Level p_277525_, BlockPos p_278107_, RandomSource p_277574_) {
       if (p_277574_.m_188503_(16) == 0) {
-         BlockPos blockpos = p_278107_.m_7495_();
+         BlockPos blockpos = p_278107_.below();
          if (FallingBlock.m_53241_(p_277525_.m_8055_(blockpos))) {
-            double d0 = (double)p_278107_.m_123341_() + p_277574_.m_188500_();
-            double d1 = (double)p_278107_.m_123342_() - 0.05D;
-            double d2 = (double)p_278107_.m_123343_() + p_277574_.m_188500_();
+            double d0 = (double)p_278107_.getX() + p_277574_.m_188500_();
+            double d1 = (double)p_278107_.getY() - 0.05D;
+            double d2 = (double)p_278107_.getZ() + p_277574_.m_188500_();
             p_277525_.m_7106_(new BlockParticleOption(ParticleTypes.f_123814_, p_277390_), d0, d1, d2, 0.0D, 0.0D, 0.0D);
          }
       }
