--- a/net/minecraft/world/level/block/BuddingAmethystBlock.java
+++ b/net/minecraft/world/level/block/BuddingAmethystBlock.java
@@ -24,7 +_,7 @@
    public void m_213898_(BlockState p_220898_, ServerLevel p_220899_, BlockPos p_220900_, RandomSource p_220901_) {
       if (p_220901_.m_188503_(5) == 0) {
          Direction direction = f_152723_[p_220901_.m_188503_(f_152723_.length)];
-         BlockPos blockpos = p_220900_.m_121945_(direction);
+         BlockPos blockpos = p_220900_.relative(direction);
          BlockState blockstate = p_220899_.m_8055_(blockpos);
          Block block = null;
          if (m_152734_(blockstate)) {
