--- a/net/minecraft/world/level/block/CampfireBlock.java
+++ b/net/minecraft/world/level/block/CampfireBlock.java
@@ -106,7 +_,7 @@
       LevelAccessor levelaccessor = p_51240_.m_43725_();
       BlockPos blockpos = p_51240_.m_8083_();
       boolean flag = levelaccessor.m_6425_(blockpos).m_76152_() == Fluids.f_76193_;
-      return this.m_49966_().m_61124_(f_51229_, Boolean.valueOf(flag)).m_61124_(f_51228_, Boolean.valueOf(this.m_51323_(levelaccessor.m_8055_(blockpos.m_7495_())))).m_61124_(f_51227_, Boolean.valueOf(!flag)).m_61124_(f_51230_, p_51240_.m_8125_());
+      return this.m_49966_().m_61124_(f_51229_, Boolean.valueOf(flag)).m_61124_(f_51228_, Boolean.valueOf(this.m_51323_(levelaccessor.m_8055_(blockpos.below())))).m_61124_(f_51227_, Boolean.valueOf(!flag)).m_61124_(f_51230_, p_51240_.m_8125_());
    }
 
    public BlockState m_7417_(BlockState p_51298_, Direction p_51299_, BlockState p_51300_, LevelAccessor p_51301_, BlockPos p_51302_, BlockPos p_51303_) {
@@ -132,12 +_,12 @@
    public void m_214162_(BlockState p_220918_, Level p_220919_, BlockPos p_220920_, RandomSource p_220921_) {
       if (p_220918_.m_61143_(f_51227_)) {
          if (p_220921_.m_188503_(10) == 0) {
-            p_220919_.m_7785_((double)p_220920_.m_123341_() + 0.5D, (double)p_220920_.m_123342_() + 0.5D, (double)p_220920_.m_123343_() + 0.5D, SoundEvents.f_11784_, SoundSource.BLOCKS, 0.5F + p_220921_.m_188501_(), p_220921_.m_188501_() * 0.7F + 0.6F, false);
+            p_220919_.m_7785_((double)p_220920_.getX() + 0.5D, (double)p_220920_.getY() + 0.5D, (double)p_220920_.getZ() + 0.5D, SoundEvents.f_11784_, SoundSource.BLOCKS, 0.5F + p_220921_.m_188501_(), p_220921_.m_188501_() * 0.7F + 0.6F, false);
          }
 
          if (this.f_51232_ && p_220921_.m_188503_(5) == 0) {
             for(int i = 0; i < p_220921_.m_188503_(1) + 1; ++i) {
-               p_220919_.m_7106_(ParticleTypes.f_123756_, (double)p_220920_.m_123341_() + 0.5D, (double)p_220920_.m_123342_() + 0.5D, (double)p_220920_.m_123343_() + 0.5D, (double)(p_220921_.m_188501_() / 2.0F), 5.0E-5D, (double)(p_220921_.m_188501_() / 2.0F));
+               p_220919_.m_7106_(ParticleTypes.f_123756_, (double)p_220920_.getX() + 0.5D, (double)p_220920_.getY() + 0.5D, (double)p_220920_.getZ() + 0.5D, (double)(p_220921_.m_188501_() / 2.0F), 5.0E-5D, (double)(p_220921_.m_188501_() / 2.0F));
             }
          }
 
@@ -189,16 +_,16 @@
    public static void m_51251_(Level p_51252_, BlockPos p_51253_, boolean p_51254_, boolean p_51255_) {
       RandomSource randomsource = p_51252_.m_213780_();
       SimpleParticleType simpleparticletype = p_51254_ ? ParticleTypes.f_123778_ : ParticleTypes.f_123777_;
-      p_51252_.m_6485_(simpleparticletype, true, (double)p_51253_.m_123341_() + 0.5D + randomsource.m_188500_() / 3.0D * (double)(randomsource.m_188499_() ? 1 : -1), (double)p_51253_.m_123342_() + randomsource.m_188500_() + randomsource.m_188500_(), (double)p_51253_.m_123343_() + 0.5D + randomsource.m_188500_() / 3.0D * (double)(randomsource.m_188499_() ? 1 : -1), 0.0D, 0.07D, 0.0D);
+      p_51252_.m_6485_(simpleparticletype, true, (double)p_51253_.getX() + 0.5D + randomsource.m_188500_() / 3.0D * (double)(randomsource.m_188499_() ? 1 : -1), (double)p_51253_.getY() + randomsource.m_188500_() + randomsource.m_188500_(), (double)p_51253_.getZ() + 0.5D + randomsource.m_188500_() / 3.0D * (double)(randomsource.m_188499_() ? 1 : -1), 0.0D, 0.07D, 0.0D);
       if (p_51255_) {
-         p_51252_.m_7106_(ParticleTypes.f_123762_, (double)p_51253_.m_123341_() + 0.5D + randomsource.m_188500_() / 4.0D * (double)(randomsource.m_188499_() ? 1 : -1), (double)p_51253_.m_123342_() + 0.4D, (double)p_51253_.m_123343_() + 0.5D + randomsource.m_188500_() / 4.0D * (double)(randomsource.m_188499_() ? 1 : -1), 0.0D, 0.005D, 0.0D);
+         p_51252_.m_7106_(ParticleTypes.f_123762_, (double)p_51253_.getX() + 0.5D + randomsource.m_188500_() / 4.0D * (double)(randomsource.m_188499_() ? 1 : -1), (double)p_51253_.getY() + 0.4D, (double)p_51253_.getZ() + 0.5D + randomsource.m_188500_() / 4.0D * (double)(randomsource.m_188499_() ? 1 : -1), 0.0D, 0.005D, 0.0D);
       }
 
    }
 
    public static boolean m_51248_(Level p_51249_, BlockPos p_51250_) {
       for(int i = 1; i <= 5; ++i) {
-         BlockPos blockpos = p_51250_.m_6625_(i);
+         BlockPos blockpos = p_51250_.below(i);
          BlockState blockstate = p_51249_.m_8055_(blockpos);
          if (m_51319_(blockstate)) {
             return true;
@@ -206,7 +_,7 @@
 
          boolean flag = Shapes.m_83157_(f_51231_, blockstate.m_60742_(p_51249_, p_51250_, CollisionContext.m_82749_()), BooleanOp.f_82689_);
          if (flag) {
-            BlockState blockstate1 = p_51249_.m_8055_(blockpos.m_7495_());
+            BlockState blockstate1 = p_51249_.m_8055_(blockpos.below());
             return m_51319_(blockstate1);
          }
       }
