--- a/net/minecraft/world/level/block/CandleCakeBlock.java
+++ b/net/minecraft/world/level/block/CandleCakeBlock.java
@@ -69,7 +_,7 @@
    }
 
    private static boolean m_152906_(BlockHitResult p_152907_) {
-      return p_152907_.m_82450_().f_82480_ - (double)p_152907_.m_82425_().m_123342_() > 0.5D;
+      return p_152907_.m_82450_().m_7098_() - (double)p_152907_.m_82425_().getY() > 0.5D;
    }
 
    protected void m_7926_(StateDefinition.Builder<Block, BlockState> p_152905_) {
@@ -85,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_152891_, LevelReader p_152892_, BlockPos p_152893_) {
-      return p_152892_.m_8055_(p_152893_.m_7495_()).m_280296_();
+      return p_152892_.m_8055_(p_152893_.below()).m_280296_();
    }
 
    public int m_6782_(BlockState p_152880_, Level p_152881_, BlockPos p_152882_) {
