--- a/net/minecraft/world/level/block/CeilingHangingSignBlock.java
+++ b/net/minecraft/world/level/block/CeilingHangingSignBlock.java
@@ -66,13 +_,13 @@
    }
 
    public boolean m_7898_(BlockState p_248994_, LevelReader p_249061_, BlockPos p_249490_) {
-      return p_249061_.m_8055_(p_249490_.m_7494_()).m_60659_(p_249061_, p_249490_.m_7494_(), Direction.DOWN, SupportType.CENTER);
+      return p_249061_.m_8055_(p_249490_.above()).m_60659_(p_249061_, p_249490_.above(), Direction.DOWN, SupportType.CENTER);
    }
 
    public BlockState m_5573_(BlockPlaceContext p_252121_) {
       Level level = p_252121_.m_43725_();
       FluidState fluidstate = level.m_6425_(p_252121_.m_8083_());
-      BlockPos blockpos = p_252121_.m_8083_().m_7494_();
+      BlockPos blockpos = p_252121_.m_8083_().above();
       BlockState blockstate = level.m_8055_(blockpos);
       boolean flag = blockstate.m_204336_(BlockTags.f_244320_);
       Direction direction = Direction.m_122364_((double)p_252121_.m_7074_());
