--- a/net/minecraft/world/level/block/ChangeOverTimeBlock.java
+++ b/net/minecraft/world/level/block/ChangeOverTimeBlock.java
@@ -28,8 +_,8 @@
       int j = 0;
       int k = 0;
 
-      for(BlockPos blockpos : BlockPos.m_121925_(p_220955_, 4, 4, 4)) {
-         int l = blockpos.m_123333_(p_220955_);
+      for(BlockPos blockpos : BlockPos.withinManhattan(p_220955_, 4, 4, 4)) {
+         int l = blockpos.distManhattan(p_220955_);
          if (l > 4) {
             break;
          }
