--- a/net/minecraft/world/level/block/CherryLeavesBlock.java
+++ b/net/minecraft/world/level/block/CherryLeavesBlock.java
@@ -17,7 +_,7 @@
    public void m_214162_(BlockState p_272714_, Level p_272837_, BlockPos p_273218_, RandomSource p_273360_) {
       super.m_214162_(p_272714_, p_272837_, p_273218_, p_273360_);
       if (p_273360_.m_188503_(10) == 0) {
-         BlockPos blockpos = p_273218_.m_7495_();
+         BlockPos blockpos = p_273218_.below();
          BlockState blockstate = p_272837_.m_8055_(blockpos);
          if (!m_49918_(blockstate.m_60812_(p_272837_, blockpos), Direction.UP)) {
             ParticleUtils.m_272037_(p_272837_, p_273218_, p_273360_, ParticleTypes.f_276452_);
