--- a/net/minecraft/world/level/block/CocoaBlock.java
+++ b/net/minecraft/world/level/block/CocoaBlock.java
@@ -57,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_51767_, LevelReader p_51768_, BlockPos p_51769_) {
-      BlockState blockstate = p_51768_.m_8055_(p_51769_.m_121945_(p_51767_.m_61143_(f_54117_)));
+      BlockState blockstate = p_51768_.m_8055_(p_51769_.relative(p_51767_.m_61143_(f_54117_)));
       return blockstate.m_204336_(BlockTags.f_13111_);
    }
 
