--- a/net/minecraft/world/level/block/CommandBlock.java
+++ b/net/minecraft/world/level/block/CommandBlock.java
@@ -56,7 +_,7 @@
             if (!flag1 && !commandblockentity.m_59143_() && commandblockentity.m_59148_() != CommandBlockEntity.Mode.SEQUENCE) {
                if (flag) {
                   commandblockentity.m_59146_();
-                  p_51839_.m_186460_(p_51840_, this, 1);
+                  p_51839_.scheduleTick(p_51840_, this, 1);
                }
 
             }
@@ -80,7 +_,7 @@
             }
 
             if (commandblockentity.m_59142_() || commandblockentity.m_59143_()) {
-               p_221006_.m_186460_(p_221007_, this, 1);
+               p_221006_.scheduleTick(p_221007_, this, 1);
             }
          } else if (commandblockentity$mode == CommandBlockEntity.Mode.REDSTONE) {
             if (flag1) {
