--- a/net/minecraft/world/level/block/ComposterBlock.java
+++ b/net/minecraft/world/level/block/ComposterBlock.java
@@ -178,7 +_,7 @@
    public static void m_51923_(Level p_51924_, BlockPos p_51925_, boolean p_51926_) {
       BlockState blockstate = p_51924_.m_8055_(p_51925_);
       p_51924_.m_245747_(p_51925_, p_51926_ ? SoundEvents.f_11765_ : SoundEvents.f_11764_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
-      double d0 = blockstate.m_60808_(p_51924_, p_51925_).m_83290_(Direction.Axis.Y, 0.5D, 0.5D) + 0.03125D;
+      double d0 = blockstate.m_60808_(p_51924_, p_51925_).max(Direction.Axis.Y, 0.5D, 0.5D).doubleValue() + 0.03125D;
       double d1 = (double)0.13125F;
       double d2 = (double)0.7375F;
       RandomSource randomsource = p_51924_.m_213780_();
@@ -187,7 +_,7 @@
          double d3 = randomsource.m_188583_() * 0.02D;
          double d4 = randomsource.m_188583_() * 0.02D;
          double d5 = randomsource.m_188583_() * 0.02D;
-         p_51924_.m_7106_(ParticleTypes.f_123749_, (double)p_51925_.m_123341_() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), (double)p_51925_.m_123342_() + d0 + (double)randomsource.m_188501_() * (1.0D - d0), (double)p_51925_.m_123343_() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), d3, d4, d5);
+         p_51924_.m_7106_(ParticleTypes.f_123749_, (double)p_51925_.getX() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), (double)p_51925_.getY() + d0 + (double)randomsource.m_188501_() * (1.0D - d0), (double)p_51925_.getZ() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), d3, d4, d5);
       }
 
    }
@@ -206,7 +_,7 @@
 
    public void m_6807_(BlockState p_51978_, Level p_51979_, BlockPos p_51980_, BlockState p_51981_, boolean p_51982_) {
       if (p_51978_.m_61143_(f_51913_) == 7) {
-         p_51979_.m_186460_(p_51980_, p_51978_.m_60734_(), 20);
+         p_51979_.scheduleTick(p_51980_, p_51978_.m_60734_(), 20);
       }
 
    }
@@ -246,7 +_,7 @@
 
    public static BlockState m_51998_(Entity p_270467_, BlockState p_51999_, Level p_52000_, BlockPos p_52001_) {
       if (!p_52000_.f_46443_) {
-         Vec3 vec3 = Vec3.m_272021_(p_52001_, 0.5D, 1.01D, 0.5D).m_272010_(p_52000_.f_46441_, 0.7F);
+         Vec3 vec3 = Vec3.atLowerCornerWithOffset(p_52001_, 0.5D, 1.01D, 0.5D).m_272010_(p_52000_.f_46441_, 0.7F);
          ItemEntity itementity = new ItemEntity(p_52000_, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), new ItemStack(Items.f_42499_));
          itementity.m_32060_();
          p_52000_.m_7967_(itementity);
@@ -275,7 +_,7 @@
          p_270151_.m_7731_(p_270547_, blockstate, 3);
          p_270151_.m_220407_(GameEvent.f_157792_, p_270547_, GameEvent.Context.m_223719_(p_270464_, blockstate));
          if (j == 7) {
-            p_270151_.m_186460_(p_270547_, p_270603_.m_60734_(), 20);
+            p_270151_.scheduleTick(p_270547_, p_270603_.m_60734_(), 20);
          }
 
          return blockstate;
