--- a/net/minecraft/world/level/block/ComposterBlock.java
+++ b/net/minecraft/world/level/block/ComposterBlock.java
@@ -167,7 +_,7 @@
 
    public static void m_51923_(Level p_51924_, BlockPos p_51925_, boolean p_51926_) {
       BlockState blockstate = p_51924_.m_8055_(p_51925_);
-      p_51924_.m_7785_((double)p_51925_.m_123341_(), (double)p_51925_.m_123342_(), (double)p_51925_.m_123343_(), p_51926_ ? SoundEvents.f_11765_ : SoundEvents.f_11764_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
+      p_51924_.m_7785_((double)p_51925_.getX(), (double)p_51925_.getY(), (double)p_51925_.getZ(), p_51926_ ? SoundEvents.f_11765_ : SoundEvents.f_11764_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
       double d0 = blockstate.m_60808_(p_51924_, p_51925_).m_83290_(Direction.Axis.Y, 0.5D, 0.5D) + 0.03125D;
       double d1 = (double)0.13125F;
       double d2 = (double)0.7375F;
@@ -177,7 +_,7 @@
          double d3 = randomsource.m_188583_() * 0.02D;
          double d4 = randomsource.m_188583_() * 0.02D;
          double d5 = randomsource.m_188583_() * 0.02D;
-         p_51924_.m_7106_(ParticleTypes.f_123749_, (double)p_51925_.m_123341_() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), (double)p_51925_.m_123342_() + d0 + (double)randomsource.m_188501_() * (1.0D - d0), (double)p_51925_.m_123343_() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), d3, d4, d5);
+         p_51924_.m_7106_(ParticleTypes.f_123749_, (double)p_51925_.getX() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), (double)p_51925_.getY() + d0 + (double)randomsource.m_188501_() * (1.0D - d0), (double)p_51925_.getZ() + (double)0.13125F + (double)0.7375F * (double)randomsource.m_188501_(), d3, d4, d5);
       }
 
    }
@@ -196,7 +_,7 @@
 
    public void m_6807_(BlockState p_51978_, Level p_51979_, BlockPos p_51980_, BlockState p_51981_, boolean p_51982_) {
       if (p_51978_.m_61143_(f_51913_) == 7) {
-         p_51979_.m_186460_(p_51980_, p_51978_.m_60734_(), 20);
+         p_51979_.scheduleTick(p_51980_, p_51978_.m_60734_(), 20);
       }
 
    }
@@ -240,7 +_,7 @@
          double d0 = (double)(p_52000_.f_46441_.m_188501_() * 0.7F) + (double)0.15F;
          double d1 = (double)(p_52000_.f_46441_.m_188501_() * 0.7F) + (double)0.060000002F + 0.6D;
          double d2 = (double)(p_52000_.f_46441_.m_188501_() * 0.7F) + (double)0.15F;
-         ItemEntity itementity = new ItemEntity(p_52000_, (double)p_52001_.m_123341_() + d0, (double)p_52001_.m_123342_() + d1, (double)p_52001_.m_123343_() + d2, new ItemStack(Items.f_42499_));
+         ItemEntity itementity = new ItemEntity(p_52000_, (double)p_52001_.getX() + d0, (double)p_52001_.getY() + d1, (double)p_52001_.getZ() + d2, new ItemStack(Items.f_42499_));
          itementity.m_32060_();
          p_52000_.m_7967_(itementity);
       }
@@ -266,7 +_,7 @@
          BlockState blockstate = p_51984_.m_61124_(f_51913_, Integer.valueOf(j));
          p_51985_.m_7731_(p_51986_, blockstate, 3);
          if (j == 7) {
-            p_51985_.m_186460_(p_51986_, p_51984_.m_60734_(), 20);
+            p_51985_.scheduleTick(p_51986_, p_51984_.m_60734_(), 20);
          }
 
          return blockstate;
