--- a/net/minecraft/world/level/block/ConcretePowderBlock.java
+++ b/net/minecraft/world/level/block/ConcretePowderBlock.java
@@ -44,7 +_,7 @@
       for(Direction direction : Direction.values()) {
          BlockState blockstate = p_52065_.m_8055_(blockpos$mutableblockpos);
          if (direction != Direction.DOWN || m_52088_(blockstate)) {
-            blockpos$mutableblockpos.m_122159_(p_52066_, direction);
+            blockpos$mutableblockpos.setWithOffset(p_52066_, direction);
             blockstate = p_52065_.m_8055_(blockpos$mutableblockpos);
             if (m_52088_(blockstate) && !blockstate.m_60783_(p_52065_, p_52066_, direction.m_122424_())) {
                flag = true;
