--- a/net/minecraft/world/level/block/CropBlock.java
+++ b/net/minecraft/world/level/block/CropBlock.java
@@ -92,12 +_,12 @@
 
    protected static float m_52272_(Block p_52273_, BlockGetter p_52274_, BlockPos p_52275_) {
       float f = 1.0F;
-      BlockPos blockpos = p_52275_.m_7495_();
+      BlockPos blockpos = p_52275_.below();
 
       for(int i = -1; i <= 1; ++i) {
          for(int j = -1; j <= 1; ++j) {
             float f1 = 0.0F;
-            BlockState blockstate = p_52274_.m_8055_(blockpos.m_7918_(i, 0, j));
+            BlockState blockstate = p_52274_.m_8055_(blockpos.offset(i, 0, j));
             if (blockstate.m_60713_(Blocks.f_50093_)) {
                f1 = 1.0F;
                if (blockstate.m_61143_(FarmBlock.f_53243_) > 0) {
@@ -113,16 +_,16 @@
          }
       }
 
-      BlockPos blockpos1 = p_52275_.m_122012_();
-      BlockPos blockpos2 = p_52275_.m_122019_();
-      BlockPos blockpos3 = p_52275_.m_122024_();
-      BlockPos blockpos4 = p_52275_.m_122029_();
+      BlockPos blockpos1 = p_52275_.north();
+      BlockPos blockpos2 = p_52275_.south();
+      BlockPos blockpos3 = p_52275_.west();
+      BlockPos blockpos4 = p_52275_.east();
       boolean flag = p_52274_.m_8055_(blockpos3).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos4).m_60713_(p_52273_);
       boolean flag1 = p_52274_.m_8055_(blockpos1).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos2).m_60713_(p_52273_);
       if (flag && flag1) {
          f /= 2.0F;
       } else {
-         boolean flag2 = p_52274_.m_8055_(blockpos3.m_122012_()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos4.m_122012_()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos4.m_122019_()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos3.m_122019_()).m_60713_(p_52273_);
+         boolean flag2 = p_52274_.m_8055_(blockpos3.north()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos4.north()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos4.south()).m_60713_(p_52273_) || p_52274_.m_8055_(blockpos3.south()).m_60713_(p_52273_);
          if (flag2) {
             f /= 2.0F;
          }
