--- a/net/minecraft/world/level/block/CryingObsidianBlock.java
+++ b/net/minecraft/world/level/block/CryingObsidianBlock.java
@@ -17,13 +_,13 @@
       if (p_221058_.m_188503_(5) == 0) {
          Direction direction = Direction.m_235672_(p_221058_);
          if (direction != Direction.UP) {
-            BlockPos blockpos = p_221057_.m_121945_(direction);
+            BlockPos blockpos = p_221057_.relative(direction);
             BlockState blockstate = p_221056_.m_8055_(blockpos);
             if (!p_221055_.m_60815_() || !blockstate.m_60783_(p_221056_, blockpos, direction.m_122424_())) {
                double d0 = direction.m_122429_() == 0 ? p_221058_.m_188500_() : 0.5D + (double)direction.m_122429_() * 0.6D;
                double d1 = direction.m_122430_() == 0 ? p_221058_.m_188500_() : 0.5D + (double)direction.m_122430_() * 0.6D;
                double d2 = direction.m_122431_() == 0 ? p_221058_.m_188500_() : 0.5D + (double)direction.m_122431_() * 0.6D;
-               p_221056_.m_7106_(ParticleTypes.f_123786_, (double)p_221057_.m_123341_() + d0, (double)p_221057_.m_123342_() + d1, (double)p_221057_.m_123343_() + d2, 0.0D, 0.0D, 0.0D);
+               p_221056_.addParticle(ParticleTypes.f_123786_, p_221057_.getBigDecX().add(d0), p_221057_.getBigDecY().add(d1), p_221057_.getBigDecZ().add(d2), 0.0D, 0.0D, 0.0D);
             }
          }
       }
