--- a/net/minecraft/world/level/block/DispenserBlock.java
+++ b/net/minecraft/world/level/block/DispenserBlock.java
@@ -94,10 +_,10 @@
    }
 
    public void m_6861_(BlockState p_52700_, Level p_52701_, BlockPos p_52702_, Block p_52703_, BlockPos p_52704_, boolean p_52705_) {
-      boolean flag = p_52701_.m_276867_(p_52702_) || p_52701_.m_276867_(p_52702_.m_7494_());
+      boolean flag = p_52701_.m_276867_(p_52702_) || p_52701_.m_276867_(p_52702_.above());
       boolean flag1 = p_52700_.m_61143_(f_52660_);
       if (flag && !flag1) {
-         p_52701_.m_186460_(p_52702_, this, 4);
+         p_52701_.scheduleTick(p_52702_, this, 4);
          p_52701_.m_7731_(p_52702_, p_52700_.m_61124_(f_52660_, Boolean.valueOf(true)), 4);
       } else if (!flag && flag1) {
          p_52701_.m_7731_(p_52702_, p_52700_.m_61124_(f_52660_, Boolean.valueOf(false)), 4);
