--- a/net/minecraft/world/level/block/DoublePlantBlock.java
+++ b/net/minecraft/world/level/block/DoublePlantBlock.java
@@ -41,11 +_,11 @@
    public BlockState m_5573_(BlockPlaceContext p_52863_) {
       BlockPos blockpos = p_52863_.m_8083_();
       Level level = p_52863_.m_43725_();
-      return blockpos.m_123342_() < level.m_151558_() - 1 && level.m_8055_(blockpos.m_7494_()).m_60629_(p_52863_) ? super.m_5573_(p_52863_) : null;
+      return blockpos.getY() < level.getMaxBuildHeight() - 1 && level.m_8055_(blockpos.above()).m_60629_(p_52863_) ? super.m_5573_(p_52863_) : null;
    }
 
    public void m_6402_(Level p_52872_, BlockPos p_52873_, BlockState p_52874_, LivingEntity p_52875_, ItemStack p_52876_) {
-      BlockPos blockpos = p_52873_.m_7494_();
+      BlockPos blockpos = p_52873_.above();
       p_52872_.m_7731_(blockpos, m_182453_(p_52872_, blockpos, this.m_49966_().m_61124_(f_52858_, DoubleBlockHalf.UPPER)), 3);
    }
 
@@ -53,13 +_,13 @@
       if (p_52887_.m_61143_(f_52858_) != DoubleBlockHalf.UPPER) {
          return super.m_7898_(p_52887_, p_52888_, p_52889_);
       } else {
-         BlockState blockstate = p_52888_.m_8055_(p_52889_.m_7495_());
+         BlockState blockstate = p_52888_.m_8055_(p_52889_.below());
          return blockstate.m_60713_(this) && blockstate.m_61143_(f_52858_) == DoubleBlockHalf.LOWER;
       }
    }
 
    public static void m_153173_(LevelAccessor p_153174_, BlockState p_153175_, BlockPos p_153176_, int p_153177_) {
-      BlockPos blockpos = p_153176_.m_7494_();
+      BlockPos blockpos = p_153176_.above();
       p_153174_.m_7731_(p_153176_, m_182453_(p_153174_, p_153176_, p_153175_.m_61124_(f_52858_, DoubleBlockHalf.LOWER)), p_153177_);
       p_153174_.m_7731_(blockpos, m_182453_(p_153174_, blockpos, p_153175_.m_61124_(f_52858_, DoubleBlockHalf.UPPER)), p_153177_);
    }
@@ -87,7 +_,7 @@
    protected static void m_52903_(Level p_52904_, BlockPos p_52905_, BlockState p_52906_, Player p_52907_) {
       DoubleBlockHalf doubleblockhalf = p_52906_.m_61143_(f_52858_);
       if (doubleblockhalf == DoubleBlockHalf.UPPER) {
-         BlockPos blockpos = p_52905_.m_7495_();
+         BlockPos blockpos = p_52905_.below();
          BlockState blockstate = p_52904_.m_8055_(blockpos);
          if (blockstate.m_60713_(p_52906_.m_60734_()) && blockstate.m_61143_(f_52858_) == DoubleBlockHalf.LOWER) {
             BlockState blockstate1 = blockstate.m_60819_().m_192917_(Fluids.f_76193_) ? Blocks.f_49990_.m_49966_() : Blocks.f_50016_.m_49966_();
@@ -103,6 +_,6 @@
    }
 
    public long m_7799_(BlockState p_52891_, BlockPos p_52892_) {
-      return Mth.m_14130_(p_52892_.m_123341_(), p_52892_.m_6625_(p_52891_.m_61143_(f_52858_) == DoubleBlockHalf.LOWER ? 0 : 1).m_123342_(), p_52892_.m_123343_());
+      return Mth.getSeed(p_52892_.getX(), p_52892_.below(p_52891_.m_61143_(f_52858_) == DoubleBlockHalf.LOWER ? 0 : 1).getY(), p_52892_.getZ());
    }
 }
