--- a/net/minecraft/world/level/block/DoubleBlockCombiner.java
+++ b/net/minecraft/world/level/block/DoubleBlockCombiner.java
@@ -24,7 +_,7 @@
          if (flag) {
             return new DoubleBlockCombiner.NeighborCombineResult.Single<>(s);
          } else {
-            BlockPos blockpos = p_52829_.m_121945_(p_52825_.apply(p_52827_));
+            BlockPos blockpos = p_52829_.relative(p_52825_.apply(p_52827_));
             BlockState blockstate = p_52828_.m_8055_(blockpos);
             if (blockstate.m_60713_(p_52827_.m_60734_())) {
                DoubleBlockCombiner.BlockType doubleblockcombiner$blocktype1 = p_52824_.apply(blockstate);
