--- a/net/minecraft/world/level/block/EnchantmentTableBlock.java
+++ b/net/minecraft/world/level/block/EnchantmentTableBlock.java
@@ -32,8 +_,8 @@
 
 public class EnchantmentTableBlock extends BaseEntityBlock {
    protected static final VoxelShape f_52950_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 12.0D, 16.0D);
-   public static final List<BlockPos> f_207902_ = BlockPos.m_121886_(-2, 0, -2, 2, 1, 2).filter((p_207914_) -> {
-      return Math.abs(p_207914_.m_123341_()) == 2 || Math.abs(p_207914_.m_123343_()) == 2;
+   public static final List<BlockPos> f_207902_ = BlockPos.betweenClosedStream(me.alphamode.mclong.math.BigInteger.valueOf(-2), me.alphamode.mclong.math.BigInteger.ZERO, me.alphamode.mclong.math.BigInteger.valueOf(-2), me.alphamode.mclong.math.BigInteger.valueOf(2), me.alphamode.mclong.math.BigInteger.ONE, me.alphamode.mclong.math.BigInteger.valueOf(2)).filter((p_207914_) -> {
+      return Math.abs(p_207914_.getX()) == 2 || Math.abs(p_207914_.getZ()) == 2;
    }).map(BlockPos::m_7949_).toList();
 
    public EnchantmentTableBlock(BlockBehaviour.Properties p_52953_) {
@@ -41,7 +_,7 @@
    }
 
    public static boolean m_207909_(Level p_207910_, BlockPos p_207911_, BlockPos p_207912_) {
-      return p_207910_.m_8055_(p_207911_.m_121955_(p_207912_)).m_204336_(BlockTags.f_278384_) && p_207910_.m_8055_(p_207911_.m_7918_(p_207912_.m_123341_() / 2, p_207912_.m_123342_(), p_207912_.m_123343_() / 2)).m_204336_(BlockTags.f_278486_);
+      return p_207910_.m_8055_(p_207911_.offset(p_207912_)).m_60713_(Blocks.f_50078_) && p_207910_.m_46859_(p_207911_.offset(p_207912_.getX() / 2, p_207912_.getY(), p_207912_.getZ() / 2));
    }
 
    public boolean m_7923_(BlockState p_52997_) {
@@ -57,7 +_,7 @@
 
       for(BlockPos blockpos : f_207902_) {
          if (p_221095_.m_188503_(16) == 0 && m_207909_(p_221093_, p_221094_, blockpos)) {
-            p_221093_.m_7106_(ParticleTypes.f_123809_, (double)p_221094_.m_123341_() + 0.5D, (double)p_221094_.m_123342_() + 2.0D, (double)p_221094_.m_123343_() + 0.5D, (double)((float)blockpos.m_123341_() + p_221095_.m_188501_()) - 0.5D, (double)((float)blockpos.m_123342_() - p_221095_.m_188501_() - 1.0F), (double)((float)blockpos.m_123343_() + p_221095_.m_188501_()) - 0.5D);
+            p_221093_.addParticle(ParticleTypes.f_123809_, p_221094_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_221094_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.TWO), p_221094_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), (double)((float)blockpos.getX() + p_221095_.m_188501_()) - 0.5D, (double)((float)blockpos.getY() - p_221095_.m_188501_() - 1.0F), (double)((float)blockpos.getZ() + p_221095_.m_188501_()) - 0.5D);
          }
       }
 
