--- a/net/minecraft/world/level/block/EnderChestBlock.java
+++ b/net/minecraft/world/level/block/EnderChestBlock.java
@@ -71,7 +_,7 @@
       PlayerEnderChestContainer playerenderchestcontainer = p_53140_.m_36327_();
       BlockEntity blockentity = p_53138_.m_7702_(p_53139_);
       if (playerenderchestcontainer != null && blockentity instanceof EnderChestBlockEntity) {
-         BlockPos blockpos = p_53139_.m_7494_();
+         BlockPos blockpos = p_53139_.above();
          if (p_53138_.m_8055_(blockpos).m_60796_(p_53138_, blockpos)) {
             return InteractionResult.m_19078_(p_53138_.f_46443_);
          } else if (p_53138_.f_46443_) {
@@ -104,13 +_,13 @@
       for(int i = 0; i < 3; ++i) {
          int j = p_221120_.m_188503_(2) * 2 - 1;
          int k = p_221120_.m_188503_(2) * 2 - 1;
-         double d0 = (double)p_221119_.m_123341_() + 0.5D + 0.25D * (double)j;
-         double d1 = (double)((float)p_221119_.m_123342_() + p_221120_.m_188501_());
-         double d2 = (double)p_221119_.m_123343_() + 0.5D + 0.25D * (double)k;
+         me.alphamode.mclong.math.BigDecimal d0 = p_221119_.getBigDecX().add(0.5D + 0.25D * (double)j);
+         me.alphamode.mclong.math.BigDecimal d1 = p_221119_.getBigDecY().add(p_221120_.m_188501_());
+         me.alphamode.mclong.math.BigDecimal d2 = p_221119_.getBigDecZ().add(0.5D + 0.25D * (double)k);
          double d3 = (double)(p_221120_.m_188501_() * (float)j);
          double d4 = ((double)p_221120_.m_188501_() - 0.5D) * 0.125D;
          double d5 = (double)(p_221120_.m_188501_() * (float)k);
-         p_221118_.m_7106_(ParticleTypes.f_123760_, d0, d1, d2, d3, d4, d5);
+         p_221118_.addParticle(ParticleTypes.f_123760_, d0, d1, d2, d3, d4, d5);
       }
 
    }
