--- a/net/minecraft/world/level/block/EndPortalBlock.java
+++ b/net/minecraft/world/level/block/EndPortalBlock.java
@@ -35,7 +_,7 @@
    }
 
    public void m_7892_(BlockState p_53025_, Level p_53026_, BlockPos p_53027_, Entity p_53028_) {
-      if (p_53026_ instanceof ServerLevel && !p_53028_.m_20159_() && !p_53028_.m_20160_() && p_53028_.m_6072_() && Shapes.m_83157_(Shapes.m_83064_(p_53028_.m_20191_().m_82386_((double)(-p_53027_.m_123341_()), (double)(-p_53027_.m_123342_()), (double)(-p_53027_.m_123343_()))), p_53025_.m_60808_(p_53026_, p_53027_), BooleanOp.f_82689_)) {
+      if (p_53026_ instanceof ServerLevel && !p_53028_.m_20159_() && !p_53028_.m_20160_() && p_53028_.m_6072_() && Shapes.m_83157_(Shapes.m_83064_(p_53028_.m_20191_().m_82386_((double)(-p_53027_.getX()), (double)(-p_53027_.getY()), (double)(-p_53027_.getZ()))), p_53025_.m_60808_(p_53026_, p_53027_), BooleanOp.f_82689_)) {
          ResourceKey<Level> resourcekey = p_53026_.m_46472_() == Level.f_46430_ ? Level.f_46428_ : Level.f_46430_;
          ServerLevel serverlevel = ((ServerLevel)p_53026_).m_7654_().m_129880_(resourcekey);
          if (serverlevel == null) {
@@ -48,9 +_,9 @@
    }
 
    public void m_214162_(BlockState p_221102_, Level p_221103_, BlockPos p_221104_, RandomSource p_221105_) {
-      double d0 = (double)p_221104_.m_123341_() + p_221105_.m_188500_();
-      double d1 = (double)p_221104_.m_123342_() + 0.8D;
-      double d2 = (double)p_221104_.m_123343_() + p_221105_.m_188500_();
+      double d0 = (double)p_221104_.getX() + p_221105_.m_188500_();
+      double d1 = (double)p_221104_.getY() + 0.8D;
+      double d2 = (double)p_221104_.getZ() + p_221105_.m_188500_();
       p_221103_.m_7106_(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
    }
 
