--- a/net/minecraft/world/level/block/FaceAttachedHorizontalDirectionalBlock.java
+++ b/net/minecraft/world/level/block/FaceAttachedHorizontalDirectionalBlock.java
@@ -24,7 +_,7 @@
    }
 
    public static boolean m_53196_(LevelReader p_53197_, BlockPos p_53198_, Direction p_53199_) {
-      BlockPos blockpos = p_53198_.m_121945_(p_53199_);
+      BlockPos blockpos = p_53198_.relative(p_53199_);
       return p_53197_.m_8055_(blockpos).m_60783_(p_53197_, blockpos, p_53199_.m_122424_());
    }
 
