--- a/net/minecraft/world/level/block/FenceBlock.java
+++ b/net/minecraft/world/level/block/FenceBlock.java
@@ -68,10 +_,10 @@
       BlockGetter blockgetter = p_53304_.m_43725_();
       BlockPos blockpos = p_53304_.m_8083_();
       FluidState fluidstate = p_53304_.m_43725_().m_6425_(p_53304_.m_8083_());
-      BlockPos blockpos1 = blockpos.m_122012_();
-      BlockPos blockpos2 = blockpos.m_122029_();
-      BlockPos blockpos3 = blockpos.m_122019_();
-      BlockPos blockpos4 = blockpos.m_122024_();
+      BlockPos blockpos1 = blockpos.north();
+      BlockPos blockpos2 = blockpos.east();
+      BlockPos blockpos3 = blockpos.south();
+      BlockPos blockpos4 = blockpos.west();
       BlockState blockstate = blockgetter.m_8055_(blockpos1);
       BlockState blockstate1 = blockgetter.m_8055_(blockpos2);
       BlockState blockstate2 = blockgetter.m_8055_(blockpos3);
