--- a/net/minecraft/world/level/block/GrassBlock.java
+++ b/net/minecraft/world/level/block/GrassBlock.java
@@ -22,7 +_,7 @@
    }
 
    public boolean m_7370_(LevelReader p_256229_, BlockPos p_256432_, BlockState p_255677_, boolean p_256630_) {
-      return p_256229_.m_8055_(p_256432_.m_7494_()).m_60795_();
+      return p_256229_.m_8055_(p_256432_.above()).m_60795_();
    }
 
    public boolean m_214167_(Level p_221275_, RandomSource p_221276_, BlockPos p_221277_, BlockState p_221278_) {
@@ -30,7 +_,7 @@
    }
 
    public void m_214148_(ServerLevel p_221270_, RandomSource p_221271_, BlockPos p_221272_, BlockState p_221273_) {
-      BlockPos blockpos = p_221272_.m_7494_();
+      BlockPos blockpos = p_221272_.above();
       BlockState blockstate = Blocks.f_50034_.m_49966_();
       Optional<Holder.Reference<PlacedFeature>> optional = p_221270_.m_9598_().m_175515_(Registries.f_256988_).m_203636_(VegetationPlacements.f_195459_);
 
@@ -39,8 +_,8 @@
          BlockPos blockpos1 = blockpos;
 
          for(int j = 0; j < i / 16; ++j) {
-            blockpos1 = blockpos1.m_7918_(p_221271_.m_188503_(3) - 1, (p_221271_.m_188503_(3) - 1) * p_221271_.m_188503_(3) / 2, p_221271_.m_188503_(3) - 1);
-            if (!p_221270_.m_8055_(blockpos1.m_7495_()).m_60713_(this) || p_221270_.m_8055_(blockpos1).m_60838_(p_221270_, blockpos1)) {
+            blockpos1 = blockpos1.offset(p_221271_.m_188503_(3) - 1, (p_221271_.m_188503_(3) - 1) * p_221271_.m_188503_(3) / 2, p_221271_.m_188503_(3) - 1);
+            if (!p_221270_.m_8055_(blockpos1.below()).m_60713_(this) || p_221270_.m_8055_(blockpos1).m_60838_(p_221270_, blockpos1)) {
                continue label49;
             }
          }
