--- a/net/minecraft/world/level/block/GrowingPlantBodyBlock.java
+++ b/net/minecraft/world/level/block/GrowingPlantBodyBlock.java
@@ -27,7 +_,7 @@
 
    public BlockState m_7417_(BlockState p_53913_, Direction p_53914_, BlockState p_53915_, LevelAccessor p_53916_, BlockPos p_53917_, BlockPos p_53918_) {
       if (p_53914_ == this.f_53859_.m_122424_() && !p_53913_.m_60710_(p_53916_, p_53917_)) {
-         p_53916_.m_186460_(p_53917_, this, 1);
+         p_53916_.scheduleTick(p_53917_, this, 1);
       }
 
       GrowingPlantHeadBlock growingplantheadblock = this.m_7272_();
@@ -48,7 +_,7 @@
 
    public boolean m_7370_(BlockGetter p_53900_, BlockPos p_53901_, BlockState p_53902_, boolean p_53903_) {
       Optional<BlockPos> optional = this.m_153322_(p_53900_, p_53901_, p_53902_.m_60734_());
-      return optional.isPresent() && this.m_7272_().m_5971_(p_53900_.m_8055_(optional.get().m_121945_(this.f_53859_)));
+      return optional.isPresent() && this.m_7272_().m_5971_(p_53900_.m_8055_(optional.get().relative(this.f_53859_)));
    }
 
    public boolean m_214167_(Level p_221290_, RandomSource p_221291_, BlockPos p_221292_, BlockState p_221293_) {
