--- a/net/minecraft/world/level/block/GrowingPlantBlock.java
+++ b/net/minecraft/world/level/block/GrowingPlantBlock.java
@@ -28,7 +_,7 @@
 
    @Nullable
    public BlockState m_5573_(BlockPlaceContext p_53868_) {
-      BlockState blockstate = p_53868_.m_43725_().m_8055_(p_53868_.m_8083_().m_121945_(this.f_53859_));
+      BlockState blockstate = p_53868_.m_43725_().m_8055_(p_53868_.m_8083_().relative(this.f_53859_));
       return !blockstate.m_60713_(this.m_7272_()) && !blockstate.m_60713_(this.m_7777_()) ? this.m_7722_(p_53868_.m_43725_()) : this.m_7777_().m_49966_();
    }
 
@@ -37,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_53876_, LevelReader p_53877_, BlockPos p_53878_) {
-      BlockPos blockpos = p_53878_.m_121945_(this.f_53859_.m_122424_());
+      BlockPos blockpos = p_53878_.relative(this.f_53859_.m_122424_());
       BlockState blockstate = p_53877_.m_8055_(blockpos);
       if (!this.m_142209_(blockstate)) {
          return false;
