--- a/net/minecraft/world/level/block/HangingRootsBlock.java
+++ b/net/minecraft/world/level/block/HangingRootsBlock.java
@@ -46,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_153347_, LevelReader p_153348_, BlockPos p_153349_) {
-      BlockPos blockpos = p_153349_.m_7494_();
+      BlockPos blockpos = p_153349_.above();
       BlockState blockstate = p_153348_.m_8055_(blockpos);
       return blockstate.m_60783_(p_153348_, blockpos, Direction.DOWN);
    }
