--- a/net/minecraft/world/level/block/HoneyBlock.java
+++ b/net/minecraft/world/level/block/HoneyBlock.java
@@ -63,13 +_,13 @@
    private boolean m_54007_(BlockPos p_54008_, Entity p_54009_) {
       if (p_54009_.m_20096_()) {
          return false;
-      } else if (p_54009_.m_20186_() > (double)p_54008_.m_123342_() + 0.9375D - 1.0E-7D) {
+      } else if (p_54009_.m_20186_() > (double)p_54008_.getY() + 0.9375D - 1.0E-7D) {
          return false;
-      } else if (p_54009_.m_20184_().f_82480_ >= -0.08D) {
+      } else if (p_54009_.m_20184_().m_7098_() >= -0.08D) {
          return false;
       } else {
-         double d0 = Math.abs((double)p_54008_.m_123341_() + 0.5D - p_54009_.m_20185_());
-         double d1 = Math.abs((double)p_54008_.m_123343_() + 0.5D - p_54009_.m_20189_());
+         double d0 = Math.abs((double)p_54008_.getX() + 0.5D - p_54009_.m_20185_());
+         double d1 = Math.abs((double)p_54008_.getZ() + 0.5D - p_54009_.m_20189_());
          double d2 = 0.4375D + (double)(p_54009_.m_20205_() / 2.0F);
          return d0 + 1.0E-7D > d2 || d1 + 1.0E-7D > d2;
       }
@@ -84,11 +_,11 @@
 
    private void m_54019_(Entity p_54020_) {
       Vec3 vec3 = p_54020_.m_20184_();
-      if (vec3.f_82480_ < -0.13D) {
-         double d0 = -0.05D / vec3.f_82480_;
-         p_54020_.m_20256_(new Vec3(vec3.f_82479_ * d0, -0.05D, vec3.f_82481_ * d0));
+      if (vec3.m_7098_() < -0.13D) {
+         double d0 = -0.05D / vec3.m_7098_();
+         p_54020_.m_20256_(new Vec3(vec3.m_7096_() * d0, -0.05D, vec3.m_7094_() * d0));
       } else {
-         p_54020_.m_20256_(new Vec3(vec3.f_82479_, -0.05D, vec3.f_82481_));
+         p_54020_.m_20256_(new Vec3(vec3.m_7096_(), -0.05D, vec3.m_7094_()));
       }
 
       p_54020_.m_183634_();
