--- a/net/minecraft/world/level/block/HoneyBlock.java
+++ b/net/minecraft/world/level/block/HoneyBlock.java
@@ -64,13 +_,13 @@
    private boolean m_54007_(BlockPos p_54008_, Entity p_54009_) {
       if (p_54009_.m_20096_()) {
          return false;
-      } else if (p_54009_.m_20186_() > (double)p_54008_.m_123342_() + 0.9375D - 1.0E-7D) {
+      } else if (p_54009_.m_20186_() > (double)p_54008_.getY() + 0.9375D - 1.0E-7D) {
          return false;
       } else if (p_54009_.m_20184_().f_82480_ >= -0.08D) {
          return false;
       } else {
-         double d0 = Math.abs((double)p_54008_.m_123341_() + 0.5D - p_54009_.m_20185_());
-         double d1 = Math.abs((double)p_54008_.m_123343_() + 0.5D - p_54009_.m_20189_());
+         double d0 = Math.abs((double)p_54008_.getX() + 0.5D - p_54009_.m_20185_());
+         double d1 = Math.abs((double)p_54008_.getZ() + 0.5D - p_54009_.m_20189_());
          double d2 = 0.4375D + (double)(p_54009_.m_20205_() / 2.0F);
          return d0 + 1.0E-7D > d2 || d1 + 1.0E-7D > d2;
       }
