--- a/net/minecraft/world/level/block/HugeMushroomBlock.java
+++ b/net/minecraft/world/level/block/HugeMushroomBlock.java
@@ -28,7 +_,7 @@
    public BlockState m_5573_(BlockPlaceContext p_54138_) {
       BlockGetter blockgetter = p_54138_.m_43725_();
       BlockPos blockpos = p_54138_.m_8083_();
-      return this.m_49966_().m_61124_(f_54132_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_7495_()).m_60713_(this))).m_61124_(f_54131_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_7494_()).m_60713_(this))).m_61124_(f_54127_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_122012_()).m_60713_(this))).m_61124_(f_54128_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_122029_()).m_60713_(this))).m_61124_(f_54129_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_122019_()).m_60713_(this))).m_61124_(f_54130_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.m_122024_()).m_60713_(this)));
+      return this.m_49966_().m_61124_(f_54132_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.below()).m_60713_(this))).m_61124_(f_54131_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.above()).m_60713_(this))).m_61124_(f_54127_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.north()).m_60713_(this))).m_61124_(f_54128_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.east()).m_60713_(this))).m_61124_(f_54129_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.south()).m_60713_(this))).m_61124_(f_54130_, Boolean.valueOf(!blockgetter.m_8055_(blockpos.west()).m_60713_(this)));
    }
 
    public BlockState m_7417_(BlockState p_54146_, Direction p_54147_, BlockState p_54148_, LevelAccessor p_54149_, BlockPos p_54150_, BlockPos p_54151_) {
