--- a/net/minecraft/world/level/block/IceBlock.java
+++ b/net/minecraft/world/level/block/IceBlock.java
@@ -29,7 +_,7 @@
             return;
          }
 
-         Material material = p_54157_.m_8055_(p_54159_.m_7495_()).m_60767_();
+         Material material = p_54157_.m_8055_(p_54159_.below()).m_60767_();
          if (material.m_76334_() || material.m_76332_()) {
             p_54157_.m_46597_(p_54159_, Blocks.f_49990_.m_49966_());
          }
