--- a/net/minecraft/world/level/block/IceBlock.java
+++ b/net/minecraft/world/level/block/IceBlock.java
@@ -31,7 +_,7 @@
             return;
          }
 
-         BlockState blockstate = p_54157_.m_8055_(p_54159_.m_7495_());
+         BlockState blockstate = p_54157_.m_8055_(p_54159_.below());
          if (blockstate.m_280555_() || blockstate.m_278721_()) {
             p_54157_.m_46597_(p_54159_, m_278844_());
          }
