--- a/net/minecraft/world/level/block/JukeboxBlock.java
+++ b/net/minecraft/world/level/block/JukeboxBlock.java
@@ -82,7 +_,7 @@
                double d1 = (double)(p_54261_.f_46441_.m_188501_() * 0.7F) + (double)0.060000002F + 0.6D;
                double d2 = (double)(p_54261_.f_46441_.m_188501_() * 0.7F) + (double)0.15F;
                ItemStack itemstack1 = itemstack.m_41777_();
-               ItemEntity itementity = new ItemEntity(p_54261_, (double)p_54262_.m_123341_() + d0, (double)p_54262_.m_123342_() + d1, (double)p_54262_.m_123343_() + d2, itemstack1);
+               ItemEntity itementity = new ItemEntity(p_54261_, (double)p_54262_.getX() + d0, (double)p_54262_.getY() + d1, (double)p_54262_.getZ() + d2, itemstack1);
                itementity.m_32060_();
                p_54261_.m_7967_(itementity);
             }
