--- a/net/minecraft/world/level/block/LanternBlock.java
+++ b/net/minecraft/world/level/block/LanternBlock.java
@@ -57,7 +_,7 @@
 
    public boolean m_7898_(BlockState p_153479_, LevelReader p_153480_, BlockPos p_153481_) {
       Direction direction = m_153495_(p_153479_).m_122424_();
-      return Block.m_49863_(p_153480_, p_153481_.m_121945_(direction), direction.m_122424_());
+      return Block.m_49863_(p_153480_, p_153481_.relative(direction), direction.m_122424_());
    }
 
    protected static Direction m_153495_(BlockState p_153496_) {
