--- a/net/minecraft/world/level/block/LeavesBlock.java
+++ b/net/minecraft/world/level/block/LeavesBlock.java
@@ -68,7 +_,7 @@
 
       int i = m_54463_(p_54442_) + 1;
       if (i != 1 || p_54440_.m_61143_(f_54418_) != i) {
-         p_54443_.m_186460_(p_54444_, this, 1);
+         p_54443_.scheduleTick(p_54444_, this, 1);
       }
 
       return p_54440_;
@@ -79,7 +_,7 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(Direction direction : Direction.values()) {
-         blockpos$mutableblockpos.m_122159_(p_54438_, direction);
+         blockpos$mutableblockpos.setWithOffset(p_54438_, direction);
          i = Math.min(i, m_54463_(p_54437_.m_8055_(blockpos$mutableblockpos)) + 1);
          if (i == 1) {
             break;
@@ -102,14 +_,14 @@
    }
 
    public void m_214162_(BlockState p_221374_, Level p_221375_, BlockPos p_221376_, RandomSource p_221377_) {
-      if (p_221375_.m_46758_(p_221376_.m_7494_())) {
+      if (p_221375_.m_46758_(p_221376_.above())) {
          if (p_221377_.m_188503_(15) == 1) {
-            BlockPos blockpos = p_221376_.m_7495_();
+            BlockPos blockpos = p_221376_.below();
             BlockState blockstate = p_221375_.m_8055_(blockpos);
             if (!blockstate.m_60815_() || !blockstate.m_60783_(p_221375_, blockpos, Direction.UP)) {
-               double d0 = (double)p_221376_.m_123341_() + p_221377_.m_188500_();
-               double d1 = (double)p_221376_.m_123342_() - 0.05D;
-               double d2 = (double)p_221376_.m_123343_() + p_221377_.m_188500_();
+               double d0 = (double)p_221376_.getX() + p_221377_.m_188500_();
+               double d1 = (double)p_221376_.getY() - 0.05D;
+               double d2 = (double)p_221376_.getZ() + p_221377_.m_188500_();
                p_221375_.m_7106_(ParticleTypes.f_123803_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
             }
          }
