--- a/net/minecraft/world/level/block/LecternBlock.java
+++ b/net/minecraft/world/level/block/LecternBlock.java
@@ -147,7 +_,7 @@
 
    public static void m_54488_(Level p_54489_, BlockPos p_54490_, BlockState p_54491_) {
       m_54553_(p_54489_, p_54490_, p_54491_, true);
-      p_54489_.m_186460_(p_54490_, p_54491_.m_60734_(), 2);
+      p_54489_.scheduleTick(p_54490_, p_54491_.m_60734_(), 2);
       p_54489_.m_46796_(1043, p_54490_, 0);
    }
 
@@ -157,7 +_,7 @@
    }
 
    private static void m_54544_(Level p_54545_, BlockPos p_54546_, BlockState p_54547_) {
-      p_54545_.m_46672_(p_54546_.m_7495_(), p_54547_.m_60734_());
+      p_54545_.m_46672_(p_54546_.below(), p_54547_.m_60734_());
    }
 
    public void m_213897_(BlockState p_221388_, ServerLevel p_221389_, BlockPos p_221390_, RandomSource p_221391_) {
@@ -171,7 +_,7 @@
          }
 
          if (p_54531_.m_61143_(f_54466_)) {
-            p_54532_.m_46672_(p_54533_.m_7495_(), this);
+            p_54532_.m_46672_(p_54533_.below(), this);
          }
 
          super.m_6810_(p_54531_, p_54532_, p_54533_, p_54534_, p_54535_);
@@ -185,7 +_,7 @@
          ItemStack itemstack = lecternblockentity.m_59566_().m_41777_();
          float f = 0.25F * (float)direction.m_122429_();
          float f1 = 0.25F * (float)direction.m_122431_();
-         ItemEntity itementity = new ItemEntity(p_54589_, (double)p_54590_.m_123341_() + 0.5D + (double)f, (double)(p_54590_.m_123342_() + 1), (double)p_54590_.m_123343_() + 0.5D + (double)f1, itemstack);
+         ItemEntity itementity = new ItemEntity(p_54589_, (double)p_54590_.getX() + 0.5D + (double)f, (double)(p_54590_.getY() + 1), (double)p_54590_.getZ() + 0.5D + (double)f1, itemstack);
          itementity.m_32060_();
          p_54589_.m_7967_(itementity);
          lecternblockentity.m_6211_();
