--- a/net/minecraft/world/level/block/LightningRodBlock.java
+++ b/net/minecraft/world/level/block/LightningRodBlock.java
@@ -72,12 +_,12 @@
    public void m_153760_(BlockState p_153761_, Level p_153762_, BlockPos p_153763_) {
       p_153762_.m_7731_(p_153763_, p_153761_.m_61124_(f_153703_, Boolean.valueOf(true)), 3);
       this.m_153764_(p_153761_, p_153762_, p_153763_);
-      p_153762_.m_186460_(p_153763_, this, 8);
+      p_153762_.scheduleTick(p_153763_, this, 8);
       p_153762_.m_46796_(3002, p_153763_, p_153761_.m_61143_(f_52588_).m_122434_().ordinal());
    }
 
    private void m_153764_(BlockState p_153765_, Level p_153766_, BlockPos p_153767_) {
-      p_153766_.m_46672_(p_153767_.m_121945_(p_153765_.m_61143_(f_52588_).m_122424_()), this);
+      p_153766_.m_46672_(p_153767_.relative(p_153765_.m_61143_(f_52588_).m_122424_()), this);
    }
 
    public void m_213897_(BlockState p_221400_, ServerLevel p_221401_, BlockPos p_221402_, RandomSource p_221403_) {
@@ -86,7 +_,7 @@
    }
 
    public void m_214162_(BlockState p_221405_, Level p_221406_, BlockPos p_221407_, RandomSource p_221408_) {
-      if (p_221406_.m_46470_() && (long)p_221406_.f_46441_.m_188503_(200) <= p_221406_.m_46467_() % 200L && p_221407_.m_123342_() == p_221406_.m_6924_(Heightmap.Types.WORLD_SURFACE, p_221407_.m_123341_(), p_221407_.m_123343_()) - 1) {
+      if (p_221406_.m_46470_() && (long)p_221406_.f_46441_.m_188503_(200) <= p_221406_.m_46467_() % 200L && p_221407_.getY() == p_221406_.getHeight(Heightmap.Types.WORLD_SURFACE, p_221407_.getBigX(), p_221407_.getBigZ()) - 1) {
          ParticleUtils.m_144967_(p_221405_.m_61143_(f_52588_).m_122434_(), p_221406_, p_221407_, 0.125D, ParticleTypes.f_175830_, UniformInt.m_146622_(1, 2));
       }
    }
@@ -116,7 +_,7 @@
          if (p_153713_.m_45527_(blockpos)) {
             LightningBolt lightningbolt = EntityType.f_20465_.m_20615_(p_153713_);
             if (lightningbolt != null) {
-               lightningbolt.m_20219_(Vec3.m_82539_(blockpos.m_7494_()));
+               lightningbolt.m_20219_(Vec3.atBottomCenterOf(blockpos.above()));
                Entity entity = p_153716_.m_19749_();
                lightningbolt.m_20879_(entity instanceof ServerPlayer ? (ServerPlayer)entity : null);
                p_153713_.m_7967_(lightningbolt);
