--- a/net/minecraft/world/level/block/LiquidBlock.java
+++ b/net/minecraft/world/level/block/LiquidBlock.java
@@ -50,7 +_,7 @@
    }
 
    public VoxelShape m_5939_(BlockState p_54760_, BlockGetter p_54761_, BlockPos p_54762_, CollisionContext p_54763_) {
-      return p_54763_.m_6513_(f_54690_, p_54762_, true) && p_54760_.m_61143_(f_54688_) == 0 && p_54763_.m_203682_(p_54761_.m_6425_(p_54762_.m_7494_()), p_54760_.m_60819_()) ? f_54690_ : Shapes.m_83040_();
+      return p_54763_.m_6513_(f_54690_, p_54762_, true) && p_54760_.m_61143_(f_54688_) == 0 && p_54763_.m_203682_(p_54761_.m_6425_(p_54762_.above()), p_54760_.m_60819_()) ? f_54690_ : Shapes.m_83040_();
    }
 
    public boolean m_6724_(BlockState p_54732_) {
@@ -114,10 +_,10 @@
 
    private boolean m_54696_(Level p_54697_, BlockPos p_54698_, BlockState p_54699_) {
       if (this.f_54689_.m_205067_(FluidTags.f_13132_)) {
-         boolean flag = p_54697_.m_8055_(p_54698_.m_7495_()).m_60713_(Blocks.f_50136_);
+         boolean flag = p_54697_.m_8055_(p_54698_.below()).m_60713_(Blocks.f_50136_);
 
          for(Direction direction : f_181233_) {
-            BlockPos blockpos = p_54698_.m_121945_(direction.m_122424_());
+            BlockPos blockpos = p_54698_.relative(direction.m_122424_());
             if (p_54697_.m_6425_(blockpos).m_205070_(FluidTags.f_13131_)) {
                Block block = p_54697_.m_6425_(p_54698_).m_76170_() ? Blocks.f_50080_ : Blocks.f_50652_;
                p_54697_.m_46597_(p_54698_, block.m_49966_());
