--- a/net/minecraft/world/level/block/MagmaBlock.java
+++ b/net/minecraft/world/level/block/MagmaBlock.java
@@ -28,18 +_,18 @@
    }
 
    public void m_213897_(BlockState p_221415_, ServerLevel p_221416_, BlockPos p_221417_, RandomSource p_221418_) {
-      BubbleColumnBlock.m_152707_(p_221416_, p_221417_.m_7494_(), p_221415_);
+      BubbleColumnBlock.m_152707_(p_221416_, p_221417_.above(), p_221415_);
    }
 
    public BlockState m_7417_(BlockState p_54811_, Direction p_54812_, BlockState p_54813_, LevelAccessor p_54814_, BlockPos p_54815_, BlockPos p_54816_) {
       if (p_54812_ == Direction.UP && p_54813_.m_60713_(Blocks.f_49990_)) {
-         p_54814_.m_186460_(p_54815_, this, 20);
+         p_54814_.scheduleTick(p_54815_, this, 20);
       }
 
       return super.m_7417_(p_54811_, p_54812_, p_54813_, p_54814_, p_54815_, p_54816_);
    }
 
    public void m_6807_(BlockState p_54823_, Level p_54824_, BlockPos p_54825_, BlockState p_54826_, boolean p_54827_) {
-      p_54824_.m_186460_(p_54825_, this, 20);
+      p_54824_.scheduleTick(p_54825_, this, 20);
    }
 }
