--- a/net/minecraft/world/level/block/MangrovePropaguleBlock.java
+++ b/net/minecraft/world/level/block/MangrovePropaguleBlock.java
@@ -60,11 +_,11 @@
          voxelshape = f_221444_[p_221468_.m_61143_(f_221441_)];
       }
 
-      return voxelshape.m_83216_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+      return voxelshape.m_83216_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
    }
 
    public boolean m_7898_(BlockState p_221473_, LevelReader p_221474_, BlockPos p_221475_) {
-      return m_221499_(p_221473_) ? p_221474_.m_8055_(p_221475_.m_7494_()).m_60713_(Blocks.f_220838_) : super.m_7898_(p_221473_, p_221474_, p_221475_);
+      return m_221499_(p_221473_) ? p_221474_.m_8055_(p_221475_.above()).m_60713_(Blocks.f_220838_) : super.m_7898_(p_221473_, p_221474_, p_221475_);
    }
 
    public BlockState m_7417_(BlockState p_221477_, Direction p_221478_, BlockState p_221479_, LevelAccessor p_221480_, BlockPos p_221481_, BlockPos p_221482_) {
