--- a/net/minecraft/world/level/block/MossBlock.java
+++ b/net/minecraft/world/level/block/MossBlock.java
@@ -16,7 +_,7 @@
    }
 
    public boolean m_7370_(LevelReader p_256507_, BlockPos p_256224_, BlockState p_256628_, boolean p_256093_) {
-      return p_256507_.m_8055_(p_256224_.m_7494_()).m_60795_();
+      return p_256507_.m_8055_(p_256224_.above()).m_60795_();
    }
 
    public boolean m_214167_(Level p_221538_, RandomSource p_221539_, BlockPos p_221540_, BlockState p_221541_) {
@@ -27,7 +_,7 @@
       p_221533_.m_9598_().m_6632_(Registries.f_256911_).flatMap((p_258973_) -> {
          return p_258973_.m_203636_(CaveFeatures.f_194951_);
       }).ifPresent((p_255669_) -> {
-         p_255669_.m_203334_().m_224953_(p_221533_, p_221533_.m_7726_().m_8481_(), p_221534_, p_221535_.m_7494_());
+         p_255669_.m_203334_().m_224953_(p_221533_, p_221533_.m_7726_().m_8481_(), p_221534_, p_221535_.above());
       });
    }
 }
