--- a/net/minecraft/world/level/block/MultifaceSpreader.java
+++ b/net/minecraft/world/level/block/MultifaceSpreader.java
@@ -159,12 +_,12 @@
       },
       SAME_PLANE {
          public MultifaceSpreader.SpreadPos m_213941_(BlockPos p_221758_, Direction p_221759_, Direction p_221760_) {
-            return new MultifaceSpreader.SpreadPos(p_221758_.m_121945_(p_221759_), p_221760_);
+            return new MultifaceSpreader.SpreadPos(p_221758_.relative(p_221759_), p_221760_);
          }
       },
       WRAP_AROUND {
          public MultifaceSpreader.SpreadPos m_213941_(BlockPos p_221765_, Direction p_221766_, Direction p_221767_) {
-            return new MultifaceSpreader.SpreadPos(p_221765_.m_121945_(p_221766_).m_121945_(p_221767_), p_221766_.m_122424_());
+            return new MultifaceSpreader.SpreadPos(p_221765_.relative(p_221766_).relative(p_221767_), p_221766_.m_122424_());
          }
       };
 
