--- a/net/minecraft/world/level/block/MushroomBlock.java
+++ b/net/minecraft/world/level/block/MushroomBlock.java
@@ -36,7 +_,7 @@
          int i = 5;
          int j = 4;
 
-         for(BlockPos blockpos : BlockPos.m_121940_(p_221786_.m_7918_(-4, -1, -4), p_221786_.m_7918_(4, 1, 4))) {
+         for(BlockPos blockpos : BlockPos.m_121940_(p_221786_.offset(-4, -1, -4), p_221786_.offset(4, 1, 4))) {
             if (p_221785_.m_8055_(blockpos).m_60713_(this)) {
                --i;
                if (i <= 0) {
@@ -45,14 +_,14 @@
             }
          }
 
-         BlockPos blockpos1 = p_221786_.m_7918_(p_221787_.m_188503_(3) - 1, p_221787_.m_188503_(2) - p_221787_.m_188503_(2), p_221787_.m_188503_(3) - 1);
+         BlockPos blockpos1 = p_221786_.offset(p_221787_.m_188503_(3) - 1, p_221787_.m_188503_(2) - p_221787_.m_188503_(2), p_221787_.m_188503_(3) - 1);
 
          for(int k = 0; k < 4; ++k) {
             if (p_221785_.m_46859_(blockpos1) && p_221784_.m_60710_(p_221785_, blockpos1)) {
                p_221786_ = blockpos1;
             }
 
-            blockpos1 = p_221786_.m_7918_(p_221787_.m_188503_(3) - 1, p_221787_.m_188503_(2) - p_221787_.m_188503_(2), p_221787_.m_188503_(3) - 1);
+            blockpos1 = p_221786_.offset(p_221787_.m_188503_(3) - 1, p_221787_.m_188503_(2) - p_221787_.m_188503_(2), p_221787_.m_188503_(3) - 1);
          }
 
          if (p_221785_.m_46859_(blockpos1) && p_221784_.m_60710_(p_221785_, blockpos1)) {
@@ -67,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_54880_, LevelReader p_54881_, BlockPos p_54882_) {
-      BlockPos blockpos = p_54882_.m_7495_();
+      BlockPos blockpos = p_54882_.below();
       BlockState blockstate = p_54881_.m_8055_(blockpos);
       if (blockstate.m_204336_(BlockTags.f_13057_)) {
          return true;
