--- a/net/minecraft/world/level/block/MushroomBlock.java
+++ b/net/minecraft/world/level/block/MushroomBlock.java
@@ -65,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_54880_, LevelReader p_54881_, BlockPos p_54882_) {
-      BlockPos blockpos = p_54882_.m_7495_();
+      BlockPos blockpos = p_54882_.below();
       BlockState blockstate = p_54881_.m_8055_(blockpos);
       if (blockstate.m_204336_(BlockTags.f_13057_)) {
          return true;
