--- a/net/minecraft/world/level/block/MyceliumBlock.java
+++ b/net/minecraft/world/level/block/MyceliumBlock.java
@@ -15,7 +_,7 @@
    public void m_214162_(BlockState p_221789_, Level p_221790_, BlockPos p_221791_, RandomSource p_221792_) {
       super.m_214162_(p_221789_, p_221790_, p_221791_, p_221792_);
       if (p_221792_.m_188503_(10) == 0) {
-         p_221790_.m_7106_(ParticleTypes.f_123757_, (double)p_221791_.m_123341_() + p_221792_.m_188500_(), (double)p_221791_.m_123342_() + 1.1D, (double)p_221791_.m_123343_() + p_221792_.m_188500_(), 0.0D, 0.0D, 0.0D);
+         p_221790_.addParticle(ParticleTypes.f_123757_, p_221791_.getBigDecX().add(p_221792_.m_188500_()), p_221791_.getBigDecY().add(1.1D), p_221791_.getBigDecZ().add(p_221792_.m_188500_()), 0.0D, 0.0D, 0.0D);
       }
 
    }
