--- a/net/minecraft/world/level/block/NetherPortalBlock.java
+++ b/net/minecraft/world/level/block/NetherPortalBlock.java
@@ -48,11 +_,11 @@
    public void m_213898_(BlockState p_221799_, ServerLevel p_221800_, BlockPos p_221801_, RandomSource p_221802_) {
       if (p_221800_.m_6042_().f_63858_() && p_221800_.m_46469_().m_46207_(GameRules.f_46134_) && p_221802_.m_188503_(2000) < p_221800_.m_46791_().m_19028_()) {
          while(p_221800_.m_8055_(p_221801_).m_60713_(this)) {
-            p_221801_ = p_221801_.m_7495_();
+            p_221801_ = p_221801_.below();
          }
 
          if (p_221800_.m_8055_(p_221801_).m_60643_(p_221800_, p_221801_, EntityType.f_20531_)) {
-            Entity entity = EntityType.f_20531_.m_262496_(p_221800_, p_221801_.m_7494_(), MobSpawnType.STRUCTURE);
+            Entity entity = EntityType.f_20531_.m_262496_(p_221800_, p_221801_.above(), MobSpawnType.STRUCTURE);
             if (entity != null) {
                entity.m_20091_();
             }
@@ -77,26 +_,26 @@
 
    public void m_214162_(BlockState p_221794_, Level p_221795_, BlockPos p_221796_, RandomSource p_221797_) {
       if (p_221797_.m_188503_(100) == 0) {
-         p_221795_.m_7785_((double)p_221796_.m_123341_() + 0.5D, (double)p_221796_.m_123342_() + 0.5D, (double)p_221796_.m_123343_() + 0.5D, SoundEvents.f_12286_, SoundSource.BLOCKS, 0.5F, p_221797_.m_188501_() * 0.4F + 0.8F, false);
+         p_221795_.playLocalSound(p_221796_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_221796_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_221796_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), SoundEvents.f_12286_, SoundSource.BLOCKS, 0.5F, p_221797_.m_188501_() * 0.4F + 0.8F, false);
       }
 
       for(int i = 0; i < 4; ++i) {
-         double d0 = (double)p_221796_.m_123341_() + p_221797_.m_188500_();
-         double d1 = (double)p_221796_.m_123342_() + p_221797_.m_188500_();
-         double d2 = (double)p_221796_.m_123343_() + p_221797_.m_188500_();
+         me.alphamode.mclong.math.BigDecimal d0 = p_221796_.getBigDecX().add(p_221797_.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d1 = p_221796_.getBigDecY().add(p_221797_.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d2 = p_221796_.getBigDecZ().add(p_221797_.m_188500_());
          double d3 = ((double)p_221797_.m_188501_() - 0.5D) * 0.5D;
          double d4 = ((double)p_221797_.m_188501_() - 0.5D) * 0.5D;
          double d5 = ((double)p_221797_.m_188501_() - 0.5D) * 0.5D;
          int j = p_221797_.m_188503_(2) * 2 - 1;
-         if (!p_221795_.m_8055_(p_221796_.m_122024_()).m_60713_(this) && !p_221795_.m_8055_(p_221796_.m_122029_()).m_60713_(this)) {
-            d0 = (double)p_221796_.m_123341_() + 0.5D + 0.25D * (double)j;
+         if (!p_221795_.m_8055_(p_221796_.west()).m_60713_(this) && !p_221795_.m_8055_(p_221796_.east()).m_60713_(this)) {
+            d0 = p_221796_.getBigDecX().add(0.5D + 0.25D * (double)j);
             d3 = (double)(p_221797_.m_188501_() * 2.0F * (float)j);
          } else {
-            d2 = (double)p_221796_.m_123343_() + 0.5D + 0.25D * (double)j;
+            d2 = p_221796_.getBigDecZ().add(0.5D + 0.25D * (double)j);
             d5 = (double)(p_221797_.m_188501_() * 2.0F * (float)j);
          }
 
-         p_221795_.m_7106_(ParticleTypes.f_123760_, d0, d1, d2, d3, d4, d5);
+         p_221795_.addParticle(ParticleTypes.f_123760_, d0, d1, d2, d3, d4, d5);
       }
 
    }
