--- a/net/minecraft/world/level/block/NetherrackBlock.java
+++ b/net/minecraft/world/level/block/NetherrackBlock.java
@@ -15,7 +_,7 @@
    }
 
    public boolean m_7370_(BlockGetter p_55002_, BlockPos p_55003_, BlockState p_55004_, boolean p_55005_) {
-      if (!p_55002_.m_8055_(p_55003_.m_7494_()).m_60631_(p_55002_, p_55003_)) {
+      if (!p_55002_.m_8055_(p_55003_.above()).m_60631_(p_55002_, p_55003_)) {
          return false;
       } else {
          for(BlockPos blockpos : BlockPos.m_121940_(p_55003_.m_7918_(-1, -1, -1), p_55003_.m_7918_(1, 1, 1))) {
