--- a/net/minecraft/world/level/block/NoteBlock.java
+++ b/net/minecraft/world/level/block/NoteBlock.java
@@ -35,7 +_,7 @@
    }
 
    public BlockState m_5573_(BlockPlaceContext p_55018_) {
-      return this.m_49966_().m_61124_(f_55011_, NoteBlockInstrument.m_61666_(p_55018_.m_43725_().m_8055_(p_55018_.m_8083_().m_7495_())));
+      return this.m_49966_().m_61124_(f_55011_, NoteBlockInstrument.m_61666_(p_55018_.m_43725_().m_8055_(p_55018_.m_8083_().below())));
    }
 
    public BlockState m_7417_(BlockState p_55048_, Direction p_55049_, BlockState p_55050_, LevelAccessor p_55051_, BlockPos p_55052_, BlockPos p_55053_) {
@@ -55,7 +_,7 @@
    }
 
    private void m_221820_(@Nullable Entity p_221821_, Level p_221822_, BlockPos p_221823_) {
-      if (p_221822_.m_8055_(p_221823_.m_7494_()).m_60795_()) {
+      if (p_221822_.m_8055_(p_221823_.above()).m_60795_()) {
          p_221822_.m_7696_(p_221823_, this, 0, 0);
          p_221822_.m_142346_(p_221821_, GameEvent.f_223699_, p_221823_);
       }
@@ -84,7 +_,7 @@
       int i = p_55023_.m_61143_(f_55013_);
       float f = (float)Math.pow(2.0D, (double)(i - 12) / 12.0D);
       p_55024_.m_5594_((Player)null, p_55025_, p_55023_.m_61143_(f_55011_).m_61668_(), SoundSource.RECORDS, 3.0F, f);
-      p_55024_.m_7106_(ParticleTypes.f_123758_, (double)p_55025_.m_123341_() + 0.5D, (double)p_55025_.m_123342_() + 1.2D, (double)p_55025_.m_123343_() + 0.5D, (double)i / 24.0D, 0.0D, 0.0D);
+      p_55024_.m_7106_(ParticleTypes.f_123758_, (double)p_55025_.getX() + 0.5D, (double)p_55025_.getY() + 1.2D, (double)p_55025_.getZ() + 0.5D, (double)i / 24.0D, 0.0D, 0.0D);
       return true;
    }
 
