--- a/net/minecraft/world/level/block/NyliumBlock.java
+++ b/net/minecraft/world/level/block/NyliumBlock.java
@@ -19,7 +_,7 @@
    }
 
    private static boolean m_55078_(BlockState p_55079_, LevelReader p_55080_, BlockPos p_55081_) {
-      BlockPos blockpos = p_55081_.m_7494_();
+      BlockPos blockpos = p_55081_.above();
       BlockState blockstate = p_55080_.m_8055_(blockpos);
       int i = LayerLightEngine.m_75667_(p_55080_, p_55079_, p_55081_, blockstate, blockpos, Direction.UP, blockstate.m_60739_(p_55080_, blockpos));
       return i < p_55080_.m_7469_();
@@ -33,7 +_,7 @@
    }
 
    public boolean m_7370_(BlockGetter p_55064_, BlockPos p_55065_, BlockState p_55066_, boolean p_55067_) {
-      return p_55064_.m_8055_(p_55065_.m_7494_()).m_60795_();
+      return p_55064_.m_8055_(p_55065_.above()).m_60795_();
    }
 
    public boolean m_214167_(Level p_221830_, RandomSource p_221831_, BlockPos p_221832_, BlockState p_221833_) {
@@ -42,7 +_,7 @@
 
    public void m_214148_(ServerLevel p_221825_, RandomSource p_221826_, BlockPos p_221827_, BlockState p_221828_) {
       BlockState blockstate = p_221825_.m_8055_(p_221827_);
-      BlockPos blockpos = p_221827_.m_7494_();
+      BlockPos blockpos = p_221827_.above();
       ChunkGenerator chunkgenerator = p_221825_.m_7726_().m_8481_();
       if (blockstate.m_60713_(Blocks.f_50699_)) {
          NetherFeatures.f_195037_.m_203334_().m_224953_(p_221825_, chunkgenerator, p_221826_, blockpos);
