--- a/net/minecraft/world/level/block/NyliumBlock.java
+++ b/net/minecraft/world/level/block/NyliumBlock.java
@@ -22,7 +_,7 @@
    }
 
    private static boolean m_55078_(BlockState p_55079_, LevelReader p_55080_, BlockPos p_55081_) {
-      BlockPos blockpos = p_55081_.m_7494_();
+      BlockPos blockpos = p_55081_.above();
       BlockState blockstate = p_55080_.m_8055_(blockpos);
       int i = LightEngine.m_284282_(p_55080_, p_55079_, p_55081_, blockstate, blockpos, Direction.UP, blockstate.m_60739_(p_55080_, blockpos));
       return i < p_55080_.m_7469_();
@@ -36,7 +_,7 @@
    }
 
    public boolean m_7370_(LevelReader p_256194_, BlockPos p_256152_, BlockState p_256389_, boolean p_255846_) {
-      return p_256194_.m_8055_(p_256152_.m_7494_()).m_60795_();
+      return p_256194_.m_8055_(p_256152_.above()).m_60795_();
    }
 
    public boolean m_214167_(Level p_221830_, RandomSource p_221831_, BlockPos p_221832_, BlockState p_221833_) {
@@ -45,7 +_,7 @@
 
    public void m_214148_(ServerLevel p_221825_, RandomSource p_221826_, BlockPos p_221827_, BlockState p_221828_) {
       BlockState blockstate = p_221825_.m_8055_(p_221827_);
-      BlockPos blockpos = p_221827_.m_7494_();
+      BlockPos blockpos = p_221827_.above();
       ChunkGenerator chunkgenerator = p_221825_.m_7726_().m_8481_();
       Registry<ConfiguredFeature<?, ?>> registry = p_221825_.m_9598_().m_175515_(Registries.f_256911_);
       if (blockstate.m_60713_(Blocks.f_50699_)) {
