--- a/net/minecraft/world/level/block/PowderSnowBlock.java
+++ b/net/minecraft/world/level/block/PowderSnowBlock.java
@@ -57,7 +_,7 @@
             RandomSource randomsource = p_154264_.m_213780_();
             boolean flag = p_154266_.f_19790_ != p_154266_.m_20185_() || p_154266_.f_19792_ != p_154266_.m_20189_();
             if (flag && randomsource.m_188499_()) {
-               p_154264_.m_7106_(ParticleTypes.f_175821_, p_154266_.m_20185_(), (double)(p_154265_.m_123342_() + 1), p_154266_.m_20189_(), (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F), (double)0.05F, (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F));
+               p_154264_.m_7106_(ParticleTypes.f_175821_, p_154266_.m_20185_(), (double)(p_154265_.getY() + 1), p_154266_.m_20189_(), (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F), (double)0.05F, (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F));
             }
          }
       }
