--- a/net/minecraft/world/level/block/PowderSnowBlock.java
+++ b/net/minecraft/world/level/block/PowderSnowBlock.java
@@ -55,9 +_,9 @@
          p_154266_.m_7601_(p_154263_, new Vec3((double)0.9F, 1.5D, (double)0.9F));
          if (p_154264_.f_46443_) {
             RandomSource randomsource = p_154264_.m_213780_();
-            boolean flag = p_154266_.f_19790_ != p_154266_.m_20185_() || p_154266_.f_19792_ != p_154266_.m_20189_();
+            boolean flag = !p_154266_.f_19790_.equals(p_154266_.getBigX()) || !p_154266_.f_19792_.equals(p_154266_.getBigZ());
             if (flag && randomsource.m_188499_()) {
-               p_154264_.m_7106_(ParticleTypes.f_175821_, p_154266_.m_20185_(), (double)(p_154265_.m_123342_() + 1), p_154266_.m_20189_(), (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F), (double)0.05F, (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F));
+               p_154264_.addParticle(ParticleTypes.f_175821_, p_154266_.getBigX(), p_154265_.getBigDecY().add(me.alphamode.mclong.math.BigDecimal.ONE), p_154266_.getBigZ(), (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F), (double)0.05F, (double)(Mth.m_216283_(randomsource, -1.0F, 1.0F) * 0.083333336F));
             }
          }
       }
