--- a/net/minecraft/world/level/block/PoweredRailBlock.java
+++ b/net/minecraft/world/level/block/PoweredRailBlock.java
@@ -24,9 +_,9 @@
       if (p_55224_ >= 8) {
          return false;
       } else {
-         int i = p_55221_.m_123341_();
-         int j = p_55221_.m_123342_();
-         int k = p_55221_.m_123343_();
+         long i = p_55221_.getX();
+         long j = p_55221_.getY();
+         long k = p_55221_.getZ();
          boolean flag = true;
          RailShape railshape = p_55222_.m_61143_(f_55214_);
          switch (railshape) {
@@ -124,9 +_,9 @@
       boolean flag1 = p_55233_.m_276867_(p_55234_) || this.m_55219_(p_55233_, p_55234_, p_55232_, true, 0) || this.m_55219_(p_55233_, p_55234_, p_55232_, false, 0);
       if (flag1 != flag) {
          p_55233_.m_7731_(p_55234_, p_55232_.m_61124_(f_55215_, Boolean.valueOf(flag1)), 3);
-         p_55233_.m_46672_(p_55234_.m_7495_(), this);
+         p_55233_.m_46672_(p_55234_.below(), this);
          if (p_55232_.m_61143_(f_55214_).m_61745_()) {
-            p_55233_.m_46672_(p_55234_.m_7494_(), this);
+            p_55233_.m_46672_(p_55234_.above(), this);
          }
       }
 
