--- a/net/minecraft/world/level/block/PumpkinBlock.java
+++ b/net/minecraft/world/level/block/PumpkinBlock.java
@@ -30,7 +_,7 @@
             Direction direction1 = direction.m_122434_() == Direction.Axis.Y ? p_55292_.m_6350_().m_122424_() : direction;
             p_55290_.m_5594_((Player)null, p_55291_, SoundEvents.f_12296_, SoundSource.BLOCKS, 1.0F, 1.0F);
             p_55290_.m_7731_(p_55291_, Blocks.f_50143_.m_49966_().m_61124_(CarvedPumpkinBlock.f_51367_, direction1), 11);
-            ItemEntity itementity = new ItemEntity(p_55290_, (double)p_55291_.m_123341_() + 0.5D + (double)direction1.m_122429_() * 0.65D, (double)p_55291_.m_123342_() + 0.1D, (double)p_55291_.m_123343_() + 0.5D + (double)direction1.m_122431_() * 0.65D, new ItemStack(Items.f_42577_, 4));
+            ItemEntity itementity = new ItemEntity(p_55290_, (double)p_55291_.getX() + 0.5D + (double)direction1.m_122429_() * 0.65D, (double)p_55291_.getY() + 0.1D, (double)p_55291_.getZ() + 0.5D + (double)direction1.m_122431_() * 0.65D, new ItemStack(Items.f_42577_, 4));
             itementity.m_20334_(0.05D * (double)direction1.m_122429_() + p_55290_.f_46441_.m_188500_() * 0.02D, 0.05D, 0.05D * (double)direction1.m_122431_() + p_55290_.f_46441_.m_188500_() * 0.02D);
             p_55290_.m_7967_(itementity);
             itemstack.m_41622_(1, p_55292_, (p_55287_) -> {
