--- a/net/minecraft/world/level/block/RedstoneLampBlock.java
+++ b/net/minecraft/world/level/block/RedstoneLampBlock.java
@@ -29,7 +_,7 @@
          boolean flag = p_55666_.m_61143_(f_55654_);
          if (flag != p_55667_.m_276867_(p_55668_)) {
             if (flag) {
-               p_55667_.m_186460_(p_55668_, this, 4);
+               p_55667_.scheduleTick(p_55668_, this, 4);
             } else {
                p_55667_.m_7731_(p_55668_, p_55666_.m_61122_(f_55654_), 2);
             }
