--- a/net/minecraft/world/level/block/RepeaterBlock.java
+++ b/net/minecraft/world/level/block/RepeaterBlock.java
@@ -61,9 +_,9 @@
    public void m_214162_(BlockState p_221964_, Level p_221965_, BlockPos p_221966_, RandomSource p_221967_) {
       if (p_221964_.m_61143_(f_52496_)) {
          Direction direction = p_221964_.m_61143_(f_54117_);
-         double d0 = (double)p_221966_.m_123341_() + 0.5D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
-         double d1 = (double)p_221966_.m_123342_() + 0.4D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
-         double d2 = (double)p_221966_.m_123343_() + 0.5D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
+         double d0 = (double)p_221966_.getX() + 0.5D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
+         double d1 = (double)p_221966_.getY() + 0.4D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
+         double d2 = (double)p_221966_.getZ() + 0.5D + (p_221967_.m_188500_() - 0.5D) * 0.2D;
          float f = -5.0F;
          if (p_221967_.m_188499_()) {
             f = (float)(p_221964_.m_61143_(f_55798_) * 2 - 1);
