--- a/net/minecraft/world/level/block/SculkCatalystBlock.java
+++ b/net/minecraft/world/level/block/SculkCatalystBlock.java
@@ -46,8 +_,8 @@
 
    public static void m_222094_(ServerLevel p_222095_, BlockPos p_222096_, BlockState p_222097_, RandomSource p_222098_) {
       p_222095_.m_7731_(p_222096_, p_222097_.m_61124_(f_222086_, Boolean.valueOf(true)), 3);
-      p_222095_.m_186460_(p_222096_, p_222097_.m_60734_(), 8);
-      p_222095_.m_8767_(ParticleTypes.f_235898_, (double)p_222096_.m_123341_() + 0.5D, (double)p_222096_.m_123342_() + 1.15D, (double)p_222096_.m_123343_() + 0.5D, 2, 0.2D, 0.0D, 0.2D, 0.0D);
+      p_222095_.scheduleTick(p_222096_, p_222097_.m_60734_(), 8);
+      p_222095_.m_8767_(ParticleTypes.f_235898_, (double)p_222096_.getX() + 0.5D, (double)p_222096_.getY() + 1.15D, (double)p_222096_.getZ() + 0.5D, 2, 0.2D, 0.0D, 0.2D, 0.0D);
       p_222095_.m_5594_((Player)null, p_222096_, SoundEvents.f_215740_, SoundSource.BLOCKS, 2.0F, 0.6F + p_222098_.m_188501_() * 0.4F);
    }
 
