--- a/net/minecraft/world/level/block/SculkSensorBlock.java
+++ b/net/minecraft/world/level/block/SculkSensorBlock.java
@@ -133,7 +_,7 @@
    private static void m_276869_(Level p_278067_, BlockPos p_277440_, BlockState p_277354_) {
       Block block = p_277354_.m_60734_();
       p_278067_.m_46672_(p_277440_, block);
-      p_278067_.m_46672_(p_277440_.m_7495_(), block);
+      p_278067_.m_46672_(p_277440_.below(), block);
    }
 
    @Nullable
@@ -178,7 +_,7 @@
 
    public static void m_154407_(Level p_154408_, BlockPos p_154409_, BlockState p_154410_) {
       p_154408_.m_7731_(p_154409_, p_154410_.m_61124_(f_154386_, SculkSensorPhase.COOLDOWN).m_61124_(f_154387_, Integer.valueOf(0)), 3);
-      p_154408_.m_186460_(p_154409_, p_154410_.m_60734_(), 10);
+      p_154408_.scheduleTick(p_154409_, p_154410_.m_60734_(), 10);
       m_276869_(p_154408_, p_154409_, p_154410_);
    }
 
@@ -189,19 +_,19 @@
 
    public void m_277033_(@Nullable Entity p_277529_, Level p_277340_, BlockPos p_277386_, BlockState p_277799_, int p_277993_, int p_278003_) {
       p_277340_.m_7731_(p_277386_, p_277799_.m_61124_(f_154386_, SculkSensorPhase.ACTIVE).m_61124_(f_154387_, Integer.valueOf(p_277993_)), 3);
-      p_277340_.m_186460_(p_277386_, p_277799_.m_60734_(), this.m_278716_());
+      p_277340_.scheduleTick(p_277386_, p_277799_.m_60734_(), this.m_278716_());
       m_276869_(p_277340_, p_277386_, p_277799_);
       m_277083_(p_277529_, p_277340_, p_277386_, p_278003_);
       p_277340_.m_142346_(p_277529_, GameEvent.f_223700_, p_277386_);
       if (!p_277799_.m_61143_(f_154388_)) {
-         p_277340_.m_6263_((Player)null, (double)p_277386_.m_123341_() + 0.5D, (double)p_277386_.m_123342_() + 0.5D, (double)p_277386_.m_123343_() + 0.5D, SoundEvents.f_144212_, SoundSource.BLOCKS, 1.0F, p_277340_.f_46441_.m_188501_() * 0.2F + 0.8F);
+         p_277340_.playSound((Player)null, p_277386_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_277386_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_277386_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), SoundEvents.f_144212_, SoundSource.BLOCKS, 1.0F, p_277340_.f_46441_.m_188501_() * 0.2F + 0.8F);
       }
 
    }
 
    public static void m_277083_(@Nullable Entity p_279315_, Level p_277804_, BlockPos p_277458_, int p_277347_) {
       for(Direction direction : Direction.values()) {
-         BlockPos blockpos = p_277458_.m_121945_(direction);
+         BlockPos blockpos = p_277458_.relative(direction);
          BlockState blockstate = p_277804_.m_8055_(blockpos);
          if (blockstate.m_204336_(BlockTags.f_276593_)) {
             p_277804_.m_220407_(VibrationSystem.m_280393_(p_277347_), blockpos, GameEvent.Context.m_223719_(p_279315_, blockstate));
@@ -216,11 +_,11 @@
       if (m_154487_(p_222148_) == SculkSensorPhase.ACTIVE) {
          Direction direction = Direction.m_235672_(p_222151_);
          if (direction != Direction.UP && direction != Direction.DOWN) {
-            double d0 = (double)p_222150_.m_123341_() + 0.5D + (direction.m_122429_() == 0 ? 0.5D - p_222151_.m_188500_() : (double)direction.m_122429_() * 0.6D);
-            double d1 = (double)p_222150_.m_123342_() + 0.25D;
-            double d2 = (double)p_222150_.m_123343_() + 0.5D + (direction.m_122431_() == 0 ? 0.5D - p_222151_.m_188500_() : (double)direction.m_122431_() * 0.6D);
+            me.alphamode.mclong.math.BigDecimal d0 = p_222150_.getBigDecX().add(0.5D + (direction.m_122429_() == 0 ? 0.5D - p_222151_.m_188500_() : (double)direction.m_122429_() * 0.6D));
+            me.alphamode.mclong.math.BigDecimal d1 = p_222150_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.PARTICLES);
+            me.alphamode.mclong.math.BigDecimal d2 = p_222150_.getBigDecZ().add(0.5D + (direction.m_122431_() == 0 ? 0.5D - p_222151_.m_188500_() : (double)direction.m_122431_() * 0.6D));
             double d3 = (double)p_222151_.m_188501_() * 0.04D;
-            p_222149_.m_7106_(DustColorTransitionOptions.f_175752_, d0, d1, d2, 0.0D, d3, 0.0D);
+            p_222149_.addParticle(DustColorTransitionOptions.f_175752_, d0, d1, d2, 0.0D, d3, 0.0D);
          }
       }
    }
