--- a/net/minecraft/world/level/block/SculkShriekerBlock.java
+++ b/net/minecraft/world/level/block/SculkShriekerBlock.java
@@ -33,7 +_,7 @@
    public static final BooleanProperty f_222153_ = BlockStateProperties.f_61362_;
    public static final BooleanProperty f_222154_ = BlockStateProperties.f_222997_;
    protected static final VoxelShape f_222155_ = Block.m_49796_(0.0D, 0.0D, 0.0D, 16.0D, 8.0D, 16.0D);
-   public static final double f_222156_ = f_222155_.m_83297_(Direction.Axis.Y);
+   public static final me.alphamode.mclong.math.BigDecimal f_222156_ = f_222155_.maxB(Direction.Axis.Y);
 
    public SculkShriekerBlock(BlockBehaviour.Properties p_222159_) {
       super(p_222159_);
