--- a/net/minecraft/world/level/block/ShulkerBoxBlock.java
+++ b/net/minecraft/world/level/block/ShulkerBoxBlock.java
@@ -120,7 +_,7 @@
                itemstack.m_41714_(shulkerboxblockentity.m_7770_());
             }
 
-            ItemEntity itementity = new ItemEntity(p_56212_, (double)p_56213_.m_123341_() + 0.5D, (double)p_56213_.m_123342_() + 0.5D, (double)p_56213_.m_123343_() + 0.5D, itemstack);
+            ItemEntity itementity = new ItemEntity(p_56212_, (double)p_56213_.getX() + 0.5D, (double)p_56213_.getY() + 0.5D, (double)p_56213_.getZ() + 0.5D, itemstack);
             itementity.m_32060_();
             p_56212_.m_7967_(itementity);
          } else {
