--- a/net/minecraft/world/level/block/SlabBlock.java
+++ b/net/minecraft/world/level/block/SlabBlock.java
@@ -64,7 +_,7 @@
          FluidState fluidstate = p_56361_.m_43725_().m_6425_(blockpos);
          BlockState blockstate1 = this.m_49966_().m_61124_(f_56353_, SlabType.BOTTOM).m_61124_(f_56354_, Boolean.valueOf(fluidstate.m_76152_() == Fluids.f_76193_));
          Direction direction = p_56361_.m_43719_();
-         return direction != Direction.DOWN && (direction == Direction.UP || !(p_56361_.m_43720_().f_82480_ - (double)blockpos.m_123342_() > 0.5D)) ? blockstate1 : blockstate1.m_61124_(f_56353_, SlabType.TOP);
+         return direction != Direction.DOWN && (direction == Direction.UP || !(p_56361_.m_43720_().f_82480_ - (double)blockpos.getY() > 0.5D)) ? blockstate1 : blockstate1.m_61124_(f_56353_, SlabType.TOP);
       }
    }
 
@@ -73,7 +_,7 @@
       SlabType slabtype = p_56373_.m_61143_(f_56353_);
       if (slabtype != SlabType.DOUBLE && itemstack.m_150930_(this.m_5456_())) {
          if (p_56374_.m_7058_()) {
-            boolean flag = p_56374_.m_43720_().f_82480_ - (double)p_56374_.m_8083_().m_123342_() > 0.5D;
+            boolean flag = p_56374_.m_43720_().f_82480_ - (double)p_56374_.m_8083_().getY() > 0.5D;
             Direction direction = p_56374_.m_43719_();
             if (slabtype == SlabType.BOTTOM) {
                return direction == Direction.UP || flag && direction.m_122434_().m_122479_();
