--- a/net/minecraft/world/level/block/SmokerBlock.java
+++ b/net/minecraft/world/level/block/SmokerBlock.java
@@ -42,14 +_,14 @@
 
    public void m_214162_(BlockState p_222443_, Level p_222444_, BlockPos p_222445_, RandomSource p_222446_) {
       if (p_222443_.m_61143_(f_48684_)) {
-         double d0 = (double)p_222445_.m_123341_() + 0.5D;
-         double d1 = (double)p_222445_.m_123342_();
-         double d2 = (double)p_222445_.m_123343_() + 0.5D;
+         me.alphamode.mclong.math.BigDecimal d0 = p_222445_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB);
+         me.alphamode.mclong.math.BigDecimal d1 = p_222445_.getBigDecY();
+         me.alphamode.mclong.math.BigDecimal d2 = p_222445_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB);
          if (p_222446_.m_188500_() < 0.1D) {
-            p_222444_.m_7785_(d0, d1, d2, SoundEvents.f_12472_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
+            p_222444_.playLocalSound(d0, d1, d2, SoundEvents.f_12472_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
          }
 
-         p_222444_.m_7106_(ParticleTypes.f_123762_, d0, d1 + 1.1D, d2, 0.0D, 0.0D, 0.0D);
+         p_222444_.addParticle(ParticleTypes.f_123762_, d0, d1.add(1.1D), d2, 0.0D, 0.0D, 0.0D);
       }
    }
 }
