--- a/net/minecraft/world/level/block/SnifferEggBlock.java
+++ b/net/minecraft/world/level/block/SnifferEggBlock.java
@@ -79,7 +_,7 @@
       int i = flag ? 12000 : 24000;
       int j = i / 3;
       p_277827_.m_220407_(GameEvent.f_157797_, p_277526_, GameEvent.Context.m_223722_(p_277964_));
-      p_277827_.m_186460_(p_277526_, this, j + p_277827_.f_46441_.m_188503_(300));
+      p_277827_.scheduleTick(p_277526_, this, j + p_277827_.f_46441_.m_188503_(300));
    }
 
    public boolean m_7357_(BlockState p_279414_, BlockGetter p_279243_, BlockPos p_279294_, PathComputationType p_279299_) {
@@ -87,6 +_,6 @@
    }
 
    public static boolean m_277163_(BlockGetter p_277485_, BlockPos p_278065_) {
-      return p_277485_.m_8055_(p_278065_.m_7495_()).m_204336_(BlockTags.f_276630_);
+      return p_277485_.m_8055_(p_278065_.below()).m_204336_(BlockTags.f_276630_);
    }
 }
