--- a/net/minecraft/world/level/block/SnowLayerBlock.java
+++ b/net/minecraft/world/level/block/SnowLayerBlock.java
@@ -70,13 +_,13 @@
    }
 
    public boolean m_7898_(BlockState p_56602_, LevelReader p_56603_, BlockPos p_56604_) {
-      BlockState blockstate = p_56603_.m_8055_(p_56604_.m_7495_());
+      BlockState blockstate = p_56603_.m_8055_(p_56604_.below());
       if (blockstate.m_204336_(BlockTags.f_215833_)) {
          return false;
       } else if (blockstate.m_204336_(BlockTags.f_215834_)) {
          return true;
       } else {
-         return Block.m_49918_(blockstate.m_60812_(p_56603_, p_56604_.m_7495_()), Direction.UP) || blockstate.m_60713_(this) && blockstate.m_61143_(f_56581_) == 8;
+         return Block.m_49918_(blockstate.m_60812_(p_56603_, p_56604_.below()), Direction.UP) || blockstate.m_60713_(this) && blockstate.m_61143_(f_56581_) == 8;
       }
    }
 
