--- a/net/minecraft/world/level/block/SnowyDirtBlock.java
+++ b/net/minecraft/world/level/block/SnowyDirtBlock.java
@@ -24,7 +_,7 @@
    }
 
    public BlockState m_5573_(BlockPlaceContext p_56642_) {
-      BlockState blockstate = p_56642_.m_43725_().m_8055_(p_56642_.m_8083_().m_7494_());
+      BlockState blockstate = p_56642_.m_43725_().m_8055_(p_56642_.m_8083_().above());
       return this.m_49966_().m_61124_(f_56637_, Boolean.valueOf(m_154648_(blockstate)));
    }
 
