--- a/net/minecraft/world/level/block/SoulFireBlock.java
+++ b/net/minecraft/world/level/block/SoulFireBlock.java
@@ -18,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_56655_, LevelReader p_56656_, BlockPos p_56657_) {
-      return m_154650_(p_56656_.m_8055_(p_56657_.m_7495_()));
+      return m_154650_(p_56656_.m_8055_(p_56657_.below()));
    }
 
    public static boolean m_154650_(BlockState p_154651_) {
