--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -51,7 +_,7 @@
          int j = tuple.m_14419_();
 
          for(Direction direction : Direction.values()) {
-            BlockPos blockpos1 = blockpos.m_121945_(direction);
+            BlockPos blockpos1 = blockpos.relative(direction);
             BlockState blockstate = p_56808_.m_8055_(blockpos1);
             FluidState fluidstate = p_56808_.m_6425_(blockpos1);
             Material material = blockstate.m_60767_();
