--- a/net/minecraft/world/level/block/SpongeBlock.java
+++ b/net/minecraft/world/level/block/SpongeBlock.java
@@ -40,7 +_,7 @@
    private boolean m_56807_(Level p_56808_, BlockPos p_56809_) {
       return BlockPos.m_276833_(p_56809_, 6, 65, (p_277519_, p_277492_) -> {
          for(Direction direction : f_276425_) {
-            p_277492_.accept(p_277519_.m_121945_(direction));
+            p_277492_.accept(p_277519_.relative(direction));
          }
 
       }, (p_279054_) -> {
