--- a/net/minecraft/world/level/block/StemBlock.java
+++ b/net/minecraft/world/level/block/StemBlock.java
@@ -52,8 +_,8 @@
                p_222539_.m_7731_(p_222540_, p_222538_, 2);
             } else {
                Direction direction = Direction.Plane.HORIZONTAL.m_235690_(p_222541_);
-               BlockPos blockpos = p_222540_.m_121945_(direction);
-               BlockState blockstate = p_222539_.m_8055_(blockpos.m_7495_());
+               BlockPos blockpos = p_222540_.relative(direction);
+               BlockState blockstate = p_222539_.m_8055_(blockpos.below());
                if (p_222539_.m_8055_(blockpos).m_60795_() && (blockstate.m_60713_(Blocks.f_50093_) || blockstate.m_204336_(BlockTags.f_144274_))) {
                   p_222539_.m_46597_(blockpos, this.f_57015_.m_49966_());
                   p_222539_.m_46597_(p_222540_, this.f_57015_.m_7810_().m_49966_().m_61124_(HorizontalDirectionalBlock.f_54117_, direction));
