--- a/net/minecraft/world/level/block/TallGrassBlock.java
+++ b/net/minecraft/world/level/block/TallGrassBlock.java
@@ -32,7 +_,7 @@
 
    public void m_214148_(ServerLevel p_222578_, RandomSource p_222579_, BlockPos p_222580_, BlockState p_222581_) {
       DoublePlantBlock doubleplantblock = (DoublePlantBlock)(p_222581_.m_60713_(Blocks.f_50035_) ? Blocks.f_50360_ : Blocks.f_50359_);
-      if (doubleplantblock.m_49966_().m_60710_(p_222578_, p_222580_) && p_222578_.m_46859_(p_222580_.m_7494_())) {
+      if (doubleplantblock.m_49966_().m_60710_(p_222578_, p_222580_) && p_222578_.m_46859_(p_222580_.above())) {
          DoublePlantBlock.m_153173_(p_222578_, doubleplantblock.m_49966_(), p_222580_, 2);
       }
 
