--- a/net/minecraft/world/level/block/TallSeagrassBlock.java
+++ b/net/minecraft/world/level/block/TallSeagrassBlock.java
@@ -44,7 +_,7 @@
    public BlockState m_5573_(BlockPlaceContext p_154747_) {
       BlockState blockstate = super.m_5573_(p_154747_);
       if (blockstate != null) {
-         FluidState fluidstate = p_154747_.m_43725_().m_6425_(p_154747_.m_8083_().m_7494_());
+         FluidState fluidstate = p_154747_.m_43725_().m_6425_(p_154747_.m_8083_().above());
          if (fluidstate.m_205070_(FluidTags.f_13131_) && fluidstate.m_76186_() == 8) {
             return blockstate;
          }
@@ -55,7 +_,7 @@
 
    public boolean m_7898_(BlockState p_154768_, LevelReader p_154769_, BlockPos p_154770_) {
       if (p_154768_.m_61143_(f_154740_) == DoubleBlockHalf.UPPER) {
-         BlockState blockstate = p_154769_.m_8055_(p_154770_.m_7495_());
+         BlockState blockstate = p_154769_.m_8055_(p_154770_.below());
          return blockstate.m_60713_(this) && blockstate.m_61143_(f_154740_) == DoubleBlockHalf.LOWER;
       } else {
          FluidState fluidstate = p_154769_.m_6425_(p_154770_);
