--- a/net/minecraft/world/level/block/TargetBlock.java
+++ b/net/minecraft/world/level/block/TargetBlock.java
@@ -72,7 +_,7 @@
 
    private static void m_57385_(LevelAccessor p_57386_, BlockState p_57387_, int p_57388_, BlockPos p_57389_, int p_57390_) {
       p_57386_.m_7731_(p_57389_, p_57387_.m_61124_(f_57376_, Integer.valueOf(p_57388_)), 3);
-      p_57386_.m_186460_(p_57389_, p_57387_.m_60734_(), p_57390_);
+      p_57386_.scheduleTick(p_57389_, p_57387_.m_60734_(), p_57390_);
    }
 
    public void m_213897_(BlockState p_222588_, ServerLevel p_222589_, BlockPos p_222590_, RandomSource p_222591_) {
