--- a/net/minecraft/world/level/block/TorchBlock.java
+++ b/net/minecraft/world/level/block/TorchBlock.java
@@ -33,13 +_,13 @@
    }
 
    public boolean m_7898_(BlockState p_57499_, LevelReader p_57500_, BlockPos p_57501_) {
-      return m_49863_(p_57500_, p_57501_.m_7495_(), Direction.UP);
+      return m_49863_(p_57500_, p_57501_.below(), Direction.UP);
    }
 
    public void m_214162_(BlockState p_222593_, Level p_222594_, BlockPos p_222595_, RandomSource p_222596_) {
-      double d0 = (double)p_222595_.m_123341_() + 0.5D;
-      double d1 = (double)p_222595_.m_123342_() + 0.7D;
-      double d2 = (double)p_222595_.m_123343_() + 0.5D;
+      double d0 = (double)p_222595_.getX() + 0.5D;
+      double d1 = (double)p_222595_.getY() + 0.7D;
+      double d2 = (double)p_222595_.getZ() + 0.5D;
       p_222594_.m_7106_(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
       p_222594_.m_7106_(this.f_57488_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
    }
