--- a/net/minecraft/world/level/block/TrapDoorBlock.java
+++ b/net/minecraft/world/level/block/TrapDoorBlock.java
@@ -121,7 +_,7 @@
       FluidState fluidstate = p_57533_.m_43725_().m_6425_(p_57533_.m_8083_());
       Direction direction = p_57533_.m_43719_();
       if (!p_57533_.m_7058_() && direction.m_122434_().m_122479_()) {
-         blockstate = blockstate.m_61124_(f_54117_, direction).m_61124_(f_57515_, p_57533_.m_43720_().f_82480_ - (double)p_57533_.m_8083_().m_123342_() > 0.5D ? Half.TOP : Half.BOTTOM);
+         blockstate = blockstate.m_61124_(f_54117_, direction).m_61124_(f_57515_, p_57533_.m_43720_().f_82480_ - (double)p_57533_.m_8083_().getY() > 0.5D ? Half.TOP : Half.BOTTOM);
       } else {
          blockstate = blockstate.m_61124_(f_54117_, p_57533_.m_8125_().m_122424_()).m_61124_(f_57515_, direction == Direction.UP ? Half.BOTTOM : Half.TOP);
       }
