--- a/net/minecraft/world/level/block/TurtleEggBlock.java
+++ b/net/minecraft/world/level/block/TurtleEggBlock.java
@@ -96,7 +_,7 @@
                Turtle turtle = EntityType.f_20490_.m_20615_(p_222645_);
                turtle.m_146762_(-24000);
                turtle.m_30219_(p_222646_);
-               turtle.m_7678_((double)p_222646_.m_123341_() + 0.3D + (double)j * 0.2D, (double)p_222646_.m_123342_(), (double)p_222646_.m_123343_() + 0.3D, 0.0F, 0.0F);
+               turtle.m_7678_((double)p_222646_.getX() + 0.3D + (double)j * 0.2D, (double)p_222646_.getY(), (double)p_222646_.getZ() + 0.3D, 0.0F, 0.0F);
                p_222645_.m_7967_(turtle);
             }
          }
@@ -105,7 +_,7 @@
    }
 
    public static boolean m_57762_(BlockGetter p_57763_, BlockPos p_57764_) {
-      return m_57800_(p_57763_, p_57764_.m_7495_());
+      return m_57800_(p_57763_, p_57764_.below());
    }
 
    public static boolean m_57800_(BlockGetter p_57801_, BlockPos p_57802_) {
