--- a/net/minecraft/world/level/block/WallTorchBlock.java
+++ b/net/minecraft/world/level/block/WallTorchBlock.java
@@ -45,7 +_,7 @@
 
    public boolean m_7898_(BlockState p_58133_, LevelReader p_58134_, BlockPos p_58135_) {
       Direction direction = p_58133_.m_61143_(f_58119_);
-      BlockPos blockpos = p_58135_.m_121945_(direction.m_122424_());
+      BlockPos blockpos = p_58135_.relative(direction.m_122424_());
       BlockState blockstate = p_58134_.m_8055_(blockpos);
       return blockstate.m_60783_(p_58134_, blockpos, direction);
    }
@@ -76,9 +_,9 @@
 
    public void m_214162_(BlockState p_222660_, Level p_222661_, BlockPos p_222662_, RandomSource p_222663_) {
       Direction direction = p_222660_.m_61143_(f_58119_);
-      double d0 = (double)p_222662_.m_123341_() + 0.5D;
-      double d1 = (double)p_222662_.m_123342_() + 0.7D;
-      double d2 = (double)p_222662_.m_123343_() + 0.5D;
+      double d0 = (double)p_222662_.getX() + 0.5D;
+      double d1 = (double)p_222662_.getY() + 0.7D;
+      double d2 = (double)p_222662_.getZ() + 0.5D;
       double d3 = 0.22D;
       double d4 = 0.27D;
       Direction direction1 = direction.m_122424_();
