--- a/net/minecraft/world/level/block/WallBlock.java
+++ b/net/minecraft/world/level/block/WallBlock.java
@@ -125,11 +_,11 @@
       LevelReader levelreader = p_57973_.m_43725_();
       BlockPos blockpos = p_57973_.m_8083_();
       FluidState fluidstate = p_57973_.m_43725_().m_6425_(p_57973_.m_8083_());
-      BlockPos blockpos1 = blockpos.m_122012_();
-      BlockPos blockpos2 = blockpos.m_122029_();
-      BlockPos blockpos3 = blockpos.m_122019_();
-      BlockPos blockpos4 = blockpos.m_122024_();
-      BlockPos blockpos5 = blockpos.m_7494_();
+      BlockPos blockpos1 = blockpos.north();
+      BlockPos blockpos2 = blockpos.east();
+      BlockPos blockpos3 = blockpos.south();
+      BlockPos blockpos4 = blockpos.west();
+      BlockPos blockpos5 = blockpos.above();
       BlockState blockstate = levelreader.m_8055_(blockpos1);
       BlockState blockstate1 = levelreader.m_8055_(blockpos2);
       BlockState blockstate2 = levelreader.m_8055_(blockpos3);
@@ -177,7 +_,7 @@
       boolean flag1 = p_57994_ == Direction.EAST ? this.m_58020_(p_57993_, p_57993_.m_60783_(p_57989_, p_57992_, direction), direction) : m_58010_(p_57991_, f_57950_);
       boolean flag2 = p_57994_ == Direction.SOUTH ? this.m_58020_(p_57993_, p_57993_.m_60783_(p_57989_, p_57992_, direction), direction) : m_58010_(p_57991_, f_57952_);
       boolean flag3 = p_57994_ == Direction.WEST ? this.m_58020_(p_57993_, p_57993_.m_60783_(p_57989_, p_57992_, direction), direction) : m_58010_(p_57991_, f_57953_);
-      BlockPos blockpos = p_57990_.m_7494_();
+      BlockPos blockpos = p_57990_.above();
       BlockState blockstate = p_57989_.m_8055_(blockpos);
       return this.m_57979_(p_57989_, p_57991_, blockpos, blockstate, flag, flag1, flag2, flag3);
    }
