--- a/net/minecraft/world/level/block/WallSkullBlock.java
+++ b/net/minecraft/world/level/block/WallSkullBlock.java
@@ -41,7 +_,7 @@
          if (direction.m_122434_().m_122479_()) {
             Direction direction1 = direction.m_122424_();
             blockstate = blockstate.m_61124_(f_58097_, direction1);
-            if (!blockgetter.m_8055_(blockpos.m_121945_(direction)).m_60629_(p_58104_)) {
+            if (!blockgetter.m_8055_(blockpos.relative(direction)).m_60629_(p_58104_)) {
                return blockstate;
             }
          }
