--- a/net/minecraft/world/level/block/WallHangingSignBlock.java
+++ b/net/minecraft/world/level/block/WallHangingSignBlock.java
@@ -93,7 +_,7 @@
    public boolean m_247551_(BlockState p_249472_, LevelReader p_249453_, BlockPos p_251235_) {
       Direction direction = p_249472_.m_61143_(f_244390_).m_122427_();
       Direction direction1 = p_249472_.m_61143_(f_244390_).m_122428_();
-      return this.m_246366_(p_249453_, p_249472_, p_251235_.m_121945_(direction), direction1) || this.m_246366_(p_249453_, p_249472_, p_251235_.m_121945_(direction1), direction);
+      return this.m_246366_(p_249453_, p_249472_, p_251235_.relative(direction), direction1) || this.m_246366_(p_249453_, p_249472_, p_251235_.relative(direction1), direction);
    }
 
    public boolean m_246366_(LevelReader p_249746_, BlockState p_251128_, BlockPos p_250583_, Direction p_250567_) {
