--- a/net/minecraft/world/level/block/WallSignBlock.java
+++ b/net/minecraft/world/level/block/WallSignBlock.java
@@ -42,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_58073_, LevelReader p_58074_, BlockPos p_58075_) {
-      return p_58074_.m_8055_(p_58075_.m_121945_(p_58073_.m_61143_(f_58064_).m_122424_())).m_280296_();
+      return p_58074_.m_8055_(p_58075_.relative(p_58073_.m_61143_(f_58064_).m_122424_())).m_280296_();
    }
 
    @Nullable
