--- a/net/minecraft/world/level/block/WallBannerBlock.java
+++ b/net/minecraft/world/level/block/WallBannerBlock.java
@@ -31,7 +_,7 @@
    }
 
    public boolean m_7898_(BlockState p_57925_, LevelReader p_57926_, BlockPos p_57927_) {
-      return p_57926_.m_8055_(p_57927_.m_121945_(p_57925_.m_61143_(f_57916_).m_122424_())).m_280296_();
+      return p_57926_.m_8055_(p_57927_.relative(p_57925_.m_61143_(f_57916_).m_122424_())).m_280296_();
    }
 
    public BlockState m_7417_(BlockState p_57935_, Direction p_57936_, BlockState p_57937_, LevelAccessor p_57938_, BlockPos p_57939_, BlockPos p_57940_) {
