--- a/net/minecraft/world/level/block/WitherRoseBlock.java
+++ b/net/minecraft/world/level/block/WitherRoseBlock.java
@@ -29,12 +_,12 @@
    public void m_214162_(BlockState p_222687_, Level p_222688_, BlockPos p_222689_, RandomSource p_222690_) {
       VoxelShape voxelshape = this.m_5940_(p_222687_, p_222688_, p_222689_, CollisionContext.m_82749_());
       Vec3 vec3 = voxelshape.m_83215_().m_82399_();
-      double d0 = (double)p_222689_.m_123341_() + vec3.f_82479_;
-      double d1 = (double)p_222689_.m_123343_() + vec3.f_82481_;
+      me.alphamode.mclong.math.BigDecimal d0 = p_222689_.getBigDecX().add(vec3.getX());
+      me.alphamode.mclong.math.BigDecimal d1 = p_222689_.getBigDecZ().add(vec3.getZ());
 
       for(int i = 0; i < 3; ++i) {
          if (p_222690_.m_188499_()) {
-            p_222688_.m_7106_(ParticleTypes.f_123762_, d0 + p_222690_.m_188500_() / 5.0D, (double)p_222689_.m_123342_() + (0.5D - p_222690_.m_188500_()), d1 + p_222690_.m_188500_() / 5.0D, 0.0D, 0.0D, 0.0D);
+            p_222688_.addParticle(ParticleTypes.f_123762_, d0.add(p_222690_.m_188500_() / 5.0D), p_222689_.getBigDecY().add((0.5D - p_222690_.m_188500_())), d1.add(p_222690_.m_188500_() / 5.0D), 0.0D, 0.0D, 0.0D);
          }
       }
 
