--- a/net/minecraft/world/level/block/WitherSkullBlock.java
+++ b/net/minecraft/world/level/block/WitherSkullBlock.java
@@ -45,14 +_,14 @@
       if (!p_58256_.f_46443_) {
          BlockState blockstate = p_58258_.m_58900_();
          boolean flag = blockstate.m_60713_(Blocks.f_50312_) || blockstate.m_60713_(Blocks.f_50313_);
-         if (flag && p_58257_.m_123342_() >= p_58256_.m_141937_() && p_58256_.m_46791_() != Difficulty.PEACEFUL) {
+         if (flag && p_58257_.getY() >= p_58256_.getMinBuildHeight() && p_58256_.m_46791_() != Difficulty.PEACEFUL) {
             BlockPattern.BlockPatternMatch blockpattern$blockpatternmatch = m_58273_().m_61184_(p_58256_, p_58257_);
             if (blockpattern$blockpatternmatch != null) {
                WitherBoss witherboss = EntityType.f_20496_.m_20615_(p_58256_);
                if (witherboss != null) {
                   CarvedPumpkinBlock.m_245585_(p_58256_, blockpattern$blockpatternmatch);
                   BlockPos blockpos = blockpattern$blockpatternmatch.m_61229_(1, 2, 0).m_61176_();
-                  witherboss.m_7678_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.55D, (double)blockpos.m_123343_() + 0.5D, blockpattern$blockpatternmatch.m_61233_().m_122434_() == Direction.Axis.X ? 0.0F : 90.0F, 0.0F);
+                  witherboss.moveTo(blockpos.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), blockpos.getBigDecY().add(0.55D), blockpos.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), blockpattern$blockpatternmatch.m_61233_().m_122434_() == Direction.Axis.X ? 0.0F : 90.0F, 0.0F);
                   witherboss.f_20883_ = blockpattern$blockpatternmatch.m_61233_().m_122434_() == Direction.Axis.X ? 0.0F : 90.0F;
                   witherboss.m_31506_();
 
@@ -70,7 +_,7 @@
    }
 
    public static boolean m_58267_(Level p_58268_, BlockPos p_58269_, ItemStack p_58270_) {
-      if (p_58270_.m_150930_(Items.f_42679_) && p_58269_.m_123342_() >= p_58268_.m_141937_() + 2 && p_58268_.m_46791_() != Difficulty.PEACEFUL && !p_58268_.f_46443_) {
+      if (p_58270_.m_150930_(Items.f_42679_) && p_58269_.getY() >= p_58268_.getMinBuildHeight() + 2 && p_58268_.m_46791_() != Difficulty.PEACEFUL && !p_58268_.f_46443_) {
          return m_58274_().m_61184_(p_58268_, p_58269_) != null;
       } else {
          return false;
