--- a/net/minecraft/world/level/levelgen/structure/structures/BuriedTreasureStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/BuriedTreasureStructure.java
@@ -22,7 +_,7 @@
    }
 
    private static void m_227391_(StructurePiecesBuilder p_227392_, Structure.GenerationContext p_227393_) {
-      BlockPos blockpos = new BlockPos(p_227393_.f_226628_().m_151382_(9), 90, p_227393_.f_226628_().m_151391_(9));
+      BlockPos blockpos = new BlockPos(p_227393_.f_226628_().getBlockX(9), 90, p_227393_.f_226628_().getBlockZ(9));
       p_227392_.m_142679_(new BuriedTreasurePieces.BuriedTreasurePiece(blockpos));
    }
 
