--- a/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidStructure.java
@@ -26,11 +_,11 @@
    public static final Codec<DesertPyramidStructure> f_227415_ = m_226607_(DesertPyramidStructure::new);
 
    public DesertPyramidStructure(Structure.StructureSettings p_227418_) {
-      super(DesertPyramidPiece::new, 21, 21, p_227418_);
+      super((p_226550_, p_226551_, p_226552_) -> new DesertPyramidPiece(p_226550_, p_226551_, p_226552_), 21, 21, p_227418_);
    }
 
    public void m_214110_(WorldGenLevel p_273644_, StructureManager p_272615_, ChunkGenerator p_273655_, RandomSource p_272939_, BoundingBox p_273179_, ChunkPos p_273334_, PiecesContainer p_273575_) {
-      Set<BlockPos> set = SortedArraySet.m_144976_(Vec3i::compareTo);
+         Set<BlockPos> set = SortedArraySet.m_144976_(me.alphamode.mclong.core.Vec3l::compareTo);
 
       for(StructurePiece structurepiece : p_273575_.f_192741_()) {
          if (structurepiece instanceof DesertPyramidPiece desertpyramidpiece) {
@@ -48,7 +_,7 @@
          if (i > 0) {
             --i;
             m_278858_(p_273179_, p_273644_, blockpos);
-         } else if (p_273179_.m_71051_(blockpos)) {
+         } else if (p_273179_.isInside(blockpos)) {
             p_273644_.m_7731_(blockpos, Blocks.f_49992_.m_49966_(), 2);
          }
       }
@@ -56,7 +_,7 @@
    }
 
    private static void m_278858_(BoundingBox p_279472_, WorldGenLevel p_279193_, BlockPos p_279136_) {
-      if (p_279472_.m_71051_(p_279136_)) {
+      if (p_279472_.isInside(p_279136_)) {
          p_279193_.m_7731_(p_279136_, Blocks.f_271439_.m_49966_(), 2);
          p_279193_.m_141902_(p_279136_, BlockEntityType.f_271323_).ifPresent((p_277328_) -> {
             p_277328_.m_277049_(BuiltInLootTables.f_276661_, p_279136_.m_121878_());
