--- a/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
@@ -22,7 +_,7 @@
    public static final int f_227396_ = 21;
    private final boolean[] f_227397_ = new boolean[4];
 
-   public DesertPyramidPiece(RandomSource p_227399_, int p_227400_, int p_227401_) {
+   public DesertPyramidPiece(RandomSource p_227399_, long p_227400_, long p_227401_) {
       super(StructurePieceType.f_210106_, p_227400_, 64, p_227401_, 21, 15, 21, m_226760_(p_227399_));
    }
 
