--- a/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidStructure.java
@@ -32,7 +_,7 @@
 
    public void m_214110_(WorldGenLevel p_273644_, StructureManager p_272615_, ChunkGenerator p_273655_, RandomSource p_272939_, BoundingBox p_273179_, ChunkPos p_273334_, PiecesContainer p_273575_) {
       if (p_273644_.m_246046_().m_245372_(FeatureFlags.f_244168_)) {
-         Set<BlockPos> set = SortedArraySet.m_144976_(Vec3i::compareTo);
+         Set<BlockPos> set = SortedArraySet.m_144976_(me.alphamode.mclong.core.Vec3l::compareTo);
 
          for(StructurePiece structurepiece : p_273575_.f_192741_()) {
             if (structurepiece instanceof DesertPyramidPiece) {
