--- a/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/DesertPyramidPiece.java
@@ -27,6 +_,10 @@
    private final List<BlockPos> f_271477_ = new ArrayList<>();
    private BlockPos f_278492_ = BlockPos.f_121853_;
 
+   public DesertPyramidPiece(RandomSource p_227399_, long p_227400_, long p_227401_) {
+      super(StructurePieceType.f_210106_, p_227400_, 64, p_227401_, 21, 15, 21, m_226760_(p_227399_));
+   }
+
    public DesertPyramidPiece(RandomSource p_227399_, int p_227400_, int p_227401_) {
       super(StructurePieceType.f_210106_, p_227400_, 64, p_227401_, 21, 15, 21, m_226760_(p_227399_));
    }
@@ -49,11 +_,11 @@
 
    public void m_213694_(WorldGenLevel p_227405_, StructureManager p_227406_, ChunkGenerator p_227407_, RandomSource p_227408_, BoundingBox p_227409_, ChunkPos p_227410_, BlockPos p_227411_) {
       if (this.m_192467_(p_227405_, -p_227408_.m_188503_(3))) {
-         this.m_73441_(p_227405_, p_227409_, 0, -4, 0, this.f_72787_ - 1, 0, this.f_72789_ - 1, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 0, -4, 0, this.f_72787_ - 1, 0, this.f_72789_ - 1, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
 
          for(int i = 1; i <= 9; ++i) {
-            this.m_73441_(p_227405_, p_227409_, i, i, i, this.f_72787_ - 1 - i, i, this.f_72789_ - 1 - i, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-            this.m_73441_(p_227405_, p_227409_, i + 1, i, i + 1, this.f_72787_ - 2 - i, i, this.f_72789_ - 2 - i, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+            this.generateBox(p_227405_, p_227409_, i, i, i, this.f_72787_ - 1 - i, i, this.f_72789_ - 1 - i, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+            this.generateBox(p_227405_, p_227409_, i + 1, i, i + 1, this.f_72787_ - 2 - i, i, this.f_72789_ - 2 - i, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
          }
 
          for(int k1 = 0; k1 < this.f_72787_; ++k1) {
@@ -67,166 +_,166 @@
          BlockState blockstate2 = Blocks.f_50263_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.SOUTH);
          BlockState blockstate3 = Blocks.f_50263_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.EAST);
          BlockState blockstate = Blocks.f_50263_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.WEST);
-         this.m_73441_(p_227405_, p_227409_, 0, 0, 0, 4, 9, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 1, 10, 1, 3, 10, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73434_(p_227405_, blockstate1, 2, 10, 0, p_227409_);
-         this.m_73434_(p_227405_, blockstate2, 2, 10, 4, p_227409_);
-         this.m_73434_(p_227405_, blockstate3, 0, 10, 2, p_227409_);
-         this.m_73434_(p_227405_, blockstate, 4, 10, 2, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 5, 0, 0, this.f_72787_ - 1, 9, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 4, 10, 1, this.f_72787_ - 2, 10, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73434_(p_227405_, blockstate1, this.f_72787_ - 3, 10, 0, p_227409_);
-         this.m_73434_(p_227405_, blockstate2, this.f_72787_ - 3, 10, 4, p_227409_);
-         this.m_73434_(p_227405_, blockstate3, this.f_72787_ - 5, 10, 2, p_227409_);
-         this.m_73434_(p_227405_, blockstate, this.f_72787_ - 1, 10, 2, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 8, 0, 0, 12, 4, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 9, 1, 0, 11, 3, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 9, 1, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 9, 2, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 9, 3, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 10, 3, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 11, 3, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 11, 2, 1, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 11, 1, 1, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 4, 1, 1, 8, 3, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 4, 1, 2, 8, 2, 2, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 12, 1, 1, 16, 3, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 12, 1, 2, 16, 2, 2, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 5, 4, 5, this.f_72787_ - 6, 4, this.f_72789_ - 6, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 9, 4, 9, 11, 4, 11, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 8, 1, 8, 8, 3, 8, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 12, 1, 8, 12, 3, 8, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 8, 1, 12, 8, 3, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 12, 1, 12, 12, 3, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 1, 1, 5, 4, 4, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 5, 1, 5, this.f_72787_ - 2, 4, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 6, 7, 9, 6, 7, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 7, 7, 9, this.f_72787_ - 7, 7, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 5, 5, 9, 5, 7, 11, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 6, 5, 9, this.f_72787_ - 6, 7, 11, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 5, 5, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 5, 6, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 6, 6, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 6, 5, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 6, 6, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 7, 6, 10, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 2, 4, 4, 2, 6, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 3, 4, 4, this.f_72787_ - 3, 6, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73434_(p_227405_, blockstate1, 2, 4, 5, p_227409_);
-         this.m_73434_(p_227405_, blockstate1, 2, 3, 4, p_227409_);
-         this.m_73434_(p_227405_, blockstate1, this.f_72787_ - 3, 4, 5, p_227409_);
-         this.m_73434_(p_227405_, blockstate1, this.f_72787_ - 3, 3, 4, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 1, 1, 3, 2, 2, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 3, 1, 3, this.f_72787_ - 2, 2, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50062_.m_49966_(), 1, 1, 2, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50062_.m_49966_(), this.f_72787_ - 2, 1, 2, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50406_.m_49966_(), 1, 2, 2, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50406_.m_49966_(), this.f_72787_ - 2, 2, 2, p_227409_);
-         this.m_73434_(p_227405_, blockstate, 2, 1, 2, p_227409_);
-         this.m_73434_(p_227405_, blockstate3, this.f_72787_ - 3, 1, 2, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 4, 3, 5, 4, 3, 17, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 5, 3, 5, this.f_72787_ - 5, 3, 17, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 3, 1, 5, 4, 2, 16, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, this.f_72787_ - 6, 1, 5, this.f_72787_ - 5, 2, 16, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 0, 0, 0, 4, 9, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 1, 10, 1, 3, 10, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.placeBlock(p_227405_, blockstate1, 2, 10, 0, p_227409_);
+         this.placeBlock(p_227405_, blockstate2, 2, 10, 4, p_227409_);
+         this.placeBlock(p_227405_, blockstate3, 0, 10, 2, p_227409_);
+         this.placeBlock(p_227405_, blockstate, 4, 10, 2, p_227409_);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 5, 0, 0, this.f_72787_ - 1, 9, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 4, 10, 1, this.f_72787_ - 2, 10, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.placeBlock(p_227405_, blockstate1, this.f_72787_ - 3, 10, 0, p_227409_);
+         this.placeBlock(p_227405_, blockstate2, this.f_72787_ - 3, 10, 4, p_227409_);
+         this.placeBlock(p_227405_, blockstate3, this.f_72787_ - 5, 10, 2, p_227409_);
+         this.placeBlock(p_227405_, blockstate, this.f_72787_ - 1, 10, 2, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 8, 0, 0, 12, 4, 4, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 9, 1, 0, 11, 3, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 9, 1, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 9, 2, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 9, 3, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 10, 3, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 11, 3, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 11, 2, 1, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 11, 1, 1, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 4, 1, 1, 8, 3, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 4, 1, 2, 8, 2, 2, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 12, 1, 1, 16, 3, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 12, 1, 2, 16, 2, 2, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 5, 4, 5, this.f_72787_ - 6, 4, this.f_72789_ - 6, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 9, 4, 9, 11, 4, 11, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 8, 1, 8, 8, 3, 8, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 12, 1, 8, 12, 3, 8, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 8, 1, 12, 8, 3, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 12, 1, 12, 12, 3, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 1, 1, 5, 4, 4, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 5, 1, 5, this.f_72787_ - 2, 4, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 6, 7, 9, 6, 7, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 7, 7, 9, this.f_72787_ - 7, 7, 11, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 5, 5, 9, 5, 7, 11, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 6, 5, 9, this.f_72787_ - 6, 7, 11, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 5, 5, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 5, 6, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 6, 6, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 6, 5, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 6, 6, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), this.f_72787_ - 7, 6, 10, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 2, 4, 4, 2, 6, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 3, 4, 4, this.f_72787_ - 3, 6, 4, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.placeBlock(p_227405_, blockstate1, 2, 4, 5, p_227409_);
+         this.placeBlock(p_227405_, blockstate1, 2, 3, 4, p_227409_);
+         this.placeBlock(p_227405_, blockstate1, this.f_72787_ - 3, 4, 5, p_227409_);
+         this.placeBlock(p_227405_, blockstate1, this.f_72787_ - 3, 3, 4, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 1, 1, 3, 2, 2, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 3, 1, 3, this.f_72787_ - 2, 2, 3, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50062_.m_49966_(), 1, 1, 2, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50062_.m_49966_(), this.f_72787_ - 2, 1, 2, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50406_.m_49966_(), 1, 2, 2, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50406_.m_49966_(), this.f_72787_ - 2, 2, 2, p_227409_);
+         this.placeBlock(p_227405_, blockstate, 2, 1, 2, p_227409_);
+         this.placeBlock(p_227405_, blockstate3, this.f_72787_ - 3, 1, 2, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 4, 3, 5, 4, 3, 17, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 5, 3, 5, this.f_72787_ - 5, 3, 17, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 3, 1, 5, 4, 2, 16, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, this.f_72787_ - 6, 1, 5, this.f_72787_ - 5, 2, 16, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
 
          for(int l = 5; l <= 17; l += 2) {
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 4, 1, l, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 4, 2, l, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), this.f_72787_ - 5, 1, l, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), this.f_72787_ - 5, 2, l, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 4, 1, l, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 4, 2, l, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), this.f_72787_ - 5, 1, l, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), this.f_72787_ - 5, 2, l, p_227409_);
          }
 
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 7, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 8, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 9, 0, 9, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 11, 0, 9, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 8, 0, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 12, 0, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 7, 0, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 13, 0, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 9, 0, 11, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 11, 0, 11, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 12, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 13, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50298_.m_49966_(), 10, 0, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 7, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 8, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 9, 0, 9, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 11, 0, 9, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 8, 0, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 12, 0, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 7, 0, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 13, 0, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 9, 0, 11, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 11, 0, 11, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 12, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 10, 0, 13, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50298_.m_49966_(), 10, 0, 10, p_227409_);
 
          for(int l1 = 0; l1 <= this.f_72787_ - 1; l1 += this.f_72787_ - 1) {
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 2, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 2, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 2, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 3, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 3, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 3, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 4, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), l1, 4, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 4, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 5, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 5, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 5, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 6, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), l1, 6, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 6, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 3, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 1, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 2, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 2, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 2, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 2, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 3, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 3, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 3, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 4, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), l1, 4, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 4, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 5, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 5, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 5, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 6, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), l1, 6, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 6, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), l1, 7, 3, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 1, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 2, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), l1, 8, 3, p_227409_);
          }
 
          for(int i2 = 2; i2 <= this.f_72787_ - 3; i2 += this.f_72787_ - 3 - 2) {
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 2, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2, 2, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 2, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 3, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2, 3, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 3, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 4, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), i2, 4, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 4, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 5, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2, 5, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 5, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 6, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), i2, 6, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 6, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 7, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2, 7, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 7, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 8, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2, 8, 0, p_227409_);
-            this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 8, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 2, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2, 2, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 2, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 3, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2, 3, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 3, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 4, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), i2, 4, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 4, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 5, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2, 5, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 5, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 6, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), i2, 6, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 6, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 - 1, 7, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2, 7, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), i2 + 1, 7, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 - 1, 8, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2, 8, 0, p_227409_);
+            this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), i2 + 1, 8, 0, p_227409_);
          }
 
-         this.m_73441_(p_227405_, p_227409_, 8, 4, 0, 12, 6, 0, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 8, 6, 0, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 12, 6, 0, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 9, 5, 0, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 10, 5, 0, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50288_.m_49966_(), 11, 5, 0, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 8, -14, 8, 12, -11, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 8, -10, 8, 12, -10, 12, Blocks.f_50063_.m_49966_(), Blocks.f_50063_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 8, -9, 8, 12, -9, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 8, -8, 8, 12, -1, 12, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
-         this.m_73441_(p_227405_, p_227409_, 9, -11, 9, 11, -1, 11, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50165_.m_49966_(), 10, -11, 10, p_227409_);
-         this.m_73441_(p_227405_, p_227409_, 9, -13, 9, 11, -13, 11, Blocks.f_50077_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 8, -11, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 8, -10, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 7, -10, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 7, -11, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 12, -11, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 12, -10, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 13, -10, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 13, -11, 10, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 10, -11, 8, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 10, -10, 8, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 10, -10, 7, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 10, -11, 7, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 10, -11, 12, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50016_.m_49966_(), 10, -10, 12, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50063_.m_49966_(), 10, -10, 13, p_227409_);
-         this.m_73434_(p_227405_, Blocks.f_50064_.m_49966_(), 10, -11, 13, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 8, 4, 0, 12, 6, 0, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 8, 6, 0, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 12, 6, 0, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 9, 5, 0, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 10, 5, 0, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50288_.m_49966_(), 11, 5, 0, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 8, -14, 8, 12, -11, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 8, -10, 8, 12, -10, 12, Blocks.f_50063_.m_49966_(), Blocks.f_50063_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 8, -9, 8, 12, -9, 12, Blocks.f_50064_.m_49966_(), Blocks.f_50064_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 8, -8, 8, 12, -1, 12, Blocks.f_50062_.m_49966_(), Blocks.f_50062_.m_49966_(), false);
+         this.generateBox(p_227405_, p_227409_, 9, -11, 9, 11, -1, 11, Blocks.f_50016_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50165_.m_49966_(), 10, -11, 10, p_227409_);
+         this.generateBox(p_227405_, p_227409_, 9, -13, 9, 11, -13, 11, Blocks.f_50077_.m_49966_(), Blocks.f_50016_.m_49966_(), false);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 8, -11, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 8, -10, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 7, -10, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 7, -11, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 12, -11, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 12, -10, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 13, -10, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 13, -11, 10, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 10, -11, 8, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 10, -10, 8, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 10, -10, 7, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 10, -11, 7, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 10, -11, 12, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50016_.m_49966_(), 10, -10, 12, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50063_.m_49966_(), 10, -10, 13, p_227409_);
+         this.placeBlock(p_227405_, Blocks.f_50064_.m_49966_(), 10, -11, 13, p_227409_);
 
          for(Direction direction : Direction.Plane.HORIZONTAL) {
             if (!this.f_227397_[direction.m_122416_()]) {
@@ -247,118 +_,119 @@
    }
 
    private void m_271901_(BlockPos p_272997_, WorldGenLevel p_272699_, BoundingBox p_273559_) {
-      int i = p_272997_.m_123341_();
-      int j = p_272997_.m_123342_();
-      int k = p_272997_.m_123343_();
+      long i = p_272997_.getX();
+      long j = p_272997_.getY();
+      long k = p_272997_.getZ();
       BlockState blockstate = Blocks.f_50263_.m_49966_();
-      this.m_73434_(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 13, -1, 17, p_273559_);
-      this.m_73434_(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 14, -2, 17, p_273559_);
-      this.m_73434_(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 15, -3, 17, p_273559_);
+      this.placeBlock(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 12, 0, 17, p_273559_);
+      this.placeBlock(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 13, -1, 17, p_273559_);
+      this.placeBlock(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 14, -2, 17, p_273559_);
+      this.placeBlock(p_272699_, blockstate.m_60717_(Rotation.COUNTERCLOCKWISE_90), 15, -3, 17, p_273559_);
       BlockState blockstate1 = Blocks.f_49992_.m_49966_();
       BlockState blockstate2 = Blocks.f_50062_.m_49966_();
       boolean flag = p_272699_.m_213780_().m_188499_();
-      this.m_73434_(p_272699_, blockstate1, i - 4, j + 4, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i - 3, j + 4, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i - 2, j + 4, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i - 1, j + 4, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i, j + 4, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i - 2, j + 3, k + 4, p_273559_);
-      this.m_73434_(p_272699_, flag ? blockstate1 : blockstate2, i - 1, j + 3, k + 4, p_273559_);
-      this.m_73434_(p_272699_, !flag ? blockstate1 : blockstate2, i, j + 3, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i - 1, j + 2, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate2, i, j + 2, k + 4, p_273559_);
-      this.m_73434_(p_272699_, blockstate1, i, j + 1, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 4, j + 4, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 3, j + 4, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 2, j + 4, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 1, j + 4, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i, j + 4, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 2, j + 3, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 1, j + 3, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i, j + 3, k + 4, p_273559_);
+      this.placeBlock(p_272699_, blockstate1, i - 1, j + 2, k + 4, p_273559_);
+      this.placeBlock(p_272699_, flag ? blockstate1 : blockstate2, i, j + 2, k + 4, p_273559_);
+      this.placeBlock(p_272699_, flag ? blockstate1 : blockstate2, i, j + 1, k + 4, p_273559_);
    }
 
    private void m_272212_(BlockPos p_272733_, WorldGenLevel p_273390_, BoundingBox p_273517_) {
-      int i = p_272733_.m_123341_();
-      int j = p_272733_.m_123342_();
-      int k = p_272733_.m_123343_();
+      long i = p_272733_.getX();
+      long j = p_272733_.getY();
+      long k = p_272733_.getZ();
       BlockState blockstate = Blocks.f_50064_.m_49966_();
       BlockState blockstate1 = Blocks.f_50063_.m_49966_();
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 1, k - 3, i - 3, j + 1, k + 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i + 3, j + 1, k - 3, i + 3, j + 1, k + 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 1, k - 3, i + 3, j + 1, k - 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 1, k + 3, i + 3, j + 1, k + 3, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 2, k - 3, i - 3, j + 2, k + 2, blockstate1, blockstate1, true);
-      this.m_73441_(p_273390_, p_273517_, i + 3, j + 2, k - 3, i + 3, j + 2, k + 2, blockstate1, blockstate1, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 2, k - 3, i + 3, j + 2, k - 2, blockstate1, blockstate1, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, j + 2, k + 3, i + 3, j + 2, k + 3, blockstate1, blockstate1, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, -1, k - 3, i - 3, -1, k + 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i + 3, -1, k - 3, i + 3, -1, k + 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, -1, k - 3, i + 3, -1, k - 2, blockstate, blockstate, true);
-      this.m_73441_(p_273390_, p_273517_, i - 3, -1, k + 3, i + 3, -1, k + 3, blockstate, blockstate, true);
-      this.m_278669_(i - 2, j + 1, k - 2, i + 2, j + 3, k + 2);
-      this.m_271833_(p_273390_, p_273517_, i - 2, j + 4, k - 2, i + 2, k + 2);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 1, k - 3, i - 3, j + 1, k + 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i + 3, j + 1, k - 3, i + 3, j + 1, k + 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 1, k - 3, i + 3, j + 1, k - 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 1, k + 3, i + 3, j + 1, k + 3, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 2, k - 3, i - 3, j + 2, k + 2, blockstate1, blockstate1, true);
+      this.generateBox(p_273390_, p_273517_, i + 3, j + 2, k - 3, i + 3, j + 2, k + 2, blockstate1, blockstate1, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 2, k - 3, i + 3, j + 2, k - 2, blockstate1, blockstate1, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, j + 2, k + 3, i + 3, j + 2, k + 3, blockstate1, blockstate1, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, -1, k - 3, i - 3, -1, k + 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i + 3, -1, k - 3, i + 3, -1, k + 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, -1, k - 3, i + 3, -1, k - 2, blockstate, blockstate, true);
+      this.generateBox(p_273390_, p_273517_, i - 3, -1, k + 3, i + 3, -1, k + 3, blockstate, blockstate, true);
+      this.placeSandBox(p_273517_, i - 2, j + 1, k - 2, i + 2, j + 3, k + 2);
+      this.placeCollapsedRoof(p_273390_, p_273517_, i - 2, j + 4, k - 2, i + 2, k + 2);
       BlockState blockstate2 = Blocks.f_50288_.m_49966_();
       BlockState blockstate3 = Blocks.f_50298_.m_49966_();
-      this.m_73434_(p_273390_, blockstate3, i, j, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i + 1, j, k - 1, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i + 1, j, k + 1, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i - 1, j, k - 1, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i - 1, j, k + 1, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i + 2, j, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i - 2, j, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i, j, k + 2, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i, j, k - 2, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i + 3, j, k, p_273517_);
-      this.m_278861_(i + 3, j + 1, k);
-      this.m_278861_(i + 3, j + 2, k);
-      this.m_73434_(p_273390_, blockstate, i + 4, j + 1, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate1, i + 4, j + 2, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i - 3, j, k, p_273517_);
-      this.m_278861_(i - 3, j + 1, k);
-      this.m_278861_(i - 3, j + 2, k);
-      this.m_73434_(p_273390_, blockstate, i - 4, j + 1, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate1, i - 4, j + 2, k, p_273517_);
-      this.m_73434_(p_273390_, blockstate2, i, j, k + 3, p_273517_);
-      this.m_278861_(i, j + 1, k + 3);
-      this.m_278861_(i, j + 2, k + 3);
-      this.m_73434_(p_273390_, blockstate2, i, j, k - 3, p_273517_);
-      this.m_278861_(i, j + 1, k - 3);
-      this.m_278861_(i, j + 2, k - 3);
-      this.m_73434_(p_273390_, blockstate, i, j + 1, k - 4, p_273517_);
-      this.m_73434_(p_273390_, blockstate1, i, -2, k - 4, p_273517_);
+      this.placeBlock(p_273390_, blockstate3, i, j, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i + 1, j, k - 1, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i + 1, j, k + 1, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i - 1, j, k - 1, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i - 1, j, k + 1, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i + 2, j, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i - 2, j, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i, j, k + 2, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i, j, k - 2, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i + 3, j, k, p_273517_);
+      this.placeSand(i + 3, j + 1, k, p_273517_);
+      this.placeSand(i + 3, j + 2, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate, i + 4, j + 1, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate1, i + 4, j + 2, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i - 3, j, k, p_273517_);
+      this.placeSand(i - 3, j + 1, k, p_273517_);
+      this.placeSand(i - 3, j + 2, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate, i - 4, j + 1, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate1, i - 4, j + 2, k, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i, j, k + 3, p_273517_);
+      this.placeSand(i, j + 1, k + 3, p_273517_);
+      this.placeSand(i, j + 2, k + 3, p_273517_);
+      this.placeBlock(p_273390_, blockstate2, i, j, k - 3, p_273517_);
+      this.placeSand(i, j + 1, k - 3, p_273517_);
+      this.placeSand(i, j + 2, k - 3, p_273517_);
+      this.placeBlock(p_273390_, blockstate, i, j + 1, k - 4, p_273517_);
+      this.placeBlock(p_273390_, blockstate1, i, -2, k - 4, p_273517_);
    }
 
-   private void m_278861_(int p_279401_, int p_279451_, int p_279265_) {
-      BlockPos blockpos = this.m_163582_(p_279401_, p_279451_, p_279265_);
+   private void placeSand(long p_273469_, long p_272983_, long p_273325_, BoundingBox p_272909_) {
+      BlockPos blockpos = this.getWorldPos(p_273469_, p_272983_, p_273325_);
       this.f_271477_.add(blockpos);
    }
 
-   private void m_278669_(int p_279483_, int p_279321_, int p_279271_, int p_279471_, int p_279229_, int p_279111_) {
-      for(int i = p_279321_; i <= p_279229_; ++i) {
-         for(int j = p_279483_; j <= p_279471_; ++j) {
-            for(int k = p_279271_; k <= p_279111_; ++k) {
-               this.m_278861_(j, i, k);
+   private void placeSandBox(BoundingBox p_272934_, long p_273391_, long p_273481_, long p_273347_, long p_273463_, long p_272814_, long p_273011_) {
+      for(long i = p_273481_; i <= p_272814_; ++i) {
+         for(long j = p_273391_; j <= p_273463_; ++j) {
+            for(long k = p_273347_; k <= p_273011_; ++k) {
+               this.placeSand(j, i, k, p_272934_);
             }
          }
       }
 
    }
 
-   private void m_271881_(WorldGenLevel p_272965_, int p_272618_, int p_273415_, int p_273110_, BoundingBox p_272645_) {
+   private void placeCollapsedRoofPiece(WorldGenLevel p_272965_, long p_272618_, long p_273415_, long p_273110_, BoundingBox p_272645_) {
       if (p_272965_.m_213780_().m_188501_() < 0.33F) {
          BlockState blockstate = Blocks.f_50062_.m_49966_();
-         this.m_73434_(p_272965_, blockstate, p_272618_, p_273415_, p_273110_, p_272645_);
+         this.placeBlock(p_272965_, blockstate, p_272618_, p_273415_, p_273110_, p_272645_);
       } else {
          BlockState blockstate1 = Blocks.f_49992_.m_49966_();
-         this.m_73434_(p_272965_, blockstate1, p_272618_, p_273415_, p_273110_, p_272645_);
+         this.placeBlock(p_272965_, blockstate1, p_272618_, p_273415_, p_273110_, p_272645_);
       }
 
    }
 
-   private void m_271833_(WorldGenLevel p_273438_, BoundingBox p_273058_, int p_272638_, int p_272826_, int p_273026_, int p_272750_, int p_272639_) {
-      for(int i = p_272638_; i <= p_272750_; ++i) {
-         for(int j = p_273026_; j <= p_272639_; ++j) {
-            this.m_271881_(p_273438_, i, p_272826_, j, p_273058_);
+   private void placeCollapsedRoof(WorldGenLevel p_273438_, BoundingBox p_273058_, long p_272638_, long p_272826_, long p_273026_, long p_272750_, long p_272639_) {
+      for(long i = p_272638_; i <= p_272750_; ++i) {
+         for(long j = p_273026_; j <= p_272639_; ++j) {
+            this.placeCollapsedRoofPiece(p_273438_, i, p_272826_, j, p_273058_);
          }
       }
 
-      RandomSource randomsource = RandomSource.m_216335_(p_273438_.m_7328_()).m_188582_().m_224542_(this.m_163582_(p_272638_, p_272826_, p_273026_));
-      int l = randomsource.m_216332_(p_272638_, p_272750_);
-      int k = randomsource.m_216332_(p_273026_, p_272639_);
-      this.f_278492_ = new BlockPos(this.m_73392_(l, k), this.m_73544_(p_272826_), this.m_73525_(l, k));
+      RandomSource randomsource = RandomSource.m_216335_(p_273438_.m_7328_()).m_188582_().m_224542_(this.getWorldPos(p_272638_, p_272826_, p_273026_));
+      long l = randomsource.nextLongBetweenInclusive(p_272638_, p_272750_);
+      long k = randomsource.nextLongBetweenInclusive(p_273026_, p_272639_);
+      this.f_278492_ = new BlockPos(this.getWorldX(l, k), this.getWorldY(p_272826_), this.getWorldZ(l, k));
    }
 
    public List<BlockPos> m_271942_() {
