--- a/net/minecraft/world/level/levelgen/structure/structures/EndCityStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/EndCityStructure.java
@@ -21,7 +_,7 @@
    public Optional<Structure.GenerationStub> m_214086_(Structure.GenerationContext p_227528_) {
       Rotation rotation = Rotation.m_221990_(p_227528_.f_226626_());
       BlockPos blockpos = this.m_226582_(p_227528_, rotation);
-      return blockpos.m_123342_() < 60 ? Optional.empty() : Optional.of(new Structure.GenerationStub(blockpos, (p_227538_) -> {
+      return blockpos.getY() < 60 ? Optional.empty() : Optional.of(new Structure.GenerationStub(blockpos, (p_227538_) -> {
          this.m_227529_(p_227538_, blockpos, rotation, p_227528_);
       }));
    }
