--- a/net/minecraft/world/level/levelgen/structure/structures/IglooStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/IglooStructure.java
@@ -27,7 +_,7 @@
    private void m_227599_(StructurePiecesBuilder p_227600_, Structure.GenerationContext p_227601_) {
       ChunkPos chunkpos = p_227601_.f_226628_();
       WorldgenRandom worldgenrandom = p_227601_.f_226626_();
-      BlockPos blockpos = new BlockPos(chunkpos.m_45604_(), 90, chunkpos.m_45605_());
+      BlockPos blockpos = new BlockPos(chunkpos.getMinBlockX(), 90, chunkpos.getMinBlockZ());
       Rotation rotation = Rotation.m_221990_(worldgenrandom);
       IglooPieces.m_227548_(p_227601_.f_226625_(), blockpos, rotation, p_227600_, worldgenrandom);
    }
