--- a/net/minecraft/world/level/levelgen/structure/structures/JungleTempleStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/JungleTempleStructure.java
@@ -9,7 +_,7 @@
    public static final Codec<JungleTempleStructure> f_227691_ = m_226607_(JungleTempleStructure::new);
 
    public JungleTempleStructure(Structure.StructureSettings p_227694_) {
-      super(JungleTemplePiece::new, 12, 15, p_227694_);
+      super((p_226550_, p_226551_, p_226552_) -> new JungleTemplePiece(p_226550_, p_226551_, p_226552_), 12, 15, p_227694_);
    }
 
    public StructureType<?> m_213658_() {
