--- a/net/minecraft/world/level/levelgen/structure/structures/JungleTemplePiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/JungleTemplePiece.java
@@ -36,7 +_,7 @@
    private boolean f_227664_;
    private static final JungleTemplePiece.MossStoneSelector f_227665_ = new JungleTemplePiece.MossStoneSelector();
 
-   public JungleTemplePiece(RandomSource p_227668_, int p_227669_, int p_227670_) {
+   public JungleTemplePiece(RandomSource p_227668_, long p_227669_, long p_227670_) {
       super(StructurePieceType.f_210101_, p_227669_, 64, p_227670_, 12, 10, 15, m_226760_(p_227668_));
    }
 
