--- a/net/minecraft/world/level/levelgen/structure/structures/MineshaftStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/MineshaftStructure.java
@@ -34,29 +_,29 @@
    public Optional<Structure.GenerationStub> m_214086_(Structure.GenerationContext p_227964_) {
       p_227964_.f_226626_().m_188500_();
       ChunkPos chunkpos = p_227964_.f_226628_();
-      BlockPos blockpos = new BlockPos(chunkpos.m_151390_(), 50, chunkpos.m_45605_());
+      BlockPos blockpos = new BlockPos(chunkpos.getMiddleBlockX(), 50, chunkpos.getMinBlockZ());
       StructurePiecesBuilder structurepiecesbuilder = new StructurePiecesBuilder();
-      int i = this.m_227965_(structurepiecesbuilder, p_227964_);
-      return Optional.of(new Structure.GenerationStub(blockpos.m_7918_(0, i, 0), Either.right(structurepiecesbuilder)));
+      long i = this.generatePiecesAndAdjust(structurepiecesbuilder, p_227964_);
+      return Optional.of(new Structure.GenerationStub(blockpos.offset(0, i, 0), Either.right(structurepiecesbuilder)));
    }
 
-   private int m_227965_(StructurePiecesBuilder p_227966_, Structure.GenerationContext p_227967_) {
+   private long generatePiecesAndAdjust(StructurePiecesBuilder p_227966_, Structure.GenerationContext p_227967_) {
       ChunkPos chunkpos = p_227967_.f_226628_();
       WorldgenRandom worldgenrandom = p_227967_.f_226626_();
       ChunkGenerator chunkgenerator = p_227967_.f_226622_();
-      MineshaftPieces.MineShaftRoom mineshaftpieces$mineshaftroom = new MineshaftPieces.MineShaftRoom(0, worldgenrandom, chunkpos.m_151382_(2), chunkpos.m_151391_(2), this.f_227958_);
+      MineshaftPieces.MineShaftRoom mineshaftpieces$mineshaftroom = new MineshaftPieces.MineShaftRoom(0, worldgenrandom, chunkpos.getBlockX(2), chunkpos.getBlockZ(2), this.f_227958_);
       p_227966_.m_142679_(mineshaftpieces$mineshaftroom);
       mineshaftpieces$mineshaftroom.m_214092_(mineshaftpieces$mineshaftroom, p_227966_, worldgenrandom);
       int i = chunkgenerator.m_6337_();
       if (this.f_227958_ == MineshaftStructure.Type.MESA) {
          BlockPos blockpos = p_227966_.m_192798_().m_162394_();
-         int j = chunkgenerator.m_214096_(blockpos.m_123341_(), blockpos.m_123343_(), Heightmap.Types.WORLD_SURFACE_WG, p_227967_.f_226629_(), p_227967_.f_226624_());
-         int k = j <= i ? i : Mth.m_216287_(worldgenrandom, i, j);
-         int l = k - blockpos.m_123342_();
-         p_227966_.m_192781_(l);
+         long j = chunkgenerator.getBaseHeight(blockpos.getX(), blockpos.getZ(), Heightmap.Types.WORLD_SURFACE_WG, p_227967_.f_226629_(), p_227967_.f_226624_());
+         int k = j <= i ? i : Mth.m_216287_(worldgenrandom, i, (int) j);
+         long l = k - blockpos.getY();
+         p_227966_.offsetPiecesVertically(l);
          return l;
       } else {
-         return p_227966_.m_226965_(i, chunkgenerator.m_142062_(), worldgenrandom, 10);
+         return p_227966_.moveBelowSeaLevel(i, chunkgenerator.getMinY(), worldgenrandom, 10);
       }
    }
 
