--- a/net/minecraft/world/level/levelgen/structure/structures/NetherFossilStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/NetherFossilStructure.java
@@ -30,19 +_,19 @@
 
    public Optional<Structure.GenerationStub> m_214086_(Structure.GenerationContext p_228576_) {
       WorldgenRandom worldgenrandom = p_228576_.f_226626_();
-      int i = p_228576_.f_226628_().m_45604_() + worldgenrandom.m_188503_(16);
-      int j = p_228576_.f_226628_().m_45605_() + worldgenrandom.m_188503_(16);
-      int k = p_228576_.f_226622_().m_6337_();
+      me.alphamode.mclong.math.BigInteger i = p_228576_.f_226628_().getMinBigBlockX().add(worldgenrandom.m_188503_(16));
+      me.alphamode.mclong.math.BigInteger j = p_228576_.f_226628_().getMinBigBlockZ().add(worldgenrandom.m_188503_(16));
+      long k = p_228576_.f_226622_().getSeaLevel();
       WorldGenerationContext worldgenerationcontext = new WorldGenerationContext(p_228576_.f_226622_(), p_228576_.f_226629_());
-      int l = this.f_228570_.m_213859_(worldgenrandom, worldgenerationcontext);
-      NoiseColumn noisecolumn = p_228576_.f_226622_().m_214184_(i, j, p_228576_.f_226629_(), p_228576_.f_226624_());
-      BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(i, l, j);
+      long l = this.f_228570_.sample(worldgenrandom, worldgenerationcontext);
+      NoiseColumn noisecolumn = p_228576_.f_226622_().getBaseColumn(i, j, p_228576_.f_226629_(), p_228576_.f_226624_());
+      BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(i, me.alphamode.mclong.math.BigInteger.valueOf(l), j);
 
       while(l > k) {
-         BlockState blockstate = noisecolumn.m_183556_(l);
+         BlockState blockstate = noisecolumn.getBlock(l);
          --l;
-         BlockState blockstate1 = noisecolumn.m_183556_(l);
-         if (blockstate.m_60795_() && (blockstate1.m_60713_(Blocks.f_50135_) || blockstate1.m_60783_(EmptyBlockGetter.INSTANCE, blockpos$mutableblockpos.m_142448_(l), Direction.UP))) {
+         BlockState blockstate1 = noisecolumn.getBlock(l);
+         if (blockstate.m_60795_() && (blockstate1.m_60713_(Blocks.f_50135_) || blockstate1.m_60783_(EmptyBlockGetter.INSTANCE, blockpos$mutableblockpos.setY(l), Direction.UP))) {
             break;
          }
       }
@@ -50,7 +_,7 @@
       if (l <= k) {
          return Optional.empty();
       } else {
-         BlockPos blockpos = new BlockPos(i, l, j);
+         BlockPos blockpos = new BlockPos(i, me.alphamode.mclong.math.BigInteger.valueOf(l), j);
          return Optional.of(new Structure.GenerationStub(blockpos, (p_228581_) -> {
             NetherFossilPieces.m_228534_(p_228576_.f_226625_(), p_228581_, worldgenrandom, blockpos);
          }));
