--- a/net/minecraft/world/level/levelgen/structure/structures/OceanRuinStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/OceanRuinStructure.java
@@ -39,7 +_,7 @@
    }
 
    private void m_229069_(StructurePiecesBuilder p_229070_, Structure.GenerationContext p_229071_) {
-      BlockPos blockpos = new BlockPos(p_229071_.f_226628_().m_45604_(), 90, p_229071_.f_226628_().m_45605_());
+      BlockPos blockpos = new BlockPos(p_229071_.f_226628_().getMinBigBlockX(), 90, p_229071_.f_226628_().getMinBigBlockZ());
       Rotation rotation = Rotation.m_221990_(p_229071_.f_226626_());
       OceanRuinPieces.m_228994_(p_229071_.f_226625_(), blockpos, rotation, p_229070_, p_229071_.f_226626_(), this);
    }
