--- a/net/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/RuinedPortalPiece.java
@@ -113,7 +_,7 @@
 
    public void m_213694_(WorldGenLevel p_229137_, StructureManager p_229138_, ChunkGenerator p_229139_, RandomSource p_229140_, BoundingBox p_229141_, ChunkPos p_229142_, BlockPos p_229143_) {
       BoundingBox boundingbox = this.f_73656_.m_74633_(this.f_73657_, this.f_73658_);
-      if (p_229141_.m_71051_(boundingbox.m_162394_())) {
+      if (p_229141_.isInside(boundingbox.m_162394_())) {
          p_229141_.m_162386_(boundingbox);
          super.m_213694_(p_229137_, p_229138_, p_229139_, p_229140_, p_229141_, p_229142_, p_229143_);
          this.m_229178_(p_229140_, p_229137_);
@@ -141,7 +_,7 @@
       BlockState blockstate = p_229122_.m_8055_(p_229123_);
       if (!blockstate.m_60795_() && !blockstate.m_60713_(Blocks.f_50191_)) {
          Direction direction = m_226760_(p_229121_);
-         BlockPos blockpos = p_229123_.m_121945_(direction);
+         BlockPos blockpos = p_229123_.relative(direction);
          BlockState blockstate1 = p_229122_.m_8055_(blockpos);
          if (blockstate1.m_60795_()) {
             if (Block.m_49918_(blockstate.m_60812_(p_229122_, p_229123_), direction)) {
@@ -153,18 +_,18 @@
    }
 
    private void m_229181_(RandomSource p_229182_, LevelAccessor p_229183_, BlockPos p_229184_) {
-      if (p_229182_.m_188501_() < 0.5F && p_229183_.m_8055_(p_229184_).m_60713_(Blocks.f_50134_) && p_229183_.m_8055_(p_229184_.m_7494_()).m_60795_()) {
-         p_229183_.m_7731_(p_229184_.m_7494_(), Blocks.f_50053_.m_49966_().m_61124_(LeavesBlock.f_54419_, Boolean.valueOf(true)), 3);
+      if (p_229182_.m_188501_() < 0.5F && p_229183_.m_8055_(p_229184_).m_60713_(Blocks.f_50134_) && p_229183_.m_8055_(p_229184_.above()).m_60795_()) {
+         p_229183_.m_7731_(p_229184_.above(), Blocks.f_50053_.m_49966_().m_61124_(LeavesBlock.f_54419_, Boolean.valueOf(true)), 3);
       }
 
    }
 
    private void m_229117_(RandomSource p_229118_, LevelAccessor p_229119_) {
-      for(int i = this.f_73383_.m_162395_() + 1; i < this.f_73383_.m_162399_(); ++i) {
-         for(int j = this.f_73383_.m_162398_() + 1; j < this.f_73383_.m_162401_(); ++j) {
-            BlockPos blockpos = new BlockPos(i, this.f_73383_.m_162396_(), j);
+      for(long i = this.f_73383_.minX() + 1; i < this.f_73383_.maxX(); ++i) {
+         for(long j = this.f_73383_.minZ() + 1; j < this.f_73383_.maxZ(); ++j) {
+            BlockPos blockpos = new BlockPos(i, this.f_73383_.minY(), j);
             if (p_229119_.m_8055_(blockpos).m_60713_(Blocks.f_50134_)) {
-               this.m_229189_(p_229118_, p_229119_, blockpos.m_7495_());
+               this.m_229189_(p_229118_, p_229119_, blockpos.below());
             }
          }
       }
@@ -187,32 +_,32 @@
    private void m_229178_(RandomSource p_229179_, LevelAccessor p_229180_) {
       boolean flag = this.f_229101_ == RuinedPortalPiece.VerticalPlacement.ON_LAND_SURFACE || this.f_229101_ == RuinedPortalPiece.VerticalPlacement.ON_OCEAN_FLOOR;
       BlockPos blockpos = this.f_73383_.m_162394_();
-      int i = blockpos.m_123341_();
-      int j = blockpos.m_123343_();
+      long i = blockpos.getX();
+      long j = blockpos.getZ();
       float[] afloat = new float[]{1.0F, 1.0F, 1.0F, 1.0F, 1.0F, 1.0F, 1.0F, 0.9F, 0.9F, 0.8F, 0.7F, 0.6F, 0.4F, 0.2F};
       int k = afloat.length;
-      int l = (this.f_73383_.m_71056_() + this.f_73383_.m_71058_()) / 2;
-      int i1 = p_229179_.m_188503_(Math.max(1, 8 - l / 2));
+      long l = (this.f_73383_.getXSpan() + this.f_73383_.getZSpan()) / 2;
+      long i1 = p_229179_.nextLong(Math.max(1, 8 - l / 2));
       int j1 = 3;
       BlockPos.MutableBlockPos blockpos$mutableblockpos = BlockPos.f_121853_.m_122032_();
 
-      for(int k1 = i - k; k1 <= i + k; ++k1) {
-         for(int l1 = j - k; l1 <= j + k; ++l1) {
-            int i2 = Math.abs(k1 - i) + Math.abs(l1 - j);
-            int j2 = Math.max(0, i2 + i1);
+      for(long k1 = i - k; k1 <= i + k; ++k1) {
+         for(long l1 = j - k; l1 <= j + k; ++l1) {
+            long i2 = Math.abs(k1 - i) + Math.abs(l1 - j);
+            long j2 = Math.max(0, i2 + i1);
             if (j2 < k) {
-               float f = afloat[j2];
+               float f = afloat[(int) j2];
                if (p_229179_.m_188500_() < (double)f) {
-                  int k2 = m_229128_(p_229180_, k1, l1, this.f_229101_);
-                  int l2 = flag ? k2 : Math.min(this.f_73383_.m_162396_(), k2);
-                  blockpos$mutableblockpos.m_122178_(k1, l2, l1);
-                  if (Math.abs(l2 - this.f_73383_.m_162396_()) <= 3 && this.m_229133_(p_229180_, blockpos$mutableblockpos)) {
+                  long k2 = getSurfaceY(p_229180_, k1, l1, this.f_229101_);
+                  long l2 = flag ? k2 : Math.min(this.f_73383_.minY(), k2);
+                  blockpos$mutableblockpos.set(k1, l2, l1);
+                  if (Math.abs(l2 - this.f_73383_.minY()) <= 3 && this.m_229133_(p_229180_, blockpos$mutableblockpos)) {
                      this.m_229193_(p_229179_, p_229180_, blockpos$mutableblockpos);
                      if (this.f_229102_.f_229201_) {
                         this.m_229181_(p_229179_, p_229180_, blockpos$mutableblockpos);
                      }
 
-                     this.m_229189_(p_229179_, p_229180_, blockpos$mutableblockpos.m_7495_());
+                     this.m_229189_(p_229179_, p_229180_, blockpos$mutableblockpos.below());
                   }
                }
             }
@@ -235,8 +_,8 @@
 
    }
 
-   private static int m_229128_(LevelAccessor p_229129_, int p_229130_, int p_229131_, RuinedPortalPiece.VerticalPlacement p_229132_) {
-      return p_229129_.m_6924_(m_229160_(p_229132_), p_229130_, p_229131_) - 1;
+   private static long getSurfaceY(LevelAccessor p_229129_, long p_229130_, long p_229131_, RuinedPortalPiece.VerticalPlacement p_229132_) {
+      return p_229129_.getHeight(m_229160_(p_229132_), p_229130_, p_229131_) - 1;
    }
 
    public static Heightmap.Types m_229160_(RuinedPortalPiece.VerticalPlacement p_229161_) {
