--- a/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
@@ -32,7 +_,7 @@
 
    private void m_229395_(StructurePiecesBuilder p_229396_, Structure.GenerationContext p_229397_) {
       Rotation rotation = Rotation.m_221990_(p_229397_.f_226626_());
-      BlockPos blockpos = new BlockPos(p_229397_.f_226628_().m_45604_(), 90, p_229397_.f_226628_().m_45605_());
+      BlockPos blockpos = new BlockPos(p_229397_.f_226628_().getMinBlockX(), 90, p_229397_.f_226628_().getMinBlockZ());
       ShipwreckPieces.m_229345_(p_229397_.f_226625_(), blockpos, rotation, p_229396_, p_229397_.f_226626_(), this.f_229385_);
    }
 
