--- a/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
@@ -27,8 +_,8 @@
    private boolean f_229949_;
    private boolean f_229950_;
 
-   public SwampHutPiece(RandomSource p_229952_, int p_229953_, int p_229954_) {
-      super(StructurePieceType.f_210105_, p_229953_, 64, p_229954_, 7, 7, 9, m_226760_(p_229952_));
+   public SwampHutPiece(RandomSource pRandom, long pX, long pZ) {
+      super(StructurePieceType.f_210105_, pX, 64, pZ, 7, 7, 9, m_226760_(pRandom));
    }
 
    public SwampHutPiece(CompoundTag p_229956_) {
@@ -45,39 +_,39 @@
 
    public void m_213694_(WorldGenLevel p_229961_, StructureManager p_229962_, ChunkGenerator p_229963_, RandomSource p_229964_, BoundingBox p_229965_, ChunkPos p_229966_, BlockPos p_229967_) {
       if (this.m_72803_(p_229961_, p_229965_, 0)) {
-         this.m_73441_(p_229961_, p_229965_, 1, 1, 1, 5, 1, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 1, 4, 2, 5, 4, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 2, 1, 0, 4, 1, 0, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 2, 2, 2, 3, 3, 2, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 1, 2, 3, 1, 3, 6, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 5, 2, 3, 5, 3, 6, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 2, 2, 7, 4, 3, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 1, 0, 2, 1, 3, 2, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 5, 0, 2, 5, 3, 2, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 1, 0, 7, 1, 3, 7, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
-         this.m_73441_(p_229961_, p_229965_, 5, 0, 7, 5, 3, 7, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
-         this.m_73434_(p_229961_, Blocks.f_50132_.m_49966_(), 2, 3, 2, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50132_.m_49966_(), 3, 3, 7, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50016_.m_49966_(), 1, 3, 4, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50016_.m_49966_(), 5, 3, 4, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50016_.m_49966_(), 5, 3, 5, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50245_.m_49966_(), 1, 3, 5, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50091_.m_49966_(), 3, 2, 6, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50256_.m_49966_(), 4, 2, 6, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50132_.m_49966_(), 1, 2, 1, p_229965_);
-         this.m_73434_(p_229961_, Blocks.f_50132_.m_49966_(), 5, 2, 1, p_229965_);
+         this.generateBox(p_229961_, p_229965_, 1, 1, 1, 5, 1, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 1, 4, 2, 5, 4, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 2, 1, 0, 4, 1, 0, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 2, 2, 2, 3, 3, 2, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 1, 2, 3, 1, 3, 6, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 5, 2, 3, 5, 3, 6, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 2, 2, 7, 4, 3, 7, Blocks.f_50741_.m_49966_(), Blocks.f_50741_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 1, 0, 2, 1, 3, 2, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 5, 0, 2, 5, 3, 2, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 1, 0, 7, 1, 3, 7, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
+         this.generateBox(p_229961_, p_229965_, 5, 0, 7, 5, 3, 7, Blocks.f_49999_.m_49966_(), Blocks.f_49999_.m_49966_(), false);
+         this.placeBlock(p_229961_, Blocks.f_50132_.m_49966_(), 2, 3, 2, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50132_.m_49966_(), 3, 3, 7, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50016_.m_49966_(), 1, 3, 4, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50016_.m_49966_(), 5, 3, 4, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50016_.m_49966_(), 5, 3, 5, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50245_.m_49966_(), 1, 3, 5, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50091_.m_49966_(), 3, 2, 6, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50256_.m_49966_(), 4, 2, 6, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50132_.m_49966_(), 1, 2, 1, p_229965_);
+         this.placeBlock(p_229961_, Blocks.f_50132_.m_49966_(), 5, 2, 1, p_229965_);
          BlockState blockstate = Blocks.f_50269_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.NORTH);
          BlockState blockstate1 = Blocks.f_50269_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.EAST);
          BlockState blockstate2 = Blocks.f_50269_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.WEST);
          BlockState blockstate3 = Blocks.f_50269_.m_49966_().m_61124_(StairBlock.f_56841_, Direction.SOUTH);
-         this.m_73441_(p_229961_, p_229965_, 0, 4, 1, 6, 4, 1, blockstate, blockstate, false);
-         this.m_73441_(p_229961_, p_229965_, 0, 4, 2, 0, 4, 7, blockstate1, blockstate1, false);
-         this.m_73441_(p_229961_, p_229965_, 6, 4, 2, 6, 4, 7, blockstate2, blockstate2, false);
-         this.m_73441_(p_229961_, p_229965_, 0, 4, 8, 6, 4, 8, blockstate3, blockstate3, false);
-         this.m_73434_(p_229961_, blockstate.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_RIGHT), 0, 4, 1, p_229965_);
-         this.m_73434_(p_229961_, blockstate.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_LEFT), 6, 4, 1, p_229965_);
-         this.m_73434_(p_229961_, blockstate3.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_LEFT), 0, 4, 8, p_229965_);
-         this.m_73434_(p_229961_, blockstate3.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_RIGHT), 6, 4, 8, p_229965_);
+         this.generateBox(p_229961_, p_229965_, 0, 4, 1, 6, 4, 1, blockstate, blockstate, false);
+         this.generateBox(p_229961_, p_229965_, 0, 4, 2, 0, 4, 7, blockstate1, blockstate1, false);
+         this.generateBox(p_229961_, p_229965_, 6, 4, 2, 6, 4, 7, blockstate2, blockstate2, false);
+         this.generateBox(p_229961_, p_229965_, 0, 4, 8, 6, 4, 8, blockstate3, blockstate3, false);
+         this.placeBlock(p_229961_, blockstate.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_RIGHT), 0, 4, 1, p_229965_);
+         this.placeBlock(p_229961_, blockstate.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_LEFT), 6, 4, 1, p_229965_);
+         this.placeBlock(p_229961_, blockstate3.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_LEFT), 0, 4, 8, p_229965_);
+         this.placeBlock(p_229961_, blockstate3.m_61124_(StairBlock.f_56843_, StairsShape.OUTER_RIGHT), 6, 4, 8, p_229965_);
 
          for(int i = 2; i <= 7; i += 5) {
             for(int j = 1; j <= 5; j += 4) {
@@ -86,13 +_,13 @@
          }
 
          if (!this.f_229949_) {
-            BlockPos blockpos = this.m_163582_(2, 2, 5);
-            if (p_229965_.m_71051_(blockpos)) {
+            BlockPos blockpos = this.getWorldPos(2, 2, 5);
+            if (p_229965_.isInside(blockpos)) {
                this.f_229949_ = true;
                Witch witch = EntityType.f_20495_.m_20615_(p_229961_.m_6018_());
                if (witch != null) {
                   witch.m_21530_();
-                  witch.m_7678_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_(), (double)blockpos.m_123343_() + 0.5D, 0.0F, 0.0F);
+                  witch.moveTo(blockpos.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), blockpos.getBigDecY(), blockpos.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), 0.0F, 0.0F);
                   witch.m_6518_(p_229961_, p_229961_.m_6436_(blockpos), MobSpawnType.STRUCTURE, (SpawnGroupData)null, (CompoundTag)null);
                   p_229961_.m_47205_(witch);
                }
@@ -105,13 +_,13 @@
 
    private void m_229957_(ServerLevelAccessor p_229958_, BoundingBox p_229959_) {
       if (!this.f_229950_) {
-         BlockPos blockpos = this.m_163582_(2, 2, 5);
-         if (p_229959_.m_71051_(blockpos)) {
+         BlockPos blockpos = this.getWorldPos(2, 2, 5);
+         if (p_229959_.isInside(blockpos)) {
             this.f_229950_ = true;
             Cat cat = EntityType.f_20553_.m_20615_(p_229958_.m_6018_());
             if (cat != null) {
                cat.m_21530_();
-               cat.m_7678_((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_(), (double)blockpos.m_123343_() + 0.5D, 0.0F, 0.0F);
+               cat.moveTo(blockpos.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), blockpos.getBigDecY(), blockpos.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), 0.0F, 0.0F);
                cat.m_6518_(p_229958_, p_229958_.m_6436_(blockpos), MobSpawnType.STRUCTURE, (SpawnGroupData)null, (CompoundTag)null);
                p_229958_.m_47205_(cat);
             }
