--- a/net/minecraft/world/level/levelgen/structure/structures/SwampHutStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/SwampHutStructure.java
@@ -21,7 +_,7 @@
    }
 
    private static void m_229980_(StructurePiecesBuilder p_229981_, Structure.GenerationContext p_229982_) {
-      p_229981_.m_142679_(new SwampHutPiece(p_229982_.f_226626_(), p_229982_.f_226628_().m_45604_(), p_229982_.f_226628_().m_45605_()));
+      p_229981_.m_142679_(new SwampHutPiece(p_229982_.f_226626_(), p_229982_.f_226628_().getMinBlockX(), p_229982_.f_226628_().getMinBlockZ()));
    }
 
    public StructureType<?> m_213658_() {
