--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/BendingTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/BendingTrunkPlacer.java
@@ -41,7 +_,7 @@
       Direction direction = Direction.Plane.HORIZONTAL.m_235690_(p_226081_);
       int i = p_226082_ - 1;
       BlockPos.MutableBlockPos blockpos$mutableblockpos = p_226083_.m_122032_();
-      BlockPos blockpos = blockpos$mutableblockpos.m_7495_();
+      BlockPos blockpos = blockpos$mutableblockpos.below();
       m_226169_(p_226079_, p_226080_, p_226081_, blockpos, p_226084_);
       List<FoliagePlacer.FoliageAttachment> list = Lists.newArrayList();
 
