--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/DarkOakTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/DarkOakTrunkPlacer.java
@@ -29,20 +_,20 @@
 
    public List<FoliagePlacer.FoliageAttachment> m_213934_(LevelSimulatedReader p_226086_, BiConsumer<BlockPos, BlockState> p_226087_, RandomSource p_226088_, int p_226089_, BlockPos p_226090_, TreeConfiguration p_226091_) {
       List<FoliagePlacer.FoliageAttachment> list = Lists.newArrayList();
-      BlockPos blockpos = p_226090_.m_7495_();
+      BlockPos blockpos = p_226090_.below();
       m_226169_(p_226086_, p_226087_, p_226088_, blockpos, p_226091_);
-      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.m_122029_(), p_226091_);
-      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.m_122019_(), p_226091_);
-      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.m_122019_().m_122029_(), p_226091_);
+      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.east(), p_226091_);
+      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.south(), p_226091_);
+      m_226169_(p_226086_, p_226087_, p_226088_, blockpos.south().east(), p_226091_);
       Direction direction = Direction.Plane.HORIZONTAL.m_235690_(p_226088_);
       int i = p_226089_ - p_226088_.m_188503_(4);
       int j = 2 - p_226088_.m_188503_(3);
-      int k = p_226090_.m_123341_();
-      int l = p_226090_.m_123342_();
-      int i1 = p_226090_.m_123343_();
-      int j1 = k;
-      int k1 = i1;
-      int l1 = l + p_226089_ - 1;
+      long k = p_226090_.getX();
+      long l = p_226090_.getY();
+      long i1 = p_226090_.getZ();
+      long j1 = k;
+      long k1 = i1;
+      long l1 = l + p_226089_ - 1;
 
       for(int i2 = 0; i2 < p_226089_; ++i2) {
          if (i2 >= i && j > 0) {
@@ -51,13 +_,13 @@
             --j;
          }
 
-         int j2 = l + i2;
+         long j2 = l + i2;
          BlockPos blockpos1 = new BlockPos(j1, j2, k1);
          if (TreeFeature.m_67267_(p_226086_, blockpos1)) {
             this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1, p_226091_);
-            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.m_122029_(), p_226091_);
-            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.m_122019_(), p_226091_);
-            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.m_122029_().m_122019_(), p_226091_);
+            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.east(), p_226091_);
+            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.south(), p_226091_);
+            this.m_226187_(p_226086_, p_226087_, p_226088_, blockpos1.east().south(), p_226091_);
          }
       }
 
