--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/MegaJungleTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/MegaJungleTrunkPlacer.java
@@ -38,11 +_,11 @@
          for(int l = 0; l < 5; ++l) {
             j = (int)(1.5F + Mth.m_14089_(f) * (float)l);
             k = (int)(1.5F + Mth.m_14031_(f) * (float)l);
-            BlockPos blockpos = p_226144_.m_7918_(j, i - 3 + l / 2, k);
+            BlockPos blockpos = p_226144_.offset(j, i - 3 + l / 2, k);
             this.m_226187_(p_226140_, p_226141_, p_226142_, blockpos, p_226145_);
          }
 
-         list.add(new FoliagePlacer.FoliageAttachment(p_226144_.m_7918_(j, i, k), -2, false));
+         list.add(new FoliagePlacer.FoliageAttachment(p_226144_.offset(j, i, k), -2, false));
       }
 
       return list;
