--- a/net/minecraft/client/renderer/entity/EndCrystalRenderer.java
+++ b/net/minecraft/client/renderer/entity/EndCrystalRenderer.java
@@ -81,9 +_,9 @@
       p_114165_.m_85849_();
       BlockPos blockpos = p_114162_.m_31064_();
       if (blockpos != null) {
-         float f3 = (float)blockpos.m_123341_() + 0.5F;
-         float f4 = (float)blockpos.m_123342_() + 0.5F;
-         float f5 = (float)blockpos.m_123343_() + 0.5F;
+         float f3 = (float)blockpos.getX() + 0.5F;
+         float f4 = (float)blockpos.getY() + 0.5F;
+         float f5 = (float)blockpos.getZ() + 0.5F;
          float f6 = (float)((double)f3 - p_114162_.m_20185_());
          float f7 = (float)((double)f4 - p_114162_.m_20186_());
          float f8 = (float)((double)f5 - p_114162_.m_20189_());
@@ -105,7 +_,7 @@
       return f_114132_;
    }
 
-   public boolean m_5523_(EndCrystal p_114169_, Frustum p_114170_, double p_114171_, double p_114172_, double p_114173_) {
-      return super.m_5523_(p_114169_, p_114170_, p_114171_, p_114172_, p_114173_) || p_114169_.m_31064_() != null;
+   public boolean shouldRender(EndCrystal pLivingEntity, Frustum pCamera, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
+      return super.shouldRender(pLivingEntity, pCamera, pCamX, pCamY, pCamZ) || pLivingEntity.m_31064_() != null;
    }
 }
