--- a/net/minecraft/client/renderer/entity/EntityRenderer.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderer.java
@@ -31,7 +_,7 @@
    }
 
    public final int m_114505_(T p_114506_, float p_114507_) {
-      BlockPos blockpos = BlockPos.m_274446_(p_114506_.m_7371_(p_114507_));
+      BlockPos blockpos = BlockPos.containing(p_114506_.m_7371_(p_114507_));
       return LightTexture.m_109885_(this.m_6086_(p_114506_, blockpos), this.m_114508_(p_114506_, blockpos));
    }
 
@@ -43,18 +_,18 @@
       return p_114496_.m_6060_() ? 15 : p_114496_.m_9236_().m_45517_(LightLayer.BLOCK, p_114497_);
    }
 
-   public boolean m_5523_(T p_114491_, Frustum p_114492_, double p_114493_, double p_114494_, double p_114495_) {
-      if (!p_114491_.m_6000_(p_114493_, p_114494_, p_114495_)) {
+   public boolean shouldRender(T pLivingEntity, Frustum pCamera, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
+      if (!pLivingEntity.shouldRender(pCamX, pCamY, pCamZ)) {
          return false;
-      } else if (p_114491_.f_19811_) {
+      } else if (pLivingEntity.f_19811_) {
          return true;
       } else {
-         AABB aabb = p_114491_.m_6921_().m_82400_(0.5D);
+         AABB aabb = pLivingEntity.m_6921_().m_82400_(0.5D);
          if (aabb.m_82392_() || aabb.m_82309_() == 0.0D) {
-            aabb = new AABB(p_114491_.m_20185_() - 2.0D, p_114491_.m_20186_() - 2.0D, p_114491_.m_20189_() - 2.0D, p_114491_.m_20185_() + 2.0D, p_114491_.m_20186_() + 2.0D, p_114491_.m_20189_() + 2.0D);
+            aabb = new AABB(pLivingEntity.getBigX().subtract(me.alphamode.mclong.core.BigConstants.TWO), pLivingEntity.getBigY().subtract(me.alphamode.mclong.core.BigConstants.TWO), pLivingEntity.getBigZ().subtract(me.alphamode.mclong.core.BigConstants.TWO), pLivingEntity.getBigX().add(me.alphamode.mclong.core.BigConstants.TWO), pLivingEntity.getBigY().add(me.alphamode.mclong.core.BigConstants.TWO), pLivingEntity.getBigZ().add(me.alphamode.mclong.core.BigConstants.TWO));
          }
 
-         return p_114492_.m_113029_(aabb);
+         return pCamera.m_113029_(aabb);
       }
    }
 
