--- a/net/minecraft/client/renderer/entity/EntityRenderer.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderer.java
@@ -31,7 +_,7 @@
    }
 
    public final int m_114505_(T p_114506_, float p_114507_) {
-      BlockPos blockpos = BlockPos.m_274446_(p_114506_.m_7371_(p_114507_));
+      BlockPos blockpos = BlockPos.containing(p_114506_.m_7371_(p_114507_));
       return LightTexture.m_109885_(this.m_6086_(p_114506_, blockpos), this.m_114508_(p_114506_, blockpos));
    }
 
