--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -122,9 +_,9 @@
       return this.f_114369_;
    }
 
-   public <E extends Entity> boolean m_114397_(E p_114398_, Frustum p_114399_, double p_114400_, double p_114401_, double p_114402_) {
+   public <E extends Entity> boolean shouldRender(E p_114398_, Frustum p_114399_, me.alphamode.mclong.math.BigDecimal p_114400_, me.alphamode.mclong.math.BigDecimal p_114401_, me.alphamode.mclong.math.BigDecimal p_114402_) {
       EntityRenderer<? super E> entityrenderer = this.m_114382_(p_114398_);
-      return entityrenderer.m_5523_(p_114398_, p_114399_, p_114400_, p_114401_, p_114402_);
+      return entityrenderer.shouldRender(p_114398_, p_114399_, p_114400_, p_114401_, p_114402_);
    }
 
    public <E extends Entity> void m_114384_(E p_114385_, double p_114386_, double p_114387_, double p_114388_, float p_114389_, float p_114390_, PoseStack p_114391_, MultiBufferSource p_114392_, int p_114393_) {
@@ -144,8 +_,8 @@
 
          p_114391_.m_85837_(-vec3.m_7096_(), -vec3.m_7098_(), -vec3.m_7094_());
          if (this.f_114360_.m_231818_().m_231551_() && this.f_114368_ && entityrenderer.f_114477_ > 0.0F && !p_114385_.m_20145_()) {
-            double d1 = this.m_114378_(p_114385_.m_20185_(), p_114385_.m_20186_(), p_114385_.m_20189_());
-            float f = (float)((1.0D - d1 / 256.0D) * (double)entityrenderer.f_114478_);
+            me.alphamode.mclong.math.BigDecimal d1 = this.distanceToSqr(p_114385_.getBigX(), p_114385_.getBigY(), p_114385_.getBigZ());
+            float f = ((me.alphamode.mclong.math.BigDecimal.ONE.subtract(d1.divide(256.0D))).multiply(entityrenderer.f_114478_)).floatValue();
             if (f > 0.0F) {
                m_114457_(p_114391_, p_114392_, p_114385_, f, p_114390_, this.f_114366_, Math.min(entityrenderer.f_114477_, 32.0F));
             }
@@ -173,15 +_,15 @@
       AABB aabb = p_114444_.m_20191_().m_82386_(-p_114444_.m_20185_(), -p_114444_.m_20186_(), -p_114444_.m_20189_());
       LevelRenderer.m_109646_(p_114442_, p_114443_, aabb, 1.0F, 1.0F, 1.0F, 1.0F);
       if (p_114444_ instanceof EnderDragon) {
-         double d0 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19790_, p_114444_.m_20185_());
-         double d1 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19791_, p_114444_.m_20186_());
-         double d2 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19792_, p_114444_.m_20189_());
+         double d0 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19790_.doubleValue(), p_114444_.m_20185_());
+         double d1 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19791_.doubleValue(), p_114444_.m_20186_());
+         double d2 = -Mth.m_14139_((double)p_114445_, p_114444_.f_19792_.doubleValue(), p_114444_.m_20189_());
 
          for(EnderDragonPart enderdragonpart : ((EnderDragon)p_114444_).m_31156_()) {
             p_114442_.m_85836_();
-            double d3 = d0 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19790_, enderdragonpart.m_20185_());
-            double d4 = d1 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19791_, enderdragonpart.m_20186_());
-            double d5 = d2 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19792_, enderdragonpart.m_20189_());
+            double d3 = d0 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19790_.doubleValue(), enderdragonpart.m_20185_());
+            double d4 = d1 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19791_.doubleValue(), enderdragonpart.m_20186_());
+            double d5 = d2 + Mth.m_14139_((double)p_114445_, enderdragonpart.f_19792_.doubleValue(), enderdragonpart.m_20189_());
             p_114442_.m_85837_(d3, d4, d5);
             LevelRenderer.m_109646_(p_114442_, p_114443_, enderdragonpart.m_20191_().m_82386_(-enderdragonpart.m_20185_(), -enderdragonpart.m_20186_(), -enderdragonpart.m_20189_()), 0.25F, 1.0F, 0.0F, 1.0F);
             p_114442_.m_85849_();
@@ -190,14 +_,14 @@
 
       if (p_114444_ instanceof LivingEntity) {
          float f = 0.01F;
-         LevelRenderer.m_109608_(p_114442_, p_114443_, aabb.f_82288_, (double)(p_114444_.m_20192_() - 0.01F), aabb.f_82290_, aabb.f_82291_, (double)(p_114444_.m_20192_() + 0.01F), aabb.f_82293_, 1.0F, 0.0F, 0.0F, 1.0F);
+         LevelRenderer.m_109608_(p_114442_, p_114443_, aabb.f_82288_.doubleValue(), (double)(p_114444_.m_20192_() - 0.01F), aabb.f_82290_.doubleValue(), aabb.f_82291_.doubleValue(), (double)(p_114444_.m_20192_() + 0.01F), aabb.f_82293_.doubleValue(), 1.0F, 0.0F, 0.0F, 1.0F);
       }
 
       Vec3 vec3 = p_114444_.m_20252_(p_114445_);
       Matrix4f matrix4f = p_114442_.m_85850_().m_252922_();
       Matrix3f matrix3f = p_114442_.m_85850_().m_252943_();
-      p_114443_.m_252986_(matrix4f, 0.0F, p_114444_.m_20192_(), 0.0F).m_6122_(0, 0, 255, 255).m_252939_(matrix3f, (float)vec3.f_82479_, (float)vec3.f_82480_, (float)vec3.f_82481_).m_5752_();
-      p_114443_.m_252986_(matrix4f, (float)(vec3.f_82479_ * 2.0D), (float)((double)p_114444_.m_20192_() + vec3.f_82480_ * 2.0D), (float)(vec3.f_82481_ * 2.0D)).m_6122_(0, 0, 255, 255).m_252939_(matrix3f, (float)vec3.f_82479_, (float)vec3.f_82480_, (float)vec3.f_82481_).m_5752_();
+      p_114443_.m_252986_(matrix4f, 0.0F, p_114444_.m_20192_(), 0.0F).m_6122_(0, 0, 255, 255).m_252939_(matrix3f, (float)vec3.m_7096_(), (float)vec3.m_7098_(), (float)vec3.m_7094_()).m_5752_();
+      p_114443_.m_252986_(matrix4f, (float)(vec3.m_7096_() * 2.0D), (float)((double)p_114444_.m_20192_() + vec3.m_7098_() * 2.0D), (float)(vec3.m_7094_() * 2.0D)).m_6122_(0, 0, 255, 255).m_252939_(matrix3f, (float)vec3.m_7096_(), (float)vec3.m_7098_(), (float)vec3.m_7094_()).m_5752_();
    }
 
    private void m_114453_(PoseStack p_114454_, MultiBufferSource p_114455_, Entity p_114456_) {
@@ -253,28 +_,28 @@
          }
       }
 
-      double d2 = Mth.m_14139_((double)p_114462_, p_114460_.f_19790_, p_114460_.m_20185_());
-      double d0 = Mth.m_14139_((double)p_114462_, p_114460_.f_19791_, p_114460_.m_20186_());
-      double d1 = Mth.m_14139_((double)p_114462_, p_114460_.f_19792_, p_114460_.m_20189_());
+      double d2 = Mth.m_14139_((double)p_114462_, p_114460_.f_19790_.doubleValue(), p_114460_.m_20185_());
+      double d0 = Mth.m_14139_((double)p_114462_, p_114460_.f_19791_.doubleValue(), p_114460_.m_20186_());
+      double d1 = Mth.m_14139_((double)p_114462_, p_114460_.f_19792_.doubleValue(), p_114460_.m_20189_());
       float f1 = Math.min(p_114461_ / 0.5F, f);
-      int i = Mth.m_14107_(d2 - (double)f);
-      int j = Mth.m_14107_(d2 + (double)f);
-      int k = Mth.m_14107_(d0 - (double)f1);
-      int l = Mth.m_14107_(d0);
-      int i1 = Mth.m_14107_(d1 - (double)f);
-      int j1 = Mth.m_14107_(d1 + (double)f);
+      long i = Mth.m_14134_(d2 - (double)f);
+      long j = Mth.m_14134_(d2 + (double)f);
+      long k = Mth.m_14134_(d0 - (double)f);
+      long l = Mth.m_14134_(d0);
+      long i1 = Mth.m_14134_(d1 - (double)f);
+      long j1 = Mth.m_14134_(d1 + (double)f);
       PoseStack.Pose posestack$pose = p_114458_.m_85850_();
       VertexConsumer vertexconsumer = p_114459_.m_6299_(f_114361_);
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-      for(int k1 = i1; k1 <= j1; ++k1) {
-         for(int l1 = i; l1 <= j; ++l1) {
-            blockpos$mutableblockpos.m_122178_(l1, 0, k1);
+      for(long k1 = i1; k1 <= j1; ++k1) {
+         for(long l1 = i; l1 <= j; ++l1) {
+            blockpos$mutableblockpos.set(l1, 0, k1);
             ChunkAccess chunkaccess = p_114463_.m_46865_(blockpos$mutableblockpos);
 
-            for(int i2 = k; i2 <= l; ++i2) {
-               blockpos$mutableblockpos.m_142448_(i2);
-               float f2 = p_114461_ - (float)(d0 - (double)blockpos$mutableblockpos.m_123342_()) * 0.5F;
+            for(long i2 = k; i2 <= l; ++i2) {
+               blockpos$mutableblockpos.setY(i2);
+               float f2 = p_114461_ - (float)(d0 - (double)blockpos$mutableblockpos.getY()) * 0.5F;
                m_277056_(posestack$pose, vertexconsumer, chunkaccess, p_114463_, blockpos$mutableblockpos, d2, d0, d1, f, f2);
             }
          }
@@ -283,7 +_,7 @@
    }
 
    private static void m_277056_(PoseStack.Pose p_277956_, VertexConsumer p_277533_, ChunkAccess p_277501_, LevelReader p_277622_, BlockPos p_277911_, double p_277682_, double p_278099_, double p_277806_, float p_277844_, float p_277496_) {
-      BlockPos blockpos = p_277911_.m_7495_();
+      BlockPos blockpos = p_277911_.below();
       BlockState blockstate = p_277501_.m_8055_(blockpos);
       if (blockstate.m_60799_() != RenderShape.INVISIBLE && p_277622_.m_46803_(p_277911_) > 3) {
          if (blockstate.m_60838_(p_277501_, blockpos)) {
@@ -297,16 +_,16 @@
                   }
 
                   AABB aabb = voxelshape.m_83215_();
-                  double d0 = (double)p_277911_.m_123341_() + aabb.f_82288_;
-                  double d1 = (double)p_277911_.m_123341_() + aabb.f_82291_;
-                  double d2 = (double)p_277911_.m_123342_() + aabb.f_82289_;
-                  double d3 = (double)p_277911_.m_123343_() + aabb.f_82290_;
-                  double d4 = (double)p_277911_.m_123343_() + aabb.f_82293_;
-                  float f2 = (float)(d0 - p_277682_);
-                  float f3 = (float)(d1 - p_277682_);
-                  float f4 = (float)(d2 - p_278099_);
-                  float f5 = (float)(d3 - p_277806_);
-                  float f6 = (float)(d4 - p_277806_);
+                  me.alphamode.mclong.math.BigDecimal d0 = p_277911_.getBigDecX().add(aabb.f_82288_);
+                  me.alphamode.mclong.math.BigDecimal d1 = p_277911_.getBigDecX().add(aabb.f_82291_);
+                  me.alphamode.mclong.math.BigDecimal d2 = p_277911_.getBigDecY().add(aabb.f_82289_);
+                  me.alphamode.mclong.math.BigDecimal d3 = p_277911_.getBigDecZ().add(aabb.f_82290_);
+                  me.alphamode.mclong.math.BigDecimal d4 = p_277911_.getBigDecZ().add(aabb.f_82293_);
+                  float f2 = (d0.subtract(me.alphamode.mclong.math.BigDecimal.valueOf(p_277682_))).floatValue();
+                  float f3 = (d1.subtract(me.alphamode.mclong.math.BigDecimal.valueOf(p_277682_))).floatValue();
+                  float f4 = (d2.subtract(me.alphamode.mclong.math.BigDecimal.valueOf(p_278099_))).floatValue();
+                  float f5 = (d3.subtract(me.alphamode.mclong.math.BigDecimal.valueOf(p_277806_))).floatValue();
+                  float f6 = (d4.subtract(me.alphamode.mclong.math.BigDecimal.valueOf(p_277806_))).floatValue();
                   float f7 = -f2 / 2.0F / p_277844_ + 0.5F;
                   float f8 = -f3 / 2.0F / p_277844_ + 0.5F;
                   float f9 = -f5 / 2.0F / p_277844_ + 0.5F;
@@ -339,8 +_,8 @@
       return this.f_114358_.m_90583_().m_82557_(p_114472_.m_20182_());
    }
 
-   public double m_114378_(double p_114379_, double p_114380_, double p_114381_) {
-      return this.f_114358_.m_90583_().m_82531_(p_114379_, p_114380_, p_114381_);
+   public me.alphamode.mclong.math.BigDecimal distanceToSqr(me.alphamode.mclong.math.BigDecimal p_114379_, me.alphamode.mclong.math.BigDecimal p_114380_, me.alphamode.mclong.math.BigDecimal p_114381_) {
+      return this.f_114358_.m_90583_().distanceToSqr(p_114379_, p_114380_, p_114381_);
    }
 
    public Quaternionf m_253208_() {
