--- a/net/minecraft/client/renderer/entity/GuardianRenderer.java
+++ b/net/minecraft/client/renderer/entity/GuardianRenderer.java
@@ -35,16 +_,16 @@
       super(p_174161_, new GuardianModel(p_174161_.m_174023_(p_174163_)), p_174162_);
    }
 
-   public boolean m_5523_(Guardian p_114836_, Frustum p_114837_, double p_114838_, double p_114839_, double p_114840_) {
-      if (super.m_5523_(p_114836_, p_114837_, p_114838_, p_114839_, p_114840_)) {
+   public boolean shouldRender(Guardian pLivingEntity, Frustum pCamera, me.alphamode.mclong.math.BigDecimal pCamX, me.alphamode.mclong.math.BigDecimal pCamY, me.alphamode.mclong.math.BigDecimal pCamZ) {
+      if (super.shouldRender(pLivingEntity, pCamera, pCamX, pCamY, pCamZ)) {
          return true;
       } else {
-         if (p_114836_.m_32855_()) {
-            LivingEntity livingentity = p_114836_.m_32856_();
+         if (pLivingEntity.m_32855_()) {
+            LivingEntity livingentity = pLivingEntity.m_32856_();
             if (livingentity != null) {
                Vec3 vec3 = this.m_114802_(livingentity, (double)livingentity.m_20206_() * 0.5D, 1.0F);
-               Vec3 vec31 = this.m_114802_(p_114836_, (double)p_114836_.m_20192_(), 1.0F);
-               return p_114837_.m_113029_(new AABB(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_));
+               Vec3 vec31 = this.m_114802_(pLivingEntity, (double)pLivingEntity.m_20192_(), 1.0F);
+               return pCamera.m_113029_(new AABB(vec31.getX(), vec31.getY(), vec31.getZ(), vec3.getX(), vec3.getY(), vec3.getZ()));
             }
          }
 
@@ -53,9 +_,10 @@
    }
 
    private Vec3 m_114802_(LivingEntity p_114803_, double p_114804_, float p_114805_) {
-      double d0 = Mth.m_14139_((double)p_114805_, p_114803_.f_19790_, p_114803_.m_20185_());
-      double d1 = Mth.m_14139_((double)p_114805_, p_114803_.f_19791_, p_114803_.m_20186_()) + p_114804_;
-      double d2 = Mth.m_14139_((double)p_114805_, p_114803_.f_19792_, p_114803_.m_20189_());
+      me.alphamode.mclong.math.BigDecimal lerp = me.alphamode.mclong.math.BigDecimal.valueOf(p_114805_);
+      me.alphamode.mclong.math.BigDecimal d0 = Mth.lerp(lerp, p_114803_.f_19790_, p_114803_.getBigX());
+      me.alphamode.mclong.math.BigDecimal d1 = Mth.lerp(lerp, p_114803_.f_19791_, p_114803_.getBigY()).add(me.alphamode.mclong.math.BigDecimal.valueOf(p_114804_));
+      me.alphamode.mclong.math.BigDecimal d2 = Mth.lerp(lerp, p_114803_.f_19792_, p_114803_.getBigZ());
       return new Vec3(d0, d1, d2);
    }
 
@@ -74,8 +_,8 @@
          Vec3 vec32 = vec3.m_82546_(vec31);
          float f4 = (float)(vec32.m_82553_() + 1.0D);
          vec32 = vec32.m_82541_();
-         float f5 = (float)Math.acos(vec32.f_82480_);
-         float f6 = (float)Math.atan2(vec32.f_82481_, vec32.f_82479_);
+         float f5 = (float)Math.acos(vec32.m_7098_());
+         float f6 = (float)Math.atan2(vec32.m_7094_(), vec32.m_7096_());
          p_114832_.m_252781_(Axis.f_252436_.m_252977_((((float)Math.PI / 2F) - f6) * (180F / (float)Math.PI)));
          p_114832_.m_252781_(Axis.f_252529_.m_252977_(f5 * (180F / (float)Math.PI)));
          int i = 1;
