--- a/net/minecraft/client/renderer/entity/IllusionerRenderer.java
+++ b/net/minecraft/client/renderer/entity/IllusionerRenderer.java
@@ -40,7 +_,7 @@
 
          for(int i = 0; i < avec3.length; ++i) {
             p_114955_.m_85836_();
-            p_114955_.m_85837_(avec3[i].f_82479_ + (double)Mth.m_14089_((float)i + f * 0.5F) * 0.025D, avec3[i].f_82480_ + (double)Mth.m_14089_((float)i + f * 0.75F) * 0.0125D, avec3[i].f_82481_ + (double)Mth.m_14089_((float)i + f * 0.7F) * 0.025D);
+            p_114955_.m_85837_(avec3[i].m_7096_() + (double)Mth.m_14089_((float)i + f * 0.5F) * 0.025D, avec3[i].m_7098_() + (double)Mth.m_14089_((float)i + f * 0.75F) * 0.0125D, avec3[i].m_7094_() + (double)Mth.m_14089_((float)i + f * 0.7F) * 0.025D);
             super.m_7392_(p_114952_, p_114953_, p_114954_, p_114955_, p_114956_, p_114957_);
             p_114955_.m_85849_();
          }
