--- a/net/minecraft/client/renderer/entity/PaintingRenderer.java
+++ b/net/minecraft/client/renderer/entity/PaintingRenderer.java
@@ -74,9 +_,9 @@
             float f16 = f + (float)(k * 16);
             float f17 = f1 + (float)((l + 1) * 16);
             float f18 = f1 + (float)(l * 16);
-            int i1 = p_115561_.m_146903_();
+            long i1 = p_115561_.getBlockX();
             int j1 = Mth.m_14107_(p_115561_.m_20186_() + (double)((f17 + f18) / 2.0F / 16.0F));
-            int k1 = p_115561_.m_146907_();
+            long k1 = p_115561_.getBlockZ();
             Direction direction = p_115561_.m_6350_();
             if (direction == Direction.NORTH) {
                i1 = Mth.m_14107_(p_115561_.m_20185_() + (double)((f15 + f16) / 2.0F / 16.0F));
