--- a/net/minecraft/client/renderer/entity/ShulkerRenderer.java
+++ b/net/minecraft/client/renderer/entity/ShulkerRenderer.java
@@ -39,8 +_,8 @@
          EntityType<?> entitytype = p_115913_.m_6095_();
          float f = entitytype.m_20679_() / 2.0F;
          float f1 = entitytype.m_20678_() / 2.0F;
-         Vec3 vec3 = Vec3.m_82539_(p_115913_.m_20183_());
-         return p_115914_.m_113029_((new AABB(p_174374_.f_82479_, p_174374_.f_82480_ + (double)f, p_174374_.f_82481_, vec3.f_82479_, vec3.f_82480_ + (double)f, vec3.f_82481_)).m_82377_((double)f1, (double)f, (double)f1));
+         Vec3 vec3 = Vec3.atBottomCenterOf(p_115913_.m_20183_());
+         return p_115914_.m_113029_((new AABB(p_174374_.m_7096_(), p_174374_.m_7098_() + (double)f, p_174374_.m_7094_(), vec3.m_7096_(), vec3.m_7098_() + (double)f, vec3.m_7094_())).m_82377_((double)f1, (double)f, (double)f1));
       }).isPresent();
    }
 
