--- a/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -108,9 +_,9 @@
       this.f_36697_ = 0;
    }
 
-   public void m_6453_(double p_36728_, double p_36729_, double p_36730_, float p_36731_, float p_36732_, int p_36733_, boolean p_36734_) {
-      this.m_6034_(p_36728_, p_36729_, p_36730_);
-      this.m_19915_(p_36731_, p_36732_);
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pYaw, float pPitch, int pPosRotationIncrements, boolean pTeleport) {
+      this.setPos(pX, pY, pZ);
+      this.m_19915_(pYaw, pPitch);
    }
 
    public void m_6001_(double p_36786_, double p_36787_, double p_36788_) {
@@ -124,8 +_,8 @@
       Vec3 vec3 = this.m_20184_();
       if (this.f_19860_ == 0.0F && this.f_19859_ == 0.0F) {
          double d0 = vec3.m_165924_();
-         this.m_146922_((float)(Mth.m_14136_(vec3.f_82479_, vec3.f_82481_) * (double)(180F / (float)Math.PI)));
-         this.m_146926_((float)(Mth.m_14136_(vec3.f_82480_, d0) * (double)(180F / (float)Math.PI)));
+         this.m_146922_((float)(Mth.m_14136_(vec3.m_7096_(), vec3.m_7094_()) * (double)(180F / (float)Math.PI)));
+         this.m_146926_((float)(Mth.m_14136_(vec3.m_7098_(), d0) * (double)(180F / (float)Math.PI)));
          this.f_19859_ = this.m_146908_();
          this.f_19860_ = this.m_146909_();
       }
@@ -199,18 +_,18 @@
          }
 
          vec3 = this.m_20184_();
-         double d5 = vec3.f_82479_;
-         double d6 = vec3.f_82480_;
-         double d1 = vec3.f_82481_;
+         double d5 = vec3.m_7096_();
+         double d6 = vec3.m_7098_();
+         double d1 = vec3.m_7094_();
          if (this.m_36792_()) {
             for(int i = 0; i < 4; ++i) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123797_, this.m_20185_() + d5 * (double)i / 4.0D, this.m_20186_() + d6 * (double)i / 4.0D, this.m_20189_() + d1 * (double)i / 4.0D, -d5, -d6 + 0.2D, -d1);
+               this.m_9236_().addParticle(ParticleTypes.f_123797_, this.getBigX().add(d5 * (double)i / 4.0D), this.getBigY().add(d6 * (double)i / 4.0D), this.getBigZ().add(d1 * (double)i / 4.0D), -d5, -d6 + 0.2D, -d1);
             }
          }
 
-         double d7 = this.m_20185_() + d5;
-         double d2 = this.m_20186_() + d6;
-         double d3 = this.m_20189_() + d1;
+         me.alphamode.mclong.math.BigDecimal d7 = this.getBigX().add(d5);
+         me.alphamode.mclong.math.BigDecimal d2 = this.getBigY().add(d6);
+         me.alphamode.mclong.math.BigDecimal d3 = this.getBigZ().add(d1);
          double d4 = vec3.m_165924_();
          if (flag) {
             this.m_146922_((float)(Mth.m_14136_(-d5, -d1) * (double)(180F / (float)Math.PI)));
@@ -226,7 +_,7 @@
          if (this.m_20069_()) {
             for(int j = 0; j < 4; ++j) {
                float f2 = 0.25F;
-               this.m_9236_().m_7106_(ParticleTypes.f_123795_, d7 - d5 * 0.25D, d2 - d6 * 0.25D, d3 - d1 * 0.25D, d5, d6, d1);
+               this.m_9236_().addParticle(ParticleTypes.f_123795_, d7.subtract(d5 * 0.25D), d2.subtract(d6 * 0.25D), d3.subtract(d1 * 0.25D), d5, d6, d1);
             }
 
             f = this.m_6882_();
@@ -235,10 +_,10 @@
          this.m_20256_(vec3.m_82490_((double)f));
          if (!this.m_20068_() && !flag) {
             Vec3 vec34 = this.m_20184_();
-            this.m_20334_(vec34.f_82479_, vec34.f_82480_ - (double)0.05F, vec34.f_82481_);
+            this.m_20334_(vec34.m_7096_(), vec34.m_7098_() - (double)0.05F, vec34.m_7094_());
          }
 
-         this.m_6034_(d7, d2, d3);
+         this.setPos(d7, d2, d3);
          this.m_20101_();
       }
    }
@@ -340,7 +_,7 @@
                double d0 = Math.max(0.0D, 1.0D - livingentity.m_21133_(Attributes.f_22278_));
                Vec3 vec3 = this.m_20184_().m_82542_(1.0D, 0.0D, 1.0D).m_82541_().m_82490_((double)this.f_36699_ * 0.6D * d0);
                if (vec3.m_82556_() > 0.0D) {
-                  livingentity.m_5997_(vec3.f_82479_, 0.1D, vec3.f_82481_);
+                  livingentity.m_5997_(vec3.m_7096_(), 0.1D, vec3.m_7094_());
                }
             }
 
@@ -391,10 +_,10 @@
    protected void m_8060_(BlockHitResult p_36755_) {
       this.f_36696_ = this.m_9236_().m_8055_(p_36755_.m_82425_());
       super.m_8060_(p_36755_);
-      Vec3 vec3 = p_36755_.m_82450_().m_82492_(this.m_20185_(), this.m_20186_(), this.m_20189_());
+      Vec3 vec3 = p_36755_.m_82450_().subtract(this.getBigX(), this.getBigY(), this.getBigZ());
       this.m_20256_(vec3);
       Vec3 vec31 = vec3.m_82541_().m_82490_((double)0.05F);
-      this.m_20343_(this.m_20185_() - vec31.f_82479_, this.m_20186_() - vec31.f_82480_, this.m_20189_() - vec31.f_82481_);
+      this.setPosRaw(this.getBigX().subtract(vec31.getX()), this.getBigY().subtract(vec31.getY()), this.getBigZ().subtract(vec31.getZ()));
       this.m_5496_(this.m_36784_(), 1.0F, 1.2F / (this.f_19796_.m_188501_() * 0.2F + 0.9F));
       this.f_36703_ = true;
       this.f_36706_ = 7;
