--- a/net/minecraft/world/entity/projectile/Arrow.java
+++ b/net/minecraft/world/entity/projectile/Arrow.java
@@ -118,7 +_,7 @@
          double d2 = (double)(i >> 0 & 255) / 255.0D;
 
          for(int j = 0; j < p_36877_; ++j) {
-            this.m_9236_().m_7106_(ParticleTypes.f_123811_, this.m_20208_(0.5D), this.m_20187_(), this.m_20262_(0.5D), d0, d1, d2);
+            this.m_9236_().addParticle(ParticleTypes.f_123811_, this.getRandomX(0.5D), this.getRandomY(), this.getRandomZ(0.5D), d0, d1, d2);
          }
 
       }
@@ -215,7 +_,7 @@
             double d2 = (double)(i >> 0 & 255) / 255.0D;
 
             for(int j = 0; j < 20; ++j) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123811_, this.m_20208_(0.5D), this.m_20187_(), this.m_20262_(0.5D), d0, d1, d2);
+               this.m_9236_().addParticle(ParticleTypes.f_123811_, this.getRandomX(0.5D), this.getRandomY(), this.getRandomZ(0.5D), d0, d1, d2);
             }
          }
       } else {
