--- a/net/minecraft/world/entity/projectile/DragonFireball.java
+++ b/net/minecraft/world/entity/projectile/DragonFireball.java
@@ -30,7 +_,7 @@
       if (p_36913_.m_6662_() != HitResult.Type.ENTITY || !this.m_150171_(((EntityHitResult)p_36913_).m_82443_())) {
          if (!this.m_9236_().f_46443_) {
             List<LivingEntity> list = this.m_9236_().m_45976_(LivingEntity.class, this.m_20191_().m_82377_(4.0D, 2.0D, 4.0D));
-            AreaEffectCloud areaeffectcloud = new AreaEffectCloud(this.m_9236_(), this.m_20185_(), this.m_20186_(), this.m_20189_());
+            AreaEffectCloud areaeffectcloud = new AreaEffectCloud(this.m_9236_(), this.getBigX(), this.getBigY(), this.getBigZ());
             Entity entity = this.m_19749_();
             if (entity instanceof LivingEntity) {
                areaeffectcloud.m_19718_((LivingEntity)entity);
