--- a/net/minecraft/world/entity/projectile/EvokerFangs.java
+++ b/net/minecraft/world/entity/projectile/EvokerFangs.java
@@ -29,12 +_,12 @@
       super(p_36923_, p_36924_);
    }
 
-   public EvokerFangs(Level p_36926_, double p_36927_, double p_36928_, double p_36929_, float p_36930_, int p_36931_, LivingEntity p_36932_) {
-      this(EntityType.f_20569_, p_36926_);
-      this.f_36916_ = p_36931_;
-      this.m_36938_(p_36932_);
-      this.m_146922_(p_36930_ * (180F / (float)Math.PI));
-      this.m_6034_(p_36927_, p_36928_, p_36929_);
+   public EvokerFangs(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pYRot, int pWarmupDelay, LivingEntity pOwner) {
+      this(EntityType.f_20569_, pLevel);
+      this.f_36916_ = pWarmupDelay;
+      this.m_36938_(pOwner);
+      this.m_146922_(pYRot * (180F / (float)Math.PI));
+      this.setPos(pX, pY, pZ);
    }
 
    protected void m_8097_() {
@@ -80,13 +_,13 @@
             --this.f_36918_;
             if (this.f_36918_ == 14) {
                for(int i = 0; i < 12; ++i) {
-                  double d0 = this.m_20185_() + (this.f_19796_.m_188500_() * 2.0D - 1.0D) * (double)this.m_20205_() * 0.5D;
-                  double d1 = this.m_20186_() + 0.05D + this.f_19796_.m_188500_();
-                  double d2 = this.m_20189_() + (this.f_19796_.m_188500_() * 2.0D - 1.0D) * (double)this.m_20205_() * 0.5D;
+                  me.alphamode.mclong.math.BigDecimal d0 = this.getBigX().add((this.f_19796_.m_188500_() * 2.0D - 1.0D) * (double)this.m_20205_() * 0.5D);
+                  me.alphamode.mclong.math.BigDecimal d1 = this.getBigY().add(0.05D + this.f_19796_.m_188500_());
+                  me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ().add((this.f_19796_.m_188500_() * 2.0D - 1.0D) * (double)this.m_20205_() * 0.5D);
                   double d3 = (this.f_19796_.m_188500_() * 2.0D - 1.0D) * 0.3D;
                   double d4 = 0.3D + this.f_19796_.m_188500_() * 0.3D;
                   double d5 = (this.f_19796_.m_188500_() * 2.0D - 1.0D) * 0.3D;
-                  this.m_9236_().m_7106_(ParticleTypes.f_123797_, d0, d1 + 1.0D, d2, d3, d4, d5);
+                  this.m_9236_().addParticle(ParticleTypes.f_123797_, d0, d1.add(me.alphamode.mclong.math.BigDecimal.ONE), d2, d3, d4, d5);
                }
             }
          }
@@ -130,7 +_,7 @@
       if (p_36935_ == 4) {
          this.f_36919_ = true;
          if (!this.m_20067_()) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_11865_, this.m_5720_(), 1.0F, this.f_19796_.m_188501_() * 0.2F + 0.85F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_11865_, this.m_5720_(), 1.0F, this.f_19796_.m_188501_() * 0.2F + 0.85F, false);
          }
       }
 
