--- a/net/minecraft/world/entity/projectile/EyeOfEnder.java
+++ b/net/minecraft/world/entity/projectile/EyeOfEnder.java
@@ -64,9 +_,9 @@
    }
 
    public void m_36967_(BlockPos p_36968_) {
-      double d0 = (double)p_36968_.m_123341_();
-      int i = p_36968_.m_123342_();
-      double d1 = (double)p_36968_.m_123343_();
+      double d0 = (double)p_36968_.getX();
+      long i = p_36968_.getY();
+      double d1 = (double)p_36968_.getZ();
       double d2 = d0 - this.m_20185_();
       double d3 = d1 - this.m_20189_();
       double d4 = Math.sqrt(d2 * d2 + d3 * d3);
@@ -99,19 +_,19 @@
    public void m_8119_() {
       super.m_8119_();
       Vec3 vec3 = this.m_20184_();
-      double d0 = this.m_20185_() + vec3.f_82479_;
-      double d1 = this.m_20186_() + vec3.f_82480_;
-      double d2 = this.m_20189_() + vec3.f_82481_;
+      double d0 = this.m_20185_() + vec3.m_7096_();
+      double d1 = this.m_20186_() + vec3.m_7098_();
+      double d2 = this.m_20189_() + vec3.m_7094_();
       double d3 = vec3.m_165924_();
-      this.m_146926_(Projectile.m_37273_(this.f_19860_, (float)(Mth.m_14136_(vec3.f_82480_, d3) * (double)(180F / (float)Math.PI))));
-      this.m_146922_(Projectile.m_37273_(this.f_19859_, (float)(Mth.m_14136_(vec3.f_82479_, vec3.f_82481_) * (double)(180F / (float)Math.PI))));
+      this.m_146926_(Projectile.m_37273_(this.f_19860_, (float)(Mth.m_14136_(vec3.m_7098_(), d3) * (double)(180F / (float)Math.PI))));
+      this.m_146922_(Projectile.m_37273_(this.f_19859_, (float)(Mth.m_14136_(vec3.m_7096_(), vec3.m_7094_()) * (double)(180F / (float)Math.PI))));
       if (!this.m_9236_().f_46443_) {
          double d4 = this.f_36950_ - d0;
          double d5 = this.f_36952_ - d2;
          float f = (float)Math.sqrt(d4 * d4 + d5 * d5);
          float f1 = (float)Mth.m_14136_(d5, d4);
          double d6 = Mth.m_14139_(0.0025D, d3, (double)f);
-         double d7 = vec3.f_82480_;
+         double d7 = vec3.m_7098_();
          if (f < 1.0F) {
             d6 *= 0.8D;
             d7 *= 0.8D;
@@ -125,10 +_,10 @@
       float f2 = 0.25F;
       if (this.m_20069_()) {
          for(int i = 0; i < 4; ++i) {
-            this.m_9236_().m_7106_(ParticleTypes.f_123795_, d0 - vec3.f_82479_ * 0.25D, d1 - vec3.f_82480_ * 0.25D, d2 - vec3.f_82481_ * 0.25D, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+            this.m_9236_().m_7106_(ParticleTypes.f_123795_, d0 - vec3.m_7096_() * 0.25D, d1 - vec3.m_7098_() * 0.25D, d2 - vec3.m_7094_() * 0.25D, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
          }
       } else {
-         this.m_9236_().m_7106_(ParticleTypes.f_123760_, d0 - vec3.f_82479_ * 0.25D + this.f_19796_.m_188500_() * 0.6D - 0.3D, d1 - vec3.f_82480_ * 0.25D - 0.5D, d2 - vec3.f_82481_ * 0.25D + this.f_19796_.m_188500_() * 0.6D - 0.3D, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+         this.m_9236_().m_7106_(ParticleTypes.f_123760_, d0 - vec3.m_7096_() * 0.25D + this.f_19796_.m_188500_() * 0.6D - 0.3D, d1 - vec3.m_7098_() * 0.25D - 0.5D, d2 - vec3.m_7094_() * 0.25D + this.f_19796_.m_188500_() * 0.6D - 0.3D, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
       }
 
       if (!this.m_9236_().f_46443_) {
