--- a/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ b/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -39,39 +_,39 @@
       super(p_37027_, p_37028_);
    }
 
-   public FireworkRocketEntity(Level p_37030_, double p_37031_, double p_37032_, double p_37033_, ItemStack p_37034_) {
-      super(EntityType.f_20451_, p_37030_);
+   public FireworkRocketEntity(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, ItemStack pStack) {
+      super(EntityType.f_20451_, pLevel);
       this.f_37022_ = 0;
-      this.m_6034_(p_37031_, p_37032_, p_37033_);
+      this.setPos(pX, pY, pZ);
       int i = 1;
-      if (!p_37034_.m_41619_() && p_37034_.m_41782_()) {
-         this.f_19804_.m_135381_(f_37019_, p_37034_.m_41777_());
-         i += p_37034_.m_41698_("Fireworks").m_128445_("Flight");
+      if (!pStack.m_41619_() && pStack.m_41782_()) {
+         this.f_19804_.m_135381_(f_37019_, pStack.m_41777_());
+         i += pStack.m_41698_("Fireworks").m_128445_("Flight");
       }
 
       this.m_20334_(this.f_19796_.m_216328_(0.0D, 0.002297D), 0.05D, this.f_19796_.m_216328_(0.0D, 0.002297D));
       this.f_37023_ = 10 * i + this.f_19796_.m_188503_(6) + this.f_19796_.m_188503_(7);
    }
 
-   public FireworkRocketEntity(Level p_37036_, @Nullable Entity p_37037_, double p_37038_, double p_37039_, double p_37040_, ItemStack p_37041_) {
-      this(p_37036_, p_37038_, p_37039_, p_37040_, p_37041_);
-      this.m_5602_(p_37037_);
+   public FireworkRocketEntity(Level pLevel, @Nullable Entity pShooter, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, ItemStack pStack) {
+      this(pLevel, pX, pY, pZ, pStack);
+      this.m_5602_(pShooter);
    }
 
    public FireworkRocketEntity(Level p_37058_, ItemStack p_37059_, LivingEntity p_37060_) {
-      this(p_37058_, p_37060_, p_37060_.m_20185_(), p_37060_.m_20186_(), p_37060_.m_20189_(), p_37059_);
+      this(p_37058_, p_37060_, p_37060_.getBigX(), p_37060_.getBigY(), p_37060_.getBigZ(), p_37059_);
       this.f_19804_.m_135381_(f_37020_, OptionalInt.of(p_37060_.m_19879_()));
       this.f_37024_ = p_37060_;
    }
 
-   public FireworkRocketEntity(Level p_37043_, ItemStack p_37044_, double p_37045_, double p_37046_, double p_37047_, boolean p_37048_) {
-      this(p_37043_, p_37045_, p_37046_, p_37047_, p_37044_);
-      this.f_19804_.m_135381_(f_37021_, p_37048_);
+   public FireworkRocketEntity(Level pLevel, ItemStack pStack, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, boolean pShotAtAngle) {
+      this(pLevel, pX, pY, pZ, pStack);
+      this.f_19804_.m_135381_(f_37021_, pShotAtAngle);
    }
 
-   public FireworkRocketEntity(Level p_37050_, ItemStack p_37051_, Entity p_37052_, double p_37053_, double p_37054_, double p_37055_, boolean p_37056_) {
-      this(p_37050_, p_37051_, p_37053_, p_37054_, p_37055_, p_37056_);
-      this.m_5602_(p_37052_);
+   public FireworkRocketEntity(Level pLevel, ItemStack pStack, Entity pShooter, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, boolean pShotAtAngle) {
+      this(pLevel, pStack, pX, pY, pZ, pShotAtAngle);
+      this.m_5602_(pShooter);
    }
 
    protected void m_8097_() {
@@ -84,8 +_,8 @@
       return p_37065_ < 4096.0D && !this.m_37088_();
    }
 
-   public boolean m_6000_(double p_37083_, double p_37084_, double p_37085_) {
-      return super.m_6000_(p_37083_, p_37084_, p_37085_) && !this.m_37088_();
+   public boolean shouldRender(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      return super.shouldRender(pX, pY, pZ) && !this.m_37088_();
    }
 
    public void m_8119_() {
@@ -108,13 +_,13 @@
                double d0 = 1.5D;
                double d1 = 0.1D;
                Vec3 vec32 = this.f_37024_.m_20184_();
-               this.f_37024_.m_20256_(vec32.m_82520_(vec31.f_82479_ * 0.1D + (vec31.f_82479_ * 1.5D - vec32.f_82479_) * 0.5D, vec31.f_82480_ * 0.1D + (vec31.f_82480_ * 1.5D - vec32.f_82480_) * 0.5D, vec31.f_82481_ * 0.1D + (vec31.f_82481_ * 1.5D - vec32.f_82481_) * 0.5D));
+               this.f_37024_.m_20256_(vec32.m_82520_(vec31.m_7096_() * 0.1D + (vec31.m_7096_() * 1.5D - vec32.m_7096_()) * 0.5D, vec31.m_7098_() * 0.1D + (vec31.m_7098_() * 1.5D - vec32.m_7098_()) * 0.5D, vec31.m_7094_() * 0.1D + (vec31.m_7094_() * 1.5D - vec32.m_7094_()) * 0.5D));
                vec3 = this.f_37024_.m_204034_(Items.f_42688_);
             } else {
                vec3 = Vec3.f_82478_;
             }
 
-            this.m_6034_(this.f_37024_.m_20185_() + vec3.f_82479_, this.f_37024_.m_20186_() + vec3.f_82480_, this.f_37024_.m_20189_() + vec3.f_82481_);
+            this.setPos(this.f_37024_.getBigX().add(vec3.getX()), this.f_37024_.getBigY().add(vec3.getY()), this.f_37024_.getBigZ().add(vec3.getZ()));
             this.m_20256_(this.f_37024_.m_20184_());
          }
       } else {
@@ -136,12 +_,12 @@
 
       this.m_37283_();
       if (this.f_37022_ == 0 && !this.m_20067_()) {
-         this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_11932_, SoundSource.AMBIENT, 3.0F, 1.0F);
+         this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_11932_, SoundSource.AMBIENT, 3.0F, 1.0F);
       }
 
       ++this.f_37022_;
       if (this.m_9236_().f_46443_ && this.f_37022_ % 2 < 2) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123815_, this.m_20185_(), this.m_20186_(), this.m_20189_(), this.f_19796_.m_188583_() * 0.05D, -this.m_20184_().f_82480_ * 0.5D, this.f_19796_.m_188583_() * 0.05D);
+         this.m_9236_().addParticle(ParticleTypes.f_123815_, this.getBigX(), this.getBigY(), this.getBigZ(), this.f_19796_.m_188583_() * 0.05D, -this.m_20184_().m_7098_() * 0.5D, this.f_19796_.m_188583_() * 0.05D);
       }
 
       if (!this.m_9236_().f_46443_ && this.f_37022_ > this.f_37023_) {
@@ -233,13 +_,13 @@
       if (p_37063_ == 17 && this.m_9236_().f_46443_) {
          if (!this.m_37086_()) {
             for(int i = 0; i < this.f_19796_.m_188503_(3) + 2; ++i) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123759_, this.m_20185_(), this.m_20186_(), this.m_20189_(), this.f_19796_.m_188583_() * 0.05D, 0.005D, this.f_19796_.m_188583_() * 0.05D);
+               this.m_9236_().addParticle(ParticleTypes.f_123759_, this.getBigX(), this.getBigY(), this.getBigZ(), this.f_19796_.m_188583_() * 0.05D, 0.005D, this.f_19796_.m_188583_() * 0.05D);
             }
          } else {
             ItemStack itemstack = this.f_19804_.m_135370_(f_37019_);
             CompoundTag compoundtag = itemstack.m_41619_() ? null : itemstack.m_41737_("Fireworks");
             Vec3 vec3 = this.m_20184_();
-            this.m_9236_().m_7228_(this.m_20185_(), this.m_20186_(), this.m_20189_(), vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, compoundtag);
+            this.m_9236_().createFireworks(this.getBigX(), this.getBigY(), this.getBigZ(), vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), compoundtag);
          }
       }
 
