--- a/net/minecraft/world/entity/projectile/FishingHook.java
+++ b/net/minecraft/world/entity/projectile/FishingHook.java
@@ -93,8 +_,8 @@
       double d3 = vec3.m_82553_();
       vec3 = vec3.m_82542_(0.6D / d3 + this.f_19796_.m_216328_(0.5D, 0.0103365D), 0.6D / d3 + this.f_19796_.m_216328_(0.5D, 0.0103365D), 0.6D / d3 + this.f_19796_.m_216328_(0.5D, 0.0103365D));
       this.m_20256_(vec3);
-      this.m_146922_((float)(Mth.m_14136_(vec3.f_82479_, vec3.f_82481_) * (double)(180F / (float)Math.PI)));
-      this.m_146926_((float)(Mth.m_14136_(vec3.f_82480_, vec3.m_165924_()) * (double)(180F / (float)Math.PI)));
+      this.m_146922_((float)(Mth.m_14136_(vec3.m_7096_(), vec3.m_7094_()) * (double)(180F / (float)Math.PI)));
+      this.m_146926_((float)(Mth.m_14136_(vec3.m_7098_(), vec3.m_165924_()) * (double)(180F / (float)Math.PI)));
       this.f_19859_ = this.m_146908_();
       this.f_19860_ = this.m_146909_();
    }
@@ -113,7 +_,7 @@
       if (f_37102_.equals(p_37153_)) {
          this.f_37099_ = this.m_20088_().m_135370_(f_37102_);
          if (this.f_37099_) {
-            this.m_20334_(this.m_20184_().f_82479_, (double)(-0.4F * Mth.m_216267_(this.f_37098_, 0.6F, 1.0F)), this.m_20184_().f_82481_);
+            this.m_20334_(this.m_20184_().m_7096_(), (double)(-0.4F * Mth.m_216267_(this.f_37098_, 0.6F, 1.0F)), this.m_20184_().m_7094_());
          }
       }
 
@@ -125,7 +_,7 @@
       return p_37125_ < 4096.0D;
    }
 
-   public void m_6453_(double p_37127_, double p_37128_, double p_37129_, float p_37130_, float p_37131_, int p_37132_, boolean p_37133_) {
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal p_37127_, me.alphamode.mclong.math.BigDecimal p_37128_, me.alphamode.mclong.math.BigDecimal p_37129_, float p_37130_, float p_37131_, int p_37132_, boolean p_37133_) {
    }
 
    public void m_8119_() {
@@ -183,12 +_,12 @@
 
             if (this.f_37095_ == FishingHook.FishHookState.BOBBING) {
                Vec3 vec3 = this.m_20184_();
-               double d0 = this.m_20186_() + vec3.f_82480_ - (double)blockpos.m_123342_() - (double)f;
+               double d0 = this.m_20186_() + vec3.m_7098_() - (double)blockpos.getY() - (double)f;
                if (Math.abs(d0) < 0.01D) {
                   d0 += Math.signum(d0) * 0.1D;
                }
 
-               this.m_20334_(vec3.f_82479_ * 0.9D, vec3.f_82480_ - d0 * (double)this.f_19796_.m_188501_() * 0.2D, vec3.f_82481_ * 0.9D);
+               this.m_20334_(vec3.m_7096_() * 0.9D, vec3.m_7098_() - d0 * (double)this.f_19796_.m_188501_() * 0.2D, vec3.m_7094_() * 0.9D);
                if (this.f_37089_ <= 0 && this.f_37091_ <= 0) {
                   this.f_37093_ = true;
                } else {
@@ -269,7 +_,7 @@
    private void m_37145_(BlockPos p_37146_) {
       ServerLevel serverlevel = (ServerLevel)this.m_9236_();
       int i = 1;
-      BlockPos blockpos = p_37146_.m_7494_();
+      BlockPos blockpos = p_37146_.above();
       if (this.f_19796_.m_188501_() < 0.25F && this.m_9236_().m_46758_(blockpos)) {
          ++i;
       }
@@ -292,25 +_,25 @@
             float f = this.f_37092_ * ((float)Math.PI / 180F);
             float f1 = Mth.m_14031_(f);
             float f2 = Mth.m_14089_(f);
-            double d0 = this.m_20185_() + (double)(f1 * (float)this.f_37091_ * 0.1F);
-            double d1 = (double)((float)Mth.m_14107_(this.m_20186_()) + 1.0F);
-            double d2 = this.m_20189_() + (double)(f2 * (float)this.f_37091_ * 0.1F);
-            BlockState blockstate = serverlevel.m_8055_(BlockPos.m_274561_(d0, d1 - 1.0D, d2));
+            me.alphamode.mclong.math.BigDecimal d0 = this.getBigX().add((double)(f1 * (float)this.f_37091_ * 0.1F));
+            me.alphamode.mclong.math.BigDecimal d1 = Mth.lfloor(this.getBigY()).toBigDecimal().add(me.alphamode.mclong.math.BigDecimal.ONE);
+            me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ().add((double)(f2 * (float)this.f_37091_ * 0.1F));
+            BlockState blockstate = serverlevel.m_8055_(BlockPos.containing(d0, d1.subtract(me.alphamode.mclong.math.BigDecimal.ONE), d2));
             if (blockstate.m_60713_(Blocks.f_49990_)) {
                if (this.f_19796_.m_188501_() < 0.15F) {
-                  serverlevel.m_8767_(ParticleTypes.f_123795_, d0, d1 - (double)0.1F, d2, 1, (double)f1, 0.1D, (double)f2, 0.0D);
+                  serverlevel.sendParticles(ParticleTypes.f_123795_, d0, d1.subtract(me.alphamode.mclong.core.BigConstants.POINT_ONE), d2, 1, (double)f1, 0.1D, (double)f2, 0.0D);
                }
 
                float f3 = f1 * 0.04F;
                float f4 = f2 * 0.04F;
-               serverlevel.m_8767_(ParticleTypes.f_123816_, d0, d1, d2, 0, (double)f4, 0.01D, (double)(-f3), 1.0D);
-               serverlevel.m_8767_(ParticleTypes.f_123816_, d0, d1, d2, 0, (double)(-f4), 0.01D, (double)f3, 1.0D);
+               serverlevel.sendParticles(ParticleTypes.f_123816_, d0, d1, d2, 0, (double)f4, 0.01D, (double)(-f3), 1.0D);
+               serverlevel.sendParticles(ParticleTypes.f_123816_, d0, d1, d2, 0, (double)(-f4), 0.01D, (double)f3, 1.0D);
             }
          } else {
             this.m_5496_(SoundEvents.f_11940_, 0.25F, 1.0F + (this.f_19796_.m_188501_() - this.f_19796_.m_188501_()) * 0.4F);
-            double d3 = this.m_20186_() + 0.5D;
-            serverlevel.m_8767_(ParticleTypes.f_123795_, this.m_20185_(), d3, this.m_20189_(), (int)(1.0F + this.m_20205_() * 20.0F), (double)this.m_20205_(), 0.0D, (double)this.m_20205_(), (double)0.2F);
-            serverlevel.m_8767_(ParticleTypes.f_123816_, this.m_20185_(), d3, this.m_20189_(), (int)(1.0F + this.m_20205_() * 20.0F), (double)this.m_20205_(), 0.0D, (double)this.m_20205_(), (double)0.2F);
+            me.alphamode.mclong.math.BigDecimal d3 = this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB);
+            serverlevel.sendParticles(ParticleTypes.f_123795_, this.getBigX(), d3, this.getBigZ(), (int)(1.0F + this.m_20205_() * 20.0F), (double)this.m_20205_(), 0.0D, (double)this.m_20205_(), (double)0.2F);
+            serverlevel.sendParticles(ParticleTypes.f_123816_, this.getBigX(), d3, this.getBigZ(), (int)(1.0F + this.m_20205_() * 20.0F), (double)this.m_20205_(), 0.0D, (double)this.m_20205_(), (double)0.2F);
             this.f_37089_ = Mth.m_216271_(this.f_19796_, 20, 40);
             this.m_20088_().m_135381_(f_37102_, true);
          }
@@ -328,12 +_,12 @@
          if (this.f_19796_.m_188501_() < f5) {
             float f6 = Mth.m_216267_(this.f_19796_, 0.0F, 360.0F) * ((float)Math.PI / 180F);
             float f7 = Mth.m_216267_(this.f_19796_, 25.0F, 60.0F);
-            double d4 = this.m_20185_() + (double)(Mth.m_14031_(f6) * f7) * 0.1D;
-            double d5 = (double)((float)Mth.m_14107_(this.m_20186_()) + 1.0F);
-            double d6 = this.m_20189_() + (double)(Mth.m_14089_(f6) * f7) * 0.1D;
-            BlockState blockstate1 = serverlevel.m_8055_(BlockPos.m_274561_(d4, d5 - 1.0D, d6));
+            me.alphamode.mclong.math.BigDecimal d4 = this.getBigX().add((double)(Mth.m_14031_(f6) * f7) * 0.1D);
+            me.alphamode.mclong.math.BigDecimal d5 = Mth.lfloor(this.getBigY()).toBigDecimal().add(me.alphamode.mclong.math.BigDecimal.ONE);
+            me.alphamode.mclong.math.BigDecimal d6 = this.getBigZ().add((double)(Mth.m_14089_(f6) * f7) * 0.1D);
+            BlockState blockstate1 = serverlevel.m_8055_(BlockPos.containing(d4, d5.subtract(me.alphamode.mclong.math.BigDecimal.ONE), d6));
             if (blockstate1.m_60713_(Blocks.f_49990_)) {
-               serverlevel.m_8767_(ParticleTypes.f_123769_, d4, d5, d6, 2 + this.f_19796_.m_188503_(2), (double)0.1F, 0.0D, (double)0.1F, 0.0D);
+               serverlevel.sendParticles(ParticleTypes.f_123769_, d4, d5, d6, 2 + this.f_19796_.m_188503_(2), (double)0.1F, 0.0D, (double)0.1F, 0.0D);
             }
          }
 
@@ -352,7 +_,7 @@
       FishingHook.OpenWaterType fishinghook$openwatertype = FishingHook.OpenWaterType.INVALID;
 
       for(int i = -1; i <= 2; ++i) {
-         FishingHook.OpenWaterType fishinghook$openwatertype1 = this.m_37147_(p_37159_.m_7918_(-2, i, -2), p_37159_.m_7918_(2, i, 2));
+         FishingHook.OpenWaterType fishinghook$openwatertype1 = this.m_37147_(p_37159_.offset(-2, i, -2), p_37159_.offset(2, i, 2));
          switch (fishinghook$openwatertype1) {
             case INVALID:
                return false;
@@ -422,7 +_,7 @@
                double d3 = 0.1D;
                itementity.m_20334_(d0 * 0.1D, d1 * 0.1D + Math.sqrt(Math.sqrt(d0 * d0 + d1 * d1 + d2 * d2)) * 0.08D, d2 * 0.1D);
                this.m_9236_().m_7967_(itementity);
-               player.m_9236_().m_7967_(new ExperienceOrb(player.m_9236_(), player.m_20185_(), player.m_20186_() + 0.5D, player.m_20189_() + 0.5D, this.f_19796_.m_188503_(6) + 1));
+               player.m_9236_().m_7967_(new ExperienceOrb(player.m_9236_(), player.getBigX(), player.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), player.getBigZ().add(me.alphamode.mclong.core.BigConstants.AABB), this.f_19796_.m_188503_(6) + 1));
                if (itemstack.m_204117_(ItemTags.f_13156_)) {
                   player.m_36222_(Stats.f_12939_, 1);
                }
