--- a/net/minecraft/world/entity/projectile/LlamaSpit.java
+++ b/net/minecraft/world/entity/projectile/LlamaSpit.java
@@ -22,7 +_,7 @@
    public LlamaSpit(Level p_37235_, Llama p_37236_) {
       this(EntityType.f_20467_, p_37235_);
       this.m_5602_(p_37236_);
-      this.m_6034_(p_37236_.m_20185_() - (double)(p_37236_.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14031_(p_37236_.f_20883_ * ((float)Math.PI / 180F)), p_37236_.m_20188_() - (double)0.1F, p_37236_.m_20189_() + (double)(p_37236_.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14089_(p_37236_.f_20883_ * ((float)Math.PI / 180F)));
+      this.setPos(p_37236_.getBigX().subtract((double)(p_37236_.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14031_(p_37236_.f_20883_ * ((float)Math.PI / 180F))), p_37236_.getEyeBigY().subtract(me.alphamode.mclong.core.BigConstants.POINT_ONE), p_37236_.getBigZ().add((double)(p_37236_.m_20205_() + 1.0F) * 0.5D * (double)Mth.m_14089_(p_37236_.f_20883_ * ((float)Math.PI / 180F))));
    }
 
    public void m_8119_() {
@@ -30,9 +_,9 @@
       Vec3 vec3 = this.m_20184_();
       HitResult hitresult = ProjectileUtil.m_278158_(this, this::m_5603_);
       this.m_6532_(hitresult);
-      double d0 = this.m_20185_() + vec3.f_82479_;
-      double d1 = this.m_20186_() + vec3.f_82480_;
-      double d2 = this.m_20189_() + vec3.f_82481_;
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX().add(vec3.getX());
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigY().add(vec3.getY());
+      me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ().add(vec3.getZ());
       this.m_37283_();
       float f = 0.99F;
       float f1 = 0.06F;
@@ -46,7 +_,7 @@
             this.m_20256_(this.m_20184_().m_82520_(0.0D, (double)-0.06F, 0.0D));
          }
 
-         this.m_6034_(d0, d1, d2);
+         this.setPos(d0, d1, d2);
       }
    }
 
@@ -78,7 +_,7 @@
 
       for(int i = 0; i < 7; ++i) {
          double d3 = 0.4D + 0.1D * (double)i;
-         this.m_9236_().m_7106_(ParticleTypes.f_123764_, this.m_20185_(), this.m_20186_(), this.m_20189_(), d0 * d3, d1, d2 * d3);
+         this.m_9236_().addParticle(ParticleTypes.f_123764_, this.getBigX(), this.getBigY(), this.getBigZ(), d0 * d3, d1, d2 * d3);
       }
 
       this.m_20334_(d0, d1, d2);
