--- a/net/minecraft/world/entity/projectile/LlamaSpit.java
+++ b/net/minecraft/world/entity/projectile/LlamaSpit.java
@@ -30,9 +_,9 @@
       Vec3 vec3 = this.m_20184_();
       HitResult hitresult = ProjectileUtil.m_278158_(this, this::m_5603_);
       this.m_6532_(hitresult);
-      double d0 = this.m_20185_() + vec3.f_82479_;
-      double d1 = this.m_20186_() + vec3.f_82480_;
-      double d2 = this.m_20189_() + vec3.f_82481_;
+      double d0 = this.m_20185_() + vec3.m_7096_();
+      double d1 = this.m_20186_() + vec3.m_7098_();
+      double d2 = this.m_20189_() + vec3.m_7094_();
       this.m_37283_();
       float f = 0.99F;
       float f1 = 0.06F;
@@ -78,7 +_,7 @@
 
       for(int i = 0; i < 7; ++i) {
          double d3 = 0.4D + 0.1D * (double)i;
-         this.m_9236_().m_7106_(ParticleTypes.f_123764_, this.m_20185_(), this.m_20186_(), this.m_20189_(), d0 * d3, d1, d2 * d3);
+         this.m_9236_().addParticle(ParticleTypes.f_123764_, this.getBigX(), this.getBigY(), this.getBigZ(), d0 * d3, d1, d2 * d3);
       }
 
       this.m_20334_(d0, d1, d2);
