--- a/net/minecraft/world/entity/projectile/Projectile.java
+++ b/net/minecraft/world/entity/projectile/Projectile.java
@@ -117,8 +_,8 @@
       Vec3 vec3 = (new Vec3(p_37266_, p_37267_, p_37268_)).m_82541_().m_82520_(this.f_19796_.m_216328_(0.0D, 0.0172275D * (double)p_37270_), this.f_19796_.m_216328_(0.0D, 0.0172275D * (double)p_37270_), this.f_19796_.m_216328_(0.0D, 0.0172275D * (double)p_37270_)).m_82490_((double)p_37269_);
       this.m_20256_(vec3);
       double d0 = vec3.m_165924_();
-      this.m_146922_((float)(Mth.m_14136_(vec3.f_82479_, vec3.f_82481_) * (double)(180F / (float)Math.PI)));
-      this.m_146926_((float)(Mth.m_14136_(vec3.f_82480_, d0) * (double)(180F / (float)Math.PI)));
+      this.m_146922_((float)(Mth.m_14136_(vec3.m_7096_(), vec3.m_7094_()) * (double)(180F / (float)Math.PI)));
+      this.m_146926_((float)(Mth.m_14136_(vec3.m_7098_(), d0) * (double)(180F / (float)Math.PI)));
       this.f_19859_ = this.m_146908_();
       this.f_19860_ = this.m_146909_();
    }
@@ -129,7 +_,7 @@
       float f2 = Mth.m_14089_(p_37254_ * ((float)Math.PI / 180F)) * Mth.m_14089_(p_37253_ * ((float)Math.PI / 180F));
       this.m_6686_((double)f, (double)f1, (double)f2, p_37256_, p_37257_);
       Vec3 vec3 = p_37252_.m_20184_();
-      this.m_20256_(this.m_20184_().m_82520_(vec3.f_82479_, p_37252_.m_20096_() ? 0.0D : vec3.f_82480_, vec3.f_82481_));
+      this.m_20256_(this.m_20184_().m_82520_(vec3.m_7096_(), p_37252_.m_20096_() ? 0.0D : vec3.m_7098_(), vec3.m_7094_()));
    }
 
    protected void m_6532_(HitResult p_37260_) {
@@ -179,8 +_,8 @@
    protected void m_37283_() {
       Vec3 vec3 = this.m_20184_();
       double d0 = vec3.m_165924_();
-      this.m_146926_(m_37273_(this.f_19860_, (float)(Mth.m_14136_(vec3.f_82480_, d0) * (double)(180F / (float)Math.PI))));
-      this.m_146922_(m_37273_(this.f_19859_, (float)(Mth.m_14136_(vec3.f_82479_, vec3.f_82481_) * (double)(180F / (float)Math.PI))));
+      this.m_146926_(m_37273_(this.f_19860_, (float)(Mth.m_14136_(vec3.m_7098_(), d0) * (double)(180F / (float)Math.PI))));
+      this.m_146922_(m_37273_(this.f_19859_, (float)(Mth.m_14136_(vec3.m_7096_(), vec3.m_7094_()) * (double)(180F / (float)Math.PI))));
    }
 
    protected static float m_37273_(float p_37274_, float p_37275_) {
