--- a/net/minecraft/world/entity/projectile/ProjectileUtil.java
+++ b/net/minecraft/world/entity/projectile/ProjectileUtil.java
@@ -114,8 +_,8 @@
       Vec3 vec3 = p_37285_.m_20184_();
       if (vec3.m_82556_() != 0.0D) {
          double d0 = vec3.m_165924_();
-         p_37285_.m_146922_((float)(Mth.m_14136_(vec3.f_82481_, vec3.f_82479_) * (double)(180F / (float)Math.PI)) + 90.0F);
-         p_37285_.m_146926_((float)(Mth.m_14136_(d0, vec3.f_82480_) * (double)(180F / (float)Math.PI)) - 90.0F);
+         p_37285_.m_146922_((float)(Mth.m_14136_(vec3.m_7094_(), vec3.m_7096_()) * (double)(180F / (float)Math.PI)) + 90.0F);
+         p_37285_.m_146926_((float)(Mth.m_14136_(d0, vec3.m_7098_()) * (double)(180F / (float)Math.PI)) - 90.0F);
 
          while(p_37285_.m_146909_() - p_37285_.f_19860_ < -180.0F) {
             p_37285_.f_19860_ -= 360.0F;
