--- a/net/minecraft/world/entity/projectile/Snowball.java
+++ b/net/minecraft/world/entity/projectile/Snowball.java
@@ -23,8 +_,8 @@
       super(EntityType.f_20477_, p_37400_, p_37399_);
    }
 
-   public Snowball(Level p_37394_, double p_37395_, double p_37396_, double p_37397_) {
-      super(EntityType.f_20477_, p_37395_, p_37396_, p_37397_, p_37394_);
+   public Snowball(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20477_, pX, pY, pZ, pLevel);
    }
 
    protected Item m_7881_() {
@@ -41,7 +_,7 @@
          ParticleOptions particleoptions = this.m_37408_();
 
          for(int i = 0; i < 8; ++i) {
-            this.m_9236_().m_7106_(particleoptions, this.m_20185_(), this.m_20186_(), this.m_20189_(), 0.0D, 0.0D, 0.0D);
+            this.m_9236_().addParticle(particleoptions, this.getBigX(), this.getBigY(), this.getBigZ(), 0.0D, 0.0D, 0.0D);
          }
       }
 
