--- a/net/minecraft/world/entity/projectile/Snowball.java
+++ b/net/minecraft/world/entity/projectile/Snowball.java
@@ -41,7 +_,7 @@
          ParticleOptions particleoptions = this.m_37408_();
 
          for(int i = 0; i < 8; ++i) {
-            this.m_9236_().m_7106_(particleoptions, this.m_20185_(), this.m_20186_(), this.m_20189_(), 0.0D, 0.0D, 0.0D);
+            this.m_9236_().addParticle(particleoptions, this.getBigX(), this.getBigY(), this.getBigZ(), 0.0D, 0.0D, 0.0D);
          }
       }
 
