--- a/net/minecraft/world/entity/projectile/SpectralArrow.java
+++ b/net/minecraft/world/entity/projectile/SpectralArrow.java
@@ -28,7 +_,7 @@
    public void m_8119_() {
       super.m_8119_();
       if (this.m_9236_().f_46443_ && !this.f_36703_) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123751_, this.m_20185_(), this.m_20186_(), this.m_20189_(), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123751_, this.getBigX(), this.getBigY(), this.getBigZ(), 0.0D, 0.0D, 0.0D);
       }
 
    }
