--- a/net/minecraft/world/entity/projectile/SpectralArrow.java
+++ b/net/minecraft/world/entity/projectile/SpectralArrow.java
@@ -21,14 +_,14 @@
       super(EntityType.f_20478_, p_37420_, p_37419_);
    }
 
-   public SpectralArrow(Level p_37414_, double p_37415_, double p_37416_, double p_37417_) {
-      super(EntityType.f_20478_, p_37415_, p_37416_, p_37417_, p_37414_);
+   public SpectralArrow(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20478_, pX, pY, pZ, pLevel);
    }
 
    public void m_8119_() {
       super.m_8119_();
       if (this.m_9236_().f_46443_ && !this.f_36703_) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123751_, this.m_20185_(), this.m_20186_(), this.m_20189_(), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123751_, this.getBigX(), this.getBigY(), this.getBigZ(), 0.0D, 0.0D, 0.0D);
       }
 
    }
