--- a/net/minecraft/world/entity/projectile/ThrownTrident.java
+++ b/net/minecraft/world/entity/projectile/ThrownTrident.java
@@ -137,7 +_,7 @@
          BlockPos blockpos = entity.m_20183_();
          if (this.f_19853_.m_45527_(blockpos)) {
             LightningBolt lightningbolt = EntityType.f_20465_.m_20615_(this.f_19853_);
-            lightningbolt.m_20219_(Vec3.m_82539_(blockpos));
+            lightningbolt.m_20219_(Vec3.atBottomCenterOf(blockpos));
             lightningbolt.m_20879_(entity1 instanceof ServerPlayer ? (ServerPlayer)entity1 : null);
             this.f_19853_.m_7967_(lightningbolt);
             soundevent = SoundEvents.f_12521_;
