--- a/net/minecraft/world/entity/projectile/ThrownExperienceBottle.java
+++ b/net/minecraft/world/entity/projectile/ThrownExperienceBottle.java
@@ -20,8 +_,8 @@
       super(EntityType.f_20485_, p_37519_, p_37518_);
    }
 
-   public ThrownExperienceBottle(Level p_37513_, double p_37514_, double p_37515_, double p_37516_) {
-      super(EntityType.f_20485_, p_37514_, p_37515_, p_37516_, p_37513_);
+   public ThrownExperienceBottle(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20485_, pX, pY, pZ, pLevel);
    }
 
    protected Item m_7881_() {
