--- a/net/minecraft/world/entity/projectile/ThrowableItemProjectile.java
+++ b/net/minecraft/world/entity/projectile/ThrowableItemProjectile.java
@@ -17,8 +_,8 @@
       super(p_37442_, p_37443_);
    }
 
-   public ThrowableItemProjectile(EntityType<? extends ThrowableItemProjectile> p_37432_, double p_37433_, double p_37434_, double p_37435_, Level p_37436_) {
-      super(p_37432_, p_37433_, p_37434_, p_37435_, p_37436_);
+   public ThrowableItemProjectile(EntityType<? extends ThrowableItemProjectile> pEntityType, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, Level pLevel) {
+      super(pEntityType, pX, pY, pZ, pLevel);
    }
 
    public ThrowableItemProjectile(EntityType<? extends ThrowableItemProjectile> p_37438_, LivingEntity p_37439_, Level p_37440_) {
