--- a/net/minecraft/world/entity/projectile/ThrownPotion.java
+++ b/net/minecraft/world/entity/projectile/ThrownPotion.java
@@ -45,8 +_,8 @@
       super(EntityType.f_20486_, p_37536_, p_37535_);
    }
 
-   public ThrownPotion(Level p_37530_, double p_37531_, double p_37532_, double p_37533_) {
-      super(EntityType.f_20486_, p_37531_, p_37532_, p_37533_, p_37530_);
+   public ThrownPotion(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20486_, pX, pY, pZ, pLevel);
    }
 
    protected Item m_7881_() {
@@ -66,13 +_,13 @@
          boolean flag = potion == Potions.f_43599_ && list.isEmpty();
          Direction direction = p_37541_.m_82434_();
          BlockPos blockpos = p_37541_.m_82425_();
-         BlockPos blockpos1 = blockpos.m_121945_(direction);
+         BlockPos blockpos1 = blockpos.relative(direction);
          if (flag) {
             this.m_150192_(blockpos1);
-            this.m_150192_(blockpos1.m_121945_(direction.m_122424_()));
+            this.m_150192_(blockpos1.relative(direction.m_122424_()));
 
             for(Direction direction1 : Direction.Plane.HORIZONTAL) {
-               this.m_150192_(blockpos1.m_121945_(direction1));
+               this.m_150192_(blockpos1.relative(direction1));
             }
          }
 
@@ -163,7 +_,7 @@
    }
 
    private void m_37537_(ItemStack p_37538_, Potion p_37539_) {
-      AreaEffectCloud areaeffectcloud = new AreaEffectCloud(this.m_9236_(), this.m_20185_(), this.m_20186_(), this.m_20189_());
+      AreaEffectCloud areaeffectcloud = new AreaEffectCloud(this.m_9236_(), this.getBigX(), this.getBigY(), this.getBigZ());
       Entity entity = this.m_19749_();
       if (entity instanceof LivingEntity) {
          areaeffectcloud.m_19718_((LivingEntity)entity);
