--- a/net/minecraft/world/entity/projectile/ThrownEgg.java
+++ b/net/minecraft/world/entity/projectile/ThrownEgg.java
@@ -20,8 +_,8 @@
       super(EntityType.f_20483_, p_37482_, p_37481_);
    }
 
-   public ThrownEgg(Level p_37476_, double p_37477_, double p_37478_, double p_37479_) {
-      super(EntityType.f_20483_, p_37477_, p_37478_, p_37479_, p_37476_);
+   public ThrownEgg(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20483_, pX, pY, pZ, pLevel);
    }
 
    public void m_7822_(byte p_37484_) {
@@ -29,7 +_,7 @@
          double d0 = 0.08D;
 
          for(int i = 0; i < 8; ++i) {
-            this.m_9236_().m_7106_(new ItemParticleOption(ParticleTypes.f_123752_, this.m_7846_()), this.m_20185_(), this.m_20186_(), this.m_20189_(), ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D);
+            this.m_9236_().addParticle(new ItemParticleOption(ParticleTypes.f_123752_, this.m_7846_()), this.getBigX(), this.getBigY(), this.getBigZ(), ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D);
          }
       }
 
@@ -53,7 +_,7 @@
                Chicken chicken = EntityType.f_20555_.m_20615_(this.m_9236_());
                if (chicken != null) {
                   chicken.m_146762_(-24000);
-                  chicken.m_7678_(this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_146908_(), 0.0F);
+                  chicken.moveTo(this.getBigX(), this.getBigY(), this.getBigZ(), this.m_146908_(), 0.0F);
                   this.m_9236_().m_7967_(chicken);
                }
             }
