--- a/net/minecraft/world/entity/projectile/ThrownTrident.java
+++ b/net/minecraft/world/entity/projectile/ThrownTrident.java
@@ -64,9 +_,9 @@
          } else {
             this.m_36790_(true);
             Vec3 vec3 = entity.m_146892_().m_82546_(this.m_20182_());
-            this.m_20343_(this.m_20185_(), this.m_20186_() + vec3.f_82480_ * 0.015D * (double)i, this.m_20189_());
+            this.m_20343_(this.m_20185_(), this.m_20186_() + vec3.m_7098_() * 0.015D * (double)i, this.m_20189_());
             if (this.m_9236_().f_46443_) {
-               this.f_19791_ = this.m_20186_();
+               this.f_19791_ = this.getBigY();
             }
 
             double d0 = 0.05D * (double)i;
@@ -138,7 +_,7 @@
          if (this.m_9236_().m_45527_(blockpos)) {
             LightningBolt lightningbolt = EntityType.f_20465_.m_20615_(this.m_9236_());
             if (lightningbolt != null) {
-               lightningbolt.m_20219_(Vec3.m_82539_(blockpos));
+               lightningbolt.m_20219_(Vec3.atBottomCenterOf(blockpos));
                lightningbolt.m_20879_(entity1 instanceof ServerPlayer ? (ServerPlayer)entity1 : null);
                this.m_9236_().m_7967_(lightningbolt);
                soundevent = SoundEvents.f_12521_;
