--- a/net/minecraft/world/entity/projectile/ThrownEgg.java
+++ b/net/minecraft/world/entity/projectile/ThrownEgg.java
@@ -29,7 +_,7 @@
          double d0 = 0.08D;
 
          for(int i = 0; i < 8; ++i) {
-            this.m_9236_().m_7106_(new ItemParticleOption(ParticleTypes.f_123752_, this.m_7846_()), this.m_20185_(), this.m_20186_(), this.m_20189_(), ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D);
+            this.m_9236_().addParticle(new ItemParticleOption(ParticleTypes.f_123752_, this.m_7846_()), this.getBigX(), this.getBigY(), this.getBigZ(), ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D, ((double)this.f_19796_.m_188501_() - 0.5D) * 0.08D);
          }
       }
 
