--- a/net/minecraft/world/level/redstone/NeighborUpdater.java
+++ b/net/minecraft/world/level/redstone/NeighborUpdater.java
@@ -25,7 +_,7 @@
    default void m_214152_(BlockPos p_230788_, Block p_230789_, @Nullable Direction p_230790_) {
       for(Direction direction : f_230761_) {
          if (direction != p_230790_) {
-            this.m_214026_(p_230788_.m_121945_(direction), p_230789_, p_230788_);
+            this.m_214026_(p_230788_.relative(direction), p_230789_, p_230788_);
          }
       }
 
