--- a/net/minecraft/util/profiling/jfr/stats/ChunkGenStat.java
+++ b/net/minecraft/util/profiling/jfr/stats/ChunkGenStat.java
@@ -8,7 +_,7 @@
 
 public record ChunkGenStat(Duration f_185592_, ChunkPos f_185593_, ColumnPos f_185594_, ChunkStatus f_185595_, String f_185596_) implements TimedStat {
    public static ChunkGenStat m_185604_(RecordedEvent p_185605_) {
-      return new ChunkGenStat(p_185605_.getDuration(), new ChunkPos(p_185605_.getInt("chunkPosX"), p_185605_.getInt("chunkPosX")), new ColumnPos(p_185605_.getInt("worldPosX"), p_185605_.getInt("worldPosZ")), ChunkStatus.m_62397_(p_185605_.getString("status")), p_185605_.getString("level"));
+      return new ChunkGenStat(p_185605_.getDuration(), new ChunkPos(p_185605_.getLong("chunkPosX"), p_185605_.getLong("chunkPosX")), new ColumnPos(p_185605_.getLong("worldPosX"), p_185605_.getLong("worldPosZ")), ChunkStatus.m_62397_(p_185605_.getString("status")), p_185605_.getString("level"));
    }
 
    public Duration m_183571_() {
