--- a/net/minecraft/network/protocol/game/ClientboundSectionBlocksUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSectionBlocksUpdatePacket.java
@@ -27,7 +_,7 @@
 
       for(short short1 : p_132987_) {
          this.f_132981_[j] = short1;
-         this.f_132982_[j] = p_132988_.m_62982_(SectionPos.m_123204_(short1), SectionPos.m_123220_(short1), SectionPos.m_123227_(short1));
+         this.f_132982_[j] = p_132988_.getBlockState(SectionPos.m_123204_(short1), SectionPos.m_123220_(short1), SectionPos.m_123227_(short1));
          ++j;
       }
 
@@ -68,7 +_,7 @@
 
       for(int i = 0; i < this.f_132981_.length; ++i) {
          short short1 = this.f_132981_[i];
-         blockpos$mutableblockpos.m_122178_(this.f_132980_.m_123232_(short1), this.f_132980_.m_123237_(short1), this.f_132980_.m_123242_(short1));
+         blockpos$mutableblockpos.m_122169_(this.f_132980_.relativeToBlockX(short1), this.f_132980_.relativeToBlockY(short1), this.f_132980_.relativeToBlockZ(short1));
          p_132993_.accept(blockpos$mutableblockpos, this.f_132982_[i]);
       }
 
