--- a/net/minecraft/network/protocol/game/ClientboundAddEntityPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundAddEntityPacket.java
@@ -36,7 +_,7 @@
    }
 
    public ClientboundAddEntityPacket(Entity p_237558_, int p_237559_, BlockPos p_237560_) {
-      this(p_237558_.m_19879_(), p_237558_.m_20148_(), (double)p_237560_.m_123341_(), (double)p_237560_.m_123342_(), (double)p_237560_.m_123343_(), p_237558_.m_146909_(), p_237558_.m_146908_(), p_237558_.m_6095_(), p_237559_, p_237558_.m_20184_(), (double)p_237558_.m_6080_());
+      this(p_237558_.m_19879_(), p_237558_.m_20148_(), (double)p_237560_.getX(), (double)p_237560_.getY(), (double)p_237560_.getZ(), p_237558_.m_146909_(), p_237558_.m_146908_(), p_237558_.m_6095_(), p_237559_, p_237558_.m_20184_(), (double)p_237558_.m_6080_());
    }
 
    public ClientboundAddEntityPacket(int p_237546_, UUID p_237547_, double p_237548_, double p_237549_, double p_237550_, float p_237551_, float p_237552_, EntityType<?> p_237553_, int p_237554_, Vec3 p_237555_, double p_237556_) {
