--- a/net/minecraft/network/protocol/game/ClientboundTeleportEntityPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundTeleportEntityPacket.java
@@ -17,9 +_,9 @@
    public ClientboundTeleportEntityPacket(Entity p_133538_) {
       this.f_133529_ = p_133538_.m_19879_();
       Vec3 vec3 = p_133538_.m_213870_();
-      this.f_133530_ = vec3.f_82479_;
-      this.f_133531_ = vec3.f_82480_;
-      this.f_133532_ = vec3.f_82481_;
+      this.f_133530_ = vec3.m_7096_();
+      this.f_133531_ = vec3.m_7098_();
+      this.f_133532_ = vec3.m_7094_();
       this.f_133533_ = (byte)((int)(p_133538_.m_146908_() * 256.0F / 360.0F));
       this.f_133534_ = (byte)((int)(p_133538_.m_146909_() * 256.0F / 360.0F));
       this.f_133535_ = p_133538_.m_20096_();
