--- a/net/minecraft/network/protocol/game/ClientboundPlayerPositionPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerPositionPacket.java
@@ -6,28 +_,28 @@
 import net.minecraft.world.entity.RelativeMovement;
 
 public class ClientboundPlayerPositionPacket implements Packet<ClientGamePacketListener> {
-   private final double f_132796_;
-   private final double f_132797_;
-   private final double f_132798_;
+   private final me.alphamode.mclong.math.BigDecimal f_132796_;
+   private final me.alphamode.mclong.math.BigDecimal f_132797_;
+   private final me.alphamode.mclong.math.BigDecimal f_132798_;
    private final float f_132799_;
    private final float f_132800_;
    private final Set<RelativeMovement> f_132801_;
    private final int f_132802_;
 
-   public ClientboundPlayerPositionPacket(double p_275438_, double p_275354_, double p_275276_, float p_275280_, float p_275203_, Set<RelativeMovement> p_275228_, int p_275614_) {
-      this.f_132796_ = p_275438_;
-      this.f_132797_ = p_275354_;
-      this.f_132798_ = p_275276_;
-      this.f_132799_ = p_275280_;
-      this.f_132800_ = p_275203_;
-      this.f_132801_ = p_275228_;
-      this.f_132802_ = p_275614_;
+   public ClientboundPlayerPositionPacket(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pYRot, float pXRot, Set<RelativeMovement> pRelativeArguments, int pId) {
+      this.f_132796_ = pX;
+      this.f_132797_ = pY;
+      this.f_132798_ = pZ;
+      this.f_132799_ = pYRot;
+      this.f_132800_ = pXRot;
+      this.f_132801_ = pRelativeArguments;
+      this.f_132802_ = pId;
    }
 
    public ClientboundPlayerPositionPacket(FriendlyByteBuf p_179158_) {
-      this.f_132796_ = p_179158_.readDouble();
-      this.f_132797_ = p_179158_.readDouble();
-      this.f_132798_ = p_179158_.readDouble();
+      this.f_132796_ = new me.alphamode.mclong.math.BigDecimal(p_179158_.m_130277_());
+      this.f_132797_ = new me.alphamode.mclong.math.BigDecimal(p_179158_.m_130277_());
+      this.f_132798_ = new me.alphamode.mclong.math.BigDecimal(p_179158_.m_130277_());
       this.f_132799_ = p_179158_.readFloat();
       this.f_132800_ = p_179158_.readFloat();
       this.f_132801_ = RelativeMovement.m_264098_(p_179158_.readUnsignedByte());
@@ -35,9 +_,9 @@
    }
 
    public void m_5779_(FriendlyByteBuf p_132820_) {
-      p_132820_.writeDouble(this.f_132796_);
-      p_132820_.writeDouble(this.f_132797_);
-      p_132820_.writeDouble(this.f_132798_);
+      p_132820_.m_130070_(this.f_132796_.toString());
+      p_132820_.m_130070_(this.f_132797_.toString());
+      p_132820_.m_130070_(this.f_132798_.toString());
       p_132820_.writeFloat(this.f_132799_);
       p_132820_.writeFloat(this.f_132800_);
       p_132820_.writeByte(RelativeMovement.m_264160_(this.f_132801_));
@@ -48,15 +_,15 @@
       p_132817_.m_5682_(this);
    }
 
-   public double m_132818_() {
+   public me.alphamode.mclong.math.BigDecimal getX() {
       return this.f_132796_;
    }
 
-   public double m_132821_() {
+   public me.alphamode.mclong.math.BigDecimal getY() {
       return this.f_132797_;
    }
 
-   public double m_132822_() {
+   public me.alphamode.mclong.math.BigDecimal getZ() {
       return this.f_132798_;
    }
 
