--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
@@ -9,8 +_,8 @@
 import net.minecraft.world.level.lighting.LevelLightEngine;
 
 public class ClientboundLevelChunkWithLightPacket implements Packet<ClientGamePacketListener> {
-   private final int f_195699_;
-   private final int f_195700_;
+   private final long f_195699_;
+   private final long f_195700_;
    private final ClientboundLevelChunkPacketData f_195701_;
    private final ClientboundLightUpdatePacketData f_195702_;
 
@@ -23,15 +_,15 @@
    }
 
    public ClientboundLevelChunkWithLightPacket(FriendlyByteBuf p_195710_) {
-      this.f_195699_ = p_195710_.readInt();
-      this.f_195700_ = p_195710_.readInt();
+      this.f_195699_ = p_195710_.readLong();
+      this.f_195700_ = p_195710_.readLong();
       this.f_195701_ = new ClientboundLevelChunkPacketData(p_195710_, this.f_195699_, this.f_195700_);
       this.f_195702_ = new ClientboundLightUpdatePacketData(p_195710_, this.f_195699_, this.f_195700_);
    }
 
    public void m_5779_(FriendlyByteBuf p_195712_) {
-      p_195712_.writeInt(this.f_195699_);
-      p_195712_.writeInt(this.f_195700_);
+      p_195712_.writeLong(this.f_195699_);
+      p_195712_.writeLong(this.f_195700_);
       this.f_195701_.m_195666_(p_195712_);
       this.f_195702_.m_195749_(p_195712_);
    }
@@ -40,11 +_,11 @@
       p_195716_.m_183388_(this);
    }
 
-   public int m_195717_() {
+   public long getX() {
       return this.f_195699_;
    }
 
-   public int m_195718_() {
+   public long getZ() {
       return this.f_195700_;
    }
 
