--- a/net/minecraft/network/protocol/game/ClientboundLightUpdatePacketData.java
+++ b/net/minecraft/network/protocol/game/ClientboundLightUpdatePacketData.java
@@ -41,7 +_,7 @@
 
    }
 
-   public ClientboundLightUpdatePacketData(FriendlyByteBuf p_195737_, int p_195738_, int p_195739_) {
+   public ClientboundLightUpdatePacketData(FriendlyByteBuf p_195737_, long p_195738_, long p_195739_) {
       this.f_195729_ = p_195737_.readBoolean();
       this.f_195723_ = p_195737_.m_178384_();
       this.f_195724_ = p_195737_.m_178384_();
