--- a/net/minecraft/network/protocol/game/ClientboundPlayerLookAtPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerLookAtPacket.java
@@ -32,9 +_,9 @@
       this.f_132771_ = p_132783_.m_19879_();
       this.f_132773_ = p_132784_;
       Vec3 vec3 = p_132784_.m_90377_(p_132783_);
-      this.f_132768_ = vec3.f_82479_;
-      this.f_132769_ = vec3.f_82480_;
-      this.f_132770_ = vec3.f_82481_;
+      this.f_132768_ = vec3.m_7096_();
+      this.f_132769_ = vec3.m_7098_();
+      this.f_132770_ = vec3.m_7094_();
       this.f_132774_ = true;
    }
 
