--- a/net/minecraft/network/protocol/game/ClientboundSetEntityMotionPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetEntityMotionPacket.java
@@ -19,9 +_,9 @@
    public ClientboundSetEntityMotionPacket(int p_133182_, Vec3 p_133183_) {
       this.f_133176_ = p_133182_;
       double d0 = 3.9D;
-      double d1 = Mth.m_14008_(p_133183_.f_82479_, -3.9D, 3.9D);
-      double d2 = Mth.m_14008_(p_133183_.f_82480_, -3.9D, 3.9D);
-      double d3 = Mth.m_14008_(p_133183_.f_82481_, -3.9D, 3.9D);
+      double d1 = Mth.m_14008_(p_133183_.m_7096_(), -3.9D, 3.9D);
+      double d2 = Mth.m_14008_(p_133183_.m_7098_(), -3.9D, 3.9D);
+      double d3 = Mth.m_14008_(p_133183_.m_7094_(), -3.9D, 3.9D);
       this.f_133177_ = (int)(d1 * 8000.0D);
       this.f_133178_ = (int)(d2 * 8000.0D);
       this.f_133179_ = (int)(d3 * 8000.0D);
