--- a/net/minecraft/network/protocol/game/VecDeltaCodec.java
+++ b/net/minecraft/network/protocol/game/VecDeltaCodec.java
@@ -8,8 +_,8 @@
    private Vec3 f_238015_ = Vec3.f_82478_;
 
    @VisibleForTesting
-   static long m_238017_(double p_238018_) {
-      return Math.round(p_238018_ * 4096.0D);
+   static long encode(me.alphamode.mclong.math.BigDecimal p_238018_) {
+      return p_238018_.multiply(4096.0D).setScale(0, java.math.RoundingMode.HALF_UP).longValue();
    }
 
    @VisibleForTesting
@@ -21,23 +_,23 @@
       if (p_238022_ == 0L && p_238023_ == 0L && p_238024_ == 0L) {
          return this.f_238015_;
       } else {
-         double d0 = p_238022_ == 0L ? this.f_238015_.f_82479_ : m_238019_(m_238017_(this.f_238015_.f_82479_) + p_238022_);
-         double d1 = p_238023_ == 0L ? this.f_238015_.f_82480_ : m_238019_(m_238017_(this.f_238015_.f_82480_) + p_238023_);
-         double d2 = p_238024_ == 0L ? this.f_238015_.f_82481_ : m_238019_(m_238017_(this.f_238015_.f_82481_) + p_238024_);
+         double d0 = p_238022_ == 0L ? this.f_238015_.m_7096_() : m_238019_(encode(this.f_238015_.getX()) + p_238022_);
+         double d1 = p_238023_ == 0L ? this.f_238015_.m_7098_() : m_238019_(encode(this.f_238015_.getY()) + p_238023_);
+         double d2 = p_238024_ == 0L ? this.f_238015_.m_7094_() : m_238019_(encode(this.f_238015_.getZ()) + p_238024_);
          return new Vec3(d0, d1, d2);
       }
    }
 
    public long m_238025_(Vec3 p_238026_) {
-      return m_238017_(p_238026_.f_82479_) - m_238017_(this.f_238015_.f_82479_);
+      return encode(p_238026_.getX()) - encode(this.f_238015_.getX());
    }
 
    public long m_238027_(Vec3 p_238028_) {
-      return m_238017_(p_238028_.f_82480_) - m_238017_(this.f_238015_.f_82480_);
+      return encode(p_238028_.getY()) - encode(this.f_238015_.getY());
    }
 
    public long m_238029_(Vec3 p_238030_) {
-      return m_238017_(p_238030_.f_82481_) - m_238017_(this.f_238015_.f_82481_);
+      return encode(p_238030_.getZ()) - encode(this.f_238015_.getZ());
    }
 
    public Vec3 m_238031_(Vec3 p_238032_) {
