--- a/net/minecraft/data/worldgen/placement/AquaticPlacements.java
+++ b/net/minecraft/data/worldgen/placement/AquaticPlacements.java
@@ -60,7 +_,7 @@
       PlacementUtils.m_254943_(p_256301_, f_195223_, reference3, m_195233_(80));
       PlacementUtils.m_254943_(p_256301_, f_195224_, reference3, m_195233_(48));
       PlacementUtils.m_254943_(p_256301_, f_195225_, reference3, m_195233_(40));
-      PlacementUtils.m_255206_(p_256301_, f_195226_, reference4, CarvingMaskPlacement.m_191590_(GenerationStep.Carving.LIQUID), RarityFilter.m_191900_(10), BlockPredicateFilter.m_191576_(BlockPredicate.m_190417_(BlockPredicate.m_224774_(Direction.DOWN.m_122436_(), Blocks.f_50069_), BlockPredicate.m_224774_(BlockPos.f_121853_, Blocks.f_49990_), BlockPredicate.m_224774_(Direction.UP.m_122436_(), Blocks.f_49990_))), BiomeFilter.m_191561_());
+      PlacementUtils.m_255206_(p_256301_, f_195226_, reference4, CarvingMaskPlacement.m_191590_(GenerationStep.Carving.LIQUID), RarityFilter.m_191900_(10), BlockPredicateFilter.m_191576_(BlockPredicate.m_190417_(BlockPredicate.m_224774_(Direction.DOWN.m_122436_(), Blocks.f_50069_), BlockPredicate.matchesBlocks(BlockPos.f_121853_, Blocks.f_49990_), BlockPredicate.m_224774_(Direction.UP.m_122436_(), Blocks.f_49990_))), BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_256301_, f_195227_, reference5, RarityFilter.m_191900_(16), InSquarePlacement.m_191715_(), PlacementUtils.f_195353_, BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_256301_, f_195228_, reference6, NoiseBasedCountPlacement.m_191731_(120, 80.0D, 0.0D), InSquarePlacement.m_191715_(), PlacementUtils.f_195353_, BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_256301_, f_195229_, reference6, NoiseBasedCountPlacement.m_191731_(80, 80.0D, 0.0D), InSquarePlacement.m_191715_(), PlacementUtils.f_195353_, BiomeFilter.m_191561_());
