--- a/net/minecraft/data/worldgen/placement/PlacementUtils.java
+++ b/net/minecraft/data/worldgen/placement/PlacementUtils.java
@@ -76,7 +_,7 @@
    }
 
    public static BlockPredicateFilter m_206493_(Block p_206494_) {
-      return BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(p_206494_.m_49966_(), BlockPos.f_121853_));
+      return BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(p_206494_.m_49966_(), BlockPos.f_121853_));
    }
 
    public static Holder<PlacedFeature> m_206506_(Holder<ConfiguredFeature<?, ?>> p_206507_, PlacementModifier... p_206508_) {
