--- a/net/minecraft/data/worldgen/placement/VegetationPlacements.java
+++ b/net/minecraft/data/worldgen/placement/VegetationPlacements.java
@@ -74,7 +_,7 @@
    public static final Holder<PlacedFeature> f_197412_ = PlacementUtils.m_206513_("flower_plains", VegetationFeatures.f_195195_, NoiseThresholdCountPlacement.m_191756_(-0.8D, 15, 4), RarityFilter.m_191900_(32), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
    public static final Holder<PlacedFeature> f_195419_ = PlacementUtils.m_206513_("flower_meadow", VegetationFeatures.f_195196_, InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
    public static final PlacementModifier f_195420_ = SurfaceWaterDepthFilter.m_191950_(0);
-   public static final Holder<PlacedFeature> f_195421_ = PlacementUtils.m_206513_("trees_plains", VegetationFeatures.f_195166_, PlacementUtils.m_195364_(0, 0.05F, 1), InSquarePlacement.m_191715_(), f_195420_, PlacementUtils.f_195355_, BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)), BiomeFilter.m_191561_());
+   public static final Holder<PlacedFeature> f_195421_ = PlacementUtils.m_206513_("trees_plains", VegetationFeatures.f_195166_, PlacementUtils.m_195364_(0, 0.05F, 1), InSquarePlacement.m_191715_(), f_195420_, PlacementUtils.f_195355_, BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)), BiomeFilter.m_191561_());
    public static final Holder<PlacedFeature> f_195422_ = PlacementUtils.m_206513_("dark_forest_vegetation", VegetationFeatures.f_195198_, CountPlacement.m_191628_(16), InSquarePlacement.m_191715_(), f_195420_, PlacementUtils.f_195355_, BiomeFilter.m_191561_());
    public static final Holder<PlacedFeature> f_195423_ = PlacementUtils.m_206513_("flower_forest_flowers", VegetationFeatures.f_195197_, RarityFilter.m_191900_(7), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, CountPlacement.m_191630_(ClampedInt.m_146395_(UniformInt.m_146622_(-1, 3), 0, 3)), BiomeFilter.m_191561_());
    public static final Holder<PlacedFeature> f_195424_ = PlacementUtils.m_206513_("forest_flowers", VegetationFeatures.f_195197_, RarityFilter.m_191900_(7), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, CountPlacement.m_191630_(ClampedInt.m_146395_(UniformInt.m_146622_(-3, 1), 0, 1)), BiomeFilter.m_191561_());
@@ -84,7 +_,7 @@
    public static final Holder<PlacedFeature> f_195429_ = PlacementUtils.m_206509_("trees_grove", VegetationFeatures.f_195160_, m_195479_(PlacementUtils.m_195364_(10, 0.1F, 1)));
    public static final Holder<PlacedFeature> f_195430_ = PlacementUtils.m_206509_("trees_badlands", TreeFeatures.f_195123_, m_195481_(PlacementUtils.m_195364_(5, 0.1F, 1), Blocks.f_50746_));
    public static final Holder<PlacedFeature> f_195431_ = PlacementUtils.m_206509_("trees_snowy", TreeFeatures.f_195127_, m_195481_(PlacementUtils.m_195364_(0, 0.1F, 1), Blocks.f_50747_));
-   public static final Holder<PlacedFeature> f_195432_ = PlacementUtils.m_206513_("trees_swamp", TreeFeatures.f_195137_, PlacementUtils.m_195364_(2, 0.1F, 1), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(2), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)));
+   public static final Holder<PlacedFeature> f_195432_ = PlacementUtils.m_206513_("trees_swamp", TreeFeatures.f_195137_, PlacementUtils.m_195364_(2, 0.1F, 1), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(2), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)));
    public static final Holder<PlacedFeature> f_195433_ = PlacementUtils.m_206509_("trees_windswept_savanna", VegetationFeatures.f_195161_, m_195479_(PlacementUtils.m_195364_(2, 0.1F, 1)));
    public static final Holder<PlacedFeature> f_195434_ = PlacementUtils.m_206509_("trees_savanna", VegetationFeatures.f_195161_, m_195479_(PlacementUtils.m_195364_(1, 0.1F, 1)));
    public static final Holder<PlacedFeature> f_195435_ = PlacementUtils.m_206509_("birch_tall", VegetationFeatures.f_195162_, m_195479_(PlacementUtils.m_195364_(10, 0.1F, 1)));
@@ -99,7 +_,7 @@
    public static final Holder<PlacedFeature> f_195444_ = PlacementUtils.m_206509_("trees_jungle", VegetationFeatures.f_195170_, m_195479_(PlacementUtils.m_195364_(50, 0.1F, 1)));
    public static final Holder<PlacedFeature> f_195445_ = PlacementUtils.m_206509_("bamboo_vegetation", VegetationFeatures.f_195171_, m_195479_(PlacementUtils.m_195364_(30, 0.1F, 1)));
    public static final Holder<PlacedFeature> f_195446_ = PlacementUtils.m_206513_("mushroom_island_vegetation", VegetationFeatures.f_195172_, InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
-   public static final Holder<PlacedFeature> f_236773_ = PlacementUtils.m_206513_("trees_mangrove", VegetationFeatures.f_236764_, CountPlacement.m_191628_(25), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(5), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_220831_.m_49966_(), BlockPos.f_121853_)));
+   public static final Holder<PlacedFeature> f_236773_ = PlacementUtils.m_206513_("trees_mangrove", VegetationFeatures.f_236764_, CountPlacement.m_191628_(25), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(5), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_220831_.m_49966_(), BlockPos.f_121853_)));
 
    public static List<PlacementModifier> m_195474_(int p_195475_) {
       return List.of(CountPlacement.m_191628_(p_195475_), InSquarePlacement.m_191715_(), PlacementUtils.f_195354_, BiomeFilter.m_191561_());
@@ -130,6 +_,6 @@
    }
 
    public static List<PlacementModifier> m_195481_(PlacementModifier p_195482_, Block p_195483_) {
-      return m_195484_(p_195482_).add(BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(p_195483_.m_49966_(), BlockPos.f_121853_))).build();
+      return m_195484_(p_195482_).add(BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(p_195483_.m_49966_(), BlockPos.f_121853_))).build();
    }
 }
