--- a/net/minecraft/data/worldgen/placement/VegetationPlacements.java
+++ b/net/minecraft/data/worldgen/placement/VegetationPlacements.java
@@ -137,7 +_,7 @@
    }
 
    public static List<PlacementModifier> m_195481_(PlacementModifier p_195482_, Block p_195483_) {
-      return m_195484_(p_195482_).add(BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(p_195483_.m_49966_(), BlockPos.f_121853_))).build();
+      return m_195484_(p_195482_).add(BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(p_195483_.m_49966_(), BlockPos.f_121853_))).build();
    }
 
    public static void m_254976_(BootstapContext<PlacedFeature> p_255657_) {
@@ -240,7 +_,7 @@
       PlacementUtils.m_255206_(p_255657_, f_271183_, holder24, NoiseThresholdCountPlacement.m_191756_(-0.8D, 5, 10), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_255657_, f_195419_, holder23, InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
       PlacementModifier placementmodifier = SurfaceWaterDepthFilter.m_191950_(0);
-      PlacementUtils.m_255206_(p_255657_, f_195421_, holder25, PlacementUtils.m_195364_(0, 0.05F, 1), InSquarePlacement.m_191715_(), placementmodifier, PlacementUtils.f_195355_, BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)), BiomeFilter.m_191561_());
+      PlacementUtils.m_255206_(p_255657_, f_195421_, holder25, PlacementUtils.m_195364_(0, 0.05F, 1), InSquarePlacement.m_191715_(), placementmodifier, PlacementUtils.f_195355_, BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)), BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_255657_, f_195422_, holder26, CountPlacement.m_191628_(16), InSquarePlacement.m_191715_(), placementmodifier, PlacementUtils.f_195355_, BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_255657_, f_195423_, holder27, RarityFilter.m_191900_(7), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, CountPlacement.m_191630_(ClampedInt.m_146395_(UniformInt.m_146622_(-1, 3), 0, 3)), BiomeFilter.m_191561_());
       PlacementUtils.m_255206_(p_255657_, f_195424_, holder27, RarityFilter.m_191900_(7), InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, CountPlacement.m_191630_(ClampedInt.m_146395_(UniformInt.m_146622_(-3, 1), 0, 1)), BiomeFilter.m_191561_());
@@ -251,7 +_,7 @@
       PlacementUtils.m_254943_(p_255657_, f_195429_, holder31, m_195479_(PlacementUtils.m_195364_(10, 0.1F, 1)));
       PlacementUtils.m_254943_(p_255657_, f_195430_, holder32, m_195481_(PlacementUtils.m_195364_(5, 0.1F, 1), Blocks.f_50746_));
       PlacementUtils.m_254943_(p_255657_, f_195431_, holder33, m_195481_(PlacementUtils.m_195364_(0, 0.1F, 1), Blocks.f_50747_));
-      PlacementUtils.m_255206_(p_255657_, f_195432_, holder35, PlacementUtils.m_195364_(2, 0.1F, 1), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(2), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)));
+      PlacementUtils.m_255206_(p_255657_, f_195432_, holder35, PlacementUtils.m_195364_(2, 0.1F, 1), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(2), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_50746_.m_49966_(), BlockPos.f_121853_)));
       PlacementUtils.m_254943_(p_255657_, f_195433_, holder36, m_195479_(PlacementUtils.m_195364_(2, 0.1F, 1)));
       PlacementUtils.m_254943_(p_255657_, f_195434_, holder36, m_195479_(PlacementUtils.m_195364_(1, 0.1F, 1)));
       PlacementUtils.m_254943_(p_255657_, f_195435_, holder37, m_195479_(PlacementUtils.m_195364_(10, 0.1F, 1)));
@@ -266,6 +_,6 @@
       PlacementUtils.m_254943_(p_255657_, f_195444_, holder45, m_195479_(PlacementUtils.m_195364_(50, 0.1F, 1)));
       PlacementUtils.m_254943_(p_255657_, f_195445_, holder46, m_195479_(PlacementUtils.m_195364_(30, 0.1F, 1)));
       PlacementUtils.m_255206_(p_255657_, f_195446_, holder47, InSquarePlacement.m_191715_(), PlacementUtils.f_195352_, BiomeFilter.m_191561_());
-      PlacementUtils.m_255206_(p_255657_, f_236773_, holder48, CountPlacement.m_191628_(25), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(5), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.m_190399_(Blocks.f_220831_.m_49966_(), BlockPos.f_121853_)));
+      PlacementUtils.m_255206_(p_255657_, f_236773_, holder48, CountPlacement.m_191628_(25), InSquarePlacement.m_191715_(), SurfaceWaterDepthFilter.m_191950_(5), PlacementUtils.f_195355_, BiomeFilter.m_191561_(), BlockPredicateFilter.m_191576_(BlockPredicate.wouldSurvive(Blocks.f_220831_.m_49966_(), BlockPos.f_121853_)));
    }
 }
