--- a/net/minecraft/server/chase/ChaseClient.java
+++ b/net/minecraft/server/chase/ChaseClient.java
@@ -168,7 +_,7 @@
          if (!list.isEmpty()) {
             ServerPlayer serverplayer = list.get(0);
             ServerLevel serverlevel = this.f_195984_.m_129783_();
-            CommandSourceStack commandsourcestack = new CommandSourceStack(serverplayer, Vec3.m_82528_(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "", CommonComponents.f_237098_, this.f_195984_, serverplayer);
+            CommandSourceStack commandsourcestack = new CommandSourceStack(serverplayer, Vec3.atLowerCornerOf(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "", CommonComponents.f_237098_, this.f_195984_, serverplayer);
             Commands commands = this.f_195984_.m_129892_();
             commands.m_230957_(commandsourcestack, p_196002_);
          }
