--- a/net/minecraft/server/chase/ChaseClient.java
+++ b/net/minecraft/server/chase/ChaseClient.java
@@ -116,7 +_,7 @@
 
    private void m_195996_(Scanner p_195997_) {
       this.m_196003_(p_195997_).ifPresent((p_195999_) -> {
-         this.m_196001_(String.format(Locale.ROOT, "execute in %s run tp @s %.3f %.3f %.3f %.3f %.3f", p_195999_.f_196008_.m_135782_(), p_195999_.f_196009_.f_82479_, p_195999_.f_196009_.f_82480_, p_195999_.f_196009_.f_82481_, p_195999_.f_196010_.f_82471_, p_195999_.f_196010_.f_82470_));
+         this.m_196001_(String.format(Locale.ROOT, "execute in %s run tp @s %.3f %.3f %.3f %.3f %.3f", p_195999_.f_196008_.m_135782_(), p_195999_.f_196009_.m_7096_(), p_195999_.f_196009_.m_7098_(), p_195999_.f_196009_.m_7094_(), p_195999_.f_196010_.f_82471_, p_195999_.f_196010_.f_82470_));
       });
    }
 
@@ -140,7 +_,7 @@
          if (!list.isEmpty()) {
             ServerPlayer serverplayer = list.get(0);
             ServerLevel serverlevel = this.f_195984_.m_129783_();
-            CommandSourceStack commandsourcestack = new CommandSourceStack(serverplayer, Vec3.m_82528_(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "", CommonComponents.f_237098_, this.f_195984_, serverplayer);
+            CommandSourceStack commandsourcestack = new CommandSourceStack(serverplayer, Vec3.atLowerCornerOf(serverlevel.m_220360_()), Vec2.f_82462_, serverlevel, 4, "", CommonComponents.f_237098_, this.f_195984_, serverplayer);
             Commands commands = this.f_195984_.m_129892_();
             commands.m_230957_(commandsourcestack, p_196002_);
          }
