--- a/net/minecraft/world/level/portal/PortalShape.java
+++ b/net/minecraft/world/level/portal/PortalShape.java
@@ -77,12 +_,12 @@
 
    @Nullable
    private BlockPos m_77733_(BlockPos p_77734_) {
-      for(int i = Math.max(this.f_77686_.m_141937_(), p_77734_.m_123342_() - 21); p_77734_.m_123342_() > i && m_77717_(this.f_77686_.m_8055_(p_77734_.m_7495_())); p_77734_ = p_77734_.m_7495_()) {
+      for(long i = Math.max(this.f_77686_.getMinBuildHeight(), p_77734_.getY() - 21); p_77734_.getY() > i && m_77717_(this.f_77686_.m_8055_(p_77734_.below())); p_77734_ = p_77734_.below()) {
       }
 
       Direction direction = this.f_77688_.m_122424_();
       int j = this.m_77735_(p_77734_, direction) - 1;
-      return j < 0 ? null : p_77734_.m_5484_(direction, j);
+      return j < 0 ? null : p_77734_.relative(direction, j);
    }
 
    private int m_77745_() {
@@ -94,7 +_,7 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(int i = 0; i <= 21; ++i) {
-         blockpos$mutableblockpos.m_122190_(p_77736_).m_122175_(p_77737_, i);
+         blockpos$mutableblockpos.set(p_77736_).move(p_77737_, i);
          BlockState blockstate = this.f_77686_.m_8055_(blockpos$mutableblockpos);
          if (!m_77717_(blockstate)) {
             if (f_77685_.m_61035_(blockstate, this.f_77686_, blockpos$mutableblockpos)) {
@@ -120,7 +_,7 @@
 
    private boolean m_77730_(BlockPos.MutableBlockPos p_77731_, int p_77732_) {
       for(int i = 0; i < this.f_77692_; ++i) {
-         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_77731_.m_122190_(this.f_77690_).m_122175_(Direction.UP, p_77732_).m_122175_(this.f_77688_, i);
+         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_77731_.set(this.f_77690_).move(Direction.UP, p_77732_).move(this.f_77688_, i);
          if (!f_77685_.m_61035_(this.f_77686_.m_8055_(blockpos$mutableblockpos), this.f_77686_, blockpos$mutableblockpos)) {
             return false;
          }
@@ -131,18 +_,18 @@
 
    private int m_77728_(BlockPos.MutableBlockPos p_77729_) {
       for(int i = 0; i < 21; ++i) {
-         p_77729_.m_122190_(this.f_77690_).m_122175_(Direction.UP, i).m_122175_(this.f_77688_, -1);
+         p_77729_.set(this.f_77690_).move(Direction.UP, i).move(this.f_77688_, -1);
          if (!f_77685_.m_61035_(this.f_77686_.m_8055_(p_77729_), this.f_77686_, p_77729_)) {
             return i;
          }
 
-         p_77729_.m_122190_(this.f_77690_).m_122175_(Direction.UP, i).m_122175_(this.f_77688_, this.f_77692_);
+         p_77729_.set(this.f_77690_).move(Direction.UP, i).move(this.f_77688_, this.f_77692_);
          if (!f_77685_.m_61035_(this.f_77686_.m_8055_(p_77729_), this.f_77686_, p_77729_)) {
             return i;
          }
 
          for(int j = 0; j < this.f_77692_; ++j) {
-            p_77729_.m_122190_(this.f_77690_).m_122175_(Direction.UP, i).m_122175_(this.f_77688_, j);
+            p_77729_.set(this.f_77690_).move(Direction.UP, i).move(this.f_77688_, j);
             BlockState blockstate = this.f_77686_.m_8055_(p_77729_);
             if (!m_77717_(blockstate)) {
                return i;
@@ -167,7 +_,7 @@
 
    public void m_77743_() {
       BlockState blockstate = Blocks.f_50142_.m_49966_().m_61124_(NetherPortalBlock.f_54904_, this.f_77687_);
-      BlockPos.m_121940_(this.f_77690_, this.f_77690_.m_5484_(Direction.UP, this.f_77691_ - 1).m_5484_(this.f_77688_, this.f_77692_ - 1)).forEach((p_77725_) -> {
+      BlockPos.m_121940_(this.f_77690_, this.f_77690_.relative(Direction.UP, this.f_77691_ - 1).relative(this.f_77688_, this.f_77692_ - 1)).forEach((p_77725_) -> {
          this.f_77686_.m_7731_(p_77725_, blockstate, 18);
       });
    }
@@ -182,7 +_,7 @@
       BlockPos blockpos = p_77739_.f_124348_;
       double d2;
       if (d0 > 0.0D) {
-         float f = (float)blockpos.m_123304_(p_77740_) + p_77742_.f_20377_ / 2.0F;
+         float f = (float)blockpos.get(p_77740_) + p_77742_.f_20377_ / 2.0F;
          d2 = Mth.m_14008_(Mth.m_14112_(p_77741_.m_82507_(p_77740_) - (double)f, 0.0D, d0), 0.0D, 1.0D);
       } else {
          d2 = 0.5D;
@@ -191,13 +_,13 @@
       double d4;
       if (d1 > 0.0D) {
          Direction.Axis direction$axis = Direction.Axis.Y;
-         d4 = Mth.m_14008_(Mth.m_14112_(p_77741_.m_82507_(direction$axis) - (double)blockpos.m_123304_(direction$axis), 0.0D, d1), 0.0D, 1.0D);
+         d4 = Mth.m_14008_(Mth.m_14112_(p_77741_.m_82507_(direction$axis) - (double)blockpos.get(direction$axis), 0.0D, d1), 0.0D, 1.0D);
       } else {
          d4 = 0.0D;
       }
 
       Direction.Axis direction$axis1 = p_77740_ == Direction.Axis.X ? Direction.Axis.Z : Direction.Axis.X;
-      double d3 = p_77741_.m_82507_(direction$axis1) - ((double)blockpos.m_123304_(direction$axis1) + 0.5D);
+      double d3 = p_77741_.m_82507_(direction$axis1) - ((double)blockpos.get(direction$axis1) + 0.5D);
       return new Vec3(d2, d4, d3);
    }
 
@@ -209,12 +_,12 @@
       double d1 = (double)p_259931_.f_124350_;
       EntityDimensions entitydimensions = p_259166_.m_6972_(p_259166_.m_20089_());
       int i = p_259901_ == direction$axis ? 0 : 90;
-      Vec3 vec3 = p_259901_ == direction$axis ? p_260043_ : new Vec3(p_260043_.f_82481_, p_260043_.f_82480_, -p_260043_.f_82479_);
+      Vec3 vec3 = p_259901_ == direction$axis ? p_260043_ : new Vec3(p_260043_.m_7094_(), p_260043_.m_7098_(), -p_260043_.m_7096_());
       double d2 = (double)entitydimensions.f_20377_ / 2.0D + (d0 - (double)entitydimensions.f_20377_) * p_259630_.m_7096_();
       double d3 = (d1 - (double)entitydimensions.f_20378_) * p_259630_.m_7098_();
       double d4 = 0.5D + p_259630_.m_7094_();
       boolean flag = direction$axis == Direction.Axis.X;
-      Vec3 vec31 = new Vec3((double)blockpos.m_123341_() + (flag ? d2 : d4), (double)blockpos.m_123342_() + d3, (double)blockpos.m_123343_() + (flag ? d4 : d2));
+      Vec3 vec31 = new Vec3((double)blockpos.getX() + (flag ? d2 : d4), (double)blockpos.getY() + d3, (double)blockpos.getZ() + (flag ? d4 : d2));
       Vec3 vec32 = m_257963_(vec31, p_259301_, p_259166_, entitydimensions);
       return new PortalInfo(vec32, vec3, p_259853_ + (float)i, p_259667_);
    }
