--- a/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.java
+++ b/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.java
@@ -56,7 +_,7 @@
       return this.f_226974_;
    }
 
-   protected boolean m_214090_(ChunkGeneratorStructureState p_256631_, int p_256202_, int p_255915_) {
+   protected boolean isPlacementChunk(ChunkGeneratorStructureState p_256631_, me.alphamode.mclong.math.BigInteger p_256202_, me.alphamode.mclong.math.BigInteger p_255915_) {
       List<ChunkPos> list = p_256631_.m_255182_(this);
       return list == null ? false : list.contains(new ChunkPos(p_256202_, p_255915_));
    }
