--- a/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.java
+++ b/net/minecraft/world/level/levelgen/structure/placement/ConcentricRingsStructurePlacement.java
@@ -57,7 +_,7 @@
       return this.f_226974_;
    }
 
-   protected boolean m_214090_(ChunkGenerator p_226991_, RandomState p_226992_, long p_226993_, int p_226994_, int p_226995_) {
+   protected boolean isPlacementChunk(ChunkGenerator p_226991_, RandomState p_226992_, long p_226993_, long p_226994_, long p_226995_) {
       List<ChunkPos> list = p_226991_.m_223119_(this, p_226992_);
       return list == null ? false : list.contains(new ChunkPos(p_226994_, p_226995_));
    }
