--- a/net/minecraft/world/level/levelgen/structure/placement/StructurePlacement.java
+++ b/net/minecraft/world/level/levelgen/structure/placement/StructurePlacement.java
@@ -59,45 +_,45 @@
       return this.f_227026_;
    }
 
-   public boolean m_227054_(ChunkGenerator p_227055_, RandomState p_227056_, long p_227057_, int p_227058_, int p_227059_) {
-      if (!this.m_214090_(p_227055_, p_227056_, p_227057_, p_227058_, p_227059_)) {
+   public boolean isStructureChunk(ChunkGenerator p_227055_, RandomState p_227056_, long p_227057_, long p_227058_, long p_227059_) {
+      if (!this.isPlacementChunk(p_227055_, p_227056_, p_227057_, p_227058_, p_227059_)) {
          return false;
-      } else if (this.f_227024_ < 1.0F && !this.f_227023_.m_227119_(p_227057_, this.f_227025_, p_227058_, p_227059_, this.f_227024_)) {
+      } else if (this.f_227024_ < 1.0F && !this.f_227023_.shouldGenerate(p_227057_, this.f_227025_, p_227058_, p_227059_, this.f_227024_)) {
          return false;
       } else {
-         return !this.f_227026_.isPresent() || !this.f_227026_.get().m_227087_(p_227055_, p_227056_, p_227057_, p_227058_, p_227059_);
+         return !this.f_227026_.isPresent() || !this.f_227026_.get().isPlacementForbidden(p_227055_, p_227056_, p_227057_, p_227058_, p_227059_);
       }
    }
 
-   protected abstract boolean m_214090_(ChunkGenerator p_227043_, RandomState p_227044_, long p_227045_, int p_227046_, int p_227047_);
+   protected abstract boolean isPlacementChunk(ChunkGenerator p_227043_, RandomState p_227044_, long p_227045_, long p_227046_, long p_227047_);
 
    public BlockPos m_227039_(ChunkPos p_227040_) {
-      return (new BlockPos(p_227040_.m_45604_(), 0, p_227040_.m_45605_())).m_121955_(this.m_227072_());
+      return (new BlockPos(p_227040_.getMinBlockX(), 0, p_227040_.getMinBlockZ())).m_121955_(this.m_227072_());
    }
 
    public abstract StructurePlacementType<?> m_203443_();
 
-   private static boolean m_227033_(long p_227034_, int p_227035_, int p_227036_, int p_227037_, float p_227038_) {
+   private static boolean probabilityReducer(long p_227034_, int p_227035_, long p_227036_, long p_227037_, float p_227038_) {
       WorldgenRandom worldgenrandom = new WorldgenRandom(new LegacyRandomSource(0L));
-      worldgenrandom.m_190058_(p_227034_, p_227035_, p_227036_, p_227037_);
+      worldgenrandom.setLargeFeatureWithSalt(p_227034_, p_227035_, p_227036_, p_227037_);
       return worldgenrandom.m_188501_() < p_227038_;
    }
 
-   private static boolean m_227048_(long p_227049_, int p_227050_, int p_227051_, int p_227052_, float p_227053_) {
+   private static boolean legacyProbabilityReducerWithDouble(long p_227049_, int p_227050_, long p_227051_, long p_227052_, float p_227053_) {
       WorldgenRandom worldgenrandom = new WorldgenRandom(new LegacyRandomSource(0L));
-      worldgenrandom.m_190068_(p_227049_, p_227051_, p_227052_);
+      worldgenrandom.setLargeFeatureSeed(p_227049_, p_227051_, p_227052_);
       return worldgenrandom.m_188500_() < (double)p_227053_;
    }
 
-   private static boolean m_227060_(long p_227061_, int p_227062_, int p_227063_, int p_227064_, float p_227065_) {
+   private static boolean legacyArbitrarySaltProbabilityReducer(long p_227061_, int p_227062_, long p_227063_, long p_227064_, float p_227065_) {
       WorldgenRandom worldgenrandom = new WorldgenRandom(new LegacyRandomSource(0L));
-      worldgenrandom.m_190058_(p_227061_, p_227063_, p_227064_, 10387320);
+      worldgenrandom.setLargeFeatureWithSalt(p_227061_, p_227063_, p_227064_, 10387320);
       return worldgenrandom.m_188501_() < p_227065_;
    }
 
-   private static boolean m_227066_(long p_227067_, int p_227068_, int p_227069_, int p_227070_, float p_227071_) {
-      int i = p_227069_ >> 4;
-      int j = p_227070_ >> 4;
+   private static boolean legacyPillagerOutpostReducer(long p_227067_, int p_227068_, long p_227069_, long p_227070_, float p_227071_) {
+      long i = p_227069_ >> 4;
+      long j = p_227070_ >> 4;
       WorldgenRandom worldgenrandom = new WorldgenRandom(new LegacyRandomSource(0L));
       worldgenrandom.m_188584_((long)(i ^ j << 4) ^ p_227067_);
       worldgenrandom.m_188502_();
@@ -111,21 +_,21 @@
          return p_227086_.group(RegistryFileCodec.m_135592_(Registry.f_211073_, StructureSet.f_210001_, false).fieldOf("other_set").forGetter(StructurePlacement.ExclusionZone::f_227078_), Codec.intRange(1, 16).fieldOf("chunk_count").forGetter(StructurePlacement.ExclusionZone::f_227079_)).apply(p_227086_, StructurePlacement.ExclusionZone::new);
       });
 
-      boolean m_227087_(ChunkGenerator p_227088_, RandomState p_227089_, long p_227090_, int p_227091_, int p_227092_) {
-         return p_227088_.m_223141_(this.f_227078_, p_227089_, p_227090_, p_227091_, p_227092_, this.f_227079_);
+      boolean isPlacementForbidden(ChunkGenerator p_227088_, RandomState p_227089_, long p_227090_, long p_227091_, long p_227092_) {
+         return p_227088_.hasStructureChunkInRange(this.f_227078_, p_227089_, p_227090_, p_227091_, p_227092_, this.f_227079_);
       }
    }
 
    @FunctionalInterface
    public interface FrequencyReducer {
-      boolean m_227098_(long p_227099_, int p_227100_, int p_227101_, int p_227102_, float p_227103_);
+      boolean shouldGenerate(long p_227099_, int p_227100_, long p_227101_, long p_227102_, float p_227103_);
    }
 
    public static enum FrequencyReductionMethod implements StringRepresentable {
-      DEFAULT("default", StructurePlacement::m_227033_),
-      LEGACY_TYPE_1("legacy_type_1", StructurePlacement::m_227066_),
-      LEGACY_TYPE_2("legacy_type_2", StructurePlacement::m_227060_),
-      LEGACY_TYPE_3("legacy_type_3", StructurePlacement::m_227048_);
+      DEFAULT("default", StructurePlacement::probabilityReducer),
+      LEGACY_TYPE_1("legacy_type_1", StructurePlacement::legacyPillagerOutpostReducer),
+      LEGACY_TYPE_2("legacy_type_2", StructurePlacement::legacyArbitrarySaltProbabilityReducer),
+      LEGACY_TYPE_3("legacy_type_3", StructurePlacement::legacyProbabilityReducerWithDouble);
 
       public static final Codec<StructurePlacement.FrequencyReductionMethod> f_227108_ = StringRepresentable.m_216439_(StructurePlacement.FrequencyReductionMethod::values);
       private final String f_227109_;
@@ -136,8 +_,8 @@
          this.f_227110_ = p_227117_;
       }
 
-      public boolean m_227119_(long p_227120_, int p_227121_, int p_227122_, int p_227123_, float p_227124_) {
-         return this.f_227110_.m_227098_(p_227120_, p_227121_, p_227122_, p_227123_, p_227124_);
+      public boolean shouldGenerate(long p_227120_, int p_227121_, long p_227122_, long p_227123_, float p_227124_) {
+         return this.f_227110_.shouldGenerate(p_227120_, p_227121_, p_227122_, p_227123_, p_227124_);
       }
 
       public String m_7912_() {
