--- a/net/minecraft/server/level/ChunkTracker.java
+++ b/net/minecraft/server/level/ChunkTracker.java
@@ -14,12 +_,12 @@
 
    protected void m_7900_(long p_140707_, int p_140708_, boolean p_140709_) {
       ChunkPos chunkpos = new ChunkPos(p_140707_);
-      int i = chunkpos.f_45578_;
-      int j = chunkpos.f_45579_;
+      long i = chunkpos.f_45578_;
+      long j = chunkpos.f_45579_;
 
       for(int k = -1; k <= 1; ++k) {
          for(int l = -1; l <= 1; ++l) {
-            long i1 = ChunkPos.m_45589_(i + k, j + l);
+            long i1 = ChunkPos.asLong(i + k, j + l);
             if (i1 != p_140707_) {
                this.m_75593_(p_140707_, i1, p_140708_, p_140709_);
             }
@@ -31,12 +_,12 @@
    protected int m_6357_(long p_140711_, long p_140712_, int p_140713_) {
       int i = p_140713_;
       ChunkPos chunkpos = new ChunkPos(p_140711_);
-      int j = chunkpos.f_45578_;
-      int k = chunkpos.f_45579_;
+      long j = chunkpos.f_45578_;
+      long k = chunkpos.f_45579_;
 
       for(int l = -1; l <= 1; ++l) {
          for(int i1 = -1; i1 <= 1; ++i1) {
-            long j1 = ChunkPos.m_45589_(j + l, k + i1);
+            long j1 = ChunkPos.asLong(j + l, k + i1);
             if (j1 == p_140711_) {
                j1 = ChunkPos.f_45577_;
             }
