--- a/net/minecraft/server/level/ColumnPos.java
+++ b/net/minecraft/server/level/ColumnPos.java
@@ -3,35 +_,19 @@
 import net.minecraft.core.SectionPos;
 import net.minecraft.world.level.ChunkPos;
 
-public record ColumnPos(int f_140723_, int f_140724_) {
+public record ColumnPos(long f_140723_, long y, long f_140724_) {
    private static final long f_143191_ = 32L;
    private static final long f_143192_ = 4294967295L;
 
    public ChunkPos m_143196_() {
-      return new ChunkPos(SectionPos.m_123171_(this.f_140723_), SectionPos.m_123171_(this.f_140724_));
-   }
-
-   public long m_143200_() {
-      return m_143197_(this.f_140723_, this.f_140724_);
-   }
-
-   public static long m_143197_(int p_143198_, int p_143199_) {
-      return (long)p_143198_ & 4294967295L | ((long)p_143199_ & 4294967295L) << 32;
-   }
-
-   public static int m_214969_(long p_214970_) {
-      return (int)(p_214970_ & 4294967295L);
-   }
-
-   public static int m_214971_(long p_214972_) {
-      return (int)(p_214972_ >>> 32 & 4294967295L);
+      return new ChunkPos(SectionPos.blockToSectionCoord(this.f_140723_), SectionPos.blockToSectionCoord(this.y), SectionPos.blockToSectionCoord(this.f_140724_));
    }
 
    public String toString() {
-      return "[" + this.f_140723_ + ", " + this.f_140724_ + "]";
+      return "[" + this.f_140723_ + ", " + this.y + ", " + this.f_140724_ + "]";
    }
 
    public int hashCode() {
-      return ChunkPos.m_220343_(this.f_140723_, this.f_140724_);
+      return ChunkPos.hash((int) this.f_140723_, (int) this.y, (int) this.f_140724_);
    }
 }
