--- a/net/minecraft/server/level/DemoMode.java
+++ b/net/minecraft/server/level/DemoMode.java
@@ -68,11 +_,11 @@
 
    }
 
-   public void m_214168_(BlockPos p_214976_, ServerboundPlayerActionPacket.Action p_214977_, Direction p_214978_, int p_214979_, int p_214980_) {
+   public void handleBlockBreakAction(BlockPos p_214976_, ServerboundPlayerActionPacket.Action p_214977_, Direction p_214978_, long p_214979_, int p_214980_) {
       if (this.f_140735_) {
          this.m_140757_();
       } else {
-         super.m_214168_(p_214976_, p_214977_, p_214978_, p_214979_, p_214980_);
+         super.handleBlockBreakAction(p_214976_, p_214977_, p_214978_, p_214979_, p_214980_);
       }
    }
 
