--- a/net/minecraft/server/level/SectionTracker.java
+++ b/net/minecraft/server/level/SectionTracker.java
@@ -16,7 +_,7 @@
       for(int i = -1; i <= 1; ++i) {
          for(int j = -1; j <= 1; ++j) {
             for(int k = -1; k <= 1; ++k) {
-               long l = SectionPos.m_123186_(p_8280_, i, j, k);
+               long l = SectionPos.offset(p_8280_, i, j, k);
                if (l != p_8280_) {
                   this.m_75593_(p_8280_, l, p_8281_, p_8282_);
                }
@@ -32,7 +_,7 @@
       for(int j = -1; j <= 1; ++j) {
          for(int k = -1; k <= 1; ++k) {
             for(int l = -1; l <= 1; ++l) {
-               long i1 = SectionPos.m_123186_(p_8284_, j, k, l);
+               long i1 = SectionPos.offset(p_8284_, j, k, l);
                if (i1 == p_8284_) {
                   i1 = Long.MAX_VALUE;
                }
