--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -81,7 +_,7 @@
          this.f_8513_.accept(new ClientboundSetPassengersPacket(this.f_8510_));
          m_277180_(list, this.f_8523_).forEach((p_289307_) -> {
             if (p_289307_ instanceof ServerPlayer serverplayer1) {
-               serverplayer1.f_8906_.m_9774_(serverplayer1.m_20185_(), serverplayer1.m_20186_(), serverplayer1.m_20189_(), serverplayer1.m_146908_(), serverplayer1.m_146909_());
+               serverplayer1.f_8906_.teleport(serverplayer1.getBigX(), serverplayer1.getBigY(), serverplayer1.getBigZ(), serverplayer1.m_146908_(), serverplayer1.m_146909_());
             }
 
          });
