--- a/net/minecraft/server/level/ThreadedLevelLightEngine.java
+++ b/net/minecraft/server/level/ThreadedLevelLightEngine.java
@@ -49,7 +_,7 @@
 
    public void m_7174_(BlockPos p_9357_) {
       BlockPos blockpos = p_9357_.m_7949_();
-      this.m_9312_(SectionPos.m_123171_(p_9357_.m_123341_()), SectionPos.m_123171_(p_9357_.m_123343_()), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
+      this.addTask(SectionPos.blockToSectionCoord(p_9357_.getX()), SectionPos.blockToSectionCoord(p_9357_.getZ()), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_7174_(blockpos);
       }, () -> {
          return "checkBlock " + blockpos;
@@ -57,19 +_,19 @@
    }
 
    protected void m_9330_(ChunkPos p_9331_) {
-      this.m_9317_(p_9331_.f_45578_, p_9331_.f_45579_, () -> {
+      this.addTask(p_9331_.x(), p_9331_.z(), () -> {
          return 0;
       }, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_6462_(p_9331_, false);
          super.m_9335_(p_9331_, false);
 
-         for(int i = this.m_164447_(); i < this.m_164448_(); ++i) {
-            super.m_284126_(LightLayer.BLOCK, SectionPos.m_123196_(p_9331_, i), (DataLayer)null);
-            super.m_284126_(LightLayer.SKY, SectionPos.m_123196_(p_9331_, i), (DataLayer)null);
+         for(long i = this.getMinLightSection(); i < this.getMaxLightSection(); ++i) {
+            super.m_284126_(LightLayer.BLOCK, SectionPos.of(p_9331_, i), (DataLayer)null);
+            super.m_284126_(LightLayer.SKY, SectionPos.of(p_9331_, i), (DataLayer)null);
          }
 
-         for(int j = this.f_164445_.m_151560_(); j < this.f_164445_.m_151561_(); ++j) {
-            super.m_6191_(SectionPos.m_123196_(p_9331_, j), true);
+         for(long j = this.f_164445_.getMinSection(); j < this.f_164445_.getMaxSection(); ++j) {
+            super.m_6191_(SectionPos.of(p_9331_, j), true);
          }
 
       }, () -> {
@@ -78,7 +_,7 @@
    }
 
    public void m_6191_(SectionPos p_9364_, boolean p_9365_) {
-      this.m_9317_(p_9364_.m_123170_(), p_9364_.m_123222_(), () -> {
+      this.addTask(p_9364_.x(), p_9364_.z(), () -> {
          return 0;
       }, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_6191_(p_9364_, p_9365_);
@@ -88,7 +_,7 @@
    }
 
    public void m_142519_(ChunkPos p_285029_) {
-      this.m_9312_(p_285029_.f_45578_, p_285029_.f_45579_, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
+      this.addTask(p_285029_.x(), p_285029_.z(), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_142519_(p_285029_);
       }, () -> {
          return "propagateLight " + p_285029_;
@@ -96,7 +_,7 @@
    }
 
    public void m_9335_(ChunkPos p_9336_, boolean p_9337_) {
-      this.m_9312_(p_9336_.f_45578_, p_9336_.f_45579_, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
+      this.addTask(p_9336_.x(), p_9336_.z(), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_9335_(p_9336_, p_9337_);
       }, () -> {
          return "enableLight " + p_9336_ + " " + p_9337_;
@@ -104,7 +_,7 @@
    }
 
    public void m_284126_(LightLayer p_285046_, SectionPos p_285496_, @Nullable DataLayer p_285495_) {
-      this.m_9317_(p_285496_.m_123170_(), p_285496_.m_123222_(), () -> {
+      this.addTask(p_285496_.x(), p_285496_.z(), () -> {
          return 0;
       }, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_284126_(p_285046_, p_285496_, p_285495_);
@@ -113,22 +_,22 @@
       }));
    }
 
-   private void m_9312_(int p_9313_, int p_9314_, ThreadedLevelLightEngine.TaskType p_9315_, Runnable p_9316_) {
-      this.m_9317_(p_9313_, p_9314_, this.f_9299_.m_140371_(ChunkPos.m_45589_(p_9313_, p_9314_)), p_9315_, p_9316_);
+   private void addTask(long p_9313_, long p_9314_, ThreadedLevelLightEngine.TaskType p_9315_, Runnable p_9316_) {
+      this.addTask(p_9313_, p_9314_, this.f_9299_.getChunkQueueLevel(new ChunkPos(p_9313_, p_9314_)), p_9315_, p_9316_);
    }
 
-   private void m_9317_(int p_9318_, int p_9319_, IntSupplier p_9320_, ThreadedLevelLightEngine.TaskType p_9321_, Runnable p_9322_) {
-      this.f_9300_.m_6937_(ChunkTaskPriorityQueueSorter.m_140624_(() -> {
+   private void addTask(long p_9318_, long p_9319_, IntSupplier p_9320_, ThreadedLevelLightEngine.TaskType p_9321_, Runnable p_9322_) {
+      this.f_9300_.m_6937_(ChunkTaskPriorityQueueSorter.message(() -> {
          this.f_9298_.add(Pair.of(p_9321_, p_9322_));
          if (this.f_9298_.size() >= 1000) {
             this.m_9366_();
          }
 
-      }, ChunkPos.m_45589_(p_9318_, p_9319_), p_9320_));
+      }, new ChunkPos(p_9318_, p_9319_), p_9320_));
    }
 
    public void m_6462_(ChunkPos p_9370_, boolean p_9371_) {
-      this.m_9317_(p_9370_.f_45578_, p_9370_.f_45579_, () -> {
+      this.addTask(p_9370_.x(), p_9370_.z(), () -> {
          return 0;
       }, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          super.m_6462_(p_9370_, p_9371_);
@@ -139,14 +_,14 @@
 
    public CompletableFuture<ChunkAccess> m_284138_(ChunkAccess p_285128_, boolean p_285441_) {
       ChunkPos chunkpos = p_285128_.m_7697_();
-      this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
+      this.addTask(chunkpos.x(), chunkpos.z(), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          LevelChunkSection[] alevelchunksection = p_285128_.m_7103_();
 
          for(int i = 0; i < p_285128_.m_151559_(); ++i) {
             LevelChunkSection levelchunksection = alevelchunksection[i];
             if (!levelchunksection.m_188008_()) {
-               int j = this.f_164445_.m_151568_(i);
-               super.m_6191_(SectionPos.m_123196_(chunkpos, j), false);
+               long j = this.f_164445_.getSectionYFromSectionIndex(i);
+               super.m_6191_(SectionPos.of(chunkpos, j), false);
             }
          }
 
@@ -158,14 +_,14 @@
          super.m_6462_(chunkpos, false);
          return p_285128_;
       }, (p_215135_) -> {
-         this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_215135_);
+         this.addTask(chunkpos.x(), chunkpos.z(), ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_215135_);
       });
    }
 
    public CompletableFuture<ChunkAccess> m_9353_(ChunkAccess p_9354_, boolean p_9355_) {
       ChunkPos chunkpos = p_9354_.m_7697_();
       p_9354_.m_8094_(false);
-      this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
+      this.addTask(chunkpos.x(), chunkpos.z(), ThreadedLevelLightEngine.TaskType.PRE_UPDATE, Util.m_137474_(() -> {
          if (!p_9355_) {
             super.m_142519_(chunkpos);
          }
@@ -178,7 +_,7 @@
          this.f_9299_.m_140375_(chunkpos);
          return p_9354_;
       }, (p_280982_) -> {
-         this.m_9312_(chunkpos.f_45578_, chunkpos.f_45579_, ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_280982_);
+         this.addTask(chunkpos.x(), chunkpos.z(), ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_280982_);
       });
    }
 
