--- a/net/minecraft/server/level/TicketType.java
+++ b/net/minecraft/server/level/TicketType.java
@@ -19,7 +_,7 @@
    public static final TicketType<ChunkPos> f_9444_ = m_9462_("player", Comparator.comparingLong(ChunkPos::m_45588_));
    public static final TicketType<ChunkPos> f_9445_ = m_9462_("forced", Comparator.comparingLong(ChunkPos::m_45588_));
    public static final TicketType<ChunkPos> f_9446_ = m_9462_("light", Comparator.comparingLong(ChunkPos::m_45588_));
-   public static final TicketType<BlockPos> f_9447_ = m_9465_("portal", Vec3i::compareTo, 300);
+   public static final TicketType<BlockPos> f_9447_ = m_9465_("portal", me.alphamode.mclong.core.Vec3l::compareTo, 300);
    public static final TicketType<Integer> f_9448_ = m_9465_("post_teleport", Integer::compareTo, 5);
    public static final TicketType<ChunkPos> f_9449_ = m_9465_("unknown", Comparator.comparingLong(ChunkPos::m_45588_), 1);
 
