--- a/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
+++ b/net/minecraft/world/item/enchantment/FrostWalkerEnchantment.java
@@ -38,15 +_,15 @@
          float f = (float)Math.min(16, 2 + p_45022_);
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-         for(BlockPos blockpos : BlockPos.m_121940_(p_45021_.m_7637_((double)(-f), -1.0D, (double)(-f)), p_45021_.m_7637_((double)f, -1.0D, (double)f))) {
-            if (blockpos.m_203195_(p_45019_.m_20182_(), (double)f)) {
-               blockpos$mutableblockpos.m_122178_(blockpos.m_123341_(), blockpos.m_123342_() + 1, blockpos.m_123343_());
+         for(BlockPos blockpos : BlockPos.m_121940_(p_45021_.offset((double)(-f), -1.0D, (double)(-f)), p_45021_.offset((double)f, -1.0D, (double)f))) {
+            if (blockpos.closerToCenterThan(p_45019_.m_20182_(), (double)f)) {
+               blockpos$mutableblockpos.set(blockpos.getX(), blockpos.getY() + 1, blockpos.getZ());
                BlockState blockstate1 = p_45020_.m_8055_(blockpos$mutableblockpos);
                if (blockstate1.m_60795_()) {
                   BlockState blockstate2 = p_45020_.m_8055_(blockpos);
                   if (blockstate2.m_60767_() == Material.f_76305_ && blockstate2.m_61143_(LiquidBlock.f_54688_) == 0 && blockstate.m_60710_(p_45020_, blockpos) && p_45020_.m_45752_(blockstate, blockpos, CollisionContext.m_82749_())) {
                      p_45020_.m_46597_(blockpos, blockstate);
-                     p_45020_.m_186460_(blockpos, Blocks.f_50449_, Mth.m_216271_(p_45019_.m_217043_(), 60, 120));
+                     p_45020_.scheduleTick(blockpos, Blocks.f_50449_, Mth.m_216271_(p_45019_.m_217043_(), 60, 120));
                   }
                }
             }
