--- a/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
@@ -34,8 +_,8 @@
       return this.f_161947_;
    }
 
-   public int m_213859_(RandomSource p_226300_, WorldGenerationContext p_226301_) {
-      return this.f_161947_.m_142322_(p_226301_);
+   public long sample(RandomSource p_226300_, WorldGenerationContext p_226301_) {
+      return this.f_161947_.resolveY(p_226301_);
    }
 
    public HeightProviderType<?> m_142002_() {
