--- a/net/minecraft/world/level/levelgen/heightproviders/HeightProvider.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/HeightProvider.java
@@ -17,7 +_,7 @@
       return p_161976_.m_142002_() == HeightProviderType.f_161981_ ? Either.left(((ConstantHeight)p_161976_).m_161963_()) : Either.right(p_161976_);
    });
 
-   public abstract int m_213859_(RandomSource p_226302_, WorldGenerationContext p_226303_);
+   public abstract long sample(RandomSource pRandom, WorldGenerationContext pContext);
 
    public abstract HeightProviderType<?> m_142002_();
 }
