--- a/net/minecraft/world/level/levelgen/heightproviders/WeightedListHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/WeightedListHeight.java
@@ -18,8 +_,8 @@
       this.f_191533_ = p_191536_;
    }
 
-   public int m_213859_(RandomSource p_226314_, WorldGenerationContext p_226315_) {
-      return this.f_191533_.m_216820_(p_226314_).orElseThrow(IllegalStateException::new).m_213859_(p_226314_, p_226315_);
+   public long sample(RandomSource p_226314_, WorldGenerationContext p_226315_) {
+      return this.f_191533_.m_216820_(p_226314_).orElseThrow(IllegalStateException::new).sample(p_226314_, p_226315_);
    }
 
    public HeightProviderType<?> m_142002_() {
