--- a/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
+++ b/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
@@ -77,16 +_,17 @@
    public static final String f_223447_ = "SkyLight";
 
    public static ProtoChunk m_188230_(ServerLevel p_188231_, PoiManager p_188232_, ChunkPos p_188233_, CompoundTag p_188234_) {
-      ChunkPos chunkpos = new ChunkPos(p_188234_.m_128451_("xPos"), p_188234_.m_128451_("zPos"));
+      ChunkPos chunkpos = new ChunkPos(p_188234_.m_128454_(f_223441_), p_188234_.m_128454_("yPos"), p_188234_.m_128454_(f_223442_));
       if (!Objects.equals(p_188233_, chunkpos)) {
          f_63449_.error("Chunk file at {} is in the wrong location; relocating. (Expected {}, got {})", p_188233_, p_188233_, chunkpos);
       }
 
       UpgradeData upgradedata = p_188234_.m_128425_("UpgradeData", 10) ? new UpgradeData(p_188234_.m_128469_("UpgradeData"), p_188231_) : UpgradeData.f_63320_;
       boolean flag = p_188234_.m_128471_("isLightOn");
-      ListTag listtag = p_188234_.m_128437_("sections", 10);
-      int i = p_188231_.m_151559_();
-      LevelChunkSection[] alevelchunksection = new LevelChunkSection[i];
+//      ListTag listtag = p_188234_.getList("sections", 10);
+      CompoundTag section = p_188234_.m_128469_("section");
+//      long i = p_188231_.getSectionsCount();
+//      LevelChunkSection[] alevelchunksection = new LevelChunkSection[(int) i];
       boolean flag1 = p_188231_.m_6042_().f_223549_();
       ChunkSource chunksource = p_188231_.m_7726_();
       LevelLightEngine levellightengine = chunksource.m_7827_();
@@ -94,11 +_,11 @@
       Codec<PalettedContainerRO<Holder<Biome>>> codec = m_188260_(registry);
       boolean flag2 = false;
 
-      for(int j = 0; j < listtag.size(); ++j) {
-         CompoundTag compoundtag = listtag.m_128728_(j);
+//      for(int j = 0; j < listtag.size(); ++j) {
+         CompoundTag compoundtag = section;//listtag.getCompound(j);
          int k = compoundtag.m_128445_("Y");
-         int l = p_188231_.m_151566_(k);
-         if (l >= 0 && l < alevelchunksection.length) {
+         long l = p_188231_.getSectionIndexFromSectionY(k);
+//         if (l >= 0 && l < alevelchunksection.length) {
             PalettedContainer<BlockState> palettedcontainer;
             if (compoundtag.m_128425_("block_states", 10)) {
                palettedcontainer = f_188227_.parse(NbtOps.f_128958_, compoundtag.m_128469_("block_states")).promotePartial((p_188283_) -> {
@@ -118,10 +_,10 @@
             }
 
             LevelChunkSection levelchunksection = new LevelChunkSection(palettedcontainer, palettedcontainerro);
-            alevelchunksection[l] = levelchunksection;
-            SectionPos sectionpos = SectionPos.m_123196_(p_188233_, k);
+            SectionPos sectionpos = SectionPos.of(p_188233_, k);
+//            alevelchunksection[(int) l] = levelchunksection;
             p_188232_.m_280570_(sectionpos, levelchunksection);
-         }
+//         }
 
          boolean flag3 = compoundtag.m_128425_("BlockLight", 7);
          boolean flag4 = flag1 && compoundtag.m_128425_("SkyLight", 7);
@@ -132,14 +_,14 @@
             }
 
             if (flag3) {
-               levellightengine.m_284126_(LightLayer.BLOCK, SectionPos.m_123196_(p_188233_, k), new DataLayer(compoundtag.m_128463_("BlockLight")));
+               levellightengine.m_284126_(LightLayer.BLOCK, SectionPos.of(p_188233_, k), new DataLayer(compoundtag.m_128463_("BlockLight")));
             }
 
             if (flag4) {
-               levellightengine.m_284126_(LightLayer.SKY, SectionPos.m_123196_(p_188233_, k), new DataLayer(compoundtag.m_128463_("SkyLight")));
+               levellightengine.m_284126_(LightLayer.SKY, SectionPos.of(p_188233_, k), new DataLayer(compoundtag.m_128463_("SkyLight")));
             }
          }
-      }
+//      }
 
       long k1 = p_188234_.m_128454_("InhabitedTime");
       ChunkStatus.ChunkType chunkstatus$chunktype = m_63485_(p_188234_);
@@ -158,7 +_,7 @@
          LevelChunkTicks<Fluid> levelchunkticks1 = LevelChunkTicks.m_193185_(p_188234_.m_128437_("fluid_ticks", 10), (p_258990_) -> {
             return BuiltInRegistries.f_257020_.m_6612_(ResourceLocation.m_135820_(p_258990_));
          }, p_188233_);
-         chunkaccess = new LevelChunk(p_188231_.m_6018_(), p_188233_, upgradedata, levelchunkticks, levelchunkticks1, k1, alevelchunksection, m_196890_(p_188231_, p_188234_), blendingdata);
+         chunkaccess = new LevelChunk(p_188231_.m_6018_(), p_188233_, upgradedata, levelchunkticks, levelchunkticks1, k1, levelchunksection, m_196890_(p_188231_, p_188234_), blendingdata);
       } else {
          ProtoChunkTicks<Block> protochunkticks = ProtoChunkTicks.m_193302_(p_188234_.m_128437_("block_ticks", 10), (p_258992_) -> {
             return BuiltInRegistries.f_256975_.m_6612_(ResourceLocation.m_135820_(p_258992_));
@@ -166,7 +_,7 @@
          ProtoChunkTicks<Fluid> protochunkticks1 = ProtoChunkTicks.m_193302_(p_188234_.m_128437_("fluid_ticks", 10), (p_258991_) -> {
             return BuiltInRegistries.f_257020_.m_6612_(ResourceLocation.m_135820_(p_258991_));
          }, p_188233_);
-         ProtoChunk protochunk = new ProtoChunk(p_188233_, upgradedata, alevelchunksection, protochunkticks, protochunkticks1, p_188231_, registry, blendingdata);
+         ProtoChunk protochunk = new ProtoChunk(p_188233_, upgradedata, levelchunksection, protochunkticks, protochunkticks1, p_188231_, registry, blendingdata);
          chunkaccess = protochunk;
          protochunk.m_6141_(k1);
          if (p_188234_.m_128425_("below_zero_retrogen", 10)) {
@@ -232,7 +_,7 @@
 
          for(String s1 : compoundtag4.m_128431_()) {
             GenerationStep.Carving generationstep$carving = GenerationStep.Carving.valueOf(s1);
-            protochunk1.m_188186_(generationstep$carving, new CarvingMask(compoundtag4.m_128467_(s1), chunkaccess.m_141937_()));
+            protochunk1.m_188186_(generationstep$carving, new CarvingMask(compoundtag4.m_128467_(s1), chunkaccess.getMinBuildHeight()));
          }
 
          return protochunk1;
@@ -250,9 +_,9 @@
    public static CompoundTag m_63454_(ServerLevel p_63455_, ChunkAccess p_63456_) {
       ChunkPos chunkpos = p_63456_.m_7697_();
       CompoundTag compoundtag = NbtUtils.m_264171_(new CompoundTag());
-      compoundtag.m_128405_("xPos", chunkpos.f_45578_);
-      compoundtag.m_128405_("yPos", p_63456_.m_151560_());
-      compoundtag.m_128405_("zPos", chunkpos.f_45579_);
+      compoundtag.m_128356_("xPos", chunkpos.f_45578_);
+      compoundtag.m_128356_("yPos", chunkpos.y);
+      compoundtag.m_128356_("zPos", chunkpos.f_45579_);
       compoundtag.m_128356_("LastUpdate", p_63455_.m_46467_());
       compoundtag.m_128356_("InhabitedTime", p_63456_.m_6319_());
       compoundtag.m_128359_("Status", BuiltInRegistries.f_256940_.m_7981_(p_63456_.m_6415_()).toString());
@@ -282,15 +_,15 @@
       Codec<PalettedContainerRO<Holder<Biome>>> codec = m_188260_(registry);
       boolean flag = p_63456_.m_6332_();
 
-      for(int i = levellightengine.m_164447_(); i < levellightengine.m_164448_(); ++i) {
-         int j = p_63456_.m_151566_(i);
+      for(long i = levellightengine.getMinLightSection(); i < levellightengine.getMaxLightSection(); ++i) {
+         long j = p_63456_.getSectionIndexFromSectionY(i);
          boolean flag1 = j >= 0 && j < alevelchunksection.length;
-         DataLayer datalayer = levellightengine.m_75814_(LightLayer.BLOCK).m_8079_(SectionPos.m_123196_(chunkpos, i));
-         DataLayer datalayer1 = levellightengine.m_75814_(LightLayer.SKY).m_8079_(SectionPos.m_123196_(chunkpos, i));
+         DataLayer datalayer = levellightengine.m_75814_(LightLayer.BLOCK).m_8079_(SectionPos.of(chunkpos, i));
+         DataLayer datalayer1 = levellightengine.m_75814_(LightLayer.SKY).m_8079_(SectionPos.of(chunkpos, i));
          if (flag1 || datalayer != null || datalayer1 != null) {
             CompoundTag compoundtag1 = new CompoundTag();
             if (flag1) {
-               LevelChunkSection levelchunksection = alevelchunksection[j];
+               LevelChunkSection levelchunksection = alevelchunksection[(int) j];
                compoundtag1.m_128365_("block_states", f_188227_.encodeStart(NbtOps.f_128958_, levelchunksection.m_63019_()).getOrThrow(false, f_63449_::error));
                compoundtag1.m_128365_("biomes", codec.encodeStart(NbtOps.f_128958_, levelchunksection.m_187996_()).getOrThrow(false, f_63449_::error));
             }
@@ -405,7 +_,7 @@
       return listtag.isEmpty() ? null : listtag;
    }
 
-   private static CompoundTag m_188249_(StructurePieceSerializationContext p_188250_, ChunkPos p_188251_, Map<Structure, StructureStart> p_188252_, Map<Structure, LongSet> p_188253_) {
+   private static CompoundTag m_188249_(StructurePieceSerializationContext p_188250_, ChunkPos p_188251_, Map<Structure, StructureStart> p_188252_, Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> p_188253_) {
       CompoundTag compoundtag = new CompoundTag();
       CompoundTag compoundtag1 = new CompoundTag();
       Registry<Structure> registry = p_188250_.f_192763_().m_175515_(Registries.f_256944_);
@@ -418,10 +_,17 @@
       compoundtag.m_128365_("starts", compoundtag1);
       CompoundTag compoundtag2 = new CompoundTag();
 
-      for(Map.Entry<Structure, LongSet> entry1 : p_188253_.entrySet()) {
+      for(Map.Entry<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> entry1 : p_188253_.entrySet()) {
          if (!entry1.getValue().isEmpty()) {
             ResourceLocation resourcelocation1 = registry.m_7981_(entry1.getKey());
-            compoundtag2.m_128365_(resourcelocation1.toString(), new LongArrayTag(entry1.getValue()));
+            net.minecraft.nbt.ListTag chunks = new net.minecraft.nbt.ListTag();
+            entry1.getValue().forEach(chunkPos -> {
+               CompoundTag chunk = new CompoundTag();
+               chunk.m_128356_("x", chunkPos.f_45578_);
+               chunk.m_128356_("z", chunkPos.f_45579_);
+               chunks.add(chunk);
+            });
+            compoundtag2.m_128365_(resourcelocation1.toString(), chunks);
          }
       }
 
@@ -450,8 +_,8 @@
       return map;
    }
 
-   private static Map<Structure, LongSet> m_208154_(RegistryAccess p_208155_, ChunkPos p_208156_, CompoundTag p_208157_) {
-      Map<Structure, LongSet> map = Maps.newHashMap();
+   private static Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> m_208154_(RegistryAccess p_208155_, ChunkPos p_208156_, CompoundTag p_208157_) {
+      Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> map = Maps.newHashMap();
       Registry<Structure> registry = p_208155_.m_175515_(Registries.f_256944_);
       CompoundTag compoundtag = p_208157_.m_128469_("References");
 
@@ -461,17 +_,18 @@
          if (structure == null) {
             f_63449_.warn("Found reference to unknown structure '{}' in chunk {}, discarding", resourcelocation, p_208156_);
          } else {
-            long[] along = compoundtag.m_128467_(s);
-            if (along.length != 0) {
-               map.put(structure, new LongOpenHashSet(Arrays.stream(along).filter((p_208153_) -> {
-                  ChunkPos chunkpos = new ChunkPos(p_208153_);
-                  if (chunkpos.m_45594_(p_208156_) > 8) {
+            net.minecraft.nbt.ListTag along = compoundtag.m_128437_(s, 7);
+            if (along.size() != 0) {
+               it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> chunks = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
+               along.forEach((p_208153_) -> {
+                  ChunkPos chunkpos = new ChunkPos(((CompoundTag) p_208153_).m_128454_("x"), ((CompoundTag) p_208153_).m_128454_("y"), ((CompoundTag) p_208153_).m_128454_("z"));
+                  if (chunkpos.getChessboardDistance(p_208156_) > 8) {
                      f_63449_.warn("Found invalid structure reference [ {} @ {} ] for chunk {}.", resourcelocation, chunkpos, p_208156_);
-                     return false;
                   } else {
-                     return true;
+                     chunks.add(chunkpos);
                   }
-               }).toArray()));
+               });
+               map.put(structure, chunks);
             }
          }
       }
