--- a/net/minecraft/world/level/chunk/storage/ChunkStorage.java
+++ b/net/minecraft/world/level/chunk/storage/ChunkStorage.java
@@ -31,8 +_,8 @@
       this.f_63495_ = new IOWorker(p_196912_, p_196914_, "chunk");
    }
 
-   public boolean m_223451_(ChunkPos p_223452_, int p_223453_) {
-      return this.f_63495_.m_223471_(p_223452_, p_223453_);
+   public boolean isOldChunkAround(ChunkPos p_223452_, long p_223453_) {
+      return this.f_63495_.isOldChunkAround(p_223452_, p_223453_);
    }
 
    public CompoundTag m_188288_(ResourceKey<Level> p_188289_, Supplier<DimensionDataStorage> p_188290_, CompoundTag p_188291_, Optional<ResourceKey<Codec<? extends ChunkGenerator>>> p_188292_) {
