--- a/net/minecraft/world/level/chunk/storage/EntityStorage.java
+++ b/net/minecraft/world/level/chunk/storage/EntityStorage.java
@@ -72,7 +_,7 @@
    }
 
    private static void m_156562_(CompoundTag p_156563_, ChunkPos p_156564_) {
-      p_156563_.m_128365_("Position", new IntArrayTag(new int[]{p_156564_.f_45578_, p_156564_.f_45579_}));
+      p_156563_.m_128365_("Position", new net.minecraft.nbt.LongArrayTag(new long[]{p_156564_.f_45578_, p_156564_.f_45579_}));
    }
 
    private static ChunkEntities<Entity> m_156568_(ChunkPos p_156569_) {
