--- a/net/minecraft/world/entity/vehicle/AbstractMinecart.java
+++ b/net/minecraft/world/entity/vehicle/AbstractMinecart.java
@@ -77,9 +_,9 @@
       p_38135_.put(RailShape.NORTH_EAST, Pair.of(vec3i2, vec3i1));
    });
    private int f_38070_;
-   private double f_38071_;
-   private double f_38072_;
-   private double f_38073_;
+   private me.alphamode.mclong.math.BigDecimal f_38071_;
+   private me.alphamode.mclong.math.BigDecimal f_38072_;
+   private me.alphamode.mclong.math.BigDecimal f_38073_;
    private double f_38074_;
    private double f_38075_;
    private double f_38076_;
@@ -94,9 +_,9 @@
    protected AbstractMinecart(EntityType<?> p_38090_, Level p_38091_, double p_38092_, double p_38093_, double p_38094_) {
       this(p_38090_, p_38091_);
       this.m_6034_(p_38092_, p_38093_, p_38094_);
-      this.f_19854_ = p_38092_;
-      this.f_19855_ = p_38093_;
-      this.f_19856_ = p_38094_;
+      this.f_19854_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_38092_);
+      this.f_19855_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_38093_);
+      this.f_19856_ = me.alphamode.mclong.math.BigDecimal.valueOf(p_38094_);
    }
 
    public static AbstractMinecart m_38119_(Level p_38120_, double p_38121_, double p_38122_, double p_38123_, AbstractMinecart.Type p_38124_) {
@@ -160,13 +_,13 @@
 
             for(int i : f_38067_.get(pose)) {
                for(int[] aint1 : aint) {
-                  blockpos$mutableblockpos.m_122178_(blockpos.m_123341_() + aint1[0], blockpos.m_123342_() + i, blockpos.m_123343_() + aint1[1]);
+                  blockpos$mutableblockpos.set(blockpos.getX() + aint1[0], blockpos.getY() + i, blockpos.getZ() + aint1[1]);
                   double d0 = this.m_9236_().m_45564_(DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos), () -> {
-                     return DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos.m_7495_());
+                     return DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos.below());
                   });
                   if (DismountHelper.m_38439_(d0)) {
                      AABB aabb = new AABB((double)(-f), 0.0D, (double)(-f), (double)f, (double)entitydimensions.f_20378_, (double)f);
-                     Vec3 vec3 = Vec3.m_82514_(blockpos$mutableblockpos, d0);
+                     Vec3 vec3 = Vec3.upFromBottomCenterOf(blockpos$mutableblockpos, d0);
                      if (DismountHelper.m_38456_(this.m_9236_(), p_38145_, aabb.m_82383_(vec3))) {
                         p_38145_.m_20124_(pose);
                         return vec3;
@@ -176,13 +_,13 @@
             }
          }
 
-         double d1 = this.m_20191_().f_82292_;
-         blockpos$mutableblockpos.m_122169_((double)blockpos.m_123341_(), d1, (double)blockpos.m_123343_());
+         double d1 = this.m_20191_().f_82292_.doubleValue();
+         blockpos$mutableblockpos.m_122169_((double)blockpos.getX(), d1, (double)blockpos.getZ());
 
          for(Pose pose1 : immutablelist) {
             double d2 = (double)p_38145_.m_6972_(pose1).f_20378_;
-            int j = Mth.m_14165_(d1 - (double)blockpos$mutableblockpos.m_123342_() + d2);
-            double d3 = DismountHelper.m_38463_(blockpos$mutableblockpos, j, (p_289495_) -> {
+            long j = Mth.lceil(d1 - (double)blockpos$mutableblockpos.getY() + d2);
+            double d3 = DismountHelper.findCeilingFrom(blockpos$mutableblockpos, j, (p_289495_) -> {
                return this.m_9236_().m_8055_(p_289495_).m_60812_(this.m_9236_(), p_289495_);
             });
             if (d1 + d2 <= d3) {
@@ -273,14 +_,14 @@
       this.m_20157_();
       if (this.m_9236_().f_46443_) {
          if (this.f_38070_ > 0) {
-            double d5 = this.m_20185_() + (this.f_38071_ - this.m_20185_()) / (double)this.f_38070_;
-            double d6 = this.m_20186_() + (this.f_38072_ - this.m_20186_()) / (double)this.f_38070_;
-            double d7 = this.m_20189_() + (this.f_38073_ - this.m_20189_()) / (double)this.f_38070_;
+            me.alphamode.mclong.math.BigDecimal d5 = this.getBigX().add((this.f_38071_.subtract(this.getBigX())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
+            me.alphamode.mclong.math.BigDecimal d6 = this.getBigY().add((this.f_38072_.subtract(this.getBigY())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
+            me.alphamode.mclong.math.BigDecimal d7 = this.getBigZ().add((this.f_38073_.subtract(this.getBigZ())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
             double d2 = Mth.m_14175_(this.f_38074_ - (double)this.m_146908_());
             this.m_146922_(this.m_146908_() + (float)d2 / (float)this.f_38070_);
             this.m_146926_(this.m_146909_() + (float)(this.f_38075_ - (double)this.m_146909_()) / (float)this.f_38070_);
             --this.f_38070_;
-            this.m_6034_(d5, d6, d7);
+            this.setPos(d5, d6, d7);
             this.m_19915_(this.m_146908_(), this.m_146909_());
          } else {
             this.m_20090_();
@@ -293,9 +_,9 @@
             this.m_20256_(this.m_20184_().m_82520_(0.0D, d0, 0.0D));
          }
 
-         int k = Mth.m_14107_(this.m_20185_());
-         int i = Mth.m_14107_(this.m_20186_());
-         int j = Mth.m_14107_(this.m_20189_());
+         long k = Mth.m_14134_(this.m_20185_());
+         long i = Mth.m_14134_(this.m_20186_());
+         long j = Mth.m_14134_(this.m_20189_());
          if (this.m_9236_().m_8055_(new BlockPos(k, i - 1, j)).m_204336_(BlockTags.f_13034_)) {
             --i;
          }
@@ -306,7 +_,7 @@
          if (this.f_287786_) {
             this.m_6401_(blockpos, blockstate);
             if (blockstate.m_60713_(Blocks.f_50285_)) {
-               this.m_6025_(k, i, j, blockstate.m_61143_(PoweredRailBlock.f_55215_));
+               this.activateMinecart(k, i, j, blockstate.m_61143_(PoweredRailBlock.f_55215_));
             }
          } else {
             this.m_38163_();
@@ -314,8 +_,8 @@
 
          this.m_20101_();
          this.m_146926_(0.0F);
-         double d1 = this.f_19854_ - this.m_20185_();
-         double d3 = this.f_19856_ - this.m_20189_();
+         double d1 = this.f_19854_.subtract(this.getBigX()).doubleValue();
+         double d3 = this.f_19856_.subtract(this.getBigZ()).doubleValue();
          if (d1 * d1 + d3 * d3 > 0.001D) {
             this.m_146922_((float)(Mth.m_14136_(d3, d1) * 180.0D / Math.PI));
             if (this.f_38068_) {
@@ -364,13 +_,13 @@
       return (this.m_20069_() ? 4.0D : 8.0D) / 20.0D;
    }
 
-   public void m_6025_(int p_38111_, int p_38112_, int p_38113_, boolean p_38114_) {
+   public void activateMinecart(long p_38111_, long p_38112_, long p_38113_, boolean p_38114_) {
    }
 
    protected void m_38163_() {
       double d0 = this.m_7097_();
       Vec3 vec3 = this.m_20184_();
-      this.m_20334_(Mth.m_14008_(vec3.f_82479_, -d0, d0), vec3.f_82480_, Mth.m_14008_(vec3.f_82481_, -d0, d0));
+      this.m_20334_(Mth.m_14008_(vec3.m_7096_(), -d0, d0), vec3.m_7098_(), Mth.m_14008_(vec3.m_7094_(), -d0, d0));
       if (this.m_20096_()) {
          this.m_20256_(this.m_20184_().m_82490_(0.5D));
       }
@@ -388,7 +_,7 @@
       double d1 = this.m_20186_();
       double d2 = this.m_20189_();
       Vec3 vec3 = this.m_38179_(d0, d1, d2);
-      d1 = (double)p_38156_.m_123342_();
+      d1 = (double)p_38156_.getY();
       boolean flag = false;
       boolean flag1 = false;
       if (p_38157_.m_60713_(Blocks.f_50030_)) {
@@ -428,14 +_,14 @@
       double d4 = (double)(vec3i1.m_123341_() - vec3i.m_123341_());
       double d5 = (double)(vec3i1.m_123343_() - vec3i.m_123343_());
       double d6 = Math.sqrt(d4 * d4 + d5 * d5);
-      double d7 = vec31.f_82479_ * d4 + vec31.f_82481_ * d5;
+      double d7 = vec31.m_7096_() * d4 + vec31.m_7094_() * d5;
       if (d7 < 0.0D) {
          d4 = -d4;
          d5 = -d5;
       }
 
       double d8 = Math.min(2.0D, vec31.m_165924_());
-      vec31 = new Vec3(d8 * d4 / d6, vec31.f_82480_, d8 * d5 / d6);
+      vec31 = new Vec3(d8 * d4 / d6, vec31.m_7098_(), d8 * d5 / d6);
       this.m_20256_(vec31);
       Entity entity = this.m_146895_();
       if (entity instanceof Player) {
@@ -443,7 +_,7 @@
          double d9 = vec32.m_165925_();
          double d11 = this.m_20184_().m_165925_();
          if (d9 > 1.0E-4D && d11 < 0.01D) {
-            this.m_20256_(this.m_20184_().m_82520_(vec32.f_82479_ * 0.1D, 0.0D, vec32.f_82481_ * 0.1D));
+            this.m_20256_(this.m_20184_().m_82520_(vec32.m_7096_() * 0.1D, 0.0D, vec32.m_7094_() * 0.1D));
             flag1 = false;
          }
       }
@@ -457,17 +_,17 @@
          }
       }
 
-      double d23 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i.m_123341_() * 0.5D;
-      double d10 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i.m_123343_() * 0.5D;
-      double d12 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
-      double d13 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
+      double d23 = (double)p_38156_.getX() + 0.5D + (double)vec3i.m_123341_() * 0.5D;
+      double d10 = (double)p_38156_.getZ() + 0.5D + (double)vec3i.m_123343_() * 0.5D;
+      double d12 = (double)p_38156_.getX() + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
+      double d13 = (double)p_38156_.getZ() + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
       d4 = d12 - d23;
       d5 = d13 - d10;
       double d14;
       if (d4 == 0.0D) {
-         d14 = d2 - (double)p_38156_.m_123343_();
+         d14 = d2 - (double)p_38156_.getZ();
       } else if (d5 == 0.0D) {
-         d14 = d0 - (double)p_38156_.m_123341_();
+         d14 = d0 - (double)p_38156_.getX();
       } else {
          double d15 = d0 - d23;
          double d16 = d2 - d10;
@@ -480,32 +_,32 @@
       double d24 = this.m_20160_() ? 0.75D : 1.0D;
       double d25 = this.m_7097_();
       vec31 = this.m_20184_();
-      this.m_6478_(MoverType.SELF, new Vec3(Mth.m_14008_(d24 * vec31.f_82479_, -d25, d25), 0.0D, Mth.m_14008_(d24 * vec31.f_82481_, -d25, d25)));
-      if (vec3i.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i.m_123343_()) {
+      this.m_6478_(MoverType.SELF, new Vec3(Mth.m_14008_(d24 * vec31.m_7096_(), -d25, d25), 0.0D, Mth.m_14008_(d24 * vec31.m_7094_(), -d25, d25)));
+      if (vec3i.m_123342_() != 0 && Mth.m_14134_(this.m_20185_()) - p_38156_.getX() == vec3i.m_123341_() && Mth.m_14134_(this.m_20189_()) - p_38156_.getZ() == vec3i.m_123343_()) {
          this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i.m_123342_(), this.m_20189_());
-      } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i1.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i1.m_123343_()) {
+      } else if (vec3i1.m_123342_() != 0 && Mth.m_14134_(this.m_20185_()) - p_38156_.getX() == vec3i1.m_123341_() && Mth.m_14134_(this.m_20189_()) - p_38156_.getZ() == vec3i1.m_123343_()) {
          this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i1.m_123342_(), this.m_20189_());
       }
 
       this.m_7114_();
       Vec3 vec33 = this.m_38179_(this.m_20185_(), this.m_20186_(), this.m_20189_());
       if (vec33 != null && vec3 != null) {
-         double d17 = (vec3.f_82480_ - vec33.f_82480_) * 0.05D;
+         double d17 = (vec3.m_7098_() - vec33.m_7098_()) * 0.05D;
          Vec3 vec34 = this.m_20184_();
          double d18 = vec34.m_165924_();
          if (d18 > 0.0D) {
             this.m_20256_(vec34.m_82542_((d18 + d17) / d18, 1.0D, (d18 + d17) / d18));
          }
 
-         this.m_6034_(this.m_20185_(), vec33.f_82480_, this.m_20189_());
+         this.m_6034_(this.m_20185_(), vec33.m_7098_(), this.m_20189_());
       }
 
-      int j = Mth.m_14107_(this.m_20185_());
-      int i = Mth.m_14107_(this.m_20189_());
-      if (j != p_38156_.m_123341_() || i != p_38156_.m_123343_()) {
+      long j = Mth.m_14134_(this.m_20185_());
+      long i = Mth.m_14134_(this.m_20189_());
+      if (j != p_38156_.getX() || i != p_38156_.getZ()) {
          Vec3 vec35 = this.m_20184_();
          double d26 = vec35.m_165924_();
-         this.m_20334_(d26 * (double)(j - p_38156_.m_123341_()), vec35.f_82480_, d26 * (double)(i - p_38156_.m_123343_()));
+         this.m_20334_(d26 * (double)(j - p_38156_.getX()), vec35.m_7098_(), d26 * (double)(i - p_38156_.getZ()));
       }
 
       if (flag) {
@@ -513,15 +_,15 @@
          double d27 = vec36.m_165924_();
          if (d27 > 0.01D) {
             double d19 = 0.06D;
-            this.m_20256_(vec36.m_82520_(vec36.f_82479_ / d27 * 0.06D, 0.0D, vec36.f_82481_ / d27 * 0.06D));
+            this.m_20256_(vec36.m_82520_(vec36.m_7096_() / d27 * 0.06D, 0.0D, vec36.m_7094_() / d27 * 0.06D));
          } else {
             Vec3 vec37 = this.m_20184_();
-            double d20 = vec37.f_82479_;
-            double d21 = vec37.f_82481_;
+            double d20 = vec37.m_7096_();
+            double d21 = vec37.m_7094_();
             if (railshape == RailShape.EAST_WEST) {
-               if (this.m_38129_(p_38156_.m_122024_())) {
+               if (this.m_38129_(p_38156_.west())) {
                   d20 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122029_())) {
+               } else if (this.m_38129_(p_38156_.east())) {
                   d20 = -0.02D;
                }
             } else {
@@ -529,14 +_,14 @@
                   return;
                }
 
-               if (this.m_38129_(p_38156_.m_122012_())) {
+               if (this.m_38129_(p_38156_.north())) {
                   d21 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122019_())) {
+               } else if (this.m_38129_(p_38156_.south())) {
                   d21 = -0.02D;
                }
             }
 
-            this.m_20334_(d20, vec37.f_82480_, d21);
+            this.m_20334_(d20, vec37.m_7098_(), d21);
          }
       }
 
@@ -563,9 +_,9 @@
 
    @Nullable
    public Vec3 m_38096_(double p_38097_, double p_38098_, double p_38099_, double p_38100_) {
-      int i = Mth.m_14107_(p_38097_);
-      int j = Mth.m_14107_(p_38098_);
-      int k = Mth.m_14107_(p_38099_);
+      long i = Mth.m_14134_(p_38097_);
+      long j = Mth.m_14134_(p_38098_);
+      long k = Mth.m_14134_(p_38099_);
       if (this.m_9236_().m_8055_(new BlockPos(i, j - 1, k)).m_204336_(BlockTags.f_13034_)) {
          --j;
       }
@@ -588,9 +_,9 @@
          d1 /= d2;
          p_38097_ += d0 * p_38100_;
          p_38099_ += d1 * p_38100_;
-         if (vec3i.m_123342_() != 0 && Mth.m_14107_(p_38097_) - i == vec3i.m_123341_() && Mth.m_14107_(p_38099_) - k == vec3i.m_123343_()) {
+         if (vec3i.m_123342_() != 0 && Mth.m_14134_(p_38097_) - i == vec3i.m_123341_() && Mth.m_14134_(p_38099_) - k == vec3i.m_123343_()) {
             p_38098_ += (double)vec3i.m_123342_();
-         } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(p_38097_) - i == vec3i1.m_123341_() && Mth.m_14107_(p_38099_) - k == vec3i1.m_123343_()) {
+         } else if (vec3i1.m_123342_() != 0 && Mth.m_14134_(p_38097_) - i == vec3i1.m_123341_() && Mth.m_14134_(p_38099_) - k == vec3i1.m_123343_()) {
             p_38098_ += (double)vec3i1.m_123342_();
          }
 
@@ -602,9 +_,9 @@
 
    @Nullable
    public Vec3 m_38179_(double p_38180_, double p_38181_, double p_38182_) {
-      int i = Mth.m_14107_(p_38180_);
-      int j = Mth.m_14107_(p_38181_);
-      int k = Mth.m_14107_(p_38182_);
+      long i = Mth.m_14134_(p_38180_);
+      long j = Mth.m_14134_(p_38181_);
+      long k = Mth.m_14134_(p_38182_);
       if (this.m_9236_().m_8055_(new BlockPos(i, j - 1, k)).m_204336_(BlockTags.f_13034_)) {
          --j;
       }
@@ -708,15 +_,15 @@
                      Vec3 vec33 = p_38165_.m_20184_();
                      if (((AbstractMinecart)p_38165_).m_6064_() == AbstractMinecart.Type.FURNACE && this.m_6064_() != AbstractMinecart.Type.FURNACE) {
                         this.m_20256_(vec32.m_82542_(0.2D, 1.0D, 0.2D));
-                        this.m_5997_(vec33.f_82479_ - d0, 0.0D, vec33.f_82481_ - d1);
+                        this.m_5997_(vec33.m_7096_() - d0, 0.0D, vec33.m_7094_() - d1);
                         p_38165_.m_20256_(vec33.m_82542_(0.95D, 1.0D, 0.95D));
                      } else if (((AbstractMinecart)p_38165_).m_6064_() != AbstractMinecart.Type.FURNACE && this.m_6064_() == AbstractMinecart.Type.FURNACE) {
                         p_38165_.m_20256_(vec33.m_82542_(0.2D, 1.0D, 0.2D));
-                        p_38165_.m_5997_(vec32.f_82479_ + d0, 0.0D, vec32.f_82481_ + d1);
+                        p_38165_.m_5997_(vec32.m_7096_() + d0, 0.0D, vec32.m_7094_() + d1);
                         this.m_20256_(vec32.m_82542_(0.95D, 1.0D, 0.95D));
                      } else {
-                        double d7 = (vec33.f_82479_ + vec32.f_82479_) / 2.0D;
-                        double d8 = (vec33.f_82481_ + vec32.f_82481_) / 2.0D;
+                        double d7 = (vec33.m_7096_() + vec32.m_7096_()) / 2.0D;
+                        double d8 = (vec33.m_7094_() + vec32.m_7094_()) / 2.0D;
                         this.m_20256_(vec32.m_82542_(0.2D, 1.0D, 0.2D));
                         this.m_5997_(d7 - d0, 0.0D, d8 - d1);
                         p_38165_.m_20256_(vec33.m_82542_(0.2D, 1.0D, 0.2D));
@@ -733,7 +_,7 @@
       }
    }
 
-   public void m_6453_(double p_38102_, double p_38103_, double p_38104_, float p_38105_, float p_38106_, int p_38107_, boolean p_38108_) {
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal p_38102_, me.alphamode.mclong.math.BigDecimal p_38103_, me.alphamode.mclong.math.BigDecimal p_38104_, float p_38105_, float p_38106_, int p_38107_, boolean p_38108_) {
       this.f_38071_ = p_38102_;
       this.f_38072_ = p_38103_;
       this.f_38073_ = p_38104_;
