--- a/net/minecraft/world/entity/vehicle/AbstractMinecart.java
+++ b/net/minecraft/world/entity/vehicle/AbstractMinecart.java
@@ -159,13 +_,13 @@
 
             for(int i : f_38067_.get(pose)) {
                for(int[] aint1 : aint) {
-                  blockpos$mutableblockpos.m_122178_(blockpos.m_123341_() + aint1[0], blockpos.m_123342_() + i, blockpos.m_123343_() + aint1[1]);
+                  blockpos$mutableblockpos.set(blockpos.getX() + aint1[0], blockpos.getY() + i, blockpos.getZ() + aint1[1]);
                   double d0 = this.f_19853_.m_45564_(DismountHelper.m_38446_(this.f_19853_, blockpos$mutableblockpos), () -> {
-                     return DismountHelper.m_38446_(this.f_19853_, blockpos$mutableblockpos.m_7495_());
+                     return DismountHelper.m_38446_(this.f_19853_, blockpos$mutableblockpos.below());
                   });
                   if (DismountHelper.m_38439_(d0)) {
                      AABB aabb = new AABB((double)(-f), 0.0D, (double)(-f), (double)f, (double)entitydimensions.f_20378_, (double)f);
-                     Vec3 vec3 = Vec3.m_82514_(blockpos$mutableblockpos, d0);
+                     Vec3 vec3 = Vec3.upFromBottomCenterOf(blockpos$mutableblockpos, d0);
                      if (DismountHelper.m_38456_(this.f_19853_, p_38145_, aabb.m_82383_(vec3))) {
                         p_38145_.m_20124_(pose);
                         return vec3;
@@ -176,12 +_,12 @@
          }
 
          double d1 = this.m_20191_().f_82292_;
-         blockpos$mutableblockpos.m_122169_((double)blockpos.m_123341_(), d1, (double)blockpos.m_123343_());
+         blockpos$mutableblockpos.m_122169_((double)blockpos.getX(), d1, (double)blockpos.getZ());
 
          for(Pose pose1 : immutablelist) {
             double d2 = (double)p_38145_.m_6972_(pose1).f_20378_;
-            int j = Mth.m_14165_(d1 - (double)blockpos$mutableblockpos.m_123342_() + d2);
-            double d3 = DismountHelper.m_38463_(blockpos$mutableblockpos, j, (p_269973_) -> {
+            long j = Mth.lceil(d1 - (double)blockpos$mutableblockpos.getY() + d2);
+            double d3 = DismountHelper.findCeilingFrom(blockpos$mutableblockpos, j, (p_269973_) -> {
                return this.f_19853_.m_8055_(p_269973_).m_60812_(this.f_19853_, p_269973_);
             });
             if (d1 + d2 <= d3) {
@@ -386,7 +_,7 @@
       double d1 = this.m_20186_();
       double d2 = this.m_20189_();
       Vec3 vec3 = this.m_38179_(d0, d1, d2);
-      d1 = (double)p_38156_.m_123342_();
+      d1 = (double)p_38156_.getY();
       boolean flag = false;
       boolean flag1 = false;
       if (p_38157_.m_60713_(Blocks.f_50030_)) {
@@ -455,17 +_,17 @@
          }
       }
 
-      double d23 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i.m_123341_() * 0.5D;
-      double d10 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i.m_123343_() * 0.5D;
-      double d12 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
-      double d13 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
+      double d23 = (double)p_38156_.getX() + 0.5D + (double)vec3i.m_123341_() * 0.5D;
+      double d10 = (double)p_38156_.getZ() + 0.5D + (double)vec3i.m_123343_() * 0.5D;
+      double d12 = (double)p_38156_.getX() + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
+      double d13 = (double)p_38156_.getZ() + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
       d4 = d12 - d23;
       d5 = d13 - d10;
       double d14;
       if (d4 == 0.0D) {
-         d14 = d2 - (double)p_38156_.m_123343_();
+         d14 = d2 - (double)p_38156_.getZ();
       } else if (d5 == 0.0D) {
-         d14 = d0 - (double)p_38156_.m_123341_();
+         d14 = d0 - (double)p_38156_.getX();
       } else {
          double d15 = d0 - d23;
          double d16 = d2 - d10;
@@ -479,9 +_,9 @@
       double d25 = this.m_7097_();
       vec31 = this.m_20184_();
       this.m_6478_(MoverType.SELF, new Vec3(Mth.m_14008_(d24 * vec31.f_82479_, -d25, d25), 0.0D, Mth.m_14008_(d24 * vec31.f_82481_, -d25, d25)));
-      if (vec3i.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i.m_123343_()) {
+      if (vec3i.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.getX() == vec3i.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.getZ() == vec3i.m_123343_()) {
          this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i.m_123342_(), this.m_20189_());
-      } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i1.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i1.m_123343_()) {
+      } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.getX() == vec3i1.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.getZ() == vec3i1.m_123343_()) {
          this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i1.m_123342_(), this.m_20189_());
       }
 
@@ -500,10 +_,10 @@
 
       int j = Mth.m_14107_(this.m_20185_());
       int i = Mth.m_14107_(this.m_20189_());
-      if (j != p_38156_.m_123341_() || i != p_38156_.m_123343_()) {
+      if (j != p_38156_.getX() || i != p_38156_.getZ()) {
          Vec3 vec35 = this.m_20184_();
          double d26 = vec35.m_165924_();
-         this.m_20334_(d26 * (double)(j - p_38156_.m_123341_()), vec35.f_82480_, d26 * (double)(i - p_38156_.m_123343_()));
+         this.m_20334_(d26 * (double)(j - p_38156_.getX()), vec35.f_82480_, d26 * (double)(i - p_38156_.getZ()));
       }
 
       if (flag) {
@@ -517,9 +_,9 @@
             double d20 = vec37.f_82479_;
             double d21 = vec37.f_82481_;
             if (railshape == RailShape.EAST_WEST) {
-               if (this.m_38129_(p_38156_.m_122024_())) {
+               if (this.m_38129_(p_38156_.west())) {
                   d20 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122029_())) {
+               } else if (this.m_38129_(p_38156_.east())) {
                   d20 = -0.02D;
                }
             } else {
@@ -527,9 +_,9 @@
                   return;
                }
 
-               if (this.m_38129_(p_38156_.m_122012_())) {
+               if (this.m_38129_(p_38156_.north())) {
                   d21 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122019_())) {
+               } else if (this.m_38129_(p_38156_.south())) {
                   d21 = -0.02D;
                }
             }
