--- a/net/minecraft/world/entity/vehicle/AbstractMinecart.java
+++ b/net/minecraft/world/entity/vehicle/AbstractMinecart.java
@@ -77,9 +_,9 @@
       p_38135_.put(RailShape.NORTH_EAST, Pair.of(vec3i2, vec3i1));
    });
    private int f_38070_;
-   private double f_38071_;
-   private double f_38072_;
-   private double f_38073_;
+   private me.alphamode.mclong.math.BigDecimal f_38071_;
+   private me.alphamode.mclong.math.BigDecimal f_38072_;
+   private me.alphamode.mclong.math.BigDecimal f_38073_;
    private double f_38074_;
    private double f_38075_;
    private double f_38076_;
@@ -91,27 +_,27 @@
       this.f_19850_ = true;
    }
 
-   protected AbstractMinecart(EntityType<?> p_38090_, Level p_38091_, double p_38092_, double p_38093_, double p_38094_) {
-      this(p_38090_, p_38091_);
-      this.m_6034_(p_38092_, p_38093_, p_38094_);
-      this.f_19854_ = p_38092_;
-      this.f_19855_ = p_38093_;
-      this.f_19856_ = p_38094_;
+   protected AbstractMinecart(EntityType<?> pEntityType, Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      this(pEntityType, pLevel);
+      this.setPos(pX, pY, pZ);
+      this.f_19854_ = pX;
+      this.f_19855_ = pY;
+      this.f_19856_ = pZ;
    }
 
-   public static AbstractMinecart m_38119_(Level p_38120_, double p_38121_, double p_38122_, double p_38123_, AbstractMinecart.Type p_38124_) {
-      if (p_38124_ == AbstractMinecart.Type.CHEST) {
-         return new MinecartChest(p_38120_, p_38121_, p_38122_, p_38123_);
-      } else if (p_38124_ == AbstractMinecart.Type.FURNACE) {
-         return new MinecartFurnace(p_38120_, p_38121_, p_38122_, p_38123_);
-      } else if (p_38124_ == AbstractMinecart.Type.TNT) {
-         return new MinecartTNT(p_38120_, p_38121_, p_38122_, p_38123_);
-      } else if (p_38124_ == AbstractMinecart.Type.SPAWNER) {
-         return new MinecartSpawner(p_38120_, p_38121_, p_38122_, p_38123_);
-      } else if (p_38124_ == AbstractMinecart.Type.HOPPER) {
-         return new MinecartHopper(p_38120_, p_38121_, p_38122_, p_38123_);
+   public static AbstractMinecart createMinecart(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, AbstractMinecart.Type pType) {
+      if (pType == AbstractMinecart.Type.CHEST) {
+         return new MinecartChest(pLevel, pX, pY, pZ);
+      } else if (pType == AbstractMinecart.Type.FURNACE) {
+         return new MinecartFurnace(pLevel, pX, pY, pZ);
+      } else if (pType == AbstractMinecart.Type.TNT) {
+         return new MinecartTNT(pLevel, pX, pY, pZ);
+      } else if (pType == AbstractMinecart.Type.SPAWNER) {
+         return new MinecartSpawner(pLevel, pX, pY, pZ);
+      } else if (pType == AbstractMinecart.Type.HOPPER) {
+         return new MinecartHopper(pLevel, pX, pY, pZ);
       } else {
-         return (AbstractMinecart)(p_38124_ == AbstractMinecart.Type.COMMAND_BLOCK ? new MinecartCommandBlock(p_38120_, p_38121_, p_38122_, p_38123_) : new Minecart(p_38120_, p_38121_, p_38122_, p_38123_));
+         return (AbstractMinecart)(pType == AbstractMinecart.Type.COMMAND_BLOCK ? new MinecartCommandBlock(pLevel, pX, pY, pZ) : new Minecart(pLevel, pX, pY, pZ));
       }
    }
 
@@ -160,13 +_,13 @@
 
             for(int i : f_38067_.get(pose)) {
                for(int[] aint1 : aint) {
-                  blockpos$mutableblockpos.m_122178_(blockpos.m_123341_() + aint1[0], blockpos.m_123342_() + i, blockpos.m_123343_() + aint1[1]);
+                  blockpos$mutableblockpos.set(blockpos.getX() + aint1[0], blockpos.getY() + i, blockpos.getZ() + aint1[1]);
                   double d0 = this.m_9236_().m_45564_(DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos), () -> {
-                     return DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos.m_7495_());
+                     return DismountHelper.m_38446_(this.m_9236_(), blockpos$mutableblockpos.below());
                   });
                   if (DismountHelper.m_38439_(d0)) {
                      AABB aabb = new AABB((double)(-f), 0.0D, (double)(-f), (double)f, (double)entitydimensions.f_20378_, (double)f);
-                     Vec3 vec3 = Vec3.m_82514_(blockpos$mutableblockpos, d0);
+                     Vec3 vec3 = Vec3.upFromBottomCenterOf(blockpos$mutableblockpos, d0);
                      if (DismountHelper.m_38456_(this.m_9236_(), p_38145_, aabb.m_82383_(vec3))) {
                         p_38145_.m_20124_(pose);
                         return vec3;
@@ -176,13 +_,13 @@
             }
          }
 
-         double d1 = this.m_20191_().f_82292_;
-         blockpos$mutableblockpos.m_122169_((double)blockpos.m_123341_(), d1, (double)blockpos.m_123343_());
+         double d1 = this.m_20191_().f_82292_.doubleValue();
+         blockpos$mutableblockpos.m_122169_((double)blockpos.getX(), d1, (double)blockpos.getZ());
 
          for(Pose pose1 : immutablelist) {
             double d2 = (double)p_38145_.m_6972_(pose1).f_20378_;
-            int j = Mth.m_14165_(d1 - (double)blockpos$mutableblockpos.m_123342_() + d2);
-            double d3 = DismountHelper.m_38463_(blockpos$mutableblockpos, j, (p_289495_) -> {
+            long j = Mth.lceil(d1 - (double)blockpos$mutableblockpos.getY() + d2);
+            double d3 = DismountHelper.findCeilingFrom(blockpos$mutableblockpos, j, (p_289495_) -> {
                return this.m_9236_().m_8055_(p_289495_).m_60812_(this.m_9236_(), p_289495_);
             });
             if (d1 + d2 <= d3) {
@@ -273,14 +_,14 @@
       this.m_20157_();
       if (this.m_9236_().f_46443_) {
          if (this.f_38070_ > 0) {
-            double d5 = this.m_20185_() + (this.f_38071_ - this.m_20185_()) / (double)this.f_38070_;
-            double d6 = this.m_20186_() + (this.f_38072_ - this.m_20186_()) / (double)this.f_38070_;
-            double d7 = this.m_20189_() + (this.f_38073_ - this.m_20189_()) / (double)this.f_38070_;
+            me.alphamode.mclong.math.BigDecimal d5 = this.getBigX().add((this.f_38071_.subtract(this.getBigX())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
+            me.alphamode.mclong.math.BigDecimal d6 = this.getBigY().add((this.f_38072_.subtract(this.getBigY())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
+            me.alphamode.mclong.math.BigDecimal d7 = this.getBigZ().add((this.f_38073_.subtract(this.getBigZ())).divide(this.f_38070_, java.math.RoundingMode.HALF_UP));
             double d2 = Mth.m_14175_(this.f_38074_ - (double)this.m_146908_());
             this.m_146922_(this.m_146908_() + (float)d2 / (float)this.f_38070_);
             this.m_146926_(this.m_146909_() + (float)(this.f_38075_ - (double)this.m_146909_()) / (float)this.f_38070_);
             --this.f_38070_;
-            this.m_6034_(d5, d6, d7);
+            this.setPos(d5, d6, d7);
             this.m_19915_(this.m_146908_(), this.m_146909_());
          } else {
             this.m_20090_();
@@ -293,9 +_,9 @@
             this.m_20256_(this.m_20184_().m_82520_(0.0D, d0, 0.0D));
          }
 
-         int k = Mth.m_14107_(this.m_20185_());
-         int i = Mth.m_14107_(this.m_20186_());
-         int j = Mth.m_14107_(this.m_20189_());
+         long k = Mth.m_14134_(this.m_20185_());
+         long i = Mth.m_14134_(this.m_20186_());
+         long j = Mth.m_14134_(this.m_20189_());
          if (this.m_9236_().m_8055_(new BlockPos(k, i - 1, j)).m_204336_(BlockTags.f_13034_)) {
             --i;
          }
@@ -306,7 +_,7 @@
          if (this.f_287786_) {
             this.m_6401_(blockpos, blockstate);
             if (blockstate.m_60713_(Blocks.f_50285_)) {
-               this.m_6025_(k, i, j, blockstate.m_61143_(PoweredRailBlock.f_55215_));
+               this.activateMinecart(k, i, j, blockstate.m_61143_(PoweredRailBlock.f_55215_));
             }
          } else {
             this.m_38163_();
@@ -314,8 +_,8 @@
 
          this.m_20101_();
          this.m_146926_(0.0F);
-         double d1 = this.f_19854_ - this.m_20185_();
-         double d3 = this.f_19856_ - this.m_20189_();
+         double d1 = this.f_19854_.subtract(this.getBigX()).doubleValue();
+         double d3 = this.f_19856_.subtract(this.getBigZ()).doubleValue();
          if (d1 * d1 + d3 * d3 > 0.001D) {
             this.m_146922_((float)(Mth.m_14136_(d3, d1) * 180.0D / Math.PI));
             if (this.f_38068_) {
@@ -364,13 +_,13 @@
       return (this.m_20069_() ? 4.0D : 8.0D) / 20.0D;
    }
 
-   public void m_6025_(int p_38111_, int p_38112_, int p_38113_, boolean p_38114_) {
+   public void activateMinecart(long pX, long pY, long pZ, boolean pPowered) {
    }
 
    protected void m_38163_() {
       double d0 = this.m_7097_();
       Vec3 vec3 = this.m_20184_();
-      this.m_20334_(Mth.m_14008_(vec3.f_82479_, -d0, d0), vec3.f_82480_, Mth.m_14008_(vec3.f_82481_, -d0, d0));
+      this.m_20334_(Mth.m_14008_(vec3.m_7096_(), -d0, d0), vec3.m_7098_(), Mth.m_14008_(vec3.m_7094_(), -d0, d0));
       if (this.m_20096_()) {
          this.m_20256_(this.m_20184_().m_82490_(0.5D));
       }
@@ -384,11 +_,11 @@
 
    protected void m_6401_(BlockPos p_38156_, BlockState p_38157_) {
       this.m_183634_();
-      double d0 = this.m_20185_();
-      double d1 = this.m_20186_();
-      double d2 = this.m_20189_();
-      Vec3 vec3 = this.m_38179_(d0, d1, d2);
-      d1 = (double)p_38156_.m_123342_();
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigY();
+      me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ();
+      Vec3 vec3 = this.getPos(d0, d1, d2);
+      d1 = p_38156_.getBigDecY();
       boolean flag = false;
       boolean flag1 = false;
       if (p_38157_.m_60713_(Blocks.f_50030_)) {
@@ -406,19 +_,19 @@
       switch (railshape) {
          case ASCENDING_EAST:
             this.m_20256_(vec31.m_82520_(-d3, 0.0D, 0.0D));
-            ++d1;
+            d1 = d1.add();
             break;
          case ASCENDING_WEST:
             this.m_20256_(vec31.m_82520_(d3, 0.0D, 0.0D));
-            ++d1;
+            d1 = d1.add();
             break;
          case ASCENDING_NORTH:
             this.m_20256_(vec31.m_82520_(0.0D, 0.0D, d3));
-            ++d1;
+            d1 = d1.add();
             break;
          case ASCENDING_SOUTH:
             this.m_20256_(vec31.m_82520_(0.0D, 0.0D, -d3));
-            ++d1;
+            d1 = d1.add();
       }
 
       vec31 = this.m_20184_();
@@ -428,14 +_,14 @@
       double d4 = (double)(vec3i1.m_123341_() - vec3i.m_123341_());
       double d5 = (double)(vec3i1.m_123343_() - vec3i.m_123343_());
       double d6 = Math.sqrt(d4 * d4 + d5 * d5);
-      double d7 = vec31.f_82479_ * d4 + vec31.f_82481_ * d5;
+      double d7 = vec31.m_7096_() * d4 + vec31.m_7094_() * d5;
       if (d7 < 0.0D) {
          d4 = -d4;
          d5 = -d5;
       }
 
       double d8 = Math.min(2.0D, vec31.m_165924_());
-      vec31 = new Vec3(d8 * d4 / d6, vec31.f_82480_, d8 * d5 / d6);
+      vec31 = new Vec3(d8 * d4 / d6, vec31.m_7098_(), d8 * d5 / d6);
       this.m_20256_(vec31);
       Entity entity = this.m_146895_();
       if (entity instanceof Player) {
@@ -443,7 +_,7 @@
          double d9 = vec32.m_165925_();
          double d11 = this.m_20184_().m_165925_();
          if (d9 > 1.0E-4D && d11 < 0.01D) {
-            this.m_20256_(this.m_20184_().m_82520_(vec32.f_82479_ * 0.1D, 0.0D, vec32.f_82481_ * 0.1D));
+            this.m_20256_(this.m_20184_().m_82520_(vec32.m_7096_() * 0.1D, 0.0D, vec32.m_7094_() * 0.1D));
             flag1 = false;
          }
       }
@@ -457,55 +_,55 @@
          }
       }
 
-      double d23 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i.m_123341_() * 0.5D;
-      double d10 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i.m_123343_() * 0.5D;
-      double d12 = (double)p_38156_.m_123341_() + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
-      double d13 = (double)p_38156_.m_123343_() + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
-      d4 = d12 - d23;
-      d5 = d13 - d10;
+      me.alphamode.mclong.math.BigDecimal d23 = p_38156_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i.m_123341_() * 0.5D);
+      me.alphamode.mclong.math.BigDecimal d10 = p_38156_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i.m_123343_() * 0.5D);
+      me.alphamode.mclong.math.BigDecimal d12 = p_38156_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i1.m_123341_() * 0.5D);
+      me.alphamode.mclong.math.BigDecimal d13 = p_38156_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB).add((double)vec3i1.m_123343_() * 0.5D);
+      d4 = d12.subtract(d23).doubleValue();
+      d5 = d13.subtract(d10).doubleValue();
       double d14;
       if (d4 == 0.0D) {
-         d14 = d2 - (double)p_38156_.m_123343_();
+         d14 = d2.subtract(p_38156_.getBigDecZ()).doubleValue();
       } else if (d5 == 0.0D) {
-         d14 = d0 - (double)p_38156_.m_123341_();
+         d14 = d0.subtract(p_38156_.getBigDecX()).doubleValue();
       } else {
-         double d15 = d0 - d23;
-         double d16 = d2 - d10;
+         double d15 = d0.subtract(d23).doubleValue();
+         double d16 = d2.subtract(d10).doubleValue();
          d14 = (d15 * d4 + d16 * d5) * 2.0D;
       }
 
-      d0 = d23 + d4 * d14;
-      d2 = d10 + d5 * d14;
-      this.m_6034_(d0, d1, d2);
+      d0 = d23.add(d4 * d14);
+      d2 = d10.add(d5 * d14);
+      this.setPos(d0, d1, d2);
       double d24 = this.m_20160_() ? 0.75D : 1.0D;
       double d25 = this.m_7097_();
       vec31 = this.m_20184_();
-      this.m_6478_(MoverType.SELF, new Vec3(Mth.m_14008_(d24 * vec31.f_82479_, -d25, d25), 0.0D, Mth.m_14008_(d24 * vec31.f_82481_, -d25, d25)));
-      if (vec3i.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i.m_123343_()) {
-         this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i.m_123342_(), this.m_20189_());
-      } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(this.m_20185_()) - p_38156_.m_123341_() == vec3i1.m_123341_() && Mth.m_14107_(this.m_20189_()) - p_38156_.m_123343_() == vec3i1.m_123343_()) {
-         this.m_6034_(this.m_20185_(), this.m_20186_() + (double)vec3i1.m_123342_(), this.m_20189_());
+      this.m_6478_(MoverType.SELF, new Vec3(Mth.m_14008_(d24 * vec31.m_7096_(), -d25, d25), 0.0D, Mth.m_14008_(d24 * vec31.m_7094_(), -d25, d25)));
+      if (vec3i.m_123342_() != 0 && Mth.m_14134_(this.m_20185_()) - p_38156_.getX() == vec3i.m_123341_() && Mth.m_14134_(this.m_20189_()) - p_38156_.getZ() == vec3i.m_123343_()) {
+         this.setPos(this.getBigX(), this.getBigY().add(vec3i.m_123342_()), this.getBigZ());
+      } else if (vec3i1.m_123342_() != 0 && Mth.lfloor(this.getBigX()).subtract(p_38156_.getBigX()).intValue() == vec3i1.m_123341_() && Mth.lfloor(this.getBigZ()).subtract(p_38156_.getBigZ()).intValue() == vec3i1.m_123343_()) {
+         this.setPos(this.getBigX(), this.getBigY().add(vec3i1.m_123342_()), this.getBigZ());
       }
 
       this.m_7114_();
-      Vec3 vec33 = this.m_38179_(this.m_20185_(), this.m_20186_(), this.m_20189_());
+      Vec3 vec33 = this.getPos(this.getBigX(), this.getBigY(), this.getBigZ());
       if (vec33 != null && vec3 != null) {
-         double d17 = (vec3.f_82480_ - vec33.f_82480_) * 0.05D;
+         double d17 = (vec3.m_7098_() - vec33.m_7098_()) * 0.05D;
          Vec3 vec34 = this.m_20184_();
          double d18 = vec34.m_165924_();
          if (d18 > 0.0D) {
             this.m_20256_(vec34.m_82542_((d18 + d17) / d18, 1.0D, (d18 + d17) / d18));
          }
 
-         this.m_6034_(this.m_20185_(), vec33.f_82480_, this.m_20189_());
+         this.setPos(this.getBigX(), vec33.getY(), this.getBigZ());
       }
 
-      int j = Mth.m_14107_(this.m_20185_());
-      int i = Mth.m_14107_(this.m_20189_());
-      if (j != p_38156_.m_123341_() || i != p_38156_.m_123343_()) {
+      me.alphamode.mclong.math.BigInteger j = Mth.lfloor(this.getBigX());
+      me.alphamode.mclong.math.BigInteger i = Mth.lfloor(this.getBigZ());
+      if (!j.equals(p_38156_.getBigX()) || !i.equals(p_38156_.getBigZ())) {
          Vec3 vec35 = this.m_20184_();
-         double d26 = vec35.m_165924_();
-         this.m_20334_(d26 * (double)(j - p_38156_.m_123341_()), vec35.f_82480_, d26 * (double)(i - p_38156_.m_123343_()));
+         me.alphamode.mclong.math.BigDecimal d26 = me.alphamode.mclong.math.BigDecimal.valueOf(vec35.m_165924_());
+         this.setDeltaMovement(d26.multiply((j.subtract(p_38156_.getBigX())).toBigDecimal()), vec35.getY(), d26.multiply((i.subtract(p_38156_.getBigZ())).toBigDecimal()));
       }
 
       if (flag) {
@@ -513,15 +_,15 @@
          double d27 = vec36.m_165924_();
          if (d27 > 0.01D) {
             double d19 = 0.06D;
-            this.m_20256_(vec36.m_82520_(vec36.f_82479_ / d27 * 0.06D, 0.0D, vec36.f_82481_ / d27 * 0.06D));
+            this.m_20256_(vec36.m_82520_(vec36.m_7096_() / d27 * 0.06D, 0.0D, vec36.m_7094_() / d27 * 0.06D));
          } else {
             Vec3 vec37 = this.m_20184_();
-            double d20 = vec37.f_82479_;
-            double d21 = vec37.f_82481_;
+            double d20 = vec37.m_7096_();
+            double d21 = vec37.m_7094_();
             if (railshape == RailShape.EAST_WEST) {
-               if (this.m_38129_(p_38156_.m_122024_())) {
+               if (this.m_38129_(p_38156_.west())) {
                   d20 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122029_())) {
+               } else if (this.m_38129_(p_38156_.east())) {
                   d20 = -0.02D;
                }
             } else {
@@ -529,14 +_,14 @@
                   return;
                }
 
-               if (this.m_38129_(p_38156_.m_122012_())) {
+               if (this.m_38129_(p_38156_.north())) {
                   d21 = 0.02D;
-               } else if (this.m_38129_(p_38156_.m_122019_())) {
+               } else if (this.m_38129_(p_38156_.south())) {
                   d21 = -0.02D;
                }
             }
 
-            this.m_20334_(d20, vec37.f_82480_, d21);
+            this.m_20334_(d20, vec37.m_7098_(), d21);
          }
       }
 
@@ -562,20 +_,20 @@
    }
 
    @Nullable
-   public Vec3 m_38096_(double p_38097_, double p_38098_, double p_38099_, double p_38100_) {
-      int i = Mth.m_14107_(p_38097_);
-      int j = Mth.m_14107_(p_38098_);
-      int k = Mth.m_14107_(p_38099_);
-      if (this.m_9236_().m_8055_(new BlockPos(i, j - 1, k)).m_204336_(BlockTags.f_13034_)) {
-         --j;
+   public Vec3 getPosOffs(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pOffset) {
+      me.alphamode.mclong.math.BigInteger i = Mth.lfloor(pX);
+      me.alphamode.mclong.math.BigInteger j = Mth.lfloor(pY);
+      me.alphamode.mclong.math.BigInteger k = Mth.lfloor(pZ);
+      if (this.m_9236_().m_8055_(new BlockPos(i, j.subtract(), k)).m_204336_(BlockTags.f_13034_)) {
+         j = j.subtract();
       }
 
       BlockState blockstate = this.m_9236_().m_8055_(new BlockPos(i, j, k));
       if (BaseRailBlock.m_49416_(blockstate)) {
          RailShape railshape = blockstate.m_61143_(((BaseRailBlock)blockstate.m_60734_()).m_7978_());
-         p_38098_ = (double)j;
+         pY = j.toBigDecimal();
          if (railshape.m_61745_()) {
-            p_38098_ = (double)(j + 1);
+            pY = j.add().toBigDecimal();
          }
 
          Pair<Vec3i, Vec3i> pair = m_38125_(railshape);
@@ -586,27 +_,27 @@
          double d2 = Math.sqrt(d0 * d0 + d1 * d1);
          d0 /= d2;
          d1 /= d2;
-         p_38097_ += d0 * p_38100_;
-         p_38099_ += d1 * p_38100_;
-         if (vec3i.m_123342_() != 0 && Mth.m_14107_(p_38097_) - i == vec3i.m_123341_() && Mth.m_14107_(p_38099_) - k == vec3i.m_123343_()) {
-            p_38098_ += (double)vec3i.m_123342_();
-         } else if (vec3i1.m_123342_() != 0 && Mth.m_14107_(p_38097_) - i == vec3i1.m_123341_() && Mth.m_14107_(p_38099_) - k == vec3i1.m_123343_()) {
-            p_38098_ += (double)vec3i1.m_123342_();
+         pX = pX.add(d0 * pOffset);
+         pZ = pZ.add(d1 * pOffset);
+         if (vec3i.m_123342_() != 0 && Mth.lfloor(pX).subtract(i).intValue() == vec3i.m_123341_() && Mth.lfloor(pZ).subtract(k).intValue() == vec3i.m_123343_()) {
+            pY = pY.add(vec3i.m_123342_());
+         } else if (vec3i1.m_123342_() != 0 && Mth.lfloor(pX).subtract(i).intValue() == vec3i1.m_123341_() && Mth.lfloor(pZ).subtract(k).intValue() == vec3i1.m_123343_()) {
+            pY = pY.add(vec3i1.m_123342_());
          }
 
-         return this.m_38179_(p_38097_, p_38098_, p_38099_);
+         return this.getPos(pX, pY, pZ);
       } else {
          return null;
       }
    }
 
    @Nullable
-   public Vec3 m_38179_(double p_38180_, double p_38181_, double p_38182_) {
-      int i = Mth.m_14107_(p_38180_);
-      int j = Mth.m_14107_(p_38181_);
-      int k = Mth.m_14107_(p_38182_);
-      if (this.m_9236_().m_8055_(new BlockPos(i, j - 1, k)).m_204336_(BlockTags.f_13034_)) {
-         --j;
+   public Vec3 getPos(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      me.alphamode.mclong.math.BigInteger i = Mth.lfloor(pX);
+      me.alphamode.mclong.math.BigInteger j = Mth.lfloor(pY);
+      me.alphamode.mclong.math.BigInteger k = Mth.lfloor(pZ);
+      if (this.m_9236_().m_8055_(new BlockPos(i, j.subtract(), k)).m_204336_(BlockTags.f_13034_)) {
+         j = j.subtract();
       }
 
       BlockState blockstate = this.m_9236_().m_8055_(new BlockPos(i, j, k));
@@ -615,36 +_,36 @@
          Pair<Vec3i, Vec3i> pair = m_38125_(railshape);
          Vec3i vec3i = pair.getFirst();
          Vec3i vec3i1 = pair.getSecond();
-         double d0 = (double)i + 0.5D + (double)vec3i.m_123341_() * 0.5D;
-         double d1 = (double)j + 0.0625D + (double)vec3i.m_123342_() * 0.5D;
-         double d2 = (double)k + 0.5D + (double)vec3i.m_123343_() * 0.5D;
-         double d3 = (double)i + 0.5D + (double)vec3i1.m_123341_() * 0.5D;
-         double d4 = (double)j + 0.0625D + (double)vec3i1.m_123342_() * 0.5D;
-         double d5 = (double)k + 0.5D + (double)vec3i1.m_123343_() * 0.5D;
-         double d6 = d3 - d0;
-         double d7 = (d4 - d1) * 2.0D;
-         double d8 = d5 - d2;
-         double d9;
-         if (d6 == 0.0D) {
-            d9 = p_38182_ - (double)k;
-         } else if (d8 == 0.0D) {
-            d9 = p_38180_ - (double)i;
+         me.alphamode.mclong.math.BigDecimal d0 = i.toBigDecimal().add(0.5D + (double)vec3i.m_123341_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d1 = j.toBigDecimal().add(0.0625D + (double)vec3i.m_123342_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d2 = k.toBigDecimal().add(0.5D + (double)vec3i.m_123343_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d3 = i.toBigDecimal().add(0.5D + (double)vec3i1.m_123341_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d4 = j.toBigDecimal().add(0.0625D + (double)vec3i1.m_123342_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d5 = k.toBigDecimal().add(0.5D + (double)vec3i1.m_123343_() * 0.5D);
+         me.alphamode.mclong.math.BigDecimal d6 = d3.subtract(d0);
+         me.alphamode.mclong.math.BigDecimal d7 = (d4.subtract(d1)).multiply(me.alphamode.mclong.core.BigConstants.TWO);
+         me.alphamode.mclong.math.BigDecimal d8 = d5.subtract(d2);
+         me.alphamode.mclong.math.BigDecimal d9;
+         if (d6.equals(me.alphamode.mclong.math.BigDecimal.ZERO)) {
+            d9 = pZ.subtract(k.toBigDecimal());
+         } else if (d8.equals(me.alphamode.mclong.math.BigDecimal.ZERO)) {
+            d9 = pX.subtract(i.toBigDecimal());
          } else {
-            double d10 = p_38180_ - d0;
-            double d11 = p_38182_ - d2;
-            d9 = (d10 * d6 + d11 * d8) * 2.0D;
-         }
-
-         p_38180_ = d0 + d6 * d9;
-         p_38181_ = d1 + d7 * d9;
-         p_38182_ = d2 + d8 * d9;
-         if (d7 < 0.0D) {
-            ++p_38181_;
-         } else if (d7 > 0.0D) {
-            p_38181_ += 0.5D;
-         }
-
-         return new Vec3(p_38180_, p_38181_, p_38182_);
+            me.alphamode.mclong.math.BigDecimal d10 = pX.subtract(d0);
+            me.alphamode.mclong.math.BigDecimal d11 = pZ.subtract(d2);
+            d9 = (d10.multiply(d6).add(d11.multiply(d8))).multiply(me.alphamode.mclong.core.BigConstants.TWO);
+         }
+
+         pX = d0.add(d6.multiply(d9));
+         pY = d1.add(d7.multiply(d9));
+         pZ = d2.add(d8.multiply(d9));
+         if (d7.compareTo(me.alphamode.mclong.math.BigDecimal.ZERO) < 0) {
+            pY = pY.add();
+         } else if (d7.compareTo(me.alphamode.mclong.math.BigDecimal.ZERO) > 0) {
+            pY = pY.add(me.alphamode.mclong.core.BigConstants.AABB);
+         }
+
+         return new Vec3(pX, pY, pZ);
       } else {
          return null;
       }
@@ -708,15 +_,15 @@
                      Vec3 vec33 = p_38165_.m_20184_();
                      if (((AbstractMinecart)p_38165_).m_6064_() == AbstractMinecart.Type.FURNACE && this.m_6064_() != AbstractMinecart.Type.FURNACE) {
                         this.m_20256_(vec32.m_82542_(0.2D, 1.0D, 0.2D));
-                        this.m_5997_(vec33.f_82479_ - d0, 0.0D, vec33.f_82481_ - d1);
+                        this.m_5997_(vec33.m_7096_() - d0, 0.0D, vec33.m_7094_() - d1);
                         p_38165_.m_20256_(vec33.m_82542_(0.95D, 1.0D, 0.95D));
                      } else if (((AbstractMinecart)p_38165_).m_6064_() != AbstractMinecart.Type.FURNACE && this.m_6064_() == AbstractMinecart.Type.FURNACE) {
                         p_38165_.m_20256_(vec33.m_82542_(0.2D, 1.0D, 0.2D));
-                        p_38165_.m_5997_(vec32.f_82479_ + d0, 0.0D, vec32.f_82481_ + d1);
+                        p_38165_.m_5997_(vec32.m_7096_() + d0, 0.0D, vec32.m_7094_() + d1);
                         this.m_20256_(vec32.m_82542_(0.95D, 1.0D, 0.95D));
                      } else {
-                        double d7 = (vec33.f_82479_ + vec32.f_82479_) / 2.0D;
-                        double d8 = (vec33.f_82481_ + vec32.f_82481_) / 2.0D;
+                        double d7 = (vec33.m_7096_() + vec32.m_7096_()) / 2.0D;
+                        double d8 = (vec33.m_7094_() + vec32.m_7094_()) / 2.0D;
                         this.m_20256_(vec32.m_82542_(0.2D, 1.0D, 0.2D));
                         this.m_5997_(d7 - d0, 0.0D, d8 - d1);
                         p_38165_.m_20256_(vec33.m_82542_(0.2D, 1.0D, 0.2D));
@@ -733,13 +_,13 @@
       }
    }
 
-   public void m_6453_(double p_38102_, double p_38103_, double p_38104_, float p_38105_, float p_38106_, int p_38107_, boolean p_38108_) {
-      this.f_38071_ = p_38102_;
-      this.f_38072_ = p_38103_;
-      this.f_38073_ = p_38104_;
-      this.f_38074_ = (double)p_38105_;
-      this.f_38075_ = (double)p_38106_;
-      this.f_38070_ = p_38107_ + 2;
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pYaw, float pPitch, int pPosRotationIncrements, boolean pTeleport) {
+      this.f_38071_ = pX;
+      this.f_38072_ = pY;
+      this.f_38073_ = pZ;
+      this.f_38074_ = (double)pYaw;
+      this.f_38075_ = (double)pPitch;
+      this.f_38070_ = pPosRotationIncrements + 2;
       this.m_20334_(this.f_38076_, this.f_38077_, this.f_38078_);
    }
 
