--- a/net/minecraft/world/entity/vehicle/ChestBoat.java
+++ b/net/minecraft/world/entity/vehicle/ChestBoat.java
@@ -34,9 +_,9 @@
       super(p_219869_, p_219870_);
    }
 
-   public ChestBoat(Level p_219872_, double p_219873_, double p_219874_, double p_219875_) {
+   public ChestBoat(Level p_219872_, java.math.BigDecimal p_219873_, java.math.BigDecimal p_219874_, java.math.BigDecimal p_219875_) {
       this(EntityType.f_217016_, p_219872_);
-      this.m_6034_(p_219873_, p_219874_, p_219875_);
+      this.setPos(p_219873_, p_219874_, p_219875_);
       this.f_19854_ = p_219873_;
       this.f_19855_ = p_219874_;
       this.f_19856_ = p_219875_;
