--- a/net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
+++ b/net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
@@ -69,7 +_,7 @@
       return this.f_38505_;
    }
 
-   public void m_6025_(int p_38517_, int p_38518_, int p_38519_, boolean p_38520_) {
+   public void activateMinecart(long p_38517_, long p_38518_, long p_38519_, boolean p_38520_) {
       if (p_38520_ && this.f_19797_ - this.f_38506_ >= 4) {
          this.m_38534_().m_45414_(this.m_9236_());
          this.f_38506_ = this.f_19797_;
