--- a/net/minecraft/world/entity/vehicle/MinecartSpawner.java
+++ b/net/minecraft/world/entity/vehicle/MinecartSpawner.java
@@ -24,7 +_,7 @@
       this.f_150333_ = this.m_150334_(p_38624_);
    }
 
-   public MinecartSpawner(Level p_38626_, double p_38627_, double p_38628_, double p_38629_) {
+   public MinecartSpawner(Level p_38626_, me.alphamode.mclong.math.BigDecimal p_38627_, me.alphamode.mclong.math.BigDecimal p_38628_, me.alphamode.mclong.math.BigDecimal p_38629_) {
       super(EntityType.f_20474_, p_38626_, p_38627_, p_38628_, p_38629_);
       this.f_150333_ = this.m_150334_(p_38626_);
    }
