--- a/net/minecraft/world/entity/vehicle/MinecartTNT.java
+++ b/net/minecraft/world/entity/vehicle/MinecartTNT.java
@@ -31,8 +_,8 @@
       super(p_38649_, p_38650_);
    }
 
-   public MinecartTNT(Level p_38652_, double p_38653_, double p_38654_, double p_38655_) {
-      super(EntityType.f_20475_, p_38652_, p_38653_, p_38654_, p_38655_);
+   public MinecartTNT(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20475_, pLevel, pX, pY, pZ);
    }
 
    public AbstractMinecart.Type m_6064_() {
@@ -47,7 +_,7 @@
       super.m_8119_();
       if (this.f_38647_ > 0) {
          --this.f_38647_;
-         this.m_9236_().m_7106_(ParticleTypes.f_123762_, this.m_20185_(), this.m_20186_() + 0.5D, this.m_20189_(), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123762_, this.getBigX(), this.getBigY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getBigZ(), 0.0D, 0.0D, 0.0D);
       } else if (this.f_38647_ == 0) {
          this.m_38688_(this.m_20184_().m_165925_());
       }
@@ -101,7 +_,7 @@
             d0 = 5.0D;
          }
 
-         this.m_9236_().m_254877_(this, p_259539_, (ExplosionDamageCalculator)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), (float)(4.0D + this.f_19796_.m_188500_() * 1.5D * d0), false, Level.ExplosionInteraction.TNT);
+         this.m_9236_().explode(this, p_259539_, (ExplosionDamageCalculator)null, this.getBigX(), this.getBigY(), this.getBigZ(), (float)(4.0D + this.f_19796_.m_188500_() * 1.5D * d0), false, Level.ExplosionInteraction.TNT);
          this.m_146870_();
       }
 
@@ -116,8 +_,8 @@
       return super.m_142535_(p_150347_, p_150348_, p_150349_);
    }
 
-   public void m_6025_(int p_38659_, int p_38660_, int p_38661_, boolean p_38662_) {
-      if (p_38662_ && this.f_38647_ < 0) {
+   public void activateMinecart(long pX, long pY, long pZ, boolean pReceivingPower) {
+      if (pReceivingPower && this.f_38647_ < 0) {
          this.m_38693_();
       }
 
@@ -137,7 +_,7 @@
       if (!this.m_9236_().f_46443_) {
          this.m_9236_().m_7605_(this, (byte)10);
          if (!this.m_20067_()) {
-            this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12512_, SoundSource.BLOCKS, 1.0F, 1.0F);
+            this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12512_, SoundSource.BLOCKS, 1.0F, 1.0F);
          }
       }
 
@@ -152,11 +_,11 @@
    }
 
    public float m_7077_(Explosion p_38675_, BlockGetter p_38676_, BlockPos p_38677_, BlockState p_38678_, FluidState p_38679_, float p_38680_) {
-      return !this.m_38695_() || !p_38678_.m_204336_(BlockTags.f_13034_) && !p_38676_.m_8055_(p_38677_.m_7494_()).m_204336_(BlockTags.f_13034_) ? super.m_7077_(p_38675_, p_38676_, p_38677_, p_38678_, p_38679_, p_38680_) : 0.0F;
+      return !this.m_38695_() || !p_38678_.m_204336_(BlockTags.f_13034_) && !p_38676_.m_8055_(p_38677_.above()).m_204336_(BlockTags.f_13034_) ? super.m_7077_(p_38675_, p_38676_, p_38677_, p_38678_, p_38679_, p_38680_) : 0.0F;
    }
 
    public boolean m_7349_(Explosion p_38669_, BlockGetter p_38670_, BlockPos p_38671_, BlockState p_38672_, float p_38673_) {
-      return !this.m_38695_() || !p_38672_.m_204336_(BlockTags.f_13034_) && !p_38670_.m_8055_(p_38671_.m_7494_()).m_204336_(BlockTags.f_13034_) ? super.m_7349_(p_38669_, p_38670_, p_38671_, p_38672_, p_38673_) : false;
+      return !this.m_38695_() || !p_38672_.m_204336_(BlockTags.f_13034_) && !p_38670_.m_8055_(p_38671_.above()).m_204336_(BlockTags.f_13034_) ? super.m_7349_(p_38669_, p_38670_, p_38671_, p_38672_, p_38673_) : false;
    }
 
    protected void m_7378_(CompoundTag p_38682_) {
