--- a/net/minecraft/world/entity/vehicle/Minecart.java
+++ b/net/minecraft/world/entity/vehicle/Minecart.java
@@ -33,7 +_,7 @@
       return Items.f_42449_;
    }
 
-   public void m_6025_(int p_38478_, int p_38479_, int p_38480_, boolean p_38481_) {
+   public void activateMinecart(long p_38478_, long p_38479_, long p_38480_, boolean p_38481_) {
       if (p_38481_) {
          if (this.m_20160_()) {
             this.m_20153_();
