--- a/net/minecraft/world/entity/vehicle/MinecartSpawner.java
+++ b/net/minecraft/world/entity/vehicle/MinecartSpawner.java
@@ -24,9 +_,9 @@
       this.f_150333_ = this.m_150334_(p_38624_);
    }
 
-   public MinecartSpawner(Level p_38626_, double p_38627_, double p_38628_, double p_38629_) {
-      super(EntityType.f_20474_, p_38626_, p_38627_, p_38628_, p_38629_);
-      this.f_150333_ = this.m_150334_(p_38626_);
+   public MinecartSpawner(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(EntityType.f_20474_, pLevel, pX, pY, pZ);
+      this.f_150333_ = this.m_150334_(pLevel);
    }
 
    protected Item m_213728_() {
