--- a/net/minecraft/world/entity/vehicle/MinecartFurnace.java
+++ b/net/minecraft/world/entity/vehicle/MinecartFurnace.java
@@ -84,8 +_,8 @@
       if (d3 > 1.0E-4D && d2 > 0.001D) {
          double d4 = Math.sqrt(d2);
          double d5 = Math.sqrt(d3);
-         this.f_38545_ = vec3.f_82479_ / d4 * d5;
-         this.f_38546_ = vec3.f_82481_ / d4 * d5;
+         this.f_38545_ = vec3.m_7096_() / d4 * d5;
+         this.f_38546_ = vec3.m_7094_() / d4 * d5;
       }
 
    }
