--- a/net/minecraft/world/entity/vehicle/MinecartTNT.java
+++ b/net/minecraft/world/entity/vehicle/MinecartTNT.java
@@ -116,7 +_,7 @@
       return super.m_142535_(p_150347_, p_150348_, p_150349_);
    }
 
-   public void m_6025_(int p_38659_, int p_38660_, int p_38661_, boolean p_38662_) {
+   public void activateMinecart(long p_38659_, long p_38660_, long p_38661_, boolean p_38662_) {
       if (p_38662_ && this.f_38647_ < 0) {
          this.m_38693_();
       }
@@ -152,11 +_,11 @@
    }
 
    public float m_7077_(Explosion p_38675_, BlockGetter p_38676_, BlockPos p_38677_, BlockState p_38678_, FluidState p_38679_, float p_38680_) {
-      return !this.m_38695_() || !p_38678_.m_204336_(BlockTags.f_13034_) && !p_38676_.m_8055_(p_38677_.m_7494_()).m_204336_(BlockTags.f_13034_) ? super.m_7077_(p_38675_, p_38676_, p_38677_, p_38678_, p_38679_, p_38680_) : 0.0F;
+      return !this.m_38695_() || !p_38678_.m_204336_(BlockTags.f_13034_) && !p_38676_.m_8055_(p_38677_.above()).m_204336_(BlockTags.f_13034_) ? super.m_7077_(p_38675_, p_38676_, p_38677_, p_38678_, p_38679_, p_38680_) : 0.0F;
    }
 
    public boolean m_7349_(Explosion p_38669_, BlockGetter p_38670_, BlockPos p_38671_, BlockState p_38672_, float p_38673_) {
-      return !this.m_38695_() || !p_38672_.m_204336_(BlockTags.f_13034_) && !p_38670_.m_8055_(p_38671_.m_7494_()).m_204336_(BlockTags.f_13034_) ? super.m_7349_(p_38669_, p_38670_, p_38671_, p_38672_, p_38673_) : false;
+      return !this.m_38695_() || !p_38672_.m_204336_(BlockTags.f_13034_) && !p_38670_.m_8055_(p_38671_.above()).m_204336_(BlockTags.f_13034_) ? super.m_7349_(p_38669_, p_38670_, p_38671_, p_38672_, p_38673_) : false;
    }
 
    protected void m_7378_(CompoundTag p_38682_) {
