--- a/net/minecraft/world/entity/vehicle/MinecartFurnace.java
+++ b/net/minecraft/world/entity/vehicle/MinecartFurnace.java
@@ -61,7 +_,7 @@
       }
 
       if (this.m_38579_() && this.f_19796_.m_188503_(4) == 0) {
-         this.m_9236_().m_7106_(ParticleTypes.f_123755_, this.m_20185_(), this.m_20186_() + 0.8D, this.m_20189_(), 0.0D, 0.0D, 0.0D);
+         this.m_9236_().addParticle(ParticleTypes.f_123755_, this.getBigX(), this.getBigY().add(me.alphamode.mclong.core.BigConstants.MINECART), this.getBigZ(), 0.0D, 0.0D, 0.0D);
       }
 
    }
@@ -84,8 +_,8 @@
       if (d3 > 1.0E-4D && d2 > 0.001D) {
          double d4 = Math.sqrt(d2);
          double d5 = Math.sqrt(d3);
-         this.f_38545_ = vec3.f_82479_ / d4 * d5;
-         this.f_38546_ = vec3.f_82481_ / d4 * d5;
+         this.f_38545_ = vec3.m_7096_() / d4 * d5;
+         this.f_38546_ = vec3.m_7094_() / d4 * d5;
       }
 
    }
