--- a/net/minecraft/world/entity/vehicle/MinecartChest.java
+++ b/net/minecraft/world/entity/vehicle/MinecartChest.java
@@ -22,7 +_,7 @@
       super(p_38487_, p_38488_);
    }
 
-   public MinecartChest(Level p_38490_, double p_38491_, double p_38492_, double p_38493_) {
+   public MinecartChest(Level p_38490_, me.alphamode.mclong.math.BigDecimal p_38491_, me.alphamode.mclong.math.BigDecimal p_38492_, me.alphamode.mclong.math.BigDecimal p_38493_) {
       super(EntityType.f_20470_, p_38491_, p_38492_, p_38493_, p_38490_);
    }
 
