--- a/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
+++ b/net/minecraft/world/entity/ai/behavior/AcquirePoi.java
@@ -34,7 +_,7 @@
       int i = 5;
       int j = 20;
       MutableLong mutablelong = new MutableLong(0L);
-      Long2ObjectMap<AcquirePoi.JitteredLinearRetry> long2objectmap = new Long2ObjectOpenHashMap<>();
+      it.unimi.dsi.fastutil.objects.Object2ObjectMap<BlockPos, AcquirePoi.JitteredLinearRetry> long2objectmap = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
       OneShot<PathfinderMob> oneshot = BehaviorBuilder.m_258034_((p_258276_) -> {
          return p_258276_.group(p_258276_.m_258080_(p_260194_)).apply(p_258276_, (p_258300_) -> {
             return (p_258292_, p_258293_, p_258294_) -> {
@@ -48,11 +_,11 @@
                } else {
                   mutablelong.setValue(p_258294_ + 20L + (long)p_258292_.m_213780_().m_188503_(20));
                   PoiManager poimanager = p_258292_.m_8904_();
-                  long2objectmap.long2ObjectEntrySet().removeIf((p_22338_) -> {
+                  long2objectmap.object2ObjectEntrySet().removeIf((p_22338_) -> {
                      return !p_22338_.getValue().m_22382_(p_258294_);
                   });
                   Predicate<BlockPos> predicate = (p_258266_) -> {
-                     AcquirePoi.JitteredLinearRetry acquirepoi$jitteredlinearretry = long2objectmap.get(p_258266_.m_121878_());
+                     AcquirePoi.JitteredLinearRetry acquirepoi$jitteredlinearretry = long2objectmap.get(p_258266_);
                      if (acquirepoi$jitteredlinearretry == null) {
                         return true;
                      } else if (!acquirepoi$jitteredlinearretry.m_22384_(p_258294_)) {
@@ -79,7 +_,7 @@
                      });
                   } else {
                      for(Pair<Holder<PoiType>, BlockPos> pair : set) {
-                        long2objectmap.computeIfAbsent(pair.getSecond().m_121878_(), (p_264881_) -> {
+                        long2objectmap.computeIfAbsent(pair.getSecond(), (p_264881_) -> {
                            return new AcquirePoi.JitteredLinearRetry(p_258292_.f_46441_, p_258294_);
                         });
                      }
