--- a/net/minecraft/world/entity/ai/behavior/AnimalPanic.java
+++ b/net/minecraft/world/entity/ai/behavior/AnimalPanic.java
@@ -55,7 +_,7 @@
    @Nullable
    private Vec3 m_196638_(PathfinderMob p_196639_, ServerLevel p_196640_) {
       if (p_196639_.m_6060_()) {
-         Optional<Vec3> optional = this.m_196641_(p_196640_, p_196639_).map(Vec3::m_82539_);
+         Optional<Vec3> optional = this.m_196641_(p_196640_, p_196639_).map(Vec3::atBottomCenterOf);
          if (optional.isPresent()) {
             return optional.get();
          }
