--- a/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
@@ -19,7 +_,7 @@
 
    protected boolean m_6114_(ServerLevel p_22450_, Villager p_22451_) {
       BlockPos blockpos = p_22451_.m_6274_().m_21952_(MemoryModuleType.f_26361_).get().m_122646_();
-      return blockpos.m_203195_(p_22451_.m_20182_(), 2.0D) || p_22451_.m_35504_();
+      return blockpos.closerToCenterThan(p_22451_.m_20182_(), 2.0D) || p_22451_.m_35504_();
    }
 
    protected void m_6735_(ServerLevel p_22453_, Villager p_22454_, long p_22455_) {
