--- a/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
+++ b/net/minecraft/world/entity/ai/behavior/AssignProfessionFromJobSite.java
@@ -15,7 +_,7 @@
          return p_258312_.group(p_258312_.m_257495_(MemoryModuleType.f_26361_), p_258312_.m_257492_(MemoryModuleType.f_26360_)).apply(p_258312_, (p_258304_, p_258305_) -> {
             return (p_258309_, p_258310_, p_258311_) -> {
                GlobalPos globalpos = p_258312_.m_258051_(p_258304_);
-               if (!globalpos.m_122646_().m_203195_(p_258310_.m_20182_(), 2.0D) && !p_258310_.m_35504_()) {
+               if (!globalpos.m_122646_().closerToCenterThan(p_258310_.m_20182_(), 2.0D) && !p_258310_.m_35504_()) {
                   return false;
                } else {
                   p_258304_.m_257971_();
