--- a/net/minecraft/world/entity/ai/behavior/CelebrateVillagersSurvivedRaid.java
+++ b/net/minecraft/world/entity/ai/behavior/CelebrateVillagersSurvivedRaid.java
@@ -51,7 +_,7 @@
          DyeColor dyecolor = Util.m_214670_(DyeColor.values(), randomsource);
          int i = randomsource.m_188503_(3);
          ItemStack itemstack = this.m_22696_(dyecolor, i);
-         FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(p_22713_.m_9236_(), p_22713_, p_22713_.m_20185_(), p_22713_.m_20188_(), p_22713_.m_20189_(), itemstack);
+         FireworkRocketEntity fireworkrocketentity = new FireworkRocketEntity(p_22713_.m_9236_(), p_22713_, p_22713_.getBigX(), p_22713_.getEyeBigY(), p_22713_.getBigZ(), itemstack);
          p_22713_.m_9236_().m_7967_(fireworkrocketentity);
       }
 
