--- a/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+++ b/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
@@ -135,7 +_,7 @@
    private boolean m_23014_(Villager p_23015_, Player p_23016_) {
       BlockPos blockpos = p_23016_.m_20183_();
       BlockPos blockpos1 = p_23015_.m_20183_();
-      return blockpos1.m_123314_(blockpos, 5.0D);
+      return blockpos1.closerThan(blockpos, 5.0D);
    }
 
    private static int m_22993_(ServerLevel p_22994_) {
