--- a/net/minecraft/world/entity/ai/behavior/GoAndGiveItemsToTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/GoAndGiveItemsToTarget.java
@@ -71,7 +_,7 @@
    }
 
    private void m_217213_(PositionTracker p_217214_, ItemStack p_217215_, ServerPlayer p_217216_) {
-      BlockPos blockpos = p_217214_.m_6675_().m_7495_();
+      BlockPos blockpos = p_217214_.m_6675_().below();
       CriteriaTriggers.f_215657_.m_285767_(p_217216_, blockpos, p_217215_);
    }
 
