--- a/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToClosestVillage.java
@@ -27,7 +_,7 @@
                         return (double)(-poimanager.m_27098_(SectionPos.m_123199_(p_147554_)));
                      });
                      if (vec31 != null) {
-                        int k = poimanager.m_27098_(SectionPos.m_123199_(BlockPos.m_274446_(vec31)));
+                        int k = poimanager.m_27098_(SectionPos.m_123199_(BlockPos.containing(vec31)));
                         if (k < i) {
                            vec3 = vec31;
                            break;
