--- a/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
+++ b/net/minecraft/world/entity/ai/behavior/GoToTargetLocation.java
@@ -22,7 +_,7 @@
 
    protected void m_6735_(ServerLevel p_217243_, Mob p_217244_, long p_217245_) {
       BlockPos blockpos = this.m_217248_(p_217244_);
-      boolean flag = blockpos.m_123314_(p_217244_.m_20183_(), (double)this.f_217232_);
+      boolean flag = blockpos.closerThan(p_217244_.m_20183_(), (double)this.f_217232_);
       if (!flag) {
          BehaviorUtils.m_22617_(p_217244_, m_217250_(p_217244_, blockpos), this.f_217233_, this.f_217232_);
       }
