--- a/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
+++ b/net/minecraft/world/entity/ai/behavior/InsideBrownianWalk.java
@@ -18,7 +_,7 @@
                   return false;
                } else {
                   BlockPos blockpos = p_258394_.m_20183_();
-                  List<BlockPos> list = BlockPos.m_121990_(blockpos.m_7918_(-1, -1, -1), blockpos.m_7918_(1, 1, 1)).map(BlockPos::m_7949_).collect(Collectors.toList());
+                  List<BlockPos> list = BlockPos.m_121990_(blockpos.offset(-1, -1, -1), blockpos.offset(1, 1, 1)).map(BlockPos::m_7949_).collect(Collectors.toList());
                   Collections.shuffle(list);
                   list.stream().filter((p_23230_) -> {
                      return !p_258393_.m_45527_(p_23230_);
