--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -120,7 +_,7 @@
       return p_259365_.isEmpty() ? false : p_259365_.get().stream().filter((p_289329_) -> {
          return p_289329_.m_6095_() == p_260091_.m_6095_();
       }).filter((p_289331_) -> {
-         return p_259764_.m_203195_(p_289331_.m_20182_(), 2.0D);
+         return p_259764_.closerToCenterThan(p_289331_.m_20182_(), 2.0D);
       }).anyMatch((p_258454_) -> {
          return m_257446_(p_258454_.m_6274_(), p_259764_);
       });
@@ -146,7 +_,7 @@
    }
 
    private static boolean m_23307_(ServerLevel p_23308_, LivingEntity p_23309_, GlobalPos p_23310_) {
-      return p_23310_.m_122640_() != p_23308_.m_46472_() || !p_23310_.m_122646_().m_203195_(p_23309_.m_20182_(), 3.0D);
+      return p_23310_.m_122640_() != p_23308_.m_46472_() || !p_23310_.m_122646_().closerToCenterThan(p_23309_.m_20182_(), 3.0D);
    }
 
    private static Optional<Set<GlobalPos>> m_261108_(MemoryAccessor<OptionalBox.Mu, Set<GlobalPos>> p_262178_, Optional<Set<GlobalPos>> p_261639_, ServerLevel p_261528_, BlockPos p_261874_) {
