--- a/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWithDoor.java
@@ -121,7 +_,7 @@
       return !brain.m_21874_(MemoryModuleType.f_148204_) ? false : brain.m_21952_(MemoryModuleType.f_148204_).get().stream().filter((p_201950_) -> {
          return p_201950_.m_6095_() == p_23305_.m_6095_();
       }).filter((p_201955_) -> {
-         return p_23306_.m_203195_(p_201955_.m_20182_(), 2.0D);
+         return p_23306_.closerToCenterThan(p_201955_.m_20182_(), 2.0D);
       }).anyMatch((p_201947_) -> {
          return m_23321_(p_23304_, p_201947_, p_23306_);
       });
@@ -147,7 +_,7 @@
    }
 
    private static boolean m_23307_(ServerLevel p_23308_, LivingEntity p_23309_, GlobalPos p_23310_) {
-      return p_23310_.m_122640_() != p_23308_.m_46472_() || !p_23310_.m_122646_().m_203195_(p_23309_.m_20182_(), 2.0D);
+      return p_23310_.m_122640_() != p_23308_.m_46472_() || !p_23310_.m_122646_().closerToCenterThan(p_23309_.m_20182_(), 2.0D);
    }
 
    private void m_23325_(ServerLevel p_23326_, LivingEntity p_23327_, BlockPos p_23328_) {
