--- a/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
+++ b/net/minecraft/world/entity/ai/behavior/JumpOnBed.java
@@ -84,7 +_,7 @@
 
    private boolean m_23379_(ServerLevel p_23380_, Mob p_23381_) {
       BlockPos blockpos = p_23381_.m_20183_();
-      BlockPos blockpos1 = blockpos.m_7495_();
+      BlockPos blockpos1 = blockpos.below();
       return this.m_23356_(p_23380_, blockpos) || this.m_23356_(p_23380_, blockpos1);
    }
 
