--- a/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
@@ -55,7 +_,7 @@
 
    public static <E extends Mob> boolean m_245086_(E p_251540_, BlockPos p_248879_) {
       Level level = p_251540_.m_9236_();
-      BlockPos blockpos = p_248879_.m_7495_();
+      BlockPos blockpos = p_248879_.below();
       return level.m_8055_(blockpos).m_60804_(level, blockpos) && p_251540_.m_21439_(WalkNodeEvaluator.m_77604_(level, p_248879_.m_122032_())) == 0.0F;
    }
 
@@ -93,13 +_,13 @@
       this.f_147633_ = 20;
       this.f_147631_ = Optional.of(p_147677_.m_20182_());
       BlockPos blockpos = p_147677_.m_20183_();
-      int i = blockpos.m_123341_();
-      int j = blockpos.m_123342_();
-      int k = blockpos.m_123343_();
-      this.f_147630_ = BlockPos.m_121886_(i - this.f_147628_, j - this.f_147627_, k - this.f_147628_, i + this.f_147628_, j + this.f_147627_, k + this.f_147628_).filter((p_217317_) -> {
+      long i = blockpos.getX();
+      long j = blockpos.getY();
+      long k = blockpos.getZ();
+      this.f_147630_ = BlockPos.betweenClosedStream(i - this.f_147628_, j - this.f_147627_, k - this.f_147628_, i + this.f_147628_, j + this.f_147627_, k + this.f_147628_).filter((p_217317_) -> {
          return !p_217317_.equals(blockpos);
       }).map((p_217314_) -> {
-         return new LongJumpToRandomPos.PossibleJump(p_217314_.m_7949_(), Mth.m_14165_(blockpos.m_123331_(p_217314_)));
+         return new LongJumpToRandomPos.PossibleJump(p_217314_.m_7949_(), Mth.m_14165_(blockpos.distSqr(p_217314_)));
       }).collect(Collectors.toCollection(Lists::newArrayList));
    }
 
@@ -135,7 +_,7 @@
                continue;
             }
 
-            Vec3 vec3 = Vec3.m_82512_(blockpos);
+            Vec3 vec3 = Vec3.atCenterOf(blockpos);
             Vec3 vec31 = this.m_217303_(p_217320_, vec3);
             if (vec31 == null) {
                continue;
@@ -165,9 +_,9 @@
 
    private boolean m_213828_(ServerLevel p_217300_, E p_217301_, BlockPos p_217302_) {
       BlockPos blockpos = p_217301_.m_20183_();
-      int i = blockpos.m_123341_();
-      int j = blockpos.m_123343_();
-      return i == p_217302_.m_123341_() && j == p_217302_.m_123343_() ? false : this.f_217290_.test(p_217301_, p_217302_);
+      long i = blockpos.getX();
+      long j = blockpos.getZ();
+      return i == p_217302_.getX() && j == p_217302_.getZ() ? false : this.f_217290_.test(p_217301_, p_217302_);
    }
 
    @Nullable
@@ -188,14 +_,14 @@
    @Nullable
    private Vec3 m_217306_(Mob p_217307_, Vec3 p_217308_, int p_217309_) {
       Vec3 vec3 = p_217307_.m_20182_();
-      Vec3 vec31 = (new Vec3(p_217308_.f_82479_ - vec3.f_82479_, 0.0D, p_217308_.f_82481_ - vec3.f_82481_)).m_82541_().m_82490_(0.5D);
+      Vec3 vec31 = (new Vec3(p_217308_.m_7096_() - vec3.m_7096_(), 0.0D, p_217308_.m_7094_() - vec3.m_7094_())).m_82541_().m_82490_(0.5D);
       p_217308_ = p_217308_.m_82546_(vec31);
       Vec3 vec32 = p_217308_.m_82546_(vec3);
       float f = (float)p_217309_ * (float)Math.PI / 180.0F;
-      double d0 = Math.atan2(vec32.f_82481_, vec32.f_82479_);
-      double d1 = vec32.m_82492_(0.0D, vec32.f_82480_, 0.0D).m_82556_();
+      double d0 = Math.atan2(vec32.m_7094_(), vec32.m_7096_());
+      double d1 = vec32.m_82492_(0.0D, vec32.m_7098_(), 0.0D).m_82556_();
       double d2 = Math.sqrt(d1);
-      double d3 = vec32.f_82480_;
+      double d3 = vec32.m_7098_();
       double d4 = Math.sin((double)(2.0F * f));
       double d5 = 0.08D;
       double d6 = Math.pow(Math.cos((double)f), 2.0D);
@@ -223,7 +_,7 @@
                double d16 = d7 / d8 * d15 - Math.pow(d15, 2.0D) * 0.08D / (2.0D * d11 * Math.pow(d8, 2.0D));
                double d17 = d15 * d10;
                double d18 = d15 * d9;
-               Vec3 vec34 = new Vec3(vec3.f_82479_ + d17, vec3.f_82480_ + d16, vec3.f_82481_ + d18);
+               Vec3 vec34 = new Vec3(vec3.m_7096_() + d17, vec3.m_7098_() + d16, vec3.m_7094_() + d18);
                if (vec33 != null && !this.m_247472_(p_217307_, entitydimensions, vec33, vec34)) {
                   return null;
                }
