--- a/net/minecraft/world/entity/ai/behavior/LongJumpToPreferredBlock.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpToPreferredBlock.java
@@ -42,7 +_,7 @@
             Optional<LongJumpToRandomPos.PossibleJump> optional = super.m_213675_(p_217273_);
             if (optional.isPresent()) {
                LongJumpToRandomPos.PossibleJump longjumptorandompos$possiblejump = optional.get();
-               if (p_217273_.m_8055_(blockpos$mutableblockpos.m_122159_(longjumptorandompos$possiblejump.m_147693_(), Direction.DOWN)).m_204336_(this.f_217259_)) {
+               if (p_217273_.m_8055_(blockpos$mutableblockpos.setWithOffset(longjumptorandompos$possiblejump.m_147693_(), Direction.DOWN)).m_204336_(this.f_217259_)) {
                   return optional;
                }
 
