--- a/net/minecraft/world/entity/ai/behavior/LongJumpToPreferredBlock.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpToPreferredBlock.java
@@ -43,7 +_,7 @@
             Optional<LongJumpToRandomPos.PossibleJump> optional = super.m_213675_(p_217273_);
             if (optional.isPresent()) {
                LongJumpToRandomPos.PossibleJump longjumptorandompos$possiblejump = optional.get();
-               if (p_217273_.m_8055_(blockpos$mutableblockpos.m_122159_(longjumptorandompos$possiblejump.m_147693_(), Direction.DOWN)).m_204336_(this.f_217259_)) {
+               if (p_217273_.m_8055_(blockpos$mutableblockpos.setWithOffset(longjumptorandompos$possiblejump.m_147693_(), Direction.DOWN)).m_204336_(this.f_217259_)) {
                   return optional;
                }
 
@@ -60,6 +_,6 @@
    }
 
    private boolean m_217286_(ServerLevel p_217287_, BlockPos p_217288_) {
-      return p_217287_.m_6425_(p_217288_).m_76178_() && p_217287_.m_6425_(p_217288_.m_7495_()).m_76178_();
+      return p_217287_.m_6425_(p_217288_).m_76178_() && p_217287_.m_6425_(p_217288_.below()).m_76178_();
    }
 }
