--- a/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
+++ b/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
@@ -48,7 +_,7 @@
                   } else {
                      for(int i = 0; i < 10; ++i) {
                         Vec3 vec3 = LandRandomPos.m_148488_(p_275029_, 20, 8);
-                        if (vec3 != null && p_275028_.m_8802_(BlockPos.m_274446_(vec3))) {
+                        if (vec3 != null && p_275028_.m_8802_(BlockPos.containing(vec3))) {
                            p_258560_.m_257512_(new WalkTarget(vec3, 0.6F, 0));
                            break;
                         }
