--- a/net/minecraft/world/entity/ai/behavior/RamTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/RamTarget.java
@@ -58,7 +_,7 @@
       BlockPos blockpos = p_217360_.m_20183_();
       Brain<?> brain = p_217360_.m_6274_();
       Vec3 vec3 = brain.m_21952_(MemoryModuleType.f_148203_).get();
-      this.f_147807_ = (new Vec3((double)blockpos.m_123341_() - vec3.m_7096_(), 0.0D, (double)blockpos.m_123343_() - vec3.m_7094_())).m_82541_();
+      this.f_147807_ = (new Vec3((double)blockpos.getX() - vec3.m_7096_(), 0.0D, (double)blockpos.getZ() - vec3.m_7094_())).m_82541_();
       brain.m_21879_(MemoryModuleType.f_26370_, new WalkTarget(vec3, this.f_147805_, 0));
    }
 
@@ -98,7 +_,7 @@
    private boolean m_217362_(ServerLevel p_217363_, Goat p_217364_) {
       Vec3 vec3 = p_217364_.m_20184_().m_82542_(1.0D, 0.0D, 1.0D).m_82541_();
       BlockPos blockpos = BlockPos.m_274446_(p_217364_.m_20182_().m_82549_(vec3));
-      return p_217363_.m_8055_(blockpos).m_204336_(BlockTags.f_215832_) || p_217363_.m_8055_(blockpos.m_7494_()).m_204336_(BlockTags.f_215832_);
+      return p_217363_.m_8055_(blockpos).m_204336_(BlockTags.f_215832_) || p_217363_.m_8055_(blockpos.above()).m_204336_(BlockTags.f_215832_);
    }
 
    protected void m_217355_(ServerLevel p_217356_, Goat p_217357_) {
