--- a/net/minecraft/world/entity/ai/behavior/RandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/RandomStroll.java
@@ -83,7 +_,7 @@
             vec31 = p_259491_.m_20182_().m_82549_(p_259491_.m_20182_().m_82505_(vec3).m_82541_().m_82542_((double)aint[0], (double)aint[1], (double)aint[0]));
          }
 
-         if (vec31 == null || p_259491_.m_9236_().m_6425_(BlockPos.m_274446_(vec31)).m_76178_()) {
+         if (vec31 == null || p_259491_.m_9236_().m_6425_(BlockPos.containing(vec31)).m_76178_()) {
             return vec3;
          }
 
@@ -96,6 +_,6 @@
    @Nullable
    private static Vec3 m_257997_(PathfinderMob p_260316_, int p_259038_, int p_259696_) {
       Vec3 vec3 = p_260316_.m_20252_(0.0F);
-      return AirAndWaterRandomPos.m_148357_(p_260316_, p_259038_, p_259696_, -2, vec3.f_82479_, vec3.f_82481_, (double)((float)Math.PI / 2F));
+      return AirAndWaterRandomPos.m_148357_(p_260316_, p_259038_, p_259696_, -2, vec3.m_7096_(), vec3.m_7094_(), (double)((float)Math.PI / 2F));
    }
 }
