--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -27,7 +_,7 @@
    protected void m_6735_(ServerLevel p_23792_, LivingEntity p_23793_, long p_23794_) {
       Brain<?> brain = p_23793_.m_6274_();
       BlockPos blockpos = brain.m_21952_(MemoryModuleType.f_26362_).get().m_122646_();
-      if (blockpos.m_123314_(p_23793_.m_20183_(), 3.0D)) {
+      if (blockpos.closerThan(p_23793_.m_20183_(), 3.0D)) {
          BlockState blockstate = p_23792_.m_8055_(blockpos);
          if (blockstate.m_60713_(Blocks.f_50680_)) {
             BellBlock bellblock = (BellBlock)blockstate.m_60734_();
