--- a/net/minecraft/world/entity/ai/behavior/RingBell.java
+++ b/net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -21,7 +_,7 @@
                   return false;
                } else {
                   BlockPos blockpos = p_259094_.m_258051_(p_259028_).m_122646_();
-                  if (blockpos.m_123314_(p_260317_.m_20183_(), 3.0D)) {
+                  if (blockpos.closerThan(p_260317_.m_20183_(), 3.0D)) {
                      BlockState blockstate = p_259026_.m_8055_(blockpos);
                      if (blockstate.m_60713_(Blocks.f_50680_)) {
                         BellBlock bellblock = (BellBlock)blockstate.m_60734_();
