--- a/net/minecraft/world/entity/ai/behavior/SetClosestHomeAsWalkTarget.java
+++ b/net/minecraft/world/entity/ai/behavior/SetClosestHomeAsWalkTarget.java
@@ -46,7 +_,7 @@
          Optional<BlockPos> optional = poimanager.m_27192_((p_217376_) -> {
             return p_217376_.m_203565_(PoiTypes.f_218060_);
          }, p_23880_.m_20183_(), 48, PoiManager.Occupancy.ANY);
-         return optional.isPresent() && !(optional.get().m_123331_(pathfindermob.m_20183_()) <= 4.0D);
+         return optional.isPresent() && !(optional.get().distSqr(pathfindermob.m_20183_()) <= 4.0D);
       }
    }
 
