--- a/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
+++ b/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
@@ -19,7 +_,7 @@
                boolean flag = j + 300L <= p_288846_;
                if (mutableint.getValue() <= i && !flag) {
                   BlockPos blockpos = p_259055_.m_258051_(p_260296_).m_122646_();
-                  if (blockpos.m_123314_(p_288845_.m_20183_(), (double)p_260263_)) {
+                  if (blockpos.closerThan(p_288845_.m_20183_(), (double)p_260263_)) {
                      mutableint.increment();
                   }
 
