--- a/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
+++ b/net/minecraft/world/entity/ai/behavior/SetHiddenState.java
@@ -28,7 +_,7 @@
       boolean flag = optional.get() + 300L <= p_23936_;
       if (this.f_23929_ <= this.f_23928_ && !flag) {
          BlockPos blockpos = brain.m_21952_(MemoryModuleType.f_26324_).get().m_122646_();
-         if (blockpos.m_123314_(p_23935_.m_20183_(), (double)this.f_23927_)) {
+         if (blockpos.closerThan(p_23935_.m_20183_(), (double)this.f_23927_)) {
             ++this.f_23929_;
          }
 
