--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.java
@@ -13,7 +_,7 @@
 
 public class SetWalkTargetAwayFrom {
    public static BehaviorControl<PathfinderMob> m_257620_(MemoryModuleType<BlockPos> p_259330_, float p_259719_, int p_259965_, boolean p_259828_) {
-      return m_257681_(p_259330_, p_259719_, p_259965_, p_259828_, Vec3::m_82539_);
+      return m_257681_(p_259330_, p_259719_, p_259965_, p_259828_, Vec3::atBottomCenterOf);
    }
 
    public static OneShot<PathfinderMob> m_257370_(MemoryModuleType<? extends Entity> p_259598_, float p_260183_, int p_260077_, boolean p_259761_) {
