--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.java
@@ -27,7 +_,7 @@
    }
 
    public static SetWalkTargetAwayFrom<BlockPos> m_24012_(MemoryModuleType<BlockPos> p_24013_, float p_24014_, int p_24015_, boolean p_24016_) {
-      return new SetWalkTargetAwayFrom<>(p_24013_, p_24014_, p_24015_, p_24016_, Vec3::m_82539_);
+      return new SetWalkTargetAwayFrom<>(p_24013_, p_24014_, p_24015_, p_24016_, Vec3::atBottomCenterOf);
    }
 
    public static SetWalkTargetAwayFrom<? extends Entity> m_24019_(MemoryModuleType<? extends Entity> p_24020_, float p_24021_, int p_24022_, boolean p_24023_) {
