--- a/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
+++ b/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.java
@@ -18,13 +_,13 @@
                GlobalPos globalpos = p_258717_.m_258051_(p_258711_);
                Optional<Long> optional = p_258717_.m_257828_(p_258709_);
                if (globalpos.m_122640_() == p_275056_.m_46472_() && (!optional.isPresent() || p_275056_.m_46467_() - optional.get() <= (long)p_259504_)) {
-                  if (globalpos.m_122646_().m_123333_(p_275057_.m_20183_()) > p_260360_) {
+                  if (globalpos.m_122646_().distManhattan(p_275057_.m_20183_()) > p_260360_) {
                      Vec3 vec3 = null;
                      int i = 0;
                      int j = 1000;
 
-                     while(vec3 == null || BlockPos.m_274446_(vec3).m_123333_(p_275057_.m_20183_()) > p_260360_) {
-                        vec3 = DefaultRandomPos.m_148412_(p_275057_, 15, 7, Vec3.m_82539_(globalpos.m_122646_()), (double)((float)Math.PI / 2F));
+                     while(vec3 == null || BlockPos.containing(vec3).distManhattan(p_275057_.m_20183_()) > p_260360_) {
+                        vec3 = DefaultRandomPos.m_148412_(p_275057_, 15, 7, Vec3.atBottomCenterOf(globalpos.m_122646_()), (double)((float)Math.PI / 2F));
                         ++i;
                         if (i == 1000) {
                            p_275057_.m_35428_(p_259685_);
@@ -35,7 +_,7 @@
                      }
 
                      p_258710_.m_257512_(new WalkTarget(vec3, p_259842_, p_259530_));
-                  } else if (globalpos.m_122646_().m_123333_(p_275057_.m_20183_()) > p_259530_) {
+                  } else if (globalpos.m_122646_().distManhattan(p_275057_.m_20183_()) > p_259530_) {
                      p_258710_.m_257512_(new WalkTarget(globalpos.m_122646_(), p_259842_, p_259530_));
                   }
                } else {
