--- a/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
+++ b/net/minecraft/world/entity/ai/behavior/SocializeAtBell.java
@@ -21,7 +_,7 @@
    protected boolean m_6114_(ServerLevel p_24170_, LivingEntity p_24171_) {
       Brain<?> brain = p_24171_.m_6274_();
       Optional<GlobalPos> optional = brain.m_21952_(MemoryModuleType.f_26362_);
-      return p_24170_.m_213780_().m_188503_(100) == 0 && optional.isPresent() && p_24170_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().m_203195_(p_24171_.m_20182_(), 4.0D) && brain.m_21952_(MemoryModuleType.f_148205_).get().m_186130_((p_24189_) -> {
+      return p_24170_.m_213780_().m_188503_(100) == 0 && optional.isPresent() && p_24170_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().closerToCenterThan(p_24171_.m_20182_(), 4.0D) && brain.m_21952_(MemoryModuleType.f_148205_).get().m_186130_((p_24189_) -> {
          return EntityType.f_20492_.equals(p_24189_.m_6095_());
       });
    }
