--- a/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
@@ -37,7 +_,7 @@
          List<GlobalPos> list = optional.get();
          if (!list.isEmpty()) {
             this.f_24360_ = list.get(p_24375_.m_213780_().m_188503_(list.size()));
-            return this.f_24360_ != null && p_24375_.m_46472_() == this.f_24360_.m_122640_() && optional1.get().m_122646_().m_203195_(p_24376_.m_20182_(), (double)this.f_24358_);
+            return this.f_24360_ != null && p_24375_.m_46472_() == this.f_24360_.m_122640_() && optional1.get().m_122646_().closerToCenterThan(p_24376_.m_20182_(), (double)this.f_24358_);
          }
       }
 
