--- a/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
@@ -29,7 +_,7 @@
 
    protected boolean m_6114_(ServerLevel p_24319_, PathfinderMob p_24320_) {
       Optional<GlobalPos> optional = p_24320_.m_6274_().m_21952_(this.f_24303_);
-      return optional.isPresent() && p_24319_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().m_203195_(p_24320_.m_20182_(), (double)this.f_24305_);
+      return optional.isPresent() && p_24319_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().closerToCenterThan(p_24320_.m_20182_(), (double)this.f_24305_);
    }
 
    protected void m_6735_(ServerLevel p_24322_, PathfinderMob p_24323_, long p_24324_) {
