--- a/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoi.java
@@ -27,7 +_,7 @@
 
    protected boolean m_6114_(ServerLevel p_24345_, PathfinderMob p_24346_) {
       Optional<GlobalPos> optional = p_24346_.m_6274_().m_21952_(this.f_24327_);
-      return optional.isPresent() && p_24345_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().m_203195_(p_24346_.m_20182_(), (double)this.f_24329_);
+      return optional.isPresent() && p_24345_.m_46472_() == optional.get().m_122640_() && optional.get().m_122646_().closerToCenterThan(p_24346_.m_20182_(), (double)this.f_24329_);
    }
 
    protected void m_6735_(ServerLevel p_24348_, PathfinderMob p_24349_, long p_24350_) {
