--- a/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollAroundPoi.java
@@ -21,7 +_,7 @@
          return p_258827_.group(p_258827_.m_257492_(MemoryModuleType.f_26370_), p_258827_.m_257495_(p_259159_)).apply(p_258827_, (p_258821_, p_258822_) -> {
             return (p_258834_, p_258835_, p_258836_) -> {
                GlobalPos globalpos = p_258827_.m_258051_(p_258822_);
-               if (p_258834_.m_46472_() == globalpos.m_122640_() && globalpos.m_122646_().m_203195_(p_258835_.m_20182_(), (double)p_260102_)) {
+               if (p_258834_.m_46472_() == globalpos.m_122640_() && globalpos.m_122646_().closerToCenterThan(p_258835_.m_20182_(), (double)p_260102_)) {
                   if (p_258836_ <= mutablelong.getValue()) {
                      return true;
                   } else {
