--- a/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
+++ b/net/minecraft/world/entity/ai/behavior/StrollToPoiList.java
@@ -20,7 +_,7 @@
                   return false;
                } else {
                   GlobalPos globalpos1 = list.get(p_259940_.m_213780_().m_188503_(list.size()));
-                  if (globalpos1 != null && p_259940_.m_46472_() == globalpos1.m_122640_() && globalpos.m_122646_().m_203195_(p_259222_.m_20182_(), (double)p_259533_)) {
+                  if (globalpos1 != null && p_259940_.m_46472_() == globalpos1.m_122640_() && globalpos.m_122646_().closerToCenterThan(p_259222_.m_20182_(), (double)p_259533_)) {
                      if (p_260161_ > mutablelong.getValue()) {
                         p_259574_.m_257512_(new WalkTarget(globalpos1.m_122646_(), p_259895_, p_260285_));
                         mutablelong.setValue(p_260161_ + 100L);
