--- a/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
+++ b/net/minecraft/world/entity/ai/behavior/TryFindLandNearWater.java
@@ -37,9 +_,9 @@
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
          for(BlockPos blockpos1 : BlockPos.m_121925_(blockpos, this.f_217442_, this.f_217442_, this.f_217442_)) {
-            if ((blockpos1.m_123341_() != blockpos.m_123341_() || blockpos1.m_123343_() != blockpos.m_123343_()) && p_217463_.m_8055_(blockpos1).m_60742_(p_217463_, blockpos1, collisioncontext).m_83281_() && !p_217463_.m_8055_(blockpos$mutableblockpos.m_122159_(blockpos1, Direction.DOWN)).m_60742_(p_217463_, blockpos1, collisioncontext).m_83281_()) {
+            if ((blockpos1.getX() != blockpos.getX() || blockpos1.getZ() != blockpos.getZ()) && p_217463_.m_8055_(blockpos1).m_60742_(p_217463_, blockpos1, collisioncontext).m_83281_() && !p_217463_.m_8055_(blockpos$mutableblockpos.setWithOffset(blockpos1, Direction.DOWN)).m_60742_(p_217463_, blockpos1, collisioncontext).m_83281_()) {
                for(Direction direction : Direction.Plane.HORIZONTAL) {
-                  blockpos$mutableblockpos.m_122159_(blockpos1, direction);
+                  blockpos$mutableblockpos.setWithOffset(blockpos1, direction);
                   if (p_217463_.m_8055_(blockpos$mutableblockpos).m_60795_() && p_217463_.m_8055_(blockpos$mutableblockpos.m_122173_(Direction.DOWN)).m_60713_(Blocks.f_49990_)) {
                      this.f_217444_ = p_217465_ + 40L;
                      BehaviorUtils.m_22617_(p_217464_, blockpos1, this.f_217443_, 0);
