--- a/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
@@ -29,7 +_,7 @@
 
    protected boolean m_6114_(ServerLevel p_24521_, LivingEntity p_24522_) {
       GlobalPos globalpos = p_24522_.m_6274_().m_21952_(this.f_24515_).get();
-      return p_24521_.m_46472_() == globalpos.m_122640_() && globalpos.m_122646_().m_203195_(p_24522_.m_20182_(), 16.0D);
+      return p_24521_.m_46472_() == globalpos.m_122640_() && globalpos.m_122646_().closerToCenterThan(p_24522_.m_20182_(), 16.0D);
    }
 
    protected void m_6735_(ServerLevel p_24524_, LivingEntity p_24525_, long p_24526_) {
