--- a/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
@@ -23,7 +_,7 @@
             return (p_259843_, p_259259_, p_260036_) -> {
                GlobalPos globalpos = p_259215_.m_258051_(p_259498_);
                BlockPos blockpos = globalpos.m_122646_();
-               if (p_259843_.m_46472_() == globalpos.m_122640_() && blockpos.m_203195_(p_259259_.m_20182_(), 16.0D)) {
+               if (p_259843_.m_46472_() == globalpos.m_122640_() && blockpos.closerToCenterThan(p_259259_.m_20182_(), 16.0D)) {
                   ServerLevel serverlevel = p_259843_.m_7654_().m_129880_(globalpos.m_122640_());
                   if (serverlevel != null && serverlevel.m_8904_().m_27091_(blockpos, p_259460_)) {
                      if (m_24530_(serverlevel, blockpos, p_259259_)) {
