--- a/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -31,7 +_,7 @@
                   SectionPos sectionpos = SectionPos.m_123199_(blockpos);
                   SectionPos sectionpos1 = BehaviorUtils.m_22581_(p_258874_, sectionpos, 2);
                   if (sectionpos1 != sectionpos) {
-                     vec3 = DefaultRandomPos.m_148412_(p_258875_, p_259708_, p_259311_, Vec3.m_82539_(sectionpos1.m_123250_()), (double)((float)Math.PI / 2F));
+                     vec3 = DefaultRandomPos.m_148412_(p_258875_, p_259708_, p_259311_, Vec3.atBottomCenterOf(sectionpos1.m_123250_()), (double)((float)Math.PI / 2F));
                   } else {
                      vec3 = LandRandomPos.m_148488_(p_258875_, p_259708_, p_259311_);
                   }
