--- a/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ b/net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -48,7 +_,7 @@
    }
 
    private void m_24561_(PathfinderMob p_24562_, SectionPos p_24563_) {
-      Optional<Vec3> optional = Optional.ofNullable(DefaultRandomPos.m_148412_(p_24562_, this.f_24543_, this.f_24544_, Vec3.m_82539_(p_24563_.m_123250_()), (double)((float)Math.PI / 2F)));
+      Optional<Vec3> optional = Optional.ofNullable(DefaultRandomPos.m_148412_(p_24562_, this.f_24543_, this.f_24544_, Vec3.atBottomCenterOf(p_24563_.m_123250_()), (double)((float)Math.PI / 2F)));
       p_24562_.m_6274_().m_21886_(MemoryModuleType.f_26370_, optional.map((p_24567_) -> {
          return new WalkTarget(p_24567_, this.f_24542_, 0);
       }));
