--- a/net/minecraft/world/entity/ai/behavior/WorkAtPoi.java
+++ b/net/minecraft/world/entity/ai/behavior/WorkAtPoi.java
@@ -26,7 +_,7 @@
       } else {
          this.f_24804_ = p_24827_.m_46467_();
          GlobalPos globalpos = p_24828_.m_6274_().m_21952_(MemoryModuleType.f_26360_).get();
-         return globalpos.m_122640_() == p_24827_.m_46472_() && globalpos.m_122646_().m_203195_(p_24828_.m_20182_(), 1.73D);
+         return globalpos.m_122640_() == p_24827_.m_46472_() && globalpos.m_122646_().closerToCenterThan(p_24828_.m_20182_(), 1.73D);
       }
    }
 
@@ -53,7 +_,7 @@
          return false;
       } else {
          GlobalPos globalpos = optional.get();
-         return globalpos.m_122640_() == p_24830_.m_46472_() && globalpos.m_122646_().m_203195_(p_24831_.m_20182_(), 1.73D);
+         return globalpos.m_122640_() == p_24830_.m_46472_() && globalpos.m_122646_().closerToCenterThan(p_24831_.m_20182_(), 1.73D);
       }
    }
 }
