--- a/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
+++ b/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
@@ -296,7 +_,7 @@
    @OnlyIn(Dist.CLIENT)
    public class RenderChunk {
       public static final int f_173716_ = 16;
-      public final int f_173717_;
+      public final long f_173717_;
       public final AtomicReference<ChunkRenderDispatcher.CompiledChunk> f_112784_ = new AtomicReference<>(ChunkRenderDispatcher.CompiledChunk.f_112748_);
       final AtomicInteger f_202433_ = new AtomicInteger(0);
       @Nullable
@@ -320,13 +_,13 @@
       });
       private boolean f_112795_;
 
-      public RenderChunk(int p_202436_, int p_202437_, int p_202438_, int p_202439_) {
+      public RenderChunk(long p_202436_, int p_202437_, int p_202438_, int p_202439_) {
          this.f_173717_ = p_202436_;
-         this.m_112801_(p_202437_, p_202438_, p_202439_);
+         this.setOrigin(p_202437_, p_202438_, p_202439_);
       }
 
       private boolean m_112822_(BlockPos p_112823_) {
-         return ChunkRenderDispatcher.this.f_112681_.m_6522_(SectionPos.m_123171_(p_112823_.m_123341_()), SectionPos.m_123171_(p_112823_.m_123343_()), ChunkStatus.f_62326_, false) != null;
+         return ChunkRenderDispatcher.this.f_112681_.getChunk(SectionPos.blockToSectionCoord(p_112823_.getX()), SectionPos.blockToSectionCoord(p_112823_.getZ()), ChunkStatus.f_62326_, false) != null;
       }
 
       public boolean m_112798_() {
@@ -346,22 +_,22 @@
          return this.f_112790_.get(p_112808_);
       }
 
-      public void m_112801_(int p_112802_, int p_112803_, int p_112804_) {
+      public void setOrigin(long p_112802_, long p_112803_, long p_112804_) {
          this.m_112846_();
-         this.f_112793_.m_122178_(p_112802_, p_112803_, p_112804_);
+         this.f_112793_.set(p_112802_, p_112803_, p_112804_);
          this.f_112785_ = new AABB((double)p_112802_, (double)p_112803_, (double)p_112804_, (double)(p_112802_ + 16), (double)(p_112803_ + 16), (double)(p_112804_ + 16));
 
          for(Direction direction : Direction.values()) {
-            this.f_112794_[direction.ordinal()].m_122190_(this.f_112793_).m_122175_(direction, 16);
+            this.f_112794_[direction.ordinal()].set(this.f_112793_).move(direction, 16);
          }
 
       }
 
       protected double m_112832_() {
          Camera camera = Minecraft.m_91087_().f_91063_.m_109153_();
-         double d0 = this.f_112785_.f_82288_ + 8.0D - camera.m_90583_().f_82479_;
-         double d1 = this.f_112785_.f_82289_ + 8.0D - camera.m_90583_().f_82480_;
-         double d2 = this.f_112785_.f_82290_ + 8.0D - camera.m_90583_().f_82481_;
+         double d0 = this.f_112785_.f_82288_.add(me.alphamode.mclong.core.BigConstants.EIGHT).subtract(camera.m_90583_().getX()).doubleValue();
+         double d1 = this.f_112785_.f_82289_.add(me.alphamode.mclong.core.BigConstants.EIGHT).subtract(camera.m_90583_().getY()).doubleValue();
+         double d2 = this.f_112785_.f_82290_.add(me.alphamode.mclong.core.BigConstants.EIGHT).subtract(camera.m_90583_().getZ()).doubleValue();
          return d0 * d0 + d1 * d1 + d2 * d2;
       }
 
@@ -446,7 +_,7 @@
          boolean flag = this.m_194419_();
          BlockPos blockpos = this.f_112793_.m_7949_();
          int i = 1;
-         RenderChunkRegion renderchunkregion = p_200438_.m_200465_(ChunkRenderDispatcher.this.f_112681_, blockpos.m_7918_(-1, -1, -1), blockpos.m_7918_(16, 16, 16), 1);
+         RenderChunkRegion renderchunkregion = p_200438_.m_200465_(ChunkRenderDispatcher.this.f_112681_, blockpos.offset(-1, -1, -1), blockpos.offset(16, 16, 16), 1);
          boolean flag1 = this.f_112784_.get() == ChunkRenderDispatcher.CompiledChunk.f_112748_;
          if (flag1 && flag) {
             this.f_202433_.incrementAndGet();
@@ -528,9 +_,9 @@
                return CompletableFuture.completedFuture(ChunkRenderDispatcher.ChunkTaskResult.CANCELLED);
             } else {
                Vec3 vec3 = ChunkRenderDispatcher.this.m_112727_();
-               float f = (float)vec3.f_82479_;
-               float f1 = (float)vec3.f_82480_;
-               float f2 = (float)vec3.f_82481_;
+               float f = (float)vec3.m_7096_();
+               float f1 = (float)vec3.m_7098_();
+               float f2 = (float)vec3.m_7094_();
                ChunkRenderDispatcher.RenderChunk.RebuildTask.CompileResults chunkrenderdispatcher$renderchunk$rebuildtask$compileresults = this.m_234467_(f, f1, f2, p_112872_);
                RenderChunk.this.m_234465_(chunkrenderdispatcher$renderchunk$rebuildtask$compileresults.f_234484_);
                if (this.f_112848_.get()) {
@@ -568,7 +_,7 @@
             ChunkRenderDispatcher.RenderChunk.RebuildTask.CompileResults chunkrenderdispatcher$renderchunk$rebuildtask$compileresults = new ChunkRenderDispatcher.RenderChunk.RebuildTask.CompileResults();
             int i = 1;
             BlockPos blockpos = RenderChunk.this.f_112793_.m_7949_();
-            BlockPos blockpos1 = blockpos.m_7918_(15, 15, 15);
+            BlockPos blockpos1 = blockpos.offset(15, 15, 15);
             VisGraph visgraph = new VisGraph();
             RenderChunkRegion renderchunkregion = this.f_112858_;
             this.f_112858_ = null;
@@ -612,7 +_,7 @@
                      }
 
                      posestack.m_85836_();
-                     posestack.m_252880_((float)(blockpos2.m_123341_() & 15), (float)(blockpos2.m_123342_() & 15), (float)(blockpos2.m_123343_() & 15));
+                     posestack.m_252880_((float)(blockpos2.getX() & 15), (float)(blockpos2.getY() & 15), (float)(blockpos2.getZ() & 15));
                      blockrenderdispatcher.m_234355_(blockstate, blockpos2, renderchunkregion, posestack, bufferbuilder2, true, randomsource);
                      posestack.m_85849_();
                   }
@@ -621,7 +_,7 @@
                if (set.contains(RenderType.m_110466_())) {
                   BufferBuilder bufferbuilder1 = p_234471_.m_108839_(RenderType.m_110466_());
                   if (!bufferbuilder1.m_231164_()) {
-                     bufferbuilder1.m_277127_(VertexSorting.m_277071_(p_234468_ - (float)blockpos.m_123341_(), p_234469_ - (float)blockpos.m_123342_(), p_234470_ - (float)blockpos.m_123343_()));
+                     bufferbuilder1.m_277127_(VertexSorting.m_277071_(p_234468_ - (float)blockpos.getX(), p_234469_ - (float)blockpos.getY(), p_234470_ - (float)blockpos.getZ()));
                      chunkrenderdispatcher$renderchunk$rebuildtask$compileresults.f_234488_ = bufferbuilder1.m_166770_();
                   }
                }
@@ -693,15 +_,15 @@
                return CompletableFuture.completedFuture(ChunkRenderDispatcher.ChunkTaskResult.CANCELLED);
             } else {
                Vec3 vec3 = ChunkRenderDispatcher.this.m_112727_();
-               float f = (float)vec3.f_82479_;
-               float f1 = (float)vec3.f_82480_;
-               float f2 = (float)vec3.f_82481_;
+               float f = (float)vec3.m_7096_();
+               float f1 = (float)vec3.m_7098_();
+               float f2 = (float)vec3.m_7094_();
                BufferBuilder.SortState bufferbuilder$sortstate = this.f_112886_.f_112754_;
                if (bufferbuilder$sortstate != null && !this.f_112886_.m_112758_(RenderType.m_110466_())) {
                   BufferBuilder bufferbuilder = p_112893_.m_108839_(RenderType.m_110466_());
                   RenderChunk.this.m_112805_(bufferbuilder);
                   bufferbuilder.m_166775_(bufferbuilder$sortstate);
-                  bufferbuilder.m_277127_(VertexSorting.m_277071_(f - (float)RenderChunk.this.f_112793_.m_123341_(), f1 - (float)RenderChunk.this.f_112793_.m_123342_(), f2 - (float)RenderChunk.this.f_112793_.m_123343_()));
+                  bufferbuilder.m_277127_(VertexSorting.m_277071_(f - (float)RenderChunk.this.f_112793_.getX(), f1 - (float)RenderChunk.this.f_112793_.getY(), f2 - (float)RenderChunk.this.f_112793_.getZ()));
                   this.f_112886_.f_112754_ = bufferbuilder.m_166770_();
                   BufferBuilder.RenderedBuffer bufferbuilder$renderedbuffer = bufferbuilder.m_231175_();
                   if (this.f_112848_.get()) {
