--- a/net/minecraft/client/renderer/chunk/RenderChunkRegion.java
+++ b/net/minecraft/client/renderer/chunk/RenderChunkRegion.java
@@ -29,14 +_,14 @@
    }
 
    public BlockState m_8055_(BlockPos p_112947_) {
-      int i = SectionPos.m_123171_(p_112947_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112947_.m_123343_()) - this.f_112900_;
+      int i = SectionPos.m_235865_(p_112947_.getX()) - this.f_112899_;
+      int j = SectionPos.m_235865_(p_112947_.getZ()) - this.f_112900_;
       return this.f_112905_[i][j].m_200453_(p_112947_);
    }
 
    public FluidState m_6425_(BlockPos p_112943_) {
-      int i = SectionPos.m_123171_(p_112943_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112943_.m_123343_()) - this.f_112900_;
+      int i = SectionPos.m_235865_(p_112943_.getX()) - this.f_112899_;
+      int j = SectionPos.m_235865_(p_112943_.getZ()) - this.f_112900_;
       return this.f_112905_[i][j].m_200453_(p_112943_).m_60819_();
    }
 
@@ -50,8 +_,8 @@
 
    @Nullable
    public BlockEntity m_7702_(BlockPos p_112945_) {
-      int i = SectionPos.m_123171_(p_112945_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112945_.m_123343_()) - this.f_112900_;
+      int i = SectionPos.m_235865_(p_112945_.getX()) - this.f_112899_;
+      int j = SectionPos.m_235865_(p_112945_.getZ()) - this.f_112900_;
       return this.f_112905_[i][j].m_200451_(p_112945_);
    }
 
@@ -59,11 +_,11 @@
       return this.f_112908_.m_6171_(p_112937_, p_112938_);
    }
 
-   public int m_141937_() {
-      return this.f_112908_.m_141937_();
+   public long getMinBuildHeight() {
+      return this.f_112908_.getMinBuildHeight();
    }
 
-   public int m_141928_() {
-      return this.f_112908_.m_141928_();
+   public long getHeight() {
+      return this.f_112908_.getHeight();
    }
 }
