--- a/net/minecraft/client/renderer/chunk/RenderChunkRegion.java
+++ b/net/minecraft/client/renderer/chunk/RenderChunkRegion.java
@@ -16,28 +_,28 @@
 
 @OnlyIn(Dist.CLIENT)
 public class RenderChunkRegion implements BlockAndTintGetter {
-   private final int f_112899_;
-   private final int f_112900_;
+   private final me.alphamode.mclong.math.BigInteger f_112899_;
+   private final me.alphamode.mclong.math.BigInteger f_112900_;
    protected final RenderChunk[][] f_112905_;
    protected final Level f_112908_;
 
-   RenderChunkRegion(Level p_200456_, int p_200457_, int p_200458_, RenderChunk[][] p_200459_) {
-      this.f_112908_ = p_200456_;
-      this.f_112899_ = p_200457_;
-      this.f_112900_ = p_200458_;
-      this.f_112905_ = p_200459_;
+   RenderChunkRegion(Level pLevel, me.alphamode.mclong.math.BigInteger pCenterX, me.alphamode.mclong.math.BigInteger pCenterZ, RenderChunk[][] pChunks) {
+      this.f_112908_ = pLevel;
+      this.f_112899_ = pCenterX;
+      this.f_112900_ = pCenterZ;
+      this.f_112905_ = pChunks;
    }
 
    public BlockState m_8055_(BlockPos p_112947_) {
-      int i = SectionPos.m_123171_(p_112947_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112947_.m_123343_()) - this.f_112900_;
+      int i = SectionPos.blockToSectionCoord(p_112947_.getBigX()).subtract(this.f_112899_).intValue();
+      int j = SectionPos.blockToSectionCoord(p_112947_.getBigZ()).subtract(this.f_112900_).intValue();
       return this.f_112905_[i][j].m_200453_(p_112947_);
    }
 
    public FluidState m_6425_(BlockPos p_112943_) {
-      int i = SectionPos.m_123171_(p_112943_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112943_.m_123343_()) - this.f_112900_;
-      return this.f_112905_[i][j].m_200453_(p_112943_).m_60819_();
+      int i = SectionPos.blockToSectionCoord(p_112943_.getBigX()).subtract(this.f_112899_).intValue();
+      int j = SectionPos.blockToSectionCoord(p_112943_.getBigZ()).subtract(this.f_112900_).intValue();
+      return this.f_112905_[(int) i][(int) j].m_200453_(p_112943_).m_60819_();
    }
 
    public float m_7717_(Direction p_112940_, boolean p_112941_) {
@@ -50,8 +_,8 @@
 
    @Nullable
    public BlockEntity m_7702_(BlockPos p_112945_) {
-      int i = SectionPos.m_123171_(p_112945_.m_123341_()) - this.f_112899_;
-      int j = SectionPos.m_123171_(p_112945_.m_123343_()) - this.f_112900_;
+      int i = SectionPos.blockToSectionCoord(p_112945_.getBigX()).subtract(this.f_112899_).intValue();
+      int j = SectionPos.blockToSectionCoord(p_112945_.getBigZ()).subtract(this.f_112900_).intValue();
       return this.f_112905_[i][j].m_200451_(p_112945_);
    }
 
@@ -59,11 +_,11 @@
       return this.f_112908_.m_6171_(p_112937_, p_112938_);
    }
 
-   public int m_141937_() {
-      return this.f_112908_.m_141937_();
+   public long getMinBuildHeight() {
+      return this.f_112908_.getMinBuildHeight();
    }
 
-   public int m_141928_() {
-      return this.f_112908_.m_141928_();
+   public long getHeight() {
+      return this.f_112908_.getHeight();
    }
 }
