--- a/net/minecraft/client/renderer/chunk/VisGraph.java
+++ b/net/minecraft/client/renderer/chunk/VisGraph.java
@@ -35,7 +_,7 @@
          for(int i1 = 0; i1 < 16; ++i1) {
             for(int j1 = 0; j1 < 16; ++j1) {
                if (l == 0 || l == 15 || i1 == 0 || i1 == 15 || j1 == 0 || j1 == 15) {
-                  p_112974_[k++] = m_112961_(l, i1, j1);
+                  p_112974_[k++] = (int) getIndex(l, i1, j1);
                }
             }
          }
@@ -45,15 +_,15 @@
    private int f_112955_ = 4096;
 
    public void m_112971_(BlockPos p_112972_) {
-      this.f_112953_.set(m_112975_(p_112972_), true);
+      this.f_112953_.set((int) getIndex(p_112972_), true);
       --this.f_112955_;
    }
 
-   private static int m_112975_(BlockPos p_112976_) {
-      return m_112961_(p_112976_.m_123341_() & 15, p_112976_.m_123342_() & 15, p_112976_.m_123343_() & 15);
+   private static long getIndex(BlockPos p_112976_) {
+      return getIndex(p_112976_.getX() & 15, p_112976_.getY() & 15, p_112976_.getZ() & 15);
    }
 
-   private static int m_112961_(int p_112962_, int p_112963_, int p_112964_) {
+   private static long getIndex(long p_112962_, long p_112963_, long p_112964_) {
       return p_112962_ << 0 | p_112963_ << 8 | p_112964_ << 4;
    }
 
