--- a/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
@@ -93,6 +_,6 @@
    }
 
    protected void m_6372_(BaseCommandBlock p_98384_) {
-      this.f_96541_.m_91403_().m_104955_(new ServerboundSetCommandBlockPacket(BlockPos.m_274446_(p_98384_.m_6607_()), this.f_97646_.m_94155_(), this.f_98378_, p_98384_.m_45440_(), this.f_98379_, this.f_98380_));
+      this.f_96541_.m_91403_().m_104955_(new ServerboundSetCommandBlockPacket(BlockPos.containing(p_98384_.m_6607_()), this.f_97646_.m_94155_(), this.f_98378_, p_98384_.m_45440_(), this.f_98379_, this.f_98380_));
    }
 }
