--- a/net/minecraft/client/gui/screens/inventory/SmithingScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/SmithingScreen.java
@@ -53,7 +_,7 @@
    }
 
    protected void m_5653_() {
-      this.f_266031_ = new ArmorStand(this.f_96541_.f_91073_, 0.0D, 0.0D, 0.0D);
+      this.f_266031_ = new ArmorStand(this.f_96541_.f_91073_, me.alphamode.mclong.math.BigDecimal.ZERO, me.alphamode.mclong.math.BigDecimal.ZERO, me.alphamode.mclong.math.BigDecimal.ZERO);
       this.f_266031_.m_31678_(true);
       this.f_266031_.m_31675_(true);
       this.f_266031_.f_20883_ = 210.0F;
