--- a/net/minecraft/client/gui/screens/inventory/StructureBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/StructureBlockEditScreen.java
@@ -184,15 +_,15 @@
       BlockPos blockpos = this.f_99391_.m_59902_();
       this.f_99358_ = new EditBox(this.f_96547_, this.f_96543_ / 2 - 152, 80, 80, 20, Component.m_237115_("structure_block.position.x"));
       this.f_99358_.m_94199_(15);
-      this.f_99358_.m_94144_(Integer.toString(blockpos.m_123341_()));
+      this.f_99358_.m_94144_(java.lang.Long.toString(blockpos.getX()));
       this.m_7787_(this.f_99358_);
       this.f_99359_ = new EditBox(this.f_96547_, this.f_96543_ / 2 - 72, 80, 80, 20, Component.m_237115_("structure_block.position.y"));
       this.f_99359_.m_94199_(15);
-      this.f_99359_.m_94144_(Integer.toString(blockpos.m_123342_()));
+      this.f_99359_.m_94144_(java.lang.Long.toString(blockpos.getY()));
       this.m_7787_(this.f_99359_);
       this.f_99360_ = new EditBox(this.f_96547_, this.f_96543_ / 2 + 8, 80, 80, 20, Component.m_237115_("structure_block.position.z"));
       this.f_99360_.m_94199_(15);
-      this.f_99360_.m_94144_(Integer.toString(blockpos.m_123343_()));
+      this.f_99360_.m_94144_(java.lang.Long.toString(blockpos.getZ()));
       this.m_7787_(this.f_99360_);
       Vec3i vec3i = this.f_99391_.m_155805_();
       this.f_99361_ = new EditBox(this.f_96547_, this.f_96543_ / 2 - 152, 120, 80, 20, Component.m_237115_("structure_block.size.x"));
