--- a/net/minecraft/world/level/levelgen/structure/pieces/PiecesContainer.java
+++ b/net/minecraft/world/level/levelgen/structure/pieces/PiecesContainer.java
@@ -31,7 +_,7 @@
 
    public boolean m_192751_(BlockPos p_192752_) {
       for(StructurePiece structurepiece : this.f_192741_) {
-         if (structurepiece.m_73547_().m_71051_(p_192752_)) {
+         if (structurepiece.m_73547_().isInside(p_192752_)) {
             return true;
          }
       }
