--- a/net/minecraft/world/level/levelgen/structure/pieces/StructurePiecesBuilder.java
+++ b/net/minecraft/world/level/levelgen/structure/pieces/StructurePiecesBuilder.java
@@ -22,41 +_,41 @@
 
    /** @deprecated */
    @Deprecated
-   public void m_192781_(int p_192782_) {
+   public void offsetPiecesVertically(long p_192782_) {
       for(StructurePiece structurepiece : this.f_192778_) {
-         structurepiece.m_6324_(0, p_192782_, 0);
+         structurepiece.move(0, p_192782_, 0);
       }
 
    }
 
    /** @deprecated */
    @Deprecated
-   public int m_226965_(int p_226966_, int p_226967_, RandomSource p_226968_, int p_226969_) {
-      int i = p_226966_ - p_226969_;
+   public long moveBelowSeaLevel(long p_226966_, long p_226967_, RandomSource p_226968_, int p_226969_) {
+      long i = p_226966_ - p_226969_;
       BoundingBox boundingbox = this.m_192798_();
-      int j = boundingbox.m_71057_() + p_226967_ + 1;
+      long j = boundingbox.getYSpan() + p_226967_ + 1;
       if (j < i) {
-         j += p_226968_.m_188503_(i - j);
+         j += p_226968_.nextLong(i - j);
       }
 
-      int k = j - boundingbox.m_162400_();
-      this.m_192781_(k);
+      long k = j - boundingbox.maxY();
+      this.offsetPiecesVertically(k);
       return k;
    }
 
    /** @deprecated */
    public void m_226970_(RandomSource p_226971_, int p_226972_, int p_226973_) {
       BoundingBox boundingbox = this.m_192798_();
-      int i = p_226973_ - p_226972_ + 1 - boundingbox.m_71057_();
-      int j;
+      long i = p_226973_ - p_226972_ + 1 - boundingbox.getYSpan();
+      long j;
       if (i > 1) {
-         j = p_226972_ + p_226971_.m_188503_(i);
+         j = p_226972_ + p_226971_.nextLong(i);
       } else {
          j = p_226972_;
       }
 
-      int k = j - boundingbox.m_162396_();
-      this.m_192781_(k);
+      long k = j - boundingbox.minY();
+      this.offsetPiecesVertically(k);
    }
 
    public PiecesContainer m_192780_() {
