--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -636,26 +_,26 @@
 
    public void m_183388_(ClientboundLevelChunkWithLightPacket p_194241_) {
       PacketUtils.m_131363_(p_194241_, this, this.f_104888_);
-      this.m_194198_(p_194241_.m_195717_(), p_194241_.m_195718_(), p_194241_.m_195719_());
-      this.m_194202_(p_194241_.m_195717_(), p_194241_.m_195718_(), p_194241_.m_195720_());
-   }
-
-   private void m_194198_(int p_194199_, int p_194200_, ClientboundLevelChunkPacketData p_194201_) {
-      this.f_104889_.m_7726_().m_194116_(p_194199_, p_194200_, p_194201_.m_195656_(), p_194201_.m_195678_(), p_194201_.m_195657_(p_194199_, p_194200_));
-   }
-
-   private void m_194202_(int p_194203_, int p_194204_, ClientboundLightUpdatePacketData p_194205_) {
+      this.updateLevelChunk(p_194241_.getX(), p_194241_.getZ(), p_194241_.m_195719_());
+      this.queueLightUpdate(p_194241_.getX(), p_194241_.getZ(), p_194241_.m_195720_());
+   }
+
+   private void updateLevelChunk(long p_194199_, long p_194200_, ClientboundLevelChunkPacketData p_194201_) {
+      this.f_104889_.m_7726_().replaceWithPacketData(p_194199_, p_194200_, p_194201_.m_195656_(), p_194201_.m_195678_(), p_194201_.getBlockEntitiesTagsConsumer(p_194199_, p_194200_));
+   }
+
+   private void queueLightUpdate(long p_194203_, long p_194204_, ClientboundLightUpdatePacketData p_194205_) {
       this.f_104889_.m_194171_(() -> {
-         this.m_194248_(p_194203_, p_194204_, p_194205_);
-         LevelChunk levelchunk = this.f_104889_.m_7726_().m_62227_(p_194203_, p_194204_, false);
+         this.applyLightData(p_194203_, p_194204_, p_194205_);
+         LevelChunk levelchunk = this.f_104889_.m_7726_().getChunk(p_194203_, p_194204_, false);
          if (levelchunk != null) {
-            this.m_194212_(levelchunk, p_194203_, p_194204_);
+            this.enableChunkLight(levelchunk, p_194203_, p_194204_);
          }
 
       });
    }
 
-   private void m_194212_(LevelChunk p_194213_, int p_194214_, int p_194215_) {
+   private void enableChunkLight(LevelChunk p_194213_, long p_194214_, long p_194215_) {
       LevelLightEngine levellightengine = this.f_104889_.m_7726_().m_7827_();
       LevelChunkSection[] alevelchunksection = p_194213_.m_7103_();
       ChunkPos chunkpos = p_194213_.m_7697_();
@@ -665,18 +_,18 @@
          LevelChunkSection levelchunksection = alevelchunksection[i];
          int j = this.f_104889_.m_151568_(i);
          levellightengine.m_6191_(SectionPos.m_123196_(chunkpos, j), levelchunksection.m_188008_());
-         this.f_104889_.m_104793_(p_194214_, j, p_194215_);
+         this.f_104889_.setSectionDirtyWithNeighbors(p_194214_, j, p_194215_);
       }
 
-      this.f_104889_.m_197405_(p_194214_, p_194215_);
+      this.f_104889_.setLightReady(p_194214_, p_194215_);
    }
 
    public void m_5729_(ClientboundForgetLevelChunkPacket p_105014_) {
       PacketUtils.m_131363_(p_105014_, this, this.f_104888_);
-      int i = p_105014_.m_132149_();
-      int j = p_105014_.m_132152_();
+      long i = p_105014_.getX();
+      long j = p_105014_.getZ();
       ClientChunkCache clientchunkcache = this.f_104889_.m_7726_();
-      clientchunkcache.m_104455_(i, j);
+      clientchunkcache.drop(i, j);
       this.m_194252_(p_105014_);
    }
 
@@ -685,11 +_,11 @@
          LevelLightEngine levellightengine = this.f_104889_.m_5518_();
 
          for(int i = this.f_104889_.m_151560_(); i < this.f_104889_.m_151561_(); ++i) {
-            levellightengine.m_6191_(SectionPos.m_123173_(p_194253_.m_132149_(), i, p_194253_.m_132152_()), true);
+            levellightengine.m_6191_(SectionPos.of(p_194253_.getX(), i, p_194253_.getZ()), true);
          }
 
-         levellightengine.m_6460_(new ChunkPos(p_194253_.m_132149_(), p_194253_.m_132152_()), false);
-         this.f_104889_.m_197405_(p_194253_.m_132149_(), p_194253_.m_132152_());
+         levellightengine.m_6460_(new ChunkPos(p_194253_.getX(), p_194253_.getZ()), false);
+         this.f_104889_.setLightReady(p_194253_.getX(), p_194253_.getZ());
       });
    }
 
@@ -2046,25 +_,25 @@
 
    public void m_183514_(ClientboundLightUpdatePacket p_194243_) {
       PacketUtils.m_131363_(p_194243_, this, this.f_104888_);
-      int i = p_194243_.m_132349_();
-      int j = p_194243_.m_132352_();
+      long i = p_194243_.getX();
+      long j = p_194243_.getZ();
       ClientboundLightUpdatePacketData clientboundlightupdatepacketdata = p_194243_.m_195722_();
       this.f_104889_.m_194171_(() -> {
-         this.m_194248_(i, j, clientboundlightupdatepacketdata);
+         this.applyLightData(i, j, clientboundlightupdatepacketdata);
       });
    }
 
-   private void m_194248_(int p_194249_, int p_194250_, ClientboundLightUpdatePacketData p_194251_) {
+   private void applyLightData(long p_194249_, long p_194250_, ClientboundLightUpdatePacketData p_194251_) {
       LevelLightEngine levellightengine = this.f_104889_.m_7726_().m_7827_();
       BitSet bitset = p_194251_.m_195740_();
       BitSet bitset1 = p_194251_.m_195751_();
       Iterator<byte[]> iterator = p_194251_.m_195754_().iterator();
-      this.m_171734_(p_194249_, p_194250_, levellightengine, LightLayer.SKY, bitset, bitset1, iterator, p_194251_.m_195760_());
+      this.readSectionList(p_194249_, p_194250_, levellightengine, LightLayer.SKY, bitset, bitset1, iterator, p_194251_.m_195760_());
       BitSet bitset2 = p_194251_.m_195757_();
       BitSet bitset3 = p_194251_.m_195758_();
       Iterator<byte[]> iterator1 = p_194251_.m_195759_().iterator();
-      this.m_171734_(p_194249_, p_194250_, levellightengine, LightLayer.BLOCK, bitset2, bitset3, iterator1, p_194251_.m_195760_());
-      this.f_104889_.m_197405_(p_194249_, p_194250_);
+      this.readSectionList(p_194249_, p_194250_, levellightengine, LightLayer.BLOCK, bitset2, bitset3, iterator1, p_194251_.m_195760_());
+      this.f_104889_.setLightReady(p_194249_, p_194250_);
    }
 
    public void m_7330_(ClientboundMerchantOffersPacket p_105034_) {
@@ -2095,7 +_,7 @@
 
    public void m_8065_(ClientboundSetChunkCacheCenterPacket p_105080_) {
       PacketUtils.m_131363_(p_105080_, this, this.f_104888_);
-      this.f_104889_.m_7726_().m_104459_(p_105080_.m_133094_(), p_105080_.m_133097_());
+      this.f_104889_.m_7726_().updateViewCenter(p_105080_.getX(), p_105080_.getZ());
    }
 
    public void m_214108_(ClientboundBlockChangedAckPacket p_233698_) {
@@ -2103,14 +_,14 @@
       this.f_104889_.m_233651_(p_233698_.f_237578_());
    }
 
-   private void m_171734_(int p_171735_, int p_171736_, LevelLightEngine p_171737_, LightLayer p_171738_, BitSet p_171739_, BitSet p_171740_, Iterator<byte[]> p_171741_, boolean p_171742_) {
+   private void readSectionList(long p_171735_, long p_171736_, LevelLightEngine p_171737_, LightLayer p_171738_, BitSet p_171739_, BitSet p_171740_, Iterator<byte[]> p_171741_, boolean p_171742_) {
       for(int i = 0; i < p_171737_.m_164446_(); ++i) {
          int j = p_171737_.m_164447_() + i;
          boolean flag = p_171739_.get(i);
          boolean flag1 = p_171740_.get(i);
          if (flag || flag1) {
-            p_171737_.m_5687_(p_171738_, SectionPos.m_123173_(p_171735_, j, p_171736_), flag ? new DataLayer((byte[])p_171741_.next().clone()) : new DataLayer(), p_171742_);
-            this.f_104889_.m_104793_(p_171735_, j, p_171736_);
+            p_171737_.m_5687_(p_171738_, SectionPos.of(p_171735_, j, p_171736_), flag ? new DataLayer((byte[])p_171741_.next().clone()) : new DataLayer(), p_171742_);
+            this.f_104889_.setSectionDirtyWithNeighbors(p_171735_, j, p_171736_);
          }
       }
 
