--- a/net/minecraft/client/multiplayer/ClientLevel.java
+++ b/net/minecraft/client/multiplayer/ClientLevel.java
@@ -140,7 +_,7 @@
          this.m_7731_(p_233648_, p_233649_, 19);
          Player player = this.f_104565_.f_91074_;
          if (this == player.m_9236_() && player.m_20039_(p_233648_, p_233649_)) {
-            player.m_20248_(p_233650_.f_82479_, p_233650_.f_82480_, p_233650_.f_82481_);
+            player.absMoveTo(p_233650_.getX(), p_233650_.getY(), p_233650_.getZ());
          }
       }
 
@@ -256,7 +_,7 @@
    }
 
    public boolean m_183599_(Entity p_194185_) {
-      return p_194185_.m_146902_().m_45594_(this.f_104565_.f_91074_.m_146902_()) <= this.f_194123_;
+      return p_194185_.m_146902_().getChessboardDistance(this.f_104565_.f_91074_.m_146902_()) <= this.f_194123_;
    }
 
    public void m_104639_(Entity p_104640_) {
@@ -299,7 +_,7 @@
 
    public void m_171649_(ChunkPos p_171650_) {
       this.f_104558_.forEach((p_194154_, p_194155_) -> {
-         p_194155_.m_92655_(p_171650_.f_45578_, p_171650_.f_45579_);
+         p_194155_.invalidateForChunk(p_171650_.x(), p_171650_.z());
       });
       this.f_171631_.m_157651_(p_171650_);
    }
@@ -310,7 +_,7 @@
       });
    }
 
-   public boolean m_7232_(int p_104737_, int p_104738_) {
+   public boolean hasChunk(long p_104737_, long p_104738_) {
       return true;
    }
 
@@ -349,15 +_,15 @@
       this.f_104561_.m_104910_().m_129507_(Component.m_237115_("multiplayer.status.quitting"));
    }
 
-   public void m_104784_(int p_104785_, int p_104786_, int p_104787_) {
+   public void animateTick(long p_104785_, long p_104786_, long p_104787_) {
       int i = 32;
       RandomSource randomsource = RandomSource.m_216327_();
       Block block = this.m_194187_();
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(int j = 0; j < 667; ++j) {
-         this.m_233612_(p_104785_, p_104786_, p_104787_, 16, randomsource, block, blockpos$mutableblockpos);
-         this.m_233612_(p_104785_, p_104786_, p_104787_, 32, randomsource, block, blockpos$mutableblockpos);
+         this.doAnimateTick(p_104785_, p_104786_, p_104787_, 16, randomsource, block, blockpos$mutableblockpos);
+         this.doAnimateTick(p_104785_, p_104786_, p_104787_, 32, randomsource, block, blockpos$mutableblockpos);
       }
 
    }
@@ -376,11 +_,11 @@
       return null;
    }
 
-   public void m_233612_(int p_233613_, int p_233614_, int p_233615_, int p_233616_, RandomSource p_233617_, @Nullable Block p_233618_, BlockPos.MutableBlockPos p_233619_) {
-      int i = p_233613_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
-      int j = p_233614_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
-      int k = p_233615_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
-      p_233619_.m_122178_(i, j, k);
+   public void doAnimateTick(long p_233613_, long p_233614_, long p_233615_, int p_233616_, RandomSource p_233617_, @Nullable Block p_233618_, BlockPos.MutableBlockPos p_233619_) {
+      long i = p_233613_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
+      long j = p_233614_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
+      long k = p_233615_ + this.f_46441_.m_188503_(p_233616_) - this.f_46441_.m_188503_(p_233616_);
+      p_233619_.set(i, j, k);
       BlockState blockstate = this.m_8055_(p_233619_);
       blockstate.m_60734_().m_214162_(blockstate, this, p_233619_, p_233617_);
       FluidState fluidstate = this.m_6425_(p_233619_);
@@ -389,7 +_,7 @@
          ParticleOptions particleoptions = fluidstate.m_76189_();
          if (particleoptions != null && this.f_46441_.m_188503_(10) == 0) {
             boolean flag = blockstate.m_60783_(this, p_233619_, Direction.DOWN);
-            BlockPos blockpos = p_233619_.m_7495_();
+            BlockPos blockpos = p_233619_.below();
             this.m_104689_(blockpos, this.m_8055_(blockpos), particleoptions, flag);
          }
       }
@@ -401,7 +_,7 @@
       if (!blockstate.m_60838_(this, p_233619_)) {
          this.m_204166_(p_233619_).m_203334_().m_47562_().ifPresent((p_264703_) -> {
             if (p_264703_.m_220527_(this.f_46441_)) {
-               this.m_7106_(p_264703_.m_47419_(), (double)p_233619_.m_123341_() + this.f_46441_.m_188500_(), (double)p_233619_.m_123342_() + this.f_46441_.m_188500_(), (double)p_233619_.m_123343_() + this.f_46441_.m_188500_(), 0.0D, 0.0D, 0.0D);
+               this.m_7106_(p_264703_.m_47419_(), (double)p_233619_.getX() + this.f_46441_.m_188500_(), (double)p_233619_.getY() + this.f_46441_.m_188500_(), (double)p_233619_.getZ() + this.f_46441_.m_188500_(), 0.0D, 0.0D, 0.0D);
             }
 
          });
@@ -415,19 +_,19 @@
          double d0 = voxelshape.m_83297_(Direction.Axis.Y);
          if (d0 < 1.0D) {
             if (p_104693_) {
-               this.m_104592_((double)p_104690_.m_123341_(), (double)(p_104690_.m_123341_() + 1), (double)p_104690_.m_123343_(), (double)(p_104690_.m_123343_() + 1), (double)(p_104690_.m_123342_() + 1) - 0.05D, p_104692_);
+               this.m_104592_((double)p_104690_.getX(), (double)(p_104690_.getX() + 1), (double)p_104690_.getZ(), (double)(p_104690_.getZ() + 1), (double)(p_104690_.getY() + 1) - 0.05D, p_104692_);
             }
          } else if (!p_104691_.m_204336_(BlockTags.f_13049_)) {
             double d1 = voxelshape.m_83288_(Direction.Axis.Y);
             if (d1 > 0.0D) {
-               this.m_104694_(p_104690_, p_104692_, voxelshape, (double)p_104690_.m_123342_() + d1 - 0.05D);
+               this.m_104694_(p_104690_, p_104692_, voxelshape, (double)p_104690_.getY() + d1 - 0.05D);
             } else {
-               BlockPos blockpos = p_104690_.m_7495_();
+               BlockPos blockpos = p_104690_.below();
                BlockState blockstate = this.m_8055_(blockpos);
                VoxelShape voxelshape1 = blockstate.m_60812_(this, blockpos);
                double d2 = voxelshape1.m_83297_(Direction.Axis.Y);
                if (d2 < 1.0D && blockstate.m_60819_().m_76178_()) {
-                  this.m_104694_(p_104690_, p_104692_, voxelshape, (double)p_104690_.m_123342_() - 0.05D);
+                  this.m_104694_(p_104690_, p_104692_, voxelshape, (double)p_104690_.getY() - 0.05D);
                }
             }
          }
@@ -436,7 +_,7 @@
    }
 
    private void m_104694_(BlockPos p_104695_, ParticleOptions p_104696_, VoxelShape p_104697_, double p_104698_) {
-      this.m_104592_((double)p_104695_.m_123341_() + p_104697_.m_83288_(Direction.Axis.X), (double)p_104695_.m_123341_() + p_104697_.m_83297_(Direction.Axis.X), (double)p_104695_.m_123343_() + p_104697_.m_83288_(Direction.Axis.Z), (double)p_104695_.m_123343_() + p_104697_.m_83297_(Direction.Axis.Z), p_104698_, p_104696_);
+      this.m_104592_((double)p_104695_.getX() + p_104697_.m_83288_(Direction.Axis.X), (double)p_104695_.getX() + p_104697_.m_83297_(Direction.Axis.X), (double)p_104695_.getZ() + p_104697_.m_83288_(Direction.Axis.Z), (double)p_104695_.getZ() + p_104697_.m_83297_(Direction.Axis.Z), p_104698_, p_104696_);
    }
 
    private void m_104592_(double p_104593_, double p_104594_, double p_104595_, double p_104596_, double p_104597_, ParticleOptions p_104598_) {
@@ -540,8 +_,8 @@
       this.f_104562_.m_109721_(p_104759_, p_104760_, p_104761_);
    }
 
-   public void m_104793_(int p_104794_, int p_104795_, int p_104796_) {
-      this.f_104562_.m_109490_(p_104794_, p_104795_, p_104796_);
+   public void setSectionDirtyWithNeighbors(long p_104794_, long p_104795_, long p_104796_) {
+      this.f_104562_.setSectionDirtyWithNeighbors(p_104794_, p_104795_, p_104796_);
    }
 
    public void m_6801_(int p_104634_, BlockPos p_104635_, int p_104636_) {
@@ -586,7 +_,7 @@
       return this.f_104566_;
    }
 
-   public Holder<Biome> m_203675_(int p_205516_, int p_205517_, int p_205518_) {
+   public Holder<Biome> getUncachedNoiseBiome(long p_205516_, long p_205517_, long p_205518_) {
       return this.m_9598_().m_175515_(Registries.f_256952_).m_246971_(Biomes.f_48202_);
    }
 
@@ -605,13 +_,13 @@
       Vec3 vec3 = p_171661_.m_82492_(2.0D, 2.0D, 2.0D).m_82490_(0.25D);
       BiomeManager biomemanager = this.m_7062_();
       Vec3 vec31 = CubicSampler.m_130038_(vec3, (p_194161_, p_194162_, p_194163_) -> {
-         return Vec3.m_82501_(biomemanager.m_204210_(p_194161_, p_194162_, p_194163_).m_203334_().m_47463_());
+         return Vec3.m_82501_(biomemanager.getNoiseBiomeAtQuart(p_194161_, p_194162_, p_194163_).m_203334_().m_47463_());
       });
       float f1 = Mth.m_14089_(f * ((float)Math.PI * 2F)) * 2.0F + 0.5F;
       f1 = Mth.m_14036_(f1, 0.0F, 1.0F);
-      float f2 = (float)vec31.f_82479_ * f1;
-      float f3 = (float)vec31.f_82480_ * f1;
-      float f4 = (float)vec31.f_82481_ * f1;
+      float f2 = (float)vec31.m_7096_() * f1;
+      float f3 = (float)vec31.m_7098_() * f1;
+      float f4 = (float)vec31.m_7094_() * f1;
       float f5 = this.m_46722_(p_171662_);
       if (f5 > 0.0F) {
          float f6 = (f2 * 0.3F + f3 * 0.59F + f4 * 0.11F) * 0.6F;
@@ -722,18 +_,19 @@
    public int m_104762_(BlockPos p_104763_, ColorResolver p_104764_) {
       int i = Minecraft.m_91087_().f_91066_.m_232121_().m_231551_();
       if (i == 0) {
-         return p_104764_.m_130045_(this.m_204166_(p_104763_).m_203334_(), (double)p_104763_.m_123341_(), (double)p_104763_.m_123343_());
+         return p_104764_.m_130045_(this.m_204166_(p_104763_).m_203334_(), (double)p_104763_.getX(), (double)p_104763_.getZ());
       } else {
          int j = (i * 2 + 1) * (i * 2 + 1);
          int k = 0;
          int l = 0;
          int i1 = 0;
-         Cursor3D cursor3d = new Cursor3D(p_104763_.m_123341_() - i, p_104763_.m_123342_(), p_104763_.m_123343_() - i, p_104763_.m_123341_() + i, p_104763_.m_123342_(), p_104763_.m_123343_() + i);
+         var blendRadius = java.math.BigInteger.valueOf(i);
+         Cursor3D cursor3d = new Cursor3D(p_104763_.getBigX().subtract(blendRadius), p_104763_.getBigY(), p_104763_.getBigZ().subtract(blendRadius), p_104763_.getBigX().add(blendRadius), p_104763_.getBigY(), p_104763_.getBigZ().add(blendRadius));
 
          int j1;
          for(BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(); cursor3d.m_122304_(); i1 += j1 & 255) {
-            blockpos$mutableblockpos.m_122178_(cursor3d.m_122305_(), cursor3d.m_122306_(), cursor3d.m_122307_());
-            j1 = p_104764_.m_130045_(this.m_204166_(blockpos$mutableblockpos).m_203334_(), (double)blockpos$mutableblockpos.m_123341_(), (double)blockpos$mutableblockpos.m_123343_());
+            blockpos$mutableblockpos.set(cursor3d.nextX(), cursor3d.nextY(), cursor3d.nextZ());
+            j1 = p_104764_.m_130045_(this.m_204166_(blockpos$mutableblockpos).m_203334_(), (double)blockpos$mutableblockpos.getX(), (double)blockpos$mutableblockpos.getZ());
             k += (j1 & 16711680) >> 16;
             l += (j1 & '\uff00') >> 8;
          }
@@ -794,9 +_,9 @@
       private final boolean f_104830_;
       private final GameRules f_104831_;
       private final boolean f_104832_;
-      private int f_104833_;
-      private int f_104834_;
-      private int f_104835_;
+      private long f_104833_;
+      private long f_104834_;
+      private long f_104835_;
       private float f_104836_;
       private long f_104837_;
       private long f_104838_;
@@ -811,15 +_,15 @@
          this.f_104831_ = new GameRules();
       }
 
-      public int m_6789_() {
+      public long getXSpawn() {
          return this.f_104833_;
       }
 
-      public int m_6527_() {
+      public long getYSpawn() {
          return this.f_104834_;
       }
 
-      public int m_6526_() {
+      public long getZSpawn() {
          return this.f_104835_;
       }
 
@@ -835,15 +_,15 @@
          return this.f_104838_;
       }
 
-      public void m_6395_(int p_104862_) {
+      public void setXSpawn(long p_104862_) {
          this.f_104833_ = p_104862_;
       }
 
-      public void m_6397_(int p_104869_) {
+      public void setYSpawn(long p_104869_) {
          this.f_104834_ = p_104869_;
       }
 
-      public void m_6400_(int p_104872_) {
+      public void setZSpawn(long p_104872_) {
          this.f_104835_ = p_104872_;
       }
 
@@ -860,9 +_,9 @@
       }
 
       public void m_7250_(BlockPos p_104854_, float p_104855_) {
-         this.f_104833_ = p_104854_.m_123341_();
-         this.f_104834_ = p_104854_.m_123342_();
-         this.f_104835_ = p_104854_.m_123343_();
+         this.f_104833_ = p_104854_.getX();
+         this.f_104834_ = p_104854_.getY();
+         this.f_104835_ = p_104854_.getZ();
          this.f_104836_ = p_104855_;
       }
 
@@ -907,7 +_,7 @@
       }
 
       public double m_171687_(LevelHeightAccessor p_171688_) {
-         return this.f_104832_ ? (double)p_171688_.m_141937_() : 63.0D;
+         return this.f_104832_ ? (double)p_171688_.getMinBuildHeight() : 63.0D;
       }
 
       public float m_205519_() {
