--- a/net/minecraft/client/multiplayer/ClientSuggestionProvider.java
+++ b/net/minecraft/client/multiplayer/ClientSuggestionProvider.java
@@ -119,7 +_,7 @@
       HitResult hitresult = this.f_105161_.f_91077_;
       if (hitresult != null && hitresult.m_6662_() == HitResult.Type.BLOCK) {
          BlockPos blockpos = ((BlockHitResult)hitresult).m_82425_();
-         return Collections.singleton(new SharedSuggestionProvider.TextCoordinates(m_105169_(blockpos.m_123341_()), m_105169_(blockpos.m_123342_()), m_105169_(blockpos.m_123343_())));
+         return Collections.singleton(new SharedSuggestionProvider.TextCoordinates(m_105167_(blockpos.getX()), m_105167_(blockpos.getY()), m_105167_(blockpos.getZ())));
       } else {
          return SharedSuggestionProvider.super.m_6265_();
       }
