--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -474,11 +_,11 @@
 
    public void m_7708_(ClientboundAddExperienceOrbPacket p_104960_) {
       PacketUtils.m_131363_(p_104960_, this, this.f_104888_);
-      double d0 = p_104960_.m_131527_();
-      double d1 = p_104960_.m_131528_();
-      double d2 = p_104960_.m_131529_();
+      me.alphamode.mclong.math.BigDecimal d0 = p_104960_.getX();
+      me.alphamode.mclong.math.BigDecimal d1 = p_104960_.getY();
+      me.alphamode.mclong.math.BigDecimal d2 = p_104960_.getZ();
       Entity entity = new ExperienceOrb(this.f_104889_, d0, d1, d2, p_104960_.m_131530_());
-      entity.m_217006_(d0, d1, d2);
+      entity.syncPacketPositionCodec(d0, d1, d2);
       entity.m_146922_(0.0F);
       entity.m_146926_(0.0F);
       entity.m_20234_(p_104960_.m_131524_());
@@ -508,16 +_,16 @@
       if (playerinfo == null) {
          f_104883_.warn("Server attempted to add player prior to sending player info (Player id {})", (Object)p_104966_.m_131606_());
       } else {
-         double d0 = p_104966_.m_131607_();
-         double d1 = p_104966_.m_131608_();
-         double d2 = p_104966_.m_131609_();
+         me.alphamode.mclong.math.BigDecimal d0 = p_104966_.getX();
+         me.alphamode.mclong.math.BigDecimal d1 = p_104966_.getY();
+         me.alphamode.mclong.math.BigDecimal d2 = p_104966_.getZ();
          float f = (float)(p_104966_.m_131610_() * 360) / 256.0F;
          float f1 = (float)(p_104966_.m_131611_() * 360) / 256.0F;
          int i = p_104966_.m_131603_();
          RemotePlayer remoteplayer = new RemotePlayer(this.f_104888_.f_91073_, playerinfo.m_105312_());
          remoteplayer.m_20234_(i);
-         remoteplayer.m_217006_(d0, d1, d2);
-         remoteplayer.m_19890_(d0, d1, d2, f, f1);
+         remoteplayer.syncPacketPositionCodec(d0, d1, d2);
+         remoteplayer.absMoveTo(d0, d1, d2, f, f1);
          remoteplayer.m_146867_();
          this.f_104889_.m_104630_(i, remoteplayer);
       }
@@ -527,14 +_,14 @@
       PacketUtils.m_131363_(p_105124_, this, this.f_104888_);
       Entity entity = this.f_104889_.m_6815_(p_105124_.m_133545_());
       if (entity != null) {
-         double d0 = p_105124_.m_133548_();
-         double d1 = p_105124_.m_133549_();
-         double d2 = p_105124_.m_133550_();
-         entity.m_217006_(d0, d1, d2);
+         me.alphamode.mclong.math.BigDecimal d0 = p_105124_.getX();
+         me.alphamode.mclong.math.BigDecimal d1 = p_105124_.getY();
+         me.alphamode.mclong.math.BigDecimal d2 = p_105124_.getZ();
+         entity.syncPacketPositionCodec(d0, d1, d2);
          if (!entity.m_6109_()) {
             float f = (float)(p_105124_.m_133551_() * 360) / 256.0F;
             float f1 = (float)(p_105124_.m_133552_() * 360) / 256.0F;
-            entity.m_6453_(d0, d1, d2, f, f1, 3, true);
+            entity.lerpTo(d0, d1, d2, f, f1, 3, true);
             entity.m_6853_(p_105124_.m_133553_());
          }
 
@@ -560,11 +_,11 @@
                vecdeltacodec.m_238033_(vec3);
                float f = p_105036_.m_132533_() ? (float)(p_105036_.m_132531_() * 360) / 256.0F : entity.m_146908_();
                float f1 = p_105036_.m_132533_() ? (float)(p_105036_.m_132532_() * 360) / 256.0F : entity.m_146909_();
-               entity.m_6453_(vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), f, f1, 3, false);
+               entity.lerpTo(vec3.getX(), vec3.getY(), vec3.getZ(), f, f1, 3, false);
             } else if (p_105036_.m_132533_()) {
                float f2 = (float)(p_105036_.m_132531_() * 360) / 256.0F;
                float f3 = (float)(p_105036_.m_132532_() * 360) / 256.0F;
-               entity.m_6453_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), f2, f3, 3, false);
+               entity.lerpTo(entity.getBigX(), entity.getBigY(), entity.getBigZ(), f2, f3, 3, false);
             }
 
             entity.m_6853_(p_105036_.m_132535_());
@@ -597,48 +_,48 @@
       boolean flag1 = p_105056_.m_132826_().contains(RelativeMovement.Y);
       boolean flag2 = p_105056_.m_132826_().contains(RelativeMovement.Z);
       double d0;
-      double d1;
+      me.alphamode.mclong.math.BigDecimal d1;
       if (flag) {
          d0 = vec3.m_7096_();
-         d1 = player.m_20185_() + p_105056_.m_132818_();
-         player.f_19790_ += p_105056_.m_132818_();
-         player.f_19854_ += p_105056_.m_132818_();
+         d1 = player.getBigX().add(p_105056_.getX());
+         player.f_19790_ = player.f_19790_.add(p_105056_.getX());
+         player.f_19854_ = player.f_19854_.add(p_105056_.getX());
       } else {
          d0 = 0.0D;
-         d1 = p_105056_.m_132818_();
+         d1 = p_105056_.getX();
          player.f_19790_ = d1;
          player.f_19854_ = d1;
       }
 
       double d2;
-      double d3;
+      me.alphamode.mclong.math.BigDecimal d3;
       if (flag1) {
          d2 = vec3.m_7098_();
-         d3 = player.m_20186_() + p_105056_.m_132821_();
-         player.f_19791_ += p_105056_.m_132821_();
-         player.f_19855_ += p_105056_.m_132821_();
+         d3 = player.getBigY().add(p_105056_.getY());
+         player.f_19791_ = player.f_19791_.add(p_105056_.getY());
+         player.f_19855_ = player.f_19855_.add(p_105056_.getY());
       } else {
          d2 = 0.0D;
-         d3 = p_105056_.m_132821_();
+         d3 = p_105056_.getY();
          player.f_19791_ = d3;
          player.f_19855_ = d3;
       }
 
       double d4;
-      double d5;
+      me.alphamode.mclong.math.BigDecimal d5;
       if (flag2) {
          d4 = vec3.m_7094_();
-         d5 = player.m_20189_() + p_105056_.m_132822_();
-         player.f_19792_ += p_105056_.m_132822_();
-         player.f_19856_ += p_105056_.m_132822_();
+         d5 = player.getBigZ().add(p_105056_.getZ());
+         player.f_19792_ = player.f_19792_.add(p_105056_.getZ());
+         player.f_19856_ = player.f_19856_.add(p_105056_.getZ());
       } else {
          d4 = 0.0D;
-         d5 = p_105056_.m_132822_();
+         d5 = p_105056_.getZ();
          player.f_19792_ = d5;
          player.f_19856_ = d5;
       }
 
-      player.m_6034_(d1, d3, d5);
+      player.setPos(d1, d3, d5);
       player.m_20334_(d0, d2, d4);
       float f = p_105056_.m_132823_();
       float f1 = p_105056_.m_132824_();
@@ -659,7 +_,7 @@
       }
 
       this.f_104885_.m_129512_(new ServerboundAcceptTeleportationPacket(p_105056_.m_132825_()));
-      this.f_104885_.m_129512_(new ServerboundMovePlayerPacket.PosRot(player.m_20185_(), player.m_20186_(), player.m_20189_(), player.m_146908_(), player.m_146909_(), false));
+      this.f_104885_.m_129512_(new ServerboundMovePlayerPacket.PosRot(player.getBigX(), player.getBigY(), player.getBigZ(), player.m_146908_(), player.m_146909_(), false));
    }
 
    public void m_5771_(ClientboundSectionBlocksUpdatePacket p_105070_) {
@@ -671,15 +_,15 @@
 
    public void m_183388_(ClientboundLevelChunkWithLightPacket p_194241_) {
       PacketUtils.m_131363_(p_194241_, this, this.f_104888_);
-      int i = p_194241_.m_195717_();
-      int j = p_194241_.m_195718_();
-      this.m_194198_(i, j, p_194241_.m_195719_());
+      me.alphamode.mclong.math.BigInteger i = p_194241_.getX();
+      me.alphamode.mclong.math.BigInteger j = p_194241_.getZ();
+      this.updateLevelChunk(i, j, p_194241_.m_195719_());
       ClientboundLightUpdatePacketData clientboundlightupdatepacketdata = p_194241_.m_195720_();
       this.f_104889_.m_194171_(() -> {
-         this.m_194248_(i, j, clientboundlightupdatepacketdata);
-         LevelChunk levelchunk = this.f_104889_.m_7726_().m_62227_(i, j, false);
+         this.applyLightData(i, j, clientboundlightupdatepacketdata);
+         LevelChunk levelchunk = this.f_104889_.m_7726_().getChunk(i, j, false);
          if (levelchunk != null) {
-            this.m_194212_(levelchunk, i, j);
+            this.enableChunkLight(levelchunk, i, j);
          }
 
       });
@@ -689,18 +_,18 @@
       PacketUtils.m_131363_(p_275437_, this, this.f_104888_);
 
       for(ClientboundChunksBiomesPacket.ChunkBiomeData clientboundchunksbiomespacket$chunkbiomedata : p_275437_.f_273816_()) {
-         this.f_104889_.m_7726_().m_274444_(clientboundchunksbiomespacket$chunkbiomedata.f_273927_().f_45578_, clientboundchunksbiomespacket$chunkbiomedata.f_273927_().f_45579_, clientboundchunksbiomespacket$chunkbiomedata.m_274543_());
+         this.f_104889_.m_7726_().replaceBiomes(clientboundchunksbiomespacket$chunkbiomedata.f_273927_().getX(), clientboundchunksbiomespacket$chunkbiomedata.f_273927_().getZ(), clientboundchunksbiomespacket$chunkbiomedata.m_274543_());
       }
 
       for(ClientboundChunksBiomesPacket.ChunkBiomeData clientboundchunksbiomespacket$chunkbiomedata1 : p_275437_.f_273816_()) {
-         this.f_104889_.m_171649_(new ChunkPos(clientboundchunksbiomespacket$chunkbiomedata1.f_273927_().f_45578_, clientboundchunksbiomespacket$chunkbiomedata1.f_273927_().f_45579_));
+         this.f_104889_.m_171649_(new ChunkPos(clientboundchunksbiomespacket$chunkbiomedata1.f_273927_().getX(), clientboundchunksbiomespacket$chunkbiomedata1.f_273927_().getZ()));
       }
 
       for(ClientboundChunksBiomesPacket.ChunkBiomeData clientboundchunksbiomespacket$chunkbiomedata2 : p_275437_.f_273816_()) {
          for(int i = -1; i <= 1; ++i) {
             for(int j = -1; j <= 1; ++j) {
-               for(int k = this.f_104889_.m_151560_(); k < this.f_104889_.m_151561_(); ++k) {
-                  this.f_104888_.f_91060_.m_109770_(clientboundchunksbiomespacket$chunkbiomedata2.f_273927_().f_45578_ + i, k, clientboundchunksbiomespacket$chunkbiomedata2.f_273927_().f_45579_ + j);
+               for(long k = this.f_104889_.getMinSection(); k < this.f_104889_.getMaxSection(); ++k) {
+                  this.f_104888_.f_91060_.setSectionDirty(clientboundchunksbiomespacket$chunkbiomedata2.f_273927_().getX().add(i), me.alphamode.mclong.math.BigInteger.valueOf(k), clientboundchunksbiomespacket$chunkbiomedata2.f_273927_().getZ().add(j));
                }
             }
          }
@@ -708,47 +_,47 @@
 
    }
 
-   private void m_194198_(int p_194199_, int p_194200_, ClientboundLevelChunkPacketData p_194201_) {
-      this.f_104889_.m_7726_().m_194116_(p_194199_, p_194200_, p_194201_.m_195656_(), p_194201_.m_195678_(), p_194201_.m_195657_(p_194199_, p_194200_));
+   private void updateLevelChunk(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ, ClientboundLevelChunkPacketData pData) {
+      this.f_104889_.m_7726_().replaceWithPacketData(pX, pZ, pData.m_195656_(), pData.m_195678_(), pData.getBlockEntitiesTagsConsumer(pX, pZ));
    }
 
-   private void m_194212_(LevelChunk p_194213_, int p_194214_, int p_194215_) {
+   private void enableChunkLight(LevelChunk pChunk, me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ) {
       LevelLightEngine levellightengine = this.f_104889_.m_7726_().m_7827_();
-      LevelChunkSection[] alevelchunksection = p_194213_.m_7103_();
-      ChunkPos chunkpos = p_194213_.m_7697_();
+      LevelChunkSection[] alevelchunksection = pChunk.m_7103_();
+      ChunkPos chunkpos = pChunk.m_7697_();
 
       for(int i = 0; i < alevelchunksection.length; ++i) {
          LevelChunkSection levelchunksection = alevelchunksection[i];
-         int j = this.f_104889_.m_151568_(i);
-         levellightengine.m_6191_(SectionPos.m_123196_(chunkpos, j), levelchunksection.m_188008_());
-         this.f_104889_.m_104793_(p_194214_, j, p_194215_);
+         long j = this.f_104889_.getSectionYFromSectionIndex(i);
+         levellightengine.m_6191_(SectionPos.of(chunkpos, j), levelchunksection.m_188008_());
+         this.f_104889_.setSectionDirtyWithNeighbors(pX, me.alphamode.mclong.math.BigInteger.valueOf(j), pZ);
       }
 
    }
 
    public void m_5729_(ClientboundForgetLevelChunkPacket p_105014_) {
       PacketUtils.m_131363_(p_105014_, this, this.f_104888_);
-      int i = p_105014_.m_132149_();
-      int j = p_105014_.m_132152_();
+      me.alphamode.mclong.math.BigInteger i = p_105014_.getX();
+      me.alphamode.mclong.math.BigInteger j = p_105014_.getZ();
       ClientChunkCache clientchunkcache = this.f_104889_.m_7726_();
-      clientchunkcache.m_104455_(i, j);
+      clientchunkcache.drop(i, j);
       this.m_194252_(p_105014_);
    }
 
    private void m_194252_(ClientboundForgetLevelChunkPacket p_194253_) {
-      ChunkPos chunkpos = new ChunkPos(p_194253_.m_132149_(), p_194253_.m_132152_());
+      ChunkPos chunkpos = new ChunkPos(p_194253_.getX(), p_194253_.getZ());
       this.f_104889_.m_194171_(() -> {
          LevelLightEngine levellightengine = this.f_104889_.m_5518_();
          levellightengine.m_9335_(chunkpos, false);
 
-         for(int i = levellightengine.m_164447_(); i < levellightengine.m_164448_(); ++i) {
-            SectionPos sectionpos = SectionPos.m_123196_(chunkpos, i);
+         for(long i = levellightengine.getMinLightSection(); i < levellightengine.getMaxLightSection(); ++i) {
+            SectionPos sectionpos = SectionPos.of(chunkpos, i);
             levellightengine.m_284126_(LightLayer.BLOCK, sectionpos, (DataLayer)null);
             levellightengine.m_284126_(LightLayer.SKY, sectionpos, (DataLayer)null);
          }
 
-         for(int j = this.f_104889_.m_151560_(); j < this.f_104889_.m_151561_(); ++j) {
-            levellightengine.m_6191_(SectionPos.m_123196_(chunkpos, j), true);
+         for(long j = this.f_104889_.getMinSection(); j < this.f_104889_.getMaxSection(); ++j) {
+            levellightengine.m_6191_(SectionPos.of(chunkpos, j), true);
          }
 
       });
@@ -792,9 +_,9 @@
 
       if (entity != null) {
          if (entity instanceof ExperienceOrb) {
-            this.f_104889_.m_7785_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), SoundEvents.f_11871_, SoundSource.PLAYERS, 0.1F, (this.f_104898_.m_188501_() - this.f_104898_.m_188501_()) * 0.35F + 0.9F, false);
+            this.f_104889_.playLocalSound(entity.getBigX(), entity.getBigY(), entity.getBigZ(), SoundEvents.f_11871_, SoundSource.PLAYERS, 0.1F, (this.f_104898_.m_188501_() - this.f_104898_.m_188501_()) * 0.35F + 0.9F, false);
          } else {
-            this.f_104889_.m_7785_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, (this.f_104898_.m_188501_() - this.f_104898_.m_188501_()) * 1.4F + 2.0F, false);
+            this.f_104889_.playLocalSound(entity.getBigX(), entity.getBigY(), entity.getBigZ(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, (this.f_104898_.m_188501_() - this.f_104898_.m_188501_()) * 1.4F + 2.0F, false);
          }
 
          this.f_104888_.f_91061_.m_107344_(new ItemPickupParticle(this.f_104888_.m_91290_(), this.f_104888_.m_91269_(), this.f_104889_, entity, livingentity));
@@ -983,7 +_,7 @@
             case 35:
                int i = 40;
                this.f_104888_.f_91061_.m_107332_(entity, ParticleTypes.f_123767_, 30);
-               this.f_104889_.m_7785_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), SoundEvents.f_12513_, entity.m_5720_(), 1.0F, 1.0F, false);
+               this.f_104889_.playLocalSound(entity.getBigX(), entity.getBigY(), entity.getBigZ(), SoundEvents.f_12513_, entity.m_5720_(), 1.0F, 1.0F, false);
                if (entity == this.f_104888_.f_91074_) {
                   this.f_104888_.f_91063_.m_109113_(m_104927_(this.f_104888_.f_91074_));
                }
@@ -1090,7 +_,7 @@
 
    public void m_7345_(ClientboundExplodePacket p_105012_) {
       PacketUtils.m_131363_(p_105012_, this, this.f_104888_);
-      Explosion explosion = new Explosion(this.f_104888_.f_91073_, (Entity)null, p_105012_.m_132132_(), p_105012_.m_132133_(), p_105012_.m_132134_(), p_105012_.m_132135_(), p_105012_.m_132136_());
+      Explosion explosion = new Explosion(this.f_104888_.f_91073_, (Entity)null, p_105012_.getX(), p_105012_.getY(), p_105012_.getZ(), p_105012_.m_132135_(), p_105012_.m_132136_());
       explosion.m_46075_(true);
       this.f_104888_.f_91074_.m_20256_(this.f_104888_.f_91074_.m_20184_().m_82520_((double)p_105012_.m_132127_(), (double)p_105012_.m_132130_(), (double)p_105012_.m_132131_()));
    }
@@ -1267,17 +_,17 @@
             this.f_104888_.f_91065_.m_93076_().m_93785_(Component.m_237110_("demo.day.6", options.f_92102_.m_90863_()));
          }
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132159_) {
-         this.f_104889_.m_6263_(player, player.m_20185_(), player.m_20188_(), player.m_20189_(), SoundEvents.f_11686_, SoundSource.PLAYERS, 0.18F, 0.45F);
+         this.f_104889_.playSound(player, player.getBigX(), player.getEyeBigY(), player.getBigZ(), SoundEvents.f_11686_, SoundSource.PLAYERS, 0.18F, 0.45F);
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132160_) {
          this.f_104889_.m_46734_(f);
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132161_) {
          this.f_104889_.m_46707_(f);
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132162_) {
-         this.f_104889_.m_6263_(player, player.m_20185_(), player.m_20186_(), player.m_20189_(), SoundEvents.f_12295_, SoundSource.NEUTRAL, 1.0F, 1.0F);
+         this.f_104889_.playSound(player, player.getBigX(), player.getBigY(), player.getBigZ(), SoundEvents.f_12295_, SoundSource.NEUTRAL, 1.0F, 1.0F);
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132163_) {
-         this.f_104889_.m_7106_(ParticleTypes.f_123807_, player.m_20185_(), player.m_20186_(), player.m_20189_(), 0.0D, 0.0D, 0.0D);
+         this.f_104889_.addParticle(ParticleTypes.f_123807_, player.getBigX(), player.getBigY(), player.getBigZ(), 0.0D, 0.0D, 0.0D);
          if (i == 1) {
-            this.f_104889_.m_6263_(player, player.m_20185_(), player.m_20186_(), player.m_20189_(), SoundEvents.f_11880_, SoundSource.HOSTILE, 1.0F, 1.0F);
+            this.f_104889_.playSound(player, player.getBigX(), player.getBigY(), player.getBigZ(), SoundEvents.f_11880_, SoundSource.HOSTILE, 1.0F, 1.0F);
          }
       } else if (clientboundgameeventpacket$type == ClientboundGameEventPacket.f_132164_) {
          this.f_104888_.f_91074_.m_108711_(f == 0.0F);
@@ -1507,7 +_,7 @@
          worldborder.m_61917_(p_171767_.m_178888_());
       }
 
-      worldborder.m_61923_(p_171767_.m_178891_());
+      worldborder.setAbsoluteMaxSize(p_171767_.getNewAbsoluteMaxSize());
       worldborder.m_61952_(p_171767_.m_178893_());
       worldborder.m_61944_(p_171767_.m_178892_());
    }
@@ -1735,7 +_,7 @@
 
    public void m_8068_(ClientboundSoundPacket p_105114_) {
       PacketUtils.m_131363_(p_105114_, this, this.f_104888_);
-      this.f_104888_.f_91073_.m_262808_(this.f_104888_.f_91074_, p_105114_.m_133459_(), p_105114_.m_133460_(), p_105114_.m_133461_(), p_105114_.m_263229_(), p_105114_.m_133458_(), p_105114_.m_133462_(), p_105114_.m_133463_(), p_105114_.m_237848_());
+      this.f_104888_.f_91073_.playSeededSound(this.f_104888_.f_91074_, p_105114_.getX(), p_105114_.getY(), p_105114_.getZ(), p_105114_.m_263229_(), p_105114_.m_133458_(), p_105114_.m_133462_(), p_105114_.m_133463_(), p_105114_.m_237848_());
    }
 
    public void m_5863_(ClientboundSoundEntityPacket p_105112_) {
@@ -1839,7 +_,7 @@
       PacketUtils.m_131363_(p_105038_, this, this.f_104888_);
       Entity entity = this.f_104888_.f_91074_.m_20201_();
       if (entity != this.f_104888_.f_91074_ && entity.m_6109_()) {
-         entity.m_19890_(p_105038_.m_132591_(), p_105038_.m_132594_(), p_105038_.m_132595_(), p_105038_.m_132596_(), p_105038_.m_132597_());
+         entity.absMoveTo(p_105038_.getX(), p_105038_.getY(), p_105038_.getZ(), p_105038_.m_132596_(), p_105038_.m_132597_());
          this.f_104885_.m_129512_(new ServerboundMoveVehiclePacket(entity));
       }
 
@@ -2163,7 +_,7 @@
          double d4 = (double)(p_105026_.m_132320_() * p_105026_.m_132319_());
 
          try {
-            this.f_104889_.m_6493_(p_105026_.m_132322_(), p_105026_.m_132311_(), p_105026_.m_132314_(), p_105026_.m_132315_(), p_105026_.m_132316_(), d0, d2, d4);
+            this.f_104889_.addParticle(p_105026_.m_132322_(), p_105026_.m_132311_(), p_105026_.getX(), p_105026_.getY(), p_105026_.getZ(), d0, d2, d4);
          } catch (Throwable throwable1) {
             f_104883_.warn("Could not spawn particle effect {}", (Object)p_105026_.m_132322_());
          }
@@ -2177,7 +_,7 @@
             double d8 = this.f_104898_.m_188583_() * (double)p_105026_.m_132320_();
 
             try {
-               this.f_104889_.m_6493_(p_105026_.m_132322_(), p_105026_.m_132311_(), p_105026_.m_132314_() + d1, p_105026_.m_132315_() + d3, p_105026_.m_132316_() + d5, d6, d7, d8);
+               this.f_104889_.addParticle(p_105026_.m_132322_(), p_105026_.m_132311_(), p_105026_.getX().add(d1), p_105026_.getY().add(d3), p_105026_.getZ().add(d5), d6, d7, d8);
             } catch (Throwable throwable) {
                f_104883_.warn("Could not spawn particle effect {}", (Object)p_105026_.m_132322_());
                return;
@@ -2235,25 +_,25 @@
 
    public void m_183514_(ClientboundLightUpdatePacket p_194243_) {
       PacketUtils.m_131363_(p_194243_, this, this.f_104888_);
-      int i = p_194243_.m_132349_();
-      int j = p_194243_.m_132352_();
+      me.alphamode.mclong.math.BigInteger i = p_194243_.getX();
+      me.alphamode.mclong.math.BigInteger j = p_194243_.getZ();
       ClientboundLightUpdatePacketData clientboundlightupdatepacketdata = p_194243_.m_195722_();
       this.f_104889_.m_194171_(() -> {
-         this.m_194248_(i, j, clientboundlightupdatepacketdata);
+         this.applyLightData(i, j, clientboundlightupdatepacketdata);
       });
    }
 
-   private void m_194248_(int p_194249_, int p_194250_, ClientboundLightUpdatePacketData p_194251_) {
+   private void applyLightData(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ, ClientboundLightUpdatePacketData pData) {
       LevelLightEngine levellightengine = this.f_104889_.m_7726_().m_7827_();
-      BitSet bitset = p_194251_.m_195740_();
-      BitSet bitset1 = p_194251_.m_195751_();
-      Iterator<byte[]> iterator = p_194251_.m_195754_().iterator();
-      this.m_171734_(p_194249_, p_194250_, levellightengine, LightLayer.SKY, bitset, bitset1, iterator);
-      BitSet bitset2 = p_194251_.m_195757_();
-      BitSet bitset3 = p_194251_.m_195758_();
-      Iterator<byte[]> iterator1 = p_194251_.m_195759_().iterator();
-      this.m_171734_(p_194249_, p_194250_, levellightengine, LightLayer.BLOCK, bitset2, bitset3, iterator1);
-      levellightengine.m_9335_(new ChunkPos(p_194249_, p_194250_), true);
+      BitSet bitset = pData.m_195740_();
+      BitSet bitset1 = pData.m_195751_();
+      Iterator<byte[]> iterator = pData.m_195754_().iterator();
+      this.readSectionList(pX, pZ, levellightengine, LightLayer.SKY, bitset, bitset1, iterator);
+      BitSet bitset2 = pData.m_195757_();
+      BitSet bitset3 = pData.m_195758_();
+      Iterator<byte[]> iterator1 = pData.m_195759_().iterator();
+      this.readSectionList(pX, pZ, levellightengine, LightLayer.BLOCK, bitset2, bitset3, iterator1);
+      levellightengine.m_9335_(new ChunkPos(pX, pZ), true);
    }
 
    public void m_7330_(ClientboundMerchantOffersPacket p_105034_) {
@@ -2284,7 +_,7 @@
 
    public void m_8065_(ClientboundSetChunkCacheCenterPacket p_105080_) {
       PacketUtils.m_131363_(p_105080_, this, this.f_104888_);
-      this.f_104889_.m_7726_().m_104459_(p_105080_.m_133094_(), p_105080_.m_133097_());
+      this.f_104889_.m_7726_().updateViewCenter(p_105080_.getX(), p_105080_.getZ());
    }
 
    public void m_214108_(ClientboundBlockChangedAckPacket p_233698_) {
@@ -2301,14 +_,14 @@
 
    }
 
-   private void m_171734_(int p_171735_, int p_171736_, LevelLightEngine p_171737_, LightLayer p_171738_, BitSet p_171739_, BitSet p_171740_, Iterator<byte[]> p_171741_) {
-      for(int i = 0; i < p_171737_.m_164446_(); ++i) {
-         int j = p_171737_.m_164447_() + i;
+   private void readSectionList(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ, LevelLightEngine pLightEngine, LightLayer pLightLayer, BitSet p_171739_, BitSet p_171740_, Iterator<byte[]> p_171741_) {
+      for(int i = 0; i < pLightEngine.getLightSectionCount(); ++i) {
+         long j = pLightEngine.getMinLightSection() + i;
          boolean flag = p_171739_.get(i);
          boolean flag1 = p_171740_.get(i);
          if (flag || flag1) {
-            p_171737_.m_284126_(p_171738_, SectionPos.m_123173_(p_171735_, j, p_171736_), flag ? new DataLayer((byte[])p_171741_.next().clone()) : new DataLayer());
-            this.f_104889_.m_104793_(p_171735_, j, p_171736_);
+            pLightEngine.m_284126_(pLightLayer, SectionPos.of(pX, me.alphamode.mclong.math.BigInteger.valueOf(j), pZ), flag ? new DataLayer((byte[])p_171741_.next().clone()) : new DataLayer());
+            this.f_104889_.setSectionDirtyWithNeighbors(pX, me.alphamode.mclong.math.BigInteger.valueOf(j), pZ);
          }
       }
 
