--- a/net/minecraft/util/datafix/fixes/BlendingDataFix.java
+++ b/net/minecraft/util/datafix/fixes/BlendingDataFix.java
@@ -39,7 +_,7 @@
          String s = NamespacedSchema.m_17311_(optional.get().asString("empty"));
          Optional<? extends Dynamic<?>> optional1 = p_240279_.get("below_zero_retrogen").result();
          if (!f_216558_.contains(s)) {
-            p_240279_ = m_216566_(p_240279_, 384, -64);
+            p_240279_ = m_216566_(p_240279_, net.minecraft.world.level.dimension.DimensionDefaults.f_156635_, net.minecraft.world.level.dimension.DimensionDefaults.f_156634_);
          } else if (optional1.isPresent()) {
             Dynamic<?> dynamic = optional1.get();
             String s1 = NamespacedSchema.m_17311_(dynamic.get("target_status").asString("empty"));
@@ -53,6 +_,6 @@
    }
 
    private static Dynamic<?> m_216566_(Dynamic<?> p_216567_, int p_216568_, int p_216569_) {
-      return p_216567_.set("blending_data", p_216567_.createMap(Map.of(p_216567_.createString("min_section"), p_216567_.createInt(SectionPos.m_123171_(p_216569_)), p_216567_.createString("max_section"), p_216567_.createInt(SectionPos.m_123171_(p_216569_ + p_216568_)))));
+      return p_216567_.set("blending_data", p_216567_.createMap(Map.of(p_216567_.createString("min_section"), p_216567_.createLong(SectionPos.blockToSectionCoord(p_216569_)), p_216567_.createString("max_section"), p_216567_.createLong(SectionPos.blockToSectionCoord(p_216569_ + p_216568_)))));
    }
 }
