--- a/com/mojang/blaze3d/audio/Channel.java
+++ b/com/mojang/blaze3d/audio/Channel.java
@@ -95,7 +_,7 @@
    }
 
    public void m_83654_(Vec3 p_83655_) {
-      AL10.alSourcefv(this.f_83642_, 4100, new float[]{(float)p_83655_.f_82479_, (float)p_83655_.f_82480_, (float)p_83655_.f_82481_});
+      AL10.alSourcefv(this.f_83642_, 4100, new float[]{(float)p_83655_.m_7096_(), (float)p_83655_.m_7098_(), (float)p_83655_.m_7094_()});
    }
 
    public void m_83650_(float p_83651_) {
