--- a/com/mojang/blaze3d/audio/Listener.java
+++ b/com/mojang/blaze3d/audio/Listener.java
@@ -13,7 +_,7 @@
 
    public void m_83739_(Vec3 p_83740_) {
       this.f_83734_ = p_83740_;
-      AL10.alListener3f(4100, (float)p_83740_.f_82479_, (float)p_83740_.f_82480_, (float)p_83740_.f_82481_);
+      AL10.alListener3f(4100, (float)p_83740_.m_7096_(), (float)p_83740_.m_7098_(), (float)p_83740_.m_7094_());
    }
 
    public Vec3 m_83736_() {
