--- a/net/minecraft/world/level/levelgen/structure/pools/EmptyPoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/EmptyPoolElement.java
@@ -24,8 +_,8 @@
       super(StructureTemplatePool.Projection.TERRAIN_MATCHING);
    }
 
-   public Vec3i m_213577_(StructureTemplateManager p_227169_, Rotation p_227170_) {
-      return Vec3i.f_123288_;
+   public me.alphamode.mclong.core.Vec3l getSize(StructureTemplateManager p_227169_, Rotation p_227170_) {
+      return me.alphamode.mclong.core.Vec3l.ZERO;
    }
 
    public List<StructureTemplate.StructureBlockInfo> m_213638_(StructureTemplateManager p_227176_, BlockPos p_227177_, Rotation p_227178_, RandomSource p_227179_) {
