--- a/net/minecraft/world/level/levelgen/structure/pools/ListPoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/ListPoolElement.java
@@ -34,19 +_,19 @@
       }
    }
 
-   public Vec3i m_213577_(StructureTemplateManager p_227283_, Rotation p_227284_) {
-      int i = 0;
-      int j = 0;
-      int k = 0;
+   public me.alphamode.mclong.core.Vec3l getSize(StructureTemplateManager pStructureTemplateManager, Rotation pRotation) {
+      long i = 0;
+      long j = 0;
+      long k = 0;
 
       for(StructurePoolElement structurepoolelement : this.f_210360_) {
-         Vec3i vec3i = structurepoolelement.m_213577_(p_227283_, p_227284_);
-         i = Math.max(i, vec3i.m_123341_());
-         j = Math.max(j, vec3i.m_123342_());
-         k = Math.max(k, vec3i.m_123343_());
+         me.alphamode.mclong.core.Vec3l vec3i = structurepoolelement.getSize(pStructureTemplateManager, pRotation);
+         i = Math.max(i, vec3i.getX());
+         j = Math.max(j, vec3i.getY());
+         k = Math.max(k, vec3i.getZ());
       }
 
-      return new Vec3i(i, j, k);
+      return new me.alphamode.mclong.core.Vec3l(i, j, k);
    }
 
    public List<StructureTemplate.StructureBlockInfo> m_213638_(StructureTemplateManager p_227290_, BlockPos p_227291_, Rotation p_227292_, RandomSource p_227293_) {
