--- a/net/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/SinglePoolElement.java
@@ -65,9 +_,9 @@
       this.f_210412_ = p_210416_;
    }
 
-   public Vec3i m_213577_(StructureTemplateManager p_227313_, Rotation p_227314_) {
+   public me.alphamode.mclong.core.Vec3l getSize(StructureTemplateManager p_227313_, Rotation p_227314_) {
       StructureTemplate structuretemplate = this.m_227299_(p_227313_);
-      return structuretemplate.m_163808_(p_227314_);
+      return structuretemplate.getSize(p_227314_);
    }
 
    private StructureTemplate m_227299_(StructureTemplateManager p_227300_) {
