--- a/net/minecraft/world/level/levelgen/structure/pools/StructurePoolElement.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/StructurePoolElement.java
@@ -38,7 +_,7 @@
       this.f_210467_ = p_210471_;
    }
 
-   public abstract Vec3i m_213577_(StructureTemplateManager p_227346_, Rotation p_227347_);
+   public abstract me.alphamode.mclong.core.Vec3l getSize(StructureTemplateManager p_227346_, Rotation p_227347_);
 
    public abstract List<StructureTemplate.StructureBlockInfo> m_213638_(StructureTemplateManager p_227351_, BlockPos p_227352_, Rotation p_227353_, RandomSource p_227354_);
 
