--- a/net/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
+++ b/net/minecraft/world/level/levelgen/structure/pools/StructureTemplatePool.java
@@ -35,7 +_,7 @@
    private final List<Pair<StructurePoolElement, Integer>> f_210559_;
    private final ObjectArrayList<StructurePoolElement> f_210560_;
    private final Holder<StructureTemplatePool> f_210561_;
-   private int f_210562_ = Integer.MIN_VALUE;
+   private long f_210562_ = java.lang.Long.MIN_VALUE;
 
    public StructureTemplatePool(Holder<StructureTemplatePool> p_255747_, List<Pair<StructurePoolElement, Integer>> p_255919_) {
       this.f_210559_ = p_255919_;
@@ -68,12 +_,12 @@
       this.f_210561_ = p_255795_;
    }
 
-   public int m_227357_(StructureTemplateManager p_227358_) {
-      if (this.f_210562_ == Integer.MIN_VALUE) {
+   public long getMaxSize(StructureTemplateManager p_227358_) {
+      if (this.f_210562_ == java.lang.Long.MIN_VALUE) {
          this.f_210562_ = this.f_210560_.stream().filter((p_210577_) -> {
             return p_210577_ != EmptyPoolElement.f_210175_;
-         }).mapToInt((p_227361_) -> {
-            return p_227361_.m_214015_(p_227358_, BlockPos.f_121853_, Rotation.NONE).m_71057_();
+         }).mapToLong((p_227361_) -> {
+            return p_227361_.m_214015_(p_227358_, BlockPos.f_121853_, Rotation.NONE).getYSpan();
          }).max().orElse(0);
       }
 
