--- a/net/minecraft/world/entity/monster/hoglin/Hoglin.java
+++ b/net/minecraft/world/entity/monster/hoglin/Hoglin.java
@@ -156,7 +_,7 @@
    }
 
    public static boolean m_219181_(EntityType<Hoglin> p_219182_, LevelAccessor p_219183_, MobSpawnType p_219184_, BlockPos p_219185_, RandomSource p_219186_) {
-      return !p_219183_.m_8055_(p_219185_.m_7495_()).m_60713_(Blocks.f_50451_);
+      return !p_219183_.m_8055_(p_219185_.below()).m_60713_(Blocks.f_50451_);
    }
 
    @Nullable
@@ -176,7 +_,7 @@
       if (HoglinAi.m_34585_(this, p_34516_)) {
          return -1.0F;
       } else {
-         return p_34517_.m_8055_(p_34516_.m_7495_()).m_60713_(Blocks.f_50699_) ? 10.0F : 0.0F;
+         return p_34517_.m_8055_(p_34516_.below()).m_60713_(Blocks.f_50699_) ? 10.0F : 0.0F;
       }
    }
 
