--- a/net/minecraft/world/entity/monster/hoglin/HoglinBase.java
+++ b/net/minecraft/world/entity/monster/hoglin/HoglinBase.java
@@ -40,7 +_,7 @@
          double d5 = d2 * (double)(p_34646_.m_9236_().f_46441_.m_188501_() * 0.5F + 0.2F);
          Vec3 vec3 = (new Vec3(d3, 0.0D, d4)).m_82541_().m_82490_(d5).m_82524_(f);
          double d6 = d2 * (double)p_34646_.m_9236_().f_46441_.m_188501_() * 0.5D;
-         p_34647_.m_5997_(vec3.f_82479_, d6, vec3.f_82481_);
+         p_34647_.m_5997_(vec3.m_7096_(), d6, vec3.m_7094_());
          p_34647_.f_19864_ = true;
       }
    }
