--- a/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
+++ b/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
@@ -133,7 +_,7 @@
 
    static boolean m_34585_(Hoglin p_34586_, BlockPos p_34587_) {
       Optional<BlockPos> optional = p_34586_.m_6274_().m_21952_(MemoryModuleType.f_26356_);
-      return optional.isPresent() && optional.get().m_123314_(p_34587_, 8.0D);
+      return optional.isPresent() && optional.get().closerThan(p_34587_, 8.0D);
    }
 
    private static boolean m_34617_(Hoglin p_34618_) {
