--- a/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
@@ -29,7 +_,7 @@
       for(int j = -4; j <= 4; ++j) {
          for(int k = -2; k <= 2; ++k) {
             for(int l = -4; l <= 4; ++l) {
-               BlockPos blockpos1 = blockpos.m_7918_(j, k, l);
+               BlockPos blockpos1 = blockpos.offset(j, k, l);
                if (p_26755_.m_7141_().m_35571_().f_35603_().contains(p_26754_.m_8055_(blockpos1).m_60734_())) {
                   list.add(GlobalPos.m_122643_(resourcekey, blockpos1));
                }
