--- a/net/minecraft/world/effect/MobEffectInstance.java
+++ b/net/minecraft/world/effect/MobEffectInstance.java
@@ -155,7 +_,7 @@
 
    public boolean m_19552_(LivingEntity p_19553_, Runnable p_19554_) {
       if (this.f_19503_ > 0) {
-         if (this.f_19502_.m_6584_(this.f_19503_, this.f_19504_)) {
+         if (this.f_19502_.isDurationEffectTick(this.f_19503_, this.f_19504_)) {
             this.m_19550_(p_19553_);
          }
 
