--- a/net/minecraft/world/level/block/grower/AbstractTreeGrower.java
+++ b/net/minecraft/world/level/block/grower/AbstractTreeGrower.java
@@ -44,7 +_,7 @@
    }
 
    private boolean m_60011_(LevelAccessor p_60012_, BlockPos p_60013_) {
-      for(BlockPos blockpos : BlockPos.MutableBlockPos.m_121940_(p_60013_.m_7495_().m_122013_(2).m_122025_(2), p_60013_.m_7494_().m_122020_(2).m_122030_(2))) {
+      for(BlockPos blockpos : BlockPos.MutableBlockPos.m_121940_(p_60013_.below().north(2).west(2), p_60013_.above().south(2).east(2))) {
          if (p_60012_.m_8055_(blockpos).m_204336_(BlockTags.f_13041_)) {
             return true;
          }
