--- a/net/minecraft/world/level/pathfinder/Node.java
+++ b/net/minecraft/world/level/pathfinder/Node.java
@@ -7,10 +_,10 @@
 import net.minecraft.world.phys.Vec3;
 
 public class Node {
-   public final int f_77271_;
-   public final int f_77272_;
-   public final int f_77273_;
-   private final int f_77283_;
+   public final long f_77271_;
+   public final long f_77272_;
+   public final long f_77273_;
+   private final long f_77283_;
    public int f_77274_ = -1;
    public float f_77275_;
    public float f_77276_;
@@ -22,14 +_,18 @@
    public float f_77281_;
    public BlockPathTypes f_77282_ = BlockPathTypes.BLOCKED;
 
-   public Node(int p_77285_, int p_77286_, int p_77287_) {
+   public Node(long p_77285_, long p_77286_, long p_77287_) {
       this.f_77271_ = p_77285_;
       this.f_77272_ = p_77286_;
       this.f_77273_ = p_77287_;
-      this.f_77283_ = m_77295_(p_77285_, p_77286_, p_77287_);
-   }
-
-   public Node m_77289_(int p_77290_, int p_77291_, int p_77292_) {
+      this.f_77283_ = createHash(p_77285_, p_77286_, p_77287_);
+   }
+
+   public Node(int p_77285_, int p_77286_, int p_77287_) {
+      this((long) p_77285_, p_77286_, p_77287_);
+   }
+
+   public Node cloneAndMove(long p_77290_, long p_77291_, long p_77292_) {
       Node node = new Node(p_77290_, p_77291_, p_77292_);
       node.f_77274_ = this.f_77274_;
       node.f_77275_ = this.f_77275_;
@@ -43,10 +_,15 @@
       return node;
    }
 
+   @Deprecated
    public static int m_77295_(int p_77296_, int p_77297_, int p_77298_) {
       return p_77297_ & 255 | (p_77296_ & 32767) << 8 | (p_77298_ & 32767) << 24 | (p_77296_ < 0 ? Integer.MIN_VALUE : 0) | (p_77298_ < 0 ? '\u8000' : 0);
    }
 
+   public static long createHash(long p_77296_, long p_77297_, long p_77298_) {
+      return p_77297_ & 255 | (p_77296_ & 32767) << 8 | (p_77298_ & 32767) << 24 | (p_77296_ < 0 ? Integer.MIN_VALUE : 0) | (p_77298_ < 0 ? '\u8000' : 0);
+   }
+
    public float m_77293_(Node p_77294_) {
       float f = (float)(p_77294_.f_77271_ - this.f_77271_);
       float f1 = (float)(p_77294_.f_77272_ - this.f_77272_);
@@ -61,9 +_,9 @@
    }
 
    public float m_164697_(BlockPos p_164698_) {
-      float f = (float)(p_164698_.m_123341_() - this.f_77271_);
-      float f1 = (float)(p_164698_.m_123342_() - this.f_77272_);
-      float f2 = (float)(p_164698_.m_123343_() - this.f_77273_);
+      float f = (float)(p_164698_.getX() - this.f_77271_);
+      float f1 = (float)(p_164698_.getY() - this.f_77272_);
+      float f2 = (float)(p_164698_.getZ() - this.f_77273_);
       return Mth.m_14116_(f * f + f1 * f1 + f2 * f2);
    }
 
@@ -75,9 +_,9 @@
    }
 
    public float m_164702_(BlockPos p_164703_) {
-      float f = (float)(p_164703_.m_123341_() - this.f_77271_);
-      float f1 = (float)(p_164703_.m_123342_() - this.f_77272_);
-      float f2 = (float)(p_164703_.m_123343_() - this.f_77273_);
+      float f = (float)(p_164703_.getX() - this.f_77271_);
+      float f1 = (float)(p_164703_.getY() - this.f_77272_);
+      float f2 = (float)(p_164703_.getZ() - this.f_77273_);
       return f * f + f1 * f1 + f2 * f2;
    }
 
@@ -89,9 +_,9 @@
    }
 
    public float m_77306_(BlockPos p_77307_) {
-      float f = (float)Math.abs(p_77307_.m_123341_() - this.f_77271_);
-      float f1 = (float)Math.abs(p_77307_.m_123342_() - this.f_77272_);
-      float f2 = (float)Math.abs(p_77307_.m_123343_() - this.f_77273_);
+      float f = (float)Math.abs(p_77307_.getX() - this.f_77271_);
+      float f1 = (float)Math.abs(p_77307_.getY() - this.f_77272_);
+      float f2 = (float)Math.abs(p_77307_.getZ() - this.f_77273_);
       return f + f1 + f2;
    }
 
@@ -112,7 +_,7 @@
    }
 
    public int hashCode() {
-      return this.f_77283_;
+      return (int) this.f_77283_;
    }
 
    public boolean m_77303_() {
@@ -124,9 +_,9 @@
    }
 
    public void m_164699_(FriendlyByteBuf p_164700_) {
-      p_164700_.writeInt(this.f_77271_);
-      p_164700_.writeInt(this.f_77272_);
-      p_164700_.writeInt(this.f_77273_);
+      p_164700_.writeLong(this.f_77271_);
+      p_164700_.writeLong(this.f_77272_);
+      p_164700_.writeLong(this.f_77273_);
       p_164700_.writeFloat(this.f_77280_);
       p_164700_.writeFloat(this.f_77281_);
       p_164700_.writeBoolean(this.f_77279_);
@@ -135,7 +_,7 @@
    }
 
    public static Node m_77301_(FriendlyByteBuf p_77302_) {
-      Node node = new Node(p_77302_.readInt(), p_77302_.readInt(), p_77302_.readInt());
+      Node node = new Node(p_77302_.readLong(), p_77302_.readLong(), p_77302_.readLong());
       m_262841_(p_77302_, node);
       return node;
    }
