--- a/net/minecraft/world/level/pathfinder/Node.java
+++ b/net/minecraft/world/level/pathfinder/Node.java
@@ -7,10 +_,10 @@
 import net.minecraft.world.phys.Vec3;
 
 public class Node {
-   public final int f_77271_;
-   public final int f_77272_;
-   public final int f_77273_;
-   private final int f_77283_;
+   public final me.alphamode.mclong.math.BigInteger f_77271_;
+   public final me.alphamode.mclong.math.BigInteger f_77272_;
+   public final me.alphamode.mclong.math.BigInteger f_77273_;
+   private final long f_77283_;
    public int f_77274_ = -1;
    public float f_77275_;
    public float f_77276_;
@@ -22,15 +_,15 @@
    public float f_77281_;
    public BlockPathTypes f_77282_ = BlockPathTypes.BLOCKED;
 
-   public Node(int p_77285_, int p_77286_, int p_77287_) {
-      this.f_77271_ = p_77285_;
-      this.f_77272_ = p_77286_;
-      this.f_77273_ = p_77287_;
-      this.f_77283_ = m_77295_(p_77285_, p_77286_, p_77287_);
+   public Node(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      this.f_77271_ = pX;
+      this.f_77272_ = pY;
+      this.f_77273_ = pZ;
+      this.f_77283_ = createHash(pX.longValue(), pY.longValue(), pZ.longValue());
    }
 
-   public Node m_77289_(int p_77290_, int p_77291_, int p_77292_) {
-      Node node = new Node(p_77290_, p_77291_, p_77292_);
+   public Node cloneAndMove(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      Node node = new Node(pX, pY, pZ);
       node.f_77274_ = this.f_77274_;
       node.f_77275_ = this.f_77275_;
       node.f_77276_ = this.f_77276_;
@@ -43,55 +_,60 @@
       return node;
    }
 
+   @Deprecated
    public static int m_77295_(int p_77296_, int p_77297_, int p_77298_) {
       return p_77297_ & 255 | (p_77296_ & 32767) << 8 | (p_77298_ & 32767) << 24 | (p_77296_ < 0 ? Integer.MIN_VALUE : 0) | (p_77298_ < 0 ? '\u8000' : 0);
    }
 
+   public static long createHash(long pX, long pY, long pZ) {
+      return pY & 255 | (pX & 32767) << 8 | (pZ & 32767) << 24 | (pX < 0 ? Integer.MIN_VALUE : 0) | (pZ < 0 ? '\u8000' : 0);
+   }
+
    public float m_77293_(Node p_77294_) {
-      float f = (float)(p_77294_.f_77271_ - this.f_77271_);
-      float f1 = (float)(p_77294_.f_77272_ - this.f_77272_);
-      float f2 = (float)(p_77294_.f_77273_ - this.f_77273_);
+      float f = (float)(p_77294_.f_77271_.subtract(this.f_77271_).floatValue());
+      float f1 = (float)(p_77294_.f_77272_.subtract(this.f_77272_).floatValue());
+      float f2 = (float)(p_77294_.f_77273_.subtract(this.f_77273_).floatValue());
       return Mth.m_14116_(f * f + f1 * f1 + f2 * f2);
    }
 
    public float m_230613_(Node p_230614_) {
-      float f = (float)(p_230614_.f_77271_ - this.f_77271_);
-      float f1 = (float)(p_230614_.f_77273_ - this.f_77273_);
+      float f = (float)(p_230614_.f_77271_.subtract(this.f_77271_).floatValue());
+      float f1 = (float)(p_230614_.f_77273_.subtract(this.f_77273_).floatValue());
       return Mth.m_14116_(f * f + f1 * f1);
    }
 
    public float m_164697_(BlockPos p_164698_) {
-      float f = (float)(p_164698_.m_123341_() - this.f_77271_);
-      float f1 = (float)(p_164698_.m_123342_() - this.f_77272_);
-      float f2 = (float)(p_164698_.m_123343_() - this.f_77273_);
+      float f = (float)(p_164698_.getBigX().subtract(this.f_77271_).floatValue());
+      float f1 = (float)(p_164698_.getBigY().subtract(this.f_77272_).floatValue());
+      float f2 = (float)(p_164698_.getBigZ().subtract(this.f_77273_).floatValue());
       return Mth.m_14116_(f * f + f1 * f1 + f2 * f2);
    }
 
    public float m_77299_(Node p_77300_) {
-      float f = (float)(p_77300_.f_77271_ - this.f_77271_);
-      float f1 = (float)(p_77300_.f_77272_ - this.f_77272_);
-      float f2 = (float)(p_77300_.f_77273_ - this.f_77273_);
+      float f = (float)(p_77300_.f_77271_.subtract(this.f_77271_).floatValue());
+      float f1 = (float)(p_77300_.f_77272_.subtract(this.f_77272_).floatValue());
+      float f2 = (float)(p_77300_.f_77273_.subtract(this.f_77273_).floatValue());
       return f * f + f1 * f1 + f2 * f2;
    }
 
    public float m_164702_(BlockPos p_164703_) {
-      float f = (float)(p_164703_.m_123341_() - this.f_77271_);
-      float f1 = (float)(p_164703_.m_123342_() - this.f_77272_);
-      float f2 = (float)(p_164703_.m_123343_() - this.f_77273_);
+      float f = (float)(p_164703_.getBigX().subtract(this.f_77271_).floatValue());
+      float f1 = (float)(p_164703_.getBigY().subtract(this.f_77272_).floatValue());
+      float f2 = (float)(p_164703_.getBigZ().subtract(this.f_77273_).floatValue());
       return f * f + f1 * f1 + f2 * f2;
    }
 
    public float m_77304_(Node p_77305_) {
-      float f = (float)Math.abs(p_77305_.f_77271_ - this.f_77271_);
-      float f1 = (float)Math.abs(p_77305_.f_77272_ - this.f_77272_);
-      float f2 = (float)Math.abs(p_77305_.f_77273_ - this.f_77273_);
+      float f = (float)Math.abs(p_77305_.f_77271_.subtract(this.f_77271_).floatValue());
+      float f1 = (float)Math.abs(p_77305_.f_77272_.subtract(this.f_77272_).floatValue());
+      float f2 = (float)Math.abs(p_77305_.f_77273_.subtract(this.f_77273_).floatValue());
       return f + f1 + f2;
    }
 
    public float m_77306_(BlockPos p_77307_) {
-      float f = (float)Math.abs(p_77307_.m_123341_() - this.f_77271_);
-      float f1 = (float)Math.abs(p_77307_.m_123342_() - this.f_77272_);
-      float f2 = (float)Math.abs(p_77307_.m_123343_() - this.f_77273_);
+      float f = (float)Math.abs(p_77307_.getBigX().subtract(this.f_77271_).floatValue());
+      float f1 = (float)Math.abs(p_77307_.getBigY().subtract(this.f_77272_).floatValue());
+      float f2 = (float)Math.abs(p_77307_.getBigZ().subtract(this.f_77273_).floatValue());
       return f + f1 + f2;
    }
 
@@ -100,19 +_,19 @@
    }
 
    public Vec3 m_164701_() {
-      return new Vec3((double)this.f_77271_, (double)this.f_77272_, (double)this.f_77273_);
+      return new Vec3(this.f_77271_.toBigDecimal(), this.f_77272_.toBigDecimal(), this.f_77273_.toBigDecimal());
    }
 
    public boolean equals(Object p_77309_) {
       if (!(p_77309_ instanceof Node node)) {
          return false;
       } else {
-         return this.f_77283_ == node.f_77283_ && this.f_77271_ == node.f_77271_ && this.f_77272_ == node.f_77272_ && this.f_77273_ == node.f_77273_;
+         return this.f_77283_ == node.f_77283_ && this.f_77271_.equals(node.f_77271_) && this.f_77272_.equals(node.f_77272_) && this.f_77273_.equals(node.f_77273_);
       }
    }
 
    public int hashCode() {
-      return this.f_77283_;
+      return (int) this.f_77283_;
    }
 
    public boolean m_77303_() {
@@ -124,9 +_,9 @@
    }
 
    public void m_164699_(FriendlyByteBuf p_164700_) {
-      p_164700_.writeInt(this.f_77271_);
-      p_164700_.writeInt(this.f_77272_);
-      p_164700_.writeInt(this.f_77273_);
+      p_164700_.m_130070_(this.f_77271_.toString());
+      p_164700_.m_130070_(this.f_77272_.toString());
+      p_164700_.m_130070_(this.f_77273_.toString());
       p_164700_.writeFloat(this.f_77280_);
       p_164700_.writeFloat(this.f_77281_);
       p_164700_.writeBoolean(this.f_77279_);
@@ -135,7 +_,7 @@
    }
 
    public static Node m_77301_(FriendlyByteBuf p_77302_) {
-      Node node = new Node(p_77302_.readInt(), p_77302_.readInt(), p_77302_.readInt());
+      Node node = new Node(new me.alphamode.mclong.math.BigInteger(p_77302_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77302_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77302_.m_130277_()));
       m_262841_(p_77302_, node);
       return node;
    }
