--- a/net/minecraft/world/level/pathfinder/PathFinder.java
+++ b/net/minecraft/world/level/pathfinder/PathFinder.java
@@ -39,7 +_,7 @@
          return null;
       } else {
          Map<Target, BlockPos> map = p_77430_.stream().collect(Collectors.toMap((p_77448_) -> {
-            return this.f_77422_.m_7568_((double)p_77448_.m_123341_(), (double)p_77448_.m_123342_(), (double)p_77448_.m_123343_());
+            return this.f_77422_.getGoal(p_77448_.getBigDecX(), p_77448_.getBigDecY(), p_77448_.getBigDecZ());
          }, Function.identity()));
          Path path = this.m_164716_(p_77428_.m_151625_(), node, map, p_77431_, p_77432_, p_77433_);
          this.f_77422_.m_6802_();
