--- a/net/minecraft/world/level/pathfinder/Path.java
+++ b/net/minecraft/world/level/pathfinder/Path.java
@@ -149,9 +_,9 @@
          this.f_77365_.forEach((p_164708_) -> {
             p_164708_.m_164699_(p_164705_);
          });
-         p_164705_.writeInt(this.f_77367_.m_123341_());
-         p_164705_.writeInt(this.f_77367_.m_123342_());
-         p_164705_.writeInt(this.f_77367_.m_123343_());
+         p_164705_.writeLong(this.f_77367_.getX());
+         p_164705_.writeLong(this.f_77367_.getY());
+         p_164705_.writeLong(this.f_77367_.getZ());
          p_164705_.writeInt(this.f_77362_.size());
 
          for(Node node : this.f_77362_) {
@@ -183,7 +_,7 @@
          set.add(Target.m_77506_(p_77391_));
       }
 
-      BlockPos blockpos = new BlockPos(p_77391_.readInt(), p_77391_.readInt(), p_77391_.readInt());
+      BlockPos blockpos = new BlockPos(p_77391_.readLong(), p_77391_.readLong(), p_77391_.readLong());
       List<Node> list = Lists.newArrayList();
       int l = p_77391_.readInt();
 
