--- a/net/minecraft/world/level/pathfinder/Path.java
+++ b/net/minecraft/world/level/pathfinder/Path.java
@@ -75,9 +_,9 @@
 
    public Vec3 m_77382_(Entity p_77383_, int p_77384_) {
       Node node = this.f_77362_.get(p_77384_);
-      double d0 = (double)node.f_77271_ + (double)((int)(p_77383_.m_20205_() + 1.0F)) * 0.5D;
-      double d1 = (double)node.f_77272_;
-      double d2 = (double)node.f_77273_ + (double)((int)(p_77383_.m_20205_() + 1.0F)) * 0.5D;
+      me.alphamode.mclong.math.BigDecimal d0 = node.f_77271_.toBigDecimal().add((double)((int)(p_77383_.m_20205_() + 1.0F)) * 0.5D);
+      me.alphamode.mclong.math.BigDecimal d1 = node.f_77272_.toBigDecimal();
+      me.alphamode.mclong.math.BigDecimal d2 = node.f_77273_.toBigDecimal().add((double)((int)(p_77383_.m_20205_() + 1.0F)) * 0.5D);
       return new Vec3(d0, d1, d2);
    }
 
@@ -149,9 +_,9 @@
          this.f_77365_.forEach((p_164708_) -> {
             p_164708_.m_164699_(p_164705_);
          });
-         p_164705_.writeInt(this.f_77367_.m_123341_());
-         p_164705_.writeInt(this.f_77367_.m_123342_());
-         p_164705_.writeInt(this.f_77367_.m_123343_());
+         p_164705_.m_130070_(this.f_77367_.getBigX().toString());
+         p_164705_.m_130070_(this.f_77367_.getBigY().toString());
+         p_164705_.m_130070_(this.f_77367_.getBigZ().toString());
          p_164705_.writeInt(this.f_77362_.size());
 
          for(Node node : this.f_77362_) {
@@ -183,7 +_,7 @@
          set.add(Target.m_77506_(p_77391_));
       }
 
-      BlockPos blockpos = new BlockPos(p_77391_.readInt(), p_77391_.readInt(), p_77391_.readInt());
+      BlockPos blockpos = new BlockPos(new me.alphamode.mclong.math.BigInteger(p_77391_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77391_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77391_.m_130277_()));
       List<Node> list = Lists.newArrayList();
       int l = p_77391_.readInt();
 
