--- a/net/minecraft/world/level/pathfinder/SwimNodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/SwimNodeEvaluator.java
@@ -35,12 +_,12 @@
 
    @Nullable
    public Node m_7171_() {
-      return super.m_5676_(Mth.m_14107_(this.f_77313_.m_20191_().f_82288_), Mth.m_14107_(this.f_77313_.m_20191_().f_82289_ + 0.5D), Mth.m_14107_(this.f_77313_.m_20191_().f_82290_));
+      return super.getNode(Mth.m_14107_(this.f_77313_.m_20191_().f_82288_), Mth.m_14107_(this.f_77313_.m_20191_().f_82289_ + 0.5D), Mth.m_14107_(this.f_77313_.m_20191_().f_82290_));
    }
 
    @Nullable
    public Target m_7568_(double p_77459_, double p_77460_, double p_77461_) {
-      return this.m_230615_(super.m_5676_(Mth.m_14107_(p_77459_), Mth.m_14107_(p_77460_), Mth.m_14107_(p_77461_)));
+      return this.m_230615_(super.getNode(Mth.m_14107_(p_77459_), Mth.m_14107_(p_77460_), Mth.m_14107_(p_77461_)));
    }
 
    public int m_6065_(Node[] p_77483_, Node p_77484_) {
@@ -48,7 +_,7 @@
       Map<Direction, Node> map = Maps.newEnumMap(Direction.class);
 
       for(Direction direction : Direction.values()) {
-         Node node = this.m_5676_(p_77484_.f_77271_ + direction.m_122429_(), p_77484_.f_77272_ + direction.m_122430_(), p_77484_.f_77273_ + direction.m_122431_());
+         Node node = this.getNode(p_77484_.f_77271_ + direction.m_122429_(), p_77484_.f_77272_ + direction.m_122430_(), p_77484_.f_77273_ + direction.m_122431_());
          map.put(direction, node);
          if (this.m_192961_(node)) {
             p_77483_[i++] = node;
@@ -57,7 +_,7 @@
 
       for(Direction direction1 : Direction.Plane.HORIZONTAL) {
          Direction direction2 = direction1.m_122427_();
-         Node node1 = this.m_5676_(p_77484_.f_77271_ + direction1.m_122429_() + direction2.m_122429_(), p_77484_.f_77272_, p_77484_.f_77273_ + direction1.m_122431_() + direction2.m_122431_());
+         Node node1 = this.getNode(p_77484_.f_77271_ + direction1.m_122429_() + direction2.m_122429_(), p_77484_.f_77272_, p_77484_.f_77273_ + direction1.m_122431_() + direction2.m_122431_());
          if (this.m_192963_(node1, map.get(direction1), map.get(direction2))) {
             p_77483_[i++] = node1;
          }
@@ -75,13 +_,13 @@
    }
 
    @Nullable
-   protected Node m_5676_(int p_77463_, int p_77464_, int p_77465_) {
+   protected Node getNode(long p_77463_, long p_77464_, long p_77465_) {
       Node node = null;
-      BlockPathTypes blockpathtypes = this.m_192967_(p_77463_, p_77464_, p_77465_);
+      BlockPathTypes blockpathtypes = this.getCachedBlockType(p_77463_, p_77464_, p_77465_);
       if (this.f_77455_ && blockpathtypes == BlockPathTypes.BREACH || blockpathtypes == BlockPathTypes.WATER) {
          float f = this.f_77313_.m_21439_(blockpathtypes);
          if (f >= 0.0F) {
-            node = super.m_5676_(p_77463_, p_77464_, p_77465_);
+            node = super.getNode(p_77463_, p_77464_, p_77465_);
             if (node != null) {
                node.f_77282_ = blockpathtypes;
                node.f_77281_ = Math.max(node.f_77281_, f);
@@ -95,25 +_,25 @@
       return node;
    }
 
-   protected BlockPathTypes m_192967_(int p_192968_, int p_192969_, int p_192970_) {
-      return this.f_192951_.computeIfAbsent(BlockPos.m_121882_(p_192968_, p_192969_, p_192970_), (p_192957_) -> {
-         return this.m_8086_(this.f_77312_, p_192968_, p_192969_, p_192970_);
+   protected BlockPathTypes getCachedBlockType(long p_192968_, long p_192969_, long p_192970_) {
+      return this.f_192951_.computeIfAbsent(BlockPos.asLong(p_192968_, p_192969_, p_192970_), (p_192957_) -> {
+         return this.getBlockPathType(this.f_77312_, p_192968_, p_192969_, p_192970_);
       });
    }
 
-   public BlockPathTypes m_8086_(BlockGetter p_77467_, int p_77468_, int p_77469_, int p_77470_) {
-      return this.m_7209_(p_77467_, p_77468_, p_77469_, p_77470_, this.f_77313_, this.f_77315_, this.f_77316_, this.f_77317_, this.m_77360_(), this.m_77357_());
+   public BlockPathTypes getBlockPathType(BlockGetter p_77467_, long p_77468_, long p_77469_, long p_77470_) {
+      return this.getBlockPathType(p_77467_, p_77468_, p_77469_, p_77470_, this.f_77313_, this.f_77315_, this.f_77316_, this.f_77317_, this.m_77360_(), this.m_77357_());
    }
 
-   public BlockPathTypes m_7209_(BlockGetter p_77472_, int p_77473_, int p_77474_, int p_77475_, Mob p_77476_, int p_77477_, int p_77478_, int p_77479_, boolean p_77480_, boolean p_77481_) {
+   public BlockPathTypes getBlockPathType(BlockGetter p_77472_, long p_77473_, long p_77474_, long p_77475_, Mob p_77476_, int p_77477_, int p_77478_, int p_77479_, boolean p_77480_, boolean p_77481_) {
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-      for(int i = p_77473_; i < p_77473_ + p_77477_; ++i) {
-         for(int j = p_77474_; j < p_77474_ + p_77478_; ++j) {
-            for(int k = p_77475_; k < p_77475_ + p_77479_; ++k) {
-               FluidState fluidstate = p_77472_.m_6425_(blockpos$mutableblockpos.m_122178_(i, j, k));
-               BlockState blockstate = p_77472_.m_8055_(blockpos$mutableblockpos.m_122178_(i, j, k));
-               if (fluidstate.m_76178_() && blockstate.m_60647_(p_77472_, blockpos$mutableblockpos.m_7495_(), PathComputationType.WATER) && blockstate.m_60795_()) {
+      for(long i = p_77473_; i < p_77473_ + p_77477_; ++i) {
+         for(long j = p_77474_; j < p_77474_ + p_77478_; ++j) {
+            for(long k = p_77475_; k < p_77475_ + p_77479_; ++k) {
+               FluidState fluidstate = p_77472_.m_6425_(blockpos$mutableblockpos.m_122169_(i, j, k));
+               BlockState blockstate = p_77472_.m_8055_(blockpos$mutableblockpos.m_122169_(i, j, k));
+               if (fluidstate.m_76178_() && blockstate.m_60647_(p_77472_, blockpos$mutableblockpos.below(), PathComputationType.WATER) && blockstate.m_60795_()) {
                   return BlockPathTypes.BREACH;
                }
 
