--- a/net/minecraft/world/level/pathfinder/Target.java
+++ b/net/minecraft/world/level/pathfinder/Target.java
@@ -11,7 +_,7 @@
       super(p_77502_.f_77271_, p_77502_.f_77272_, p_77502_.f_77273_);
    }
 
-   public Target(int p_77498_, int p_77499_, int p_77500_) {
+   public Target(long p_77498_, long p_77499_, long p_77500_) {
       super(p_77498_, p_77499_, p_77500_);
    }
 
@@ -36,7 +_,7 @@
    }
 
    public static Target m_77506_(FriendlyByteBuf p_77507_) {
-      Target target = new Target(p_77507_.readInt(), p_77507_.readInt(), p_77507_.readInt());
+      Target target = new Target(p_77507_.readLong(), p_77507_.readLong(), p_77507_.readLong());
       m_262841_(p_77507_, target);
       return target;
    }
