--- a/net/minecraft/world/level/pathfinder/Target.java
+++ b/net/minecraft/world/level/pathfinder/Target.java
@@ -11,8 +_,8 @@
       super(p_77502_.f_77271_, p_77502_.f_77272_, p_77502_.f_77273_);
    }
 
-   public Target(int p_77498_, int p_77499_, int p_77500_) {
-      super(p_77498_, p_77499_, p_77500_);
+   public Target(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      super(pX, pY, pZ);
    }
 
    public void m_77503_(float p_77504_, Node p_77505_) {
@@ -36,7 +_,7 @@
    }
 
    public static Target m_77506_(FriendlyByteBuf p_77507_) {
-      Target target = new Target(p_77507_.readInt(), p_77507_.readInt(), p_77507_.readInt());
+      Target target = new Target(new me.alphamode.mclong.math.BigInteger(p_77507_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77507_.m_130277_()), new me.alphamode.mclong.math.BigInteger(p_77507_.m_130277_()));
       m_262841_(p_77507_, target);
       return target;
    }
