--- a/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelectorParser.java
@@ -144,7 +_,7 @@
          };
       } else {
          function = (p_121258_) -> {
-            return new Vec3(this.f_121208_ == null ? p_121258_.f_82479_ : this.f_121208_, this.f_121209_ == null ? p_121258_.f_82480_ : this.f_121209_, this.f_121210_ == null ? p_121258_.f_82481_ : this.f_121210_);
+            return new Vec3(this.f_121208_ == null ? p_121258_.m_7096_() : this.f_121208_, this.f_121209_ == null ? p_121258_.m_7098_() : this.f_121209_, this.f_121210_ == null ? p_121258_.m_7094_() : this.f_121210_);
          };
       }
 
