--- a/net/minecraft/server/commands/ExecuteCommand.java
+++ b/net/minecraft/server/commands/ExecuteCommand.java
@@ -151,10 +_,10 @@
          ServerLevel serverlevel = p_274814_.getSource().m_81372_();
          double d0 = vec3.m_7096_();
          double d1 = vec3.m_7094_();
-         if (!serverlevel.m_7232_(SectionPos.m_235865_(d0), SectionPos.m_235865_(d1))) {
+         if (!serverlevel.hasChunk(SectionPos.blockToSectionCoord(d0), SectionPos.blockToSectionCoord(d1))) {
             throw BlockPosArgument.f_118234_.create();
          } else {
-            int i = serverlevel.m_6924_(HeightmapTypeArgument.m_274370_(p_274814_, "heightmap"), Mth.m_14107_(d0), Mth.m_14107_(d1));
+            long i = serverlevel.getHeight(HeightmapTypeArgument.m_274370_(p_274814_, "heightmap"), Mth.m_14107_(d0), Mth.m_14107_(d1));
             return p_274814_.getSource().m_81348_(new Vec3(d0, (double)i, d1));
          }
       })))).then(Commands.m_82127_("rotated").then(Commands.m_82129_("rot", RotationArgument.m_120479_()).redirect(literalcommandnode, (p_137291_) -> {
@@ -271,11 +_,11 @@
 
    private static boolean m_264155_(ServerLevel p_265261_, BlockPos p_265260_) {
       ChunkPos chunkpos = new ChunkPos(p_265260_);
-      LevelChunk levelchunk = p_265261_.m_7726_().m_7131_(chunkpos.f_45578_, chunkpos.f_45579_);
+      LevelChunk levelchunk = p_265261_.m_7726_().getChunkNow(chunkpos.f_45578_, chunkpos.f_45579_);
       if (levelchunk == null) {
          return false;
       } else {
-         return levelchunk.m_287138_() == FullChunkStatus.ENTITY_TICKING && p_265261_.m_143319_(chunkpos.m_45588_());
+         return levelchunk.m_287138_() == FullChunkStatus.ENTITY_TICKING && p_265261_.areEntitiesLoaded(chunkpos);
       }
    }
 
@@ -445,20 +_,20 @@
    }
 
    private static OptionalInt m_137036_(ServerLevel p_137037_, BlockPos p_137038_, BlockPos p_137039_, BlockPos p_137040_, boolean p_137041_) throws CommandSyntaxException {
-      BoundingBox boundingbox = BoundingBox.m_162375_(p_137038_, p_137039_);
-      BoundingBox boundingbox1 = BoundingBox.m_162375_(p_137040_, p_137040_.m_121955_(boundingbox.m_71053_()));
-      BlockPos blockpos = new BlockPos(boundingbox1.m_162395_() - boundingbox.m_162395_(), boundingbox1.m_162396_() - boundingbox.m_162396_(), boundingbox1.m_162398_() - boundingbox.m_162398_());
-      int i = boundingbox.m_71056_() * boundingbox.m_71057_() * boundingbox.m_71058_();
+      BoundingBox boundingbox = BoundingBox.fromCorners(p_137038_, p_137039_);
+      BoundingBox boundingbox1 = BoundingBox.fromCorners(p_137040_, p_137040_.offset(boundingbox.getLength()));
+      BlockPos blockpos = new BlockPos(boundingbox1.minX() - boundingbox.minX(), boundingbox1.minY() - boundingbox.minY(), boundingbox1.minZ() - boundingbox.minZ());
+      long i = boundingbox.getXSpan() * boundingbox.getYSpan() * boundingbox.getZSpan();
       if (i > 32768) {
          throw f_137030_.create(32768, i);
       } else {
          int j = 0;
 
-         for(int k = boundingbox.m_162398_(); k <= boundingbox.m_162401_(); ++k) {
-            for(int l = boundingbox.m_162396_(); l <= boundingbox.m_162400_(); ++l) {
-               for(int i1 = boundingbox.m_162395_(); i1 <= boundingbox.m_162399_(); ++i1) {
+         for(long k = boundingbox.minZ(); k <= boundingbox.maxZ(); ++k) {
+            for(long l = boundingbox.minY(); l <= boundingbox.maxY(); ++l) {
+               for(long i1 = boundingbox.minX(); i1 <= boundingbox.maxX(); ++i1) {
                   BlockPos blockpos1 = new BlockPos(i1, l, k);
-                  BlockPos blockpos2 = blockpos1.m_121955_(blockpos);
+                  BlockPos blockpos2 = blockpos1.offset(blockpos);
                   BlockState blockstate = p_137037_.m_8055_(blockpos1);
                   if (!p_137041_ || !blockstate.m_60713_(Blocks.f_50016_)) {
                      if (blockstate != p_137037_.m_8055_(blockpos2)) {
