--- a/net/minecraft/server/commands/GiveCommand.java
+++ b/net/minecraft/server/commands/GiveCommand.java
@@ -54,7 +_,7 @@
                      itementity1.m_32065_();
                   }
 
-                  serverplayer.m_9236_().m_6263_((Player)null, serverplayer.m_20185_(), serverplayer.m_20186_(), serverplayer.m_20189_(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, ((serverplayer.m_217043_().m_188501_() - serverplayer.m_217043_().m_188501_()) * 0.7F + 1.0F) * 2.0F);
+                  serverplayer.m_9236_().playSound((Player)null, serverplayer.getBigX(), serverplayer.getBigY(), serverplayer.getBigZ(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, ((serverplayer.m_217043_().m_188501_() - serverplayer.m_217043_().m_188501_()) * 0.7F + 1.0F) * 2.0F);
                   serverplayer.f_36096_.m_38946_();
                } else {
                   ItemEntity itementity = serverplayer.m_36176_(itemstack1, false);
