--- a/net/minecraft/server/commands/LocateCommand.java
+++ b/net/minecraft/server/commands/LocateCommand.java
@@ -140,10 +_,10 @@
 
    private static int m_262858_(CommandSourceStack p_262983_, BlockPos p_263016_, Pair<BlockPos, ? extends Holder<?>> p_262941_, String p_263083_, boolean p_263010_, String p_263048_, Duration p_263040_) {
       BlockPos blockpos = p_262941_.getFirst();
-      int i = p_263010_ ? Mth.m_14143_(Mth.m_14116_((float)p_263016_.m_123331_(blockpos))) : Mth.m_14143_(m_137853_(p_263016_.m_123341_(), p_263016_.m_123343_(), blockpos.m_123341_(), blockpos.m_123343_()));
-      String s = p_263010_ ? String.valueOf(blockpos.m_123342_()) : "~";
-      Component component = ComponentUtils.m_130748_(Component.m_237110_("chat.coordinates", blockpos.m_123341_(), s, blockpos.m_123343_())).m_130938_((p_214489_) -> {
-         return p_214489_.m_131140_(ChatFormatting.GREEN).m_131142_(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, "/tp @s " + blockpos.m_123341_() + " " + s + " " + blockpos.m_123343_())).m_131144_(new HoverEvent(HoverEvent.Action.f_130831_, Component.m_237115_("chat.coordinates.tooltip")));
+      int i = p_263010_ ? Mth.m_14143_(Mth.m_14116_((float)p_263016_.distSqr(blockpos))) : Mth.m_14143_(dist(p_263016_.getX(), p_263016_.getZ(), blockpos.getX(), blockpos.getZ()));
+      String s = p_263010_ ? String.valueOf(blockpos.getY()) : "~";
+      Component component = ComponentUtils.m_130748_(Component.m_237110_("chat.coordinates", blockpos.getX(), s, blockpos.getZ())).m_130938_((p_214489_) -> {
+         return p_214489_.m_131140_(ChatFormatting.GREEN).m_131142_(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, "/tp @s " + blockpos.getX() + " " + s + " " + blockpos.getZ())).m_131144_(new HoverEvent(HoverEvent.Action.f_130831_, Component.m_237115_("chat.coordinates.tooltip")));
       });
       p_262983_.m_288197_(() -> {
          return Component.m_237110_(p_263083_, p_263048_, component, i);
@@ -152,9 +_,9 @@
       return i;
    }
 
-   private static float m_137853_(int p_137854_, int p_137855_, int p_137856_, int p_137857_) {
-      int i = p_137856_ - p_137854_;
-      int j = p_137857_ - p_137855_;
+   private static float dist(long p_137854_, long p_137855_, long p_137856_, long p_137857_) {
+      long i = p_137856_ - p_137854_;
+      long j = p_137857_ - p_137855_;
       return Mth.m_14116_((float)(i * i + j * j));
    }
 }
