--- a/net/minecraft/server/commands/ParticleCommand.java
+++ b/net/minecraft/server/commands/ParticleCommand.java
@@ -45,7 +_,7 @@
       int i = 0;
 
       for(ServerPlayer serverplayer : p_138136_) {
-         if (p_138129_.m_81372_().m_8624_(serverplayer, p_138130_, p_138135_, p_138131_.f_82479_, p_138131_.f_82480_, p_138131_.f_82481_, p_138134_, p_138132_.f_82479_, p_138132_.f_82480_, p_138132_.f_82481_, (double)p_138133_)) {
+         if (p_138129_.m_81372_().sendParticles(serverplayer, p_138130_, p_138135_, p_138131_.getX(), p_138131_.getY(), p_138131_.getZ(), p_138134_, p_138132_.m_7096_(), p_138132_.m_7098_(), p_138132_.m_7094_(), (double)p_138133_)) {
             ++i;
          }
       }
