--- a/net/minecraft/server/commands/PlaceCommand.java
+++ b/net/minecraft/server/commands/PlaceCommand.java
@@ -94,7 +_,7 @@
          String s = p_214577_.m_203543_().map((p_214584_) -> {
             return p_214584_.m_135782_().toString();
          }).orElse("[unregistered]");
-         p_214576_.m_81354_(Component.m_237110_("commands.place.feature.success", s, p_214578_.m_123341_(), p_214578_.m_123342_(), p_214578_.m_123343_()), true);
+         p_214576_.m_81354_(Component.m_237110_("commands.place.feature.success", s, p_214578_.getX(), p_214578_.getY(), p_214578_.getZ()), true);
          return 1;
       }
    }
@@ -104,7 +_,7 @@
       if (!JigsawPlacement.m_227203_(serverlevel, p_214571_, p_214572_, p_214573_, p_214574_, false)) {
          throw f_214531_.create();
       } else {
-         p_214570_.m_81354_(Component.m_237110_("commands.place.jigsaw.success", p_214574_.m_123341_(), p_214574_.m_123342_(), p_214574_.m_123343_()), true);
+         p_214570_.m_81354_(Component.m_237110_("commands.place.jigsaw.success", p_214574_.getX(), p_214574_.getY(), p_214574_.getZ()), true);
          return 1;
       }
    }
@@ -120,16 +_,16 @@
          throw f_214532_.create();
       } else {
          BoundingBox boundingbox = structurestart.m_73601_();
-         ChunkPos chunkpos = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162395_()), SectionPos.m_123171_(boundingbox.m_162398_()));
-         ChunkPos chunkpos1 = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162399_()), SectionPos.m_123171_(boundingbox.m_162401_()));
+         ChunkPos chunkpos = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.minX()), SectionPos.blockToSectionCoord(boundingbox.minZ()));
+         ChunkPos chunkpos1 = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.maxX()), SectionPos.blockToSectionCoord(boundingbox.maxZ()));
          m_214543_(serverlevel, chunkpos, chunkpos1);
          ChunkPos.m_45599_(chunkpos, chunkpos1).forEach((p_214558_) -> {
-            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox(p_214558_.m_45604_(), serverlevel.m_141937_(), p_214558_.m_45605_(), p_214558_.m_45608_(), serverlevel.m_151558_(), p_214558_.m_45609_()), p_214558_);
+            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox((int) p_214558_.getMinBlockX(), (int) serverlevel.getMinBuildHeight(), (int) p_214558_.getMinBlockZ(), (int) p_214558_.getMaxBlockX(), (int) serverlevel.getMaxBuildHeight(), (int) p_214558_.getMaxBlockZ()), p_214558_);
          });
          String s = p_214589_.m_203543_().map((p_214539_) -> {
             return p_214539_.m_135782_().toString();
          }).orElse("[unregistered]");
-         p_214588_.m_81354_(Component.m_237110_("commands.place.structure.success", s, p_214590_.m_123341_(), p_214590_.m_123342_(), p_214590_.m_123343_()), true);
+         p_214588_.m_81354_(Component.m_237110_("commands.place.structure.success", s, p_214590_.getX(), p_214590_.getY(), p_214590_.getZ()), true);
          return 1;
       }
    }
@@ -159,7 +_,7 @@
          if (!flag) {
             throw f_214534_.create();
          } else {
-            p_214562_.m_81354_(Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.m_123341_(), p_214564_.m_123342_(), p_214564_.m_123343_()), true);
+            p_214562_.m_81354_(Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.getX(), p_214564_.getY(), p_214564_.getZ()), true);
             return 1;
          }
       }
