--- a/net/minecraft/server/commands/PlaceCommand.java
+++ b/net/minecraft/server/commands/PlaceCommand.java
@@ -57,19 +_,19 @@
       p_214548_.register(Commands.m_82127_("place").requires((p_214560_) -> {
          return p_214560_.m_6761_(2);
       }).then(Commands.m_82127_("feature").then(Commands.m_82129_("feature", ResourceKeyArgument.m_212386_(Registries.f_256911_)).executes((p_274824_) -> {
-         return m_214575_(p_274824_.getSource(), ResourceKeyArgument.m_245341_(p_274824_, "feature"), BlockPos.m_274446_(p_274824_.getSource().m_81371_()));
+         return m_214575_(p_274824_.getSource(), ResourceKeyArgument.m_245341_(p_274824_, "feature"), BlockPos.containing(p_274824_.getSource().m_81371_()));
       }).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_248163_) -> {
          return m_214575_(p_248163_.getSource(), ResourceKeyArgument.m_245341_(p_248163_, "feature"), BlockPosArgument.m_118242_(p_248163_, "pos"));
       })))).then(Commands.m_82127_("jigsaw").then(Commands.m_82129_("pool", ResourceKeyArgument.m_212386_(Registries.f_256948_)).then(Commands.m_82129_("target", ResourceLocationArgument.m_106984_()).then(Commands.m_82129_("max_depth", IntegerArgumentType.integer(1, 7)).executes((p_274825_) -> {
-         return m_214569_(p_274825_.getSource(), ResourceKeyArgument.m_247435_(p_274825_, "pool"), ResourceLocationArgument.m_107011_(p_274825_, "target"), IntegerArgumentType.getInteger(p_274825_, "max_depth"), BlockPos.m_274446_(p_274825_.getSource().m_81371_()));
+         return m_214569_(p_274825_.getSource(), ResourceKeyArgument.m_247435_(p_274825_, "pool"), ResourceLocationArgument.m_107011_(p_274825_, "target"), IntegerArgumentType.getInteger(p_274825_, "max_depth"), BlockPos.containing(p_274825_.getSource().m_81371_()));
       }).then(Commands.m_82129_("position", BlockPosArgument.m_118239_()).executes((p_248167_) -> {
          return m_214569_(p_248167_.getSource(), ResourceKeyArgument.m_247435_(p_248167_, "pool"), ResourceLocationArgument.m_107011_(p_248167_, "target"), IntegerArgumentType.getInteger(p_248167_, "max_depth"), BlockPosArgument.m_118242_(p_248167_, "position"));
       })))))).then(Commands.m_82127_("structure").then(Commands.m_82129_("structure", ResourceKeyArgument.m_212386_(Registries.f_256944_)).executes((p_274826_) -> {
-         return m_214587_(p_274826_.getSource(), ResourceKeyArgument.m_247318_(p_274826_, "structure"), BlockPos.m_274446_(p_274826_.getSource().m_81371_()));
+         return m_214587_(p_274826_.getSource(), ResourceKeyArgument.m_247318_(p_274826_, "structure"), BlockPos.containing(p_274826_.getSource().m_81371_()));
       }).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_248168_) -> {
          return m_214587_(p_248168_.getSource(), ResourceKeyArgument.m_247318_(p_248168_, "structure"), BlockPosArgument.m_118242_(p_248168_, "pos"));
       })))).then(Commands.m_82127_("template").then(Commands.m_82129_("template", ResourceLocationArgument.m_106984_()).suggests(f_214535_).executes((p_274827_) -> {
-         return m_214561_(p_274827_.getSource(), ResourceLocationArgument.m_107011_(p_274827_, "template"), BlockPos.m_274446_(p_274827_.getSource().m_81371_()), Rotation.NONE, Mirror.NONE, 1.0F, 0);
+         return m_214561_(p_274827_.getSource(), ResourceLocationArgument.m_107011_(p_274827_, "template"), BlockPos.containing(p_274827_.getSource().m_81371_()), Rotation.NONE, Mirror.NONE, 1.0F, 0);
       }).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_214596_) -> {
          return m_214561_(p_214596_.getSource(), ResourceLocationArgument.m_107011_(p_214596_, "template"), BlockPosArgument.m_118242_(p_214596_, "pos"), Rotation.NONE, Mirror.NONE, 1.0F, 0);
       }).then(Commands.m_82129_("rotation", TemplateRotationArgument.m_234414_()).executes((p_214594_) -> {
@@ -87,13 +_,13 @@
       ServerLevel serverlevel = p_214576_.m_81372_();
       ConfiguredFeature<?, ?> configuredfeature = p_248822_.m_203334_();
       ChunkPos chunkpos = new ChunkPos(p_214578_);
-      m_214543_(serverlevel, new ChunkPos(chunkpos.f_45578_ - 1, chunkpos.f_45579_ - 1), new ChunkPos(chunkpos.f_45578_ + 1, chunkpos.f_45579_ + 1));
+      m_214543_(serverlevel, new ChunkPos(chunkpos.getX().subtract(), chunkpos.getZ().subtract()), new ChunkPos(chunkpos.getX().add(), chunkpos.getZ().add()));
       if (!configuredfeature.m_224953_(serverlevel, serverlevel.m_7726_().m_8481_(), serverlevel.m_213780_(), p_214578_)) {
          throw f_214530_.create();
       } else {
          String s = p_248822_.m_205785_().m_135782_().toString();
          p_214576_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.feature.success", s, p_214578_.m_123341_(), p_214578_.m_123342_(), p_214578_.m_123343_());
+            return Component.m_237110_("commands.place.feature.success", s, p_214578_.getX(), p_214578_.getY(), p_214578_.getZ());
          }, true);
          return 1;
       }
@@ -105,7 +_,7 @@
          throw f_214531_.create();
       } else {
          p_214570_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.jigsaw.success", p_214574_.m_123341_(), p_214574_.m_123342_(), p_214574_.m_123343_());
+            return Component.m_237110_("commands.place.jigsaw.success", p_214574_.getX(), p_214574_.getY(), p_214574_.getZ());
          }, true);
          return 1;
       }
@@ -122,15 +_,15 @@
          throw f_214532_.create();
       } else {
          BoundingBox boundingbox = structurestart.m_73601_();
-         ChunkPos chunkpos = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162395_()), SectionPos.m_123171_(boundingbox.m_162398_()));
-         ChunkPos chunkpos1 = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162399_()), SectionPos.m_123171_(boundingbox.m_162401_()));
+         ChunkPos chunkpos = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.minX()), SectionPos.blockToSectionCoord(boundingbox.minZ()));
+         ChunkPos chunkpos1 = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.maxX()), SectionPos.blockToSectionCoord(boundingbox.maxZ()));
          m_214543_(serverlevel, chunkpos, chunkpos1);
          ChunkPos.m_45599_(chunkpos, chunkpos1).forEach((p_289290_) -> {
-            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox(p_289290_.m_45604_(), serverlevel.m_141937_(), p_289290_.m_45605_(), p_289290_.m_45608_(), serverlevel.m_151558_(), p_289290_.m_45609_()), p_289290_);
+            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox(p_289290_.getMinBlockX(), serverlevel.getMinBuildHeight(), p_289290_.getMinBlockZ(), p_289290_.getMaxBlockX(), serverlevel.getMaxBuildHeight(), p_289290_.getMaxBlockZ()), p_289290_);
          });
          String s = p_251799_.m_205785_().m_135782_().toString();
          p_214588_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.structure.success", s, p_214590_.m_123341_(), p_214590_.m_123342_(), p_214590_.m_123343_());
+            return Component.m_237110_("commands.place.structure.success", s, p_214590_.getX(), p_214590_.getY(), p_214590_.getZ());
          }, true);
          return 1;
       }
@@ -151,7 +_,7 @@
          throw f_214533_.create(p_214563_);
       } else {
          StructureTemplate structuretemplate = optional.get();
-         m_214543_(serverlevel, new ChunkPos(p_214564_), new ChunkPos(p_214564_.m_121955_(structuretemplate.m_163801_())));
+         m_214543_(serverlevel, new ChunkPos(p_214564_), new ChunkPos(p_214564_.offset(structuretemplate.getSize())));
          StructurePlaceSettings structureplacesettings = (new StructurePlaceSettings()).m_74377_(p_214566_).m_74379_(p_214565_);
          if (p_214567_ < 1.0F) {
             structureplacesettings.m_74394_().m_74383_(new BlockRotProcessor(p_214567_)).m_230324_(StructureBlockEntity.m_222888_((long)p_214568_));
@@ -162,7 +_,7 @@
             throw f_214534_.create();
          } else {
             p_214562_.m_288197_(() -> {
-               return Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.m_123341_(), p_214564_.m_123342_(), p_214564_.m_123343_());
+               return Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.getX(), p_214564_.getY(), p_214564_.getZ());
             }, true);
             return 1;
          }
