--- a/net/minecraft/server/commands/PlaceCommand.java
+++ b/net/minecraft/server/commands/PlaceCommand.java
@@ -87,13 +_,13 @@
       ServerLevel serverlevel = p_214576_.m_81372_();
       ConfiguredFeature<?, ?> configuredfeature = p_248822_.m_203334_();
       ChunkPos chunkpos = new ChunkPos(p_214578_);
-      m_214543_(serverlevel, new ChunkPos(chunkpos.f_45578_ - 1, chunkpos.f_45579_ - 1), new ChunkPos(chunkpos.f_45578_ + 1, chunkpos.f_45579_ + 1));
+      m_214543_(serverlevel, new ChunkPos(chunkpos.f_45578_ - 1, chunkpos.y - 1, chunkpos.f_45579_ - 1), new ChunkPos(chunkpos.f_45578_ + 1, chunkpos.y + 1, chunkpos.f_45579_ + 1));
       if (!configuredfeature.m_224953_(serverlevel, serverlevel.m_7726_().m_8481_(), serverlevel.m_213780_(), p_214578_)) {
          throw f_214530_.create();
       } else {
          String s = p_248822_.m_205785_().m_135782_().toString();
          p_214576_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.feature.success", s, p_214578_.m_123341_(), p_214578_.m_123342_(), p_214578_.m_123343_());
+            return Component.m_237110_("commands.place.feature.success", s, p_214578_.getX(), p_214578_.getY(), p_214578_.getZ());
          }, true);
          return 1;
       }
@@ -105,7 +_,7 @@
          throw f_214531_.create();
       } else {
          p_214570_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.jigsaw.success", p_214574_.m_123341_(), p_214574_.m_123342_(), p_214574_.m_123343_());
+            return Component.m_237110_("commands.place.jigsaw.success", p_214574_.getX(), p_214574_.getY(), p_214574_.getZ());
          }, true);
          return 1;
       }
@@ -122,15 +_,15 @@
          throw f_214532_.create();
       } else {
          BoundingBox boundingbox = structurestart.m_73601_();
-         ChunkPos chunkpos = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162395_()), SectionPos.m_123171_(boundingbox.m_162398_()));
-         ChunkPos chunkpos1 = new ChunkPos(SectionPos.m_123171_(boundingbox.m_162399_()), SectionPos.m_123171_(boundingbox.m_162401_()));
+         ChunkPos chunkpos = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.minX()), SectionPos.blockToSectionCoord(boundingbox.minY()), SectionPos.blockToSectionCoord(boundingbox.minZ()));
+         ChunkPos chunkpos1 = new ChunkPos(SectionPos.blockToSectionCoord(boundingbox.maxX()), SectionPos.blockToSectionCoord(boundingbox.maxY()), SectionPos.blockToSectionCoord(boundingbox.maxZ()));
          m_214543_(serverlevel, chunkpos, chunkpos1);
          ChunkPos.m_45599_(chunkpos, chunkpos1).forEach((p_289290_) -> {
-            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox(p_289290_.m_45604_(), serverlevel.m_141937_(), p_289290_.m_45605_(), p_289290_.m_45608_(), serverlevel.m_151558_(), p_289290_.m_45609_()), p_289290_);
+            structurestart.m_226850_(serverlevel, serverlevel.m_215010_(), chunkgenerator, serverlevel.m_213780_(), new BoundingBox(p_289290_.getMinBlockX(), serverlevel.getMinBuildHeight(), p_289290_.getMinBlockZ(), p_289290_.getMaxBlockX(), serverlevel.getMaxBuildHeight(), p_289290_.getMaxBlockZ()), p_289290_);
          });
          String s = p_251799_.m_205785_().m_135782_().toString();
          p_214588_.m_288197_(() -> {
-            return Component.m_237110_("commands.place.structure.success", s, p_214590_.m_123341_(), p_214590_.m_123342_(), p_214590_.m_123343_());
+            return Component.m_237110_("commands.place.structure.success", s, p_214590_.getX(), p_214590_.getY(), p_214590_.getZ());
          }, true);
          return 1;
       }
@@ -151,7 +_,7 @@
          throw f_214533_.create(p_214563_);
       } else {
          StructureTemplate structuretemplate = optional.get();
-         m_214543_(serverlevel, new ChunkPos(p_214564_), new ChunkPos(p_214564_.m_121955_(structuretemplate.m_163801_())));
+         m_214543_(serverlevel, new ChunkPos(p_214564_), new ChunkPos(p_214564_.offset(structuretemplate.getSize())));
          StructurePlaceSettings structureplacesettings = (new StructurePlaceSettings()).m_74377_(p_214566_).m_74379_(p_214565_);
          if (p_214567_ < 1.0F) {
             structureplacesettings.m_74394_().m_74383_(new BlockRotProcessor(p_214567_)).m_230324_(StructureBlockEntity.m_222888_((long)p_214568_));
@@ -162,7 +_,7 @@
             throw f_214534_.create();
          } else {
             p_214562_.m_288197_(() -> {
-               return Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.m_123341_(), p_214564_.m_123342_(), p_214564_.m_123343_());
+               return Component.m_237110_("commands.place.template.success", p_214563_, p_214564_.getX(), p_214564_.getY(), p_214564_.getZ());
             }, true);
             return 1;
          }
