--- a/net/minecraft/server/commands/PlaySoundCommand.java
+++ b/net/minecraft/server/commands/PlaySoundCommand.java
@@ -59,9 +_,9 @@
       long j = p_138161_.m_81372_().m_213780_().m_188505_();
 
       for(ServerPlayer serverplayer : p_138162_) {
-         double d1 = p_138165_.f_82479_ - serverplayer.m_20185_();
-         double d2 = p_138165_.f_82480_ - serverplayer.m_20186_();
-         double d3 = p_138165_.f_82481_ - serverplayer.m_20189_();
+         double d1 = p_138165_.m_7096_() - serverplayer.m_20185_();
+         double d2 = p_138165_.m_7098_() - serverplayer.m_20186_();
+         double d3 = p_138165_.m_7094_() - serverplayer.m_20189_();
          double d4 = d1 * d1 + d2 * d2 + d3 * d3;
          Vec3 vec3 = p_138165_;
          float f = p_138166_;
@@ -75,7 +_,7 @@
             f = p_138168_;
          }
 
-         serverplayer.f_8906_.m_9829_(new ClientboundSoundPacket(holder, p_138164_, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), f, p_138167_, j));
+         serverplayer.f_8906_.m_9829_(new ClientboundSoundPacket(holder, p_138164_, vec3.getX(), vec3.getY(), vec3.getZ(), f, p_138167_, j));
          ++i;
       }
 
