--- a/net/minecraft/server/commands/RaidCommand.java
+++ b/net/minecraft/server/commands/RaidCommand.java
@@ -89,8 +_,8 @@
       } else {
          raider.m_33075_(true);
          raider.m_8061_(EquipmentSlot.HEAD, Raid.m_37779_());
-         raider.m_6034_(p_180483_.m_81371_().f_82479_, p_180483_.m_81371_().f_82480_, p_180483_.m_81371_().f_82481_);
-         raider.m_6518_(p_180483_.m_81372_(), p_180483_.m_81372_().m_6436_(BlockPos.m_274446_(p_180483_.m_81371_())), MobSpawnType.COMMAND, (SpawnGroupData)null, (CompoundTag)null);
+         raider.m_6034_(p_180483_.m_81371_().m_7096_(), p_180483_.m_81371_().m_7098_(), p_180483_.m_81371_().m_7094_());
+         raider.m_6518_(p_180483_.m_81372_(), p_180483_.m_81372_().m_6436_(BlockPos.containing(p_180483_.m_81371_())), MobSpawnType.COMMAND, (SpawnGroupData)null, (CompoundTag)null);
          p_180483_.m_81372_().m_47205_(raider);
          return 1;
       }
@@ -100,7 +_,7 @@
       if (p_180479_ != null && p_180479_.getString().equals("local")) {
          ServerLevel serverlevel = p_180478_.m_81372_();
          Vec3 vec3 = p_180478_.m_81371_().m_82520_(5.0D, 0.0D, 0.0D);
-         serverlevel.m_262808_((Player)null, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, SoundEvents.f_12355_, SoundSource.NEUTRAL, 2.0F, 1.0F, serverlevel.f_46441_.m_188505_());
+         serverlevel.m_262808_((Player)null, vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_(), SoundEvents.f_12355_, SoundSource.NEUTRAL, 2.0F, 1.0F, serverlevel.f_46441_.m_188505_());
       }
 
       return 1;
