--- a/net/minecraft/server/commands/SetBlockCommand.java
+++ b/net/minecraft/server/commands/SetBlockCommand.java
@@ -57,7 +_,7 @@
             throw f_138597_.create();
          } else {
             serverlevel.m_6289_(p_138609_, p_138610_.m_114669_().m_60734_());
-            p_138608_.m_81354_(Component.m_237110_("commands.setblock.success", p_138609_.m_123341_(), p_138609_.m_123342_(), p_138609_.m_123343_()), true);
+            p_138608_.m_81354_(Component.m_237110_("commands.setblock.success", p_138609_.getX(), p_138609_.getY(), p_138609_.getZ()), true);
             return 1;
          }
       }
