--- a/net/minecraft/server/commands/SetSpawnCommand.java
+++ b/net/minecraft/server/commands/SetSpawnCommand.java
@@ -19,9 +_,9 @@
       p_138644_.register(Commands.m_82127_("spawnpoint").requires((p_138648_) -> {
          return p_138648_.m_6761_(2);
       }).executes((p_274828_) -> {
-         return m_138649_(p_274828_.getSource(), Collections.singleton(p_274828_.getSource().m_81375_()), BlockPos.m_274446_(p_274828_.getSource().m_81371_()), 0.0F);
+         return m_138649_(p_274828_.getSource(), Collections.singleton(p_274828_.getSource().m_81375_()), BlockPos.containing(p_274828_.getSource().m_81371_()), 0.0F);
       }).then(Commands.m_82129_("targets", EntityArgument.m_91470_()).executes((p_274829_) -> {
-         return m_138649_(p_274829_.getSource(), EntityArgument.m_91477_(p_274829_, "targets"), BlockPos.m_274446_(p_274829_.getSource().m_81371_()), 0.0F);
+         return m_138649_(p_274829_.getSource(), EntityArgument.m_91477_(p_274829_, "targets"), BlockPos.containing(p_274829_.getSource().m_81371_()), 0.0F);
       }).then(Commands.m_82129_("pos", BlockPosArgument.m_118239_()).executes((p_138655_) -> {
          return m_138649_(p_138655_.getSource(), EntityArgument.m_91477_(p_138655_, "targets"), BlockPosArgument.m_174395_(p_138655_, "pos"), 0.0F);
       }).then(Commands.m_82129_("angle", AngleArgument.m_83807_()).executes((p_138646_) -> {
@@ -39,11 +_,11 @@
       String s = resourcekey.m_135782_().toString();
       if (p_138651_.size() == 1) {
          p_138650_.m_288197_(() -> {
-            return Component.m_237110_("commands.spawnpoint.success.single", p_138652_.m_123341_(), p_138652_.m_123342_(), p_138652_.m_123343_(), p_138653_, s, p_138651_.iterator().next().m_5446_());
+            return Component.m_237110_("commands.spawnpoint.success.single", p_138652_.getX(), p_138652_.getY(), p_138652_.getZ(), p_138653_, s, p_138651_.iterator().next().m_5446_());
          }, true);
       } else {
          p_138650_.m_288197_(() -> {
-            return Component.m_237110_("commands.spawnpoint.success.multiple", p_138652_.m_123341_(), p_138652_.m_123342_(), p_138652_.m_123343_(), p_138653_, s, p_138651_.size());
+            return Component.m_237110_("commands.spawnpoint.success.multiple", p_138652_.getX(), p_138652_.getY(), p_138652_.getZ(), p_138653_, s, p_138651_.size());
          }, true);
       }
 
