--- a/net/minecraft/server/commands/SetSpawnCommand.java
+++ b/net/minecraft/server/commands/SetSpawnCommand.java
@@ -38,9 +_,9 @@
 
       String s = resourcekey.m_135782_().toString();
       if (p_138651_.size() == 1) {
-         p_138650_.m_81354_(Component.m_237110_("commands.spawnpoint.success.single", p_138652_.m_123341_(), p_138652_.m_123342_(), p_138652_.m_123343_(), p_138653_, s, p_138651_.iterator().next().m_5446_()), true);
+         p_138650_.m_81354_(Component.m_237110_("commands.spawnpoint.success.single", p_138652_.getX(), p_138652_.getY(), p_138652_.getZ(), p_138653_, s, p_138651_.iterator().next().m_5446_()), true);
       } else {
-         p_138650_.m_81354_(Component.m_237110_("commands.spawnpoint.success.multiple", p_138652_.m_123341_(), p_138652_.m_123342_(), p_138652_.m_123343_(), p_138653_, s, p_138651_.size()), true);
+         p_138650_.m_81354_(Component.m_237110_("commands.spawnpoint.success.multiple", p_138652_.getX(), p_138652_.getY(), p_138652_.getZ(), p_138653_, s, p_138651_.size()), true);
       }
 
       return p_138651_.size();
