--- a/net/minecraft/server/commands/SetWorldSpawnCommand.java
+++ b/net/minecraft/server/commands/SetWorldSpawnCommand.java
@@ -23,7 +_,7 @@
 
    private static int m_138666_(CommandSourceStack p_138667_, BlockPos p_138668_, float p_138669_) {
       p_138667_.m_81372_().m_8733_(p_138668_, p_138669_);
-      p_138667_.m_81354_(Component.m_237110_("commands.setworldspawn.success", p_138668_.m_123341_(), p_138668_.m_123342_(), p_138668_.m_123343_(), p_138669_), true);
+      p_138667_.m_81354_(Component.m_237110_("commands.setworldspawn.success", p_138668_.getX(), p_138668_.getY(), p_138668_.getZ(), p_138669_), true);
       return 1;
    }
 }
