--- a/net/minecraft/world/level/levelgen/feature/ChorusPlantFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ChorusPlantFeature.java
@@ -17,7 +_,7 @@
       WorldGenLevel worldgenlevel = p_159521_.m_159774_();
       BlockPos blockpos = p_159521_.m_159777_();
       RandomSource randomsource = p_159521_.m_225041_();
-      if (worldgenlevel.m_46859_(blockpos) && worldgenlevel.m_8055_(blockpos.m_7495_()).m_60713_(Blocks.f_50259_)) {
+      if (worldgenlevel.m_46859_(blockpos) && worldgenlevel.m_8055_(blockpos.below()).m_60713_(Blocks.f_50259_)) {
          ChorusFlowerBlock.m_220962_(worldgenlevel, blockpos, randomsource, 8);
          return true;
       } else {
