--- a/net/minecraft/world/level/levelgen/feature/CoralTreeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/CoralTreeFeature.java
@@ -31,7 +_,7 @@
       List<Direction> list = Direction.Plane.HORIZONTAL.m_235694_(p_224988_);
 
       for(Direction direction : list.subList(0, k)) {
-         blockpos$mutableblockpos.m_122190_(blockpos);
+         blockpos$mutableblockpos.set(blockpos);
          blockpos$mutableblockpos.m_122173_(direction);
          int l = p_224988_.m_188503_(5) + 2;
          int i1 = 0;
