--- a/net/minecraft/world/level/levelgen/feature/DripstoneUtils.java
+++ b/net/minecraft/world/level/levelgen/feature/DripstoneUtils.java
@@ -38,7 +_,7 @@
          for(float f2 = 0.0F; f2 < ((float)Math.PI * 2F); f2 += f1) {
             int i = (int)(Mth.m_14089_(f2) * (float)p_159642_);
             int j = (int)(Mth.m_14031_(f2) * (float)p_159642_);
-            if (m_159659_(p_159640_, p_159641_.m_7918_(i, 0, j))) {
+            if (m_159659_(p_159640_, p_159641_.offset(i, 0, j))) {
                return false;
             }
          }
@@ -55,35 +_,35 @@
       return p_159660_.m_7433_(p_159661_, DripstoneUtils::m_159666_);
    }
 
-   protected static void m_159651_(Direction p_159652_, int p_159653_, boolean p_159654_, Consumer<BlockState> p_159655_) {
-      if (p_159653_ >= 3) {
-         p_159655_.accept(m_159656_(p_159652_, DripstoneThickness.BASE));
+   protected static void buildBaseToTipColumn(Direction pDirection, long pHeight, boolean pMergeTip, Consumer<BlockState> pBlockSetter) {
+      if (pHeight >= 3) {
+         pBlockSetter.accept(m_159656_(pDirection, DripstoneThickness.BASE));
 
-         for(int i = 0; i < p_159653_ - 3; ++i) {
-            p_159655_.accept(m_159656_(p_159652_, DripstoneThickness.MIDDLE));
+         for(int i = 0; i < pHeight - 3; ++i) {
+            pBlockSetter.accept(m_159656_(pDirection, DripstoneThickness.MIDDLE));
          }
       }
 
-      if (p_159653_ >= 2) {
-         p_159655_.accept(m_159656_(p_159652_, DripstoneThickness.FRUSTUM));
+      if (pHeight >= 2) {
+         pBlockSetter.accept(m_159656_(pDirection, DripstoneThickness.FRUSTUM));
       }
 
-      if (p_159653_ >= 1) {
-         p_159655_.accept(m_159656_(p_159652_, p_159654_ ? DripstoneThickness.TIP_MERGE : DripstoneThickness.TIP));
+      if (pHeight >= 1) {
+         pBlockSetter.accept(m_159656_(pDirection, pMergeTip ? DripstoneThickness.TIP_MERGE : DripstoneThickness.TIP));
       }
 
    }
 
-   protected static void m_190847_(LevelAccessor p_190848_, BlockPos p_190849_, Direction p_190850_, int p_190851_, boolean p_190852_) {
-      if (m_159662_(p_190848_.m_8055_(p_190849_.m_121945_(p_190850_.m_122424_())))) {
-         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_190849_.m_122032_();
-         m_159651_(p_190850_, p_190851_, p_190852_, (p_277326_) -> {
+   protected static void growPointedDripstone(LevelAccessor pLevel, BlockPos pPos, Direction pDirection, long pHeight, boolean pMergeTip) {
+      if (m_159662_(pLevel.m_8055_(pPos.relative(pDirection.m_122424_())))) {
+         BlockPos.MutableBlockPos blockpos$mutableblockpos = pPos.m_122032_();
+         buildBaseToTipColumn(pDirection, pHeight, pMergeTip, (p_277326_) -> {
             if (p_277326_.m_60713_(Blocks.f_152588_)) {
-               p_277326_ = p_277326_.m_61124_(PointedDripstoneBlock.f_154011_, Boolean.valueOf(p_190848_.m_46801_(blockpos$mutableblockpos)));
+               p_277326_ = p_277326_.m_61124_(PointedDripstoneBlock.f_154011_, Boolean.valueOf(pLevel.m_46801_(blockpos$mutableblockpos)));
             }
 
-            p_190848_.m_7731_(blockpos$mutableblockpos, p_277326_, 2);
-            blockpos$mutableblockpos.m_122173_(p_190850_);
+            pLevel.m_7731_(blockpos$mutableblockpos, p_277326_, 2);
+            blockpos$mutableblockpos.m_122173_(pDirection);
          });
       }
    }
