--- a/net/minecraft/world/level/levelgen/feature/EndGatewayFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/EndGatewayFeature.java
@@ -19,10 +_,10 @@
       EndGatewayConfiguration endgatewayconfiguration = p_159715_.m_159778_();
 
       for(BlockPos blockpos1 : BlockPos.m_121940_(blockpos.m_7918_(-1, -2, -1), blockpos.m_7918_(1, 2, 1))) {
-         boolean flag = blockpos1.m_123341_() == blockpos.m_123341_();
-         boolean flag1 = blockpos1.m_123342_() == blockpos.m_123342_();
-         boolean flag2 = blockpos1.m_123343_() == blockpos.m_123343_();
-         boolean flag3 = Math.abs(blockpos1.m_123342_() - blockpos.m_123342_()) == 2;
+         boolean flag = blockpos1.getX() == blockpos.getX();
+         boolean flag1 = blockpos1.getY() == blockpos.getY();
+         boolean flag2 = blockpos1.getZ() == blockpos.getZ();
+         boolean flag3 = Math.abs(blockpos1.getY() - blockpos.getY()) == 2;
          if (flag && flag1 && flag2) {
             BlockPos blockpos2 = blockpos1.m_7949_();
             this.m_5974_(worldgenlevel, blockpos2, Blocks.f_50446_.m_49966_());
