--- a/net/minecraft/world/level/levelgen/feature/EndIslandFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/EndIslandFeature.java
@@ -23,7 +_,7 @@
          for(int j = Mth.m_14143_(-f); j <= Mth.m_14167_(f); ++j) {
             for(int k = Mth.m_14143_(-f); k <= Mth.m_14167_(f); ++k) {
                if ((float)(j * j + k * k) <= (f + 1.0F) * (f + 1.0F)) {
-                  this.m_5974_(worldgenlevel, blockpos.m_7918_(j, i, k), Blocks.f_50259_.m_49966_());
+                  this.m_5974_(worldgenlevel, blockpos.offset(j, i, k), Blocks.f_50259_.m_49966_());
                }
             }
          }
