--- a/net/minecraft/world/level/levelgen/feature/Feature.java
+++ b/net/minecraft/world/level/levelgen/feature/Feature.java
@@ -170,7 +_,7 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(Direction direction : Direction.values()) {
-         blockpos$mutableblockpos.m_122159_(p_159755_, direction);
+         blockpos$mutableblockpos.setWithOffset(p_159755_, direction);
          if (p_159756_.test(p_159754_.apply(blockpos$mutableblockpos))) {
             return true;
          }
