--- a/net/minecraft/world/level/levelgen/feature/FillLayerFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/FillLayerFeature.java
@@ -18,10 +_,10 @@
 
       for(int i = 0; i < 16; ++i) {
          for(int j = 0; j < 16; ++j) {
-            int k = blockpos.m_123341_() + i;
-            int l = blockpos.m_123343_() + j;
-            int i1 = worldgenlevel.m_141937_() + layerconfiguration.f_67768_;
-            blockpos$mutableblockpos.m_122178_(k, i1, l);
+            long k = blockpos.getX() + i;
+            long l = blockpos.getZ() + j;
+            long i1 = worldgenlevel.getMinBuildHeight() + layerconfiguration.f_67768_;
+            blockpos$mutableblockpos.m_122169_(k, i1, l);
             if (worldgenlevel.m_8055_(blockpos$mutableblockpos).m_60795_()) {
                worldgenlevel.m_7731_(blockpos$mutableblockpos, layerconfiguration.f_67769_, 2);
             }
