--- a/net/minecraft/world/level/levelgen/feature/GlowstoneFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/GlowstoneFeature.java
@@ -21,7 +_,7 @@
       if (!worldgenlevel.m_46859_(blockpos)) {
          return false;
       } else {
-         BlockState blockstate = worldgenlevel.m_8055_(blockpos.m_7494_());
+         BlockState blockstate = worldgenlevel.m_8055_(blockpos.above());
          if (!blockstate.m_60713_(Blocks.f_50134_) && !blockstate.m_60713_(Blocks.f_50137_) && !blockstate.m_60713_(Blocks.f_50730_)) {
             return false;
          } else {
@@ -33,7 +_,7 @@
                   int j = 0;
 
                   for(Direction direction : Direction.values()) {
-                     if (worldgenlevel.m_8055_(blockpos1.m_121945_(direction)).m_60713_(Blocks.f_50141_)) {
+                     if (worldgenlevel.m_8055_(blockpos1.relative(direction)).m_60713_(Blocks.f_50141_)) {
                         ++j;
                      }
 
