--- a/net/minecraft/world/level/levelgen/feature/HugeBrownMushroomFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/HugeBrownMushroomFeature.java
@@ -25,7 +_,7 @@
             boolean flag4 = flag || flag1;
             boolean flag5 = flag2 || flag3;
             if (!flag4 || !flag5) {
-               p_225047_.m_122154_(p_225045_, j, p_225046_, k);
+               p_225047_.setWithOffset(p_225045_, j, p_225046_, k);
                if (!p_225043_.m_8055_(p_225047_).m_60804_(p_225043_, p_225047_)) {
                   boolean flag6 = flag || flag5 && j == 1 - i;
                   boolean flag7 = flag1 || flag5 && j == i - 1;
