--- a/net/minecraft/world/level/levelgen/feature/IceSpikeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/IceSpikeFeature.java
@@ -19,17 +_,17 @@
       RandomSource randomsource = p_159882_.m_225041_();
 
       WorldGenLevel worldgenlevel;
-      for(worldgenlevel = p_159882_.m_159774_(); worldgenlevel.m_46859_(blockpos) && blockpos.m_123342_() > worldgenlevel.m_141937_() + 2; blockpos = blockpos.m_7495_()) {
+      for(worldgenlevel = p_159882_.m_159774_(); worldgenlevel.m_46859_(blockpos) && blockpos.getY() > worldgenlevel.getMinBuildHeight() + 2; blockpos = blockpos.below()) {
       }
 
       if (!worldgenlevel.m_8055_(blockpos).m_60713_(Blocks.f_50127_)) {
          return false;
       } else {
-         blockpos = blockpos.m_6630_(randomsource.m_188503_(4));
+         blockpos = blockpos.above(randomsource.m_188503_(4));
          int i = randomsource.m_188503_(4) + 7;
          int j = i / 4 + randomsource.m_188503_(2);
          if (j > 1 && randomsource.m_188503_(60) == 0) {
-            blockpos = blockpos.m_6630_(10 + randomsource.m_188503_(30));
+            blockpos = blockpos.above(10 + randomsource.m_188503_(30));
          }
 
          for(int k = 0; k < i; ++k) {
@@ -42,15 +_,15 @@
                for(int j1 = -l; j1 <= l; ++j1) {
                   float f2 = (float)Mth.m_14040_(j1) - 0.25F;
                   if ((i1 == 0 && j1 == 0 || !(f1 * f1 + f2 * f2 > f * f)) && (i1 != -l && i1 != l && j1 != -l && j1 != l || !(randomsource.m_188501_() > 0.75F))) {
-                     BlockState blockstate = worldgenlevel.m_8055_(blockpos.m_7918_(i1, k, j1));
+                     BlockState blockstate = worldgenlevel.m_8055_(blockpos.offset(i1, k, j1));
                      if (blockstate.m_60795_() || m_159759_(blockstate) || blockstate.m_60713_(Blocks.f_50127_) || blockstate.m_60713_(Blocks.f_50126_)) {
-                        this.m_5974_(worldgenlevel, blockpos.m_7918_(i1, k, j1), Blocks.f_50354_.m_49966_());
+                        this.m_5974_(worldgenlevel, blockpos.offset(i1, k, j1), Blocks.f_50354_.m_49966_());
                      }
 
                      if (k != 0 && l > 1) {
-                        blockstate = worldgenlevel.m_8055_(blockpos.m_7918_(i1, -k, j1));
+                        blockstate = worldgenlevel.m_8055_(blockpos.offset(i1, -k, j1));
                         if (blockstate.m_60795_() || m_159759_(blockstate) || blockstate.m_60713_(Blocks.f_50127_) || blockstate.m_60713_(Blocks.f_50126_)) {
-                           this.m_5974_(worldgenlevel, blockpos.m_7918_(i1, -k, j1), Blocks.f_50354_.m_49966_());
+                           this.m_5974_(worldgenlevel, blockpos.offset(i1, -k, j1), Blocks.f_50354_.m_49966_());
                         }
                      }
                   }
@@ -67,23 +_,23 @@
 
          for(int l1 = -k1; l1 <= k1; ++l1) {
             for(int i2 = -k1; i2 <= k1; ++i2) {
-               BlockPos blockpos1 = blockpos.m_7918_(l1, -1, i2);
+               BlockPos blockpos1 = blockpos.offset(l1, -1, i2);
                int j2 = 50;
                if (Math.abs(l1) == 1 && Math.abs(i2) == 1) {
                   j2 = randomsource.m_188503_(5);
                }
 
-               while(blockpos1.m_123342_() > 50) {
+               while(blockpos1.getY() > 50) {
                   BlockState blockstate1 = worldgenlevel.m_8055_(blockpos1);
                   if (!blockstate1.m_60795_() && !m_159759_(blockstate1) && !blockstate1.m_60713_(Blocks.f_50127_) && !blockstate1.m_60713_(Blocks.f_50126_) && !blockstate1.m_60713_(Blocks.f_50354_)) {
                      break;
                   }
 
                   this.m_5974_(worldgenlevel, blockpos1, Blocks.f_50354_.m_49966_());
-                  blockpos1 = blockpos1.m_7495_();
+                  blockpos1 = blockpos1.below();
                   --j2;
                   if (j2 <= 0) {
-                     blockpos1 = blockpos1.m_6625_(randomsource.m_188503_(5) + 1);
+                     blockpos1 = blockpos1.below(randomsource.m_188503_(5) + 1);
                      j2 = randomsource.m_188503_(5);
                   }
                }
