--- a/net/minecraft/world/level/levelgen/feature/RandomPatchFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/RandomPatchFeature.java
@@ -22,7 +_,7 @@
       int k = randompatchconfiguration.f_191303_() + 1;
 
       for(int l = 0; l < randompatchconfiguration.f_67907_(); ++l) {
-         blockpos$mutableblockpos.m_122154_(blockpos, randomsource.m_188503_(j) - randomsource.m_188503_(j), randomsource.m_188503_(k) - randomsource.m_188503_(k), randomsource.m_188503_(j) - randomsource.m_188503_(j));
+         blockpos$mutableblockpos.setWithOffset(blockpos, randomsource.m_188503_(j) - randomsource.m_188503_(j), randomsource.m_188503_(k) - randomsource.m_188503_(k), randomsource.m_188503_(j) - randomsource.m_188503_(j));
          if (randompatchconfiguration.f_191304_().m_203334_().m_226357_(worldgenlevel, p_160210_.m_159775_(), randomsource, blockpos$mutableblockpos)) {
             ++i;
          }
