--- a/net/minecraft/world/level/levelgen/feature/ScatteredOreFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/ScatteredOreFeature.java
@@ -41,7 +_,7 @@
       int i = this.m_225228_(p_225233_, p_225235_);
       int j = this.m_225228_(p_225233_, p_225235_);
       int k = this.m_225228_(p_225233_, p_225235_);
-      p_225232_.m_122154_(p_225234_, i, j, k);
+      p_225232_.setWithOffset(p_225234_, i, j, k);
    }
 
    private int m_225228_(RandomSource p_225229_, int p_225230_) {
