--- a/net/minecraft/world/level/levelgen/feature/SeagrassFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SeagrassFeature.java
@@ -24,15 +_,15 @@
       ProbabilityFeatureConfiguration probabilityfeatureconfiguration = p_160318_.m_159778_();
       int i = randomsource.m_188503_(8) - randomsource.m_188503_(8);
       int j = randomsource.m_188503_(8) - randomsource.m_188503_(8);
-      int k = worldgenlevel.m_6924_(Heightmap.Types.OCEAN_FLOOR, blockpos.m_123341_() + i, blockpos.m_123343_() + j);
-      BlockPos blockpos1 = new BlockPos(blockpos.m_123341_() + i, k, blockpos.m_123343_() + j);
+      long k = worldgenlevel.getHeight(Heightmap.Types.OCEAN_FLOOR, blockpos.getBigX().add(i), blockpos.getBigZ().add(j));
+      BlockPos blockpos1 = new BlockPos(blockpos.getBigX().add(i), k, blockpos.getBigZ().add(j));
       if (worldgenlevel.m_8055_(blockpos1).m_60713_(Blocks.f_49990_)) {
          boolean flag1 = randomsource.m_188500_() < (double)probabilityfeatureconfiguration.f_67859_;
          BlockState blockstate = flag1 ? Blocks.f_50038_.m_49966_() : Blocks.f_50037_.m_49966_();
          if (blockstate.m_60710_(worldgenlevel, blockpos1)) {
             if (flag1) {
                BlockState blockstate1 = blockstate.m_61124_(TallSeagrassBlock.f_154740_, DoubleBlockHalf.UPPER);
-               BlockPos blockpos2 = blockpos1.m_7494_();
+               BlockPos blockpos2 = blockpos1.above();
                if (worldgenlevel.m_8055_(blockpos2).m_60713_(Blocks.f_49990_)) {
                   worldgenlevel.m_7731_(blockpos1, blockstate, 2);
                   worldgenlevel.m_7731_(blockpos2, blockstate1, 2);
