--- a/net/minecraft/world/level/levelgen/feature/SeaPickleFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SeaPickleFeature.java
@@ -25,8 +_,8 @@
       for(int k = 0; k < j; ++k) {
          int l = randomsource.m_188503_(8) - randomsource.m_188503_(8);
          int i1 = randomsource.m_188503_(8) - randomsource.m_188503_(8);
-         int j1 = worldgenlevel.m_6924_(Heightmap.Types.OCEAN_FLOOR, blockpos.m_123341_() + l, blockpos.m_123343_() + i1);
-         BlockPos blockpos1 = new BlockPos(blockpos.m_123341_() + l, j1, blockpos.m_123343_() + i1);
+         long j1 = worldgenlevel.getHeight(Heightmap.Types.OCEAN_FLOOR, blockpos.getX() + l, blockpos.getZ() + i1);
+         BlockPos blockpos1 = new BlockPos(blockpos.getX() + l, j1, blockpos.getZ() + i1);
          BlockState blockstate = Blocks.f_50567_.m_49966_().m_61124_(SeaPickleBlock.f_56074_, Integer.valueOf(randomsource.m_188503_(4) + 1));
          if (worldgenlevel.m_8055_(blockpos1).m_60713_(Blocks.f_49990_) && blockstate.m_60710_(worldgenlevel, blockpos1)) {
             worldgenlevel.m_7731_(blockpos1, blockstate, 2);
