--- a/net/minecraft/world/level/levelgen/feature/SimpleBlockFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SimpleBlockFeature.java
@@ -19,7 +_,7 @@
       BlockState blockstate = simpleblockconfiguration.f_68069_().m_213972_(p_160341_.m_225041_(), blockpos);
       if (blockstate.m_60710_(worldgenlevel, blockpos)) {
          if (blockstate.m_60734_() instanceof DoublePlantBlock) {
-            if (!worldgenlevel.m_46859_(blockpos.m_7494_())) {
+            if (!worldgenlevel.m_46859_(blockpos.above())) {
                return false;
             }
 
