--- a/net/minecraft/world/level/levelgen/feature/SpikeFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SpikeFeature.java
@@ -63,10 +_,10 @@
    private void m_225246_(ServerLevelAccessor p_225247_, RandomSource p_225248_, SpikeConfiguration p_225249_, SpikeFeature.EndSpike p_225250_) {
       int i = p_225250_.m_66896_();
 
-      for(BlockPos blockpos : BlockPos.m_121940_(new BlockPos(p_225250_.m_66886_() - i, p_225247_.m_141937_(), p_225250_.m_66893_() - i), new BlockPos(p_225250_.m_66886_() + i, p_225250_.m_66899_() + 10, p_225250_.m_66893_() + i))) {
-         if (blockpos.m_203202_((double)p_225250_.m_66886_(), (double)blockpos.m_123342_(), (double)p_225250_.m_66893_()) <= (double)(i * i + 1) && blockpos.m_123342_() < p_225250_.m_66899_()) {
+      for(BlockPos blockpos : BlockPos.m_121940_(new BlockPos(p_225250_.m_66886_() - i, p_225247_.getMinBuildHeight(), p_225250_.m_66893_() - i), new BlockPos(p_225250_.m_66886_() + i, p_225250_.m_66899_() + 10, p_225250_.m_66893_() + i))) {
+         if (blockpos.distToLowCornerSqr((double)p_225250_.m_66886_(), (double)blockpos.getY(), (double)p_225250_.m_66893_()) <= (double)(i * i + 1) && blockpos.getY() < p_225250_.m_66899_()) {
             this.m_5974_(p_225247_, blockpos, Blocks.f_50080_.m_49966_());
-         } else if (blockpos.m_123342_() > 65) {
+         } else if (blockpos.getY() > 65) {
             this.m_5974_(p_225247_, blockpos, Blocks.f_50016_.m_49966_());
          }
       }
@@ -87,7 +_,7 @@
                      boolean flag3 = k == -2 || k == 2 || flag2;
                      boolean flag4 = l == -2 || l == 2 || flag2;
                      BlockState blockstate = Blocks.f_50183_.m_49966_().m_61124_(IronBarsBlock.f_52309_, Boolean.valueOf(flag3 && l != -2)).m_61124_(IronBarsBlock.f_52311_, Boolean.valueOf(flag3 && l != 2)).m_61124_(IronBarsBlock.f_52312_, Boolean.valueOf(flag4 && k != -2)).m_61124_(IronBarsBlock.f_52310_, Boolean.valueOf(flag4 && k != 2));
-                     this.m_5974_(p_225247_, blockpos$mutableblockpos.m_122178_(p_225250_.m_66886_() + k, p_225250_.m_66899_() + i1, p_225250_.m_66893_() + l), blockstate);
+                     this.m_5974_(p_225247_, blockpos$mutableblockpos.set(p_225250_.m_66886_() + k, p_225250_.m_66899_() + i1, p_225250_.m_66893_() + l), blockstate);
                   }
                }
             }
@@ -133,7 +_,7 @@
       }
 
       public boolean m_66891_(BlockPos p_66892_) {
-         return SectionPos.m_123171_(p_66892_.m_123341_()) == SectionPos.m_123171_(this.f_66873_) && SectionPos.m_123171_(p_66892_.m_123343_()) == SectionPos.m_123171_(this.f_66874_);
+         return SectionPos.blockToSectionCoord(p_66892_.getX()) == SectionPos.blockToSectionCoord(this.f_66873_) && SectionPos.blockToSectionCoord(p_66892_.getZ()) == SectionPos.blockToSectionCoord(this.f_66874_);
       }
 
       public int m_66886_() {
