--- a/net/minecraft/world/level/levelgen/feature/SpringFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/SpringFeature.java
@@ -15,9 +_,9 @@
       SpringConfiguration springconfiguration = p_160404_.m_159778_();
       WorldGenLevel worldgenlevel = p_160404_.m_159774_();
       BlockPos blockpos = p_160404_.m_159777_();
-      if (!worldgenlevel.m_8055_(blockpos.m_7494_()).m_204341_(springconfiguration.f_68128_)) {
+      if (!worldgenlevel.m_8055_(blockpos.above()).m_204341_(springconfiguration.f_68128_)) {
          return false;
-      } else if (springconfiguration.f_68125_ && !worldgenlevel.m_8055_(blockpos.m_7495_()).m_204341_(springconfiguration.f_68128_)) {
+      } else if (springconfiguration.f_68125_ && !worldgenlevel.m_8055_(blockpos.below()).m_204341_(springconfiguration.f_68128_)) {
          return false;
       } else {
          BlockState blockstate = worldgenlevel.m_8055_(blockpos);
@@ -26,44 +_,44 @@
          } else {
             int i = 0;
             int j = 0;
-            if (worldgenlevel.m_8055_(blockpos.m_122024_()).m_204341_(springconfiguration.f_68128_)) {
-               ++j;
-            }
-
-            if (worldgenlevel.m_8055_(blockpos.m_122029_()).m_204341_(springconfiguration.f_68128_)) {
-               ++j;
-            }
-
-            if (worldgenlevel.m_8055_(blockpos.m_122012_()).m_204341_(springconfiguration.f_68128_)) {
-               ++j;
-            }
-
-            if (worldgenlevel.m_8055_(blockpos.m_122019_()).m_204341_(springconfiguration.f_68128_)) {
-               ++j;
-            }
-
-            if (worldgenlevel.m_8055_(blockpos.m_7495_()).m_204341_(springconfiguration.f_68128_)) {
+            if (worldgenlevel.m_8055_(blockpos.west()).m_204341_(springconfiguration.f_68128_)) {
+               ++j;
+            }
+
+            if (worldgenlevel.m_8055_(blockpos.east()).m_204341_(springconfiguration.f_68128_)) {
+               ++j;
+            }
+
+            if (worldgenlevel.m_8055_(blockpos.north()).m_204341_(springconfiguration.f_68128_)) {
+               ++j;
+            }
+
+            if (worldgenlevel.m_8055_(blockpos.south()).m_204341_(springconfiguration.f_68128_)) {
+               ++j;
+            }
+
+            if (worldgenlevel.m_8055_(blockpos.below()).m_204341_(springconfiguration.f_68128_)) {
                ++j;
             }
 
             int k = 0;
-            if (worldgenlevel.m_46859_(blockpos.m_122024_())) {
-               ++k;
-            }
-
-            if (worldgenlevel.m_46859_(blockpos.m_122029_())) {
-               ++k;
-            }
-
-            if (worldgenlevel.m_46859_(blockpos.m_122012_())) {
-               ++k;
-            }
-
-            if (worldgenlevel.m_46859_(blockpos.m_122019_())) {
-               ++k;
-            }
-
-            if (worldgenlevel.m_46859_(blockpos.m_7495_())) {
+            if (worldgenlevel.m_46859_(blockpos.west())) {
+               ++k;
+            }
+
+            if (worldgenlevel.m_46859_(blockpos.east())) {
+               ++k;
+            }
+
+            if (worldgenlevel.m_46859_(blockpos.north())) {
+               ++k;
+            }
+
+            if (worldgenlevel.m_46859_(blockpos.south())) {
+               ++k;
+            }
+
+            if (worldgenlevel.m_46859_(blockpos.below())) {
                ++k;
             }
 
