--- a/net/minecraft/world/level/levelgen/feature/VinesFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/VinesFeature.java
@@ -21,7 +_,7 @@
          return false;
       } else {
          for(Direction direction : Direction.values()) {
-            if (direction != Direction.DOWN && VineBlock.m_57853_(worldgenlevel, blockpos.m_121945_(direction), direction)) {
+            if (direction != Direction.DOWN && VineBlock.m_57853_(worldgenlevel, blockpos.relative(direction), direction)) {
                worldgenlevel.m_7731_(blockpos, Blocks.f_50191_.m_49966_().m_61124_(VineBlock.m_57883_(direction), Boolean.valueOf(true)), 2);
                return true;
             }
