--- a/net/minecraft/client/gui/Gui.java
+++ b/net/minecraft/client/gui/Gui.java
@@ -932,7 +_,7 @@
 
    private void m_93020_(Entity p_93021_) {
       if (p_93021_ != null) {
-         BlockPos blockpos = BlockPos.m_274561_(p_93021_.m_20185_(), p_93021_.m_20188_(), p_93021_.m_20189_());
+         BlockPos blockpos = BlockPos.containing(p_93021_.getBigX(), p_93021_.getEyeBigY(), p_93021_.getBigZ());
          float f = LightTexture.m_234316_(p_93021_.m_9236_().m_6042_(), p_93021_.m_9236_().m_46803_(blockpos));
          float f1 = Mth.m_14036_(1.0F - f, 0.0F, 1.0F);
          this.f_92980_ += (f1 - this.f_92980_) * 0.01F;
