--- a/net/minecraft/client/particle/AttackSweepParticle.java
+++ b/net/minecraft/client/particle/AttackSweepParticle.java
@@ -9,7 +_,7 @@
 public class AttackSweepParticle extends TextureSheetParticle {
    private final SpriteSet f_105544_;
 
-   AttackSweepParticle(ClientLevel p_105546_, double p_105547_, double p_105548_, double p_105549_, double p_105550_, SpriteSet p_105551_) {
+   AttackSweepParticle(ClientLevel p_105546_, me.alphamode.mclong.math.BigDecimal p_105547_, me.alphamode.mclong.math.BigDecimal p_105548_, me.alphamode.mclong.math.BigDecimal p_105549_, double p_105550_, SpriteSet p_105551_) {
       super(p_105546_, p_105547_, p_105548_, p_105549_, 0.0D, 0.0D, 0.0D);
       this.f_105544_ = p_105551_;
       this.f_107225_ = 4;
@@ -48,7 +_,7 @@
          this.f_105564_ = p_105566_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_105577_, ClientLevel p_105578_, double p_105579_, double p_105580_, double p_105581_, double p_105582_, double p_105583_, double p_105584_) {
+      public Particle createParticle(SimpleParticleType p_105577_, ClientLevel p_105578_, me.alphamode.mclong.math.BigDecimal p_105579_, me.alphamode.mclong.math.BigDecimal p_105580_, me.alphamode.mclong.math.BigDecimal p_105581_, double p_105582_, double p_105583_, double p_105584_) {
          return new AttackSweepParticle(p_105578_, p_105579_, p_105580_, p_105581_, p_105582_, this.f_105564_);
       }
    }
