--- a/net/minecraft/client/particle/BaseAshSmokeParticle.java
+++ b/net/minecraft/client/particle/BaseAshSmokeParticle.java
@@ -9,7 +_,7 @@
 public class BaseAshSmokeParticle extends TextureSheetParticle {
    private final SpriteSet f_105620_;
 
-   protected BaseAshSmokeParticle(ClientLevel p_171904_, double p_171905_, double p_171906_, double p_171907_, float p_171908_, float p_171909_, float p_171910_, double p_171911_, double p_171912_, double p_171913_, float p_171914_, SpriteSet p_171915_, float p_171916_, int p_171917_, float p_171918_, boolean p_171919_) {
+   protected BaseAshSmokeParticle(ClientLevel p_171904_, me.alphamode.mclong.math.BigDecimal p_171905_, me.alphamode.mclong.math.BigDecimal p_171906_, me.alphamode.mclong.math.BigDecimal p_171907_, float p_171908_, float p_171909_, float p_171910_, double p_171911_, double p_171912_, double p_171913_, float p_171914_, SpriteSet p_171915_, float p_171916_, int p_171917_, float p_171918_, boolean p_171919_) {
       super(p_171904_, p_171905_, p_171906_, p_171907_, 0.0D, 0.0D, 0.0D);
       this.f_172258_ = 0.96F;
       this.f_107226_ = p_171918_;
