--- a/net/minecraft/client/particle/BlockMarker.java
+++ b/net/minecraft/client/particle/BlockMarker.java
@@ -9,9 +_,9 @@
 
 @OnlyIn(Dist.CLIENT)
 public class BlockMarker extends TextureSheetParticle {
-   BlockMarker(ClientLevel p_194267_, double p_194268_, double p_194269_, double p_194270_, BlockState p_194271_) {
-      super(p_194267_, p_194268_, p_194269_, p_194270_);
-      this.m_108337_(Minecraft.m_91087_().m_91289_().m_110907_().m_110882_(p_194271_));
+   BlockMarker(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, BlockState pState) {
+      super(pLevel, pX, pY, pZ);
+      this.m_108337_(Minecraft.m_91087_().m_91289_().m_110907_().m_110882_(pState));
       this.f_107226_ = 0.0F;
       this.f_107225_ = 80;
       this.f_107219_ = false;
@@ -27,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class Provider implements ParticleProvider<BlockParticleOption> {
-      public Particle m_6966_(BlockParticleOption p_194277_, ClientLevel p_194278_, double p_194279_, double p_194280_, double p_194281_, double p_194282_, double p_194283_, double p_194284_) {
+      public Particle createParticle(BlockParticleOption p_194277_, ClientLevel p_194278_, me.alphamode.mclong.math.BigDecimal p_194279_, me.alphamode.mclong.math.BigDecimal p_194280_, me.alphamode.mclong.math.BigDecimal p_194281_, double p_194282_, double p_194283_, double p_194284_) {
          return new BlockMarker(p_194278_, p_194279_, p_194280_, p_194281_, p_194277_.m_123642_());
       }
    }
