--- a/net/minecraft/client/particle/BreakingItemParticle.java
+++ b/net/minecraft/client/particle/BreakingItemParticle.java
@@ -15,23 +_,23 @@
    private final float f_105643_;
    private final float f_105644_;
 
-   BreakingItemParticle(ClientLevel p_105646_, double p_105647_, double p_105648_, double p_105649_, double p_105650_, double p_105651_, double p_105652_, ItemStack p_105653_) {
-      this(p_105646_, p_105647_, p_105648_, p_105649_, p_105653_);
+   BreakingItemParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, ItemStack pStack) {
+      this(pLevel, pX, pY, pZ, pStack);
       this.f_107215_ *= (double)0.1F;
       this.f_107216_ *= (double)0.1F;
       this.f_107217_ *= (double)0.1F;
-      this.f_107215_ += p_105650_;
-      this.f_107216_ += p_105651_;
-      this.f_107217_ += p_105652_;
+      this.f_107215_ += pXSpeed;
+      this.f_107216_ += pYSpeed;
+      this.f_107217_ += pZSpeed;
    }
 
    public ParticleRenderType m_7556_() {
       return ParticleRenderType.f_107429_;
    }
 
-   protected BreakingItemParticle(ClientLevel p_105665_, double p_105666_, double p_105667_, double p_105668_, ItemStack p_105669_) {
-      super(p_105665_, p_105666_, p_105667_, p_105668_, 0.0D, 0.0D, 0.0D);
-      this.m_108337_(Minecraft.m_91087_().m_91291_().m_174264_(p_105669_, p_105665_, (LivingEntity)null, 0).m_6160_());
+   protected BreakingItemParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, ItemStack pStack) {
+      super(pLevel, pX, pY, pZ, 0.0D, 0.0D, 0.0D);
+      this.m_108337_(Minecraft.m_91087_().m_91291_().m_174264_(pStack, pLevel, (LivingEntity)null, 0).m_6160_());
       this.f_107226_ = 1.0F;
       this.f_107663_ /= 2.0F;
       this.f_105643_ = this.f_107223_.m_188501_() * 3.0F;
@@ -56,22 +_,22 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class Provider implements ParticleProvider<ItemParticleOption> {
-      public Particle m_6966_(ItemParticleOption p_105677_, ClientLevel p_105678_, double p_105679_, double p_105680_, double p_105681_, double p_105682_, double p_105683_, double p_105684_) {
-         return new BreakingItemParticle(p_105678_, p_105679_, p_105680_, p_105681_, p_105682_, p_105683_, p_105684_, p_105677_.m_123718_());
+      public Particle createParticle(ItemParticleOption pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new BreakingItemParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, pType.m_123718_());
       }
    }
 
    @OnlyIn(Dist.CLIENT)
    public static class SlimeProvider implements ParticleProvider<SimpleParticleType> {
-      public Particle m_6966_(SimpleParticleType p_105705_, ClientLevel p_105706_, double p_105707_, double p_105708_, double p_105709_, double p_105710_, double p_105711_, double p_105712_) {
-         return new BreakingItemParticle(p_105706_, p_105707_, p_105708_, p_105709_, new ItemStack(Items.f_42518_));
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new BreakingItemParticle(pLevel, pX, pY, pZ, new ItemStack(Items.f_42518_));
       }
    }
 
    @OnlyIn(Dist.CLIENT)
    public static class SnowballProvider implements ParticleProvider<SimpleParticleType> {
-      public Particle m_6966_(SimpleParticleType p_105724_, ClientLevel p_105725_, double p_105726_, double p_105727_, double p_105728_, double p_105729_, double p_105730_, double p_105731_) {
-         return new BreakingItemParticle(p_105725_, p_105726_, p_105727_, p_105728_, new ItemStack(Items.f_42452_));
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new BreakingItemParticle(pLevel, pX, pY, pZ, new ItemStack(Items.f_42452_));
       }
    }
 }
