--- a/net/minecraft/client/particle/BubblePopParticle.java
+++ b/net/minecraft/client/particle/BubblePopParticle.java
@@ -9,7 +_,7 @@
 public class BubblePopParticle extends TextureSheetParticle {
    private final SpriteSet f_105812_;
 
-   BubblePopParticle(ClientLevel p_105814_, double p_105815_, double p_105816_, double p_105817_, double p_105818_, double p_105819_, double p_105820_, SpriteSet p_105821_) {
+   BubblePopParticle(ClientLevel p_105814_, me.alphamode.mclong.math.BigDecimal p_105815_, me.alphamode.mclong.math.BigDecimal p_105816_, me.alphamode.mclong.math.BigDecimal p_105817_, double p_105818_, double p_105819_, double p_105820_, SpriteSet p_105821_) {
       super(p_105814_, p_105815_, p_105816_, p_105817_);
       this.f_105812_ = p_105821_;
       this.f_107225_ = 4;
@@ -45,7 +_,7 @@
          this.f_105834_ = p_105836_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_105847_, ClientLevel p_105848_, double p_105849_, double p_105850_, double p_105851_, double p_105852_, double p_105853_, double p_105854_) {
+      public Particle createParticle(SimpleParticleType p_105847_, ClientLevel p_105848_, me.alphamode.mclong.math.BigDecimal p_105849_, me.alphamode.mclong.math.BigDecimal p_105850_, me.alphamode.mclong.math.BigDecimal p_105851_, double p_105852_, double p_105853_, double p_105854_) {
          return new BubblePopParticle(p_105848_, p_105849_, p_105850_, p_105851_, p_105852_, p_105853_, p_105854_, this.f_105834_);
       }
    }
