--- a/net/minecraft/client/particle/BubbleColumnUpParticle.java
+++ b/net/minecraft/client/particle/BubbleColumnUpParticle.java
@@ -9,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class BubbleColumnUpParticle extends TextureSheetParticle {
-   BubbleColumnUpParticle(ClientLevel p_105733_, double p_105734_, double p_105735_, double p_105736_, double p_105737_, double p_105738_, double p_105739_) {
+   BubbleColumnUpParticle(ClientLevel p_105733_, me.alphamode.mclong.math.BigDecimal p_105734_, me.alphamode.mclong.math.BigDecimal p_105735_, me.alphamode.mclong.math.BigDecimal p_105736_, double p_105737_, double p_105738_, double p_105739_) {
       super(p_105733_, p_105734_, p_105735_, p_105736_);
       this.f_107226_ = -0.125F;
       this.f_172258_ = 0.85F;
@@ -23,7 +_,7 @@
 
    public void m_5989_() {
       super.m_5989_();
-      if (!this.f_107220_ && !this.f_107208_.m_6425_(BlockPos.m_274561_(this.f_107212_, this.f_107213_, this.f_107214_)).m_205070_(FluidTags.f_13131_)) {
+      if (!this.f_107220_ && !this.f_107208_.m_6425_(BlockPos.containing(this.f_107212_, this.f_107213_, this.f_107214_)).m_205070_(FluidTags.f_13131_)) {
          this.m_107274_();
       }
 
@@ -41,7 +_,7 @@
          this.f_105751_ = p_105753_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_105764_, ClientLevel p_105765_, double p_105766_, double p_105767_, double p_105768_, double p_105769_, double p_105770_, double p_105771_) {
+      public Particle createParticle(SimpleParticleType p_105764_, ClientLevel p_105765_, me.alphamode.mclong.math.BigDecimal p_105766_, me.alphamode.mclong.math.BigDecimal p_105767_, me.alphamode.mclong.math.BigDecimal p_105768_, double p_105769_, double p_105770_, double p_105771_) {
          BubbleColumnUpParticle bubblecolumnupparticle = new BubbleColumnUpParticle(p_105765_, p_105766_, p_105767_, p_105768_, p_105769_, p_105770_, p_105771_);
          bubblecolumnupparticle.m_108335_(this.f_105751_);
          return bubblecolumnupparticle;
