--- a/net/minecraft/client/particle/CampfireSmokeParticle.java
+++ b/net/minecraft/client/particle/CampfireSmokeParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class CampfireSmokeParticle extends TextureSheetParticle {
-   CampfireSmokeParticle(ClientLevel p_105856_, double p_105857_, double p_105858_, double p_105859_, double p_105860_, double p_105861_, double p_105862_, boolean p_105863_) {
+   CampfireSmokeParticle(ClientLevel p_105856_, me.alphamode.mclong.math.BigDecimal p_105857_, me.alphamode.mclong.math.BigDecimal p_105858_, me.alphamode.mclong.math.BigDecimal p_105859_, double p_105860_, double p_105861_, double p_105862_, boolean p_105863_) {
       super(p_105856_, p_105857_, p_105858_, p_105859_);
       this.m_6569_(3.0F);
       this.m_107250_(0.25F, 0.25F);
@@ -53,7 +_,7 @@
          this.f_105876_ = p_105878_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_105889_, ClientLevel p_105890_, double p_105891_, double p_105892_, double p_105893_, double p_105894_, double p_105895_, double p_105896_) {
+      public Particle createParticle(SimpleParticleType p_105889_, ClientLevel p_105890_, me.alphamode.mclong.math.BigDecimal p_105891_, me.alphamode.mclong.math.BigDecimal p_105892_, me.alphamode.mclong.math.BigDecimal p_105893_, double p_105894_, double p_105895_, double p_105896_) {
          CampfireSmokeParticle campfiresmokeparticle = new CampfireSmokeParticle(p_105890_, p_105891_, p_105892_, p_105893_, p_105894_, p_105895_, p_105896_, false);
          campfiresmokeparticle.m_107271_(0.9F);
          campfiresmokeparticle.m_108335_(this.f_105876_);
@@ -69,7 +_,7 @@
          this.f_105897_ = p_105899_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_105910_, ClientLevel p_105911_, double p_105912_, double p_105913_, double p_105914_, double p_105915_, double p_105916_, double p_105917_) {
+      public Particle createParticle(SimpleParticleType p_105910_, ClientLevel p_105911_, me.alphamode.mclong.math.BigDecimal p_105912_, me.alphamode.mclong.math.BigDecimal p_105913_, me.alphamode.mclong.math.BigDecimal p_105914_, double p_105915_, double p_105916_, double p_105917_) {
          CampfireSmokeParticle campfiresmokeparticle = new CampfireSmokeParticle(p_105911_, p_105912_, p_105913_, p_105914_, p_105915_, p_105916_, p_105917_, true);
          campfiresmokeparticle.m_107271_(0.95F);
          campfiresmokeparticle.m_108335_(this.f_105897_);
