--- a/net/minecraft/client/particle/CherryParticle.java
+++ b/net/minecraft/client/particle/CherryParticle.java
@@ -15,7 +_,7 @@
    private final float f_276535_;
    private final float f_276513_;
 
-   protected CherryParticle(ClientLevel p_277612_, double p_278010_, double p_277614_, double p_277673_, SpriteSet p_277465_) {
+   protected CherryParticle(ClientLevel p_277612_, me.alphamode.mclong.math.BigDecimal p_278010_, me.alphamode.mclong.math.BigDecimal p_277614_, me.alphamode.mclong.math.BigDecimal p_277673_, SpriteSet p_277465_) {
       super(p_277612_, p_278010_, p_277614_, p_277673_);
       this.m_108337_(p_277465_.m_5819_(this.f_107223_.m_188503_(12), 12));
       this.f_276609_ = (float)Math.toRadians(this.f_107223_.m_188499_() ? -30.0D : 30.0D);
