--- a/net/minecraft/client/particle/DragonBreathParticle.java
+++ b/net/minecraft/client/particle/DragonBreathParticle.java
@@ -19,12 +_,12 @@
    private boolean f_106002_;
    private final SpriteSet f_106003_;
 
-   DragonBreathParticle(ClientLevel p_106005_, double p_106006_, double p_106007_, double p_106008_, double p_106009_, double p_106010_, double p_106011_, SpriteSet p_106012_) {
-      super(p_106005_, p_106006_, p_106007_, p_106008_);
+   DragonBreathParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ);
       this.f_172258_ = 0.96F;
-      this.f_107215_ = p_106009_;
-      this.f_107216_ = p_106010_;
-      this.f_107217_ = p_106011_;
+      this.f_107215_ = pXSpeed;
+      this.f_107216_ = pYSpeed;
+      this.f_107217_ = pZSpeed;
       this.f_107227_ = Mth.m_216267_(this.f_107223_, 0.7176471F, 0.8745098F);
       this.f_107228_ = Mth.m_216267_(this.f_107223_, 0.0F, 0.0F);
       this.f_107229_ = Mth.m_216267_(this.f_107223_, 0.8235294F, 0.9764706F);
@@ -32,8 +_,8 @@
       this.f_107225_ = (int)(20.0D / ((double)this.f_107223_.m_188501_() * 0.8D + 0.2D));
       this.f_106002_ = false;
       this.f_107219_ = false;
-      this.f_106003_ = p_106012_;
-      this.m_108339_(p_106012_);
+      this.f_106003_ = pSprites;
+      this.m_108339_(pSprites);
    }
 
    public void m_5989_() {
@@ -84,8 +_,8 @@
          this.f_106027_ = p_106029_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106040_, ClientLevel p_106041_, double p_106042_, double p_106043_, double p_106044_, double p_106045_, double p_106046_, double p_106047_) {
-         return new DragonBreathParticle(p_106041_, p_106042_, p_106043_, p_106044_, p_106045_, p_106046_, p_106047_, this.f_106027_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new DragonBreathParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, this.f_106027_);
       }
    }
 }
