--- a/net/minecraft/client/particle/DragonBreathParticle.java
+++ b/net/minecraft/client/particle/DragonBreathParticle.java
@@ -19,7 +_,7 @@
    private boolean f_106002_;
    private final SpriteSet f_106003_;
 
-   DragonBreathParticle(ClientLevel p_106005_, double p_106006_, double p_106007_, double p_106008_, double p_106009_, double p_106010_, double p_106011_, SpriteSet p_106012_) {
+   DragonBreathParticle(ClientLevel p_106005_, me.alphamode.mclong.math.BigDecimal p_106006_, me.alphamode.mclong.math.BigDecimal p_106007_, me.alphamode.mclong.math.BigDecimal p_106008_, double p_106009_, double p_106010_, double p_106011_, SpriteSet p_106012_) {
       super(p_106005_, p_106006_, p_106007_, p_106008_);
       this.f_172258_ = 0.96F;
       this.f_107215_ = p_106009_;
@@ -84,7 +_,7 @@
          this.f_106027_ = p_106029_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106040_, ClientLevel p_106041_, double p_106042_, double p_106043_, double p_106044_, double p_106045_, double p_106046_, double p_106047_) {
+      public Particle createParticle(SimpleParticleType p_106040_, ClientLevel p_106041_, me.alphamode.mclong.math.BigDecimal p_106042_, me.alphamode.mclong.math.BigDecimal p_106043_, me.alphamode.mclong.math.BigDecimal p_106044_, double p_106045_, double p_106046_, double p_106047_) {
          return new DragonBreathParticle(p_106041_, p_106042_, p_106043_, p_106044_, p_106045_, p_106046_, p_106047_, this.f_106027_);
       }
    }
