--- a/net/minecraft/client/particle/DripParticle.java
+++ b/net/minecraft/client/particle/DripParticle.java
@@ -55,7 +_,7 @@
             this.f_107217_ *= (double)0.98F;
             BlockPos blockpos = new BlockPos(this.f_107212_, this.f_107213_, this.f_107214_);
             FluidState fluidstate = this.f_107208_.m_6425_(blockpos);
-            if (fluidstate.m_76152_() == this.f_106049_ && this.f_107213_ < (double)((float)blockpos.m_123342_() + fluidstate.m_76155_(this.f_107208_, blockpos))) {
+            if (fluidstate.m_76152_() == this.f_106049_ && this.f_107213_ < (double)((float)blockpos.getY() + fluidstate.m_76155_(this.f_107208_, blockpos))) {
                this.m_107274_();
             }
 
