--- a/net/minecraft/client/particle/DustParticleBase.java
+++ b/net/minecraft/client/particle/DustParticleBase.java
@@ -10,7 +_,7 @@
 public class DustParticleBase<T extends DustParticleOptionsBase> extends TextureSheetParticle {
    private final SpriteSet f_172092_;
 
-   protected DustParticleBase(ClientLevel p_172094_, double p_172095_, double p_172096_, double p_172097_, double p_172098_, double p_172099_, double p_172100_, T p_172101_, SpriteSet p_172102_) {
+   protected DustParticleBase(ClientLevel p_172094_, me.alphamode.mclong.math.BigDecimal p_172095_, me.alphamode.mclong.math.BigDecimal p_172096_, me.alphamode.mclong.math.BigDecimal p_172097_, double p_172098_, double p_172099_, double p_172100_, T p_172101_, SpriteSet p_172102_) {
       super(p_172094_, p_172095_, p_172096_, p_172097_, p_172098_, p_172099_, p_172100_);
       this.f_172258_ = 0.96F;
       this.f_172259_ = true;
