--- a/net/minecraft/client/particle/DustParticle.java
+++ b/net/minecraft/client/particle/DustParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class DustParticle extends DustParticleBase<DustParticleOptions> {
-   protected DustParticle(ClientLevel p_106415_, double p_106416_, double p_106417_, double p_106418_, double p_106419_, double p_106420_, double p_106421_, DustParticleOptions p_106422_, SpriteSet p_106423_) {
+   protected DustParticle(ClientLevel p_106415_, me.alphamode.mclong.math.BigDecimal p_106416_, me.alphamode.mclong.math.BigDecimal p_106417_, me.alphamode.mclong.math.BigDecimal p_106418_, double p_106419_, double p_106420_, double p_106421_, DustParticleOptions p_106422_, SpriteSet p_106423_) {
       super(p_106415_, p_106416_, p_106417_, p_106418_, p_106419_, p_106420_, p_106421_, p_106422_, p_106423_);
    }
 
@@ -19,7 +_,7 @@
          this.f_106439_ = p_106441_;
       }
 
-      public Particle m_6966_(DustParticleOptions p_106443_, ClientLevel p_106444_, double p_106445_, double p_106446_, double p_106447_, double p_106448_, double p_106449_, double p_106450_) {
+      public Particle createParticle(DustParticleOptions p_106443_, ClientLevel p_106444_, me.alphamode.mclong.math.BigDecimal p_106445_, me.alphamode.mclong.math.BigDecimal p_106446_, me.alphamode.mclong.math.BigDecimal p_106447_, double p_106448_, double p_106449_, double p_106450_) {
          return new DustParticle(p_106444_, p_106445_, p_106446_, p_106447_, p_106448_, p_106449_, p_106450_, p_106443_, this.f_106439_);
       }
    }
