--- a/net/minecraft/client/particle/EndRodParticle.java
+++ b/net/minecraft/client/particle/EndRodParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class EndRodParticle extends SimpleAnimatedParticle {
-   EndRodParticle(ClientLevel p_106531_, double p_106532_, double p_106533_, double p_106534_, double p_106535_, double p_106536_, double p_106537_, SpriteSet p_106538_) {
+   EndRodParticle(ClientLevel p_106531_, me.alphamode.mclong.math.BigDecimal p_106532_, me.alphamode.mclong.math.BigDecimal p_106533_, me.alphamode.mclong.math.BigDecimal p_106534_, double p_106535_, double p_106536_, double p_106537_, SpriteSet p_106538_) {
       super(p_106531_, p_106532_, p_106533_, p_106534_, p_106538_, 0.0125F);
       this.f_107215_ = p_106535_;
       this.f_107216_ = p_106536_;
@@ -31,7 +_,7 @@
          this.f_106553_ = p_106555_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106566_, ClientLevel p_106567_, double p_106568_, double p_106569_, double p_106570_, double p_106571_, double p_106572_, double p_106573_) {
+      public Particle createParticle(SimpleParticleType p_106566_, ClientLevel p_106567_, me.alphamode.mclong.math.BigDecimal p_106568_, me.alphamode.mclong.math.BigDecimal p_106569_, me.alphamode.mclong.math.BigDecimal p_106570_, double p_106571_, double p_106572_, double p_106573_) {
          return new EndRodParticle(p_106567_, p_106568_, p_106569_, p_106570_, p_106571_, p_106572_, p_106573_, this.f_106553_);
       }
    }
