--- a/net/minecraft/client/particle/EndRodParticle.java
+++ b/net/minecraft/client/particle/EndRodParticle.java
@@ -7,15 +_,15 @@
 
 @OnlyIn(Dist.CLIENT)
 public class EndRodParticle extends SimpleAnimatedParticle {
-   EndRodParticle(ClientLevel p_106531_, double p_106532_, double p_106533_, double p_106534_, double p_106535_, double p_106536_, double p_106537_, SpriteSet p_106538_) {
-      super(p_106531_, p_106532_, p_106533_, p_106534_, p_106538_, 0.0125F);
-      this.f_107215_ = p_106535_;
-      this.f_107216_ = p_106536_;
-      this.f_107217_ = p_106537_;
+   EndRodParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ, pSprites, 0.0125F);
+      this.f_107215_ = pXSpeed;
+      this.f_107216_ = pYSpeed;
+      this.f_107217_ = pZSpeed;
       this.f_107663_ *= 0.75F;
       this.f_107225_ = 60 + this.f_107223_.m_188503_(12);
       this.m_107659_(15916745);
-      this.m_108339_(p_106538_);
+      this.m_108339_(pSprites);
    }
 
    public void m_6257_(double p_106550_, double p_106551_, double p_106552_) {
@@ -31,8 +_,8 @@
          this.f_106553_ = p_106555_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106566_, ClientLevel p_106567_, double p_106568_, double p_106569_, double p_106570_, double p_106571_, double p_106572_, double p_106573_) {
-         return new EndRodParticle(p_106567_, p_106568_, p_106569_, p_106570_, p_106571_, p_106572_, p_106573_, this.f_106553_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new EndRodParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, this.f_106553_);
       }
    }
 }
