--- a/net/minecraft/client/particle/ExplodeParticle.java
+++ b/net/minecraft/client/particle/ExplodeParticle.java
@@ -9,7 +_,7 @@
 public class ExplodeParticle extends TextureSheetParticle {
    private final SpriteSet f_106574_;
 
-   protected ExplodeParticle(ClientLevel p_106576_, double p_106577_, double p_106578_, double p_106579_, double p_106580_, double p_106581_, double p_106582_, SpriteSet p_106583_) {
+   protected ExplodeParticle(ClientLevel p_106576_, me.alphamode.mclong.math.BigDecimal p_106577_, me.alphamode.mclong.math.BigDecimal p_106578_, me.alphamode.mclong.math.BigDecimal p_106579_, double p_106580_, double p_106581_, double p_106582_, SpriteSet p_106583_) {
       super(p_106576_, p_106577_, p_106578_, p_106579_);
       this.f_107226_ = -0.1F;
       this.f_172258_ = 0.9F;
@@ -43,7 +_,7 @@
          this.f_106586_ = p_106588_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106599_, ClientLevel p_106600_, double p_106601_, double p_106602_, double p_106603_, double p_106604_, double p_106605_, double p_106606_) {
+      public Particle createParticle(SimpleParticleType p_106599_, ClientLevel p_106600_, me.alphamode.mclong.math.BigDecimal p_106601_, me.alphamode.mclong.math.BigDecimal p_106602_, me.alphamode.mclong.math.BigDecimal p_106603_, double p_106604_, double p_106605_, double p_106606_) {
          return new ExplodeParticle(p_106600_, p_106601_, p_106602_, p_106603_, p_106604_, p_106605_, p_106606_, this.f_106586_);
       }
    }
