--- a/net/minecraft/client/particle/FallingDustParticle.java
+++ b/net/minecraft/client/particle/FallingDustParticle.java
@@ -17,17 +_,17 @@
    private final float f_106607_;
    private final SpriteSet f_106608_;
 
-   FallingDustParticle(ClientLevel p_106610_, double p_106611_, double p_106612_, double p_106613_, float p_106614_, float p_106615_, float p_106616_, SpriteSet p_106617_) {
-      super(p_106610_, p_106611_, p_106612_, p_106613_);
-      this.f_106608_ = p_106617_;
-      this.f_107227_ = p_106614_;
-      this.f_107228_ = p_106615_;
-      this.f_107229_ = p_106616_;
+   FallingDustParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pXSpeed, float pYSpeed, float pZSpeed, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ);
+      this.f_106608_ = pSprites;
+      this.f_107227_ = pXSpeed;
+      this.f_107228_ = pYSpeed;
+      this.f_107229_ = pZSpeed;
       float f = 0.9F;
       this.f_107663_ *= 0.67499995F;
       int i = (int)(32.0D / (Math.random() * 0.8D + 0.2D));
       this.f_107225_ = (int)Math.max((float)i * 0.9F, 1.0F);
-      this.m_108339_(p_106617_);
+      this.m_108339_(pSprites);
       this.f_106607_ = ((float)Math.random() - 0.5F) * 0.1F;
       this.f_107231_ = (float)Math.random() * ((float)Math.PI * 2F);
    }
@@ -69,21 +_,21 @@
       }
 
       @Nullable
-      public Particle m_6966_(BlockParticleOption p_106636_, ClientLevel p_106637_, double p_106638_, double p_106639_, double p_106640_, double p_106641_, double p_106642_, double p_106643_) {
-         BlockState blockstate = p_106636_.m_123642_();
+      public Particle createParticle(BlockParticleOption pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         BlockState blockstate = pType.m_123642_();
          if (!blockstate.m_60795_() && blockstate.m_60799_() == RenderShape.INVISIBLE) {
             return null;
          } else {
-            BlockPos blockpos = BlockPos.m_274561_(p_106638_, p_106639_, p_106640_);
-            int i = Minecraft.m_91087_().m_91298_().m_92582_(blockstate, p_106637_, blockpos);
+            BlockPos blockpos = BlockPos.containing(pX, pY, pZ);
+            int i = Minecraft.m_91087_().m_91298_().m_92582_(blockstate, pLevel, blockpos);
             if (blockstate.m_60734_() instanceof FallingBlock) {
-               i = ((FallingBlock)blockstate.m_60734_()).m_6248_(blockstate, p_106637_, blockpos);
+               i = ((FallingBlock)blockstate.m_60734_()).m_6248_(blockstate, pLevel, blockpos);
             }
 
             float f = (float)(i >> 16 & 255) / 255.0F;
             float f1 = (float)(i >> 8 & 255) / 255.0F;
             float f2 = (float)(i & 255) / 255.0F;
-            return new FallingDustParticle(p_106637_, p_106638_, p_106639_, p_106640_, f, f1, f2, this.f_106632_);
+            return new FallingDustParticle(pLevel, pX, pY, pZ, f, f1, f2, this.f_106632_);
          }
       }
    }
