--- a/net/minecraft/client/particle/FireworkParticles.java
+++ b/net/minecraft/client/particle/FireworkParticles.java
@@ -29,7 +_,7 @@
          this.f_106655_ = p_106657_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106668_, ClientLevel p_106669_, double p_106670_, double p_106671_, double p_106672_, double p_106673_, double p_106674_, double p_106675_) {
+      public Particle createParticle(SimpleParticleType p_106668_, ClientLevel p_106669_, me.alphamode.mclong.math.BigDecimal p_106670_, me.alphamode.mclong.math.BigDecimal p_106671_, me.alphamode.mclong.math.BigDecimal p_106672_, double p_106673_, double p_106674_, double p_106675_) {
          FireworkParticles.OverlayParticle fireworkparticles$overlayparticle = new FireworkParticles.OverlayParticle(p_106669_, p_106670_, p_106671_, p_106672_);
          fireworkparticles$overlayparticle.m_108335_(this.f_106655_);
          return fireworkparticles$overlayparticle;
@@ -38,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class OverlayParticle extends TextureSheetParticle {
-      OverlayParticle(ClientLevel p_106677_, double p_106678_, double p_106679_, double p_106680_) {
+      OverlayParticle(ClientLevel p_106677_, me.alphamode.mclong.math.BigDecimal p_106678_, me.alphamode.mclong.math.BigDecimal p_106679_, me.alphamode.mclong.math.BigDecimal p_106680_) {
          super(p_106677_, p_106678_, p_106679_, p_106680_);
          this.f_107225_ = 4;
       }
@@ -67,7 +_,7 @@
       private float f_106698_;
       private boolean f_106699_;
 
-      SparkParticle(ClientLevel p_106702_, double p_106703_, double p_106704_, double p_106705_, double p_106706_, double p_106707_, double p_106708_, ParticleEngine p_106709_, SpriteSet p_106710_) {
+      SparkParticle(ClientLevel p_106702_, me.alphamode.mclong.math.BigDecimal p_106703_, me.alphamode.mclong.math.BigDecimal p_106704_, me.alphamode.mclong.math.BigDecimal p_106705_, double p_106706_, double p_106707_, double p_106708_, ParticleEngine p_106709_, SpriteSet p_106710_) {
          super(p_106702_, p_106703_, p_106704_, p_106705_, p_106710_, 0.1F);
          this.f_107215_ = p_106706_;
          this.f_107216_ = p_106707_;
@@ -122,7 +_,7 @@
          this.f_106731_ = p_106733_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106744_, ClientLevel p_106745_, double p_106746_, double p_106747_, double p_106748_, double p_106749_, double p_106750_, double p_106751_) {
+      public Particle createParticle(SimpleParticleType p_106744_, ClientLevel p_106745_, me.alphamode.mclong.math.BigDecimal p_106746_, me.alphamode.mclong.math.BigDecimal p_106747_, me.alphamode.mclong.math.BigDecimal p_106748_, double p_106749_, double p_106750_, double p_106751_) {
          FireworkParticles.SparkParticle fireworkparticles$sparkparticle = new FireworkParticles.SparkParticle(p_106745_, p_106746_, p_106747_, p_106748_, p_106749_, p_106750_, p_106751_, Minecraft.m_91087_().f_91061_, this.f_106731_);
          fireworkparticles$sparkparticle.m_107271_(0.99F);
          return fireworkparticles$sparkparticle;
@@ -136,7 +_,7 @@
       private ListTag f_106752_;
       private boolean f_106753_;
 
-      public Starter(ClientLevel p_106757_, double p_106758_, double p_106759_, double p_106760_, double p_106761_, double p_106762_, double p_106763_, ParticleEngine p_106764_, @Nullable CompoundTag p_106765_) {
+      public Starter(ClientLevel p_106757_, me.alphamode.mclong.math.BigDecimal p_106758_, me.alphamode.mclong.math.BigDecimal p_106759_, me.alphamode.mclong.math.BigDecimal p_106760_, double p_106761_, double p_106762_, double p_106763_, ParticleEngine p_106764_, @Nullable CompoundTag p_106765_) {
          super(p_106757_, p_106758_, p_106759_, p_106760_);
          this.f_107215_ = p_106761_;
          this.f_107216_ = p_106762_;
@@ -186,7 +_,7 @@
                soundevent1 = flag ? SoundEvents.f_11929_ : SoundEvents.f_11928_;
             }
 
-            this.f_107208_.m_7785_(this.f_107212_, this.f_107213_, this.f_107214_, soundevent1, SoundSource.AMBIENT, 20.0F, 0.95F + this.f_107223_.m_188501_() * 0.1F, true);
+            this.f_107208_.playLocalSound(this.f_107212_, this.f_107213_, this.f_107214_, soundevent1, SoundSource.AMBIENT, 20.0F, 0.95F + this.f_107223_.m_188501_() * 0.1F, true);
          }
 
          if (this.f_106754_ % 2 == 0 && this.f_106752_ != null && this.f_106754_ / 2 < this.f_106752_.size()) {
@@ -223,7 +_,7 @@
             float f = (float)((j & 16711680) >> 16) / 255.0F;
             float f1 = (float)((j & '\uff00') >> 8) / 255.0F;
             float f2 = (float)((j & 255) >> 0) / 255.0F;
-            Particle particle = this.f_106755_.m_107370_(ParticleTypes.f_123747_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
+            Particle particle = this.f_106755_.createParticle(ParticleTypes.f_123747_, this.f_107212_, this.f_107213_, this.f_107214_, 0.0D, 0.0D, 0.0D);
             particle.m_107253_(f, f1, f2);
          }
 
@@ -232,7 +_,7 @@
             if (this.f_106753_) {
                boolean flag3 = this.m_106798_();
                SoundEvent soundevent = flag3 ? SoundEvents.f_11935_ : SoundEvents.f_11934_;
-               this.f_107208_.m_7785_(this.f_107212_, this.f_107213_, this.f_107214_, soundevent, SoundSource.AMBIENT, 20.0F, 0.9F + this.f_107223_.m_188501_() * 0.15F, true);
+               this.f_107208_.playLocalSound(this.f_107212_, this.f_107213_, this.f_107214_, soundevent, SoundSource.AMBIENT, 20.0F, 0.9F + this.f_107223_.m_188501_() * 0.15F, true);
             }
 
             this.m_107274_();
@@ -242,11 +_,11 @@
 
       private boolean m_106798_() {
          Minecraft minecraft = Minecraft.m_91087_();
-         return minecraft.f_91063_.m_109153_().m_90583_().m_82531_(this.f_107212_, this.f_107213_, this.f_107214_) >= 256.0D;
+         return minecraft.f_91063_.m_109153_().m_90583_().distanceToSqr(this.f_107212_, this.f_107213_, this.f_107214_).compareTo(me.alphamode.mclong.core.BigConstants.DISTANCE) >= 0;
       }
 
-      private void m_106767_(double p_106768_, double p_106769_, double p_106770_, double p_106771_, double p_106772_, double p_106773_, int[] p_106774_, int[] p_106775_, boolean p_106776_, boolean p_106777_) {
-         FireworkParticles.SparkParticle fireworkparticles$sparkparticle = (FireworkParticles.SparkParticle)this.f_106755_.m_107370_(ParticleTypes.f_123815_, p_106768_, p_106769_, p_106770_, p_106771_, p_106772_, p_106773_);
+      private void createParticle(me.alphamode.mclong.math.BigDecimal p_106768_, me.alphamode.mclong.math.BigDecimal p_106769_, me.alphamode.mclong.math.BigDecimal p_106770_, double p_106771_, double p_106772_, double p_106773_, int[] p_106774_, int[] p_106775_, boolean p_106776_, boolean p_106777_) {
+         FireworkParticles.SparkParticle fireworkparticles$sparkparticle = (FireworkParticles.SparkParticle)this.f_106755_.createParticle(ParticleTypes.f_123815_, p_106768_, p_106769_, p_106770_, p_106771_, p_106772_, p_106773_);
          fireworkparticles$sparkparticle.m_106727_(p_106776_);
          fireworkparticles$sparkparticle.m_106729_(p_106777_);
          fireworkparticles$sparkparticle.m_107271_(0.99F);
@@ -259,9 +_,9 @@
       }
 
       private void m_106778_(double p_106779_, int p_106780_, int[] p_106781_, int[] p_106782_, boolean p_106783_, boolean p_106784_) {
-         double d0 = this.f_107212_;
-         double d1 = this.f_107213_;
-         double d2 = this.f_107214_;
+         me.alphamode.mclong.math.BigDecimal d0 = this.f_107212_;
+         me.alphamode.mclong.math.BigDecimal d1 = this.f_107213_;
+         me.alphamode.mclong.math.BigDecimal d2 = this.f_107214_;
 
          for(int i = -p_106780_; i <= p_106780_; ++i) {
             for(int j = -p_106780_; j <= p_106780_; ++j) {
@@ -270,7 +_,7 @@
                   double d4 = (double)i + (this.f_107223_.m_188500_() - this.f_107223_.m_188500_()) * 0.5D;
                   double d5 = (double)k + (this.f_107223_.m_188500_() - this.f_107223_.m_188500_()) * 0.5D;
                   double d6 = Math.sqrt(d3 * d3 + d4 * d4 + d5 * d5) / p_106779_ + this.f_107223_.m_188583_() * 0.05D;
-                  this.m_106767_(d0, d1, d2, d3 / d6, d4 / d6, d5 / d6, p_106781_, p_106782_, p_106783_, p_106784_);
+                  this.createParticle(d0, d1, d2, d3 / d6, d4 / d6, d5 / d6, p_106781_, p_106782_, p_106783_, p_106784_);
                   if (i != -p_106780_ && i != p_106780_ && j != -p_106780_ && j != p_106780_) {
                      k += p_106780_ * 2 - 1;
                   }
@@ -283,7 +_,7 @@
       private void m_106785_(double p_106786_, double[][] p_106787_, int[] p_106788_, int[] p_106789_, boolean p_106790_, boolean p_106791_, boolean p_106792_) {
          double d0 = p_106787_[0][0];
          double d1 = p_106787_[0][1];
-         this.m_106767_(this.f_107212_, this.f_107213_, this.f_107214_, d0 * p_106786_, d1 * p_106786_, 0.0D, p_106788_, p_106789_, p_106790_, p_106791_);
+         this.createParticle(this.f_107212_, this.f_107213_, this.f_107214_, d0 * p_106786_, d1 * p_106786_, 0.0D, p_106788_, p_106789_, p_106790_, p_106791_);
          float f = this.f_107223_.m_188501_() * (float)Math.PI;
          double d2 = p_106792_ ? 0.034D : 0.34D;
 
@@ -303,7 +_,7 @@
                   d9 *= Math.cos(d3);
 
                   for(double d12 = -1.0D; d12 <= 1.0D; d12 += 2.0D) {
-                     this.m_106767_(this.f_107212_, this.f_107213_, this.f_107214_, d9 * d12, d10, d11 * d12, p_106788_, p_106789_, p_106790_, p_106791_);
+                     this.createParticle(this.f_107212_, this.f_107213_, this.f_107214_, d9 * d12, d10, d11 * d12, p_106788_, p_106789_, p_106790_, p_106791_);
                   }
                }
 
@@ -322,7 +_,7 @@
             double d2 = this.f_107215_ * 0.5D + this.f_107223_.m_188583_() * 0.15D + d0;
             double d3 = this.f_107217_ * 0.5D + this.f_107223_.m_188583_() * 0.15D + d1;
             double d4 = this.f_107216_ * 0.5D + this.f_107223_.m_188500_() * 0.5D;
-            this.m_106767_(this.f_107212_, this.f_107213_, this.f_107214_, d2, d4, d3, p_106794_, p_106795_, p_106796_, p_106797_);
+            this.createParticle(this.f_107212_, this.f_107213_, this.f_107214_, d2, d4, d3, p_106794_, p_106795_, p_106796_, p_106797_);
          }
 
       }
