--- a/net/minecraft/client/particle/FlameParticle.java
+++ b/net/minecraft/client/particle/FlameParticle.java
@@ -8,8 +_,8 @@
 
 @OnlyIn(Dist.CLIENT)
 public class FlameParticle extends RisingParticle {
-   FlameParticle(ClientLevel p_106800_, double p_106801_, double p_106802_, double p_106803_, double p_106804_, double p_106805_, double p_106806_) {
-      super(p_106800_, p_106801_, p_106802_, p_106803_, p_106804_, p_106805_, p_106806_);
+   FlameParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
    }
 
    public ParticleRenderType m_7556_() {
@@ -48,8 +_,8 @@
          this.f_106825_ = p_106827_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106838_, ClientLevel p_106839_, double p_106840_, double p_106841_, double p_106842_, double p_106843_, double p_106844_, double p_106845_) {
-         FlameParticle flameparticle = new FlameParticle(p_106839_, p_106840_, p_106841_, p_106842_, p_106843_, p_106844_, p_106845_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         FlameParticle flameparticle = new FlameParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
          flameparticle.m_108335_(this.f_106825_);
          return flameparticle;
       }
@@ -63,8 +_,8 @@
          this.f_172111_ = p_172113_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_172124_, ClientLevel p_172125_, double p_172126_, double p_172127_, double p_172128_, double p_172129_, double p_172130_, double p_172131_) {
-         FlameParticle flameparticle = new FlameParticle(p_172125_, p_172126_, p_172127_, p_172128_, p_172129_, p_172130_, p_172131_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         FlameParticle flameparticle = new FlameParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
          flameparticle.m_108335_(this.f_172111_);
          flameparticle.m_6569_(0.5F);
          return flameparticle;
