--- a/net/minecraft/client/particle/HeartParticle.java
+++ b/net/minecraft/client/particle/HeartParticle.java
@@ -8,8 +_,8 @@
 
 @OnlyIn(Dist.CLIENT)
 public class HeartParticle extends TextureSheetParticle {
-   HeartParticle(ClientLevel p_106847_, double p_106848_, double p_106849_, double p_106850_) {
-      super(p_106847_, p_106848_, p_106849_, p_106850_, 0.0D, 0.0D, 0.0D);
+   HeartParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(pLevel, pX, pY, pZ, 0.0D, 0.0D, 0.0D);
       this.f_172259_ = true;
       this.f_172258_ = 0.86F;
       this.f_107215_ *= (double)0.01F;
@@ -37,8 +_,8 @@
          this.f_106861_ = p_106863_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106874_, ClientLevel p_106875_, double p_106876_, double p_106877_, double p_106878_, double p_106879_, double p_106880_, double p_106881_) {
-         HeartParticle heartparticle = new HeartParticle(p_106875_, p_106876_, p_106877_ + 0.5D, p_106878_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         HeartParticle heartparticle = new HeartParticle(pLevel, pX, pY.add(me.alphamode.mclong.core.BigConstants.AABB), pZ);
          heartparticle.m_108335_(this.f_106861_);
          heartparticle.m_107253_(1.0F, 1.0F, 1.0F);
          return heartparticle;
@@ -53,8 +_,8 @@
          this.f_106882_ = p_106884_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106895_, ClientLevel p_106896_, double p_106897_, double p_106898_, double p_106899_, double p_106900_, double p_106901_, double p_106902_) {
-         HeartParticle heartparticle = new HeartParticle(p_106896_, p_106897_, p_106898_, p_106899_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         HeartParticle heartparticle = new HeartParticle(pLevel, pX, pY, pZ);
          heartparticle.m_108335_(this.f_106882_);
          return heartparticle;
       }
