--- a/net/minecraft/client/particle/HugeExplosionParticle.java
+++ b/net/minecraft/client/particle/HugeExplosionParticle.java
@@ -9,16 +_,16 @@
 public class HugeExplosionParticle extends TextureSheetParticle {
    private final SpriteSet f_106903_;
 
-   protected HugeExplosionParticle(ClientLevel p_106905_, double p_106906_, double p_106907_, double p_106908_, double p_106909_, SpriteSet p_106910_) {
-      super(p_106905_, p_106906_, p_106907_, p_106908_, 0.0D, 0.0D, 0.0D);
+   protected HugeExplosionParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pQuadSizeMultiplier, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ, 0.0D, 0.0D, 0.0D);
       this.f_107225_ = 6 + this.f_107223_.m_188503_(4);
       float f = this.f_107223_.m_188501_() * 0.6F + 0.4F;
       this.f_107227_ = f;
       this.f_107228_ = f;
       this.f_107229_ = f;
-      this.f_107663_ = 2.0F * (1.0F - (float)p_106909_ * 0.5F);
-      this.f_106903_ = p_106910_;
-      this.m_108339_(p_106910_);
+      this.f_107663_ = 2.0F * (1.0F - (float)pQuadSizeMultiplier * 0.5F);
+      this.f_106903_ = pSprites;
+      this.m_108339_(pSprites);
    }
 
    public int m_6355_(float p_106921_) {
@@ -48,8 +_,8 @@
          this.f_106923_ = p_106925_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_106936_, ClientLevel p_106937_, double p_106938_, double p_106939_, double p_106940_, double p_106941_, double p_106942_, double p_106943_) {
-         return new HugeExplosionParticle(p_106937_, p_106938_, p_106939_, p_106940_, p_106941_, this.f_106923_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new HugeExplosionParticle(pLevel, pX, pY, pZ, pXSpeed, this.f_106923_);
       }
    }
 }
