--- a/net/minecraft/client/particle/ItemPickupParticle.java
+++ b/net/minecraft/client/particle/ItemPickupParticle.java
@@ -28,7 +_,7 @@
    }
 
    private ItemPickupParticle(EntityRenderDispatcher p_107029_, RenderBuffers p_107030_, ClientLevel p_107031_, Entity p_107032_, Entity p_107033_, Vec3 p_107034_) {
-      super(p_107031_, p_107032_.m_20185_(), p_107032_.m_20186_(), p_107032_.m_20189_(), p_107034_.f_82479_, p_107034_.f_82480_, p_107034_.f_82481_);
+      super(p_107031_, p_107032_.m_20185_(), p_107032_.m_20186_(), p_107032_.m_20189_(), p_107034_.m_7096_(), p_107034_.m_7098_(), p_107034_.m_7094_());
       this.f_107020_ = p_107030_;
       this.f_107021_ = this.m_107036_(p_107032_);
       this.f_107017_ = p_107033_;
@@ -46,15 +_,17 @@
    public void m_5744_(VertexConsumer p_107039_, Camera p_107040_, float p_107041_) {
       float f = ((float)this.f_107018_ + p_107041_) / 3.0F;
       f *= f;
-      double d0 = Mth.m_14139_((double)p_107041_, this.f_107017_.f_19790_, this.f_107017_.m_20185_());
-      double d1 = Mth.m_14139_((double)p_107041_, this.f_107017_.f_19791_, (this.f_107017_.m_20186_() + this.f_107017_.m_20188_()) / 2.0D);
-      double d2 = Mth.m_14139_((double)p_107041_, this.f_107017_.f_19792_, this.f_107017_.m_20189_());
-      double d3 = Mth.m_14139_((double)f, this.f_107021_.m_20185_(), d0);
-      double d4 = Mth.m_14139_((double)f, this.f_107021_.m_20186_(), d1);
-      double d5 = Mth.m_14139_((double)f, this.f_107021_.m_20189_(), d2);
+      java.math.BigDecimal lerp = java.math.BigDecimal.valueOf(p_107041_);
+      java.math.BigDecimal d0 = Mth.lerp(lerp, this.f_107017_.f_19790_, this.f_107017_.getBigX());
+      java.math.BigDecimal d1 = Mth.lerp(lerp, this.f_107017_.f_19791_, (this.f_107017_.getBigY().add(java.math.BigDecimal.valueOf(this.f_107017_.m_20188_()))).divide(me.alphamode.mclong.core.BigConstants.TWO));
+      java.math.BigDecimal d2 = Mth.lerp(lerp, this.f_107017_.f_19792_, this.f_107017_.getBigZ());
+      java.math.BigDecimal lerp2 = java.math.BigDecimal.valueOf(f);
+      java.math.BigDecimal d3 = Mth.lerp(lerp2, this.f_107021_.getBigX(), d0);
+      java.math.BigDecimal d4 = Mth.lerp(lerp2, this.f_107021_.getBigY(), d1);
+      java.math.BigDecimal d5 = Mth.lerp(lerp2, this.f_107021_.getBigZ(), d2);
       MultiBufferSource.BufferSource multibuffersource$buffersource = this.f_107020_.m_110104_();
       Vec3 vec3 = p_107040_.m_90583_();
-      this.f_107019_.m_114384_(this.f_107021_, d3 - vec3.m_7096_(), d4 - vec3.m_7098_(), d5 - vec3.m_7094_(), this.f_107021_.m_146908_(), p_107041_, new PoseStack(), multibuffersource$buffersource, this.f_107019_.m_114394_(this.f_107021_, p_107041_));
+      this.f_107019_.m_114384_(this.f_107021_, d3.subtract(vec3.getX()).doubleValue(), d4.subtract(vec3.getY()).doubleValue(), d5.subtract(vec3.getZ()).doubleValue(), this.f_107021_.m_146908_(), p_107041_, new PoseStack(), multibuffersource$buffersource, this.f_107019_.m_114394_(this.f_107021_, p_107041_));
       multibuffersource$buffersource.m_109911_();
    }
 
