--- a/net/minecraft/client/particle/LargeSmokeParticle.java
+++ b/net/minecraft/client/particle/LargeSmokeParticle.java
@@ -7,8 +_,8 @@
 
 @OnlyIn(Dist.CLIENT)
 public class LargeSmokeParticle extends SmokeParticle {
-   protected LargeSmokeParticle(ClientLevel p_107044_, double p_107045_, double p_107046_, double p_107047_, double p_107048_, double p_107049_, double p_107050_, SpriteSet p_107051_) {
-      super(p_107044_, p_107045_, p_107046_, p_107047_, p_107048_, p_107049_, p_107050_, 2.5F, p_107051_);
+   protected LargeSmokeParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, 2.5F, pSprites);
    }
 
    @OnlyIn(Dist.CLIENT)
@@ -19,8 +_,8 @@
          this.f_107052_ = p_107054_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_107065_, ClientLevel p_107066_, double p_107067_, double p_107068_, double p_107069_, double p_107070_, double p_107071_, double p_107072_) {
-         return new LargeSmokeParticle(p_107066_, p_107067_, p_107068_, p_107069_, p_107070_, p_107071_, p_107072_, this.f_107052_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new LargeSmokeParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, this.f_107052_);
       }
    }
 }
