--- a/net/minecraft/client/particle/MobAppearanceParticle.java
+++ b/net/minecraft/client/particle/MobAppearanceParticle.java
@@ -23,7 +_,7 @@
    private final Model f_107111_;
    private final RenderType f_107112_ = RenderType.m_110473_(ElderGuardianRenderer.f_114116_);
 
-   MobAppearanceParticle(ClientLevel p_107114_, double p_107115_, double p_107116_, double p_107117_) {
+   MobAppearanceParticle(ClientLevel p_107114_, me.alphamode.mclong.math.BigDecimal p_107115_, me.alphamode.mclong.math.BigDecimal p_107116_, me.alphamode.mclong.math.BigDecimal p_107117_) {
       super(p_107114_, p_107115_, p_107116_, p_107117_);
       this.f_107111_ = new GuardianModel(Minecraft.m_91087_().m_167973_().m_171103_(ModelLayers.f_171140_));
       this.f_107226_ = 0.0F;
@@ -50,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public static class Provider implements ParticleProvider<SimpleParticleType> {
-      public Particle m_6966_(SimpleParticleType p_107140_, ClientLevel p_107141_, double p_107142_, double p_107143_, double p_107144_, double p_107145_, double p_107146_, double p_107147_) {
+      public Particle createParticle(SimpleParticleType p_107140_, ClientLevel p_107141_, me.alphamode.mclong.math.BigDecimal p_107142_, me.alphamode.mclong.math.BigDecimal p_107143_, me.alphamode.mclong.math.BigDecimal p_107144_, double p_107145_, double p_107146_, double p_107147_) {
          return new MobAppearanceParticle(p_107141_, p_107142_, p_107143_, p_107144_);
       }
    }
