--- a/net/minecraft/client/particle/NoRenderParticle.java
+++ b/net/minecraft/client/particle/NoRenderParticle.java
@@ -8,12 +_,12 @@
 
 @OnlyIn(Dist.CLIENT)
 public class NoRenderParticle extends Particle {
-   protected NoRenderParticle(ClientLevel p_107149_, double p_107150_, double p_107151_, double p_107152_) {
-      super(p_107149_, p_107150_, p_107151_, p_107152_);
+   protected NoRenderParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(pLevel, pX, pY, pZ);
    }
 
-   protected NoRenderParticle(ClientLevel p_107154_, double p_107155_, double p_107156_, double p_107157_, double p_107158_, double p_107159_, double p_107160_) {
-      super(p_107154_, p_107155_, p_107156_, p_107157_, p_107158_, p_107159_, p_107160_);
+   protected NoRenderParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
    }
 
    public final void m_5744_(VertexConsumer p_107162_, Camera p_107163_, float p_107164_) {
