--- a/net/minecraft/client/particle/NoteParticle.java
+++ b/net/minecraft/client/particle/NoteParticle.java
@@ -8,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class NoteParticle extends TextureSheetParticle {
-   NoteParticle(ClientLevel p_107167_, double p_107168_, double p_107169_, double p_107170_, double p_107171_) {
+   NoteParticle(ClientLevel p_107167_, me.alphamode.mclong.math.BigDecimal p_107168_, me.alphamode.mclong.math.BigDecimal p_107169_, me.alphamode.mclong.math.BigDecimal p_107170_, double p_107171_) {
       super(p_107167_, p_107168_, p_107169_, p_107170_, 0.0D, 0.0D, 0.0D);
       this.f_172258_ = 0.66F;
       this.f_172259_ = true;
@@ -39,7 +_,7 @@
          this.f_107183_ = p_107185_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_107196_, ClientLevel p_107197_, double p_107198_, double p_107199_, double p_107200_, double p_107201_, double p_107202_, double p_107203_) {
+      public Particle createParticle(SimpleParticleType p_107196_, ClientLevel p_107197_, me.alphamode.mclong.math.BigDecimal p_107198_, me.alphamode.mclong.math.BigDecimal p_107199_, me.alphamode.mclong.math.BigDecimal p_107200_, double p_107201_, double p_107202_, double p_107203_) {
          NoteParticle noteparticle = new NoteParticle(p_107197_, p_107198_, p_107199_, p_107200_, p_107201_);
          noteparticle.m_108335_(this.f_107183_);
          return noteparticle;
