--- a/net/minecraft/client/particle/ParticleProvider.java
+++ b/net/minecraft/client/particle/ParticleProvider.java
@@ -9,11 +_,11 @@
 @OnlyIn(Dist.CLIENT)
 public interface ParticleProvider<T extends ParticleOptions> {
    @Nullable
-   Particle m_6966_(T p_107421_, ClientLevel p_107422_, double p_107423_, double p_107424_, double p_107425_, double p_107426_, double p_107427_, double p_107428_);
+   Particle createParticle(T pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed);
 
    @OnlyIn(Dist.CLIENT)
    public interface Sprite<T extends ParticleOptions> {
       @Nullable
-      TextureSheetParticle m_272232_(T p_273550_, ClientLevel p_273071_, double p_273160_, double p_273576_, double p_272710_, double p_273652_, double p_273457_, double p_272840_);
+      TextureSheetParticle createParticle(T pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed);
    }
 }
