--- a/net/minecraft/client/particle/Particle.java
+++ b/net/minecraft/client/particle/Particle.java
@@ -21,12 +_,12 @@
    private static final AABB f_107206_ = new AABB(0.0D, 0.0D, 0.0D, 0.0D, 0.0D, 0.0D);
    private static final double f_197408_ = Mth.m_144952_(100.0D);
    protected final ClientLevel f_107208_;
-   protected double f_107209_;
-   protected double f_107210_;
-   protected double f_107211_;
-   protected double f_107212_;
-   protected double f_107213_;
-   protected double f_107214_;
+   protected me.alphamode.mclong.math.BigDecimal f_107209_;
+   protected me.alphamode.mclong.math.BigDecimal f_107210_;
+   protected me.alphamode.mclong.math.BigDecimal f_107211_;
+   protected me.alphamode.mclong.math.BigDecimal f_107212_;
+   protected me.alphamode.mclong.math.BigDecimal f_107213_;
+   protected me.alphamode.mclong.math.BigDecimal f_107214_;
    protected double f_107215_;
    protected double f_107216_;
    protected double f_107217_;
@@ -50,17 +_,17 @@
    protected float f_172258_ = 0.98F;
    protected boolean f_172259_ = false;
 
-   protected Particle(ClientLevel p_107234_, double p_107235_, double p_107236_, double p_107237_) {
+   protected Particle(ClientLevel p_107234_, me.alphamode.mclong.math.BigDecimal p_107235_, me.alphamode.mclong.math.BigDecimal p_107236_, me.alphamode.mclong.math.BigDecimal p_107237_) {
       this.f_107208_ = p_107234_;
       this.m_107250_(0.2F, 0.2F);
-      this.m_107264_(p_107235_, p_107236_, p_107237_);
+      this.setPos(p_107235_, p_107236_, p_107237_);
       this.f_107209_ = p_107235_;
       this.f_107210_ = p_107236_;
       this.f_107211_ = p_107237_;
       this.f_107225_ = (int)(4.0F / (this.f_107223_.m_188501_() * 0.9F + 0.1F));
    }
 
-   public Particle(ClientLevel p_107239_, double p_107240_, double p_107241_, double p_107242_, double p_107243_, double p_107244_, double p_107245_) {
+   public Particle(ClientLevel p_107239_, me.alphamode.mclong.math.BigDecimal p_107240_, me.alphamode.mclong.math.BigDecimal p_107241_, me.alphamode.mclong.math.BigDecimal p_107242_, double p_107243_, double p_107244_, double p_107245_) {
       this(p_107239_, p_107240_, p_107241_, p_107242_);
       this.f_107215_ = p_107243_ + (Math.random() * 2.0D - 1.0D) * (double)0.4F;
       this.f_107216_ = p_107244_ + (Math.random() * 2.0D - 1.0D) * (double)0.4F;
@@ -150,20 +_,22 @@
          this.f_107221_ = p_107251_;
          this.f_107222_ = p_107252_;
          AABB aabb = this.m_107277_();
-         double d0 = (aabb.f_82288_ + aabb.f_82291_ - (double)p_107251_) / 2.0D;
-         double d1 = (aabb.f_82290_ + aabb.f_82293_ - (double)p_107251_) / 2.0D;
-         this.m_107259_(new AABB(d0, aabb.f_82289_, d1, d0 + (double)this.f_107221_, aabb.f_82289_ + (double)this.f_107222_, d1 + (double)this.f_107221_));
+         var val = me.alphamode.mclong.math.BigDecimal.valueOf(p_107251_);
+         me.alphamode.mclong.math.BigDecimal d0 = (aabb.f_82288_.add(aabb.f_82291_).subtract(val)).divide(me.alphamode.mclong.core.BigConstants.TWO);
+         me.alphamode.mclong.math.BigDecimal d1 = (aabb.f_82290_.add(aabb.f_82293_).subtract(val)).divide(me.alphamode.mclong.core.BigConstants.TWO);
+         var width = me.alphamode.mclong.math.BigDecimal.valueOf(this.f_107221_);
+         this.m_107259_(new AABB(d0, aabb.f_82289_, d1, d0.add(width), aabb.f_82289_.add(me.alphamode.mclong.math.BigDecimal.valueOf(this.f_107222_)), d1.add(width)));
       }
 
    }
 
-   public void m_107264_(double p_107265_, double p_107266_, double p_107267_) {
+   public void setPos(me.alphamode.mclong.math.BigDecimal p_107265_, me.alphamode.mclong.math.BigDecimal p_107266_, me.alphamode.mclong.math.BigDecimal p_107267_) {
       this.f_107212_ = p_107265_;
       this.f_107213_ = p_107266_;
       this.f_107214_ = p_107267_;
       float f = this.f_107221_ / 2.0F;
       float f1 = this.f_107222_;
-      this.m_107259_(new AABB(p_107265_ - (double)f, p_107266_, p_107267_ - (double)f, p_107265_ + (double)f, p_107266_ + (double)f1, p_107267_ + (double)f));
+      this.m_107259_(new AABB(p_107265_.subtract(f), p_107266_, p_107267_.subtract(f), p_107265_.add(f), p_107266_.add(f1), p_107267_.add(f)));
    }
 
    public void m_6257_(double p_107246_, double p_107247_, double p_107248_) {
@@ -173,9 +_,9 @@
          double d2 = p_107248_;
          if (this.f_107219_ && (p_107246_ != 0.0D || p_107247_ != 0.0D || p_107248_ != 0.0D) && p_107246_ * p_107246_ + p_107247_ * p_107247_ + p_107248_ * p_107248_ < f_197408_) {
             Vec3 vec3 = Entity.m_198894_((Entity)null, new Vec3(p_107246_, p_107247_, p_107248_), this.m_107277_(), this.f_107208_, List.of());
-            p_107246_ = vec3.f_82479_;
-            p_107247_ = vec3.f_82480_;
-            p_107248_ = vec3.f_82481_;
+            p_107246_ = vec3.m_7096_();
+            p_107247_ = vec3.m_7098_();
+            p_107248_ = vec3.m_7094_();
          }
 
          if (p_107246_ != 0.0D || p_107247_ != 0.0D || p_107248_ != 0.0D) {
@@ -201,13 +_,13 @@
 
    protected void m_107275_() {
       AABB aabb = this.m_107277_();
-      this.f_107212_ = (aabb.f_82288_ + aabb.f_82291_) / 2.0D;
+      this.f_107212_ = (aabb.f_82288_.add(aabb.f_82291_)).divide(me.alphamode.mclong.core.BigConstants.TWO);
       this.f_107213_ = aabb.f_82289_;
-      this.f_107214_ = (aabb.f_82290_ + aabb.f_82293_) / 2.0D;
+      this.f_107214_ = (aabb.f_82290_.add(aabb.f_82293_)).divide(me.alphamode.mclong.core.BigConstants.TWO);
    }
 
    protected int m_6355_(float p_107249_) {
-      BlockPos blockpos = BlockPos.m_274561_(this.f_107212_, this.f_107213_, this.f_107214_);
+      BlockPos blockpos = BlockPos.containing(this.f_107212_, this.f_107213_, this.f_107214_);
       return this.f_107208_.m_46805_(blockpos) ? LevelRenderer.m_109541_(this.f_107208_, blockpos) : 0;
    }
 
