--- a/net/minecraft/client/particle/PlayerCloudParticle.java
+++ b/net/minecraft/client/particle/PlayerCloudParticle.java
@@ -50,7 +_,7 @@
             double d0 = player.m_20186_();
             if (this.f_107213_ > d0) {
                this.f_107213_ += (d0 - this.f_107213_) * 0.2D;
-               this.f_107216_ += (player.m_20184_().f_82480_ - this.f_107216_) * 0.2D;
+               this.f_107216_ += (player.m_20184_().m_7098_() - this.f_107216_) * 0.2D;
                this.m_107264_(this.f_107212_, this.f_107213_, this.f_107214_);
             }
          }
