--- a/net/minecraft/client/particle/SculkChargeParticle.java
+++ b/net/minecraft/client/particle/SculkChargeParticle.java
@@ -9,13 +_,13 @@
 public class SculkChargeParticle extends TextureSheetParticle {
    private final SpriteSet f_233890_;
 
-   SculkChargeParticle(ClientLevel p_233892_, double p_233893_, double p_233894_, double p_233895_, double p_233896_, double p_233897_, double p_233898_, SpriteSet p_233899_) {
-      super(p_233892_, p_233893_, p_233894_, p_233895_, p_233896_, p_233897_, p_233898_);
+   SculkChargeParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, SpriteSet pSprite) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
       this.f_172258_ = 0.96F;
-      this.f_233890_ = p_233899_;
+      this.f_233890_ = pSprite;
       this.m_6569_(1.5F);
       this.f_107219_ = false;
-      this.m_108339_(p_233899_);
+      this.m_108339_(pSprite);
    }
 
    public int m_6355_(float p_233902_) {
@@ -33,7 +_,7 @@
 
    @OnlyIn(Dist.CLIENT)
    public static record Provider(SpriteSet f_233904_) implements ParticleProvider<SculkChargeParticleOptions> {
-      public Particle m_6966_(SculkChargeParticleOptions p_233918_, ClientLevel p_233919_, double p_233920_, double p_233921_, double p_233922_, double p_233923_, double p_233924_, double p_233925_) {
+      public Particle createParticle(SculkChargeParticleOptions p_233918_, ClientLevel p_233919_, me.alphamode.mclong.math.BigDecimal p_233920_, me.alphamode.mclong.math.BigDecimal p_233921_, me.alphamode.mclong.math.BigDecimal p_233922_, double p_233923_, double p_233924_, double p_233925_) {
          SculkChargeParticle sculkchargeparticle = new SculkChargeParticle(p_233919_, p_233920_, p_233921_, p_233922_, p_233923_, p_233924_, p_233925_, this.f_233904_);
          sculkchargeparticle.m_107271_(1.0F);
          sculkchargeparticle.m_172260_(p_233923_, p_233924_, p_233925_);
