--- a/net/minecraft/client/particle/SimpleAnimatedParticle.java
+++ b/net/minecraft/client/particle/SimpleAnimatedParticle.java
@@ -12,7 +_,7 @@
    private float f_107642_;
    private boolean f_107643_;
 
-   protected SimpleAnimatedParticle(ClientLevel p_107647_, double p_107648_, double p_107649_, double p_107650_, SpriteSet p_107651_, float p_107652_) {
+   protected SimpleAnimatedParticle(ClientLevel p_107647_, me.alphamode.mclong.math.BigDecimal p_107648_, me.alphamode.mclong.math.BigDecimal p_107649_, me.alphamode.mclong.math.BigDecimal p_107650_, SpriteSet p_107651_, float p_107652_) {
       super(p_107647_, p_107648_, p_107649_, p_107650_);
       this.f_172258_ = 0.91F;
       this.f_107226_ = p_107652_;
