--- a/net/minecraft/client/particle/SimpleAnimatedParticle.java
+++ b/net/minecraft/client/particle/SimpleAnimatedParticle.java
@@ -12,11 +_,11 @@
    private float f_107642_;
    private boolean f_107643_;
 
-   protected SimpleAnimatedParticle(ClientLevel p_107647_, double p_107648_, double p_107649_, double p_107650_, SpriteSet p_107651_, float p_107652_) {
-      super(p_107647_, p_107648_, p_107649_, p_107650_);
+   protected SimpleAnimatedParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, SpriteSet pSprites, float pGravity) {
+      super(pLevel, pX, pY, pZ);
       this.f_172258_ = 0.91F;
-      this.f_107226_ = p_107652_;
-      this.f_107644_ = p_107651_;
+      this.f_107226_ = pGravity;
+      this.f_107644_ = pSprites;
    }
 
    public void m_107657_(int p_107658_) {
