--- a/net/minecraft/client/particle/SmokeParticle.java
+++ b/net/minecraft/client/particle/SmokeParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SmokeParticle extends BaseAshSmokeParticle {
-   protected SmokeParticle(ClientLevel p_107685_, double p_107686_, double p_107687_, double p_107688_, double p_107689_, double p_107690_, double p_107691_, float p_107692_, SpriteSet p_107693_) {
+   protected SmokeParticle(ClientLevel p_107685_, me.alphamode.mclong.math.BigDecimal p_107686_, me.alphamode.mclong.math.BigDecimal p_107687_, me.alphamode.mclong.math.BigDecimal p_107688_, double p_107689_, double p_107690_, double p_107691_, float p_107692_, SpriteSet p_107693_) {
       super(p_107685_, p_107686_, p_107687_, p_107688_, 0.1F, 0.1F, 0.1F, p_107689_, p_107690_, p_107691_, p_107692_, p_107693_, 0.3F, 8, -0.1F, true);
    }
 
@@ -19,7 +_,7 @@
          this.f_107694_ = p_107696_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_107707_, ClientLevel p_107708_, double p_107709_, double p_107710_, double p_107711_, double p_107712_, double p_107713_, double p_107714_) {
+      public Particle createParticle(SimpleParticleType p_107707_, ClientLevel p_107708_, me.alphamode.mclong.math.BigDecimal p_107709_, me.alphamode.mclong.math.BigDecimal p_107710_, me.alphamode.mclong.math.BigDecimal p_107711_, double p_107712_, double p_107713_, double p_107714_) {
          return new SmokeParticle(p_107708_, p_107709_, p_107710_, p_107711_, p_107712_, p_107713_, p_107714_, 1.0F, this.f_107694_);
       }
    }
