--- a/net/minecraft/client/particle/SnowflakeParticle.java
+++ b/net/minecraft/client/particle/SnowflakeParticle.java
@@ -9,7 +_,7 @@
 public class SnowflakeParticle extends TextureSheetParticle {
    private final SpriteSet f_172290_;
 
-   protected SnowflakeParticle(ClientLevel p_172292_, double p_172293_, double p_172294_, double p_172295_, double p_172296_, double p_172297_, double p_172298_, SpriteSet p_172299_) {
+   protected SnowflakeParticle(ClientLevel p_172292_, me.alphamode.mclong.math.BigDecimal p_172293_, me.alphamode.mclong.math.BigDecimal p_172294_, me.alphamode.mclong.math.BigDecimal p_172295_, double p_172296_, double p_172297_, double p_172298_, SpriteSet p_172299_) {
       super(p_172292_, p_172293_, p_172294_, p_172295_);
       this.f_107226_ = 0.225F;
       this.f_172258_ = 1.0F;
@@ -42,7 +_,7 @@
          this.f_172302_ = p_172304_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_172315_, ClientLevel p_172316_, double p_172317_, double p_172318_, double p_172319_, double p_172320_, double p_172321_, double p_172322_) {
+      public Particle createParticle(SimpleParticleType p_172315_, ClientLevel p_172316_, me.alphamode.mclong.math.BigDecimal p_172317_, me.alphamode.mclong.math.BigDecimal p_172318_, me.alphamode.mclong.math.BigDecimal p_172319_, double p_172320_, double p_172321_, double p_172322_) {
          SnowflakeParticle snowflakeparticle = new SnowflakeParticle(p_172316_, p_172317_, p_172318_, p_172319_, p_172320_, p_172321_, p_172322_, this.f_172302_);
          snowflakeparticle.m_107253_(0.923F, 0.964F, 0.999F);
          return snowflakeparticle;
