--- a/net/minecraft/client/particle/SpitParticle.java
+++ b/net/minecraft/client/particle/SpitParticle.java
@@ -7,8 +_,8 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SpitParticle extends ExplodeParticle {
-   SpitParticle(ClientLevel p_107888_, double p_107889_, double p_107890_, double p_107891_, double p_107892_, double p_107893_, double p_107894_, SpriteSet p_107895_) {
-      super(p_107888_, p_107889_, p_107890_, p_107891_, p_107892_, p_107893_, p_107894_, p_107895_);
+   SpitParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, pSprites);
       this.f_107226_ = 0.5F;
    }
 
@@ -20,8 +_,8 @@
          this.f_107907_ = p_107909_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_107920_, ClientLevel p_107921_, double p_107922_, double p_107923_, double p_107924_, double p_107925_, double p_107926_, double p_107927_) {
-         return new SpitParticle(p_107921_, p_107922_, p_107923_, p_107924_, p_107925_, p_107926_, p_107927_, this.f_107907_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new SpitParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, this.f_107907_);
       }
    }
 }
