--- a/net/minecraft/client/particle/SplashParticle.java
+++ b/net/minecraft/client/particle/SplashParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SplashParticle extends WaterDropParticle {
-   SplashParticle(ClientLevel p_107929_, double p_107930_, double p_107931_, double p_107932_, double p_107933_, double p_107934_, double p_107935_) {
+   SplashParticle(ClientLevel p_107929_, me.alphamode.mclong.math.BigDecimal p_107930_, me.alphamode.mclong.math.BigDecimal p_107931_, me.alphamode.mclong.math.BigDecimal p_107932_, double p_107933_, double p_107934_, double p_107935_) {
       super(p_107929_, p_107930_, p_107931_, p_107932_);
       this.f_107226_ = 0.04F;
       if (p_107934_ == 0.0D && (p_107933_ != 0.0D || p_107935_ != 0.0D)) {
@@ -26,8 +_,8 @@
          this.f_107945_ = p_107947_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_107958_, ClientLevel p_107959_, double p_107960_, double p_107961_, double p_107962_, double p_107963_, double p_107964_, double p_107965_) {
-         SplashParticle splashparticle = new SplashParticle(p_107959_, p_107960_, p_107961_, p_107962_, p_107963_, p_107964_, p_107965_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         SplashParticle splashparticle = new SplashParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
          splashparticle.m_108335_(this.f_107945_);
          return splashparticle;
       }
