--- a/net/minecraft/client/particle/SquidInkParticle.java
+++ b/net/minecraft/client/particle/SquidInkParticle.java
@@ -9,18 +_,18 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SquidInkParticle extends SimpleAnimatedParticle {
-   SquidInkParticle(ClientLevel p_172325_, double p_172326_, double p_172327_, double p_172328_, double p_172329_, double p_172330_, double p_172331_, int p_172332_, SpriteSet p_172333_) {
-      super(p_172325_, p_172326_, p_172327_, p_172328_, p_172333_, 0.0F);
+   SquidInkParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed, int pPackedColor, SpriteSet pSprites) {
+      super(pLevel, pX, pY, pZ, pSprites, 0.0F);
       this.f_172258_ = 0.92F;
       this.f_107663_ = 0.5F;
       this.m_107271_(1.0F);
-      this.m_107253_((float)FastColor.ARGB32.m_13665_(p_172332_), (float)FastColor.ARGB32.m_13667_(p_172332_), (float)FastColor.ARGB32.m_13669_(p_172332_));
+      this.m_107253_((float)FastColor.ARGB32.m_13665_(pPackedColor), (float)FastColor.ARGB32.m_13667_(pPackedColor), (float)FastColor.ARGB32.m_13669_(pPackedColor));
       this.f_107225_ = (int)((double)(this.f_107663_ * 12.0F) / (Math.random() * (double)0.8F + (double)0.2F));
-      this.m_108339_(p_172333_);
+      this.m_108339_(pSprites);
       this.f_107219_ = false;
-      this.f_107215_ = p_172329_;
-      this.f_107216_ = p_172330_;
-      this.f_107217_ = p_172331_;
+      this.f_107215_ = pXSpeed;
+      this.f_107216_ = pYSpeed;
+      this.f_107217_ = pZSpeed;
    }
 
    public void m_5989_() {
@@ -31,7 +_,7 @@
             this.m_107271_(1.0F - ((float)this.f_107224_ - (float)(this.f_107225_ / 2)) / (float)this.f_107225_);
          }
 
-         if (this.f_107208_.m_8055_(BlockPos.m_274561_(this.f_107212_, this.f_107213_, this.f_107214_)).m_60795_()) {
+         if (this.f_107208_.m_8055_(BlockPos.containing(this.f_107212_, this.f_107213_, this.f_107214_)).m_60795_()) {
             this.f_107216_ -= (double)0.0074F;
          }
       }
@@ -46,8 +_,8 @@
          this.f_172334_ = p_172336_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_172347_, ClientLevel p_172348_, double p_172349_, double p_172350_, double p_172351_, double p_172352_, double p_172353_, double p_172354_) {
-         return new SquidInkParticle(p_172348_, p_172349_, p_172350_, p_172351_, p_172352_, p_172353_, p_172354_, FastColor.ARGB32.m_13660_(255, 204, 31, 102), this.f_172334_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new SquidInkParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, FastColor.ARGB32.m_13660_(255, 204, 31, 102), this.f_172334_);
       }
    }
 
@@ -59,8 +_,8 @@
          this.f_107989_ = p_107991_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108002_, ClientLevel p_108003_, double p_108004_, double p_108005_, double p_108006_, double p_108007_, double p_108008_, double p_108009_) {
-         return new SquidInkParticle(p_108003_, p_108004_, p_108005_, p_108006_, p_108007_, p_108008_, p_108009_, FastColor.ARGB32.m_13660_(255, 255, 255, 255), this.f_107989_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         return new SquidInkParticle(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed, FastColor.ARGB32.m_13660_(255, 255, 255, 255), this.f_107989_);
       }
    }
 }
