--- a/net/minecraft/client/particle/SuspendedParticle.java
+++ b/net/minecraft/client/particle/SuspendedParticle.java
@@ -11,10 +_,10 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SuspendedParticle extends TextureSheetParticle {
-   SuspendedParticle(ClientLevel p_172403_, SpriteSet p_172404_, double p_172405_, double p_172406_, double p_172407_) {
-      super(p_172403_, p_172405_, p_172406_ - 0.125D, p_172407_);
+   SuspendedParticle(ClientLevel pLevel, SpriteSet pSprites, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(pLevel, pX, pY.subtract(me.alphamode.mclong.core.BigConstants.PARTICLE), pZ);
       this.m_107250_(0.01F, 0.01F);
-      this.m_108335_(p_172404_);
+      this.m_108335_(pSprites);
       this.f_107663_ *= this.f_107223_.m_188501_() * 0.6F + 0.2F;
       this.f_107225_ = (int)(16.0D / (Math.random() * 0.8D + 0.2D));
       this.f_107219_ = false;
@@ -22,10 +_,10 @@
       this.f_107226_ = 0.0F;
    }
 
-   SuspendedParticle(ClientLevel p_172409_, SpriteSet p_172410_, double p_172411_, double p_172412_, double p_172413_, double p_172414_, double p_172415_, double p_172416_) {
-      super(p_172409_, p_172411_, p_172412_ - 0.125D, p_172413_, p_172414_, p_172415_, p_172416_);
+   SuspendedParticle(ClientLevel pLevel, SpriteSet pSprites, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+      super(pLevel, pX, pY.subtract(me.alphamode.mclong.core.BigConstants.PARTICLE), pZ, pXSpeed, pYSpeed, pZSpeed);
       this.m_107250_(0.01F, 0.01F);
-      this.m_108335_(p_172410_);
+      this.m_108335_(pSprites);
       this.f_107663_ *= this.f_107223_.m_188501_() * 0.6F + 0.6F;
       this.f_107225_ = (int)(16.0D / (Math.random() * 0.8D + 0.2D));
       this.f_107219_ = false;
@@ -45,12 +_,12 @@
          this.f_108040_ = p_108042_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108053_, ClientLevel p_108054_, double p_108055_, double p_108056_, double p_108057_, double p_108058_, double p_108059_, double p_108060_) {
-         RandomSource randomsource = p_108054_.f_46441_;
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         RandomSource randomsource = pLevel.f_46441_;
          double d0 = randomsource.m_188583_() * (double)1.0E-6F;
          double d1 = randomsource.m_188583_() * (double)1.0E-4F;
          double d2 = randomsource.m_188583_() * (double)1.0E-6F;
-         SuspendedParticle suspendedparticle = new SuspendedParticle(p_108054_, this.f_108040_, p_108055_, p_108056_, p_108057_, d0, d1, d2);
+         SuspendedParticle suspendedparticle = new SuspendedParticle(pLevel, this.f_108040_, pX, pY, pZ, d0, d1, d2);
          suspendedparticle.m_107253_(0.9F, 0.4F, 0.5F);
          return suspendedparticle;
       }
@@ -64,13 +_,13 @@
          this.f_172417_ = p_172419_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_172430_, ClientLevel p_172431_, double p_172432_, double p_172433_, double p_172434_, double p_172435_, double p_172436_, double p_172437_) {
-         SuspendedParticle suspendedparticle = new SuspendedParticle(p_172431_, this.f_172417_, p_172432_, p_172433_, p_172434_, 0.0D, (double)-0.8F, 0.0D) {
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         SuspendedParticle suspendedparticle = new SuspendedParticle(pLevel, this.f_172417_, pX, pY, pZ, 0.0D, (double)-0.8F, 0.0D) {
             public Optional<ParticleGroup> m_142654_() {
                return Optional.of(ParticleGroup.f_175814_);
             }
          };
-         suspendedparticle.f_107225_ = Mth.m_216287_(p_172431_.f_46441_, 500, 1000);
+         suspendedparticle.f_107225_ = Mth.m_216287_(pLevel.f_46441_, 500, 1000);
          suspendedparticle.f_107226_ = 0.01F;
          suspendedparticle.m_107253_(0.32F, 0.5F, 0.22F);
          return suspendedparticle;
@@ -85,8 +_,8 @@
          this.f_108061_ = p_108063_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108074_, ClientLevel p_108075_, double p_108076_, double p_108077_, double p_108078_, double p_108079_, double p_108080_, double p_108081_) {
-         SuspendedParticle suspendedparticle = new SuspendedParticle(p_108075_, this.f_108061_, p_108076_, p_108077_, p_108078_);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         SuspendedParticle suspendedparticle = new SuspendedParticle(pLevel, this.f_108061_, pX, pY, pZ);
          suspendedparticle.m_107253_(0.4F, 0.4F, 0.7F);
          return suspendedparticle;
       }
@@ -100,9 +_,9 @@
          this.f_108082_ = p_108084_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108095_, ClientLevel p_108096_, double p_108097_, double p_108098_, double p_108099_, double p_108100_, double p_108101_, double p_108102_) {
-         double d0 = (double)p_108096_.f_46441_.m_188501_() * -1.9D * (double)p_108096_.f_46441_.m_188501_() * 0.1D;
-         SuspendedParticle suspendedparticle = new SuspendedParticle(p_108096_, this.f_108082_, p_108097_, p_108098_, p_108099_, 0.0D, d0, 0.0D);
+      public Particle createParticle(SimpleParticleType pType, ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+         double d0 = (double)pLevel.f_46441_.m_188501_() * -1.9D * (double)pLevel.f_46441_.m_188501_() * 0.1D;
+         SuspendedParticle suspendedparticle = new SuspendedParticle(pLevel, this.f_108082_, pX, pY, pZ, 0.0D, d0, 0.0D);
          suspendedparticle.m_107253_(0.1F, 0.1F, 0.3F);
          suspendedparticle.m_107250_(0.001F, 0.001F);
          return suspendedparticle;
