--- a/net/minecraft/client/particle/SuspendedParticle.java
+++ b/net/minecraft/client/particle/SuspendedParticle.java
@@ -11,8 +_,8 @@
 
 @OnlyIn(Dist.CLIENT)
 public class SuspendedParticle extends TextureSheetParticle {
-   SuspendedParticle(ClientLevel p_172403_, SpriteSet p_172404_, double p_172405_, double p_172406_, double p_172407_) {
-      super(p_172403_, p_172405_, p_172406_ - 0.125D, p_172407_);
+   SuspendedParticle(ClientLevel p_172403_, SpriteSet p_172404_, me.alphamode.mclong.math.BigDecimal p_172405_, me.alphamode.mclong.math.BigDecimal p_172406_, me.alphamode.mclong.math.BigDecimal p_172407_) {
+      super(p_172403_, p_172405_, p_172406_.subtract(me.alphamode.mclong.core.BigConstants.PARTICLE), p_172407_);
       this.m_107250_(0.01F, 0.01F);
       this.m_108335_(p_172404_);
       this.f_107663_ *= this.f_107223_.m_188501_() * 0.6F + 0.2F;
@@ -22,8 +_,8 @@
       this.f_107226_ = 0.0F;
    }
 
-   SuspendedParticle(ClientLevel p_172409_, SpriteSet p_172410_, double p_172411_, double p_172412_, double p_172413_, double p_172414_, double p_172415_, double p_172416_) {
-      super(p_172409_, p_172411_, p_172412_ - 0.125D, p_172413_, p_172414_, p_172415_, p_172416_);
+   SuspendedParticle(ClientLevel p_172409_, SpriteSet p_172410_, me.alphamode.mclong.math.BigDecimal p_172411_, me.alphamode.mclong.math.BigDecimal p_172412_, me.alphamode.mclong.math.BigDecimal p_172413_, double p_172414_, double p_172415_, double p_172416_) {
+      super(p_172409_, p_172411_, p_172412_.subtract(me.alphamode.mclong.core.BigConstants.PARTICLE), p_172413_, p_172414_, p_172415_, p_172416_);
       this.m_107250_(0.01F, 0.01F);
       this.m_108335_(p_172410_);
       this.f_107663_ *= this.f_107223_.m_188501_() * 0.6F + 0.6F;
@@ -45,7 +_,7 @@
          this.f_108040_ = p_108042_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108053_, ClientLevel p_108054_, double p_108055_, double p_108056_, double p_108057_, double p_108058_, double p_108059_, double p_108060_) {
+      public Particle createParticle(SimpleParticleType p_108053_, ClientLevel p_108054_, me.alphamode.mclong.math.BigDecimal p_108055_, me.alphamode.mclong.math.BigDecimal p_108056_, me.alphamode.mclong.math.BigDecimal p_108057_, double p_108058_, double p_108059_, double p_108060_) {
          RandomSource randomsource = p_108054_.f_46441_;
          double d0 = randomsource.m_188583_() * (double)1.0E-6F;
          double d1 = randomsource.m_188583_() * (double)1.0E-4F;
@@ -64,7 +_,7 @@
          this.f_172417_ = p_172419_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_172430_, ClientLevel p_172431_, double p_172432_, double p_172433_, double p_172434_, double p_172435_, double p_172436_, double p_172437_) {
+      public Particle createParticle(SimpleParticleType p_172430_, ClientLevel p_172431_, me.alphamode.mclong.math.BigDecimal p_172432_, me.alphamode.mclong.math.BigDecimal p_172433_, me.alphamode.mclong.math.BigDecimal p_172434_, double p_172435_, double p_172436_, double p_172437_) {
          SuspendedParticle suspendedparticle = new SuspendedParticle(p_172431_, this.f_172417_, p_172432_, p_172433_, p_172434_, 0.0D, (double)-0.8F, 0.0D) {
             public Optional<ParticleGroup> m_142654_() {
                return Optional.of(ParticleGroup.f_175814_);
@@ -85,7 +_,7 @@
          this.f_108061_ = p_108063_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108074_, ClientLevel p_108075_, double p_108076_, double p_108077_, double p_108078_, double p_108079_, double p_108080_, double p_108081_) {
+      public Particle createParticle(SimpleParticleType p_108074_, ClientLevel p_108075_, me.alphamode.mclong.math.BigDecimal p_108076_, me.alphamode.mclong.math.BigDecimal p_108077_, me.alphamode.mclong.math.BigDecimal p_108078_, double p_108079_, double p_108080_, double p_108081_) {
          SuspendedParticle suspendedparticle = new SuspendedParticle(p_108075_, this.f_108061_, p_108076_, p_108077_, p_108078_);
          suspendedparticle.m_107253_(0.4F, 0.4F, 0.7F);
          return suspendedparticle;
@@ -100,7 +_,7 @@
          this.f_108082_ = p_108084_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108095_, ClientLevel p_108096_, double p_108097_, double p_108098_, double p_108099_, double p_108100_, double p_108101_, double p_108102_) {
+      public Particle createParticle(SimpleParticleType p_108095_, ClientLevel p_108096_, me.alphamode.mclong.math.BigDecimal p_108097_, me.alphamode.mclong.math.BigDecimal p_108098_, me.alphamode.mclong.math.BigDecimal p_108099_, double p_108100_, double p_108101_, double p_108102_) {
          double d0 = (double)p_108096_.f_46441_.m_188501_() * -1.9D * (double)p_108096_.f_46441_.m_188501_() * 0.1D;
          SuspendedParticle suspendedparticle = new SuspendedParticle(p_108096_, this.f_108082_, p_108097_, p_108098_, p_108099_, 0.0D, d0, 0.0D);
          suspendedparticle.m_107253_(0.1F, 0.1F, 0.3F);
