--- a/net/minecraft/client/particle/TextureSheetParticle.java
+++ b/net/minecraft/client/particle/TextureSheetParticle.java
@@ -9,12 +_,12 @@
 public abstract class TextureSheetParticle extends SingleQuadParticle {
    protected TextureAtlasSprite f_108321_;
 
-   protected TextureSheetParticle(ClientLevel p_108323_, double p_108324_, double p_108325_, double p_108326_) {
-      super(p_108323_, p_108324_, p_108325_, p_108326_);
+   protected TextureSheetParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super(pLevel, pX, pY, pZ);
    }
 
-   protected TextureSheetParticle(ClientLevel p_108328_, double p_108329_, double p_108330_, double p_108331_, double p_108332_, double p_108333_, double p_108334_) {
-      super(p_108328_, p_108329_, p_108330_, p_108331_, p_108332_, p_108333_, p_108334_);
+   protected TextureSheetParticle(ClientLevel pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+      super(pLevel, pX, pY, pZ, pXSpeed, pYSpeed, pZSpeed);
    }
 
    protected void m_108337_(TextureAtlasSprite p_108338_) {
