--- a/net/minecraft/client/particle/TotemParticle.java
+++ b/net/minecraft/client/particle/TotemParticle.java
@@ -7,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class TotemParticle extends SimpleAnimatedParticle {
-   TotemParticle(ClientLevel p_108346_, double p_108347_, double p_108348_, double p_108349_, double p_108350_, double p_108351_, double p_108352_, SpriteSet p_108353_) {
+   TotemParticle(ClientLevel p_108346_, me.alphamode.mclong.math.BigDecimal p_108347_, me.alphamode.mclong.math.BigDecimal p_108348_, me.alphamode.mclong.math.BigDecimal p_108349_, double p_108350_, double p_108351_, double p_108352_, SpriteSet p_108353_) {
       super(p_108346_, p_108347_, p_108348_, p_108349_, p_108353_, 1.25F);
       this.f_172258_ = 0.6F;
       this.f_107215_ = p_108350_;
@@ -32,7 +_,7 @@
          this.f_108364_ = p_108366_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108377_, ClientLevel p_108378_, double p_108379_, double p_108380_, double p_108381_, double p_108382_, double p_108383_, double p_108384_) {
+      public Particle createParticle(SimpleParticleType p_108377_, ClientLevel p_108378_, me.alphamode.mclong.math.BigDecimal p_108379_, me.alphamode.mclong.math.BigDecimal p_108380_, me.alphamode.mclong.math.BigDecimal p_108381_, double p_108382_, double p_108383_, double p_108384_) {
          return new TotemParticle(p_108378_, p_108379_, p_108380_, p_108381_, p_108382_, p_108383_, p_108384_, this.f_108364_);
       }
    }
