--- a/net/minecraft/client/particle/TrackingEmitter.java
+++ b/net/minecraft/client/particle/TrackingEmitter.java
@@ -23,7 +_,7 @@
    }
 
    private TrackingEmitter(ClientLevel p_108399_, Entity p_108400_, ParticleOptions p_108401_, int p_108402_, Vec3 p_108403_) {
-      super(p_108399_, p_108400_.m_20185_(), p_108400_.m_20227_(0.5D), p_108400_.m_20189_(), p_108403_.f_82479_, p_108403_.f_82480_, p_108403_.f_82481_);
+      super(p_108399_, p_108400_.m_20185_(), p_108400_.m_20227_(0.5D), p_108400_.m_20189_(), p_108403_.m_7096_(), p_108403_.m_7098_(), p_108403_.m_7094_());
       this.f_108387_ = p_108400_;
       this.f_108385_ = p_108402_;
       this.f_108386_ = p_108401_;
