--- a/net/minecraft/client/particle/VibrationSignalParticle.java
+++ b/net/minecraft/client/particle/VibrationSignalParticle.java
@@ -22,7 +_,7 @@
    private float f_244335_;
    private float f_243886_;
 
-   VibrationSignalParticle(ClientLevel p_234105_, double p_234106_, double p_234107_, double p_234108_, PositionSource p_234109_, int p_234110_) {
+   VibrationSignalParticle(ClientLevel p_234105_, me.alphamode.mclong.math.BigDecimal p_234106_, me.alphamode.mclong.math.BigDecimal p_234107_, me.alphamode.mclong.math.BigDecimal p_234108_, PositionSource p_234109_, int p_234110_) {
       super(p_234105_, p_234106_, p_234107_, p_234108_, 0.0D, 0.0D, 0.0D);
       this.f_107663_ = 0.3F;
       this.f_234103_ = p_234109_;
@@ -30,9 +_,9 @@
       Optional<Vec3> optional = p_234109_.m_142502_(p_234105_);
       if (optional.isPresent()) {
          Vec3 vec3 = optional.get();
-         double d0 = p_234106_ - vec3.m_7096_();
-         double d1 = p_234107_ - vec3.m_7098_();
-         double d2 = p_234108_ - vec3.m_7094_();
+         double d0 = p_234106_.subtract(vec3.getX()).doubleValue();
+         double d1 = p_234107_.subtract(vec3.getY()).doubleValue();
+         double d2 = p_234108_.subtract(vec3.getZ()).doubleValue();
          this.f_244341_ = this.f_243819_ = (float)Mth.m_14136_(d0, d2);
          this.f_243886_ = this.f_244335_ = (float)Mth.m_14136_(d1, Math.sqrt(d0 * d0 + d2 * d2));
       }
@@ -53,9 +_,9 @@
 
    private void m_172478_(VertexConsumer p_172479_, Camera p_172480_, float p_172481_, Consumer<Quaternionf> p_172482_) {
       Vec3 vec3 = p_172480_.m_90583_();
-      float f = (float)(Mth.m_14139_((double)p_172481_, this.f_107209_, this.f_107212_) - vec3.m_7096_());
-      float f1 = (float)(Mth.m_14139_((double)p_172481_, this.f_107210_, this.f_107213_) - vec3.m_7098_());
-      float f2 = (float)(Mth.m_14139_((double)p_172481_, this.f_107211_, this.f_107214_) - vec3.m_7094_());
+      float f = (float)(Mth.lerp((double)p_172481_, this.f_107209_, this.f_107212_).subtract(vec3.getX()).floatValue());
+      float f1 = (float)(Mth.lerp((double)p_172481_, this.f_107210_, this.f_107213_).subtract(vec3.getY())).floatValue();
+      float f2 = (float)(Mth.lerp((double)p_172481_, this.f_107211_, this.f_107214_).subtract(vec3.getZ())).floatValue();
       Vector3f vector3f = (new Vector3f(0.5F, 0.5F, 0.5F)).normalize();
       Quaternionf quaternionf = (new Quaternionf()).setAngleAxis(0.0F, vector3f.x(), vector3f.y(), vector3f.z());
       p_172482_.accept(quaternionf);
@@ -102,12 +_,12 @@
             int i = this.f_107225_ - this.f_107224_;
             double d0 = 1.0D / (double)i;
             Vec3 vec3 = optional.get();
-            this.f_107212_ = Mth.m_14139_(d0, this.f_107212_, vec3.m_7096_());
-            this.f_107213_ = Mth.m_14139_(d0, this.f_107213_, vec3.m_7098_());
-            this.f_107214_ = Mth.m_14139_(d0, this.f_107214_, vec3.m_7094_());
-            double d1 = this.f_107212_ - vec3.m_7096_();
-            double d2 = this.f_107213_ - vec3.m_7098_();
-            double d3 = this.f_107214_ - vec3.m_7094_();
+            this.f_107212_ = Mth.lerp(d0, this.f_107212_, vec3.getX());
+            this.f_107213_ = Mth.lerp(d0, this.f_107213_, vec3.getY());
+            this.f_107214_ = Mth.lerp(d0, this.f_107214_, vec3.getZ());
+            double d1 = this.f_107212_.subtract(vec3.getX()).doubleValue();
+            double d2 = this.f_107213_.subtract(vec3.getY()).doubleValue();
+            double d3 = this.f_107214_.subtract(vec3.getZ()).doubleValue();
             this.f_244341_ = this.f_243819_;
             this.f_243819_ = (float)Mth.m_14136_(d1, d3);
             this.f_243886_ = this.f_244335_;
@@ -124,7 +_,7 @@
          this.f_172488_ = p_172490_;
       }
 
-      public Particle m_6966_(VibrationParticleOption p_172501_, ClientLevel p_172502_, double p_172503_, double p_172504_, double p_172505_, double p_172506_, double p_172507_, double p_172508_) {
+      public Particle createParticle(VibrationParticleOption p_172501_, ClientLevel p_172502_, me.alphamode.mclong.math.BigDecimal p_172503_, me.alphamode.mclong.math.BigDecimal p_172504_, me.alphamode.mclong.math.BigDecimal p_172505_, double p_172506_, double p_172507_, double p_172508_) {
          VibrationSignalParticle vibrationsignalparticle = new VibrationSignalParticle(p_172502_, p_172503_, p_172504_, p_172505_, p_172501_.m_235983_(), p_172501_.m_235984_());
          vibrationsignalparticle.m_108335_(this.f_172488_);
          vibrationsignalparticle.m_107271_(1.0F);
