--- a/net/minecraft/client/particle/WakeParticle.java
+++ b/net/minecraft/client/particle/WakeParticle.java
@@ -9,7 +_,7 @@
 public class WakeParticle extends TextureSheetParticle {
    private final SpriteSet f_108405_;
 
-   WakeParticle(ClientLevel p_108407_, double p_108408_, double p_108409_, double p_108410_, double p_108411_, double p_108412_, double p_108413_, SpriteSet p_108414_) {
+   WakeParticle(ClientLevel p_108407_, me.alphamode.mclong.math.BigDecimal p_108408_, me.alphamode.mclong.math.BigDecimal p_108409_, me.alphamode.mclong.math.BigDecimal p_108410_, double p_108411_, double p_108412_, double p_108413_, SpriteSet p_108414_) {
       super(p_108407_, p_108408_, p_108409_, p_108410_, 0.0D, 0.0D, 0.0D);
       this.f_108405_ = p_108414_;
       this.f_107215_ *= (double)0.3F;
@@ -55,7 +_,7 @@
          this.f_108427_ = p_108429_;
       }
 
-      public Particle m_6966_(SimpleParticleType p_108440_, ClientLevel p_108441_, double p_108442_, double p_108443_, double p_108444_, double p_108445_, double p_108446_, double p_108447_) {
+      public Particle createParticle(SimpleParticleType p_108440_, ClientLevel p_108441_, me.alphamode.mclong.math.BigDecimal p_108442_, me.alphamode.mclong.math.BigDecimal p_108443_, me.alphamode.mclong.math.BigDecimal p_108444_, double p_108445_, double p_108446_, double p_108447_) {
          return new WakeParticle(p_108441_, p_108442_, p_108443_, p_108444_, p_108445_, p_108446_, p_108447_, this.f_108427_);
       }
    }
