--- a/net/minecraft/client/particle/WaterDropParticle.java
+++ b/net/minecraft/client/particle/WaterDropParticle.java
@@ -45,8 +_,8 @@
          }
 
          BlockPos blockpos = new BlockPos(this.f_107212_, this.f_107213_, this.f_107214_);
-         double d0 = Math.max(this.f_107208_.m_8055_(blockpos).m_60812_(this.f_107208_, blockpos).m_83290_(Direction.Axis.Y, this.f_107212_ - (double)blockpos.m_123341_(), this.f_107214_ - (double)blockpos.m_123343_()), (double)this.f_107208_.m_6425_(blockpos).m_76155_(this.f_107208_, blockpos));
-         if (d0 > 0.0D && this.f_107213_ < (double)blockpos.m_123342_() + d0) {
+         double d0 = Math.max(this.f_107208_.m_8055_(blockpos).m_60812_(this.f_107208_, blockpos).m_83290_(Direction.Axis.Y, this.f_107212_ - (double)blockpos.getX(), this.f_107214_ - (double)blockpos.getZ()), (double)this.f_107208_.m_6425_(blockpos).m_76155_(this.f_107208_, blockpos));
+         if (d0 > 0.0D && this.f_107213_ < (double)blockpos.getY() + d0) {
             this.m_107274_();
          }
 
