--- a/net/minecraft/world/entity/decoration/ArmorStand.java
+++ b/net/minecraft/world/entity/decoration/ArmorStand.java
@@ -91,17 +_,17 @@
       this.m_274367_(0.0F);
    }
 
-   public ArmorStand(Level p_31556_, double p_31557_, double p_31558_, double p_31559_) {
-      this(EntityType.f_20529_, p_31556_);
-      this.m_6034_(p_31557_, p_31558_, p_31559_);
+   public ArmorStand(Level pLevel, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      this(EntityType.f_20529_, pLevel);
+      this.setPos(pX, pY, pZ);
    }
 
    public void m_6210_() {
-      double d0 = this.m_20185_();
-      double d1 = this.m_20186_();
-      double d2 = this.m_20189_();
+      me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+      me.alphamode.mclong.math.BigDecimal d1 = this.getBigY();
+      me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ();
       super.m_6210_();
-      this.m_6034_(d0, d1, d2);
+      this.setPos(d0, d1, d2);
    }
 
    private boolean m_31560_() {
@@ -328,7 +_,7 @@
    private EquipmentSlot m_31659_(Vec3 p_31660_) {
       EquipmentSlot equipmentslot = EquipmentSlot.MAINHAND;
       boolean flag = this.m_31666_();
-      double d0 = flag ? p_31660_.f_82480_ * 2.0D : p_31660_.f_82480_;
+      double d0 = flag ? p_31660_.m_7098_() * 2.0D : p_31660_.m_7098_();
       EquipmentSlot equipmentslot1 = EquipmentSlot.FEET;
       if (d0 >= 0.1D && d0 < 0.1D + (flag ? 0.8D : 0.45D) && this.m_21033_(equipmentslot1)) {
          equipmentslot = EquipmentSlot.FEET;
@@ -440,7 +_,7 @@
    public void m_7822_(byte p_31568_) {
       if (p_31568_ == 32) {
          if (this.m_9236_().f_46443_) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_11683_, this.m_5720_(), 0.3F, 1.0F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_11683_, this.m_5720_(), 0.3F, 1.0F, false);
             this.f_31528_ = this.m_9236_().m_46467_();
          }
       } else {
@@ -461,7 +_,7 @@
 
    private void m_31565_() {
       if (this.m_9236_() instanceof ServerLevel) {
-         ((ServerLevel)this.m_9236_()).m_8767_(new BlockParticleOption(ParticleTypes.f_123794_, Blocks.f_50705_.m_49966_()), this.m_20185_(), this.m_20227_(0.6666666666666666D), this.m_20189_(), 10, (double)(this.m_20205_() / 4.0F), (double)(this.m_20206_() / 4.0F), (double)(this.m_20205_() / 4.0F), 0.05D);
+         ((ServerLevel)this.m_9236_()).sendParticles(new BlockParticleOption(ParticleTypes.f_123794_, Blocks.f_50705_.m_49966_()), this.getBigX(), this.getBigY(0.6666666666666666D), this.getBigZ(), 10, (double)(this.m_20205_() / 4.0F), (double)(this.m_20206_() / 4.0F), (double)(this.m_20205_() / 4.0F), 0.05D);
       }
 
    }
@@ -496,7 +_,7 @@
       for(int i = 0; i < this.f_31538_.size(); ++i) {
          ItemStack itemstack = this.f_31538_.get(i);
          if (!itemstack.m_41619_()) {
-            Block.m_49840_(this.m_9236_(), this.m_20183_().m_7494_(), itemstack);
+            Block.m_49840_(this.m_9236_(), this.m_20183_().above(), itemstack);
             this.f_31538_.set(i, ItemStack.f_41583_);
          }
       }
@@ -504,7 +_,7 @@
       for(int j = 0; j < this.f_31539_.size(); ++j) {
          ItemStack itemstack1 = this.f_31539_.get(j);
          if (!itemstack1.m_41619_()) {
-            Block.m_49840_(this.m_9236_(), this.m_20183_().m_7494_(), itemstack1);
+            Block.m_49840_(this.m_9236_(), this.m_20183_().above(), itemstack1);
             this.f_31539_.set(j, ItemStack.f_41583_);
          }
       }
@@ -512,7 +_,7 @@
    }
 
    private void m_31566_() {
-      this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_11681_, this.m_5720_(), 1.0F, 1.0F);
+      this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_11681_, this.m_5720_(), 1.0F, 1.0F);
    }
 
    protected float m_5632_(float p_31644_, float p_31645_) {
@@ -769,10 +_,10 @@
          BlockPos blockpos = this.m_20183_();
          int i = Integer.MIN_VALUE;
 
-         for(BlockPos blockpos1 : BlockPos.m_121940_(BlockPos.m_274561_(aabb.f_82288_, aabb.f_82289_, aabb.f_82290_), BlockPos.m_274561_(aabb.f_82291_, aabb.f_82292_, aabb.f_82293_))) {
+         for(BlockPos blockpos1 : BlockPos.m_121940_(BlockPos.containing(aabb.f_82288_, aabb.f_82289_, aabb.f_82290_), BlockPos.containing(aabb.f_82291_, aabb.f_82292_, aabb.f_82293_))) {
             int j = Math.max(this.m_9236_().m_45517_(LightLayer.BLOCK, blockpos1), this.m_9236_().m_45517_(LightLayer.SKY, blockpos1));
             if (j == 15) {
-               return Vec3.m_82512_(blockpos1);
+               return Vec3.atCenterOf(blockpos1);
             }
 
             if (j > i) {
@@ -781,7 +_,7 @@
             }
          }
 
-         return Vec3.m_82512_(blockpos);
+         return Vec3.atCenterOf(blockpos);
       } else {
          return super.m_7371_(p_31665_);
       }
