--- a/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -87,9 +_,9 @@
    protected void m_7087_() {
       if (this.f_31699_ != null) {
          double d0 = 0.46875D;
-         double d1 = (double)this.f_31698_.m_123341_() + 0.5D - (double)this.f_31699_.m_122429_() * 0.46875D;
-         double d2 = (double)this.f_31698_.m_123342_() + 0.5D - (double)this.f_31699_.m_122430_() * 0.46875D;
-         double d3 = (double)this.f_31698_.m_123343_() + 0.5D - (double)this.f_31699_.m_122431_() * 0.46875D;
+         double d1 = (double)this.f_31698_.getX() + 0.5D - (double)this.f_31699_.m_122429_() * 0.46875D;
+         double d2 = (double)this.f_31698_.getY() + 0.5D - (double)this.f_31699_.m_122430_() * 0.46875D;
+         double d3 = (double)this.f_31698_.getZ() + 0.5D - (double)this.f_31699_.m_122431_() * 0.46875D;
          this.m_20343_(d1, d2, d3);
          double d4 = (double)this.m_7076_();
          double d5 = (double)this.m_7068_();
@@ -119,7 +_,7 @@
       } else if (!this.f_19853_.m_45786_(this)) {
          return false;
       } else {
-         BlockState blockstate = this.f_19853_.m_8055_(this.f_31698_.m_121945_(this.f_31699_.m_122424_()));
+         BlockState blockstate = this.f_19853_.m_8055_(this.f_31698_.relative(this.f_31699_.m_122424_()));
          return blockstate.m_60767_().m_76333_() || this.f_31699_.m_122434_().m_122479_() && DiodeBlock.m_52586_(blockstate) ? this.f_19853_.m_6249_(this, this.m_20191_(), f_31697_).isEmpty() : false;
       }
    }
