--- a/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java
+++ b/net/minecraft/world/entity/decoration/LeashFenceKnotEntity.java
@@ -32,11 +_,11 @@
 
    public LeashFenceKnotEntity(Level p_31831_, BlockPos p_31832_) {
       super(EntityType.f_20464_, p_31831_, p_31832_);
-      this.m_6034_((double)p_31832_.m_123341_(), (double)p_31832_.m_123342_(), (double)p_31832_.m_123343_());
+      this.m_6034_((double)p_31832_.getX(), (double)p_31832_.getY(), (double)p_31832_.getZ());
    }
 
    protected void m_7087_() {
-      this.m_20343_((double)this.f_31698_.m_123341_() + 0.5D, (double)this.f_31698_.m_123342_() + 0.375D, (double)this.f_31698_.m_123343_() + 0.5D);
+      this.m_20343_((double)this.f_31698_.getX() + 0.5D, (double)this.f_31698_.getY() + 0.375D, (double)this.f_31698_.getZ() + 0.5D);
       double d0 = (double)this.m_6095_().m_20678_() / 2.0D;
       double d1 = (double)this.m_6095_().m_20679_();
       this.m_20011_(new AABB(this.m_20185_() - d0, this.m_20186_(), this.m_20189_() - d0, this.m_20185_() + d0, this.m_20186_() + d1, this.m_20189_() + d0));
@@ -106,9 +_,9 @@
    }
 
    public static LeashFenceKnotEntity m_31844_(Level p_31845_, BlockPos p_31846_) {
-      int i = p_31846_.m_123341_();
-      int j = p_31846_.m_123342_();
-      int k = p_31846_.m_123343_();
+      long i = p_31846_.getX();
+      long j = p_31846_.getY();
+      long k = p_31846_.getZ();
 
       for(LeashFenceKnotEntity leashfenceknotentity : p_31845_.m_45976_(LeashFenceKnotEntity.class, new AABB((double)i - 1.0D, (double)j - 1.0D, (double)k - 1.0D, (double)i + 1.0D, (double)j + 1.0D, (double)k + 1.0D))) {
          if (leashfenceknotentity.m_31748_().equals(p_31846_)) {
