--- a/net/minecraft/world/entity/decoration/Painting.java
+++ b/net/minecraft/world/entity/decoration/Painting.java
@@ -162,12 +_,16 @@
       this.m_6034_(p_31929_, p_31930_, p_31931_);
    }
 
-   public void m_6453_(double p_31917_, double p_31918_, double p_31919_, float p_31920_, float p_31921_, int p_31922_, boolean p_31923_) {
-      this.m_6034_(p_31917_, p_31918_, p_31919_);
+   public void moveTo(me.alphamode.mclong.math.BigDecimal x, me.alphamode.mclong.math.BigDecimal y, me.alphamode.mclong.math.BigDecimal z, float yRot, float xRot) {
+      this.setPos(x, y, z);
+   }
+
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pYaw, float pPitch, int pPosRotationIncrements, boolean pTeleport) {
+      this.setPos(pX, pY, pZ);
    }
 
    public Vec3 m_213870_() {
-      return Vec3.m_82528_(this.f_31698_);
+      return Vec3.atLowerCornerOf(this.f_31698_);
    }
 
    public Packet<ClientGamePacketListener> m_5654_() {
