--- a/net/minecraft/world/item/context/DirectionalPlaceContext.java
+++ b/net/minecraft/world/item/context/DirectionalPlaceContext.java
@@ -13,7 +_,7 @@
    private final Direction f_43648_;
 
    public DirectionalPlaceContext(Level p_43650_, BlockPos p_43651_, Direction p_43652_, ItemStack p_43653_, Direction p_43654_) {
-      super(p_43650_, (Player)null, InteractionHand.MAIN_HAND, p_43653_, new BlockHitResult(Vec3.m_82539_(p_43651_), p_43654_, p_43651_, false));
+      super(p_43650_, (Player)null, InteractionHand.MAIN_HAND, p_43653_, new BlockHitResult(Vec3.atBottomCenterOf(p_43651_), p_43654_, p_43651_, false));
       this.f_43648_ = p_43652_;
    }
 
