--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/AcaciaFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/AcaciaFoliagePlacer.java
@@ -25,7 +_,7 @@
 
    protected void m_213633_(LevelSimulatedReader p_225499_, BiConsumer<BlockPos, BlockState> p_225500_, RandomSource p_225501_, TreeConfiguration p_225502_, int p_225503_, FoliagePlacer.FoliageAttachment p_225504_, int p_225505_, int p_225506_, int p_225507_) {
       boolean flag = p_225504_.m_68590_();
-      BlockPos blockpos = p_225504_.m_161451_().m_6630_(p_225507_);
+      BlockPos blockpos = p_225504_.m_161451_().above(p_225507_);
       this.m_225628_(p_225499_, p_225500_, p_225501_, p_225502_, blockpos, p_225506_ + p_225504_.m_68589_(), -1 - p_225505_, flag);
       this.m_225628_(p_225499_, p_225500_, p_225501_, p_225502_, blockpos, p_225506_ - 1, -p_225505_, flag);
       this.m_225628_(p_225499_, p_225500_, p_225501_, p_225502_, blockpos, p_225506_ + p_225504_.m_68589_() - 1, 0, flag);
