--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/CherryFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/CherryFoliagePlacer.java
@@ -43,7 +_,7 @@
 
    protected void m_213633_(LevelSimulatedReader p_272723_, FoliagePlacer.FoliageSetter p_273410_, RandomSource p_273057_, TreeConfiguration p_273220_, int p_272975_, FoliagePlacer.FoliageAttachment p_273037_, int p_273647_, int p_273700_, int p_273188_) {
       boolean flag = p_273037_.m_68590_();
-      BlockPos blockpos = p_273037_.m_161451_().m_6630_(p_273188_);
+      BlockPos blockpos = p_273037_.m_161451_().above(p_273188_);
       int i = p_273700_ + p_273037_.m_68589_() - 1;
       this.m_225628_(p_272723_, p_273410_, p_273057_, p_273220_, blockpos, i - 2, p_273647_ - 3, flag);
       this.m_225628_(p_272723_, p_273410_, p_273057_, p_273220_, blockpos, i - 1, p_273647_ - 4, flag);
