--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/DarkOakFoliagePlacer.java
@@ -24,7 +_,7 @@
    }
 
    protected void m_213633_(LevelSimulatedReader p_225558_, BiConsumer<BlockPos, BlockState> p_225559_, RandomSource p_225560_, TreeConfiguration p_225561_, int p_225562_, FoliagePlacer.FoliageAttachment p_225563_, int p_225564_, int p_225565_, int p_225566_) {
-      BlockPos blockpos = p_225563_.m_161451_().m_6630_(p_225566_);
+      BlockPos blockpos = p_225563_.m_161451_().above(p_225566_);
       boolean flag = p_225563_.m_68590_();
       if (flag) {
          this.m_225628_(p_225558_, p_225559_, p_225560_, p_225561_, blockpos, p_225565_ + 2, -1, flag);
