--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacer.java
@@ -74,7 +_,7 @@
       for(int j = -p_225634_; j <= p_225634_ + i; ++j) {
          for(int k = -p_225634_; k <= p_225634_ + i; ++k) {
             if (!this.m_214202_(p_225631_, j, p_225635_, k, p_225634_, p_225636_)) {
-               blockpos$mutableblockpos.m_122154_(p_225633_, j, p_225635_, k);
+               blockpos$mutableblockpos.setWithOffset(p_225633_, j, p_225635_, k);
                m_225622_(p_225629_, p_225630_, p_225631_, p_225632_, blockpos$mutableblockpos);
             }
          }
