--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/MegaPineFoliagePlacer.java
@@ -30,8 +_,8 @@
       BlockPos blockpos = p_225683_.m_161451_();
       int i = 0;
 
-      for(int j = blockpos.m_123342_() - p_225684_ + p_225686_; j <= blockpos.m_123342_() + p_225686_; ++j) {
-         int k = blockpos.m_123342_() - j;
+      for(long j = blockpos.getY() - p_225684_ + p_225686_; j <= blockpos.getY() + p_225686_; ++j) {
+         long k = blockpos.getY() - j;
          int l = p_225685_ + p_225683_.m_68589_() + Mth.m_14143_((float)k / (float)p_225684_ * 3.5F);
          int i1;
          if (k > 0 && l == i && (j & 1) == 0) {
@@ -40,7 +_,7 @@
             i1 = l;
          }
 
-         this.m_225628_(p_225678_, p_273345_, p_225680_, p_225681_, new BlockPos(blockpos.m_123341_(), j, blockpos.m_123343_()), i1, 0, p_225683_.m_68590_());
+         this.m_225628_(p_225678_, p_273345_, p_225680_, p_225681_, new BlockPos(blockpos.getBigX(), j, blockpos.getBigZ()), i1, 0, p_225683_.m_68590_());
          i = l;
       }
 
