--- a/net/minecraft/world/level/levelgen/feature/foliageplacers/RandomSpreadFoliagePlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/foliageplacers/RandomSpreadFoliagePlacer.java
@@ -34,7 +_,7 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = blockpos.m_122032_();
 
       for(int i = 0; i < this.f_161503_; ++i) {
-         blockpos$mutableblockpos.m_122154_(blockpos, p_225725_.m_188503_(p_225730_) - p_225725_.m_188503_(p_225730_), p_225725_.m_188503_(p_225729_) - p_225725_.m_188503_(p_225729_), p_225725_.m_188503_(p_225730_) - p_225725_.m_188503_(p_225730_));
+         blockpos$mutableblockpos.setWithOffset(blockpos, p_225725_.m_188503_(p_225730_) - p_225725_.m_188503_(p_225730_), p_225725_.m_188503_(p_225729_) - p_225725_.m_188503_(p_225729_), p_225725_.m_188503_(p_225730_) - p_225725_.m_188503_(p_225730_));
          m_272253_(p_225723_, p_272842_, p_225725_, p_225726_, blockpos$mutableblockpos);
       }
 
