--- a/net/minecraft/world/level/border/WorldBorder.java
+++ b/net/minecraft/world/level/border/WorldBorder.java
@@ -29,11 +_,11 @@
    public static final WorldBorder.Settings f_61907_ = new WorldBorder.Settings(0.0D, 0.0D, 0.2D, 5.0D, 5, 15, 5.9999968E7D, 0L, 0.0D);
 
    public boolean m_61937_(BlockPos p_61938_) {
-      return (double)(p_61938_.m_123341_() + 1) > this.m_61955_() && (double)p_61938_.m_123341_() < this.m_61957_() && (double)(p_61938_.m_123343_() + 1) > this.m_61956_() && (double)p_61938_.m_123343_() < this.m_61958_();
+      return (double)(p_61938_.getX() + 1) > this.m_61955_() && (double)p_61938_.getX() < this.m_61957_() && (double)(p_61938_.getZ() + 1) > this.m_61956_() && (double)p_61938_.getZ() < this.m_61958_();
    }
 
    public boolean m_61927_(ChunkPos p_61928_) {
-      return (double)p_61928_.m_45608_() > this.m_61955_() && (double)p_61928_.m_45604_() < this.m_61957_() && (double)p_61928_.m_45609_() > this.m_61956_() && (double)p_61928_.m_45605_() < this.m_61958_();
+      return (double)p_61928_.getMaxBlockX() > this.m_61955_() && (double)p_61928_.getMinBlockX() < this.m_61957_() && (double)p_61928_.getMaxBlockZ() > this.m_61956_() && (double)p_61928_.getMinBlockZ() < this.m_61958_();
    }
 
    public boolean m_156093_(double p_156094_, double p_156095_) {
