--- a/net/minecraft/world/level/border/WorldBorder.java
+++ b/net/minecraft/world/level/border/WorldBorder.java
@@ -24,28 +_,28 @@
    private int f_61910_ = 5;
    private double f_61911_;
    private double f_61912_;
-   int f_61913_ = 29999984;
-   private WorldBorder.BorderExtent f_61914_ = new WorldBorder.StaticBorderExtent((double)5.999997E7F);
-   public static final WorldBorder.Settings f_61907_ = new WorldBorder.Settings(0.0D, 0.0D, 0.2D, 5.0D, 5, 15, (double)5.999997E7F, 0L, 0.0D);
+   long f_61913_ = 29999984;
+   private WorldBorder.BorderExtent f_61914_ = new WorldBorder.StaticBorderExtent(java.lang.Long.MAX_VALUE);
+   public static final WorldBorder.Settings f_61907_ = new WorldBorder.Settings(0.0D, 0.0D, 0.2D, 5.0D, 5, 15, java.lang.Long.MAX_VALUE, 0L, 0.0D);
 
    public boolean m_61937_(BlockPos p_61938_) {
-      return (double)(p_61938_.m_123341_() + 1) > this.m_61955_() && (double)p_61938_.m_123341_() < this.m_61957_() && (double)(p_61938_.m_123343_() + 1) > this.m_61956_() && (double)p_61938_.m_123343_() < this.m_61958_();
+      return true;//(double)(pPos.getX() + 1) > this.getMinX() && (double)pPos.getX() < this.getMaxX() && (double)(pPos.getZ() + 1) > this.getMinZ() && (double)pPos.getZ() < this.getMaxZ();
    }
 
    public boolean m_61927_(ChunkPos p_61928_) {
-      return (double)p_61928_.m_45608_() > this.m_61955_() && (double)p_61928_.m_45604_() < this.m_61957_() && (double)p_61928_.m_45609_() > this.m_61956_() && (double)p_61928_.m_45605_() < this.m_61958_();
+      return true;//(double)pChunkPos.getMaxBlockX() > this.getMinX() && (double)pChunkPos.getMinBlockX() < this.getMaxX() && (double)pChunkPos.getMaxBlockZ() > this.getMinZ() && (double)pChunkPos.getMinBlockZ() < this.getMaxZ();
    }
 
    public boolean m_156093_(double p_156094_, double p_156095_) {
-      return p_156094_ > this.m_61955_() && p_156094_ < this.m_61957_() && p_156095_ > this.m_61956_() && p_156095_ < this.m_61958_();
+      return true;//pX > this.getMinX() && pX < this.getMaxX() && pZ > this.getMinZ() && pZ < this.getMaxZ();
    }
 
    public boolean m_187562_(double p_187563_, double p_187564_, double p_187565_) {
-      return p_187563_ > this.m_61955_() - p_187565_ && p_187563_ < this.m_61957_() + p_187565_ && p_187564_ > this.m_61956_() - p_187565_ && p_187564_ < this.m_61958_() + p_187565_;
+      return true;//pX > this.getMinX() - pOffset && pX < this.getMaxX() + pOffset && pZ > this.getMinZ() - pOffset && pZ < this.getMaxZ() + pOffset;
    }
 
    public boolean m_61935_(AABB p_61936_) {
-      return p_61936_.f_82291_ > this.m_61955_() && p_61936_.f_82288_ < this.m_61957_() && p_61936_.f_82293_ > this.m_61956_() && p_61936_.f_82290_ < this.m_61958_();
+      return true;//pBox.maxX > this.getMinX() && pBox.minX < this.getMaxX() && pBox.maxZ > this.getMinZ() && pBox.minZ < this.getMaxZ();
    }
 
    public BlockPos m_187569_(double p_187570_, double p_187571_, double p_187572_) {
@@ -53,7 +_,7 @@
    }
 
    public double m_61925_(Entity p_61926_) {
-      return this.m_61941_(p_61926_.m_20185_(), p_61926_.m_20189_());
+      return java.lang.Double.MAX_VALUE;//this.getDistanceToBorder(pEntity.getX(), pEntity.getZ());
    }
 
    public VoxelShape m_61946_() {
@@ -72,7 +_,7 @@
 
    public boolean m_187566_(Entity p_187567_, AABB p_187568_) {
       double d0 = Math.max(Mth.m_14005_(p_187568_.m_82362_(), p_187568_.m_82385_()), 1.0D);
-      return this.m_61925_(p_187567_) < d0 * 2.0D && this.m_187562_(p_187567_.m_20185_(), p_187567_.m_20189_(), d0);
+      return false;//this.getDistanceToBorder(pEntity) < d0 * 2.0D && this.isWithinBounds(pEntity.getX(), pEntity.getZ(), d0);
    }
 
    public BorderStatus m_61954_() {
@@ -156,12 +_,12 @@
       this.f_61905_.remove(p_156097_);
    }
 
-   public void m_61923_(int p_61924_) {
-      this.f_61913_ = p_61924_;
+   public void setAbsoluteMaxSize(long pSize) {
+      this.f_61913_ = pSize;
       this.f_61914_.m_6623_();
    }
 
-   public int m_61963_() {
+   public long getAbsoluteMaxSize() {
       return this.f_61913_;
    }
 
@@ -485,11 +_,11 @@
       }
 
       private void m_62073_() {
-         this.f_62052_ = Mth.m_14008_(WorldBorder.this.m_6347_() - this.f_62051_ / 2.0D, (double)(-WorldBorder.this.f_61913_), (double)WorldBorder.this.f_61913_);
-         this.f_62053_ = Mth.m_14008_(WorldBorder.this.m_6345_() - this.f_62051_ / 2.0D, (double)(-WorldBorder.this.f_61913_), (double)WorldBorder.this.f_61913_);
-         this.f_62054_ = Mth.m_14008_(WorldBorder.this.m_6347_() + this.f_62051_ / 2.0D, (double)(-WorldBorder.this.f_61913_), (double)WorldBorder.this.f_61913_);
-         this.f_62055_ = Mth.m_14008_(WorldBorder.this.m_6345_() + this.f_62051_ / 2.0D, (double)(-WorldBorder.this.f_61913_), (double)WorldBorder.this.f_61913_);
-         this.f_62056_ = Shapes.m_83113_(Shapes.f_83036_, Shapes.m_83048_(Math.floor(this.m_6613_()), Double.NEGATIVE_INFINITY, Math.floor(this.m_6606_()), Math.ceil(this.m_6608_()), Double.POSITIVE_INFINITY, Math.ceil(this.m_6619_())), BooleanOp.f_82685_);
+//         this.minX = Mth.clamp(WorldBorder.this.getCenterX() - this.size / 2.0D, (double)(-WorldBorder.this.absoluteMaxSize), (double)WorldBorder.this.absoluteMaxSize);
+//         this.minZ = Mth.clamp(WorldBorder.this.getCenterZ() - this.size / 2.0D, (double)(-WorldBorder.this.absoluteMaxSize), (double)WorldBorder.this.absoluteMaxSize);
+//         this.maxX = Mth.clamp(WorldBorder.this.getCenterX() + this.size / 2.0D, (double)(-WorldBorder.this.absoluteMaxSize), (double)WorldBorder.this.absoluteMaxSize);
+//         this.maxZ = Mth.clamp(WorldBorder.this.getCenterZ() + this.size / 2.0D, (double)(-WorldBorder.this.absoluteMaxSize), (double)WorldBorder.this.absoluteMaxSize);
+//         this.shape = Shapes.join(Shapes.INFINITY, Shapes.box(Math.floor(this.getMinX()), Double.MIN_VALUE, Math.floor(this.getMinZ()), Math.ceil(this.getMaxX()), Double.MAX_VALUE, Math.ceil(this.getMaxZ())), BooleanOp.ONLY_FIRST);
       }
 
       public void m_6623_() {
