--- a/net/minecraft/world/level/lighting/BlockLightEngine.java
+++ b/net/minecraft/world/level/lighting/BlockLightEngine.java
@@ -22,48 +_,48 @@
       super(p_278252_, p_278255_);
    }
 
-   protected void m_75858_(long p_285169_) {
-      long i = SectionPos.m_123235_(p_285169_);
-      if (this.f_283849_.m_75791_(i)) {
-         BlockState blockstate = this.m_284512_(this.f_75489_.m_122188_(p_285169_));
-         int j = this.m_284436_(p_285169_, blockstate);
-         int k = this.f_283849_.m_75795_(p_285169_);
+   protected void checkNode(BlockPos p_285169_) {
+      SectionPos i = SectionPos.m_123199_(p_285169_);
+      if (this.f_283849_.storingLightForSection(i)) {
+         BlockState blockstate = this.m_284512_(this.f_75489_.set(p_285169_));
+         int j = this.getEmission(p_285169_, blockstate);
+         int k = this.f_283849_.getStoredLevel(p_285169_);
          if (j < k) {
-            this.f_283849_.m_75772_(p_285169_, 0);
-            this.m_284343_(p_285169_, LightEngine.QueueEntry.m_284290_(k));
+            this.f_283849_.setStoredLevel(p_285169_, 0);
+            this.enqueueDecrease(p_285169_, new BlockPos(LightEngine.QueueEntry.m_284290_(k), 0, 0));
          } else {
-            this.m_284343_(p_285169_, f_283854_);
+            this.enqueueDecrease(p_285169_, new BlockPos(f_283854_, 0, 0));
          }
 
          if (j > 0) {
-            this.m_284218_(p_285169_, LightEngine.QueueEntry.m_284185_(j, m_284265_(blockstate)));
+            this.enqueueIncrease(p_285169_, new BlockPos(LightEngine.QueueEntry.m_284185_(j, m_284265_(blockstate)), 0, 0));
          }
 
       }
    }
 
-   protected void m_284316_(long p_285500_, long p_285410_, int p_285492_) {
+   protected void propagateIncrease(BlockPos p_285500_, long p_285410_, int p_285492_) {
       BlockState blockstate = null;
 
       for(Direction direction : f_283814_) {
          if (LightEngine.QueueEntry.m_284416_(p_285410_, direction)) {
-            long i = BlockPos.m_121915_(p_285500_, direction);
-            if (this.f_283849_.m_75791_(SectionPos.m_123235_(i))) {
-               int j = this.f_283849_.m_75795_(i);
+            BlockPos i = p_285500_.relative(direction);
+            if (this.f_283849_.storingLightForSection(SectionPos.m_123199_(i))) {
+               int j = this.f_283849_.getStoredLevel(i);
                int k = p_285492_ - 1;
                if (k > j) {
-                  this.f_75489_.m_122188_(i);
+                  this.f_75489_.set(i);
                   BlockState blockstate1 = this.m_284512_(this.f_75489_);
                   int l = p_285492_ - this.m_284404_(blockstate1, this.f_75489_);
                   if (l > j) {
                      if (blockstate == null) {
-                        blockstate = LightEngine.QueueEntry.m_284390_(p_285410_) ? Blocks.f_50016_.m_49966_() : this.m_284512_(this.f_75489_.m_122188_(p_285500_));
+                        blockstate = LightEngine.QueueEntry.m_284390_(p_285410_) ? Blocks.f_50016_.m_49966_() : this.m_284512_(this.f_75489_.set(p_285500_));
                      }
 
-                     if (!this.m_284187_(p_285500_, blockstate, i, blockstate1, direction)) {
-                        this.f_283849_.m_75772_(i, l);
+                     if (!this.shapeOccludes(p_285500_, blockstate, i, blockstate1, direction)) {
+                        this.f_283849_.setStoredLevel(i, l);
                         if (l > 1) {
-                           this.m_284218_(i, LightEngine.QueueEntry.m_284188_(l, m_284265_(blockstate1), direction.m_122424_()));
+                           this.enqueueIncrease(i, new BlockPos(LightEngine.QueueEntry.m_284188_(l, m_284265_(blockstate1), direction.m_122424_()), 0, 0));
                         }
                      }
                   }
@@ -74,28 +_,28 @@
 
    }
 
-   protected void m_284321_(long p_285435_, long p_285230_) {
+   protected void propagateDecrease(BlockPos p_285435_, long p_285230_) {
       int i = LightEngine.QueueEntry.m_284170_(p_285230_);
 
       for(Direction direction : f_283814_) {
          if (LightEngine.QueueEntry.m_284416_(p_285230_, direction)) {
-            long j = BlockPos.m_121915_(p_285435_, direction);
-            if (this.f_283849_.m_75791_(SectionPos.m_123235_(j))) {
-               int k = this.f_283849_.m_75795_(j);
+            BlockPos j = p_285435_.relative(direction);
+            if (this.f_283849_.storingLightForSection(SectionPos.m_123199_(j))) {
+               int k = this.f_283849_.getStoredLevel(j);
                if (k != 0) {
                   if (k <= i - 1) {
-                     BlockState blockstate = this.m_284512_(this.f_75489_.m_122188_(j));
-                     int l = this.m_284436_(j, blockstate);
-                     this.f_283849_.m_75772_(j, 0);
+                     BlockState blockstate = this.m_284512_(this.f_75489_.set(j));
+                     int l = this.getEmission(j, blockstate);
+                     this.f_283849_.setStoredLevel(j, 0);
                      if (l < k) {
-                        this.m_284343_(j, LightEngine.QueueEntry.m_284546_(k, direction.m_122424_()));
+                        this.enqueueDecrease(j, new BlockPos(LightEngine.QueueEntry.m_284546_(k, direction.m_122424_()), 0, 0));
                      }
 
                      if (l > 0) {
-                        this.m_284218_(j, LightEngine.QueueEntry.m_284185_(l, m_284265_(blockstate)));
+                        this.enqueueIncrease(j, new BlockPos(LightEngine.QueueEntry.m_284185_(l, m_284265_(blockstate)), 0, 0));
                      }
                   } else {
-                     this.m_284218_(j, LightEngine.QueueEntry.m_284128_(k, false, direction.m_122424_()));
+                     this.enqueueIncrease(j, new BlockPos(LightEngine.QueueEntry.m_284128_(k, false, direction.m_122424_()), 0, 0));
                   }
                }
             }
@@ -104,18 +_,18 @@
 
    }
 
-   private int m_284436_(long p_285243_, BlockState p_284973_) {
+   private int getEmission(BlockPos p_285243_, BlockState p_284973_) {
       int i = p_284973_.m_60791_();
-      return i > 0 && this.f_283849_.m_284382_(SectionPos.m_123235_(p_285243_)) ? i : 0;
+      return i > 0 && this.f_283849_.lightOnInSection(SectionPos.m_123199_(p_285243_)) ? i : 0;
    }
 
    public void m_142519_(ChunkPos p_285274_) {
       this.m_9335_(p_285274_, true);
-      LightChunk lightchunk = this.f_283884_.m_6196_(p_285274_.f_45578_, p_285274_.f_45579_);
+      LightChunk lightchunk = this.f_283884_.getChunkForLighting(p_285274_.f_45578_, p_285274_.f_45579_);
       if (lightchunk != null) {
          lightchunk.m_284254_((p_285266_, p_285452_) -> {
             int i = p_285452_.m_60791_();
-            this.m_284218_(p_285266_.m_121878_(), LightEngine.QueueEntry.m_284185_(i, m_284265_(p_285452_)));
+            this.enqueueIncrease(p_285266_, new BlockPos(LightEngine.QueueEntry.m_284185_(i, m_284265_(p_285452_)), 0, 0));
          });
       }
 
