--- a/net/minecraft/world/level/lighting/BlockLightSectionStorage.java
+++ b/net/minecraft/world/level/lighting/BlockLightSectionStorage.java
@@ -15,7 +_,7 @@
    protected int m_6181_(long p_75513_) {
       long i = SectionPos.m_123235_(p_75513_);
       DataLayer datalayer = this.m_75758_(i, false);
-      return datalayer == null ? 0 : datalayer.m_62560_(SectionPos.m_123207_(BlockPos.m_121983_(p_75513_)), SectionPos.m_123207_(BlockPos.m_122008_(p_75513_)), SectionPos.m_123207_(BlockPos.m_122015_(p_75513_)));
+      return datalayer == null ? 0 : datalayer.get(SectionPos.sectionRelative(BlockPos.getX(p_75513_)), SectionPos.sectionRelative(BlockPos.getY(p_75513_)), SectionPos.sectionRelative(BlockPos.getZ(p_75513_)));
    }
 
    protected static final class BlockDataLayerStorageMap extends DataLayerStorageMap<BlockLightSectionStorage.BlockDataLayerStorageMap> {
