--- a/net/minecraft/world/level/lighting/ChunkSkyLightSources.java
+++ b/net/minecraft/world/level/lighting/ChunkSkyLightSources.java
@@ -24,8 +_,8 @@
    private final BlockPos.MutableBlockPos f_283783_ = new BlockPos.MutableBlockPos();
 
    public ChunkSkyLightSources(LevelHeightAccessor p_285502_) {
-      this.f_283758_ = p_285502_.m_141937_() - 1;
-      int i = p_285502_.m_151558_();
+      this.f_283758_ = (int) p_285502_.getMinBuildHeight() - 1;
+      int i = (int) p_285502_.getMaxBuildHeight();
       int j = Mth.m_14163_(i - this.f_283758_ + 1);
       this.f_283905_ = new SimpleBitStorage(j, 256);
    }
@@ -38,7 +_,7 @@
          for(int j = 0; j < 16; ++j) {
             for(int k = 0; k < 16; ++k) {
                int l = Math.max(this.m_284264_(p_285152_, i, k, j), this.f_283758_);
-               this.m_284514_(m_284186_(k, j), l);
+               this.set(index(k, j), l);
             }
          }
 
@@ -46,27 +_,27 @@
    }
 
    private int m_284264_(ChunkAccess p_285214_, int p_285171_, int p_285021_, int p_285226_) {
-      int i = SectionPos.m_123223_(p_285214_.m_151568_(p_285171_) + 1);
-      BlockPos.MutableBlockPos blockpos$mutableblockpos = this.f_283859_.m_122178_(p_285021_, i, p_285226_);
-      BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.f_283783_.m_122159_(blockpos$mutableblockpos, Direction.DOWN);
+      long i = SectionPos.sectionToBlockCoord(p_285214_.getSectionYFromSectionIndex(p_285171_) + 1);
+      BlockPos.MutableBlockPos blockpos$mutableblockpos = this.f_283859_.set(p_285021_, i, p_285226_);
+      BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.f_283783_.setWithOffset(blockpos$mutableblockpos, Direction.DOWN);
       BlockState blockstate = Blocks.f_50016_.m_49966_();
 
       for(int j = p_285171_; j >= 0; --j) {
          LevelChunkSection levelchunksection = p_285214_.m_183278_(j);
          if (levelchunksection.m_188008_()) {
             blockstate = Blocks.f_50016_.m_49966_();
-            int l = p_285214_.m_151568_(j);
-            blockpos$mutableblockpos.m_142448_(SectionPos.m_123223_(l));
-            blockpos$mutableblockpos1.m_142448_(blockpos$mutableblockpos.m_123342_() - 1);
+            long l = p_285214_.getSectionYFromSectionIndex(j);
+            blockpos$mutableblockpos.setY(SectionPos.sectionToBlockCoord(l));
+            blockpos$mutableblockpos1.setY(blockpos$mutableblockpos.getY() - 1);
          } else {
             for(int k = 15; k >= 0; --k) {
-               BlockState blockstate1 = levelchunksection.m_62982_(p_285021_, k, p_285226_);
+               BlockState blockstate1 = levelchunksection.getBlockState(p_285021_, k, p_285226_);
                if (m_284529_(p_285214_, blockpos$mutableblockpos, blockstate, blockpos$mutableblockpos1, blockstate1)) {
-                  return blockpos$mutableblockpos.m_123342_();
+                  return (int) blockpos$mutableblockpos.getY();
                }
 
                blockstate = blockstate1;
-               blockpos$mutableblockpos.m_122190_(blockpos$mutableblockpos1);
+               blockpos$mutableblockpos.set(blockpos$mutableblockpos1);
                blockpos$mutableblockpos1.m_122173_(Direction.DOWN);
             }
          }
@@ -75,36 +_,36 @@
       return this.f_283758_;
    }
 
-   public boolean m_284521_(BlockGetter p_285514_, int p_284999_, int p_285358_, int p_284944_) {
-      int i = p_285358_ + 1;
-      int j = m_284186_(p_284999_, p_284944_);
-      int k = this.m_284474_(j);
-      if (i < k) {
+   public boolean update(BlockGetter p_285514_, me.alphamode.mclong.math.BigInteger p_284999_, me.alphamode.mclong.math.BigInteger p_285358_, me.alphamode.mclong.math.BigInteger p_284944_) {
+      me.alphamode.mclong.math.BigInteger i = p_285358_.add();
+      long j = index(p_284999_.longValue(), p_284944_.longValue());
+      int k = this.get(j);
+      if (i.compareTo(k) < 0) {
          return false;
       } else {
-         BlockPos blockpos = this.f_283859_.m_122178_(p_284999_, p_285358_ + 1, p_284944_);
+         BlockPos blockpos = this.f_283859_.set(p_284999_, p_285358_.add(), p_284944_);
          BlockState blockstate = p_285514_.m_8055_(blockpos);
-         BlockPos blockpos1 = this.f_283783_.m_122178_(p_284999_, p_285358_, p_284944_);
+         BlockPos blockpos1 = this.f_283783_.set(p_284999_, p_285358_, p_284944_);
          BlockState blockstate1 = p_285514_.m_8055_(blockpos1);
-         if (this.m_284301_(p_285514_, j, k, blockpos, blockstate, blockpos1, blockstate1)) {
+         if (this.updateEdge(p_285514_, j, k, blockpos, blockstate, blockpos1, blockstate1)) {
             return true;
          } else {
-            BlockPos blockpos2 = this.f_283859_.m_122178_(p_284999_, p_285358_ - 1, p_284944_);
+            BlockPos blockpos2 = this.f_283859_.set(p_284999_, p_285358_.subtract(me.alphamode.mclong.math.BigInteger.ONE), p_284944_);
             BlockState blockstate2 = p_285514_.m_8055_(blockpos2);
-            return this.m_284301_(p_285514_, j, k, blockpos1, blockstate1, blockpos2, blockstate2);
+            return this.updateEdge(p_285514_, j, k, blockpos1, blockstate1, blockpos2, blockstate2);
          }
       }
    }
 
-   private boolean m_284301_(BlockGetter p_285066_, int p_285184_, int p_285101_, BlockPos p_285446_, BlockState p_285185_, BlockPos p_285103_, BlockState p_285009_) {
-      int i = p_285446_.m_123342_();
+   private boolean updateEdge(BlockGetter p_285066_, long p_285184_, int p_285101_, BlockPos p_285446_, BlockState p_285185_, BlockPos p_285103_, BlockState p_285009_) {
+      int i = (int) p_285446_.getY();
       if (m_284529_(p_285066_, p_285446_, p_285185_, p_285103_, p_285009_)) {
          if (i > p_285101_) {
-            this.m_284514_(p_285184_, i);
+            this.set(p_285184_, i);
             return true;
          }
       } else if (i == p_285101_) {
-         this.m_284514_(p_285184_, this.m_284300_(p_285066_, p_285103_, p_285009_));
+         this.set(p_285184_, this.m_284300_(p_285066_, p_285103_, p_285009_));
          return true;
       }
 
@@ -112,18 +_,18 @@
    }
 
    private int m_284300_(BlockGetter p_285279_, BlockPos p_285119_, BlockState p_285096_) {
-      BlockPos.MutableBlockPos blockpos$mutableblockpos = this.f_283859_.m_122190_(p_285119_);
-      BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.f_283783_.m_122159_(p_285119_, Direction.DOWN);
+      BlockPos.MutableBlockPos blockpos$mutableblockpos = this.f_283859_.set(p_285119_);
+      BlockPos.MutableBlockPos blockpos$mutableblockpos1 = this.f_283783_.setWithOffset(p_285119_, Direction.DOWN);
       BlockState blockstate = p_285096_;
 
-      while(blockpos$mutableblockpos1.m_123342_() >= this.f_283758_) {
+      while(blockpos$mutableblockpos1.getY() >= this.f_283758_) {
          BlockState blockstate1 = p_285279_.m_8055_(blockpos$mutableblockpos1);
          if (m_284529_(p_285279_, blockpos$mutableblockpos, blockstate, blockpos$mutableblockpos1, blockstate1)) {
-            return blockpos$mutableblockpos.m_123342_();
+            return (int) blockpos$mutableblockpos.getY();
          }
 
          blockstate = blockstate1;
-         blockpos$mutableblockpos.m_122190_(blockpos$mutableblockpos1);
+         blockpos$mutableblockpos.set(blockpos$mutableblockpos1);
          blockpos$mutableblockpos1.m_122173_(Direction.DOWN);
       }
 
@@ -140,8 +_,8 @@
       }
    }
 
-   public int m_284402_(int p_285247_, int p_285082_) {
-      int i = this.m_284474_(m_284186_(p_285247_, p_285082_));
+   public int getLowestSourceY(long p_285247_, long p_285082_) {
+      int i = this.get(index(p_285247_, p_285082_));
       return this.m_284557_(i);
    }
 
@@ -149,7 +_,7 @@
       int i = Integer.MIN_VALUE;
 
       for(int j = 0; j < this.f_283905_.m_13521_(); ++j) {
-         int k = this.f_283905_.m_13514_(j);
+         int k = this.f_283905_.get(j);
          if (k > i) {
             i = k;
          }
@@ -162,24 +_,24 @@
       int i = p_285311_ - this.f_283758_;
 
       for(int j = 0; j < this.f_283905_.m_13521_(); ++j) {
-         this.f_283905_.m_13524_(j, i);
+         this.f_283905_.set(j, i);
       }
 
    }
 
-   private void m_284514_(int p_285323_, int p_285220_) {
-      this.f_283905_.m_13524_(p_285323_, p_285220_ - this.f_283758_);
+   private void set(long p_285323_, int p_285220_) {
+      this.f_283905_.set(p_285323_, p_285220_ - this.f_283758_);
    }
 
-   private int m_284474_(int p_284951_) {
-      return this.f_283905_.m_13514_(p_284951_) + this.f_283758_;
+   private int get(long p_284951_) {
+      return this.f_283905_.get(p_284951_) + this.f_283758_;
    }
 
    private int m_284557_(int p_284953_) {
       return p_284953_ == this.f_283758_ ? Integer.MIN_VALUE : p_284953_;
    }
 
-   private static int m_284186_(int p_284980_, int p_285277_) {
+   private static long index(long p_284980_, long p_285277_) {
       return p_284980_ + p_285277_ * 16;
    }
 }
