--- a/net/minecraft/world/level/lighting/LayerLightSectionStorage.java
+++ b/net/minecraft/world/level/lighting/LayerLightSectionStorage.java
@@ -19,15 +_,15 @@
 public abstract class LayerLightSectionStorage<M extends DataLayerStorageMap<M>> {
    private final LightLayer f_75738_;
    protected final LightChunkGetter f_75739_;
-   protected final Long2ByteMap f_283872_ = new Long2ByteOpenHashMap();
-   private final LongSet f_283775_ = new LongOpenHashSet();
+   protected final it.unimi.dsi.fastutil.objects.Object2ByteMap<SectionPos> f_283872_ = new it.unimi.dsi.fastutil.objects.Object2ByteOpenHashMap<>();
+   protected final it.unimi.dsi.fastutil.objects.ObjectSet<SectionPos> f_283775_ = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
    protected volatile M f_75731_;
    protected final M f_75732_;
-   protected final LongSet f_75733_ = new LongOpenHashSet();
-   protected final LongSet f_75734_ = new LongOpenHashSet();
-   protected final Long2ObjectMap<DataLayer> f_75735_ = Long2ObjectMaps.synchronize(new Long2ObjectOpenHashMap<>());
-   private final LongSet f_75741_ = new LongOpenHashSet();
-   private final LongSet f_75742_ = new LongOpenHashSet();
+   protected final it.unimi.dsi.fastutil.objects.ObjectSet<SectionPos> f_75733_ = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
+   protected final it.unimi.dsi.fastutil.objects.ObjectSet<SectionPos> f_75734_ = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
+   protected final it.unimi.dsi.fastutil.objects.Object2ObjectMap<SectionPos, DataLayer> f_75735_ = it.unimi.dsi.fastutil.objects.Object2ObjectMaps.synchronize(new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>());
+   private final it.unimi.dsi.fastutil.objects.ObjectSet<SectionPos> f_75741_ = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
+   private final it.unimi.dsi.fastutil.objects.ObjectSet<SectionPos> f_75742_ = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
    protected volatile boolean f_283847_;
 
    protected LayerLightSectionStorage(LightLayer p_75745_, LightChunkGetter p_75746_, M p_75747_) {
@@ -39,29 +_,29 @@
       this.f_283872_.defaultReturnValue((byte)0);
    }
 
-   protected boolean m_75791_(long p_75792_) {
-      return this.m_75758_(p_75792_, true) != null;
-   }
-
-   @Nullable
-   protected DataLayer m_75758_(long p_75759_, boolean p_75760_) {
-      return this.m_75761_((M)(p_75760_ ? this.f_75732_ : this.f_75731_), p_75759_);
-   }
-
-   @Nullable
-   protected DataLayer m_75761_(M p_75762_, long p_75763_) {
-      return p_75762_.m_75532_(p_75763_);
-   }
-
-   @Nullable
-   protected DataLayer m_284157_(long p_285278_) {
-      DataLayer datalayer = this.f_75732_.m_75532_(p_285278_);
+   protected boolean storingLightForSection(SectionPos p_75792_) {
+      return this.getDataLayer(p_75792_, true) != null;
+   }
+
+   @Nullable
+   protected DataLayer getDataLayer(SectionPos p_75759_, boolean p_75760_) {
+      return this.getDataLayer((M)(p_75760_ ? this.f_75732_ : this.f_75731_), p_75759_);
+   }
+
+   @Nullable
+   protected DataLayer getDataLayer(M p_75762_, SectionPos p_75763_) {
+      return p_75762_.getLayer(p_75763_);
+   }
+
+   @Nullable
+   protected DataLayer getDataLayerToWrite(SectionPos p_285278_) {
+      DataLayer datalayer = this.f_75732_.getLayer(p_285278_);
       if (datalayer == null) {
          return null;
       } else {
          if (this.f_75733_.add(p_285278_)) {
             datalayer = datalayer.m_62569_();
-            this.f_75732_.m_75526_(p_285278_, datalayer);
+            this.f_75732_.setLayer(p_285278_, datalayer);
             this.f_75732_.m_75531_();
          }
 
@@ -70,48 +_,48 @@
    }
 
    @Nullable
-   public DataLayer m_75793_(long p_75794_) {
+   public DataLayer getDataLayerData(SectionPos p_75794_) {
       DataLayer datalayer = this.f_75735_.get(p_75794_);
-      return datalayer != null ? datalayer : this.m_75758_(p_75794_, false);
-   }
-
-   protected abstract int m_6181_(long p_75786_);
-
-   protected int m_75795_(long p_75796_) {
-      long i = SectionPos.m_123235_(p_75796_);
-      DataLayer datalayer = this.m_75758_(i, true);
-      return datalayer.m_62560_(SectionPos.m_123207_(BlockPos.m_121983_(p_75796_)), SectionPos.m_123207_(BlockPos.m_122008_(p_75796_)), SectionPos.m_123207_(BlockPos.m_122015_(p_75796_)));
-   }
-
-   protected void m_75772_(long p_75773_, int p_75774_) {
-      long i = SectionPos.m_123235_(p_75773_);
+      return datalayer != null ? datalayer : this.getDataLayer(p_75794_, false);
+   }
+
+   protected abstract int getLightValue(BlockPos p_75786_);
+
+   protected int getStoredLevel(BlockPos p_75796_) {
+      SectionPos i = SectionPos.m_123199_(p_75796_);
+      DataLayer datalayer = this.getDataLayer(i, true);
+      return datalayer.get(SectionPos.sectionRelative(p_75796_.getX()), SectionPos.sectionRelative(p_75796_.getY()), SectionPos.sectionRelative(p_75796_.getZ()));
+   }
+
+   protected void setStoredLevel(BlockPos p_75773_, int p_75774_) {
+      SectionPos i = SectionPos.m_123199_(p_75773_);
       DataLayer datalayer;
       if (this.f_75733_.add(i)) {
-         datalayer = this.f_75732_.m_75524_(i);
+         datalayer = this.f_75732_.copyDataLayer(i);
       } else {
-         datalayer = this.m_75758_(i, true);
+         datalayer = this.getDataLayer(i, true);
       }
 
-      datalayer.m_62564_(SectionPos.m_123207_(BlockPos.m_121983_(p_75773_)), SectionPos.m_123207_(BlockPos.m_122008_(p_75773_)), SectionPos.m_123207_(BlockPos.m_122015_(p_75773_)), p_75774_);
-      SectionPos.m_194639_(p_75773_, this.f_75734_::add);
+      datalayer.set(SectionPos.sectionRelative(p_75773_.getX()), SectionPos.sectionRelative(p_75773_.getY()), SectionPos.sectionRelative(p_75773_.getZ()), p_75774_);
+      SectionPos.aroundAndAtBlockPos(p_75773_, this.f_75734_::add);
    }
 
-   protected void m_280483_(long p_281610_) {
-      int i = SectionPos.m_123213_(p_281610_);
-      int j = SectionPos.m_123225_(p_281610_);
-      int k = SectionPos.m_123230_(p_281610_);
+   protected void markSectionAndNeighborsAsAffected(SectionPos p_281610_) {
+      long i = p_281610_.x();
+      long j = p_281610_.y();
+      long k = p_281610_.z();
 
       for(int l = -1; l <= 1; ++l) {
          for(int i1 = -1; i1 <= 1; ++i1) {
             for(int j1 = -1; j1 <= 1; ++j1) {
-               this.f_75734_.add(SectionPos.m_123209_(i + i1, j + j1, k + l));
+               this.f_75734_.add(SectionPos.of(i + i1, j + j1, k + l));
             }
          }
       }
 
    }
 
-   protected DataLayer m_7667_(long p_75797_) {
+   protected DataLayer createDataLayer(SectionPos p_75797_) {
       DataLayer datalayer = this.f_75735_.get(p_75797_);
       return datalayer != null ? datalayer : new DataLayer();
    }
@@ -124,10 +_,10 @@
       if (this.f_283847_) {
          this.f_283847_ = false;
 
-         for(long i : this.f_75742_) {
+         for(net.minecraft.core.SectionPos i : this.f_75742_) {
             DataLayer datalayer = this.f_75735_.remove(i);
-            DataLayer datalayer1 = this.f_75732_.m_75535_(i);
-            if (this.f_75741_.contains(SectionPos.m_123240_(i))) {
+            DataLayer datalayer1 = this.f_75732_.removeLayer(i);
+            if (this.f_75741_.contains(i)) {
                if (datalayer != null) {
                   this.f_75735_.put(i, datalayer);
                } else if (datalayer1 != null) {
@@ -138,21 +_,20 @@
 
          this.f_75732_.m_75531_();
 
-         for(long k : this.f_75742_) {
-            this.m_6187_(k);
+         for(net.minecraft.core.SectionPos k : this.f_75742_) {
+            this.onNodeRemoved(k);
             this.f_75733_.add(k);
          }
 
          this.f_75742_.clear();
-         ObjectIterator<Long2ObjectMap.Entry<DataLayer>> objectiterator = Long2ObjectMaps.fastIterator(this.f_75735_);
+         ObjectIterator<it.unimi.dsi.fastutil.objects.Object2ObjectMap.Entry<SectionPos, DataLayer>> objectiterator = it.unimi.dsi.fastutil.objects.Object2ObjectMaps.fastIterator(this.f_75735_);
 
-         while(objectiterator.hasNext()) {
-            Long2ObjectMap.Entry<DataLayer> entry = objectiterator.next();
-            long j = entry.getLongKey();
-            if (this.m_75791_(j)) {
+         for(it.unimi.dsi.fastutil.objects.Object2ObjectMap.Entry<net.minecraft.core.SectionPos, DataLayer> entry : this.f_75735_.object2ObjectEntrySet()) {
+            net.minecraft.core.SectionPos j = entry.getKey();
+            if (this.storingLightForSection(j)) {
                DataLayer datalayer2 = entry.getValue();
-               if (this.f_75732_.m_75532_(j) != datalayer2) {
-                  this.f_75732_.m_75526_(j, datalayer2);
+               if (this.f_75732_.getLayer(j) != datalayer2) {
+                  this.f_75732_.setLayer(j, datalayer2);
                   this.f_75733_.add(j);
                }
 
@@ -164,13 +_,13 @@
       }
    }
 
-   protected void m_6177_(long p_75798_) {
-   }
-
-   protected void m_6187_(long p_75799_) {
-   }
-
-   protected void m_284259_(long p_285065_, boolean p_284938_) {
+   protected void onNodeAdded(net.minecraft.core.SectionPos p_75798_) {
+   }
+
+   protected void onNodeRemoved(net.minecraft.core.SectionPos p_75799_) {
+   }
+
+   protected void setLightEnabled(SectionPos p_285065_, boolean p_284938_) {
       if (p_284938_) {
          this.f_283775_.add(p_285065_);
       } else {
@@ -179,12 +_,11 @@
 
    }
 
-   protected boolean m_284382_(long p_285433_) {
-      long i = SectionPos.m_123240_(p_285433_);
-      return this.f_283775_.contains(i);
+   protected boolean lightOnInSection(SectionPos p_285433_) {
+      return this.f_283775_.contains(p_285433_);
    }
 
-   public void m_75782_(long p_75783_, boolean p_75784_) {
+   public void retainData(SectionPos p_75783_, boolean p_75784_) {
       if (p_75784_) {
          this.f_75741_.add(p_75783_);
       } else {
@@ -193,7 +_,7 @@
 
    }
 
-   protected void m_284542_(long p_285403_, @Nullable DataLayer p_285498_) {
+   protected void queueSectionData(net.minecraft.core.SectionPos p_285403_, @Nullable DataLayer p_285498_) {
       if (p_285498_ != null) {
          this.f_75735_.put(p_285403_, p_285498_);
          this.f_283847_ = true;
@@ -203,20 +_,20 @@
 
    }
 
-   protected void m_75787_(long p_75788_, boolean p_75789_) {
-      byte b0 = this.f_283872_.get(p_75788_);
+   protected void updateSectionStatus(SectionPos p_75788_, boolean p_75789_) {
+      byte b0 = this.f_283872_.getByte(p_75788_);
       byte b1 = LayerLightSectionStorage.SectionState.m_284365_(b0, !p_75789_);
       if (b0 != b1) {
-         this.m_284336_(p_75788_, b1);
+         this.putSectionState(p_75788_, b1);
          int i = p_75789_ ? -1 : 1;
 
          for(int j = -1; j <= 1; ++j) {
             for(int k = -1; k <= 1; ++k) {
                for(int l = -1; l <= 1; ++l) {
                   if (j != 0 || k != 0 || l != 0) {
-                     long i1 = SectionPos.m_123186_(p_75788_, j, k, l);
-                     byte b2 = this.f_283872_.get(i1);
-                     this.m_284336_(i1, LayerLightSectionStorage.SectionState.m_284236_(b2, LayerLightSectionStorage.SectionState.m_284522_(b2) + i));
+                     SectionPos i1 = p_75788_.offset(j, k, l);
+                     byte b2 = this.f_283872_.getByte(i1);
+                     this.putSectionState(i1, LayerLightSectionStorage.SectionState.m_284236_(b2, LayerLightSectionStorage.SectionState.m_284522_(b2) + i));
                   }
                }
             }
@@ -225,29 +_,29 @@
       }
    }
 
-   protected void m_284336_(long p_285451_, byte p_285078_) {
+   protected void putSectionState(SectionPos p_285451_, byte p_285078_) {
       if (p_285078_ != 0) {
          if (this.f_283872_.put(p_285451_, p_285078_) == 0) {
-            this.m_284497_(p_285451_);
+            this.initializeSection(p_285451_);
          }
       } else if (this.f_283872_.remove(p_285451_) != 0) {
-         this.m_284475_(p_285451_);
+         this.removeSection(p_285451_);
       }
 
    }
 
-   private void m_284497_(long p_285124_) {
+   private void initializeSection(SectionPos p_285124_) {
       if (!this.f_75742_.remove(p_285124_)) {
-         this.f_75732_.m_75526_(p_285124_, this.m_7667_(p_285124_));
+         this.f_75732_.setLayer(p_285124_, this.createDataLayer(p_285124_));
          this.f_75733_.add(p_285124_);
-         this.m_6177_(p_285124_);
-         this.m_280483_(p_285124_);
+         this.onNodeAdded(p_285124_);
+         this.markSectionAndNeighborsAsAffected(p_285124_);
          this.f_283847_ = true;
       }
 
    }
 
-   private void m_284475_(long p_285477_) {
+   private void removeSection(SectionPos p_285477_) {
       this.f_75742_.add(p_285477_);
       this.f_283847_ = true;
    }
@@ -261,11 +_,11 @@
       }
 
       if (!this.f_75734_.isEmpty()) {
-         LongIterator longiterator = this.f_75734_.iterator();
+         it.unimi.dsi.fastutil.objects.ObjectIterator<SectionPos> longiterator = this.f_75734_.iterator();
 
          while(longiterator.hasNext()) {
-            long i = longiterator.nextLong();
-            this.f_75739_.m_6506_(this.f_75738_, SectionPos.m_123184_(i));
+            SectionPos i = longiterator.next();
+            this.f_75739_.m_6506_(this.f_75738_, i);
          }
 
          this.f_75734_.clear();
