--- a/net/minecraft/world/level/lighting/LevelLightEngine.java
+++ b/net/minecraft/world/level/lighting/LevelLightEngine.java
@@ -99,10 +_,10 @@
    public String m_75816_(LightLayer p_75817_, SectionPos p_75818_) {
       if (p_75817_ == LightLayer.BLOCK) {
          if (this.f_75802_ != null) {
-            return this.f_75802_.m_284216_(p_75818_.m_123252_());
+            return this.f_75802_.getDebugData(p_75818_);
          }
       } else if (this.f_75803_ != null) {
-         return this.f_75803_.m_284216_(p_75818_.m_123252_());
+         return this.f_75803_.getDebugData(p_75818_);
       }
 
       return "n/a";
@@ -111,10 +_,10 @@
    public LayerLightSectionStorage.SectionType m_284493_(LightLayer p_285008_, SectionPos p_285336_) {
       if (p_285008_ == LightLayer.BLOCK) {
          if (this.f_75802_ != null) {
-            return this.f_75802_.m_284437_(p_285336_.m_123252_());
+            return this.f_75802_.getDebugSectionType(p_285336_);
          }
       } else if (this.f_75803_ != null) {
-         return this.f_75803_.m_284437_(p_285336_.m_123252_());
+         return this.f_75803_.getDebugSectionType(p_285336_);
       }
 
       return LayerLightSectionStorage.SectionType.EMPTY;
@@ -123,10 +_,10 @@
    public void m_284126_(LightLayer p_285328_, SectionPos p_284962_, @Nullable DataLayer p_285035_) {
       if (p_285328_ == LightLayer.BLOCK) {
          if (this.f_75802_ != null) {
-            this.f_75802_.m_284203_(p_284962_.m_123252_(), p_285035_);
+            this.f_75802_.queueSectionData(p_284962_, p_285035_);
          }
       } else if (this.f_75803_ != null) {
-         this.f_75803_.m_284203_(p_284962_.m_123252_(), p_285035_);
+         this.f_75803_.queueSectionData(p_284962_, p_285035_);
       }
 
    }
@@ -149,19 +_,18 @@
    }
 
    public boolean m_284439_(SectionPos p_285319_) {
-      long i = p_285319_.m_123252_();
-      return this.f_75802_ == null || this.f_75802_.f_283849_.m_284382_(i) && (this.f_75803_ == null || this.f_75803_.f_283849_.m_284382_(i));
+      return this.f_75802_ == null || this.f_75802_.f_283849_.lightOnInSection(p_285319_) && (this.f_75803_ == null || this.f_75803_.f_283849_.lightOnInSection(p_285319_));
    }
 
-   public int m_164446_() {
+   public long getLightSectionCount() {
       return this.f_164445_.m_151559_() + 2;
    }
 
-   public int m_164447_() {
-      return this.f_164445_.m_151560_() - 1;
+   public long getMinLightSection() {
+      return this.f_164445_.getMinSection() - 1;
    }
 
-   public int m_164448_() {
-      return this.m_164447_() + this.m_164446_();
+   public long getMaxLightSection() {
+      return this.getMinLightSection() + this.getLightSectionCount();
    }
 }
