--- a/net/minecraft/world/level/lighting/SkyLightSectionStorage.java
+++ b/net/minecraft/world/level/lighting/SkyLightSectionStorage.java
@@ -11,84 +_,84 @@
 
 public class SkyLightSectionStorage extends LayerLightSectionStorage<SkyLightSectionStorage.SkyDataLayerStorageMap> {
    protected SkyLightSectionStorage(LightChunkGetter p_75868_) {
-      super(LightLayer.SKY, p_75868_, new SkyLightSectionStorage.SkyDataLayerStorageMap(new Long2ObjectOpenHashMap<>(), new Long2IntOpenHashMap(), Integer.MAX_VALUE));
-   }
-
-   protected int m_6181_(long p_75880_) {
-      return this.m_164457_(p_75880_, false);
-   }
-
-   protected int m_164457_(long p_164458_, boolean p_164459_) {
-      long i = SectionPos.m_123235_(p_164458_);
-      int j = SectionPos.m_123225_(i);
+      super(LightLayer.SKY, p_75868_, new SkyLightSectionStorage.SkyDataLayerStorageMap(new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>(), new it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap<>(), Integer.MAX_VALUE));
+   }
+
+   protected int getLightValue(BlockPos p_75880_) {
+      return this.getLightValue(p_75880_, false);
+   }
+
+   protected int getLightValue(BlockPos p_164458_, boolean p_164459_) {
+      SectionPos i = SectionPos.m_123199_(p_164458_);
+      long j = i.y();
       SkyLightSectionStorage.SkyDataLayerStorageMap skylightsectionstorage$skydatalayerstoragemap = p_164459_ ? this.f_75732_ : this.f_75731_;
-      int k = skylightsectionstorage$skydatalayerstoragemap.f_75901_.get(SectionPos.m_123240_(i));
+      long k = skylightsectionstorage$skydatalayerstoragemap.f_75901_.getLong(SectionPos.getZeroNode(i.getBigX(), i.getBigZ()));
       if (k != skylightsectionstorage$skydatalayerstoragemap.f_75900_ && j < k) {
-         DataLayer datalayer = this.m_75761_(skylightsectionstorage$skydatalayerstoragemap, i);
+         DataLayer datalayer = this.getDataLayer(skylightsectionstorage$skydatalayerstoragemap, i);
          if (datalayer == null) {
-            for(p_164458_ = BlockPos.m_122027_(p_164458_); datalayer == null; datalayer = this.m_75761_(skylightsectionstorage$skydatalayerstoragemap, i)) {
+            for(p_164458_ = new BlockPos(p_164458_.getBigX(), me.alphamode.mclong.math.BigInteger.ZERO, p_164458_.getBigZ()); datalayer == null; datalayer = this.getDataLayer(skylightsectionstorage$skydatalayerstoragemap, i)) {
                ++j;
                if (j >= k) {
                   return 15;
                }
 
-               i = SectionPos.m_123191_(i, Direction.UP);
+               i = i.relative(Direction.UP);
             }
          }
 
-         return datalayer.m_62560_(SectionPos.m_123207_(BlockPos.m_121983_(p_164458_)), SectionPos.m_123207_(BlockPos.m_122008_(p_164458_)), SectionPos.m_123207_(BlockPos.m_122015_(p_164458_)));
+         return datalayer.get(SectionPos.sectionRelative(p_164458_.getX()), SectionPos.sectionRelative(p_164458_.getY()), SectionPos.sectionRelative(p_164458_.getZ()));
       } else {
-         return p_164459_ && !this.m_284382_(i) ? 0 : 15;
+         return p_164459_ && !this.lightOnInSection(i) ? 0 : 15;
       }
    }
 
-   protected void m_6177_(long p_75885_) {
-      int i = SectionPos.m_123225_(p_75885_);
+   protected void onNodeAdded(SectionPos p_75885_) {
+      long i = p_75885_.y();
       if ((this.f_75732_).f_75900_ > i) {
          (this.f_75732_).f_75900_ = i;
          (this.f_75732_).f_75901_.defaultReturnValue((this.f_75732_).f_75900_);
       }
 
-      long j = SectionPos.m_123240_(p_75885_);
-      int k = (this.f_75732_).f_75901_.get(j);
+      SectionPos j = SectionPos.getZeroNode(p_75885_.getBigX(), p_75885_.getBigZ());
+      long k = (this.f_75732_).f_75901_.getLong(j);
       if (k < i + 1) {
          (this.f_75732_).f_75901_.put(j, i + 1);
       }
 
    }
 
-   protected void m_6187_(long p_75887_) {
-      long i = SectionPos.m_123240_(p_75887_);
-      int j = SectionPos.m_123225_(p_75887_);
-      if ((this.f_75732_).f_75901_.get(i) == j + 1) {
-         long k;
-         for(k = p_75887_; !this.m_75791_(k) && this.m_75870_(j); k = SectionPos.m_123191_(k, Direction.DOWN)) {
+   protected void onNodeRemoved(SectionPos p_75887_) {
+      SectionPos i = SectionPos.getZeroNode(p_75887_.getBigX(), p_75887_.getBigZ());
+      long j = p_75887_.y();
+      if ((this.f_75732_).f_75901_.getLong(i) == j + 1) {
+         SectionPos k;
+         for(k = p_75887_; !this.storingLightForSection(k) && this.hasLightDataAtOrBelow(j); k = k.relative(Direction.DOWN)) {
             --j;
          }
 
-         if (this.m_75791_(k)) {
+         if (this.storingLightForSection(k)) {
             (this.f_75732_).f_75901_.put(i, j + 1);
          } else {
-            (this.f_75732_).f_75901_.remove(i);
+            (this.f_75732_).f_75901_.removeLong(i);
          }
       }
 
    }
 
-   protected DataLayer m_7667_(long p_75883_) {
+   protected DataLayer createDataLayer(SectionPos p_75883_) {
       DataLayer datalayer = this.f_75735_.get(p_75883_);
       if (datalayer != null) {
          return datalayer;
       } else {
-         int i = (this.f_75732_).f_75901_.get(SectionPos.m_123240_(p_75883_));
-         if (i != (this.f_75732_).f_75900_ && SectionPos.m_123225_(p_75883_) < i) {
+         long i = (this.f_75732_).f_75901_.getLong(SectionPos.getZeroNode(p_75883_.getBigX(), p_75883_.getBigZ()));
+         if (i != (this.f_75732_).f_75900_ && p_75883_.y() < i) {
             DataLayer datalayer1;
-            for(long j = SectionPos.m_123191_(p_75883_, Direction.UP); (datalayer1 = this.m_75758_(j, true)) == null; j = SectionPos.m_123191_(j, Direction.UP)) {
+            for(SectionPos j = p_75883_.relative(Direction.UP); (datalayer1 = this.getDataLayer(j, true)) == null; j = j.relative(Direction.UP)) {
             }
 
             return m_182512_(datalayer1);
          } else {
-            return this.m_284382_(p_75883_) ? new DataLayer(15) : new DataLayer();
+            return this.lightOnInSection(p_75883_) ? new DataLayer(15) : new DataLayer();
          }
       }
    }
@@ -108,29 +_,29 @@
       }
    }
 
-   protected boolean m_75870_(int p_278270_) {
+   protected boolean hasLightDataAtOrBelow(long p_278270_) {
       return p_278270_ >= (this.f_75732_).f_75900_;
    }
 
-   protected boolean m_75890_(long p_75891_) {
-      long i = SectionPos.m_123240_(p_75891_);
-      int j = (this.f_75732_).f_75901_.get(i);
-      return j == (this.f_75732_).f_75900_ || SectionPos.m_123225_(p_75891_) >= j;
-   }
-
-   protected int m_284384_(long p_285094_) {
-      return (this.f_75732_).f_75901_.get(p_285094_);
-   }
-
-   protected int m_284293_() {
+   protected boolean isAboveData(SectionPos p_75891_) {
+      SectionPos i = SectionPos.getZeroNode(p_75891_.getBigX(), p_75891_.getBigZ());
+      long j = (this.f_75732_).f_75901_.getLong(i);
+      return j == (this.f_75732_).f_75900_ || p_75891_.y() >= j;
+   }
+
+   protected long getTopSectionY(SectionPos p_285094_) {
+      return (this.f_75732_).f_75901_.getLong(p_285094_);
+   }
+
+   protected long getBottomSectionY() {
       return (this.f_75732_).f_75900_;
    }
 
    protected static final class SkyDataLayerStorageMap extends DataLayerStorageMap<SkyLightSectionStorage.SkyDataLayerStorageMap> {
-      int f_75900_;
-      final Long2IntOpenHashMap f_75901_;
+      long f_75900_;
+      final it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap<SectionPos> f_75901_;
 
-      public SkyDataLayerStorageMap(Long2ObjectOpenHashMap<DataLayer> p_75903_, Long2IntOpenHashMap p_75904_, int p_75905_) {
+      public SkyDataLayerStorageMap(it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<SectionPos, DataLayer> p_75903_, it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap<SectionPos> p_75904_, long p_75905_) {
          super(p_75903_);
          this.f_75901_ = p_75904_;
          p_75904_.defaultReturnValue(p_75905_);
