--- a/net/minecraft/world/level/lighting/SkyLightEngine.java
+++ b/net/minecraft/world/level/lighting/SkyLightEngine.java
@@ -64,7 +_,7 @@
    protected void m_7900_(long p_75845_, int p_75846_, boolean p_75847_) {
       long i = SectionPos.m_123235_(p_75845_);
       int j = BlockPos.m_122008_(p_75845_);
-      int k = SectionPos.m_123207_(j);
+      long k = SectionPos.sectionRelative(j);
       int l = SectionPos.m_123171_(j);
       int i1;
       if (k != 0) {
