--- a/net/minecraft/world/inventory/GrindstoneMenu.java
+++ b/net/minecraft/world/inventory/GrindstoneMenu.java
@@ -60,7 +_,7 @@
          public void m_142406_(Player p_150574_, ItemStack p_150575_) {
             p_39568_.m_39292_((p_39634_, p_39635_) -> {
                if (p_39634_ instanceof ServerLevel) {
-                  ExperienceOrb.m_147082_((ServerLevel)p_39634_, Vec3.m_82512_(p_39635_), this.m_39631_(p_39634_));
+                  ExperienceOrb.m_147082_((ServerLevel)p_39634_, Vec3.atCenterOf(p_39635_), this.m_39631_(p_39634_));
                }
 
                p_39634_.m_46796_(1042, p_39635_, 0);
