--- a/net/minecraft/world/inventory/MerchantMenu.java
+++ b/net/minecraft/world/inventory/MerchantMenu.java
@@ -146,7 +_,7 @@
    private void m_40077_() {
       if (!this.f_40027_.m_183595_()) {
          Entity entity = (Entity)this.f_40027_;
-         entity.m_9236_().m_7785_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), this.f_40027_.m_7596_(), SoundSource.NEUTRAL, 1.0F, 1.0F, false);
+         entity.m_9236_().playLocalSound(entity.getBigX(), entity.getBigY(), entity.getBigZ(), this.f_40027_.m_7596_(), SoundSource.NEUTRAL, 1.0F, 1.0F, false);
       }
 
    }
