--- a/net/minecraft/data/advancements/packs/VanillaAdventureAdvancements.java
+++ b/net/minecraft/data/advancements/packs/VanillaAdventureAdvancements.java
@@ -128,7 +_,7 @@
       LootItemCondition.Builder[] alootitemcondition$builder = ComparatorBlock.f_54117_.m_6908_().stream().map((p_286187_) -> {
          StatePropertiesPredicate statepropertiespredicate = StatePropertiesPredicate.Builder.m_67693_().m_67697_(ComparatorBlock.f_54117_, p_286187_).m_67706_();
          BlockPredicate blockpredicate = BlockPredicate.Builder.m_17924_().m_146726_(Blocks.f_50328_).m_17929_(statepropertiespredicate).m_17931_();
-         return LocationCheck.m_81727_(LocationPredicate.Builder.m_52651_().m_52652_(blockpredicate), new BlockPos(p_286187_.m_122424_().m_122436_()));
+         return LocationCheck.m_81727_(LocationPredicate.Builder.m_52651_().m_52652_(blockpredicate), new BlockPos(p_286187_.m_122424_().m_122436_().toLong()));
       }).toArray((p_286188_) -> {
          return new LootItemCondition.Builder[p_286188_];
       });
@@ -139,7 +_,7 @@
       LootItemCondition.Builder[] alootitemcondition$builder = ComparatorBlock.f_54117_.m_6908_().stream().map((p_286190_) -> {
          StatePropertiesPredicate.Builder statepropertiespredicate$builder = StatePropertiesPredicate.Builder.m_67693_().m_67697_(ComparatorBlock.f_54117_, p_286190_);
          LootItemBlockStatePropertyCondition.Builder lootitemblockstatepropertycondition$builder = (new LootItemBlockStatePropertyCondition.Builder(Blocks.f_50328_)).m_81784_(statepropertiespredicate$builder);
-         LootItemCondition.Builder lootitemcondition$builder = LocationCheck.m_81727_(LocationPredicate.Builder.m_52651_().m_52652_(BlockPredicate.Builder.m_17924_().m_146726_(p_286250_).m_17931_()), new BlockPos(p_286190_.m_122436_()));
+         LootItemCondition.Builder lootitemcondition$builder = LocationCheck.m_81727_(LocationPredicate.Builder.m_52651_().m_52652_(BlockPredicate.Builder.m_17924_().m_146726_(p_286250_).m_17931_()), new BlockPos(p_286190_.m_122436_().toLong()));
          return AllOfCondition.m_285871_(lootitemblockstatepropertycondition$builder, lootitemcondition$builder);
       }).toArray((p_286191_) -> {
          return new LootItemCondition.Builder[p_286191_];
