--- a/net/minecraft/world/level/levelgen/carver/CanyonCarverConfiguration.java
+++ b/net/minecraft/world/level/levelgen/carver/CanyonCarverConfiguration.java
@@ -15,25 +_,25 @@
          return p_158990_;
       }), FloatProvider.f_146502_.fieldOf("vertical_rotation").forGetter((p_158988_) -> {
          return p_158988_.f_158967_;
-      }), CanyonCarverConfiguration.CanyonShapeConfiguration.f_158991_.fieldOf("shape").forGetter((p_158986_) -> {
+      }), CanyonShapeConfiguration.f_158991_.fieldOf("shape").forGetter((p_158986_) -> {
          return p_158986_.f_158968_;
       })).apply(p_158984_, CanyonCarverConfiguration::new);
    });
    public final FloatProvider f_158967_;
-   public final CanyonCarverConfiguration.CanyonShapeConfiguration f_158968_;
+   public final CanyonShapeConfiguration f_158968_;
 
-   public CanyonCarverConfiguration(float p_224788_, HeightProvider p_224789_, FloatProvider p_224790_, VerticalAnchor p_224791_, CarverDebugSettings p_224792_, HolderSet<Block> p_224793_, FloatProvider p_224794_, CanyonCarverConfiguration.CanyonShapeConfiguration p_224795_) {
+   public CanyonCarverConfiguration(float p_224788_, HeightProvider p_224789_, FloatProvider p_224790_, VerticalAnchor p_224791_, CarverDebugSettings p_224792_, HolderSet<Block> p_224793_, FloatProvider p_224794_, CanyonShapeConfiguration p_224795_) {
       super(p_224788_, p_224789_, p_224790_, p_224791_, p_224792_, p_224793_);
       this.f_158967_ = p_224794_;
       this.f_158968_ = p_224795_;
    }
 
-   public CanyonCarverConfiguration(CarverConfiguration p_158980_, FloatProvider p_158981_, CanyonCarverConfiguration.CanyonShapeConfiguration p_158982_) {
+   public CanyonCarverConfiguration(CarverConfiguration p_158980_, FloatProvider p_158981_, CanyonShapeConfiguration p_158982_) {
       this(p_158980_.f_67859_, p_158980_.f_159088_, p_158980_.f_159089_, p_158980_.f_159090_, p_158980_.f_159092_, p_158980_.f_224830_, p_158981_, p_158982_);
    }
 
    public static class CanyonShapeConfiguration {
-      public static final Codec<CanyonCarverConfiguration.CanyonShapeConfiguration> f_158991_ = RecordCodecBuilder.create((p_159007_) -> {
+      public static final Codec<CanyonShapeConfiguration> f_158991_ = RecordCodecBuilder.create((p_159007_) -> {
          return p_159007_.group(FloatProvider.f_146502_.fieldOf("distance_factor").forGetter((p_159019_) -> {
             return p_159019_.f_158992_;
          }), FloatProvider.f_146502_.fieldOf("thickness").forGetter((p_159017_) -> {
@@ -46,7 +_,7 @@
             return p_159011_.f_158996_;
          }), Codec.FLOAT.fieldOf("vertical_radius_center_factor").forGetter((p_159009_) -> {
             return p_159009_.f_158997_;
-         })).apply(p_159007_, CanyonCarverConfiguration.CanyonShapeConfiguration::new);
+         })).apply(p_159007_, CanyonShapeConfiguration::new);
       });
       public final FloatProvider f_158992_;
       public final FloatProvider f_158993_;
