--- a/net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
+++ b/net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
@@ -23,17 +_,17 @@
    }
 
    public boolean m_213788_(CarvingContext p_224885_, CaveCarverConfiguration p_224886_, ChunkAccess p_224887_, Function<BlockPos, Holder<Biome>> p_224888_, RandomSource p_224889_, Aquifer p_224890_, ChunkPos p_224891_, CarvingMask p_224892_) {
-      int i = SectionPos.m_123223_(this.m_65073_() * 2 - 1);
+      int i = (int) SectionPos.sectionToBlockCoord(this.m_65073_() * 2 - 1);
       int j = p_224889_.m_188503_(p_224889_.m_188503_(p_224889_.m_188503_(this.m_6208_()) + 1) + 1);
 
       for(int k = 0; k < j; ++k) {
-         double d0 = (double)p_224891_.m_151382_(p_224889_.m_188503_(16));
-         double d1 = (double)p_224886_.f_159088_.m_213859_(p_224889_, p_224885_);
-         double d2 = (double)p_224891_.m_151391_(p_224889_.m_188503_(16));
+         double d0 = (double)p_224891_.getBlockX(p_224889_.m_188503_(16));
+         double d1 = (double)p_224886_.f_159088_.sample(p_224889_, p_224885_);
+         double d2 = (double)p_224891_.getBlockZ(p_224889_.m_188503_(16));
          double d3 = (double)p_224886_.f_159155_.m_214084_(p_224889_);
          double d4 = (double)p_224886_.f_159156_.m_214084_(p_224889_);
          double d5 = (double)p_224886_.f_159157_.m_214084_(p_224889_);
-         WorldCarver.CarveSkipChecker worldcarver$carveskipchecker = (p_159202_, p_159203_, p_159204_, p_159205_, p_159206_) -> {
+         CarveSkipChecker worldcarver$carveskipchecker = (p_159202_, p_159203_, p_159204_, p_159205_, p_159206_) -> {
             return m_159195_(p_159203_, p_159204_, p_159205_, d5);
          };
          int l = 1;
@@ -74,13 +_,13 @@
       return 1.0D;
    }
 
-   protected void m_190690_(CarvingContext p_190691_, CaveCarverConfiguration p_190692_, ChunkAccess p_190693_, Function<BlockPos, Holder<Biome>> p_190694_, Aquifer p_190695_, double p_190696_, double p_190697_, double p_190698_, float p_190699_, double p_190700_, CarvingMask p_190701_, WorldCarver.CarveSkipChecker p_190702_) {
+   protected void m_190690_(CarvingContext p_190691_, CaveCarverConfiguration p_190692_, ChunkAccess p_190693_, Function<BlockPos, Holder<Biome>> p_190694_, Aquifer p_190695_, double p_190696_, double p_190697_, double p_190698_, float p_190699_, double p_190700_, CarvingMask p_190701_, CarveSkipChecker p_190702_) {
       double d0 = 1.5D + (double)(Mth.m_14031_(((float)Math.PI / 2F)) * p_190699_);
       double d1 = d0 * p_190700_;
       this.m_190753_(p_190691_, p_190692_, p_190693_, p_190694_, p_190695_, p_190696_ + 1.0D, p_190697_, p_190698_, d0, d1, p_190701_, p_190702_);
    }
 
-   protected void m_190670_(CarvingContext p_190671_, CaveCarverConfiguration p_190672_, ChunkAccess p_190673_, Function<BlockPos, Holder<Biome>> p_190674_, long p_190675_, Aquifer p_190676_, double p_190677_, double p_190678_, double p_190679_, double p_190680_, double p_190681_, float p_190682_, float p_190683_, float p_190684_, int p_190685_, int p_190686_, double p_190687_, CarvingMask p_190688_, WorldCarver.CarveSkipChecker p_190689_) {
+   protected void m_190670_(CarvingContext p_190671_, CaveCarverConfiguration p_190672_, ChunkAccess p_190673_, Function<BlockPos, Holder<Biome>> p_190674_, long p_190675_, Aquifer p_190676_, double p_190677_, double p_190678_, double p_190679_, double p_190680_, double p_190681_, float p_190682_, float p_190683_, float p_190684_, int p_190685_, int p_190686_, double p_190687_, CarvingMask p_190688_, CarveSkipChecker p_190689_) {
       RandomSource randomsource = RandomSource.m_216335_(p_190675_);
       int i = randomsource.m_188503_(p_190686_ / 2) + p_190686_ / 4;
       boolean flag = randomsource.m_188503_(6) == 0;
