--- a/net/minecraft/world/level/levelgen/carver/NetherWorldCarver.java
+++ b/net/minecraft/world/level/levelgen/carver/NetherWorldCarver.java
@@ -35,7 +_,7 @@
    protected boolean m_183633_(CarvingContext p_190731_, CaveCarverConfiguration p_190732_, ChunkAccess p_190733_, Function<BlockPos, Holder<Biome>> p_190734_, CarvingMask p_190735_, BlockPos.MutableBlockPos p_190736_, BlockPos.MutableBlockPos p_190737_, Aquifer p_190738_, MutableBoolean p_190739_) {
       if (this.m_224910_(p_190732_, p_190733_.m_8055_(p_190736_))) {
          BlockState blockstate;
-         if (p_190736_.m_123342_() <= p_190731_.m_142201_() + 31) {
+         if (p_190736_.getY() <= p_190731_.getMinGenY() + 31) {
             blockstate = f_64982_.m_76188_();
          } else {
             blockstate = f_64980_;
