--- a/net/minecraft/world/level/levelgen/feature/treedecorators/CocoaDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/CocoaDecorator.java
@@ -26,9 +_,9 @@
       RandomSource randomsource = p_226028_.m_226067_();
       if (!(randomsource.m_188501_() >= this.f_69973_)) {
          List<BlockPos> list = p_226028_.m_226068_();
-         int i = list.get(0).m_123342_();
+         long i = list.get(0).getY();
          list.stream().filter((p_69980_) -> {
-            return p_69980_.m_123342_() - i <= 2;
+            return p_69980_.getY() - i <= 2;
          }).forEach((p_226026_) -> {
             for(Direction direction : Direction.Plane.HORIZONTAL) {
                if (randomsource.m_188501_() <= 0.25F) {
