--- a/net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecorator.java
@@ -37,9 +_,9 @@
          this.f_226050_ = new ObjectArrayList<>(p_226057_);
          this.f_226048_ = new ObjectArrayList<>(p_226055_);
          this.f_226049_ = new ObjectArrayList<>(p_226056_);
-         this.f_226048_.sort(Comparator.comparingInt(Vec3i::m_123342_));
-         this.f_226049_.sort(Comparator.comparingInt(Vec3i::m_123342_));
-         this.f_226050_.sort(Comparator.comparingInt(Vec3i::m_123342_));
+         this.f_226048_.sort(Comparator.comparingLong(me.alphamode.mclong.core.Vec3l::getY));
+         this.f_226049_.sort(Comparator.comparingLong(me.alphamode.mclong.core.Vec3l::getY));
+         this.f_226050_.sort(Comparator.comparingLong(me.alphamode.mclong.core.Vec3l::getY));
       }
 
       public void m_226064_(BlockPos p_226065_, BooleanProperty p_226066_) {
