--- a/net/minecraft/world/level/levelgen/feature/treedecorators/TrunkVineDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/TrunkVineDecorator.java
@@ -19,28 +_,28 @@
       RandomSource randomsource = p_226077_.m_226067_();
       p_226077_.m_226068_().forEach((p_226075_) -> {
          if (randomsource.m_188503_(3) > 0) {
-            BlockPos blockpos = p_226075_.m_122024_();
+            BlockPos blockpos = p_226075_.west();
             if (p_226077_.m_226059_(blockpos)) {
                p_226077_.m_226064_(blockpos, VineBlock.f_57835_);
             }
          }
 
          if (randomsource.m_188503_(3) > 0) {
-            BlockPos blockpos1 = p_226075_.m_122029_();
+            BlockPos blockpos1 = p_226075_.east();
             if (p_226077_.m_226059_(blockpos1)) {
                p_226077_.m_226064_(blockpos1, VineBlock.f_57837_);
             }
          }
 
          if (randomsource.m_188503_(3) > 0) {
-            BlockPos blockpos2 = p_226075_.m_122012_();
+            BlockPos blockpos2 = p_226075_.north();
             if (p_226077_.m_226059_(blockpos2)) {
                p_226077_.m_226064_(blockpos2, VineBlock.f_57836_);
             }
          }
 
          if (randomsource.m_188503_(3) > 0) {
-            BlockPos blockpos3 = p_226075_.m_122019_();
+            BlockPos blockpos3 = p_226075_.south();
             if (p_226077_.m_226059_(blockpos3)) {
                p_226077_.m_226064_(blockpos3, VineBlock.f_57834_);
             }
