--- a/net/minecraft/client/resources/sounds/AbstractSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/AbstractSoundInstance.java
@@ -16,9 +_,9 @@
    protected final ResourceLocation f_119572_;
    protected float f_119573_ = 1.0F;
    protected float f_119574_ = 1.0F;
-   protected double f_119575_;
-   protected double f_119576_;
-   protected double f_119577_;
+   protected me.alphamode.mclong.math.BigDecimal f_119575_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   protected me.alphamode.mclong.math.BigDecimal f_119576_ = me.alphamode.mclong.math.BigDecimal.ZERO;
+   protected me.alphamode.mclong.math.BigDecimal f_119577_ = me.alphamode.mclong.math.BigDecimal.ZERO;
    protected boolean f_119578_;
    protected int f_119579_;
    protected SoundInstance.Attenuation f_119580_ = SoundInstance.Attenuation.LINEAR;
@@ -79,15 +_,15 @@
       return this.f_119574_ * this.f_119570_.m_235147_().m_214084_(this.f_235066_);
    }
 
-   public double m_7772_() {
+   public me.alphamode.mclong.math.BigDecimal getX() {
       return this.f_119575_;
    }
 
-   public double m_7780_() {
+   public me.alphamode.mclong.math.BigDecimal getY() {
       return this.f_119576_;
    }
 
-   public double m_7778_() {
+   public me.alphamode.mclong.math.BigDecimal getZ() {
       return this.f_119577_;
    }
 
