--- a/net/minecraft/client/resources/sounds/BeeSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/BeeSoundInstance.java
@@ -19,9 +_,9 @@
    public BeeSoundInstance(Bee p_119621_, SoundEvent p_119622_, SoundSource p_119623_) {
       super(p_119622_, p_119623_, SoundInstance.m_235150_());
       this.f_119618_ = p_119621_;
-      this.f_119575_ = (double)((float)p_119621_.m_20185_());
-      this.f_119576_ = (double)((float)p_119621_.m_20186_());
-      this.f_119577_ = (double)((float)p_119621_.m_20189_());
+      this.f_119575_ = p_119621_.getBigX();
+      this.f_119576_ = p_119621_.getBigY();
+      this.f_119577_ = p_119621_.getBigZ();
       this.f_119578_ = true;
       this.f_119579_ = 0;
       this.f_119573_ = 0.0F;
@@ -35,9 +_,9 @@
       }
 
       if (!this.f_119618_.m_213877_() && !this.f_119619_) {
-         this.f_119575_ = (double)((float)this.f_119618_.m_20185_());
-         this.f_119576_ = (double)((float)this.f_119618_.m_20186_());
-         this.f_119577_ = (double)((float)this.f_119618_.m_20189_());
+         this.f_119575_ = this.f_119618_.getBigX();
+         this.f_119576_ = this.f_119618_.getBigY();
+         this.f_119577_ = this.f_119618_.getBigZ();
          float f = (float)this.f_119618_.m_20184_().m_165924_();
          if (f >= 0.01F) {
             this.f_119574_ = Mth.m_14179_(Mth.m_14036_(f, this.m_119627_(), this.m_119628_()), this.m_119627_(), this.m_119628_());
