--- a/net/minecraft/client/resources/sounds/ElytraOnPlayerSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/ElytraOnPlayerSoundInstance.java
@@ -24,9 +_,9 @@
    public void m_7788_() {
       ++this.f_119671_;
       if (!this.f_119670_.m_213877_() && (this.f_119671_ <= 20 || this.f_119670_.m_21255_())) {
-         this.f_119575_ = (double)((float)this.f_119670_.m_20185_());
-         this.f_119576_ = (double)((float)this.f_119670_.m_20186_());
-         this.f_119577_ = (double)((float)this.f_119670_.m_20189_());
+         this.f_119575_ = this.f_119670_.getBigX();
+         this.f_119576_ = this.f_119670_.getBigY();
+         this.f_119577_ = this.f_119670_.getBigZ();
          float f = (float)this.f_119670_.m_20184_().m_82556_();
          if ((double)f >= 1.0E-7D) {
             this.f_119573_ = Mth.m_14036_(f / 4.0F, 0.0F, 1.0F);
