--- a/net/minecraft/client/resources/sounds/GuardianAttackSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/GuardianAttackSoundInstance.java
@@ -28,9 +_,9 @@
 
    public void m_7788_() {
       if (!this.f_119688_.m_213877_() && this.f_119688_.m_5448_() == null) {
-         this.f_119575_ = (double)((float)this.f_119688_.m_20185_());
-         this.f_119576_ = (double)((float)this.f_119688_.m_20186_());
-         this.f_119577_ = (double)((float)this.f_119688_.m_20189_());
+         this.f_119575_ = this.f_119688_.getBigX();
+         this.f_119576_ = this.f_119688_.getBigY();
+         this.f_119577_ = this.f_119688_.getBigZ();
          float f = this.f_119688_.m_32812_(0.0F);
          this.f_119573_ = 0.0F + 1.0F * f * f;
          this.f_119574_ = 0.7F + 0.5F * f;
