--- a/net/minecraft/client/resources/sounds/MinecartSoundInstance.java
+++ b/net/minecraft/client/resources/sounds/MinecartSoundInstance.java
@@ -23,9 +_,9 @@
       this.f_119578_ = true;
       this.f_119579_ = 0;
       this.f_119573_ = 0.0F;
-      this.f_119575_ = (double)((float)p_119696_.m_20185_());
-      this.f_119576_ = (double)((float)p_119696_.m_20186_());
-      this.f_119577_ = (double)((float)p_119696_.m_20189_());
+      this.f_119575_ = p_119696_.getBigX();
+      this.f_119576_ = p_119696_.getBigY();
+      this.f_119577_ = p_119696_.getBigZ();
    }
 
    public boolean m_7767_() {
@@ -40,9 +_,9 @@
       if (this.f_119693_.m_213877_()) {
          this.m_119609_();
       } else {
-         this.f_119575_ = (double)((float)this.f_119693_.m_20185_());
-         this.f_119576_ = (double)((float)this.f_119693_.m_20186_());
-         this.f_119577_ = (double)((float)this.f_119693_.m_20189_());
+         this.f_119575_ = this.f_119693_.getBigX();
+         this.f_119576_ = this.f_119693_.getBigY();
+         this.f_119577_ = this.f_119693_.getBigZ();
          float f = (float)this.f_119693_.m_20184_().m_165924_();
          if (f >= 0.01F) {
             this.f_119694_ = Mth.m_14036_(this.f_119694_ + 0.0025F, 0.0F, 1.0F);
