--- a/net/minecraft/client/resources/sounds/SoundInstance.java
+++ b/net/minecraft/client/resources/sounds/SoundInstance.java
@@ -30,11 +_,11 @@
 
    float m_7783_();
 
-   double m_7772_();
-
-   double m_7780_();
-
-   double m_7778_();
+   me.alphamode.mclong.math.BigDecimal getX();
+
+   me.alphamode.mclong.math.BigDecimal getY();
+
+   me.alphamode.mclong.math.BigDecimal getZ();
 
    SoundInstance.Attenuation m_7438_();
 
