--- a/net/minecraft/gametest/framework/GameTestAssertPosException.java
+++ b/net/minecraft/gametest/framework/GameTestAssertPosException.java
@@ -16,7 +_,7 @@
    }
 
    public String getMessage() {
-      String s = this.f_127493_.m_123341_() + "," + this.f_127493_.m_123342_() + "," + this.f_127493_.m_123343_() + " (relative: " + this.f_127494_.m_123341_() + "," + this.f_127494_.m_123342_() + "," + this.f_127494_.m_123343_() + ")";
+      String s = this.f_127493_.getX() + "," + this.f_127493_.getY() + "," + this.f_127493_.getZ() + " (relative: " + this.f_127494_.getX() + "," + this.f_127494_.getY() + "," + this.f_127494_.getZ() + ")";
       return super.getMessage() + " at " + s + " (t=" + this.f_127495_ + ")";
    }
 
