--- a/net/minecraft/gametest/framework/GameTestBatchRunner.java
+++ b/net/minecraft/gametest/framework/GameTestBatchRunner.java
@@ -99,10 +_,10 @@
          gametestinfo.m_127617_(structureblockentity.m_58899_());
          map.put(gametestinfo, new BlockPos(this.f_127560_));
          aabb = aabb.m_82367_(aabb1);
-         this.f_127560_.m_122184_((int)aabb1.m_82362_() + 5, 0, 0);
+         this.f_127560_.move((int)aabb1.m_82362_() + 5, 0, 0);
          if (i++ % this.f_127554_ == this.f_127554_ - 1) {
-            this.f_127560_.m_122184_(0, 0, (int)aabb.m_82385_() + 6);
-            this.f_127560_.m_142451_(this.f_127551_.m_123341_());
+            this.f_127560_.move(0, 0, (int)aabb.m_82385_() + 6);
+            this.f_127560_.setX(this.f_127551_.getX());
             aabb = new AABB(this.f_127560_);
          }
       }
