--- a/net/minecraft/gametest/framework/GameTestInfo.java
+++ b/net/minecraft/gametest/framework/GameTestInfo.java
@@ -228,7 +_,7 @@
          throw new IllegalStateException("Expected structure to be initialized, but it was null");
       } else {
          BoundingBox boundingbox = StructureUtils.m_127904_(this.f_177469_);
-         StructureUtils.m_127849_(boundingbox, this.f_127599_.m_123342_(), this.f_127600_);
+         StructureUtils.clearSpaceForStructure(boundingbox, this.f_127599_.getY(), this.f_127600_);
       }
    }
 
