--- a/net/minecraft/gametest/framework/GameTestRunner.java
+++ b/net/minecraft/gametest/framework/GameTestRunner.java
@@ -57,15 +_,15 @@
 
    public static void m_127694_(ServerLevel p_127695_, BlockPos p_127696_, GameTestTicker p_127697_, int p_127698_) {
       p_127697_.m_127787_();
-      BlockPos blockpos = p_127696_.m_7918_(-p_127698_, 0, -p_127698_);
-      BlockPos blockpos1 = p_127696_.m_7918_(p_127698_, 0, p_127698_);
+      BlockPos blockpos = p_127696_.offset(-p_127698_, 0, -p_127698_);
+      BlockPos blockpos1 = p_127696_.offset(p_127698_, 0, p_127698_);
       BlockPos.m_121990_(blockpos, blockpos1).filter((p_177540_) -> {
          return p_127695_.m_8055_(p_177540_).m_60713_(Blocks.f_50677_);
       }).forEach((p_177529_) -> {
          StructureBlockEntity structureblockentity = (StructureBlockEntity)p_127695_.m_7702_(p_177529_);
          BlockPos blockpos2 = structureblockentity.m_58899_();
          BoundingBox boundingbox = StructureUtils.m_127904_(structureblockentity);
-         StructureUtils.m_127849_(boundingbox, blockpos2.m_123342_(), p_127695_);
+         StructureUtils.clearSpaceForStructure(boundingbox, blockpos2.getY(), p_127695_);
       });
    }
 
