--- a/net/minecraft/gametest/framework/GameTestInfo.java
+++ b/net/minecraft/gametest/framework/GameTestInfo.java
@@ -140,9 +_,9 @@
    }
 
    @Nullable
-   public Vec3i m_177483_() {
+   public me.alphamode.mclong.core.Vec3l getStructureSize() {
       StructureBlockEntity structureblockentity = this.m_177471_();
-      return structureblockentity == null ? null : structureblockentity.m_155805_();
+      return structureblockentity == null ? null : structureblockentity.getStructureSize();
    }
 
    @Nullable
@@ -228,7 +_,7 @@
          throw new IllegalStateException("Expected structure to be initialized, but it was null");
       } else {
          BoundingBox boundingbox = StructureUtils.m_127904_(this.f_177469_);
-         StructureUtils.m_127849_(boundingbox, this.f_127599_.m_123342_(), this.f_127600_);
+         StructureUtils.clearSpaceForStructure(boundingbox, this.f_127599_.getY(), this.f_127600_);
       }
    }
 
