--- a/net/minecraft/gametest/framework/GameTestRunner.java
+++ b/net/minecraft/gametest/framework/GameTestRunner.java
@@ -65,7 +_,7 @@
          StructureBlockEntity structureblockentity = (StructureBlockEntity)p_127695_.m_7702_(p_177529_);
          BlockPos blockpos2 = structureblockentity.m_58899_();
          BoundingBox boundingbox = StructureUtils.m_127904_(structureblockentity);
-         StructureUtils.m_127849_(boundingbox, blockpos2.m_123342_(), p_127695_);
+         StructureUtils.clearSpaceForStructure(boundingbox, blockpos2.getY(), p_127695_);
       });
    }
 
