--- a/net/minecraft/gametest/framework/ReportGameListener.java
+++ b/net/minecraft/gametest/framework/ReportGameListener.java
@@ -117,7 +_,7 @@
       ServerLevel serverlevel = p_177720_.m_127637_();
       BlockPos blockpos = p_177720_.m_127636_();
       BlockPos blockpos1 = new BlockPos(-1, -1, -1);
-      BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos.m_121955_(blockpos1), Mirror.NONE, p_177720_.m_127646_(), blockpos);
+      BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos.offset(blockpos1), Mirror.NONE, p_177720_.m_127646_(), blockpos);
       serverlevel.m_46597_(blockpos2, Blocks.f_50273_.m_49966_().m_60717_(p_177720_.m_127646_()));
       BlockPos blockpos3 = blockpos2.m_7918_(0, 1, 0);
       serverlevel.m_46597_(blockpos3, p_177721_.m_49966_());
@@ -135,7 +_,7 @@
       ServerLevel serverlevel = p_177739_.m_127637_();
       BlockPos blockpos = p_177739_.m_127636_();
       BlockPos blockpos1 = new BlockPos(-1, 1, -1);
-      BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos.m_121955_(blockpos1), Mirror.NONE, p_177739_.m_127646_(), blockpos);
+      BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos.offset(blockpos1), Mirror.NONE, p_177739_.m_127646_(), blockpos);
       serverlevel.m_46597_(blockpos2, Blocks.f_50624_.m_49966_().m_60717_(p_177739_.m_127646_()));
       BlockState blockstate = serverlevel.m_8055_(blockpos2);
       ItemStack itemstack = m_177710_(p_177739_.m_127633_(), p_177739_.m_127643_(), p_177740_);
