--- a/net/minecraft/gametest/framework/StructureUtils.java
+++ b/net/minecraft/gametest/framework/StructureUtils.java
@@ -102,59 +_,59 @@
 
    public static AABB m_127847_(StructureBlockEntity p_127848_) {
       BlockPos blockpos = p_127848_.m_58899_();
-      BlockPos blockpos1 = blockpos.m_121955_(p_127848_.m_155805_().m_7918_(-1, -1, -1));
+      BlockPos blockpos1 = blockpos.offset(p_127848_.getStructureSize().offset(-1, -1, -1));
       BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos1, Mirror.NONE, p_127848_.m_59906_(), blockpos);
       return new AABB(blockpos, blockpos2);
    }
 
    public static BoundingBox m_127904_(StructureBlockEntity p_127905_) {
       BlockPos blockpos = p_127905_.m_58899_();
-      BlockPos blockpos1 = blockpos.m_121955_(p_127905_.m_155805_().m_7918_(-1, -1, -1));
+      BlockPos blockpos1 = blockpos.offset(p_127905_.getStructureSize().offset(-1, -1, -1));
       BlockPos blockpos2 = StructureTemplate.m_74593_(blockpos1, Mirror.NONE, p_127905_.m_59906_(), blockpos);
-      return BoundingBox.m_162375_(blockpos, blockpos2);
+      return BoundingBox.fromCorners(blockpos, blockpos2);
    }
 
    public static void m_127875_(BlockPos p_127876_, BlockPos p_127877_, Rotation p_127878_, ServerLevel p_127879_) {
-      BlockPos blockpos = StructureTemplate.m_74593_(p_127876_.m_121955_(p_127877_), Mirror.NONE, p_127878_, p_127876_);
+      BlockPos blockpos = StructureTemplate.m_74593_(p_127876_.offset(p_127877_), Mirror.NONE, p_127878_, p_127876_);
       p_127879_.m_46597_(blockpos, Blocks.f_50272_.m_49966_());
       CommandBlockEntity commandblockentity = (CommandBlockEntity)p_127879_.m_7702_(blockpos);
       commandblockentity.m_59141_().m_6590_("test runthis");
-      BlockPos blockpos1 = StructureTemplate.m_74593_(blockpos.m_7918_(0, 0, -1), Mirror.NONE, p_127878_, blockpos);
+      BlockPos blockpos1 = StructureTemplate.m_74593_(blockpos.offset(0, 0, -1), Mirror.NONE, p_127878_, blockpos);
       p_127879_.m_46597_(blockpos1, Blocks.f_50124_.m_49966_().m_60717_(p_127878_));
    }
 
-   public static void m_177764_(String p_177765_, BlockPos p_177766_, Vec3i p_177767_, Rotation p_177768_, ServerLevel p_177769_) {
-      BoundingBox boundingbox = m_177760_(p_177766_, p_177767_, p_177768_);
-      m_127849_(boundingbox, p_177766_.m_123342_(), p_177769_);
+   public static void createNewEmptyStructureBlock(String p_177765_, BlockPos p_177766_, me.alphamode.mclong.core.Vec3l p_177767_, Rotation p_177768_, ServerLevel p_177769_) {
+      BoundingBox boundingbox = getStructureBoundingBox(p_177766_, p_177767_, p_177768_);
+      clearSpaceForStructure(boundingbox, p_177766_.getY(), p_177769_);
       p_177769_.m_46597_(p_177766_, Blocks.f_50677_.m_49966_());
       StructureBlockEntity structureblockentity = (StructureBlockEntity)p_177769_.m_7702_(p_177766_);
       structureblockentity.m_59876_(false);
       structureblockentity.m_59874_(new ResourceLocation(p_177765_));
-      structureblockentity.m_155797_(p_177767_);
+      structureblockentity.setStructureSize(p_177767_);
       structureblockentity.m_59860_(StructureMode.SAVE);
       structureblockentity.m_59898_(true);
    }
 
    public static StructureBlockEntity m_127883_(String p_127884_, BlockPos p_127885_, Rotation p_127886_, int p_127887_, ServerLevel p_127888_, boolean p_127889_) {
-      Vec3i vec3i = m_127880_(p_127884_, p_127888_).m_163801_();
-      BoundingBox boundingbox = m_177760_(p_127885_, vec3i, p_127886_);
+      me.alphamode.mclong.core.Vec3l vec3i = m_127880_(p_127884_, p_127888_).getSize();
+      BoundingBox boundingbox = getStructureBoundingBox(p_127885_, vec3i, p_127886_);
       BlockPos blockpos;
       if (p_127886_ == Rotation.NONE) {
          blockpos = p_127885_;
       } else if (p_127886_ == Rotation.CLOCKWISE_90) {
-         blockpos = p_127885_.m_7918_(vec3i.m_123343_() - 1, 0, 0);
+         blockpos = p_127885_.offset(vec3i.getZ() - 1, 0, 0);
       } else if (p_127886_ == Rotation.CLOCKWISE_180) {
-         blockpos = p_127885_.m_7918_(vec3i.m_123341_() - 1, 0, vec3i.m_123343_() - 1);
+         blockpos = p_127885_.offset(vec3i.getX() - 1, 0, vec3i.getZ() - 1);
       } else {
          if (p_127886_ != Rotation.COUNTERCLOCKWISE_90) {
             throw new IllegalArgumentException("Invalid rotation: " + p_127886_);
          }
 
-         blockpos = p_127885_.m_7918_(0, 0, vec3i.m_123341_() - 1);
+         blockpos = p_127885_.offset(0, 0, vec3i.getX() - 1);
       }
 
       m_127857_(p_127885_, p_127888_);
-      m_127849_(boundingbox, p_127885_.m_123342_(), p_127888_);
+      clearSpaceForStructure(boundingbox, p_127885_.getY(), p_127888_);
       StructureBlockEntity structureblockentity = m_127890_(p_127884_, blockpos, p_127886_, p_127888_, p_127889_);
       p_127888_.m_183326_().m_193234_(boundingbox);
       p_127888_.m_8722_(boundingbox);
@@ -166,35 +_,35 @@
 
       for(int i = -1; i < 4; ++i) {
          for(int j = -1; j < 4; ++j) {
-            int k = chunkpos.f_45578_ + i;
-            int l = chunkpos.f_45579_ + j;
-            p_127859_.m_8602_(k, l, true);
+            long k = chunkpos.f_45578_ + i;
+            long l = chunkpos.f_45579_ + j;
+            p_127859_.setChunkForced(k, l, true);
          }
       }
 
    }
 
-   public static void m_127849_(BoundingBox p_127850_, int p_127851_, ServerLevel p_127852_) {
-      BoundingBox boundingbox = new BoundingBox(p_127850_.m_162395_() - 2, p_127850_.m_162396_() - 3, p_127850_.m_162398_() - 3, p_127850_.m_162399_() + 3, p_127850_.m_162400_() + 20, p_127850_.m_162401_() + 3);
+   public static void clearSpaceForStructure(BoundingBox p_127850_, long p_127851_, ServerLevel p_127852_) {
+      BoundingBox boundingbox = new BoundingBox(p_127850_.minX() - 2, p_127850_.minY() - 3, p_127850_.minZ() - 3, p_127850_.maxX() + 3, p_127850_.maxY() + 20, p_127850_.maxZ() + 3);
       BlockPos.m_121919_(boundingbox).forEach((p_177748_) -> {
-         m_127841_(p_127851_, p_177748_, p_127852_);
+         clearBlock(p_127851_, p_177748_, p_127852_);
       });
       p_127852_.m_183326_().m_193234_(boundingbox);
       p_127852_.m_8722_(boundingbox);
-      AABB aabb = new AABB((double)boundingbox.m_162395_(), (double)boundingbox.m_162396_(), (double)boundingbox.m_162398_(), (double)boundingbox.m_162399_(), (double)boundingbox.m_162400_(), (double)boundingbox.m_162401_());
+      AABB aabb = new AABB((double)boundingbox.minX(), (double)boundingbox.minY(), (double)boundingbox.minZ(), (double)boundingbox.maxX(), (double)boundingbox.maxY(), (double)boundingbox.maxZ());
       List<Entity> list = p_127852_.m_6443_(Entity.class, aabb, (p_177750_) -> {
          return !(p_177750_ instanceof Player);
       });
       list.forEach(Entity::m_146870_);
    }
 
-   public static BoundingBox m_177760_(BlockPos p_177761_, Vec3i p_177762_, Rotation p_177763_) {
-      BlockPos blockpos = p_177761_.m_121955_(p_177762_).m_7918_(-1, -1, -1);
+   public static BoundingBox getStructureBoundingBox(BlockPos p_177761_, me.alphamode.mclong.core.Vec3l p_177762_, Rotation p_177763_) {
+      BlockPos blockpos = p_177761_.offset(p_177762_).offset(-1, -1, -1);
       BlockPos blockpos1 = StructureTemplate.m_74593_(blockpos, Mirror.NONE, p_177763_, p_177761_);
-      BoundingBox boundingbox = BoundingBox.m_162375_(p_177761_, blockpos1);
-      int i = Math.min(boundingbox.m_162395_(), boundingbox.m_162399_());
-      int j = Math.min(boundingbox.m_162398_(), boundingbox.m_162401_());
-      return boundingbox.m_162367_(p_177761_.m_123341_() - i, 0, p_177761_.m_123343_() - j);
+      BoundingBox boundingbox = BoundingBox.fromCorners(p_177761_, blockpos1);
+      long i = Math.min(boundingbox.minX(), boundingbox.maxX());
+      long j = Math.min(boundingbox.minZ(), boundingbox.maxZ());
+      return boundingbox.move(p_177761_.getX() - i, 0, p_177761_.getZ() - j);
    }
 
    public static Optional<BlockPos> m_127853_(BlockPos p_127854_, int p_127855_, ServerLevel p_127856_) {
@@ -206,7 +_,7 @@
    @Nullable
    public static BlockPos m_127906_(BlockPos p_127907_, int p_127908_, ServerLevel p_127909_) {
       Comparator<BlockPos> comparator = Comparator.comparingInt((p_177759_) -> {
-         return p_177759_.m_123333_(p_127907_);
+         return p_177759_.distManhattan(p_127907_);
       });
       Collection<BlockPos> collection = m_127910_(p_127907_, p_127908_, p_127909_);
       Optional<BlockPos> optional = collection.stream().min(comparator);
@@ -218,9 +_,9 @@
       AABB aabb = new AABB(p_127911_);
       aabb = aabb.m_82400_((double)p_127912_);
 
-      for(int i = (int)aabb.f_82288_; i <= (int)aabb.f_82291_; ++i) {
-         for(int j = (int)aabb.f_82289_; j <= (int)aabb.f_82292_; ++j) {
-            for(int k = (int)aabb.f_82290_; k <= (int)aabb.f_82293_; ++k) {
+      for(int i = (int)aabb.f_82288_.doubleValue(); i <= (int)aabb.f_82291_.doubleValue(); ++i) {
+         for(int j = (int)aabb.f_82289_.doubleValue(); j <= (int)aabb.f_82292_.doubleValue(); ++j) {
+            for(int k = (int)aabb.f_82290_.doubleValue(); k <= (int)aabb.f_82293_.doubleValue(); ++k) {
                BlockPos blockpos = new BlockPos(i, j, k);
                BlockState blockstate = p_127913_.m_8055_(blockpos);
                if (blockstate.m_60713_(Blocks.f_50677_)) {
@@ -258,12 +_,12 @@
       structureblockentity.m_59876_(false);
       structureblockentity.m_59874_(new ResourceLocation(p_127891_));
       structureblockentity.m_59844_(p_127894_, p_127895_);
-      if (structureblockentity.m_155805_() != Vec3i.f_123288_) {
+      if (structureblockentity.getStructureSize() != me.alphamode.mclong.core.Vec3l.ZERO) {
          return structureblockentity;
       } else {
          StructureTemplate structuretemplate = m_127880_(p_127891_, p_127894_);
          structureblockentity.m_59847_(p_127894_, p_127895_, structuretemplate);
-         if (structureblockentity.m_155805_() == Vec3i.f_123288_) {
+         if (structureblockentity.getStructureSize() == me.alphamode.mclong.core.Vec3l.ZERO) {
             throw new RuntimeException("Failed to load structure " + p_127891_);
          } else {
             return structureblockentity;
@@ -284,14 +_,14 @@
       }
    }
 
-   private static void m_127841_(int p_127842_, BlockPos p_127843_, ServerLevel p_127844_) {
+   private static void clearBlock(long p_127842_, BlockPos p_127843_, ServerLevel p_127844_) {
       BlockState blockstate = null;
       RegistryAccess registryaccess = p_127844_.m_9598_();
       FlatLevelGeneratorSettings flatlevelgeneratorsettings = FlatLevelGeneratorSettings.m_254980_(registryaccess.m_255025_(Registries.f_256952_), registryaccess.m_255025_(Registries.f_256998_), registryaccess.m_255025_(Registries.f_256988_));
       List<BlockState> list = flatlevelgeneratorsettings.m_161917_();
-      int i = p_127843_.m_123342_() - p_127844_.m_141937_();
-      if (p_127843_.m_123342_() < p_127842_ && i > 0 && i <= list.size()) {
-         blockstate = list.get(i - 1);
+      long i = p_127843_.getY() - p_127844_.getMinBuildHeight();
+      if (p_127843_.getY() < p_127842_ && i > 0 && i <= list.size()) {
+         blockstate = list.get((int) (i - 1));
       }
 
       if (blockstate == null) {
@@ -306,6 +_,6 @@
    private static boolean m_127867_(BlockPos p_127868_, BlockPos p_127869_, ServerLevel p_127870_) {
       StructureBlockEntity structureblockentity = (StructureBlockEntity)p_127870_.m_7702_(p_127868_);
       AABB aabb = m_127847_(structureblockentity).m_82400_(1.0D);
-      return aabb.m_82390_(Vec3.m_82512_(p_127869_));
+      return aabb.m_82390_(Vec3.atCenterOf(p_127869_));
    }
 }
