--- a/net/minecraft/advancements/AdvancementRewards.java
+++ b/net/minecraft/advancements/AdvancementRewards.java
@@ -49,7 +_,7 @@
       for(ResourceLocation resourcelocation : this.f_9980_) {
          for(ItemStack itemstack : p_9990_.f_8924_.m_278653_().m_278676_(resourcelocation).m_287195_(lootparams)) {
             if (p_9990_.m_36356_(itemstack)) {
-               p_9990_.m_9236_().m_6263_((Player)null, p_9990_.m_20185_(), p_9990_.m_20186_(), p_9990_.m_20189_(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, ((p_9990_.m_217043_().m_188501_() - p_9990_.m_217043_().m_188501_()) * 0.7F + 1.0F) * 2.0F);
+               p_9990_.m_9236_().playSound((Player)null, p_9990_.getBigX(), p_9990_.getBigY(), p_9990_.getBigZ(), SoundEvents.f_12019_, SoundSource.PLAYERS, 0.2F, ((p_9990_.m_217043_().m_188501_() - p_9990_.m_217043_().m_188501_()) * 0.7F + 1.0F) * 2.0F);
                flag = true;
             } else {
                ItemEntity itementity = p_9990_.m_36176_(itemstack, false);
