--- a/net/minecraft/world/level/BaseSpawner.java
+++ b/net/minecraft/world/level/BaseSpawner.java
@@ -49,7 +_,7 @@
    }
 
    private boolean m_151343_(Level p_151344_, BlockPos p_151345_) {
-      return p_151344_.m_45914_((double)p_151345_.m_123341_() + 0.5D, (double)p_151345_.m_123342_() + 0.5D, (double)p_151345_.m_123343_() + 0.5D, (double)this.f_45452_);
+      return p_151344_.hasNearbyAlivePlayer(p_151345_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_151345_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_151345_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), (double)this.f_45452_);
    }
 
    public void m_151319_(Level p_151320_, BlockPos p_151321_) {
@@ -57,11 +_,11 @@
          this.f_45446_ = this.f_45445_;
       } else if (this.f_45450_ != null) {
          RandomSource randomsource = p_151320_.m_213780_();
-         double d0 = (double)p_151321_.m_123341_() + randomsource.m_188500_();
-         double d1 = (double)p_151321_.m_123342_() + randomsource.m_188500_();
-         double d2 = (double)p_151321_.m_123343_() + randomsource.m_188500_();
-         p_151320_.m_7106_(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
-         p_151320_.m_7106_(ParticleTypes.f_123744_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
+         me.alphamode.mclong.math.BigDecimal d0 = p_151321_.getBigDecX().add(randomsource.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d1 = p_151321_.getBigDecY().add(randomsource.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d2 = p_151321_.getBigDecZ().add(randomsource.m_188500_());
+         p_151320_.addParticle(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
+         p_151320_.addParticle(ParticleTypes.f_123744_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
          if (this.f_45442_ > 0) {
             --this.f_45442_;
          }
@@ -95,11 +_,11 @@
 
                ListTag listtag = compoundtag.m_128437_("Pos", 6);
                int j = listtag.size();
-               double d0 = j >= 1 ? listtag.m_128772_(0) : (double)p_151313_.m_123341_() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
-               double d1 = j >= 2 ? listtag.m_128772_(1) : (double)(p_151313_.m_123342_() + randomsource.m_188503_(3) - 1);
-               double d2 = j >= 3 ? listtag.m_128772_(2) : (double)p_151313_.m_123343_() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
-               if (p_151312_.m_45772_(optional.get().m_20585_(d0, d1, d2))) {
-                  BlockPos blockpos = BlockPos.m_274561_(d0, d1, d2);
+               me.alphamode.mclong.math.BigDecimal d0 = j >= 1 ? new me.alphamode.mclong.math.BigDecimal(listtag.m_128778_(0)) : p_151313_.getBigDecX().add((randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D);
+               me.alphamode.mclong.math.BigDecimal d1 = j >= 2 ? new me.alphamode.mclong.math.BigDecimal(listtag.m_128778_(1)) : (p_151313_.getBigDecY().add(randomsource.m_188503_(3) - 1));
+               me.alphamode.mclong.math.BigDecimal d2 = j >= 3 ? new me.alphamode.mclong.math.BigDecimal(listtag.m_128778_(2)) : p_151313_.getBigDecZ().add((randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D);
+               if (p_151312_.m_45772_(optional.get().getAABB(d0, d1, d2))) {
+                  BlockPos blockpos = BlockPos.containing(d0, d1, d2);
                   if (spawndata.m_186574_().isPresent()) {
                      if (!optional.get().m_20674_().m_21609_() && p_151312_.m_46791_() == Difficulty.PEACEFUL) {
                         continue;
@@ -114,7 +_,7 @@
                   }
 
                   Entity entity = EntityType.m_20645_(compoundtag, p_151312_, (p_151310_) -> {
-                     p_151310_.m_7678_(d0, d1, d2, p_151310_.m_146908_(), p_151310_.m_146909_());
+                     p_151310_.moveTo(d0, d1, d2, p_151310_.m_146908_(), p_151310_.m_146909_());
                      return p_151310_;
                   });
                   if (entity == null) {
@@ -122,13 +_,13 @@
                      return;
                   }
 
-                  int k = p_151312_.m_45976_(entity.getClass(), (new AABB((double)p_151313_.m_123341_(), (double)p_151313_.m_123342_(), (double)p_151313_.m_123343_(), (double)(p_151313_.m_123341_() + 1), (double)(p_151313_.m_123342_() + 1), (double)(p_151313_.m_123343_() + 1))).m_82400_((double)this.f_45453_)).size();
+                  int k = p_151312_.m_45976_(entity.getClass(), (new AABB(p_151313_.getBigDecX(), p_151313_.getBigDecY(), p_151313_.getBigDecZ(), (p_151313_.getBigDecX().add()), (p_151313_.getBigDecY().add()), (p_151313_.getBigDecZ().add()))).m_82400_((double)this.f_45453_)).size();
                   if (k >= this.f_45451_) {
                      this.m_151350_(p_151312_, p_151313_);
                      return;
                   }
 
-                  entity.m_7678_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), randomsource.m_188501_() * 360.0F, 0.0F);
+                  entity.moveTo(entity.getBigX(), entity.getBigY(), entity.getBigZ(), randomsource.m_188501_() * 360.0F, 0.0F);
                   if (entity instanceof Mob) {
                      Mob mob = (Mob)entity;
                      if (spawndata.m_186574_().isEmpty() && !mob.m_5545_(p_151312_, MobSpawnType.SPAWNER) || !mob.m_6914_(p_151312_)) {
