--- a/net/minecraft/world/level/BaseSpawner.java
+++ b/net/minecraft/world/level/BaseSpawner.java
@@ -49,7 +_,7 @@
    }
 
    private boolean m_151343_(Level p_151344_, BlockPos p_151345_) {
-      return p_151344_.m_45914_((double)p_151345_.m_123341_() + 0.5D, (double)p_151345_.m_123342_() + 0.5D, (double)p_151345_.m_123343_() + 0.5D, (double)this.f_45452_);
+      return p_151344_.m_45914_((double)p_151345_.getX() + 0.5D, (double)p_151345_.getY() + 0.5D, (double)p_151345_.getZ() + 0.5D, (double)this.f_45452_);
    }
 
    public void m_151319_(Level p_151320_, BlockPos p_151321_) {
@@ -57,11 +_,11 @@
          this.f_45446_ = this.f_45445_;
       } else if (this.f_45450_ != null) {
          RandomSource randomsource = p_151320_.m_213780_();
-         double d0 = (double)p_151321_.m_123341_() + randomsource.m_188500_();
-         double d1 = (double)p_151321_.m_123342_() + randomsource.m_188500_();
-         double d2 = (double)p_151321_.m_123343_() + randomsource.m_188500_();
-         p_151320_.m_7106_(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
-         p_151320_.m_7106_(ParticleTypes.f_123744_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
+         me.alphamode.mclong.math.BigDecimal d0 = p_151321_.getBigDecX().add(randomsource.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d1 = p_151321_.getBigDecY().add(randomsource.m_188500_());
+         me.alphamode.mclong.math.BigDecimal d2 = p_151321_.getBigDecZ().add(randomsource.m_188500_());
+         p_151320_.addParticle(ParticleTypes.f_123762_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
+         p_151320_.addParticle(ParticleTypes.f_123744_, d0, d1, d2, 0.0D, 0.0D, 0.0D);
          if (this.f_45442_ > 0) {
             --this.f_45442_;
          }
@@ -95,9 +_,9 @@
 
                ListTag listtag = compoundtag.m_128437_("Pos", 6);
                int j = listtag.size();
-               double d0 = j >= 1 ? listtag.m_128772_(0) : (double)p_151313_.m_123341_() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
-               double d1 = j >= 2 ? listtag.m_128772_(1) : (double)(p_151313_.m_123342_() + randomsource.m_188503_(3) - 1);
-               double d2 = j >= 3 ? listtag.m_128772_(2) : (double)p_151313_.m_123343_() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
+               double d0 = j >= 1 ? listtag.m_128772_(0) : (double)p_151313_.getX() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
+               double d1 = j >= 2 ? listtag.m_128772_(1) : (double)(p_151313_.getY() + randomsource.m_188503_(3) - 1);
+               double d2 = j >= 3 ? listtag.m_128772_(2) : (double)p_151313_.getZ() + (randomsource.m_188500_() - randomsource.m_188500_()) * (double)this.f_45453_ + 0.5D;
                if (p_151312_.m_45772_(optional.get().m_20585_(d0, d1, d2))) {
                   BlockPos blockpos = BlockPos.m_274561_(d0, d1, d2);
                   if (spawndata.m_186574_().isPresent()) {
@@ -122,7 +_,7 @@
                      return;
                   }
 
-                  int k = p_151312_.m_45976_(entity.getClass(), (new AABB((double)p_151313_.m_123341_(), (double)p_151313_.m_123342_(), (double)p_151313_.m_123343_(), (double)(p_151313_.m_123341_() + 1), (double)(p_151313_.m_123342_() + 1), (double)(p_151313_.m_123343_() + 1))).m_82400_((double)this.f_45453_)).size();
+                  int k = p_151312_.m_45976_(entity.getClass(), (new AABB((double)p_151313_.getX(), (double)p_151313_.getY(), (double)p_151313_.getZ(), (double)(p_151313_.getX() + 1), (double)(p_151313_.getY() + 1), (double)(p_151313_.getZ() + 1))).m_82400_((double)this.f_45453_)).size();
                   if (k >= this.f_45451_) {
                      this.m_151350_(p_151312_, p_151313_);
                      return;
