--- a/net/minecraft/world/level/BlockGetter.java
+++ b/net/minecraft/world/level/BlockGetter.java
@@ -97,7 +_,7 @@
 
    default double m_45573_(BlockPos p_45574_) {
       return this.m_45564_(this.m_8055_(p_45574_).m_60812_(this, p_45574_), () -> {
-         BlockPos blockpos = p_45574_.m_7495_();
+         BlockPos blockpos = p_45574_.below();
          return this.m_8055_(blockpos).m_60812_(this, blockpos);
       });
    }
@@ -112,9 +_,9 @@
          double d3 = Mth.m_14139_(-1.0E-7D, p_151362_.f_82479_, p_151363_.f_82479_);
          double d4 = Mth.m_14139_(-1.0E-7D, p_151362_.f_82480_, p_151363_.f_82480_);
          double d5 = Mth.m_14139_(-1.0E-7D, p_151362_.f_82481_, p_151363_.f_82481_);
-         int i = Mth.m_14107_(d3);
-         int j = Mth.m_14107_(d4);
-         int k = Mth.m_14107_(d5);
+         long i = Mth.m_14134_(d3);
+         long j = Mth.m_14134_(d4);
+         long k = Mth.m_14134_(d5);
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(i, j, k);
          T t = p_151365_.apply(p_151364_, blockpos$mutableblockpos);
          if (t != null) {
@@ -123,9 +_,9 @@
             double d6 = d0 - d3;
             double d7 = d1 - d4;
             double d8 = d2 - d5;
-            int l = Mth.m_14205_(d6);
-            int i1 = Mth.m_14205_(d7);
-            int j1 = Mth.m_14205_(d8);
+            long l = Mth.lsign(d6);
+            long i1 = Mth.lsign(d7);
+            long j1 = Mth.lsign(d8);
             double d9 = l == 0 ? Double.MAX_VALUE : (double)l / d6;
             double d10 = i1 == 0 ? Double.MAX_VALUE : (double)i1 / d7;
             double d11 = j1 == 0 ? Double.MAX_VALUE : (double)j1 / d8;
@@ -150,7 +_,7 @@
                   d14 += d11;
                }
 
-               T t1 = p_151365_.apply(p_151364_, blockpos$mutableblockpos.m_122178_(i, j, k));
+               T t1 = p_151365_.apply(p_151364_, blockpos$mutableblockpos.set(i, j, k));
                if (t1 != null) {
                   return t1;
                }
