--- a/net/minecraft/world/level/BlockGetter.java
+++ b/net/minecraft/world/level/BlockGetter.java
@@ -97,7 +_,7 @@
 
    default double m_45573_(BlockPos p_45574_) {
       return this.m_45564_(this.m_8055_(p_45574_).m_60812_(this, p_45574_), () -> {
-         BlockPos blockpos = p_45574_.m_7495_();
+         BlockPos blockpos = p_45574_.below();
          return this.m_8055_(blockpos).m_60812_(this, blockpos);
       });
    }
@@ -150,7 +_,7 @@
                   d14 += d11;
                }
 
-               T t1 = p_151365_.apply(p_151364_, blockpos$mutableblockpos.m_122178_(i, j, k));
+               T t1 = p_151365_.apply(p_151364_, blockpos$mutableblockpos.set(i, j, k));
                if (t1 != null) {
                   return t1;
                }
