--- a/net/minecraft/world/level/BlockCollisions.java
+++ b/net/minecraft/world/level/BlockCollisions.java
@@ -26,8 +_,9 @@
    private final boolean f_186398_;
    @Nullable
    private BlockGetter f_186399_;
-   private long f_186400_;
    private final BiFunction<BlockPos.MutableBlockPos, VoxelShape, T> f_285595_;
+   @Nullable
+   private ChunkPos f_186400_;
 
    public BlockCollisions(CollisionGetter p_286817_, @Nullable Entity p_286246_, AABB p_286624_, boolean p_286354_, BiFunction<BlockPos.MutableBlockPos, VoxelShape, T> p_286303_) {
       this.f_186393_ = p_286246_ == null ? CollisionContext.m_82749_() : CollisionContext.m_82750_(p_286246_);
@@ -37,24 +_,25 @@
       this.f_186392_ = p_286624_;
       this.f_186398_ = p_286354_;
       this.f_285595_ = p_286303_;
-      int i = Mth.m_14107_(p_286624_.f_82288_ - 1.0E-7D) - 1;
-      int j = Mth.m_14107_(p_286624_.f_82291_ + 1.0E-7D) + 1;
-      int k = Mth.m_14107_(p_286624_.f_82289_ - 1.0E-7D) - 1;
-      int l = Mth.m_14107_(p_286624_.f_82292_ + 1.0E-7D) + 1;
-      int i1 = Mth.m_14107_(p_286624_.f_82290_ - 1.0E-7D) - 1;
-      int j1 = Mth.m_14107_(p_286624_.f_82293_ + 1.0E-7D) + 1;
+      long i = Mth.m_14134_(p_286624_.f_82288_ - 1.0E-7D) - 1;
+      long j = Mth.m_14134_(p_286624_.f_82291_ + 1.0E-7D) + 1;
+      long k = Mth.m_14134_(p_286624_.f_82289_ - 1.0E-7D) - 1;
+      long l = Mth.m_14134_(p_286624_.f_82292_ + 1.0E-7D) + 1;
+      long i1 = Mth.m_14134_(p_286624_.f_82290_ - 1.0E-7D) - 1;
+      long j1 = Mth.m_14134_(p_286624_.f_82293_ + 1.0E-7D) + 1;
       this.f_186394_ = new Cursor3D(i, k, i1, j, l, j1);
    }
 
    @Nullable
-   private BlockGetter m_186411_(int p_186412_, int p_186413_) {
-      int i = SectionPos.m_123171_(p_186412_);
-      int j = SectionPos.m_123171_(p_186413_);
-      long k = ChunkPos.m_45589_(i, j);
-      if (this.f_186399_ != null && this.f_186400_ == k) {
+   private BlockGetter getChunk(long p_186412_, long y, long p_186413_) {
+      long i = SectionPos.blockToSectionCoord(p_186412_);
+      long chunkY = SectionPos.blockToSectionCoord(y);
+      long j = SectionPos.blockToSectionCoord(p_186413_);
+      ChunkPos k = new ChunkPos(i, chunkY, j);
+      if (this.f_186399_ != null && this.f_186400_.equals(k)) {
          return this.f_186399_;
       } else {
-         BlockGetter blockgetter = this.f_186397_.m_7925_(i, j);
+         BlockGetter blockgetter = this.f_186397_.getChunkForCollisions(i, chunkY, j);
          this.f_186399_ = blockgetter;
          this.f_186400_ = k;
          return blockgetter;
@@ -64,20 +_,20 @@
    protected T computeNext() {
       while(true) {
          if (this.f_186394_.m_122304_()) {
-            int i = this.f_186394_.m_122305_();
-            int j = this.f_186394_.m_122306_();
-            int k = this.f_186394_.m_122307_();
+            long i = this.f_186394_.nextX();
+            long j = this.f_186394_.nextY();
+            long k = this.f_186394_.nextZ();
             int l = this.f_186394_.m_122308_();
             if (l == 3) {
                continue;
             }
 
-            BlockGetter blockgetter = this.m_186411_(i, k);
+            BlockGetter blockgetter = this.getChunk(i, j, k);
             if (blockgetter == null) {
                continue;
             }
 
-            this.f_186395_.m_122178_(i, j, k);
+            this.f_186395_.set(i, j, k);
             BlockState blockstate = blockgetter.m_8055_(this.f_186395_);
             if (this.f_186398_ && !blockstate.m_60828_(blockgetter, this.f_186395_) || l == 1 && !blockstate.m_60779_() || l == 2 && !blockstate.m_60713_(Blocks.f_50110_)) {
                continue;
