--- a/net/minecraft/world/level/ChunkPos.java
+++ b/net/minecraft/world/level/ChunkPos.java
@@ -8,55 +_,49 @@
 import net.minecraft.core.BlockPos;
 import net.minecraft.core.SectionPos;
 
-public class ChunkPos {
+public class ChunkPos implements java.lang.Comparable<ChunkPos> {
    private static final int f_199440_ = 1056;
-   public static final long f_45577_ = m_45589_(1875066, 1875066);
-   public static final ChunkPos f_186419_ = new ChunkPos(0, 0);
+   public static final ChunkPos f_45577_ = new ChunkPos(java.lang.Long.MAX_VALUE, java.lang.Long.MAX_VALUE, java.lang.Long.MAX_VALUE);
+   public static final ChunkPos f_186419_ = new ChunkPos(0, 0, 0);
    private static final long f_151375_ = 32L;
    private static final long f_151376_ = 4294967295L;
    private static final int f_151377_ = 5;
    public static final int f_220335_ = 32;
    private static final int f_151378_ = 31;
    public static final int f_220336_ = 31;
-   public final int f_45578_;
-   public final int f_45579_;
+   public final long f_45578_;
+   public final long y;
+   public final long f_45579_;
    private static final int f_151379_ = 1664525;
    private static final int f_151380_ = 1013904223;
    private static final int f_151381_ = -559038737;
 
-   public ChunkPos(int p_45582_, int p_45583_) {
+   public ChunkPos(long p_45582_, long y, long p_45583_) {
       this.f_45578_ = p_45582_;
+      this.y = y;
       this.f_45579_ = p_45583_;
    }
 
    public ChunkPos(BlockPos p_45587_) {
-      this.f_45578_ = SectionPos.m_123171_(p_45587_.m_123341_());
-      this.f_45579_ = SectionPos.m_123171_(p_45587_.m_123343_());
-   }
-
-   public ChunkPos(long p_45585_) {
-      this.f_45578_ = (int)p_45585_;
-      this.f_45579_ = (int)(p_45585_ >> 32);
-   }
-
-   public static ChunkPos m_220337_(int p_220338_, int p_220339_) {
-      return new ChunkPos(p_220338_ << 5, p_220339_ << 5);
-   }
-
-   public static ChunkPos m_220340_(int p_220341_, int p_220342_) {
-      return new ChunkPos((p_220341_ << 5) + 31, (p_220342_ << 5) + 31);
-   }
-
-   public long m_45588_() {
-      return m_45589_(this.f_45578_, this.f_45579_);
-   }
-
-   public static long m_45589_(int p_45590_, int p_45591_) {
+      this.f_45578_ = SectionPos.blockToSectionCoord(p_45587_.getX());
+      this.y = SectionPos.blockToSectionCoord(p_45587_.getY());
+      this.f_45579_ = SectionPos.blockToSectionCoord(p_45587_.getZ());
+   }
+
+   public static ChunkPos minFromRegion(long p_220338_, long y, long p_220339_) {
+      return new ChunkPos(p_220338_ << f_151377_, y << f_151377_, p_220339_ << f_151377_);
+   }
+
+   public static ChunkPos maxFromRegion(long p_220341_, long y, long p_220342_) {
+      return new ChunkPos((p_220341_ << f_151377_) + f_220336_, (y << f_151377_) + f_220336_, (p_220342_ << f_151377_) + f_220336_);
+   }
+
+   public static long asLong(long p_45590_, long p_45591_) {
       return (long)p_45590_ & 4294967295L | ((long)p_45591_ & 4294967295L) << 32;
    }
 
    public static long m_151388_(BlockPos p_151389_) {
-      return m_45589_(SectionPos.m_123171_(p_151389_.m_123341_()), SectionPos.m_123171_(p_151389_.m_123343_()));
+      return asLong(SectionPos.blockToSectionCoord(p_151389_.getX()), SectionPos.blockToSectionCoord(p_151389_.getZ()));
    }
 
    public static int m_45592_(long p_45593_) {
@@ -68,13 +_,14 @@
    }
 
    public int hashCode() {
-      return m_220343_(this.f_45578_, this.f_45579_);
+      return hash((int) this.f_45578_, (int) this.y, (int) this.f_45579_);
    }
 
-   public static int m_220343_(int p_220344_, int p_220345_) {
-      int i = 1664525 * p_220344_ + 1013904223;
-      int j = 1664525 * (p_220345_ ^ -559038737) + 1013904223;
-      return i ^ j;
+   public static int hash(int p_220344_, int p_220345_, int p_220346_) {
+      int i = f_151379_ * p_220344_ + f_151380_;
+      int j = f_151379_ * (p_220345_ ^ f_151381_) + f_151380_;
+      int k = f_151379_ * (p_220346_) + f_151380_;
+      return i ^ j ^ k;
    }
 
    public boolean equals(Object p_45607_) {
@@ -84,88 +_,123 @@
          return false;
       } else {
          ChunkPos chunkpos = (ChunkPos)p_45607_;
-         return this.f_45578_ == chunkpos.f_45578_ && this.f_45579_ == chunkpos.f_45579_;
-      }
-   }
-
-   public int m_151390_() {
-      return this.m_151382_(8);
-   }
-
-   public int m_151393_() {
-      return this.m_151391_(8);
-   }
-
-   public int m_45604_() {
-      return SectionPos.m_123223_(this.f_45578_);
-   }
-
-   public int m_45605_() {
-      return SectionPos.m_123223_(this.f_45579_);
-   }
-
-   public int m_45608_() {
-      return this.m_151382_(15);
-   }
-
-   public int m_45609_() {
-      return this.m_151391_(15);
-   }
-
-   public int m_45610_() {
+         return this.f_45578_ == chunkpos.f_45578_ && this.y == chunkpos.y && this.f_45579_ == chunkpos.f_45579_;
+      }
+   }
+
+   @Override
+   public int compareTo(ChunkPos o) {
+      if (this.y == o.y) {
+         return (int) (this.f_45579_ == o.f_45579_ ? this.f_45578_ - o.f_45578_ : this.f_45579_ - o.f_45579_);
+      } else {
+         return (int) (this.y - o.y);
+      }
+   }
+
+   public long getMiddleBlockX() {
+      return this.getBlockX(8);
+   }
+
+   public long getMiddleBlockY() {
+      return this.getBlockY(8);
+   }
+
+   public long getMiddleBlockZ() {
+      return this.getBlockZ(8);
+   }
+
+   public long getMinBlockX() {
+      return SectionPos.sectionToBlockCoord(this.f_45578_);
+   }
+
+   public long getMinBlockY() {
+      return SectionPos.sectionToBlockCoord(this.y);
+   }
+
+   public long getMinBlockZ() {
+      return SectionPos.sectionToBlockCoord(this.f_45579_);
+   }
+
+   public long getMaxBlockX() {
+      return this.getBlockX(15);
+   }
+
+   public long getMaxBlockY() {
+      return this.getBlockY(15);
+   }
+
+   public long getMaxBlockZ() {
+      return this.getBlockZ(15);
+   }
+
+   public long getRegionX() {
       return this.f_45578_ >> 5;
    }
 
-   public int m_45612_() {
+   public long getRegionY() {
+      return this.y >> f_151377_;
+   }
+
+   public long getRegionZ() {
       return this.f_45579_ >> 5;
    }
 
-   public int m_45613_() {
+   public long getRegionLocalX() {
       return this.f_45578_ & 31;
    }
 
-   public int m_45614_() {
+   public long getRegionLocalY() {
+      return this.y & f_151378_;
+   }
+
+   public long getRegionLocalZ() {
       return this.f_45579_ & 31;
    }
 
-   public BlockPos m_151384_(int p_151385_, int p_151386_, int p_151387_) {
-      return new BlockPos(this.m_151382_(p_151385_), p_151386_, this.m_151391_(p_151387_));
-   }
-
-   public int m_151382_(int p_151383_) {
-      return SectionPos.m_175554_(this.f_45578_, p_151383_);
-   }
-
-   public int m_151391_(int p_151392_) {
-      return SectionPos.m_175554_(this.f_45579_, p_151392_);
-   }
-
-   public BlockPos m_151394_(int p_151395_) {
-      return new BlockPos(this.m_151390_(), p_151395_, this.m_151393_());
+   public BlockPos getBlockAt(long p_151385_, long p_151386_, long p_151387_) {
+      return new BlockPos(this.getBlockX(p_151385_), p_151386_, this.getBlockZ(p_151387_));
+   }
+
+   public long getBlockX(long p_151383_) {
+      return SectionPos.sectionToBlockCoord(this.f_45578_, p_151383_);
+   }
+
+   public long getBlockY(long mask) {
+      return SectionPos.sectionToBlockCoord(this.y, mask);
+   }
+
+   public long getBlockZ(long p_151392_) {
+      return SectionPos.sectionToBlockCoord(this.f_45579_, p_151392_);
+   }
+
+   public BlockPos getMiddleBlockPosition(long p_151395_) {
+      return new BlockPos(this.getMiddleBlockX(), p_151395_, this.getMiddleBlockZ());
    }
 
    public String toString() {
-      return "[" + this.f_45578_ + ", " + this.f_45579_ + "]";
+      return "[" + this.f_45578_ + ", " + this.y + ", " + this.f_45579_ + "]";
    }
 
    public BlockPos m_45615_() {
-      return new BlockPos(this.m_45604_(), 0, this.m_45605_());
+      return new BlockPos(this.getMinBlockX(), this.getMinBlockY(), this.getMinBlockZ());
    }
 
-   public int m_45594_(ChunkPos p_45595_) {
+   public long getChessboardDistance(ChunkPos p_45595_) {
       return Math.max(Math.abs(this.f_45578_ - p_45595_.f_45578_), Math.abs(this.f_45579_ - p_45595_.f_45579_));
    }
 
    public static Stream<ChunkPos> m_45596_(ChunkPos p_45597_, int p_45598_) {
-      return m_45599_(new ChunkPos(p_45597_.f_45578_ - p_45598_, p_45597_.f_45579_ - p_45598_), new ChunkPos(p_45597_.f_45578_ + p_45598_, p_45597_.f_45579_ + p_45598_));
+      return m_45599_(new ChunkPos(p_45597_.f_45578_ - p_45598_, p_45597_.y - p_45598_, p_45597_.f_45579_ - p_45598_), new ChunkPos(p_45597_.f_45578_ + p_45598_, p_45597_.y + p_45598_, p_45597_.f_45579_ + p_45598_));
    }
 
    public static Stream<ChunkPos> m_45599_(final ChunkPos p_45600_, final ChunkPos p_45601_) {
-      int i = Math.abs(p_45600_.f_45578_ - p_45601_.f_45578_) + 1;
-      int j = Math.abs(p_45600_.f_45579_ - p_45601_.f_45579_) + 1;
+      long i = Math.abs(p_45600_.f_45578_ - p_45601_.f_45578_) + 1;
+      long y = Math.abs(p_45600_.y - p_45601_.y) + 1;
+      long j = Math.abs(p_45600_.f_45579_ - p_45601_.f_45579_) + 1;
       final int k = p_45600_.f_45578_ < p_45601_.f_45578_ ? 1 : -1;
+      final int yOffset = p_45600_.y < p_45601_.y ? 1 : -1;
       final int l = p_45600_.f_45579_ < p_45601_.f_45579_ ? 1 : -1;
-      return StreamSupport.stream(new Spliterators.AbstractSpliterator<ChunkPos>((long)(i * j), 64) {
+      return StreamSupport.stream(new Spliterators.AbstractSpliterator<ChunkPos>((long)(i * y * j), 64) {
          @Nullable
          private ChunkPos f_45621_;
 
@@ -173,16 +_,22 @@
             if (this.f_45621_ == null) {
                this.f_45621_ = p_45600_;
             } else {
-               int i1 = this.f_45621_.f_45578_;
-               int j1 = this.f_45621_.f_45579_;
+               long i1 = this.f_45621_.f_45578_;
+               long chunkY = this.f_45621_.y;
+               long j1 = this.f_45621_.f_45579_;
                if (i1 == p_45601_.f_45578_) {
+                  if (chunkY == p_45601_.y) {
                   if (j1 == p_45601_.f_45579_) {
                      return false;
                   }
-
-                  this.f_45621_ = new ChunkPos(p_45600_.f_45578_, j1 + l);
+                     this.f_45621_ = new ChunkPos(p_45600_.f_45578_, p_45600_.y, j1 + l);
+                  } else {
+                     this.f_45621_ = new ChunkPos(p_45600_.f_45578_, chunkY + yOffset, j1);
+                  }
+
+
                } else {
-                  this.f_45621_ = new ChunkPos(i1 + k, j1);
+                  this.f_45621_ = new ChunkPos(i1 + k, chunkY, j1);
                }
             }
 
