--- a/net/minecraft/world/level/CollisionGetter.java
+++ b/net/minecraft/world/level/CollisionGetter.java
@@ -20,7 +_,7 @@
    WorldBorder m_6857_();
 
    @Nullable
-   BlockGetter m_7925_(int p_45774_, int p_45775_);
+   BlockGetter getChunkForCollisions(long p_45774_, long p_45775_);
 
    default boolean m_5450_(@Nullable Entity p_45750_, VoxelShape p_45751_) {
       return true;
@@ -28,7 +_,7 @@
 
    default boolean m_45752_(BlockState p_45753_, BlockPos p_45754_, CollisionContext p_45755_) {
       VoxelShape voxelshape = p_45753_.m_60742_(this, p_45754_, p_45755_);
-      return voxelshape.m_83281_() || this.m_5450_((Entity)null, voxelshape.m_83216_((double)p_45754_.m_123341_(), (double)p_45754_.m_123342_(), (double)p_45754_.m_123343_()));
+      return voxelshape.m_83281_() || this.m_5450_((Entity)null, voxelshape.m_83216_((double)p_45754_.getX(), (double)p_45754_.getY(), (double)p_45754_.getZ()));
    }
 
    default boolean m_45784_(Entity p_45785_) {
