--- a/net/minecraft/world/level/Explosion.java
+++ b/net/minecraft/world/level/Explosion.java
@@ -89,9 +_,9 @@
 
    public static float m_46064_(Vec3 p_46065_, Entity p_46066_) {
       AABB aabb = p_46066_.m_20191_();
-      double d0 = 1.0D / ((aabb.f_82291_ - aabb.f_82288_) * 2.0D + 1.0D);
-      double d1 = 1.0D / ((aabb.f_82292_ - aabb.f_82289_) * 2.0D + 1.0D);
-      double d2 = 1.0D / ((aabb.f_82293_ - aabb.f_82290_) * 2.0D + 1.0D);
+      double d0 = 1.0D / ((aabb.f_82291_.doubleValue() - aabb.f_82288_.doubleValue()) * 2.0D + 1.0D);
+      double d1 = 1.0D / ((aabb.f_82292_.doubleValue() - aabb.f_82289_.doubleValue()) * 2.0D + 1.0D);
+      double d2 = 1.0D / ((aabb.f_82293_.doubleValue() - aabb.f_82290_.doubleValue()) * 2.0D + 1.0D);
       double d3 = (1.0D - Math.floor(1.0D / d0) * d0) / 2.0D;
       double d4 = (1.0D - Math.floor(1.0D / d2) * d2) / 2.0D;
       if (!(d0 < 0.0D) && !(d1 < 0.0D) && !(d2 < 0.0D)) {
@@ -101,9 +_,9 @@
          for(double d5 = 0.0D; d5 <= 1.0D; d5 += d0) {
             for(double d6 = 0.0D; d6 <= 1.0D; d6 += d1) {
                for(double d7 = 0.0D; d7 <= 1.0D; d7 += d2) {
-                  double d8 = Mth.m_14139_(d5, aabb.f_82288_, aabb.f_82291_);
-                  double d9 = Mth.m_14139_(d6, aabb.f_82289_, aabb.f_82292_);
-                  double d10 = Mth.m_14139_(d7, aabb.f_82290_, aabb.f_82293_);
+                  double d8 = Mth.m_14139_(d5, aabb.f_82288_.doubleValue(), aabb.f_82291_.doubleValue());
+                  double d9 = Mth.m_14139_(d6, aabb.f_82289_.doubleValue(), aabb.f_82292_.doubleValue());
+                  double d10 = Mth.m_14139_(d7, aabb.f_82290_.doubleValue(), aabb.f_82293_.doubleValue());
                   Vec3 vec3 = new Vec3(d8 + d3, d9, d10 + d4);
                   if (p_46066_.m_9236_().m_45547_(new ClipContext(vec3, p_46065_, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, p_46066_)).m_6662_() == HitResult.Type.MISS) {
                      ++i;
@@ -169,12 +_,12 @@
 
       this.f_46020_.addAll(set);
       float f2 = this.f_46017_ * 2.0F;
-      int k1 = Mth.m_14107_(this.f_46013_ - (double)f2 - 1.0D);
-      int l1 = Mth.m_14107_(this.f_46013_ + (double)f2 + 1.0D);
-      int i2 = Mth.m_14107_(this.f_46014_ - (double)f2 - 1.0D);
-      int i1 = Mth.m_14107_(this.f_46014_ + (double)f2 + 1.0D);
-      int j2 = Mth.m_14107_(this.f_46015_ - (double)f2 - 1.0D);
-      int j1 = Mth.m_14107_(this.f_46015_ + (double)f2 + 1.0D);
+      long k1 = Mth.m_14134_(this.f_46013_ - (double)f2 - 1.0D);
+      long l1 = Mth.m_14134_(this.f_46013_ + (double)f2 + 1.0D);
+      long i2 = Mth.m_14134_(this.f_46014_ - (double)f2 - 1.0D);
+      long i1 = Mth.m_14134_(this.f_46014_ + (double)f2 + 1.0D);
+      long j2 = Mth.m_14134_(this.f_46015_ - (double)f2 - 1.0D);
+      long j1 = Mth.m_14134_(this.f_46015_ + (double)f2 + 1.0D);
       List<Entity> list = this.f_46012_.m_45933_(this.f_46016_, new AABB((double)k1, (double)i2, (double)j2, (double)l1, (double)i1, (double)j1));
       Vec3 vec3 = new Vec3(this.f_46013_, this.f_46014_, this.f_46015_);
 
@@ -250,7 +_,7 @@
                   if ($$9 instanceof ServerLevel) {
                      ServerLevel serverlevel = (ServerLevel)$$9;
                      BlockEntity blockentity = blockstate.m_155947_() ? this.f_46012_.m_7702_(blockpos) : null;
-                     LootParams.Builder lootparams$builder = (new LootParams.Builder(serverlevel)).m_287286_(LootContextParams.f_81460_, Vec3.m_82512_(blockpos)).m_287286_(LootContextParams.f_81463_, ItemStack.f_41583_).m_287289_(LootContextParams.f_81462_, blockentity).m_287289_(LootContextParams.f_81455_, this.f_46016_);
+                     LootParams.Builder lootparams$builder = (new LootParams.Builder(serverlevel)).m_287286_(LootContextParams.f_81460_, Vec3.atCenterOf(blockpos)).m_287286_(LootContextParams.f_81463_, ItemStack.f_41583_).m_287289_(LootContextParams.f_81462_, blockentity).m_287289_(LootContextParams.f_81455_, this.f_46016_);
                      if (this.f_46010_ == Explosion.BlockInteraction.DESTROY_WITH_DECAY) {
                         lootparams$builder.m_287286_(LootContextParams.f_81464_, this.f_46017_);
                      }
@@ -275,7 +_,7 @@
 
       if (this.f_46009_) {
          for(BlockPos blockpos2 : this.f_46020_) {
-            if (this.f_46011_.m_188503_(3) == 0 && this.f_46012_.m_8055_(blockpos2).m_60795_() && this.f_46012_.m_8055_(blockpos2.m_7495_()).m_60804_(this.f_46012_, blockpos2.m_7495_())) {
+            if (this.f_46011_.m_188503_(3) == 0 && this.f_46012_.m_8055_(blockpos2).m_60795_() && this.f_46012_.m_8055_(blockpos2.below()).m_60804_(this.f_46012_, blockpos2.below())) {
                this.f_46012_.m_46597_(blockpos2, BaseFireBlock.m_49245_(this.f_46012_, blockpos2));
             }
          }
