--- a/net/minecraft/world/level/Level.java
+++ b/net/minecraft/world/level/Level.java
@@ -153,33 +_,33 @@
    }
 
    public boolean m_46739_(BlockPos p_46740_) {
-      return !this.m_151570_(p_46740_) && m_46457_(p_46740_);
+      return true;//!this.isOutsideBuildHeight(pPos) && isInWorldBoundsHorizontal(pPos);
    }
 
    public static boolean m_46741_(BlockPos p_46742_) {
-      return !m_46724_(p_46742_.m_123342_()) && m_46457_(p_46742_);
+      return true;//isOutsideSpawnableHeight(pPos.getY()) && isInWorldBoundsHorizontal(pPos);
    }
 
    private static boolean m_46457_(BlockPos p_46458_) {
-      return p_46458_.m_123341_() >= -30000000 && p_46458_.m_123343_() >= -30000000 && p_46458_.m_123341_() < 30000000 && p_46458_.m_123343_() < 30000000;
+      return true;//pPos.getX() >= -30000000 && pPos.getZ() >= -30000000 && pPos.getX() < 30000000 && pPos.getZ() < 30000000;
    }
 
-   private static boolean m_46724_(int p_46725_) {
-      return p_46725_ < -20000000 || p_46725_ >= 20000000;
+   private static boolean isOutsideSpawnableHeight(long pY) {
+      return pY < -20000000 || pY >= 20000000;
    }
 
    public LevelChunk m_46745_(BlockPos p_46746_) {
-      return this.m_6325_(SectionPos.m_123171_(p_46746_.m_123341_()), SectionPos.m_123171_(p_46746_.m_123343_()));
+      return this.getChunk(SectionPos.blockToSectionCoord(p_46746_.getBigX()), SectionPos.blockToSectionCoord(p_46746_.getBigZ()));
    }
 
-   public LevelChunk m_6325_(int p_46727_, int p_46728_) {
-      return (LevelChunk)this.m_46819_(p_46727_, p_46728_, ChunkStatus.f_62326_);
+   public LevelChunk getChunk(me.alphamode.mclong.math.BigInteger pChunkX, me.alphamode.mclong.math.BigInteger pChunkZ) {
+      return (LevelChunk)this.getChunk(pChunkX, pChunkZ, ChunkStatus.f_62326_);
    }
 
    @Nullable
-   public ChunkAccess m_6522_(int p_46502_, int p_46503_, ChunkStatus p_46504_, boolean p_46505_) {
-      ChunkAccess chunkaccess = this.m_7726_().m_7587_(p_46502_, p_46503_, p_46504_, p_46505_);
-      if (chunkaccess == null && p_46505_) {
+   public ChunkAccess getChunk(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ, ChunkStatus pRequiredStatus, boolean pNonnull) {
+      ChunkAccess chunkaccess = this.m_7726_().getChunk(pX, pZ, pRequiredStatus, pNonnull);
+      if (chunkaccess == null && pNonnull) {
          throw new IllegalStateException("Should always be able to create a chunk!");
       } else {
          return chunkaccess;
@@ -294,17 +_,13 @@
       this.f_220350_.m_213547_(p_220385_, p_220386_, p_220387_, p_220388_, p_220389_, p_220390_);
    }
 
-   public int m_6924_(Heightmap.Types p_46571_, int p_46572_, int p_46573_) {
-      int i;
-      if (p_46572_ >= -30000000 && p_46573_ >= -30000000 && p_46572_ < 30000000 && p_46573_ < 30000000) {
-         if (this.m_7232_(SectionPos.m_123171_(p_46572_), SectionPos.m_123171_(p_46573_))) {
-            i = this.m_6325_(SectionPos.m_123171_(p_46572_), SectionPos.m_123171_(p_46573_)).m_5885_(p_46571_, p_46572_ & 15, p_46573_ & 15) + 1;
+   public long getHeight(Heightmap.Types pHeightmapType, me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pZ) {
+      long i;
+         if (this.hasChunk(SectionPos.blockToSectionCoord(pX), SectionPos.blockToSectionCoord(pZ))) {
+            i = this.getChunk(SectionPos.blockToSectionCoord(pX), SectionPos.blockToSectionCoord(pZ)).getHeight(pHeightmapType, pX.and(me.alphamode.mclong.core.BigConstants.Ints.FIFTEEN), pZ.and(me.alphamode.mclong.core.BigConstants.Ints.FIFTEEN)) + 1;
          } else {
-            i = this.m_141937_();
+            i = this.getMinBuildHeight();
          }
-      } else {
-         i = this.m_5736_() + 1;
-      }
 
       return i;
    }
@@ -317,7 +_,7 @@
       if (this.m_151570_(p_46732_)) {
          return Blocks.f_50626_.m_49966_();
       } else {
-         LevelChunk levelchunk = this.m_6325_(SectionPos.m_123171_(p_46732_.m_123341_()), SectionPos.m_123171_(p_46732_.m_123343_()));
+         LevelChunk levelchunk = this.getChunk(SectionPos.blockToSectionCoord(p_46732_.getBigX()), SectionPos.blockToSectionCoord(p_46732_.getBigZ()));
          return levelchunk.m_8055_(p_46732_);
       }
    }
@@ -351,19 +_,19 @@
    }
 
    public void m_5594_(@Nullable Player p_46560_, BlockPos p_46561_, SoundEvent p_46562_, SoundSource p_46563_, float p_46564_, float p_46565_) {
-      this.m_6263_(p_46560_, (double)p_46561_.m_123341_() + 0.5D, (double)p_46561_.m_123342_() + 0.5D, (double)p_46561_.m_123343_() + 0.5D, p_46562_, p_46563_, p_46564_, p_46565_);
+      this.playSound(p_46560_, p_46561_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_46561_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_46561_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), p_46562_, p_46563_, p_46564_, p_46565_);
    }
 
-   public abstract void m_262808_(@Nullable Player p_262953_, double p_263004_, double p_263398_, double p_263376_, Holder<SoundEvent> p_263359_, SoundSource p_263020_, float p_263055_, float p_262914_, long p_262991_);
+   public abstract void playSeededSound(@Nullable Player pPlayer, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, Holder<SoundEvent> pSound, SoundSource pSource, float pVolume, float pPitch, long pSeed);
 
-   public void m_214150_(@Nullable Player p_220363_, double p_220364_, double p_220365_, double p_220366_, SoundEvent p_220367_, SoundSource p_220368_, float p_220369_, float p_220370_, long p_220371_) {
-      this.m_262808_(p_220363_, p_220364_, p_220365_, p_220366_, BuiltInRegistries.f_256894_.m_263177_(p_220367_), p_220368_, p_220369_, p_220370_, p_220371_);
+   public void playSeededSound(@Nullable Player pPlayer, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, SoundEvent pSound, SoundSource pSource, float pVolume, float pPitch, long pSeed) {
+      this.playSeededSound(pPlayer, pX, pY, pZ, BuiltInRegistries.f_256894_.m_263177_(pSound), pSource, pVolume, pPitch, pSeed);
    }
 
    public abstract void m_213890_(@Nullable Player p_220372_, Entity p_220373_, Holder<SoundEvent> p_263500_, SoundSource p_220375_, float p_220376_, float p_220377_, long p_220378_);
 
-   public void m_6263_(@Nullable Player p_46543_, double p_46544_, double p_46545_, double p_46546_, SoundEvent p_46547_, SoundSource p_46548_, float p_46549_, float p_46550_) {
-      this.m_214150_(p_46543_, p_46544_, p_46545_, p_46546_, p_46547_, p_46548_, p_46549_, p_46550_, this.f_220348_.m_188505_());
+   public void playSound(@Nullable Player pPlayer, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, SoundEvent pSound, SoundSource pCategory, float pVolume, float pPitch) {
+      this.playSeededSound(pPlayer, pX, pY, pZ, pSound, pCategory, pVolume, pPitch, this.f_220348_.m_188505_());
    }
 
    public void m_6269_(@Nullable Player p_46551_, Entity p_46552_, SoundEvent p_46553_, SoundSource p_46554_, float p_46555_, float p_46556_) {
@@ -371,22 +_,22 @@
    }
 
    public void m_245747_(BlockPos p_250938_, SoundEvent p_252209_, SoundSource p_249161_, float p_249980_, float p_250277_, boolean p_250151_) {
-      this.m_7785_((double)p_250938_.m_123341_() + 0.5D, (double)p_250938_.m_123342_() + 0.5D, (double)p_250938_.m_123343_() + 0.5D, p_252209_, p_249161_, p_249980_, p_250277_, p_250151_);
-   }
-
-   public void m_7785_(double p_46482_, double p_46483_, double p_46484_, SoundEvent p_46485_, SoundSource p_46486_, float p_46487_, float p_46488_, boolean p_46489_) {
-   }
-
-   public void m_7106_(ParticleOptions p_46631_, double p_46632_, double p_46633_, double p_46634_, double p_46635_, double p_46636_, double p_46637_) {
-   }
-
-   public void m_6493_(ParticleOptions p_46638_, boolean p_46639_, double p_46640_, double p_46641_, double p_46642_, double p_46643_, double p_46644_, double p_46645_) {
-   }
-
-   public void m_7107_(ParticleOptions p_46684_, double p_46685_, double p_46686_, double p_46687_, double p_46688_, double p_46689_, double p_46690_) {
-   }
-
-   public void m_6485_(ParticleOptions p_46691_, boolean p_46692_, double p_46693_, double p_46694_, double p_46695_, double p_46696_, double p_46697_, double p_46698_) {
+      this.playLocalSound(p_250938_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_250938_.getBigDecY().add(me.alphamode.mclong.core.BigConstants.AABB), p_250938_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), p_252209_, p_249161_, p_249980_, p_250277_, p_250151_);
+   }
+
+   public void playLocalSound(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, SoundEvent pSound, SoundSource pCategory, float pVolume, float pPitch, boolean pDistanceDelay) {
+   }
+
+   public void addParticle(ParticleOptions pParticleData, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+   }
+
+   public void addParticle(ParticleOptions pParticleData, boolean pForceAlwaysRender, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+   }
+
+   public void addAlwaysVisibleParticle(ParticleOptions pParticleData, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
+   }
+
+   public void addAlwaysVisibleParticle(ParticleOptions pParticleData, boolean pIgnoreRange, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pXSpeed, double pYSpeed, double pZSpeed) {
    }
 
    public float m_46490_(float p_46491_) {
@@ -437,33 +_,37 @@
       return true;
    }
 
-   public boolean m_183438_(long p_186456_) {
+   public boolean shouldTickBlocksAt(ChunkPos pChunkPos) {
       return true;
    }
 
    public boolean m_220393_(BlockPos p_220394_) {
-      return this.m_183438_(ChunkPos.m_151388_(p_220394_));
+      return this.shouldTickBlocksAt(new ChunkPos(p_220394_));
    }
 
    public Explosion m_254849_(@Nullable Entity p_256599_, double p_255914_, double p_255684_, double p_255843_, float p_256310_, Level.ExplosionInteraction p_256178_) {
-      return this.m_254877_(p_256599_, (DamageSource)null, (ExplosionDamageCalculator)null, p_255914_, p_255684_, p_255843_, p_256310_, false, p_256178_);
-   }
-
-   public Explosion m_255391_(@Nullable Entity p_255682_, double p_255803_, double p_256403_, double p_256538_, float p_255674_, boolean p_256634_, Level.ExplosionInteraction p_256111_) {
-      return this.m_254877_(p_255682_, (DamageSource)null, (ExplosionDamageCalculator)null, p_255803_, p_256403_, p_256538_, p_255674_, p_256634_, p_256111_);
+      return this.explode(p_256599_, (DamageSource)null, (ExplosionDamageCalculator)null, me.alphamode.mclong.math.BigDecimal.valueOf(p_255914_), me.alphamode.mclong.math.BigDecimal.valueOf(p_255684_), me.alphamode.mclong.math.BigDecimal.valueOf(p_255843_), p_256310_, false, p_256178_);
+   }
+
+   public Explosion explode(@Nullable Entity pSource, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pRadius, Level.ExplosionInteraction pExplosionInteraction) {
+      return this.explode(pSource, (DamageSource)null, (ExplosionDamageCalculator)null, pX, pY, pZ, pRadius, false, pExplosionInteraction);
+   }
+
+   public Explosion explode(@Nullable Entity pSource, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pRadius, boolean pFire, Level.ExplosionInteraction pExplosionInteraction) {
+      return this.explode(pSource, (DamageSource)null, (ExplosionDamageCalculator)null, pX, pY, pZ, pRadius, pFire, pExplosionInteraction);
    }
 
    public Explosion m_254951_(@Nullable Entity p_255653_, @Nullable DamageSource p_256558_, @Nullable ExplosionDamageCalculator p_255929_, Vec3 p_256001_, float p_255963_, boolean p_256099_, Level.ExplosionInteraction p_256371_) {
-      return this.m_254877_(p_255653_, p_256558_, p_255929_, p_256001_.m_7096_(), p_256001_.m_7098_(), p_256001_.m_7094_(), p_255963_, p_256099_, p_256371_);
-   }
-
-   public Explosion m_254877_(@Nullable Entity p_256145_, @Nullable DamageSource p_256004_, @Nullable ExplosionDamageCalculator p_255696_, double p_256208_, double p_256036_, double p_255746_, float p_256647_, boolean p_256098_, Level.ExplosionInteraction p_256104_) {
-      return this.m_255278_(p_256145_, p_256004_, p_255696_, p_256208_, p_256036_, p_255746_, p_256647_, p_256098_, p_256104_, true);
-   }
-
-   public Explosion m_255278_(@Nullable Entity p_256233_, @Nullable DamageSource p_255861_, @Nullable ExplosionDamageCalculator p_255867_, double p_256447_, double p_255732_, double p_255717_, float p_256013_, boolean p_256228_, Level.ExplosionInteraction p_255784_, boolean p_256377_) {
+      return this.explode(p_255653_, p_256558_, p_255929_, p_256001_.getX(), p_256001_.getY(), p_256001_.getZ(), p_255963_, p_256099_, p_256371_);
+   }
+
+   public Explosion explode(@Nullable Entity pSource, @Nullable DamageSource pDamageSource, @Nullable ExplosionDamageCalculator pDamageCalculator, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pRadius, boolean pFire, Level.ExplosionInteraction pExplosionInteraction) {
+      return this.explode(pSource, pDamageSource, pDamageCalculator, pX, pY, pZ, pRadius, pFire, pExplosionInteraction, true);
+   }
+
+   public Explosion explode(@Nullable Entity pSource, @Nullable DamageSource pDamageSource, @Nullable ExplosionDamageCalculator pDamageCalculator, me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, float pRadius, boolean pFire, Level.ExplosionInteraction pExplosionInteraction, boolean pSpawnParticles) {
       Explosion.BlockInteraction explosion$blockinteraction1;
-      switch (p_255784_) {
+      switch (pExplosionInteraction) {
          case NONE:
             explosion$blockinteraction1 = Explosion.BlockInteraction.KEEP;
             break;
@@ -481,9 +_,9 @@
       }
 
       Explosion.BlockInteraction explosion$blockinteraction = explosion$blockinteraction1;
-      Explosion explosion = new Explosion(this, p_256233_, p_255861_, p_255867_, p_256447_, p_255732_, p_255717_, p_256013_, p_256228_, explosion$blockinteraction);
+      Explosion explosion = new Explosion(this, pSource, pDamageSource, pDamageCalculator, pX, pY, pZ, pRadius, pFire, explosion$blockinteraction);
       explosion.m_46061_();
-      explosion.m_46075_(p_256377_);
+      explosion.m_46075_(pSpawnParticles);
       return explosion;
    }
 
@@ -516,14 +_,14 @@
    }
 
    public boolean m_46749_(BlockPos p_46750_) {
-      return this.m_151570_(p_46750_) ? false : this.m_7726_().m_5563_(SectionPos.m_123171_(p_46750_.m_123341_()), SectionPos.m_123171_(p_46750_.m_123343_()));
+      return this.m_151570_(p_46750_) ? false : this.m_7726_().hasChunk(SectionPos.blockToSectionCoord(p_46750_.getBigX()), SectionPos.blockToSectionCoord(p_46750_.getBigZ()));
    }
 
    public boolean m_46578_(BlockPos p_46579_, Entity p_46580_, Direction p_46581_) {
       if (this.m_151570_(p_46579_)) {
          return false;
       } else {
-         ChunkAccess chunkaccess = this.m_6522_(SectionPos.m_123171_(p_46579_.m_123341_()), SectionPos.m_123171_(p_46579_.m_123343_()), ChunkStatus.f_62326_, false);
+         ChunkAccess chunkaccess = this.getChunk(SectionPos.blockToSectionCoord(p_46579_.getBigX()), SectionPos.blockToSectionCoord(p_46579_.getBigZ()), ChunkStatus.f_62326_, false);
          return chunkaccess == null ? false : chunkaccess.m_8055_(p_46579_).m_60638_(this, p_46579_, p_46580_, p_46581_);
       }
    }
@@ -544,7 +_,7 @@
    }
 
    public BlockPos m_220360_() {
-      BlockPos blockpos = new BlockPos(this.f_46442_.m_6789_(), this.f_46442_.m_6527_(), this.f_46442_.m_6526_());
+      BlockPos blockpos = new BlockPos(this.f_46442_.getXSpawn(), this.f_46442_.getYSpawn(), this.f_46442_.getZSpawn());
       if (!this.m_6857_().m_61937_(blockpos)) {
          blockpos = this.m_5452_(Heightmap.Types.MOTION_BLOCKING, BlockPos.m_274561_(this.m_6857_().m_6347_(), 0.0D, this.m_6857_().m_6345_()));
       }
@@ -571,8 +_,8 @@
    }
 
    @Nullable
-   public BlockGetter m_7925_(int p_46711_, int p_46712_) {
-      return this.m_6522_(p_46711_, p_46712_, ChunkStatus.f_62326_, false);
+   public BlockGetter getChunkForCollisions(me.alphamode.mclong.math.BigInteger pChunkX, me.alphamode.mclong.math.BigInteger pChunkZ) {
+      return this.getChunk(pChunkX, pChunkZ, ChunkStatus.f_62326_, false);
    }
 
    public List<Entity> m_6249_(@Nullable Entity p_46536_, AABB p_46537_, Predicate<? super Entity> p_46538_) {
@@ -715,7 +_,7 @@
          return false;
       } else if (!this.m_45527_(p_46759_)) {
          return false;
-      } else if (this.m_5452_(Heightmap.Types.MOTION_BLOCKING, p_46759_).m_123342_() > p_46759_.m_123342_()) {
+      } else if (this.m_5452_(Heightmap.Types.MOTION_BLOCKING, p_46759_).getY() > p_46759_.getY()) {
          return false;
       } else {
          Biome biome = this.m_204166_(p_46759_).m_203334_();
@@ -754,20 +_,20 @@
 
    public abstract void m_6801_(int p_46506_, BlockPos p_46507_, int p_46508_);
 
-   public void m_7228_(double p_46475_, double p_46476_, double p_46477_, double p_46478_, double p_46479_, double p_46480_, @Nullable CompoundTag p_46481_) {
+   public void createFireworks(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ, double pMotionX, double pMotionY, double pMotionZ, @Nullable CompoundTag pCompound) {
    }
 
    public abstract Scoreboard m_6188_();
 
    public void m_46717_(BlockPos p_46718_, Block p_46719_) {
       for(Direction direction : Direction.Plane.HORIZONTAL) {
-         BlockPos blockpos = p_46718_.m_121945_(direction);
+         BlockPos blockpos = p_46718_.relative(direction);
          if (this.m_46805_(blockpos)) {
             BlockState blockstate = this.m_8055_(blockpos);
             if (blockstate.m_60713_(Blocks.f_50328_)) {
                this.m_213960_(blockstate, blockpos, p_46719_, p_46718_, false);
             } else if (blockstate.m_60796_(this, blockpos)) {
-               blockpos = blockpos.m_121945_(direction);
+               blockpos = blockpos.relative(direction);
                blockstate = this.m_8055_(blockpos);
                if (blockstate.m_60713_(Blocks.f_50328_)) {
                   this.m_213960_(blockstate, blockpos, p_46719_, p_46718_, false);
@@ -834,10 +_,10 @@
 
    public abstract RecipeManager m_7465_();
 
-   public BlockPos m_46496_(int p_46497_, int p_46498_, int p_46499_, int p_46500_) {
+   public BlockPos getBlockRandomPos(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ, int pYMask) {
       this.f_46435_ = this.f_46435_ * 3 + 1013904223;
       int i = this.f_46435_ >> 2;
-      return new BlockPos(p_46497_ + (i & 15), p_46498_ + (i >> 16 & p_46500_), p_46499_ + (i >> 8 & 15));
+      return new BlockPos(pX.add((i & 15)), pY.add((i >> 16 & pYMask)), pZ.add((i >> 8 & 15)));
    }
 
    public boolean m_7441_() {
