--- a/net/minecraft/world/level/NaturalSpawner.java
+++ b/net/minecraft/world/level/NaturalSpawner.java
@@ -77,7 +_,7 @@
          MobCategory mobcategory = entity.m_6095_().m_20674_();
          if (mobcategory != MobCategory.MISC) {
             BlockPos blockpos = entity.m_20183_();
-            p_186527_.m_47103_(ChunkPos.m_151388_(blockpos), (p_275163_) -> {
+            p_186527_.query(new ChunkPos(blockpos), (p_275163_) -> {
                MobSpawnSettings.MobSpawnCost mobspawnsettings$mobspawncost = m_47095_(blockpos, p_275163_).m_47518_().m_48345_(entity.m_6095_());
                if (mobspawnsettings$mobspawncost != null) {
                   potentialcalculator.m_47192_(entity.m_20183_(), mobspawnsettings$mobspawncost.f_48386_());
@@ -96,7 +_,7 @@
    }
 
    static Biome m_47095_(BlockPos p_47096_, ChunkAccess p_47097_) {
-      return p_47097_.m_203495_(QuartPos.m_175400_(p_47096_.m_123341_()), QuartPos.m_175400_(p_47096_.m_123342_()), QuartPos.m_175400_(p_47096_.m_123343_())).m_203334_();
+      return p_47097_.getNoiseBiome(QuartPos.fromBlock(p_47096_.getX()), QuartPos.fromBlock(p_47096_.getY()), QuartPos.fromBlock(p_47096_.getZ())).m_203334_();
    }
 
    public static void m_47029_(ServerLevel p_47030_, LevelChunk p_47031_, NaturalSpawner.SpawnState p_47032_, boolean p_47033_, boolean p_47034_, boolean p_47035_) {
@@ -113,7 +_,7 @@
 
    public static void m_47045_(MobCategory p_47046_, ServerLevel p_47047_, LevelChunk p_47048_, NaturalSpawner.SpawnPredicate p_47049_, NaturalSpawner.AfterSpawnCallback p_47050_) {
       BlockPos blockpos = m_47062_(p_47047_, p_47048_);
-      if (blockpos.m_123342_() >= p_47047_.m_141937_() + 1) {
+      if (blockpos.getY() >= p_47047_.getMinBuildHeight() + 1) {
          m_47038_(p_47046_, p_47047_, p_47048_, blockpos, p_47049_, p_47050_);
       }
    }
@@ -129,15 +_,15 @@
    public static void m_47038_(MobCategory p_47039_, ServerLevel p_47040_, ChunkAccess p_47041_, BlockPos p_47042_, NaturalSpawner.SpawnPredicate p_47043_, NaturalSpawner.AfterSpawnCallback p_47044_) {
       StructureManager structuremanager = p_47040_.m_215010_();
       ChunkGenerator chunkgenerator = p_47040_.m_7726_().m_8481_();
-      int i = p_47042_.m_123342_();
+      long i = p_47042_.getY();
       BlockState blockstate = p_47041_.m_8055_(p_47042_);
       if (!blockstate.m_60796_(p_47041_, p_47042_)) {
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
          int j = 0;
 
          for(int k = 0; k < 3; ++k) {
-            int l = p_47042_.m_123341_();
-            int i1 = p_47042_.m_123343_();
+            long l = p_47042_.getX();
+            long i1 = p_47042_.getZ();
             int j1 = 6;
             MobSpawnSettings.SpawnerData mobspawnsettings$spawnerdata = null;
             SpawnGroupData spawngroupdata = null;
@@ -147,7 +_,7 @@
             for(int i2 = 0; i2 < k1; ++i2) {
                l += p_47040_.f_46441_.m_188503_(6) - p_47040_.f_46441_.m_188503_(6);
                i1 += p_47040_.f_46441_.m_188503_(6) - p_47040_.f_46441_.m_188503_(6);
-               blockpos$mutableblockpos.m_122178_(l, i, i1);
+               blockpos$mutableblockpos.set(l, i, i1);
                double d0 = (double)l + 0.5D;
                double d1 = (double)i1 + 0.5D;
                Player player = p_47040_.m_45924_(d0, (double)i, d1, -1.0D, false);
@@ -197,7 +_,7 @@
    private static boolean m_47024_(ServerLevel p_47025_, ChunkAccess p_47026_, BlockPos.MutableBlockPos p_47027_, double p_47028_) {
       if (p_47028_ <= 576.0D) {
          return false;
-      } else if (p_47025_.m_220360_().m_203195_(new Vec3((double)p_47027_.m_123341_() + 0.5D, (double)p_47027_.m_123342_(), (double)p_47027_.m_123343_() + 0.5D), 24.0D)) {
+      } else if (p_47025_.m_220360_().closerToCenterThan(new Vec3((double)p_47027_.getX() + 0.5D, (double)p_47027_.getY(), (double)p_47027_.getZ() + 0.5D), 24.0D)) {
          return false;
       } else {
          return Objects.equals(new ChunkPos(p_47027_), p_47026_.m_7697_()) || p_47025_.m_201918_(p_47027_);
@@ -217,7 +_,7 @@
          } else if (!SpawnPlacements.m_217074_(entitytype, p_220422_, MobSpawnType.NATURAL, p_220427_, p_220422_.f_46441_)) {
             return false;
          } else {
-            return p_220422_.m_45772_(entitytype.m_20585_((double)p_220427_.m_123341_() + 0.5D, (double)p_220427_.m_123342_(), (double)p_220427_.m_123343_() + 0.5D));
+            return p_220422_.m_45772_(entitytype.m_20585_((double)p_220427_.getX() + 0.5D, (double)p_220427_.getY(), (double)p_220427_.getZ() + 0.5D));
          }
       } else {
          return false;
@@ -262,7 +_,7 @@
    }
 
    public static boolean m_220455_(BlockPos p_220456_, ServerLevel p_220457_, MobCategory p_220458_, StructureManager p_220459_) {
-      if (p_220458_ == MobCategory.MONSTER && p_220457_.m_8055_(p_220456_.m_7495_()).m_60713_(Blocks.f_50197_)) {
+      if (p_220458_ == MobCategory.MONSTER && p_220457_.m_8055_(p_220456_.below()).m_60713_(Blocks.f_50197_)) {
          Structure structure = p_220459_.m_220521_().m_175515_(Registries.f_256944_).m_6246_(BuiltinStructures.f_209859_);
          return structure == null ? false : p_220459_.m_220494_(p_220456_, structure).m_73603_();
       } else {
@@ -272,10 +_,10 @@
 
    private static BlockPos m_47062_(Level p_47063_, LevelChunk p_47064_) {
       ChunkPos chunkpos = p_47064_.m_7697_();
-      int i = chunkpos.m_45604_() + p_47063_.f_46441_.m_188503_(16);
-      int j = chunkpos.m_45605_() + p_47063_.f_46441_.m_188503_(16);
-      int k = p_47064_.m_5885_(Heightmap.Types.WORLD_SURFACE, i, j) + 1;
-      int l = Mth.m_216287_(p_47063_.f_46441_, p_47063_.m_141937_(), k);
+      long i = chunkpos.getMinBlockX() + p_47063_.f_46441_.m_188503_(16);
+      long j = chunkpos.getMinBlockZ() + p_47063_.f_46441_.m_188503_(16);
+      long k = p_47064_.getHeight(Heightmap.Types.WORLD_SURFACE, i, j) + 1;
+      long l = Mth.randomBetweenInclusive(p_47063_.f_46441_, p_47063_.getMinBuildHeight(), k);
       return new BlockPos(i, l, j);
    }
 
@@ -299,8 +_,8 @@
       } else if (p_47055_ != null && p_47053_.m_6857_().m_61937_(p_47054_)) {
          BlockState blockstate = p_47053_.m_8055_(p_47054_);
          FluidState fluidstate = p_47053_.m_6425_(p_47054_);
-         BlockPos blockpos = p_47054_.m_7494_();
-         BlockPos blockpos1 = p_47054_.m_7495_();
+         BlockPos blockpos = p_47054_.above();
+         BlockPos blockpos1 = p_47054_.below();
          switch (p_47052_) {
             case IN_WATER:
                return fluidstate.m_205070_(FluidTags.f_13131_) && !p_47053_.m_8055_(blockpos).m_60796_(p_47053_, blockpos);
@@ -324,8 +_,9 @@
       MobSpawnSettings mobspawnsettings = p_220452_.m_203334_().m_47518_();
       WeightedRandomList<MobSpawnSettings.SpawnerData> weightedrandomlist = mobspawnsettings.m_151798_(MobCategory.CREATURE);
       if (!weightedrandomlist.m_146337_()) {
-         int i = p_220453_.m_45604_();
-         int j = p_220453_.m_45605_();
+         long i = p_220453_.getMinBlockX();
+         long minY = p_220453_.getMinBlockY();
+         long j = p_220453_.getMinBlockZ();
 
          while(p_220454_.m_188501_() < mobspawnsettings.m_48344_()) {
             Optional<MobSpawnSettings.SpawnerData> optional = weightedrandomlist.m_216829_(p_220454_);
@@ -333,21 +_,22 @@
                MobSpawnSettings.SpawnerData mobspawnsettings$spawnerdata = optional.get();
                int k = mobspawnsettings$spawnerdata.f_48405_ + p_220454_.m_188503_(1 + mobspawnsettings$spawnerdata.f_48406_ - mobspawnsettings$spawnerdata.f_48405_);
                SpawnGroupData spawngroupdata = null;
-               int l = i + p_220454_.m_188503_(16);
-               int i1 = j + p_220454_.m_188503_(16);
-               int j1 = l;
-               int k1 = i1;
+               long l = i + p_220454_.m_188503_(16);
+               long y = minY + p_220454_.m_188503_(16);
+               long i1 = j + p_220454_.m_188503_(16);
+               long j1 = l;
+               long k1 = i1;
 
                for(int l1 = 0; l1 < k; ++l1) {
                   boolean flag = false;
 
                   for(int i2 = 0; !flag && i2 < 4; ++i2) {
-                     BlockPos blockpos = m_47065_(p_220451_, mobspawnsettings$spawnerdata.f_48404_, l, i1);
+                     BlockPos blockpos = getTopNonCollidingPos(p_220451_, mobspawnsettings$spawnerdata.f_48404_, l, y, i1);
                      if (mobspawnsettings$spawnerdata.f_48404_.m_20654_() && m_47051_(SpawnPlacements.m_21752_(mobspawnsettings$spawnerdata.f_48404_), p_220451_, blockpos, mobspawnsettings$spawnerdata.f_48404_)) {
                         float f = mobspawnsettings$spawnerdata.f_48404_.m_20678_();
                         double d0 = Mth.m_14008_((double)l, (double)i + (double)f, (double)i + 16.0D - (double)f);
                         double d1 = Mth.m_14008_((double)i1, (double)j + (double)f, (double)j + 16.0D - (double)f);
-                        if (!p_220451_.m_45772_(mobspawnsettings$spawnerdata.f_48404_.m_20585_(d0, (double)blockpos.m_123342_(), d1)) || !SpawnPlacements.m_217074_(mobspawnsettings$spawnerdata.f_48404_, p_220451_, MobSpawnType.CHUNK_GENERATION, BlockPos.m_274561_(d0, (double)blockpos.m_123342_(), d1), p_220451_.m_213780_())) {
+                        if (!p_220451_.m_45772_(mobspawnsettings$spawnerdata.f_48404_.m_20585_(d0, (double)blockpos.getY(), d1)) || !SpawnPlacements.m_217074_(mobspawnsettings$spawnerdata.f_48404_, p_220451_, MobSpawnType.CHUNK_GENERATION, BlockPos.m_274561_(d0, (double)blockpos.getY(), d1), p_220451_.m_213780_())) {
                            continue;
                         }
 
@@ -363,7 +_,7 @@
                            continue;
                         }
 
-                        entity.m_7678_(d0, (double)blockpos.m_123342_(), d1, p_220454_.m_188501_() * 360.0F, 0.0F);
+                        entity.m_7678_(d0, (double)blockpos.getY(), d1, p_220454_.m_188501_() * 360.0F, 0.0F);
                         if (entity instanceof Mob) {
                            Mob mob = (Mob)entity;
                            if (mob.m_5545_(p_220451_, MobSpawnType.CHUNK_GENERATION) && mob.m_6914_(p_220451_)) {
@@ -387,8 +_,8 @@
       }
    }
 
-   private static BlockPos m_47065_(LevelReader p_47066_, EntityType<?> p_47067_, int p_47068_, int p_47069_) {
-      int i = p_47066_.m_6924_(SpawnPlacements.m_21765_(p_47067_), p_47068_, p_47069_);
+   private static BlockPos getTopNonCollidingPos(LevelReader p_47066_, EntityType<?> p_47067_, long p_47068_, long y, long p_47069_) {
+      long i = p_47066_.getHeight(SpawnPlacements.m_21765_(p_47067_), p_47068_, y, p_47069_);
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(p_47068_, i, p_47069_);
       if (p_47066_.m_6042_().f_63856_()) {
          do {
@@ -397,11 +_,11 @@
 
          do {
             blockpos$mutableblockpos.m_122173_(Direction.DOWN);
-         } while(p_47066_.m_8055_(blockpos$mutableblockpos).m_60795_() && blockpos$mutableblockpos.m_123342_() > p_47066_.m_141937_());
+         } while(p_47066_.m_8055_(blockpos$mutableblockpos).m_60795_() && blockpos$mutableblockpos.getY() > p_47066_.getMinBuildHeight());
       }
 
       if (SpawnPlacements.m_21752_(p_47067_) == SpawnPlacements.Type.ON_GROUND) {
-         BlockPos blockpos = blockpos$mutableblockpos.m_7495_();
+         BlockPos blockpos = blockpos$mutableblockpos.below();
          if (p_47066_.m_8055_(blockpos).m_60647_(p_47066_, blockpos, PathComputationType.LAND)) {
             return blockpos;
          }
@@ -417,7 +_,7 @@
 
    @FunctionalInterface
    public interface ChunkGetter {
-      void m_47103_(long p_47104_, Consumer<LevelChunk> p_47105_);
+      void query(ChunkPos p_47104_, Consumer<LevelChunk> p_47105_);
    }
 
    @FunctionalInterface
