--- a/net/minecraft/world/level/PathNavigationRegion.java
+++ b/net/minecraft/world/level/PathNavigationRegion.java
@@ -37,10 +_,10 @@
       this.f_204180_ = Suppliers.memoize(() -> {
          return p_47164_.m_5962_().m_175515_(Registry.f_122885_).m_206081_(Biomes.f_48202_);
       });
-      this.f_47158_ = SectionPos.m_123171_(p_47165_.m_123341_());
-      this.f_47159_ = SectionPos.m_123171_(p_47165_.m_123343_());
-      int i = SectionPos.m_123171_(p_47166_.m_123341_());
-      int j = SectionPos.m_123171_(p_47166_.m_123343_());
+      this.f_47158_ = SectionPos.m_235865_(p_47165_.getX());
+      this.f_47159_ = SectionPos.m_235865_(p_47165_.getZ());
+      int i = SectionPos.m_235865_(p_47166_.getX());
+      int j = SectionPos.m_235865_(p_47166_.getZ());
       this.f_47160_ = new ChunkAccess[i - this.f_47158_ + 1][j - this.f_47159_ + 1];
       ChunkSource chunksource = p_47164_.m_7726_();
       this.f_47161_ = true;
@@ -51,10 +_,10 @@
          }
       }
 
-      for(int i1 = SectionPos.m_123171_(p_47165_.m_123341_()); i1 <= SectionPos.m_123171_(p_47166_.m_123341_()); ++i1) {
-         for(int j1 = SectionPos.m_123171_(p_47165_.m_123343_()); j1 <= SectionPos.m_123171_(p_47166_.m_123343_()); ++j1) {
+      for(int i1 = SectionPos.m_235865_(p_47165_.getX()); i1 <= SectionPos.m_235865_(p_47166_.getX()); ++i1) {
+         for(int j1 = SectionPos.m_235865_(p_47165_.getZ()); j1 <= SectionPos.m_235865_(p_47166_.getZ()); ++j1) {
             ChunkAccess chunkaccess = this.f_47160_[i1 - this.f_47158_][j1 - this.f_47159_];
-            if (chunkaccess != null && !chunkaccess.m_5566_(p_47165_.m_123342_(), p_47166_.m_123342_())) {
+            if (chunkaccess != null && !chunkaccess.isYSpaceEmpty(p_47165_.getY(), p_47166_.getY())) {
                this.f_47161_ = false;
                return;
             }
@@ -64,7 +_,7 @@
    }
 
    private ChunkAccess m_47185_(BlockPos p_47186_) {
-      return this.m_47167_(SectionPos.m_123171_(p_47186_.m_123341_()), SectionPos.m_123171_(p_47186_.m_123343_()));
+      return this.m_47167_(SectionPos.m_235865_(p_47186_.getX()), SectionPos.m_235865_(p_47186_.getZ()));
    }
 
    private ChunkAccess m_47167_(int p_47168_, int p_47169_) {
@@ -114,12 +_,12 @@
       }
    }
 
-   public int m_141937_() {
-      return this.f_47162_.m_141937_();
+   public long getMinBuildHeight() {
+      return this.f_47162_.getMinBuildHeight();
    }
 
-   public int m_141928_() {
-      return this.f_47162_.m_141928_();
+   public long getHeight() {
+      return this.f_47162_.getHeight();
    }
 
    public ProfilerFiller m_151625_() {
