--- a/net/minecraft/world/level/PotentialCalculator.java
+++ b/net/minecraft/world/level/PotentialCalculator.java
@@ -38,7 +_,7 @@
       }
 
       public double m_47203_(BlockPos p_47204_) {
-         double d0 = this.f_47198_.m_123331_(p_47204_);
+         double d0 = this.f_47198_.distSqr(p_47204_);
          return d0 == 0.0D ? Double.POSITIVE_INFINITY : this.f_47199_ / Math.sqrt(d0);
       }
    }
