--- a/net/minecraft/world/level/StructureManager.java
+++ b/net/minecraft/world/level/StructureManager.java
@@ -43,7 +_,7 @@
    }
 
    public List<StructureStart> m_220477_(ChunkPos p_220478_, Predicate<Structure> p_220479_) {
-      Map<Structure, LongSet> map = this.f_220460_.m_46819_(p_220478_.f_45578_, p_220478_.f_45579_, ChunkStatus.f_62316_).m_62769_();
+      Map<Structure, LongSet> map = this.f_220460_.getChunk(p_220478_.f_45578_, p_220478_.f_45579_, ChunkStatus.f_62316_).m_62769_();
       ImmutableList.Builder<StructureStart> builder = ImmutableList.builder();
 
       for(Map.Entry<Structure, LongSet> entry : map.entrySet()) {
@@ -57,7 +_,7 @@
    }
 
    public List<StructureStart> m_220504_(SectionPos p_220505_, Structure p_220506_) {
-      LongSet longset = this.f_220460_.m_46819_(p_220505_.m_123170_(), p_220505_.m_123222_(), ChunkStatus.f_62316_).m_213649_(p_220506_);
+      LongSet longset = this.f_220460_.getChunk(p_220505_.x(), p_220505_.z(), ChunkStatus.f_62316_).m_213649_(p_220506_);
       ImmutableList.Builder<StructureStart> builder = ImmutableList.builder();
       this.m_220480_(p_220506_, longset, builder::add);
       return builder.build();
@@ -65,8 +_,8 @@
 
    public void m_220480_(Structure p_220481_, LongSet p_220482_, Consumer<StructureStart> p_220483_) {
       for(long i : p_220482_) {
-         SectionPos sectionpos = SectionPos.m_123196_(new ChunkPos(i), this.f_220460_.m_151560_());
-         StructureStart structurestart = this.m_220512_(sectionpos, p_220481_, this.f_220460_.m_46819_(sectionpos.m_123170_(), sectionpos.m_123222_(), ChunkStatus.f_62315_));
+         SectionPos sectionpos = SectionPos.of(new ChunkPos(i), this.f_220460_.getMinSection());
+         StructureStart structurestart = this.m_220512_(sectionpos, p_220481_, this.f_220460_.getChunk(sectionpos.x(), sectionpos.z(), ChunkStatus.f_62315_));
          if (structurestart != null && structurestart.m_73603_()) {
             p_220483_.accept(structurestart);
          }
@@ -93,7 +_,7 @@
 
    public StructureStart m_220494_(BlockPos p_220495_, Structure p_220496_) {
       for(StructureStart structurestart : this.m_220504_(SectionPos.m_123199_(p_220495_), p_220496_)) {
-         if (structurestart.m_73601_().m_71051_(p_220495_)) {
+         if (structurestart.m_73601_().isInside(p_220495_)) {
             return structurestart;
          }
       }
@@ -134,7 +_,7 @@
 
    public boolean m_220497_(BlockPos p_220498_, StructureStart p_220499_) {
       for(StructurePiece structurepiece : p_220499_.m_73602_()) {
-         if (structurepiece.m_73547_().m_71051_(p_220498_)) {
+         if (structurepiece.m_73547_().isInside(p_220498_)) {
             return true;
          }
       }
@@ -144,12 +_,12 @@
 
    public boolean m_220486_(BlockPos p_220487_) {
       SectionPos sectionpos = SectionPos.m_123199_(p_220487_);
-      return this.f_220460_.m_46819_(sectionpos.m_123170_(), sectionpos.m_123222_(), ChunkStatus.f_62316_).m_187678_();
+      return this.f_220460_.getChunk(sectionpos.x(), sectionpos.z(), ChunkStatus.f_62316_).m_187678_();
    }
 
    public Map<Structure, LongSet> m_220522_(BlockPos p_220523_) {
       SectionPos sectionpos = SectionPos.m_123199_(p_220523_);
-      return this.f_220460_.m_46819_(sectionpos.m_123170_(), sectionpos.m_123222_(), ChunkStatus.f_62316_).m_62769_();
+      return this.f_220460_.getChunk(sectionpos.x(), sectionpos.z(), ChunkStatus.f_62316_).m_62769_();
    }
 
    public StructureCheckResult m_220473_(ChunkPos p_220474_, Structure p_220475_, boolean p_220476_) {
