--- a/net/minecraft/client/sounds/SoundEngine.java
+++ b/net/minecraft/client/sounds/SoundEngine.java
@@ -240,7 +_,7 @@
          } else {
             float f = this.m_120327_(tickablesoundinstance);
             float f1 = this.m_120324_(tickablesoundinstance);
-            Vec3 vec3 = new Vec3(tickablesoundinstance.m_7772_(), tickablesoundinstance.m_7780_(), tickablesoundinstance.m_7778_());
+            Vec3 vec3 = new Vec3(tickablesoundinstance.getX(), tickablesoundinstance.getY(), tickablesoundinstance.getZ());
             ChannelAccess.ChannelHandle channelaccess$channelhandle = this.f_120226_.get(tickablesoundinstance);
             if (channelaccess$channelhandle != null) {
                channelaccess$channelhandle.m_120154_((p_194478_) -> {
@@ -351,7 +_,7 @@
                      if (f2 == 0.0F && !p_120313_.m_7784_()) {
                         f_120215_.debug(f_120214_, "Skipped playing sound {}, volume was zero.", (Object)sound.m_119787_());
                      } else {
-                        Vec3 vec3 = new Vec3(p_120313_.m_7772_(), p_120313_.m_7780_(), p_120313_.m_7778_());
+                        Vec3 vec3 = new Vec3(p_120313_.getX(), p_120313_.getY(), p_120313_.getZ());
                         if (!this.f_120231_.isEmpty()) {
                            boolean flag1 = flag || soundinstance$attenuation == SoundInstance.Attenuation.NONE || this.f_120221_.m_83736_().m_82557_(vec3) < (double)(f1 * f1);
                            if (flag1) {
