--- a/net/minecraft/world/entity/animal/frog/Frog.java
+++ b/net/minecraft/world/entity/animal/frog/Frog.java
@@ -347,7 +_,7 @@
    }
 
    public static boolean m_218511_(EntityType<? extends Animal> p_218512_, LevelAccessor p_218513_, MobSpawnType p_218514_, BlockPos p_218515_, RandomSource p_218516_) {
-      return p_218513_.m_8055_(p_218515_.m_7495_()).m_204336_(BlockTags.f_215827_) && m_186209_(p_218513_, p_218515_);
+      return p_218513_.m_8055_(p_218515_.below()).m_204336_(BlockTags.f_215827_) && m_186209_(p_218513_, p_218515_);
    }
 
    class FrogLookControl extends LookControl {
@@ -372,10 +_,10 @@
          return this.m_230631_(new BlockPos(Mth.m_14107_(this.f_77313_.m_20191_().f_82288_), Mth.m_14107_(this.f_77313_.m_20191_().f_82289_), Mth.m_14107_(this.f_77313_.m_20191_().f_82290_)));
       }
 
-      public BlockPathTypes m_8086_(BlockGetter p_218551_, int p_218552_, int p_218553_, int p_218554_) {
-         this.f_218546_.m_122178_(p_218552_, p_218553_ - 1, p_218554_);
+      public BlockPathTypes getBlockPathType(BlockGetter p_218551_, long p_218552_, long p_218553_, long p_218554_) {
+         this.f_218546_.m_122169_(p_218552_, p_218553_ - 1, p_218554_);
          BlockState blockstate = p_218551_.m_8055_(this.f_218546_);
-         return blockstate.m_204336_(BlockTags.f_215837_) ? BlockPathTypes.OPEN : super.m_8086_(p_218551_, p_218552_, p_218553_, p_218554_);
+         return blockstate.m_204336_(BlockTags.f_215837_) ? BlockPathTypes.OPEN : super.getBlockPathType(p_218551_, p_218552_, p_218553_, p_218554_);
       }
    }
 
