--- a/net/minecraft/world/entity/animal/frog/FrogAi.java
+++ b/net/minecraft/world/entity/animal/frog/FrogAi.java
@@ -112,8 +_,8 @@
 
    private static <E extends Mob> boolean m_246199_(E p_249699_, BlockPos p_250057_) {
       Level level = p_249699_.f_19853_;
-      BlockPos blockpos = p_250057_.m_7495_();
-      if (level.m_6425_(p_250057_).m_76178_() && level.m_6425_(blockpos).m_76178_() && level.m_6425_(p_250057_.m_7494_()).m_76178_()) {
+      BlockPos blockpos = p_250057_.below();
+      if (level.m_6425_(p_250057_).m_76178_() && level.m_6425_(blockpos).m_76178_() && level.m_6425_(p_250057_.above()).m_76178_()) {
          BlockState blockstate = level.m_8055_(p_250057_);
          BlockState blockstate1 = level.m_8055_(blockpos);
          if (!blockstate.m_204336_(BlockTags.f_215837_) && !blockstate1.m_204336_(BlockTags.f_215837_)) {
