--- a/net/minecraft/world/entity/animal/frog/Tadpole.java
+++ b/net/minecraft/world/entity/animal/frog/Tadpole.java
@@ -168,7 +_,7 @@
    private void m_218690_(Player p_218691_, ItemStack p_218692_) {
       this.m_218705_(p_218691_, p_218692_);
       this.m_218700_(AgeableMob.m_216967_(this.m_218723_()));
-      this.m_9236_().m_7106_(ParticleTypes.f_123748_, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), 0.0D, 0.0D, 0.0D);
+      this.m_9236_().addParticle(ParticleTypes.f_123748_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), 0.0D, 0.0D, 0.0D);
    }
 
    private void m_218705_(Player p_218706_, ItemStack p_218707_) {
