--- a/net/minecraft/world/level/storage/DerivedLevelData.java
+++ b/net/minecraft/world/level/storage/DerivedLevelData.java
@@ -20,16 +_,16 @@
       this.f_78077_ = p_78080_;
    }
 
-   public int m_6789_() {
-      return this.f_78077_.m_6789_();
-   }
-
-   public int m_6527_() {
-      return this.f_78077_.m_6527_();
-   }
-
-   public int m_6526_() {
-      return this.f_78077_.m_6526_();
+   public long getXSpawn() {
+      return this.f_78077_.getXSpawn();
+   }
+
+   public long getYSpawn() {
+      return this.f_78077_.getYSpawn();
+   }
+
+   public long getZSpawn() {
+      return this.f_78077_.getZSpawn();
    }
 
    public float m_6790_() {
@@ -75,13 +_,13 @@
       return this.f_78076_.m_5464_();
    }
 
-   public void m_6395_(int p_78103_) {
-   }
-
-   public void m_6397_(int p_78110_) {
-   }
-
-   public void m_6400_(int p_78115_) {
+   public void setXSpawn(long pX) {
+   }
+
+   public void setYSpawn(long pY) {
+   }
+
+   public void setZSpawn(long pZ) {
    }
 
    public void m_7113_(float p_78083_) {
