--- a/net/minecraft/world/level/storage/LevelData.java
+++ b/net/minecraft/world/level/storage/LevelData.java
@@ -7,11 +_,11 @@
 import net.minecraft.world.level.LevelHeightAccessor;
 
 public interface LevelData {
-   int m_6789_();
-
-   int m_6527_();
-
-   int m_6526_();
+   long getXSpawn();
+
+   long getYSpawn();
+
+   long getZSpawn();
 
    float m_6790_();
 
@@ -35,7 +_,7 @@
 
    default void m_142471_(CrashReportCategory p_164873_, LevelHeightAccessor p_164874_) {
       p_164873_.m_128165_("Level spawn location", () -> {
-         return CrashReportCategory.m_178942_(p_164874_, this.m_6789_(), this.m_6527_(), this.m_6526_());
+         return CrashReportCategory.formatLocation(p_164874_, this.getXSpawn(), this.getYSpawn(), this.getZSpawn());
       });
       p_164873_.m_128165_("Level time", () -> {
          return String.format(Locale.ROOT, "%d game time, %d day time", this.m_6793_(), this.m_6792_());
