--- a/net/minecraft/world/level/storage/PrimaryLevelData.java
+++ b/net/minecraft/world/level/storage/PrimaryLevelData.java
@@ -50,9 +_,9 @@
    private final WorldOptions f_244409_;
    private final PrimaryLevelData.SpecialWorldProperty f_244368_;
    private final Lifecycle f_78445_;
-   private int f_78446_;
-   private int f_78447_;
-   private int f_78448_;
+   private long f_78446_;
+   private long f_78447_;
+   private long f_78448_;
    private float f_78449_;
    private long f_78450_;
    private long f_78451_;
@@ -160,9 +_,9 @@
          p_78547_.m_128365_("WorldGenSettings", p_78574_);
       });
       p_78547_.m_128405_("GameType", this.f_78443_.m_46929_().m_46392_());
-      p_78547_.m_128405_("SpawnX", this.f_78446_);
-      p_78547_.m_128405_("SpawnY", this.f_78447_);
-      p_78547_.m_128405_("SpawnZ", this.f_78448_);
+      p_78547_.m_128356_("SpawnX", this.f_78446_);
+      p_78547_.m_128356_("SpawnY", this.f_78447_);
+      p_78547_.m_128356_("SpawnZ", this.f_78448_);
       p_78547_.m_128350_("SpawnAngle", this.f_78449_);
       p_78547_.m_128356_("Time", this.f_78450_);
       p_78547_.m_128356_("DayTime", this.f_78451_);
@@ -211,15 +_,15 @@
       return listtag;
    }
 
-   public int m_6789_() {
+   public long getXSpawn() {
       return this.f_78446_;
    }
 
-   public int m_6527_() {
+   public long getYSpawn() {
       return this.f_78447_;
    }
 
-   public int m_6526_() {
+   public long getZSpawn() {
       return this.f_78448_;
    }
 
@@ -254,15 +_,15 @@
       return this.f_78455_;
    }
 
-   public void m_6395_(int p_78565_) {
+   public void setXSpawn(long p_78565_) {
       this.f_78446_ = p_78565_;
    }
 
-   public void m_6397_(int p_78579_) {
+   public void setYSpawn(long p_78579_) {
       this.f_78447_ = p_78579_;
    }
 
-   public void m_6400_(int p_78584_) {
+   public void setZSpawn(long p_78584_) {
       this.f_78448_ = p_78584_;
    }
 
@@ -279,9 +_,9 @@
    }
 
    public void m_7250_(BlockPos p_78540_, float p_78541_) {
-      this.f_78446_ = p_78540_.m_123341_();
-      this.f_78447_ = p_78540_.m_123342_();
-      this.f_78448_ = p_78540_.m_123343_();
+      this.f_78446_ = p_78540_.getX();
+      this.f_78447_ = p_78540_.getY();
+      this.f_78448_ = p_78540_.getZ();
       this.f_78449_ = p_78541_;
    }
 
