--- a/net/minecraft/world/entity/ai/village/poi/PoiManager.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiManager.java
@@ -47,11 +_,11 @@
    }
 
    public void m_217919_(BlockPos p_217920_, Holder<PoiType> p_217921_) {
-      this.m_63827_(SectionPos.m_175568_(p_217920_)).m_218021_(p_217920_, p_217921_);
+      this.getOrCreate(SectionPos.m_123199_(p_217920_)).m_218021_(p_217920_, p_217921_);
    }
 
    public void m_27079_(BlockPos p_27080_) {
-      this.m_63823_(SectionPos.m_175568_(p_27080_)).ifPresent((p_148657_) -> {
+      this.getOrLoad(SectionPos.m_123199_(p_27080_)).ifPresent((p_148657_) -> {
          p_148657_.m_27279_(p_27080_);
       });
    }
@@ -72,21 +_,21 @@
          return this.m_27117_(p_27167_, p_217938_, p_27170_);
       }).filter((p_217971_) -> {
          BlockPos blockpos = p_217971_.m_27257_();
-         return Math.abs(blockpos.m_123341_() - p_27168_.m_123341_()) <= p_27169_ && Math.abs(blockpos.m_123343_() - p_27168_.m_123343_()) <= p_27169_;
+         return Math.abs(blockpos.getX() - p_27168_.getX()) <= p_27169_ && Math.abs(blockpos.getZ() - p_27168_.getZ()) <= p_27169_;
       });
    }
 
    public Stream<PoiRecord> m_27181_(Predicate<Holder<PoiType>> p_27182_, BlockPos p_27183_, int p_27184_, PoiManager.Occupancy p_27185_) {
       int i = p_27184_ * p_27184_;
       return this.m_27166_(p_27182_, p_27183_, p_27184_, p_27185_).filter((p_217906_) -> {
-         return p_217906_.m_27257_().m_123331_(p_27183_) <= (double)i;
+         return p_217906_.m_27257_().distSqr(p_27183_) <= (double)i;
       });
    }
 
    @VisibleForDebug
    public Stream<PoiRecord> m_27117_(Predicate<Holder<PoiType>> p_27118_, ChunkPos p_27119_, PoiManager.Occupancy p_27120_) {
-      return IntStream.range(this.f_156618_.m_151560_(), this.f_156618_.m_151561_()).boxed().map((p_217886_) -> {
-         return this.m_63823_(SectionPos.m_123196_(p_27119_, p_217886_).m_123252_());
+      return java.util.stream.LongStream.range(this.f_156618_.getMinSection(), this.f_156618_.getMaxSection()).boxed().map((p_217886_) -> {
+         return this.getOrLoad(SectionPos.of(p_27119_, p_217886_));
       }).filter(Optional::isPresent).flatMap((p_217942_) -> {
          return p_217942_.get().m_27304_(p_27118_, p_27120_);
       });
@@ -106,7 +_,7 @@
 
    public Stream<Pair<Holder<PoiType>, BlockPos>> m_217994_(Predicate<Holder<PoiType>> p_217995_, Predicate<BlockPos> p_217996_, BlockPos p_217997_, int p_217998_, PoiManager.Occupancy p_217999_) {
       return this.m_217983_(p_217995_, p_217996_, p_217997_, p_217998_, p_217999_).sorted(Comparator.comparingDouble((p_217915_) -> {
-         return p_217915_.getSecond().m_123331_(p_217997_);
+         return p_217915_.getSecond().distSqr(p_217997_);
       }));
    }
 
@@ -116,13 +_,13 @@
 
    public Optional<BlockPos> m_27192_(Predicate<Holder<PoiType>> p_27193_, BlockPos p_27194_, int p_27195_, PoiManager.Occupancy p_27196_) {
       return this.m_27181_(p_27193_, p_27194_, p_27195_, p_27196_).map(PoiRecord::m_27257_).min(Comparator.comparingDouble((p_217977_) -> {
-         return p_217977_.m_123331_(p_27194_);
+         return p_217977_.distSqr(p_27194_);
       }));
    }
 
    public Optional<Pair<Holder<PoiType>, BlockPos>> m_218002_(Predicate<Holder<PoiType>> p_218003_, BlockPos p_218004_, int p_218005_, PoiManager.Occupancy p_218006_) {
       return this.m_27181_(p_218003_, p_218004_, p_218005_, p_218006_).min(Comparator.comparingDouble((p_217909_) -> {
-         return p_217909_.m_27257_().m_123331_(p_218004_);
+         return p_217909_.m_27257_().distSqr(p_218004_);
       })).map((p_217959_) -> {
          return Pair.of(p_217959_.m_218018_(), p_217959_.m_27257_());
       });
@@ -130,7 +_,7 @@
 
    public Optional<BlockPos> m_148658_(Predicate<Holder<PoiType>> p_148659_, Predicate<BlockPos> p_148660_, BlockPos p_148661_, int p_148662_, PoiManager.Occupancy p_148663_) {
       return this.m_27181_(p_148659_, p_148661_, p_148662_, p_148663_).map(PoiRecord::m_27257_).filter(p_148660_).min(Comparator.comparingDouble((p_217918_) -> {
-         return p_217918_.m_123331_(p_148661_);
+         return p_217918_.distSqr(p_148661_);
       }));
    }
 
@@ -151,7 +_,7 @@
    }
 
    public boolean m_27154_(BlockPos p_27155_) {
-      return this.m_63823_(SectionPos.m_175568_(p_27155_)).map((p_217993_) -> {
+      return this.getOrLoad(SectionPos.m_123199_(p_27155_)).map((p_217993_) -> {
          return p_217993_.m_27317_(p_27155_);
       }).orElseThrow(() -> {
          return Util.m_137570_(new IllegalStateException("POI never registered at " + p_27155_));
@@ -159,13 +_,13 @@
    }
 
    public boolean m_27091_(BlockPos p_27092_, Predicate<Holder<PoiType>> p_27093_) {
-      return this.m_63823_(SectionPos.m_175568_(p_27092_)).map((p_217925_) -> {
+      return this.getOrLoad(SectionPos.m_123199_(p_27092_)).map((p_217925_) -> {
          return p_217925_.m_27288_(p_27092_, p_27093_);
       }).orElse(false);
    }
 
    public Optional<Holder<PoiType>> m_27177_(BlockPos p_27178_) {
-      return this.m_63823_(SectionPos.m_175568_(p_27178_)).flatMap((p_217974_) -> {
+      return this.getOrLoad(SectionPos.m_123199_(p_27178_)).flatMap((p_217974_) -> {
          return p_217974_.m_27319_(p_27178_);
       });
    }
@@ -174,18 +_,18 @@
    @Deprecated
    @VisibleForDebug
    public int m_148653_(BlockPos p_148654_) {
-      return this.m_63823_(SectionPos.m_175568_(p_148654_)).map((p_217912_) -> {
+      return this.getOrLoad(SectionPos.m_123199_(p_148654_)).map((p_217912_) -> {
          return p_217912_.m_148682_(p_148654_);
       }).orElse(0);
    }
 
    public int m_27098_(SectionPos p_27099_) {
       this.f_27029_.m_27203_();
-      return this.f_27029_.m_6172_(p_27099_.m_123252_());
+      return this.f_27029_.getLevel(p_27099_);
    }
 
-   boolean m_27197_(long p_27198_) {
-      Optional<PoiSection> optional = this.m_63818_(p_27198_);
+   boolean isVillageCenter(SectionPos p_27198_) {
+      Optional<PoiSection> optional = this.get(p_27198_);
       return optional == null ? false : optional.map((p_217883_) -> {
          return p_217883_.m_27304_((p_217927_) -> {
             return p_217927_.m_203656_(PoiTypeTags.f_215876_);
@@ -198,17 +_,17 @@
       this.f_27029_.m_27203_();
    }
 
-   protected void m_5838_(long p_27036_) {
-      super.m_5838_(p_27036_);
-      this.f_27029_.m_8288_(p_27036_, this.f_27029_.m_7409_(p_27036_), false);
+   protected void setDirty(SectionPos p_27036_) {
+      super.setDirty(p_27036_);
+      this.f_27029_.update(p_27036_, this.f_27029_.getLevelFromSource(p_27036_), false);
    }
 
-   protected void m_5839_(long p_27145_) {
-      this.f_27029_.m_8288_(p_27145_, this.f_27029_.m_7409_(p_27145_), false);
+   protected void onSectionLoad(SectionPos p_27145_) {
+      this.f_27029_.update(p_27145_, this.f_27029_.getLevelFromSource(p_27145_), false);
    }
 
    public void m_280570_(SectionPos p_281731_, LevelChunkSection p_281893_) {
-      Util.m_137521_(this.m_63823_(p_281731_.m_123252_()), (p_217898_) -> {
+      Util.m_137521_(this.getOrLoad(p_281731_), (p_217898_) -> {
          p_217898_.m_27302_((p_217967_) -> {
             if (m_27060_(p_281893_)) {
                this.m_27069_(p_281893_, p_281731_, p_217967_);
@@ -217,7 +_,7 @@
          });
       }, () -> {
          if (m_27060_(p_281893_)) {
-            PoiSection poisection = this.m_63827_(p_281731_.m_123252_());
+            PoiSection poisection = this.getOrCreate(p_281731_);
             this.m_27069_(p_281893_, p_281731_, poisection::m_218021_);
          }
 
@@ -230,7 +_,7 @@
 
    private void m_27069_(LevelChunkSection p_27070_, SectionPos p_27071_, BiConsumer<BlockPos, Holder<PoiType>> p_27072_) {
       p_27071_.m_123253_().forEach((p_217902_) -> {
-         BlockState blockstate = p_27070_.m_62982_(SectionPos.m_123207_(p_217902_.m_123341_()), SectionPos.m_123207_(p_217902_.m_123342_()), SectionPos.m_123207_(p_217902_.m_123343_()));
+         BlockState blockstate = p_27070_.getBlockState(SectionPos.sectionRelative(p_217902_.getX()), SectionPos.sectionRelative(p_217902_.getY()), SectionPos.sectionRelative(p_217902_.getZ()));
          PoiTypes.m_218075_(blockstate).ifPresent((p_217931_) -> {
             p_27072_.accept(p_217902_, p_217931_);
          });
@@ -238,8 +_,8 @@
    }
 
    public void m_27056_(LevelReader p_27057_, BlockPos p_27058_, int p_27059_) {
-      SectionPos.m_175557_(new ChunkPos(p_27058_), Math.floorDiv(p_27059_, 16), this.f_156618_.m_151560_(), this.f_156618_.m_151561_()).map((p_217979_) -> {
-         return Pair.of(p_217979_, this.m_63823_(p_217979_.m_123252_()));
+      SectionPos.aroundChunk(new ChunkPos(p_27058_), Math.floorDiv(p_27059_, 16), this.f_156618_.getMinSection(), this.f_156618_.getMaxSection()).map((p_217979_) -> {
+         return Pair.of(p_217979_, this.getOrLoad(p_217979_));
       }).filter((p_217963_) -> {
          return !p_217963_.getSecond().map(PoiSection::m_27272_).orElse(false);
       }).map((p_217891_) -> {
@@ -247,29 +_,29 @@
       }).filter((p_217961_) -> {
          return this.f_27030_.add(p_217961_.m_45588_());
       }).forEach((p_217889_) -> {
-         p_27057_.m_46819_(p_217889_.f_45578_, p_217889_.f_45579_, ChunkStatus.f_62314_);
+         p_27057_.getChunk(p_217889_.f_45578_, p_217889_.f_45579_, ChunkStatus.f_62314_);
       });
    }
 
    final class DistanceTracker extends SectionTracker {
-      private final Long2ByteMap f_27200_ = new Long2ByteOpenHashMap();
+      private final it.unimi.dsi.fastutil.objects.Object2ByteMap<net.minecraft.core.SectionPos> f_27200_ = new it.unimi.dsi.fastutil.objects.Object2ByteOpenHashMap<>();
 
       protected DistanceTracker() {
          super(7, 16, 256);
          this.f_27200_.defaultReturnValue((byte)7);
       }
 
-      protected int m_7409_(long p_27208_) {
-         return PoiManager.this.m_27197_(p_27208_) ? 0 : 7;
-      }
-
-      protected int m_6172_(long p_27210_) {
-         return this.f_27200_.get(p_27210_);
-      }
-
-      protected void m_7351_(long p_27205_, int p_27206_) {
+      protected int getLevelFromSource(SectionPos p_27208_) {
+         return PoiManager.this.isVillageCenter(p_27208_) ? 0 : 7;
+      }
+
+      protected int getLevel(net.minecraft.core.SectionPos p_27210_) {
+         return this.f_27200_.getByte(p_27210_);
+      }
+
+      protected void setLevel(net.minecraft.core.SectionPos p_27205_, int p_27206_) {
          if (p_27206_ > 6) {
-            this.f_27200_.remove(p_27205_);
+            this.f_27200_.removeByte(p_27205_);
          } else {
             this.f_27200_.put(p_27205_, (byte)p_27206_);
          }
