--- a/net/minecraft/util/profiling/jfr/event/ChunkGenerationEvent.java
+++ b/net/minecraft/util/profiling/jfr/event/ChunkGenerationEvent.java
@@ -23,16 +_,16 @@
    public static final EventType f_195534_ = EventType.getEventType(ChunkGenerationEvent.class);
    @Name("worldPosX")
    @Label("First Block X World Position")
-   public final int f_195539_;
+   public final long f_195539_;
    @Name("worldPosZ")
    @Label("First Block Z World Position")
-   public final int f_195540_;
+   public final long f_195540_;
    @Name("chunkPosX")
    @Label("Chunk X Position")
-   public final int f_195535_;
+   public final long f_195535_;
    @Name("chunkPosZ")
    @Label("Chunk Z Position")
-   public final int f_195536_;
+   public final long f_195536_;
    @Name("status")
    @Label("Status")
    public final String f_195538_;
@@ -45,8 +_,8 @@
       this.f_195537_ = p_195544_.toString();
       this.f_195535_ = p_195543_.f_45578_;
       this.f_195536_ = p_195543_.f_45579_;
-      this.f_195539_ = p_195543_.m_45604_();
-      this.f_195540_ = p_195543_.m_45605_();
+      this.f_195539_ = p_195543_.getMinBlockX();
+      this.f_195540_ = p_195543_.getMinBlockZ();
    }
 
    public static class Fields {
