--- a/net/minecraft/util/profiling/jfr/event/ChunkGenerationEvent.java
+++ b/net/minecraft/util/profiling/jfr/event/ChunkGenerationEvent.java
@@ -23,16 +_,16 @@
    public static final EventType TYPE = EventType.getEventType(ChunkGenerationEvent.class);
    @Name("worldPosX")
    @Label("First Block X World Position")
-   public final int worldPosX;
+   public final long worldPosX;
    @Name("worldPosZ")
    @Label("First Block Z World Position")
-   public final int worldPosZ;
+   public final long worldPosZ;
    @Name("chunkPosX")
    @Label("Chunk X Position")
-   public final int chunkPosX;
+   public final long chunkPosX;
    @Name("chunkPosZ")
    @Label("Chunk Z Position")
-   public final int chunkPosZ;
+   public final long chunkPosZ;
    @Name("status")
    @Label("Status")
    public final String targetStatus;
@@ -43,10 +_,10 @@
    public ChunkGenerationEvent(ChunkPos p_195543_, ResourceKey<Level> p_195544_, String p_195545_) {
       this.targetStatus = p_195545_;
       this.level = p_195544_.toString();
-      this.chunkPosX = p_195543_.f_45578_;
-      this.chunkPosZ = p_195543_.f_45579_;
-      this.worldPosX = p_195543_.m_45604_();
-      this.worldPosZ = p_195543_.m_45605_();
+      this.chunkPosX = p_195543_.x();
+      this.chunkPosZ = p_195543_.z();
+      this.worldPosX = p_195543_.getMinBlockX();
+      this.worldPosZ = p_195543_.getMinBlockZ();
    }
 
    public static class Fields {
