--- a/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
+++ b/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
@@ -28,7 +_,7 @@
 
    public static ColumnPos m_118992_(CommandContext<CommandSourceStack> p_118993_, String p_118994_) {
       BlockPos blockpos = p_118993_.getArgument(p_118994_, Coordinates.class).m_119568_(p_118993_.getSource());
-      return new ColumnPos(blockpos.m_123341_(), blockpos.m_123343_());
+      return new ColumnPos(blockpos.getBigX(), blockpos.getBigZ());
    }
 
    public Coordinates parse(StringReader p_118991_) throws CommandSyntaxException {
@@ -40,7 +_,7 @@
          if (p_118991_.canRead() && p_118991_.peek() == ' ') {
             p_118991_.skip();
             WorldCoordinate worldcoordinate1 = WorldCoordinate.m_120869_(p_118991_);
-            return new WorldCoordinates(worldcoordinate, new WorldCoordinate(true, 0.0D), worldcoordinate1);
+            return new WorldCoordinates(worldcoordinate, new WorldCoordinate(true, me.alphamode.mclong.math.BigDecimal.ZERO), worldcoordinate1);
          } else {
             p_118991_.setCursor(i);
             throw f_118985_.createWithContext(p_118991_);
