--- a/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
+++ b/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
@@ -28,7 +_,7 @@
 
    public static ColumnPos m_118992_(CommandContext<CommandSourceStack> p_118993_, String p_118994_) {
       BlockPos blockpos = p_118993_.getArgument(p_118994_, Coordinates.class).m_119568_(p_118993_.getSource());
-      return new ColumnPos(blockpos.m_123341_(), blockpos.m_123343_());
+      return new ColumnPos(blockpos.getX(), blockpos.getY(), blockpos.getZ());
    }
 
    public Coordinates parse(StringReader p_118991_) throws CommandSyntaxException {
