--- a/net/minecraft/commands/arguments/coordinates/Coordinates.java
+++ b/net/minecraft/commands/arguments/coordinates/Coordinates.java
@@ -11,7 +_,7 @@
    Vec2 m_6970_(CommandSourceStack p_119567_);
 
    default BlockPos m_119568_(CommandSourceStack p_119569_) {
-      return BlockPos.m_274446_(this.m_6955_(p_119569_));
+      return BlockPos.containing(this.m_6955_(p_119569_));
    }
 
    boolean m_6888_();
