--- a/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java
+++ b/net/minecraft/commands/arguments/coordinates/LocalCoordinates.java
@@ -32,10 +_,10 @@
       Vec3 vec31 = new Vec3((double)(f * f2), (double)f3, (double)(f1 * f2));
       Vec3 vec32 = new Vec3((double)(f * f4), (double)f5, (double)(f1 * f4));
       Vec3 vec33 = vec31.m_82537_(vec32).m_82490_(-1.0D);
-      double d0 = vec31.f_82479_ * this.f_119900_ + vec32.f_82479_ * this.f_119899_ + vec33.f_82479_ * this.f_119898_;
-      double d1 = vec31.f_82480_ * this.f_119900_ + vec32.f_82480_ * this.f_119899_ + vec33.f_82480_ * this.f_119898_;
-      double d2 = vec31.f_82481_ * this.f_119900_ + vec32.f_82481_ * this.f_119899_ + vec33.f_82481_ * this.f_119898_;
-      return new Vec3(vec3.f_82479_ + d0, vec3.f_82480_ + d1, vec3.f_82481_ + d2);
+      double d0 = vec31.m_7096_() * this.f_119900_ + vec32.m_7096_() * this.f_119899_ + vec33.m_7096_() * this.f_119898_;
+      double d1 = vec31.m_7098_() * this.f_119900_ + vec32.m_7098_() * this.f_119899_ + vec33.m_7098_() * this.f_119898_;
+      double d2 = vec31.m_7094_() * this.f_119900_ + vec32.m_7094_() * this.f_119899_ + vec33.m_7094_() * this.f_119898_;
+      return new Vec3(vec3.m_7096_() + d0, vec3.m_7098_() + d1, vec3.m_7094_() + d2);
    }
 
    public Vec2 m_6970_(CommandSourceStack p_119915_) {
