--- a/net/minecraft/commands/arguments/coordinates/WorldCoordinates.java
+++ b/net/minecraft/commands/arguments/coordinates/WorldCoordinates.java
@@ -19,12 +_,12 @@
 
    public Vec3 m_6955_(CommandSourceStack p_120893_) {
       Vec3 vec3 = p_120893_.m_81371_();
-      return new Vec3(this.f_120879_.m_120867_(vec3.f_82479_), this.f_120880_.m_120867_(vec3.f_82480_), this.f_120881_.m_120867_(vec3.f_82481_));
+      return new Vec3(this.f_120879_.get(vec3.getX()), this.f_120880_.get(vec3.getY()), this.f_120881_.get(vec3.getZ()));
    }
 
    public Vec2 m_6970_(CommandSourceStack p_120896_) {
       Vec2 vec2 = p_120896_.m_81376_();
-      return new Vec2((float)this.f_120879_.m_120867_((double)vec2.f_82470_), (float)this.f_120880_.m_120867_((double)vec2.f_82471_));
+      return new Vec2(this.f_120879_.get(me.alphamode.mclong.math.BigDecimal.valueOf(vec2.f_82470_)).floatValue(), this.f_120880_.get(me.alphamode.mclong.math.BigDecimal.valueOf(vec2.f_82471_)).floatValue());
    }
 
    public boolean m_6888_() {
@@ -95,15 +_,15 @@
    }
 
    public static WorldCoordinates m_175085_(double p_175086_, double p_175087_, double p_175088_) {
-      return new WorldCoordinates(new WorldCoordinate(false, p_175086_), new WorldCoordinate(false, p_175087_), new WorldCoordinate(false, p_175088_));
+      return new WorldCoordinates(new WorldCoordinate(false, me.alphamode.mclong.math.BigDecimal.valueOf(p_175086_)), new WorldCoordinate(false, me.alphamode.mclong.math.BigDecimal.valueOf(p_175087_)), new WorldCoordinate(false, me.alphamode.mclong.math.BigDecimal.valueOf(p_175088_)));
    }
 
    public static WorldCoordinates m_175089_(Vec2 p_175090_) {
-      return new WorldCoordinates(new WorldCoordinate(false, (double)p_175090_.f_82470_), new WorldCoordinate(false, (double)p_175090_.f_82471_), new WorldCoordinate(true, 0.0D));
+      return new WorldCoordinates(new WorldCoordinate(false, me.alphamode.mclong.math.BigDecimal.valueOf(p_175090_.f_82470_)), new WorldCoordinate(false, me.alphamode.mclong.math.BigDecimal.valueOf(p_175090_.f_82471_)), new WorldCoordinate(true, me.alphamode.mclong.math.BigDecimal.ZERO));
    }
 
    public static WorldCoordinates m_120898_() {
-      return new WorldCoordinates(new WorldCoordinate(true, 0.0D), new WorldCoordinate(true, 0.0D), new WorldCoordinate(true, 0.0D));
+      return new WorldCoordinates(new WorldCoordinate(true, me.alphamode.mclong.math.BigDecimal.ZERO), new WorldCoordinate(true, me.alphamode.mclong.math.BigDecimal.ZERO), new WorldCoordinate(true, me.alphamode.mclong.math.BigDecimal.ZERO));
    }
 
    public int hashCode() {
