--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -386,7 +_,7 @@
    }
 
    private boolean m_239812_() {
-      return this.f_238682_ == null || !this.f_238682_.m_203195_(this.m_20182_(), (double)GameEvent.f_238690_.m_157827_()) || !this.m_9236_().m_8055_(this.f_238682_).m_60713_(Blocks.f_50131_);
+      return this.f_238682_ == null || !this.f_238682_.closerToCenterThan(this.m_20182_(), (double)GameEvent.f_238690_.m_157827_()) || !this.m_9236_().m_8055_(this.f_238682_).m_60713_(Blocks.f_50131_);
    }
 
    public float m_218394_(float p_218395_) {
@@ -544,10 +_,10 @@
 
       public boolean m_214068_(ServerLevel p_250009_, GameEvent p_251513_, GameEvent.Context p_249478_, Vec3 p_250852_) {
          if (p_251513_ == GameEvent.f_238690_) {
-            Allay.this.m_240101_(BlockPos.m_274446_(p_250852_), true);
+            Allay.this.m_240101_(BlockPos.containing(p_250852_), true);
             return true;
          } else if (p_251513_ == GameEvent.f_238649_) {
-            Allay.this.m_240101_(BlockPos.m_274446_(p_250852_), false);
+            Allay.this.m_240101_(BlockPos.containing(p_250852_), false);
             return true;
          } else {
             return false;
