--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -362,7 +_,7 @@
    }
 
    private boolean m_239812_() {
-      return this.f_238682_ == null || !this.f_238682_.m_203195_(this.m_20182_(), (double)GameEvent.f_238690_.m_157827_()) || !this.f_19853_.m_8055_(this.f_238682_).m_60713_(Blocks.f_50131_);
+      return this.f_238682_ == null || !this.f_238682_.closerToCenterThan(this.m_20182_(), (double)GameEvent.f_238690_.m_157827_()) || !this.f_19853_.m_8055_(this.f_238682_).m_60713_(Blocks.f_50131_);
    }
 
    public float m_218394_(float p_218395_) {
@@ -428,7 +_,7 @@
       int l = Mth.m_14107_(aabb.f_82293_ + 0.5D);
       int i1 = Mth.m_14107_(aabb.f_82289_ - 0.5D);
       int j1 = Mth.m_14107_(aabb.f_82292_ + 0.5D);
-      return BlockPos.m_121976_(i, i1, k, j, j1, l);
+      return BlockPos.betweenClosed(i, i1, k, j, j1, l);
    }
 
    private void m_218375_() {
