--- a/net/minecraft/world/entity/animal/allay/AllayAi.java
+++ b/net/minecraft/world/entity/animal/allay/AllayAi.java
@@ -92,7 +_,7 @@
       if (optional.isPresent()) {
          GlobalPos globalpos = optional.get();
          if (m_218412_(p_218424_, brain, globalpos)) {
-            return Optional.of(new BlockPosTracker(globalpos.m_122646_().m_7494_()));
+            return Optional.of(new BlockPosTracker(globalpos.m_122646_().above()));
          }
 
          brain.m_21936_(MemoryModuleType.f_217779_);
