--- a/net/minecraft/world/level/gameevent/vibrations/VibrationListener.java
+++ b/net/minecraft/world/level/gameevent/vibrations/VibrationListener.java
@@ -186,8 +_,8 @@
    }
 
    private static boolean m_223775_(Level p_223776_, Vec3 p_223777_, Vec3 p_223778_) {
-      Vec3 vec3 = new Vec3((double)Mth.m_14107_(p_223777_.f_82479_) + 0.5D, (double)Mth.m_14107_(p_223777_.f_82480_) + 0.5D, (double)Mth.m_14107_(p_223777_.f_82481_) + 0.5D);
-      Vec3 vec31 = new Vec3((double)Mth.m_14107_(p_223778_.f_82479_) + 0.5D, (double)Mth.m_14107_(p_223778_.f_82480_) + 0.5D, (double)Mth.m_14107_(p_223778_.f_82481_) + 0.5D);
+      Vec3 vec3 = new Vec3((double)Mth.m_14134_(p_223777_.f_82479_) + 0.5D, (double)Mth.m_14134_(p_223777_.f_82480_) + 0.5D, (double)Mth.m_14134_(p_223777_.f_82481_) + 0.5D);
+      Vec3 vec31 = new Vec3((double)Mth.m_14134_(p_223778_.f_82479_) + 0.5D, (double)Mth.m_14134_(p_223778_.f_82480_) + 0.5D, (double)Mth.m_14134_(p_223778_.f_82481_) + 0.5D);
 
       for(Direction direction : Direction.values()) {
          Vec3 vec32 = vec3.m_231075_(direction, (double)1.0E-5F);
