--- a/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
@@ -54,6 +_,6 @@
    }
 
    protected double m_191406_(BlockPos p_191407_) {
-      return this.f_191393_.m_75380_((double)((float)p_191407_.m_123341_() * this.f_191392_), (double)((float)p_191407_.m_123342_() * this.f_191392_), (double)((float)p_191407_.m_123343_() * this.f_191392_));
+      return this.f_191393_.m_75380_((double)((float)p_191407_.getX() * this.f_191392_), (double)((float)p_191407_.getY() * this.f_191392_), (double)((float)p_191407_.getZ() * this.f_191392_));
    }
 }
