--- a/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
@@ -47,13 +_,13 @@
       List<BlockState> list = Lists.newArrayListWithCapacity(i);
 
       for(int j = 0; j < i; ++j) {
-         list.add(this.m_191449_(this.f_191439_, this.m_191406_(p_225911_.m_7918_(j * '\ud511', 0, j * '\u85ba'))));
+         list.add(this.m_191449_(this.f_191439_, this.m_191406_(p_225911_.offset(j * '\ud511', 0, j * '\u85ba'))));
       }
 
       return this.m_191452_(list, p_225911_, (double)this.f_191419_);
    }
 
    protected double m_191406_(BlockPos p_191407_) {
-      return this.f_191393_.m_75380_((double)((float)p_191407_.m_123341_() * this.f_191392_), (double)((float)p_191407_.m_123342_() * this.f_191392_), (double)((float)p_191407_.m_123343_() * this.f_191392_));
+      return this.f_191393_.m_75380_((double)((float)p_191407_.getX() * this.f_191392_), (double)((float)p_191407_.getY() * this.f_191392_), (double)((float)p_191407_.getZ() * this.f_191392_));
    }
 }
