--- a/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseBasedStateProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/NoiseBasedStateProvider.java
@@ -33,6 +_,6 @@
    }
 
    protected double m_191429_(BlockPos p_191430_, double p_191431_) {
-      return this.f_191420_.m_75380_((double)p_191430_.m_123341_() * p_191431_, (double)p_191430_.m_123342_() * p_191431_, (double)p_191430_.m_123343_() * p_191431_);
+      return this.f_191420_.getValue(p_191430_.getBigDecX().multiply(p_191431_), p_191430_.getBigDecY().multiply(p_191431_), p_191430_.getBigDecZ().multiply(p_191431_));
    }
 }
