--- a/net/minecraft/world/level/levelgen/placement/EnvironmentScanPlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/EnvironmentScanPlacement.java
@@ -56,7 +_,7 @@
                }
 
                blockpos$mutableblockpos.m_122173_(this.f_191639_);
-               if (worldgenlevel.m_151562_(blockpos$mutableblockpos.m_123342_())) {
+               if (worldgenlevel.isOutsideBuildHeight(blockpos$mutableblockpos.getY())) {
                   return Stream.of();
                }
 
