--- a/net/minecraft/world/level/levelgen/placement/HeightRangePlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/HeightRangePlacement.java
@@ -35,7 +_,7 @@
    }
 
    public Stream<BlockPos> m_213676_(PlacementContext p_226340_, RandomSource p_226341_, BlockPos p_226342_) {
-      return Stream.of(p_226342_.m_175288_(this.f_191674_.m_213859_(p_226341_, p_226340_)));
+      return Stream.of(p_226342_.atY(this.f_191674_.sample(p_226341_, p_226340_)));
    }
 
    public PlacementModifierType<?> m_183327_() {
