--- a/net/minecraft/world/level/levelgen/placement/InSquarePlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/InSquarePlacement.java
@@ -16,9 +_,9 @@
    }
 
    public Stream<BlockPos> m_213676_(PlacementContext p_226348_, RandomSource p_226349_, BlockPos p_226350_) {
-      int i = p_226349_.m_188503_(16) + p_226350_.m_123341_();
-      int j = p_226349_.m_188503_(16) + p_226350_.m_123343_();
-      return Stream.of(new BlockPos(i, p_226350_.m_123342_(), j));
+      me.alphamode.mclong.math.BigInteger i = me.alphamode.mclong.math.BigInteger.valueOf(p_226349_.m_188503_(16)).add(p_226350_.getBigX());
+      me.alphamode.mclong.math.BigInteger j = me.alphamode.mclong.math.BigInteger.valueOf(p_226349_.m_188503_(16)).add(p_226350_.getBigZ());
+      return Stream.of(new BlockPos(i, p_226350_.getBigY(), j));
    }
 
    public PlacementModifierType<?> m_183327_() {
