--- a/net/minecraft/world/level/levelgen/placement/NoiseBasedCountPlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/NoiseBasedCountPlacement.java
@@ -31,7 +_,7 @@
    }
 
    protected int m_213944_(RandomSource p_226352_, BlockPos p_226353_) {
-      double d0 = Biome.f_47433_.m_75449_((double)p_226353_.m_123341_() / this.f_191724_, (double)p_226353_.m_123343_() / this.f_191724_, false);
+      double d0 = Biome.f_47433_.m_75449_((double)p_226353_.getX() / this.f_191724_, (double)p_226353_.getZ() / this.f_191724_, false);
       return (int)Math.ceil((d0 + this.f_191725_) * (double)this.f_191723_);
    }
 
