--- a/net/minecraft/world/level/levelgen/placement/NoiseThresholdCountPlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/NoiseThresholdCountPlacement.java
@@ -31,7 +_,7 @@
    }
 
    protected int m_213944_(RandomSource p_226355_, BlockPos p_226356_) {
-      double d0 = Biome.f_47433_.m_75449_((double)p_226356_.m_123341_() / 200.0D, (double)p_226356_.m_123343_() / 200.0D, false);
+      double d0 = Biome.f_47433_.m_75449_((double)p_226356_.getX() / 200.0D, (double)p_226356_.getZ() / 200.0D, false);
       return d0 < this.f_191748_ ? this.f_191749_ : this.f_191750_;
    }
 
