--- a/net/minecraft/world/level/levelgen/placement/PlacementContext.java
+++ b/net/minecraft/world/level/levelgen/placement/PlacementContext.java
@@ -24,20 +_,20 @@
       this.f_191816_ = p_191820_;
    }
 
-   public int m_191824_(Heightmap.Types p_191825_, int p_191826_, int p_191827_) {
-      return this.f_191814_.m_6924_(p_191825_, p_191826_, p_191827_);
+   public long getHeight(Heightmap.Types p_191825_, long p_191826_, long p_191827_) {
+      return this.f_191814_.getHeight(p_191825_, p_191826_, p_191827_);
    }
 
    public CarvingMask m_191821_(ChunkPos p_191822_, GenerationStep.Carving p_191823_) {
-      return ((ProtoChunk)this.f_191814_.m_6325_(p_191822_.f_45578_, p_191822_.f_45579_)).m_183613_(p_191823_);
+      return ((ProtoChunk)this.f_191814_.getChunk(p_191822_.x(), p_191822_.z())).m_183613_(p_191823_);
    }
 
    public BlockState m_191828_(BlockPos p_191829_) {
       return this.f_191814_.m_8055_(p_191829_);
    }
 
-   public int m_191830_() {
-      return this.f_191814_.m_141937_();
+   public long getMinBuildHeight() {
+      return this.f_191814_.getMinBuildHeight();
    }
 
    public WorldGenLevel m_191831_() {
