--- a/net/minecraft/world/level/levelgen/placement/RandomOffsetPlacement.java
+++ b/net/minecraft/world/level/levelgen/placement/RandomOffsetPlacement.java
@@ -37,9 +_,9 @@
    }
 
    public Stream<BlockPos> m_213676_(PlacementContext p_226393_, RandomSource p_226394_, BlockPos p_226395_) {
-      int i = p_226395_.m_123341_() + this.f_191871_.m_214085_(p_226394_);
-      int j = p_226395_.m_123342_() + this.f_191872_.m_214085_(p_226394_);
-      int k = p_226395_.m_123343_() + this.f_191871_.m_214085_(p_226394_);
+      me.alphamode.mclong.math.BigInteger i = p_226395_.getBigX().add(this.f_191871_.m_214085_(p_226394_));
+      me.alphamode.mclong.math.BigInteger j = p_226395_.getBigY().add(this.f_191872_.m_214085_(p_226394_));
+      me.alphamode.mclong.math.BigInteger k = p_226395_.getBigZ().add(this.f_191871_.m_214085_(p_226394_));
       return Stream.of(new BlockPos(i, j, k));
    }
 
