--- a/net/minecraft/world/level/levelgen/placement/SurfaceWaterDepthFilter.java
+++ b/net/minecraft/world/level/levelgen/placement/SurfaceWaterDepthFilter.java
@@ -23,8 +_,8 @@
    }
 
    protected boolean m_213917_(PlacementContext p_226411_, RandomSource p_226412_, BlockPos p_226413_) {
-      int i = p_226411_.m_191824_(Heightmap.Types.OCEAN_FLOOR, p_226413_.m_123341_(), p_226413_.m_123343_());
-      int j = p_226411_.m_191824_(Heightmap.Types.WORLD_SURFACE, p_226413_.m_123341_(), p_226413_.m_123343_());
+      long i = p_226411_.getHeight(Heightmap.Types.OCEAN_FLOOR, p_226413_.getBigX(), p_226413_.getBigZ());
+      long j = p_226411_.getHeight(Heightmap.Types.WORLD_SURFACE, p_226413_.getBigX(), p_226413_.getBigZ());
       return j - i <= this.f_191946_;
    }
 
