--- a/net/minecraft/client/renderer/entity/layers/ArrowLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/ArrowLayer.java
@@ -28,7 +_,7 @@
 
    protected void m_5558_(PoseStack p_116569_, MultiBufferSource p_116570_, int p_116571_, Entity p_116572_, float p_116573_, float p_116574_, float p_116575_, float p_116576_) {
       float f = Mth.m_14116_(p_116573_ * p_116573_ + p_116575_ * p_116575_);
-      Arrow arrow = new Arrow(p_116572_.m_9236_(), p_116572_.m_20185_(), p_116572_.m_20186_(), p_116572_.m_20189_());
+      Arrow arrow = new Arrow(p_116572_.m_9236_(), p_116572_.getBigX(), p_116572_.getBigY(), p_116572_.getBigZ());
       arrow.m_146922_((float)(Math.atan2((double)p_116573_, (double)p_116575_) * (double)(180F / (float)Math.PI)));
       arrow.m_146926_((float)(Math.atan2((double)p_116574_, (double)f) * (double)(180F / (float)Math.PI)));
       arrow.f_19859_ = arrow.m_146908_();
