--- a/net/minecraft/world/entity/ai/navigation/AmphibiousPathNavigation.java
+++ b/net/minecraft/world/entity/ai/navigation/AmphibiousPathNavigation.java
@@ -26,8 +_,8 @@
       return new Vec3(this.f_26494_.m_20185_(), this.f_26494_.m_20227_(0.5D), this.f_26494_.m_20189_());
    }
 
-   protected double m_183345_(Vec3 p_217794_) {
-      return p_217794_.f_82480_;
+   protected me.alphamode.mclong.math.BigDecimal getGroundY(Vec3 p_217794_) {
+      return p_217794_.getY();
    }
 
    protected boolean m_183431_(Vec3 p_217796_, Vec3 p_217797_) {
@@ -35,7 +_,7 @@
    }
 
    public boolean m_6342_(BlockPos p_217799_) {
-      return !this.f_26495_.m_8055_(p_217799_.m_7495_()).m_60795_();
+      return !this.f_26495_.m_8055_(p_217799_.below()).m_60795_();
    }
 
    public void m_7008_(boolean p_217801_) {
