--- a/net/minecraft/world/entity/ai/navigation/FlyingPathNavigation.java
+++ b/net/minecraft/world/entity/ai/navigation/FlyingPathNavigation.java
@@ -49,7 +_,7 @@
             this.m_7636_();
          } else if (this.f_26496_ != null && !this.f_26496_.m_77392_()) {
             Vec3 vec3 = this.f_26496_.m_77380_(this.f_26494_);
-            if (this.f_26494_.m_146903_() == Mth.m_14107_(vec3.f_82479_) && this.f_26494_.m_146904_() == Mth.m_14107_(vec3.f_82480_) && this.f_26494_.m_146907_() == Mth.m_14107_(vec3.f_82481_)) {
+            if (this.f_26494_.getBlockX() == Mth.m_14107_(vec3.m_7096_()) && this.f_26494_.getBlockY() == Mth.m_14107_(vec3.m_7098_()) && this.f_26494_.getBlockZ() == Mth.m_14107_(vec3.m_7094_())) {
                this.f_26496_.m_77374_();
             }
          }
@@ -57,7 +_,7 @@
          DebugPackets.m_133703_(this.f_26495_, this.f_26494_, this.f_26496_, this.f_26505_);
          if (!this.m_26571_()) {
             Vec3 vec31 = this.f_26496_.m_77380_(this.f_26494_);
-            this.f_26494_.m_21566_().m_6849_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, this.f_26497_);
+            this.f_26494_.m_21566_().m_6849_(vec31.m_7096_(), vec31.m_7098_(), vec31.m_7094_(), this.f_26497_);
          }
       }
    }
