--- a/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
+++ b/net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
@@ -41,8 +_,8 @@
          super.m_7638_();
       } else {
          if (this.f_26578_ != null) {
-            if (!this.f_26578_.m_203195_(this.f_26494_.m_20182_(), (double)this.f_26494_.m_20205_()) && (!(this.f_26494_.m_20186_() > (double)this.f_26578_.m_123342_()) || !(new BlockPos((double)this.f_26578_.m_123341_(), this.f_26494_.m_20186_(), (double)this.f_26578_.m_123343_())).m_203195_(this.f_26494_.m_20182_(), (double)this.f_26494_.m_20205_()))) {
-               this.f_26494_.m_21566_().m_6849_((double)this.f_26578_.m_123341_(), (double)this.f_26578_.m_123342_(), (double)this.f_26578_.m_123343_(), this.f_26497_);
+            if (!this.f_26578_.closerToCenterThan(this.f_26494_.m_20182_(), (double)this.f_26494_.m_20205_()) && (!(this.f_26494_.m_20186_() > (double)this.f_26578_.getY()) || !(new BlockPos((double)this.f_26578_.getX(), this.f_26494_.m_20186_(), (double)this.f_26578_.getZ())).closerToCenterThan(this.f_26494_.m_20182_(), (double)this.f_26494_.m_20205_()))) {
+               this.f_26494_.m_21566_().m_6849_((double)this.f_26578_.getX(), (double)this.f_26578_.getY(), (double)this.f_26578_.getZ(), this.f_26497_);
             } else {
                this.f_26578_ = null;
             }
