--- a/net/minecraft/world/entity/ai/navigation/WaterBoundPathNavigation.java
+++ b/net/minecraft/world/entity/ai/navigation/WaterBoundPathNavigation.java
@@ -26,11 +_,11 @@
    }
 
    protected Vec3 m_7475_() {
-      return new Vec3(this.f_26494_.m_20185_(), this.f_26494_.m_20227_(0.5D), this.f_26494_.m_20189_());
+      return new Vec3(this.f_26494_.getBigX(), this.f_26494_.getBigY(0.5D), this.f_26494_.getBigZ());
    }
 
-   protected double m_183345_(Vec3 p_186136_) {
-      return p_186136_.f_82480_;
+   protected me.alphamode.mclong.math.BigDecimal getGroundY(Vec3 p_186136_) {
+      return p_186136_.getY();
    }
 
    protected boolean m_183431_(Vec3 p_186138_, Vec3 p_186139_) {
