--- a/net/minecraft/world/entity/raid/Raids.java
+++ b/net/minecraft/world/entity/raid/Raids.java
@@ -95,7 +_,7 @@
 
             for(PoiRecord poirecord : list) {
                BlockPos blockpos2 = poirecord.m_27257_();
-               vec3 = vec3.m_82520_((double)blockpos2.m_123341_(), (double)blockpos2.m_123342_(), (double)blockpos2.m_123343_());
+               vec3 = vec3.m_82520_((double)blockpos2.getX(), (double)blockpos2.getY(), (double)blockpos2.getZ());
                ++i;
             }
 
@@ -186,7 +_,7 @@
       double d0 = (double)p_37972_;
 
       for(Raid raid1 : this.f_37951_.values()) {
-         double d1 = raid1.m_37780_().m_123331_(p_37971_);
+         double d1 = raid1.m_37780_().distSqr(p_37971_);
          if (raid1.m_37782_() && d1 < d0) {
             raid = raid1;
             d0 = d1;
