--- a/net/minecraft/world/level/saveddata/maps/MapBanner.java
+++ b/net/minecraft/world/level/saveddata/maps/MapBanner.java
@@ -120,6 +_,6 @@
    }
 
    public String m_77787_() {
-      return "banner-" + this.f_77766_.m_123341_() + "," + this.f_77766_.m_123342_() + "," + this.f_77766_.m_123343_();
+      return "banner-" + this.f_77766_.getX() + "," + this.f_77766_.getY() + "," + this.f_77766_.getZ();
    }
 }
