--- a/net/minecraft/world/level/saveddata/maps/MapFrame.java
+++ b/net/minecraft/world/level/saveddata/maps/MapFrame.java
@@ -47,6 +_,6 @@
    }
 
    public static String m_77870_(BlockPos p_77871_) {
-      return "frame-" + p_77871_.m_123341_() + "," + p_77871_.m_123342_() + "," + p_77871_.m_123343_();
+      return "frame-" + p_77871_.getX() + "," + p_77871_.getY() + "," + p_77871_.getZ();
    }
 }
