--- a/net/minecraft/client/renderer/BiomeColors.java
+++ b/net/minecraft/client/renderer/BiomeColors.java
@@ -9,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class BiomeColors {
-   public static final ColorResolver f_108789_ = Biome::m_47464_;
+   public static final ColorResolver f_108789_ = Biome::getGrassColor;
    public static final ColorResolver f_108790_ = (p_108808_, p_108809_, p_108810_) -> {
       return p_108808_.m_47542_();
    };
