--- a/net/minecraft/client/renderer/DimensionSpecialEffects.java
+++ b/net/minecraft/client/renderer/DimensionSpecialEffects.java
@@ -69,7 +_,7 @@
 
    public abstract Vec3 m_5927_(Vec3 p_108878_, float p_108879_);
 
-   public abstract boolean m_5781_(int p_108874_, int p_108875_);
+   public abstract boolean isFoggyAt(me.alphamode.mclong.math.BigInteger p_108874_, me.alphamode.mclong.math.BigInteger p_108875_);
 
    public DimensionSpecialEffects.SkyType m_108883_() {
       return this.f_108861_;
@@ -93,7 +_,7 @@
          return p_108894_.m_82490_((double)0.15F);
       }
 
-      public boolean m_5781_(int p_108891_, int p_108892_) {
+      public boolean isFoggyAt(me.alphamode.mclong.math.BigInteger p_108891_, me.alphamode.mclong.math.BigInteger p_108892_) {
          return false;
       }
 
@@ -113,7 +_,7 @@
          return p_108901_;
       }
 
-      public boolean m_5781_(int p_108898_, int p_108899_) {
+      public boolean isFoggyAt(me.alphamode.mclong.math.BigInteger p_108898_, me.alphamode.mclong.math.BigInteger p_108899_) {
          return true;
       }
    }
@@ -130,7 +_,7 @@
          return p_108908_.m_82542_((double)(p_108909_ * 0.94F + 0.06F), (double)(p_108909_ * 0.94F + 0.06F), (double)(p_108909_ * 0.91F + 0.09F));
       }
 
-      public boolean m_5781_(int p_108905_, int p_108906_) {
+      public boolean isFoggyAt(me.alphamode.mclong.math.BigInteger p_108905_, me.alphamode.mclong.math.BigInteger p_108906_) {
          return false;
       }
    }
