--- a/net/minecraft/client/renderer/LevelRenderer.java
+++ b/net/minecraft/client/renderer/LevelRenderer.java
@@ -189,7 +_,7 @@
    private final RunningTrimmedMean f_109476_ = new RunningTrimmedMean(100);
    private int f_109477_;
    private final Int2ObjectMap<BlockDestructionProgress> f_109408_ = new Int2ObjectOpenHashMap<>();
-   private final Long2ObjectMap<SortedSet<BlockDestructionProgress>> f_109409_ = new Long2ObjectOpenHashMap<>();
+   private final it.unimi.dsi.fastutil.objects.Object2ObjectMap<BlockPos, SortedSet<BlockDestructionProgress>> f_109409_ = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
    private final Map<BlockPos, SoundInstance> f_109410_ = Maps.newHashMap();
    @Nullable
    private RenderTarget f_109411_;
@@ -210,17 +_,17 @@
    private double f_109419_ = Double.MIN_VALUE;
    private double f_109420_ = Double.MIN_VALUE;
    private double f_109421_ = Double.MIN_VALUE;
-   private int f_109422_ = Integer.MIN_VALUE;
-   private int f_109423_ = Integer.MIN_VALUE;
-   private int f_109424_ = Integer.MIN_VALUE;
+   private long f_109422_ = java.lang.Long.MIN_VALUE;
+   private long f_109423_ = java.lang.Long.MIN_VALUE;
+   private long f_109424_ = java.lang.Long.MIN_VALUE;
    private double f_109425_ = Double.MIN_VALUE;
    private double f_109426_ = Double.MIN_VALUE;
    private double f_109427_ = Double.MIN_VALUE;
    private double f_109428_ = Double.MIN_VALUE;
    private double f_109429_ = Double.MIN_VALUE;
-   private int f_109430_ = Integer.MIN_VALUE;
-   private int f_109431_ = Integer.MIN_VALUE;
-   private int f_109432_ = Integer.MIN_VALUE;
+   private long f_109430_ = java.lang.Long.MIN_VALUE;
+   private long f_109431_ = java.lang.Long.MIN_VALUE;
+   private long f_109432_ = java.lang.Long.MIN_VALUE;
    private Vec3 f_109433_ = Vec3.f_82478_;
    @Nullable
    private CloudStatus f_109435_;
@@ -271,9 +_,9 @@
       if (!(f <= 0.0F)) {
          p_109704_.m_109896_();
          Level level = this.f_109461_.f_91073_;
-         int i = Mth.m_14107_(p_109706_);
-         int j = Mth.m_14107_(p_109707_);
-         int k = Mth.m_14107_(p_109708_);
+         long i = Mth.m_14134_(p_109706_);
+         long j = Mth.m_14134_(p_109707_);
+         long k = Mth.m_14134_(p_109708_);
          Tesselator tesselator = Tesselator.m_85913_();
          BufferBuilder bufferbuilder = tesselator.m_85915_();
          RenderSystem.disableCull();
@@ -290,17 +_,18 @@
          RenderSystem.setShader(GameRenderer::m_172829_);
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-         for(int j1 = k - l; j1 <= k + l; ++j1) {
-            for(int k1 = i - l; k1 <= i + l; ++k1) {
-               int l1 = (j1 - k + 16) * 32 + k1 - i + 16;
-               double d0 = (double)this.f_109451_[l1] * 0.5D;
-               double d1 = (double)this.f_109452_[l1] * 0.5D;
+         for(long j1 = k - l; j1 <= k + l; ++j1) {
+            for(long y = j - l; y <= j + l; ++y) {
+            for(long k1 = i - l; k1 <= i + l; ++k1) {
+               long l1 = (j1 - k + 16) * 32 + k1 - i + 16;
+               double d0 = (double)this.f_109451_[(int) l1] * 0.5D;
+               double d1 = (double)this.f_109452_[(int) l1] * 0.5D;
                blockpos$mutableblockpos.m_122169_((double)k1, p_109707_, (double)j1);
                Biome biome = level.m_204166_(blockpos$mutableblockpos).m_203334_();
                if (biome.m_264473_()) {
-                  int i2 = level.m_6924_(Heightmap.Types.MOTION_BLOCKING, k1, j1);
-                  int j2 = j - l;
-                  int k2 = j + l;
+                  long i2 = level.getHeight(Heightmap.Types.MOTION_BLOCKING, k1, y, j1);
+                  long j2 = j - l;
+                  long k2 = j + l;
                   if (j2 < i2) {
                      j2 = i2;
                   }
@@ -309,14 +_,14 @@
                      k2 = i2;
                   }
 
-                  int l2 = i2;
+                  long l2 = i2;
                   if (i2 < j) {
                      l2 = j;
                   }
 
                   if (j2 != k2) {
                      RandomSource randomsource = RandomSource.m_216335_((long)(k1 * k1 * 3121 + k1 * 45238971 ^ j1 * j1 * 418711 + j1 * 13761));
-                     blockpos$mutableblockpos.m_122178_(k1, j2, j1);
+                     blockpos$mutableblockpos.set(k1, j2, j1);
                      Biome.Precipitation biome$precipitation = biome.m_264600_(blockpos$mutableblockpos);
                      if (biome$precipitation == Biome.Precipitation.RAIN) {
                         if (i1 != 0) {
@@ -329,13 +_,13 @@
                            bufferbuilder.m_166779_(VertexFormat.Mode.QUADS, DefaultVertexFormat.f_85813_);
                         }
 
-                        int i3 = this.f_109477_ + k1 * k1 * 3121 + k1 * 45238971 + j1 * j1 * 418711 + j1 * 13761 & 31;
+                        long i3 = this.f_109477_ + k1 * k1 * 3121 + k1 * 45238971 + j1 * j1 * 418711 + j1 * 13761 & 31;
                         float f2 = -((float)i3 + p_109705_) / 32.0F * (3.0F + randomsource.m_188501_());
                         double d2 = (double)k1 + 0.5D - p_109706_;
                         double d4 = (double)j1 + 0.5D - p_109708_;
                         float f3 = (float)Math.sqrt(d2 * d2 + d4 * d4) / (float)l;
                         float f4 = ((1.0F - f3 * f3) * 0.5F + 0.5F) * f;
-                        blockpos$mutableblockpos.m_122178_(k1, l2, j1);
+                        blockpos$mutableblockpos.set(k1, l2, j1);
                         int j3 = m_109541_(level, blockpos$mutableblockpos);
                         bufferbuilder.m_5483_((double)k1 - p_109706_ - d0 + 0.5D, (double)k2 - p_109707_, (double)j1 - p_109708_ - d1 + 0.5D).m_7421_(0.0F, (float)j2 * 0.25F + f2).m_85950_(1.0F, 1.0F, 1.0F, f4).m_85969_(j3).m_5752_();
                         bufferbuilder.m_5483_((double)k1 - p_109706_ + d0 + 0.5D, (double)k2 - p_109707_, (double)j1 - p_109708_ + d1 + 0.5D).m_7421_(1.0F, (float)j2 * 0.25F + f2).m_85950_(1.0F, 1.0F, 1.0F, f4).m_85969_(j3).m_5752_();
@@ -359,7 +_,7 @@
                         double d5 = (double)j1 + 0.5D - p_109708_;
                         float f8 = (float)Math.sqrt(d3 * d3 + d5 * d5) / (float)l;
                         float f9 = ((1.0F - f8 * f8) * 0.3F + 0.5F) * f;
-                        blockpos$mutableblockpos.m_122178_(k1, l2, j1);
+                        blockpos$mutableblockpos.set(k1, l2, j1);
                         int k3 = m_109541_(level, blockpos$mutableblockpos);
                         int l3 = k3 >> 16 & '\uffff';
                         int i4 = k3 & '\uffff';
@@ -372,6 +_,7 @@
                      }
                   }
                }
+               }
             }
          }
 
@@ -397,11 +_,11 @@
          for(int j = 0; j < i; ++j) {
             int k = randomsource.m_188503_(21) - 10;
             int l = randomsource.m_188503_(21) - 10;
-            BlockPos blockpos2 = levelreader.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos.m_7918_(k, 0, l));
-            if (blockpos2.m_123342_() > levelreader.m_141937_() && blockpos2.m_123342_() <= blockpos.m_123342_() + 10 && blockpos2.m_123342_() >= blockpos.m_123342_() - 10) {
+            BlockPos blockpos2 = levelreader.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos.offset(k, 0, l));
+            if (blockpos2.getY() > levelreader.getMinBuildHeight() && blockpos2.getY() <= blockpos.getY() + 10 && blockpos2.getY() >= blockpos.getY() - 10) {
                Biome biome = levelreader.m_204166_(blockpos2).m_203334_();
                if (biome.m_264600_(blockpos2) == Biome.Precipitation.RAIN) {
-                  blockpos1 = blockpos2.m_7495_();
+                  blockpos1 = blockpos2.below();
                   if (this.f_109461_.f_91066_.m_231929_().m_231551_() == ParticleStatus.MINIMAL) {
                      break;
                   }
@@ -415,14 +_,14 @@
                   double d3 = (double)fluidstate.m_76155_(levelreader, blockpos1);
                   double d4 = Math.max(d2, d3);
                   ParticleOptions particleoptions = !fluidstate.m_205070_(FluidTags.f_13132_) && !blockstate.m_60713_(Blocks.f_50450_) && !CampfireBlock.m_51319_(blockstate) ? ParticleTypes.f_123761_ : ParticleTypes.f_123762_;
-                  this.f_109461_.f_91073_.m_7106_(particleoptions, (double)blockpos1.m_123341_() + d0, (double)blockpos1.m_123342_() + d4, (double)blockpos1.m_123343_() + d1, 0.0D, 0.0D, 0.0D);
+                  this.f_109461_.f_91073_.m_7106_(particleoptions, (double)blockpos1.getX() + d0, (double)blockpos1.getY() + d4, (double)blockpos1.getZ() + d1, 0.0D, 0.0D, 0.0D);
                }
             }
          }
 
          if (blockpos1 != null && randomsource.m_188503_(3) < this.f_109450_++) {
             this.f_109450_ = 0;
-            if (blockpos1.m_123342_() > blockpos.m_123342_() + 1 && levelreader.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos).m_123342_() > Mth.m_14143_((float)blockpos.m_123342_())) {
+            if (blockpos1.getY() > blockpos.getY() + 1 && levelreader.m_5452_(Heightmap.Types.MOTION_BLOCKING, blockpos).getY() > Mth.m_14143_((float)blockpos.getY())) {
                this.f_109461_.f_91073_.m_245747_(blockpos1, SoundEvents.f_12542_, SoundSource.WEATHER, 0.1F, 0.5F, false);
             } else {
                this.f_109461_.f_91073_.m_245747_(blockpos1, SoundEvents.f_12541_, SoundSource.WEATHER, 0.2F, 1.0F, false);
@@ -655,9 +_,9 @@
       this.f_109419_ = Double.MIN_VALUE;
       this.f_109420_ = Double.MIN_VALUE;
       this.f_109421_ = Double.MIN_VALUE;
-      this.f_109422_ = Integer.MIN_VALUE;
-      this.f_109423_ = Integer.MIN_VALUE;
-      this.f_109424_ = Integer.MIN_VALUE;
+      this.f_109422_ = java.lang.Long.MIN_VALUE;
+      this.f_109423_ = java.lang.Long.MIN_VALUE;
+      this.f_109424_ = java.lang.Long.MIN_VALUE;
       this.f_109463_.m_114406_(p_109702_);
       this.f_109465_ = p_109702_;
       if (p_109702_ != null) {
@@ -789,9 +_,9 @@
       double d0 = this.f_109461_.f_91074_.m_20185_();
       double d1 = this.f_109461_.f_91074_.m_20186_();
       double d2 = this.f_109461_.f_91074_.m_20189_();
-      int i = SectionPos.m_175552_(d0);
-      int j = SectionPos.m_175552_(d1);
-      int k = SectionPos.m_175552_(d2);
+      long i = SectionPos.posToSectionCoord(d0);
+      long j = SectionPos.posToSectionCoord(d1);
+      long k = SectionPos.posToSectionCoord(d2);
       if (this.f_109422_ != i || this.f_109423_ != j || this.f_109424_ != k) {
          this.f_109419_ = d0;
          this.f_109420_ = d1;
@@ -896,15 +_,15 @@
       BlockPos blockpos = p_194344_.m_90588_();
       ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk = this.f_109469_.m_110866_(blockpos);
       if (chunkrenderdispatcher$renderchunk == null) {
-         boolean flag = blockpos.m_123342_() > this.f_109465_.m_141937_();
-         int j = flag ? this.f_109465_.m_151558_() - 8 : this.f_109465_.m_141937_() + 8;
-         int k = Mth.m_14107_(vec3.f_82479_ / 16.0D) * 16;
-         int l = Mth.m_14107_(vec3.f_82481_ / 16.0D) * 16;
+         boolean flag = blockpos.getY() > this.f_109465_.getMinBuildHeight();
+         long j = flag ? this.f_109465_.getMaxBuildHeight() - 8 : this.f_109465_.getMinBuildHeight() + 8;
+         long k = Mth.m_14134_(vec3.f_82479_ / 16.0D) * 16;
+         long l = Mth.m_14134_(vec3.f_82481_ / 16.0D) * 16;
          List<LevelRenderer.RenderChunkInfo> list = Lists.newArrayList();
 
          for(int i1 = -this.f_109438_; i1 <= this.f_109438_; ++i1) {
             for(int j1 = -this.f_109438_; j1 <= this.f_109438_; ++j1) {
-               ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk1 = this.f_109469_.m_110866_(new BlockPos(k + SectionPos.m_175554_(i1, 8), j, l + SectionPos.m_175554_(j1, 8)));
+               ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk1 = this.f_109469_.m_110866_(new BlockPos(k + SectionPos.sectionToBlockCoord(i1, 8), j, l + SectionPos.sectionToBlockCoord(j1, 8)));
                if (chunkrenderdispatcher$renderchunk1 != null) {
                   list.add(new LevelRenderer.RenderChunkInfo(chunkrenderdispatcher$renderchunk1, (Direction)null, 0));
                }
@@ -912,7 +_,7 @@
          }
 
          list.sort(Comparator.comparingDouble((p_234303_) -> {
-            return blockpos.m_123331_(p_234303_.f_109839_.m_112839_().m_7918_(8, 8, 8));
+            return blockpos.distSqr(p_234303_.f_109839_.m_112839_().offset(8, 8, 8));
          }));
          p_194345_.addAll(list);
       } else {
@@ -927,15 +_,15 @@
 
    private void m_194362_(LinkedHashSet<LevelRenderer.RenderChunkInfo> p_194363_, LevelRenderer.RenderInfoMap p_194364_, Vec3 p_194365_, Queue<LevelRenderer.RenderChunkInfo> p_194366_, boolean p_194367_) {
       int i = 16;
-      BlockPos blockpos = new BlockPos(Mth.m_14107_(p_194365_.f_82479_ / 16.0D) * 16, Mth.m_14107_(p_194365_.f_82480_ / 16.0D) * 16, Mth.m_14107_(p_194365_.f_82481_ / 16.0D) * 16);
-      BlockPos blockpos1 = blockpos.m_7918_(8, 8, 8);
+      BlockPos blockpos = new BlockPos(Mth.m_14134_(p_194365_.f_82479_ / 16.0D) * 16, Mth.m_14134_(p_194365_.f_82480_ / 16.0D) * 16, Mth.m_14134_(p_194365_.f_82481_ / 16.0D) * 16);
+      BlockPos blockpos1 = blockpos.offset(8, 8, 8);
       Entity.m_20103_(Mth.m_14008_((double)this.f_109461_.f_91066_.m_193772_() / 8.0D, 1.0D, 2.5D) * this.f_109461_.f_91066_.m_232018_().m_231551_());
 
       while(!p_194366_.isEmpty()) {
          LevelRenderer.RenderChunkInfo levelrenderer$renderchunkinfo = p_194366_.poll();
          ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk = levelrenderer$renderchunkinfo.f_109839_;
          p_194363_.add(levelrenderer$renderchunkinfo);
-         boolean flag = Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().m_123341_() - blockpos.m_123341_()) > 60 || Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().m_123342_() - blockpos.m_123342_()) > 60 || Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().m_123343_() - blockpos.m_123343_()) > 60;
+         boolean flag = Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().getX() - blockpos.getX()) > 60 || Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().getY() - blockpos.getY()) > 60 || Math.abs(chunkrenderdispatcher$renderchunk.m_112839_().getZ() - blockpos.getZ()) > 60;
 
          for(Direction direction : f_109434_) {
             ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk1 = this.m_109728_(blockpos, chunkrenderdispatcher$renderchunk, direction);
@@ -963,10 +_,10 @@
                      label125: {
                         blockpos2 = chunkrenderdispatcher$renderchunk1.m_112839_();
                         if (direction.m_122434_() == Direction.Axis.X) {
-                           if (blockpos1.m_123341_() > blockpos2.m_123341_()) {
+                           if (blockpos1.getX() > blockpos2.getX()) {
                               break label125;
                            }
-                        } else if (blockpos1.m_123341_() < blockpos2.m_123341_()) {
+                        } else if (blockpos1.getX() < blockpos2.getX()) {
                            break label125;
                         }
 
@@ -981,10 +_,10 @@
                   label118: {
                      label117: {
                         if (direction.m_122434_() == Direction.Axis.Y) {
-                           if (blockpos1.m_123342_() > blockpos2.m_123342_()) {
+                           if (blockpos1.getY() > blockpos2.getY()) {
                               break label117;
                            }
-                        } else if (blockpos1.m_123342_() < blockpos2.m_123342_()) {
+                        } else if (blockpos1.getY() < blockpos2.getY()) {
                            break label117;
                         }
 
@@ -999,10 +_,10 @@
                   label110: {
                      label109: {
                         if (direction.m_122434_() == Direction.Axis.Z) {
-                           if (blockpos1.m_123343_() > blockpos2.m_123343_()) {
+                           if (blockpos1.getZ() > blockpos2.getZ()) {
                               break label109;
                            }
-                        } else if (blockpos1.m_123343_() < blockpos2.m_123343_()) {
+                        } else if (blockpos1.getZ() < blockpos2.getZ()) {
                            break label109;
                         }
 
@@ -1013,14 +_,14 @@
                      b2 = 16;
                   }
 
-                  BlockPos blockpos3 = blockpos2.m_7918_(b0, b1, b2);
-                  Vec3 vec31 = new Vec3((double)blockpos3.m_123341_(), (double)blockpos3.m_123342_(), (double)blockpos3.m_123343_());
+                  BlockPos blockpos3 = blockpos2.offset(b0, b1, b2);
+                  Vec3 vec31 = new Vec3((double)blockpos3.getX(), (double)blockpos3.getY(), (double)blockpos3.getZ());
                   Vec3 vec3 = p_194365_.m_82546_(vec31).m_82541_().m_82490_(f_194304_);
                   boolean flag2 = true;
 
                   while(p_194365_.m_82546_(vec31).m_82556_() > 3600.0D) {
                      vec31 = vec31.m_82549_(vec3);
-                     if (vec31.f_82480_ > (double)this.f_109465_.m_151558_() || vec31.f_82480_ < (double)this.f_109465_.m_141937_()) {
+                     if (vec31.f_82480_ > (double)this.f_109465_.getMaxBuildHeight() || vec31.f_82480_ < (double)this.f_109465_.getMinBuildHeight()) {
                         break;
                      }
 
@@ -1058,22 +_,22 @@
    @Nullable
    private ChunkRenderDispatcher.RenderChunk m_109728_(BlockPos p_109729_, ChunkRenderDispatcher.RenderChunk p_109730_, Direction p_109731_) {
       BlockPos blockpos = p_109730_.m_112824_(p_109731_);
-      if (Mth.m_14040_(p_109729_.m_123341_() - blockpos.m_123341_()) > this.f_109438_ * 16) {
+      if (Mth.abs(p_109729_.getX() - blockpos.getX()) > this.f_109438_ * 16) {
          return null;
-      } else if (Mth.m_14040_(p_109729_.m_123342_() - blockpos.m_123342_()) <= this.f_109438_ * 16 && blockpos.m_123342_() >= this.f_109465_.m_141937_() && blockpos.m_123342_() < this.f_109465_.m_151558_()) {
-         return Mth.m_14040_(p_109729_.m_123343_() - blockpos.m_123343_()) > this.f_109438_ * 16 ? null : this.f_109469_.m_110866_(blockpos);
+      } else if (Mth.abs(p_109729_.getY() - blockpos.getY()) <= this.f_109438_ * 16 && blockpos.getY() >= this.f_109465_.getMinBuildHeight() && blockpos.getY() < this.f_109465_.getMaxBuildHeight()) {
+         return Mth.abs(p_109729_.getZ() - blockpos.getZ()) > this.f_109438_ * 16 ? null : this.f_109469_.m_110866_(blockpos);
       } else {
          return null;
       }
    }
 
    private boolean m_194359_(BlockPos p_194360_, ChunkRenderDispatcher.RenderChunk p_194361_) {
-      int i = SectionPos.m_123171_(p_194360_.m_123341_());
-      int j = SectionPos.m_123171_(p_194360_.m_123343_());
+      long i = SectionPos.blockToSectionCoord(p_194360_.getX());
+      long j = SectionPos.blockToSectionCoord(p_194360_.getZ());
       BlockPos blockpos = p_194361_.m_112839_();
-      int k = SectionPos.m_123171_(blockpos.m_123341_());
-      int l = SectionPos.m_123171_(blockpos.m_123343_());
-      return !ChunkMap.m_200878_(k, l, i, j, this.f_109438_ - 3);
+      long k = SectionPos.blockToSectionCoord(blockpos.getX());
+      long l = SectionPos.blockToSectionCoord(blockpos.getZ());
+      return !ChunkMap.isChunkInRange(k, SectionPos.blockToSectionCoord(blockpos.getY()), l, i, SectionPos.blockToSectionCoord(p_194360_.getY()), j, this.f_109438_ - 2);
    }
 
    private void m_252964_(Matrix4f p_253756_, Matrix4f p_253787_, double p_254187_, double p_253833_, double p_254547_, Frustum p_253954_) {
@@ -1144,7 +_,7 @@
       FogRenderer.m_109036_();
       RenderSystem.clear(16640, Minecraft.f_91002_);
       float f = p_109605_.m_109152_();
-      boolean flag1 = this.f_109461_.f_91073_.m_104583_().m_5781_(Mth.m_14107_(d0), Mth.m_14107_(d1)) || this.f_109461_.f_91065_.m_93090_().m_93715_();
+      boolean flag1 = this.f_109461_.f_91073_.m_104583_().isFoggyAt(Mth.m_14134_(d0), Mth.m_14134_(d1)) || this.f_109461_.f_91065_.m_93090_().m_93715_();
       profilerfiller.m_6182_("sky");
       RenderSystem.setShader(GameRenderer::m_172808_);
       this.m_202423_(p_109600_, p_254120_, p_109601_, p_109604_, flag1, () -> {
@@ -1190,7 +_,7 @@
       for(Entity entity : this.f_109465_.m_104735_()) {
          if (this.f_109463_.m_114397_(entity, frustum, d0, d1, d2) || entity.m_20367_(this.f_109461_.f_91074_)) {
             BlockPos blockpos = entity.m_20183_();
-            if ((this.f_109465_.m_151562_(blockpos.m_123342_()) || this.m_202430_(blockpos)) && (entity != p_109604_.m_90592_() || p_109604_.m_90594_() || p_109604_.m_90592_() instanceof LivingEntity && ((LivingEntity)p_109604_.m_90592_()).m_5803_()) && (!(entity instanceof LocalPlayer) || p_109604_.m_90592_() == entity)) {
+            if ((this.f_109465_.isOutsideBuildHeight(blockpos.getY()) || this.m_202430_(blockpos)) && (entity != p_109604_.m_90592_() || p_109604_.m_90594_() || p_109604_.m_90592_() instanceof LivingEntity && ((LivingEntity)p_109604_.m_90592_()).m_5803_()) && (!(entity instanceof LocalPlayer) || p_109604_.m_90592_() == entity)) {
                ++this.f_109439_;
                if (entity.f_19797_ == 0) {
                   entity.f_19790_ = entity.m_20185_();
@@ -1229,8 +_,8 @@
                BlockPos blockpos4 = blockentity1.m_58899_();
                MultiBufferSource multibuffersource1 = multibuffersource$buffersource;
                p_109600_.m_85836_();
-               p_109600_.m_85837_((double)blockpos4.m_123341_() - d0, (double)blockpos4.m_123342_() - d1, (double)blockpos4.m_123343_() - d2);
-               SortedSet<BlockDestructionProgress> sortedset = this.f_109409_.get(blockpos4.m_121878_());
+               p_109600_.m_85837_((double)blockpos4.getX() - d0, (double)blockpos4.getY() - d1, (double)blockpos4.getZ() - d2);
+               SortedSet<BlockDestructionProgress> sortedset = this.f_109409_.get(blockpos4);
                if (sortedset != null && !sortedset.isEmpty()) {
                   int j = sortedset.last().m_139988_();
                   if (j >= 0) {
@@ -1253,7 +_,7 @@
          for(BlockEntity blockentity : this.f_109468_) {
             BlockPos blockpos3 = blockentity.m_58899_();
             p_109600_.m_85836_();
-            p_109600_.m_85837_((double)blockpos3.m_123341_() - d0, (double)blockpos3.m_123342_() - d1, (double)blockpos3.m_123343_() - d2);
+            p_109600_.m_85837_((double)blockpos3.getX() - d0, (double)blockpos3.getY() - d1, (double)blockpos3.getZ() - d2);
             this.f_172946_.m_112267_(blockentity, p_109601_, p_109600_, multibuffersource$buffersource);
             p_109600_.m_85849_();
          }
@@ -1278,17 +_,17 @@
 
       profilerfiller.m_6182_("destroyProgress");
 
-      for(Long2ObjectMap.Entry<SortedSet<BlockDestructionProgress>> entry : this.f_109409_.long2ObjectEntrySet()) {
-         BlockPos blockpos2 = BlockPos.m_122022_(entry.getLongKey());
-         double d3 = (double)blockpos2.m_123341_() - d0;
-         double d4 = (double)blockpos2.m_123342_() - d1;
-         double d5 = (double)blockpos2.m_123343_() - d2;
+      for(it.unimi.dsi.fastutil.objects.Object2ObjectMap.Entry<BlockPos, SortedSet<BlockDestructionProgress>> entry : this.f_109409_.object2ObjectEntrySet()) {
+         BlockPos blockpos2 = entry.getKey();
+         double d3 = (double)blockpos2.getX() - d0;
+         double d4 = (double)blockpos2.getY() - d1;
+         double d5 = (double)blockpos2.getZ() - d2;
          if (!(d3 * d3 + d4 * d4 + d5 * d5 > 1024.0D)) {
             SortedSet<BlockDestructionProgress> sortedset1 = entry.getValue();
             if (sortedset1 != null && !sortedset1.isEmpty()) {
                int k = sortedset1.last().m_139988_();
                p_109600_.m_85836_();
-               p_109600_.m_85837_((double)blockpos2.m_123341_() - d0, (double)blockpos2.m_123342_() - d1, (double)blockpos2.m_123343_() - d2);
+               p_109600_.m_85837_((double)blockpos2.getX() - d0, (double)blockpos2.getY() - d1, (double)blockpos2.getZ() - d2);
                PoseStack.Pose posestack$pose1 = p_109600_.m_85850_();
                VertexConsumer vertexconsumer1 = new SheetedDecalTextureGenerator(this.f_109464_.m_110108_().m_6299_(ModelBakery.f_119229_.get(k)), posestack$pose1.m_252922_(), posestack$pose1.m_252943_(), 1.0F);
                this.f_109461_.m_91289_().m_110918_(this.f_109465_.m_8055_(blockpos2), blockpos2, this.f_109465_, p_109600_, vertexconsumer1);
@@ -1420,17 +_,17 @@
          double d1 = p_172997_ - this.f_109446_;
          double d2 = p_172998_ - this.f_109447_;
          if (d0 * d0 + d1 * d1 + d2 * d2 > 1.0D) {
-            int j = SectionPos.m_175552_(p_172996_);
-            int k = SectionPos.m_175552_(p_172997_);
-            int l = SectionPos.m_175552_(p_172998_);
-            boolean flag = j != SectionPos.m_175552_(this.f_109445_) || l != SectionPos.m_175552_(this.f_109447_) || k != SectionPos.m_175552_(this.f_109446_);
+            long j = SectionPos.posToSectionCoord(p_172996_);
+            long k = SectionPos.posToSectionCoord(p_172997_);
+            long l = SectionPos.posToSectionCoord(p_172998_);
+            boolean flag = j != SectionPos.posToSectionCoord(this.f_109445_) || l != SectionPos.posToSectionCoord(this.f_109447_) || k != SectionPos.posToSectionCoord(this.f_109446_);
             this.f_109445_ = p_172996_;
             this.f_109446_ = p_172997_;
             this.f_109447_ = p_172998_;
             int i1 = 0;
 
             for(LevelRenderer.RenderChunkInfo levelrenderer$renderchunkinfo : this.f_194297_) {
-               if (i1 < 15 && (flag || levelrenderer$renderchunkinfo.m_274540_(j, k, l)) && levelrenderer$renderchunkinfo.f_109839_.m_112809_(p_172994_, this.f_109436_)) {
+               if (i1 < 15 && (flag || levelrenderer$renderchunkinfo.isAxisAlignedWith(j, k, l)) && levelrenderer$renderchunkinfo.f_109839_.m_112809_(p_172994_, this.f_109436_)) {
                   ++i1;
                }
             }
@@ -1511,7 +_,7 @@
             VertexBuffer vertexbuffer = chunkrenderdispatcher$renderchunk.m_112807_(p_172994_);
             BlockPos blockpos = chunkrenderdispatcher$renderchunk.m_112839_();
             if (uniform != null) {
-               uniform.m_5889_((float)((double)blockpos.m_123341_() - p_172996_), (float)((double)blockpos.m_123342_() - p_172997_), (float)((double)blockpos.m_123343_() - p_172998_));
+               uniform.m_5889_((float)((double)blockpos.getX() - p_172996_), (float)((double)blockpos.getY() - p_172997_), (float)((double)blockpos.getZ() - p_172998_));
                uniform.m_85633_();
             }
 
@@ -1540,7 +_,7 @@
             ChunkRenderDispatcher.RenderChunk chunkrenderdispatcher$renderchunk = levelrenderer$renderchunkinfo.f_109839_;
             BlockPos blockpos = chunkrenderdispatcher$renderchunk.m_112839_();
             p_271014_.m_85836_();
-            p_271014_.m_85837_((double)blockpos.m_123341_() - d0, (double)blockpos.m_123342_() - d1, (double)blockpos.m_123343_() - d2);
+            p_271014_.m_85837_((double)blockpos.getX() - d0, (double)blockpos.getY() - d1, (double)blockpos.getZ() - d2);
             Matrix4f matrix4f = p_271014_.m_85850_().m_252922_();
             if (this.f_109461_.f_90978_) {
                VertexConsumer vertexconsumer1 = p_270107_.m_6299_(RenderType.m_110504_());
@@ -1687,7 +_,7 @@
    }
 
    private void m_109765_(BlockDestructionProgress p_109766_) {
-      long i = p_109766_.m_139985_().m_121878_();
+      BlockPos i = p_109766_.m_139985_();
       Set<BlockDestructionProgress> set = this.f_109409_.get(i);
       set.remove(p_109766_);
       if (set.isEmpty()) {
@@ -1876,15 +_,15 @@
          double d2 = (p_253843_ + d1) / 12.0D;
          double d3 = (double)(f - (float)p_253663_ + 0.33F);
          double d4 = p_253795_ / 12.0D + (double)0.33F;
-         d2 -= (double)(Mth.m_14107_(d2 / 2048.0D) * 2048);
-         d4 -= (double)(Mth.m_14107_(d4 / 2048.0D) * 2048);
-         float f3 = (float)(d2 - (double)Mth.m_14107_(d2));
-         float f4 = (float)(d3 / 4.0D - (double)Mth.m_14107_(d3 / 4.0D)) * 4.0F;
-         float f5 = (float)(d4 - (double)Mth.m_14107_(d4));
+         d2 -= (double)(Mth.m_14134_(d2 / 2048.0D) * 2048);
+         d4 -= (double)(Mth.m_14134_(d4 / 2048.0D) * 2048);
+         float f3 = (float)(d2 - (double)Mth.m_14134_(d2));
+         float f4 = (float)(d3 / 4.0D - (double)Mth.m_14134_(d3 / 4.0D)) * 4.0F;
+         float f5 = (float)(d4 - (double)Mth.m_14134_(d4));
          Vec3 vec3 = this.f_109465_.m_104808_(p_254364_);
-         int i = (int)Math.floor(d2);
-         int j = (int)Math.floor(d3 / 4.0D);
-         int k = (int)Math.floor(d4);
+         long i = (long) Math.floor(d2);
+         long j = (long) Math.floor(d3 / 4.0D);
+         long k = (long) Math.floor(d4);
          if (i != this.f_109430_ || j != this.f_109431_ || k != this.f_109432_ || this.f_109461_.f_91066_.m_92174_() != this.f_109435_ || this.f_109433_.m_82557_(vec3) > 2.0E-4D) {
             this.f_109430_ = i;
             this.f_109431_ = j;
@@ -2048,8 +_,8 @@
          if (chunkrenderdispatcher$renderchunk.m_112841_() && levellightengine.m_284439_(sectionpos)) {
             boolean flag = false;
             if (this.f_109461_.f_91066_.m_232080_().m_231551_() == PrioritizeChunkUpdates.NEARBY) {
-               BlockPos blockpos1 = chunkrenderdispatcher$renderchunk.m_112839_().m_7918_(8, 8, 8);
-               flag = blockpos1.m_123331_(blockpos) < 768.0D || chunkrenderdispatcher$renderchunk.m_112842_();
+               BlockPos blockpos1 = chunkrenderdispatcher$renderchunk.m_112839_().offset(8, 8, 8);
+               flag = blockpos1.distSqr(blockpos) < 768.0D || chunkrenderdispatcher$renderchunk.m_112842_();
             } else if (this.f_109461_.f_91066_.m_232080_().m_231551_() == PrioritizeChunkUpdates.PLAYER_AFFECTED) {
                flag = chunkrenderdispatcher$renderchunk.m_112842_();
             }
@@ -2109,8 +_,8 @@
          float f4 = (float)(-Mth.m_14185_(p_173013_.m_90583_().f_82480_ * 0.5D));
          float f5 = f4 + (float)d4;
          bufferbuilder.m_166779_(VertexFormat.Mode.QUADS, DefaultVertexFormat.f_85817_);
-         double d5 = Math.max((double)Mth.m_14107_(d3 - d0), worldborder.m_61956_());
-         double d6 = Math.min((double)Mth.m_14165_(d3 + d0), worldborder.m_61958_());
+         double d5 = Math.max((double)Mth.m_14134_(d3 - d0), worldborder.m_61956_());
+         double d6 = Math.min((double)Mth.lceil(d3 + d0), worldborder.m_61958_());
          float f6 = (float)(Mth.m_14107_(d5) & 1) * 0.5F;
          if (d2 > worldborder.m_61957_() - d0) {
             float f7 = f6;
@@ -2140,8 +_,8 @@
             }
          }
 
-         d5 = Math.max((double)Mth.m_14107_(d2 - d0), worldborder.m_61955_());
-         d6 = Math.min((double)Mth.m_14165_(d2 + d0), worldborder.m_61957_());
+         d5 = Math.max((double)Mth.m_14134_(d2 - d0), worldborder.m_61955_());
+         d6 = Math.min((double)Mth.lceil(d2 + d0), worldborder.m_61957_());
          f6 = (float)(Mth.m_14107_(d5) & 1) * 0.5F;
          if (d3 > worldborder.m_61958_() - d0) {
             float f10 = f6;
@@ -2185,7 +_,7 @@
    }
 
    private void m_109637_(PoseStack p_109638_, VertexConsumer p_109639_, Entity p_109640_, double p_109641_, double p_109642_, double p_109643_, BlockPos p_109644_, BlockState p_109645_) {
-      m_109782_(p_109638_, p_109639_, p_109645_.m_60651_(this.f_109465_, p_109644_, CollisionContext.m_82750_(p_109640_)), (double)p_109644_.m_123341_() - p_109641_, (double)p_109644_.m_123342_() - p_109642_, (double)p_109644_.m_123343_() - p_109643_, 0.0F, 0.0F, 0.0F, 0.4F);
+      m_109782_(p_109638_, p_109639_, p_109645_.m_60651_(this.f_109465_, p_109644_, CollisionContext.m_82750_(p_109640_)), (double)p_109644_.getX() - p_109641_, (double)p_109644_.getY() - p_109642_, (double)p_109644_.getZ() - p_109643_, 0.0F, 0.0F, 0.0F, 0.4F);
    }
 
    private static Vec3 m_285956_(float p_286899_) {
@@ -2349,21 +_,21 @@
    }
 
    private void m_109732_(BlockPos p_109733_, boolean p_109734_) {
-      for(int i = p_109733_.m_123343_() - 1; i <= p_109733_.m_123343_() + 1; ++i) {
-         for(int j = p_109733_.m_123341_() - 1; j <= p_109733_.m_123341_() + 1; ++j) {
-            for(int k = p_109733_.m_123342_() - 1; k <= p_109733_.m_123342_() + 1; ++k) {
-               this.m_109501_(SectionPos.m_123171_(j), SectionPos.m_123171_(k), SectionPos.m_123171_(i), p_109734_);
+      for(long i = p_109733_.getZ() - 1; i <= p_109733_.getZ() + 1; ++i) {
+         for(long j = p_109733_.getX() - 1; j <= p_109733_.getX() + 1; ++j) {
+            for(long k = p_109733_.getY() - 1; k <= p_109733_.getY() + 1; ++k) {
+               this.setSectionDirty(SectionPos.blockToSectionCoord(j), SectionPos.blockToSectionCoord(k), SectionPos.blockToSectionCoord(i), p_109734_);
             }
          }
       }
 
    }
 
-   public void m_109494_(int p_109495_, int p_109496_, int p_109497_, int p_109498_, int p_109499_, int p_109500_) {
-      for(int i = p_109497_ - 1; i <= p_109500_ + 1; ++i) {
-         for(int j = p_109495_ - 1; j <= p_109498_ + 1; ++j) {
-            for(int k = p_109496_ - 1; k <= p_109499_ + 1; ++k) {
-               this.m_109770_(SectionPos.m_123171_(j), SectionPos.m_123171_(k), SectionPos.m_123171_(i));
+   public void setBlocksDirty(long p_109495_, long p_109496_, long p_109497_, long p_109498_, long p_109499_, long p_109500_) {
+      for(long i = p_109497_ - 1; i <= p_109500_ + 1; ++i) {
+         for(long j = p_109495_ - 1; j <= p_109498_ + 1; ++j) {
+            for(long k = p_109496_ - 1; k <= p_109499_ + 1; ++k) {
+               this.setSectionDirty(SectionPos.blockToSectionCoord(j), SectionPos.blockToSectionCoord(k), SectionPos.blockToSectionCoord(i));
             }
          }
       }
@@ -2372,28 +_,28 @@
 
    public void m_109721_(BlockPos p_109722_, BlockState p_109723_, BlockState p_109724_) {
       if (this.f_109461_.m_91304_().m_119415_(p_109723_, p_109724_)) {
-         this.m_109494_(p_109722_.m_123341_(), p_109722_.m_123342_(), p_109722_.m_123343_(), p_109722_.m_123341_(), p_109722_.m_123342_(), p_109722_.m_123343_());
+         this.setBlocksDirty(p_109722_.getX(), p_109722_.getY(), p_109722_.getZ(), p_109722_.getX(), p_109722_.getY(), p_109722_.getZ());
       }
 
    }
 
-   public void m_109490_(int p_109491_, int p_109492_, int p_109493_) {
-      for(int i = p_109493_ - 1; i <= p_109493_ + 1; ++i) {
-         for(int j = p_109491_ - 1; j <= p_109491_ + 1; ++j) {
-            for(int k = p_109492_ - 1; k <= p_109492_ + 1; ++k) {
-               this.m_109770_(j, k, i);
+   public void setSectionDirtyWithNeighbors(long p_109491_, long p_109492_, long p_109493_) {
+      for(long i = p_109493_ - 1; i <= p_109493_ + 1; ++i) {
+         for(long j = p_109491_ - 1; j <= p_109491_ + 1; ++j) {
+            for(long k = p_109492_ - 1; k <= p_109492_ + 1; ++k) {
+               this.setSectionDirty(j, k, i);
             }
          }
       }
 
    }
 
-   public void m_109770_(int p_109771_, int p_109772_, int p_109773_) {
-      this.m_109501_(p_109771_, p_109772_, p_109773_, false);
+   public void setSectionDirty(long p_109771_, long p_109772_, long p_109773_) {
+      this.setSectionDirty(p_109771_, p_109772_, p_109773_, false);
    }
 
-   private void m_109501_(int p_109502_, int p_109503_, int p_109504_, boolean p_109505_) {
-      this.f_109469_.m_110859_(p_109502_, p_109503_, p_109504_, p_109505_);
+   private void setSectionDirty(long p_109502_, long p_109503_, long p_109504_, boolean p_109505_) {
+      this.f_109469_.setDirty(p_109502_, p_109503_, p_109504_, p_109505_);
    }
 
    public void m_109514_(@Nullable SoundEvent p_109515_, BlockPos p_109516_) {
@@ -2409,7 +_,7 @@
             this.f_109461_.f_91065_.m_93055_(recorditem.m_43050_());
          }
 
-         SoundInstance simplesoundinstance = SimpleSoundInstance.m_246411_(p_109515_, Vec3.m_82512_(p_109516_));
+         SoundInstance simplesoundinstance = SimpleSoundInstance.m_246411_(p_109515_, Vec3.atCenterOf(p_109516_));
          this.f_109410_.put(p_109516_, simplesoundinstance);
          this.f_109461_.m_91106_().m_120367_(simplesoundinstance);
       }
@@ -2488,9 +_,9 @@
          case 1038:
             Camera camera = this.f_109461_.f_91063_.m_109153_();
             if (camera.m_90593_()) {
-               double d0 = (double)p_109508_.m_123341_() - camera.m_90583_().f_82479_;
-               double d1 = (double)p_109508_.m_123342_() - camera.m_90583_().f_82480_;
-               double d2 = (double)p_109508_.m_123343_() - camera.m_90583_().f_82481_;
+               double d0 = (double)p_109508_.getX() - camera.m_90583_().f_82479_;
+               double d1 = (double)p_109508_.getY() - camera.m_90583_().f_82480_;
+               double d2 = (double)p_109508_.getZ() - camera.m_90583_().f_82481_;
                double d3 = Math.sqrt(d0 * d0 + d1 * d1 + d2 * d2);
                double d4 = camera.m_90583_().f_82479_;
                double d5 = camera.m_90583_().f_82480_;
@@ -2641,16 +_,16 @@
             this.f_109465_.m_245747_(p_234306_, SoundEvents.f_12031_, SoundSource.BLOCKS, 0.5F, 2.6F + (randomsource.m_188501_() - randomsource.m_188501_()) * 0.8F, false);
 
             for(int j3 = 0; j3 < 8; ++j3) {
-               this.f_109465_.m_7106_(ParticleTypes.f_123755_, (double)p_234306_.m_123341_() + randomsource.m_188500_(), (double)p_234306_.m_123342_() + 1.2D, (double)p_234306_.m_123343_() + randomsource.m_188500_(), 0.0D, 0.0D, 0.0D);
+               this.f_109465_.m_7106_(ParticleTypes.f_123755_, (double)p_234306_.getX() + randomsource.m_188500_(), (double)p_234306_.getY() + 1.2D, (double)p_234306_.getZ() + randomsource.m_188500_(), 0.0D, 0.0D, 0.0D);
             }
             break;
          case 1502:
             this.f_109465_.m_245747_(p_234306_, SoundEvents.f_12374_, SoundSource.BLOCKS, 0.5F, 2.6F + (randomsource.m_188501_() - randomsource.m_188501_()) * 0.8F, false);
 
             for(int i3 = 0; i3 < 5; ++i3) {
-               double d16 = (double)p_234306_.m_123341_() + randomsource.m_188500_() * 0.6D + 0.2D;
-               double d22 = (double)p_234306_.m_123342_() + randomsource.m_188500_() * 0.6D + 0.2D;
-               double d27 = (double)p_234306_.m_123343_() + randomsource.m_188500_() * 0.6D + 0.2D;
+               double d16 = (double)p_234306_.getX() + randomsource.m_188500_() * 0.6D + 0.2D;
+               double d22 = (double)p_234306_.getY() + randomsource.m_188500_() * 0.6D + 0.2D;
+               double d27 = (double)p_234306_.getZ() + randomsource.m_188500_() * 0.6D + 0.2D;
                this.f_109465_.m_7106_(ParticleTypes.f_123762_, d16, d22, d27, 0.0D, 0.0D, 0.0D);
             }
             break;
@@ -2658,9 +_,9 @@
             this.f_109465_.m_245747_(p_234306_, SoundEvents.f_11859_, SoundSource.BLOCKS, 1.0F, 1.0F, false);
 
             for(int l2 = 0; l2 < 16; ++l2) {
-               double d15 = (double)p_234306_.m_123341_() + (5.0D + randomsource.m_188500_() * 6.0D) / 16.0D;
-               double d21 = (double)p_234306_.m_123342_() + 0.8125D;
-               double d26 = (double)p_234306_.m_123343_() + (5.0D + randomsource.m_188500_() * 6.0D) / 16.0D;
+               double d15 = (double)p_234306_.getX() + (5.0D + randomsource.m_188500_() * 6.0D) / 16.0D;
+               double d21 = (double)p_234306_.getY() + 0.8125D;
+               double d26 = (double)p_234306_.getZ() + (5.0D + randomsource.m_188500_() * 6.0D) / 16.0D;
                this.f_109465_.m_7106_(ParticleTypes.f_123762_, d15, d21, d26, 0.0D, 0.0D, 0.0D);
             }
             break;
@@ -2676,9 +_,9 @@
             int j = direction1.m_122429_();
             int i1 = direction1.m_122430_();
             int k1 = direction1.m_122431_();
-            double d10 = (double)p_234306_.m_123341_() + (double)j * 0.6D + 0.5D;
-            double d14 = (double)p_234306_.m_123342_() + (double)i1 * 0.6D + 0.5D;
-            double d20 = (double)p_234306_.m_123343_() + (double)k1 * 0.6D + 0.5D;
+            double d10 = (double)p_234306_.getX() + (double)j * 0.6D + 0.5D;
+            double d14 = (double)p_234306_.getY() + (double)i1 * 0.6D + 0.5D;
+            double d20 = (double)p_234306_.getZ() + (double)k1 * 0.6D + 0.5D;
 
             for(int i4 = 0; i4 < 10; ++i4) {
                double d28 = randomsource.m_188500_() * 0.2D + 0.01D;
@@ -2702,7 +_,7 @@
             break;
          case 2002:
          case 2007:
-            Vec3 vec3 = Vec3.m_82539_(p_234306_);
+            Vec3 vec3 = Vec3.atBottomCenterOf(p_234306_);
 
             for(int i = 0; i < 8; ++i) {
                this.m_109735_(new ItemParticleOption(ParticleTypes.f_123752_, new ItemStack(Items.f_42736_)), vec3.f_82479_, vec3.f_82480_, vec3.f_82481_, randomsource.m_188583_() * 0.15D, randomsource.m_188500_() * 0.2D, randomsource.m_188583_() * 0.15D);
@@ -2730,9 +_,9 @@
             this.f_109465_.m_245747_(p_234306_, SoundEvents.f_12436_, SoundSource.NEUTRAL, 1.0F, randomsource.m_188501_() * 0.1F + 0.9F, false);
             break;
          case 2003:
-            double d0 = (double)p_234306_.m_123341_() + 0.5D;
-            double d7 = (double)p_234306_.m_123342_();
-            double d9 = (double)p_234306_.m_123343_() + 0.5D;
+            double d0 = p_234306_.getX() + 0.5D;
+            double d7 = p_234306_.getY();
+            double d9 = p_234306_.getZ() + 0.5D;
 
             for(int k3 = 0; k3 < 8; ++k3) {
                this.m_109735_(new ItemParticleOption(ParticleTypes.f_123752_, new ItemStack(Items.f_42545_)), d0, d7, d9, randomsource.m_188583_() * 0.15D, randomsource.m_188500_() * 0.2D, randomsource.m_188583_() * 0.15D);
@@ -2745,9 +_,9 @@
             break;
          case 2004:
             for(int l = 0; l < 20; ++l) {
-               double d8 = (double)p_234306_.m_123341_() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
-               double d11 = (double)p_234306_.m_123342_() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
-               double d17 = (double)p_234306_.m_123343_() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
+               double d8 = (double)p_234306_.getX() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
+               double d11 = (double)p_234306_.getY() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
+               double d17 = (double)p_234306_.getZ() + 0.5D + (randomsource.m_188500_() - 0.5D) * 2.0D;
                this.f_109465_.m_7106_(ParticleTypes.f_123762_, d8, d11, d17, 0.0D, 0.0D, 0.0D);
                this.f_109465_.m_7106_(ParticleTypes.f_123744_, d8, d11, d17, 0.0D, 0.0D, 0.0D);
             }
@@ -2762,7 +_,7 @@
                double d18 = (double)(Mth.m_14089_(f11) * f10);
                double d24 = 0.01D + randomsource.m_188500_() * 0.5D;
                double d29 = (double)(Mth.m_14031_(f11) * f10);
-               Particle particle = this.m_109795_(ParticleTypes.f_123799_, false, (double)p_234306_.m_123341_() + d18 * 0.1D, (double)p_234306_.m_123342_() + 0.3D, (double)p_234306_.m_123343_() + d29 * 0.1D, d18, d24, d29);
+               Particle particle = this.m_109795_(ParticleTypes.f_123799_, false, (double)p_234306_.getX() + d18 * 0.1D, (double)p_234306_.getY() + 0.3D, (double)p_234306_.getZ() + d29 * 0.1D, d18, d24, d29);
                if (particle != null) {
                   particle.m_107268_(f10);
                }
@@ -2773,15 +_,15 @@
             }
             break;
          case 2008:
-            this.f_109465_.m_7106_(ParticleTypes.f_123813_, (double)p_234306_.m_123341_() + 0.5D, (double)p_234306_.m_123342_() + 0.5D, (double)p_234306_.m_123343_() + 0.5D, 0.0D, 0.0D, 0.0D);
+            this.f_109465_.m_7106_(ParticleTypes.f_123813_, (double)p_234306_.getX() + 0.5D, (double)p_234306_.getY() + 0.5D, (double)p_234306_.getZ() + 0.5D, 0.0D, 0.0D, 0.0D);
             break;
          case 2009:
             for(int i2 = 0; i2 < 8; ++i2) {
-               this.f_109465_.m_7106_(ParticleTypes.f_123796_, (double)p_234306_.m_123341_() + randomsource.m_188500_(), (double)p_234306_.m_123342_() + 1.2D, (double)p_234306_.m_123343_() + randomsource.m_188500_(), 0.0D, 0.0D, 0.0D);
+               this.f_109465_.m_7106_(ParticleTypes.f_123796_, (double)p_234306_.getX() + randomsource.m_188500_(), (double)p_234306_.getY() + 1.2D, (double)p_234306_.getZ() + randomsource.m_188500_(), 0.0D, 0.0D, 0.0D);
             }
             break;
          case 3000:
-            this.f_109465_.m_6493_(ParticleTypes.f_123812_, true, (double)p_234306_.m_123341_() + 0.5D, (double)p_234306_.m_123342_() + 0.5D, (double)p_234306_.m_123343_() + 0.5D, 0.0D, 0.0D, 0.0D);
+            this.f_109465_.m_6493_(ParticleTypes.f_123812_, true, (double)p_234306_.getX() + 0.5D, (double)p_234306_.getY() + 0.5D, (double)p_234306_.getZ() + 0.5D, 0.0D, 0.0D, 0.0D);
             this.f_109465_.m_245747_(p_234306_, SoundEvents.f_11858_, SoundSource.BLOCKS, 10.0F, (1.0F + (this.f_109465_.f_46441_.m_188501_() - this.f_109465_.f_46441_.m_188501_()) * 0.2F) * 0.7F, false);
             break;
          case 3001:
@@ -2843,20 +_,20 @@
                   float f12 = 2.0F * randomsource.m_188501_() - 1.0F;
                   float f14 = 2.0F * randomsource.m_188501_() - 1.0F;
                   float f15 = 2.0F * randomsource.m_188501_() - 1.0F;
-                  this.f_109465_.m_7106_(ParticleTypes.f_235900_, (double)p_234306_.m_123341_() + 0.5D + (double)(f12 * f8), (double)p_234306_.m_123342_() + 0.5D + (double)(f14 * f8), (double)p_234306_.m_123343_() + 0.5D + (double)(f15 * f8), (double)(f12 * 0.07F), (double)(f14 * 0.07F), (double)(f15 * 0.07F));
+                  this.f_109465_.m_7106_(ParticleTypes.f_235900_, (double)p_234306_.getX() + 0.5D + (double)(f12 * f9), (double)p_234306_.getY() + 0.5D + (double)(f14 * f9), (double)p_234306_.getZ() + 0.5D + (double)(f15 * f9), (double)(f12 * 0.07F), (double)(f14 * 0.07F), (double)(f15 * 0.07F));
                }
             }
             break;
          case 3007:
             for(int j1 = 0; j1 < 10; ++j1) {
-               this.f_109465_.m_6493_(new ShriekParticleOption(j1 * 5), false, (double)p_234306_.m_123341_() + 0.5D, (double)p_234306_.m_123342_() + SculkShriekerBlock.f_222156_, (double)p_234306_.m_123343_() + 0.5D, 0.0D, 0.0D, 0.0D);
+               this.f_109465_.m_6493_(new ShriekParticleOption(j1 * 5), false, (double)p_234306_.getX() + 0.5D, (double)p_234306_.getY() + SculkShriekerBlock.f_222156_, (double)p_234306_.getZ() + 0.5D, 0.0D, 0.0D, 0.0D);
             }
 
             BlockState blockstate2 = this.f_109465_.m_8055_(p_234306_);
             boolean flag = blockstate2.m_61138_(BlockStateProperties.f_61362_) && blockstate2.m_61143_(BlockStateProperties.f_61362_);
             if (!flag) {
-               this.f_109465_.m_7785_((double)p_234306_.m_123341_() + 0.5D, (double)p_234306_.m_123342_() + SculkShriekerBlock.f_222156_, (double)p_234306_.m_123343_() + 0.5D, SoundEvents.f_215750_, SoundSource.BLOCKS, 2.0F, 0.6F + this.f_109465_.f_46441_.m_188501_() * 0.4F, false);
             }
+            this.f_109465_.m_7785_((double)p_234306_.getX() + 0.5D, (double)p_234306_.getY() + SculkShriekerBlock.f_222156_, (double)p_234306_.getZ() + 0.5D, SoundEvents.f_215750_, SoundSource.BLOCKS, 2.0F, 0.6F + this.f_109465_.f_46441_.m_188501_() * 0.4F, false);
             break;
          case 3008:
             BlockState blockstate = Block.m_49803_(p_234307_);
@@ -2880,14 +_,14 @@
             this.m_109765_(blockdestructionprogress1);
          }
 
-         if (blockdestructionprogress1 == null || blockdestructionprogress1.m_139985_().m_123341_() != p_109776_.m_123341_() || blockdestructionprogress1.m_139985_().m_123342_() != p_109776_.m_123342_() || blockdestructionprogress1.m_139985_().m_123343_() != p_109776_.m_123343_()) {
+         if (blockdestructionprogress1 == null || blockdestructionprogress1.m_139985_().getX() != p_109776_.getX() || blockdestructionprogress1.m_139985_().getY() != p_109776_.getY() || blockdestructionprogress1.m_139985_().getZ() != p_109776_.getZ()) {
             blockdestructionprogress1 = new BlockDestructionProgress(p_109775_, p_109776_);
             this.f_109408_.put(p_109775_, blockdestructionprogress1);
          }
 
          blockdestructionprogress1.m_139981_(p_109777_);
          blockdestructionprogress1.m_139986_(this.f_109477_);
-         this.f_109409_.computeIfAbsent(blockdestructionprogress1.m_139985_().m_121878_(), (p_234254_) -> {
+         this.f_109409_.computeIfAbsent(blockdestructionprogress1.m_139985_(), (p_234254_) -> {
             return Sets.newTreeSet();
          }).add(blockdestructionprogress1);
       } else {
@@ -3005,9 +_,9 @@
          return this.f_173020_ != 0;
       }
 
-      public boolean m_274540_(int p_275489_, int p_275424_, int p_275596_) {
+      public boolean isAxisAlignedWith(long p_275489_, long p_275424_, long p_275596_) {
          BlockPos blockpos = this.f_109839_.m_112839_();
-         return p_275489_ == blockpos.m_123341_() / 16 || p_275596_ == blockpos.m_123343_() / 16 || p_275424_ == blockpos.m_123342_() / 16;
+         return p_275489_ == blockpos.getX() / 16 || p_275596_ == blockpos.getZ() / 16 || p_275424_ == blockpos.getY() / 16;
       }
 
       public int hashCode() {
@@ -3043,13 +_,13 @@
       }
 
       public void m_173037_(ChunkRenderDispatcher.RenderChunk p_173038_, LevelRenderer.RenderChunkInfo p_173039_) {
-         this.f_173030_[p_173038_.f_173717_] = p_173039_;
+         this.f_173030_[(int) p_173038_.f_173717_] = p_173039_;
       }
 
       @Nullable
       public LevelRenderer.RenderChunkInfo m_173035_(ChunkRenderDispatcher.RenderChunk p_173036_) {
-         int i = p_173036_.f_173717_;
-         return i >= 0 && i < this.f_173030_.length ? this.f_173030_[i] : null;
+         long i = p_173036_.f_173717_;
+         return i >= 0 && i < this.f_173030_.length ? this.f_173030_[(int) i] : null;
       }
    }
 
