--- a/net/minecraft/client/renderer/ScreenEffectRenderer.java
+++ b/net/minecraft/client/renderer/ScreenEffectRenderer.java
@@ -53,10 +_,10 @@
       BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
       for(int i = 0; i < 8; ++i) {
-         double d0 = p_110717_.m_20185_() + (double)(((float)((i >> 0) % 2) - 0.5F) * p_110717_.m_20205_() * 0.8F);
-         double d1 = p_110717_.m_20188_() + (double)(((float)((i >> 1) % 2) - 0.5F) * 0.1F);
-         double d2 = p_110717_.m_20189_() + (double)(((float)((i >> 2) % 2) - 0.5F) * p_110717_.m_20205_() * 0.8F);
-         blockpos$mutableblockpos.m_122169_(d0, d1, d2);
+         me.alphamode.mclong.math.BigDecimal d0 = p_110717_.getBigX().add((double)(((float)((i >> 0) % 2) - 0.5F) * p_110717_.m_20205_() * 0.8F));
+         me.alphamode.mclong.math.BigDecimal d1 = p_110717_.getEyeBigY().add((double)(((float)((i >> 1) % 2) - 0.5F) * 0.1F));
+         me.alphamode.mclong.math.BigDecimal d2 = p_110717_.getBigZ().add((double)(((float)((i >> 2) % 2) - 0.5F) * p_110717_.m_20205_() * 0.8F));
+         blockpos$mutableblockpos.set(d0, d1, d2);
          BlockState blockstate = p_110717_.m_9236_().m_8055_(blockpos$mutableblockpos);
          if (blockstate.m_60799_() != RenderShape.INVISIBLE && blockstate.m_60831_(p_110717_.m_9236_(), blockpos$mutableblockpos)) {
             return blockstate;
@@ -93,7 +_,7 @@
       RenderSystem.setShader(GameRenderer::m_172817_);
       RenderSystem.setShaderTexture(0, f_110714_);
       BufferBuilder bufferbuilder = Tesselator.m_85913_().m_85915_();
-      BlockPos blockpos = BlockPos.m_274561_(p_110726_.f_91074_.m_20185_(), p_110726_.f_91074_.m_20188_(), p_110726_.f_91074_.m_20189_());
+      BlockPos blockpos = BlockPos.containing(p_110726_.f_91074_.getBigX(), p_110726_.f_91074_.getEyeBigY(), p_110726_.f_91074_.getBigZ());
       float f = LightTexture.m_234316_(p_110726_.f_91074_.m_9236_().m_6042_(), p_110726_.f_91074_.m_9236_().m_46803_(blockpos));
       RenderSystem.enableBlend();
       RenderSystem.setShaderColor(f, f, f, 0.1F);
