--- a/net/minecraft/world/entity/ai/memory/WalkTarget.java
+++ b/net/minecraft/world/entity/ai/memory/WalkTarget.java
@@ -17,7 +_,7 @@
    }
 
    public WalkTarget(Vec3 p_26413_, float p_26414_, int p_26415_) {
-      this(new BlockPosTracker(BlockPos.m_274446_(p_26413_)), p_26414_, p_26415_);
+      this(new BlockPosTracker(BlockPos.containing(p_26413_)), p_26414_, p_26415_);
    }
 
    public WalkTarget(Entity p_148209_, float p_148210_, int p_148211_) {
