--- a/net/minecraft/world/phys/AABB.java
+++ b/net/minecraft/world/phys/AABB.java
@@ -26,11 +_,11 @@
    }
 
    public AABB(BlockPos p_82305_) {
-      this((double)p_82305_.m_123341_(), (double)p_82305_.m_123342_(), (double)p_82305_.m_123343_(), (double)(p_82305_.m_123341_() + 1), (double)(p_82305_.m_123342_() + 1), (double)(p_82305_.m_123343_() + 1));
+      this((double)p_82305_.getX(), (double)p_82305_.getY(), (double)p_82305_.getZ(), (double)(p_82305_.getX() + 1), (double)(p_82305_.getY() + 1), (double)(p_82305_.getZ() + 1));
    }
 
    public AABB(BlockPos p_82307_, BlockPos p_82308_) {
-      this((double)p_82307_.m_123341_(), (double)p_82307_.m_123342_(), (double)p_82307_.m_123343_(), (double)p_82308_.m_123341_(), (double)p_82308_.m_123342_(), (double)p_82308_.m_123343_());
+      this((double)p_82307_.getX(), (double)p_82307_.getY(), (double)p_82307_.getZ(), (double)p_82308_.getX(), (double)p_82308_.getY(), (double)p_82308_.getZ());
    }
 
    public AABB(Vec3 p_82302_, Vec3 p_82303_) {
@@ -38,7 +_,7 @@
    }
 
    public static AABB m_82321_(BoundingBox p_82322_) {
-      return new AABB((double)p_82322_.m_162395_(), (double)p_82322_.m_162396_(), (double)p_82322_.m_162398_(), (double)(p_82322_.m_162399_() + 1), (double)(p_82322_.m_162400_() + 1), (double)(p_82322_.m_162401_() + 1));
+      return new AABB((double)p_82322_.minX(), (double)p_82322_.minY(), (double)p_82322_.minZ(), (double)(p_82322_.maxX() + 1), (double)(p_82322_.maxY() + 1), (double)(p_82322_.maxZ() + 1));
    }
 
    public static AABB m_82333_(Vec3 p_82334_) {
@@ -214,7 +_,7 @@
    }
 
    public AABB m_82338_(BlockPos p_82339_) {
-      return new AABB(this.f_82288_ + (double)p_82339_.m_123341_(), this.f_82289_ + (double)p_82339_.m_123342_(), this.f_82290_ + (double)p_82339_.m_123343_(), this.f_82291_ + (double)p_82339_.m_123341_(), this.f_82292_ + (double)p_82339_.m_123342_(), this.f_82293_ + (double)p_82339_.m_123343_());
+      return new AABB(this.f_82288_ + (double)p_82339_.getX(), this.f_82289_ + (double)p_82339_.getY(), this.f_82290_ + (double)p_82339_.getZ(), this.f_82291_ + (double)p_82339_.getX(), this.f_82292_ + (double)p_82339_.getY(), this.f_82293_ + (double)p_82339_.getZ());
    }
 
    public AABB m_82383_(Vec3 p_82384_) {
