--- a/net/minecraft/world/phys/HitResult.java
+++ b/net/minecraft/world/phys/HitResult.java
@@ -10,9 +_,9 @@
    }
 
    public double m_82448_(Entity p_82449_) {
-      double d0 = this.f_82445_.f_82479_ - p_82449_.m_20185_();
-      double d1 = this.f_82445_.f_82480_ - p_82449_.m_20186_();
-      double d2 = this.f_82445_.f_82481_ - p_82449_.m_20189_();
+      double d0 = this.f_82445_.getX().subtract(p_82449_.getBigX()).doubleValue();
+      double d1 = this.f_82445_.getY().subtract(p_82449_.getBigY()).doubleValue();
+      double d2 = this.f_82445_.getZ().subtract(p_82449_.getBigZ()).doubleValue();
       return d0 * d0 + d1 * d1 + d2 * d2;
    }
 
