--- a/net/minecraft/world/phys/Vec3.java
+++ b/net/minecraft/world/phys/Vec3.java
@@ -31,24 +_,32 @@
       return new Vec3(d0, d1, d2);
    }
 
-   public static Vec3 m_82528_(Vec3i p_82529_) {
-      return new Vec3((double)p_82529_.m_123341_(), (double)p_82529_.m_123342_(), (double)p_82529_.m_123343_());
+   public static Vec3 atLowerCornerOf(me.alphamode.mclong.core.Vec3l p_82529_) {
+      return new Vec3(p_82529_.getX(), p_82529_.getY(), p_82529_.getZ());
    }
 
    public static Vec3 m_272021_(Vec3i p_272866_, double p_273680_, double p_273668_, double p_273687_) {
       return new Vec3((double)p_272866_.m_123341_() + p_273680_, (double)p_272866_.m_123342_() + p_273668_, (double)p_272866_.m_123343_() + p_273687_);
    }
 
+   public static Vec3 atLowerCornerWithOffset(me.alphamode.mclong.core.Vec3l p_272866_, double p_273680_, double p_273668_, double p_273687_) {
+      return new Vec3(p_272866_.getX() + p_273680_, p_272866_.getY() + p_273668_, p_272866_.getZ() + p_273687_);
+   }
+
    public static Vec3 m_82512_(Vec3i p_82513_) {
       return m_272021_(p_82513_, 0.5D, 0.5D, 0.5D);
    }
 
-   public static Vec3 m_82539_(Vec3i p_82540_) {
-      return m_272021_(p_82540_, 0.5D, 0.0D, 0.5D);
-   }
-
-   public static Vec3 m_82514_(Vec3i p_82515_, double p_82516_) {
-      return m_272021_(p_82515_, 0.5D, p_82516_, 0.5D);
+   public static Vec3 atCenterOf(me.alphamode.mclong.core.Vec3l p_82513_) {
+      return atLowerCornerWithOffset(p_82513_, 0.5D, 0.5D, 0.5D);
+   }
+
+   public static Vec3 atBottomCenterOf(me.alphamode.mclong.core.Vec3l p_82540_) {
+      return atLowerCornerWithOffset(p_82540_, 0.5D, 0.0D, 0.5D);
+   }
+
+   public static Vec3 upFromBottomCenterOf(me.alphamode.mclong.core.Vec3l p_82515_, double p_82516_) {
+      return atLowerCornerWithOffset(p_82515_, 0.5D, p_82516_, 0.5D);
    }
 
    public Vec3(double p_82484_, double p_82485_, double p_82486_) {
@@ -229,9 +_,9 @@
    }
 
    public Vec3 m_82517_(EnumSet<Direction.Axis> p_82518_) {
-      double d0 = p_82518_.contains(Direction.Axis.X) ? (double)Mth.m_14107_(this.f_82479_) : this.f_82479_;
-      double d1 = p_82518_.contains(Direction.Axis.Y) ? (double)Mth.m_14107_(this.f_82480_) : this.f_82480_;
-      double d2 = p_82518_.contains(Direction.Axis.Z) ? (double)Mth.m_14107_(this.f_82481_) : this.f_82481_;
+      double d0 = p_82518_.contains(Direction.Axis.X) ? (double)Mth.m_14134_(this.f_82479_) : this.f_82479_;
+      double d1 = p_82518_.contains(Direction.Axis.Y) ? (double)Mth.m_14134_(this.f_82480_) : this.f_82480_;
+      double d2 = p_82518_.contains(Direction.Axis.Z) ? (double)Mth.m_14134_(this.f_82481_) : this.f_82481_;
       return new Vec3(d0, d1, d2);
    }
 
