--- a/net/minecraft/world/phys/Vec3.java
+++ b/net/minecraft/world/phys/Vec3.java
@@ -30,20 +_,40 @@
       return new Vec3(d0, d1, d2);
    }
 
+   @Deprecated
    public static Vec3 m_82512_(Vec3i p_82513_) {
       return new Vec3((double)p_82513_.m_123341_() + 0.5D, (double)p_82513_.m_123342_() + 0.5D, (double)p_82513_.m_123343_() + 0.5D);
    }
 
+   public static Vec3 atCenterOf(me.alphamode.mclong.core.Vec3l p_82513_) {
+      return new Vec3((double)p_82513_.getX() + 0.5D, (double)p_82513_.getY() + 0.5D, (double)p_82513_.getZ() + 0.5D);
+   }
+
+   @Deprecated
    public static Vec3 m_82528_(Vec3i p_82529_) {
       return new Vec3((double)p_82529_.m_123341_(), (double)p_82529_.m_123342_(), (double)p_82529_.m_123343_());
    }
 
+   public static Vec3 atLowerCornerOf(me.alphamode.mclong.core.Vec3l p_82529_) {
+      return new Vec3((double)p_82529_.getX(), (double)p_82529_.getY(), (double)p_82529_.getZ());
+   }
+
+   @Deprecated
    public static Vec3 m_82539_(Vec3i p_82540_) {
       return new Vec3((double)p_82540_.m_123341_() + 0.5D, (double)p_82540_.m_123342_(), (double)p_82540_.m_123343_() + 0.5D);
    }
 
+   public static Vec3 atBottomCenterOf(me.alphamode.mclong.core.Vec3l p_82540_) {
+      return new Vec3((double)p_82540_.getX() + 0.5D, (double)p_82540_.getY(), (double)p_82540_.getZ() + 0.5D);
+   }
+
+   @Deprecated
    public static Vec3 m_82514_(Vec3i p_82515_, double p_82516_) {
       return new Vec3((double)p_82515_.m_123341_() + 0.5D, (double)p_82515_.m_123342_() + p_82516_, (double)p_82515_.m_123343_() + 0.5D);
+   }
+
+   public static Vec3 upFromBottomCenterOf(me.alphamode.mclong.core.Vec3l p_82515_, double p_82516_) {
+      return new Vec3((double)p_82515_.getX() + 0.5D, (double)p_82515_.getY() + p_82516_, (double)p_82515_.getZ() + 0.5D);
    }
 
    public Vec3(double p_82484_, double p_82485_, double p_82486_) {
