--- a/net/minecraft/world/level/chunk/BlockColumn.java
+++ b/net/minecraft/world/level/chunk/BlockColumn.java
@@ -3,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public interface BlockColumn {
-   BlockState m_183556_(int p_187573_);
+   BlockState getBlock(long p_187573_);
 
-   void m_183639_(int p_187574_, BlockState p_187575_);
+   void setBlock(long p_187574_, BlockState p_187575_);
 }
