--- a/net/minecraft/world/level/chunk/BulkSectionAccess.java
+++ b/net/minecraft/world/level/chunk/BulkSectionAccess.java
@@ -22,12 +_,12 @@
 
    @Nullable
    public LevelChunkSection m_156104_(BlockPos p_156105_) {
-      int i = this.f_156098_.m_151564_(p_156105_.m_123342_());
+      int i = this.f_156098_.getSectionIndex(p_156105_.getY());
       if (i >= 0 && i < this.f_156098_.m_151559_()) {
          long j = SectionPos.m_175568_(p_156105_);
          if (this.f_156100_ == null || this.f_156101_ != j) {
             this.f_156100_ = this.f_156099_.computeIfAbsent(j, (p_156109_) -> {
-               ChunkAccess chunkaccess = this.f_156098_.m_6325_(SectionPos.m_123171_(p_156105_.m_123341_()), SectionPos.m_123171_(p_156105_.m_123343_()));
+               ChunkAccess chunkaccess = this.f_156098_.getChunk(SectionPos.m_235865_(p_156105_.getX()), SectionPos.m_235865_(p_156105_.getZ()));
                LevelChunkSection levelchunksection = chunkaccess.m_183278_(i);
                levelchunksection.m_62981_();
                return levelchunksection;
@@ -46,10 +_,10 @@
       if (levelchunksection == null) {
          return Blocks.f_50016_.m_49966_();
       } else {
-         int i = SectionPos.m_123207_(p_156111_.m_123341_());
-         int j = SectionPos.m_123207_(p_156111_.m_123342_());
-         int k = SectionPos.m_123207_(p_156111_.m_123343_());
-         return levelchunksection.m_62982_(i, j, k);
+         long i = SectionPos.sectionRelative(p_156111_.getX());
+         long j = SectionPos.sectionRelative(p_156111_.getY());
+         long k = SectionPos.sectionRelative(p_156111_.getZ());
+         return levelchunksection.getBlockState(i, j, k);
       }
    }
 
