--- a/net/minecraft/world/level/chunk/ChunkSource.java
+++ b/net/minecraft/world/level/chunk/ChunkSource.java
@@ -8,26 +_,26 @@
 
 public abstract class ChunkSource implements LightChunkGetter, AutoCloseable {
    @Nullable
-   public LevelChunk m_62227_(int p_62228_, int p_62229_, boolean p_62230_) {
-      return (LevelChunk)this.m_7587_(p_62228_, p_62229_, ChunkStatus.f_62326_, p_62230_);
-   }
-
-   @Nullable
-   public LevelChunk m_7131_(int p_62221_, int p_62222_) {
-      return this.m_62227_(p_62221_, p_62222_, false);
-   }
-
-   @Nullable
-   public LightChunk m_6196_(int p_62241_, int p_62242_) {
-      return this.m_7587_(p_62241_, p_62242_, ChunkStatus.f_62314_, false);
-   }
-
-   public boolean m_5563_(int p_62238_, int p_62239_) {
-      return this.m_7587_(p_62238_, p_62239_, ChunkStatus.f_62326_, false) != null;
-   }
-
-   @Nullable
-   public abstract ChunkAccess m_7587_(int p_62223_, int p_62224_, ChunkStatus p_62225_, boolean p_62226_);
+   public LevelChunk getChunk(me.alphamode.mclong.math.BigInteger p_62228_, me.alphamode.mclong.math.BigInteger p_62229_, boolean p_62230_) {
+      return (LevelChunk)this.getChunk(p_62228_, p_62229_, ChunkStatus.f_62326_, p_62230_);
+   }
+
+   @Nullable
+   public LevelChunk getChunkNow(me.alphamode.mclong.math.BigInteger p_62221_, me.alphamode.mclong.math.BigInteger p_62222_) {
+      return this.getChunk(p_62221_, p_62222_, false);
+   }
+
+   @Nullable
+   public LightChunk getChunkForLighting(me.alphamode.mclong.math.BigInteger p_62241_, me.alphamode.mclong.math.BigInteger p_62242_) {
+      return this.getChunk(p_62241_, p_62242_, ChunkStatus.f_62314_, false);
+   }
+
+   public boolean hasChunk(me.alphamode.mclong.math.BigInteger p_62238_, me.alphamode.mclong.math.BigInteger p_62239_) {
+      return this.getChunk(p_62238_, p_62239_, ChunkStatus.f_62326_, false) != null;
+   }
+
+   @Nullable
+   public abstract ChunkAccess getChunk(me.alphamode.mclong.math.BigInteger p_62223_, me.alphamode.mclong.math.BigInteger p_62224_, ChunkStatus p_62225_, boolean p_62226_);
 
    public abstract void m_201698_(BooleanSupplier p_202162_, boolean p_202163_);
 
