--- a/net/minecraft/world/level/chunk/ChunkStatus.java
+++ b/net/minecraft/world/level/chunk/ChunkStatus.java
@@ -43,7 +_,7 @@
    };
    public static final ChunkStatus f_62314_ = m_62414_("empty", (ChunkStatus)null, -1, f_62327_, ChunkStatus.ChunkType.PROTOCHUNK, (p_156307_, p_156308_, p_156309_, p_156310_, p_156311_) -> {
    });
-   public static final ChunkStatus f_62315_ = m_62406_("structure_starts", f_62314_, 0, f_62327_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223361_, p_223362_, p_223363_, p_223364_, p_223365_, p_223366_, p_223367_, p_223368_, p_223369_, p_223370_) -> {
+   public static final ChunkStatus f_62315_ = register("structure_starts", f_62314_, 0, f_62327_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223361_, p_223362_, p_223363_, p_223364_, p_223365_, p_223366_, p_223367_, p_223368_, p_223369_, p_223370_) -> {
       if (!p_223369_.m_6415_().m_62427_(p_223361_)) {
          if (p_223363_.m_7654_().m_129910_().m_246337_().m_247749_()) {
             p_223364_.m_255037_(p_223363_.m_9598_(), p_223363_.m_7726_().m_255415_(), p_223363_.m_215010_(), p_223369_, p_223365_);
@@ -124,7 +_,7 @@
    });
    public static final ChunkStatus f_62321_ = m_62414_("liquid_carvers", f_62320_, 8, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_196805_, p_196806_, p_196807_, p_196808_, p_196809_) -> {
    });
-   public static final ChunkStatus f_62322_ = m_62406_("features", f_62321_, 8, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223314_, p_223315_, p_223316_, p_223317_, p_223318_, p_223319_, p_223320_, p_223321_, p_223322_, p_223323_) -> {
+   public static final ChunkStatus f_62322_ = register("features", f_62321_, 8, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223314_, p_223315_, p_223316_, p_223317_, p_223318_, p_223319_, p_223320_, p_223321_, p_223322_, p_223323_) -> {
       ProtoChunk protochunk = (ProtoChunk)p_223322_;
       protochunk.m_63209_(p_223319_);
       if (p_223323_ || !p_223322_.m_6415_().m_62427_(p_223314_)) {
@@ -139,7 +_,7 @@
    }, (p_223307_, p_223308_, p_223309_, p_223310_, p_223311_, p_223312_) -> {
       return p_223310_.m_215136_(p_223312_).thenApply(Either::left);
    });
-   public static final ChunkStatus f_62323_ = m_62406_("light", f_62322_, 1, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223296_, p_223297_, p_223298_, p_223299_, p_223300_, p_223301_, p_223302_, p_223303_, p_223304_, p_223305_) -> {
+   public static final ChunkStatus f_62323_ = register("light", f_62322_, 1, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223296_, p_223297_, p_223298_, p_223299_, p_223300_, p_223301_, p_223302_, p_223303_, p_223304_, p_223305_) -> {
       return m_62388_(p_223296_, p_223301_, p_223304_);
    }, (p_223289_, p_223290_, p_223291_, p_223292_, p_223293_, p_223294_) -> {
       return m_62388_(p_223289_, p_223292_, p_223294_);
@@ -152,7 +_,7 @@
    });
    public static final ChunkStatus f_62325_ = m_62414_("heightmaps", f_62324_, 0, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_223254_, p_223255_, p_223256_, p_223257_, p_223258_) -> {
    });
-   public static final ChunkStatus f_62326_ = m_62406_("full", f_62325_, 0, f_62328_, ChunkStatus.ChunkType.LEVELCHUNK, (p_223267_, p_223268_, p_223269_, p_223270_, p_223271_, p_223272_, p_223273_, p_223274_, p_223275_, p_223276_) -> {
+   public static final ChunkStatus f_62326_ = register("full", f_62325_, 0, f_62328_, ChunkStatus.ChunkType.LEVELCHUNK, (p_223267_, p_223268_, p_223269_, p_223270_, p_223271_, p_223272_, p_223273_, p_223274_, p_223275_, p_223276_) -> {
       return p_223273_.apply(p_223275_);
    }, (p_223260_, p_223261_, p_223262_, p_223263_, p_223264_, p_223265_) -> {
       return p_223264_.apply(p_223265_);
@@ -175,7 +_,7 @@
    private final ChunkStatus f_62334_;
    private final ChunkStatus.GenerationTask f_62335_;
    private final ChunkStatus.LoadingTask f_62336_;
-   private final int f_62337_;
+   private final long f_62337_;
    private final ChunkStatus.ChunkType f_62338_;
    private final EnumSet<Heightmap.Types> f_62339_;
 
@@ -193,10 +_,10 @@
    }
 
    private static ChunkStatus m_62399_(String p_62400_, @Nullable ChunkStatus p_62401_, int p_62402_, EnumSet<Heightmap.Types> p_62403_, ChunkStatus.ChunkType p_62404_, ChunkStatus.GenerationTask p_62405_) {
-      return m_62406_(p_62400_, p_62401_, p_62402_, p_62403_, p_62404_, p_62405_, f_62329_);
+      return register(p_62400_, p_62401_, p_62402_, p_62403_, p_62404_, p_62405_, f_62329_);
    }
 
-   private static ChunkStatus m_62406_(String p_62407_, @Nullable ChunkStatus p_62408_, int p_62409_, EnumSet<Heightmap.Types> p_62410_, ChunkStatus.ChunkType p_62411_, ChunkStatus.GenerationTask p_62412_, ChunkStatus.LoadingTask p_62413_) {
+   private static ChunkStatus register(String p_62407_, @Nullable ChunkStatus p_62408_, long p_62409_, EnumSet<Heightmap.Types> p_62410_, ChunkStatus.ChunkType p_62411_, ChunkStatus.GenerationTask p_62412_, ChunkStatus.LoadingTask p_62413_) {
       return Registry.m_122961_(BuiltInRegistries.f_256940_, p_62407_, new ChunkStatus(p_62407_, p_62408_, p_62409_, p_62410_, p_62411_, p_62412_, p_62413_));
    }
 
@@ -217,11 +_,11 @@
       return p_62394_.m_6415_().m_62427_(p_62393_) && p_62394_.m_6332_();
    }
 
-   public static ChunkStatus m_156185_(int p_156186_) {
+   public static ChunkStatus getStatusAroundFullChunk(long p_156186_) {
       if (p_156186_ >= f_62330_.size()) {
          return f_62314_;
       } else {
-         return p_156186_ < 0 ? f_62326_ : f_62330_.get(p_156186_);
+         return p_156186_ < 0 ? f_62326_ : f_62330_.get((int) p_156186_);
       }
    }
 
@@ -233,7 +_,7 @@
       return f_62331_.getInt(p_62371_.m_62445_());
    }
 
-   public ChunkStatus(String p_62342_, @Nullable ChunkStatus p_62343_, int p_62344_, EnumSet<Heightmap.Types> p_62345_, ChunkStatus.ChunkType p_62346_, ChunkStatus.GenerationTask p_62347_, ChunkStatus.LoadingTask p_62348_) {
+   public ChunkStatus(String p_62342_, @Nullable ChunkStatus p_62343_, long p_62344_, EnumSet<Heightmap.Types> p_62345_, ChunkStatus.ChunkType p_62346_, ChunkStatus.GenerationTask p_62347_, ChunkStatus.LoadingTask p_62348_) {
       this.f_62332_ = p_62342_;
       this.f_62334_ = p_62343_ == null ? this : p_62343_;
       this.f_62335_ = p_62347_;
@@ -270,7 +_,7 @@
       return this.f_62336_.m_223381_(this, p_223245_, p_223246_, p_223247_, p_223248_, p_223249_);
    }
 
-   public int m_62488_() {
+   public long getRange() {
       return this.f_62337_;
    }
 
