--- a/net/minecraft/world/level/chunk/ChunkStatus.java
+++ b/net/minecraft/world/level/chunk/ChunkStatus.java
@@ -89,7 +_,7 @@
    public static final ChunkStatus f_62322_ = m_62414_("features", f_62320_, 8, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_281188_, p_281189_, p_281190_, p_281191_, p_281192_) -> {
       Heightmap.m_64256_(p_281192_, EnumSet.of(Heightmap.Types.MOTION_BLOCKING, Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, Heightmap.Types.OCEAN_FLOOR, Heightmap.Types.WORLD_SURFACE));
       WorldGenRegion worldgenregion = new WorldGenRegion(p_281189_, p_281191_, p_281188_, 1);
-      p_281190_.m_213609_(worldgenregion, p_281192_, p_281189_.m_215010_().m_220468_(worldgenregion));
+//      p_281190_.applyBiomeDecoration(worldgenregion, p_281192_, p_281189_.structureManager().forWorldGenRegion(worldgenregion));
       Blender.m_197031_(worldgenregion, p_281192_);
    });
    public static final ChunkStatus f_279614_ = m_280108_("initialize_light", f_62322_, 0, false, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_281179_, p_281180_, p_281181_, p_281182_, p_281183_, p_281184_, p_281185_, p_281186_, p_281187_) -> {
@@ -104,7 +_,7 @@
    });
    public static final ChunkStatus f_62324_ = m_62414_("spawn", f_62323_, 0, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_196758_, p_196759_, p_196760_, p_196761_, p_196762_) -> {
       if (!p_196762_.m_187679_()) {
-         p_196760_.m_6929_(new WorldGenRegion(p_196759_, p_196761_, p_196758_, -1));
+//         p_196760_.spawnOriginalMobs(new WorldGenRegion(p_196759_, p_196761_, p_196758_, -1));
       }
 
    });
