--- a/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java
+++ b/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java
@@ -125,10 +_,10 @@
             int l1 = (int)Math.round(Math.sin(d0) * d1);
             RandomSource randomsource1 = randomsource.m_213769_();
             list.add(CompletableFuture.supplyAsync(() -> {
-               Pair<BlockPos, Holder<Biome>> pair = this.f_254681_.m_220570_(SectionPos.m_175554_(k1, 8), 0, SectionPos.m_175554_(l1, 8), 112, holderset::m_203333_, randomsource1, this.f_254677_.m_224579_());
+               Pair<BlockPos, Holder<Biome>> pair = this.f_254681_.findBiomeHorizontal(SectionPos.sectionToBlockCoord(k1, 8), 0, SectionPos.sectionToBlockCoord(l1, 8), 112, holderset::m_203333_, randomsource1, this.f_254677_.m_224579_());
                if (pair != null) {
                   BlockPos blockpos = pair.getFirst();
-                  return new ChunkPos(SectionPos.m_123171_(blockpos.m_123341_()), SectionPos.m_123171_(blockpos.m_123343_()));
+                  return new ChunkPos(SectionPos.blockToSectionCoord(blockpos.getBigX()), SectionPos.blockToSectionCoord(blockpos.getBigZ()));
                } else {
                   return new ChunkPos(k1, l1);
                }
@@ -176,12 +_,12 @@
       return this.f_254677_;
    }
 
-   public boolean m_254936_(Holder<StructureSet> p_256489_, int p_256593_, int p_256115_, int p_256619_) {
+   public boolean hasStructureChunkInRange(Holder<StructureSet> p_256489_, me.alphamode.mclong.math.BigInteger p_256593_, me.alphamode.mclong.math.BigInteger p_256115_, long p_256619_) {
       StructurePlacement structureplacement = p_256489_.m_203334_().f_210004_();
 
-      for(int i = p_256593_ - p_256619_; i <= p_256593_ + p_256619_; ++i) {
-         for(int j = p_256115_ - p_256619_; j <= p_256115_ + p_256619_; ++j) {
-            if (structureplacement.m_255071_(this, i, j)) {
+      for(me.alphamode.mclong.math.BigInteger i = p_256593_.subtract(p_256619_); i.compareTo(p_256593_.add(p_256619_)) <= 0; i = i.add()) {
+         for(me.alphamode.mclong.math.BigInteger j = p_256115_.subtract(p_256619_); j.compareTo(p_256115_.add(p_256619_)) <= 0; j = j.add()) {
+            if (structureplacement.isStructureChunk(this, i, j)) {
                return true;
             }
          }
