--- a/net/minecraft/world/level/chunk/ChunkStatus.java
+++ b/net/minecraft/world/level/chunk/ChunkStatus.java
@@ -102,7 +_,7 @@
    }, (p_284898_, p_284899_, p_284900_, p_284901_, p_284902_, p_284903_) -> {
       return m_284159_(p_284901_, p_284903_);
    });
-   public static final ChunkStatus f_62324_ = m_62414_("spawn", f_62323_, 0, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_196758_, p_196759_, p_196760_, p_196761_, p_196762_) -> {
+   public static final ChunkStatus f_62324_ = m_62414_("spawn", f_62323_, 3, f_62328_, ChunkStatus.ChunkType.PROTOCHUNK, (p_196758_, p_196759_, p_196760_, p_196761_, p_196762_) -> {
       if (!p_196762_.m_187679_()) {
          p_196760_.m_6929_(new WorldGenRegion(p_196759_, p_196761_, p_196758_, -1));
       }
