--- a/net/minecraft/world/level/chunk/DataLayer.java
+++ b/net/minecraft/world/level/chunk/DataLayer.java
@@ -30,16 +_,16 @@
       }
    }
 
-   public int m_62560_(int p_62561_, int p_62562_, int p_62563_) {
-      return this.m_62570_(m_6406_(p_62561_, p_62562_, p_62563_));
-   }
-
-   public void m_62564_(int p_62565_, int p_62566_, int p_62567_, int p_62568_) {
-      this.m_62557_(m_6406_(p_62565_, p_62566_, p_62567_), p_62568_);
-   }
-
-   private static int m_6406_(int p_62572_, int p_62573_, int p_62574_) {
-      return p_62573_ << 8 | p_62574_ << 4 | p_62572_;
+   public int get(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      return this.m_62570_(getIndex(pX, pY, pZ));
+   }
+
+   public void set(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ, int pValue) {
+      this.m_62557_(getIndex(pX, pY, pZ), pValue);
+   }
+
+   private static int getIndex(me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      return pY.shiftLeft(8).or(pZ.shiftLeft(4)).or(pX).intValue();
    }
 
    private int m_62570_(int p_62571_) {
