--- a/net/minecraft/world/level/chunk/EmptyLevelChunk.java
+++ b/net/minecraft/world/level/chunk/EmptyLevelChunk.java
@@ -56,7 +_,7 @@
       return true;
    }
 
-   public boolean m_5566_(int p_62587_, int p_62588_) {
+   public boolean isYSpaceEmpty(long p_62587_, long p_62588_) {
       return true;
    }
 
@@ -64,7 +_,31 @@
       return ChunkHolder.FullChunkStatus.BORDER;
    }
 
-   public Holder<Biome> m_203495_(int p_204426_, int p_204427_, int p_204428_) {
+   public Holder<Biome> getNoiseBiome(long p_204426_, long p_204427_, long p_204428_) {
       return this.f_204420_;
    }
+
+   public ca.spottedleaf.starlight.common.light.SWMRNibbleArray[] getBlockNibbles() {
+      return ca.spottedleaf.starlight.common.light.StarLightEngine.getFilledEmptyLight(this.m_62953_());
+   }
+
+   public void setBlockNibbles(final ca.spottedleaf.starlight.common.light.SWMRNibbleArray[] nibbles) {}
+
+   public ca.spottedleaf.starlight.common.light.SWMRNibbleArray[] getSkyNibbles() {
+      return ca.spottedleaf.starlight.common.light.StarLightEngine.getFilledEmptyLight(this.m_62953_());
+   }
+
+   public void setSkyNibbles(final ca.spottedleaf.starlight.common.light.SWMRNibbleArray[] nibbles) {}
+
+   public boolean[] getSkyEmptinessMap() {
+      return null;
+   }
+
+   public void setSkyEmptinessMap(final boolean[] emptinessMap) {}
+
+   public boolean[] getBlockEmptinessMap() {
+      return null;
+   }
+
+   public void setBlockEmptinessMap(final boolean[] emptinessMap) {}
 }
