--- a/net/minecraft/world/level/chunk/GlobalPalette.java
+++ b/net/minecraft/world/level/chunk/GlobalPalette.java
@@ -12,7 +_,7 @@
       this.f_62639_ = p_187897_;
    }
 
-   public static <A> Palette<A> m_187898_(int p_187899_, IdMap<A> p_187900_, PaletteResize<A> p_187901_, List<A> p_187902_) {
+   public static <A> Palette<A> create(long p_187899_, IdMap<A> p_187900_, PaletteResize<A> p_187901_, List<A> p_187902_) {
       return new GlobalPalette<>(p_187900_);
    }
 
