--- a/net/minecraft/world/level/chunk/GlobalPalette.java
+++ b/net/minecraft/world/level/chunk/GlobalPalette.java
@@ -12,8 +_,8 @@
       this.f_62639_ = p_187897_;
    }
 
-   public static <A> Palette<A> m_187898_(int p_187899_, IdMap<A> p_187900_, PaletteResize<A> p_187901_, List<A> p_187902_) {
-      return new GlobalPalette<>(p_187900_);
+   public static <A> Palette<A> create(long pBits, IdMap<A> pRegistry, PaletteResize<A> pResizeHandler, List<A> pValues) {
+      return new GlobalPalette<>(pRegistry);
    }
 
    public int m_6796_(T p_62648_) {
