--- a/net/minecraft/world/level/chunk/ImposterProtoChunk.java
+++ b/net/minecraft/world/level/chunk/ImposterProtoChunk.java
@@ -103,12 +_,12 @@
       return this.f_62685_.m_6005_(p_187928_);
    }
 
-   public int m_5885_(Heightmap.Types p_62702_, int p_62703_, int p_62704_) {
-      return this.f_62685_.m_5885_(this.m_62741_(p_62702_), p_62703_, p_62704_);
+   public long getHeight(Heightmap.Types p_62702_, long p_62703_, long p_62704_) {
+      return this.f_62685_.getHeight(this.m_62741_(p_62702_), p_62703_, p_62704_);
    }
 
-   public Holder<Biome> m_203495_(int p_204430_, int p_204431_, int p_204432_) {
-      return this.f_62685_.m_203495_(p_204430_, p_204431_, p_204432_);
+   public Holder<Biome> getNoiseBiome(long p_204430_, long p_204431_, long p_204432_) {
+      return this.f_62685_.getNoiseBiome(p_204430_, p_204431_, p_204432_);
    }
 
    public ChunkPos m_7697_() {
