--- a/net/minecraft/world/level/chunk/LevelChunkSection.java
+++ b/net/minecraft/world/level/chunk/LevelChunkSection.java
@@ -35,12 +_,12 @@
       this.f_187995_ = new PalettedContainer<>(p_282873_.m_206115_(), p_282873_.m_246971_(Biomes.f_48202_), PalettedContainer.Strategy.f_188138_);
    }
 
-   public BlockState m_62982_(int p_62983_, int p_62984_, int p_62985_) {
-      return this.f_62972_.m_63087_(p_62983_, p_62984_, p_62985_);
+   public BlockState getBlockState(long p_62983_, long p_62984_, long p_62985_) {
+      return this.f_62972_.get(p_62983_, p_62984_, p_62985_);
    }
 
-   public FluidState m_63007_(int p_63008_, int p_63009_, int p_63010_) {
-      return this.f_62972_.m_63087_(p_63008_, p_63009_, p_63010_).m_60819_();
+   public FluidState getFluidState(long p_63008_, long p_63009_, long p_63010_) {
+      return this.f_62972_.get(p_63008_, p_63009_, p_63010_).m_60819_();
    }
 
    public void m_62981_() {
@@ -51,16 +_,16 @@
       this.f_62972_.m_63120_();
    }
 
-   public BlockState m_62986_(int p_62987_, int p_62988_, int p_62989_, BlockState p_62990_) {
-      return this.m_62991_(p_62987_, p_62988_, p_62989_, p_62990_, true);
+   public BlockState setBlockState(long p_62987_, long p_62988_, long p_62989_, BlockState p_62990_) {
+      return this.setBlockState(p_62987_, p_62988_, p_62989_, p_62990_, true);
    }
 
-   public BlockState m_62991_(int p_62992_, int p_62993_, int p_62994_, BlockState p_62995_, boolean p_62996_) {
+   public BlockState setBlockState(long p_62992_, long p_62993_, long p_62994_, BlockState p_62995_, boolean p_62996_) {
       BlockState blockstate;
       if (p_62996_) {
-         blockstate = this.f_62972_.m_63091_(p_62992_, p_62993_, p_62994_, p_62995_);
+         blockstate = this.f_62972_.getAndSet(p_62992_, p_62993_, p_62994_, p_62995_);
       } else {
-         blockstate = this.f_62972_.m_63127_(p_62992_, p_62993_, p_62994_, p_62995_);
+         blockstate = this.f_62972_.getAndSetUnchecked(p_62992_, p_62993_, p_62994_, p_62995_);
       }
 
       FluidState fluidstate = blockstate.m_60819_();
@@ -174,18 +_,18 @@
       return this.f_62972_.m_63109_(p_63003_);
    }
 
-   public Holder<Biome> m_204433_(int p_204434_, int p_204435_, int p_204436_) {
-      return this.f_187995_.m_63087_(p_204434_, p_204435_, p_204436_);
+   public Holder<Biome> getNoiseBiome(long p_204434_, long p_204435_, long p_204436_) {
+      return this.f_187995_.get(p_204434_, p_204435_, p_204436_);
    }
 
-   public void m_280631_(BiomeResolver p_282075_, Climate.Sampler p_283084_, int p_282310_, int p_281510_, int p_283057_) {
+   public void fillBiomesFromNoise(BiomeResolver p_282075_, Climate.Sampler p_283084_, long p_282310_, long p_281510_, long p_283057_) {
       PalettedContainer<Holder<Biome>> palettedcontainer = this.f_187995_.m_238334_();
       int i = 4;
 
       for(int j = 0; j < 4; ++j) {
          for(int k = 0; k < 4; ++k) {
             for(int l = 0; l < 4; ++l) {
-               palettedcontainer.m_63127_(j, k, l, p_282075_.m_203407_(p_282310_ + j, p_281510_ + k, p_283057_ + l, p_283084_));
+               palettedcontainer.getAndSetUnchecked(j, k, l, p_282075_.getNoiseBiome(p_282310_ + j, p_281510_ + k, p_283057_ + l, p_283084_));
             }
          }
       }
