--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -72,7 +_,7 @@
    private Supplier<FullChunkStatus> f_62790_;
    @Nullable
    private LevelChunk.PostLoadProcessor f_62791_;
-   private final Int2ObjectMap<GameEventListenerRegistry> f_244451_;
+   private final it.unimi.dsi.fastutil.longs.Long2ObjectMap<GameEventListenerRegistry> f_244451_;
    private final LevelChunkTicks<Block> f_62784_;
    private final LevelChunkTicks<Fluid> f_187943_;
 
@@ -83,7 +_,7 @@
    public LevelChunk(Level p_196854_, ChunkPos p_196855_, UpgradeData p_196856_, LevelChunkTicks<Block> p_196857_, LevelChunkTicks<Fluid> p_196858_, long p_196859_, @Nullable LevelChunkSection[] p_196860_, @Nullable LevelChunk.PostLoadProcessor p_196861_, @Nullable BlendingData p_196862_) {
       super(p_196855_, p_196856_, p_196854_, p_196854_.m_9598_().m_175515_(Registries.f_256952_), p_196859_, p_196860_, p_196862_);
       this.f_62776_ = p_196854_;
-      this.f_244451_ = new Int2ObjectOpenHashMap<>();
+      this.f_244451_ = new it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap<>();
 
       for(Heightmap.Types heightmap$types : Heightmap.Types.values()) {
          if (ChunkStatus.f_62326_.m_62500_().contains(heightmap$types)) {
@@ -135,21 +_,21 @@
       return new ChunkAccess.TicksToSave(this.f_62784_, this.f_187943_);
    }
 
-   public GameEventListenerRegistry m_246686_(int p_251193_) {
+   public GameEventListenerRegistry getListenerRegistry(long p_251193_) {
       Level level = this.f_62776_;
       if (level instanceof ServerLevel serverlevel) {
          return this.f_244451_.computeIfAbsent(p_251193_, (p_281221_) -> {
-            return new EuclideanGameEventListenerRegistry(serverlevel, p_251193_, this::m_280642_);
+            return new EuclideanGameEventListenerRegistry(serverlevel, p_251193_, this::removeGameEventListenerRegistry);
          });
       } else {
-         return super.m_246686_(p_251193_);
+         return super.getListenerRegistry(p_251193_);
       }
    }
 
    public BlockState m_8055_(BlockPos p_62923_) {
-      int i = p_62923_.m_123341_();
-      int j = p_62923_.m_123342_();
-      int k = p_62923_.m_123343_();
+      long i = p_62923_.getX();
+      long j = p_62923_.getY();
+      long k = p_62923_.getZ();
       if (this.f_62776_.m_46659_()) {
          BlockState blockstate = null;
          if (j == 60) {
@@ -157,17 +_,17 @@
          }
 
          if (j == 70) {
-            blockstate = DebugLevelSource.m_64148_(i, k);
+            blockstate = DebugLevelSource.getBlockStateFor(i, k);
          }
 
          return blockstate == null ? Blocks.f_50016_.m_49966_() : blockstate;
       } else {
          try {
-            int l = this.m_151564_(j);
+            long l = this.getSectionIndex(j);
             if (l >= 0 && l < this.f_187612_.length) {
-               LevelChunkSection levelchunksection = this.f_187612_[l];
+               LevelChunkSection levelchunksection = this.f_187612_[(int) l]; // TODO: long support
                if (!levelchunksection.m_188008_()) {
-                  return levelchunksection.m_62982_(i & 15, j & 15, k & 15);
+                  return levelchunksection.getBlockState(i & 15, j & 15, k & 15);
                }
             }
 
@@ -176,7 +_,7 @@
             CrashReport crashreport = CrashReport.m_127521_(throwable, "Getting block state");
             CrashReportCategory crashreportcategory = crashreport.m_127514_("Block being got");
             crashreportcategory.m_128165_("Location", () -> {
-               return CrashReportCategory.m_178942_(this, i, j, k);
+               return CrashReportCategory.formatLocation(this, i, j, k);
             });
             throw new ReportedException(crashreport);
          }
@@ -184,16 +_,16 @@
    }
 
    public FluidState m_6425_(BlockPos p_62895_) {
-      return this.m_62814_(p_62895_.m_123341_(), p_62895_.m_123342_(), p_62895_.m_123343_());
+      return this.getFluidState(p_62895_.getX(), p_62895_.getY(), p_62895_.getZ());
    }
 
-   public FluidState m_62814_(int p_62815_, int p_62816_, int p_62817_) {
+   public FluidState getFluidState(long p_62815_, long p_62816_, long p_62817_) {
       try {
-         int i = this.m_151564_(p_62816_);
+         long i = this.getSectionIndex(p_62816_);
          if (i >= 0 && i < this.f_187612_.length) {
-            LevelChunkSection levelchunksection = this.f_187612_[i];
+            LevelChunkSection levelchunksection = this.f_187612_[(int) i]; // TODO: long support
             if (!levelchunksection.m_188008_()) {
-               return levelchunksection.m_63007_(p_62815_ & 15, p_62816_ & 15, p_62817_ & 15);
+               return levelchunksection.getFluidState(p_62815_ & 15, p_62816_ & 15, p_62817_ & 15);
             }
          }
 
@@ -202,7 +_,7 @@
          CrashReport crashreport = CrashReport.m_127521_(throwable, "Getting fluid state");
          CrashReportCategory crashreportcategory = crashreport.m_127514_("Block being got");
          crashreportcategory.m_128165_("Location", () -> {
-            return CrashReportCategory.m_178942_(this, p_62815_, p_62816_, p_62817_);
+            return CrashReportCategory.formatLocation(this, p_62815_, p_62816_, p_62817_);
          });
          throw new ReportedException(crashreport);
       }
@@ -210,24 +_,24 @@
 
    @Nullable
    public BlockState m_6978_(BlockPos p_62865_, BlockState p_62866_, boolean p_62867_) {
-      int i = p_62865_.m_123342_();
-      LevelChunkSection levelchunksection = this.m_183278_(this.m_151564_(i));
+      long i = p_62865_.getY();
+      LevelChunkSection levelchunksection = this.getSection(this.getSectionIndex(i));
       boolean flag = levelchunksection.m_188008_();
       if (flag && p_62866_.m_60795_()) {
          return null;
       } else {
-         int j = p_62865_.m_123341_() & 15;
-         int k = i & 15;
-         int l = p_62865_.m_123343_() & 15;
-         BlockState blockstate = levelchunksection.m_62986_(j, k, l, p_62866_);
+         long j = p_62865_.getX() & 15;
+         long k = i & 15;
+         long l = p_62865_.getZ() & 15;
+         BlockState blockstate = levelchunksection.setBlockState(j, k, l, p_62866_);
          if (blockstate == p_62866_) {
             return null;
          } else {
             Block block = p_62866_.m_60734_();
-            this.f_187608_.get(Heightmap.Types.MOTION_BLOCKING).m_64249_(j, i, l, p_62866_);
-            this.f_187608_.get(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES).m_64249_(j, i, l, p_62866_);
-            this.f_187608_.get(Heightmap.Types.OCEAN_FLOOR).m_64249_(j, i, l, p_62866_);
-            this.f_187608_.get(Heightmap.Types.WORLD_SURFACE).m_64249_(j, i, l, p_62866_);
+            this.f_187608_.get(Heightmap.Types.MOTION_BLOCKING).update(j, i, l, p_62866_);
+            this.f_187608_.get(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES).update(j, i, l, p_62866_);
+            this.f_187608_.get(Heightmap.Types.OCEAN_FLOOR).update(j, i, l, p_62866_);
+            this.f_187608_.get(Heightmap.Types.WORLD_SURFACE).update(j, i, l, p_62866_);
             boolean flag1 = levelchunksection.m_188008_();
             if (flag != flag1) {
                this.f_62776_.m_7726_().m_7827_().m_75834_(p_62865_, flag1);
@@ -236,7 +_,7 @@
             if (LightEngine.m_284387_(this, p_62865_, blockstate, p_62866_)) {
                ProfilerFiller profilerfiller = this.f_62776_.m_46473_();
                profilerfiller.m_6180_("updateSkyLightSources");
-               this.f_283754_.m_284521_(this, j, i, l);
+               this.f_283754_.update(this, j, i, l);
                profilerfiller.m_6182_("queueCheckLight");
                this.f_62776_.m_7726_().m_7827_().m_7174_(p_62865_);
                profilerfiller.m_7238_();
@@ -249,7 +_,7 @@
                this.m_8114_(p_62865_);
             }
 
-            if (!levelchunksection.m_62982_(j, k, l).m_60713_(block)) {
+            if (!levelchunksection.getBlockState(j, k, l).m_60713_(block)) {
                return null;
             } else {
                if (!this.f_62776_.f_46443_) {
@@ -347,7 +_,7 @@
             return true;
          } else {
             ServerLevel serverlevel = (ServerLevel)level;
-            return this.m_287138_().m_287205_(FullChunkStatus.BLOCK_TICKING) && serverlevel.m_143319_(ChunkPos.m_151388_(p_156411_));
+            return this.m_287138_().m_287205_(FullChunkStatus.BLOCK_TICKING) && serverlevel.areEntitiesLoaded(new ChunkPos(p_156411_));
          }
       }
    }
@@ -405,15 +_,15 @@
       if (block instanceof EntityBlock) {
          GameEventListener gameeventlistener = ((EntityBlock)block).m_214009_(p_223414_, p_223413_);
          if (gameeventlistener != null) {
-            int i = SectionPos.m_123171_(p_223413_.m_58899_().m_123342_());
-            GameEventListenerRegistry gameeventlistenerregistry = this.m_246686_(i);
+            long i = SectionPos.blockToSectionCoord(p_223413_.m_58899_().getY());
+            GameEventListenerRegistry gameeventlistenerregistry = this.getListenerRegistry(i);
             gameeventlistenerregistry.m_246052_(gameeventlistener);
          }
       }
 
    }
 
-   private void m_280642_(int p_283355_) {
+   private void removeGameEventListenerRegistry(long p_283355_) {
       this.f_244451_.remove(p_283355_);
    }
 
@@ -486,7 +_,7 @@
       for(int i = 0; i < this.f_187602_.length; ++i) {
          if (this.f_187602_[i] != null) {
             for(Short oshort : this.f_187602_[i]) {
-               BlockPos blockpos = ProtoChunk.m_63227_(oshort, this.m_151568_(i), chunkpos);
+               BlockPos blockpos = ProtoChunk.unpackOffsetCoordinates(oshort, this.getSectionYFromSectionIndex(i), chunkpos);
                BlockState blockstate = this.m_8055_(blockpos);
                FluidState fluidstate = blockstate.m_60819_();
                if (!fluidstate.m_76178_()) {
@@ -588,7 +_,7 @@
       if (block instanceof EntityBlock) {
          GameEventListener gameeventlistener = ((EntityBlock)block).m_214009_(p_223417_, p_223416_);
          if (gameeventlistener != null) {
-            this.m_246686_(SectionPos.m_123171_(p_223416_.m_58899_().m_123342_())).m_245531_(gameeventlistener);
+            this.getListenerRegistry(SectionPos.blockToSectionCoord(p_223416_.m_58899_().getY())).m_245531_(gameeventlistener);
          }
       }
 
