--- a/net/minecraft/world/level/chunk/LightChunkGetter.java
+++ b/net/minecraft/world/level/chunk/LightChunkGetter.java
@@ -7,7 +_,7 @@
 
 public interface LightChunkGetter {
    @Nullable
-   LightChunk m_6196_(int p_63023_, int p_63024_);
+   LightChunk getChunkForLighting(me.alphamode.mclong.math.BigInteger p_63023_, me.alphamode.mclong.math.BigInteger p_63024_);
 
    default void m_6506_(LightLayer p_63021_, SectionPos p_63022_) {
    }
