--- a/net/minecraft/world/level/chunk/PalettedContainerRO.java
+++ b/net/minecraft/world/level/chunk/PalettedContainerRO.java
@@ -10,7 +_,7 @@
 import net.minecraft.network.FriendlyByteBuf;
 
 public interface PalettedContainerRO<T> {
-   T m_63087_(int p_238291_, int p_238292_, int p_238293_);
+   T get(me.alphamode.mclong.math.BigInteger p_238291_, me.alphamode.mclong.math.BigInteger p_238292_, me.alphamode.mclong.math.BigInteger p_238293_);
 
    void m_196879_(Consumer<T> p_238353_);
 
