--- a/net/minecraft/world/level/chunk/SingleValuePalette.java
+++ b/net/minecraft/world/level/chunk/SingleValuePalette.java
@@ -23,7 +_,7 @@
 
    }
 
-   public static <A> Palette<A> m_188213_(int p_188214_, IdMap<A> p_188215_, PaletteResize<A> p_188216_, List<A> p_188217_) {
+   public static <A> Palette<A> create(long p_188214_, IdMap<A> p_188215_, PaletteResize<A> p_188216_, List<A> p_188217_) {
       return new SingleValuePalette<>(p_188215_, p_188216_, p_188217_);
    }
 
