--- a/net/minecraft/world/level/chunk/StructureAccess.java
+++ b/net/minecraft/world/level/chunk/StructureAccess.java
@@ -12,11 +_,11 @@
 
    void m_213792_(Structure p_223437_, StructureStart p_223438_);
 
-   LongSet m_213649_(Structure p_223439_);
-
-   void m_213843_(Structure p_223435_, long p_223436_);
-
-   Map<Structure, LongSet> m_62769_();
-
-   void m_62737_(Map<Structure, LongSet> p_223440_);
+   it.unimi.dsi.fastutil.objects.ObjectSet<net.minecraft.world.level.ChunkPos> getReferencesForStructure(Structure p_223439_);
+
+   void addReferenceForStructure(Structure p_223435_, net.minecraft.world.level.ChunkPos p_223436_);
+
+   Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<net.minecraft.world.level.ChunkPos>> m_62769_();
+
+   void m_62737_(Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<net.minecraft.world.level.ChunkPos>> p_223440_);
 }
