--- a/net/minecraft/client/gui/components/SubtitleOverlay.java
+++ b/net/minecraft/client/gui/components/SubtitleOverlay.java
@@ -97,13 +_,13 @@
          if (!this.f_94638_.isEmpty()) {
             for(SubtitleOverlay.Subtitle subtitleoverlay$subtitle : this.f_94638_) {
                if (subtitleoverlay$subtitle.m_94655_().equals(component)) {
-                  subtitleoverlay$subtitle.m_94656_(new Vec3(p_94645_.m_7772_(), p_94645_.m_7780_(), p_94645_.m_7778_()));
+                  subtitleoverlay$subtitle.m_94656_(new Vec3(p_94645_.getX(), p_94645_.getY(), p_94645_.getZ()));
                   return;
                }
             }
          }
 
-         this.f_94638_.add(new SubtitleOverlay.Subtitle(component, new Vec3(p_94645_.m_7772_(), p_94645_.m_7780_(), p_94645_.m_7778_())));
+         this.f_94638_.add(new SubtitleOverlay.Subtitle(component, new Vec3(p_94645_.getX(), p_94645_.getY(), p_94645_.getZ())));
       }
    }
 
