--- a/net/minecraft/world/entity/ai/util/AirAndWaterRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/AirAndWaterRandomPos.java
@@ -22,7 +_,7 @@
       } else {
          BlockPos blockpos1 = RandomPos.m_217863_(p_148366_, p_148367_, p_148366_.m_217043_(), blockpos);
          if (!GoalUtils.m_148451_(blockpos1, p_148366_) && !GoalUtils.m_148454_(p_148373_, p_148366_, blockpos1)) {
-            blockpos1 = RandomPos.m_148545_(blockpos1, p_148366_.f_19853_.m_151558_(), (p_148376_) -> {
+            blockpos1 = RandomPos.moveUpOutOfSolid(blockpos1, p_148366_.f_19853_.getMaxBuildHeight(), (p_148376_) -> {
                return GoalUtils.m_148461_(p_148366_, p_148376_);
             });
             return GoalUtils.m_148458_(p_148366_, blockpos1) ? null : blockpos1;
