--- a/net/minecraft/world/entity/ai/util/AirRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/AirRandomPos.java
@@ -11,7 +_,7 @@
       Vec3 vec3 = p_148392_.m_82492_(p_148388_.m_20185_(), p_148388_.m_20186_(), p_148388_.m_20189_());
       boolean flag = GoalUtils.m_148442_(p_148388_, p_148389_);
       return RandomPos.m_148542_(p_148388_, () -> {
-         BlockPos blockpos = AirAndWaterRandomPos.m_148365_(p_148388_, p_148389_, p_148390_, p_148391_, vec3.f_82479_, vec3.f_82481_, p_148393_, flag);
+         BlockPos blockpos = AirAndWaterRandomPos.m_148365_(p_148388_, p_148389_, p_148390_, p_148391_, vec3.m_7096_(), vec3.m_7094_(), p_148393_, flag);
          return blockpos != null && !GoalUtils.m_148445_(p_148388_, blockpos) ? blockpos : null;
       });
    }
