--- a/net/minecraft/world/entity/ai/util/DefaultRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/DefaultRandomPos.java
@@ -20,7 +_,7 @@
       Vec3 vec3 = p_148416_.m_82492_(p_148413_.m_20185_(), p_148413_.m_20186_(), p_148413_.m_20189_());
       boolean flag = GoalUtils.m_148442_(p_148413_, p_148414_);
       return RandomPos.m_148542_(p_148413_, () -> {
-         BlockPos blockpos = RandomPos.m_217855_(p_148413_.m_217043_(), p_148414_, p_148415_, 0, vec3.f_82479_, vec3.f_82481_, p_148417_);
+         BlockPos blockpos = RandomPos.m_217855_(p_148413_.m_217043_(), p_148414_, p_148415_, 0, vec3.m_7096_(), vec3.m_7094_(), p_148417_);
          return blockpos == null ? null : m_148436_(p_148413_, p_148414_, flag, blockpos);
       });
    }
@@ -30,7 +_,7 @@
       Vec3 vec3 = p_148408_.m_20182_().m_82546_(p_148411_);
       boolean flag = GoalUtils.m_148442_(p_148408_, p_148409_);
       return RandomPos.m_148542_(p_148408_, () -> {
-         BlockPos blockpos = RandomPos.m_217855_(p_148408_.m_217043_(), p_148409_, p_148410_, 0, vec3.f_82479_, vec3.f_82481_, (double)((float)Math.PI / 2F));
+         BlockPos blockpos = RandomPos.m_217855_(p_148408_.m_217043_(), p_148409_, p_148410_, 0, vec3.m_7096_(), vec3.m_7094_(), (double)((float)Math.PI / 2F));
          return blockpos == null ? null : m_148436_(p_148408_, p_148409_, flag, blockpos);
       });
    }
