--- a/net/minecraft/world/entity/ai/util/HoverRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/HoverRandomPos.java
@@ -18,7 +_,7 @@
             if (blockpos1 == null) {
                return null;
             } else {
-               blockpos1 = RandomPos.m_26947_(blockpos1, p_148466_.m_217043_().m_188503_(p_148472_ - p_148473_ + 1) + p_148473_, p_148466_.m_9236_().m_151558_(), (p_148486_) -> {
+               blockpos1 = RandomPos.moveUpToAboveSolid(blockpos1, p_148466_.m_217043_().m_188503_(p_148472_ - p_148473_ + 1) + p_148473_, p_148466_.m_9236_().getMaxBuildHeight(), (p_148486_) -> {
                   return GoalUtils.m_148461_(p_148466_, p_148486_);
                });
                return !GoalUtils.m_148445_(p_148466_, blockpos1) && !GoalUtils.m_148458_(p_148466_, blockpos1) ? blockpos1 : null;
