--- a/net/minecraft/world/entity/ai/util/LandRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/LandRandomPos.java
@@ -39,7 +_,7 @@
    @Nullable
    private static Vec3 m_148497_(PathfinderMob p_148498_, int p_148499_, int p_148500_, Vec3 p_148501_, boolean p_148502_) {
       return RandomPos.m_148542_(p_148498_, () -> {
-         BlockPos blockpos = RandomPos.m_217855_(p_148498_.m_217043_(), p_148499_, p_148500_, 0, p_148501_.f_82479_, p_148501_.f_82481_, (double)((float)Math.PI / 2F));
+         BlockPos blockpos = RandomPos.m_217855_(p_148498_.m_217043_(), p_148499_, p_148500_, 0, p_148501_.m_7096_(), p_148501_.m_7094_(), (double)((float)Math.PI / 2F));
          if (blockpos == null) {
             return null;
          } else {
@@ -51,7 +_,7 @@
 
    @Nullable
    public static BlockPos m_148518_(PathfinderMob p_148519_, BlockPos p_148520_) {
-      p_148520_ = RandomPos.m_148545_(p_148520_, p_148519_.m_9236_().m_151558_(), (p_148534_) -> {
+      p_148520_ = RandomPos.moveUpOutOfSolid(p_148520_, p_148519_.m_9236_().getMaxBuildHeight(), (p_148534_) -> {
          return GoalUtils.m_148461_(p_148519_, p_148534_);
       });
       return !GoalUtils.m_148445_(p_148519_, p_148520_) && !GoalUtils.m_148458_(p_148519_, p_148520_) ? p_148520_ : null;
