--- a/net/minecraft/world/entity/ai/util/LandRandomPos.java
+++ b/net/minecraft/world/entity/ai/util/LandRandomPos.java
@@ -51,7 +_,7 @@
 
    @Nullable
    public static BlockPos m_148518_(PathfinderMob p_148519_, BlockPos p_148520_) {
-      p_148520_ = RandomPos.m_148545_(p_148520_, p_148519_.f_19853_.m_151558_(), (p_148534_) -> {
+      p_148520_ = RandomPos.moveUpOutOfSolid(p_148520_, p_148519_.f_19853_.getMaxBuildHeight(), (p_148534_) -> {
          return GoalUtils.m_148461_(p_148519_, p_148534_);
       });
       return !GoalUtils.m_148445_(p_148519_, p_148520_) && !GoalUtils.m_148458_(p_148519_, p_148520_) ? p_148520_ : null;
