--- a/net/minecraft/world/entity/AgeableMob.java
+++ b/net/minecraft/world/entity/AgeableMob.java
@@ -119,7 +_,7 @@
       if (this.m_9236_().f_46443_) {
          if (this.f_146735_ > 0) {
             if (this.f_146735_ % 4 == 0) {
-               this.m_9236_().m_7106_(ParticleTypes.f_123748_, this.m_20208_(1.0D), this.m_20187_() + 0.5D, this.m_20262_(1.0D), 0.0D, 0.0D, 0.0D);
+               this.m_9236_().addParticle(ParticleTypes.f_123748_, this.getRandomX(1.0D), this.getRandomY().add(me.alphamode.mclong.core.BigConstants.AABB), this.getRandomZ(1.0D), 0.0D, 0.0D, 0.0D);
             }
 
             --this.f_146735_;
