--- a/net/minecraft/world/entity/Display.java
+++ b/net/minecraft/world/entity/Display.java
@@ -381,8 +_,8 @@
       return this.f_19804_.m_135370_(f_268639_);
    }
 
-   public void m_6034_(double p_270091_, double p_270983_, double p_270419_) {
-      super.m_6034_(p_270091_, p_270983_, p_270419_);
+   public void setPos(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      super.setPos(pX, pY, pZ);
       this.m_269361_();
    }
 
