--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -394,13 +_,13 @@
       } else {
          double d0;
          if (p_262685_) {
-            t.m_6034_((double)p_262595_.m_123341_() + 0.5D, (double)(p_262595_.m_123342_() + 1), (double)p_262595_.m_123343_() + 0.5D);
+            t.setPos(p_262595_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), (p_262595_.getBigDecY().add()), p_262595_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB));
             d0 = m_20625_(p_262637_, p_262595_, p_262588_, t.m_20191_());
          } else {
             d0 = 0.0D;
          }
 
-         t.m_7678_((double)p_262595_.m_123341_() + 0.5D, (double)p_262595_.m_123342_() + d0, (double)p_262595_.m_123343_() + 0.5D, Mth.m_14177_(p_262637_.f_46441_.m_188501_() * 360.0F), 0.0F);
+         t.moveTo(p_262595_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB), p_262595_.getBigDecY().add(d0), p_262595_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB), Mth.m_14177_(p_262637_.f_46441_.m_188501_() * 360.0F), 0.0F);
          if (t instanceof Mob) {
             Mob mob = (Mob)t;
             mob.f_20885_ = mob.m_146908_();
@@ -424,7 +_,7 @@
       }
 
       Iterable<VoxelShape> iterable = p_20626_.m_186431_((Entity)null, aabb);
-      return 1.0D + Shapes.m_193135_(Direction.Axis.Y, p_20629_, iterable, p_20628_ ? -2.0D : -1.0D);
+      return 1.0D + Shapes.collide(Direction.Axis.Y, p_20629_, iterable, me.alphamode.mclong.math.BigDecimal.valueOf(p_20628_ ? -2.0D : -1.0D)).doubleValue();
    }
 
    public static void m_20620_(Level p_20621_, @Nullable Player p_20622_, @Nullable Entity p_20623_, @Nullable CompoundTag p_20624_) {
@@ -523,9 +_,9 @@
       });
    }
 
-   public AABB m_20585_(double p_20586_, double p_20587_, double p_20588_) {
-      float f = this.m_20678_() / 2.0F;
-      return new AABB(p_20586_ - (double)f, p_20587_, p_20588_ - (double)f, p_20586_ + (double)f, p_20587_ + (double)this.m_20679_(), p_20588_ + (double)f);
+   public AABB getAABB(me.alphamode.mclong.math.BigDecimal pX, me.alphamode.mclong.math.BigDecimal pY, me.alphamode.mclong.math.BigDecimal pZ) {
+      me.alphamode.mclong.math.BigDecimal f = me.alphamode.mclong.math.BigDecimal.valueOf(this.m_20678_() / 2.0F);
+      return new AABB(pX.subtract(f), pY, pZ.subtract(f), pX.add(f), pY.add(this.m_20679_()), pZ.add(f));
    }
 
    public boolean m_20630_(BlockState p_20631_) {
