--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -340,13 +_,13 @@
       } else {
          double d0;
          if (p_20662_) {
-            t.m_6034_((double)p_20660_.m_123341_() + 0.5D, (double)(p_20660_.m_123342_() + 1), (double)p_20660_.m_123343_() + 0.5D);
+            t.m_6034_((double)p_20660_.getX() + 0.5D, (double)(p_20660_.getY() + 1), (double)p_20660_.getZ() + 0.5D);
             d0 = m_20625_(p_20656_, p_20660_, p_20663_, t.m_20191_());
          } else {
             d0 = 0.0D;
          }
 
-         t.m_7678_((double)p_20660_.m_123341_() + 0.5D, (double)p_20660_.m_123342_() + d0, (double)p_20660_.m_123343_() + 0.5D, Mth.m_14177_(p_20656_.f_46441_.m_188501_() * 360.0F), 0.0F);
+         t.m_7678_((double)p_20660_.getX() + 0.5D, (double)p_20660_.getY() + d0, (double)p_20660_.getZ() + 0.5D, Mth.m_14177_(p_20656_.f_46441_.m_188501_() * 360.0F), 0.0F);
          if (t instanceof Mob) {
             Mob mob = (Mob)t;
             mob.f_20885_ = mob.m_146908_();
