--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -394,13 +_,13 @@
       } else {
          double d0;
          if (p_262685_) {
-            t.m_6034_((double)p_262595_.m_123341_() + 0.5D, (double)(p_262595_.m_123342_() + 1), (double)p_262595_.m_123343_() + 0.5D);
+            t.m_6034_((double)p_262595_.getX() + 0.5D, (double)(p_262595_.getY() + 1), (double)p_262595_.getZ() + 0.5D);
             d0 = m_20625_(p_262637_, p_262595_, p_262588_, t.m_20191_());
          } else {
             d0 = 0.0D;
          }
 
-         t.m_7678_((double)p_262595_.m_123341_() + 0.5D, (double)p_262595_.m_123342_() + d0, (double)p_262595_.m_123343_() + 0.5D, Mth.m_14177_(p_262637_.f_46441_.m_188501_() * 360.0F), 0.0F);
+         t.m_7678_((double)p_262595_.getX() + 0.5D, (double)p_262595_.getY() + d0, (double)p_262595_.getZ() + 0.5D, Mth.m_14177_(p_262637_.f_46441_.m_188501_() * 360.0F), 0.0F);
          if (t instanceof Mob) {
             Mob mob = (Mob)t;
             mob.f_20885_ = mob.m_146908_();
@@ -424,7 +_,7 @@
       }
 
       Iterable<VoxelShape> iterable = p_20626_.m_186431_((Entity)null, aabb);
-      return 1.0D + Shapes.m_193135_(Direction.Axis.Y, p_20629_, iterable, p_20628_ ? -2.0D : -1.0D);
+      return 1.0D + Shapes.collide(Direction.Axis.Y, p_20629_, iterable, java.math.BigDecimal.valueOf(p_20628_ ? -2.0D : -1.0D)).doubleValue();
    }
 
    public static void m_20620_(Level p_20621_, @Nullable Player p_20622_, @Nullable Entity p_20623_, @Nullable CompoundTag p_20624_) {
