--- a/net/minecraft/world/entity/GlowSquid.java
+++ b/net/minecraft/world/entity/GlowSquid.java
@@ -65,7 +_,7 @@
          this.m_147119_(i - 1);
       }
 
-      this.m_9236_().m_7106_(ParticleTypes.f_175827_, this.m_20208_(0.6D), this.m_20187_(), this.m_20262_(0.6D), 0.0D, 0.0D, 0.0D);
+      this.m_9236_().addParticle(ParticleTypes.f_175827_, this.getRandomX(0.6D), this.getRandomY(), this.getRandomZ(0.6D), 0.0D, 0.0D, 0.0D);
    }
 
    public boolean m_6469_(DamageSource p_147114_, float p_147115_) {
@@ -86,6 +_,6 @@
    }
 
    public static boolean m_217017_(EntityType<? extends LivingEntity> p_217018_, ServerLevelAccessor p_217019_, MobSpawnType p_217020_, BlockPos p_217021_, RandomSource p_217022_) {
-      return p_217021_.m_123342_() <= p_217019_.m_5736_() - 33 && p_217019_.m_45524_(p_217021_, 0) == 0 && p_217019_.m_8055_(p_217021_).m_60713_(Blocks.f_49990_);
+      return p_217021_.getY() <= p_217019_.m_5736_() - 33 && p_217019_.m_45524_(p_217021_, 0) == 0 && p_217019_.m_8055_(p_217021_).m_60713_(Blocks.f_49990_);
    }
 }
