--- a/net/minecraft/world/entity/GlowSquid.java
+++ b/net/minecraft/world/entity/GlowSquid.java
@@ -86,6 +_,6 @@
    }
 
    public static boolean m_217017_(EntityType<? extends LivingEntity> p_217018_, ServerLevelAccessor p_217019_, MobSpawnType p_217020_, BlockPos p_217021_, RandomSource p_217022_) {
-      return p_217021_.m_123342_() <= p_217019_.m_5736_() - 33 && p_217019_.m_45524_(p_217021_, 0) == 0 && p_217019_.m_8055_(p_217021_).m_60713_(Blocks.f_49990_);
+      return p_217021_.getY() <= p_217019_.m_5736_() - 33 && p_217019_.m_45524_(p_217021_, 0) == 0 && p_217019_.m_8055_(p_217021_).m_60713_(Blocks.f_49990_);
    }
 }
