--- a/net/minecraft/world/entity/LightningBolt.java
+++ b/net/minecraft/world/entity/LightningBolt.java
@@ -148,7 +_,7 @@
          }
 
          for(int i = 0; i < p_20871_; ++i) {
-            BlockPos blockpos1 = blockpos.m_7918_(this.f_19796_.m_188503_(3) - 1, this.f_19796_.m_188503_(3) - 1, this.f_19796_.m_188503_(3) - 1);
+            BlockPos blockpos1 = blockpos.offset(this.f_19796_.m_188503_(3) - 1, this.f_19796_.m_188503_(3) - 1, this.f_19796_.m_188503_(3) - 1);
             blockstate = BaseFireBlock.m_49245_(this.m_9236_(), blockpos1);
             if (this.m_9236_().m_8055_(blockpos1).m_60795_() && blockstate.m_60710_(this.m_9236_(), blockpos1)) {
                this.m_9236_().m_46597_(blockpos1, blockstate);
@@ -164,7 +_,7 @@
       BlockPos blockpos;
       BlockState blockstate1;
       if (blockstate.m_60713_(Blocks.f_152587_)) {
-         blockpos = p_147152_.m_121945_(blockstate.m_61143_(LightningRodBlock.f_52588_).m_122424_());
+         blockpos = p_147152_.relative(blockstate.m_61143_(LightningRodBlock.f_52588_).m_122424_());
          blockstate1 = p_147151_.m_8055_(blockpos);
       } else {
          blockpos = p_147152_;
@@ -185,7 +_,7 @@
    }
 
    private static void m_147145_(Level p_147146_, BlockPos p_147147_, BlockPos.MutableBlockPos p_147148_, int p_147149_) {
-      p_147148_.m_122190_(p_147147_);
+      p_147148_.set(p_147147_);
 
       for(int i = 0; i < p_147149_; ++i) {
          Optional<BlockPos> optional = m_147153_(p_147146_, p_147148_);
@@ -193,7 +_,7 @@
             break;
          }
 
-         p_147148_.m_122190_(optional.get());
+         p_147148_.set(optional.get());
       }
 
    }
