--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -197,9 +_,9 @@
    public float f_20901_;
    public float f_20902_;
    protected int f_20903_;
-   protected double f_20904_;
-   protected double f_20905_;
-   protected double f_20906_;
+   protected me.alphamode.mclong.math.BigDecimal f_20904_;
+   protected me.alphamode.mclong.math.BigDecimal f_20905_;
+   protected me.alphamode.mclong.math.BigDecimal f_20906_;
    protected double f_20907_;
    protected double f_20908_;
    protected double f_20933_;
@@ -287,22 +_,22 @@
       }
 
       if (!this.m_9236_().f_46443_ && this.f_19789_ > 3.0F && p_20991_ && !p_20992_.m_60795_()) {
-         double d0 = this.m_20185_();
-         double d1 = this.m_20186_();
-         double d2 = this.m_20189_();
+         me.alphamode.mclong.math.BigDecimal d0 = this.getBigX();
+         me.alphamode.mclong.math.BigDecimal d1 = this.getBigY();
+         me.alphamode.mclong.math.BigDecimal d2 = this.getBigZ();
          BlockPos blockpos = this.m_20183_();
-         if (p_20993_.m_123341_() != blockpos.m_123341_() || p_20993_.m_123343_() != blockpos.m_123343_()) {
-            double d3 = d0 - (double)p_20993_.m_123341_() - 0.5D;
-            double d5 = d2 - (double)p_20993_.m_123343_() - 0.5D;
-            double d6 = Math.max(Math.abs(d3), Math.abs(d5));
-            d0 = (double)p_20993_.m_123341_() + 0.5D + d3 / d6 * 0.5D;
-            d2 = (double)p_20993_.m_123343_() + 0.5D + d5 / d6 * 0.5D;
+         if (p_20993_.getX() != blockpos.getX() || p_20993_.getZ() != blockpos.getZ()) {
+            me.alphamode.mclong.math.BigDecimal d3 = d0.subtract(p_20993_.getBigDecX()).subtract(me.alphamode.mclong.core.BigConstants.AABB);
+            me.alphamode.mclong.math.BigDecimal d5 = d2.subtract(p_20993_.getBigDecZ()).subtract(me.alphamode.mclong.core.BigConstants.AABB);
+            me.alphamode.mclong.math.BigDecimal d6 = d3.abs().max(d5.abs());
+            d0 = p_20993_.getBigDecX().add(me.alphamode.mclong.core.BigConstants.AABB).add(d3.divide(d6, java.math.RoundingMode.HALF_UP).multiply(me.alphamode.mclong.core.BigConstants.AABB));
+            d2 = p_20993_.getBigDecZ().add(me.alphamode.mclong.core.BigConstants.AABB).add(d5.divide(d6, java.math.RoundingMode.HALF_UP).multiply(me.alphamode.mclong.core.BigConstants.AABB));
          }
 
          float f = (float)Mth.m_14167_(this.f_19789_ - 3.0F);
          double d4 = Math.min((double)(0.2F + f / 15.0F), 2.5D);
          int i = (int)(150.0D * d4);
-         ((ServerLevel)this.m_9236_()).m_8767_(new BlockParticleOption(ParticleTypes.f_123794_, p_20992_), d0, d1, d2, i, 0.0D, 0.0D, 0.0D, (double)0.15F);
+         ((ServerLevel)this.m_9236_()).sendParticles(new BlockParticleOption(ParticleTypes.f_123794_, p_20992_), d0, d1, d2, i, 0.0D, 0.0D, 0.0D, (double)0.15F);
       }
 
       super.m_7840_(p_20990_, p_20991_, p_20992_, p_20993_);
@@ -346,13 +_,13 @@
                if (d0 < 0.0D) {
                   double d1 = this.m_9236_().m_6857_().m_61965_();
                   if (d1 > 0.0D) {
-                     this.m_6469_(this.m_269291_().m_287287_(), (float)Math.max(1, Mth.m_14107_(-d0 * d1)));
+                     this.m_6469_(this.m_269291_().m_269318_(), (float)Math.max(1, Mth.m_14134_(-d0 * d1)));
                   }
                }
             }
          }
 
-         if (this.m_204029_(FluidTags.f_13131_) && !this.m_9236_().m_8055_(BlockPos.m_274561_(this.m_20185_(), this.m_20188_(), this.m_20189_())).m_60713_(Blocks.f_50628_)) {
+         if (this.m_204029_(FluidTags.f_13131_) && !this.m_9236_().m_8055_(BlockPos.containing(this.getBigX(), this.getEyeBigY(), this.getBigZ())).m_60713_(Blocks.f_50628_)) {
             boolean flag1 = !this.m_6040_() && !MobEffectUtil.m_19588_(this) && (!flag || !((Player)this).m_150110_().f_35934_);
             if (flag1) {
                this.m_20301_(this.m_7302_(this.m_20146_()));
@@ -364,7 +_,7 @@
                      double d2 = this.f_19796_.m_188500_() - this.f_19796_.m_188500_();
                      double d3 = this.f_19796_.m_188500_() - this.f_19796_.m_188500_();
                      double d4 = this.f_19796_.m_188500_() - this.f_19796_.m_188500_();
-                     this.m_9236_().m_7106_(ParticleTypes.f_123795_, this.m_20185_() + d2, this.m_20186_() + d3, this.m_20189_() + d4, vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+                     this.m_9236_().addParticle(ParticleTypes.f_123795_, this.getBigX().add(d2), this.getBigY().add(d3), this.getBigZ().add(d4), vec3.m_7096_(), vec3.m_7098_(), vec3.m_7094_());
                   }
 
                   this.m_6469_(this.m_269291_().m_269063_(), 2.0F);
@@ -431,12 +_,12 @@
    }
 
    public boolean m_6039_() {
-      return this.f_19797_ % 5 == 0 && this.m_20184_().f_82479_ != 0.0D && this.m_20184_().f_82481_ != 0.0D && !this.m_5833_() && EnchantmentHelper.m_44942_(this) && this.m_6046_();
+      return this.f_19797_ % 5 == 0 && this.m_20184_().m_7096_() != 0.0D && this.m_20184_().m_7094_() != 0.0D && !this.m_5833_() && EnchantmentHelper.m_44942_(this) && this.m_6046_();
    }
 
    protected void m_21184_() {
       Vec3 vec3 = this.m_20184_();
-      this.m_9236_().m_7106_(ParticleTypes.f_123746_, this.m_20185_() + (this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_(), this.m_20186_() + 0.1D, this.m_20189_() + (this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_(), vec3.f_82479_ * -0.2D, 0.1D, vec3.f_82481_ * -0.2D);
+      this.m_9236_().addParticle(ParticleTypes.f_123746_, this.getBigX().add((this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_()), this.getBigY().add(me.alphamode.mclong.core.BigConstants.POINT_ONE), this.getBigZ().add((this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_()), vec3.m_7096_() * -0.2D, 0.1D, vec3.m_7094_() * -0.2D);
       float f = this.f_19796_.m_188501_() * 0.4F + this.f_19796_.m_188501_() > 0.9F ? 0.6F : 0.0F;
       this.m_5496_(SoundEvents.f_12404_, f, 0.6F + this.f_19796_.m_188501_() * 0.4F);
    }
@@ -641,7 +_,7 @@
          Equipable equipable = Equipable.m_269088_(p_238395_);
          if (equipable != null && !this.m_5833_() && equipable.m_40402_() == p_238393_) {
             if (!this.m_9236_().m_5776_() && !this.m_20067_()) {
-               this.m_9236_().m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), equipable.m_150681_(), this.m_5720_(), 1.0F, 1.0F);
+               this.m_9236_().playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), equipable.m_150681_(), this.m_5720_(), 1.0F, 1.0F);
             }
 
             if (this.m_213772_(p_238393_)) {
@@ -676,9 +_,9 @@
 
       p_21145_.m_128379_("FallFlying", this.m_21255_());
       this.m_21257_().ifPresent((p_21099_) -> {
-         p_21145_.m_128405_("SleepingX", p_21099_.m_123341_());
-         p_21145_.m_128405_("SleepingY", p_21099_.m_123342_());
-         p_21145_.m_128405_("SleepingZ", p_21099_.m_123343_());
+         p_21145_.m_128356_("SleepingX", p_21099_.getX());
+         p_21145_.m_128356_("SleepingY", p_21099_.getY());
+         p_21145_.m_128356_("SleepingZ", p_21099_.getZ());
       });
       DataResult<Tag> dataresult = this.f_20939_.m_21914_(NbtOps.f_128958_);
       dataresult.resultOrPartial(f_201943_::error).ifPresent((p_21102_) -> {
@@ -725,7 +_,7 @@
       }
 
       if (p_21096_.m_128425_("SleepingX", 99) && p_21096_.m_128425_("SleepingY", 99) && p_21096_.m_128425_("SleepingZ", 99)) {
-         BlockPos blockpos = new BlockPos(p_21096_.m_128451_("SleepingX"), p_21096_.m_128451_("SleepingY"), p_21096_.m_128451_("SleepingZ"));
+         BlockPos blockpos = new BlockPos(p_21096_.m_128454_("SleepingX"), p_21096_.m_128454_("SleepingY"), p_21096_.m_128454_("SleepingZ"));
          this.m_21250_(blockpos);
          this.f_19804_.m_135381_(f_19806_, Pose.SLEEPING);
          if (!this.f_19803_) {
@@ -787,7 +_,7 @@
             double d0 = (double)(i >> 16 & 255) / 255.0D;
             double d1 = (double)(i >> 8 & 255) / 255.0D;
             double d2 = (double)(i >> 0 & 255) / 255.0D;
-            this.m_9236_().m_7106_(flag1 ? ParticleTypes.f_123770_ : ParticleTypes.f_123811_, this.m_20208_(0.5D), this.m_20187_(), this.m_20262_(0.5D), d0, d1, d2);
+            this.m_9236_().addParticle(flag1 ? ParticleTypes.f_123770_ : ParticleTypes.f_123811_, this.getRandomX(0.5D), this.getRandomY(), this.getRandomZ(0.5D), d0, d1, d2);
          }
       }
 
@@ -1256,7 +_,7 @@
          if (vec32 != null) {
             Vec3 vec3 = this.m_20252_(1.0F);
             Vec3 vec31 = vec32.m_82505_(this.m_20182_()).m_82541_();
-            vec31 = new Vec3(vec31.f_82479_, 0.0D, vec31.f_82481_);
+            vec31 = new Vec3(vec31.getX(), me.alphamode.mclong.math.BigDecimal.ZERO, vec31.getZ());
             if (vec31.m_82526_(vec3) < 0.0D) {
                return true;
             }
@@ -1269,7 +_,7 @@
    private void m_21278_(ItemStack p_21279_) {
       if (!p_21279_.m_41619_()) {
          if (!this.m_20067_()) {
-            this.m_9236_().m_7785_(this.m_20185_(), this.m_20186_(), this.m_20189_(), SoundEvents.f_12018_, this.m_5720_(), 0.8F, 0.8F + this.m_9236_().f_46441_.m_188501_() * 0.4F, false);
+            this.m_9236_().playLocalSound(this.getBigX(), this.getBigY(), this.getBigZ(), SoundEvents.f_12018_, this.m_5720_(), 0.8F, 0.8F + this.m_9236_().f_46441_.m_188501_() * 0.4F, false);
          }
 
          this.m_21060_(p_21279_, 5);
@@ -1391,7 +_,7 @@
          this.f_19812_ = true;
          Vec3 vec3 = this.m_20184_();
          Vec3 vec31 = (new Vec3(p_147242_, 0.0D, p_147243_)).m_82541_().m_82490_(p_147241_);
-         this.m_20334_(vec3.f_82479_ / 2.0D - vec31.f_82479_, this.m_20096_() ? Math.min(0.4D, vec3.f_82480_ / 2.0D + p_147241_) : vec3.f_82480_, vec3.f_82481_ / 2.0D - vec31.f_82481_);
+         this.m_20334_(vec3.m_7096_() / 2.0D - vec31.m_7096_(), this.m_20096_() ? Math.min(0.4D, vec3.m_7098_() / 2.0D + p_147241_) : vec3.m_7098_(), vec3.m_7094_() / 2.0D - vec31.m_7094_());
       }
    }
 
@@ -1469,7 +_,7 @@
 
    private boolean m_21176_(BlockPos p_21177_, BlockState p_21178_) {
       if (p_21178_.m_61143_(TrapDoorBlock.f_57514_)) {
-         BlockState blockstate = this.m_9236_().m_8055_(p_21177_.m_7495_());
+         BlockState blockstate = this.m_9236_().m_8055_(p_21177_.below());
          if (blockstate.m_60713_(Blocks.f_50155_) && blockstate.m_61143_(LadderBlock.f_54337_) == p_21178_.m_61143_(TrapDoorBlock.f_54117_)) {
             return true;
          }
@@ -1507,9 +_,9 @@
 
    protected void m_21229_() {
       if (!this.m_20067_()) {
-         int i = Mth.m_14107_(this.m_20185_());
-         int j = Mth.m_14107_(this.m_20186_() - (double)0.2F);
-         int k = Mth.m_14107_(this.m_20189_());
+         long i = Mth.m_14134_(this.m_20185_());
+         long j = Mth.m_14134_(this.m_20186_() - (double)0.2F);
+         long k = Mth.m_14134_(this.m_20189_());
          BlockState blockstate = this.m_9236_().m_8055_(new BlockPos(i, j, k));
          if (!blockstate.m_60795_()) {
             SoundType soundtype = blockstate.m_60827_();
@@ -1706,14 +_,14 @@
             int i = 128;
 
             for(int j = 0; j < 128; ++j) {
-               double d0 = (double)j / 127.0D;
+               me.alphamode.mclong.math.BigDecimal d0 = me.alphamode.mclong.math.BigDecimal.valueOf(j / 127.0D);
                float f = (this.f_19796_.m_188501_() - 0.5F) * 0.2F;
                float f1 = (this.f_19796_.m_188501_() - 0.5F) * 0.2F;
                float f2 = (this.f_19796_.m_188501_() - 0.5F) * 0.2F;
-               double d1 = Mth.m_14139_(d0, this.f_19854_, this.m_20185_()) + (this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_() * 2.0D;
-               double d2 = Mth.m_14139_(d0, this.f_19855_, this.m_20186_()) + this.f_19796_.m_188500_() * (double)this.m_20206_();
-               double d3 = Mth.m_14139_(d0, this.f_19856_, this.m_20189_()) + (this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_() * 2.0D;
-               this.m_9236_().m_7106_(ParticleTypes.f_123760_, d1, d2, d3, (double)f, (double)f1, (double)f2);
+               me.alphamode.mclong.math.BigDecimal d1 = Mth.lerp(d0, this.f_19854_, this.getBigX()).add((this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_() * 2.0D);
+               me.alphamode.mclong.math.BigDecimal d2 = Mth.lerp(d0, this.f_19855_, this.getBigY()).add(this.f_19796_.m_188500_() * (double)this.m_20206_());
+               me.alphamode.mclong.math.BigDecimal d3 = Mth.lerp(d0, this.f_19856_, this.getBigZ()).add((this.f_19796_.m_188500_() - 0.5D) * (double)this.m_20205_() * 2.0D);
+               this.m_9236_().addParticle(ParticleTypes.f_123760_, d1, d2, d3, (double)f, (double)f1, (double)f2);
             }
             break;
          case 47:
@@ -1754,7 +_,7 @@
          double d0 = this.f_19796_.m_188583_() * 0.02D;
          double d1 = this.f_19796_.m_188583_() * 0.02D;
          double d2 = this.f_19796_.m_188583_() * 0.02D;
-         this.m_9236_().m_7106_(ParticleTypes.f_123759_, this.m_20208_(1.0D), this.m_20187_(), this.m_20262_(1.0D), d0, d1, d2);
+         this.m_9236_().addParticle(ParticleTypes.f_123759_, this.getRandomX(1.0D), this.getRandomY(), this.getRandomZ(1.0D), d0, d1, d2);
       }
 
    }
@@ -1931,7 +_,7 @@
          vec3 = new Vec3(this.m_20185_(), d0, this.m_20189_());
       }
 
-      this.m_142098_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+      this.dismountTo(vec3.getX(), vec3.getY(), vec3.getZ());
    }
 
    public boolean m_6052_() {
@@ -1948,7 +_,7 @@
 
    protected void m_6135_() {
       Vec3 vec3 = this.m_20184_();
-      this.m_20334_(vec3.f_82479_, (double)this.m_6118_(), vec3.f_82481_);
+      this.m_20334_(vec3.m_7096_(), (double)this.m_6118_(), vec3.m_7094_());
       if (this.m_20142_()) {
          float f = this.m_146908_() * ((float)Math.PI / 180F);
          this.m_20256_(this.m_20184_().m_82520_((double)(-Mth.m_14031_(f) * 0.2F), 0.0D, (double)(Mth.m_14089_(f) * 0.2F)));
@@ -1976,7 +_,7 @@
    public void m_7023_(Vec3 p_21280_) {
       if (this.m_6109_()) {
          double d0 = 0.08D;
-         boolean flag = this.m_20184_().f_82480_ <= 0.0D;
+         boolean flag = this.m_20184_().m_7098_() <= 0.0D;
          if (flag && this.m_21023_(MobEffects.f_19591_)) {
             d0 = 0.01D;
          }
@@ -2008,14 +_,14 @@
             this.m_6478_(MoverType.SELF, this.m_20184_());
             Vec3 vec36 = this.m_20184_();
             if (this.f_19862_ && this.m_6147_()) {
-               vec36 = new Vec3(vec36.f_82479_, 0.2D, vec36.f_82481_);
+               vec36 = new Vec3(vec36.m_7096_(), 0.2D, vec36.m_7094_());
             }
 
             this.m_20256_(vec36.m_82542_((double)f4, (double)0.8F, (double)f4));
             Vec3 vec32 = this.m_20994_(d0, flag, this.m_20184_());
             this.m_20256_(vec32);
-            if (this.f_19862_ && this.m_20229_(vec32.f_82479_, vec32.f_82480_ + (double)0.6F - this.m_20186_() + d9, vec32.f_82481_)) {
-               this.m_20334_(vec32.f_82479_, (double)0.3F, vec32.f_82481_);
+            if (this.f_19862_ && this.m_20229_(vec32.m_7096_(), vec32.m_7098_() + (double)0.6F - this.m_20186_() + d9, vec32.m_7094_())) {
+               this.m_20334_(vec32.m_7096_(), (double)0.3F, vec32.m_7094_());
             }
          } else if (this.m_20077_() && this.m_6129_() && !this.m_203441_(fluidstate)) {
             double d8 = this.m_20186_();
@@ -2034,32 +_,32 @@
             }
 
             Vec3 vec34 = this.m_20184_();
-            if (this.f_19862_ && this.m_20229_(vec34.f_82479_, vec34.f_82480_ + (double)0.6F - this.m_20186_() + d8, vec34.f_82481_)) {
-               this.m_20334_(vec34.f_82479_, (double)0.3F, vec34.f_82481_);
+            if (this.f_19862_ && this.m_20229_(vec34.m_7096_(), vec34.m_7098_() + (double)0.6F - this.m_20186_() + d8, vec34.m_7094_())) {
+               this.m_20334_(vec34.m_7096_(), (double)0.3F, vec34.m_7094_());
             }
          } else if (this.m_21255_()) {
             this.m_245125_();
             Vec3 vec3 = this.m_20184_();
             Vec3 vec31 = this.m_20154_();
             float f = this.m_146909_() * ((float)Math.PI / 180F);
-            double d1 = Math.sqrt(vec31.f_82479_ * vec31.f_82479_ + vec31.f_82481_ * vec31.f_82481_);
+            double d1 = Math.sqrt(vec31.m_7096_() * vec31.m_7096_() + vec31.m_7094_() * vec31.m_7094_());
             double d3 = vec3.m_165924_();
             double d4 = vec31.m_82553_();
             double d5 = Math.cos((double)f);
             d5 = d5 * d5 * Math.min(1.0D, d4 / 0.4D);
             vec3 = this.m_20184_().m_82520_(0.0D, d0 * (-1.0D + d5 * 0.75D), 0.0D);
-            if (vec3.f_82480_ < 0.0D && d1 > 0.0D) {
-               double d6 = vec3.f_82480_ * -0.1D * d5;
-               vec3 = vec3.m_82520_(vec31.f_82479_ * d6 / d1, d6, vec31.f_82481_ * d6 / d1);
+            if (vec3.m_7098_() < 0.0D && d1 > 0.0D) {
+               double d6 = vec3.m_7098_() * -0.1D * d5;
+               vec3 = vec3.m_82520_(vec31.m_7096_() * d6 / d1, d6, vec31.m_7094_() * d6 / d1);
             }
 
             if (f < 0.0F && d1 > 0.0D) {
                double d10 = d3 * (double)(-Mth.m_14031_(f)) * 0.04D;
-               vec3 = vec3.m_82520_(-vec31.f_82479_ * d10 / d1, d10 * 3.2D, -vec31.f_82481_ * d10 / d1);
+               vec3 = vec3.m_82520_(-vec31.m_7096_() * d10 / d1, d10 * 3.2D, -vec31.m_7094_() * d10 / d1);
             }
 
             if (d1 > 0.0D) {
-               vec3 = vec3.m_82520_((vec31.f_82479_ / d1 * d3 - vec3.f_82479_) * 0.1D, 0.0D, (vec31.f_82481_ / d1 * d3 - vec3.f_82481_) * 0.1D);
+               vec3 = vec3.m_82520_((vec31.m_7096_() / d1 * d3 - vec3.m_7096_()) * 0.1D, 0.0D, (vec31.m_7094_() / d1 * d3 - vec3.m_7094_()) * 0.1D);
             }
 
             this.m_20256_(vec3.m_82542_((double)0.99F, (double)0.98F, (double)0.99F));
@@ -2082,11 +_,11 @@
             float f2 = this.m_9236_().m_8055_(blockpos).m_60734_().m_49958_();
             float f3 = this.m_20096_() ? f2 * 0.91F : 0.91F;
             Vec3 vec35 = this.m_21074_(p_21280_, f2);
-            double d2 = vec35.f_82480_;
+            double d2 = vec35.m_7098_();
             if (this.m_21023_(MobEffects.f_19620_)) {
-               d2 += (0.05D * (double)(this.m_21124_(MobEffects.f_19620_).m_19564_() + 1) - vec35.f_82480_) * 0.2D;
+               d2 += (0.05D * (double)(this.m_21124_(MobEffects.f_19620_).m_19564_() + 1) - vec35.m_7098_()) * 0.2D;
             } else if (this.m_9236_().f_46443_ && !this.m_9236_().m_46805_(blockpos)) {
-               if (this.m_20186_() > (double)this.m_9236_().m_141937_()) {
+               if (this.m_20186_() > (double)this.m_9236_().getMinBuildHeight()) {
                   d2 = -0.1D;
                } else {
                   d2 = 0.0D;
@@ -2096,9 +_,9 @@
             }
 
             if (this.m_147223_()) {
-               this.m_20334_(vec35.f_82479_, d2, vec35.f_82481_);
+               this.m_20334_(vec35.m_7096_(), d2, vec35.m_7094_());
             } else {
-               this.m_20334_(vec35.f_82479_ * (double)f3, d2 * (double)0.98F, vec35.f_82481_ * (double)f3);
+               this.m_20334_(vec35.m_7096_() * (double)f3, d2 * (double)0.98F, vec35.m_7094_() * (double)f3);
             }
          }
       }
@@ -2132,7 +_,7 @@
    }
 
    public void m_267651_(boolean p_268129_) {
-      float f = (float)Mth.m_184648_(this.m_20185_() - this.f_19854_, p_268129_ ? this.m_20186_() - this.f_19855_ : 0.0D, this.m_20189_() - this.f_19856_);
+      float f = (float)Mth.m_184648_(this.getBigX().subtract(this.f_19854_).doubleValue(), p_268129_ ? this.getBigY().subtract(this.f_19855_).doubleValue() : 0.0D, this.getBigZ().subtract(this.f_19856_).doubleValue());
       this.m_267689_(f);
    }
 
@@ -2147,7 +_,7 @@
       this.m_6478_(MoverType.SELF, this.m_20184_());
       Vec3 vec3 = this.m_20184_();
       if ((this.f_19862_ || this.f_20899_) && (this.m_6147_() || this.m_146900_().m_60713_(Blocks.f_152499_) && PowderSnowBlock.m_154255_(this))) {
-         vec3 = new Vec3(vec3.f_82479_, 0.2D, vec3.f_82481_);
+         vec3 = new Vec3(vec3.m_7096_(), 0.2D, vec3.m_7094_());
       }
 
       return vec3;
@@ -2156,13 +_,13 @@
    public Vec3 m_20994_(double p_20995_, boolean p_20996_, Vec3 p_20997_) {
       if (!this.m_20068_() && !this.m_20142_()) {
          double d0;
-         if (p_20996_ && Math.abs(p_20997_.f_82480_ - 0.005D) >= 0.003D && Math.abs(p_20997_.f_82480_ - p_20995_ / 16.0D) < 0.003D) {
+         if (p_20996_ && Math.abs(p_20997_.m_7098_() - 0.005D) >= 0.003D && Math.abs(p_20997_.m_7098_() - p_20995_ / 16.0D) < 0.003D) {
             d0 = -0.003D;
          } else {
-            d0 = p_20997_.f_82480_ - p_20995_ / 16.0D;
+            d0 = p_20997_.m_7098_() - p_20995_ / 16.0D;
          }
 
-         return new Vec3(p_20997_.f_82479_, d0, p_20997_.f_82481_);
+         return new Vec3(p_20997_.m_7096_(), d0, p_20997_.m_7094_());
       } else {
          return p_20997_;
       }
@@ -2172,9 +_,9 @@
       if (this.m_6147_()) {
          this.m_183634_();
          float f = 0.15F;
-         double d0 = Mth.m_14008_(p_21298_.f_82479_, (double)-0.15F, (double)0.15F);
-         double d1 = Mth.m_14008_(p_21298_.f_82481_, (double)-0.15F, (double)0.15F);
-         double d2 = Math.max(p_21298_.f_82480_, (double)-0.15F);
+         double d0 = Mth.m_14008_(p_21298_.m_7096_(), (double)-0.15F, (double)0.15F);
+         double d1 = Mth.m_14008_(p_21298_.m_7094_(), (double)-0.15F, (double)0.15F);
+         double d2 = Math.max(p_21298_.m_7098_(), (double)-0.15F);
          if (d2 < 0.0D && !this.m_146900_().m_60713_(Blocks.f_50616_) && this.m_5791_() && this instanceof Player) {
             d2 = 0.0D;
          }
@@ -2249,8 +_,8 @@
          this.m_8107_();
       }
 
-      double d1 = this.m_20185_() - this.f_19854_;
-      double d0 = this.m_20189_() - this.f_19856_;
+      double d1 = this.getBigX().subtract(this.f_19854_).doubleValue();
+      double d0 = this.getBigZ().subtract(this.f_19856_).doubleValue();
       float f = (float)(d1 * d1 + d0 * d0);
       float f1 = this.f_20883_;
       float f2 = 0.0F;
@@ -2449,18 +_,18 @@
 
       if (this.m_6109_()) {
          this.f_20903_ = 0;
-         this.m_217006_(this.m_20185_(), this.m_20186_(), this.m_20189_());
+         this.syncPacketPositionCodec(this.getBigX(), this.getBigY(), this.getBigZ());
       }
 
       if (this.f_20903_ > 0) {
-         double d0 = this.m_20185_() + (this.f_20904_ - this.m_20185_()) / (double)this.f_20903_;
-         double d2 = this.m_20186_() + (this.f_20905_ - this.m_20186_()) / (double)this.f_20903_;
-         double d4 = this.m_20189_() + (this.f_20906_ - this.m_20189_()) / (double)this.f_20903_;
+         me.alphamode.mclong.math.BigDecimal d0 = this.getBigX().add((this.f_20904_.subtract(this.getBigX())).divide(this.f_20903_, java.math.RoundingMode.HALF_UP));
+         me.alphamode.mclong.math.BigDecimal d2 = this.getBigY().add((this.f_20905_.subtract(this.getBigY())).divide(this.f_20903_, java.math.RoundingMode.HALF_UP));
+         me.alphamode.mclong.math.BigDecimal d4 = this.getBigZ().add((this.f_20906_.subtract(this.getBigZ())).divide(this.f_20903_, java.math.RoundingMode.HALF_UP));
          double d6 = Mth.m_14175_(this.f_20907_ - (double)this.m_146908_());
          this.m_146922_(this.m_146908_() + (float)d6 / (float)this.f_20903_);
          this.m_146926_(this.m_146909_() + (float)(this.f_20908_ - (double)this.m_146909_()) / (float)this.f_20903_);
          --this.f_20903_;
-         this.m_6034_(d0, d2, d4);
+         this.setPos(d0, d2, d4);
          this.m_19915_(this.m_146908_(), this.m_146909_());
       } else if (!this.m_21515_()) {
          this.m_20256_(this.m_20184_().m_82490_(0.98D));
@@ -2472,18 +_,18 @@
       }
 
       Vec3 vec31 = this.m_20184_();
-      double d1 = vec31.f_82479_;
-      double d3 = vec31.f_82480_;
-      double d5 = vec31.f_82481_;
-      if (Math.abs(vec31.f_82479_) < 0.003D) {
+      double d1 = vec31.m_7096_();
+      double d3 = vec31.m_7098_();
+      double d5 = vec31.m_7094_();
+      if (Math.abs(vec31.m_7096_()) < 0.003D) {
          d1 = 0.0D;
       }
 
-      if (Math.abs(vec31.f_82480_) < 0.003D) {
+      if (Math.abs(vec31.m_7098_()) < 0.003D) {
          d3 = 0.0D;
       }
 
-      if (Math.abs(vec31.f_82481_) < 0.003D) {
+      if (Math.abs(vec31.m_7094_()) < 0.003D) {
          d5 = 0.0D;
       }
 
@@ -2699,7 +_,7 @@
       this.m_183634_();
    }
 
-   public void m_6453_(double p_20977_, double p_20978_, double p_20979_, float p_20980_, float p_20981_, int p_20982_, boolean p_20983_) {
+   public void lerpTo(me.alphamode.mclong.math.BigDecimal p_20977_, me.alphamode.mclong.math.BigDecimal p_20978_, me.alphamode.mclong.math.BigDecimal p_20979_, float p_20980_, float p_20981_, int p_20982_, boolean p_20983_) {
       this.f_20904_ = p_20977_;
       this.f_20905_ = p_20978_;
       this.f_20906_ = p_20979_;
@@ -2784,7 +_,7 @@
    }
 
    public static Vec3 m_21289_(Vec3 p_21290_) {
-      return new Vec3(p_21290_.f_82479_, p_21290_.f_82480_, 0.0D);
+      return new Vec3(p_21290_.getX(), p_21290_.getY(), me.alphamode.mclong.math.BigDecimal.ZERO);
    }
 
    public float m_6103_() {
@@ -2937,7 +_,7 @@
          vec31 = vec31.m_82496_(-this.m_146909_() * ((float)Math.PI / 180F));
          vec31 = vec31.m_82524_(-this.m_146908_() * ((float)Math.PI / 180F));
          vec31 = vec31.m_82520_(this.m_20185_(), this.m_20188_(), this.m_20189_());
-         this.m_9236_().m_7106_(new ItemParticleOption(ParticleTypes.f_123752_, p_21061_), vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, vec3.f_82479_, vec3.f_82480_ + 0.05D, vec3.f_82481_);
+         this.m_9236_().addParticle(new ItemParticleOption(ParticleTypes.f_123752_, p_21061_), vec31.getX(), vec31.getY(), vec31.getZ(), vec3.m_7096_(), vec3.m_7098_() + 0.05D, vec3.m_7094_());
       }
 
    }
@@ -3038,8 +_,8 @@
       if (level.m_46805_(blockpos)) {
          boolean flag1 = false;
 
-         while(!flag1 && blockpos.m_123342_() > level.m_141937_()) {
-            BlockPos blockpos1 = blockpos.m_7495_();
+         while(!flag1 && blockpos.getY() > level.getMinBuildHeight()) {
+            BlockPos blockpos1 = blockpos.below();
             BlockState blockstate = level.m_8055_(blockpos1);
             if (blockstate.m_280555_()) {
                flag1 = true;
@@ -3139,7 +_,7 @@
    }
 
    private void m_21080_(BlockPos p_21081_) {
-      this.m_6034_((double)p_21081_.m_123341_() + 0.5D, (double)p_21081_.m_123342_() + 0.6875D, (double)p_21081_.m_123343_() + 0.5D);
+      this.m_6034_((double)p_21081_.getX() + 0.5D, (double)p_21081_.getY() + 0.6875D, (double)p_21081_.getZ() + 0.5D);
    }
 
    private boolean m_21334_() {
@@ -3155,12 +_,12 @@
             Direction direction = blockstate.m_61143_(BedBlock.f_54117_);
             this.m_9236_().m_7731_(p_261435_, blockstate.m_61124_(BedBlock.f_49441_, Boolean.valueOf(false)), 3);
             Vec3 vec31 = BedBlock.m_260958_(this.m_6095_(), this.m_9236_(), p_261435_, direction, this.m_146908_()).orElseGet(() -> {
-               BlockPos blockpos = p_261435_.m_7494_();
-               return new Vec3((double)blockpos.m_123341_() + 0.5D, (double)blockpos.m_123342_() + 0.1D, (double)blockpos.m_123343_() + 0.5D);
+               BlockPos blockpos = p_261435_.above();
+               return new Vec3((double)blockpos.getX() + 0.5D, (double)blockpos.getY() + 0.1D, (double)blockpos.getZ() + 0.5D);
             });
-            Vec3 vec32 = Vec3.m_82539_(p_261435_).m_82546_(vec31).m_82541_();
-            float f = (float)Mth.m_14175_(Mth.m_14136_(vec32.f_82481_, vec32.f_82479_) * (double)(180F / (float)Math.PI) - 90.0D);
-            this.m_6034_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_);
+            Vec3 vec32 = Vec3.atBottomCenterOf(p_261435_).m_82546_(vec31).m_82541_();
+            float f = (float)Mth.m_14175_(Mth.m_14136_(vec32.m_7094_(), vec32.m_7096_()) * (double)(180F / (float)Math.PI) - 90.0D);
+            this.setPos(vec31.getX(), vec31.getY(), vec31.getZ());
             this.m_146922_(f);
             this.m_146926_(0.0F);
          }
@@ -3168,7 +_,7 @@
       });
       Vec3 vec3 = this.m_20182_();
       this.m_20124_(Pose.STANDING);
-      this.m_6034_(vec3.f_82479_, vec3.f_82480_, vec3.f_82481_);
+      this.setPos(vec3.getX(), vec3.getY(), vec3.getZ());
       this.m_21258_();
    }
 
@@ -3196,7 +_,7 @@
 
    public ItemStack m_5584_(Level p_21067_, ItemStack p_21068_) {
       if (p_21068_.m_41614_()) {
-         p_21067_.m_6263_((Player)null, this.m_20185_(), this.m_20186_(), this.m_20189_(), this.m_7866_(p_21068_), SoundSource.NEUTRAL, 1.0F, 1.0F + (p_21067_.f_46441_.m_188501_() - p_21067_.f_46441_.m_188501_()) * 0.4F);
+         p_21067_.playSound((Player)null, this.getBigX(), this.getBigY(), this.getBigZ(), this.m_7866_(p_21068_), SoundSource.NEUTRAL, 1.0F, 1.0F + (p_21067_.f_46441_.m_188501_() - p_21067_.f_46441_.m_188501_()) * 0.4F);
          this.m_21063_(p_21068_, p_21067_, this);
          if (!(this instanceof Player) || !((Player)this).m_150110_().f_35937_) {
             p_21068_.m_41774_(1);
@@ -3307,19 +_,19 @@
    }
 
    public void m_141965_(ClientboundAddEntityPacket p_217037_) {
-      double d0 = p_217037_.m_131500_();
-      double d1 = p_217037_.m_131501_();
-      double d2 = p_217037_.m_131502_();
+      me.alphamode.mclong.math.BigDecimal d0 = p_217037_.getX();
+      me.alphamode.mclong.math.BigDecimal d1 = p_217037_.getY();
+      me.alphamode.mclong.math.BigDecimal d2 = p_217037_.getZ();
       float f = p_217037_.m_237567_();
       float f1 = p_217037_.m_237566_();
-      this.m_217006_(d0, d1, d2);
+      this.syncPacketPositionCodec(d0, d1, d2);
       this.f_20883_ = p_217037_.m_237568_();
       this.f_20885_ = p_217037_.m_237568_();
       this.f_20884_ = this.f_20883_;
       this.f_20886_ = this.f_20885_;
       this.m_20234_(p_217037_.m_131496_());
       this.m_20084_(p_217037_.m_131499_());
-      this.m_19890_(d0, d1, d2, f, f1);
+      this.absMoveTo(d0, d1, d2, f, f1);
       this.m_20334_(p_217037_.m_131503_(), p_217037_.m_131504_(), p_217037_.m_131505_());
    }
 
