--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -310,7 +_,7 @@
             double d1 = this.f_19796_.m_188583_() * 0.02D;
             double d2 = this.f_19796_.m_188583_() * 0.02D;
             double d3 = 10.0D;
-            this.m_9236_().m_7106_(ParticleTypes.f_123759_, this.m_20165_(1.0D) - d0 * 10.0D, this.m_20187_() - d1 * 10.0D, this.m_20262_(1.0D) - d2 * 10.0D, d0, d1, d2);
+            this.m_9236_().addParticle(ParticleTypes.f_123759_, this.getBigX(1.0D).subtract(d0 * 10.0D), this.getRandomY().subtract(d1 * 10.0D), this.getRandomZ(1.0D).subtract(d2 * 10.0D), d0, d1, d2);
          }
       } else {
          this.m_9236_().m_7605_(this, (byte)20);
@@ -405,9 +_,9 @@
             compoundtag2.m_128362_("UUID", uuid);
          } else if (this.f_21357_ instanceof HangingEntity) {
             BlockPos blockpos = ((HangingEntity)this.f_21357_).m_31748_();
-            compoundtag2.m_128405_("X", blockpos.m_123341_());
-            compoundtag2.m_128405_("Y", blockpos.m_123342_());
-            compoundtag2.m_128405_("Z", blockpos.m_123343_());
+            compoundtag2.m_128359_("X", blockpos.getBigX().toString());
+            compoundtag2.m_128359_("Y", blockpos.getBigY().toString());
+            compoundtag2.m_128359_("Z", blockpos.getBigZ().toString());
          }
 
          p_21484_.m_128365_("Leash", compoundtag2);
@@ -780,7 +_,7 @@
       if (p_21392_ instanceof LivingEntity livingentity) {
          d1 = livingentity.m_20188_() - this.m_20188_();
       } else {
-         d1 = (p_21392_.m_20191_().f_82289_ + p_21392_.m_20191_().f_82292_) / 2.0D - this.m_20188_();
+         d1 = (p_21392_.m_20191_().f_82289_.doubleValue() + p_21392_.m_20191_().f_82292_.doubleValue()) / 2.0D - this.m_20188_();
       }
 
       double d3 = Math.sqrt(d0 * d0 + d2 * d2);
@@ -804,7 +_,7 @@
    }
 
    public static boolean m_217057_(EntityType<? extends Mob> p_217058_, LevelAccessor p_217059_, MobSpawnType p_217060_, BlockPos p_217061_, RandomSource p_217062_) {
-      BlockPos blockpos = p_217061_.m_7495_();
+      BlockPos blockpos = p_217061_.below();
       return p_217060_ == MobSpawnType.SPAWNER || p_217059_.m_8055_(blockpos).m_60643_(p_217059_, blockpos, p_217058_);
    }
 
@@ -1140,7 +_,7 @@
       if (this.f_21341_ == -1.0F) {
          return true;
       } else {
-         return this.f_21360_.m_123331_(p_21445_) < (double)(this.f_21341_ * this.f_21341_);
+         return this.f_21360_.distSqr(p_21445_) < (double)(this.f_21341_ * this.f_21341_);
       }
    }
 
@@ -1404,7 +_,7 @@
    protected boolean m_21527_() {
       if (this.m_9236_().m_46461_() && !this.m_9236_().f_46443_) {
          float f = this.m_213856_();
-         BlockPos blockpos = BlockPos.m_274561_(this.m_20185_(), this.m_20188_(), this.m_20189_());
+         BlockPos blockpos = BlockPos.containing(this.getBigX(), this.getEyeBigY(), this.getBigZ());
          boolean flag = this.m_20071_() || this.f_146808_ || this.f_146809_;
          if (f > 0.5F && this.f_19796_.m_188501_() * 30.0F < (f - 0.4F) * 2.0F && !flag && this.m_9236_().m_45527_(blockpos)) {
             return true;
