--- a/net/minecraft/world/entity/PathfinderMob.java
+++ b/net/minecraft/world/entity/PathfinderMob.java
@@ -58,7 +_,7 @@
             this.f_21345_.m_25374_(Goal.Flag.MOVE);
             float f1 = 2.0F;
             Vec3 vec3 = (new Vec3(entity.m_20185_() - this.m_20185_(), entity.m_20186_() - this.m_20186_(), entity.m_20189_() - this.m_20189_())).m_82541_().m_82490_((double)Math.max(f - 2.0F, 0.0F));
-            this.m_21573_().m_26519_(this.m_20185_() + vec3.f_82479_, this.m_20186_() + vec3.f_82480_, this.m_20189_() + vec3.f_82481_, this.m_5823_());
+            this.m_21573_().m_26519_(this.m_20185_() + vec3.m_7096_(), this.m_20186_() + vec3.m_7098_(), this.m_20189_() + vec3.m_7094_(), this.m_5823_());
          }
       }
 
