--- a/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
@@ -54,11 +_,11 @@
 
    protected void m_26073_() {
       if (this.f_26048_ != Player.class && this.f_26048_ != ServerPlayer.class) {
-         this.f_26050_ = this.f_26135_.m_9236_().m_45982_(this.f_26135_.m_9236_().m_6443_(this.f_26048_, this.m_7255_(this.m_7623_()), (p_148152_) -> {
+         this.f_26050_ = this.f_26135_.m_9236_().getNearestEntity(this.f_26135_.m_9236_().m_6443_(this.f_26048_, this.m_7255_(this.m_7623_()), (p_148152_) -> {
             return true;
-         }), this.f_26051_, this.f_26135_, this.f_26135_.m_20185_(), this.f_26135_.m_20188_(), this.f_26135_.m_20189_());
+         }), this.f_26051_, this.f_26135_, this.f_26135_.getBigX(), this.f_26135_.getEyeBigY(), this.f_26135_.getBigZ());
       } else {
-         this.f_26050_ = this.f_26135_.m_9236_().m_45949_(this.f_26051_, this.f_26135_, this.f_26135_.m_20185_(), this.f_26135_.m_20188_(), this.f_26135_.m_20189_());
+         this.f_26050_ = this.f_26135_.m_9236_().getNearestPlayer(this.f_26051_, this.f_26135_, this.f_26135_.getBigX(), this.f_26135_.getEyeBigY(), this.f_26135_.getBigZ());
       }
 
    }
