--- a/net/minecraft/world/entity/ai/goal/target/TargetGoal.java
+++ b/net/minecraft/world/entity/ai/goal/target/TargetGoal.java
@@ -120,8 +_,8 @@
          if (node == null) {
             return false;
          } else {
-            int i = node.f_77271_ - p_26149_.m_146903_();
-            int j = node.f_77273_ - p_26149_.m_146907_();
+            long i = node.f_77271_ - p_26149_.getBlockX();
+            long j = node.f_77273_ - p_26149_.getBlockZ();
             return (double)(i * i + j * j) <= 2.25D;
          }
       }
