--- a/net/minecraft/world/level/levelgen/synth/BlendedNoise.java
+++ b/net/minecraft/world/level/levelgen/synth/BlendedNoise.java
@@ -68,9 +_,9 @@
    }
 
    public double m_207386_(DensityFunction.FunctionContext p_210621_) {
-      double d0 = (double)p_210621_.m_207115_() * this.f_230456_;
-      double d1 = (double)p_210621_.m_207114_() * this.f_230457_;
-      double d2 = (double)p_210621_.m_207113_() * this.f_230456_;
+      double d0 = (double)p_210621_.blockX() * this.f_230456_;
+      double d1 = (double)p_210621_.blockY() * this.f_230457_;
+      double d2 = (double)p_210621_.blockZ() * this.f_230456_;
       double d3 = d0 / this.f_230458_;
       double d4 = d1 / this.f_230459_;
       double d5 = d2 / this.f_230458_;
