--- a/net/minecraft/world/level/levelgen/synth/NormalNoise.java
+++ b/net/minecraft/world/level/levelgen/synth/NormalNoise.java
@@ -73,11 +_,11 @@
       return 0.1D * (1.0D + 1.0D / (double)(p_75385_ + 1));
    }
 
-   public double m_75380_(double p_75381_, double p_75382_, double p_75383_) {
-      double d0 = p_75381_ * 1.0181268882175227D;
-      double d1 = p_75382_ * 1.0181268882175227D;
-      double d2 = p_75383_ * 1.0181268882175227D;
-      return (this.f_75374_.m_75408_(p_75381_, p_75382_, p_75383_) + this.f_75375_.m_75408_(d0, d1, d2)) * this.f_75373_;
+   public double getValue(me.alphamode.mclong.math.BigDecimal p_75381_, me.alphamode.mclong.math.BigDecimal p_75382_, me.alphamode.mclong.math.BigDecimal p_75383_) {
+      me.alphamode.mclong.math.BigDecimal d0 = p_75381_.multiply(1.0181268882175227D);
+      me.alphamode.mclong.math.BigDecimal d1 = p_75382_.multiply(1.0181268882175227D);
+      me.alphamode.mclong.math.BigDecimal d2 = p_75383_.multiply(1.0181268882175227D);
+      return (this.f_75374_.getValue(p_75381_, p_75382_, p_75383_) + this.f_75375_.getValue(d0, d1, d2)) * this.f_75373_;
    }
 
    public NormalNoise.NoiseParameters m_192842_() {
