--- a/net/minecraft/world/level/levelgen/synth/PerlinNoise.java
+++ b/net/minecraft/world/level/levelgen/synth/PerlinNoise.java
@@ -141,13 +_,13 @@
       p_230519_.m_190110_(262);
    }
 
-   public double m_75408_(double p_75409_, double p_75410_, double p_75411_) {
-      return this.m_75417_(p_75409_, p_75410_, p_75411_, 0.0D, 0.0D, false);
+   public double getValue(me.alphamode.mclong.math.BigDecimal p_75409_, me.alphamode.mclong.math.BigDecimal p_75410_, me.alphamode.mclong.math.BigDecimal p_75411_) {
+      return this.getValue(p_75409_, p_75410_, p_75411_, 0.0D, 0.0D, false);
    }
 
    /** @deprecated */
    @Deprecated
-   public double m_75417_(double p_75418_, double p_75419_, double p_75420_, double p_75421_, double p_75422_, boolean p_75423_) {
+   public double getValue(me.alphamode.mclong.math.BigDecimal p_75418_, me.alphamode.mclong.math.BigDecimal p_75419_, me.alphamode.mclong.math.BigDecimal p_75420_, double p_75421_, double p_75422_, boolean p_75423_) {
       double d0 = 0.0D;
       double d1 = this.f_75393_;
       double d2 = this.f_75392_;
@@ -155,7 +_,7 @@
       for(int i = 0; i < this.f_75390_.length; ++i) {
          ImprovedNoise improvednoise = this.f_75390_[i];
          if (improvednoise != null) {
-            double d3 = improvednoise.m_75327_(m_75406_(p_75418_ * d1), p_75423_ ? -improvednoise.f_75322_ : m_75406_(p_75419_ * d1), m_75406_(p_75420_ * d1), p_75421_ * d1, p_75422_ * d1);
+            double d3 = improvednoise.noise(wrap(p_75418_.multiply(d1)), p_75423_ ? me.alphamode.mclong.math.BigDecimal.valueOf(-improvednoise.f_75322_) : wrap(p_75419_.multiply(d1)), wrap(p_75420_.multiply(d1)), p_75421_ * d1, me.alphamode.mclong.math.BigDecimal.valueOf(p_75422_ * d1));
             d0 += this.f_75391_.getDouble(i) * d3 * d2;
          }
 
@@ -191,8 +_,8 @@
       return this.f_75390_[this.f_75390_.length - 1 - p_75425_];
    }
 
-   public static double m_75406_(double p_75407_) {
-      return p_75407_ - (double)Mth.m_14134_(p_75407_ / 3.3554432E7D + 0.5D) * 3.3554432E7D;
+   public static me.alphamode.mclong.math.BigDecimal wrap(me.alphamode.mclong.math.BigDecimal p_75407_) {
+      return p_75407_.subtract(Mth.lfloor(p_75407_.divide(3.3554432E7D).add(me.alphamode.mclong.core.BigConstants.AABB)).toBigDecimal().multiply(3.3554432E7D));
    }
 
    protected int m_192872_() {
