--- a/net/minecraft/client/multiplayer/chat/report/ReportEnvironment.java
+++ b/net/minecraft/client/multiplayer/chat/report/ReportEnvironment.java
@@ -52,7 +_,7 @@
 
    private static String m_239334_() {
       StringBuilder stringbuilder = new StringBuilder();
-      stringbuilder.append("1.20.1");
+      stringbuilder.append(net.minecraft.SharedConstants.f_142952_);
       if (Minecraft.m_193589_().m_184597_()) {
          stringbuilder.append(" (modded)");
       }
