--- a/net/minecraft/world/entity/animal/axolotl/Axolotl.java
+++ b/net/minecraft/world/entity/animal/axolotl/Axolotl.java
@@ -447,7 +_,7 @@
    }
 
    public static boolean m_218437_(EntityType<? extends LivingEntity> p_218438_, ServerLevelAccessor p_218439_, MobSpawnType p_218440_, BlockPos p_218441_, RandomSource p_218442_) {
-      return p_218439_.m_8055_(p_218441_.m_7495_()).m_204336_(BlockTags.f_184229_);
+      return p_218439_.m_8055_(p_218441_.below()).m_204336_(BlockTags.f_184229_);
    }
 
    public static class AxolotlGroupData extends AgeableMob.AgeableMobGroupData {
