--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingAttackingPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingAttackingPhase.java
@@ -12,7 +_,7 @@
    }
 
    public void m_6991_() {
-      this.f_31176_.m_9236_().m_7785_(this.f_31176_.m_20185_(), this.f_31176_.m_20186_(), this.f_31176_.m_20189_(), SoundEvents.f_11894_, this.f_31176_.m_5720_(), 2.5F, 0.8F + this.f_31176_.m_217043_().m_188501_() * 0.3F, false);
+      this.f_31176_.m_9236_().playLocalSound(this.f_31176_.getBigX(), this.f_31176_.getBigY(), this.f_31176_.getBigZ(), SoundEvents.f_11894_, this.f_31176_.m_5720_(), 2.5F, 0.8F + this.f_31176_.m_217043_().m_188501_() * 0.3F, false);
    }
 
    public void m_6989_() {
