--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingPhase.java
@@ -29,7 +_,7 @@
          double d4 = d1 + randomsource.m_188583_() / 2.0D;
          double d5 = d2 + randomsource.m_188583_() / 2.0D;
          Vec3 vec31 = this.f_31176_.m_20184_();
-         this.f_31176_.m_9236_().m_7106_(ParticleTypes.f_123799_, d3, d4, d5, -vec3.f_82479_ * (double)0.08F + vec31.f_82479_, -vec3.f_82480_ * (double)0.3F + vec31.f_82480_, -vec3.f_82481_ * (double)0.08F + vec31.f_82481_);
+         this.f_31176_.m_9236_().m_7106_(ParticleTypes.f_123799_, d3, d4, d5, -vec3.m_7096_() * (double)0.08F + vec31.m_7096_(), -vec3.m_7098_() * (double)0.3F + vec31.m_7098_(), -vec3.m_7094_() * (double)0.08F + vec31.m_7094_());
          vec3.m_82524_(0.19634955F);
       }
 
@@ -37,7 +_,7 @@
 
    public void m_6989_() {
       if (this.f_31303_ == null) {
-         this.f_31303_ = Vec3.m_82539_(this.f_31176_.m_9236_().m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.m_287210_(this.f_31176_.m_287165_())));
+         this.f_31303_ = Vec3.atBottomCenterOf(this.f_31176_.m_9236_().m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.m_287210_(this.f_31176_.m_287165_())));
       }
 
       if (this.f_31303_.m_82531_(this.f_31176_.m_20185_(), this.f_31176_.m_20186_(), this.f_31176_.m_20189_()) < 1.0D) {
