--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
@@ -61,18 +_,18 @@
                if (this.f_31350_ >= 5 && f >= 0.0F && f < 10.0F) {
                   double d14 = 1.0D;
                   Vec3 vec32 = this.f_31176_.m_20252_(1.0F);
-                  double d6 = this.f_31176_.f_31080_.m_20185_() - vec32.f_82479_ * 1.0D;
-                  double d7 = this.f_31176_.f_31080_.m_20227_(0.5D) + 0.5D;
-                  double d8 = this.f_31176_.f_31080_.m_20189_() - vec32.f_82481_ * 1.0D;
-                  double d9 = this.f_31353_.m_20185_() - d6;
-                  double d10 = this.f_31353_.m_20227_(0.5D) - d7;
-                  double d11 = this.f_31353_.m_20189_() - d8;
+                  me.alphamode.mclong.math.BigDecimal d6 = this.f_31176_.f_31080_.getBigX().subtract(vec32.getX().multiply(me.alphamode.mclong.math.BigDecimal.ONE));
+                  me.alphamode.mclong.math.BigDecimal d7 = this.f_31176_.f_31080_.getBigY(0.5D).add(me.alphamode.mclong.core.BigConstants.AABB);
+                  me.alphamode.mclong.math.BigDecimal d8 = this.f_31176_.f_31080_.getBigZ().subtract(vec32.getZ().multiply(me.alphamode.mclong.math.BigDecimal.ONE));
+                  double d9 = this.f_31353_.getBigX().subtract(d6).doubleValue();
+                  double d10 = this.f_31353_.getBigY(0.5D).subtract(d7).doubleValue();
+                  double d11 = this.f_31353_.getBigZ().subtract(d8).doubleValue();
                   if (!this.f_31176_.m_20067_()) {
                      this.f_31176_.m_9236_().m_5898_((Player)null, 1017, this.f_31176_.m_20183_(), 0);
                   }
 
                   DragonFireball dragonfireball = new DragonFireball(this.f_31176_.m_9236_(), this.f_31176_, d9, d10, d11);
-                  dragonfireball.m_7678_(d6, d7, d8, 0.0F, 0.0F);
+                  dragonfireball.moveTo(d6, d7, d8, 0.0F, 0.0F);
                   this.f_31176_.m_9236_().m_7967_(dragonfireball);
                   this.f_31350_ = 0;
                   if (this.f_31351_ != null) {
@@ -130,15 +_,15 @@
 
    private void m_31365_() {
       if (this.f_31351_ != null && !this.f_31351_.m_77392_()) {
-         Vec3i vec3i = this.f_31351_.m_77400_();
+         me.alphamode.mclong.core.Vec3l vec3i = this.f_31351_.m_77400_();
          this.f_31351_.m_77374_();
-         double d0 = (double)vec3i.m_123341_();
-         double d2 = (double)vec3i.m_123343_();
+         double d0 = (double)vec3i.getX();
+         double d2 = (double)vec3i.getZ();
 
          double d1;
          do {
-            d1 = (double)((float)vec3i.m_123342_() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
-         } while(d1 < (double)vec3i.m_123342_());
+            d1 = (double)((float)vec3i.getY() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
+         } while(d1 < (double)vec3i.getY());
 
          this.f_31352_ = new Vec3(d0, d1, d2);
       }
@@ -156,8 +_,8 @@
       this.f_31353_ = p_31359_;
       int i = this.f_31176_.m_31155_();
       int j = this.f_31176_.m_31170_(this.f_31353_.m_20185_(), this.f_31353_.m_20186_(), this.f_31353_.m_20189_());
-      int k = this.f_31353_.m_146903_();
-      int l = this.f_31353_.m_146907_();
+      long k = this.f_31353_.getBlockX();
+      long l = this.f_31353_.getBlockZ();
       double d0 = (double)k - this.f_31176_.m_20185_();
       double d1 = (double)l - this.f_31176_.m_20189_();
       double d2 = Math.sqrt(d0 * d0 + d1 * d1);
