--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
@@ -130,15 +_,15 @@
 
    private void m_31365_() {
       if (this.f_31351_ != null && !this.f_31351_.m_77392_()) {
-         Vec3i vec3i = this.f_31351_.m_77400_();
+         me.alphamode.mclong.core.Vec3l vec3i = this.f_31351_.m_77400_();
          this.f_31351_.m_77374_();
-         double d0 = (double)vec3i.m_123341_();
-         double d2 = (double)vec3i.m_123343_();
+         double d0 = (double)vec3i.getX();
+         double d2 = (double)vec3i.getZ();
 
          double d1;
          do {
-            d1 = (double)((float)vec3i.m_123342_() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
-         } while(d1 < (double)vec3i.m_123342_());
+            d1 = (double)((float)vec3i.getY() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
+         } while(d1 < (double)vec3i.getY());
 
          this.f_31352_ = new Vec3(d0, d1, d2);
       }
@@ -156,8 +_,8 @@
       this.f_31353_ = p_31359_;
       int i = this.f_31176_.m_31155_();
       int j = this.f_31176_.m_31170_(this.f_31353_.m_20185_(), this.f_31353_.m_20186_(), this.f_31353_.m_20189_());
-      int k = this.f_31353_.m_146903_();
-      int l = this.f_31353_.m_146907_();
+      long k = this.f_31353_.getBlockX();
+      long l = this.f_31353_.getBlockZ();
       double d0 = (double)k - this.f_31176_.m_20185_();
       double d1 = (double)l - this.f_31176_.m_20189_();
       double d2 = Math.sqrt(d0 * d0 + d1 * d1);
