--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonDeathPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonDeathPhase.java
@@ -31,7 +_,7 @@
       ++this.f_31215_;
       if (this.f_31214_ == null) {
          BlockPos blockpos = this.f_31176_.f_19853_.m_5452_(Heightmap.Types.MOTION_BLOCKING, EndPodiumFeature.f_65714_);
-         this.f_31214_ = Vec3.m_82539_(blockpos);
+         this.f_31214_ = Vec3.atBottomCenterOf(blockpos);
       }
 
       double d0 = this.f_31214_.m_82531_(this.f_31176_.m_20185_(), this.f_31176_.m_20186_(), this.f_31176_.m_20189_());
