--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonTakeoffPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonTakeoffPhase.java
@@ -24,7 +_,7 @@
    public void m_6989_() {
       if (!this.f_31366_ && this.f_31367_ != null) {
          BlockPos blockpos = this.f_31176_.m_9236_().m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.m_287210_(this.f_31176_.m_287165_()));
-         if (!blockpos.m_203195_(this.f_31176_.m_20182_(), 10.0D)) {
+         if (!blockpos.closerToCenterThan(this.f_31176_.m_20182_(), 10.0D)) {
             this.f_31176_.m_31157_().m_31416_(EnderDragonPhase.f_31377_);
          }
       } else {
@@ -63,15 +_,15 @@
       if (this.f_31367_ != null) {
          this.f_31367_.m_77374_();
          if (!this.f_31367_.m_77392_()) {
-            Vec3i vec3i = this.f_31367_.m_77400_();
+            me.alphamode.mclong.core.Vec3l vec3i = this.f_31367_.m_77400_();
             this.f_31367_.m_77374_();
 
             double d0;
             do {
-               d0 = (double)((float)vec3i.m_123342_() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
-            } while(d0 < (double)vec3i.m_123342_());
+               d0 = (double)((float)vec3i.getY() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
+            } while(d0 < (double)vec3i.getY());
 
-            this.f_31368_ = new Vec3((double)vec3i.m_123341_(), d0, (double)vec3i.m_123343_());
+            this.f_31368_ = new Vec3((double)vec3i.getX(), d0, (double)vec3i.getZ());
          }
       }
 
