--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonDeathPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonDeathPhase.java
@@ -22,7 +_,7 @@
          float f = (this.f_31176_.m_217043_().m_188501_() - 0.5F) * 8.0F;
          float f1 = (this.f_31176_.m_217043_().m_188501_() - 0.5F) * 4.0F;
          float f2 = (this.f_31176_.m_217043_().m_188501_() - 0.5F) * 8.0F;
-         this.f_31176_.m_9236_().m_7106_(ParticleTypes.f_123812_, this.f_31176_.m_20185_() + (double)f, this.f_31176_.m_20186_() + 2.0D + (double)f1, this.f_31176_.m_20189_() + (double)f2, 0.0D, 0.0D, 0.0D);
+         this.f_31176_.m_9236_().addParticle(ParticleTypes.f_123812_, this.f_31176_.getBigX().add(f), this.f_31176_.getBigY().add(2.0D + (double)f1), this.f_31176_.getBigZ().add(f2), 0.0D, 0.0D, 0.0D);
       }
 
    }
@@ -31,7 +_,7 @@
       ++this.f_31215_;
       if (this.f_31214_ == null) {
          BlockPos blockpos = this.f_31176_.m_9236_().m_5452_(Heightmap.Types.MOTION_BLOCKING, EndPodiumFeature.m_287210_(this.f_31176_.m_287165_()));
-         this.f_31214_ = Vec3.m_82539_(blockpos);
+         this.f_31214_ = Vec3.atBottomCenterOf(blockpos);
       }
 
       double d0 = this.f_31214_.m_82531_(this.f_31176_.m_20185_(), this.f_31176_.m_20186_(), this.f_31176_.m_20189_());
