--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingFlamingPhase.java
@@ -38,7 +_,7 @@
             double d5 = d2 + this.f_31176_.m_217043_().m_188583_() / 2.0D;
 
             for(int j = 0; j < 6; ++j) {
-               this.f_31176_.m_9236_().m_7106_(ParticleTypes.f_123799_, d3, d4, d5, -vec3.f_82479_ * (double)0.08F * (double)j, -vec3.f_82480_ * (double)0.6F, -vec3.f_82481_ * (double)0.08F * (double)j);
+               this.f_31176_.m_9236_().m_7106_(ParticleTypes.f_123799_, d3, d4, d5, -vec3.m_7096_() * (double)0.08F * (double)j, -vec3.m_7098_() * (double)0.6F, -vec3.m_7094_() * (double)0.08F * (double)j);
             }
 
             vec3.m_82524_(0.19634955F);
@@ -58,8 +_,8 @@
       } else if (this.f_31326_ == 10) {
          Vec3 vec3 = (new Vec3(this.f_31176_.f_31080_.m_20185_() - this.f_31176_.m_20185_(), 0.0D, this.f_31176_.f_31080_.m_20189_() - this.f_31176_.m_20189_())).m_82541_();
          float f = 5.0F;
-         double d0 = this.f_31176_.f_31080_.m_20185_() + vec3.f_82479_ * 5.0D / 2.0D;
-         double d1 = this.f_31176_.f_31080_.m_20189_() + vec3.f_82481_ * 5.0D / 2.0D;
+         double d0 = this.f_31176_.f_31080_.m_20185_() + vec3.m_7096_() * 5.0D / 2.0D;
+         double d1 = this.f_31176_.f_31080_.m_20189_() + vec3.m_7094_() * 5.0D / 2.0D;
          double d2 = this.f_31176_.f_31080_.m_20227_(0.5D);
          double d3 = d2;
          BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(d0, d2, d1);
