--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonLandingApproachPhase.java
@@ -49,16 +_,16 @@
       if (this.f_31254_ == null || this.f_31254_.m_77392_()) {
          int i = this.f_31176_.m_31155_();
          BlockPos blockpos = this.f_31176_.f_19853_.m_5452_(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, EndPodiumFeature.f_65714_);
-         Player player = this.f_31176_.f_19853_.m_45949_(f_31253_, this.f_31176_, (double)blockpos.m_123341_(), (double)blockpos.m_123342_(), (double)blockpos.m_123343_());
+         Player player = this.f_31176_.f_19853_.m_45949_(f_31253_, this.f_31176_, (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
          int j;
          if (player != null) {
             Vec3 vec3 = (new Vec3(player.m_20185_(), 0.0D, player.m_20189_())).m_82541_();
             j = this.f_31176_.m_31170_(-vec3.f_82479_ * 40.0D, 105.0D, -vec3.f_82481_ * 40.0D);
          } else {
-            j = this.f_31176_.m_31170_(40.0D, (double)blockpos.m_123342_(), 0.0D);
+            j = this.f_31176_.m_31170_(40.0D, (double)blockpos.getY(), 0.0D);
          }
 
-         Node node = new Node(blockpos.m_123341_(), blockpos.m_123342_(), blockpos.m_123343_());
+         Node node = new Node(blockpos.getX(), blockpos.getY(), blockpos.getZ());
          this.f_31254_ = this.f_31176_.m_31104_(i, j, node);
          if (this.f_31254_ != null) {
             this.f_31254_.m_77374_();
@@ -74,15 +_,15 @@
 
    private void m_31264_() {
       if (this.f_31254_ != null && !this.f_31254_.m_77392_()) {
-         Vec3i vec3i = this.f_31254_.m_77400_();
+         me.alphamode.mclong.core.Vec3l vec3i = this.f_31254_.m_77400_();
          this.f_31254_.m_77374_();
-         double d0 = (double)vec3i.m_123341_();
-         double d1 = (double)vec3i.m_123343_();
+         double d0 = (double)vec3i.getX();
+         double d1 = (double)vec3i.getZ();
 
          double d2;
          do {
-            d2 = (double)((float)vec3i.m_123342_() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
-         } while(d2 < (double)vec3i.m_123342_());
+            d2 = (double)((float)vec3i.getY() + this.f_31176_.m_217043_().m_188501_() * 20.0F);
+         } while(d2 < (double)vec3i.getY());
 
          this.f_31255_ = new Vec3(d0, d2, d1);
       }
