--- a/net/minecraft/world/level/levelgen/structure/templatesystem/AxisAlignedLinearPosTest.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/AxisAlignedLinearPosTest.java
@@ -41,9 +_,9 @@
 
    public boolean m_213782_(BlockPos p_230251_, BlockPos p_230252_, BlockPos p_230253_, RandomSource p_230254_) {
       Direction direction = Direction.m_122390_(Direction.AxisDirection.POSITIVE, this.f_73967_);
-      float f = (float)Math.abs((p_230252_.m_123341_() - p_230253_.m_123341_()) * direction.m_122429_());
-      float f1 = (float)Math.abs((p_230252_.m_123342_() - p_230253_.m_123342_()) * direction.m_122430_());
-      float f2 = (float)Math.abs((p_230252_.m_123343_() - p_230253_.m_123343_()) * direction.m_122431_());
+      float f = (float)Math.abs((p_230252_.getX() - p_230253_.getX()) * direction.m_122429_());
+      float f1 = (float)Math.abs((p_230252_.getY() - p_230253_.getY()) * direction.m_122430_());
+      float f2 = (float)Math.abs((p_230252_.getZ() - p_230253_.getZ()) * direction.m_122431_());
       int i = (int)(f + f1 + f2);
       float f3 = p_230254_.m_188501_();
       return f3 <= Mth.m_144920_(this.f_73963_, this.f_73964_, Mth.m_184655_((float)i, (float)this.f_73965_, (float)this.f_73966_));
