--- a/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
@@ -40,9 +_,9 @@
       }
 
       BlockPos blockpos = p_74113_.f_74675_();
-      int i = p_74109_.m_6924_(heightmap$types, blockpos.m_123341_(), blockpos.m_123343_()) + this.f_74102_;
-      int j = p_74112_.f_74675_().m_123342_();
-      return new StructureTemplate.StructureBlockInfo(new BlockPos(blockpos.m_123341_(), i + j, blockpos.m_123343_()), p_74113_.f_74676_(), p_74113_.f_74677_());
+      long i = p_74109_.getHeight(heightmap$types, blockpos.getBigX(), blockpos.getBigZ()) + this.f_74102_;
+      long j = p_74112_.f_74675_().getY();
+      return new StructureTemplate.StructureBlockInfo(new BlockPos(blockpos.getBigX(), me.alphamode.mclong.math.BigInteger.valueOf(i + j), blockpos.getBigZ()), p_74113_.f_74676_(), p_74113_.f_74677_());
    }
 
    protected StructureProcessorType<?> m_6953_() {
