--- a/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
@@ -39,9 +_,9 @@
          heightmap$types = this.f_74101_;
       }
 
-      int i = p_74109_.m_6924_(heightmap$types, p_74113_.f_74675_.m_123341_(), p_74113_.f_74675_.m_123343_()) + this.f_74102_;
-      int j = p_74112_.f_74675_.m_123342_();
-      return new StructureTemplate.StructureBlockInfo(new BlockPos(p_74113_.f_74675_.m_123341_(), i + j, p_74113_.f_74675_.m_123343_()), p_74113_.f_74676_, p_74113_.f_74677_);
+      long i = p_74109_.getHeight(heightmap$types, p_74113_.f_74675_.getX(), p_74113_.f_74675_.getZ()) + this.f_74102_;
+      long j = p_74112_.f_74675_.getY();
+      return new StructureTemplate.StructureBlockInfo(new BlockPos(p_74113_.f_74675_.getX(), i + j, p_74113_.f_74675_.getZ()), p_74113_.f_74676_, p_74113_.f_74677_);
    }
 
    protected StructureProcessorType<?> m_6953_() {
