--- a/net/minecraft/world/level/levelgen/structure/templatesystem/RuleProcessor.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/RuleProcessor.java
@@ -22,7 +_,7 @@
 
    @Nullable
    public StructureTemplate.StructureBlockInfo m_7382_(LevelReader p_74299_, BlockPos p_74300_, BlockPos p_74301_, StructureTemplate.StructureBlockInfo p_74302_, StructureTemplate.StructureBlockInfo p_74303_, StructurePlaceSettings p_74304_) {
-      RandomSource randomsource = RandomSource.m_216335_(Mth.m_14057_(p_74303_.f_74675_()));
+      RandomSource randomsource = RandomSource.m_216335_(Mth.getSeed(p_74303_.f_74675_()));
       BlockState blockstate = p_74299_.m_8055_(p_74303_.f_74675_());
 
       for(ProcessorRule processorrule : this.f_74293_) {
