--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructurePlaceSettings.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructurePlaceSettings.java
@@ -113,7 +_,7 @@
       if (this.f_74368_ != null) {
          return this.f_74368_;
       } else {
-         return p_230327_ == null ? RandomSource.m_216335_(Util.m_137550_()) : RandomSource.m_216335_(Mth.m_14057_(p_230327_));
+         return p_230327_ == null ? RandomSource.m_216335_(Util.m_137550_()) : RandomSource.m_216335_(Mth.getSeed(p_230327_));
       }
    }
 
