--- a/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
+++ b/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
@@ -66,10 +_,10 @@
    public static final String f_163799_ = "size";
    private final List<StructureTemplate.Palette> f_74482_ = Lists.newArrayList();
    private final List<StructureTemplate.StructureEntityInfo> f_74483_ = Lists.newArrayList();
-   private Vec3i f_74484_ = Vec3i.f_123288_;
+   private me.alphamode.mclong.core.Vec3l f_74484_ = me.alphamode.mclong.core.Vec3l.ZERO;
    private String f_74485_ = "?";
 
-   public Vec3i m_163801_() {
+   public me.alphamode.mclong.core.Vec3l getSize() {
       return this.f_74484_;
    }
 
@@ -81,21 +_,21 @@
       return this.f_74485_;
    }
 
-   public void m_163802_(Level p_163803_, BlockPos p_163804_, Vec3i p_163805_, boolean p_163806_, @Nullable Block p_163807_) {
-      if (p_163805_.m_123341_() >= 1 && p_163805_.m_123342_() >= 1 && p_163805_.m_123343_() >= 1) {
-         BlockPos blockpos = p_163804_.m_121955_(p_163805_).m_7918_(-1, -1, -1);
+   public void fillFromWorld(Level pLevel, BlockPos pPos, me.alphamode.mclong.core.Vec3l pSize, boolean pWithEntities, @Nullable Block pToIgnore) {
+      if (pSize.getBigX().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0 && pSize.getBigY().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0 && pSize.getBigZ().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0) {
+         BlockPos blockpos = pPos.offset(pSize).offset(-1, -1, -1);
          List<StructureTemplate.StructureBlockInfo> list = Lists.newArrayList();
          List<StructureTemplate.StructureBlockInfo> list1 = Lists.newArrayList();
          List<StructureTemplate.StructureBlockInfo> list2 = Lists.newArrayList();
-         BlockPos blockpos1 = new BlockPos(Math.min(p_163804_.m_123341_(), blockpos.m_123341_()), Math.min(p_163804_.m_123342_(), blockpos.m_123342_()), Math.min(p_163804_.m_123343_(), blockpos.m_123343_()));
-         BlockPos blockpos2 = new BlockPos(Math.max(p_163804_.m_123341_(), blockpos.m_123341_()), Math.max(p_163804_.m_123342_(), blockpos.m_123342_()), Math.max(p_163804_.m_123343_(), blockpos.m_123343_()));
-         this.f_74484_ = p_163805_;
+         BlockPos blockpos1 = new BlockPos(pPos.getBigX().min(blockpos.getBigX()), pPos.getBigY().min(blockpos.getBigY()), pPos.getBigZ().min(blockpos.getBigZ()));
+         BlockPos blockpos2 = new BlockPos(pPos.getBigX().max(blockpos.getBigX()), pPos.getBigY().max(blockpos.getBigY()), pPos.getBigZ().max(blockpos.getBigZ()));
+         this.f_74484_ = pSize;
 
          for(BlockPos blockpos3 : BlockPos.m_121940_(blockpos1, blockpos2)) {
-            BlockPos blockpos4 = blockpos3.m_121996_(blockpos1);
-            BlockState blockstate = p_163803_.m_8055_(blockpos3);
-            if (p_163807_ == null || !blockstate.m_60713_(p_163807_)) {
-               BlockEntity blockentity = p_163803_.m_7702_(blockpos3);
+            BlockPos blockpos4 = blockpos3.subtract(blockpos1);
+            BlockState blockstate = pLevel.m_8055_(blockpos3);
+            if (pToIgnore == null || !blockstate.m_60713_(pToIgnore)) {
+               BlockEntity blockentity = pLevel.m_7702_(blockpos3);
                StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo;
                if (blockentity != null) {
                   structuretemplate$structureblockinfo = new StructureTemplate.StructureBlockInfo(blockpos4, blockstate, blockentity.m_187481_());
@@ -110,8 +_,8 @@
          List<StructureTemplate.StructureBlockInfo> list3 = m_74614_(list, list1, list2);
          this.f_74482_.clear();
          this.f_74482_.add(new StructureTemplate.Palette(list3));
-         if (p_163806_) {
-            this.m_74500_(p_163803_, blockpos1, blockpos2.m_7918_(1, 1, 1));
+         if (pWithEntities) {
+            this.m_74500_(pLevel, blockpos1, blockpos2.offset(1, 1, 1));
          } else {
             this.f_74483_.clear();
          }
@@ -131,12 +_,12 @@
    }
 
    private static List<StructureTemplate.StructureBlockInfo> m_74614_(List<StructureTemplate.StructureBlockInfo> p_74615_, List<StructureTemplate.StructureBlockInfo> p_74616_, List<StructureTemplate.StructureBlockInfo> p_74617_) {
-      Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.<StructureTemplate.StructureBlockInfo>comparingInt((p_74641_) -> {
-         return p_74641_.f_74675_.m_123342_();
-      }).thenComparingInt((p_74637_) -> {
-         return p_74637_.f_74675_.m_123341_();
-      }).thenComparingInt((p_74572_) -> {
-         return p_74572_.f_74675_.m_123343_();
+      Comparator<StructureTemplate.StructureBlockInfo> comparator = Comparator.<StructureTemplate.StructureBlockInfo>comparingLong((p_74641_) -> {
+         return p_74641_.f_74675_.getY();
+      }).thenComparingLong((p_74637_) -> {
+         return p_74637_.f_74675_.getX();
+      }).thenComparingLong((p_74572_) -> {
+         return p_74572_.f_74675_.getZ();
       });
       p_74615_.sort(comparator);
       p_74617_.sort(comparator);
@@ -155,14 +_,14 @@
       this.f_74483_.clear();
 
       for(Entity entity : list) {
-         Vec3 vec3 = new Vec3(entity.m_20185_() - (double)p_74502_.m_123341_(), entity.m_20186_() - (double)p_74502_.m_123342_(), entity.m_20189_() - (double)p_74502_.m_123343_());
+         Vec3 vec3 = new Vec3(entity.getBigX().subtract(p_74502_.getBigDecX()), entity.getBigY().subtract(p_74502_.getBigDecY()), entity.getBigZ().subtract(p_74502_.getBigDecZ()));
          CompoundTag compoundtag = new CompoundTag();
          entity.m_20223_(compoundtag);
          BlockPos blockpos;
          if (entity instanceof Painting) {
-            blockpos = ((Painting)entity).m_31748_().m_121996_(p_74502_);
+            blockpos = ((Painting)entity).m_31748_().subtract(p_74502_);
          } else {
-            blockpos = BlockPos.m_274446_(vec3);
+            blockpos = BlockPos.containing(vec3);
          }
 
          this.f_74483_.add(new StructureTemplate.StructureEntityInfo(vec3, blockpos, compoundtag.m_6426_()));
@@ -181,8 +_,8 @@
          return objectarraylist;
       } else {
          for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : p_230337_.m_74387_(this.f_74482_, p_230336_).m_74653_(p_230338_)) {
-            BlockPos blockpos = p_230339_ ? m_74563_(p_230337_, structuretemplate$structureblockinfo.f_74675_).m_121955_(p_230336_) : structuretemplate$structureblockinfo.f_74675_;
-            if (boundingbox == null || boundingbox.m_71051_(blockpos)) {
+            BlockPos blockpos = p_230339_ ? m_74563_(p_230337_, structuretemplate$structureblockinfo.f_74675_).offset(p_230336_) : structuretemplate$structureblockinfo.f_74675_;
+            if (boundingbox == null || boundingbox.isInside(blockpos)) {
                objectarraylist.add(new StructureTemplate.StructureBlockInfo(blockpos, structuretemplate$structureblockinfo.f_74676_.m_60717_(p_230337_.m_74404_()), structuretemplate$structureblockinfo.f_74677_));
             }
          }
@@ -194,7 +_,7 @@
    public BlockPos m_74566_(StructurePlaceSettings p_74567_, BlockPos p_74568_, StructurePlaceSettings p_74569_, BlockPos p_74570_) {
       BlockPos blockpos = m_74563_(p_74567_, p_74568_);
       BlockPos blockpos1 = m_74563_(p_74569_, p_74570_);
-      return blockpos.m_121996_(blockpos1);
+      return blockpos.subtract(blockpos1);
    }
 
    public static BlockPos m_74563_(StructurePlaceSettings p_74564_, BlockPos p_74565_) {
@@ -206,21 +_,21 @@
          return false;
       } else {
          List<StructureTemplate.StructureBlockInfo> list = p_230332_.m_74387_(this.f_74482_, p_230330_).m_74652_();
-         if ((!list.isEmpty() || !p_230332_.m_74408_() && !this.f_74483_.isEmpty()) && this.f_74484_.m_123341_() >= 1 && this.f_74484_.m_123342_() >= 1 && this.f_74484_.m_123343_() >= 1) {
+         if ((!list.isEmpty() || !p_230332_.m_74408_() && !this.f_74483_.isEmpty()) && this.f_74484_.getBigX().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0 && this.f_74484_.getBigY().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0 && this.f_74484_.getBigZ().compareTo(me.alphamode.mclong.math.BigInteger.ONE) >= 0) {
             BoundingBox boundingbox = p_230332_.m_74409_();
             List<BlockPos> list1 = Lists.newArrayListWithCapacity(p_230332_.m_74413_() ? list.size() : 0);
             List<BlockPos> list2 = Lists.newArrayListWithCapacity(p_230332_.m_74413_() ? list.size() : 0);
             List<Pair<BlockPos, CompoundTag>> list3 = Lists.newArrayListWithCapacity(list.size());
-            int i = Integer.MAX_VALUE;
-            int j = Integer.MAX_VALUE;
-            int k = Integer.MAX_VALUE;
-            int l = Integer.MIN_VALUE;
-            int i1 = Integer.MIN_VALUE;
-            int j1 = Integer.MIN_VALUE;
+            me.alphamode.mclong.math.BigInteger i = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+            me.alphamode.mclong.math.BigInteger j = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+            me.alphamode.mclong.math.BigInteger k = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MAX_VALUE);
+            me.alphamode.mclong.math.BigInteger l = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MIN_VALUE);
+            me.alphamode.mclong.math.BigInteger i1 = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MIN_VALUE);
+            me.alphamode.mclong.math.BigInteger j1 = me.alphamode.mclong.math.BigInteger.valueOf(java.lang.Long.MIN_VALUE);
 
             for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : m_74517_(p_230329_, p_230330_, p_230331_, p_230332_, list)) {
                BlockPos blockpos = structuretemplate$structureblockinfo.f_74675_;
-               if (boundingbox == null || boundingbox.m_71051_(blockpos)) {
+               if (boundingbox == null || boundingbox.isInside(blockpos)) {
                   FluidState fluidstate = p_230332_.m_74413_() ? p_230329_.m_6425_(blockpos) : null;
                   BlockState blockstate = structuretemplate$structureblockinfo.f_74676_.m_60715_(p_230332_.m_74401_()).m_60717_(p_230332_.m_74404_());
                   if (structuretemplate$structureblockinfo.f_74677_ != null) {
@@ -230,12 +_,12 @@
                   }
 
                   if (p_230329_.m_7731_(blockpos, blockstate, p_230334_)) {
-                     i = Math.min(i, blockpos.m_123341_());
-                     j = Math.min(j, blockpos.m_123342_());
-                     k = Math.min(k, blockpos.m_123343_());
-                     l = Math.max(l, blockpos.m_123341_());
-                     i1 = Math.max(i1, blockpos.m_123342_());
-                     j1 = Math.max(j1, blockpos.m_123343_());
+                     i = i.min(blockpos.getBigX());
+                     j = j.min(blockpos.getBigY());
+                     k = k.min(blockpos.getBigZ());
+                     l = l.max(blockpos.getBigX());
+                     i1 = i1.max(blockpos.getBigY());
+                     j1 = j1.max(blockpos.getBigZ());
                      list3.add(Pair.of(blockpos, structuretemplate$structureblockinfo.f_74677_));
                      if (structuretemplate$structureblockinfo.f_74677_ != null) {
                         BlockEntity blockentity1 = p_230329_.m_7702_(blockpos);
@@ -274,7 +_,7 @@
                   FluidState fluidstate2 = p_230329_.m_6425_(blockpos3);
 
                   for(int i2 = 0; i2 < adirection.length && !fluidstate2.m_76170_(); ++i2) {
-                     BlockPos blockpos1 = blockpos3.m_121945_(adirection[i2]);
+                     BlockPos blockpos1 = blockpos3.relative(adirection[i2]);
                      FluidState fluidstate1 = p_230329_.m_6425_(blockpos1);
                      if (fluidstate1.m_76170_() && !list2.contains(blockpos1)) {
                         fluidstate2 = fluidstate1;
@@ -293,19 +_,19 @@
                }
             }
 
-            if (i <= l) {
+            if (i.compareTo(l) <= 0) {
                if (!p_230332_.m_74410_()) {
-                  DiscreteVoxelShape discretevoxelshape = new BitSetDiscreteVoxelShape(l - i + 1, i1 - j + 1, j1 - k + 1);
-                  int k1 = i;
-                  int l1 = j;
-                  int j2 = k;
+                  DiscreteVoxelShape discretevoxelshape = new BitSetDiscreteVoxelShape((int) (l.subtract(i).add().intValue()), (int) (i1.subtract(j).add().intValue()), (int) (j1.subtract(k).add().intValue()));
+                  me.alphamode.mclong.math.BigInteger k1 = i;
+                  me.alphamode.mclong.math.BigInteger l1 = j;
+                  me.alphamode.mclong.math.BigInteger j2 = k;
 
                   for(Pair<BlockPos, CompoundTag> pair1 : list3) {
                      BlockPos blockpos2 = pair1.getFirst();
-                     discretevoxelshape.m_142703_(blockpos2.m_123341_() - k1, blockpos2.m_123342_() - l1, blockpos2.m_123343_() - j2);
+                     discretevoxelshape.fill(blockpos2.getBigX().subtract(k1).longValue(), blockpos2.getBigY().subtract(l1).longValue(), blockpos2.getBigZ().subtract(j2).longValue());
                   }
 
-                  m_74510_(p_230329_, p_230334_, discretevoxelshape, k1, l1, j2);
+                  updateShapeAtEdge(p_230329_, p_230334_, discretevoxelshape, k1, l1, j2);
                }
 
                for(Pair<BlockPos, CompoundTag> pair : list3) {
@@ -340,20 +_,20 @@
       }
    }
 
-   public static void m_74510_(LevelAccessor p_74511_, int p_74512_, DiscreteVoxelShape p_74513_, int p_74514_, int p_74515_, int p_74516_) {
-      p_74513_.m_82810_((p_74494_, p_74495_, p_74496_, p_74497_) -> {
-         BlockPos blockpos = new BlockPos(p_74514_ + p_74495_, p_74515_ + p_74496_, p_74516_ + p_74497_);
-         BlockPos blockpos1 = blockpos.m_121945_(p_74494_);
-         BlockState blockstate = p_74511_.m_8055_(blockpos);
-         BlockState blockstate1 = p_74511_.m_8055_(blockpos1);
-         BlockState blockstate2 = blockstate.m_60728_(p_74494_, blockstate1, p_74511_, blockpos, blockpos1);
+   public static void updateShapeAtEdge(LevelAccessor pLevel, int pFlags, DiscreteVoxelShape pShape, me.alphamode.mclong.math.BigInteger pX, me.alphamode.mclong.math.BigInteger pY, me.alphamode.mclong.math.BigInteger pZ) {
+      pShape.m_82810_((p_74494_, p_74495_, p_74496_, p_74497_) -> {
+         BlockPos blockpos = new BlockPos(pX.add(p_74495_), pY.add(p_74496_), pZ.add(p_74497_));
+         BlockPos blockpos1 = blockpos.relative(p_74494_);
+         BlockState blockstate = pLevel.m_8055_(blockpos);
+         BlockState blockstate1 = pLevel.m_8055_(blockpos1);
+         BlockState blockstate2 = blockstate.m_60728_(p_74494_, blockstate1, pLevel, blockpos, blockpos1);
          if (blockstate != blockstate2) {
-            p_74511_.m_7731_(blockpos, blockstate2, p_74512_ & -2);
+            pLevel.m_7731_(blockpos, blockstate2, pFlags & -2);
          }
 
-         BlockState blockstate3 = blockstate1.m_60728_(p_74494_.m_122424_(), blockstate2, p_74511_, blockpos1, blockpos);
+         BlockState blockstate3 = blockstate1.m_60728_(p_74494_.m_122424_(), blockstate2, pLevel, blockpos1, blockpos);
          if (blockstate1 != blockstate3) {
-            p_74511_.m_7731_(blockpos1, blockstate3, p_74512_ & -2);
+            pLevel.m_7731_(blockpos1, blockstate3, pFlags & -2);
          }
 
       });
@@ -364,7 +_,7 @@
       List<StructureTemplate.StructureBlockInfo> list1 = new ArrayList<>();
 
       for(StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo : p_74522_) {
-         BlockPos blockpos = m_74563_(p_74521_, structuretemplate$structureblockinfo.f_74675_).m_121955_(p_74519_);
+         BlockPos blockpos = m_74563_(p_74521_, structuretemplate$structureblockinfo.f_74675_).offset(p_74519_);
          StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo1 = new StructureTemplate.StructureBlockInfo(blockpos, structuretemplate$structureblockinfo.f_74676_, structuretemplate$structureblockinfo.f_74677_ != null ? structuretemplate$structureblockinfo.f_74677_.m_6426_() : null);
 
          for(Iterator<StructureProcessor> iterator = p_74521_.m_74411_().iterator(); structuretemplate$structureblockinfo1 != null && iterator.hasNext(); structuretemplate$structureblockinfo1 = iterator.next().m_7382_(p_278297_, p_74519_, p_74520_, structuretemplate$structureblockinfo, structuretemplate$structureblockinfo1, p_74521_)) {
@@ -385,23 +_,23 @@
 
    private void m_74523_(ServerLevelAccessor p_74524_, BlockPos p_74525_, Mirror p_74526_, Rotation p_74527_, BlockPos p_74528_, @Nullable BoundingBox p_74529_, boolean p_74530_) {
       for(StructureTemplate.StructureEntityInfo structuretemplate$structureentityinfo : this.f_74483_) {
-         BlockPos blockpos = m_74593_(structuretemplate$structureentityinfo.f_74684_, p_74526_, p_74527_, p_74528_).m_121955_(p_74525_);
-         if (p_74529_ == null || p_74529_.m_71051_(blockpos)) {
+         BlockPos blockpos = m_74593_(structuretemplate$structureentityinfo.f_74684_, p_74526_, p_74527_, p_74528_).offset(p_74525_);
+         if (p_74529_ == null || p_74529_.isInside(blockpos)) {
             CompoundTag compoundtag = structuretemplate$structureentityinfo.f_74685_.m_6426_();
             Vec3 vec3 = m_74578_(structuretemplate$structureentityinfo.f_74683_, p_74526_, p_74527_, p_74528_);
-            Vec3 vec31 = vec3.m_82520_((double)p_74525_.m_123341_(), (double)p_74525_.m_123342_(), (double)p_74525_.m_123343_());
+            Vec3 vec31 = vec3.add(p_74525_.getBigDecX(), p_74525_.getBigDecY(), p_74525_.getBigDecZ());
             ListTag listtag = new ListTag();
-            listtag.add(DoubleTag.m_128500_(vec31.f_82479_));
-            listtag.add(DoubleTag.m_128500_(vec31.f_82480_));
-            listtag.add(DoubleTag.m_128500_(vec31.f_82481_));
+            listtag.add(net.minecraft.nbt.StringTag.m_129297_(vec31.getX().toString()));
+            listtag.add(net.minecraft.nbt.StringTag.m_129297_(vec31.getY().toString()));
+            listtag.add(net.minecraft.nbt.StringTag.m_129297_(vec31.getZ().toString()));
             compoundtag.m_128365_("Pos", listtag);
             compoundtag.m_128473_("UUID");
             m_74543_(p_74524_, compoundtag).ifPresent((p_275190_) -> {
                float f = p_275190_.m_7890_(p_74527_);
                f += p_275190_.m_6961_(p_74526_) - p_275190_.m_146908_();
-               p_275190_.m_7678_(vec31.f_82479_, vec31.f_82480_, vec31.f_82481_, f, p_275190_.m_146909_());
+               p_275190_.moveTo(vec31.getX(), vec31.getY(), vec31.getZ(), f, p_275190_.m_146909_());
                if (p_74530_ && p_275190_ instanceof Mob) {
-                  ((Mob)p_275190_).m_6518_(p_74524_, p_74524_.m_6436_(BlockPos.m_274446_(vec31)), MobSpawnType.STRUCTURE, (SpawnGroupData)null, compoundtag);
+                  ((Mob)p_275190_).m_6518_(p_74524_, p_74524_.m_6436_(BlockPos.containing(vec31)), MobSpawnType.STRUCTURE, (SpawnGroupData)null, compoundtag);
                }
 
                p_74524_.m_47205_(p_275190_);
@@ -419,98 +_,98 @@
       }
    }
 
-   public Vec3i m_163808_(Rotation p_163809_) {
-      switch (p_163809_) {
+   public me.alphamode.mclong.core.Vec3l getSize(Rotation pRotation) {
+      switch (pRotation) {
          case COUNTERCLOCKWISE_90:
          case CLOCKWISE_90:
-            return new Vec3i(this.f_74484_.m_123343_(), this.f_74484_.m_123342_(), this.f_74484_.m_123341_());
+            return new me.alphamode.mclong.core.Vec3l(this.f_74484_.getBigZ(), this.f_74484_.getBigY(), this.f_74484_.getBigX());
          default:
             return this.f_74484_;
       }
    }
 
    public static BlockPos m_74593_(BlockPos p_74594_, Mirror p_74595_, Rotation p_74596_, BlockPos p_74597_) {
-      int i = p_74594_.m_123341_();
-      int j = p_74594_.m_123342_();
-      int k = p_74594_.m_123343_();
+      me.alphamode.mclong.math.BigInteger i = p_74594_.getBigX();
+      me.alphamode.mclong.math.BigInteger j = p_74594_.getBigY();
+      me.alphamode.mclong.math.BigInteger k = p_74594_.getBigZ();
       boolean flag = true;
       switch (p_74595_) {
          case LEFT_RIGHT:
-            k = -k;
+            k = k.negate();
             break;
          case FRONT_BACK:
-            i = -i;
+            i = i.negate();
             break;
          default:
             flag = false;
       }
 
-      int l = p_74597_.m_123341_();
-      int i1 = p_74597_.m_123343_();
+      me.alphamode.mclong.math.BigInteger l = p_74597_.getBigX();
+      me.alphamode.mclong.math.BigInteger i1 = p_74597_.getBigZ();
       switch (p_74596_) {
          case COUNTERCLOCKWISE_90:
-            return new BlockPos(l - i1 + k, j, l + i1 - i);
+            return new BlockPos(l.subtract(i1).add(k), j, l.add(i1).subtract(i));
          case CLOCKWISE_90:
-            return new BlockPos(l + i1 - k, j, i1 - l + i);
+            return new BlockPos(l.add(i1).subtract(k), j, i1.subtract(l).add(i));
          case CLOCKWISE_180:
-            return new BlockPos(l + l - i, j, i1 + i1 - k);
+            return new BlockPos(l.add(l).subtract(i), j, i1.add(i1).subtract(k));
          default:
             return flag ? new BlockPos(i, j, k) : p_74594_;
       }
    }
 
    public static Vec3 m_74578_(Vec3 p_74579_, Mirror p_74580_, Rotation p_74581_, BlockPos p_74582_) {
-      double d0 = p_74579_.f_82479_;
-      double d1 = p_74579_.f_82480_;
-      double d2 = p_74579_.f_82481_;
+      me.alphamode.mclong.math.BigDecimal d0 = p_74579_.getX();
+      me.alphamode.mclong.math.BigDecimal d1 = p_74579_.getY();
+      me.alphamode.mclong.math.BigDecimal d2 = p_74579_.getZ();
       boolean flag = true;
       switch (p_74580_) {
          case LEFT_RIGHT:
-            d2 = 1.0D - d2;
+            d2 = me.alphamode.mclong.math.BigDecimal.ONE.subtract(d2);
             break;
          case FRONT_BACK:
-            d0 = 1.0D - d0;
+            d0 = me.alphamode.mclong.math.BigDecimal.ONE.subtract(d0);
             break;
          default:
             flag = false;
       }
 
-      int i = p_74582_.m_123341_();
-      int j = p_74582_.m_123343_();
+      me.alphamode.mclong.math.BigInteger i = p_74582_.getBigX();
+      me.alphamode.mclong.math.BigInteger j = p_74582_.getBigZ();
       switch (p_74581_) {
          case COUNTERCLOCKWISE_90:
-            return new Vec3((double)(i - j) + d2, d1, (double)(i + j + 1) - d0);
+            return new Vec3((i.subtract(j).toBigDecimal()).add(d2), d1, (i.add(j).add().toBigDecimal()).subtract(d0));
          case CLOCKWISE_90:
-            return new Vec3((double)(i + j + 1) - d2, d1, (double)(j - i) + d0);
+            return new Vec3((i.add(j).add().toBigDecimal()).subtract(d2), d1, (j.subtract(i).toBigDecimal()).add(d0));
          case CLOCKWISE_180:
-            return new Vec3((double)(i + i + 1) - d0, d1, (double)(j + j + 1) - d2);
+            return new Vec3((i.add(i).add().toBigDecimal()).subtract(d0), d1, (j.add(j).add().toBigDecimal()).subtract(d2));
          default:
             return flag ? new Vec3(d0, d1, d2) : p_74579_;
       }
    }
 
    public BlockPos m_74583_(BlockPos p_74584_, Mirror p_74585_, Rotation p_74586_) {
-      return m_74587_(p_74584_, p_74585_, p_74586_, this.m_163801_().m_123341_(), this.m_163801_().m_123343_());
+      return getZeroPositionWithTransform(p_74584_, p_74585_, p_74586_, this.getSize().getBigX(), this.getSize().getBigZ());
    }
 
-   public static BlockPos m_74587_(BlockPos p_74588_, Mirror p_74589_, Rotation p_74590_, int p_74591_, int p_74592_) {
-      --p_74591_;
-      --p_74592_;
-      int i = p_74589_ == Mirror.FRONT_BACK ? p_74591_ : 0;
-      int j = p_74589_ == Mirror.LEFT_RIGHT ? p_74592_ : 0;
-      BlockPos blockpos = p_74588_;
-      switch (p_74590_) {
+   public static BlockPos getZeroPositionWithTransform(BlockPos pPos, Mirror pMirror, Rotation pRotation, me.alphamode.mclong.math.BigInteger pSizeX, me.alphamode.mclong.math.BigInteger pSizeZ) {
+      pSizeX = pSizeX.subtract();
+      pSizeZ = pSizeZ.subtract();
+      me.alphamode.mclong.math.BigInteger i = pMirror == Mirror.FRONT_BACK ? pSizeX : me.alphamode.mclong.math.BigInteger.ZERO;
+      me.alphamode.mclong.math.BigInteger j = pMirror == Mirror.LEFT_RIGHT ? pSizeZ : me.alphamode.mclong.math.BigInteger.ZERO;
+      BlockPos blockpos = pPos;
+      switch (pRotation) {
          case COUNTERCLOCKWISE_90:
-            blockpos = p_74588_.m_7918_(j, 0, p_74591_ - i);
+            blockpos = pPos.offset(j, me.alphamode.mclong.math.BigInteger.ZERO, pSizeX.subtract(i));
             break;
          case CLOCKWISE_90:
-            blockpos = p_74588_.m_7918_(p_74592_ - j, 0, i);
+            blockpos = pPos.offset(pSizeZ.subtract(j), me.alphamode.mclong.math.BigInteger.ZERO, i);
             break;
          case CLOCKWISE_180:
-            blockpos = p_74588_.m_7918_(p_74591_ - i, 0, p_74592_ - j);
+            blockpos = pPos.offset(pSizeX.subtract(i), me.alphamode.mclong.math.BigInteger.ZERO, pSizeZ.subtract(j));
             break;
          case NONE:
-            blockpos = p_74588_.m_7918_(i, 0, j);
+            blockpos = pPos.offset(i, me.alphamode.mclong.math.BigInteger.ZERO, j);
       }
 
       return blockpos;
@@ -521,15 +_,15 @@
    }
 
    public BoundingBox m_74598_(BlockPos p_74599_, Rotation p_74600_, BlockPos p_74601_, Mirror p_74602_) {
-      return m_163810_(p_74599_, p_74600_, p_74601_, p_74602_, this.f_74484_);
+      return getBoundingBox(p_74599_, p_74600_, p_74601_, p_74602_, this.f_74484_);
    }
 
    @VisibleForTesting
-   protected static BoundingBox m_163810_(BlockPos p_163811_, Rotation p_163812_, BlockPos p_163813_, Mirror p_163814_, Vec3i p_163815_) {
-      Vec3i vec3i = p_163815_.m_7918_(-1, -1, -1);
-      BlockPos blockpos = m_74593_(BlockPos.f_121853_, p_163814_, p_163812_, p_163813_);
-      BlockPos blockpos1 = m_74593_(BlockPos.f_121853_.m_121955_(vec3i), p_163814_, p_163812_, p_163813_);
-      return BoundingBox.m_162375_(blockpos, blockpos1).m_162373_(p_163811_);
+   protected static BoundingBox getBoundingBox(BlockPos pStartPos, Rotation pRotation, BlockPos pPivotPos, Mirror pMirror, me.alphamode.mclong.core.Vec3l pSize) {
+      me.alphamode.mclong.core.Vec3l vec3i = pSize.offset(-1, -1, -1);
+      BlockPos blockpos = m_74593_(BlockPos.f_121853_, pMirror, pRotation, pPivotPos);
+      BlockPos blockpos1 = m_74593_(BlockPos.f_121853_.offset(vec3i), pMirror, pRotation, pPivotPos);
+      return BoundingBox.fromCorners(blockpos, blockpos1).move(pStartPos);
    }
 
    public CompoundTag m_74618_(CompoundTag p_74619_) {
@@ -551,7 +_,7 @@
          for(int j = 0; j < list1.size(); ++j) {
             StructureTemplate.StructureBlockInfo structuretemplate$structureblockinfo = list1.get(j);
             CompoundTag compoundtag = new CompoundTag();
-            compoundtag.m_128365_("pos", this.m_74625_(structuretemplate$structureblockinfo.f_74675_.m_123341_(), structuretemplate$structureblockinfo.f_74675_.m_123342_(), structuretemplate$structureblockinfo.f_74675_.m_123343_()));
+            compoundtag.m_128365_("pos", this.newLongList(structuretemplate$structureblockinfo.f_74675_.getBigX().toString(), structuretemplate$structureblockinfo.f_74675_.getBigY().toString(), structuretemplate$structureblockinfo.f_74675_.getBigZ().toString()));
             int k = structuretemplate$simplepalette.m_74669_(structuretemplate$structureblockinfo.f_74676_);
             compoundtag.m_128405_("state", k);
             if (structuretemplate$structureblockinfo.f_74677_ != null) {
@@ -596,8 +_,8 @@
 
       for(StructureTemplate.StructureEntityInfo structuretemplate$structureentityinfo : this.f_74483_) {
          CompoundTag compoundtag1 = new CompoundTag();
-         compoundtag1.m_128365_("pos", this.m_74623_(structuretemplate$structureentityinfo.f_74683_.f_82479_, structuretemplate$structureentityinfo.f_74683_.f_82480_, structuretemplate$structureentityinfo.f_74683_.f_82481_));
-         compoundtag1.m_128365_("blockPos", this.m_74625_(structuretemplate$structureentityinfo.f_74684_.m_123341_(), structuretemplate$structureentityinfo.f_74684_.m_123342_(), structuretemplate$structureentityinfo.f_74684_.m_123343_()));
+         compoundtag1.m_128365_("pos", this.m_74623_(structuretemplate$structureentityinfo.f_74683_.m_7096_(), structuretemplate$structureentityinfo.f_74683_.m_7098_(), structuretemplate$structureentityinfo.f_74683_.m_7094_()));
+         compoundtag1.m_128365_("blockPos", this.newLongList(structuretemplate$structureentityinfo.f_74684_.getBigX().toString(), structuretemplate$structureentityinfo.f_74684_.getBigY().toString(), structuretemplate$structureentityinfo.f_74684_.getBigZ().toString()));
          if (structuretemplate$structureentityinfo.f_74685_ != null) {
             compoundtag1.m_128365_("nbt", structuretemplate$structureentityinfo.f_74685_);
          }
@@ -606,7 +_,7 @@
       }
 
       p_74619_.m_128365_("entities", listtag);
-      p_74619_.m_128365_("size", this.m_74625_(this.f_74484_.m_123341_(), this.f_74484_.m_123342_(), this.f_74484_.m_123343_()));
+      p_74619_.m_128365_("size", this.newLongList(this.f_74484_.getBigX().toString(), this.f_74484_.getBigY().toString(), this.f_74484_.getBigZ().toString()));
       return NbtUtils.m_264171_(p_74619_);
    }
 
@@ -614,7 +_,7 @@
       this.f_74482_.clear();
       this.f_74483_.clear();
       ListTag listtag = p_248574_.m_128437_("size", 3);
-      this.f_74484_ = new Vec3i(listtag.m_128763_(0), listtag.m_128763_(1), listtag.m_128763_(2));
+      this.f_74484_ = new me.alphamode.mclong.core.Vec3l(new me.alphamode.mclong.math.BigInteger(listtag.m_128778_(0)), new me.alphamode.mclong.math.BigInteger(listtag.m_128778_(1)), new me.alphamode.mclong.math.BigInteger(listtag.m_128778_(2)));
       ListTag listtag1 = p_248574_.m_128437_("blocks", 10);
       if (p_248574_.m_128425_("palettes", 9)) {
          ListTag listtag2 = p_248574_.m_128437_("palettes", 9);
@@ -678,6 +_,16 @@
 
       for(int i : p_74626_) {
          listtag.add(IntTag.m_128679_(i));
+      }
+
+      return listtag;
+   }
+
+   private ListTag newLongList(String... longs) {
+      ListTag listtag = new ListTag();
+
+      for(String i : longs) {
+         listtag.add(net.minecraft.nbt.StringTag.m_129297_(i));
       }
 
       return listtag;
